--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -893,7 +893,7 @@
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
-        ActionListType actionListType = mock(ActionListType.class);
+        ActionListType actionListType = createMockActionListType();
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
@@ -907,7 +907,6 @@
 
         when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
         when(policyRuleType.getActionList()).thenReturn(actionListType);
-        when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
         when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
@@ -1000,7 +999,35 @@
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
-        ActionListType actionListType = mock(ActionListType.class);
+        AddressType addressType = mock(AddressType.class);
+        SubnetType subnetType = mock(SubnetType.class);
+        PortType portType = mock(PortType.class);
+        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+        when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenPolicyRuleCommand.class), anyLong())).thenReturn(listTungstenPolicyRuleAnswer);
+        doReturn(networkPolicy).when(listTungstenPolicyRuleAnswer).getApiObjectBase();
+        when(networkPolicy.getEntries()).thenReturn(policyEntriesType);
+        when(policyEntriesType.getPolicyRule()).thenReturn(List.of(policyRuleType));
+        when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
+        ActionListType actionListType = createMockActionListType();
+        when(policyRuleType.getActionList()).thenReturn(actionListType);
+        when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
+        when(addressType.getSubnet()).thenReturn(subnetType);
+        when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
+        when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
+        when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
+
+        assertNotNull(tungstenService.listTungstenPolicyRule(1L, "948f421c-edde-4518-a391-09299cc25dc2", "8b4637b6-5629-46de-8fb2-d0b0502bfa85"));
+    }
+
+    @Test
+    public void listTungstenPolicyRuleWithAllRuleTest() {
+        TungstenAnswer listTungstenPolicyRuleAnswer = MockTungstenAnswerFactory.get(true);
+        NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
+        PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
+        PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
+        ActionListType actionListType = createMockActionListType();
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
@@ -1013,36 +1040,6 @@
         when(policyEntriesType.getPolicyRule()).thenReturn(List.of(policyRuleType));
         when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
         when(policyRuleType.getActionList()).thenReturn(actionListType);
-        when(actionListType.getSimpleAction()).thenReturn("pass");
-        when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
-        when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
-        when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
-        when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
-
-        assertNotNull(tungstenService.listTungstenPolicyRule(1L, "948f421c-edde-4518-a391-09299cc25dc2", "8b4637b6-5629-46de-8fb2-d0b0502bfa85"));
-    }
-
-    @Test
-    public void listTungstenPolicyRuleWithAllRuleTest() {
-        TungstenAnswer listTungstenPolicyRuleAnswer = MockTungstenAnswerFactory.get(true);
-        NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
-        PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
-        PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
-        ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
-        SubnetType subnetType = mock(SubnetType.class);
-        PortType portType = mock(PortType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
-        when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenPolicyRuleCommand.class), anyLong())).thenReturn(listTungstenPolicyRuleAnswer);
-        doReturn(networkPolicy).when(listTungstenPolicyRuleAnswer).getApiObjectBase();
-        when(networkPolicy.getEntries()).thenReturn(policyEntriesType);
-        when(policyEntriesType.getPolicyRule()).thenReturn(List.of(policyRuleType));
-        when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
-        when(policyRuleType.getActionList()).thenReturn(actionListType);
-        when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
         when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
@@ -1287,7 +1284,7 @@
     public void createTungstenFirewallRuleTest() {
         TungstenAnswer createTungstenFirewallRuleAnswer = MockTungstenAnswerFactory.get(true);
         net.juniper.tungsten.api.types.FirewallRule firewallRule = mock(net.juniper.tungsten.api.types.FirewallRule.class);
-        ActionListType actionListType = mock(ActionListType.class);
+        ActionListType actionListType = createMockActionListType();
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
@@ -1298,7 +1295,6 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenFirewallRuleCommand.class), anyLong())).thenReturn(createTungstenFirewallRuleAnswer);
         when(createTungstenFirewallRuleAnswer.getApiObjectBase()).thenReturn(firewallRule);
         when(firewallRule.getActionList()).thenReturn(actionListType);
-        when(actionListType.getSimpleAction()).thenReturn("pass");
         when(firewallRule.getServiceGroup()).thenReturn(List.of(serviceGroup));
         when(serviceGroup.getReferredName()).thenReturn(List.of("service"));
         when(firewallRule.getEndpoint1()).thenReturn(firewallRuleEndpointType1);
@@ -1387,7 +1383,7 @@
     public void listTungstenFirewallRuleTest() {
         TungstenAnswer listTungstenFirewallRuleAnswer = MockTungstenAnswerFactory.get(true);
         net.juniper.tungsten.api.types.FirewallRule firewallRule = mock(net.juniper.tungsten.api.types.FirewallRule.class);
-        ActionListType actionListType = mock(ActionListType.class);
+        ActionListType actionListType = createMockActionListType();
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
@@ -1398,7 +1394,6 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenFirewallRuleCommand.class), anyLong())).thenReturn(listTungstenFirewallRuleAnswer);
         doReturn(List.of(firewallRule)).when(listTungstenFirewallRuleAnswer).getApiObjectBaseList();
         when(firewallRule.getActionList()).thenReturn(actionListType);
-        when(actionListType.getSimpleAction()).thenReturn("pass");
         when(firewallRule.getServiceGroup()).thenReturn(List.of(serviceGroup));
         when(serviceGroup.getReferredName()).thenReturn(List.of("service"));
         when(firewallRule.getEndpoint1()).thenReturn(firewallRuleEndpointType1);
@@ -1668,4 +1663,10 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static ActionListType createMockActionListType() {
+        ActionListType actionListType = mock(ActionListType.class);
+        when(actionListType.getSimpleAction()).thenReturn("pass");
+        return actionListType;
+    }
 }
