--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -185,6 +185,8 @@
 
     TungstenResource tungstenResource;
 
+    private ApplicationPolicySet applicationPolicySetMock;
+
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
 
     AutoCloseable closeable;
@@ -196,6 +198,8 @@
         tungstenResource.tungstenApi = tungstenApi;
         ReflectionTestUtils.setField(tungstenResource, "vrouterPort", "9091");
         tungstenVRouterApiMocked = Mockito.mockStatic(TungstenVRouterApi.class);
+
+        applicationPolicySetMock = mock(ApplicationPolicySet.class);
 
         Project project = mock(Project.class);
         when(project.getUuid()).thenReturn("065eab99-b819-4f3f-8e97-99c2ab22e6ed");
@@ -1119,8 +1123,7 @@
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
-        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
+        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySetMock);
         when(tungstenApi.applyTungstenPolicyTag(anyString(), anyString())).thenReturn(true);
         when(tungstenApi.getTungstenObject(eq(Tag.class), anyString())).thenReturn(tag);
         when(tungstenApi.getBackRefFromVirtualNetwork(eq(VirtualNetwork.class), anyList())).thenAnswer(virtualNetworksAnswer);
@@ -1170,8 +1173,7 @@
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
-        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
+        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySetMock);
         when(tungstenApi.removeTungstenTag(anyList(), anyList(), anyList(), anyString(), anyString(), anyString())).thenReturn(tag);
         when(tungstenApi.getBackRefFromVirtualNetwork(eq(VirtualNetwork.class), anyList())).thenAnswer(virtualNetworksAnswer);
         when(tungstenApi.getBackRefFromVirtualMachine(eq(VirtualMachine.class), anyList())).thenAnswer(virtualMachinesAnswer);
@@ -1242,8 +1244,7 @@
     public void executeRequestListTungstenApplicationPolicySetCommandTest() {
         TungstenCommand command = new ListTungstenApplicationPolicySetCommand("d64e6082-10b3-11ec-82a8-0242ac130003");
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
-        Answer<List<ApiObjectBase>> applicationPolicySetsAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
+        Answer<List<ApiObjectBase>> applicationPolicySetsAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySetMock);
         when(tungstenApi.listTungstenApplicationPolicySet(anyString())).thenAnswer(applicationPolicySetsAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
