--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxApiClientTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxApiClientTest.java
@@ -104,10 +104,7 @@
 
     @Test
     public void testGetGroupsForTrafficIngress() {
-        SDNProviderNetworkRule rule = Mockito.mock(SDNProviderNetworkRule.class);
-        Mockito.when(rule.getSourceCidrList()).thenReturn(List.of("ANY"));
-        Mockito.when(rule.getTrafficType()).thenReturn("Ingress");
-        Mockito.when(rule.getService()).thenReturn(Network.Service.NetworkACL);
+        SDNProviderNetworkRule rule = createMockSdnProviderNetworkRule("Ingress");
         String segmentName = "segment";
         List<String> sourceGroups = client.getGroupsForTraffic(rule, segmentName, true);
         List<String> destinationGroups = client.getGroupsForTraffic(rule, segmentName, false);
@@ -117,10 +114,7 @@
 
     @Test
     public void testGetGroupsForTrafficEgress() {
-        SDNProviderNetworkRule rule = Mockito.mock(SDNProviderNetworkRule.class);
-        Mockito.when(rule.getSourceCidrList()).thenReturn(List.of("ANY"));
-        Mockito.when(rule.getTrafficType()).thenReturn("Egress");
-        Mockito.when(rule.getService()).thenReturn(Network.Service.NetworkACL);
+        SDNProviderNetworkRule rule = createMockSdnProviderNetworkRule("Egress");
         String segmentName = "segment";
         List<String> sourceGroups = client.getGroupsForTraffic(rule, segmentName, true);
         List<String> destinationGroups = client.getGroupsForTraffic(rule, segmentName, false);
@@ -391,6 +385,14 @@
         assertTrue(thrownException.getMessage().startsWith("Failed to create NSX LB server pool, due to"));
     }
 
+    private static SDNProviderNetworkRule createMockSdnProviderNetworkRule(String trafficType) {
+        SDNProviderNetworkRule rule = Mockito.mock(SDNProviderNetworkRule.class);
+        Mockito.when(rule.getSourceCidrList()).thenReturn(List.of("ANY"));
+        Mockito.when(rule.getTrafficType()).thenReturn(trafficType);
+        Mockito.when(rule.getService()).thenReturn(Network.Service.NetworkACL);
+        return rule;
+    }
+
     private LbMonitorProfiles mockLbMonitorProfiles() {
         LbMonitorProfiles lbMonitorProfiles = Mockito.mock(LbMonitorProfiles.class);
         Structure monitorStructure = Mockito.mock(Structure.class, Mockito.RETURNS_DEEP_STUBS);
