--- a/server/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateManagerImplTest.java
@@ -287,20 +287,16 @@
     @Test
     public void testGetOpenPortsForVnfAppliance() {
         when(template.getId()).thenReturn(templateId);
-        VnfTemplateDetailVO accessMethodsDetail = Mockito.mock(VnfTemplateDetailVO.class);
-        when(accessMethodsDetail.getValue()).thenReturn("console,ssh-password,http,https");
+        VnfTemplateDetailVO accessMethodsDetail = createMockVnfTemplateDetail("console,ssh-password,http,https");
         when(vnfTemplateDetailsDao.findDetail(templateId, VNF.AccessDetail.ACCESS_METHODS.name().toLowerCase())).thenReturn(accessMethodsDetail);
 
-        VnfTemplateDetailVO sshPortDetail = Mockito.mock(VnfTemplateDetailVO.class);
-        when(sshPortDetail.getValue()).thenReturn(String.valueOf(sshPort));
+        VnfTemplateDetailVO sshPortDetail = createMockVnfTemplateDetail(String.valueOf(sshPort));
         when(vnfTemplateDetailsDao.findDetail(templateId, VNF.AccessDetail.SSH_PORT.name().toLowerCase())).thenReturn(sshPortDetail);
 
-        VnfTemplateDetailVO httpPortDetail = Mockito.mock(VnfTemplateDetailVO.class);
-        when(httpPortDetail.getValue()).thenReturn(String.valueOf(httpPort));
+        VnfTemplateDetailVO httpPortDetail = createMockVnfTemplateDetail(String.valueOf(httpPort));
         when(vnfTemplateDetailsDao.findDetail(templateId, VNF.AccessDetail.HTTP_PORT.name().toLowerCase())).thenReturn(httpPortDetail);
 
-        VnfTemplateDetailVO httpsPortDetail = Mockito.mock(VnfTemplateDetailVO.class);
-        when(httpsPortDetail.getValue()).thenReturn(String.valueOf(httpsPort));
+        VnfTemplateDetailVO httpsPortDetail = createMockVnfTemplateDetail(String.valueOf(httpsPort));
         when(vnfTemplateDetailsDao.findDetail(templateId, VNF.AccessDetail.HTTPS_PORT.name().toLowerCase())).thenReturn(httpsPortDetail);
 
         Set<Integer> ports = vnfTemplateManagerImpl.getOpenPortsForVnfAppliance(template);
@@ -386,4 +382,10 @@
         Mockito.verify(firewallRulesDao, Mockito.times(3)).persist(any());
         Mockito.verify(firewallService, Mockito.times(1)).applyIngressFwRules(publicIpId, owner);
     }
+
+    private static VnfTemplateDetailVO createMockVnfTemplateDetail(String value) {
+        VnfTemplateDetailVO httpsPortDetail = Mockito.mock(VnfTemplateDetailVO.class);
+        when(httpsPortDetail.getValue()).thenReturn(value);
+        return httpsPortDetail;
+    }
 }
