--- a/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
@@ -160,6 +160,12 @@
         }
     }
 
+    private static VirtualMachine createMockVirtualMachine() {
+        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
+        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        return vm;
+    }
+
 
     @Test
     public void testPurgeLinkedSnapshotEntitiesNoSnapshots() {
@@ -586,8 +592,7 @@
 
     @Test
     public void testExpungedVmResourcesLaterIfNeededFalse() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        VirtualMachine vm = createMockVirtualMachine();
         Mockito.doReturn(false).when(resourceCleanupService).isVmOfferingPurgeResourcesEnabled(1L);
         resourceCleanupService.purgeExpungedVmResourcesLaterIfNeeded(vm);
         Mockito.verify(resourceCleanupService, Mockito.never()).purgeExpungedResourceLater(Mockito.anyLong(), Mockito.any());
@@ -595,8 +600,7 @@
 
     @Test
     public void testExpungedVmResourcesLaterIfNeeded() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        VirtualMachine vm = createMockVirtualMachine();
         Mockito.doReturn(true).when(resourceCleanupService).isVmOfferingPurgeResourcesEnabled(1L);
         Mockito.doNothing().when(resourceCleanupService).purgeExpungedResourceLater(Mockito.anyLong(), Mockito.any());
         resourceCleanupService.purgeExpungedVmResourcesLaterIfNeeded(vm);
