--- a/server/src/test/java/com/cloud/consoleproxy/ConsoleProxyManagerTest.java
+++ b/server/src/test/java/com/cloud/consoleproxy/ConsoleProxyManagerTest.java
@@ -67,6 +67,8 @@
     DataCenterDao dataCenterDaoMock;
     @Mock
     NetworkDao networkDaoMock;
+    @Mock
+    NetworkVO badNetworkMock;
 
     @Mock
     Logger loggerMock;
@@ -143,21 +145,20 @@
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Public)))
                     .thenReturn(Collections.singletonList(network));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Public))))
-        .thenReturn(Collections.singletonList(badNetwork));
+        .thenReturn(Collections.singletonList(badNetworkMock));
 
         when(networkDaoMock.listByZoneSecurityGroup(anyLong()))
-                    .thenReturn(Collections.singletonList(badNetwork));
+                    .thenReturn(Collections.singletonList(badNetworkMock));
 
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForAdvancedZone(dc);
 
         Assert.assertNotNull(returnedNetwork);
         Assert.assertEquals(network, returnedNetwork);
-        Assert.assertNotEquals(badNetwork, returnedNetwork);
+        Assert.assertNotEquals(badNetworkMock, returnedNetwork);
     }
 
     @Test
@@ -169,9 +170,8 @@
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
-                    .thenReturn(Collections.singletonList(badNetwork));
+                    .thenReturn(Collections.singletonList(badNetworkMock));
 
         when(networkDaoMock.listByZoneSecurityGroup(anyLong()))
                     .thenReturn(Collections.singletonList(network));
@@ -179,7 +179,7 @@
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForAdvancedZone(dc);
 
         Assert.assertEquals(network, returnedNetwork);
-        Assert.assertNotEquals(badNetwork, returnedNetwork);
+        Assert.assertNotEquals(badNetworkMock, returnedNetwork);
     }
 
     @Test
@@ -191,17 +191,16 @@
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Guest))))
-                    .thenReturn(Collections.singletonList(badNetwork));
+                    .thenReturn(Collections.singletonList(badNetworkMock));
 
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForBasicZone(dc);
         Assert.assertNotNull(returnedNetwork);
         Assert.assertEquals(network, returnedNetwork);
-        Assert.assertNotEquals(badNetwork, returnedNetwork);
+        Assert.assertNotEquals(badNetworkMock, returnedNetwork);
     }
 
     @Test
@@ -213,18 +212,17 @@
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Guest))))
-                    .thenReturn(Collections.singletonList(badNetwork));
+                    .thenReturn(Collections.singletonList(badNetworkMock));
 
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForBasicZone(dc);
 
         Assert.assertNotNull(returnedNetwork);
         Assert.assertEquals(network, returnedNetwork);
-        Assert.assertNotEquals(badNetwork, returnedNetwork);
+        Assert.assertNotEquals(badNetworkMock, returnedNetwork);
     }
 
     //also test invalid input
@@ -237,12 +235,11 @@
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Guest))))
-                    .thenReturn(Collections.singletonList(badNetwork));
+                    .thenReturn(Collections.singletonList(badNetworkMock));
 
         consoleProxyManagerImplMock.getDefaultNetworkForBasicZone(dc);
     }
@@ -256,9 +253,8 @@
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
-                    .thenReturn(Collections.singletonList(badNetwork));
+                    .thenReturn(Collections.singletonList(badNetworkMock));
 
         when(networkDaoMock.listByZoneSecurityGroup(anyLong()))
                     .thenReturn(Collections.singletonList(network));
