--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -95,6 +95,7 @@
     NetworkDao netdao = mock(NetworkDao.class);
     GuestNetworkGuru guestGuru = mock(GuestNetworkGuru.class);
     BrocadeVcsGuestNetworkGuru guru;
+    private final DeploymentPlan deploymentPlan = mock(DeploymentPlan.class);
 
     @Before
     public void setUp() {
@@ -166,11 +167,10 @@
 
         when(netmodel.listNetworkOfferingServices(NETWORK_ID)).thenReturn(Arrays.asList(Service.Connectivity));
 
-        final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designednetwork = guru.design(offering, deploymentPlan, network, "", 1L, account);
         assertTrue(designednetwork != null);
         assertTrue(designednetwork.getBroadcastDomainType() == BroadcastDomainType.Vcs);
     }
@@ -187,11 +187,10 @@
         when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
         when(offering.getGuestType()).thenReturn(GuestType.Isolated);
 
-        final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designednetwork = guru.design(offering, deploymentPlan, network, "", 1L, account);
         assertTrue(designednetwork == null);
     }
 
@@ -209,11 +208,10 @@
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
-        final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designednetwork = guru.design(offering, deploymentPlan, network, "", 1L, account);
         assertTrue(designednetwork == null);
     }
 
