--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -193,10 +193,9 @@
         String hypervisorVersion = "default";
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
-        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
+        VMSnapshotVO vmSnapshot = createMockVMSnapshotVO(vmId);
         UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
         Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
-        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
         Mockito.when(userVmDao.findById(ArgumentMatchers.anyLong())).thenReturn(userVmVO);
@@ -209,8 +208,6 @@
         VMSnapshotTO vmSnapshotTO = Mockito.mock(VMSnapshotTO.class);
         Mockito.when(vmSnapshotHelper.getSnapshotWithParents(ArgumentMatchers.any(VMSnapshotVO.class))).thenReturn(vmSnapshotTO);
         Mockito.when(vmSnapshotDao.findById(ArgumentMatchers.anyLong())).thenReturn(vmSnapshot);
-        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
-        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
         HostVO hostVO = Mockito.mock(HostVO.class);
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
@@ -220,6 +217,14 @@
         Mockito.when(answer.getResult()).thenReturn(Boolean.TRUE);
         boolean result = vmStrategy.revertVMSnapshot(vmSnapshot);
         assertTrue(result);
+    }
+
+    public static VMSnapshotVO createMockVMSnapshotVO(Long vmId) {
+        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
+        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
+        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
+        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
+        return vmSnapshot;
     }
 
     @Test
@@ -270,10 +275,9 @@
         String hypervisorVersion = "default";
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
-        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
         UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
         Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
-        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
+        VMSnapshotVO vmSnapshot = createMockVMSnapshotVO(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
         Mockito.when(userVmDao.findById(ArgumentMatchers.anyLong())).thenReturn(userVmVO);
@@ -286,8 +290,6 @@
         VMSnapshotTO vmSnapshotTO = Mockito.mock(VMSnapshotTO.class);
         Mockito.when(vmSnapshotHelper.getSnapshotWithParents(ArgumentMatchers.any(VMSnapshotVO.class))).thenReturn(vmSnapshotTO);
         Mockito.when(vmSnapshotDao.findById(ArgumentMatchers.anyLong())).thenReturn(vmSnapshot);
-        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
-        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
         HostVO hostVO = Mockito.mock(HostVO.class);
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
@@ -150,6 +150,14 @@
         assertNotNull(snapshot);
     }
 
+    public static VMSnapshotVO createMockVMSnapshotVO(Long vmId) {
+        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
+        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
+        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
+        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
+        return vmSnapshot;
+    }
+
     @Test
     public void testRevertSnapshot() throws AgentUnavailableException, OperationTimedoutException {
         Long hostId = 1L;
@@ -159,10 +167,9 @@
         String hypervisorVersion = "default";
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
-        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
+        VMSnapshotVO vmSnapshot = createMockVMSnapshotVO(vmId);
         UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
         Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
-        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
         Mockito.when(userVmDao.findById(ArgumentMatchers.anyLong())).thenReturn(userVmVO);
@@ -175,8 +182,6 @@
         VMSnapshotTO vmSnapshotTO = Mockito.mock(VMSnapshotTO.class);
         Mockito.when(vmSnapshotHelper.getSnapshotWithParents(ArgumentMatchers.any(VMSnapshotVO.class))).thenReturn(vmSnapshotTO);
         Mockito.when(vmSnapshotDao.findById(ArgumentMatchers.anyLong())).thenReturn(vmSnapshot);
-        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
-        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(null);
         HostVO hostVO = Mockito.mock(HostVO.class);
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
@@ -207,10 +212,9 @@
         String hypervisorVersion = "default";
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
-        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
+        VMSnapshotVO vmSnapshot = createMockVMSnapshotVO(vmId);
         UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
         Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
-        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
         Mockito.when(userVmDao.findById(ArgumentMatchers.anyLong())).thenReturn(userVmVO);
@@ -223,8 +227,6 @@
         VMSnapshotTO vmSnapshotTO = Mockito.mock(VMSnapshotTO.class);
         Mockito.when(vmSnapshotHelper.getSnapshotWithParents(ArgumentMatchers.any(VMSnapshotVO.class))).thenReturn(vmSnapshotTO);
         Mockito.when(vmSnapshotDao.findById(ArgumentMatchers.anyLong())).thenReturn(vmSnapshot);
-        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
-        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(null);
         HostVO hostVO = Mockito.mock(HostVO.class);
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/MockVMSnapshotVOFactory.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/MockVMSnapshotVOFactory.java
@@ -0,0 +1,17 @@
+package org.apache.cloudstack.storage.vmsnapshot;
+
+import java.util.Date;
+
+import org.mockito.Mockito;
+
+import com.cloud.vm.snapshot.VMSnapshotVO;
+
+public class MockVMSnapshotVOFactory {
+    public static VMSnapshotVO createMockVMSnapshotVO(Long vmId) {
+        VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
+        Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
+        Mockito.when(vmSnapshot.getId()).thenReturn(1L);
+        Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
+        return vmSnapshot;
+    }
+}
