--- a/api/src/test/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmdTest.java
@@ -43,6 +43,8 @@
     public EntityManager _entityMgr;
     @Mock
     public NetworkService networkService;
+    @Mock
+    private Network mockCreatedNetwork;
     private ResponseGenerator responseGenerator;
 
     @InjectMocks
@@ -227,10 +229,9 @@
         ReflectionTestUtils.setField(cmd, "networkOfferingId", networkOfferingId);
         ReflectionTestUtils.setField(cmd, "associatedNetworkId", associatedNetworkId);
         NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
-        Network network = Mockito.mock(Network.class);
-        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
-        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
-        Mockito.when(_entityMgr.findById(Network.class, associatedNetworkId)).thenReturn(network);
+        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
+        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
+        Mockito.when(_entityMgr.findById(Network.class, associatedNetworkId)).thenReturn(mockCreatedNetwork);
         Mockito.when(networkOffering.getGuestType()).thenReturn(Network.GuestType.Isolated);
         try {
             cmd.getPhysicalNetworkId();
@@ -335,11 +336,10 @@
         ReflectionTestUtils.setField(cmd, "name", "testNetwork");
         ReflectionTestUtils.setField(cmd, "networkOfferingId", 1L);
         ReflectionTestUtils.setField(cmd, "zoneId", 1L);
-        Network createdNetwork = Mockito.mock(Network.class);
         NetworkResponse response = Mockito.mock(NetworkResponse.class);
         responseGenerator = Mockito.mock(ResponseGenerator.class);
-        Mockito.when(networkService.createGuestNetwork(cmd)).thenReturn(createdNetwork);
-        Mockito.when(responseGenerator.createNetworkResponse(ResponseObject.ResponseView.Restricted, createdNetwork)).thenReturn(response);
+        Mockito.when(networkService.createGuestNetwork(cmd)).thenReturn(mockCreatedNetwork);
+        Mockito.when(responseGenerator.createNetworkResponse(ResponseObject.ResponseView.Restricted, mockCreatedNetwork)).thenReturn(response);
         cmd._responseGenerator = responseGenerator;
 
         try {
