--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -252,6 +252,8 @@
     @Mock
     HostDao hostDao;
     @Mock
+    private HostVO mockedHostVO;
+    @Mock
     NetworkDetailsDao networkDetailsDao;
     @Mock
     SecurityGroupDao securityGroupDao;
@@ -643,7 +645,6 @@
         LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
         Network publicNetwork = mock(Network.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        HostVO hostVO = mock(HostVO.class);
         TungstenAnswer getTungstenLoadBalancerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateLoadBalancerServiceInstanceAnswer = MockTungstenAnswerFactory.get(true);
         Answer updateTungstenLoadbalancerStatsAnswer = mock(Answer.class);
@@ -658,7 +659,7 @@
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(publicNetwork);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
-        when(hostDao.listAllHostsByZoneAndHypervisorType(anyLong(), eq(Hypervisor.HypervisorType.KVM))).thenReturn(List.of(hostVO));
+        when(hostDao.listAllHostsByZoneAndHypervisorType(anyLong(), eq(Hypervisor.HypervisorType.KVM))).thenReturn(List.of(mockedHostVO));
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenLoadBalancerCommand.class), anyLong())).thenReturn(getTungstenLoadBalancerAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(UpdateLoadBalancerServiceInstanceCommand.class), anyLong())).thenReturn(updateLoadBalancerServiceInstanceAnswer);
         when(agentMgr.easySend(anyLong(), any(UpdateTungstenLoadbalancerStatsCommand.class))).thenReturn(updateTungstenLoadbalancerStatsAnswer);
@@ -1504,7 +1505,6 @@
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         NetworkDetailVO networkDetailVO = mock(NetworkDetailVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
-        HostVO hostVO = mock(HostVO.class);
         Answer answer = mock(Answer.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         ApiObjectBase apiObjectBase = mock(ApiObjectBase.class);
@@ -1520,7 +1520,7 @@
         when(vlan.getIp6Gateway()).thenReturn("fd00::1");
         when(networkDetailsDao.persist(any(NetworkDetailVO.class))).thenReturn(networkDetailVO);
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
-        when(hostDao.findByPublicIp(anyString())).thenReturn(hostVO);
+        when(hostDao.findByPublicIp(anyString())).thenReturn(mockedHostVO);
         when(agentMgr.easySend(anyLong(), any(SetupTungstenVRouterCommand.class))).thenReturn(answer);
         when(answer.getResult()).thenReturn(true);
         when(vlan.getIp6Cidr()).thenReturn("fd00::1/64");
