--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -150,6 +150,9 @@
     @Mock
     private VMInstanceVO vmInstanceVOMock;
 
+    @Mock
+    private NicSecondaryIp nicSecondaryIpResult;
+
     @Spy
     @InjectMocks
     ApiResponseHelper apiResponseHelper = new ApiResponseHelper();
@@ -245,11 +248,10 @@
 
     @Test
     public void setResponseIpAddressTestIpv4() {
-        NicSecondaryIp result = Mockito.mock(NicSecondaryIp.class);
         NicSecondaryIpResponse response = new NicSecondaryIpResponse();
-        setResult(result, "ipv4", "ipv6");
-
-        ApiResponseHelper.setResponseIpAddress(result, response);
+        setResult(nicSecondaryIpResult, "ipv4", "ipv6");
+
+        ApiResponseHelper.setResponseIpAddress(nicSecondaryIpResult, response);
 
         assertTrue(response.getIpAddr().equals("ipv4"));
     }
@@ -261,11 +263,10 @@
 
     @Test
     public void setResponseIpAddressTestIpv6() {
-        NicSecondaryIp result = Mockito.mock(NicSecondaryIp.class);
         NicSecondaryIpResponse response = new NicSecondaryIpResponse();
-        setResult(result, null, "ipv6");
-
-        ApiResponseHelper.setResponseIpAddress(result, response);
+        setResult(nicSecondaryIpResult, null, "ipv6");
+
+        ApiResponseHelper.setResponseIpAddress(nicSecondaryIpResult, response);
 
         assertTrue(response.getIpAddr().equals("ipv6"));
     }
