--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -1778,10 +1778,9 @@
         try {
             Mockito.when(migrateVolumeCmd.getVolumeId()).thenReturn(1L);
             Mockito.when(migrateVolumeCmd.getStoragePoolId()).thenReturn(2L);
-            VolumeVO vol = Mockito.mock(VolumeVO.class);
+            VolumeVO vol = createMockVolumeVo();
             Mockito.when(volumeDaoMock.findById(1L)).thenReturn(vol);
             Mockito.when(vol.getState()).thenReturn(Volume.State.Ready);
-            Mockito.when(vol.getPoolId()).thenReturn(1L);
             Mockito.when(vol.getInstanceId()).thenReturn(null);
             Mockito.when(vol.getDiskOfferingId()).thenReturn(1L);
             DiskOfferingVO diskOffering = Mockito.mock(DiskOfferingVO.class);
@@ -2215,9 +2214,8 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         Mockito.when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        VolumeVO existingVolume = Mockito.mock(VolumeVO.class);
+        VolumeVO existingVolume = createMockVolumeVo();
         Mockito.when(existingVolume.getState()).thenReturn(Volume.State.Ready);
-        when(existingVolume.getPoolId()).thenReturn(1L);
         StoragePoolVO destPrimaryStorage = Mockito.mock(StoragePoolVO.class);
         Mockito.when(destPrimaryStorage.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
         Mockito.when(primaryDataStoreDaoMock.findById(1L)).thenReturn(destPrimaryStorage);
@@ -2705,6 +2703,12 @@
     }
 
 
+    private static VolumeVO createMockVolumeVo() {
+        VolumeVO vol = Mockito.mock(VolumeVO.class);
+        Mockito.when(vol.getPoolId()).thenReturn(1L);
+        return vol;
+    }
+
     private <T> T invokePrivateMethod(String methodName, Class<?>[] paramTypes, Object... params) {
         try {
             java.lang.reflect.Method method = VolumeApiServiceImpl.class.getDeclaredMethod(methodName, paramTypes);
