--- a/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
@@ -67,6 +67,9 @@
     @Mock
     VMTemplateZoneDao templateZoneDao;
 
+    @Mock
+    private SearchCriteria<VMTemplateVO> sc;
+
     @Spy
     @InjectMocks
     VMTemplateDaoImpl templateDao = new VMTemplateDaoImpl();
@@ -346,7 +349,6 @@
         VMTemplateVO expectedTemplate = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(expectedTemplate);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         List<VMTemplateVO> templates = Collections.singletonList(expectedTemplate);
@@ -364,7 +366,6 @@
         VMTemplateVO template = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(template);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         doReturn(Collections.emptyList()).when(templateDao).listBy(any(SearchCriteria.class), any(Filter.class));
@@ -380,7 +381,6 @@
         VMTemplateVO expectedTemplate = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(expectedTemplate);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         List<VMTemplateVO> templates = Collections.singletonList(expectedTemplate);
@@ -398,7 +398,6 @@
         VMTemplateVO expectedTemplate = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(expectedTemplate);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         List<VMTemplateVO> templates = Collections.singletonList(expectedTemplate);
