--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -348,6 +348,12 @@
         f.set(configKey, o);
     }
 
+    private static NetworkVO createMockNetworkVO() {
+        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
+        Mockito.when(networkVO.getVpcId()).thenReturn(null);
+        return networkVO;
+    }
+
     @Test
     public void testGetPrivateVlanPairNoVlans() {
         Pair<String, Network.PVlanType> pair = service.getPrivateVlanPair(null, null, null);
@@ -526,7 +532,7 @@
         User callingUser = mock(User.class);
         UserVO userVO = mock(UserVO.class);
         Account callingAccount = mock(Account.class);
-        NetworkVO networkVO = mock(NetworkVO.class);
+        NetworkVO networkVO = createMockNetworkVO();
         NicVO nicVO = mock(NicVO.class);
         List<IPAddressVO> addresses = new ArrayList<>();
         List<IpAddressTO> ips = new ArrayList<>();
@@ -536,7 +542,6 @@
 
         CallContext.register(callingUser, callingAccount);
         Mockito.when(CallContext.current()).thenReturn(callContextMock);
-        Mockito.when(networkVO.getVpcId()).thenReturn(null);
 
         Pair<Integer, Integer> updatedMtus = service.validateMtuOnUpdate(networkVO, zoneId, publicMtu, privateMtu);
         Assert.assertEquals(publicMtu, updatedMtus.first());
@@ -704,8 +709,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testCheckAndUpdateNetworkDnsServiceFailure() {
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
-        Mockito.when(networkVO.getVpcId()).thenReturn(null);
+        NetworkVO networkVO = createMockNetworkVO();
         Long offeringId = 1L;
         NetworkOffering offering = Mockito.mock(NetworkOffering.class);
         Mockito.when(offering.getId()).thenReturn(offeringId);
