--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -162,10 +162,8 @@
         when(networkModel.isProviderForNetwork(Provider.NiciraNvp, NETWORK_ID)).thenReturn(true);
         when(ntwkSrvcDao.canProviderSupportServiceInNetwork(NETWORK_ID, Service.Connectivity, Provider.NiciraNvp)).thenReturn(true);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDevice();
         when(niciraNvpDao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
-        when(device.getHostId()).thenReturn(NICIRA_NVP_HOST_ID);
 
         HostVO niciraNvpHost = mock(HostVO.class);
         when(niciraNvpHost.getId()).thenReturn(NICIRA_NVP_HOST_ID);
@@ -278,10 +276,8 @@
         when(networkModel.isProviderForNetwork(Provider.NiciraNvp, NETWORK_ID)).thenReturn(true);
         when(ntwkSrvcDao.canProviderSupportServiceInNetwork(NETWORK_ID, Service.Connectivity, Provider.NiciraNvp)).thenReturn(true);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDevice();
         when(niciraNvpDao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
-        when(device.getHostId()).thenReturn(NICIRA_NVP_HOST_ID);
 
         HostVO niciraNvpHost = mock(HostVO.class);
         when(niciraNvpHost.getId()).thenReturn(NICIRA_NVP_HOST_ID);
@@ -331,10 +327,8 @@
         when(networkModel.isProviderForNetwork(Provider.NiciraNvp, NETWORK_ID)).thenReturn(true);
         when(ntwkSrvcDao.canProviderSupportServiceInNetwork(NETWORK_ID, Service.Connectivity, Provider.NiciraNvp)).thenReturn(true);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDevice();
         when(niciraNvpDao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
-        when(device.getHostId()).thenReturn(NICIRA_NVP_HOST_ID);
 
         HostVO niciraNvpHost = mock(HostVO.class);
         when(niciraNvpHost.getId()).thenReturn(NICIRA_NVP_HOST_ID);
@@ -383,10 +377,8 @@
         when(networkModel.isProviderForNetwork(Provider.NiciraNvp, NETWORK_ID)).thenReturn(true);
         when(ntwkSrvcDao.canProviderSupportServiceInNetwork(NETWORK_ID, Service.Connectivity, Provider.NiciraNvp)).thenReturn(true);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDevice();
         when(niciraNvpDao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
-        when(device.getHostId()).thenReturn(NICIRA_NVP_HOST_ID);
 
         HostVO niciraNvpHost = mock(HostVO.class);
         when(niciraNvpHost.getId()).thenReturn(NICIRA_NVP_HOST_ID);
@@ -418,4 +410,11 @@
 
         element.implement(network, offering, dest, context);
     }
+
+    private static NiciraNvpDeviceVO createMockNiciraNvpDevice() {
+        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        when(device.getId()).thenReturn(1L);
+        when(device.getHostId()).thenReturn(NICIRA_NVP_HOST_ID);
+        return device;
+    }
 }
