--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -220,16 +220,10 @@
         when(offering.getGuestType()).thenReturn(GuestType.Isolated);
 
         final List<PublicIpAddress> ipAddresses = new ArrayList<PublicIpAddress>();
-        final PublicIpAddress pipReleased = mock(PublicIpAddress.class);
-        final PublicIpAddress pipAllocated = mock(PublicIpAddress.class);
         final Ip ipReleased = new Ip("42.10.10.10");
         final Ip ipAllocated = new Ip("10.10.10.10");
-        when(pipAllocated.getState()).thenReturn(IpAddress.State.Allocated);
-        when(pipAllocated.getAddress()).thenReturn(ipAllocated);
-        when(pipAllocated.getNetmask()).thenReturn("255.255.255.0");
-        when(pipReleased.getState()).thenReturn(IpAddress.State.Releasing);
-        when(pipReleased.getAddress()).thenReturn(ipReleased);
-        when(pipReleased.getNetmask()).thenReturn("255.255.255.0");
+        final PublicIpAddress pipReleased = createMockPublicIpAddress(ipReleased, IpAddress.State.Releasing);
+        final PublicIpAddress pipAllocated = createMockPublicIpAddress(ipAllocated, IpAddress.State.Allocated);
         ipAddresses.add(pipAllocated);
         ipAddresses.add(pipReleased);
 
@@ -418,4 +412,12 @@
 
         element.implement(network, offering, dest, context);
     }
+
+    private static PublicIpAddress createMockPublicIpAddress(final Ip address, final IpAddress.State state) {
+        final PublicIpAddress publicIpAddress = mock(PublicIpAddress.class);
+        when(publicIpAddress.getState()).thenReturn(state);
+        when(publicIpAddress.getAddress()).thenReturn(address);
+        when(publicIpAddress.getNetmask()).thenReturn("255.255.255.0");
+        return publicIpAddress;
+    }
 }
