--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -150,6 +150,8 @@
     NicDao nicDao;
     @Mock
     AgentManager agentMgr;
+    @Mock
+    private VirtualMachine virtualMachine;
 
     TungstenGuestNetworkGuru guru;
 
@@ -497,11 +499,10 @@
         final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
-        final VirtualMachine vm = mock(VirtualMachine.class);
         final HostVO hostVO = mock(HostVO.class);
 
         when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        when(vmProfile.getVirtualMachine()).thenReturn(virtualMachine);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
         guru.rollbackMigration(nic, network, vmProfile, context, context);
@@ -515,11 +516,10 @@
         final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
-        final VirtualMachine vm = mock(VirtualMachine.class);
         final HostVO hostVO = mock(HostVO.class);
 
         when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        when(vmProfile.getVirtualMachine()).thenReturn(virtualMachine);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
         guru.commitMigration(nic, network, vmProfile, context, context);
