--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -157,6 +157,12 @@
         Assert.assertEquals(expected, accountPSResourceCount);
     }
 
+    private static UserVm createMockUserVm(VirtualMachine.State state) {
+        UserVm vm = Mockito.mock(UserVm.class);
+        Mockito.when(vm.getState()).thenReturn(state);
+        return vm;
+    }
+
     @Test
     public void testCheckAndUpdateVolumeAccountResourceCountSameSize() {
         runCheckAndUpdateVolumeAccountResourceCountTest(10L, 10L);
@@ -442,8 +448,7 @@
     public void testVolumeInactiveVmStopped() {
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
-        UserVm vm = Mockito.mock(UserVm.class);
-        Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Stopped);
+        UserVm vm = createMockUserVm(VirtualMachine.State.Stopped);
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(vm);
         assertTrue(volumeOrchestrator.volumeInactive(volume));
     }
@@ -452,8 +457,7 @@
     public void testVolumeInactiveVmDestroyed() {
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
-        UserVm vm = Mockito.mock(UserVm.class);
-        Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Destroyed);
+        UserVm vm = createMockUserVm(VirtualMachine.State.Destroyed);
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(vm);
         assertTrue(volumeOrchestrator.volumeInactive(volume));
     }
@@ -462,8 +466,7 @@
     public void testVolumeInactiveVmRunning() {
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
-        UserVm vm = Mockito.mock(UserVm.class);
-        Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Running); // Active state
+        UserVm vm = createMockUserVm(VirtualMachine.State.Running); // Active state
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(vm);
         Assert.assertFalse(volumeOrchestrator.volumeInactive(volume));
     }
