--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
@@ -147,8 +147,8 @@
     public void finalizeMigrateForLocalStorageToHaveTargetHostGuid(){
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Map<Volume, StoragePool> volumeToPool = new HashMap<>();
-        Volume rootVolume = Mockito.mock(Volume.class);
-        Volume dataVolume = Mockito.mock(Volume.class);
+        Volume rootVolume = createMockDataVolume();
+        Volume dataVolume = createMockDataVolume();
         StoragePool localStorage = Mockito.mock(StoragePool.class);
         volumeToPool.put(rootVolume, localStorage);
         volumeToPool.put(dataVolume, localStorage);
@@ -161,7 +161,6 @@
         Mockito.when(localStorage.getId()).thenReturn(1L);
         Mockito.when(_storagePoolDao.findById(1L)).thenReturn(storagePoolVO);
         Mockito.when(rootVolume.getVolumeType()).thenReturn(Volume.Type.ROOT);
-        Mockito.when(dataVolume.getVolumeType()).thenReturn(Volume.Type.DATADISK);
         Mockito.when(localStorage.isLocal()).thenReturn(true);
         Pair<Long, Long> clusterAndHost = new Pair<>(1L, 1L);
 
@@ -647,4 +646,10 @@
         Mockito.when(configSummary.getMemorySizeMB()).thenReturn(1024);
         vMwareGuru.checkSourceVmResourcesAgainstSelectedOfferingResources(virtualMachineMO, 2, 1500, 2048);
     }
+
+    private static Volume createMockDataVolume() {
+        Volume dataVolume = Mockito.mock(Volume.class);
+        Mockito.when(dataVolume.getVolumeType()).thenReturn(Volume.Type.DATADISK);
+        return dataVolume;
+    }
 }
