--- a/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
@@ -446,8 +446,7 @@
 
     @Test
     public void getHostAndPortForKVMMaintenanceHostIfNeededReturnsNullForNonKVMHypervisor() {
-        HostVO host = Mockito.mock(HostVO.class);
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.XenServer);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.XenServer);
 
         Pair<String, Integer> result = consoleAccessManager.getHostAndPortForKVMMaintenanceHostIfNeeded(host, Map.of());
 
@@ -456,8 +455,7 @@
 
     @Test
     public void getHostAndPortForKVMMaintenanceHostIfNeededReturnsNullForNonMaintenanceResourceState() {
-        HostVO host = Mockito.mock(HostVO.class);
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.KVM);
         Mockito.when(host.getResourceState()).thenReturn(ResourceState.Enabled);
 
         Pair<String, Integer> result = consoleAccessManager.getHostAndPortForKVMMaintenanceHostIfNeeded(host, Map.of());
@@ -467,8 +465,7 @@
 
     @Test
     public void getHostAndPortForKVMMaintenanceHostIfNeededReturnsHostAndPortForValidKVMInMaintenance() {
-        HostVO host = Mockito.mock(HostVO.class);
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.KVM);
         Mockito.when(host.getResourceState()).thenReturn(ResourceState.ErrorInMaintenance);
 
         String address = "192.168.1.100";
@@ -501,13 +498,12 @@
     @Test
     public void getConsoleConnectionDetailsReturnsDetailsForExternalHypervisor() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        HostVO host = Mockito.mock(HostVO.class);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.External);
         ConsoleConnectionDetails details = Mockito.mock(ConsoleConnectionDetails.class);
 
         Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
         Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
         Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
 
         Mockito.doReturn(details).when(consoleAccessManager).getConsoleConnectionDetailsForExternalVm(Mockito.any(), Mockito.eq(vm), Mockito.eq(host));
@@ -521,7 +517,6 @@
     @Test
     public void getConsoleConnectionDetailsReturnsDetailsForKVMHypervisor() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         int port = 5900;
         String vmUuid = "vm-uuid";
@@ -533,7 +528,7 @@
         Mockito.when(vm.getUuid()).thenReturn(vmUuid);
         Mockito.when(vm.getHostName()).thenReturn(vmHostName);
         Mockito.when(vm.getVncPassword()).thenReturn(vncPassword);
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.KVM);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(hostPortInfo);
         Mockito.doReturn(new Ternary<>(hostAddress, null, null)).when(consoleAccessManager).parseHostInfo(Mockito.anyString());
@@ -548,14 +543,13 @@
     @Test
     public void getConsoleConnectionDetailsReturnsDetailsWithRDPForHyperV() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         Pair<String, Integer> hostPortInfo = new Pair<>(hostAddress, -9);
 
         Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
         Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
         Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.Hyperv);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.Hyperv);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(hostPortInfo);
         int port = 3389;
@@ -574,12 +568,11 @@
     @Test
     public void getConsoleConnectionDetailsReturnsNullHostInvalidPortWhenVncPortInfoIsMissing() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        HostVO host = Mockito.mock(HostVO.class);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.KVM);
 
         Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
         Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
         Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(new Pair<>(null, -1));
 
@@ -592,14 +585,13 @@
     @Test
     public void getConsoleConnectionDetailsSetsLocaleWhenKeyboardDetailIsPresent() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         Pair<String, Integer> hostPortInfo = new Pair<>(hostAddress, 5900);
 
         Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
         Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
         Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
-        Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
+        HostVO host = createMockHost(Hypervisor.HypervisorType.KVM);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of(VmDetailConstants.KEYBOARD, "en-us"));
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(hostPortInfo);
         Mockito.doReturn(new Ternary<>(hostAddress, null, null)).when(consoleAccessManager).parseHostInfo(Mockito.anyString());
@@ -787,4 +779,10 @@
         Assert.assertNull(endpoint.getWebsocketExtra());
         Assert.assertNull(endpoint.getWebsocketHost());
     }
+
+    private static HostVO createMockHost(Hypervisor.HypervisorType hypervisorType) {
+        HostVO host = Mockito.mock(HostVO.class);
+        Mockito.when(host.getHypervisorType()).thenReturn(hypervisorType);
+        return host;
+    }
 }
