--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,7 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    private final VirtualMachineTO virtualMachineTO = Mockito.mock(VirtualMachineTO.class);
 
     @Test
     public void testWrapperInstance() {
@@ -327,8 +328,7 @@
 
     @Test
     public void testPrepareForMigrationCommand() {
-        final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
-        final PrepareForMigrationCommand prepareCommand = new PrepareForMigrationCommand(machineTO);
+        final PrepareForMigrationCommand prepareCommand = new PrepareForMigrationCommand(virtualMachineTO);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -341,8 +341,7 @@
 
     @Test
     public void testMigrateCommand() {
-        final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
-        final MigrateCommand migrateCommand = new MigrateCommand("Test", "127.0.0.1", false, machineTO, false);
+        final MigrateCommand migrateCommand = new MigrateCommand("Test", "127.0.0.1", false, virtualMachineTO, false);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -635,10 +634,9 @@
 
     @Test
     public void testStartCommand() {
-        final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
         final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
 
-        final StartCommand startCommand = new StartCommand(vm, host, false);
+        final StartCommand startCommand = new StartCommand(virtualMachineTO, host, false);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -1614,12 +1612,11 @@
     public void testScaleVmCommand() {
         final String uuid = "6172d8b7-ba10-4a70-93f9-ecaf41f51d53";
 
-        final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
         final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
         final Host host = Mockito.mock(Host.class);
 
-        final ScaleVmCommand scaleVm = new ScaleVmCommand(machineTO);
+        final ScaleVmCommand scaleVm = new ScaleVmCommand(virtualMachineTO);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
