--- a/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
@@ -121,10 +121,9 @@
 
     @Test
     public void testGetSortedTemplatesListWithPreferredArch_NoPreferred() {
-        VMTemplateVO template1 = Mockito.mock(VMTemplateVO.class);
+        VMTemplateVO template1 = createMockVmTemplateWithId();
         when(template1.getId()).thenReturn(1L);
-        VMTemplateVO template2 = Mockito.mock(VMTemplateVO.class);
-        when(template2.getId()).thenReturn(2L);
+        VMTemplateVO template2 = createMockVmTemplateWithId();
         Map<Pair<Hypervisor.HypervisorType, CPU.CPUArch>, VMTemplateVO> uniqueTemplates = new HashMap<>();
         uniqueTemplates.put(new Pair<>(Hypervisor.HypervisorType.KVM, CPU.CPUArch.amd64), template1);
         uniqueTemplates.put(new Pair<>(Hypervisor.HypervisorType.KVM, CPU.CPUArch.arm64), template2);
@@ -202,6 +201,12 @@
         assertNull(result);
     }
 
+    private static VMTemplateVO createMockVmTemplateWithId() {
+        VMTemplateVO template2 = Mockito.mock(VMTemplateVO.class);
+        when(template2.getId()).thenReturn(2L);
+        return template2;
+    }
+
     private void mockTemplateZoneJoin() {
         VMTemplateZoneVO templateZoneVO = mock(VMTemplateZoneVO.class);
         SearchBuilder<VMTemplateZoneVO> templateZoneVOSearchBuilder = mock(SearchBuilder.class);
