--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -261,6 +261,8 @@
     @Mock
     private Map<String, String> details;
 
+    private final KVMStoragePoolManager kvmStoragePoolManager = Mockito.mock(KVMStoragePoolManager.class);
+
     private static final String PHYSICAL_DISK_PATH = "/path/to/disk";
     private static final int DEV_ID = 1;
     private static final DiskDef.DiskBus DISK_BUS_TYPE = DiskDef.DiskBus.VIRTIO;
@@ -1507,7 +1509,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
-        final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final DiskTO diskTO = Mockito.mock(DiskTO.class);
         final VifDriver vifDriver = Mockito.mock(VifDriver.class);
@@ -1528,7 +1529,7 @@
         when(diskTO.getType()).thenReturn(Volume.Type.ISO);
 
         when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(vifDriver);
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -1604,7 +1605,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
-        final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
@@ -1619,7 +1619,7 @@
 
         when(vm.getNics()).thenReturn(new NicTO[]{nicTO});
 
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -1645,7 +1645,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
-        final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final DiskTO volume = Mockito.mock(DiskTO.class);
         final VifDriver vifDriver = Mockito.mock(VifDriver.class);
@@ -1666,7 +1665,7 @@
         when(volume.getType()).thenReturn(Volume.Type.ISO);
 
         when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(vifDriver);
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
         try {
             when(libvirtComputingResourceMock.getVolumePath(conn, volume)).thenThrow(URISyntaxException.class);
         } catch (final LibvirtException e) {
@@ -1701,7 +1700,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
-        final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final DiskTO volume = Mockito.mock(DiskTO.class);
 
@@ -1718,7 +1716,7 @@
         when(nicTO.getType()).thenReturn(TrafficType.Guest);
 
         BDDMockito.given(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).willAnswer(invocationOnMock -> {throw new InternalErrorException("Exception Occurred");});
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -5161,7 +5159,6 @@
 
         final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
 
-        final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
@@ -5170,7 +5167,7 @@
 
         final String vmName = "Test";
 
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
         when(vmSpec.getName()).thenReturn(vmName);
         when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
 
@@ -5204,7 +5201,6 @@
 
         final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
 
-        final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
@@ -5214,7 +5210,7 @@
 
         final String vmName = "Test";
 
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
@@ -5255,7 +5251,6 @@
 
         final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
 
-        final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
@@ -5265,7 +5260,7 @@
 
         final String vmName = "Test";
 
-        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(kvmStoragePoolManager);
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
