--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -249,6 +249,9 @@
     private NetworkVO networkMock;
 
     @Mock
+    private NetworkVO network2Mock;
+
+    @Mock
     private GuestOSDao guestOSDao;
 
     @Mock
@@ -3596,7 +3599,6 @@
         when(backupManager.getRootDiskInfoFromBackup(backup)).thenReturn(rootVmDiskInfo);
 
         NetworkVO network1 = mock(NetworkVO.class);
-        NetworkVO network2 = mock(NetworkVO.class);
         when(backupManager.getDataDiskInfoListFromBackup(backup)).thenReturn(List.of(new VmDiskInfo(diskOffering, 10L, 1000L, 2000L)));
         Mockito.when(backupManager.canCreateInstanceFromBackup(backupId)).thenReturn(true);
 
@@ -3775,7 +3777,6 @@
         when(backupManager.getRootDiskInfoFromBackup(backup)).thenReturn(rootVmDiskInfo);
 
         NetworkVO network1 = mock(NetworkVO.class);
-        NetworkVO network2 = mock(NetworkVO.class);
         when(backupManager.getDataDiskInfoListFromBackup(backup)).thenReturn(List.of(new VmDiskInfo(diskOffering, 10L, 1000L, 2000L)));
         Mockito.when(backupManager.canCreateInstanceFromBackup(backupId)).thenReturn(true);
 
