--- a/server/src/test/java/com/cloud/consoleproxy/ConsoleProxyManagerTest.java
+++ b/server/src/test/java/com/cloud/consoleproxy/ConsoleProxyManagerTest.java
@@ -67,6 +67,8 @@
     DataCenterDao dataCenterDaoMock;
     @Mock
     NetworkDao networkDaoMock;
+    @Mock
+    NetworkVO defaultNetwork;
 
     @Mock
     Logger loggerMock;
@@ -142,10 +144,9 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Public)))
-                    .thenReturn(Collections.singletonList(network));
+                    .thenReturn(Collections.singletonList(defaultNetwork));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Public))))
         .thenReturn(Collections.singletonList(badNetwork));
@@ -156,7 +157,7 @@
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForAdvancedZone(dc);
 
         Assert.assertNotNull(returnedNetwork);
-        Assert.assertEquals(network, returnedNetwork);
+        Assert.assertEquals(defaultNetwork, returnedNetwork);
         Assert.assertNotEquals(badNetwork, returnedNetwork);
     }
 
@@ -168,17 +169,16 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
 
         when(networkDaoMock.listByZoneSecurityGroup(anyLong()))
-                    .thenReturn(Collections.singletonList(network));
+                    .thenReturn(Collections.singletonList(defaultNetwork));
 
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForAdvancedZone(dc);
 
-        Assert.assertEquals(network, returnedNetwork);
+        Assert.assertEquals(defaultNetwork, returnedNetwork);
         Assert.assertNotEquals(badNetwork, returnedNetwork);
     }
 
@@ -190,17 +190,16 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
-                    .thenReturn(Collections.singletonList(network));
+                    .thenReturn(Collections.singletonList(defaultNetwork));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Guest))))
                     .thenReturn(Collections.singletonList(badNetwork));
 
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForBasicZone(dc);
         Assert.assertNotNull(returnedNetwork);
-        Assert.assertEquals(network, returnedNetwork);
+        Assert.assertEquals(defaultNetwork, returnedNetwork);
         Assert.assertNotEquals(badNetwork, returnedNetwork);
     }
 
@@ -212,10 +211,9 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
-                    .thenReturn(Collections.singletonList(network));
+                    .thenReturn(Collections.singletonList(defaultNetwork));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Guest))))
                     .thenReturn(Collections.singletonList(badNetwork));
@@ -223,7 +221,7 @@
         NetworkVO returnedNetwork = consoleProxyManagerImplMock.getDefaultNetworkForBasicZone(dc);
 
         Assert.assertNotNull(returnedNetwork);
-        Assert.assertEquals(network, returnedNetwork);
+        Assert.assertEquals(defaultNetwork, returnedNetwork);
         Assert.assertNotEquals(badNetwork, returnedNetwork);
     }
 
@@ -236,10 +234,9 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
-                    .thenReturn(Collections.singletonList(network));
+                    .thenReturn(Collections.singletonList(defaultNetwork));
 
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), not(eq(TrafficType.Guest))))
                     .thenReturn(Collections.singletonList(badNetwork));
@@ -255,13 +252,12 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
 
         when(networkDaoMock.listByZoneSecurityGroup(anyLong()))
-                    .thenReturn(Collections.singletonList(network));
+                    .thenReturn(Collections.singletonList(defaultNetwork));
 
         consoleProxyManagerImplMock.getDefaultNetworkForAdvancedZone(dc);
     }
