--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -146,11 +146,13 @@
     StorageVMSnapshotStrategy vmStrategy;
     @Inject
     VMSnapshotDetailsDao vmSnapshotDetailsDao;
+    private Snapshot diskSnapshot;
 
     @Override
     @Before
     public void setUp() throws Exception {
         ComponentContext.initComponentsLifeCycle();
+        diskSnapshot = Mockito.mock(Snapshot.class);
     }
 
     @Test
@@ -227,7 +229,6 @@
         VMSnapshot vmSnapshot = Mockito.mock(VMSnapshot.class);
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
         SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
-        Snapshot snap = Mockito.mock(Snapshot.class);
         DataStore dataStore = Mockito.mock(DataStore.class);
 
         String volUuid = UUID.randomUUID().toString();
@@ -237,7 +238,7 @@
         when(vmSnapshot.getUuid()).thenReturn(vmUuid);
         when(vol.getDataStore()).thenReturn(dataStore);
         when(snapshotVO.getId()).thenReturn(1L);
-        when(_snapshotService.revertSnapshot(snapshotVO.getId())).thenReturn(snap);
+        when(_snapshotService.revertSnapshot(snapshotVO.getId())).thenReturn(diskSnapshot);
         vmStrategy.revertDiskSnapshot(vmSnapshot);
     }
 
@@ -246,7 +247,6 @@
         VMSnapshot vmSnapshot = Mockito.mock(VMSnapshot.class);
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
         SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
-        Snapshot snap = Mockito.mock(Snapshot.class);
         SnapshotInfo info = Mockito.mock(SnapshotInfo.class);
         SnapshotStrategy strategy = Mockito.mock(SnapshotStrategy.class);
         String volUuid = UUID.randomUUID().toString();
