--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -722,10 +722,9 @@
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(vm.getId()).thenReturn(1L);
 
-        Host srcHost = Mockito.mock(Host.class);
-        Mockito.when(srcHost.getId()).thenReturn(1L);
-
-        Host destHost = Mockito.mock(Host.class);
+        Host srcHost = createMockSrcHost();
+
+        Host destHost = createMockSrcHost();
         Mockito.when(destHost.getId()).thenReturn(2L);
 
         Mockito.when(clusterDao.findById(1L)).thenReturn(cluster);
@@ -951,4 +950,10 @@
 
         Mockito.verify(clusterDrsService, Mockito.times(2)).executeDrsPlan(Mockito.any(ClusterDrsPlanVO.class));
     }
+
+    private static Host createMockSrcHost() {
+        Host srcHost = Mockito.mock(Host.class);
+        Mockito.when(srcHost.getId()).thenReturn(1L);
+        return srcHost;
+    }
 }
