--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -657,13 +657,14 @@
         long domainId = 4L;
         long backupOfferingId = 5L;
 
+        vmInstanceVOMock = createMockVMInstanceVO(zoneId);
+
         when(createBackupScheduleCmdMock.getVmId()).thenReturn(vmId);
         when(createBackupScheduleCmdMock.getTimezone()).thenReturn("GMT");
         when(createBackupScheduleCmdMock.getIntervalType()).thenReturn(DateUtil.IntervalType.DAILY);
         when(createBackupScheduleCmdMock.getMaxBackups()).thenReturn(8);
 
         when(vmInstanceDao.findById(vmId)).thenReturn(vmInstanceVOMock);
-        when(vmInstanceVOMock.getDataCenterId()).thenReturn(zoneId);
         when(vmInstanceVOMock.getAccountId()).thenReturn(accountId);
         when(vmInstanceVOMock.getBackupOfferingId()).thenReturn(backupOfferingId);
 
@@ -933,9 +934,10 @@
         long vmId = 1L;
         long dataCenterId = 2L;
 
+        vmInstanceVOMock = createMockVMInstanceVO(dataCenterId);
+
         try (MockedStatic<CallContext> mockedCallContext = Mockito.mockStatic(CallContext.class)) {
             Mockito.when(vmInstanceDao.findById(vmId)).thenReturn(vmInstanceVOMock);
-            Mockito.when(vmInstanceVOMock.getDataCenterId()).thenReturn(dataCenterId);
             Mockito.when(backupManager.isDisabled(dataCenterId)).thenReturn(false);
 
             mockedCallContext.when(CallContext::current).thenReturn(callContextMock);
@@ -1948,9 +1950,8 @@
         Mockito.when(cmd.getId()).thenReturn(1L);
 
         // Mock VM for validation
-        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
+        VMInstanceVO vm = createMockVMInstanceVO(1L);
         Mockito.when(vmInstanceDao.findById(vmId)).thenReturn(vm);
-        Mockito.when(vm.getDataCenterId()).thenReturn(1L);
         overrideBackupFrameworkConfigValue();
         Mockito.doNothing().when(accountManager).checkAccess(Mockito.any(), Mockito.any(), Mockito.anyBoolean(), Mockito.any());
 
@@ -1983,9 +1984,8 @@
         Mockito.when(cmd.getId()).thenReturn(1L);
 
         // Mock VM for validation
-        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
+        VMInstanceVO vm = createMockVMInstanceVO(1L);
         Mockito.when(vmInstanceDao.findById(vmId)).thenReturn(vm);
-        Mockito.when(vm.getDataCenterId()).thenReturn(1L);
         overrideBackupFrameworkConfigValue();
         Mockito.doNothing().when(accountManager).checkAccess(Mockito.any(), Mockito.any(), Mockito.anyBoolean(), Mockito.any());
 
@@ -2623,6 +2623,12 @@
         when(offering.getId()).thenReturn(id);
         when(offering.getName()).thenReturn(name);
         return offering;
+    }
+
+    private static VMInstanceVO createMockVMInstanceVO(long dataCenterId) {
+        VMInstanceVO vmInstanceVOMock = Mockito.mock(VMInstanceVO.class);
+        Mockito.when(vmInstanceVOMock.getDataCenterId()).thenReturn(dataCenterId);
+        return vmInstanceVOMock;
     }
 
     @Test
