--- a/server/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateManagerImplTest.java
@@ -259,14 +259,11 @@
 
         UserVm vm = Mockito.mock(UserVm.class);
         when(vm.getId()).thenReturn(vmId);
-        NicVO nic1 = Mockito.mock(NicVO.class);
-        NicVO nic2 = Mockito.mock(NicVO.class);
-        NicVO nic3 = Mockito.mock(NicVO.class);
-        when(nic1.getDeviceId()).thenReturn(0);
+        NicVO nic1 = createMockNicVO(0);
+        NicVO nic3 = createMockNicVO(2);
         when(nic1.getIPv4Address()).thenReturn(ipAddress);
         when(nic1.getNetworkId()).thenReturn(networkId);
-        when(nic2.getDeviceId()).thenReturn(1);
-        when(nic3.getDeviceId()).thenReturn(2);
+        NicVO nic2 = createMockNicVO(1);
         Mockito.doReturn(Arrays.asList(nic1, nic2, nic3)).when(nicDao).listByVmId(vmId);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
@@ -386,4 +383,10 @@
         Mockito.verify(firewallRulesDao, Mockito.times(3)).persist(any());
         Mockito.verify(firewallService, Mockito.times(1)).applyIngressFwRules(publicIpId, owner);
     }
+
+    private static NicVO createMockNicVO(int deviceId) {
+        NicVO nic = Mockito.mock(NicVO.class);
+        when(nic.getDeviceId()).thenReturn(deviceId);
+        return nic;
+    }
 }
