--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,14 @@
         return method;
     }
 
+    private static DeployDestination createMockDeployDestination(Host destHost, Pod destPod) {
+        final DeployDestination dest = mock(DeployDestination.class);
+        when(dest.getHost()).thenReturn(destHost);
+        when(dest.getPod()).thenReturn(destPod);
+        when(dest.getCluster()).thenReturn(mock(Cluster.class));
+        return dest;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1159,10 +1167,7 @@
 
         Host destHost = mock(Host.class);
         Pod destPod = mock(Pod.class);
-        DeployDestination dest = mock(DeployDestination.class);
-        when(dest.getHost()).thenReturn(destHost);
-        when(dest.getPod()).thenReturn(destPod);
-        when(dest.getCluster()).thenReturn(mock(Cluster.class));
+        DeployDestination dest = createMockDeployDestination(destHost, destPod);
         when(destHost.getId()).thenReturn(1L);
         when(destPod.getId()).thenReturn(2L);
         when(_dpMgr.planDeployment(any(VirtualMachineProfileImpl.class), any(DataCenterDeployment.class), any(ExcludeList.class), any(DeploymentPlanner.class))).thenReturn(dest);
@@ -1255,10 +1260,7 @@
 
         Host destHost = mock(Host.class);
         Pod destPod = mock(Pod.class);
-        DeployDestination dest = mock(DeployDestination.class);
-        when(dest.getHost()).thenReturn(destHost);
-        when(dest.getPod()).thenReturn(destPod);
-        when(dest.getCluster()).thenReturn(mock(Cluster.class));
+        DeployDestination dest = createMockDeployDestination(destHost, destPod);
         when(destHost.getId()).thenReturn(1L);
         when(destPod.getId()).thenReturn(2L);
         when(_dpMgr.planDeployment(any(VirtualMachineProfileImpl.class), any(DataCenterDeployment.class), any(ExcludeList.class), any(DeploymentPlanner.class))).thenReturn(dest);
