--- a/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
@@ -104,6 +104,9 @@
     @Mock
     VpcManager vpcManager;
 
+    @Mock
+    private Account loadBalancerOwner;
+
     @Spy
     @InjectMocks
     LoadBalancingRulesManagerImpl lbr = new LoadBalancingRulesManagerImpl();
@@ -326,7 +329,6 @@
         UserVm userVm = Mockito.mock(UserVm.class);
         LoadBalancerVO loadBalancer = Mockito.mock(LoadBalancerVO.class);
         Network loadBalancerNetwork = Mockito.mock(Network.class);
-        Account owner = Mockito.mock(Account.class);
 
         when(vpcManager.isNetworkOnVpcEnabledConserveMode(Mockito.eq(loadBalancerNetwork))).thenReturn(false);
 
@@ -335,7 +337,7 @@
         when(nic.getNetworkId()).thenReturn(networkId);
         List<? extends Nic> nics = Collections.singletonList(nic);
         Mockito.doReturn(nics).when(_networkModel).getNics(anyLong());
-        Nic nicInLb = lbr.getVmNicInLoadBalancer(userVm, loadBalancer, loadBalancerNetwork, null, owner);
+        Nic nicInLb = lbr.getVmNicInLoadBalancer(userVm, loadBalancer, loadBalancerNetwork, null, loadBalancerOwner);
         Assert.assertEquals(nic, nicInLb);
     }
 
@@ -346,7 +348,6 @@
         when(userVm.getId()).thenReturn(vmId);
         LoadBalancerVO loadBalancer = Mockito.mock(LoadBalancerVO.class);
         Network loadBalancerNetwork = Mockito.mock(Network.class);
-        Account owner = Mockito.mock(Account.class);
 
         long networkTier2Id = 20L;
         NetworkVO networkTier2 = Mockito.mock(NetworkVO.class);
@@ -360,7 +361,7 @@
         Nic nic = Mockito.mock(Nic.class);
         when(_networkModel.getNicInNetwork(Mockito.eq(vmId), Mockito.eq(networkTier2Id))).thenReturn(nic);
 
-        Nic nicInLb = lbr.getVmNicInLoadBalancer(userVm, loadBalancer, loadBalancerNetwork, vmIdNetworkIdMap, owner);
+        Nic nicInLb = lbr.getVmNicInLoadBalancer(userVm, loadBalancer, loadBalancerNetwork, vmIdNetworkIdMap, loadBalancerOwner);
         Assert.assertEquals(nic, nicInLb);
     }
 
