--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -112,6 +112,7 @@
     private String dhcpProvider = "VirtualRouter";
     private NetworkGuru guru = mock(NetworkGuru.class);
     private NetworkExtensionElement networkExtensionElement;
+    private Network.IpAddresses nicImportIpAddresses;
 
     NetworkOfferingVO networkOffering = mock(NetworkOfferingVO.class);
 
@@ -126,6 +127,7 @@
     @Before
     public void setUp() {
         // make class-scope mocks
+        nicImportIpAddresses = Mockito.mock(Network.IpAddresses.class);
         testOrchestrator._nicDao = mock(NicDao.class);
         testOrchestrator._networksDao = mock(NetworkDao.class);
         testOrchestrator._networkModel = mock(NetworkModel.class);
@@ -774,9 +776,8 @@
     public void testGetGuestIpForNicImportL2Network() {
         Network network = Mockito.mock(Network.class);
         DataCenter dataCenter = Mockito.mock(DataCenter.class);
-        Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(network.getGuestType()).thenReturn(GuestType.L2);
-        Assert.assertNull(testOrchestrator.getSelectedIpForNicImport(network, dataCenter, ipAddresses));
+        Assert.assertNull(testOrchestrator.getSelectedIpForNicImport(network, dataCenter, nicImportIpAddresses));
     }
 
     @Test
@@ -797,7 +798,6 @@
     public void testGetGuestIpForNicImportBasicZoneAutomaticIP() {
         Network network = Mockito.mock(Network.class);
         DataCenter dataCenter = Mockito.mock(DataCenter.class);
-        Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(network.getGuestType()).thenReturn(GuestType.Shared);
         Mockito.when(dataCenter.getNetworkType()).thenReturn(DataCenter.NetworkType.Basic);
         long networkId = 1L;
@@ -811,7 +811,7 @@
         Mockito.when(network.getId()).thenReturn(networkId);
         Mockito.when(dataCenter.getId()).thenReturn(dataCenterId);
         Mockito.when(testOrchestrator._ipAddressDao.findBySourceNetworkIdAndDatacenterIdAndState(networkId, dataCenterId, State.Free)).thenReturn(ipAddressVO);
-        String ipAddress = testOrchestrator.getSelectedIpForNicImport(network, dataCenter, ipAddresses);
+        String ipAddress = testOrchestrator.getSelectedIpForNicImport(network, dataCenter, nicImportIpAddresses);
         Assert.assertEquals(freeIp, ipAddress);
     }
 
