--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -268,8 +268,8 @@
         final VDI vdi1 = Mockito.mock(VDI.class);
         final VDI vdi2 = Mockito.mock(VDI.class);
 
-        final NicTO nic1 = Mockito.mock(NicTO.class);
-        final NicTO nic2 = Mockito.mock(NicTO.class);
+        final NicTO nic1 = createMockNicTO(mac);
+        final NicTO nic2 = createMockNicTO(mac);
 
         final Network network1 = Mockito.mock(Network.class);
         final Network network2 = Mockito.mock(Network.class);
@@ -294,9 +294,6 @@
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
         when(vmSpec.getName()).thenReturn(vmName);
-
-        when(nic1.getMac()).thenReturn(mac);
-        when(nic2.getMac()).thenReturn(mac);
 
         when(xenServer610Resource.getVDIbyUuid(conn, volume1.getPath())).thenReturn(vdi1);
         when(xenServer610Resource.getVDIbyUuid(conn, volume2.getPath())).thenReturn(vdi2);
@@ -465,6 +462,12 @@
         assertFalse(answer.getResult());
     }
 
+    private static NicTO createMockNicTO(final String mac) {
+        final NicTO nic1 = Mockito.mock(NicTO.class);
+        when(nic1.getMac()).thenReturn(mac);
+        return nic1;
+    }
+
     VolumeTO MockVolumeTO(String path){
         VolumeTO vol = Mockito.mock(VolumeTO.class);
         when(vol.getPath()).thenReturn(path);
