--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -1270,7 +1270,6 @@
     @SuppressWarnings("unchecked")
     @Test
     public void testUpdateHostPasswordCommand() {
-        final XenServerUtilitiesHelper xenServerUtilitiesHelper = Mockito.mock(XenServerUtilitiesHelper.class);
         final Pair<Boolean, String> result = Mockito.mock(Pair.class);
 
         final UpdateHostPasswordCommand updatePwd = new UpdateHostPasswordCommand("test", "123", "127.0.0.1");
@@ -1285,12 +1284,10 @@
         final StringBuilder cmdLine = new StringBuilder();
         cmdLine.append(XenServerUtilitiesHelper.SCRIPT_CMD_PATH).append(VRScripts.UPDATE_HOST_PASSWD).append(' ').append(username).append(' ').append(newPassword);
 
+        final XenServerUtilitiesHelper xenServerUtilitiesHelper = createMockXenServerUtilitiesHelper(username, newPassword, cmdLine.toString(), hostIp, hostPasswd, result);
         when(citrixResourceBase.getXenServerUtilitiesHelper()).thenReturn(xenServerUtilitiesHelper);
-        when(xenServerUtilitiesHelper.buildCommandLine(XenServerUtilitiesHelper.SCRIPT_CMD_PATH, VRScripts.UPDATE_HOST_PASSWD, username, newPassword)).thenReturn(
-                        cmdLine.toString());
-
-        try {
-            when(xenServerUtilitiesHelper.executeSshWrapper(hostIp, 22, username, null, hostPasswd, cmdLine.toString())).thenReturn(result);
+
+        try {
             when(result.first()).thenReturn(true);
             when(result.second()).thenReturn("");
         } catch (final Exception e) {
@@ -1319,7 +1316,6 @@
     @SuppressWarnings("unchecked")
     @Test
     public void testUpdateHostPasswordCommandFail() {
-        final XenServerUtilitiesHelper xenServerUtilitiesHelper = Mockito.mock(XenServerUtilitiesHelper.class);
         final Pair<Boolean, String> result = Mockito.mock(Pair.class);
 
         final UpdateHostPasswordCommand updatePwd = new UpdateHostPasswordCommand("test", "123", "127.0.0.1");
@@ -1334,12 +1330,10 @@
         final StringBuilder cmdLine = new StringBuilder();
         cmdLine.append(XenServerUtilitiesHelper.SCRIPT_CMD_PATH).append(VRScripts.UPDATE_HOST_PASSWD).append(' ').append(username).append(' ').append(newPassword);
 
+        final XenServerUtilitiesHelper xenServerUtilitiesHelper = createMockXenServerUtilitiesHelper(username, newPassword, cmdLine.toString(), hostIp, hostPasswd, result);
         when(citrixResourceBase.getXenServerUtilitiesHelper()).thenReturn(xenServerUtilitiesHelper);
-        when(xenServerUtilitiesHelper.buildCommandLine(XenServerUtilitiesHelper.SCRIPT_CMD_PATH, VRScripts.UPDATE_HOST_PASSWD, username, newPassword)).thenReturn(
-                        cmdLine.toString());
-
-        try {
-            when(xenServerUtilitiesHelper.executeSshWrapper(hostIp, 22, username, null, hostPasswd, cmdLine.toString())).thenReturn(result);
+
+        try {
             when(result.first()).thenReturn(false);
             when(result.second()).thenReturn("");
         } catch (final Exception e) {
@@ -1920,6 +1914,19 @@
         assertTrue(answer.getResult());
     }
 
+    private static XenServerUtilitiesHelper createMockXenServerUtilitiesHelper(final String username, final String newPassword, final String cmdLine,
+                    final String hostIp, final String hostPasswd, final Pair<Boolean, String> result) {
+        final XenServerUtilitiesHelper xenServerUtilitiesHelper = Mockito.mock(XenServerUtilitiesHelper.class);
+        when(xenServerUtilitiesHelper.buildCommandLine(XenServerUtilitiesHelper.SCRIPT_CMD_PATH, VRScripts.UPDATE_HOST_PASSWD, username, newPassword)).thenReturn(
+                        cmdLine);
+        try {
+            when(xenServerUtilitiesHelper.executeSshWrapper(hostIp, 22, username, null, hostPasswd, cmdLine)).thenReturn(result);
+        } catch (final Exception e) {
+            fail(e.getMessage());
+        }
+        return xenServerUtilitiesHelper;
+    }
+
 }
 
 class NotAValidCommand extends Command {
