--- a/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
+++ b/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
@@ -167,9 +167,8 @@
         _resource.configure("BigSwitchBcfResource", _parameters);
 
         NetworkData networkdata = mock(NetworkData.class);
-        NetworkData.Network network = mock(NetworkData.Network.class);
+        NetworkData.Network network = createMockNetwork();
         when(networkdata.getNetwork()).thenReturn(network);
-        when(network.getId()).thenReturn("cccc");
         when(_bigswitchBcfApi.createNetwork((NetworkData)any())).thenThrow(new BigSwitchBcfApiException())
         .thenReturn(UUID.randomUUID().toString());
 
@@ -184,9 +183,8 @@
         _resource.configure("BigSwitchBcfResource", _parameters);
 
         NetworkData networkdata = mock(NetworkData.class);
-        NetworkData.Network network = mock(NetworkData.Network.class);
+        NetworkData.Network network = createMockNetwork();
         when(networkdata.getNetwork()).thenReturn(network);
-        when(network.getId()).thenReturn("cccc");
         doThrow(new BigSwitchBcfApiException()).when(_bigswitchBcfApi).createNetwork((NetworkData)any());
 
         CreateBcfSegmentCommand cmd = new CreateBcfSegmentCommand("tenantid", "tenantname",
@@ -428,4 +426,10 @@
         assertTrue(ans.getResult());
     }
 
+    private static NetworkData.Network createMockNetwork() {
+        NetworkData.Network network = mock(NetworkData.Network.class);
+        when(network.getId()).thenReturn("cccc");
+        return network;
+    }
+
 }
