--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -600,6 +600,12 @@
 
         Assert.assertEquals(expectedIsolation, resultUri.getScheme());
         Assert.assertEquals(expectedUri, resultUri.toString());
+    }
+
+    private static Ip createMockIp() {
+        Ip ip = Mockito.mock(Ip.class);
+        Mockito.when(ip.addr()).thenReturn("172.10.10.10");
+        return ip;
     }
 
     private NicProfile prepareMocksAndRunPrepareNic(VirtualMachine.Type vmType, boolean isDefaultNic, boolean isVpcRouter, boolean routerResourceHasCustomDns) {
@@ -804,8 +810,7 @@
         long dataCenterId = 1L;
         String freeIp = "172.10.10.10";
         IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
-        Ip ip = mock(Ip.class);
-        Mockito.when(ip.addr()).thenReturn(freeIp);
+        Ip ip = createMockIp();
         Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
         Mockito.when(ipAddressVO.getState()).thenReturn(State.Free);
         Mockito.when(network.getId()).thenReturn(networkId);
@@ -826,8 +831,7 @@
         long dataCenterId = 1L;
         String requestedIp = "172.10.10.10";
         IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
-        Ip ip = mock(Ip.class);
-        Mockito.when(ip.addr()).thenReturn(requestedIp);
+        Ip ip = createMockIp();
         Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
         Mockito.when(ipAddressVO.getState()).thenReturn(State.Free);
         Mockito.when(network.getId()).thenReturn(networkId);
@@ -849,8 +853,7 @@
         long dataCenterId = 1L;
         String requestedIp = "172.10.10.10";
         IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
-        Ip ip = mock(Ip.class);
-        Mockito.when(ip.addr()).thenReturn(requestedIp);
+        Ip ip = createMockIp();
         Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
         Mockito.when(ipAddressVO.getState()).thenReturn(State.Allocated);
         Mockito.when(network.getId()).thenReturn(networkId);
