--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -424,8 +424,7 @@
 
     @Test
     public void testVolumeInactiveNoVmId() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(null);
+        VolumeVO volume = createMockVolumeWithNoInstanceId();
         assertTrue(volumeOrchestrator.volumeInactive(volume));
         Mockito.verify(entityMgr, Mockito.never()).findById(Mockito.eq(UserVm.class), Mockito.anyLong());
     }
@@ -470,8 +469,7 @@
 
     @Test
     public void testGetVmNameOnVolumeNoVmId() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(null);
+        VolumeVO volume = createMockVolumeWithNoInstanceId();
         Assert.assertNull(volumeOrchestrator.getVmNameOnVolume(volume));
         Mockito.verify(entityMgr, Mockito.never()).findById(Mockito.eq(VirtualMachine.class), Mockito.anyLong());
     }
@@ -945,4 +943,10 @@
         return null;
     }
 
+    private static VolumeVO createMockVolumeWithNoInstanceId() {
+        VolumeVO volume = Mockito.mock(VolumeVO.class);
+        Mockito.when(volume.getInstanceId()).thenReturn(null);
+        return volume;
+    }
+
 }
