--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -249,6 +249,8 @@
     private ImportVmTasksManager importVmTasksManager;
     @Mock
     private SnapshotDao snapshotDao;
+    @Mock
+    private DiskOfferingVO dataDiskOfferingMock;
 
     @Mock
     private VMInstanceVO virtualMachine;
@@ -1043,14 +1045,13 @@
         StoragePoolVO selectedPool = mock(StoragePoolVO.class);
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
         DiskOfferingVO rootDiskOffering = mock(DiskOfferingVO.class);
-        DiskOfferingVO dataDiskOffering = mock(DiskOfferingVO.class);
 
         when(serviceOffering.getDiskOfferingId()).thenReturn(21L);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(selectedPool);
         when(diskOfferingDao.findById(21L)).thenReturn(rootDiskOffering);
-        when(diskOfferingDao.findById(22L)).thenReturn(dataDiskOffering);
+        when(diskOfferingDao.findById(22L)).thenReturn(dataDiskOfferingMock);
         when(volumeApiService.doesStoragePoolSupportDiskOffering(selectedPool, rootDiskOffering)).thenReturn(true);
-        when(volumeApiService.doesStoragePoolSupportDiskOffering(selectedPool, dataDiskOffering)).thenReturn(false);
+        when(volumeApiService.doesStoragePoolSupportDiskOffering(selectedPool, dataDiskOfferingMock)).thenReturn(false);
 
         unmanagedVMsManager.validateSelectedConversionStoragePoolForVddk(true, poolId,
                 serviceOffering, Map.of("1000-2", 22L));
@@ -1062,14 +1063,13 @@
         StoragePoolVO selectedPool = mock(StoragePoolVO.class);
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
         DiskOfferingVO rootDiskOffering = mock(DiskOfferingVO.class);
-        DiskOfferingVO dataDiskOffering = mock(DiskOfferingVO.class);
 
         when(serviceOffering.getDiskOfferingId()).thenReturn(31L);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(selectedPool);
         when(diskOfferingDao.findById(31L)).thenReturn(rootDiskOffering);
-        when(diskOfferingDao.findById(32L)).thenReturn(dataDiskOffering);
+        when(diskOfferingDao.findById(32L)).thenReturn(dataDiskOfferingMock);
         when(volumeApiService.doesStoragePoolSupportDiskOffering(selectedPool, rootDiskOffering)).thenReturn(true);
-        when(volumeApiService.doesStoragePoolSupportDiskOffering(selectedPool, dataDiskOffering)).thenReturn(true);
+        when(volumeApiService.doesStoragePoolSupportDiskOffering(selectedPool, dataDiskOfferingMock)).thenReturn(true);
 
         unmanagedVMsManager.validateSelectedConversionStoragePoolForVddk(true, poolId,
                 serviceOffering, Map.of("1000-2", 32L));
