--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,7 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    private final com.cloud.host.Host cloudHost = Mockito.mock(com.cloud.host.Host.class);
 
     @Test
     public void testWrapperInstance() {
@@ -610,8 +611,7 @@
 
     @Test
     public void testCheckOnHostCommand() {
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
-        final CheckOnHostCommand onHostCommand = new CheckOnHostCommand(host);
+        final CheckOnHostCommand onHostCommand = new CheckOnHostCommand(cloudHost);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -636,9 +636,8 @@
     @Test
     public void testStartCommand() {
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
-
-        final StartCommand startCommand = new StartCommand(vm, host, false);
+
+        final StartCommand startCommand = new StartCommand(vm, cloudHost, false);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
