--- a/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
+++ b/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
@@ -166,7 +166,7 @@
         UserVmVO vm = new UserVmVO(2L, "test", "test", 101L, Hypervisor.HypervisorType.Any, 21L, false, false, domainId, 200L, 1, 5L, "", 1L, null, "test");
 
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
-        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
+        LoadBalancerVMMapDao lb2VmMapDao = createMockLoadBalancerVMMapDao();
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
@@ -215,7 +215,7 @@
         UserVmVO vm = new UserVmVO(2L, "test", "test", 101L, Hypervisor.HypervisorType.Any, 21L, false, false, domainId, 200L, 1, 5L, "", 1L, null, "test");
 
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
-        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
+        LoadBalancerVMMapDao lb2VmMapDao = createMockLoadBalancerVMMapDao();
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
@@ -266,7 +266,7 @@
         UserVmVO vm = new UserVmVO(2L, "test", "test", 101L, Hypervisor.HypervisorType.Any, 21L, false, false, domainId, 200L, 1, 5L, "", 1L, null, "test");
 
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
-        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
+        LoadBalancerVMMapDao lb2VmMapDao = createMockLoadBalancerVMMapDao();
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
@@ -305,4 +305,8 @@
         CallContext.unregister();
     }
 
+    private static LoadBalancerVMMapDao createMockLoadBalancerVMMapDao() {
+        return Mockito.mock(LoadBalancerVMMapDao.class);
+    }
+
 }
