--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -95,10 +95,12 @@
     NetworkDao netdao = mock(NetworkDao.class);
     GuestNetworkGuru guestGuru = mock(GuestNetworkGuru.class);
     BrocadeVcsGuestNetworkGuru guru;
+    private DataCenter dcMock;
 
     @Before
     public void setUp() {
 
+        dcMock = mock(DataCenter.class);
         guru = new BrocadeVcsGuestNetworkGuru();
         ((GuestNetworkGuru)guru)._physicalNetworkDao = physnetdao;
         guru._brocadeVcsDao = vcsdao;
@@ -240,8 +242,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenter dc = mock(DataCenter.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dcMock);
 
         final HostVO brocadeHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
@@ -299,8 +300,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenter dc = mock(DataCenter.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dcMock);
 
         final HostVO brocadeHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
