--- a/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
+++ b/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
@@ -1238,9 +1238,8 @@
         Long accountId = 1L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
-
-        Mockito.doReturn(accountId).when(account).getId();
+        AccountVO account = createMockAccount();
+
         Mockito.when(cmd.getQuotaEnforce()).thenReturn(Boolean.TRUE);
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
 
@@ -1255,10 +1254,9 @@
         Long accountId = 1L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
+        AccountVO account = createMockAccount();
 
         Mockito.when(cmd.getMinBalance()).thenReturn(50.0);
-        Mockito.doReturn(accountId).when(account).getId();
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
 
         quotaResponseBuilderSpy.persistQuotaCredits(cmd, value, depositedOn, account, false);
@@ -1368,4 +1366,10 @@
         Assert.assertEquals(record2.getStartDate(), secondHistory.getStartDate());
         Assert.assertEquals(record2.getEndDate(), secondHistory.getEndDate());
     }
+
+    private static AccountVO createMockAccount() {
+        AccountVO account = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(1L).when(account).getId();
+        return account;
+    }
 }
