--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
@@ -647,4 +647,9 @@
         Mockito.when(configSummary.getMemorySizeMB()).thenReturn(1024);
         vMwareGuru.checkSourceVmResourcesAgainstSelectedOfferingResources(virtualMachineMO, 2, 1500, 2048);
     }
+
+    private static HostMO createMockHostMo(ManagedObjectReference mor) throws Exception {
+        HostMO hostMo = createMockHostMo(mor);
+        return hostMo;
+    }
 }
