--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -223,6 +223,9 @@
     DataCenterDao dcDao;
 
     @Mock
+    DataCenterVO dataCenter;
+
+    @Mock
     ConfigurationDao configDao;
 
     @Mock
@@ -1600,8 +1603,7 @@
         // Setup other mocks
         Mockito.when(dpdkHelper.isVMDpdkEnabled(vm.getId())).thenReturn(false);
         Mockito.when(affinityGroupVMMapDao.countAffinityGroupsForVm(vm.getId())).thenReturn(0L);
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
-        Mockito.when(dcDao.findById(srcHost.getDataCenterId())).thenReturn(dc);
+        Mockito.when(dcDao.findById(srcHost.getDataCenterId())).thenReturn(dataCenter);
         Mockito.doNothing().when(dpMgr).checkForNonDedicatedResources(Mockito.any(), Mockito.any(), Mockito.any());
         Mockito.doNothing().when(dpMgr).reorderHostsByPriority(Mockito.any(), Mockito.anyList());
 
@@ -2144,8 +2146,7 @@
             Mockito.anyList(), Mockito.anyList(), Mockito.any());
         Mockito.when(dpdkHelper.isVMDpdkEnabled(vm.getId())).thenReturn(false);
         Mockito.when(affinityGroupVMMapDao.countAffinityGroupsForVm(vm.getId())).thenReturn(0L);
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
-        Mockito.when(dcDao.findById(1L)).thenReturn(dc);
+        Mockito.when(dcDao.findById(1L)).thenReturn(dataCenter);
         Mockito.doNothing().when(dpMgr).checkForNonDedicatedResources(Mockito.any(), Mockito.any(), Mockito.any());
         Mockito.doNothing().when(dpMgr).reorderHostsByPriority(Mockito.any(), Mockito.anyList());
 
