--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1620,13 +1620,12 @@
 
     @Test
     public void isNativeTrue() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
         when(counterTO.getSource()).thenReturn(Counter.Source.CPU);
@@ -1637,13 +1636,12 @@
 
     @Test
     public void isNativeFalse() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
         when(counterTO.getSource()).thenReturn(Counter.Source.VIRTUALROUTER);
@@ -1654,13 +1652,12 @@
 
     @Test
     public void hasSourceVirtualRouterTrue() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
         when(counterTO.getSource()).thenReturn(Counter.Source.VIRTUALROUTER);
@@ -1671,13 +1668,12 @@
 
     @Test
     public void hasSourceVirtualRouterFalse() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
         when(counterTO.getSource()).thenReturn(Counter.Source.CPU);
@@ -1705,14 +1701,13 @@
 
     @Test
     public void getPolicyCounters() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO1 = Mockito.mock(ConditionTO.class);
         CounterTO counterTO1 = Mockito.mock(CounterTO.class);
         ConditionTO conditionTO2 = Mockito.mock(ConditionTO.class);
         CounterTO counterTO2 = Mockito.mock(CounterTO.class);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO1, conditionTO2));
         when(conditionTO1.getCounter()).thenReturn(counterTO1);
         when(conditionTO2.getCounter()).thenReturn(counterTO2);
@@ -1729,9 +1724,8 @@
 
     @Test
     public void getAutoscaleAction() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
 
         Map<String, Double> countersMap = new HashMap<>();
         Map<String, Integer> countersNumberMap = new HashMap<>();
@@ -1844,14 +1838,13 @@
 
     @Test
     public void setPerformanceMonitorCommandParams() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO1 = Mockito.mock(ConditionTO.class);
         CounterTO counterTO1 = Mockito.mock(CounterTO.class);
         ConditionTO conditionTO2 = Mockito.mock(ConditionTO.class);
         CounterTO counterTO2 = Mockito.mock(CounterTO.class);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO1, conditionTO2));
         when(policyTO.getDuration()).thenReturn(scaleUpPolicyDuration);
         when(conditionTO1.getCounter()).thenReturn(counterTO1);
@@ -2019,9 +2012,6 @@
 
     @Test
     public void updateCountersMap1() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        when(groupTO.getId()).thenReturn(vmGroupId);
-
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
         when(scaleUpPolicyTO.getDuration()).thenReturn(scaleUpPolicyDuration);
@@ -2042,7 +2032,8 @@
         when(scaleDownConditionTO.getCounter()).thenReturn(scaleDownCounterTO);
         when(scaleDownCounterTO.getId()).thenReturn(scaleDownCounterId);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
+        when(groupTO.getId()).thenReturn(vmGroupId);
 
         when(asGroupStatisticsDao.listDummyRecordsByVmGroup(eq(vmGroupId), any()))
                 .thenReturn(new ArrayList<>())
@@ -2074,9 +2065,6 @@
 
     @Test
     public void updateCountersMap2() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        when(groupTO.getId()).thenReturn(vmGroupId);
-
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
         when(scaleUpPolicyTO.getDuration()).thenReturn(scaleUpPolicyDuration);
@@ -2087,7 +2075,8 @@
         when(scaleUpConditionTO.getCounter()).thenReturn(scaleUpCounterTO);
         when(scaleUpCounterTO.getId()).thenReturn(scaleUpCounterId);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(scaleUpPolicyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(scaleUpPolicyTO));
+        when(groupTO.getId()).thenReturn(vmGroupId);
 
         when(asGroupStatisticsDao.listDummyRecordsByVmGroup(eq(vmGroupId), any()))
                 .thenReturn(new ArrayList<>())
@@ -2205,8 +2194,6 @@
 
     @Test
     public void monitorVirtualRouterAsGroup() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO scaleUpConditionTO = Mockito.mock(ConditionTO.class);
         CounterTO scaleUpCounterTO = Mockito.mock(CounterTO.class);
@@ -2221,7 +2208,7 @@
         when(scaleDownConditionTO.getCounter()).thenReturn(scaleDownCounterTO);
         when(scaleDownCounterTO.getSource()).thenReturn(Counter.Source.VIRTUALROUTER);
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
 
         when(asVmGroupMock.getId()).thenReturn(vmGroupId);
         when(asVmGroupMock.getMinMembers()).thenReturn(minMembers);
@@ -2363,8 +2350,6 @@
 
     @Test
     public void setGetAutoScaleMetricsCommandMetrics() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO scaleUpConditionTO = Mockito.mock(ConditionTO.class);
         CounterTO scaleUpCounterTO = Mockito.mock(CounterTO.class);
@@ -2383,7 +2368,7 @@
         when(scaleDownCounterTO.getProvider()).thenReturn(Network.Provider.VirtualRouter.getName());
         when(scaleDownCounterTO.getValue()).thenReturn(VirtualRouterAutoScaleCounter.NETWORK_TRANSMIT_AVERAGE_MBPS.toString());
 
-        when(groupTO.getPolicies()).thenReturn(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
 
         List<VirtualRouterAutoScale.AutoScaleMetrics> result = autoScaleManagerImplSpy.setGetAutoScaleMetricsCommandMetrics(groupTO);
 
@@ -2512,6 +2497,12 @@
                 "-" + asVmGroupMock.getNextVmSeq() + "-[a-z]{6}";
         Assert.assertTrue(result.first().matches(vmHostNamePattern));
         Assert.assertEquals(result.first(), result.second());
+    }
+
+    private static AutoScaleVmGroupTO createMockAutoScaleVmGroupTO(List<AutoScalePolicyTO> policies) {
+        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
+        when(groupTO.getPolicies()).thenReturn(policies);
+        return groupTO;
     }
 
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
