--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VmwareVmImplementerTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VmwareVmImplementerTest.java
@@ -84,6 +84,12 @@
         assertEquals(expectedResult, result);
     }
 
+    private static GuestOSVO createMockGuestOs() {
+        GuestOSVO guestOs = Mockito.mock(GuestOSVO.class);
+        Mockito.when(guestOs.getId()).thenReturn(200L);
+        return guestOs;
+    }
+
     @Test
     public void testConfigureNestedVirtualization(){
         setConfigValues(true, true, null);
@@ -159,9 +165,8 @@
 
     @Test
     public void testGetGuestOsMapping1() {
-        GuestOSVO guestOs = Mockito.mock(GuestOSVO.class);
+        GuestOSVO guestOs = createMockGuestOs();
         GuestOSHypervisorVO guestOsMapping = Mockito.mock(GuestOSHypervisorVO.class);
-        Mockito.when(guestOs.getId()).thenReturn(200L);
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(200L, "VMware", "8.0.1.0")).thenReturn(guestOsMapping);
         GuestOSHypervisorVO result = implementer.getGuestOsMapping(guestOs, "8.0.1.0");
         Assert.assertEquals(guestOsMapping, result);
@@ -169,9 +174,8 @@
 
     @Test
     public void testGetGuestOsMapping2() {
-        GuestOSVO guestOs = Mockito.mock(GuestOSVO.class);
+        GuestOSVO guestOs = createMockGuestOs();
         GuestOSHypervisorVO guestOsMapping = Mockito.mock(GuestOSHypervisorVO.class);
-        Mockito.when(guestOs.getId()).thenReturn(200L);
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(200L, "VMware", "8.0.1.0")).thenReturn(null);
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(200L, "VMware", "8.0.1")).thenReturn(guestOsMapping);
         GuestOSHypervisorVO result = implementer.getGuestOsMapping(guestOs, "8.0.1.0");
@@ -180,8 +184,7 @@
 
     @Test
     public void testGetGuestOsMapping3() {
-        GuestOSVO guestOs = Mockito.mock(GuestOSVO.class);
-        Mockito.when(guestOs.getId()).thenReturn(200L);
+        GuestOSVO guestOs = createMockGuestOs();
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(200L, "VMware", "8.0.1.0")).thenReturn(null);
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(200L, "VMware", "8.0.1")).thenReturn(null);
         GuestOSHypervisorVO result = implementer.getGuestOsMapping(guestOs, "8.0.1.0");
@@ -190,8 +193,7 @@
 
     @Test
     public void testGetGuestOsMapping4() {
-        GuestOSVO guestOs = Mockito.mock(GuestOSVO.class);
-        Mockito.when(guestOs.getId()).thenReturn(200L);
+        GuestOSVO guestOs = createMockGuestOs();
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(200L, "VMware", "8.0")).thenReturn(null);
         GuestOSHypervisorVO result = implementer.getGuestOsMapping(guestOs, "8.0");
         Assert.assertNull(result);
