--- a/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
@@ -334,6 +334,12 @@
         return nic;
     }
 
+    private static NetworkVO createMockPublicNetwork(Networks.BroadcastDomainType broadcastDomainType) {
+        NetworkVO publicNetworkMock = Mockito.mock(NetworkVO.class);
+        Mockito.when(publicNetworkMock.getBroadcastDomainType()).thenReturn(broadcastDomainType);
+        return publicNetworkMock;
+    }
+
     @Test
     public void configurePublicVrNicBasedOnSourceNatIpTestConfigureVxLanNic() {
         String vlanTag = "200";
@@ -370,7 +376,7 @@
         NicProfile expected = getExpectedNicProfile(false, vlanTag);
 
         Ip ipMock = Mockito.mock(Ip.class);
-        NetworkVO publicNetworkMock = Mockito.mock(NetworkVO.class);
+        NetworkVO publicNetworkMock = createMockPublicNetwork(Networks.BroadcastDomainType.Vlan);
         long networkId = 1L;
         Mockito.when(publicIpMock.getAddress()).thenReturn(ipMock);
         Mockito.when(ipMock.addr()).thenReturn(expected.getIPv4Address());
@@ -379,7 +385,6 @@
         Mockito.when(publicIpMock.getMacAddress()).thenReturn(expected.getMacAddress());
         Mockito.when(publicIpMock.getNetworkId()).thenReturn(networkId);
         Mockito.when(networkDao.findById(networkId)).thenReturn(publicNetworkMock);
-        Mockito.when(publicNetworkMock.getBroadcastDomainType()).thenReturn(Networks.BroadcastDomainType.Vlan);
         Mockito.when(publicIpMock.getVlanTag()).thenReturn(vlanTag);
 
         networkHelperSpy.configurePublicVrNicBasedOnSourceNatIp(nicProfile, publicIpMock);
@@ -400,7 +405,6 @@
         NicProfile expected = getExpectedNicProfile(false, vlanTag);
 
         Ip ipMock = Mockito.mock(Ip.class);
-        NetworkVO publicNetworkMock = Mockito.mock(NetworkVO.class);
         long networkId = 1L;
         Mockito.when(publicIpMock.getAddress()).thenReturn(ipMock);
         Mockito.when(ipMock.addr()).thenReturn(expected.getIPv4Address());
@@ -408,8 +412,8 @@
         Mockito.when(publicIpMock.getNetmask()).thenReturn(expected.getIPv4Netmask());
         Mockito.when(publicIpMock.getMacAddress()).thenReturn(expected.getMacAddress());
         Mockito.when(publicIpMock.getNetworkId()).thenReturn(networkId);
+        NetworkVO publicNetworkMock = createMockPublicNetwork(Networks.BroadcastDomainType.Vlan);
         Mockito.when(networkDao.findById(networkId)).thenReturn(publicNetworkMock);
-        Mockito.when(publicNetworkMock.getBroadcastDomainType()).thenReturn(Networks.BroadcastDomainType.Vlan);
         Mockito.when(publicIpMock.getVlanTag()).thenReturn(vlanTag);
 
         networkHelperSpy.configurePublicVrNicBasedOnSourceNatIp(nicProfile, publicIpMock);
