--- a/server/src/test/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocatorTest.java
+++ b/server/src/test/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocatorTest.java
@@ -344,10 +344,9 @@
   @Test
   public void filterHostsWithUefiEnabledTestDetailWithUefiWithInvalidModeShouldNotFilterAnyHost() {
     List<HostVO> suitableHosts = new ArrayList<>(Arrays.asList(host1, host2, host3));
-    VMInstanceDetailVO userVmDetailVO = mock(VMInstanceDetailVO.class);
     String bootMode = "Invalid mode";
 
-    Mockito.doReturn(bootMode).when(userVmDetailVO).getValue();
+    VMInstanceDetailVO userVmDetailVO = createMockUserVmDetailVO(bootMode);
     Mockito.doReturn(userVmDetailVO).when(userVmDetailsDaoMock).findDetail(Mockito.anyLong(), Mockito.anyString());
     firstFitAllocatorSpy.filterHostsWithUefiEnabled(type, virtualMachineProfile, clusterId, podId, dcId, suitableHosts);
 
@@ -361,10 +360,9 @@
   public void filterHostsWithUefiEnabledTestDetailWithUefiWithLegacyModeShouldFilterHost() {
     List<HostVO> suitableHosts = new ArrayList<>(Arrays.asList(host1, host2, host3));
     List<HostVO> uefiHosts = new ArrayList<>(Arrays.asList(host2, host3));
-    VMInstanceDetailVO userVmDetailVO = mock(VMInstanceDetailVO.class);
     String bootMode = ApiConstants.BootMode.LEGACY.toString();
 
-    Mockito.doReturn(bootMode).when(userVmDetailVO).getValue();
+    VMInstanceDetailVO userVmDetailVO = createMockUserVmDetailVO(bootMode);
     Mockito.doReturn(userVmDetailVO).when(userVmDetailsDaoMock).findDetail(Mockito.anyLong(), Mockito.anyString());
     Mockito.doReturn(uefiHosts).when(hostDaoMock).listByHostCapability(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString());
     firstFitAllocatorSpy.filterHostsWithUefiEnabled(type, virtualMachineProfile, clusterId, podId, dcId, suitableHosts);
@@ -378,10 +376,9 @@
   public void filterHostsWithUefiEnabledTestDetailWithUefiWithSecureModeShouldFilterHost() {
     List<HostVO> suitableHosts = new ArrayList<>(Arrays.asList(host1, host2, host3));
     List<HostVO> uefiHosts = new ArrayList<>(Arrays.asList(host2, host3));
-    VMInstanceDetailVO userVmDetailVO = mock(VMInstanceDetailVO.class);
     String bootMode = ApiConstants.BootMode.SECURE.toString();
-
-    Mockito.doReturn(bootMode).when(userVmDetailVO).getValue();
+    VMInstanceDetailVO userVmDetailVO = createMockUserVmDetailVO(bootMode);
+
     Mockito.doReturn(userVmDetailVO).when(userVmDetailsDaoMock).findDetail(Mockito.anyLong(), Mockito.anyString());
     Mockito.doReturn(uefiHosts).when(hostDaoMock).listByHostCapability(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString());
     firstFitAllocatorSpy.filterHostsWithUefiEnabled(type, virtualMachineProfile, clusterId, podId, dcId, suitableHosts);
@@ -584,6 +581,12 @@
         0.9484375, hostByCombinedCapacity.get(1L), TOLERANCE);
   }
 
+  private static VMInstanceDetailVO createMockUserVmDetailVO(String bootMode) {
+    VMInstanceDetailVO userVmDetailVO = mock(VMInstanceDetailVO.class);
+    Mockito.doReturn(bootMode).when(userVmDetailVO).getValue();
+    return userVmDetailVO;
+  }
+
   List<CapacityVO> getHostCapacities() {
     CapacityVO cpuCapacity1 = mock(CapacityVO.class);
     when(cpuCapacity1.getHostOrPoolId()).thenReturn(1L);
