--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -179,6 +179,8 @@
     TungstenFabricLBHealthMonitorDao tungstenFabricLBHealthMonitorDao;
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
+    @Mock
+    private ReservationContext sharedReservationContext;
 
     TungstenElement tungstenElement;
 
@@ -584,7 +586,6 @@
         NicProfile nicProfile = new NicProfile();
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
         TungstenAnswer createTungstenVMAnswer = MockTungstenAnswerFactory.get(true);
@@ -600,7 +601,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(createTungstenNetworkPolicyAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
 
-        assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext));
+        assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, sharedReservationContext));
         assertEquals(Nic.ReservationStrategy.Create, nicProfile.getReservationStrategy());
         assertEquals(Networks.BroadcastDomainType.TUNGSTEN, nicProfile.getBroadcastType());
         assertEquals(Networks.BroadcastDomainType.TUNGSTEN.toUri("tf"), nicProfile.getBroadCastUri());
@@ -613,7 +614,6 @@
         NicProfile nicProfile = new NicProfile();
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
         TungstenAnswer createTungstenVMAnswer = MockTungstenAnswerFactory.get(true);
@@ -625,7 +625,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
 
-        assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext));
+        assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, sharedReservationContext));
         assertEquals(Nic.ReservationStrategy.Create, nicProfile.getReservationStrategy());
         assertEquals(Networks.BroadcastDomainType.TUNGSTEN, nicProfile.getBroadcastType());
         assertEquals(Networks.BroadcastDomainType.TUNGSTEN.toUri("tf"), nicProfile.getBroadCastUri());
@@ -638,7 +638,6 @@
         NicProfile nicProfile = new NicProfile();
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
         TungstenAnswer createTungstenVMAnswer = mock(TungstenAnswer.class);
@@ -650,7 +649,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
 
-        tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext);
+        tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, sharedReservationContext);
     }
 
     @Test
@@ -658,7 +657,6 @@
         Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
@@ -678,7 +676,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(deleteTungstenNetworkPolicyAnswer);
 
-        assertTrue(tungstenElement.release(network, nicProfile, virtualMachineProfile, reservationContext));
+        assertTrue(tungstenElement.release(network, nicProfile, virtualMachineProfile, sharedReservationContext));
     }
 
     @Test
@@ -686,7 +684,6 @@
         Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
@@ -701,7 +698,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswer);
 
-        assertTrue(tungstenElement.release(network, nicProfile, virtualMachineProfile, reservationContext));
+        assertTrue(tungstenElement.release(network, nicProfile, virtualMachineProfile, sharedReservationContext));
     }
 
     @Test(expected = CloudRuntimeException.class)
@@ -709,7 +706,6 @@
         Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
@@ -723,7 +719,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenThrow(IllegalArgumentException.class);
 
-        tungstenElement.release(network, nicProfile, virtualMachineProfile, reservationContext);
+        tungstenElement.release(network, nicProfile, virtualMachineProfile, sharedReservationContext);
     }
 
     @Test
@@ -731,7 +727,6 @@
         IPAddressVO ipAddressVO1 = mock(IPAddressVO.class);
         IPAddressVO ipAddressVO2 = mock(IPAddressVO.class);
         Network network = mock(Network.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         List<IPAddressVO> ipAddressVOList = Arrays.asList(ipAddressVO1, ipAddressVO2);
         TungstenAnswer tungstenDeleteFIPAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer tungstenDeleteNPAnswer = MockTungstenAnswerFactory.get(true);
@@ -740,7 +735,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenFloatingIpCommand.class), anyLong())).thenReturn(tungstenDeleteFIPAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(tungstenDeleteNPAnswer);
 
-        assertTrue(tungstenElement.destroy(network, reservationContext));
+        assertTrue(tungstenElement.destroy(network, sharedReservationContext));
     }
 
     @Test
@@ -756,7 +751,6 @@
     @Test
     public void shutdownProviderInstancesTest() throws ConcurrentOperationException {
         PhysicalNetworkServiceProvider physicalNetworkServiceProvider = mock(PhysicalNetworkServiceProvider.class);
-        ReservationContext reservationContext = mock(ReservationContext.class);
         PhysicalNetworkVO physicalNetworkVO = mock(PhysicalNetworkVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -775,7 +769,7 @@
         when(vlanDao.listVlansByNetworkIdIncludingRemoved(anyLong())).thenReturn(vlanVOList);
         when(hostPodDao.listByDataCenterId(anyLong())).thenReturn(hostPodVOList);
 
-        assertTrue(tungstenElement.shutdownProviderInstances(physicalNetworkServiceProvider, reservationContext));
+        assertTrue(tungstenElement.shutdownProviderInstances(physicalNetworkServiceProvider, sharedReservationContext));
         verify(networkServiceMapDao, times(1)).deleteByNetworkId(anyLong());
         verify(tungstenService, times(2)).removePublicNetworkSubnet(any(VlanVO.class));
         verify(tungstenService, times(1)).deletePublicNetwork(anyLong());
@@ -913,7 +907,6 @@
         Network network = mock(Network.class);
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         DeployDestination dest = mock(DeployDestination.class);
-        ReservationContext context = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO hostVO = mock(HostVO.class);
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(true);
@@ -924,7 +917,7 @@
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(tungstenAnswer);
 
-        assertTrue(tungstenElement.prepareMigration(nic, network, vm, dest, context));
+        assertTrue(tungstenElement.prepareMigration(nic, network, vm, dest, sharedReservationContext));
     }
 
     @Test
@@ -932,7 +925,6 @@
         NicProfile nic = mock(NicProfile.class);
         Network network = mock(Network.class);
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
-        ReservationContext src = mock(ReservationContext.class);
         ReservationContext dest = mock(ReservationContext.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         HostVO hostVO = mock(HostVO.class);
@@ -941,7 +933,7 @@
         when(vm.getVirtualMachine()).thenReturn(virtualMachine);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        tungstenElement.rollbackMigration(nic, network, vm, src, dest);
+        tungstenElement.rollbackMigration(nic, network, vm, sharedReservationContext, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
 
@@ -950,7 +942,6 @@
         NicProfile nic = mock(NicProfile.class);
         Network network = mock(Network.class);
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
-        ReservationContext src = mock(ReservationContext.class);
         ReservationContext dest = mock(ReservationContext.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         HostVO hostVO = mock(HostVO.class);
@@ -959,7 +950,7 @@
         when(vm.getVirtualMachine()).thenReturn(virtualMachine);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        tungstenElement.commitMigration(nic, network, vm, src, dest);
+        tungstenElement.commitMigration(nic, network, vm, sharedReservationContext, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
 }
