--- a/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
+++ b/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
@@ -110,6 +110,7 @@
     private AssignToLoadBalancerRuleCmd assignToLbRuleCmd;
     private ResponseGenerator responseGenerator;
     private SuccessResponse successResponseGenerator;
+    private AccountDao sharedAccountDao;
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
@@ -120,6 +121,7 @@
 
     @Before
     public void setUp() {
+        sharedAccountDao = Mockito.mock(AccountDao.class);
         assignToLbRuleCmd = new AssignToLoadBalancerRuleCmd() {
         };
 
@@ -217,7 +219,6 @@
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
         LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
-        AccountDao accountDao = Mockito.mock(AccountDao.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
         NicSecondaryIpDao nicSecIpDao =  Mockito.mock(NicSecondaryIpDao.class);
@@ -227,7 +228,7 @@
         _lbMgr._lbDao = lbDao;
         _lbMgr._lb2VmMapDao = lb2VmMapDao;
         _lbMgr._vmDao = userVmDao;
-        _lbMgr._accountDao = accountDao;
+        _lbMgr._accountDao = sharedAccountDao;
         _lbMgr._accountMgr = accountMgr;
         _lbMgr._networkDao = networkDao;
         _lbMgr._nicSecondaryIpDao = nicSecIpDao;
@@ -268,7 +269,6 @@
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
         LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
-        AccountDao accountDao = Mockito.mock(AccountDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         NicSecondaryIpDao nicSecIpDao =  Mockito.mock(NicSecondaryIpDao.class);
@@ -279,7 +279,7 @@
         _lbMgr._lbDao = lbDao;
         _lbMgr._lb2VmMapDao = lb2VmMapDao;
         _lbMgr._vmDao = userVmDao;
-        _lbMgr._accountDao = accountDao;
+        _lbMgr._accountDao = sharedAccountDao;
         _lbMgr._accountMgr = accountMgr;
         _lbMgr._networkDao = networkDao;
         _lbMgr._nicSecondaryIpDao = nicSecIpDao;
