--- a/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
@@ -130,6 +130,10 @@
 
     @Mock
     private Network networkMock;
+
+    private NetworkOffering createNetworkOfferingMock() {
+        return Mockito.mock(NetworkOffering.class);
+    }
 
     private NicProfile nicProfile = new NicProfile();
 
@@ -498,7 +502,7 @@
     @Test
     public void configurePublicNicTestConfigureDeviceId() throws InsufficientAddressCapacityException {
         PublicIp publicIpMock = Mockito.mock(PublicIp.class);
-        NetworkOffering networkOfferingMock = Mockito.mock(NetworkOffering.class);
+        NetworkOffering networkOfferingMock = createNetworkOfferingMock();
 
         Mockito.when(routerDeploymentDefinitionMock.isPublicNetwork()).thenReturn(true);
         Mockito.when(routerDeploymentDefinitionMock.getSourceNatIP()).thenReturn(publicIpMock);
@@ -517,7 +521,7 @@
     @Test
     public void configurePublicNicTestUpdateGuestNetworksIpv6Nic() throws InsufficientAddressCapacityException {
         PublicIp publicIpMock = Mockito.mock(PublicIp.class);
-        NetworkOffering networkOfferingMock = Mockito.mock(NetworkOffering.class);
+        NetworkOffering networkOfferingMock = createNetworkOfferingMock();
         DeployDestination deployDestinationMock = Mockito.mock(DeployDestination.class);
 
         Mockito.when(routerDeploymentDefinitionMock.isPublicNetwork()).thenReturn(true);
