--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -242,6 +242,9 @@
     HostDao hostDao;
 
     @Mock
+    private HostVO hostMock;
+
+    @Mock
     private NetworkDao networkDao;
 
     @Mock
@@ -1368,8 +1371,7 @@
 
         VolumeVO rootVolume = mock(VolumeVO.class);
         when(rootVolume.getPoolId()).thenReturn(poolId);
-        HostVO host = mock(HostVO.class);
-        when(hostDao.findById(hostId)).thenReturn(host);
+        when(hostDao.findById(hostId)).thenReturn(hostMock);
         StoragePoolVO pool = mock(StoragePoolVO.class);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
@@ -1424,8 +1426,7 @@
 
         VolumeVO rootVolume = mock(VolumeVO.class);
         when(rootVolume.getPoolId()).thenReturn(poolId);
-        HostVO host = mock(HostVO.class);
-        when(hostDao.findById(hostId)).thenReturn(host);
+        when(hostDao.findById(hostId)).thenReturn(hostMock);
         StoragePoolVO pool = mock(StoragePoolVO.class);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
