--- a/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
+++ b/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
@@ -370,6 +370,17 @@
         Mockito.doNothing().when(presetVariableHelperSpy).validateIfObjectIsNull(Mockito.any(), Mockito.anyLong(), Mockito.anyString());
     }
 
+    private static VolumeVO createMockVolumeVo(String uuid, String name, ProvisioningType provisioningType, Volume.Type volumeType, Long size, ImageFormat imageFormat) {
+        VolumeVO volumeVoMock = Mockito.mock(VolumeVO.class);
+        Mockito.doReturn(imageFormat).when(volumeVoMock).getFormat();
+        Mockito.doReturn(provisioningType).when(volumeVoMock).getProvisioningType();
+        Mockito.doReturn(volumeType).when(volumeVoMock).getVolumeType();
+        Mockito.doReturn(size).when(volumeVoMock).getSize();
+        Mockito.doReturn(name).when(volumeVoMock).getName();
+        Mockito.doReturn(uuid).when(volumeVoMock).getUuid();
+        return volumeVoMock;
+    }
+
     private GenericPresetVariable getGenericPresetVariableForTests() {
         GenericPresetVariable gpv = new GenericPresetVariable();
         gpv.setId("test_id");
@@ -775,21 +786,15 @@
         for (ImageFormat imageFormat : ImageFormat.values()) {
             Value expected = getValueForTests();
 
-            VolumeVO volumeVoMock = Mockito.mock(VolumeVO.class);
+            VolumeVO volumeVoMock = createMockVolumeVo(expected.getId(), expected.getName(), ProvisioningType.getProvisioningType(expected.getProvisioningType()), Volume.Type.valueOf(expected.getVolumeType()), expected.getSize(), imageFormat);
             Mockito.doReturn(volumeVoMock).when(volumeDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
             Mockito.doReturn(1L).when(volumeVoMock).getPoolId();
 
             mockMethodValidateIfObjectIsNull();
 
-            Mockito.doReturn(expected.getId()).when(volumeVoMock).getUuid();
-            Mockito.doReturn(expected.getName()).when(volumeVoMock).getName();
             Mockito.doReturn(expected.getDiskOffering()).when(presetVariableHelperSpy).getPresetVariableValueDiskOffering(Mockito.anyLong());
-            Mockito.doReturn(ProvisioningType.getProvisioningType(expected.getProvisioningType())).when(volumeVoMock).getProvisioningType();
-            Mockito.doReturn(Volume.Type.valueOf(expected.getVolumeType())).when(volumeVoMock).getVolumeType();
             Mockito.doReturn(expected.getStorage()).when(presetVariableHelperSpy).getPresetVariableValueStorage(Mockito.anyLong(), Mockito.anyInt());
             Mockito.doReturn(expected.getTags()).when(presetVariableHelperSpy).getPresetVariableValueResourceTags(Mockito.anyLong(), Mockito.any(ResourceObjectType.class));
-            Mockito.doReturn(expected.getSize()).when(volumeVoMock).getSize();
-            Mockito.doReturn(imageFormat).when(volumeVoMock).getFormat();
 
             Mockito.doReturn(UsageTypes.VOLUME).when(usageVoMock).getUsageType();
 
@@ -817,20 +822,15 @@
         for (ImageFormat imageFormat : ImageFormat.values()) {
             Value expected = getValueForTests();
 
-            VolumeVO volumeVoMock = Mockito.mock(VolumeVO.class);
+            VolumeVO volumeVoMock = createMockVolumeVo(expected.getId(), expected.getName(), ProvisioningType.getProvisioningType(expected.getProvisioningType()),
+                    Volume.Type.valueOf(expected.getVolumeType()), expected.getSize(), imageFormat);
             Mockito.doReturn(volumeVoMock).when(volumeDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
             Mockito.doReturn(null).when(volumeVoMock).getPoolId();
 
             mockMethodValidateIfObjectIsNull();
 
-            Mockito.doReturn(expected.getId()).when(volumeVoMock).getUuid();
-            Mockito.doReturn(expected.getName()).when(volumeVoMock).getName();
             Mockito.doReturn(expected.getDiskOffering()).when(presetVariableHelperSpy).getPresetVariableValueDiskOffering(Mockito.anyLong());
-            Mockito.doReturn(Volume.Type.valueOf(expected.getVolumeType())).when(volumeVoMock).getVolumeType();
-            Mockito.doReturn(ProvisioningType.getProvisioningType(expected.getProvisioningType())).when(volumeVoMock).getProvisioningType();
             Mockito.doReturn(expected.getTags()).when(presetVariableHelperSpy).getPresetVariableValueResourceTags(Mockito.anyLong(), Mockito.any(ResourceObjectType.class));
-            Mockito.doReturn(expected.getSize()).when(volumeVoMock).getSize();
-            Mockito.doReturn(imageFormat).when(volumeVoMock).getFormat();
 
             Mockito.doReturn(UsageTypes.VOLUME).when(usageVoMock).getUsageType();
 
