--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -316,11 +316,14 @@
 
     final OneLineParser statsParserMock = Mockito.mock(OneLineParser.class);
 
+    private VirtualMachineTO vmToMock;
+
     @Before
     public void setup() throws Exception {
         libvirtComputingResourceSpy.qemuSocketsPath = new File("/var/run/qemu");
         libvirtComputingResourceSpy.parser = parserMock;
         LibvirtComputingResource.LOGGER = loggerMock;
+        vmToMock = Mockito.mock(VirtualMachineTO.class);
     }
 
     /**
@@ -3469,8 +3472,7 @@
         cidrs.add("0.0.0.0/0");
 
         final SecurityGroupRulesCmd command = new SecurityGroupRulesCmd(guestIp, guestIp6, guestMac, vmName, vmId, signature, seqNum, ingressRuleSet, egressRuleSet, secIps);
-        final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
-        command.setVmTO(vm);
+        command.setVmTO(vmToMock);
 
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
@@ -3503,7 +3505,7 @@
         when(egressRuleSet[0].getEndPort()).thenReturn(22);
         when(egressRuleSet[0].getAllowedCidrs()).thenReturn(cidrs);
 
-        when(libvirtComputingResourceMock.applyDefaultNetworkRules(conn, vm, true)).thenReturn(true);
+        when(libvirtComputingResourceMock.applyDefaultNetworkRules(conn, vmToMock, true)).thenReturn(true);
         when(libvirtComputingResourceMock.addNetworkRules(command.getVmName(), Long.toString(command.getVmId()), command.getGuestIp(), command.getGuestIp6(), command.getSignature(),
                 Long.toString(command.getSeqNum()), command.getGuestMac(), command.stringifyRules(), vif, brname, command.getSecIpsString())).thenReturn(true);
 
@@ -5710,13 +5712,12 @@
 
     @Test
     public void getCurrentMemAccordingToMemBallooningTestValidateCurrentMemoryWithoutMemBallooning(){
-        VirtualMachineTO vmTo = Mockito.mock(VirtualMachineTO.class);
         LibvirtComputingResource libvirtComputingResource = new LibvirtComputingResource();
         libvirtComputingResource.noMemBalloon = true;
         long requestedMemory = 1024 * 1024;
         long minMemory = 512 * 1024;
 
-        long currentMemory = libvirtComputingResource.getCurrentMemAccordingToMemBallooning(vmTo, requestedMemory, minMemory);
+        long currentMemory = libvirtComputingResource.getCurrentMemAccordingToMemBallooning(vmToMock, requestedMemory, minMemory);
         Assert.assertEquals(requestedMemory, currentMemory);
     }
 
