--- a/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
@@ -893,8 +893,7 @@
     @Test
     public void testGetResourceCount() {
         long accountId = 1L;
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getId()).thenReturn(accountId);
+        Account account = createMockAccount(accountId);
         resourceLimitManager.getResourceCount(account, Resource.ResourceType.user_vm, "tag");
         Mockito.verify(resourceCountDao, Mockito.times(1))
                 .getResourceCount(accountId, Resource.ResourceOwnerType.Account, Resource.ResourceType.user_vm, "tag");
@@ -903,8 +902,7 @@
     @Test
     public void testGetTaggedResourceLimitAndCountResponse() {
         long accountId = 1L;
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getId()).thenReturn(accountId);
+        Account account = createMockAccount(accountId);
         Long accountLimit = 10L;
         Mockito.doReturn(accountLimit).when(resourceLimitManager)
                 .findCorrectResourceLimitForAccount(Mockito.any(Account.class), Mockito.any(Resource.ResourceType.class), Mockito.anyString());
@@ -969,6 +967,12 @@
                 Mockito.anyLong(), Mockito.any(Resource.ResourceType.class), Mockito.anyString(), Mockito.anyLong());
     }
 
+    private static Account createMockAccount(long accountId) {
+        Account account = Mockito.mock(Account.class);
+        Mockito.when(account.getId()).thenReturn(accountId);
+        return account;
+    }
+
     private void mockDecrementResourceCountWithTag() {
         Mockito.doNothing().when(resourceLimitManager).decrementResourceCountWithTag(
                 Mockito.anyLong(), Mockito.any(Resource.ResourceType.class), Mockito.anyString());
