--- a/server/src/test/java/com/cloud/bgp/BGPServiceImplTest.java
+++ b/server/src/test/java/com/cloud/bgp/BGPServiceImplTest.java
@@ -74,6 +74,9 @@
     AccountDao accountDao;
 
     @Mock
+    AccountVO accountOwner;
+
+    @Mock
     VpcServiceMapDao vpcServiceMapDao;
 
     @Test
@@ -127,13 +130,12 @@
 
         when(bgpPeerDao.listNonRevokeByVpcId(vpcId)).thenReturn(new ArrayList<>());
 
-        AccountVO owner = Mockito.mock(AccountVO.class);
-        when(accountDao.findByIdIncludingRemoved(accountId)).thenReturn(owner);
+        when(accountDao.findByIdIncludingRemoved(accountId)).thenReturn(accountOwner);
 
         Long bgpPeerId1 = 14L;
         BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
         when(bgpPeerDao.findById(bgpPeerId1)).thenReturn(bgpPeer1);
-        when(routedIpv4Manager.getBgpPeerIdsForAccount(owner, zoneId)).thenReturn(Arrays.asList(bgpPeerId1));
+        when(routedIpv4Manager.getBgpPeerIdsForAccount(accountOwner, zoneId)).thenReturn(Arrays.asList(bgpPeerId1));
 
         doReturn(true).when(virtualRouterElement).applyBgpPeers(eq(null), eq(network), any());
 
@@ -159,13 +161,12 @@
 
         when(bgpPeerDao.listNonRevokeByVpcId(vpcId)).thenReturn(new ArrayList<>());
 
-        AccountVO owner = Mockito.mock(AccountVO.class);
-        when(accountDao.findByIdIncludingRemoved(accountId)).thenReturn(owner);
+        when(accountDao.findByIdIncludingRemoved(accountId)).thenReturn(accountOwner);
 
         Long bgpPeerId1 = 14L;
         BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
         when(bgpPeerDao.findById(bgpPeerId1)).thenReturn(bgpPeer1);
-        when(routedIpv4Manager.getBgpPeerIdsForAccount(owner, zoneId)).thenReturn(Arrays.asList(bgpPeerId1));
+        when(routedIpv4Manager.getBgpPeerIdsForAccount(accountOwner, zoneId)).thenReturn(Arrays.asList(bgpPeerId1));
 
         doReturn(true).when(vpcVirtualRouterElement).applyBgpPeers(eq(vpc), eq(null), any());
 
