--- a/plugins/network-elements/globodns/src/test/java/com/globo/globodns/cloudstack/element/GloboDnsElementTest.java
+++ b/plugins/network-elements/globodns/src/test/java/com/globo/globodns/cloudstack/element/GloboDnsElementTest.java
@@ -132,9 +132,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testUpperCaseCharactersAreNotAllowed() throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
-        Network network = mock(Network.class);
-        when(network.getDataCenterId()).thenReturn(zoneId);
-        when(network.getId()).thenReturn(1l);
+        Network network = createMockNetwork();
         NicProfile nic = new NicProfile();
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         when(vm.getHostName()).thenReturn("UPPERCASENAME");
@@ -147,9 +145,7 @@
 
     @Test
     public void testPrepareMethodCallGloboDnsToRegisterHostName() throws Exception {
-        Network network = mock(Network.class);
-        when(network.getDataCenterId()).thenReturn(zoneId);
-        when(network.getId()).thenReturn(1l);
+        Network network = createMockNetwork();
         NicProfile nic = new NicProfile();
         nic.setIPv4Address("10.11.12.13");
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
@@ -180,9 +176,7 @@
 
     @Test
     public void testReleaseMethodCallResource() throws Exception {
-        Network network = mock(Network.class);
-        when(network.getDataCenterId()).thenReturn(zoneId);
-        when(network.getId()).thenReturn(1l);
+        Network network = createMockNetwork();
         NicProfile nic = new NicProfile();
         nic.setIPv4Address("10.11.12.13");
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
@@ -210,6 +204,13 @@
         verify(_agentMgr, times(1)).easySend(eq(globoDnsHostId), isA(RemoveRecordCommand.class));
     }
 
+    private static Network createMockNetwork() {
+        Network network = mock(Network.class);
+        when(network.getDataCenterId()).thenReturn(zoneId);
+        when(network.getId()).thenReturn(1l);
+        return network;
+    }
+
     @Configuration
     @ComponentScan(basePackageClasses = {GloboDnsElement.class}, includeFilters = {@Filter(value = TestConfiguration.Library.class, type = FilterType.CUSTOM)}, useDefaultFilters = false)
     public static class TestConfiguration extends SpringUtils.CloudStackTestConfiguration {
