--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
@@ -80,8 +80,7 @@
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getId()).thenReturn(1L);
         Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
-        KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(map.getClusterId()).thenReturn(1L);
+        KubernetesClusterVmMapVO map = createMockKubernetesClusterVmMapVO();
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(map);
         KubernetesClusterVO kubernetesCluster = Mockito.mock(KubernetesClusterVO.class);
         Mockito.when(kubernetesClusterDao.findById(1L)).thenReturn(kubernetesCluster);
@@ -94,8 +93,7 @@
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getId()).thenReturn(1L);
         Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
-        KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(map.getClusterId()).thenReturn(1L);
+        KubernetesClusterVmMapVO map = createMockKubernetesClusterVmMapVO();
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(map);
         KubernetesClusterVO kubernetesCluster = Mockito.mock(KubernetesClusterVO.class);
         Mockito.when(kubernetesClusterDao.findById(1L)).thenReturn(kubernetesCluster);
@@ -123,6 +121,12 @@
         Map<String, String> map = new HashMap<>();
         map.put(VmDetailConstants.CKS_NODE_TYPE, nodeType.name().toLowerCase());
         map.put(VmDetailConstants.OFFERING, offeringUuid);
+        return map;
+    }
+
+    private static KubernetesClusterVmMapVO createMockKubernetesClusterVmMapVO() {
+        KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
+        Mockito.when(map.getClusterId()).thenReturn(1L);
         return map;
     }
 
