--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -270,6 +270,9 @@
     DiskTO diskToMock;
 
     @Mock
+    private NicTO nicMock;
+
+    @Mock
     private VolumeObjectTO volumeObjectToMock;
 
 
@@ -1605,7 +1608,6 @@
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
         final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
-        final NicTO nicTO = Mockito.mock(NicTO.class);
         final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
         final PrepareForMigrationCommand command = new PrepareForMigrationCommand(vm);
@@ -1617,7 +1619,7 @@
             fail(e.getMessage());
         }
 
-        when(vm.getNics()).thenReturn(new NicTO[]{nicTO});
+        when(vm.getNics()).thenReturn(new NicTO[]{nicMock});
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
 
@@ -3676,11 +3678,10 @@
     @SuppressWarnings("unchecked")
     @Test
     public void testPlugNicCommandLibvirtException() {
-        final NicTO nic = Mockito.mock(NicTO.class);
         final String instanceName = "Test";
         final Type vmtype = Type.DomainRouter;
 
-        final PlugNicCommand command = new PlugNicCommand(nic, instanceName, vmtype);
+        final PlugNicCommand command = new PlugNicCommand(nicMock, instanceName, vmtype);
 
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
@@ -3817,10 +3818,9 @@
 
     @Test
     public void testUnPlugNicCommandNoNics() {
-        final NicTO nic = Mockito.mock(NicTO.class);
         final String instanceName = "Test";
 
-        final UnPlugNicCommand command = new UnPlugNicCommand(nic, instanceName);
+        final UnPlugNicCommand command = new UnPlugNicCommand(nicMock, instanceName);
 
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
@@ -3860,10 +3860,9 @@
     @SuppressWarnings("unchecked")
     @Test
     public void testUnPlugNicCommandLibvirtException() {
-        final NicTO nic = Mockito.mock(NicTO.class);
         final String instanceName = "Test";
 
-        final UnPlugNicCommand command = new UnPlugNicCommand(nic, instanceName);
+        final UnPlugNicCommand command = new UnPlugNicCommand(nicMock, instanceName);
 
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
@@ -5112,8 +5111,7 @@
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
-        final NicTO nic = Mockito.mock(NicTO.class);
-        final NicTO[] nics = new NicTO[]{nic};
+        final NicTO[] nics = new NicTO[]{nicMock};
 
         final String vmName = "Test";
 
@@ -5165,8 +5163,7 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
-        final NicTO nic = Mockito.mock(NicTO.class);
-        final NicTO[] nics = new NicTO[]{nic};
+        final NicTO[] nics = new NicTO[]{nicMock};
 
         final String vmName = "Test";
 
@@ -5209,8 +5206,7 @@
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
-        final NicTO nic = Mockito.mock(NicTO.class);
-        final NicTO[] nics = new NicTO[]{nic};
+        final NicTO[] nics = new NicTO[]{nicMock};
 
         final String vmName = "Test";
 
@@ -5260,8 +5256,7 @@
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
-        final NicTO nic = Mockito.mock(NicTO.class);
-        final NicTO[] nics = new NicTO[]{nic};
+        final NicTO[] nics = new NicTO[]{nicMock};
 
         final String vmName = "Test";
 
@@ -5470,8 +5465,7 @@
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
-        final NicTO nic = Mockito.mock(NicTO.class);
-        final NicTO[] nics = new NicTO[]{nic};
+        final NicTO[] nics = new NicTO[]{nicMock};
         final String vmName = "Test";
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
