--- a/server/src/test/java/com/cloud/network/router/VirtualNetworkApplianceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/VirtualNetworkApplianceManagerImplTest.java
@@ -237,6 +237,9 @@
     private DomainRouterVO _router;
 
     @Mock
+    private DomainRouterVO domainRouter;
+
+    @Mock
     private AlertManager _alertMgr;
 
     @Mock
@@ -365,7 +368,6 @@
     public void testFinalizeNetworkRulesForNetwork() {
         Long guestNetworkId = 10L;
         Commands cmds = new Commands(Command.OnError.Stop);
-        DomainRouterVO router = Mockito.mock(DomainRouterVO.class);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
         when(_networkDao.findById(guestNetworkId)).thenReturn(network);
@@ -373,9 +375,9 @@
         when(routedIpv4Manager.isDynamicRoutedNetwork(network)).thenReturn(true);
         List<? extends BgpPeer> bgpPeers = Mockito.mock(List.class);
         doReturn(bgpPeers).when(bgpService).getBgpPeersForNetwork(network);
-        virtualNetworkApplianceManagerImpl.finalizeNetworkRulesForNetwork(cmds, router, Network.Provider.VirtualRouter, guestNetworkId);
-
-        Mockito.verify(_commandSetupHelper).createBgpPeersCommands(bgpPeers, router, cmds, network);
+        virtualNetworkApplianceManagerImpl.finalizeNetworkRulesForNetwork(cmds, domainRouter, Network.Provider.VirtualRouter, guestNetworkId);
+
+        Mockito.verify(_commandSetupHelper).createBgpPeersCommands(bgpPeers, domainRouter, cmds, network);
     }
 
     @Test
@@ -383,7 +385,6 @@
         Long guestNetworkId = 10L;
         Long vpcId = 11L;
         Commands cmds = new Commands(Command.OnError.Stop);
-        DomainRouterVO router = Mockito.mock(DomainRouterVO.class);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
         when(_networkDao.findById(guestNetworkId)).thenReturn(network);
@@ -394,9 +395,9 @@
         List<? extends BgpPeer> bgpPeers = Mockito.mock(List.class);
         doReturn(bgpPeers).when(bgpService).getBgpPeersForVpc(vpc);
 
-        virtualNetworkApplianceManagerImpl.finalizeNetworkRulesForNetwork(cmds, router, Network.Provider.VirtualRouter, guestNetworkId);
-
-        Mockito.verify(_commandSetupHelper).createBgpPeersCommands(bgpPeers, router, cmds, network);
+        virtualNetworkApplianceManagerImpl.finalizeNetworkRulesForNetwork(cmds, domainRouter, Network.Provider.VirtualRouter, guestNetworkId);
+
+        Mockito.verify(_commandSetupHelper).createBgpPeersCommands(bgpPeers, domainRouter, cmds, network);
     }
 
     @Test
