--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxProviderServiceImplTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxProviderServiceImplTest.java
@@ -133,9 +133,8 @@
 
     @Test
     public void testListNsxControllers() {
-        NsxProviderVO nsxProviderVO = Mockito.mock(NsxProviderVO.class);
+        NsxProviderVO nsxProviderVO = createMockNsxProviderVO();
 
-        when(nsxProviderVO.getZoneId()).thenReturn(1L);
         when(dataCenterDao.findById(1L)).thenReturn(mock(DataCenterVO.class));
         when(nsxProviderDao.findByZoneId(anyLong())).thenReturn(nsxProviderVO);
 
@@ -145,13 +144,12 @@
 
     @Test
     public void testDeleteNsxController() {
-        NsxProviderVO nsxProviderVO = Mockito.mock(NsxProviderVO.class);
+        NsxProviderVO nsxProviderVO = createMockNsxProviderVO();
         PhysicalNetworkVO physicalNetworkVO = mock(PhysicalNetworkVO.class);
         List<PhysicalNetworkVO> physicalNetworkVOList = List.of(physicalNetworkVO);
         NetworkVO networkVO = mock(NetworkVO.class);
         List<NetworkVO> networkVOList = List.of(networkVO);
 
-        when(nsxProviderVO.getZoneId()).thenReturn(1L);
         when(physicalNetworkVO.getId()).thenReturn(2L);
         when(physicalNetworkDao.listByZone(1L)).thenReturn(physicalNetworkVOList);
         when(nsxProviderDao.findById(anyLong())).thenReturn(nsxProviderVO);
@@ -171,4 +169,10 @@
 
         assertThrows(CloudRuntimeException.class, () -> nsxProviderService.validateNetworkState(networkVOList));
     }
+
+    private static NsxProviderVO createMockNsxProviderVO() {
+        NsxProviderVO nsxProviderVO = Mockito.mock(NsxProviderVO.class);
+        when(nsxProviderVO.getZoneId()).thenReturn(1L);
+        return nsxProviderVO;
+    }
 }
