--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1036,6 +1036,12 @@
         return expectedUserPasswordAfterEncoded;
     }
 
+    private static AccountVO createMockAccountVO() {
+        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(14L).when(accountVoMock).getDomainId();
+        return accountVoMock;
+    }
+
     @Test(expected = InvalidParameterValueException.class)
     public void validateCurrentPasswordTestUserNotAuthenticatedWithProvidedCurrentPassword() {
         Mockito.doReturn(Mockito.mock(AccountVO.class)).when(_accountDao).findById(accountMockId);
@@ -1047,9 +1053,8 @@
 
     @Test
     public void validateCurrentPasswordTestUserAuthenticatedWithProvidedCurrentPasswordViaFirstAuthenticator() {
-        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
+        AccountVO accountVoMock = createMockAccountVO();
         long domainId = 14L;
-        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
 
         Mockito.doReturn(accountVoMock).when(_accountDao).findById(accountMockId);
         String username = "username";
@@ -1076,9 +1081,8 @@
 
     @Test
     public void validateCurrentPasswordTestUserAuthenticatedWithProvidedCurrentPasswordViaSecondAuthenticator() {
-        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
         long domainId = 14L;
-        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
+        AccountVO accountVoMock = createMockAccountVO();
 
         Mockito.doReturn(accountVoMock).when(_accountDao).findById(accountMockId);
         String username = "username";
