--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
@@ -293,10 +293,9 @@
 
     @Test
     public void testCreateNsxSegmentForVpc() {
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
+        NetworkVO networkVO = createMockNetworkVO();
         DataCenter dataCenter = Mockito.mock(DataCenter.class);
 
-        when(networkVO.getAccountId()).thenReturn(1L);
         when(nsxControllerUtils.sendNsxCommand(any(CreateNsxSegmentCommand.class),
                 anyLong())).thenReturn(new NsxAnswer(new NsxCommand(), true, ""));
         guru.createNsxSegment(networkVO, dataCenter);
@@ -307,11 +306,10 @@
 
     @Test
     public void testCreateNsxSegmentForIsolatedNetwork() {
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
         NetworkOfferingVO offeringVO = Mockito.mock(NetworkOfferingVO.class);
         DataCenter dataCenter = Mockito.mock(DataCenter.class);
 
-        when(networkVO.getAccountId()).thenReturn(1L);
+        NetworkVO networkVO = createMockNetworkVO();
         when(networkVO.getVpcId()).thenReturn(null);
         when(nsxControllerUtils.sendNsxCommand(any(CreateNsxTier1GatewayCommand.class),
                 anyLong())).thenReturn(new NsxAnswer(new NsxCommand(), true, ""));
@@ -326,4 +324,10 @@
         verify(nsxControllerUtils, times(1)).sendNsxCommand(any(CreateNsxSegmentCommand.class),
                 anyLong());
     }
+
+    private static NetworkVO createMockNetworkVO() {
+        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
+        when(networkVO.getAccountId()).thenReturn(1L);
+        return networkVO;
+    }
 }
