--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -1831,8 +1831,7 @@
 
     @Test
     public void cleanupValidationTestDestroyVmThrowsAndCleanupMailIsSent() throws ResourceUnavailableException {
-        VolumeVO dataVolume = Mockito.mock(VolumeVO.class);
-        doReturn(Volume.Type.DATADISK).when(dataVolume).getVolumeType();
+        VolumeVO dataVolume = createMockDataVolume();
         doReturn(88L).when(userVmVOMock).getHostId();
         doReturn(Set.of("secondary-storage-1")).when(kbossBackupProviderSpy).getSecondaryStorageUrls(userVmVOMock);
         doThrow(new RuntimeException("boom")).when(userVmManagerMock).destroyVm(any(DestroyVMCmd.class), Mockito.eq(false));
@@ -1854,8 +1853,7 @@
 
     @Test
     public void cleanupValidationTestCleanupCommandFails() {
-        VolumeVO dataVolume = Mockito.mock(VolumeVO.class);
-        doReturn(Volume.Type.DATADISK).when(dataVolume).getVolumeType();
+        VolumeVO dataVolume = createMockDataVolume();
         doReturn(88L).when(userVmVOMock).getHostId();
         doReturn(Set.of("secondary-storage-1")).when(kbossBackupProviderSpy).getSecondaryStorageUrls(userVmVOMock);
         doReturn(null).when(agentManagerMock).easySend(anyLong(), any());
@@ -3113,4 +3111,10 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+    private static VolumeVO createMockDataVolume() {
+        VolumeVO dataVolume = Mockito.mock(VolumeVO.class);
+        doReturn(Volume.Type.DATADISK).when(dataVolume).getVolumeType();
+        return dataVolume;
+    }
+
 }
