--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -814,6 +814,12 @@
         return to;
     }
 
+    private static KVMStoragePool createMockKVMStoragePool(String physicalDiskName, KVMPhysicalDisk physicalDiskToReturn) {
+        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
+        when(storagePool.getPhysicalDisk(physicalDiskName)).thenReturn(physicalDiskToReturn);
+        return storagePool;
+    }
+
     private void verifyGuestResourceDef(GuestResourceDef guestResourceDef, VirtualMachineTO to) {
         String xml = "<domain>" + guestResourceDef.toString() + "</domain>";
         Document domainDoc = parse(xml);
@@ -2234,14 +2240,13 @@
         final CreateCommand command = new CreateCommand(diskCharacteristics, templateUrl, pool, executeInSequence );
 
         final KVMStoragePoolManager poolManager = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool primary = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
         final KVMPhysicalDisk baseVol = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool primary = createMockKVMStoragePool(command.getTemplateUrl(), baseVol);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(poolManager);
         when(poolManager.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(primary);
 
-        when(primary.getPhysicalDisk(command.getTemplateUrl())).thenReturn(baseVol);
         when(poolManager.createDiskFromTemplate(baseVol, diskCharacteristics.getPath(), diskCharacteristics.getProvisioningType(), primary, baseVol.getSize(), 0,null)).thenReturn(vol);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
@@ -2433,7 +2438,6 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk tmplVol = Mockito.mock(KVMPhysicalDisk.class);
         final KVMPhysicalDisk primaryVol = Mockito.mock(KVMPhysicalDisk.class);
 
@@ -2441,8 +2445,8 @@
         final String mountpoint = url.substring(0, index);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool secondaryPool = createMockKVMStoragePool("template.qcow2", tmplVol);
         when(storagePoolMgr.getStoragePoolByURI(mountpoint)).thenReturn(secondaryPool);
-        when(secondaryPool.getPhysicalDisk("template.qcow2")).thenReturn(tmplVol);
         when(storagePoolMgr.getStoragePool(command.getPool().getType(), command.getPoolUuid())).thenReturn(primaryPool);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
@@ -3309,7 +3313,6 @@
         final CreateVolumeFromSnapshotCommand command = new CreateVolumeFromSnapshotCommand(pool, secondaryStoragePoolURL, dcId, accountId, volumeId, backedUpSnapshotUuid, backedUpSnapshotName, wait);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk snapshot = Mockito.mock(KVMPhysicalDisk.class);
         final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
 
@@ -3320,8 +3323,8 @@
         final String primaryUuid = command.getPrimaryStoragePoolNameLabel();
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool secondaryPool = createMockKVMStoragePool(command.getSnapshotName(), snapshot);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(secondaryPool);
-        when(secondaryPool.getPhysicalDisk(command.getSnapshotName())).thenReturn(snapshot);
         when(storagePoolMgr.getStoragePool(command.getPool().getType(), primaryUuid)).thenReturn(primaryPool);
 
         //when(storagePoolMgr.copyPhysicalDisk(snapshot, volUuid, primaryPool, 0)).thenReturn(disk);
@@ -3354,7 +3357,6 @@
         final CreateVolumeFromSnapshotCommand command = new CreateVolumeFromSnapshotCommand(pool, secondaryStoragePoolURL, dcId, accountId, volumeId, backedUpSnapshotUuid, backedUpSnapshotName, wait);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk snapshot = Mockito.mock(KVMPhysicalDisk.class);
 
         String snapshotPath = command.getSnapshotUuid();
@@ -3364,8 +3366,8 @@
         final String primaryUuid = command.getPrimaryStoragePoolNameLabel();
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool secondaryPool = createMockKVMStoragePool(command.getSnapshotName(), snapshot);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(secondaryPool);
-        when(secondaryPool.getPhysicalDisk(command.getSnapshotName())).thenReturn(snapshot);
         when(storagePoolMgr.getStoragePool(command.getPool().getType(), primaryUuid)).thenThrow(CloudRuntimeException.class);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
@@ -4113,7 +4115,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
         final Domain vm = Mockito.mock(Domain.class);
         final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
@@ -4133,8 +4134,8 @@
         }
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool primaryPool = createMockKVMStoragePool(command.getVolumePath(), disk);
         when(storagePoolMgr.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(primaryPool);
-        when(primaryPool.getPhysicalDisk(command.getVolumePath())).thenReturn(disk);
         when(primaryPool.isExternalSnapshot()).thenReturn(false);
 
         try {
@@ -4231,8 +4232,8 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool snapshotPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk snapshot = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool snapshotPool = createMockKVMStoragePool(command.getSnapshotName(), snapshot);
         final StorageLayer storage = Mockito.mock(StorageLayer.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
@@ -4247,7 +4248,6 @@
 
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(snapshotPool);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl())).thenReturn(secondaryPool);
-        when(snapshotPool.getPhysicalDisk(command.getSnapshotName())).thenReturn(snapshot);
         when(secondaryPool.getLocalPath()).thenReturn(localPath);
         when(libvirtComputingResourceMock.getStorage()).thenReturn(storage);
 
@@ -4305,8 +4305,8 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool snapshotPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk snapshot = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool snapshotPool = createMockKVMStoragePool(command.getSnapshotName(), snapshot);
         final StorageLayer storage = Mockito.mock(StorageLayer.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
@@ -4321,7 +4321,6 @@
 
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(snapshotPool);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl())).thenReturn(secondaryPool);
-        when(snapshotPool.getPhysicalDisk(command.getSnapshotName())).thenReturn(snapshot);
         when(secondaryPool.getLocalPath()).thenReturn(localPath);
         when(libvirtComputingResourceMock.getStorage()).thenReturn(storage);
 
@@ -4375,8 +4374,8 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool snapshotPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk snapshot = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool snapshotPool = createMockKVMStoragePool(command.getSnapshotName(), snapshot);
         final StorageLayer storage = Mockito.mock(StorageLayer.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
@@ -4390,7 +4389,6 @@
 
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(snapshotPool);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl())).thenReturn(secondaryPool);
-        when(snapshotPool.getPhysicalDisk(command.getSnapshotName())).thenReturn(snapshot);
         when(secondaryPool.getLocalPath()).thenReturn(localPath);
         when(libvirtComputingResourceMock.getStorage()).thenReturn(storage);
 
@@ -4447,8 +4445,8 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool snapshotPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk snapshot = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool snapshotPool = createMockKVMStoragePool(command.getSnapshotName(), snapshot);
         final StorageLayer storage = Mockito.mock(StorageLayer.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
@@ -4463,7 +4461,6 @@
 
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(snapshotPool);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl())).thenReturn(secondaryPool);
-        when(snapshotPool.getPhysicalDisk(command.getSnapshotName())).thenReturn(snapshot);
         when(secondaryPool.getLocalPath()).thenReturn(localPath);
         when(libvirtComputingResourceMock.getStorage()).thenReturn(storage);
 
@@ -4568,9 +4565,8 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool secondary = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool primary = Mockito.mock(KVMStoragePool.class);
-
         final KVMPhysicalDisk disk = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool primary = createMockKVMStoragePool(command.getVolumePath(), disk);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final StorageFilerTO pool = command.getPool();
@@ -4580,7 +4576,6 @@
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtUtilitiesHelper.generateUUIDName()).thenReturn(destVolumeName);
-        when(primary.getPhysicalDisk(command.getVolumePath())).thenReturn(disk);
         when(storagePoolMgr.getStoragePoolByURI(secondaryStoragePoolURL)).thenReturn(secondary);
         when(secondary.getType()).thenReturn(StoragePoolType.ManagedNFS);
         when(secondary.getUuid()).thenReturn("60d979d8-d132-4181-8eca-8dfde50d7df6");
@@ -4838,7 +4833,6 @@
         final ResizeVolumeCommand command = new ResizeVolumeCommand(path, pool, currentSize, newSize, shrinkOk, vmInstance, null);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
@@ -4850,8 +4844,8 @@
 
         when(pool.getType()).thenReturn(StoragePoolType.RBD);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool storagePool = createMockKVMStoragePool(path, vol);
         when(storagePoolMgr.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(storagePool);
-        when(storagePool.getPhysicalDisk(path)).thenReturn(vol);
         when(vol.getPath()).thenReturn(path);
         when(storagePool.getType()).thenReturn(StoragePoolType.RBD);
         when(vol.getFormat()).thenReturn(PhysicalDiskFormat.FILE);
@@ -4898,7 +4892,6 @@
         final ResizeVolumeCommand command = new ResizeVolumeCommand(path, pool, currentSize, newSize, shrinkOk, vmInstance, null);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
@@ -4909,8 +4902,8 @@
 
         when(pool.getType()).thenReturn(StoragePoolType.Linstor);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool storagePool = createMockKVMStoragePool(path, vol);
         when(storagePoolMgr.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(storagePool);
-        when(storagePool.getPhysicalDisk(path)).thenReturn(vol);
         when(vol.getPath()).thenReturn(path);
         when(storagePool.getType()).thenReturn(StoragePoolType.Linstor);
 
@@ -4971,13 +4964,12 @@
         final ResizeVolumeCommand command = new ResizeVolumeCommand(path, pool, currentSize, newSize, shrinkOk, vmInstance, null);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
+        final KVMStoragePool storagePool = createMockKVMStoragePool(path, vol);
 
         when(pool.getType()).thenReturn(StoragePoolType.Filesystem);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
         when(storagePoolMgr.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(storagePool);
-        when(storagePool.getPhysicalDisk(path)).thenReturn(vol);
         when(vol.getPath()).thenReturn(path);
         when(libvirtComputingResourceMock.getResizeScriptType(storagePool, vol)).thenReturn("QCOW2");
 
@@ -5001,14 +4993,13 @@
         final ResizeVolumeCommand command = new ResizeVolumeCommand(path, pool, currentSize, newSize, shrinkOk, vmInstance, null);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         when(pool.getType()).thenReturn(StoragePoolType.RBD);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool storagePool = createMockKVMStoragePool(path, vol);
         when(storagePoolMgr.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(storagePool);
-        when(storagePool.getPhysicalDisk(path)).thenReturn(vol);
         when(vol.getPath()).thenReturn(path);
         when(storagePool.getType()).thenReturn(StoragePoolType.RBD);
         when(vol.getFormat()).thenReturn(PhysicalDiskFormat.FILE);
