--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -260,6 +260,7 @@
     private KVMPhysicalDisk physicalDisk;
     @Mock
     private Map<String, String> details;
+    private final InterfaceDef interfaceDefMock = Mockito.mock(InterfaceDef.class);
 
     private static final String PHYSICAL_DISK_PATH = "/path/to/disk";
     private static final int DEV_ID = 1;
@@ -1761,9 +1762,8 @@
             fail(e.getMessage());
         }
 
-        final InterfaceDef interfaceDef = Mockito.mock(InterfaceDef.class);
         final List<InterfaceDef> ifaces = new ArrayList<InterfaceDef>();
-        ifaces.add(interfaceDef);
+        ifaces.add(interfaceDefMock);
 
         when(libvirtComputingResourceMock.getInterfaces(conn, vmName)).thenReturn(ifaces);
 
@@ -3427,8 +3427,7 @@
         final Connect conn = Mockito.mock(Connect.class);
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
-        final InterfaceDef interfaceDef = Mockito.mock(InterfaceDef.class);
-        nics.add(interfaceDef);
+        nics.add(interfaceDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
@@ -3541,8 +3540,7 @@
         final Connect conn = Mockito.mock(Connect.class);
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
-        final InterfaceDef interfaceDef = Mockito.mock(InterfaceDef.class);
-        nics.add(interfaceDef);
+        nics.add(interfaceDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
