--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -1944,13 +1944,12 @@
         ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
 
-        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
+        VMInstanceVO vm = createMockVmInstance();
         HostVO destinationHostVO = Mockito.mock(HostVO.class);
 
         Mockito.when(_serviceOfferingDao.findByIdIncludingRemoved(1L)).thenReturn(serviceOffering);
         Mockito.when(templateDao.findByIdIncludingRemoved(2L)).thenReturn(template);
 
-        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
         Mockito.when(vm.getTemplateId()).thenReturn(2L);
 
         Mockito.when(destinationHostVO.checkHostServiceOfferingAndTemplateTags(Mockito.any(ServiceOffering.class), Mockito.any(VirtualMachineTemplate.class), Mockito.anySet())).thenReturn(true);
@@ -1967,13 +1966,12 @@
         ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
 
-        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
         HostVO destinationHostVO = Mockito.mock(HostVO.class);
 
         Mockito.when(_serviceOfferingDao.findByIdIncludingRemoved(1L)).thenReturn(serviceOffering);
         Mockito.when(templateDao.findByIdIncludingRemoved(2L)).thenReturn(template);
 
-        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        VMInstanceVO vm = createMockVmInstance();
         Mockito.when(vm.getTemplateId()).thenReturn(2L);
 
         Mockito.when(destinationHostVO.checkHostServiceOfferingAndTemplateTags(Mockito.any(ServiceOffering.class), Mockito.any(VirtualMachineTemplate.class), Mockito.anySet())).thenReturn(false);
@@ -4652,6 +4650,12 @@
         return serviceOffering;
     }
 
+    private static VMInstanceVO createMockVmInstance() {
+        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
+        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        return vm;
+    }
+
     @Test
     public void customOfferingNeedsCustomizationThrowsException() {
         ServiceOfferingVO serviceOffering = getMockedServiceOffering(true, true);
