--- a/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
@@ -355,10 +355,9 @@
 
         List<SnapshotDataStoreVO> snapshotDataStoreList = List.of(snapshotDataStore1, snapshotDataStore2);
 
-        SnapshotVO snapshot1 = Mockito.mock(SnapshotVO.class);
-        Mockito.when(snapshot1.getId()).thenReturn(1L);
-
-        SnapshotVO snapshot2 = Mockito.mock(SnapshotVO.class);
+        SnapshotVO snapshot1 = createMockSnapshot();
+
+        SnapshotVO snapshot2 = createMockSnapshot();
         Mockito.when(snapshot2.getId()).thenReturn(2L);
 
         List<SnapshotVO> snapshots = List.of(snapshot1, snapshot2);
@@ -391,10 +390,9 @@
 
         List<SnapshotDataStoreVO> snapshotDataStoreList = List.of(snapshotDataStore1, snapshotDataStore2);
 
-        SnapshotVO snapshot1 = Mockito.mock(SnapshotVO.class);
-        Mockito.when(snapshot1.getId()).thenReturn(1L);
-
-        SnapshotVO snapshot2 = Mockito.mock(SnapshotVO.class);
+        SnapshotVO snapshot1 = createMockSnapshot();
+
+        SnapshotVO snapshot2 = createMockSnapshot();
         Mockito.when(snapshot2.getId()).thenReturn(2L);
 
         List<SnapshotVO> snapshots = List.of(snapshot1, snapshot2);
@@ -456,4 +454,10 @@
         Assert.assertEquals(200L, response.getResponses().get(1).getSize());
         Assert.assertNotNull(response.getResponses().get(1).getLastUpdated());
     }
+
+    private static SnapshotVO createMockSnapshot() {
+        SnapshotVO snapshot = Mockito.mock(SnapshotVO.class);
+        Mockito.when(snapshot.getId()).thenReturn(1L);
+        return snapshot;
+    }
 }
