--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -1152,9 +1152,8 @@
         Mockito.when(host2.getId()).thenReturn(host2Id);
 
         HostPodVO pod1 = Mockito.mock(HostPodVO.class);
-        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
         Mockito.when(pod1.getId()).thenReturn(podId);
-        Mockito.when(cluster1.getId()).thenReturn(clusterId);
+        ClusterVO cluster1 = createMockCluster(clusterId);
         Mockito.when(podDao.findById(podId)).thenReturn(pod1);
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(cluster1);
         Mockito.when(podDao.listByDataCenterId(zoneId)).thenReturn(Collections.singletonList(pod1));
@@ -1189,9 +1188,8 @@
         Mockito.when(host2.getId()).thenReturn(host2Id);
 
         HostPodVO pod1 = Mockito.mock(HostPodVO.class);
-        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
         Mockito.when(pod1.getStorageAccessGroups()).thenReturn("group1,group3");
-        Mockito.when(cluster1.getId()).thenReturn(clusterId);
+        ClusterVO cluster1 = createMockCluster(clusterId);
         Mockito.when(podDao.findById(podId)).thenReturn(pod1);
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(cluster1);
         Mockito.when(clusterDao.listByPodId(podId)).thenReturn(Collections.singletonList(cluster1));
@@ -1223,9 +1221,8 @@
         Mockito.when(host1.getId()).thenReturn(host1Id);
         Mockito.when(host2.getId()).thenReturn(host2Id);
 
-        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
+        ClusterVO cluster1 = createMockCluster(clusterId);
         Mockito.when(cluster1.getStorageAccessGroups()).thenReturn("group1,group3");
-        Mockito.when(cluster1.getId()).thenReturn(clusterId);
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(cluster1);
         Mockito.when(hostDao.findHypervisorHostInCluster(clusterId)).thenReturn(Arrays.asList(host1, host2));
         Mockito.when(hostDao.findByClusterId(clusterId)).thenReturn(Arrays.asList(host1, host2));
@@ -1393,4 +1390,10 @@
         Mockito.verify(hostDao).findByClusterId(clusterId, Host.Type.Routing);
         Mockito.verify(hostDao).findByPodId(podId, Host.Type.Routing);
     }
+
+    private static ClusterVO createMockCluster(long clusterId) {
+        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
+        Mockito.when(cluster1.getId()).thenReturn(clusterId);
+        return cluster1;
+    }
 }
