--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -1117,8 +1117,7 @@
         when(cmd.getResourceId()).thenReturn(UUID.randomUUID().toString());
         when(cmd.getExtensionId()).thenReturn(1L);
         ExtensionVO extension = mock(ExtensionVO.class);
-        ClusterVO clusterVO = mock(ClusterVO.class);
-        when(clusterVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
+        ClusterVO clusterVO = createMockClusterVO();
         when(clusterDao.findByUuid(anyString())).thenReturn(clusterVO);
         ExtensionResourceMapVO resourceMap = mock(ExtensionResourceMapVO.class);
         when(extensionResourceMapDao.persist(any())).thenReturn(resourceMap);
@@ -1150,8 +1149,7 @@
         when(cmd.getResourceType()).thenReturn(ExtensionResourceMap.ResourceType.Cluster.name());
         when(cmd.getResourceId()).thenReturn(UUID.randomUUID().toString());
         when(cmd.getExtensionId()).thenReturn(1L);
-        ClusterVO clusterVO = mock(ClusterVO.class);
-        when(clusterVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
+        ClusterVO clusterVO = createMockClusterVO();
         when(clusterDao.findByUuid(anyString())).thenReturn(clusterVO);
         ExtensionVO extension = mock(ExtensionVO.class);
         when(extensionDao.findById(1L)).thenReturn(extension);
@@ -1547,6 +1545,12 @@
         extensionsManager.deleteCustomAction(cmd);
         verify(extensionCustomActionDetailsDao).removeDetails(actionId);
         verify(extensionCustomActionDao).remove(actionId);
+    }
+
+    private static ClusterVO createMockClusterVO() {
+        ClusterVO clusterVO = mock(ClusterVO.class);
+        when(clusterVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
+        return clusterVO;
     }
 
     private void mockCallerRole(RoleType roleType) {
