--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -118,10 +118,7 @@
 
     @Test
     public void testCanHandle() {
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
@@ -157,10 +154,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(true);
 
@@ -182,10 +176,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
@@ -202,10 +193,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
@@ -224,10 +212,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(true);
 
@@ -283,10 +268,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
@@ -519,10 +501,7 @@
         when(network.getId()).thenReturn(NETWORK_ID);
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final HostVO brocadeHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
@@ -561,10 +540,7 @@
         when(network.getId()).thenReturn(NETWORK_ID);
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final HostVO brocadeHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -108,10 +108,7 @@
 
     @Test
     public void testCanHandle() {
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
@@ -155,10 +152,7 @@
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
         when(device.getId()).thenReturn(1L);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(true);
 
@@ -182,10 +176,7 @@
 
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Collections.<NiciraNvpDeviceVO> emptyList());
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
@@ -204,10 +195,7 @@
 
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Collections.<NiciraNvpDeviceVO> emptyList());
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
@@ -228,10 +216,7 @@
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
         when(device.getId()).thenReturn(1L);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
@@ -254,10 +239,7 @@
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
         when(device.getId()).thenReturn(1L);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
@@ -308,10 +290,7 @@
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
         when(device.getId()).thenReturn(1L);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
@@ -366,10 +345,7 @@
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
         when(device.getId()).thenReturn(1L);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
@@ -420,10 +396,7 @@
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
         when(device.getId()).thenReturn(1L);
 
-        final NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(NETWORK_ID);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        final NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(NETWORK_ID, GuestType.Isolated);
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
--- a/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/VxlanGuestNetworkGuruTest.java
+++ b/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/VxlanGuestNetworkGuruTest.java
@@ -92,10 +92,7 @@
 
     @Test
     public void testCanHandle() {
-        NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(42L);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(42L, GuestType.Isolated);
 
         PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VXLAN"}));
@@ -129,10 +126,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VXLAN"}));
         when(physnet.getId()).thenReturn(42L);
 
-        NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(42L);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(42L, GuestType.Isolated);
 
         DeploymentPlan plan = mock(DeploymentPlan.class);
         Network network = mock(Network.class);
@@ -150,10 +144,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VXLAN"}));
         when(physnet.getId()).thenReturn(42L);
 
-        NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(42L);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(42L, GuestType.Isolated);
 
         NetworkVO network = mock(NetworkVO.class);
         when(network.getName()).thenReturn("testnetwork");
@@ -192,10 +183,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VXLAN"}));
         when(physnet.getId()).thenReturn(42L);
 
-        NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(42L);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(42L, GuestType.Isolated);
 
         NetworkVO network = mock(NetworkVO.class);
         when(network.getName()).thenReturn("testnetwork");
@@ -239,10 +227,7 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VXLAN"}));
         when(physnet.getId()).thenReturn(42L);
 
-        NetworkOffering offering = mock(NetworkOffering.class);
-        when(offering.getId()).thenReturn(42L);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getGuestType()).thenReturn(GuestType.Isolated);
+        NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(42L, GuestType.Isolated);
 
         NetworkVO network = mock(NetworkVO.class);
         when(network.getName()).thenReturn("testnetwork");
--- a/server/src/test/java/com/cloud/network/guru/DirectNetworkGuruTest.java
+++ b/server/src/test/java/com/cloud/network/guru/DirectNetworkGuruTest.java
@@ -100,9 +100,7 @@
 
         when(dc.getNetworkType()).thenReturn(NetworkType.Advanced);
         when(dc.getId()).thenReturn(1l);
-        when(offering.getGuestType()).thenReturn(GuestType.Shared);
-        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
-        when(offering.getId()).thenReturn(42l);
+        offering = MockNetworkOffering.createMockNetworkOffering(42l, GuestType.Shared);
         when(ntwkOfferingSrvcDao.isProviderForNetworkOffering(offering.getId(), Network.Provider.NiciraNvp)).thenReturn(false);
     }
 
--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
@@ -0,0 +1,19 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+import com.cloud.network.Network.GuestType;
+import com.cloud.network.Networks.TrafficType;
+import com.cloud.offering.NetworkOffering;
+
+public class MockNetworkOffering {
+    public static NetworkOffering createMockNetworkOffering(long offeringId, GuestType guestType) {
+        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
+        when(offering.getGuestType()).thenReturn(guestType);
+        when(offering.getId()).thenReturn(offeringId);
+        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
+        return offering;
+    }
+}
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
@@ -0,0 +1,19 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+import com.cloud.network.Network.GuestType;
+import com.cloud.network.Networks.TrafficType;
+import com.cloud.offering.NetworkOffering;
+
+public class MockNetworkOffering {
+    public static NetworkOffering createMockNetworkOffering(long offeringId, GuestType guestType) {
+        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
+        when(offering.getGuestType()).thenReturn(guestType);
+        when(offering.getId()).thenReturn(offeringId);
+        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
+        return offering;
+    }
+}
--- a/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
+++ b/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
@@ -0,0 +1,19 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+import com.cloud.network.Network.GuestType;
+import com.cloud.network.Networks.TrafficType;
+import com.cloud.offering.NetworkOffering;
+
+public class MockNetworkOffering {
+    public static NetworkOffering createMockNetworkOffering(long offeringId, GuestType guestType) {
+        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
+        when(offering.getGuestType()).thenReturn(guestType);
+        when(offering.getId()).thenReturn(offeringId);
+        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
+        return offering;
+    }
+}
--- a/server/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
+++ b/server/src/test/java/com/cloud/network/guru/MockNetworkOffering.java
@@ -0,0 +1,19 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+import com.cloud.network.Network.GuestType;
+import com.cloud.network.Networks.TrafficType;
+import com.cloud.offering.NetworkOffering;
+
+public class MockNetworkOffering {
+    public static NetworkOffering createMockNetworkOffering(long offeringId, GuestType guestType) {
+        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
+        when(offering.getGuestType()).thenReturn(guestType);
+        when(offering.getId()).thenReturn(offeringId);
+        when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
+        return offering;
+    }
+}
