--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -276,8 +276,7 @@
     @Test
     public void prepareExtensionPathOnMSPeerReturnsTrueOnSuccess() {
         Extension ext = mock(Extension.class);
-        ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
-        when(msHost.getMsid()).thenReturn(1L);
+        ManagementServerHostVO msHost = createMockManagementServerHost(1L);
         when(clusterManager.execute(anyString(), anyLong(), anyString(), eq(true)))
                 .thenReturn("answer");
         doReturn(new Pair<>(true, "ok")).when(extensionsManager).getResultFromAnswersString(anyString(), eq(ext), eq(msHost), anyString());
@@ -295,8 +294,7 @@
     @Test
     public void cleanupExtensionFilesOnMSPeerReturnsTrueOnSuccess() {
         Extension ext = mock(Extension.class);
-        ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
-        when(msHost.getMsid()).thenReturn(1L);
+        ManagementServerHostVO msHost = createMockManagementServerHost(1L);
         when(clusterManager.execute(anyString(), anyLong(), anyString(), eq(true)))
                 .thenReturn("answer");
         doReturn(new Pair<>(true, "ok")).when(extensionsManager).getResultFromAnswersString(anyString(), eq(ext), eq(msHost), anyString());
@@ -623,8 +621,7 @@
         when(ext.getName()).thenReturn("ext");
         when(ext.getRelativePath()).thenReturn("entry.sh");
         when(externalProvisioner.getChecksumForExtensionPath("ext", "entry.sh")).thenReturn("checksum123");
-        ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
-        when(msHost.getMsid()).thenReturn(1L);
+        ManagementServerHostVO msHost = createMockManagementServerHost(1L);
         doReturn(new Pair<>(true, "checksum456")).when(extensionsManager).getChecksumForExtensionPathOnMSPeer(ext, msHost);
         extensionsManager.checkExtensionPathState(ext, Collections.singletonList(msHost));
         verify(extensionsManager).updateExtensionPathReady(ext, false);
@@ -637,8 +634,7 @@
         when(ext.getRelativePath()).thenReturn("entry.sh");
         when(externalProvisioner.getChecksumForExtensionPath("ext", "entry.sh")).thenReturn("checksum123");
 
-        ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
-        when(msHost.getMsid()).thenReturn(1L);
+        ManagementServerHostVO msHost = createMockManagementServerHost(1L);
         doReturn(new Pair<>(false, null)).when(extensionsManager).getChecksumForExtensionPathOnMSPeer(ext, msHost);
 
         extensionsManager.checkExtensionPathState(ext, Collections.singletonList(msHost));
@@ -698,10 +694,8 @@
         when(ext.getId()).thenReturn(1L);
         when(ext.isPathReady()).thenReturn(false);
 
-        ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
-        when(msHost1.getMsid()).thenReturn(100L);
-        when(msHost2.getMsid()).thenReturn(200L);
+        ManagementServerHostVO msHost2 = createMockManagementServerHost(200L);
+        ManagementServerHostVO msHost1 = createMockManagementServerHost(100L);
 
         when(managementServerHostDao.listBy(any())).thenReturn(Arrays.asList(msHost1, msHost2));
 
@@ -733,10 +727,8 @@
         when(ext.getId()).thenReturn(1L);
         when(ext.isPathReady()).thenReturn(true);
 
-        ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
-        when(msHost1.getMsid()).thenReturn(101L);
-        when(msHost2.getMsid()).thenReturn(200L);
+        ManagementServerHostVO msHost1 = createMockManagementServerHost(101L);
+        ManagementServerHostVO msHost2 = createMockManagementServerHost(200L);
 
         when(managementServerHostDao.listBy(any())).thenReturn(Arrays.asList(msHost1, msHost2));
 
@@ -763,8 +755,7 @@
         when(ext.getRelativePath()).thenReturn("entry.sh");
         when(ext.isPathReady()).thenReturn(true);
 
-        ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
-        when(msHost.getMsid()).thenReturn(101L);
+        ManagementServerHostVO msHost = createMockManagementServerHost(101L);
 
         when(managementServerHostDao.listBy(any())).thenReturn(Collections.singletonList(msHost));
 
@@ -3062,4 +3053,10 @@
         assertTrue(resp.getResult().get(ApiConstants.DETAILS).contains("Action failed"));
     }
 
+    private static ManagementServerHostVO createMockManagementServerHost(long msid) {
+        ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
+        when(msHost.getMsid()).thenReturn(msid);
+        return msHost;
+    }
+
 }
