--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -618,12 +618,9 @@
             when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
-            SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
+            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
+            SearchBuilder<UserDataVO> sb = createMockSearchBuilder(userData, sc);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
-            when(sb.entity()).thenReturn(userData);
-
-            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
-            when(sb.create()).thenReturn(sc);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
             userDataList.add(userData);
@@ -651,12 +648,9 @@
             when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
-            SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
+            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
+            SearchBuilder<UserDataVO> sb = createMockSearchBuilder(userData, sc);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
-            when(sb.entity()).thenReturn(userData);
-
-            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
-            when(sb.create()).thenReturn(sc);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
             userDataList.add(userData);
@@ -684,12 +678,9 @@
             when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
-            SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
+            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
+            SearchBuilder<UserDataVO> sb = createMockSearchBuilder(userData, sc);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
-            when(sb.entity()).thenReturn(userData);
-
-            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
-            when(sb.create()).thenReturn(sc);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
             userDataList.add(userData);
@@ -700,6 +691,13 @@
 
             Assert.assertEquals(userdataResultList.first().get(0), userDataList.get(0));
         }
+    }
+
+    private static SearchBuilder<UserDataVO> createMockSearchBuilder(UserDataVO userData, SearchCriteria<UserDataVO> sc) {
+        SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
+        when(sb.entity()).thenReturn(userData);
+        when(sb.create()).thenReturn(sc);
+        return sb;
     }
 
     private UserVmVO mockFilterUefiHostsTestVm(String uefiValue) {
