--- a/plugins/integrations/veeam-control-service/src/test/java/org/apache/cloudstack/veeam/adapter/ServerAdapterTest.java
+++ b/plugins/integrations/veeam-control-service/src/test/java/org/apache/cloudstack/veeam/adapter/ServerAdapterTest.java
@@ -115,6 +115,7 @@
     @InjectMocks
     ServerAdapter serverAdapter;
 
+    @Mock private DataCenterVO zoneMock;
     @Mock AccountService accountService;
     @Mock DataCenterDao dataCenterDao;
     @Mock DataCenterJoinDao dataCenterJoinDao;
@@ -659,9 +660,8 @@
         when(offering.isCustomized()).thenReturn(false);
         when(offering.getCpu()).thenReturn(2);
         when(offering.getRamSize()).thenReturn(1024);
-        DataCenterVO zone = mock(DataCenterVO.class);
-
-        assertEquals(offering, serverAdapter.getServiceOfferingFromRequest(zone, account, "uuid4", 2, 1024));
+
+        assertEquals(offering, serverAdapter.getServiceOfferingFromRequest(zoneMock, account, "uuid4", 2, 1024));
     }
 
     @Test
@@ -672,9 +672,8 @@
         doNothing().when(accountService).checkAccess(eq(account), eq(offering), any());
         when(offering.isCustomized()).thenReturn(true);
         doNothing().when(userVmManager).validateCustomParameters(eq(offering), any());
-        DataCenterVO zone = mock(DataCenterVO.class);
-
-        ServiceOfferingVO result = serverAdapter.getServiceOfferingFromRequest(zone, account, "uuid5", 2, 1024);
+
+        ServiceOfferingVO result = serverAdapter.getServiceOfferingFromRequest(zoneMock, account, "uuid5", 2, 1024);
 
         assertEquals(offering, result);
         verify(offering).setCpu(2);
