--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
@@ -260,7 +260,6 @@
     @Test
     public void testAllocateForDomainRouter() throws InsufficientVirtualNetworkCapacityException, InsufficientAddressCapacityException {
         Network network = Mockito.mock(Network.class);
-        NicProfile nicProfile = Mockito.mock(NicProfile.class);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         VirtualMachine virtualMachine = Mockito.mock(VirtualMachine.class);
         Pair<String, String> dns = new Pair<>("10.1.5.1", "8.8.8.8");
@@ -271,6 +270,7 @@
         when(vmProfile.getVirtualMachine()).thenReturn(virtualMachine);
         when(virtualMachine.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(network.getId()).thenReturn(2L);
+        NicProfile nicProfile = MockNicProfile.createMockNicProfile("10.1.13.10");
         when(nicProfile.getMacAddress()).thenReturn(macAddress);
         when(networkOfferingDao.isIpv6Supported(anyLong())).thenReturn(false);
         when(network.getDataCenterId()).thenReturn(1L);
@@ -280,7 +280,6 @@
         when(accountDao.findById(anyLong())).thenReturn(Mockito.mock(AccountVO.class));
         when(vpcDao.findById(anyLong())).thenReturn(Mockito.mock(VpcVO.class));
         when(domainDao.findById(anyLong())).thenReturn(Mockito.mock(DomainVO.class));
-        when(nicProfile.getIPv4Address()).thenReturn("10.1.13.10");
         when(nsxControllerUtils.sendNsxCommand(any(CreateNsxDhcpRelayConfigCommand.class),
                 anyLong())).thenReturn(new NsxAnswer(new NsxCommand(), true, ""));
 
--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
@@ -146,7 +146,6 @@
         String publicIpVR = "10.1.12.10";
         String publicIpNSX = "10.1.13.10";
         Network network = Mockito.mock(Network.class);
-        NicProfile profile = Mockito.mock(NicProfile.class);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         IPAddressVO srcNatIpOnVR = new IPAddressVO(new Ip(publicIpVR), 2L , 0xaabbccddeeffL, 2L, true);
         srcNatIpOnVR.setVpcId(12L);
@@ -158,7 +157,7 @@
         VpcOfferingVO vpcOffering = Mockito.mock(VpcOfferingVO.class);
 
 
-        when(profile.getIPv4Address()).thenReturn(publicIpVR);
+        NicProfile profile = MockNicProfile.createMockNicProfile(publicIpVR);
         when(ipAddressDao.findByIp(anyString())).thenReturn(srcNatIpOnVR);
         when(vpcDao.findById(anyLong())).thenReturn(vpcVO);
         when(ipAddressDao.listByAssociatedVpc(12L, true)).thenReturn(sourceNatList);
--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/MockNicProfile.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/MockNicProfile.java
@@ -0,0 +1,15 @@
+package org.apache.cloudstack.service;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+import com.cloud.vm.NicProfile;
+
+public class MockNicProfile {
+    public static NicProfile createMockNicProfile(String ipv4Address) {
+        NicProfile profile = Mockito.mock(NicProfile.class);
+        when(profile.getIPv4Address()).thenReturn(ipv4Address);
+        return profile;
+    }
+}
