--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -224,9 +224,8 @@
         when(network.getId()).thenReturn(networkId);
         ASNumberVO asNumberVO = Mockito.mock(ASNumberVO.class);
         when(asNumberDao.findByZoneAndNetworkId(zoneId, networkId)).thenReturn(asNumberVO);
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
+        DataCenterVO dc = createMockDataCenterVO();
         when(dcDao.findById(zoneId)).thenReturn(dc);
-        when(dc.getNetworkType()).thenReturn(DataCenter.NetworkType.Advanced);
 
         commandSetupHelper.createBgpPeersCommands(bgpPeers, router, cmds, network);
 
@@ -262,9 +261,8 @@
         when(network2.getNetworkOfferingId()).thenReturn(networkOfferingId);
         when(networkDao.listByVpc(vpcId)).thenReturn(Arrays.asList(network1, network2));
 
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
+        DataCenterVO dc = createMockDataCenterVO();
         when(dcDao.findById(zoneId)).thenReturn(dc);
-        when(dc.getNetworkType()).thenReturn(DataCenter.NetworkType.Advanced);
 
         commandSetupHelper.createBgpPeersCommands(bgpPeers, router, cmds, network);
 
@@ -311,4 +309,10 @@
         ConfigKey<Integer> configKey = NetworkOrchestrationService.DhcpLeaseTimeout;
         Assert.assertTrue("ConfigKey should be dynamic", configKey.isDynamic());
     }
+
+    private static DataCenterVO createMockDataCenterVO() {
+        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
+        when(dc.getNetworkType()).thenReturn(DataCenter.NetworkType.Advanced);
+        return dc;
+    }
 }
