--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -273,6 +273,8 @@
     DataCenterDao dataCenterDao;
     @Mock
     IpAddressManager ipAddressManager;
+    @Mock
+    private SecurityGroup securityGroupMock;
 
     TungstenServiceImpl tungstenService;
 
@@ -682,7 +684,6 @@
 
     @Test
     public void createTungstenSecurityGroupTest() {
-        SecurityGroup securityGroup = mock(SecurityGroup.class);
         ProjectVO projectVO = mock(ProjectVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         DomainVO domainVO = mock(DomainVO.class);
@@ -698,20 +699,19 @@
 
         try (MockedStatic<Transaction> transactionMocked = Mockito.mockStatic(Transaction.class)) {
             transactionMocked.when(() -> Transaction.execute(any(TransactionCallback.class))).thenReturn(List.of(tungstenSecurityGroupRuleVO));
-            assertTrue(tungstenService.createTungstenSecurityGroup(securityGroup));
+            assertTrue(tungstenService.createTungstenSecurityGroup(securityGroupMock));
         }
     }
 
     @Test
     public void deleteTungstenSecurityGroupTest() {
-        SecurityGroup securityGroup = mock(SecurityGroup.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenAnswer deleteTungstenSecurityGroupAnswer = MockTungstenAnswerFactory.get(true);
 
         when(tungstenProviderDao.findAll()).thenReturn(List.of(tungstenProviderVO));
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenSecurityGroupCommand.class), anyLong())).thenReturn(deleteTungstenSecurityGroupAnswer);
 
-        assertTrue(tungstenService.deleteTungstenSecurityGroup(securityGroup));
+        assertTrue(tungstenService.deleteTungstenSecurityGroup(securityGroupMock));
     }
 
     @Test
