--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -276,6 +276,9 @@
     @Mock
     NetworkOrchestrationService networkOrchestrationService;
 
+    @Mock
+    private SearchCriteria<Long> searchCriteriaMockForCondition;
+
     AccountVO account;
     UserVO user;
 
@@ -607,12 +610,11 @@
     @Test
     public void testUpdateCondition() throws ResourceInUseException {
         GenericSearchBuilder<AutoScalePolicyConditionMapVO, Long> searchBuilderMock = Mockito.mock(GenericSearchBuilder.class);
-        SearchCriteria<Long> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
 
         Mockito.doReturn(searchBuilderMock).when(autoScalePolicyConditionMapDao).createSearchBuilder(any());
         when(searchBuilderMock.entity()).thenReturn(autoScalePolicyConditionMapVOMock);
-        Mockito.doReturn(searchCriteriaMock).when(searchBuilderMock).create();
-        Mockito.doReturn(Arrays.asList()).when(autoScalePolicyConditionMapDao).customSearch(searchCriteriaMock, null);
+        Mockito.doReturn(searchCriteriaMockForCondition).when(searchBuilderMock).create();
+        Mockito.doReturn(Arrays.asList()).when(autoScalePolicyConditionMapDao).customSearch(searchCriteriaMockForCondition, null);
 
         when(conditionDao.update(eq(conditionId), any())).thenReturn(true);
 
@@ -657,12 +659,11 @@
     @Test(expected = ResourceInUseException.class)
     public void testUpdateConditionWithPolicies() throws ResourceInUseException {
         GenericSearchBuilder<AutoScalePolicyConditionMapVO, Long> genericSearchBuilderMock = Mockito.mock(GenericSearchBuilder.class);
-        SearchCriteria<Long> searchCriteriaLongMock = Mockito.mock(SearchCriteria.class);
 
         Mockito.doReturn(genericSearchBuilderMock).when(autoScalePolicyConditionMapDao).createSearchBuilder(Long.class);
         when(genericSearchBuilderMock.entity()).thenReturn(autoScalePolicyConditionMapVOMock);
-        Mockito.doReturn(searchCriteriaLongMock).when(genericSearchBuilderMock).create();
-        Mockito.doReturn(Arrays.asList(scaleUpPolicyId)).when(autoScalePolicyConditionMapDao).customSearch(searchCriteriaLongMock, null);
+        Mockito.doReturn(searchCriteriaMockForCondition).when(genericSearchBuilderMock).create();
+        Mockito.doReturn(Arrays.asList(scaleUpPolicyId)).when(autoScalePolicyConditionMapDao).customSearch(searchCriteriaMockForCondition, null);
 
         SearchBuilder<AutoScaleVmGroupPolicyMapVO> searchBuilderMock1 = Mockito.mock(SearchBuilder.class);
         Mockito.doReturn(searchBuilderMock1).when(autoScaleVmGroupPolicyMapDao).createSearchBuilder();
