--- a/api/src/test/java/org/apache/cloudstack/api/command/user/firewall/CreateEgressFirewallRuleCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/firewall/CreateEgressFirewallRuleCmdTest.java
@@ -66,10 +66,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn(cidrMock).when(networkMock).getCidr();
+        Network networkMock = createMockNetwork(cidrMock);
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(cidrMock)).thenReturn(true);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(cidrMock)).thenReturn(false);
@@ -98,10 +96,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn("10.1.1.0/24").when(networkMock).getCidr();
+        Network networkMock = createMockNetwork("10.1.1.0/24");
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(Mockito.eq("10.1.1.0/24"))).thenReturn(true);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(Mockito.eq("10.1.1.0/24"))).thenReturn(false);
@@ -131,10 +127,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn("10.1.1.0/24").when(networkMock).getCidr();
+        Network networkMock = createMockNetwork("10.1.1.0/24");
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(sourceCidrMock)).thenReturn(false);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(sourceCidrMock)).thenReturn(false);
@@ -165,10 +159,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn(sourceCidrMock).when(networkMock).getCidr();
+        Network networkMock = createMockNetwork(sourceCidrMock);
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(Mockito.eq(sourceCidrMock))).thenReturn(true);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(Mockito.eq(sourceCidrMock))).thenReturn(false);
@@ -209,10 +201,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn(sourceCidrMock).when(networkMock).getCidr();
+        Network networkMock = createMockNetwork(sourceCidrMock);
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(Mockito.eq(sourceCidrMock))).thenReturn(true);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(Mockito.eq(sourceCidrMock))).thenReturn(false);
@@ -247,10 +237,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn(cidrMock).when(networkMock).getCidr();
+        Network networkMock = createMockNetwork(cidrMock);
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(cidrMock)).thenReturn(true);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(cidrMock)).thenReturn(false);
@@ -279,10 +267,8 @@
         long networkIdMock = 1234;
         Mockito.doReturn(networkIdMock).when(cmdMock).getNetworkId();
 
-        Network networkMock = Mockito.mock(Network.class);
-        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
-
-        Mockito.doReturn(sourceCidrMock).when(networkMock).getCidr();
+        Network networkMock = createMockNetwork(sourceCidrMock);
+        Mockito.doReturn(networkMock).when(networkServiceMock).getNetwork(networkIdMock);
 
         netUtilsMocked.when(() -> NetUtils.isValidIp4Cidr(Mockito.eq(sourceCidrMock))).thenReturn(true);
         netUtilsMocked.when(() -> NetUtils.isValidIp6Cidr(Mockito.eq(sourceCidrMock))).thenReturn(false);
@@ -303,4 +289,10 @@
         netUtilsMocked.verify(() -> NetUtils.isValidIp4Cidr(null), Mockito.never());
         netUtilsMocked.verify(() -> NetUtils.isValidIp6Cidr(null), Mockito.never());
     }
+
+    private static Network createMockNetwork(String cidrReturn) {
+        Network networkMock = Mockito.mock(Network.class);
+        Mockito.doReturn(cidrReturn).when(networkMock).getCidr();
+        return networkMock;
+    }
 }
