--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -1082,6 +1082,13 @@
         return cluster;
     }
 
+    private static HostVO createMockHost(Status status) {
+        HostVO host = Mockito.mock(HostVO.class);
+        when(host.getResourceState()).thenReturn(ResourceState.Enabled);
+        when(host.getStatus()).thenReturn(status);
+        return host;
+    }
+
     @Test(expected = CloudRuntimeException.class)
     public void testSelectInstanceConversionTemporaryLocationInvalidStorage() {
         ClusterVO cluster = getClusterForTests();
@@ -1199,9 +1206,7 @@
     public void testSelectKVMHostForImportingInClusterWithImportInstanceIdInvalidType() {
         Long hostId = 1L;
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
-        when(host.getResourceState()).thenReturn(ResourceState.Enabled);
-        when(host.getStatus()).thenReturn(Status.Up);
+        HostVO host = createMockHost(Status.Up);
         when(host.getType()).thenReturn(Host.Type.Storage);
 
         when(hostDao.findById(hostId)).thenReturn(host);
@@ -1347,9 +1352,7 @@
     public void testSelectKVMHostForConversionInClusterWithImportInstanceIdInvalidStatus() {
         Long hostId = 1L;
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
-        when(host.getResourceState()).thenReturn(ResourceState.Enabled);
-        when(host.getStatus()).thenReturn(Status.Down);
+        HostVO host = createMockHost(Status.Down);
 
         when(hostDao.findById(hostId)).thenReturn(host);
 
