--- a/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/OAuth2UserAuthenticatorTest.java
+++ b/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/OAuth2UserAuthenticatorTest.java
@@ -59,6 +59,9 @@
     private UserDao userDao;
 
     @Mock
+    private UserAccount userAccountMock;
+
+    @Mock
     private OAuth2AuthManager userOAuth2mgr;
 
     @InjectMocks
@@ -86,12 +89,11 @@
         String[] email = {"testemail"};
         String[] secretCode = {"testsecretcode"};
 
-        UserAccount userAccount = mock(UserAccount.class);
         UserVO user = mock(UserVO.class);
         UserOAuth2Authenticator userOAuth2Authenticator = mock(UserOAuth2Authenticator.class);
 
-        when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccount);
-        when(userDao.getUser(userAccount.getId())).thenReturn(user);
+        when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccountMock);
+        when(userDao.getUser(userAccountMock.getId())).thenReturn(user);
         when(userOAuth2mgr.getUserOAuth2AuthenticationProvider(provider[0])).thenReturn(userOAuth2Authenticator);
         when(userOAuth2Authenticator.verifyUser(email[0], secretCode[0], domainId)).thenReturn(true);
 
@@ -106,7 +108,7 @@
         assertNull(result.second());
 
         verify(userAccountDao).getUserAccount(username, domainId);
-        verify(userDao).getUser(userAccount.getId());
+        verify(userDao).getUser(userAccountMock.getId());
         verify(userOAuth2mgr).getUserOAuth2AuthenticationProvider(provider[0]);
         verify(userOAuth2Authenticator).verifyUser(email[0], secretCode[0], domainId);
     }
@@ -119,12 +121,11 @@
         String[] email = {"testemail"};
         String[] secretCode = {"testsecretcode"};
 
-        UserAccount userAccount = mock(UserAccount.class);
         UserVO user = mock(UserVO.class);
         UserOAuth2Authenticator userOAuth2Authenticator = mock(UserOAuth2Authenticator.class);
 
-        when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccount);
-        when(userDao.getUser(userAccount.getId())).thenReturn(user);
+        when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccountMock);
+        when(userDao.getUser(userAccountMock.getId())).thenReturn(user);
         when(userOAuth2mgr.getUserOAuth2AuthenticationProvider(provider[0])).thenReturn(userOAuth2Authenticator);
         when(userOAuth2Authenticator.verifyUser(email[0], secretCode[0], domainId)).thenReturn(false);
 
@@ -139,7 +140,7 @@
         assertEquals(OAuth2UserAuthenticator.ActionOnFailedAuthentication.INCREMENT_INCORRECT_LOGIN_ATTEMPT_COUNT, result.second());
 
         verify(userAccountDao).getUserAccount(username, domainId);
-        verify(userDao).getUser(userAccount.getId());
+        verify(userDao).getUser(userAccountMock.getId());
         verify(userOAuth2mgr).getUserOAuth2AuthenticationProvider(provider[0]);
         verify(userOAuth2Authenticator).verifyUser(email[0], secretCode[0], domainId);
     }
@@ -196,9 +197,8 @@
         String username = "testuser";
         Long domainId = 1L;
 
-        UserAccount userAccount = mock(UserAccount.class);
-        when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccount);
-        when(userDao.getUser(userAccount.getId())).thenReturn(mock(UserVO.class));
+        when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccountMock);
+        when(userDao.getUser(userAccountMock.getId())).thenReturn(mock(UserVO.class));
 
         Map<String, Object[]> requestParameters = new HashMap<>();
         requestParameters.put("email", new String[]{"test@email.com"});
