--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -99,6 +99,8 @@
 
     @Mock
     Account caller;
+    @Mock
+    private SearchCriteria<WebhookJoinVO> webhookSearchCriteria;
 
     @Spy
     @InjectMocks
@@ -782,8 +784,7 @@
         Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         SearchBuilder<WebhookJoinVO> sb = Mockito.mock(SearchBuilder.class);
         Mockito.when(sb.entity()).thenReturn(Mockito.mock(WebhookJoinVO.class));
-        SearchCriteria<WebhookJoinVO> sc = Mockito.mock(SearchCriteria.class);
-        Mockito.when(sb.create()).thenReturn(sc);
+        Mockito.when(sb.create()).thenReturn(webhookSearchCriteria);
         Mockito.when(webhookJoinDao.createSearchBuilder()).thenReturn(sb);
         Mockito.when(webhookJoinDao.searchAndCount(Mockito.any(), Mockito.any())).thenReturn(new Pair<>(List.of(), 0));
 
@@ -825,8 +826,7 @@
         Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         SearchBuilder<WebhookJoinVO> sb = Mockito.mock(SearchBuilder.class);
         Mockito.when(sb.entity()).thenReturn(Mockito.mock(WebhookJoinVO.class));
-        SearchCriteria<WebhookJoinVO> sc = Mockito.mock(SearchCriteria.class);
-        Mockito.when(sb.create()).thenReturn(sc);
+        Mockito.when(sb.create()).thenReturn(webhookSearchCriteria);
         Mockito.when(webhookJoinDao.createSearchBuilder()).thenReturn(sb);
         Mockito.when(webhookJoinDao.searchAndCount(Mockito.any(), Mockito.any())).thenReturn(new Pair<>(List.of(webhook), 1));
 
