--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -273,6 +273,8 @@
     DataCenterDao dataCenterDao;
     @Mock
     IpAddressManager ipAddressManager;
+    @Mock
+    private NetworkDetailVO networkDetailVOMock;
 
     TungstenServiceImpl tungstenService;
 
@@ -475,7 +477,6 @@
     public void createPublicNetworkTest() {
         Network publicNetwork = mock(Network.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        NetworkDetailVO networkDetailVO = mock(NetworkDetailVO.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         TungstenAnswer createPublicNetworkAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer createFloatingIpPoolAnswer = MockTungstenAnswerFactory.get(true);
@@ -489,7 +490,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(applyTungstenPolicyAnswer);
         when(createPublicNetworkAnswer.getApiObjectBase()).thenReturn(virtualNetwork);
         when(createTungstenPolicyAnswer.getApiObjectBase()).thenReturn(networkPolicy);
-        when(networkDetailsDao.persist(any(NetworkDetailVO.class))).thenReturn(networkDetailVO);
+        when(networkDetailsDao.persist(any(NetworkDetailVO.class))).thenReturn(networkDetailVOMock);
         when(virtualNetwork.getQualifiedName()).thenReturn(Arrays.asList("default-domain", "default-project", "publicNetwork"));
 
         assertTrue(tungstenService.createPublicNetwork(1L));
@@ -1502,7 +1503,6 @@
         TungstenAnswer createTungstenSharedNetworkAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer addNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
-        NetworkDetailVO networkDetailVO = mock(NetworkDetailVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         HostVO hostVO = mock(HostVO.class);
         Answer answer = mock(Answer.class);
@@ -1518,7 +1518,7 @@
         when(vlan.getIpRange()).thenReturn("192.168.100.100-192.168.100.200");
         when(vlan.getVlanGateway()).thenReturn("192.168.100.1");
         when(vlan.getIp6Gateway()).thenReturn("fd00::1");
-        when(networkDetailsDao.persist(any(NetworkDetailVO.class))).thenReturn(networkDetailVO);
+        when(networkDetailsDao.persist(any(NetworkDetailVO.class))).thenReturn(networkDetailVOMock);
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
         when(hostDao.findByPublicIp(anyString())).thenReturn(hostVO);
         when(agentMgr.easySend(anyLong(), any(SetupTungstenVRouterCommand.class))).thenReturn(answer);
