--- a/api/src/test/java/org/apache/cloudstack/api/command/user/vpc/CreateVPCCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/vpc/CreateVPCCmdTest.java
@@ -51,6 +51,10 @@
     @Mock
     private Object job;
 
+    private Vpc mockVpc() {
+        return Mockito.mock(Vpc.class);
+    }
+
     @InjectMocks
     CreateVPCCmd cmd;
 
@@ -168,7 +172,7 @@
 
     @Test
     public void testCreate() throws ResourceAllocationException {
-        Vpc vpc = Mockito.mock(Vpc.class);
+        Vpc vpc = mockVpc();
         ReflectionTestUtils.setField(cmd, "zoneId", 1L);
         ReflectionTestUtils.setField(cmd, "vpcOffering", 1L);
         ReflectionTestUtils.setField(cmd, "vpcName", "testVpc");
@@ -184,7 +188,7 @@
 
     @Test
     public void testExecute() throws ResourceUnavailableException, InsufficientCapacityException {
-        Vpc vpc = Mockito.mock(Vpc.class);
+        Vpc vpc = mockVpc();
         VpcResponse response = Mockito.mock(VpcResponse.class);
 
         ReflectionTestUtils.setField(cmd, "id", 1L);
