--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -321,9 +321,9 @@
     @Test
     public void assignVMToBackupOfferingTestKvmWithUnsupportedDiskOnlyVmSnapshot() {
         doReturn(Hypervisor.HypervisorType.KVM).when(virtualMachineMock).getHypervisorType();
+        long vmSnapId = 921;
+        vmSnapshotVoMock = createMockVmSnapshotVo(vmSnapId);
         doReturn(List.of(vmSnapshotVoMock)).when(vmSnapshotDaoMock).findByVmAndByType(vmId, VMSnapshot.Type.Disk);
-        long vmSnapId = 921;
-        doReturn(vmSnapId).when(vmSnapshotVoMock).getId();
         doReturn(List.of(vmSnapshotDetailsVoMock)).when(vmSnapshotDetailsDaoMock).listDetails(vmSnapId);
         doReturn("Anything").when(vmSnapshotDetailsVoMock).getName();
 
@@ -334,9 +334,9 @@
     @Test
     public void assignVMToBackupOfferingTestKvmWithSupportedDiskOnlyVmSnapshotAndDiskAndMemoryVmSnapshot() {
         doReturn(Hypervisor.HypervisorType.KVM).when(virtualMachineMock).getHypervisorType();
+        long vmSnapId = 921;
+        vmSnapshotVoMock = createMockVmSnapshotVo(vmSnapId);
         doReturn(List.of(vmSnapshotVoMock)).when(vmSnapshotDaoMock).findByVmAndByType(vmId, VMSnapshot.Type.Disk);
-        long vmSnapId = 921;
-        doReturn(vmSnapId).when(vmSnapshotVoMock).getId();
         doReturn(List.of(vmSnapshotDetailsVoMock)).when(vmSnapshotDetailsDaoMock).listDetails(vmSnapId);
         doReturn(VolumeApiServiceImpl.KVM_FILE_BASED_STORAGE_SNAPSHOT).when(vmSnapshotDetailsVoMock).getName();
         doReturn(List.of(vmSnapshotVoMock)).when(vmSnapshotDaoMock).findByVmAndByType(vmId, VMSnapshot.Type.DiskAndMemory);
@@ -349,9 +349,9 @@
     @Test
     public void assignVMToBackupOfferingTestKvmWithSupportedDiskOnlyVmSnapshotAndNoDiskAndMemoryVmSnapshot() {
         doReturn(Hypervisor.HypervisorType.KVM).when(virtualMachineMock).getHypervisorType();
+        long vmSnapId = 921;
+        vmSnapshotVoMock = createMockVmSnapshotVo(vmSnapId);
         doReturn(List.of(vmSnapshotVoMock)).when(vmSnapshotDaoMock).findByVmAndByType(vmId, VMSnapshot.Type.Disk);
-        long vmSnapId = 921;
-        doReturn(vmSnapId).when(vmSnapshotVoMock).getId();
         doReturn(List.of(vmSnapshotDetailsVoMock)).when(vmSnapshotDetailsDaoMock).listDetails(vmSnapId);
         doReturn(VolumeApiServiceImpl.KVM_FILE_BASED_STORAGE_SNAPSHOT).when(vmSnapshotDetailsVoMock).getName();
 
@@ -676,9 +676,8 @@
 
     @Test
     public void mapVolumesToVmSnapshotReferencesTestVmSnapshotVOListHasTwoElements() {
-        VMSnapshotVO vmSnapshotVoMock1 = Mockito.mock(VMSnapshotVO.class);
-        doReturn(1L).when(vmSnapshotVoMock).getId();
-        doReturn(2L).when(vmSnapshotVoMock1).getId();
+        VMSnapshotVO vmSnapshotVoMock = createMockVmSnapshotVo(1L);
+        VMSnapshotVO vmSnapshotVoMock1 = createMockVmSnapshotVo(2L);
         doNothing().when(kbossBackupProviderSpy).mapVolumesToSnapshotReferences(Mockito.anyList(), Mockito.anyList(), anyMap());
 
         kbossBackupProviderSpy.mapVolumesToVmSnapshotReferences(List.of(), List.of(vmSnapshotVoMock, vmSnapshotVoMock1));
@@ -2200,7 +2199,7 @@
 
     @Test
     public void gatherSnapshotReferencesOfChildrenSnapshotTestChildrenListIsEmpty() {
-        doReturn(100L).when(vmSnapshotVoMock).getId();
+        vmSnapshotVoMock = createMockVmSnapshotVo(100L);
         doReturn(List.of()).when(vmSnapshotDaoMock).listByParent(100L);
 
         List<VolumeObjectTO> volumeObjectTOs = List.of(volumeObjectToMock);
@@ -2215,12 +2214,11 @@
 
     @Test
     public void gatherSnapshotReferencesOfChildrenSnapshotTestSingleChildWithSingleSnapshotReference() {
-        VMSnapshotVO childSnapshot = Mockito.mock(VMSnapshotVO.class);
+        VMSnapshotVO childSnapshot = createMockVmSnapshotVo(200L);
         SnapshotDataStoreVO snapshotDataStoreVO = Mockito.mock(SnapshotDataStoreVO.class);
 
-        doReturn(100L).when(vmSnapshotVoMock).getId();
+        vmSnapshotVoMock = createMockVmSnapshotVo(100L);
         doReturn(List.of(childSnapshot)).when(vmSnapshotDaoMock).listByParent(100L);
-        doReturn(200L).when(childSnapshot).getId();
         doReturn(List.of(snapshotDataStoreVO)).when(vmSnapshotHelperMock).getVolumeSnapshotsAssociatedWithKvmDiskOnlyVmSnapshot(200L);
         doNothing().when(kbossBackupProviderSpy).mapVolumesToSnapshotReferences(anyList(), anyList(), anyMap());
 
@@ -2237,15 +2235,13 @@
 
     @Test
     public void gatherSnapshotReferencesOfChildrenSnapshotTestMultipleChildrenAggregatesSnapshotReferences() {
-        VMSnapshotVO childSnapshot1 = Mockito.mock(VMSnapshotVO.class);
-        VMSnapshotVO childSnapshot2 = Mockito.mock(VMSnapshotVO.class);
+        VMSnapshotVO childSnapshot1 = createMockVmSnapshotVo(201L);
+        VMSnapshotVO childSnapshot2 = createMockVmSnapshotVo(202L);
         SnapshotDataStoreVO snapshotDataStoreVO1 = Mockito.mock(SnapshotDataStoreVO.class);
         SnapshotDataStoreVO snapshotDataStoreVO2 = Mockito.mock(SnapshotDataStoreVO.class);
 
-        doReturn(100L).when(vmSnapshotVoMock).getId();
+        vmSnapshotVoMock = createMockVmSnapshotVo(100L);
         doReturn(List.of(childSnapshot1, childSnapshot2)).when(vmSnapshotDaoMock).listByParent(100L);
-        doReturn(201L).when(childSnapshot1).getId();
-        doReturn(202L).when(childSnapshot2).getId();
         doReturn(List.of(snapshotDataStoreVO1)).when(vmSnapshotHelperMock)
                 .getVolumeSnapshotsAssociatedWithKvmDiskOnlyVmSnapshot(201L);
         doReturn(List.of(snapshotDataStoreVO2)).when(vmSnapshotHelperMock)
@@ -3113,4 +3109,10 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+    private static VMSnapshotVO createMockVmSnapshotVo(long snapshotId) {
+        VMSnapshotVO vmSnapshotVoMock = Mockito.mock(VMSnapshotVO.class);
+        doReturn(snapshotId).when(vmSnapshotVoMock).getId();
+        return vmSnapshotVoMock;
+    }
+
 }
