--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
@@ -269,9 +269,8 @@
 
         long networkId = 1L;
         Mockito.when(createNetworkAclCmdMock.getNetworkId()).thenReturn(networkId);
-        Network networkMock = Mockito.mock(Network.class);
-
-        Mockito.when(networkMock.getVpcId()).thenReturn(12L);
+        Network networkMock = MockNetworkWithVpcId.createMockNetwork(12L);
+
         Long expectedAclListId = 15L;
         Mockito.when(networkMock.getNetworkACLId()).thenReturn(expectedAclListId);
 
--- a/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
@@ -599,13 +599,12 @@
 
     @Test
     public void testIsNetworkOnVpcEnabledConserveModeVpcNetworkConserveMode() {
-        Network network = mock(Network.class);
         Vpc vpc = mock(Vpc.class);
         VpcOfferingVO vpcOffering = mock(VpcOfferingVO.class);
         long vpcId = 10L;
+        Network network = MockNetworkWithVpcId.createMockNetwork(vpcId);
         long vpcOfferingId = 11L;
 
-        Mockito.when(network.getVpcId()).thenReturn(vpcId);
         Mockito.when(vpcDao.getActiveVpcById(Mockito.eq(vpcId))).thenReturn(vpc);
         Mockito.when(vpc.getVpcOfferingId()).thenReturn(vpcOfferingId);
         Mockito.when(vpcOfferingDao.findById(Mockito.eq(vpcOfferingId))).thenReturn(vpcOffering);
--- a/server/src/test/java/com/cloud/network/vpc/MockNetworkWithVpcId.java
+++ b/server/src/test/java/com/cloud/network/vpc/MockNetworkWithVpcId.java
@@ -0,0 +1,14 @@
+package com.cloud.network.vpc;
+
+import org.mockito.Mockito;
+
+import com.cloud.network.Network;
+
+public class MockNetworkWithVpcId {
+
+    public static Network createMockNetwork(long vpcId) {
+        Network networkMock = Mockito.mock(Network.class);
+        Mockito.when(networkMock.getVpcId()).thenReturn(vpcId);
+        return networkMock;
+    }
+}
