--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -2324,8 +2324,7 @@
         when(cmd.getId()).thenReturn(id);
         when(cmd.getDomainIds()).thenReturn(List.of(domainId));
 
-        DomainVO domain = Mockito.mock(DomainVO.class);
-        when(domainDao.findById(domainId)).thenReturn(domain);
+        when(domainDao.findById(domainId)).thenReturn(domainVOMock);
 
         when(domainHelper.filterChildSubDomains(List.of(domainId))).thenReturn(new ArrayList<>(List.of(domainId)));
         when(backupOfferingDetailsDao.findDomainIds(id)).thenReturn(new ArrayList<>());
@@ -2659,8 +2658,7 @@
         when(savedOffering.getId()).thenReturn(savedOfferingId);
         when(backupOfferingDao.persist(any(BackupOfferingVO.class))).thenReturn(savedOffering);
 
-        DomainVO domain = Mockito.mock(DomainVO.class);
-        when(domainDao.findById(11L)).thenReturn(domain);
+        when(domainDao.findById(11L)).thenReturn(domainVOMock);
 
         overrideBackupFrameworkConfigValue();
 
@@ -2708,9 +2706,8 @@
         when(backupOfferingDao.persist(any(BackupOfferingVO.class))).thenReturn(savedOffering);
 
         // domain handling
-        DomainVO domain21 = Mockito.mock(DomainVO.class);
         DomainVO domain22 = Mockito.mock(DomainVO.class);
-        when(domainDao.findById(21L)).thenReturn(domain21);
+        when(domainDao.findById(21L)).thenReturn(domainVOMock);
         when(domainDao.findById(22L)).thenReturn(domain22);
         when(domainHelper.filterChildSubDomains(sourceDomainIds)).thenReturn(new ArrayList<>(sourceDomainIds));
 
