--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -151,6 +151,8 @@
     @Mock
     AgentManager agentMgr;
 
+    private final Network networkMock = mock(Network.class);
+
     TungstenGuestNetworkGuru guru;
 
     AutoCloseable closeable;
@@ -230,10 +232,9 @@
                 Network.Service.SourceNat, Network.Service.StaticNat, Network.Service.UserData));
         when(networkModel.listNetworkOfferingServices(anyLong())).thenReturn(networkOfferingServiceList);
 
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designedNetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designedNetwork = guru.design(offering, plan, networkMock, "", 1L, account);
         assertNotNull(designedNetwork);
         assertSame(Networks.BroadcastDomainType.TUNGSTEN, designedNetwork.getBroadcastDomainType());
         assertSame(Network.State.Allocated, designedNetwork.getState());
@@ -435,9 +436,8 @@
 
     @Test
     public void testTrash() {
-        final Network network = mock(Network.class);
-
-        guru.trash(network, offering);
+
+        guru.trash(networkMock, offering);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenNetworkPolicyCommand.class),
             anyLong());
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(ClearTungstenNetworkGatewayCommand.class),
@@ -494,7 +494,6 @@
     @Test
     public void testRollbackMigration() {
         final NicProfile nic = mock(NicProfile.class);
-        final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
         final VirtualMachine vm = mock(VirtualMachine.class);
@@ -504,7 +503,7 @@
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        guru.rollbackMigration(nic, network, vmProfile, context, context);
+        guru.rollbackMigration(nic, networkMock, vmProfile, context, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
@@ -512,7 +511,6 @@
     @Test
     public void testCommitMigration() {
         final NicProfile nic = mock(NicProfile.class);
-        final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
         final VirtualMachine vm = mock(VirtualMachine.class);
@@ -522,7 +520,7 @@
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        guru.commitMigration(nic, network, vmProfile, context, context);
+        guru.commitMigration(nic, networkMock, vmProfile, context, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
