--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
@@ -65,6 +65,8 @@
 public class NsxPublicNetworkGuruTest {
 
     NetworkOffering offering;
+    @Mock
+    DeploymentPlan deploymentPlan;
 
     NsxPublicNetworkGuru guru;
     @Mock
@@ -119,25 +121,23 @@
 
     @Test
     public void testDesign() {
-        DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
         Network network = Mockito.mock(Network.class);
         Account account = Mockito.mock(Account.class);
 
 //        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
 
-        Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, account);
+        Network designedNetwork = guru.design(offering, deploymentPlan, network, "net1", 1L, account);
         Assert.assertEquals(Networks.TrafficType.Public, designedNetwork.getTrafficType());
     }
 
     @Test
     public void testDesign_whenOfferingIsForGuestTraffic() {
-        DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
         Network network = Mockito.mock(Network.class);
         Account account = Mockito.mock(Account.class);
 
         when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
 
-        Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, account);
+        Network designedNetwork = guru.design(offering, deploymentPlan, network, "net1", 1L, account);
         Assert.assertNull(designedNetwork);
     }
 
