--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -2121,11 +2121,10 @@
     public void updateCountersMapWithInstantDataForMemory() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
-        AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
+        AutoScalePolicyTO scaleUpPolicyTO = createMockAutoScalePolicyTO();
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
 
         when(conditionTO.getId()).thenReturn(conditionId);
-        when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
 
         when(counterDao.findById(counterId)).thenReturn(counterMock);
         when(counterMock.getSource()).thenReturn(Counter.Source.MEMORY);
@@ -2157,11 +2156,10 @@
     public void updateCountersMapWithInstantDataForCPU() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
-        AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
+        AutoScalePolicyTO scaleUpPolicyTO = createMockAutoScalePolicyTO();
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
 
         when(conditionTO.getId()).thenReturn(conditionId);
-        when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
         when(counterDao.findById(counterId)).thenReturn(counterMock);
         when(counterMock.getSource()).thenReturn(Counter.Source.CPU);
 
@@ -2425,10 +2423,9 @@
     @Test
     public void cleanupAsVmGroupStatistics() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
+        AutoScalePolicyTO policyTO = createMockAutoScalePolicyTO();
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(groupTO.getId()).thenReturn(vmGroupId);
-        when(policyTO.getId()).thenReturn(scaleUpPolicyId);
 
         autoScaleManagerImplSpy.cleanupAsVmGroupStatistics(groupTO);
 
@@ -2581,4 +2578,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static AutoScalePolicyTO createMockAutoScalePolicyTO() {
+        AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
+        when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
+        return scaleUpPolicyTO;
+    }
 }
