--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -422,9 +422,14 @@
         when(serviceOfferingDao.findByIdIncludingRemoved(offeringIdForVmsOfThisAccount)).thenReturn(offeringForVmOfThisAccount);
         when(offeringForVmOfThisAccount.getDeploymentPlanner()).thenReturn(planner.getName());
 
-        ServiceOfferingVO offeringForVMOfOtherAccount = mock(ServiceOfferingVO.class);
+        ServiceOfferingVO offeringForVMOfOtherAccount = createMockServiceOfferingForVmOfOtherAccount();
         when(serviceOfferingDao.findByIdIncludingRemoved(offeringIdForVmsOfOtherAccount)).thenReturn(offeringForVMOfOtherAccount);
+    }
+
+    private static ServiceOfferingVO createMockServiceOfferingForVmOfOtherAccount() {
+        ServiceOfferingVO offeringForVMOfOtherAccount = Mockito.mock(ServiceOfferingVO.class);
         when(offeringForVMOfOtherAccount.getDeploymentPlanner()).thenReturn("FirstFitPlanner");
+        return offeringForVMOfOtherAccount;
     }
 
     @Configuration
