--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -81,6 +81,7 @@
     AgentManager agentmgr = mock(AgentManager.class);
     NetworkOrchestrationService netmgr = mock(NetworkOrchestrationService.class);
     NetworkModel netmodel = mock(NetworkModel.class);
+    private final DataCenter dataCenter = mock(DataCenter.class);
 
     HostDao hostdao = mock(HostDao.class);
     NetworkDao netdao = mock(NetworkDao.class);
@@ -269,8 +270,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenter dc = mock(DataCenter.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
 
         final HostVO niciraHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
@@ -325,8 +325,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenter dc = mock(DataCenter.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
 
         final HostVO niciraHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
@@ -381,8 +380,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenter dc = mock(DataCenter.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
 
         final HostVO niciraHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
@@ -438,8 +436,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenter dc = mock(DataCenter.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
 
         final HostVO niciraHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
