--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
@@ -83,6 +83,8 @@
     NsxControllerUtils nsxControllerUtils;
     @Mock
     NetworkModel networkModel;
+    @Mock
+    private Account accountMock;
 
     @Before
     public void setup() {
@@ -121,11 +123,10 @@
     public void testDesign() {
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
         Network network = Mockito.mock(Network.class);
-        Account account = Mockito.mock(Account.class);
 
 //        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
 
-        Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, account);
+        Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, accountMock);
         Assert.assertEquals(Networks.TrafficType.Public, designedNetwork.getTrafficType());
     }
 
@@ -133,11 +134,10 @@
     public void testDesign_whenOfferingIsForGuestTraffic() {
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
         Network network = Mockito.mock(Network.class);
-        Account account = Mockito.mock(Account.class);
 
         when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
 
-        Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, account);
+        Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, accountMock);
         Assert.assertNull(designedNetwork);
     }
 
