--- a/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedSANStrategyTest.java
+++ b/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedSANStrategyTest.java
@@ -395,8 +395,7 @@
         lenient().when(storagePoolDetailsDao.listDetailsKeyPairs(1L)).thenReturn(details);
 
         List<HostVO> hosts = new ArrayList<>();
-        HostVO host1 = mock(HostVO.class);
-        lenient().when(host1.getUuid()).thenReturn("host1");
+        HostVO host1 = createMockHostVO();
         hosts.add(host1);
         accessGroup.setHostsToConnect(hosts);
 
@@ -443,8 +442,7 @@
         lenient().when(storagePoolDetailsDao.listDetailsKeyPairs(1L)).thenReturn(details);
 
         List<HostVO> hosts = new ArrayList<>();
-        HostVO host1 = mock(HostVO.class);
-        lenient().when(host1.getUuid()).thenReturn("host1");
+        HostVO host1 = createMockHostVO();
         hosts.add(host1);
         accessGroup.setHostsToConnect(hosts);
 
@@ -1802,4 +1800,10 @@
             verify(sanFeignClient, never()).createLunMap(any(), anyBoolean(), any(LunMap.class));
         }
     }
+
+    private static HostVO createMockHostVO() {
+        HostVO host1 = mock(HostVO.class);
+        lenient().when(host1.getUuid()).thenReturn("host1");
+        return host1;
+    }
 }
