--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -74,6 +74,7 @@
 
 public class NiciraNvpGuestNetworkGuruTest {
     private static final long NETWORK_ID = 42L;
+    private final Account designAccount = mock(Account.class);
     PhysicalNetworkDao physnetdao = mock(PhysicalNetworkDao.class);
     NiciraNvpDao nvpdao = mock(NiciraNvpDao.class);
     DataCenterDao dcdao = mock(DataCenterDao.class);
@@ -166,9 +167,8 @@
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
-        final Account account = mock(Account.class);
-
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+
+        final Network designednetwork = guru.design(offering, plan, network, "", 1L, designAccount);
         assertTrue(designednetwork != null);
         assertTrue(designednetwork.getBroadcastDomainType() == BroadcastDomainType.Lswitch);
     }
@@ -189,9 +189,8 @@
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
-        final Account account = mock(Account.class);
-
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+
+        final Network designednetwork = guru.design(offering, plan, network, "", 1L, designAccount);
         assertTrue(designednetwork == null);
     }
 
@@ -211,9 +210,8 @@
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
-        final Account account = mock(Account.class);
-
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+
+        final Network designednetwork = guru.design(offering, plan, network, "", 1L, designAccount);
         assertTrue(designednetwork == null);
     }
 
@@ -237,9 +235,8 @@
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
-        final Account account = mock(Account.class);
-
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+
+        final Network designednetwork = guru.design(offering, plan, network, "", 1L, designAccount);
         assertTrue(designednetwork == null);
     }
 
