--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -967,13 +967,18 @@
         return newRootDiskOffering;
     }
 
+    private static VirtualMachineTemplate createMockVirtualMachineTemplate(UserData.UserDataOverridePolicy userDataOverridePolicy) {
+        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
+        when(template.getUserDataId()).thenReturn(2L);
+        when(template.getUserDataOverridePolicy()).thenReturn(userDataOverridePolicy);
+        return template;
+    }
+
     @Test (expected = CloudRuntimeException.class)
     public void testUserDataDenyOverride() {
         Long userDataId = 1L;
 
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        when(template.getUserDataId()).thenReturn(2L);
-        when(template.getUserDataOverridePolicy()).thenReturn(UserData.UserDataOverridePolicy.DENYOVERRIDE);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate(UserData.UserDataOverridePolicy.DENYOVERRIDE);
 
         userVmManagerImpl.finalizeUserData(null, userDataId, template);
     }
@@ -983,9 +988,7 @@
         String templateUserData = "testTemplateUserdata";
         Long userDataId = 1L;
 
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        when(template.getUserDataId()).thenReturn(2L);
-        when(template.getUserDataOverridePolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
         UserDataVO apiUserDataVO = Mockito.mock(UserDataVO.class);
         doReturn(apiUserDataVO).when(userDataDao).findById(userDataId);
@@ -1017,9 +1020,7 @@
     public void testUserDataAllowOverrideWithoutAPIuserdata() {
         String templateUserData = "testTemplateUserdata";
 
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        when(template.getUserDataId()).thenReturn(2L);
-        when(template.getUserDataOverridePolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
         UserDataVO templateUserDataVO = Mockito.mock(UserDataVO.class);
         doReturn(templateUserDataVO).when(userDataDao).findById(2L);
         when(templateUserDataVO.getUserData()).thenReturn(templateUserData);
