--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -272,6 +272,8 @@
     private VMTemplateVO templateMock;
     @Mock
     private UnmanagedInstanceTO unmanagedInstanceMock;
+    @Mock
+    private UserVmVO userVmVOMock;
 
     private static final long virtualMachineId = 1L;
 
@@ -584,8 +586,7 @@
         when(virtualMachine.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(virtualMachine.getType()).thenReturn(VirtualMachine.Type.User);
         when(virtualMachine.getState()).thenReturn(VirtualMachine.State.Stopped);
-        UserVmVO userVmVO = mock(UserVmVO.class);
-        when(userVmDao.findById(anyLong())).thenReturn(userVmVO);
+        when(userVmDao.findById(anyLong())).thenReturn(userVmVOMock);
         Mockito.doNothing().when(unmanagedVMsManager).performUnmanageVMInstancePrechecks(any());
         unmanagedVMsManager.unmanageVMInstance(virtualMachineId, null, false);
     }
@@ -596,8 +597,7 @@
         when(virtualMachine.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(virtualMachine.getState()).thenReturn(VirtualMachine.State.Stopped);
         when(virtualMachine.getId()).thenReturn(virtualMachineId);
-        UserVmVO userVmVO = mock(UserVmVO.class);
-        when(userVmDao.findById(anyLong())).thenReturn(userVmVO);
+        when(userVmDao.findById(anyLong())).thenReturn(userVmVOMock);
         when(vmDao.findById(virtualMachineId)).thenReturn(virtualMachine);
         VolumeVO volumeVO = mock(VolumeVO.class);
         long volumeId = 20L;
