--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -351,7 +351,6 @@
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
         TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer createTungstenNetworkLoadbalancerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateTungstenLoadBalancerPoolAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateTungstenLoadBalancerMemberAnswer = MockTungstenAnswerFactory.get(true);
@@ -377,7 +376,7 @@
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(ipAddressVO.getAddress()).thenReturn(ip);
         when(lbVmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(loadBalancerVMMapVOList);
-        when(tungstenGuestNetworkIpAddressVO.getGuestIpAddress()).thenReturn(ip);
+        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = MockTungstenGuestNetworkIpAddressVOWithGuestIp.createMockTungstenGuestNetworkIpAddressVO(ip);
         when(ip.addr()).thenReturn("10.10.10.10");
         when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(), anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkLoadbalancerCommand.class), anyLong())).thenReturn(createTungstenNetworkLoadbalancerAnswer);
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -651,8 +651,8 @@
         FirewallRuleVO firewallRuleVO = mock(FirewallRuleVO.class);
         LoadBalancerCertMapVO loadBalancerCertMapVO = mock(LoadBalancerCertMapVO.class);
         SslCertVO sslCertVO = mock(SslCertVO.class);
-        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         Ip ip = mock(Ip.class);
+        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = MockTungstenGuestNetworkIpAddressVOWithGuestIp.createMockTungstenGuestNetworkIpAddressVO(ip);
         AccountVO accountVO = mock(AccountVO.class);
         Loadbalancer loadbalancer = mock(Loadbalancer.class);
 
@@ -675,7 +675,6 @@
         when(ip.addr()).thenReturn("192.168.100.100");
         when(getTungstenLoadBalancerAnswer.getApiObjectBase()).thenReturn(loadbalancer);
         when(ipAddressVO.getAddress()).thenReturn(ip);
-        when(tungstenGuestNetworkIpAddressVO.getGuestIpAddress()).thenReturn(ip);
 
         assertTrue(tungstenService.updateLoadBalancer(network, loadBalancingRule));
     }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/MockTungstenGuestNetworkIpAddressVOWithGuestIp.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/MockTungstenGuestNetworkIpAddressVOWithGuestIp.java
@@ -0,0 +1,15 @@
+package org.apache.cloudstack.network.tungsten.service;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.cloud.network.TungstenGuestNetworkIpAddressVO;
+import com.cloud.utils.net.Ip;
+
+public class MockTungstenGuestNetworkIpAddressVOWithGuestIp {
+    public static TungstenGuestNetworkIpAddressVO createMockTungstenGuestNetworkIpAddressVO(Ip guestIpAddress) {
+        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
+        when(tungstenGuestNetworkIpAddressVO.getGuestIpAddress()).thenReturn(guestIpAddress);
+        return tungstenGuestNetworkIpAddressVO;
+    }
+}
