--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpRequestWrapperTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpRequestWrapperTest.java
@@ -78,7 +78,7 @@
 
     @Test
     public void testCreateLogicalSwitchCommandWrapper() {
-        final NiciraNvpApi niciraNvpApi = Mockito.mock(NiciraNvpApi.class);
+        final NiciraNvpApi niciraNvpApi = MockNiciraNvpApi.createMockNiciraNvpApi();
         final NiciraNvpUtilities niciraNvpUtilities = Mockito.mock(NiciraNvpUtilities.class);
         final LogicalSwitch logicalSwitch = Mockito.mock(LogicalSwitch.class);
 
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
@@ -87,7 +87,7 @@
 import com.cloud.network.utils.CommandRetryUtility;
 
 public class NiciraNvpResourceTest {
-    NiciraNvpApi nvpApi = mock(NiciraNvpApi.class);
+    NiciraNvpApi nvpApi = MockNiciraNvpApi.createMockNiciraNvpApi();
     NiciraNvpResource resource;
     Map<String, Object> parameters;
 
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/MockNiciraNvpApi.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/MockNiciraNvpApi.java
@@ -0,0 +1,13 @@
+package com.cloud.network.resource;
+
+import com.cloud.network.nicira.NiciraNvpApi;
+import org.mockito.Mockito;
+
+public final class MockNiciraNvpApi {
+    private MockNiciraNvpApi() {
+    }
+
+    public static NiciraNvpApi createMockNiciraNvpApi() {
+        return Mockito.mock(NiciraNvpApi.class);
+    }
+}
