--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -203,6 +203,12 @@
         return prefix;
     }
 
+    private static DataCenterVO createMockDataCenter() {
+        DataCenterVO zoneMock = Mockito.mock(DataCenterVO.class);
+        Mockito.when(zoneMock.getUuid()).thenReturn("uuid");
+        return zoneMock;
+    }
+
     @Test
     public void testGetUsedTotalIpv6SubnetForPrefix() {
         DataCenterGuestIpv6PrefixVO prefix = prepareMocksForIpv6Subnet();
@@ -419,9 +425,8 @@
         Nic nic = Mockito.mock(Nic.class);
         Mockito.when(nic.getIPv6Address()).thenReturn(null);
         Mockito.when(nic.getBroadcastUri()).thenReturn(URI.create(vlan));
-        DataCenterVO zoneMock = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneMock = createMockDataCenter();
         Mockito.when(zoneDao.findById(Mockito.anyLong())).thenReturn(zoneMock);
-        Mockito.when(zoneMock.getUuid()).thenReturn("uuid");
         try (TransactionLegacy txn = TransactionLegacy.open("testNewErrorAssignPublicIpv6ToNetwork")) {
            ipv6Service.assignPublicIpv6ToNetwork(Mockito.mock(Network.class), nic);
         }
@@ -625,9 +630,8 @@
         Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
         Mockito.when(vlanDao.findById(Mockito.anyLong())).thenReturn(vlanVO);
         Mockito.when(vlanDao.listIpv6RangeByZoneIdAndVlanId(Mockito.anyLong(), Mockito.anyString())).thenReturn(new ArrayList<>());
-        DataCenterVO zoneMock = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneMock = createMockDataCenter();
         Mockito.when(zoneDao.findById(zoneId)).thenReturn(zoneMock);
-        Mockito.when(zoneMock.getUuid()).thenReturn("uuid");
         try {
             ipv6Service.checkNetworkIpv6Upgrade(network);
             Assert.fail("No InsufficientAddressCapacityException");
