--- a/server/src/test/java/com/cloud/capacity/CapacityManagerTest.java
+++ b/server/src/test/java/com/cloud/capacity/CapacityManagerTest.java
@@ -70,11 +70,18 @@
         when(svo.getRamSize()).thenReturn(512);
         when(cvoCpu.getReservedCapacity()).thenReturn(0l);
         when(cvoRam.getReservedCapacity()).thenReturn(0l);
+        clusterDetailRam = createMockClusterDetailsVO();
         when(clusterDetailRam.getValue()).thenReturn("1.5");
-        when(clusterDetailCpu.getValue()).thenReturn("2");
+        clusterDetailCpu = createMockClusterDetailsVO();
         when(CDao.update(anyLong(), isA(CapacityVO.class))).thenReturn(true);
         boolean hasCapacity = capMgr.checkIfHostHasCapacity(host, 500, 1024 * 1024 * 1024, false, 2, 2, false);
         Assert.assertTrue(hasCapacity);
 
     }
+
+    private static ClusterDetailsVO createMockClusterDetailsVO() {
+        ClusterDetailsVO clusterDetailCpu = mock(ClusterDetailsVO.class);
+        when(clusterDetailCpu.getValue()).thenReturn("2");
+        return clusterDetailCpu;
+    }
 }
