--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -216,6 +216,8 @@
     @Mock
     VolumeVO volumeVO;
     @Mock
+    private VolumeVO volumeVOMock;
+    @Mock
     StoragePoolVO storagePoolVO;
     @Mock
     DataCenterGuestIpv6PrefixDao dataCenterGuestIpv6PrefixDao;
@@ -767,9 +769,8 @@
         Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         ArrayList<VolumeVO> arrayList = new ArrayList<VolumeVO>();
-        arrayList.add(volumeVO);
+        arrayList.add(volumeVOMock);
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(0);
         Mockito.when(_volumeDao.findByPod(anyLong())).thenReturn(arrayList);
         Mockito.when(_hostDao.findByPodId(anyLong())).thenReturn(new ArrayList<HostVO>());
@@ -927,9 +928,8 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfZoneIsDeletableFailureOnVolumeTest() {
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         ArrayList<VolumeVO> arrayList = new ArrayList<VolumeVO>();
-        arrayList.add(volumeVO);
+        arrayList.add(volumeVOMock);
 
         Mockito.when(_hostDao.listEnabledIdsByDataCenterId(anyLong())).thenReturn(new ArrayList<>());
         Mockito.when(_podDao.listByDataCenterId(anyLong())).thenReturn(new ArrayList<HostPodVO>());
