--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorkerTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorkerTest.java
@@ -136,20 +136,11 @@
         KubernetesClusterScaleWorker worker = new KubernetesClusterScaleWorker(kubernetesCluster, new java.util.HashMap<>(), 2L, null, false, null, null, clusterManager);
         KubernetesClusterScaleWorker spyWorker = Mockito.spy(worker);
 
-        KubernetesClusterVmMapVO vm1 = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(vm1.isExternalNode()).thenReturn(false);
-        Mockito.when(vm1.isControlNode()).thenReturn(false);
-        Mockito.when(vm1.isEtcdNode()).thenReturn(false);
+        KubernetesClusterVmMapVO vm1 = createMockKubernetesClusterVmMapVO();
 
-        KubernetesClusterVmMapVO vm2 = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(vm2.isExternalNode()).thenReturn(false);
-        Mockito.when(vm2.isControlNode()).thenReturn(false);
-        Mockito.when(vm2.isEtcdNode()).thenReturn(false);
+        KubernetesClusterVmMapVO vm2 = createMockKubernetesClusterVmMapVO();
 
-        KubernetesClusterVmMapVO vm3 = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(vm3.isExternalNode()).thenReturn(false);
-        Mockito.when(vm3.isControlNode()).thenReturn(false);
-        Mockito.when(vm3.isEtcdNode()).thenReturn(false);
+        KubernetesClusterVmMapVO vm3 = createMockKubernetesClusterVmMapVO();
 
         Mockito.doReturn(Arrays.asList(vm1, vm2, vm3)).when(spyWorker).getKubernetesClusterVMMaps();
 
@@ -166,20 +157,11 @@
         KubernetesClusterScaleWorker worker = new KubernetesClusterScaleWorker(kubernetesCluster, new java.util.HashMap<>(), 3L, null, false, null, null, clusterManager);
         KubernetesClusterScaleWorker spyWorker = Mockito.spy(worker);
 
-        KubernetesClusterVmMapVO vm1 = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(vm1.isExternalNode()).thenReturn(false);
-        Mockito.when(vm1.isControlNode()).thenReturn(false);
-        Mockito.when(vm1.isEtcdNode()).thenReturn(false);
+        KubernetesClusterVmMapVO vm1 = createMockKubernetesClusterVmMapVO();
 
-        KubernetesClusterVmMapVO vm2 = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(vm2.isExternalNode()).thenReturn(false);
-        Mockito.when(vm2.isControlNode()).thenReturn(false);
-        Mockito.when(vm2.isEtcdNode()).thenReturn(false);
+        KubernetesClusterVmMapVO vm2 = createMockKubernetesClusterVmMapVO();
 
-        KubernetesClusterVmMapVO vm3 = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(vm3.isExternalNode()).thenReturn(false);
-        Mockito.when(vm3.isControlNode()).thenReturn(false);
-        Mockito.when(vm3.isEtcdNode()).thenReturn(false);
+        KubernetesClusterVmMapVO vm3 = createMockKubernetesClusterVmMapVO();
 
         Mockito.doReturn(Arrays.asList(vm1, vm2, vm3)).when(spyWorker).getKubernetesClusterVMMaps();
 
@@ -187,4 +169,12 @@
 
         Assert.assertTrue(toRemove.isEmpty());
     }
+
+    private static KubernetesClusterVmMapVO createMockKubernetesClusterVmMapVO() {
+        KubernetesClusterVmMapVO vm = Mockito.mock(KubernetesClusterVmMapVO.class);
+        Mockito.when(vm.isExternalNode()).thenReturn(false);
+        Mockito.when(vm.isControlNode()).thenReturn(false);
+        Mockito.when(vm.isEtcdNode()).thenReturn(false);
+        return vm;
+    }
 }
