--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
@@ -118,6 +118,9 @@
     @Mock
     private HostDao hostDao;
 
+    @Mock
+    private ServiceOfferingVO workerOfferingMock;
+
     @Spy
     @InjectMocks
     KubernetesClusterManagerImpl kubernetesClusterManager;
@@ -411,12 +414,11 @@
     public void testCreateNodeTypeToServiceOfferingMapNullMap() {
         KubernetesClusterVO clusterVO = Mockito.mock(KubernetesClusterVO.class);
         Mockito.when(clusterVO.getServiceOfferingId()).thenReturn(1L);
-        ServiceOfferingVO offering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOfferingDao.findById(1L)).thenReturn(offering);
+        Mockito.when(serviceOfferingDao.findById(1L)).thenReturn(workerOfferingMock);
         Map<String, ServiceOffering> mapping = kubernetesClusterManager.createNodeTypeToServiceOfferingMap(new HashMap<>(), null, clusterVO);
         Assert.assertFalse(MapUtils.isEmpty(mapping));
         Assert.assertTrue(mapping.containsKey(DEFAULT.name()));
-        Assert.assertEquals(offering, mapping.get(DEFAULT.name()));
+        Assert.assertEquals(workerOfferingMock, mapping.get(DEFAULT.name()));
     }
 
     @Test
@@ -427,15 +429,14 @@
         idsMap.put(WORKER.name(), workerOfferingId);
         idsMap.put(CONTROL.name(), controlOfferingId);
 
-        ServiceOfferingVO workerOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOfferingDao.findById(workerOfferingId)).thenReturn(workerOffering);
+        Mockito.when(serviceOfferingDao.findById(workerOfferingId)).thenReturn(workerOfferingMock);
         ServiceOfferingVO controlOffering = Mockito.mock(ServiceOfferingVO.class);
         Mockito.when(serviceOfferingDao.findById(controlOfferingId)).thenReturn(controlOffering);
 
         Map<String, ServiceOffering> mapping = kubernetesClusterManager.createNodeTypeToServiceOfferingMap(idsMap, null, null);
         Assert.assertEquals(2, mapping.size());
         Assert.assertTrue(mapping.containsKey(WORKER.name()) && mapping.containsKey(CONTROL.name()));
-        Assert.assertEquals(workerOffering, mapping.get(WORKER.name()));
+        Assert.assertEquals(workerOfferingMock, mapping.get(WORKER.name()));
         Assert.assertEquals(controlOffering, mapping.get(CONTROL.name()));
     }
 
