--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtDiskOnlyVMSnapshotCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtDiskOnlyVMSnapshotCommandWrapperTest.java
@@ -485,12 +485,8 @@
         LibvirtUtilitiesHelper libvirtUtilitiesHelper = mock(LibvirtUtilitiesHelper.class);
         org.libvirt.Connect connect = mock(org.libvirt.Connect.class);
         Domain domain = mock(Domain.class);
-        CreateDiskOnlyVmSnapshotCommand command = mock(CreateDiskOnlyVmSnapshotCommand.class);
-        VMSnapshotTO target = mock(VMSnapshotTO.class);
-
-        when(command.getVmName()).thenReturn("vm-name");
-        when(command.getVolumeTOs()).thenReturn(List.of());
-        when(command.getTarget()).thenReturn(target);
+        VMSnapshotTO target = mock(VMSnapshotTO.class);
+        CreateDiskOnlyVmSnapshotCommand command = createMockCreateDiskOnlyVmSnapshotCommand(target);
         when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtUtilitiesHelper.getConnection()).thenReturn(connect);
         when(resource.getDisks(connect, "vm-name")).thenReturn(List.of());
@@ -558,12 +554,8 @@
         LibvirtUtilitiesHelper libvirtUtilitiesHelper = mock(LibvirtUtilitiesHelper.class);
         org.libvirt.Connect connect = mock(org.libvirt.Connect.class);
         Domain domain = mock(Domain.class);
-        CreateDiskOnlyVmSnapshotCommand command = mock(CreateDiskOnlyVmSnapshotCommand.class);
-        VMSnapshotTO target = mock(VMSnapshotTO.class);
-
-        when(command.getVmName()).thenReturn("vm-name");
-        when(command.getVolumeTOs()).thenReturn(List.of());
-        when(command.getTarget()).thenReturn(target);
+        VMSnapshotTO target = mock(VMSnapshotTO.class);
+        CreateDiskOnlyVmSnapshotCommand command = createMockCreateDiskOnlyVmSnapshotCommand(target);
         when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtUtilitiesHelper.getConnection()).thenReturn(connect);
         when(resource.getDisks(connect, "vm-name")).thenReturn(List.of());
@@ -628,12 +620,8 @@
         LibvirtUtilitiesHelper libvirtUtilitiesHelper = mock(LibvirtUtilitiesHelper.class);
         org.libvirt.Connect connect = mock(org.libvirt.Connect.class);
         Domain domain = mock(Domain.class);
-        CreateDiskOnlyVmSnapshotCommand command = mock(CreateDiskOnlyVmSnapshotCommand.class);
-        VMSnapshotTO target = mock(VMSnapshotTO.class);
-
-        when(command.getVmName()).thenReturn("vm-name");
-        when(command.getVolumeTOs()).thenReturn(List.of());
-        when(command.getTarget()).thenReturn(target);
+        VMSnapshotTO target = mock(VMSnapshotTO.class);
+        CreateDiskOnlyVmSnapshotCommand command = createMockCreateDiskOnlyVmSnapshotCommand(target);
         when(command.isUefiEnabled()).thenReturn(true);
         when(target.getQuiescevm()).thenReturn(true);
         when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
@@ -695,12 +683,8 @@
         LibvirtUtilitiesHelper libvirtUtilitiesHelper = mock(LibvirtUtilitiesHelper.class);
         org.libvirt.Connect connect = mock(org.libvirt.Connect.class);
         Domain domain = mock(Domain.class);
-        CreateDiskOnlyVmSnapshotCommand command = mock(CreateDiskOnlyVmSnapshotCommand.class);
-        VMSnapshotTO target = mock(VMSnapshotTO.class);
-
-        when(command.getVmName()).thenReturn("vm-name");
-        when(command.getVolumeTOs()).thenReturn(List.of());
-        when(command.getTarget()).thenReturn(target);
+        VMSnapshotTO target = mock(VMSnapshotTO.class);
+        CreateDiskOnlyVmSnapshotCommand command = createMockCreateDiskOnlyVmSnapshotCommand(target);
         when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtUtilitiesHelper.getConnection()).thenReturn(connect);
         when(resource.getDisks(connect, "vm-name")).thenReturn(List.of());
@@ -756,12 +740,8 @@
         LibvirtUtilitiesHelper libvirtUtilitiesHelper = mock(LibvirtUtilitiesHelper.class);
         org.libvirt.Connect connect = mock(org.libvirt.Connect.class);
         Domain domain = mock(Domain.class);
-        CreateDiskOnlyVmSnapshotCommand command = mock(CreateDiskOnlyVmSnapshotCommand.class);
-        VMSnapshotTO target = mock(VMSnapshotTO.class);
-
-        when(command.getVmName()).thenReturn("vm-name");
-        when(command.getVolumeTOs()).thenReturn(List.of());
-        when(command.getTarget()).thenReturn(target);
+        VMSnapshotTO target = mock(VMSnapshotTO.class);
+        CreateDiskOnlyVmSnapshotCommand command = createMockCreateDiskOnlyVmSnapshotCommand(target);
         when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtUtilitiesHelper.getConnection()).thenReturn(connect);
         when(resource.getDisks(connect, "vm-name")).thenReturn(List.of());
@@ -802,20 +782,24 @@
         LibvirtUtilitiesHelper libvirtUtilitiesHelper = mock(LibvirtUtilitiesHelper.class);
         org.libvirt.Connect connect = mock(org.libvirt.Connect.class);
         Domain domain = mock(Domain.class);
+        VMSnapshotTO target = mock(VMSnapshotTO.class);
+        CreateDiskOnlyVmSnapshotCommand command = createMockCreateDiskOnlyVmSnapshotCommand(target);
+        when(command.isUefiEnabled()).thenReturn(true);
+        when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
+        when(libvirtUtilitiesHelper.getConnection()).thenReturn(connect);
+        when(resource.getDisks(connect, "vm-name")).thenReturn(List.of());
+        when(resource.getDomain(connect, "vm-name")).thenReturn(domain);
+
+        CreateDiskOnlyVmSnapshotAnswer answer = (CreateDiskOnlyVmSnapshotAnswer) wrapper.takeDiskOnlyVmSnapshotOfRunningVm(command, resource);
+
+        assertFalse(answer.getResult());
+    }
+
+    private static CreateDiskOnlyVmSnapshotCommand createMockCreateDiskOnlyVmSnapshotCommand(VMSnapshotTO target) {
         CreateDiskOnlyVmSnapshotCommand command = mock(CreateDiskOnlyVmSnapshotCommand.class);
-        VMSnapshotTO target = mock(VMSnapshotTO.class);
-
         when(command.getVmName()).thenReturn("vm-name");
         when(command.getVolumeTOs()).thenReturn(List.of());
         when(command.getTarget()).thenReturn(target);
-        when(command.isUefiEnabled()).thenReturn(true);
-        when(resource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
-        when(libvirtUtilitiesHelper.getConnection()).thenReturn(connect);
-        when(resource.getDisks(connect, "vm-name")).thenReturn(List.of());
-        when(resource.getDomain(connect, "vm-name")).thenReturn(domain);
-
-        CreateDiskOnlyVmSnapshotAnswer answer = (CreateDiskOnlyVmSnapshotAnswer) wrapper.takeDiskOnlyVmSnapshotOfRunningVm(command, resource);
-
-        assertFalse(answer.getResult());
+        return command;
     }
 }
