--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -814,6 +814,12 @@
         return to;
     }
 
+    private static KVMStoragePool createMockKVMStoragePool(String diskPath) {
+        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
+        when(storagePool.getPhysicalDisk(diskPath)).thenThrow(CloudRuntimeException.class);
+        return storagePool;
+    }
+
     private void verifyGuestResourceDef(GuestResourceDef guestResourceDef, VirtualMachineTO to) {
         String xml = "<domain>" + guestResourceDef.toString() + "</domain>";
         Document domainDoc = parse(xml);
@@ -4522,7 +4528,7 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool snapshotPool = Mockito.mock(KVMStoragePool.class);
+        final KVMStoragePool snapshotPool = createMockKVMStoragePool(command.getSnapshotName());
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final String tmplName = "ce97bbc1-34fe-4259-9202-74bbce2562ab";
@@ -4538,7 +4544,6 @@
 
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl() + snapshotPath)).thenReturn(snapshotPool);
         when(storagePoolMgr.getStoragePoolByURI(command.getSecondaryStorageUrl())).thenReturn(secondaryPool);
-        when(snapshotPool.getPhysicalDisk(command.getSnapshotName())).thenThrow(CloudRuntimeException.class);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -5049,12 +5054,11 @@
         final ResizeVolumeCommand command = new ResizeVolumeCommand(path, pool, currentSize, newSize, shrinkOk, vmInstance, null);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
-        final KVMStoragePool storagePool = Mockito.mock(KVMStoragePool.class);
 
         when(pool.getType()).thenReturn(StoragePoolType.RBD);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool storagePool = createMockKVMStoragePool(path);
         when(storagePoolMgr.getStoragePool(pool.getType(), pool.getUuid())).thenReturn(storagePool);
-        when(storagePool.getPhysicalDisk(path)).thenThrow(CloudRuntimeException.class);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
         assertNotNull(wrapper);
