--- a/server/src/test/java/com/cloud/ha/HighAvailabilityManagerImplTest.java
+++ b/server/src/test/java/com/cloud/ha/HighAvailabilityManagerImplTest.java
@@ -250,9 +250,7 @@
 
     @Test
     public void scheduleRestartHADisabled() {
-        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vm.getId()).thenReturn(1L);
-        Mockito.when(vm.getDataCenterId()).thenReturn(1L);
+        VMInstanceVO vm = createMockVmInstanceVO();
 
         ConfigKey<Boolean> haEnabled = Mockito.mock(ConfigKey.class);
         highAvailabilityManager.VmHaEnabled = haEnabled;
@@ -290,9 +288,7 @@
 
     @Test
     public void scheduleStopHADisabled() {
-        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vm.getId()).thenReturn(1L);
-        Mockito.when(vm.getDataCenterId()).thenReturn(1L);
+        VMInstanceVO vm = createMockVmInstanceVO();
         Mockito.when(_haDao.hasBeenScheduled(vm.getId(), WorkType.Stop)).thenReturn(false);
 
         ConfigKey<Boolean> haEnabled = Mockito.mock(ConfigKey.class);
@@ -408,6 +404,13 @@
         Mockito.when(highAvailabilityManager.VmHaEnabled.valueIn(1L)).thenReturn(false);
 
         assertTrue(highAvailabilityManager.investigate(1L) == Status.Alert);
+    }
+
+    private static VMInstanceVO createMockVmInstanceVO() {
+        VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
+        Mockito.when(vm.getId()).thenReturn(1L);
+        Mockito.when(vm.getDataCenterId()).thenReturn(1L);
+        return vm;
     }
 
     private void processWorkWithRetryCount(int count, Step expectedStep) {
