--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
@@ -496,21 +496,26 @@
 
     @Test (expected = CloudRuntimeException.class)
     public void getDiskLabelToSnapshotTestDiskPathDoesNotMatch() throws LibvirtException {
+        LibvirtVMDef.DiskDef diskDefMock1 = createMockDiskDef("test");
+
+        storageProcessorSpy.getDiskLabelToSnapshot(List.of(diskDefMock1), "Path", Mockito.mock(Domain.class));
+    }
+
+    @Test
+    public void getDiskLabelToSnapshotTestDiskMatches() throws LibvirtException {
+        LibvirtVMDef.DiskDef diskDefMock1 = createMockDiskDef("Path");
+        Mockito.doReturn("vda").when(diskDefMock1).getDiskLabel();
+
+        String result = storageProcessorSpy.getDiskLabelToSnapshot(List.of(diskDefMock1), "Path", Mockito.mock(Domain.class));
+
+        Assert.assertEquals("vda", result);
+    }
+
+    private static LibvirtVMDef.DiskDef createMockDiskDef(String diskPath) {
         LibvirtVMDef.DiskDef diskDefMock1 = Mockito.mock(LibvirtVMDef.DiskDef.class);
-        Mockito.doReturn("test").when(diskDefMock1).getDiskPath();
-
-        storageProcessorSpy.getDiskLabelToSnapshot(List.of(diskDefMock1), "Path", Mockito.mock(Domain.class));
-    }
-
-    @Test
-    public void getDiskLabelToSnapshotTestDiskMatches() throws LibvirtException {
-        LibvirtVMDef.DiskDef diskDefMock1 = Mockito.mock(LibvirtVMDef.DiskDef.class);
-        Mockito.doReturn("Path").when(diskDefMock1).getDiskPath();
-        Mockito.doReturn("vda").when(diskDefMock1).getDiskLabel();
-
-        String result = storageProcessorSpy.getDiskLabelToSnapshot(List.of(diskDefMock1), "Path", Mockito.mock(Domain.class));
-
-        Assert.assertEquals("vda", result);
+        Mockito.doReturn(diskPath).when(diskDefMock1).getDiskPath();
+
+        return diskDefMock1;
     }
 
     /**
