--- a/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
+++ b/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
@@ -220,7 +220,7 @@
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
-        NicSecondaryIpDao nicSecIpDao =  Mockito.mock(NicSecondaryIpDao.class);
+        NicSecondaryIpDao nicSecIpDao = createMockNicSecondaryIpDao();
         AutoScaleVmGroupDao autoScaleVmGroupDao = Mockito.mock(AutoScaleVmGroupDao.class);
         VpcManager vpcMgr = Mockito.mock(VpcManager.class);
 
@@ -240,7 +240,6 @@
         when(lbDao.findById(anyLong())).thenReturn(lbVO);
         when(userVmDao.findById(anyLong())).thenReturn(vm);
         when(lb2VmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(_lbvmMapList);
-        when (nicSecIpDao.findByIp4AddressAndNicId(anyString(), anyLong())).thenReturn(null);
         when(autoScaleVmGroupDao.isAutoScaleLoadBalancer(anyLong())).thenReturn(Boolean.FALSE);
         when(vpcMgr.isNetworkOnVpcEnabledConserveMode(any())).thenReturn(false);
 
@@ -271,7 +270,7 @@
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
-        NicSecondaryIpDao nicSecIpDao =  Mockito.mock(NicSecondaryIpDao.class);
+        NicSecondaryIpDao nicSecIpDao = createMockNicSecondaryIpDao();
         LoadBalancerVMMapVO lbVmMapVO = new LoadBalancerVMMapVO(1L, 1L, "10.1.1.175", false);
         AutoScaleVmGroupDao autoScaleVmGroupDao = Mockito.mock(AutoScaleVmGroupDao.class);
         VpcManager vpcMgr = Mockito.mock(VpcManager.class);
@@ -293,11 +292,16 @@
         when(lbDao.findById(anyLong())).thenReturn(lbVO);
         when(userVmDao.findById(anyLong())).thenReturn(vm);
         when(lb2VmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(_lbvmMapList);
-        when (nicSecIpDao.findByIp4AddressAndNicId(anyString(), anyLong())).thenReturn(null);
         when(autoScaleVmGroupDao.isAutoScaleLoadBalancer(anyLong())).thenReturn(Boolean.FALSE);
         when(vpcMgr.isNetworkOnVpcEnabledConserveMode(any())).thenReturn(false);
 
         _lbMgr.assignToLoadBalancer(1L, null, vmIdIpMap, null, false);
+    }
+
+    private static NicSecondaryIpDao createMockNicSecondaryIpDao() {
+        NicSecondaryIpDao nicSecIpDao = Mockito.mock(NicSecondaryIpDao.class);
+        when (nicSecIpDao.findByIp4AddressAndNicId(anyString(), anyLong())).thenReturn(null);
+        return nicSecIpDao;
     }
 
     @After
