--- a/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
@@ -59,6 +59,8 @@
     private SearchBuilder<HostVO> mockSearchBuilder;
     @Mock
     private SearchCriteria<HostVO> mockSearchCriteria;
+    @Mock
+    private HostVO mockHost;
 
     @Test
     public void testCountUpAndEnabledHostsInZone() {
@@ -82,9 +84,8 @@
         GenericDaoBase.SumCount mockSumCount = new GenericDaoBase.SumCount();
         mockSumCount.count = 10;
         mockSumCount.sum = 20;
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, GenericDaoBase.SumCount> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        when(sb.entity()).thenReturn(mockHost);
         doReturn(sb).when(hostDao).createSearchBuilder(GenericDaoBase.SumCount.class);
         SearchCriteria<GenericDaoBase.SumCount> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
@@ -99,9 +100,8 @@
     public void testIsHostUp() {
         long testHostId = 101L;
         List<Status> statuses = List.of(Status.Up);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Status> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        when(sb.entity()).thenReturn(mockHost);
         SearchCriteria<Status> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         doReturn(sb).when(hostDao).createSearchBuilder(Status.class);
@@ -121,9 +121,8 @@
         List<Host.Type> types = List.of(Host.Type.Routing);
         List<Hypervisor.HypervisorType> hypervisorTypes = List.of(Hypervisor.HypervisorType.KVM);
         List<Long> mockResults = List.of(1001L, 1002L); // Mocked result
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        when(sb.entity()).thenReturn(mockHost);
         SearchCriteria<Long> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(sb.and()).thenReturn(sb);
@@ -144,9 +143,8 @@
     public void testListDistinctHypervisorTypes() {
         Long zoneId = 1L;
         List<Hypervisor.HypervisorType> mockResults = List.of(Hypervisor.HypervisorType.KVM, Hypervisor.HypervisorType.XenServer);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, String> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        when(sb.entity()).thenReturn(mockHost);
         SearchCriteria<String> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         doReturn(sb).when(hostDao).createSearchBuilder(String.class);
@@ -179,9 +177,8 @@
         Hypervisor.HypervisorType hypervisorType = Hypervisor.HypervisorType.KVM;
         Long zoneId = 1L, podId = 2L, clusterId = 3L;
         List<Long> mockResults = List.of(1001L, 1002L);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        when(sb.entity()).thenReturn(mockHost);
         SearchCriteria<Long> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         doReturn(sb).when(hostDao).createSearchBuilder(Long.class);
@@ -236,9 +233,8 @@
     public void testListDistinctArchTypes() {
         Long clusterId = 1L;
         List<CPU.CPUArch> mockResults = List.of(CPU.CPUArch.amd64, CPU.CPUArch.arm64);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, String> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        when(sb.entity()).thenReturn(mockHost);
         SearchCriteria<String> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         doReturn(sb).when(hostDao).createSearchBuilder(String.class);
