--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -238,6 +238,9 @@
     GuestOSDao guestOSDao;
 
     @Mock
+    private SearchCriteria<GuestOSCategoryVO> guestOSCategorySearchCriteria;
+
+    @Mock
     ExtensionsManager extensionManager;
 
     @Spy
@@ -1023,20 +1026,19 @@
         Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
-        SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
-        Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
+        Mockito.when(searchBuilder.create()).thenReturn(guestOSCategorySearchCriteria);
         Mockito.when(templateDao.listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf)).thenReturn(Arrays.asList(1L, 2L));
         Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(guestOSCategory), 1);
         mockGuestOsJoin();
-        Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(searchCriteria), Mockito.any())).thenReturn(mockResult);
+        Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any())).thenReturn(mockResult);
         Pair<List<? extends GuestOsCategory>, Integer> result = spy.listGuestOSCategoriesByCriteria(listCmd);
         Assert.assertNotNull(result);
         Assert.assertEquals(1, result.second().intValue());
         Assert.assertEquals(1, result.first().size());
         Mockito.verify(guestOSCategoryDao, Mockito.times(1)).createSearchBuilder();
         Mockito.verify(templateDao, Mockito.times(1)).listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf);
-        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(searchCriteria), Mockito.any());
+        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any());
     }
 
     @Test
@@ -1061,12 +1063,11 @@
         Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
-        SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
-        Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
+        Mockito.when(searchBuilder.create()).thenReturn(guestOSCategorySearchCriteria);
         Mockito.when(templateDao.listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf)).thenReturn(Arrays.asList(1L, 2L));
         Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(), 0);
-        Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(searchCriteria), Mockito.any())).thenReturn(mockResult);
+        Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any())).thenReturn(mockResult);
         mockGuestOsJoin();
         Pair<List<? extends GuestOsCategory>, Integer> result = spy.listGuestOSCategoriesByCriteria(listCmd);
         Assert.assertNotNull(result);
@@ -1074,7 +1075,7 @@
         Assert.assertEquals(0, result.first().size());
         Mockito.verify(guestOSCategoryDao, Mockito.times(1)).createSearchBuilder();
         Mockito.verify(templateDao, Mockito.times(1)).listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf);
-        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(searchCriteria), Mockito.any());
+        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any());
     }
 
     @Test
@@ -1099,12 +1100,11 @@
         Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
-        SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
-        Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
+        Mockito.when(searchBuilder.create()).thenReturn(guestOSCategorySearchCriteria);
         Mockito.when(templateDao.listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf)).thenReturn(Arrays.asList(1L, 2L));
         Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(), 0);
-        when(guestOSCategoryDao.searchAndCount(Mockito.eq(searchCriteria), Mockito.any())).thenReturn(mockResult);
+        when(guestOSCategoryDao.searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any())).thenReturn(mockResult);
         mockGuestOsJoin();
         Pair<List<? extends GuestOsCategory>, Integer> result = spy.listGuestOSCategoriesByCriteria(listCmd);
         Assert.assertNotNull(result);
@@ -1112,7 +1112,7 @@
         Assert.assertEquals(0, result.first().size());
         Mockito.verify(guestOSCategoryDao, Mockito.times(1)).createSearchBuilder();
         Mockito.verify(templateDao, Mockito.times(1)).listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf);
-        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(searchCriteria), Mockito.any());
+        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any());
     }
 
     @Test
@@ -1126,19 +1126,18 @@
         Mockito.when(listCmd.isVnf()).thenReturn(null);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
-        SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
-        Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
+        Mockito.when(searchBuilder.create()).thenReturn(guestOSCategorySearchCriteria);
         Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(guestOSCategory), 1);
-        Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(searchCriteria), Mockito.any())).thenReturn(mockResult);
+        Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any())).thenReturn(mockResult);
         mockGuestOsJoin();
         Pair<List<? extends GuestOsCategory>, Integer> result = spy.listGuestOSCategoriesByCriteria(listCmd);
         Assert.assertNotNull(result);
         Assert.assertEquals(1, result.second().intValue());
         Assert.assertEquals(1, result.first().size());
         Mockito.verify(guestOSCategoryDao, Mockito.times(1)).createSearchBuilder();
-        Mockito.verify(searchCriteria, Mockito.times(1)).setParameters("id", id);
-        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(searchCriteria), Mockito.any());
+        Mockito.verify(guestOSCategorySearchCriteria, Mockito.times(1)).setParameters("id", id);
+        Mockito.verify(guestOSCategoryDao, Mockito.times(1)).searchAndCount(Mockito.eq(guestOSCategorySearchCriteria), Mockito.any());
 
     }
 
