--- a/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
@@ -362,8 +362,7 @@
         limits = new ArrayList<>();
         limits.add(Mockito.mock(ResourceLimitVO.class));
         int size = limits.size();
-        AccountVO account = Mockito.mock(AccountVO.class);
-        Mockito.when(account.getId()).thenReturn(1L);
+        AccountVO account = createMockAccountVO();
         Mockito.when(accountDao.findById(1L)).thenReturn(account);
         Mockito.when(accountManager.isRootAdmin(1L)).thenReturn(true);
         resourceLimitManager.addTaggedResourceLimits(limits, List.of(Resource.ResourceType.cpu), hostTags, Resource.ResourceOwnerType.Account, 1L);
@@ -372,8 +371,7 @@
 
     @Test
     public void testFindCorrectResourceLimitForAccount() {
-        AccountVO account = Mockito.mock(AccountVO.class);
-        Mockito.when(account.getId()).thenReturn(1L);
+        AccountVO account = createMockAccountVO();
         Mockito.when(accountManager.isRootAdmin(1L)).thenReturn(true);
         long result = resourceLimitManager.findCorrectResourceLimitForAccount(account, Resource.ResourceType.cpu, hostTags.get(0));
         Assert.assertEquals(Resource.RESOURCE_UNLIMITED, result);
@@ -399,8 +397,7 @@
 
     @Test
     public void testFindCorrectResourceLimitForAccountProjects() {
-            AccountVO account = Mockito.mock(AccountVO.class);
-            Mockito.when(account.getId()).thenReturn(1L);
+            AccountVO account = createMockAccountVO();
             Mockito.when(accountManager.isRootAdmin(1L)).thenReturn(true);
 
             long result = resourceLimitManager.findCorrectResourceLimitForAccount(account,
@@ -552,8 +549,7 @@
 
     @Test
     public void testCheckResourceLimitWithTag() {
-        AccountVO account = Mockito.mock(AccountVO.class);
-        Mockito.when(account.getId()).thenReturn(1L);
+        AccountVO account = createMockAccountVO();
         Mockito.when(accountManager.isRootAdmin(1L)).thenReturn(true);
         try {
             resourceLimitManager.checkResourceLimitWithTag(account, Resource.ResourceType.cpu, hostTags.get(0), 1);
@@ -564,8 +560,7 @@
 
     @Test
     public void testCheckResourceLimitWithTagNonAdmin() throws ResourceAllocationException {
-        AccountVO account = Mockito.mock(AccountVO.class);
-        Mockito.when(account.getId()).thenReturn(1L);
+        AccountVO account = createMockAccountVO();
         Mockito.when(account.getDomainId()).thenReturn(1L);
         Mockito.when(accountManager.isRootAdmin(1L)).thenReturn(false);
         Mockito.doReturn(new ArrayList<ResourceLimitVO>()).when(resourceLimitManager).lockAccountAndOwnerDomainRows(Mockito.anyLong(), Mockito.any(Resource.ResourceType.class), Mockito.anyString());
@@ -580,8 +575,7 @@
 
     @Test
     public void testCheckResourceLimitWithTagProject() throws ResourceAllocationException {
-        AccountVO account = Mockito.mock(AccountVO.class);
-        Mockito.when(account.getId()).thenReturn(1L);
+        AccountVO account = createMockAccountVO();
         Mockito.when(account.getType()).thenReturn(Account.Type.PROJECT);
         Mockito.when(accountManager.isRootAdmin(1L)).thenReturn(false);
         ProjectVO projectVO = Mockito.mock(ProjectVO.class);
@@ -962,6 +956,12 @@
                         Mockito.anyList(), Mockito.eq(tag));
     }
 
+    private static AccountVO createMockAccountVO() {
+        AccountVO account = Mockito.mock(AccountVO.class);
+        Mockito.when(account.getId()).thenReturn(1L);
+        return account;
+    }
+
     private void mockIncrementResourceCountWithTag() {
         Mockito.doNothing().when(resourceLimitManager).incrementResourceCountWithTag(
                 Mockito.anyLong(), Mockito.any(Resource.ResourceType.class), Mockito.anyString());
