--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer56WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer56WrapperTest.java
@@ -57,14 +57,15 @@
     @Mock
     private XenServer56Resource xenServer56Resource;
 
+    private final Host mockedHost = Mockito.mock(Host.class);
+
     final static long[] vpcStats = { 1000L, 2000L };
     final static long[] networkStats = { 3000L, 4000L };
     final static long[] lbStats = { 5L };
 
     @Test
     public void testCheckOnHostCommand() {
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
-        final CheckOnHostCommand onHostCommand = new CheckOnHostCommand(host);
+        final CheckOnHostCommand onHostCommand = new CheckOnHostCommand(mockedHost);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -77,11 +78,10 @@
     @Test
     public void testFenceCommand() {
         final VMInstanceVO vm = Mockito.mock(VMInstanceVO.class);
-        final Host host = Mockito.mock(Host.class);
-
-        final Connection conn = Mockito.mock(Connection.class);
-
-        final FenceCommand fenceCommand = new FenceCommand(vm, host);
+
+        final Connection conn = Mockito.mock(Connection.class);
+
+        final FenceCommand fenceCommand = new FenceCommand(vm, mockedHost);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
