--- a/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
@@ -172,13 +172,11 @@
 
         Commands commands = mock(Commands.class);
         when(commands.size()).thenReturn(3);
-        Answer answer1 = mock(Answer.class);
-        Answer answer2 = mock(Answer.class);
+        Answer answer1 = createMockAnswer(true);
+        Answer answer2 = createMockAnswer(false);
         Answer answer3 = mock(Answer.class);
         // In the second iteration it should match and return, without invoking the third
         Answer[] answers = {answer1, answer2, answer3};
-        when(answer1.getResult()).thenReturn(true);
-        when(answer2.getResult()).thenReturn(false);
         lenient().when(answer3.getResult()).thenReturn(false);
         when(this.agentManager.send(HOST_ID, commands)).thenReturn(answers);
 
@@ -210,14 +208,11 @@
 
         Commands commands = mock(Commands.class);
         when(commands.size()).thenReturn(3);
-        Answer answer1 = mock(Answer.class);
-        Answer answer2 = mock(Answer.class);
-        Answer answer3 = mock(Answer.class);
+        Answer answer1 = createMockAnswer(true);
+        Answer answer2 = createMockAnswer(true);
+        Answer answer3 = createMockAnswer(true);
         // In the second iteration it should match and return, without invoking the third
         Answer[] answers = {answer1, answer2, answer3};
-        when(answer1.getResult()).thenReturn(true);
-        when(answer2.getResult()).thenReturn(true);
-        when(answer3.getResult()).thenReturn(true);
         when(this.agentManager.send(HOST_ID, commands)).thenReturn(answers);
 
         // Execute
@@ -313,6 +308,12 @@
     }
 
 
+    private static Answer createMockAnswer(boolean result) {
+        Answer answer = mock(Answer.class);
+        when(answer.getResult()).thenReturn(result);
+        return answer;
+    }
+
     private NicProfile getExpectedNicProfile(boolean vxlan, String vlanTag) {
         NicProfile nic = new NicProfile();
         nic.setDefaultNic(true);
