--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -226,7 +226,7 @@
     public void testRevertDiskSnapshot() throws Exception {
         VMSnapshot vmSnapshot = Mockito.mock(VMSnapshot.class);
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
-        SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
+        SnapshotVO snapshotVO = createMockSnapshotVO();
         Snapshot snap = Mockito.mock(Snapshot.class);
         DataStore dataStore = Mockito.mock(DataStore.class);
 
@@ -236,7 +236,6 @@
         when(vol.getUuid()).thenReturn(volUuid);
         when(vmSnapshot.getUuid()).thenReturn(vmUuid);
         when(vol.getDataStore()).thenReturn(dataStore);
-        when(snapshotVO.getId()).thenReturn(1L);
         when(_snapshotService.revertSnapshot(snapshotVO.getId())).thenReturn(snap);
         vmStrategy.revertDiskSnapshot(vmSnapshot);
     }
@@ -245,7 +244,6 @@
     public void testDeleteDiskSnapshot() {
         VMSnapshot vmSnapshot = Mockito.mock(VMSnapshot.class);
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
-        SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
         Snapshot snap = Mockito.mock(Snapshot.class);
         SnapshotInfo info = Mockito.mock(SnapshotInfo.class);
         SnapshotStrategy strategy = Mockito.mock(SnapshotStrategy.class);
@@ -254,7 +252,7 @@
         String name = vmUuid + "_" + volUuid;
         when(vol.getUuid()).thenReturn(volUuid);
         when(vmSnapshot.getUuid()).thenReturn(vmUuid);
-        when(snapshotVO.getId()).thenReturn(1L);
+        SnapshotVO snapshotVO = createMockSnapshotVO();
         when( _snapshotDataFactory.getSnapshot(snapshotVO.getId(), vol.getDataStore())).thenReturn(info);
         when(_storageStrategyFactory.getSnapshotStrategy(info, SnapshotOperation.DELETE)).thenReturn(strategy);
         testFindSnapshotByName(name);
@@ -299,6 +297,12 @@
         assertTrue(result);
     }
 
+    private static SnapshotVO createMockSnapshotVO() {
+        SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
+        when(snapshotVO.getId()).thenReturn(1L);
+        return snapshotVO;
+    }
+
     @SuppressWarnings("unchecked")
     private SnapshotVO testFindSnapshotByName(String snapshotName) {
         SearchBuilder<SnapshotVO> sb = Mockito.mock(SearchBuilder.class);
