--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1847,9 +1847,9 @@
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO1 = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO1 = Mockito.mock(CounterTO.class);
+        CounterTO counterTO1 = createMockCounterTO(counterName + "-1");
         ConditionTO conditionTO2 = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO2 = Mockito.mock(CounterTO.class);
+        CounterTO counterTO2 = createMockCounterTO(counterName + "-2");
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO1, conditionTO2));
@@ -1858,8 +1858,6 @@
         when(conditionTO2.getCounter()).thenReturn(counterTO2);
         when(conditionTO1.getId()).thenReturn(conditionId);
         when(conditionTO2.getId()).thenReturn(conditionId + 1L);
-        when(counterTO1.getName()).thenReturn(counterName + "-1");
-        when(counterTO2.getName()).thenReturn(counterName + "-2");
 
         Map<String, String> params = new LinkedHashMap<>();
 
@@ -2512,6 +2510,12 @@
                 "-" + asVmGroupMock.getNextVmSeq() + "-[a-z]{6}";
         Assert.assertTrue(result.first().matches(vmHostNamePattern));
         Assert.assertEquals(result.first(), result.second());
+    }
+
+    private static CounterTO createMockCounterTO(String counterName) {
+        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        when(counterTO.getName()).thenReturn(counterName);
+        return counterTO;
     }
 
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
