--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static Cluster createMockCluster() {
+        Cluster cluster = mock(Cluster.class);
+        when(cluster.getId()).thenReturn(1L);
+        return cluster;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1174,11 +1180,10 @@
         when(_stateMachine.transitTo(vmInstance, VirtualMachine.Event.OperationFailed, new Pair(vmInstance.getHostId(), null), vmInstanceDaoMock)).thenReturn(true);
 
 
-        Cluster cluster = mock(Cluster.class);
+        Cluster cluster = createMockCluster();
         when(dest.getCluster()).thenReturn(cluster);
         ClusterDetailsVO cluster_detail_cpu = mock(ClusterDetailsVO.class);
         ClusterDetailsVO cluster_detail_ram = mock(ClusterDetailsVO.class);
-        when(cluster.getId()).thenReturn(1L);
         when(_clusterDetailsDao.findDetail(1L, VmDetailConstants.CPU_OVER_COMMIT_RATIO)).thenReturn(cluster_detail_cpu);
         when(_clusterDetailsDao.findDetail(1L, VmDetailConstants.MEMORY_OVER_COMMIT_RATIO)).thenReturn(cluster_detail_ram);
         when(vmInstanceDetailsDao.findDetail(anyLong(), Mockito.anyString())).thenReturn(null);
@@ -1270,11 +1275,10 @@
         when(_stateMachine.transitTo(vmInstance, VirtualMachine.Event.OperationFailed, new Pair(vmInstance.getHostId(), null), vmInstanceDaoMock)).thenReturn(true);
 
 
-        Cluster cluster = mock(Cluster.class);
+        Cluster cluster = createMockCluster();
         when(dest.getCluster()).thenReturn(cluster);
         ClusterDetailsVO cluster_detail_cpu = mock(ClusterDetailsVO.class);
         ClusterDetailsVO cluster_detail_ram = mock(ClusterDetailsVO.class);
-        when(cluster.getId()).thenReturn(1L);
         when(_clusterDetailsDao.findDetail(1L, VmDetailConstants.CPU_OVER_COMMIT_RATIO)).thenReturn(cluster_detail_cpu);
         when(_clusterDetailsDao.findDetail(1L, VmDetailConstants.MEMORY_OVER_COMMIT_RATIO)).thenReturn(cluster_detail_ram);
         when(vmInstanceDetailsDao.findDetail(anyLong(), Mockito.anyString())).thenReturn(null);
