--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -266,6 +266,8 @@
     @Mock
     private DeploymentPlanningManager _dpMgr;
     @Mock
+    private DeploymentPlanner deploymentPlannerMock;
+    @Mock
     private HypervisorGuruManager _hvGuruMgr;
     @Mock
     private ClusterDao clusterDao;
@@ -1141,7 +1143,6 @@
 
         Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
 
-        DeploymentPlanner planner = mock(DeploymentPlanner.class);
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
@@ -1193,7 +1194,7 @@
             when(CallContext.current()).thenReturn(callContext);
 
             try {
-                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, planner);
+                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, deploymentPlannerMock);
             } catch (CloudRuntimeException e) {
                 assertEquals(e.getMessage(), "Error while transitioning");
             }
@@ -1237,7 +1238,6 @@
 
         Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
 
-        DeploymentPlanner planner = mock(DeploymentPlanner.class);
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
@@ -1289,7 +1289,7 @@
             when(CallContext.current()).thenReturn(callContext);
 
             try {
-                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, planner);
+                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, deploymentPlannerMock);
             } catch (CloudRuntimeException e) {
                 assertEquals(e.getMessage(), "Error while transitioning");
             }
