--- a/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/service/NetrisProviderServiceImplTest.java
+++ b/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/service/NetrisProviderServiceImplTest.java
@@ -60,6 +60,8 @@
     @Mock
     private PhysicalNetworkDao physicalNetworkDao;
     @Mock
+    private PhysicalNetworkVO physicalNetworkMock;
+    @Mock
     private NetworkDao networkDao;
     @Mock
     private NetrisService netrisService;
@@ -138,8 +140,7 @@
         Mockito.when(providerVO.getZoneId()).thenReturn(ZONE_ID);
         Mockito.when(netrisProviderDao.findById(PROVIDER_ID)).thenReturn(providerVO);
 
-        PhysicalNetworkVO physicalNetwork = Mockito.mock(PhysicalNetworkVO.class);
-        Mockito.when(physicalNetworkDao.listByZone(ZONE_ID)).thenReturn(Arrays.asList(physicalNetwork));
+        Mockito.when(physicalNetworkDao.listByZone(ZONE_ID)).thenReturn(Arrays.asList(physicalNetworkMock));
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
         Mockito.when(networkDao.listByPhysicalNetwork(Mockito.anyLong())).thenReturn(Arrays.asList(network));
@@ -170,8 +171,7 @@
         Mockito.when(providerVO.getZoneId()).thenReturn(ZONE_ID);
         Mockito.when(netrisProviderDao.findById(PROVIDER_ID)).thenReturn(providerVO);
 
-        PhysicalNetworkVO physicalNetwork = Mockito.mock(PhysicalNetworkVO.class);
-        Mockito.when(physicalNetworkDao.listByZone(ZONE_ID)).thenReturn(Arrays.asList(physicalNetwork));
+        Mockito.when(physicalNetworkDao.listByZone(ZONE_ID)).thenReturn(Arrays.asList(physicalNetworkMock));
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
         Mockito.when(networkDao.listByPhysicalNetwork(Mockito.anyLong())).thenReturn(Arrays.asList(network));
