--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1353,6 +1353,13 @@
         });
     }
 
+    private static DataCenterVO createMockEdgeZone() {
+        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
+        Mockito.when(zone.getNetworkType()).thenReturn(NetworkType.Advanced);
+        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Edge);
+        return zone;
+    }
+
     @Test
     public void testCreateEdgeZone() {
         CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
@@ -1407,9 +1414,7 @@
         UpdateZoneCmd cmd = Mockito.mock(UpdateZoneCmd.class);
         Mockito.when(cmd.getId()).thenReturn(zoneId);
         Mockito.when(cmd.getZoneName()).thenReturn("NewName");
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getNetworkType()).thenReturn(NetworkType.Advanced);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Edge);
+        DataCenterVO zone = createMockEdgeZone();
         Mockito.when(zone.getId()).thenReturn(zoneId);
         Mockito.when(_zoneDao.findById(Mockito.anyLong())).thenReturn(zone);
         Mockito.when(_networkModel.getDefaultPhysicalNetworkByZoneAndTrafficType(zoneId, Networks.TrafficType.Public)).thenReturn(null);
@@ -1420,9 +1425,7 @@
     @Test
     public void testEdgeZoneCreatePod() {
         final long zoneId = 1L;
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getNetworkType()).thenReturn(NetworkType.Advanced);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Edge);
+        DataCenterVO zone = createMockEdgeZone();
         Mockito.when(zone.getId()).thenReturn(1L);
         Mockito.when(_zoneDao.findById(Mockito.anyLong())).thenReturn(zone);
         Mockito.when(_configDao.getValue(Config.ControlCidr.key())).thenReturn(Config.ControlCidr.getDefaultValue());
