--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -1586,11 +1586,17 @@
         assertTrue(proxyIdList.stream().anyMatch( p -> p.getUuid().equals(vmId)));
     }
 
+    private static HostVO createMockLastHost() {
+        HostVO lastHost = Mockito.mock(HostVO.class);
+        Mockito.when(lastHost.getId()).thenReturn(2L);
+        return lastHost;
+    }
+
     @Test
     public void testSetVmRequiredFieldsForImportFromLastHost() {
         HostVO lastHost = Mockito.mock(HostVO.class);
-        HostVO host = Mockito.mock(HostVO.class);
         Mockito.when(_dcMock.getId()).thenReturn(1L);
+        HostVO host = createMockLastHost();
         Mockito.when(host.getId()).thenReturn(1L);
         Mockito.when(lastHost.getId()).thenReturn(2L);
         userVmManagerImpl.setVmRequiredFieldsForImport(true, userVmVoMock, _dcMock,
