--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -179,6 +179,8 @@
     TungstenFabricLBHealthMonitorDao tungstenFabricLBHealthMonitorDao;
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
+    @Mock
+    private PhysicalNetworkServiceProvider physicalNetworkServiceProviderMock;
 
     TungstenElement tungstenElement;
 
@@ -745,17 +747,15 @@
 
     @Test
     public void isReadyTest() {
-        PhysicalNetworkServiceProvider physicalNetworkServiceProvider = mock(PhysicalNetworkServiceProvider.class);
         PhysicalNetworkTrafficTypeVO physicalNetworkTrafficTypeVO = mock(PhysicalNetworkTrafficTypeVO.class);
 
         when(physicalNetworkTrafficTypeDao.findBy(anyLong(), eq(Networks.TrafficType.Management))).thenReturn(physicalNetworkTrafficTypeVO);
 
-        assertTrue(tungstenElement.isReady(physicalNetworkServiceProvider));
+        assertTrue(tungstenElement.isReady(physicalNetworkServiceProviderMock));
     }
 
     @Test
     public void shutdownProviderInstancesTest() throws ConcurrentOperationException {
-        PhysicalNetworkServiceProvider physicalNetworkServiceProvider = mock(PhysicalNetworkServiceProvider.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
         PhysicalNetworkVO physicalNetworkVO = mock(PhysicalNetworkVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
@@ -775,7 +775,7 @@
         when(vlanDao.listVlansByNetworkIdIncludingRemoved(anyLong())).thenReturn(vlanVOList);
         when(hostPodDao.listByDataCenterId(anyLong())).thenReturn(hostPodVOList);
 
-        assertTrue(tungstenElement.shutdownProviderInstances(physicalNetworkServiceProvider, reservationContext));
+        assertTrue(tungstenElement.shutdownProviderInstances(physicalNetworkServiceProviderMock, reservationContext));
         verify(networkServiceMapDao, times(1)).deleteByNetworkId(anyLong());
         verify(tungstenService, times(2)).removePublicNetworkSubnet(any(VlanVO.class));
         verify(tungstenService, times(1)).deletePublicNetwork(anyLong());
