--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenProviderServiceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenProviderServiceTest.java
@@ -71,6 +71,8 @@
     HostDetailsDao hostDetailsDao;
     @Mock
     MessageBus messageBus;
+    @Mock
+    private TungstenProviderVO tungstenProviderMock;
 
     TungstenProviderServiceImpl tungstenProviderService;
 
@@ -121,9 +123,8 @@
 
     @Test
     public void listTungstenProviderWithZoneIdTest() {
-        TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
 
-        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
+        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderMock);
 
         List<BaseResponse> baseResponseList = tungstenProviderService.listTungstenProvider(1L);
         assertEquals(1, baseResponseList.size());
@@ -131,10 +132,9 @@
 
     @Test
     public void listTungstenProviderWithoutZoneIdTest() {
-        TungstenProviderVO tungstenProviderVO1 = Mockito.mock(TungstenProviderVO.class);
         TungstenProviderVO tungstenProviderVO2 = Mockito.mock(TungstenProviderVO.class);
 
-        when(tungstenProviderDao.listAll()).thenReturn(Arrays.asList(tungstenProviderVO1, tungstenProviderVO2));
+        when(tungstenProviderDao.listAll()).thenReturn(Arrays.asList(tungstenProviderMock, tungstenProviderVO2));
 
         List<BaseResponse> baseResponseList = tungstenProviderService.listTungstenProvider(null);
         assertEquals(2, baseResponseList.size());
