--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpRequestWrapperTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpRequestWrapperTest.java
@@ -78,9 +78,9 @@
 
     @Test
     public void testCreateLogicalSwitchCommandWrapper() {
-        final NiciraNvpApi niciraNvpApi = Mockito.mock(NiciraNvpApi.class);
         final NiciraNvpUtilities niciraNvpUtilities = Mockito.mock(NiciraNvpUtilities.class);
         final LogicalSwitch logicalSwitch = Mockito.mock(LogicalSwitch.class);
+        final NiciraNvpApi niciraNvpApi = MockNiciraNvpApiFactory.createMockNiciraNvpApi(logicalSwitch);
 
         final String transportUuid = "d2e05a9e-7120-4487-a5fc-414ab36d9345";
         final String transportType = "stt";
@@ -95,13 +95,7 @@
         when(niciraNvpUtilities.createLogicalSwitch()).thenReturn(logicalSwitch);
         when(niciraNvpResource.truncate("lswitch-" + command.getName(), NiciraNvpResource.NAME_MAX_LEN)).thenReturn(truncated);
         when(niciraNvpResource.getNiciraNvpApi()).thenReturn(niciraNvpApi);
-
-        try {
-            when(niciraNvpApi.createLogicalSwitch(logicalSwitch)).thenReturn(logicalSwitch);
-            when(logicalSwitch.getUuid()).thenReturn(transportUuid);
-        } catch (final NiciraNvpApiException e) {
-            fail(e.getMessage());
-        }
+        when(logicalSwitch.getUuid()).thenReturn(transportUuid);
 
         final NiciraNvpRequestWrapper wrapper = NiciraNvpRequestWrapper.getInstance();
         assertNotNull(wrapper);
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
@@ -187,7 +187,7 @@
 
         final LogicalSwitch ls = mock(LogicalSwitch.class);
         when(ls.getUuid()).thenReturn("cccc").thenReturn("cccc");
-        when(nvpApi.createLogicalSwitch((LogicalSwitch) any())).thenReturn(ls);
+        nvpApi = MockNiciraNvpApiFactory.createMockNiciraNvpApi(ls);
 
         final CreateLogicalSwitchCommand clsc = new CreateLogicalSwitchCommand((String) parameters.get("guid"), "stt", "loigicalswitch", "owner");
         final CreateLogicalSwitchAnswer clsa = (CreateLogicalSwitchAnswer) resource.executeRequest(clsc);
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/MockNiciraNvpApiFactory.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/MockNiciraNvpApiFactory.java
@@ -0,0 +1,21 @@
+package com.cloud.network.resource;
+
+import static org.mockito.Mockito.when;
+
+import com.cloud.network.nicira.LogicalSwitch;
+import com.cloud.network.nicira.NiciraNvpApi;
+import com.cloud.network.nicira.NiciraNvpApiException;
+
+import org.mockito.Mockito;
+
+public class MockNiciraNvpApiFactory {
+    public static NiciraNvpApi createMockNiciraNvpApi(final LogicalSwitch logicalSwitch) {
+        final NiciraNvpApi niciraNvpApi = Mockito.mock(NiciraNvpApi.class);
+        try {
+            when(niciraNvpApi.createLogicalSwitch((LogicalSwitch) Mockito.any())).thenReturn(logicalSwitch);
+        } catch (final NiciraNvpApiException e) {
+            throw new RuntimeException(e);
+        }
+        return niciraNvpApi;
+    }
+}
