--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -105,7 +105,7 @@
         final String uuid = "206b21a7-c6ec-40e2-b5e2-f861b9612f04";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
+        final VirtualMachineTO vmSpec = createMockVirtualMachineTO();
 
         final VolumeTO vol1 = Mockito.mock(VolumeTO.class);
         final VolumeTO vol2 = Mockito.mock(VolumeTO.class);
@@ -131,7 +131,6 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(vmSpec.getName()).thenReturn(vmName);
         when(vmSpec.getNics()).thenReturn(nicTOs);
 
 
@@ -178,7 +177,7 @@
         final String uuid = "206b21a7-c6ec-40e2-b5e2-f861b9612f04";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
+        final VirtualMachineTO vmSpec = createMockVirtualMachineTO();
 
         final VolumeTO vol1 = Mockito.mock(VolumeTO.class);
         final VolumeTO vol2 = Mockito.mock(VolumeTO.class);
@@ -208,7 +207,6 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(vmSpec.getName()).thenReturn(vmName);
         when(vmSpec.getNics()).thenReturn(nicTOs);
 
         try {
@@ -257,7 +255,7 @@
         final String mac = "3c:15:c2:c4:4f:18";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
+        final VirtualMachineTO vmSpec = createMockVirtualMachineTO();
 
         final VolumeTO volume1 = MockVolumeTO(path);
         final VolumeTO volume2 = MockVolumeTO(path);
@@ -293,7 +291,6 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(vmSpec.getName()).thenReturn(vmName);
 
         when(nic1.getMac()).thenReturn(mac);
         when(nic2.getMac()).thenReturn(mac);
@@ -331,7 +328,7 @@
         final String vmName = "small";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
+        final VirtualMachineTO vmSpec = createMockVirtualMachineTO();
 
         final VolumeTO volume1 = Mockito.mock(VolumeTO.class);
         final VolumeTO volume2 = Mockito.mock(VolumeTO.class);
@@ -349,7 +346,6 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(vmSpec.getName()).thenReturn(vmName);
 
         final Answer answer = wrapper.execute(migrateStorageCommand, xenServer610Resource);
 
@@ -364,7 +360,7 @@
         final String path = "/";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
+        final VirtualMachineTO vmSpec = createMockVirtualMachineTO();
 
         final VolumeTO volume1 = MockVolumeTO(path);
         final VolumeTO volume2 = MockVolumeTO(path);
@@ -394,7 +390,6 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(vmSpec.getName()).thenReturn(vmName);
 
         when(xenServer610Resource.getVDIbyUuid(conn, volume1.getPath())).thenReturn(vdi1);
         when(xenServer610Resource.getVDIbyUuid(conn, volume2.getPath())).thenReturn(vdi2);
@@ -414,7 +409,7 @@
         final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
 
-        final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
+        final VirtualMachineTO vm = createMockVirtualMachineTO();
 
         final MigrateWithStorageCompleteCommand createStorageCommand = new MigrateWithStorageCompleteCommand(vm);
 
@@ -422,7 +417,6 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(vm.getName()).thenReturn(vmName);
         when(xenServer610Resource.getHost()).thenReturn(xsHost);
         when(xsHost.getUuid()).thenReturn(uuid);
 
@@ -465,6 +459,12 @@
         assertFalse(answer.getResult());
     }
 
+    private static VirtualMachineTO createMockVirtualMachineTO() {
+        final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
+        when(vmSpec.getName()).thenReturn("small");
+        return vmSpec;
+    }
+
     VolumeTO MockVolumeTO(String path){
         VolumeTO vol = Mockito.mock(VolumeTO.class);
         when(vol.getPath()).thenReturn(path);
