--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -97,6 +97,8 @@
     HostDao hostDao;
     @Mock
     ConfigurationDao configDao;
+    @Mock
+    private DataTO copyOfflineSrcDataTO;
 
     static MockedStatic<RemoteHostEndPoint> remoteHostEndPointMock;
 
@@ -515,8 +517,7 @@
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
         DataObject srcData = Mockito.mock(DataObject.class);
-        DataTO srcDataTO = Mockito.mock(DataTO.class);
-        when(srcData.getTO()).thenReturn(srcDataTO);
+        when(srcData.getTO()).thenReturn(copyOfflineSrcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
         DataTO destDataTO = Mockito.mock(DataTO.class);
         when(destData.getTO()).thenReturn(destDataTO);
@@ -538,8 +539,7 @@
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
         DataObject srcData = Mockito.mock(DataObject.class);
-        DataTO srcDataTO = Mockito.mock(DataTO.class);
-        when(srcData.getTO()).thenReturn(srcDataTO);
+        when(srcData.getTO()).thenReturn(copyOfflineSrcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
         DataTO destDataTO = Mockito.mock(DataTO.class);
         when(destData.getTO()).thenReturn(destDataTO);
