--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -203,6 +203,12 @@
         return volumeVO;
     }
 
+    private static VolumeVO createMockVolumeVO() {
+        VolumeVO volumeVOMock = Mockito.mock(VolumeVO.class);
+        Mockito.doReturn(null).when(volumeVOMock).getInstanceId();
+        return volumeVOMock;
+    }
+
     @Test
     public void testIsVolumeSuspectedDestroyDuplicateNoPool() {
         VolumeVO volume = mockVolumeForIsVolumeSuspectedDestroyDuplicateTest();
@@ -595,7 +601,7 @@
     @Test
     public void getStoragePoolNonDestroyedVolumesLogTestNonDestroyedVolumes_VMLogForOneVolume() {
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
-        Mockito.doReturn(null).when(volume1VOMock).getInstanceId();
+        volume1VOMock = createMockVolumeVO();
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
         Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
@@ -613,9 +619,9 @@
     @Test
     public void getStoragePoolNonDestroyedVolumesLogTestNonDestroyedVolumes_NotAttachedLogs() {
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
-        Mockito.doReturn(null).when(volume1VOMock).getInstanceId();
+        volume1VOMock = createMockVolumeVO();
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(null).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO();
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
 
         Mockito.when(_volumeDao.findNonDestroyedVolumesByPoolId(storagePoolVOMock.getId(), null)).thenReturn(List.of(volume1VOMock, volume2VOMock));
