--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -194,6 +194,8 @@
     @Mock
     VMInstanceDao _vmInstanceDao;
     @Mock
+    VMInstanceVO vmInstanceVOMock;
+    @Mock
     ClusterDao _clusterDao;
     @Mock
     HostPodDao _podDao;
@@ -803,9 +805,8 @@
         Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
-        VMInstanceVO vMInstanceVO = Mockito.mock(VMInstanceVO.class);
         ArrayList<VMInstanceVO> arrayList = new ArrayList<VMInstanceVO>();
-        arrayList.add(vMInstanceVO);
+        arrayList.add(vmInstanceVOMock);
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(0);
         Mockito.when(_volumeDao.findByPod(anyLong())).thenReturn(new ArrayList<VolumeVO>());
         Mockito.when(_hostDao.findByPodId(anyLong())).thenReturn(new ArrayList<HostVO>());
@@ -909,9 +910,8 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfZoneIsDeletableFailureOnVmInstanceTest() {
-        VMInstanceVO vMInstanceVO = Mockito.mock(VMInstanceVO.class);
         ArrayList<VMInstanceVO> arrayList = new ArrayList<VMInstanceVO>();
-        arrayList.add(vMInstanceVO);
+        arrayList.add(vmInstanceVOMock);
 
         Mockito.when(_hostDao.listEnabledIdsByDataCenterId(anyLong())).thenReturn(new ArrayList<>());
         Mockito.when(_podDao.listByDataCenterId(anyLong())).thenReturn(new ArrayList<HostPodVO>());
