--- a/api/src/test/java/org/apache/cloudstack/api/command/test/CreateSnapshotCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/CreateSnapshotCmdTest.java
@@ -82,9 +82,8 @@
     @Test
     public void testCreateSuccess() {
 
-        AccountService accountService = Mockito.mock(AccountService.class);
         Account account = Mockito.mock(Account.class);
-        Mockito.when(accountService.getAccount(anyLong())).thenReturn(account);
+        AccountService accountService = createMockAccountService(account);
 
         VolumeApiService volumeApiService = Mockito.mock(VolumeApiService.class);
         Snapshot snapshot = Mockito.mock(Snapshot.class);
@@ -115,9 +114,8 @@
     @Test
     public void testCreateFailure() {
 
-        AccountService accountService = Mockito.mock(AccountService.class);
         Account account = Mockito.mock(Account.class);
-        Mockito.when(accountService.getAccount(anyLong())).thenReturn(account);
+        AccountService accountService = createMockAccountService(account);
         String volumeUuid = UUID.randomUUID().toString();
 
         CallContext.current().putApiResourceUuid("volumeid", volumeUuid);
@@ -170,4 +168,10 @@
         Assert.assertEquals(ids.get(0), cmd.getZoneIds().get(0));
         Assert.assertEquals(ids.get(1), cmd.getZoneIds().get(1));
     }
+
+    private static AccountService createMockAccountService(Account account) {
+        AccountService accountService = Mockito.mock(AccountService.class);
+        Mockito.when(accountService.getAccount(anyLong())).thenReturn(account);
+        return accountService;
+    }
 }
