--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -814,6 +814,12 @@
         return to;
     }
 
+    private static VirtualMachineTO createMockVirtualMachineTO(Type type) {
+        VirtualMachineTO vmTo = Mockito.mock(VirtualMachineTO.class);
+        Mockito.when(vmTo.getType()).thenReturn(type);
+        return vmTo;
+    }
+
     private void verifyGuestResourceDef(GuestResourceDef guestResourceDef, VirtualMachineTO to) {
         String xml = "<domain>" + guestResourceDef.toString() + "</domain>";
         Document domainDoc = parse(xml);
@@ -5725,8 +5731,7 @@
         LibvirtComputingResource libvirtComputingResource = new LibvirtComputingResource();
         libvirtComputingResource.noMemBalloon = false;
 
-        VirtualMachineTO vmTo = Mockito.mock(VirtualMachineTO.class);
-        Mockito.when(vmTo.getType()).thenReturn(Type.User);
+        VirtualMachineTO vmTo = createMockVirtualMachineTO(Type.User);
         long requestedMemory = 1024 * 1024;
         long minMemory = 512 * 1024;
 
@@ -5739,8 +5744,7 @@
         LibvirtComputingResource libvirtComputingResource = new LibvirtComputingResource();
         libvirtComputingResource.noMemBalloon = false;
 
-        VirtualMachineTO vmTo = Mockito.mock(VirtualMachineTO.class);
-        Mockito.when(vmTo.getType()).thenReturn(Type.SecondaryStorageVm);
+        VirtualMachineTO vmTo = createMockVirtualMachineTO(Type.SecondaryStorageVm);
         long requestedMemory = 1024 * 1024;
         long minMemory = 512 * 1024;
 
