--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
@@ -93,6 +93,9 @@
     @Mock
     SnapshotService snapshotService;
 
+    @Mock
+    private Snapshot snapshotMock;
+
     List<SnapshotInfo> mockSnapshotInfos = new ArrayList<>();
 
     @Before
@@ -326,32 +329,28 @@
 
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_NullVolume() {
-        Snapshot snapshot = Mockito.mock(Snapshot.class);
-        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, null));
+        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshotMock, null));
     }
 
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_NullPoolId() {
-        Snapshot snapshot = Mockito.mock(Snapshot.class);
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getPoolId()).thenReturn(null);
 
-        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, volumeVO));
+        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshotMock, volumeVO));
     }
 
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_NullPool() {
-        Snapshot snapshot = Mockito.mock(Snapshot.class);
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
         Mockito.when(dataStoreManager.getDataStore(10L, DataStoreRole.Primary)).thenReturn(null);
 
-        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, volumeVO));
+        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshotMock, volumeVO));
     }
 
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_NonCLVMPool() {
-        Snapshot snapshot = Mockito.mock(Snapshot.class);
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
 
@@ -359,12 +358,11 @@
         Mockito.when(pool.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
         Mockito.when(dataStoreManager.getDataStore(10L, DataStoreRole.Primary)).thenReturn((DataStore) pool);
 
-        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, volumeVO));
+        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshotMock, volumeVO));
     }
 
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_RBDPool() {
-        Snapshot snapshot = Mockito.mock(Snapshot.class);
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
 
@@ -372,7 +370,7 @@
         Mockito.when(pool.getPoolType()).thenReturn(Storage.StoragePoolType.RBD);
         Mockito.when(dataStoreManager.getDataStore(10L, DataStoreRole.Primary)).thenReturn((DataStore) pool);
 
-        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, volumeVO));
+        Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshotMock, volumeVO));
     }
 
     @Test
