--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -549,8 +549,7 @@
         Mockito.when(nicDao.listByVmId(Mockito.anyLong())).thenReturn(new ArrayList<>());
         addresses = ipv6Service.getPublicIpv6AddressesForNetwork(Mockito.mock(Network.class));
         Assert.assertTrue(CollectionUtils.isEmpty(addresses));
-        NicVO nic = Mockito.mock(NicVO.class);
-        Mockito.when(nic.getIPv6Address()).thenReturn(null);
+        NicVO nic = createMockNicWithNoIpv6Address();
         List<NicVO> nics = List.of(nic);
         Mockito.when(nicDao.listByVmId(Mockito.anyLong())).thenReturn(nics);
         addresses = ipv6Service.getPublicIpv6AddressesForNetwork(Mockito.mock(Network.class));
@@ -795,10 +794,15 @@
         Assert.assertEquals(1, removedNics.size());
         Assert.assertEquals(nicId, removedNics.get(0));
         removedNics.clear();
-        NicVO nic1 = Mockito.mock(NicVO.class);
-        Mockito.when(nic1.getIPv6Address()).thenReturn(null);
+        NicVO nic1 = createMockNicWithNoIpv6Address();
         Mockito.when(nicDao.listPlaceholderNicsByNetworkId(Mockito.anyLong())).thenReturn(List.of(nic1));
         ipv6Service.removePublicIpv6PlaceholderNics(network);
         Assert.assertEquals(0, removedNics.size());
     }
+
+    private static NicVO createMockNicWithNoIpv6Address() {
+        NicVO nic = Mockito.mock(NicVO.class);
+        Mockito.when(nic.getIPv6Address()).thenReturn(null);
+        return nic;
+    }
 }
