--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -179,6 +179,8 @@
     TungstenFabricLBHealthMonitorDao tungstenFabricLBHealthMonitorDao;
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
+    @Mock
+    private LoadBalancerVMMapVO loadBalancerVMMapMock;
 
     TungstenElement tungstenElement;
 
@@ -342,11 +344,10 @@
         Network publicNetwork = mock(Network.class);
         Ip ip = mock(Ip.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
-        LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
         LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
-        List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
+        List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapMock);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
         TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
@@ -398,11 +399,10 @@
         Network publicNetwork = mock(Network.class);
         Ip ip = mock(Ip.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
-        LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
         LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
-        List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
+        List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapMock);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
