--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
@@ -278,8 +278,7 @@
 
     @Test
     public void testIsSnapshotStoredOnSameZoneStoreForQCOW2VolumeVHD() {
-        VolumeVO volumeVO = Mockito.mock((VolumeVO.class));
-        Mockito.when(volumeVO.getFormat()).thenReturn(Storage.ImageFormat.VHD);
+        VolumeVO volumeVO = createMockVolumeVO(Storage.ImageFormat.VHD);
         Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSameZoneStoreForQCOW2Volume(Mockito.mock(Snapshot.class), volumeVO));
     }
 
@@ -300,12 +299,17 @@
 
     }
 
+    private static VolumeVO createMockVolumeVO(Storage.ImageFormat imageFormat) {
+        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
+        Mockito.when(volumeVO.getFormat()).thenReturn(imageFormat);
+        return volumeVO;
+    }
+
     @Test
     public void testIsSnapshotStoredOnSameZoneStoreForQCOW2VolumeNoRef() {
         Snapshot snapshot = Mockito.mock((Snapshot.class));
         Mockito.when(snapshot.getId()).thenReturn(1L);
-        VolumeVO volumeVO = Mockito.mock((VolumeVO.class));
-        Mockito.when(volumeVO.getFormat()).thenReturn(Storage.ImageFormat.QCOW2);
+        VolumeVO volumeVO = createMockVolumeVO(Storage.ImageFormat.QCOW2);
         Mockito.when(snapshotDataStoreDao.listBySnapshotIdAndState(1L, ObjectInDataStoreStateMachine.State.Ready)).thenReturn(new ArrayList<>());
         Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSameZoneStoreForQCOW2Volume(snapshot, volumeVO));
 
