--- a/server/src/test/java/com/cloud/network/guru/DirectNetworkGuruTest.java
+++ b/server/src/test/java/com/cloud/network/guru/DirectNetworkGuruTest.java
@@ -152,10 +152,7 @@
         when(plan.getPhysicalNetworkId()).thenReturn(1l);
         when(physicalNetworkDao.findById(physicalNetwork.getId())).thenReturn(physicalNetwork);
         when(offering.isRedundantRouter()).thenReturn(false);
-        when(network.getDns1()).thenReturn(ip4Dns[0]);
-        when(network.getDns2()).thenReturn(ip4Dns[1]);
-        when(network.getIp6Dns1()).thenReturn(ip6Dns[0]);
-        when(network.getIp6Dns2()).thenReturn(ip6Dns[1]);
+        network = MockNetwork.createMockNetwork(ip4Dns[0], ip4Dns[1], ip6Dns[0], ip6Dns[1]);
 
         when(networkModel.areServicesSupportedByNetworkOffering(offering.getId(), Network.Service.SecurityGroup)).thenReturn(false);
 
--- a/server/src/test/java/com/cloud/network/guru/ExternalGuestNetworkGuruTest.java
+++ b/server/src/test/java/com/cloud/network/guru/ExternalGuestNetworkGuruTest.java
@@ -69,11 +69,7 @@
         PhysicalNetworkVO physicalNetwork = Mockito.mock(PhysicalNetworkVO.class);
         Mockito.when(physicalNetworkDao.findById(Mockito.anyLong())).thenReturn(physicalNetwork);
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getDns1()).thenReturn(ip4Dns[0]);
-        Mockito.when(network.getDns2()).thenReturn(ip4Dns[1]);
-        Mockito.when(network.getIp6Dns1()).thenReturn(ip6Dns[0]);
-        Mockito.when(network.getIp6Dns2()).thenReturn(ip6Dns[1]);
+        Network network = MockNetwork.createMockNetwork(ip4Dns[0], ip4Dns[1], ip6Dns[0], ip6Dns[1]);
         Account owner = Mockito.mock(Account.class);
         Network config = guru.design(networkOffering, plan, network, "", 1L, owner);
         assertNotNull(config);
--- a/server/src/test/java/com/cloud/network/guru/MockNetwork.java
+++ b/server/src/test/java/com/cloud/network/guru/MockNetwork.java
@@ -0,0 +1,15 @@
+package com.cloud.network.guru;
+
+import com.cloud.network.Network;
+import org.mockito.Mockito;
+
+public class MockNetwork {
+    public static Network createMockNetwork(String dns1, String dns2, String ip6Dns1, String ip6Dns2) {
+        Network network = Mockito.mock(Network.class);
+        Mockito.when(network.getDns1()).thenReturn(dns1);
+        Mockito.when(network.getDns2()).thenReturn(dns2);
+        Mockito.when(network.getIp6Dns1()).thenReturn(ip6Dns1);
+        Mockito.when(network.getIp6Dns2()).thenReturn(ip6Dns2);
+        return network;
+    }
+}
