--- a/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/service/MockNetworkVO.java
+++ b/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/service/MockNetworkVO.java
@@ -0,0 +1,16 @@
+package org.apache.cloudstack.service;
+
+import com.cloud.network.Network;
+import com.cloud.network.Networks;
+import com.cloud.network.dao.NetworkVO;
+import org.mockito.Mockito;
+
+public class MockNetworkVO {
+
+    public static NetworkVO createMockNetworkVO(Networks.BroadcastDomainType broadcastDomainType, Network.State state) {
+        NetworkVO network = Mockito.mock(NetworkVO.class);
+        Mockito.when(network.getBroadcastDomainType()).thenReturn(broadcastDomainType);
+        Mockito.when(network.getState()).thenReturn(state);
+        return network;
+    }
+}
