--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -267,6 +267,9 @@
     private AccountService accountService;
 
     @Mock
+    private Account ownerMock;
+
+    @Mock
     private EntityManager entityManager;
 
     @Mock
@@ -3571,8 +3574,7 @@
         when(cmd.getDataDiskInfoList()).thenReturn(null);
         when(cmd.getOverrideDiskOfferingId()).thenReturn(null);
 
-        Account owner = mock(Account.class);
-        when(accountService.getActiveAccountById(accountId)).thenReturn(owner);
+        when(accountService.getActiveAccountById(accountId)).thenReturn(ownerMock);
 
         DataCenterVO zone = mock(DataCenterVO.class);
         when(_dcDao.findById(zoneId)).thenReturn(zone);
@@ -3747,8 +3749,7 @@
         when(cmd.getIpToNetworkMap()).thenReturn(null);
         when(cmd.getDataDiskInfoList()).thenReturn(null);
 
-        Account owner = mock(Account.class);
-        when(accountService.getActiveAccountById(accountId)).thenReturn(owner);
+        when(accountService.getActiveAccountById(accountId)).thenReturn(ownerMock);
 
         DataCenterVO zone = mock(DataCenterVO.class);
         when(_dcDao.findById(zoneId)).thenReturn(zone);
@@ -3831,8 +3832,7 @@
 
         ReflectionTestUtils.setField(userVmManagerImpl, "_uuidMgr", uuidMgr);
         CallContext callContext = mock(CallContext.class);
-        Account callingAccount = mock(Account.class);
-        when(callContext.getCallingAccount()).thenReturn(callingAccount);
+        when(callContext.getCallingAccount()).thenReturn(ownerMock);
         try (MockedStatic<CallContext> mockedCallContext = mockStatic(CallContext.class)) {
             mockedCallContext.when(CallContext::current).thenReturn(callContext);
             mockedCallContext.when(() -> CallContext.register(callContext, ApiCommandResourceType.Volume)).thenReturn(callContext);
