--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -156,12 +156,8 @@
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
-        VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
+        VMInstanceVO vm2 = MockVMInstanceVO.createMockVMInstanceVO(host2Id, new Date());
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
-        when(vm2.getHostId()).thenReturn(null);
-        when(vm2.getLastHostId()).thenReturn(host2Id);
-        when(vm2.getState()).thenReturn(VirtualMachine.State.Starting);
-        when(vm2.getUpdateTime()).thenReturn(new Date());
 
         ReflectionTestUtils.setField(processor, "vmCapacityReleaseInterval", 3600);
         processor.process(vmProfile, plan, avoid);
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -155,12 +155,8 @@
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
-        VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
+        VMInstanceVO vm2 = MockVMInstanceVO.createMockVMInstanceVO(host2Id, new Date());
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
-        when(vm2.getHostId()).thenReturn(null);
-        when(vm2.getLastHostId()).thenReturn(host2Id);
-        when(vm2.getState()).thenReturn(VirtualMachine.State.Starting);
-        when(vm2.getUpdateTime()).thenReturn(new Date());
 
         ReflectionTestUtils.setField(processor, "vmCapacityReleaseInterval", 3600);
         processor.process(vmProfile, plan, avoid);
--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/MockVMInstanceVO.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/MockVMInstanceVO.java
@@ -0,0 +1,20 @@
+package org.apache.cloudstack.affinity;
+
+import static org.mockito.Mockito.when;
+
+import java.util.Date;
+
+import com.cloud.vm.VMInstanceVO;
+import com.cloud.vm.VirtualMachine;
+import org.mockito.Mockito;
+
+public class MockVMInstanceVO {
+    public static VMInstanceVO createMockVMInstanceVO(Long lastHostId, Date updateTime) {
+        VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
+        when(vm2.getHostId()).thenReturn(null);
+        when(vm2.getLastHostId()).thenReturn(lastHostId);
+        when(vm2.getState()).thenReturn(VirtualMachine.State.Starting);
+        when(vm2.getUpdateTime()).thenReturn(updateTime);
+        return vm2;
+    }
+}
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/MockVMInstanceVO.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/MockVMInstanceVO.java
@@ -0,0 +1,20 @@
+package org.apache.cloudstack.affinity;
+
+import static org.mockito.Mockito.when;
+
+import java.util.Date;
+
+import com.cloud.vm.VMInstanceVO;
+import com.cloud.vm.VirtualMachine;
+import org.mockito.Mockito;
+
+public class MockVMInstanceVO {
+    public static VMInstanceVO createMockVMInstanceVO(Long lastHostId, Date updateTime) {
+        VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
+        when(vm2.getHostId()).thenReturn(null);
+        when(vm2.getLastHostId()).thenReturn(lastHostId);
+        when(vm2.getState()).thenReturn(VirtualMachine.State.Starting);
+        when(vm2.getUpdateTime()).thenReturn(updateTime);
+        return vm2;
+    }
+}
