--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -194,8 +194,7 @@
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
         VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
-        UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
-        Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
+        UserVmVO userVmVO = MockUserVmVO.createMockUserVmVO(guestOsId);
         Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
@@ -271,8 +270,7 @@
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
         VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
-        UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
-        Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
+        UserVmVO userVmVO = MockUserVmVO.createMockUserVmVO(guestOsId);
         Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
@@ -116,8 +116,7 @@
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
         VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
-        UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
-        Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
+        UserVmVO userVmVO = MockUserVmVO.createMockUserVmVO(guestOsId);
         Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
@@ -160,8 +159,7 @@
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
         VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
-        UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
-        Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
+        UserVmVO userVmVO = MockUserVmVO.createMockUserVmVO(guestOsId);
         Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
@@ -208,8 +206,7 @@
         String guestOsName = "Other";
         List<VolumeObjectTO> volumeObjectTOs = new ArrayList<VolumeObjectTO>();
         VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
-        UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
-        Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
+        UserVmVO userVmVO = MockUserVmVO.createMockUserVmVO(guestOsId);
         Mockito.when(vmSnapshot.getVmId()).thenReturn(vmId);
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/MockUserVmVO.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/MockUserVmVO.java
@@ -0,0 +1,13 @@
+package org.apache.cloudstack.storage.vmsnapshot;
+
+import org.mockito.Mockito;
+
+import com.cloud.vm.UserVmVO;
+
+public class MockUserVmVO {
+    public static UserVmVO createMockUserVmVO(Long guestOsId) {
+        UserVmVO userVmVO = Mockito.mock(UserVmVO.class);
+        Mockito.when(userVmVO.getGuestOSId()).thenReturn(guestOsId);
+        return userVmVO;
+    }
+}
