--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -341,13 +341,13 @@
         Network network = mock(Network.class);
         Network publicNetwork = mock(Network.class);
         Ip ip = mock(Ip.class);
-        LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
+        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
+        List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
+        LoadBalancingRule loadBalancingRule1 = createMockLoadBalancingRule(443, ip, lbStickinessPolicyList);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
-        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
-        List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
         TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
@@ -366,13 +366,6 @@
         latest.setSecretKey("secretkey");
         when(ApiDBUtils.searchForLatestUserKeyPair(Mockito.anyLong())).thenReturn(latest);
         when(lbStickinessPolicy.getParams()).thenReturn(pairList);
-        when(loadBalancingRule1.getId()).thenReturn(1L);
-        when(loadBalancingRule1.getState()).thenReturn(FirewallRule.State.Add);
-        when(loadBalancingRule1.getAlgorithm()).thenReturn("roundrobin");
-        when(loadBalancingRule1.getSourcePortStart()).thenReturn(443);
-        when(loadBalancingRule1.getDefaultPortStart()).thenReturn(443);
-        when(loadBalancingRule1.getStickinessPolicies()).thenReturn(lbStickinessPolicyList);
-        when(loadBalancingRule1.getSourceIp()).thenReturn(ip);
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(ipAddressVO.getAddress()).thenReturn(ip);
@@ -397,13 +390,13 @@
         Network network = mock(Network.class);
         Network publicNetwork = mock(Network.class);
         Ip ip = mock(Ip.class);
-        LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
+        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
+        List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
+        LoadBalancingRule loadBalancingRule1 = createMockLoadBalancingRule(80, ip, lbStickinessPolicyList);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
-        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
-        List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
@@ -416,13 +409,6 @@
 
         when(lbStickinessPolicy.getMethodName()).thenReturn("AppCookie");
         when(lbStickinessPolicy.getParams()).thenReturn(pairList);
-        when(loadBalancingRule1.getId()).thenReturn(1L);
-        when(loadBalancingRule1.getState()).thenReturn(FirewallRule.State.Add);
-        when(loadBalancingRule1.getAlgorithm()).thenReturn("roundrobin");
-        when(loadBalancingRule1.getSourcePortStart()).thenReturn(80);
-        when(loadBalancingRule1.getDefaultPortStart()).thenReturn(443);
-        when(loadBalancingRule1.getStickinessPolicies()).thenReturn(lbStickinessPolicyList);
-        when(loadBalancingRule1.getSourceIp()).thenReturn(ip);
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(ipAddressVO.getAddress()).thenReturn(ip);
@@ -962,4 +948,16 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    private static LoadBalancingRule createMockLoadBalancingRule(int sourcePortStart, Ip sourceIp, List<LoadBalancingRule.LbStickinessPolicy> stickinessPolicies) {
+        LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
+        when(loadBalancingRule.getId()).thenReturn(1L);
+        when(loadBalancingRule.getState()).thenReturn(FirewallRule.State.Add);
+        when(loadBalancingRule.getAlgorithm()).thenReturn("roundrobin");
+        when(loadBalancingRule.getSourcePortStart()).thenReturn(sourcePortStart);
+        when(loadBalancingRule.getDefaultPortStart()).thenReturn(443);
+        when(loadBalancingRule.getStickinessPolicies()).thenReturn(stickinessPolicies);
+        when(loadBalancingRule.getSourceIp()).thenReturn(sourceIp);
+        return loadBalancingRule;
+    }
 }
