--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -179,6 +179,8 @@
     TungstenFabricLBHealthMonitorDao tungstenFabricLBHealthMonitorDao;
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
+    @Mock
+    private UserVm userVmMock;
 
     TungstenElement tungstenElement;
 
@@ -500,7 +502,6 @@
         Network publicNetwork = mock(Network.class);
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        UserVm userVm = mock(UserVm.class);
         Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(true);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
@@ -508,7 +509,7 @@
         when(portForwardingRule.getState()).thenReturn(FirewallRule.State.Add);
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(ApiDBUtils.findIpAddressById(anyLong())).thenReturn(ipAddressVO);
-        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVm);
+        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVmMock);
         when(networkModel.getNicInNetwork(anyLong(), anyLong())).thenReturn(nic);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenPortForwardingCommand.class), anyLong())).thenReturn(applyTungstenPortForwardingAnswer);
 
@@ -521,7 +522,6 @@
         Network publicNetwork = mock(Network.class);
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        UserVm userVm = mock(UserVm.class);
         Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(false);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
@@ -529,7 +529,7 @@
         when(portForwardingRule.getState()).thenReturn(FirewallRule.State.Add);
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(ApiDBUtils.findIpAddressById(anyLong())).thenReturn(ipAddressVO);
-        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVm);
+        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVmMock);
         when(networkModel.getNicInNetwork(anyLong(), anyLong())).thenReturn(nic);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenPortForwardingCommand.class), anyLong())).thenReturn(applyTungstenPortForwardingAnswer);
 
@@ -542,7 +542,6 @@
         Network publicNetwork = mock(Network.class);
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        UserVm userVm = mock(UserVm.class);
         Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(true);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
@@ -550,7 +549,7 @@
         when(portForwardingRule.getState()).thenReturn(FirewallRule.State.Revoke);
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(ApiDBUtils.findIpAddressById(anyLong())).thenReturn(ipAddressVO);
-        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVm);
+        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVmMock);
         when(networkModel.getNicInNetwork(anyLong(), anyLong())).thenReturn(nic);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenPortForwardingCommand.class), anyLong())).thenReturn(applyTungstenPortForwardingAnswer);
 
@@ -563,7 +562,6 @@
         Network publicNetwork = mock(Network.class);
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        UserVm userVm = mock(UserVm.class);
         Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(false);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
@@ -571,7 +569,7 @@
         when(portForwardingRule.getState()).thenReturn(FirewallRule.State.Revoke);
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(ApiDBUtils.findIpAddressById(anyLong())).thenReturn(ipAddressVO);
-        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVm);
+        when(ApiDBUtils.findUserVmById(anyLong())).thenReturn(userVmMock);
         when(networkModel.getNicInNetwork(anyLong(), anyLong())).thenReturn(nic);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenPortForwardingCommand.class), anyLong())).thenReturn(applyTungstenPortForwardingAnswer);
 
