--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -133,6 +133,8 @@
     @Mock
     AccountManager accountManager;
     @Mock
+    private NetworkOffering networkOfferingMock;
+    @Mock
     NetworkOfferingDao networkOfferingDao;
     @Mock
     PhysicalNetworkDao physicalNetworkDao;
@@ -924,8 +926,7 @@
             Assert.fail(String.format("Unable to set network guestType, %s", e.getMessage()));
         }
         networkVO.setIp6Cidr("cidr");
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        boolean updated = service.checkAndUpdateNetworkDns(networkVO, offering, "", null, "", null);
+        boolean updated = service.checkAndUpdateNetworkDns(networkVO, networkOfferingMock, "", null, "", null);
         Assert.assertTrue(updated);
         Assert.assertNull(networkVO.getDns1());
         Assert.assertNull(networkVO.getDns2());
@@ -1207,12 +1208,11 @@
         ReflectionTestUtils.setField(createNetworkCmd, "physicalNetworkId", null);
         NetworkVO networkVO = Mockito.spy(NetworkVO.class);
         IpAddress ipAddress = Mockito.mock(IPAddressVO.class);
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
         Long networkId = 7l;
         when(networkVO.getId()).thenReturn(networkId);
         when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
         when(networkDao.findById(networkId)).thenReturn(networkVO);
-        when(entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(ntwkOff);
+        when(entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOfferingMock);
         when(entityMgr.findById(eq(DataCenter.class), anyLong())).thenReturn(dc);
         when(ipAddress.getId()).thenReturn(5l);
         when(networkVO.getId()).thenReturn(networkId);
