--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -315,6 +315,7 @@
     final static Integer port = 8080;
 
     final OneLineParser statsParserMock = Mockito.mock(OneLineParser.class);
+    private final VifDriver sharedVifDriver = Mockito.mock(VifDriver.class);
 
     @Before
     public void setup() throws Exception {
@@ -1510,7 +1511,6 @@
         final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final DiskTO diskTO = Mockito.mock(DiskTO.class);
-        final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
         final PrepareForMigrationCommand command = new PrepareForMigrationCommand(vm);
 
@@ -1527,7 +1527,7 @@
         when(nicTO.getType()).thenReturn(TrafficType.Guest);
         when(diskTO.getType()).thenReturn(Volume.Type.ISO);
 
-        when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(vifDriver);
+        when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(sharedVifDriver);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
@@ -1558,7 +1558,6 @@
         final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final DiskTO diskTO = Mockito.mock(DiskTO.class);
-        final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
         final PrepareForMigrationCommand command = new PrepareForMigrationCommand(vm);
 
@@ -1575,7 +1574,7 @@
         when(nicTO.getType()).thenReturn(TrafficType.Guest);
         when(diskTO.getType()).thenReturn(Volume.Type.ISO);
 
-        when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(vifDriver);
+        when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(sharedVifDriver);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
         when(storagePoolManager.connectPhysicalDisksViaVmSpec(vm, true)).thenReturn(true);
 
@@ -1606,7 +1605,6 @@
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
         final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
-        final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
         final PrepareForMigrationCommand command = new PrepareForMigrationCommand(vm);
 
@@ -1648,7 +1646,6 @@
         final KVMStoragePoolManager storagePoolManager = Mockito.mock(KVMStoragePoolManager.class);
         final NicTO nicTO = Mockito.mock(NicTO.class);
         final DiskTO volume = Mockito.mock(DiskTO.class);
-        final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
         final PrepareForMigrationCommand command = new PrepareForMigrationCommand(vm);
 
@@ -1665,7 +1662,7 @@
         when(nicTO.getType()).thenReturn(TrafficType.Guest);
         when(volume.getType()).thenReturn(Volume.Type.ISO);
 
-        when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(vifDriver);
+        when(libvirtComputingResourceMock.getVifDriver(nicTO.getType(), nicTO.getName())).thenReturn(sharedVifDriver);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
         try {
             when(libvirtComputingResourceMock.getVolumePath(conn, volume)).thenThrow(URISyntaxException.class);
@@ -3779,9 +3776,8 @@
         final InterfaceDef intDef = Mockito.mock(InterfaceDef.class);
         nics.add(intDef);
 
-        final VifDriver vifDriver = Mockito.mock(VifDriver.class);
         final List<VifDriver> drivers = new ArrayList<VifDriver>();
-        drivers.add(vifDriver);
+        drivers.add(sharedVifDriver);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
@@ -3828,9 +3824,8 @@
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
 
-        final VifDriver vifDriver = Mockito.mock(VifDriver.class);
         final List<VifDriver> drivers = new ArrayList<VifDriver>();
-        drivers.add(vifDriver);
+        drivers.add(sharedVifDriver);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
