--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -159,10 +159,12 @@
     int cpuSpeedInOffering = 500;
     int ramInOffering = 512;
     AccountVO acct = new AccountVO(accountId);
+    private VirtualMachineProfileImpl implicitPlannerVmProfile;
 
     @Before
     public void setUp() {
         ComponentContext.initComponentsLifeCycle();
+        implicitPlannerVmProfile = mock(VirtualMachineProfileImpl.class);
 
         acct.setType(Account.Type.NORMAL);
         acct.setAccountName("user1");
@@ -197,15 +199,14 @@
 
     @Test
     public void checkStrictModeWithCurrentAccountVmsPresent() throws InsufficientServerCapacityException {
-        VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         ExcludeList avoids = new ExcludeList();
 
-        initializeForTest(vmProfile, plan);
+        initializeForTest(implicitPlannerVmProfile, plan);
 
         initializeForImplicitPlannerTest(false);
 
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(implicitPlannerVmProfile, plan, avoids);
 
         // Validations.
         // Check cluster 2 and 3 are not in the cluster list.
@@ -224,17 +225,16 @@
     @Test
     public void checkStrictModeHostWithCurrentAccountVmsFull() throws InsufficientServerCapacityException {
         @SuppressWarnings("unchecked")
-        VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         ExcludeList avoids = new ExcludeList();
 
-        initializeForTest(vmProfile, plan);
+        initializeForTest(implicitPlannerVmProfile, plan);
 
         initializeForImplicitPlannerTest(false);
 
         // Mark the host 5 with current account vms to be in avoid list.
         avoids.addHost(5L);
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(implicitPlannerVmProfile, plan, avoids);
 
         // Validations.
         // Check cluster 1 and 3 are not in the cluster list.
@@ -253,18 +253,17 @@
     @Test
     public void checkStrictModeNoHostsAvailable() throws InsufficientServerCapacityException {
         @SuppressWarnings("unchecked")
-        VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         ExcludeList avoids = new ExcludeList();
 
-        initializeForTest(vmProfile, plan);
+        initializeForTest(implicitPlannerVmProfile, plan);
 
         initializeForImplicitPlannerTest(false);
 
         // Mark the host 5 and 6 to be in avoid list.
         avoids.addHost(5L);
         avoids.addHost(6L);
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(implicitPlannerVmProfile, plan, avoids);
 
         // Validations.
         // Check cluster list is empty.
@@ -274,18 +273,17 @@
     @Test
     public void checkPreferredModePreferredHostAvailable() throws InsufficientServerCapacityException {
         @SuppressWarnings("unchecked")
-        VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         ExcludeList avoids = new ExcludeList();
 
-        initializeForTest(vmProfile, plan);
+        initializeForTest(implicitPlannerVmProfile, plan);
 
         initializeForImplicitPlannerTest(true);
 
         // Mark the host 5 and 6 to be in avoid list.
         avoids.addHost(5L);
         avoids.addHost(6L);
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(implicitPlannerVmProfile, plan, avoids);
 
         // Validations.
         // Check cluster 1 and 2 are not in the cluster list.
@@ -304,11 +302,10 @@
     @Test
     public void checkPreferredModeNoHostsAvailable() throws InsufficientServerCapacityException {
         @SuppressWarnings("unchecked")
-        VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         ExcludeList avoids = new ExcludeList();
 
-        initializeForTest(vmProfile, plan);
+        initializeForTest(implicitPlannerVmProfile, plan);
 
         initializeForImplicitPlannerTest(false);
 
@@ -316,7 +313,7 @@
         avoids.addHost(5L);
         avoids.addHost(6L);
         avoids.addHost(7L);
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(implicitPlannerVmProfile, plan, avoids);
 
         // Validations.
         // Check cluster list is empty.
