--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -243,9 +243,8 @@
         final DataCenter dc = mock(DataCenter.class);
         when(dest.getDataCenter()).thenReturn(dc);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -302,9 +301,8 @@
         final DataCenter dc = mock(DataCenter.class);
         when(dest.getDataCenter()).thenReturn(dc);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -356,9 +354,8 @@
         when(dest.getDataCenter()).thenReturn(dc);
         when(dcdao.findById((long)anyInt())).thenReturn(dc);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -408,9 +405,8 @@
         when(dest.getDataCenter()).thenReturn(dc);
         when(dcdao.findById((long)anyInt())).thenReturn(dc);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -453,9 +449,8 @@
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -489,9 +484,8 @@
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -524,9 +518,8 @@
         when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
         when(offering.getGuestType()).thenReturn(GuestType.Isolated);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -566,9 +559,8 @@
         when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
         when(offering.getGuestType()).thenReturn(GuestType.Isolated);
 
-        final HostVO brocadeHost = mock(HostVO.class);
-        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
-        when(brocadeHost.getId()).thenReturn(NETWORK_ID);
+        final HostVO brocadeHost = createMockHostVO(NETWORK_ID);
+        when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
@@ -588,4 +580,9 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+    private HostVO createMockHostVO(final long hostId) {
+        final HostVO hostVO = mock(HostVO.class);
+        when(hostVO.getId()).thenReturn(hostId);
+        return hostVO;
+    }
 }
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -272,11 +272,10 @@
         final DataCenter dc = mock(DataCenter.class);
         when(dest.getDataCenter()).thenReturn(dc);
 
-        final HostVO niciraHost = mock(HostVO.class);
+        final HostVO niciraHost = createMockHostVO(NETWORK_ID);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
         when(niciraHost.getDetail("transportzoneuuid")).thenReturn("aaaa");
         when(niciraHost.getDetail("transportzoneisotype")).thenReturn("stt");
-        when(niciraHost.getId()).thenReturn(NETWORK_ID);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
@@ -328,11 +327,10 @@
         final DataCenter dc = mock(DataCenter.class);
         when(dest.getDataCenter()).thenReturn(dc);
 
-        final HostVO niciraHost = mock(HostVO.class);
+        final HostVO niciraHost = createMockHostVO(NETWORK_ID);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
         when(niciraHost.getDetail("transportzoneuuid")).thenReturn("aaaa");
         when(niciraHost.getDetail("transportzoneisotype")).thenReturn("stt");
-        when(niciraHost.getId()).thenReturn(NETWORK_ID);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
@@ -384,11 +382,10 @@
         final DataCenter dc = mock(DataCenter.class);
         when(dest.getDataCenter()).thenReturn(dc);
 
-        final HostVO niciraHost = mock(HostVO.class);
+        final HostVO niciraHost = createMockHostVO(NETWORK_ID);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
         when(niciraHost.getDetail("transportzoneuuid")).thenReturn("aaaa");
         when(niciraHost.getDetail("transportzoneisotype")).thenReturn("stt");
-        when(niciraHost.getId()).thenReturn(NETWORK_ID);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
@@ -441,11 +438,10 @@
         final DataCenter dc = mock(DataCenter.class);
         when(dest.getDataCenter()).thenReturn(dc);
 
-        final HostVO niciraHost = mock(HostVO.class);
+        final HostVO niciraHost = createMockHostVO(NETWORK_ID);
         when(hostdao.findById(anyLong())).thenReturn(niciraHost);
         when(niciraHost.getDetail("transportzoneuuid")).thenReturn("aaaa");
         when(niciraHost.getDetail("transportzoneisotype")).thenReturn("stt");
-        when(niciraHost.getId()).thenReturn(NETWORK_ID);
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
@@ -469,4 +465,9 @@
         verify(agentmgr, times(1)).easySend(eq(NETWORK_ID), (Command)any());
         verify(implementednetwork, times(1)).setBroadcastUri(null);
     }
+    private HostVO createMockHostVO(final long hostId) {
+        final HostVO hostVO = mock(HostVO.class);
+        when(hostVO.getId()).thenReturn(hostId);
+        return hostVO;
+    }
 }
