--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
@@ -126,17 +126,21 @@
         return map;
     }
 
+    private static ServiceOfferingVO createMockServiceOffering(long serviceOfferingId) {
+        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
+        Mockito.when(serviceOffering.getId()).thenReturn(serviceOfferingId);
+        return serviceOffering;
+    }
+
     @Test
     public void testGetServiceOfferingNodeTypeMap() {
         String workerOfferingUuid = UUID.randomUUID().toString();
         String controlOfferingUuid = UUID.randomUUID().toString();
 
-        ServiceOfferingVO workerOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(workerOffering.getId()).thenReturn(1L);
+        ServiceOfferingVO workerOffering = createMockServiceOffering(1L);
         Mockito.when(serviceOfferingDao.findByUuid(workerOfferingUuid)).thenReturn(workerOffering);
 
-        ServiceOfferingVO controlOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(controlOffering.getId()).thenReturn(2L);
+        ServiceOfferingVO controlOffering = createMockServiceOffering(2L);
         Mockito.when(serviceOfferingDao.findByUuid(controlOfferingUuid)).thenReturn(controlOffering);
 
         Map<String, Map<String, String>> serviceOfferingNodeTypeMap = new HashMap<>();
@@ -163,8 +167,7 @@
     public void testGetServiceOfferingNodeTypeMapEtcdNodes() {
         String etcdOfferingUuid = UUID.randomUUID().toString();
 
-        ServiceOfferingVO etcdOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(etcdOffering.getId()).thenReturn(3L);
+        ServiceOfferingVO etcdOffering = createMockServiceOffering(3L);
         Mockito.when(serviceOfferingDao.findByUuid(etcdOfferingUuid)).thenReturn(etcdOffering);
 
         Map<String, Map<String, String>> serviceOfferingNodeTypeMap = new HashMap<>();
