--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -70,6 +70,8 @@
     @Mock
     protected XenServer610Resource xenServer610Resource;
 
+    private final VolumeTO sharedVol1 = Mockito.mock(VolumeTO.class);
+
     @Test
     public void testCheckNetworkCommandFailure() {
         final XenServer610Resource xenServer610Resource = new XenServer610Resource();
@@ -107,13 +109,12 @@
         final Connection conn = Mockito.mock(Connection.class);
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
 
-        final VolumeTO vol1 = Mockito.mock(VolumeTO.class);
         final VolumeTO vol2 = Mockito.mock(VolumeTO.class);
         final StorageFilerTO storage1 = Mockito.mock(StorageFilerTO.class);
         final StorageFilerTO storage2 = Mockito.mock(StorageFilerTO.class);
 
         final Map<VolumeTO, StorageFilerTO> volumeToFiler = new  HashMap<VolumeTO, StorageFilerTO>();
-        volumeToFiler.put(vol1, storage1);
+        volumeToFiler.put(sharedVol1, storage1);
         volumeToFiler.put(vol2, storage2);
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
@@ -180,13 +181,12 @@
         final Connection conn = Mockito.mock(Connection.class);
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
 
-        final VolumeTO vol1 = Mockito.mock(VolumeTO.class);
         final VolumeTO vol2 = Mockito.mock(VolumeTO.class);
         final StorageFilerTO storage1 = Mockito.mock(StorageFilerTO.class);
         final StorageFilerTO storage2 = Mockito.mock(StorageFilerTO.class);
 
         final List<Pair<VolumeTO, String>> volumeToFiler = new ArrayList<>();
-        volumeToFiler.add(new Pair<>(vol1, storage1.getPath()));
+        volumeToFiler.add(new Pair<>(sharedVol1, storage1.getPath()));
         volumeToFiler.add(new Pair<>(vol2, storage2.getPath()));
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
@@ -333,11 +333,10 @@
         final Connection conn = Mockito.mock(Connection.class);
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
 
-        final VolumeTO volume1 = Mockito.mock(VolumeTO.class);
         final VolumeTO volume2 = Mockito.mock(VolumeTO.class);
 
         final List<Pair<VolumeTO, Object>> volumeToSr = new ArrayList<Pair<VolumeTO, Object>>();
-        volumeToSr.add(new Pair<VolumeTO, Object>(volume1, new String("a")));
+        volumeToSr.add(new Pair<VolumeTO, Object>(sharedVol1, new String("a")));
         volumeToSr.add(new Pair<VolumeTO, Object>(volume2, new String("b")));
 
         final List<Pair<NicTO, Object>> nicToNetwork = new ArrayList<Pair<NicTO, Object>>();
