--- a/server/src/test/java/com/cloud/deploy/DeploymentPlanningManagerImplTest.java
+++ b/server/src/test/java/com/cloud/deploy/DeploymentPlanningManagerImplTest.java
@@ -456,8 +456,7 @@
 
     @Test
     public void avoidDisabledDataCentersTest() {
-        DataCenter dc = Mockito.mock(DataCenter.class);
-        Mockito.when(dc.getId()).thenReturn(123l);
+        DataCenter dc = createMockDataCenter();
 
         ExcludeList avoids = new ExcludeList();
         AllocationState[] allocationStates = AllocationState.values();
@@ -980,6 +979,12 @@
         return planner;
     }
 
+    private static DataCenter createMockDataCenter() {
+        DataCenter dc = Mockito.mock(DataCenter.class);
+        Mockito.when(dc.getId()).thenReturn(123l);
+        return dc;
+    }
+
     private DataCenter prepareAvoidDisabledTests() {
         DataCenter dc = Mockito.mock(DataCenter.class);
         Mockito.when(dc.getId()).thenReturn(123l);
