--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -7289,6 +7289,12 @@
         Mockito.verify(diskDef).defFileBasedDisk(PHYSICAL_DISK_PATH, DEV_ID, DISK_BUS_TYPE_DATA, DiskDef.DiskFmtType.QCOW2);
     }
 
+    private static InterfaceDef createMockInterfaceDef() {
+        final InterfaceDef interfaceDef = Mockito.mock(InterfaceDef.class);
+        Mockito.doReturn("a0:90:27:a9:9e:62").when(interfaceDef).getMacAddress();
+        return interfaceDef;
+    }
+
     @Test
     public void getInterfaceTestValidMacAddressReturnInterface() {
         String macAddress = "a0:90:27:a9:9e:62";
@@ -7310,11 +7316,10 @@
         String invalidMacAddress = "ea:57:5d:f1:64:05";
         String macAddress = "a0:90:27:a9:9e:62";
         final String vmName = "Test";
-        final InterfaceDef interfaceDef = Mockito.mock(InterfaceDef.class);
+        final InterfaceDef interfaceDef = createMockInterfaceDef();
         final List<InterfaceDef> interfaces = new ArrayList<>();
         interfaces.add(interfaceDef);
 
-        Mockito.doReturn(macAddress).when(interfaceDef).getMacAddress();
         Mockito.doReturn(interfaces).when(libvirtComputingResourceSpy).getInterfaces(Mockito.any(), Mockito.anyString());
 
         libvirtComputingResourceSpy.getInterface(connMock, vmName, invalidMacAddress);
