--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -90,6 +90,7 @@
     AgentManager agentmgr = mock(AgentManager.class);
     NetworkOrchestrationService netmgr = mock(NetworkOrchestrationService.class);
     NetworkModel netmodel = mock(NetworkModel.class);
+    private final Network networkMock = mock(Network.class);
 
     HostDao hostdao = mock(HostDao.class);
     NetworkDao netdao = mock(NetworkDao.class);
@@ -167,10 +168,9 @@
         when(netmodel.listNetworkOfferingServices(NETWORK_ID)).thenReturn(Arrays.asList(Service.Connectivity));
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designednetwork = guru.design(offering, plan, networkMock, "", 1L, account);
         assertTrue(designednetwork != null);
         assertTrue(designednetwork.getBroadcastDomainType() == BroadcastDomainType.Vcs);
     }
@@ -188,10 +188,9 @@
         when(offering.getGuestType()).thenReturn(GuestType.Isolated);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designednetwork = guru.design(offering, plan, networkMock, "", 1L, account);
         assertTrue(designednetwork == null);
     }
 
@@ -210,10 +209,9 @@
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
-        final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
+        final Network designednetwork = guru.design(offering, plan, networkMock, "", 1L, account);
         assertTrue(designednetwork == null);
     }
 
