--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -760,11 +760,9 @@
         long newVmId = 2L;
         long networkId = 1L;
 
-        ClusterVO cluster = mock(ClusterVO.class);
-        when(cluster.getId()).thenReturn(clusterId);
+        ClusterVO cluster = createMockCluster();
         when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
         when(cluster.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
-        when(cluster.getDataCenterId()).thenReturn(zoneId);
         when(clusterDao.findById(clusterId)).thenReturn(cluster);
 
         ImportVmCmd importVmCmd = Mockito.mock(ImportVmCmd.class);
@@ -1076,6 +1074,11 @@
     }
 
     private ClusterVO getClusterForTests() {
+        ClusterVO cluster = createMockCluster();
+        return cluster;
+    }
+
+    private static ClusterVO createMockCluster() {
         ClusterVO cluster = mock(ClusterVO.class);
         when(cluster.getId()).thenReturn(1L);
         when(cluster.getDataCenterId()).thenReturn(1L);
