--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -246,6 +246,7 @@
     VirtualMachineTO vmTO;
     @Mock
     LibvirtVMDef vmDef;
+    private final LibvirtVMDef vmDefMock = Mockito.mock(LibvirtVMDef.class);
     @Mock
     Logger loggerMock;
     @Mock
@@ -5110,7 +5111,6 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5121,12 +5121,12 @@
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenReturn(conn);
-            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDef);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenReturn(conn);
+            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDefMock);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
@@ -5146,7 +5146,7 @@
         verify(libvirtComputingResourceMock, times(1)).getStoragePoolMgr();
         verify(libvirtComputingResourceMock, times(1)).getLibvirtUtilitiesHelper();
         try {
-            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDef.getHvsType());
+            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDefMock.getHvsType());
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -5163,7 +5163,6 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5172,11 +5171,11 @@
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
         when(vmSpec.getName()).thenReturn(vmName);
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenThrow(LibvirtException.class);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenThrow(LibvirtException.class);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -5190,7 +5189,7 @@
         verify(libvirtComputingResourceMock, times(1)).getStoragePoolMgr();
         verify(libvirtComputingResourceMock, times(1)).getLibvirtUtilitiesHelper();
         try {
-            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDef.getHvsType());
+            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDefMock.getHvsType());
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -5207,7 +5206,6 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5218,12 +5216,12 @@
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenReturn(conn);
-            doThrow(InternalErrorException.class).when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDef);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenReturn(conn);
+            doThrow(InternalErrorException.class).when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDefMock);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
@@ -5241,7 +5239,7 @@
         verify(libvirtComputingResourceMock, times(1)).getStoragePoolMgr();
         verify(libvirtComputingResourceMock, times(1)).getLibvirtUtilitiesHelper();
         try {
-            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDef.getHvsType());
+            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDefMock.getHvsType());
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -5258,7 +5256,6 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5269,12 +5266,12 @@
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenReturn(conn);
-            doThrow(URISyntaxException.class).when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDef);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenReturn(conn);
+            doThrow(URISyntaxException.class).when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDefMock);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
@@ -5292,7 +5289,7 @@
         verify(libvirtComputingResourceMock, times(1)).getStoragePoolMgr();
         verify(libvirtComputingResourceMock, times(1)).getLibvirtUtilitiesHelper();
         try {
-            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDef.getHvsType());
+            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDefMock.getHvsType());
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -5309,7 +5306,6 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
         final VirtualRoutingResource virtRouterResource = Mockito.mock(VirtualRoutingResource.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
@@ -5322,12 +5318,12 @@
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenReturn(conn);
-            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDef);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenReturn(conn);
+            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDefMock);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
@@ -5338,9 +5334,9 @@
 
         when(storagePoolMgr.connectPhysicalDisksViaVmSpec(vmSpec, false)).thenReturn(true);
         try {
-            doNothing().when(libvirtComputingResourceMock).createVifs(vmSpec, vmDef);
-
-            when(libvirtComputingResourceMock.startVM(conn, vmName, vmDef.toString())).thenReturn("SUCCESS");
+            doNothing().when(libvirtComputingResourceMock).createVifs(vmSpec, vmDefMock);
+
+            when(libvirtComputingResourceMock.startVM(conn, vmName, vmDefMock.toString())).thenReturn("SUCCESS");
 
             when(vmSpec.getBootArgs()).thenReturn("ls -lart");
             when(libvirtComputingResourceMock.passCmdLine(vmName, vmSpec.getBootArgs())).thenReturn(true);
@@ -5371,7 +5367,7 @@
         verify(libvirtComputingResourceMock, times(1)).getStoragePoolMgr();
         verify(libvirtComputingResourceMock, times(1)).getLibvirtUtilitiesHelper();
         try {
-            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDef.getHvsType());
+            verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDefMock.getHvsType());
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -5388,7 +5384,6 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
         final VirtualRoutingResource virtRouterResource = Mockito.mock(VirtualRoutingResource.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
@@ -5401,12 +5396,12 @@
         when(vmSpec.getNics()).thenReturn(nics);
         when(vmSpec.getType()).thenReturn(VirtualMachine.Type.DomainRouter);
         when(vmSpec.getName()).thenReturn(vmName);
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenReturn(conn);
-            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDef);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenReturn(conn);
+            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDefMock);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
@@ -5417,9 +5412,9 @@
 
         when(storagePoolMgr.connectPhysicalDisksViaVmSpec(vmSpec, false)).thenReturn(true);
         try {
-            doNothing().when(libvirtComputingResourceMock).createVifs(vmSpec, vmDef);
-
-            when(libvirtComputingResourceMock.startVM(conn, vmName, vmDef.toString())).thenReturn("SUCCESS");
+            doNothing().when(libvirtComputingResourceMock).createVifs(vmSpec, vmDefMock);
+
+            when(libvirtComputingResourceMock.startVM(conn, vmName, vmDefMock.toString())).thenReturn("SUCCESS");
 
 
             when(vmSpec.getBootArgs()).thenReturn("ls -lart");
@@ -5450,7 +5445,7 @@
             verify(libvirtComputingResourceMock, times(1)).getStoragePoolMgr();
             verify(libvirtComputingResourceMock, times(1)).getLibvirtUtilitiesHelper();
             try {
-                verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDef.getHvsType());
+                verify(libvirtUtilitiesHelper, times(1)).getConnectionByType(vmDefMock.getHvsType());
             } catch (final LibvirtException e) {
                 fail(e.getMessage());
             }
@@ -5468,7 +5463,6 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
-        final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5480,14 +5474,14 @@
         when(vmSpec.getName()).thenReturn(vmName);
         when(vmSpec.getDisks()).thenReturn(new DiskTO[]{diskToMock});
         when(diskToMock.getData()).thenReturn(new VolumeObjectTO());
-        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDef);
+        when(libvirtComputingResourceMock.createVMFromSpec(vmSpec)).thenReturn(vmDefMock);
 
         when(libvirtComputingResourceMock.recreateCheckpointsOnVm(any(), any(), any())).thenReturn(true);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         try {
-            when(libvirtUtilitiesHelper.getConnectionByType(vmDef.getHvsType())).thenReturn(conn);
-            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDef);
+            when(libvirtUtilitiesHelper.getConnectionByType(vmDefMock.getHvsType())).thenReturn(conn);
+            doNothing().when(libvirtComputingResourceMock).createVbd(conn, vmSpec, vmName, vmDefMock);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
