--- a/api/src/test/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmdTest.java
@@ -321,12 +321,11 @@
 
         ReflectionTestUtils.setField(cmd, "dataDisksDetails", dataDisksDetails);
 
-        EntityManager entityMgr = mock(EntityManager.class);
+        DiskOffering diskOffering = mock(DiskOffering.class);
+        EntityManager entityMgr = createMockEntityManager(diskOffering);
         ReflectionTestUtils.setField(cmd, "_entityMgr", entityMgr);
-        DiskOffering diskOffering = mock(DiskOffering.class);
         when(diskOffering.getDiskSize()).thenReturn(1024 * 1024 * 1024L);
         when(diskOffering.isCustomizedIops()).thenReturn(true);
-        when(entityMgr.findByUuid(DiskOffering.class, "offering-uuid")).thenReturn(diskOffering);
 
         List<VmDiskInfo> result = cmd.getDataDiskInfoList();
 
@@ -624,4 +623,10 @@
     public void testIsBlankInstance() {
         assertFalse(cmd.isBlankInstance());
     }
+
+    private static EntityManager createMockEntityManager(DiskOffering diskOffering) {
+        EntityManager entityMgr = mock(EntityManager.class);
+        when(entityMgr.findByUuid(DiskOffering.class, "offering-uuid")).thenReturn(diskOffering);
+        return entityMgr;
+    }
 }
