--- a/server/src/test/java/com/cloud/template/HypervisorTemplateAdapterTest.java
+++ b/server/src/test/java/com/cloud/template/HypervisorTemplateAdapterTest.java
@@ -332,11 +332,10 @@
 
     @Test
     public void createTemplateWithinZonesTestZoneIdsNotNullShouldNotCallListAllZones() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
         VMTemplateVO vmTemplateVOMock = Mockito.mock(VMTemplateVO.class);
         List<Long> zoneIds = List.of(1L);
-
-        Mockito.when(templateProfileMock.getZoneIdList()).thenReturn(zoneIds);
+        TemplateProfile templateProfileMock = createMockTemplateProfile(zoneIds);
+
         Mockito.doReturn(null).when(_adapter).getImageStoresThrowsExceptionIfNotFound(Mockito.any(Long.class), Mockito.any(TemplateProfile.class));
         Mockito.doReturn(null).when(_templateMgr).verifyHeuristicRulesForZone(Mockito.any(VMTemplateVO.class), Mockito.anyLong());
         Mockito.doNothing().when(_adapter).standardImageStoreAllocation(Mockito.isNull(), Mockito.any(VMTemplateVO.class), Mockito.anyLong());
@@ -348,11 +347,10 @@
 
     @Test
     public void createTemplateWithinZonesTestZoneDoesNotHaveActiveHeuristicRulesShouldCallStandardImageStoreAllocation() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
         VMTemplateVO vmTemplateVOMock = Mockito.mock(VMTemplateVO.class);
         List<Long> zoneIds = List.of(zoneId);
-
-        Mockito.when(templateProfileMock.getZoneIdList()).thenReturn(zoneIds);
+        TemplateProfile templateProfileMock = createMockTemplateProfile(zoneIds);
+
         Mockito.doReturn(null).when(_adapter).getImageStoresThrowsExceptionIfNotFound(Mockito.any(Long.class), Mockito.any(TemplateProfile.class));
         Mockito.doReturn(null).when(_templateMgr).verifyHeuristicRulesForZone(Mockito.any(VMTemplateVO.class), Mockito.anyLong());
         Mockito.doNothing().when(_adapter).standardImageStoreAllocation(Mockito.isNull(), Mockito.any(VMTemplateVO.class), Mockito.anyLong());
@@ -364,12 +362,11 @@
 
     @Test
     public void createTemplateWithinZonesTestZoneWithHeuristicRuleShouldCallValidateSecondaryStorageAndCreateTemplate() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
         VMTemplateVO vmTemplateVOMock = Mockito.mock(VMTemplateVO.class);
         DataStore dataStoreMock = Mockito.mock(DataStore.class);
         List<Long> zoneIds = List.of(1L);
-
-        Mockito.when(templateProfileMock.getZoneIdList()).thenReturn(zoneIds);
+        TemplateProfile templateProfileMock = createMockTemplateProfile(zoneIds);
+
         Mockito.doReturn(dataStoreMock).when(_templateMgr).verifyHeuristicRulesForZone(Mockito.any(VMTemplateVO.class), Mockito.anyLong());
         Mockito.doNothing().when(_adapter).validateSecondaryStorageAndCreateTemplate(Mockito.any(List.class), Mockito.any(VMTemplateVO.class), Mockito.any(Map.class), Mockito.anyInt());
 
@@ -555,4 +552,10 @@
         Mockito.when(dataStoreManagerMock.getImageStoresByScope(Mockito.any())).thenReturn(new ArrayList<>());
         _adapter.checkZoneImageStores(templateVO, List.of(1L));
     }
+
+    private static TemplateProfile createMockTemplateProfile(List<Long> zoneIds) {
+        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
+        Mockito.when(templateProfileMock.getZoneIdList()).thenReturn(zoneIds);
+        return templateProfileMock;
+    }
 }
