--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -939,7 +939,7 @@
         HostVO host = Mockito.mock(HostVO.class);
         ClusterVO cluster = Mockito.mock(ClusterVO.class);
         HostPodVO pod = Mockito.mock(HostPodVO.class);
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenterVO("sag4");
 
         Mockito.when(hostDao.findById(hostId)).thenReturn(host);
         Mockito.when(host.getClusterId()).thenReturn(2L);
@@ -952,7 +952,6 @@
         Mockito.when(host.getStorageAccessGroups()).thenReturn("sag1");
         Mockito.when(cluster.getStorageAccessGroups()).thenReturn("sag2");
         Mockito.when(pod.getStorageAccessGroups()).thenReturn("sag3");
-        Mockito.when(zone.getStorageAccessGroups()).thenReturn("sag4");
 
         String[] sags = storageManagerImpl.getStorageAccessGroups(null, null, null, hostId);
 
@@ -1301,8 +1300,7 @@
         Mockito.when(cmd.getStorageId()).thenReturn(null);
         Mockito.when(cmd.getStorageAccessGroups()).thenReturn(storageAccessGroups);
 
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getStorageAccessGroups()).thenReturn("sag2,sag1");
+        DataCenterVO zone = createMockDataCenterVO("sag2,sag1");
         Mockito.when(dataCenterDao.findById(zoneId)).thenReturn(zone);
 
         boolean result = storageManagerImpl.configureStorageAccess(cmd);
@@ -1436,9 +1434,8 @@
         long zoneId = 1L;
         List<String> newStorageAccessGroups = Arrays.asList("group1", "group2");
 
-        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneVO = createMockDataCenterVO("group3,group4");
         Mockito.when(dataCenterDao.findById(zoneId)).thenReturn(zoneVO);
-        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn("group3,group4");
 
         storageManagerImpl.checkIfStorageAccessGroupsExistsOnZone(zoneId, newStorageAccessGroups);
     }
@@ -1448,9 +1445,8 @@
         long zoneId = 1L;
         List<String> newStorageAccessGroups = Arrays.asList("group1", "group2", "group3");
 
-        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneVO = createMockDataCenterVO("group3,group4");
         Mockito.when(dataCenterDao.findById(zoneId)).thenReturn(zoneVO);
-        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn("group3,group4");
 
         CloudRuntimeException thrownException = assertThrows(CloudRuntimeException.class, () -> {
             storageManagerImpl.checkIfStorageAccessGroupsExistsOnZone(zoneId, newStorageAccessGroups);
@@ -1466,12 +1462,10 @@
         List<String> newStorageAccessGroups = Arrays.asList("group1", "group2");
 
         HostPodVO podVO = Mockito.mock(HostPodVO.class);
-        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneVO = createMockDataCenterVO("group5,group6");
 
         Mockito.when(podVO.getDataCenterId()).thenReturn(zoneId);
         Mockito.when(podVO.getStorageAccessGroups()).thenReturn("group3,group4");
-
-        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn("group5,group6");
 
         Mockito.when(podDao.findById(podId)).thenReturn(podVO);
         Mockito.when(dataCenterDao.findById(zoneId)).thenReturn(zoneVO);
@@ -1486,12 +1480,11 @@
         List<String> newStorageAccessGroups = Arrays.asList("group1", "group2", "group3");
 
         HostPodVO podVO = Mockito.mock(HostPodVO.class);
-        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
 
         Mockito.when(podVO.getDataCenterId()).thenReturn(zoneId);
         Mockito.when(podVO.getStorageAccessGroups()).thenReturn("group3,group4");
 
-        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn("group5,group6");
+        DataCenterVO zoneVO = createMockDataCenterVO("group5,group6");
 
         Mockito.when(podDao.findById(podId)).thenReturn(podVO);
         Mockito.when(dataCenterDao.findById(zoneId)).thenReturn(zoneVO);
@@ -1512,15 +1505,13 @@
 
         ClusterVO clusterVO = Mockito.mock(ClusterVO.class);
         HostPodVO podVO = Mockito.mock(HostPodVO.class);
-        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneVO = createMockDataCenterVO("group8,group9");
 
         Mockito.when(clusterVO.getPodId()).thenReturn(podId);
         Mockito.when(clusterVO.getStorageAccessGroups()).thenReturn("group4,group5");
 
         Mockito.when(podVO.getDataCenterId()).thenReturn(zoneId);
         Mockito.when(podVO.getStorageAccessGroups()).thenReturn("group6,group7");
-
-        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn("group8,group9");
 
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(clusterVO);
         Mockito.when(podDao.findById(podId)).thenReturn(podVO);
@@ -1538,15 +1529,13 @@
 
         ClusterVO clusterVO = Mockito.mock(ClusterVO.class);
         HostPodVO podVO = Mockito.mock(HostPodVO.class);
-        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zoneVO = createMockDataCenterVO("group8,group9");
 
         Mockito.when(clusterVO.getPodId()).thenReturn(podId);
         Mockito.when(clusterVO.getStorageAccessGroups()).thenReturn("group4,group5");
 
         Mockito.when(podVO.getDataCenterId()).thenReturn(zoneId);
         Mockito.when(podVO.getStorageAccessGroups()).thenReturn("group6,group7");
-
-        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn("group8,group9");
 
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(clusterVO);
         Mockito.when(podDao.findById(podId)).thenReturn(podVO);
@@ -1727,4 +1716,10 @@
 
         Assert.assertFalse(result);
     }
+
+    private static DataCenterVO createMockDataCenterVO(String storageAccessGroups) {
+        DataCenterVO zoneVO = Mockito.mock(DataCenterVO.class);
+        Mockito.when(zoneVO.getStorageAccessGroups()).thenReturn(storageAccessGroups);
+        return zoneVO;
+    }
 }
