--- a/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
+++ b/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
@@ -144,8 +144,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateVnfNicsStartWithNonzero() {
-        VnfNic nic1 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic1.getDeviceId()).thenReturn(1L);
+        VnfNic nic1 = createMockVnfNic(1L);
 
         VnfNic nic2 = Mockito.mock(VnfNic.class);
 
@@ -162,8 +161,7 @@
         Mockito.when(nic1.getDeviceId()).thenReturn(0L);
         Mockito.when(nic1.isRequired()).thenReturn(true);
 
-        VnfNic nic2 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic2.getDeviceId()).thenReturn(2L);
+        VnfNic nic2 = createMockVnfNic(2L);
 
         VnfNic nic3 = Mockito.mock(VnfNic.class);
 
@@ -258,4 +256,10 @@
         cidrList.add("10.10.10.0/33");
         VnfTemplateUtils.validateVnfCidrList(cidrList);
     }
+
+    private static VnfNic createMockVnfNic(long deviceIdReturn) {
+        VnfNic nic2 = Mockito.mock(VnfNic.class);
+        Mockito.when(nic2.getDeviceId()).thenReturn(deviceIdReturn);
+        return nic2;
+    }
 }
