--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxElementTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxElementTest.java
@@ -374,12 +374,10 @@
         LoadBalancingRule.LbDestination destination = new LoadBalancingRule.LbDestination(6443, 6443, "172.30.110.11", false);
         LoadBalancingRule rule = new LoadBalancingRule(lb, List.of(destination), null, null, new Ip("10.1.13.10"), null, "TCP");
 
-        VpcVO vpc = Mockito.mock(VpcVO.class);
+        VpcVO vpc = MockVpcVO.createMockVpcVO(5L, 2L);
 
         IPAddressVO ipAddress = new IPAddressVO(new Ip("10.1.13.10"), 1L, 1L, 1L,false);
         when(vpcDao.findById(anyLong())).thenReturn(vpc);
-        when(vpc.getDomainId()).thenReturn(2L);
-        when(vpc.getAccountId()).thenReturn(5L);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddress);
         when(nsxService.createLbRule(any(NsxNetworkRule.class))).thenReturn(true);
 
@@ -397,12 +395,10 @@
         LoadBalancingRule.LbDestination destination = new LoadBalancingRule.LbDestination(6443, 6443, "172.30.110.11", false);
         LoadBalancingRule rule = new LoadBalancingRule(lb, List.of(destination), null, null, new Ip("10.1.13.10"), null, "TCP");
 
-        VpcVO vpc = Mockito.mock(VpcVO.class);
+        VpcVO vpc = MockVpcVO.createMockVpcVO(5L, 2L);
 
         IPAddressVO ipAddress = new IPAddressVO(new Ip("10.1.13.10"), 1L, 1L, 1L,false);
         when(vpcDao.findById(anyLong())).thenReturn(vpc);
-        when(vpc.getDomainId()).thenReturn(2L);
-        when(vpc.getAccountId()).thenReturn(5L);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddress);
         when(nsxService.deleteLbRule(any(NsxNetworkRule.class))).thenReturn(true);
 
--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxServiceImplTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxServiceImplTest.java
@@ -117,14 +117,12 @@
 
     @Test
     public void testUpdateVpcSourceNatIp() {
-        VpcVO vpc = mock(VpcVO.class);
         IpAddress ipAddress = mock(IpAddress.class);
         Ip ip = Mockito.mock(Ip.class);
         when(ip.addr()).thenReturn("10.1.10.10");
         when(ipAddress.getAddress()).thenReturn(ip);
         long vpcId = 1L;
-        when(vpc.getAccountId()).thenReturn(accountId);
-        when(vpc.getDomainId()).thenReturn(domainId);
+        VpcVO vpc = MockVpcVO.createMockVpcVO(accountId, domainId);
         when(vpc.getZoneId()).thenReturn(zoneId);
         when(vpc.getId()).thenReturn(vpcId);
         NsxAnswer answer = mock(NsxAnswer.class);
--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/MockVpcVO.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/MockVpcVO.java
@@ -0,0 +1,16 @@
+package org.apache.cloudstack.service;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.cloud.network.vpc.VpcVO;
+
+public class MockVpcVO {
+
+    public static VpcVO createMockVpcVO(long accountId, long domainId) {
+        VpcVO vpc = mock(VpcVO.class);
+        when(vpc.getAccountId()).thenReturn(accountId);
+        when(vpc.getDomainId()).thenReturn(domainId);
+        return vpc;
+    }
+}
