--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -295,6 +295,8 @@
     private HighAvailabilityManager _haMgr;
     @Mock
     VirtualMachineGuru guru;
+    @Mock
+    private ItWorkVO itWork;
 
     private ConfigDepotImpl configDepotImpl;
     private boolean updatedConfigKeyDepot = false;
@@ -1128,7 +1130,6 @@
 
         ReservationContext ctx = mock(ReservationContext.class);
 
-        ItWorkVO work = mock(ItWorkVO.class);
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
@@ -1147,7 +1148,7 @@
 
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
-        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
+        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, itWork);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
@@ -1224,7 +1225,6 @@
 
         ReservationContext ctx = mock(ReservationContext.class);
 
-        ItWorkVO work = mock(ItWorkVO.class);
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
@@ -1243,7 +1243,7 @@
 
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
-        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
+        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, itWork);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
