--- a/server/src/test/java/com/cloud/resource/RollingMaintenanceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/RollingMaintenanceManagerImplTest.java
@@ -124,6 +124,12 @@
         closeable.close();
     }
 
+    private static ServiceOfferingVO createMockServiceOffering(boolean dynamic) {
+        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
+        Mockito.when(serviceOffering.isDynamic()).thenReturn(dynamic);
+        return serviceOffering;
+    }
+
     private void checkResults(Map<Long, List<Host>> result) {
         Assert.assertEquals(2, result.size());
         Assert.assertTrue(result.containsKey(clusterId1));
