--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachineTemplate template) {
+        VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
+        when(vmProfile.getTemplate()).thenReturn(template);
+        return vmProfile;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1669,9 +1675,8 @@
     @Test
     public void processPrepareExternalProvisioning_nonExternalHypervisor_noAction() throws OperationTimedoutException, AgentUnavailableException {
         Host host = mock(Host.class);
-        VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
-        when(vmProfile.getTemplate()).thenReturn(template);
+        VirtualMachineProfile vmProfile = createMockVirtualMachineProfile(template);
         when(host.getHypervisorType()).thenReturn(HypervisorType.KVM);
         virtualMachineManagerImpl.processPrepareExternalProvisioning(true, host, vmProfile, mock(DataCenter.class));
         verify(agentManagerMock, never()).send(anyLong(), any(Command.class));
@@ -1680,9 +1685,8 @@
     @Test
     public void processPrepareExternalProvisioning_externalHypervisor_sendsCommand() throws OperationTimedoutException, AgentUnavailableException {
         Host host = mock(Host.class);
-        VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
-        when(vmProfile.getTemplate()).thenReturn(template);
+        VirtualMachineProfile vmProfile = createMockVirtualMachineProfile(template);
         NicTO[] nics = new NicTO[]{mock(NicTO.class)};
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         when(vmTO.getNics()).thenReturn(nics);
