--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -151,6 +151,8 @@
     @Mock
     AgentManager agentMgr;
 
+    private final NicVO deallocatedNicVO = mock(NicVO.class);
+
     TungstenGuestNetworkGuru guru;
 
     AutoCloseable closeable;
@@ -244,12 +246,11 @@
         final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        final NicVO nicVO = mock(NicVO.class);
 
         when(network.getDataCenterId()).thenReturn(1L);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
-        when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
+        when(nicDao.listByVmId(anyLong())).thenReturn(List.of(deallocatedNicVO));
 
         guru.deallocate(network, nicProfile, virtualMachineProfile);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong());
@@ -261,11 +262,10 @@
         final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        final NicVO nicVO = mock(NicVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenThrow(
             new IllegalArgumentException());
-        when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
+        when(nicDao.listByVmId(anyLong())).thenReturn(List.of(deallocatedNicVO));
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
 
