--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
@@ -143,12 +143,9 @@
     public void implementTest() throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
         URI uri = URI.create("vlan://123");
 
-        Network network = mock(Network.class);
-        when(network.getId()).thenReturn(1L);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Vlan);
+        Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Vlan, uri, 1L);
         when(network.getDataCenterId()).thenReturn(1L);
         when(network.getGateway()).thenReturn("1.1.1.1");
-        when(network.getBroadcastUri()).thenReturn(uri);
         when(network.getCidr()).thenReturn("1.1.1.0/24");
 
         NetworkOffering offering = mock(NetworkOffering.class);
@@ -236,11 +233,8 @@
     public void shutdownTest() throws ConcurrentOperationException, ResourceUnavailableException {
         URI uri = URI.create("vlan://123");
 
-        Network network = mock(Network.class);
-        when(network.getId()).thenReturn(1L);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Vlan);
-        when(network.getDataCenterId()).thenReturn(1L);
-        when(network.getBroadcastUri()).thenReturn(uri);
+        Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Vlan, uri, 1L);
+        when(network.getDataCenterId()).thenReturn(1L);
 
         ReservationContext context = mock(ReservationContext.class);
 
@@ -266,11 +260,8 @@
     public void applyFWRulesTest() throws ResourceUnavailableException {
         URI uri = URI.create("vlan://123");
 
-        Network network = mock(Network.class);
-        when(network.getId()).thenReturn(1L);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Vlan);
-        when(network.getDataCenterId()).thenReturn(1L);
-        when(network.getBroadcastUri()).thenReturn(uri);
+        Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Vlan, uri, 1L);
+        when(network.getDataCenterId()).thenReturn(1L);
         when(network.getCidr()).thenReturn("1.1.1.0/24");
         when(network.getState()).thenReturn(Network.State.Implemented);
 
@@ -310,11 +301,8 @@
     public void applyPRulesTest() throws ResourceUnavailableException {
         URI uri = URI.create("vlan://123");
 
-        Network network = mock(Network.class);
-        when(network.getId()).thenReturn(1L);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Vlan);
-        when(network.getDataCenterId()).thenReturn(1L);
-        when(network.getBroadcastUri()).thenReturn(uri);
+        Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Vlan, uri, 1L);
+        when(network.getDataCenterId()).thenReturn(1L);
         when(network.getCidr()).thenReturn("1.1.1.0/24");
         when(network.getState()).thenReturn(Network.State.Implemented);
 
@@ -360,11 +348,8 @@
     public void applyStaticNatsTest() throws ResourceUnavailableException {
         URI uri = URI.create("vlan://123");
 
-        Network network = mock(Network.class);
-        when(network.getId()).thenReturn(1L);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Vlan);
-        when(network.getDataCenterId()).thenReturn(1L);
-        when(network.getBroadcastUri()).thenReturn(uri);
+        Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Vlan, uri, 1L);
+        when(network.getDataCenterId()).thenReturn(1L);
         when(network.getCidr()).thenReturn("1.1.1.0/24");
         when(network.getState()).thenReturn(Network.State.Implemented);
 
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -152,10 +152,7 @@
 
     @Test
     public void implementIsolatedNetworkTest() throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException, URISyntaxException {
-        final Network network = mock(Network.class);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Lswitch);
-        when(network.getBroadcastUri()).thenReturn(new URI("lswitch:aaaaa"));
-        when(network.getId()).thenReturn(NETWORK_ID);
+        final Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Lswitch, new URI("lswitch:aaaaa"), NETWORK_ID);
         when(network.getPhysicalNetworkId()).thenReturn(NETWORK_ID);
         when(network.getGuestType()).thenReturn(GuestType.Isolated);
 
@@ -268,10 +265,7 @@
     @Test
     public void implementSharedNetworkUuidVlanIdTest() throws URISyntaxException, ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
         // SHARED NETWORKS CASE 1: loggerICAL ROUTER'S UUID AS VLAN ID
-        final Network network = mock(Network.class);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Lswitch);
-        when(network.getBroadcastUri()).thenReturn(new URI("lswitch:aaaaa"));
-        when(network.getId()).thenReturn(NETWORK_ID);
+        final Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Lswitch, new URI("lswitch:aaaaa"), NETWORK_ID);
         when(network.getPhysicalNetworkId()).thenReturn(NETWORK_ID);
         when(network.getGuestType()).thenReturn(GuestType.Shared);
 
@@ -321,10 +315,7 @@
     @Test
     public void implementSharedNetworkNumericalVlanIdTest() throws URISyntaxException, ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
         // SHARED NETWORKS CASE 2: NUMERICAL VLAN ID
-        final Network network = mock(Network.class);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Lswitch);
-        when(network.getBroadcastUri()).thenReturn(new URI("lswitch:aaaaa"));
-        when(network.getId()).thenReturn(NETWORK_ID);
+        final Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Lswitch, new URI("lswitch:aaaaa"), NETWORK_ID);
         when(network.getPhysicalNetworkId()).thenReturn(NETWORK_ID);
         when(network.getGuestType()).thenReturn(GuestType.Shared);
 
@@ -373,10 +364,7 @@
 
     @Test(expected=CloudRuntimeException.class)
     public void implementSharedNetworkNumericalVlanIdWithoutL2GatewayService() throws URISyntaxException, ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException{
-        final Network network = mock(Network.class);
-        when(network.getBroadcastDomainType()).thenReturn(BroadcastDomainType.Lswitch);
-        when(network.getBroadcastUri()).thenReturn(new URI("lswitch:aaaaa"));
-        when(network.getId()).thenReturn(NETWORK_ID);
+        final Network network = MockNetwork.createMockNetwork(BroadcastDomainType.Lswitch, new URI("lswitch:aaaaa"), NETWORK_ID);
         when(network.getPhysicalNetworkId()).thenReturn(NETWORK_ID);
         when(network.getGuestType()).thenReturn(GuestType.Shared);
 
--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/MockNetwork.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/MockNetwork.java
@@ -0,0 +1,19 @@
+package com.cloud.network.element;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+
+import com.cloud.network.Network;
+import com.cloud.network.Networks.BroadcastDomainType;
+
+public class MockNetwork {
+    public static Network createMockNetwork(BroadcastDomainType broadcastDomainType, URI broadcastUri, long networkId) {
+        Network network = mock(Network.class);
+        when(network.getBroadcastDomainType()).thenReturn(broadcastDomainType);
+        when(network.getBroadcastUri()).thenReturn(broadcastUri);
+        when(network.getId()).thenReturn(networkId);
+        return network;
+    }
+}
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/MockNetwork.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/MockNetwork.java
@@ -0,0 +1,19 @@
+package com.cloud.network.element;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+
+import com.cloud.network.Network;
+import com.cloud.network.Networks.BroadcastDomainType;
+
+public class MockNetwork {
+    public static Network createMockNetwork(BroadcastDomainType broadcastDomainType, URI broadcastUri, long networkId) {
+        Network network = mock(Network.class);
+        when(network.getBroadcastDomainType()).thenReturn(broadcastDomainType);
+        when(network.getBroadcastUri()).thenReturn(broadcastUri);
+        when(network.getId()).thenReturn(networkId);
+        return network;
+    }
+}
