--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -273,6 +273,8 @@
     DataCenterDao dataCenterDao;
     @Mock
     IpAddressManager ipAddressManager;
+    @Mock
+    private DataCenterVO dataCenterVOMock;
 
     TungstenServiceImpl tungstenService;
 
@@ -877,10 +879,9 @@
         TungstenAnswer createTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenPolicyCommand.class), anyLong())).thenReturn(createTungstenPolicyAnswer);
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(createTungstenPolicyAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
         when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
 
@@ -897,10 +898,9 @@
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(AddTungstenPolicyRuleCommand.class), anyLong())).thenReturn(addTungstenPolicyRuleAnswer);
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(addTungstenPolicyRuleAnswer.getApiObjectBase()).thenReturn(networkPolicy);
         when(networkPolicy.getEntries()).thenReturn(policyEntriesType);
         when(policyEntriesType.getPolicyRule()).thenReturn(List.of(policyRuleType));
@@ -932,9 +932,8 @@
         TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(networkDao.findById(anyLong())).thenReturn(networkVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenPolicyCommand.class), anyLong())).thenReturn(listTungstenPolicyAnswer);
         when(listTungstenPolicyAnswer.getTungstenModelList()).thenReturn(List.of(tungstenNetworkPolicy));
@@ -949,9 +948,8 @@
     public void listTungstenNetworkTest() {
         TungstenAnswer listTungstenNetworkAnswer = MockTungstenAnswerFactory.get(true);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenNetworkCommand.class), anyLong())).thenReturn(listTungstenNetworkAnswer);
         doReturn(List.of(virtualNetwork)).when(listTungstenNetworkAnswer).getApiObjectBaseList();
         when((virtualNetwork.getName())).thenReturn("guestNetwork1");
@@ -963,9 +961,8 @@
     public void listTungstenNicTest() {
         TungstenAnswer listTungstenNicAnswer = MockTungstenAnswerFactory.get(true);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenNicCommand.class), anyLong())).thenReturn(listTungstenNicAnswer);
         doReturn(List.of(virtualMachineInterface)).when(listTungstenNicAnswer).getApiObjectBaseList();
 
@@ -976,9 +973,8 @@
     public void listTungstenVmTest() {
         TungstenAnswer listTungstenVmAnswer = MockTungstenAnswerFactory.get(true);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenVmCommand.class), anyLong())).thenReturn(listTungstenVmAnswer);
         doReturn(List.of(virtualMachine)).when(listTungstenVmAnswer).getApiObjectBaseList();
 
@@ -1004,9 +1000,8 @@
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenPolicyRuleCommand.class), anyLong())).thenReturn(listTungstenPolicyRuleAnswer);
         doReturn(networkPolicy).when(listTungstenPolicyRuleAnswer).getApiObjectBase();
         when(networkPolicy.getEntries()).thenReturn(policyEntriesType);
@@ -1033,9 +1028,8 @@
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenPolicyRuleCommand.class), anyLong())).thenReturn(listTungstenPolicyRuleAnswer);
         doReturn(networkPolicy).when(listTungstenPolicyRuleAnswer).getApiObjectBase();
         when(networkPolicy.getEntries()).thenReturn(policyEntriesType);
@@ -1058,9 +1052,8 @@
         TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenPolicyRuleCommand.class), anyLong())).thenReturn(removeTungstenPolicyRuleAnswer);
         when(removeTungstenPolicyRuleAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
         when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
@@ -1078,9 +1071,8 @@
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenTagCommand.class), anyLong())).thenReturn(createTungstenTagAnswer);
         when(createTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
         when(tungstenTag.getTag()).thenReturn(tag);
@@ -1096,10 +1088,9 @@
     public void createTungstenTagTypeTest() {
         TungstenAnswer createTungstenTagTypeAnswer = MockTungstenAnswerFactory.get(true);
         TagType tagtype = mock(TagType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenTagTypeCommand.class), anyLong())).thenReturn(createTungstenTagTypeAnswer);
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(createTungstenTagTypeAnswer.getApiObjectBase()).thenReturn(tagtype);
 
         assertNotNull(tungstenService.createTungstenTagType(1L, "testTagType"));
@@ -1135,9 +1126,8 @@
     public void listTungstenTagTypesTest() {
         TungstenAnswer listTungstenTagTypeAnswer = MockTungstenAnswerFactory.get(true);
         TagType tagtype = mock(TagType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenTagTypeCommand.class), anyLong())).thenReturn(listTungstenTagTypeAnswer);
         doReturn(List.of(tagtype)).when(listTungstenTagTypeAnswer).getApiObjectBaseList();
 
@@ -1168,9 +1158,8 @@
         TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(applyTungstenPolicyAnswer);
         when(applyTungstenPolicyAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
         when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
@@ -1188,9 +1177,8 @@
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenTagCommand.class), anyLong())).thenReturn(applyTungstenTagAnswer);
         when(applyTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
         when(tungstenTag.getTag()).thenReturn(tag);
@@ -1210,9 +1198,8 @@
         TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenPolicyCommand.class), anyLong())).thenReturn(removeTungstenPolicyAnswer);
         when(removeTungstenPolicyAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
         when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
@@ -1230,9 +1217,8 @@
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenTagCommand.class), anyLong())).thenReturn(removeTungstenTagAnswer);
         when(removeTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
         when(tungstenTag.getTag()).thenReturn(tag);
@@ -1253,9 +1239,8 @@
         AddressGroup addressGroup = mock(AddressGroup.class);
         SubnetListType subnetListType = mock(SubnetListType.class);
         SubnetType subnetType = mock(SubnetType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenAddressGroupCommand.class), anyLong())).thenReturn(createTungstenAddressGroupAnswer);
         when(createTungstenAddressGroupAnswer.getApiObjectBase()).thenReturn(addressGroup);
         when(addressGroup.getPrefix()).thenReturn(subnetListType);
@@ -1271,9 +1256,8 @@
         FirewallServiceGroupType firewallServiceGroupType = mock(FirewallServiceGroupType.class);
         FirewallServiceType firewallServiceType = mock(FirewallServiceType.class);
         PortType portType = mock(PortType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenServiceGroupCommand.class), anyLong())).thenReturn(createTungstenServiceGroupAnswer);
         when(createTungstenServiceGroupAnswer.getApiObjectBase()).thenReturn(serviceGroup);
         when(serviceGroup.getFirewallServiceList()).thenReturn(firewallServiceGroupType);
@@ -1292,9 +1276,8 @@
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
         FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenFirewallRuleCommand.class), anyLong())).thenReturn(createTungstenFirewallRuleAnswer);
         when(createTungstenFirewallRuleAnswer.getApiObjectBase()).thenReturn(firewallRule);
         when(firewallRule.getActionList()).thenReturn(actionListType);
@@ -1318,9 +1301,8 @@
         TungstenAnswer createTungstenFirewallPolicyAnswer = MockTungstenAnswerFactory.get(true);
         FirewallPolicy firewallPolicy = mock(FirewallPolicy.class);
         ObjectReference<FirewallSequence> firewallSequenceObjectReference = mock(ObjectReference.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenFirewallPolicyCommand.class), anyLong())).thenReturn(createTungstenFirewallPolicyAnswer);
         when(createTungstenFirewallPolicyAnswer.getApiObjectBase()).thenReturn(firewallPolicy);
         when(firewallPolicy.getFirewallRule()).thenReturn(List.of(firewallSequenceObjectReference));
@@ -1335,9 +1317,8 @@
         ApplicationPolicySet applicationPolicySet = mock(ApplicationPolicySet.class);
         ObjectReference<ApiPropertyBase> objectReference = mock(ObjectReference.class);
         ObjectReference<FirewallSequence> firewallSequenceObjectReference = mock(ObjectReference.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenApplicationPolicySetCommand.class), anyLong())).thenReturn(createTungstenApplicationPolicySetAnswer);
         when(createTungstenApplicationPolicySetAnswer.getApiObjectBase()).thenReturn(applicationPolicySet);
         when(applicationPolicySet.getTag()).thenReturn(List.of(objectReference));
@@ -1354,9 +1335,8 @@
         ApplicationPolicySet applicationPolicySet = mock(ApplicationPolicySet.class);
         ObjectReference<ApiPropertyBase> objectReference = mock(ObjectReference.class);
         ObjectReference<FirewallSequence> firewallSequenceObjectReference = mock(ObjectReference.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenApplicationPolicySetCommand.class), anyLong())).thenReturn(listTungstenApplicationPolicySetAnswer);
         doReturn(List.of(applicationPolicySet)).when(listTungstenApplicationPolicySetAnswer).getApiObjectBaseList();
         when(applicationPolicySet.getTag()).thenReturn(List.of(objectReference));
@@ -1372,9 +1352,8 @@
         TungstenAnswer listTungstenFirewallPolicyAnswer = MockTungstenAnswerFactory.get(true);
         FirewallPolicy firewallPolicy = mock(FirewallPolicy.class);
         ObjectReference<FirewallSequence> firewallSequenceObjectReference = mock(ObjectReference.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenFirewallPolicyCommand.class), anyLong())).thenReturn(listTungstenFirewallPolicyAnswer);
         doReturn(List.of(firewallPolicy)).when(listTungstenFirewallPolicyAnswer).getApiObjectBaseList();
         when(firewallPolicy.getFirewallRule()).thenReturn(List.of(firewallSequenceObjectReference));
@@ -1392,9 +1371,8 @@
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
         FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenFirewallRuleCommand.class), anyLong())).thenReturn(listTungstenFirewallRuleAnswer);
         doReturn(List.of(firewallRule)).when(listTungstenFirewallRuleAnswer).getApiObjectBaseList();
         when(firewallRule.getActionList()).thenReturn(actionListType);
@@ -1418,9 +1396,8 @@
         AddressGroup addressGroup = mock(AddressGroup.class);
         SubnetListType subnetListType = mock(SubnetListType.class);
         SubnetType subnetType = mock(SubnetType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenAddressGroupCommand.class), anyLong())).thenReturn(listTungstenAddressGroupAnswer);
         doReturn(List.of(addressGroup)).when(listTungstenAddressGroupAnswer).getApiObjectBaseList();
         when(addressGroup.getPrefix()).thenReturn(subnetListType);
@@ -1437,9 +1414,8 @@
         FirewallServiceGroupType firewallServiceGroupType = mock(FirewallServiceGroupType.class);
         FirewallServiceType firewallServiceType = mock(FirewallServiceType.class);
         PortType portType = mock(PortType.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenServiceGroupCommand.class), anyLong())).thenReturn(listTungstenServiceGroupAnswer);
         doReturn(List.of(serviceGroup)).when(listTungstenServiceGroupAnswer).getApiObjectBaseList();
         when(serviceGroup.getFirewallServiceList()).thenReturn(firewallServiceGroupType);
@@ -1536,7 +1512,6 @@
     @Test
     public void addTungstenVmSecurityGroupTest() {
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
         SecurityGroupVO securityGroupVO = mock(SecurityGroupVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenAnswer getTungstenSecurityGroupAnswer = mock(TungstenAnswer.class);
@@ -1547,8 +1522,8 @@
         TungstenSecurityGroupRuleVO tungstenSecurityGroupRuleVO = mock(TungstenSecurityGroupRuleVO.class);
         TungstenAnswer addTungstenSecurityGroupRuleAnswer = MockTungstenAnswerFactory.get(true);
 
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
-        when(dataCenterVO.isSecurityGroupEnabled()).thenReturn(true);
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
+        when(dataCenterVOMock.isSecurityGroupEnabled()).thenReturn(true);
         when(securityGroupManager.getSecurityGroupsForVm(anyLong())).thenReturn(List.of(securityGroupVO));
         when(tungstenProviderDao.findAll()).thenReturn(List.of(tungstenProviderVO));
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenSecurityGroupCommand.class), anyLong())).thenReturn(getTungstenSecurityGroupAnswer);
@@ -1591,9 +1566,8 @@
         TungstenLogicalRouter tungstenLogicalRouter = mock(TungstenLogicalRouter.class);
         LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenRoutingLogicalRouterCommand.class), anyLong())).thenReturn(tungstenAnswer);
         when(tungstenAnswer.getTungstenModel()).thenReturn(tungstenLogicalRouter);
         when(tungstenLogicalRouter.getLogicalRouter()).thenReturn(logicalRouter);
@@ -1609,9 +1583,8 @@
         TungstenLogicalRouter tungstenLogicalRouter = mock(TungstenLogicalRouter.class);
         LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(networkDao.findByUuid(anyString())).thenReturn(networkVO);
         when(ipAddressManager.acquireLastGuestIpAddress(any(Network.class))).thenReturn("192.168.100.100");
         when(tungstenFabricUtils.sendTungstenCommand(any(AddTungstenNetworkGatewayToLogicalRouterCommand.class), anyLong())).thenReturn(tungstenAnswer);
@@ -1628,9 +1601,8 @@
         TungstenLogicalRouter tungstenLogicalRouter = mock(TungstenLogicalRouter.class);
         LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenRoutingLogicalRouterCommand.class), anyLong())).thenReturn(tungstenAnswer);
         when(tungstenAnswer.getTungstenModelList()).thenReturn(List.of(tungstenLogicalRouter));
         when(tungstenLogicalRouter.getLogicalRouter()).thenReturn(logicalRouter);
@@ -1647,9 +1619,8 @@
         TungstenLogicalRouter tungstenLogicalRouter = mock(TungstenLogicalRouter.class);
         LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-
-        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+
+        when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVOMock);
         when(networkDao.findByUuid(anyString())).thenReturn(networkVO);
         when(tungstenGuestNetworkIpAddressDao.findByNetworkAndLogicalRouter(anyLong(), anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenNetworkGatewayFromLogicalRouterCommand.class), anyLong())).thenReturn(tungstenAnswer);
