--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -261,6 +261,8 @@
     @Mock
     private Map<String, String> details;
 
+    private final FormatInfo formatInfo = Mockito.mock(FormatInfo.class);
+
     private static final String PHYSICAL_DISK_PATH = "/path/to/disk";
     private static final int DEV_ID = 1;
     private static final DiskDef.DiskBus DISK_BUS_TYPE = DiskDef.DiskBus.VIRTIO;
@@ -4237,7 +4239,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
         final Processor qcow2Processor = Mockito.mock(Processor.class);
-        final FormatInfo info = Mockito.mock(FormatInfo.class);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
 
@@ -4264,7 +4265,7 @@
 
         try {
             when(libvirtUtilitiesHelper.buildQCOW2Processor(storage)).thenReturn(qcow2Processor);
-            when(qcow2Processor.process(tmplPath, null, tmplName)).thenReturn(info);
+            when(qcow2Processor.process(tmplPath, null, tmplName)).thenReturn(formatInfo);
         } catch (final ConfigurationException e) {
             fail(e.getMessage());
         } catch (final InternalErrorException e) {
@@ -4311,7 +4312,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
         final Processor qcow2Processor = Mockito.mock(Processor.class);
-        final FormatInfo info = Mockito.mock(FormatInfo.class);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
 
@@ -4453,7 +4453,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final TemplateLocation location = Mockito.mock(TemplateLocation.class);
         final Processor qcow2Processor = Mockito.mock(Processor.class);
-        final FormatInfo info = Mockito.mock(FormatInfo.class);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
 
@@ -4480,7 +4479,7 @@
 
         try {
             when(libvirtUtilitiesHelper.buildQCOW2Processor(storage)).thenReturn(qcow2Processor);
-            when(qcow2Processor.process(tmplPath, null, tmplName)).thenReturn(info);
+            when(qcow2Processor.process(tmplPath, null, tmplName)).thenReturn(formatInfo);
 
             when(location.create(1, true, tmplName)).thenThrow(IOException.class);
 
