--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -995,10 +995,9 @@
 
     @Test
     public void testValidateAutoScaleCounters() {
-        Counter counterCpuMock = Mockito.mock(Counter.class);
+        Counter counterCpuMock = createMockMemoryCounter();
         when(counterCpuMock.getSource()).thenReturn(Counter.Source.CPU);
-        Counter counterMemoryMock = Mockito.mock(Counter.class);
-        when(counterMemoryMock.getSource()).thenReturn(Counter.Source.MEMORY);
+        Counter counterMemoryMock = createMockMemoryCounter();
 
         List<Counter> counters = Arrays.asList(counterCpuMock, counterMemoryMock);
 
@@ -1016,8 +1015,7 @@
     public void testValidateAutoScaleCountersFail() {
         Counter counterCpuMock = Mockito.mock(Counter.class);
         when(counterCpuMock.getSource()).thenReturn(Counter.Source.CPU);
-        Counter counterMemoryMock = Mockito.mock(Counter.class);
-        when(counterMemoryMock.getSource()).thenReturn(Counter.Source.MEMORY);
+        Counter counterMemoryMock = createMockMemoryCounter();
 
         List<Counter> counters = Arrays.asList(counterCpuMock, counterMemoryMock);
 
@@ -2581,4 +2579,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static Counter createMockMemoryCounter() {
+        Counter counterMemoryMock = Mockito.mock(Counter.class);
+        when(counterMemoryMock.getSource()).thenReturn(Counter.Source.MEMORY);
+        return counterMemoryMock;
+    }
 }
