--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessorTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessorTest.java
@@ -23,16 +23,23 @@
 import java.util.List;
 
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
 import com.cloud.agent.api.to.SwiftTO;
 
 public class XenServerStorageProcessorTest {
+    private CitrixResourceBase resourceMock;
+
+    @Before
+    public void setUp() {
+        resourceMock = Mockito.mock(CitrixResourceBase.class);
+    }
+
     @Test
     public void testOrderOfSwiftUplodScriptParamsWithoutStoragePolicy() {
-        CitrixResourceBase resource = Mockito.mock(CitrixResourceBase.class);
-        XenServerStorageProcessor mock = new XenServerStorageProcessor(resource);
+        XenServerStorageProcessor mock = new XenServerStorageProcessor(resourceMock);
 
         SwiftTO swift = Mockito.mock(SwiftTO.class);
         when(swift.getStoragePolicy()).thenReturn(null);
@@ -71,8 +78,7 @@
 
     @Test
     public void testOrderOfSwiftUplodScriptParamsWithStoragePolicy() {
-        CitrixResourceBase resource = Mockito.mock(CitrixResourceBase.class);
-        XenServerStorageProcessor mock = new XenServerStorageProcessor(resource);
+        XenServerStorageProcessor mock = new XenServerStorageProcessor(resourceMock);
 
         SwiftTO swift = Mockito.mock(SwiftTO.class);
         when(swift.getStoragePolicy()).thenReturn("sample-storagepolicy");
