--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -1077,13 +1077,9 @@
         when(volumeInfo2.getDeviceId()).thenReturn(2L);
         when(backup.getBackedUpVolumes()).thenReturn(List.of(volumeInfo0, volumeInfo1, volumeInfo2));
 
-        DiskOfferingVO diskOffering1 = mock(DiskOfferingVO.class);
-        when(diskOffering1.getUuid()).thenReturn("disk-offering-uuid-1");
-        when(diskOffering1.getState()).thenReturn(DiskOffering.State.Active);
-
-        DiskOfferingVO diskOffering2 = mock(DiskOfferingVO.class);
-        when(diskOffering2.getUuid()).thenReturn("disk-offering-uuid-2");
-        when(diskOffering2.getState()).thenReturn(DiskOffering.State.Active);
+        DiskOfferingVO diskOffering1 = createMockDiskOffering("disk-offering-uuid-1");
+
+        DiskOfferingVO diskOffering2 = createMockDiskOffering("disk-offering-uuid-2");
 
         when(diskOfferingDao.findByUuid("disk-offering-uuid-1")).thenReturn(diskOffering1);
         when(diskOfferingDao.findByUuid("disk-offering-uuid-2")).thenReturn(diskOffering2);
@@ -1117,9 +1113,7 @@
         when(volumeInfo1.getDeviceId()).thenReturn(1L);
         when(backup.getBackedUpVolumes()).thenReturn(List.of(volumeInfo1));
 
-        DiskOfferingVO diskOffering = mock(DiskOfferingVO.class);
-        when(diskOffering.getUuid()).thenReturn("disk-offering-uuid-1");
-        when(diskOffering.getState()).thenReturn(DiskOffering.State.Active);
+        DiskOfferingVO diskOffering = createMockDiskOffering("disk-offering-uuid-1");
 
         when(diskOfferingDao.findByUuid("disk-offering-uuid-1")).thenReturn(diskOffering);
 
@@ -2623,6 +2617,13 @@
         when(offering.getId()).thenReturn(id);
         when(offering.getName()).thenReturn(name);
         return offering;
+    }
+
+    private static DiskOfferingVO createMockDiskOffering(String uuid) {
+        DiskOfferingVO diskOffering = mock(DiskOfferingVO.class);
+        when(diskOffering.getUuid()).thenReturn(uuid);
+        when(diskOffering.getState()).thenReturn(DiskOffering.State.Active);
+        return diskOffering;
     }
 
     @Test
