--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorkerTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorkerTest.java
@@ -96,11 +96,9 @@
         Mockito.when(kubernetesCluster.getId()).thenReturn(kubernetesClusterId);
         Mockito.when(kubernetesCluster.getControlNodeCount()).thenReturn(controlNodes);
 
-        ServiceOfferingVO existingOffering = Mockito.mock(ServiceOfferingVO.class);
         int existingCores = 2;
         int existingMemory = 2048;
-        Mockito.when(existingOffering.getCpu()).thenReturn(existingCores);
-        Mockito.when(existingOffering.getRamSize()).thenReturn(existingMemory);
+        ServiceOfferingVO existingOffering = createMockServiceOffering(existingCores, existingMemory);
         int remainingClusterCpu = 8;
         int remainingClusterMemory = 12288;
         Mockito.when(kubernetesCluster.getCores()).thenReturn(remainingClusterCpu + (controlNodes * existingCores));
@@ -108,11 +106,9 @@
 
         Mockito.when(serviceOfferingDao.findById(1L)).thenReturn(existingOffering);
 
-        ServiceOfferingVO newOffering = Mockito.mock(ServiceOfferingVO.class);
         int newCores = 4;
         int newMemory = 2048;
-        Mockito.when(newOffering.getCpu()).thenReturn(newCores);
-        Mockito.when(newOffering.getRamSize()).thenReturn(newMemory);
+        ServiceOfferingVO newOffering = createMockServiceOffering(newCores, newMemory);
 
         KubernetesClusterVmMapVO controlNodeVM1 = Mockito.mock(KubernetesClusterVmMapVO.class);
         Mockito.when(controlNodeVM1.getVmId()).thenReturn(10L);
@@ -187,4 +183,11 @@
 
         Assert.assertTrue(toRemove.isEmpty());
     }
+
+    private static ServiceOfferingVO createMockServiceOffering(int existingCores, int existingMemory) {
+        ServiceOfferingVO existingOffering = Mockito.mock(ServiceOfferingVO.class);
+        Mockito.when(existingOffering.getCpu()).thenReturn(existingCores);
+        Mockito.when(existingOffering.getRamSize()).thenReturn(existingMemory);
+        return existingOffering;
+    }
 }
