--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -153,7 +153,7 @@
     @Test
     public void testDesign() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -178,7 +178,7 @@
     @Test
     public void testDesignNoIsolationMethodVCS() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -198,7 +198,7 @@
     @Test
     public void testDesignNoConnectivityInOffering() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -220,7 +220,7 @@
     @Test
     public void testImplement() throws InsufficientVirtualNetworkCapacityException, URISyntaxException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -279,7 +279,7 @@
     @Test
     public void testImplementFail() throws InsufficientVirtualNetworkCapacityException, URISyntaxException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -147,7 +147,7 @@
     @Test
     public void testDesign() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -176,7 +176,7 @@
     @Test
     public void testDesignNoElementOnPhysicalNetwork() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -198,7 +198,7 @@
     @Test
     public void testDesignNoIsolationMethodSTT() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -220,7 +220,7 @@
     @Test
     public void testDesignNoConnectivityInOffering() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -246,7 +246,7 @@
     @Test
     public void testImplement() throws InsufficientVirtualNetworkCapacityException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -300,7 +300,7 @@
     @Test
     public void testImplementWithCidr() throws InsufficientVirtualNetworkCapacityException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -358,7 +358,7 @@
     @Test
     public void testImplementURIException() throws InsufficientVirtualNetworkCapacityException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
@@ -412,7 +412,7 @@
     @Test
     public void testShutdown() throws InsufficientVirtualNetworkCapacityException, URISyntaxException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
-        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+        MockPhysicalNetworkDao.configureMockPhysicalNetworkDao(physnetdao, physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
@@ -0,0 +1,13 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+
+import com.cloud.network.dao.PhysicalNetworkDao;
+import com.cloud.network.dao.PhysicalNetworkVO;
+
+public class MockPhysicalNetworkDao {
+    public static void configureMockPhysicalNetworkDao(PhysicalNetworkDao physnetdao, PhysicalNetworkVO physnet) {
+        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+    }
+}
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
@@ -0,0 +1,13 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+
+import com.cloud.network.dao.PhysicalNetworkDao;
+import com.cloud.network.dao.PhysicalNetworkVO;
+
+public class MockPhysicalNetworkDao {
+    public static void configureMockPhysicalNetworkDao(PhysicalNetworkDao physnetdao, PhysicalNetworkVO physnet) {
+        when(physnetdao.findById((Long)any())).thenReturn(physnet);
+    }
+}
