--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1723,11 +1723,10 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void testValidateRoleChangeUnknownCaller() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Role role = Mockito.mock(Role.class);
         Mockito.when(role.getRoleType()).thenReturn(RoleType.Unknown);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(role);
         accountManagerImpl.validateRoleChange(account, Mockito.mock(Role.class), caller);
@@ -1735,13 +1734,12 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void testValidateRoleChangeUnknownNewRole() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.Unknown);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
         accountManagerImpl.validateRoleChange(account, newRole, caller);
@@ -1749,8 +1747,7 @@
 
     @Test
     public void testValidateRoleNewRoleSameCaller() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Role currentRole = Mockito.mock(Role.class);
         Mockito.when(currentRole.getRoleType()).thenReturn(RoleType.User);
         Mockito.when(roleService.findRole(1L)).thenReturn(currentRole);
@@ -1758,7 +1755,7 @@
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
         accountManagerImpl.validateRoleChange(account, newRole, caller);
@@ -1766,15 +1763,14 @@
 
     @Test
     public void testValidateRoleCurrentRoleSameCaller() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Role accountRole = Mockito.mock(Role.class);
         Mockito.when(accountRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.User);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(accountRole);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
@@ -1783,13 +1779,12 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void testValidateRoleNewRoleHigherCaller() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.Admin);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
         accountManagerImpl.validateRoleChange(account, newRole, caller);
@@ -1797,15 +1792,14 @@
 
     @Test
     public void testValidateRoleNewRoleLowerCaller() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.User);
         Role accountRole = Mockito.mock(Role.class);
         Mockito.when(accountRole.getRoleType()).thenReturn(RoleType.User);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(accountRole);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
@@ -1814,15 +1808,14 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void testValidateRoleAdminCannotEscalateAdminFromNonRootDomain() {
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
+        Account account = createMockAccountWithRoleId();
         Mockito.when(account.getDomainId()).thenReturn(2L);
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.Admin);
         Role accountRole = Mockito.mock(Role.class);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.Admin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(accountRole);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
@@ -1831,18 +1824,23 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void testValidateRoleAdminCannotChangeDefaultAdmin() {
-        Account account = Mockito.mock(Account.class);
+        Account account = createMockAccountWithRoleId();
         Mockito.when(account.isDefault()).thenReturn(true);
-        Mockito.when(account.getRoleId()).thenReturn(1L);
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.User);
         Role callerRole = Mockito.mock(Role.class);
         Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.Admin);
-        Account caller = Mockito.mock(Account.class);
+        Account caller = createMockAccountWithRoleId();
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(Mockito.mock(Role.class));
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
         accountManagerImpl.validateRoleChange(account, newRole, caller);
+    }
+
+    private static Account createMockAccountWithRoleId() {
+        Account account = Mockito.mock(Account.class);
+        Mockito.when(account.getRoleId()).thenReturn(1L);
+        return account;
     }
 
     @Test
