--- a/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
@@ -59,6 +59,8 @@
     private SearchBuilder<HostVO> mockSearchBuilder;
     @Mock
     private SearchCriteria<HostVO> mockSearchCriteria;
+    @Mock
+    private SearchCriteria<Long> mockLongSearchCriteria;
 
     @Test
     public void testCountUpAndEnabledHostsInZone() {
@@ -124,20 +126,19 @@
         HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
-        SearchCriteria<Long> sc = mock(SearchCriteria.class);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(mockLongSearchCriteria);
         when(sb.and()).thenReturn(sb);
         doReturn(sb).when(hostDao).createSearchBuilder(Long.class);
         doReturn(mockResults).when(hostDao).customSearch(any(SearchCriteria.class), any());
         List<Long> hostIds = hostDao.findHostIdsByZoneClusterResourceStateTypeAndHypervisorType(
                 zoneId, clusterId, msId, resourceStates, types, hypervisorTypes);
         assertEquals(mockResults, hostIds);
-        Mockito.verify(sc).setParameters("zoneId", zoneId);
-        Mockito.verify(sc).setParameters("clusterId", clusterId);
-        Mockito.verify(sc).setParameters("msId", msId);
-        Mockito.verify(sc).setParameters("resourceState", resourceStates.toArray());
-        Mockito.verify(sc).setParameters("type", types.toArray());
-        Mockito.verify(sc).setParameters("hypervisorTypes", hypervisorTypes.toArray());
+        Mockito.verify(mockLongSearchCriteria).setParameters("zoneId", zoneId);
+        Mockito.verify(mockLongSearchCriteria).setParameters("clusterId", clusterId);
+        Mockito.verify(mockLongSearchCriteria).setParameters("msId", msId);
+        Mockito.verify(mockLongSearchCriteria).setParameters("resourceState", resourceStates.toArray());
+        Mockito.verify(mockLongSearchCriteria).setParameters("type", types.toArray());
+        Mockito.verify(mockLongSearchCriteria).setParameters("hypervisorTypes", hypervisorTypes.toArray());
     }
 
     @Test
@@ -182,19 +183,18 @@
         HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
-        SearchCriteria<Long> sc = mock(SearchCriteria.class);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(mockLongSearchCriteria);
         doReturn(sb).when(hostDao).createSearchBuilder(Long.class);
         doReturn(mockResults).when(hostDao).customSearch(any(SearchCriteria.class), any());
         List<Long> hostIds = hostDao.listIdsBy(type, status, resourceState, hypervisorType, zoneId, podId, clusterId);
         assertEquals(mockResults, hostIds);
-        Mockito.verify(sc).setParameters("type", type);
-        Mockito.verify(sc).setParameters("status", status);
-        Mockito.verify(sc).setParameters("resourceState", resourceState);
-        Mockito.verify(sc).setParameters("hypervisorType", hypervisorType);
-        Mockito.verify(sc).setParameters("zoneId", zoneId);
-        Mockito.verify(sc).setParameters("podId", podId);
-        Mockito.verify(sc).setParameters("clusterId", clusterId);
+        Mockito.verify(mockLongSearchCriteria).setParameters("type", type);
+        Mockito.verify(mockLongSearchCriteria).setParameters("status", status);
+        Mockito.verify(mockLongSearchCriteria).setParameters("resourceState", resourceState);
+        Mockito.verify(mockLongSearchCriteria).setParameters("hypervisorType", hypervisorType);
+        Mockito.verify(mockLongSearchCriteria).setParameters("zoneId", zoneId);
+        Mockito.verify(mockLongSearchCriteria).setParameters("podId", podId);
+        Mockito.verify(mockLongSearchCriteria).setParameters("clusterId", clusterId);
     }
 
     @Test
