--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -76,6 +76,9 @@
     @Mock
     ManagementServerHostDao msHostDao;
 
+    @Mock
+    ManagementServerHostVO msHost3Mock;
+
     private long prepareCountPendingJobs() {
         long expectedCount = 1L;
         Mockito.doReturn(expectedCount).when(jobManagerMock).countPendingNonPseudoJobs(1L);
@@ -393,10 +396,9 @@
     @Test
     public void prepareForMaintenanceCmdNoMsHost() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
-        msHostList.add(msHost2);
+        msHostList.add(msHost3Mock);
         Mockito.when(msHostDao.listBy(ManagementServerHost.State.Up)).thenReturn(msHostList);
         Mockito.when(msHostDao.findById(1L)).thenReturn(null);
         PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
@@ -433,9 +435,8 @@
         List<ManagementServerHostVO> msHostList1 = new ArrayList<>();
         msHostList1.add(msHost1);
         msHostList1.add(msHost2);
-        ManagementServerHostVO msHost3 = mock(ManagementServerHostVO.class);
         List<ManagementServerHostVO> msHostList2 = new ArrayList<>();
-        msHostList2.add(msHost3);
+        msHostList2.add(msHost3Mock);
         Mockito.when(msHostDao.listBy(ManagementServerHost.State.Up)).thenReturn(msHostList1);
         Mockito.when(msHostDao.listBy(ManagementServerHost.State.PreparingForMaintenance, ManagementServerHost.State.PreparingForShutDown)).thenReturn(msHostList2);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost1);
