--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -3578,13 +3578,11 @@
         final Domain vm = Mockito.mock(Domain.class);
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
-        final InterfaceDef intDef = Mockito.mock(InterfaceDef.class);
+        final InterfaceDef intDef = createMockInterfaceDef();
         nics.add(intDef);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
-
-        when(intDef.getMacAddress()).thenReturn("00:00:00:00");
 
         when(nic.getMac()).thenReturn("00:00:00:00");
 
@@ -3625,13 +3623,11 @@
         final InterfaceDef interfaceDef = Mockito.mock(InterfaceDef.class);
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
-        final InterfaceDef intDef = Mockito.mock(InterfaceDef.class);
+        final InterfaceDef intDef = createMockInterfaceDef();
         nics.add(intDef);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
-
-        when(intDef.getMacAddress()).thenReturn("00:00:00:00");
 
         when(nic.getMac()).thenReturn("00:00:00:01");
         when(nic.getName()).thenReturn("br0");
@@ -3721,13 +3717,11 @@
         final VifDriver vifDriver = Mockito.mock(VifDriver.class);
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
-        final InterfaceDef intDef = Mockito.mock(InterfaceDef.class);
+        final InterfaceDef intDef = createMockInterfaceDef();
         nics.add(intDef);
 
         when(libvirtComputingResourceMock.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
-
-        when(intDef.getMacAddress()).thenReturn("00:00:00:00");
 
         when(nic.getMac()).thenReturn("00:00:00:01");
 
@@ -3776,7 +3770,7 @@
         final Domain vm = Mockito.mock(Domain.class);
 
         final List<InterfaceDef> nics = new ArrayList<InterfaceDef>();
-        final InterfaceDef intDef = Mockito.mock(InterfaceDef.class);
+        final InterfaceDef intDef = createMockInterfaceDef();
         nics.add(intDef);
 
         final VifDriver vifDriver = Mockito.mock(VifDriver.class);
@@ -3787,7 +3781,6 @@
         when(libvirtComputingResourceMock.getInterfaces(conn, command.getVmName())).thenReturn(nics);
 
         when(intDef.getBrName()).thenReturn("br0");
-        when(intDef.getMacAddress()).thenReturn("00:00:00:00");
 
         when(nic.getMac()).thenReturn("00:00:00:00");
 
@@ -5613,6 +5606,12 @@
         long memoryFreeInKBs = libvirtComputingResource.getMemoryFreeInKBs(domainMock);
 
         Assert.assertEquals(-1, memoryFreeInKBs);
+    }
+
+    private static InterfaceDef createMockInterfaceDef() {
+        final InterfaceDef intDef = Mockito.mock(InterfaceDef.class);
+        when(intDef.getMacAddress()).thenReturn("00:00:00:00");
+        return intDef;
     }
 
     private MemoryStatistic[] createMemoryStatisticFreeMemory100() {
