--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
@@ -185,23 +185,27 @@
 
     @Test (expected = CloudRuntimeException.class)
     public void validateValidateAvailableSizeOnPoolToTakeVolumeSnapshotAvailabeSizeLessThanMinRateThrowCloudRuntimeException(){
-        KVMPhysicalDisk kvmPhysicalDiskMock = Mockito.mock(KVMPhysicalDisk.class);
+        KVMPhysicalDisk kvmPhysicalDiskMock = createMockKvmPhysicalDisk();
 
         Mockito.doReturn(104l).when(kvmStoragePoolMock).getAvailable();
-        Mockito.doReturn(100l).when(kvmPhysicalDiskMock).getSize();
 
         storageProcessor.validateAvailableSizeOnPoolToTakeVolumeSnapshot(kvmStoragePoolMock, kvmPhysicalDiskMock);
     }
 
     @Test
     public void validateValidateAvailableSizeOnPoolToTakeVolumeSnapshotAvailabeSizeEqualOrHigherThanMinRateDoNothing(){
-        KVMPhysicalDisk kvmPhysicalDiskMock = Mockito.mock(KVMPhysicalDisk.class);
+        KVMPhysicalDisk kvmPhysicalDiskMock = createMockKvmPhysicalDisk();
 
         Mockito.doReturn(105l, 106l).when(kvmStoragePoolMock).getAvailable();
-        Mockito.doReturn(100l).when(kvmPhysicalDiskMock).getSize();
 
         storageProcessor.validateAvailableSizeOnPoolToTakeVolumeSnapshot(kvmStoragePoolMock, kvmPhysicalDiskMock);
         storageProcessor.validateAvailableSizeOnPoolToTakeVolumeSnapshot(kvmStoragePoolMock, kvmPhysicalDiskMock);
+    }
+
+    private static KVMPhysicalDisk createMockKvmPhysicalDisk() {
+        KVMPhysicalDisk kvmPhysicalDiskMock = Mockito.mock(KVMPhysicalDisk.class);
+        Mockito.doReturn(100l).when(kvmPhysicalDiskMock).getSize();
+        return kvmPhysicalDiskMock;
     }
 
     private List<LibvirtVMDef.DiskDef> createDiskDefs(int iterations, boolean duplicatePath) {
