--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -259,6 +259,9 @@
     @Mock
     private UserDataVO userDataMock;
 
+    @Mock
+    private VolumeVO imageStoreVolumeVO;
+
     public class CustomThreadPoolExecutor extends ThreadPoolExecutor {
         AtomicInteger ai = new AtomicInteger(0);
         public CustomThreadPoolExecutor(int corePoolSize, int maximumPoolSize, long keepAliveTime, TimeUnit unit,
@@ -630,42 +633,38 @@
     @Test
     public void getImageStoreTestStoreUuidIsNotNullShouldReturnAValidImageStoreIfValidUuid() {
         DataStore dataStore = Mockito.mock(DataStore.class);
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
 
         Mockito.when(dataStoreManager.getDataStore(Mockito.anyString(), Mockito.any(DataStoreRole.class))).thenReturn(dataStore);
 
-        templateManager.getImageStore("UUID", 1L, volumeVO);
+        templateManager.getImageStore("UUID", 1L, imageStoreVolumeVO);
     }
 
     @Test(expected = CloudRuntimeException.class)
     public void getImageStoreTestStoreUuidIsNotNullShouldThrowCloudRuntimeExceptionIfInvalidUuid() {
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
 
         Mockito.when(dataStoreManager.getDataStore(Mockito.anyString(), Mockito.any(DataStoreRole.class))).thenReturn(null);
 
-        templateManager.getImageStore("UUID", 1L, volumeVO);
+        templateManager.getImageStore("UUID", 1L, imageStoreVolumeVO);
     }
 
     @Test
     public void getImageStoreTestStoreUuidIsNullAndThereIsNoActiveHeuristicRulesShouldCallGetImageStoreWithFreeCapacity() {
         DataStore dataStore = Mockito.mock(DataStore.class);
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
 
         Mockito.when(heuristicRuleHelperMock.getImageStoreIfThereIsHeuristicRule(Mockito.anyLong(), Mockito.any(HeuristicType.class), Mockito.any(VolumeVO.class))).thenReturn(null);
         Mockito.when(dataStoreManager.getImageStoreWithFreeCapacity(Mockito.anyLong())).thenReturn(dataStore);
 
-        templateManager.getImageStore(null, 1L, volumeVO);
+        templateManager.getImageStore(null, 1L, imageStoreVolumeVO);
         Mockito.verify(dataStoreManager, Mockito.times(1)).getImageStoreWithFreeCapacity(Mockito.anyLong());
     }
 
     @Test
     public void getImageStoreTestStoreUuidIsNullAndThereIsActiveHeuristicRulesShouldNotCallGetImageStoreWithFreeCapacity() {
         DataStore dataStore = Mockito.mock(DataStore.class);
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
 
         Mockito.when(heuristicRuleHelperMock.getImageStoreIfThereIsHeuristicRule(Mockito.anyLong(), Mockito.any(HeuristicType.class), Mockito.any(VolumeVO.class))).thenReturn(dataStore);
 
-        templateManager.getImageStore(null, 1L, volumeVO);
+        templateManager.getImageStore(null, 1L, imageStoreVolumeVO);
         Mockito.verify(dataStoreManager, Mockito.times(0)).getImageStoreWithFreeCapacity(Mockito.anyLong());
     }
 
