--- a/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
+++ b/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
@@ -110,6 +110,7 @@
     private AssignToLoadBalancerRuleCmd assignToLbRuleCmd;
     private ResponseGenerator responseGenerator;
     private SuccessResponse successResponseGenerator;
+    private NetworkDao sharedNetworkDao;
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
@@ -120,6 +121,7 @@
 
     @Before
     public void setUp() {
+        sharedNetworkDao = Mockito.mock(NetworkDao.class);
         assignToLbRuleCmd = new AssignToLoadBalancerRuleCmd() {
         };
 
@@ -169,7 +171,6 @@
         LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
-        NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
         AutoScaleVmGroupDao autoScaleVmGroupDao = Mockito.mock(AutoScaleVmGroupDao.class);
         VpcManager vpcMgr = Mockito.mock(VpcManager.class);
@@ -179,7 +180,7 @@
         _lbMgr._vmDao = userVmDao;
         _lbMgr._accountDao = accountDao;
         _lbMgr._accountMgr = accountMgr;
-        _lbMgr._networkDao = networkDao;
+        _lbMgr._networkDao = sharedNetworkDao;
         _lbvmMapList = new ArrayList<>();
         _lbMgr._rulesMgr = _rulesMgr;
         _lbMgr._networkModel = _networkModel;
@@ -218,7 +219,6 @@
         LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
-        NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
         NicSecondaryIpDao nicSecIpDao =  Mockito.mock(NicSecondaryIpDao.class);
         AutoScaleVmGroupDao autoScaleVmGroupDao = Mockito.mock(AutoScaleVmGroupDao.class);
@@ -229,7 +229,7 @@
         _lbMgr._vmDao = userVmDao;
         _lbMgr._accountDao = accountDao;
         _lbMgr._accountMgr = accountMgr;
-        _lbMgr._networkDao = networkDao;
+        _lbMgr._networkDao = sharedNetworkDao;
         _lbMgr._nicSecondaryIpDao = nicSecIpDao;
         _lbvmMapList = new ArrayList<>();
         _lbMgr._rulesMgr = _rulesMgr;
@@ -270,7 +270,6 @@
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
-        NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         NicSecondaryIpDao nicSecIpDao =  Mockito.mock(NicSecondaryIpDao.class);
         LoadBalancerVMMapVO lbVmMapVO = new LoadBalancerVMMapVO(1L, 1L, "10.1.1.175", false);
         AutoScaleVmGroupDao autoScaleVmGroupDao = Mockito.mock(AutoScaleVmGroupDao.class);
@@ -281,7 +280,7 @@
         _lbMgr._vmDao = userVmDao;
         _lbMgr._accountDao = accountDao;
         _lbMgr._accountMgr = accountMgr;
-        _lbMgr._networkDao = networkDao;
+        _lbMgr._networkDao = sharedNetworkDao;
         _lbMgr._nicSecondaryIpDao = nicSecIpDao;
         _lbvmMapList = new ArrayList<>();
         _lbvmMapList.add(lbVmMapVO);
