--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -226,6 +226,8 @@
     @Mock
     AccountDao accountDao;
     @Mock
+    private AccountVO accountVOMock;
+    @Mock
     NetworkDao networkDao;
     @Mock
     ConfigurationDao configDao;
@@ -653,7 +655,6 @@
         SslCertVO sslCertVO = mock(SslCertVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         Ip ip = mock(Ip.class);
-        AccountVO accountVO = mock(AccountVO.class);
         Loadbalancer loadbalancer = mock(Loadbalancer.class);
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(publicNetwork);
@@ -671,7 +672,7 @@
         when(entityMgr.findById(eq(SslCertVO.class), anyLong())).thenReturn(sslCertVO);
         when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(), anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(loadBalancingRule.getSourceIp()).thenReturn(ip);
-        when(accountDao.findById(anyLong())).thenReturn(accountVO);
+        when(accountDao.findById(anyLong())).thenReturn(accountVOMock);
         when(ip.addr()).thenReturn("192.168.100.100");
         when(getTungstenLoadBalancerAnswer.getApiObjectBase()).thenReturn(loadbalancer);
         when(ipAddressVO.getAddress()).thenReturn(ip);
@@ -1498,7 +1499,6 @@
     public void createSharedNetworkTest() {
         Network network = mock(Network.class);
         Vlan vlan = mock(Vlan.class);
-        AccountVO accountVO = mock(AccountVO.class);
         TungstenAnswer createTungstenSharedNetworkAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer addNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
@@ -1509,7 +1509,7 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         ApiObjectBase apiObjectBase = mock(ApiObjectBase.class);
 
-        when(accountDao.findById(anyLong())).thenReturn(accountVO);
+        when(accountDao.findById(anyLong())).thenReturn(accountVOMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkCommand.class), anyLong())).thenReturn(createTungstenSharedNetworkAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(AddTungstenNetworkSubnetCommand.class), anyLong())).thenReturn(addNetworkSubnetAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenNetworkDnsCommand.class), anyLong())).thenReturn(getTungstenNetworkDnsAnswer);
