--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/lifecycle/ScaleIOPrimaryDataStoreLifeCycleTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/lifecycle/ScaleIOPrimaryDataStoreLifeCycleTest.java
@@ -143,11 +143,8 @@
 
         final ZoneScope scope = new ZoneScope(1L);
 
-        HostVO host1 = Mockito.mock(HostVO.class);
-        HostVO host2 = Mockito.mock(HostVO.class);
-
-        Mockito.when(host1.getId()).thenReturn(1L);
-        Mockito.when(host2.getId()).thenReturn(2L);
+        HostVO host1 = createMockHostVO(1L);
+        HostVO host2 = createMockHostVO(2L);
 
         when(resourceManager.getEligibleUpAndEnabledHostsInZoneForStorageConnection(dataStore, scope.getScopeId(), Hypervisor.HypervisorType.KVM))
                 .thenReturn(Arrays.asList(host1, host2));
@@ -227,4 +224,10 @@
         final boolean result = scaleIOPrimaryDataStoreLifeCycleTest.deleteDataStore(store);
         assertThat(result).isTrue();
     }
+
+    private static HostVO createMockHostVO(long id) {
+        HostVO host = Mockito.mock(HostVO.class);
+        Mockito.when(host.getId()).thenReturn(id);
+        return host;
+    }
 }
