--- a/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
@@ -130,6 +130,9 @@
 
     @Mock
     private Network networkMock;
+
+    @Mock
+    private NetworkOffering publicNicNetworkOfferingMock;
 
     private NicProfile nicProfile = new NicProfile();
 
@@ -498,12 +501,11 @@
     @Test
     public void configurePublicNicTestConfigureDeviceId() throws InsufficientAddressCapacityException {
         PublicIp publicIpMock = Mockito.mock(PublicIp.class);
-        NetworkOffering networkOfferingMock = Mockito.mock(NetworkOffering.class);
 
         Mockito.when(routerDeploymentDefinitionMock.isPublicNetwork()).thenReturn(true);
         Mockito.when(routerDeploymentDefinitionMock.getSourceNatIP()).thenReturn(publicIpMock);
         Mockito.doNothing().when(networkHelperSpy).configurePublicVrNicBasedOnSourceNatIp(Mockito.any(), Mockito.any());
-        Mockito.doReturn(List.of(networkOfferingMock)).when(networkModel).getSystemAccountNetworkOfferings(Mockito.any());
+        Mockito.doReturn(List.of(publicNicNetworkOfferingMock)).when(networkModel).getSystemAccountNetworkOfferings(Mockito.any());
         Mockito.doReturn(List.of(networkMock)).when(networkOrchestrationService).setupNetwork(
                 Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.anyBoolean()
         );
@@ -517,13 +519,12 @@
     @Test
     public void configurePublicNicTestUpdateGuestNetworksIpv6Nic() throws InsufficientAddressCapacityException {
         PublicIp publicIpMock = Mockito.mock(PublicIp.class);
-        NetworkOffering networkOfferingMock = Mockito.mock(NetworkOffering.class);
         DeployDestination deployDestinationMock = Mockito.mock(DeployDestination.class);
 
         Mockito.when(routerDeploymentDefinitionMock.isPublicNetwork()).thenReturn(true);
         Mockito.when(routerDeploymentDefinitionMock.getSourceNatIP()).thenReturn(publicIpMock);
         Mockito.doNothing().when(networkHelperSpy).configurePublicVrNicBasedOnSourceNatIp(Mockito.any(), Mockito.any());
-        Mockito.doReturn(List.of(networkOfferingMock)).when(networkModel).getSystemAccountNetworkOfferings(Mockito.any());
+        Mockito.doReturn(List.of(publicNicNetworkOfferingMock)).when(networkModel).getSystemAccountNetworkOfferings(Mockito.any());
         Mockito.doReturn(List.of(networkMock)).when(networkOrchestrationService).setupNetwork(
                 Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.anyBoolean()
         );
