--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -344,9 +344,7 @@
         when(network.getBroadcastUri()).thenReturn(new URI("vlan://14"));
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
-        final NicProfile nic = mock(NicProfile.class);
-        when(nic.getMacAddress()).thenReturn("macaddress");
-        when(nic.getReservationStrategy()).thenReturn(ReservationStrategy.Start);
+        final NicProfile nic = createMockNicProfile();
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
@@ -396,9 +394,7 @@
         when(network.getBroadcastUri()).thenReturn(new URI("vlan://14"));
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
-        final NicProfile nic = mock(NicProfile.class);
-        when(nic.getMacAddress()).thenReturn("macaddress");
-        when(nic.getReservationStrategy()).thenReturn(ReservationStrategy.Start);
+        final NicProfile nic = createMockNicProfile();
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
@@ -447,9 +443,7 @@
         when(network.getPhysicalNetworkId()).thenReturn(NETWORK_ID);
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
-        final NicProfile nic = mock(NicProfile.class);
-        when(nic.getMacAddress()).thenReturn("macaddress");
-        when(nic.getReservationStrategy()).thenReturn(ReservationStrategy.Start);
+        final NicProfile nic = createMockNicProfile();
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
@@ -483,9 +477,7 @@
         when(network.getPhysicalNetworkId()).thenReturn(NETWORK_ID);
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
-        final NicProfile nic = mock(NicProfile.class);
-        when(nic.getMacAddress()).thenReturn("macaddress");
-        when(nic.getReservationStrategy()).thenReturn(ReservationStrategy.Start);
+        final NicProfile nic = createMockNicProfile();
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
@@ -588,4 +580,11 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    private static NicProfile createMockNicProfile() {
+        final NicProfile nic = mock(NicProfile.class);
+        when(nic.getMacAddress()).thenReturn("macaddress");
+        when(nic.getReservationStrategy()).thenReturn(ReservationStrategy.Start);
+        return nic;
+    }
 }
