--- a/api/src/test/java/org/apache/cloudstack/api/command/test/ActivateProjectCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/ActivateProjectCmdTest.java
@@ -72,8 +72,7 @@
         Project project = Mockito.mock(Project.class);
         Mockito.when(project.getId()).thenReturn(2L);
         ProjectService projectService = Mockito.mock(ProjectService.class);
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getId()).thenReturn(2L);
+        Account account = MockProjectOwnerAccount.createMockAccount();
         Mockito.when(projectService.getProject(ArgumentMatchers.anyLong())).thenReturn(project);
 
         Mockito.when(projectService.getProjectOwner(ArgumentMatchers.anyLong())).thenReturn(account);
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/AddAccountToProjectCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/AddAccountToProjectCmdTest.java
@@ -99,9 +99,8 @@
         Mockito.when(project.getId()).thenReturn(2L);
 
         ProjectService projectService = Mockito.mock(ProjectService.class);
-        Account account = Mockito.mock(Account.class);
+        Account account = MockProjectOwnerAccount.createMockAccount();
 
-        Mockito.when(account.getId()).thenReturn(2L);
         Mockito.when(projectService.getProject(ArgumentMatchers.anyLong())).thenReturn(project);
 
         Mockito.when(projectService.getProjectOwner(ArgumentMatchers.anyLong())).thenReturn(account);
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/MockProjectOwnerAccount.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/MockProjectOwnerAccount.java
@@ -0,0 +1,12 @@
+package org.apache.cloudstack.api.command.test;
+
+import com.cloud.user.Account;
+import org.mockito.Mockito;
+
+public class MockProjectOwnerAccount {
+    public static Account createMockAccount() {
+        Account account = Mockito.mock(Account.class);
+        Mockito.when(account.getId()).thenReturn(2L);
+        return account;
+    }
+}
