--- a/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
+++ b/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
@@ -224,9 +224,8 @@
         _resource.configure("BigSwitchBcfResource", _parameters);
 
         AttachmentData attachmentData = mock(AttachmentData.class);
-        AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
+        AttachmentData.Attachment attachment = createMockAttachment();
         when(attachmentData.getAttachment()).thenReturn(attachment);
-        when(attachment.getId()).thenReturn("eeee");
         when(_bigswitchBcfApi.createAttachment((String)any(), (String)any(), (AttachmentData)any())).thenThrow(new BigSwitchBcfApiException())
         .thenReturn(UUID.randomUUID().toString());
 
@@ -241,9 +240,8 @@
         _resource.configure("BigSwitchBcfResource", _parameters);
 
         AttachmentData attachmentData = mock(AttachmentData.class);
-        AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
+        AttachmentData.Attachment attachment = createMockAttachment();
         when(attachmentData.getAttachment()).thenReturn(attachment);
-        when(attachment.getId()).thenReturn("eeee");
         doThrow(new BigSwitchBcfApiException()).when(_bigswitchBcfApi).createAttachment((String)any(), (String)any(), (AttachmentData)any());
 
         CreateBcfAttachmentCommand cmd = new CreateBcfAttachmentCommand("tenantid", "tenantname",
@@ -428,4 +426,9 @@
         assertTrue(ans.getResult());
     }
 
+    private static AttachmentData.Attachment createMockAttachment() {
+        AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
+        when(attachment.getId()).thenReturn("eeee");
+        return attachment;
+    }
 }
