--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -40,7 +40,6 @@
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
-import static org.mockito.Mockito.withSettings;
 
 import java.io.File;
 import java.security.InvalidParameterException;
@@ -213,6 +212,10 @@
     private VpcServiceMapDao vpcServiceMapDao;
     @Mock
     private NetworkModel networkModel;
+    @Mock
+    private NetworkElement networkElement;
+    @Mock(extraInterfaces = NetworkCustomActionProvider.class)
+    private NetworkElement networkCustomActionElement;
 
     @Mock
     private PhysicalNetworkServiceProviderDao physicalNetworkServiceProviderDao;
@@ -1983,9 +1986,8 @@
         when(network.getId()).thenReturn(13L);
         when(networkServiceMapDao.getProviderForServiceInNetwork(13L, Network.Service.CustomAction)).thenReturn("ExtProvider");
 
-        NetworkElement element = mock(NetworkElement.class, withSettings().extraInterfaces(NetworkCustomActionProvider.class));
-        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) element;
-        when(networkModel.getElementImplementingProvider("ExtProvider")).thenReturn(element);
+        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) networkCustomActionElement;
+        when(networkModel.getElementImplementingProvider("ExtProvider")).thenReturn(networkCustomActionElement);
         when(provider.canHandleCustomAction(network)).thenReturn(false);
 
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
@@ -2011,8 +2013,7 @@
         when(network.getId()).thenReturn(131L);
         when(networkServiceMapDao.getProviderForServiceInNetwork(131L, Network.Service.CustomAction)).thenReturn("ExtProvider");
 
-        NetworkElement element = mock(NetworkElement.class);
-        when(networkModel.getElementImplementingProvider("ExtProvider")).thenReturn(element);
+        when(networkModel.getElementImplementingProvider("ExtProvider")).thenReturn(networkElement);
 
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
         when(actionVO.getId()).thenReturn(121L);
@@ -2037,9 +2038,8 @@
         when(network.getId()).thenReturn(14L);
         when(networkServiceMapDao.getProviderForServiceInNetwork(14L, Network.Service.CustomAction)).thenReturn("ExtProvider");
 
-        NetworkElement element = mock(NetworkElement.class, withSettings().extraInterfaces(NetworkCustomActionProvider.class));
-        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) element;
-        when(networkModel.getElementImplementingProvider("ExtProvider")).thenReturn(element);
+        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) networkCustomActionElement;
+        when(networkModel.getElementImplementingProvider("ExtProvider")).thenReturn(networkCustomActionElement);
         when(provider.canHandleCustomAction(network)).thenReturn(true);
         when(provider.runCustomAction(eq(network), eq("dump-config"), any())).thenReturn("dump-output");
 
@@ -2066,8 +2066,7 @@
         when(vpc.getId()).thenReturn(21L);
         when(vpcServiceMapDao.getProviderForServiceInVpc(21L, Network.Service.CustomAction)).thenReturn("VpcProvider");
 
-        NetworkElement element = mock(NetworkElement.class);
-        when(networkModel.getElementImplementingProvider("VpcProvider")).thenReturn(element);
+        when(networkModel.getElementImplementingProvider("VpcProvider")).thenReturn(networkElement);
 
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
         when(actionVO.getId()).thenReturn(20L);
@@ -2113,9 +2112,8 @@
         when(vpc.getId()).thenReturn(212L);
         when(vpcServiceMapDao.getProviderForServiceInVpc(212L, Network.Service.CustomAction)).thenReturn("VpcProvider");
 
-        NetworkElement element = mock(NetworkElement.class, withSettings().extraInterfaces(NetworkCustomActionProvider.class));
-        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) element;
-        when(networkModel.getElementImplementingProvider("VpcProvider")).thenReturn(element);
+        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) networkCustomActionElement;
+        when(networkModel.getElementImplementingProvider("VpcProvider")).thenReturn(networkCustomActionElement);
         when(provider.canHandleVpcCustomAction(vpc)).thenReturn(false);
 
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
@@ -2141,9 +2139,8 @@
         when(vpc.getId()).thenReturn(22L);
         when(vpcServiceMapDao.getProviderForServiceInVpc(22L, Network.Service.CustomAction)).thenReturn("VpcProvider");
 
-        NetworkElement element = mock(NetworkElement.class, withSettings().extraInterfaces(NetworkCustomActionProvider.class));
-        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) element;
-        when(networkModel.getElementImplementingProvider("VpcProvider")).thenReturn(element);
+        NetworkCustomActionProvider provider = (NetworkCustomActionProvider) networkCustomActionElement;
+        when(networkModel.getElementImplementingProvider("VpcProvider")).thenReturn(networkCustomActionElement);
         when(provider.canHandleVpcCustomAction(vpc)).thenReturn(true);
         when(provider.runCustomAction(eq(vpc), eq("dump-config"), any())).thenReturn("vpc-dump-output");
 
