--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1420,9 +1420,8 @@
 
     @Test
     public void getVmOverrideDiskOfferingId() {
-        DiskOfferingVO diskOfferingMock = Mockito.mock(DiskOfferingVO.class);
+        DiskOfferingVO diskOfferingMock = createMockDiskOffering(overrideDiskOfferingId);
         when(diskOfferingDao.findByUuid(overrideDiskOfferingUuid)).thenReturn(diskOfferingMock);
-        when(diskOfferingMock.getId()).thenReturn(overrideDiskOfferingId);
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("overridediskofferingid", overrideDiskOfferingUuid);
@@ -1434,9 +1433,8 @@
 
     @Test
     public void getVmDiskOfferingId() {
-        DiskOfferingVO diskOfferingMock = Mockito.mock(DiskOfferingVO.class);
+        DiskOfferingVO diskOfferingMock = createMockDiskOffering(diskOfferingId);
         when(diskOfferingDao.findByUuid(diskOfferingUuid)).thenReturn(diskOfferingMock);
-        when(diskOfferingMock.getId()).thenReturn(diskOfferingId);
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("diskofferingid", diskOfferingUuid);
@@ -2512,6 +2510,12 @@
                 "-" + asVmGroupMock.getNextVmSeq() + "-[a-z]{6}";
         Assert.assertTrue(result.first().matches(vmHostNamePattern));
         Assert.assertEquals(result.first(), result.second());
+    }
+
+    private static DiskOfferingVO createMockDiskOffering(Long diskOfferingId) {
+        DiskOfferingVO diskOfferingMock = Mockito.mock(DiskOfferingVO.class);
+        when(diskOfferingMock.getId()).thenReturn(diskOfferingId);
+        return diskOfferingMock;
     }
 
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
