--- a/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
+++ b/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
@@ -22,6 +22,7 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.InjectMocks;
@@ -40,16 +41,22 @@
     @InjectMocks
     private ElasticLoadBalancerManagerImpl elasticLoadBalancerManagerImpl;
 
+    private ReservationContext reservationContext;
+
+    @Before
+    public void setUp() {
+        reservationContext = mock(ReservationContext.class);
+    }
+
     @Test
     public void testFinalizeStartWhenCmdsAnswerIsNull() throws Exception {
         VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
         when(cmds.getAnswer("checkSsh")).thenReturn(null);
-        ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = false;
-        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, context);
+        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, reservationContext);
 
         assertEquals(expected, actual);
     }
@@ -62,10 +69,9 @@
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
         when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
-        ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = false;
-        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, context);
+        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, reservationContext);
 
         assertEquals(expected, actual);
     }
@@ -78,10 +84,9 @@
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
         when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
-        ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = true;
-        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, context);
+        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, reservationContext);
 
         assertEquals(expected, actual);
     }
