--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -240,9 +240,8 @@
 
         final List<NiciraNvpDeviceVO> deviceList = new ArrayList<NiciraNvpDeviceVO>();
         final NiciraNvpDeviceVO nndVO = mock(NiciraNvpDeviceVO.class);
-        final NiciraNvpRouterMappingVO nnrmVO = mock(NiciraNvpRouterMappingVO.class);
+        final NiciraNvpRouterMappingVO nnrmVO = createMockNiciraNvpRouterMappingVO("abcde");
         when(niciraNvpRouterMappingDao.findByNetworkId(NETWORK_ID)).thenReturn(nnrmVO);
-        when(nnrmVO.getLogicalRouterUuid()).thenReturn("abcde");
         when(nndVO.getHostId()).thenReturn(NETWORK_ID);
         final HostVO hvo = mock(HostVO.class);
         when(hvo.getId()).thenReturn(NETWORK_ID);
@@ -307,8 +306,7 @@
         when(network.getCidr()).thenReturn(NETWORK_CIDR);
         when(network.getGateway()).thenReturn(NETWORK_GATEWAY);
 
-        NiciraNvpRouterMappingVO mapping = mock(NiciraNvpRouterMappingVO.class);
-        when(mapping.getLogicalRouterUuid()).thenReturn("xxxx-xxxx-xxxx");
+        NiciraNvpRouterMappingVO mapping = createMockNiciraNvpRouterMappingVO("xxxx-xxxx-xxxx");
         when(niciraNvpRouterMappingDao.findByNetworkId(NETWORK_ID)).thenReturn(mapping);
 
         final ConfigureSharedNetworkUuidAnswer answer = mock(ConfigureSharedNetworkUuidAnswer.class);
@@ -418,4 +416,10 @@
 
         element.implement(network, offering, dest, context);
     }
+
+    private static NiciraNvpRouterMappingVO createMockNiciraNvpRouterMappingVO(final String logicalRouterUuid) {
+        final NiciraNvpRouterMappingVO nnrmVO = mock(NiciraNvpRouterMappingVO.class);
+        when(nnrmVO.getLogicalRouterUuid()).thenReturn(logicalRouterUuid);
+        return nnrmVO;
+    }
 }
