--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1286,8 +1286,7 @@
         final Long prefixId = 1L;
         DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
         Mockito.when(cmd.getId()).thenReturn(prefixId);
-        DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
-        Mockito.when(prefixVO.getId()).thenReturn(prefixId);
+        DataCenterGuestIpv6PrefixVO prefixVO = createMockDataCenterGuestIpv6PrefixVO();
         Mockito.when(dataCenterGuestIpv6PrefixDao.findById(Mockito.anyLong())).thenReturn(prefixVO);
         Mockito.when(ipv6GuestPrefixSubnetNetworkMapDao.listUsedByPrefix(Mockito.anyLong()))
                 .thenReturn(List.of(Mockito.mock(Ipv6GuestPrefixSubnetNetworkMapVO.class)));
@@ -1299,8 +1298,7 @@
         final Long prefixId = 1L;
         DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
         Mockito.when(cmd.getId()).thenReturn(prefixId);
-        DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
-        Mockito.when(prefixVO.getId()).thenReturn(prefixId);
+        DataCenterGuestIpv6PrefixVO prefixVO = createMockDataCenterGuestIpv6PrefixVO();
         Mockito.when(dataCenterGuestIpv6PrefixDao.findById(Mockito.anyLong())).thenReturn(prefixVO);
         Mockito.when(ipv6GuestPrefixSubnetNetworkMapDao.listUsedByPrefix(Mockito.anyLong())).thenReturn(new ArrayList<>());
         final List<Long> removedPrefix = new ArrayList<>();
@@ -1342,6 +1340,12 @@
         } catch (InsufficientCapacityException | ResourceUnavailableException | ResourceAllocationException e) {
             throw new RuntimeException(e);
         }
+    }
+
+    private static DataCenterGuestIpv6PrefixVO createMockDataCenterGuestIpv6PrefixVO() {
+        DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
+        Mockito.when(prefixVO.getId()).thenReturn(1L);
+        return prefixVO;
     }
 
     private void mockPersistDatacenterForCreateZone() {
