--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static VirtualMachineTO createMockUpdatedVirtualMachineTO(String vncPassword) {
+        VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
+        when(updatedTO.getVncPassword()).thenReturn(vncPassword);
+        return updatedTO;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1569,11 +1575,9 @@
 
     @Test
     public void updateExternalVmDataFromPrepareAnswer_vncPasswordUpdated_updatesPassword() {
-        VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
-        VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
+        VirtualMachineTO updatedTO = createMockUpdatedVirtualMachineTO("newPassword");
         UserVmVO userVmVO = mock(UserVmVO.class);
-        when(updatedTO.getVncPassword()).thenReturn("newPassword");
-        when(vmTO.getVncPassword()).thenReturn("oldPassword");
+        VirtualMachineTO vmTO = createMockUpdatedVirtualMachineTO("oldPassword");
         when(userVmDaoMock.findById(anyLong())).thenReturn(userVmVO);
         virtualMachineManagerImpl.updateExternalVmDataFromPrepareAnswer(vmTO, updatedTO);
         verify(userVmVO).setVncPassword("newPassword");
