--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -228,6 +228,8 @@
     @Mock
     HostPodDao podDao;
     @Mock
+    private HostPodVO podMock;
+    @Mock
     ClusterDao clusterDao;
     @Mock
     VolumeOrchestrationService volumeOrchestrationService;
@@ -2143,7 +2145,6 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
         ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        HostPodVO pod = Mockito.mock(HostPodVO.class);
         DataCenterVO zone = mockZone();
         mockDiskOffering();
         StoragePool pool = Mockito.mock(StoragePool.class);
@@ -2151,9 +2152,9 @@
         when(virtualMachineManager.findClusterAndHostIdForVm(vm, false)).thenReturn(new Pair<>(1L, 2L));
         when(clusterDao.findById(1L)).thenReturn(cluster);
         when(cluster.getPodId()).thenReturn(1L);
-        when(podDao.findById(1L)).thenReturn(pod);
+        when(podDao.findById(1L)).thenReturn(podMock);
         when(volumeToAttach.getDiskOfferingId()).thenReturn(1L);
-        when(volumeOrchestrationService.findStoragePool(any(DiskProfile.class), eq(zone), eq(pod), eq(1L), eq(2L), eq(vm), eq(Collections.emptySet())))
+        when(volumeOrchestrationService.findStoragePool(any(DiskProfile.class), eq(zone), eq(podMock), eq(1L), eq(2L), eq(vm), eq(Collections.emptySet())))
                 .thenReturn(pool);
         StoragePool result = volumeApiServiceImpl.getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm);
         Assert.assertNotNull(result);
@@ -2167,14 +2168,13 @@
         DataCenterVO zone = mockZone();
         Pair<Long, Long> clusterHostId = new Pair<>(1L, 2L);
         ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        HostPodVO pod = Mockito.mock(HostPodVO.class);
         mockDiskOffering();
         when(vm.getDataCenterId()).thenReturn(1L);
         when(clusterDao.findById(1L)).thenReturn(cluster);
         when(virtualMachineManager.findClusterAndHostIdForVm(vm, false)).thenReturn(clusterHostId);
-        when(podDao.findById(anyLong())).thenReturn(pod);
+        when(podDao.findById(anyLong())).thenReturn(podMock);
         when(volumeToAttach.getDiskOfferingId()).thenReturn(1L);
-        when(volumeOrchestrationService.findStoragePool(any(DiskProfile.class), eq(zone), eq(pod), eq(1L), eq(2L), eq(vm), eq(Collections.emptySet())))
+        when(volumeOrchestrationService.findStoragePool(any(DiskProfile.class), eq(zone), eq(podMock), eq(1L), eq(2L), eq(vm), eq(Collections.emptySet())))
                 .thenReturn(null);
         Assert.assertNull(volumeApiServiceImpl.getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm));
     }
@@ -2184,15 +2184,14 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
         DataCenterVO zone = mockZone();
-        HostPodVO pod = Mockito.mock(HostPodVO.class);
         mockDiskOffering();
         StoragePool pool = Mockito.mock(StoragePool.class);
         when(vm.getDataCenterId()).thenReturn(1L);
         when(vm.getPodIdToDeployIn()).thenReturn(2L);
         when(virtualMachineManager.findClusterAndHostIdForVm(vm, false)).thenReturn(new Pair<>(null, 2L));
-        when(podDao.findById(2L)).thenReturn(pod);
+        when(podDao.findById(2L)).thenReturn(podMock);
         when(volumeToAttach.getDiskOfferingId()).thenReturn(1L);
-        when(volumeOrchestrationService.findStoragePool(any(DiskProfile.class), eq(zone), eq(pod), eq(null), eq(2L), eq(vm), eq(Collections.emptySet())))
+        when(volumeOrchestrationService.findStoragePool(any(DiskProfile.class), eq(zone), eq(podMock), eq(null), eq(2L), eq(vm), eq(Collections.emptySet())))
                 .thenReturn(pool);
         StoragePool result = volumeApiServiceImpl.getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm);
         Assert.assertNotNull(result);
