--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -180,6 +180,8 @@
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
 
+    LoadBalancerVO loadBalancerVOMock;
+
     TungstenElement tungstenElement;
 
     MockedStatic<ApiDBUtils> apiDBUtilsMocked;
@@ -222,6 +224,7 @@
         encryptionUtilMocked = Mockito.mockStatic(EncryptionUtil.class);
 
         when(tungstenService.getTungstenProjectFqn(any())).thenReturn("default-domain:default-project");
+        loadBalancerVOMock = mock(LoadBalancerVO.class);
     }
 
     @After
@@ -343,12 +346,11 @@
         Ip ip = mock(Ip.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
-        LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
         LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
-        List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
+        List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVOMock);
         TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
@@ -399,12 +401,11 @@
         Ip ip = mock(Ip.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
-        LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
         LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
-        List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
+        List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVOMock);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer createTungstenNetworkLoadbalancerAnswer = MockTungstenAnswerFactory.get(true);
@@ -446,9 +447,8 @@
         Ip ip1 = mock(Ip.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
-        LoadBalancerVO loadBalancerVO1 = mock(LoadBalancerVO.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
-        List<LoadBalancerVO> loadBalancerVOList1 = List.of(loadBalancerVO1);
+        List<LoadBalancerVO> loadBalancerVOList1 = List.of(loadBalancerVOMock);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer deleteTungstenLoadBalancerListenerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenLoadBalancerCommand = MockTungstenAnswerFactory.get(true);
@@ -474,13 +474,12 @@
         Ip ip = mock(Ip.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
-        LoadBalancerVO loadBalancerVO1 = mock(LoadBalancerVO.class);
         LoadBalancerVO loadBalancerVO2 = mock(LoadBalancerVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer deleteTungstenLoadBalancerListenerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenLoadBalancerCommand = MockTungstenAnswerFactory.get(true);
         List<LoadBalancingRule> loadBalancingRuleList = List.of(loadBalancingRule);
-        List<LoadBalancerVO> loadBalancerVOList = Arrays.asList(loadBalancerVO1, loadBalancerVO2);
+        List<LoadBalancerVO> loadBalancerVOList = Arrays.asList(loadBalancerVOMock, loadBalancerVO2);
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), any())).thenReturn(publicNetwork);
         when(loadBalancingRule.getSourceIp()).thenReturn(ip);
