--- a/engine/storage/configdrive/src/test/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilderTest.java
+++ b/engine/storage/configdrive/src/test/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilderTest.java
@@ -152,8 +152,7 @@
 
             configDriveBuilderMocked.when(() -> ConfigDriveBuilder.generateAndRetrieveIsoAsBase64Iso(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).thenAnswer(invocation -> "mockIsoDataBase64");
 
-            NicProfile mockedNicProfile = Mockito.mock(NicProfile.class);
-            Mockito.when(mockedNicProfile.getId()).thenReturn(1L);
+            NicProfile mockedNicProfile = createMockNicProfile();
 
             //force execution of real method
             Mockito.when(ConfigDriveBuilder.buildConfigDrive(List.of(mockedNicProfile), new ArrayList<>(), "teste", "C:", null, supportedServices)).thenCallRealMethod();
@@ -524,8 +523,7 @@
     @Test
     public void testWriteNetworkData() throws Exception {
         // Setup
-        NicProfile nicp = mock(NicProfile.class);
-        Mockito.when(nicp.getId()).thenReturn(1L);
+        NicProfile nicp = createMockNicProfile();
 
         Mockito.when(nicp.getMacAddress()).thenReturn("00:00:00:00:00:00");
         Mockito.when(nicp.getMtu()).thenReturn(2000);
@@ -659,4 +657,10 @@
         Assert.assertEquals(expectedJsonObject, actualJson);
         folder.delete();
     }
+
+    private static NicProfile createMockNicProfile() {
+        NicProfile mockedNicProfile = Mockito.mock(NicProfile.class);
+        Mockito.when(mockedNicProfile.getId()).thenReturn(1L);
+        return mockedNicProfile;
+    }
 }
