--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -785,7 +785,7 @@
         String userName = "username";
         Mockito.doReturn(userName).when(UpdateUserCmdMock).getUsername();
         Mockito.lenient().doReturn(userName).when(userVoMock).getUsername();
-        Mockito.doReturn(domanIdCurrentUser).when(accountMock).getDomainId();
+        accountMock = createMockAccountWithDomainId(domanIdCurrentUser);
 
         long userVoDuplicatedMockId = 67L;
         UserVO userVoDuplicatedMock = Mockito.mock(UserVO.class);
@@ -795,9 +795,8 @@
 
         Mockito.doReturn(accountIdUserDuplicated).when(userVoDuplicatedMock).getAccountId();
 
-        Account accountUserDuplicatedMock = Mockito.mock(AccountVO.class);
+        Account accountUserDuplicatedMock = createMockAccountWithDomainId(domanIdCurrentUser);
         Mockito.lenient().doReturn(accountIdUserDuplicated).when(accountUserDuplicatedMock).getId();
-        Mockito.doReturn(domanIdCurrentUser).when(accountUserDuplicatedMock).getDomainId();
 
         List<UserVO> usersWithSameUserName = new ArrayList<>();
         usersWithSameUserName.add(userVoMock);
@@ -820,7 +819,7 @@
         String userName = "username";
         Mockito.doReturn(userName).when(UpdateUserCmdMock).getUsername();
         Mockito.lenient().doReturn(userName).when(userVoMock).getUsername();
-        Mockito.doReturn(domanIdCurrentUser).when(accountMock).getDomainId();
+        accountMock = createMockAccountWithDomainId(domanIdCurrentUser);
 
         long userVoDuplicatedMockId = 67L;
         UserVO userVoDuplicatedMock = Mockito.mock(UserVO.class);
@@ -830,9 +829,8 @@
         long accountIdUserDuplicated = 98L;
         Mockito.doReturn(accountIdUserDuplicated).when(userVoDuplicatedMock).getAccountId();
 
-        Account accountUserDuplicatedMock = Mockito.mock(AccountVO.class);
+        Account accountUserDuplicatedMock = createMockAccountWithDomainId(45L);
         Mockito.lenient().doReturn(accountIdUserDuplicated).when(accountUserDuplicatedMock).getId();
-        Mockito.doReturn(45L).when(accountUserDuplicatedMock).getDomainId();
 
         List<UserVO> usersWithSameUserName = new ArrayList<>();
         usersWithSameUserName.add(userVoMock);
@@ -2289,4 +2287,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static Account createMockAccountWithDomainId(long domainId) {
+        Account accountUserDuplicatedMock = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(domainId).when(accountUserDuplicatedMock).getDomainId();
+        return accountUserDuplicatedMock;
+    }
 }
