--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerTest.java
@@ -18,16 +18,23 @@
 
 import org.apache.cloudstack.framework.config.ConfigKey;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
 public class AutoScaleManagerTest {
 
+    private AutoScaleManager autoScaleManager;
+
+    @Before
+    public void setUp() {
+        autoScaleManager = Mockito.mock(AutoScaleManager.class);
+    }
+
     @Test
     public void testAutoScaleStatsInterval() {
-        AutoScaleManager asManager = Mockito.mock(AutoScaleManager.class);
 
-        ConfigKey config = asManager.AutoScaleStatsInterval;
+        ConfigKey config = autoScaleManager.AutoScaleStatsInterval;
         Assert.assertEquals("Advanced", config.category());
         Assert.assertEquals(Integer.class, config.type());
         Assert.assertEquals("autoscale.stats.interval", config.key());
@@ -37,9 +44,8 @@
 
     @Test
     public void testAutoScaleStatsCleanupDelay() {
-        AutoScaleManager asManager = Mockito.mock(AutoScaleManager.class);
 
-        ConfigKey config = asManager.AutoScaleStatsCleanupDelay;
+        ConfigKey config = autoScaleManager.AutoScaleStatsCleanupDelay;
         Assert.assertEquals("Advanced", config.category());
         Assert.assertEquals(Integer.class, config.type());
         Assert.assertEquals("autoscale.stats.cleanup.delay", config.key());
@@ -49,9 +55,8 @@
 
     @Test
     public void testAutoScaleStatsWorker() {
-        AutoScaleManager asManager = Mockito.mock(AutoScaleManager.class);
 
-        ConfigKey config = asManager.AutoScaleStatsWorker;
+        ConfigKey config = autoScaleManager.AutoScaleStatsWorker;
         Assert.assertEquals("Advanced", config.category());
         Assert.assertEquals(Integer.class, config.type());
         Assert.assertEquals("autoscale.stats.worker", config.key());
