--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -242,6 +242,7 @@
 
     @Mock
     private LibvirtComputingResource libvirtComputingResourceMock;
+    private final com.cloud.host.Host hostMock = Mockito.mock(com.cloud.host.Host.class);
     @Mock
     VirtualMachineTO vmTO;
     @Mock
@@ -3185,9 +3186,8 @@
 
     @Test
     public void testCheckOnHostCommand() {
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);;
-
-        final CheckOnHostCommand command = new CheckOnHostCommand(host);
+
+        final CheckOnHostCommand command = new CheckOnHostCommand(hostMock);
 
         final KVMHAMonitor monitor = Mockito.mock(KVMHAMonitor.class);
 
@@ -3383,9 +3383,8 @@
     @Test
     public void testFenceCommand() {
         final VirtualMachine vm = Mockito.mock(VirtualMachine.class);;
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
-
-        final FenceCommand command = new FenceCommand(vm, host);
+
+        final FenceCommand command = new FenceCommand(vm, hostMock);
 
         final KVMHAMonitor monitor = Mockito.mock(KVMHAMonitor.class);
 
@@ -5102,10 +5101,9 @@
     @Test
     public void testStartCommandFailedConnect() {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
@@ -5156,10 +5154,9 @@
     @Test
     public void testStartCommandLibvirtException() {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
@@ -5199,10 +5196,9 @@
     @Test
     public void testStartCommandInternalError() {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
@@ -5250,10 +5246,9 @@
     @Test
     public void testStartCommandUriException() {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
@@ -5301,10 +5296,9 @@
     @Test
     public void testStartCommand() throws Exception {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
@@ -5380,10 +5374,9 @@
     @Test
     public void testStartCommandIsolationEc2() throws Exception {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
@@ -5460,10 +5453,9 @@
     @Test
     public void testStartCommandHostMemory() {
         final VirtualMachineTO vmSpec = Mockito.mock(VirtualMachineTO.class);
-        final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
         final boolean executeInSequence = false;
 
-        final StartCommand command = new StartCommand(vmSpec, host, executeInSequence);
+        final StartCommand command = new StartCommand(vmSpec, hostMock, executeInSequence);
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
