--- a/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
@@ -110,6 +110,8 @@
     TaggedResourceService taggedResourceService;
     @Mock
     PrimaryDataStoreDao primaryDataStoreDao;
+    @Mock
+    private SnapshotPolicyVO policy1Mock;
     @InjectMocks
     SnapshotManagerImpl snapshotManager = new SnapshotManagerImpl();
 
@@ -444,9 +446,8 @@
         Mockito.when(caller.getId()).thenReturn(1L);
         CallContext.register(Mockito.mock(User.class), caller);
 
-        SnapshotPolicyVO policy1 = Mockito.mock(SnapshotPolicyVO.class);
         SnapshotPolicyVO policy2 = Mockito.mock(SnapshotPolicyVO.class);
-        List<SnapshotPolicyVO> mockPolicies = List.of(policy1, policy2);
+        List<SnapshotPolicyVO> mockPolicies = List.of(policy1Mock, policy2);
 
         SearchBuilder<SnapshotPolicyVO> mockSearchBuilder = Mockito.mock(SearchBuilder.class);
         SearchBuilder<VolumeVO> mockVolumeSearchBuilder = Mockito.mock(SearchBuilder.class);
@@ -479,9 +480,8 @@
         Mockito.when(caller.getId()).thenReturn(2L);
         CallContext.register(Mockito.mock(User.class), caller);
 
-        SnapshotPolicyVO policy1 = Mockito.mock(SnapshotPolicyVO.class);
         SnapshotPolicyVO policy2 = Mockito.mock(SnapshotPolicyVO.class);
-        List<SnapshotPolicyVO> mockPolicies = List.of(policy1, policy2);
+        List<SnapshotPolicyVO> mockPolicies = List.of(policy1Mock, policy2);
 
         SearchBuilder<SnapshotPolicyVO> mockSearchBuilder = Mockito.mock(SearchBuilder.class);
         SearchBuilder<VolumeVO> mockVolumeSearchBuilder = Mockito.mock(SearchBuilder.class);
@@ -514,7 +514,6 @@
         Mockito.when(caller.getId()).thenReturn(1L);
         CallContext.register(Mockito.mock(User.class), caller);
 
-        SnapshotPolicyVO policy = Mockito.mock(SnapshotPolicyVO.class);
         SearchBuilder<SnapshotPolicyVO> mockSearchBuilder = Mockito.mock(SearchBuilder.class);
         SearchBuilder<VolumeVO> mockVolumeSearchBuilder = Mockito.mock(SearchBuilder.class);
         SearchCriteria<SnapshotPolicyVO> mockSearchCriteria = Mockito.mock(SearchCriteria.class);
@@ -524,7 +523,7 @@
         Mockito.when(mockSearchBuilder.create()).thenReturn(mockSearchCriteria);
         Mockito.when(volumeDao.createSearchBuilder()).thenReturn(mockVolumeSearchBuilder);
         Mockito.when(mockVolumeSearchBuilder.entity()).thenReturn(Mockito.mock(VolumeVO.class));
-        Mockito.when(snapshotPolicyDao.searchAndCount(Mockito.any(), Mockito.any())).thenReturn(new Pair<>(List.of(policy), 1));
+        Mockito.when(snapshotPolicyDao.searchAndCount(Mockito.any(), Mockito.any())).thenReturn(new Pair<>(List.of(policy1Mock), 1));
 
         Pair<List<? extends SnapshotPolicy>, Integer> result = snapshotManager.listSnapshotPolicies(cmd);
 
