--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -254,6 +254,8 @@
     @Mock
     private AccountDao accountDao;
     @Mock
+    private Account orchestrateStartAccount;
+    @Mock
     private DomainDao domainDao;
     @Mock
     private DataCenterDao dcDao;
@@ -1123,7 +1125,6 @@
 
         User user = mock(User.class);
 
-        Account account = mock(Account.class);
         Account owner = mock(Account.class);
 
         ReservationContext ctx = mock(ReservationContext.class);
@@ -1148,7 +1149,7 @@
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
-        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
+        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, orchestrateStartAccount, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
@@ -1187,7 +1188,7 @@
         doReturn(false).when(virtualMachineManagerImpl).areAllVolumesAllocated(Mockito.anyLong());
 
         CallContext callContext = mock(CallContext.class);
-        when(callContext.getCallingAccount()).thenReturn(account);
+        when(callContext.getCallingAccount()).thenReturn(orchestrateStartAccount);
         when(callContext.getCallingUser()).thenReturn(user);
         try (MockedStatic<CallContext> ignored = Mockito.mockStatic(CallContext.class)) {
             when(CallContext.current()).thenReturn(callContext);
@@ -1219,7 +1220,6 @@
 
         User user = mock(User.class);
 
-        Account account = mock(Account.class);
         Account owner = mock(Account.class);
 
         ReservationContext ctx = mock(ReservationContext.class);
@@ -1244,7 +1244,7 @@
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
-        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
+        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, orchestrateStartAccount, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
@@ -1283,7 +1283,7 @@
         doReturn(true).when(virtualMachineManagerImpl).areAllVolumesAllocated(Mockito.anyLong());
 
         CallContext callContext = mock(CallContext.class);
-        when(callContext.getCallingAccount()).thenReturn(account);
+        when(callContext.getCallingAccount()).thenReturn(orchestrateStartAccount);
         when(callContext.getCallingUser()).thenReturn(user);
         try (MockedStatic<CallContext> ignored = Mockito.mockStatic(CallContext.class)) {
             when(CallContext.current()).thenReturn(callContext);
