--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -250,6 +250,8 @@
     @Mock
     private UserVmVO userVmMock;
     @Mock
+    private UserVmVO userVmVOMock;
+    @Mock
     private NetworkDao networkDao;
     @Mock
     private AccountDao accountDao;
@@ -1548,22 +1550,20 @@
     @Test
     public void updateExternalVmDetailsFromPrepareAnswer_nullDetails_noAction() {
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
-        UserVmVO userVmVO = mock(UserVmVO.class);
-        virtualMachineManagerImpl.updateExternalVmDetailsFromPrepareAnswer(vmTO, userVmVO, null);
+        virtualMachineManagerImpl.updateExternalVmDetailsFromPrepareAnswer(vmTO, userVmVOMock, null);
         verify(vmTO, never()).setDetails(any());
-        verify(userVmVO, never()).setDetails(any());
+        verify(userVmVOMock, never()).setDetails(any());
         verify(userVmDaoMock, never()).saveDetails(any());
     }
 
     @Test
     public void updateExternalVmDetailsFromPrepareAnswer_sameDetails_noAction() {
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
-        UserVmVO userVmVO = mock(UserVmVO.class);
         Map<String, String> details = new HashMap<>();
         when(vmTO.getDetails()).thenReturn(details);
-        virtualMachineManagerImpl.updateExternalVmDetailsFromPrepareAnswer(vmTO, userVmVO, details);
+        virtualMachineManagerImpl.updateExternalVmDetailsFromPrepareAnswer(vmTO, userVmVOMock, details);
         verify(vmTO, never()).setDetails(any());
-        verify(userVmVO, never()).setDetails(any());
+        verify(userVmVOMock, never()).setDetails(any());
         verify(userVmDaoMock, never()).saveDetails(any());
     }
 
@@ -1571,12 +1571,11 @@
     public void updateExternalVmDataFromPrepareAnswer_vncPasswordUpdated_updatesPassword() {
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
-        UserVmVO userVmVO = mock(UserVmVO.class);
         when(updatedTO.getVncPassword()).thenReturn("newPassword");
         when(vmTO.getVncPassword()).thenReturn("oldPassword");
-        when(userVmDaoMock.findById(anyLong())).thenReturn(userVmVO);
+        when(userVmDaoMock.findById(anyLong())).thenReturn(userVmVOMock);
         virtualMachineManagerImpl.updateExternalVmDataFromPrepareAnswer(vmTO, updatedTO);
-        verify(userVmVO).setVncPassword("newPassword");
+        verify(userVmVOMock).setVncPassword("newPassword");
         verify(vmTO).setVncPassword("newPassword");
     }
 
