--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -823,7 +823,7 @@
         DataCenter dataCenter = mock(DataCenter.class);
         Nic nic = mock(Nic.class);
         SecurityGroupVO securityGroupVO = mock(SecurityGroupVO.class);
-        SecurityGroupRuleVO securityGroupRuleVO = mock(SecurityGroupRuleVO.class);
+        SecurityGroupRuleVO securityGroupRuleVO = createMockSecurityGroupRuleVO();
         TungstenSecurityGroupRuleVO tungstenSecurityGroupRuleVO = mock(TungstenSecurityGroupRuleVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenAnswer addTungstenSecondaryIpAddressAnswer = MockTungstenAnswerFactory.get(true);
@@ -841,7 +841,6 @@
         when(securityGroupManager.getSecurityGroupsForVm(anyLong())).thenReturn(List.of(securityGroupVO));
         when(securityGroupRuleDao.listByAllowedSecurityGroupId(anyLong())).thenReturn(List.of(securityGroupRuleVO));
         when(tungstenSecurityGroupRuleDao.persist(any(TungstenSecurityGroupRuleVO.class))).thenReturn(tungstenSecurityGroupRuleVO);
-        when(securityGroupRuleVO.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
         when(tungstenProviderDao.findAll()).thenReturn(List.of(tungstenProviderVO));
 
         assertTrue(tungstenService.addTungstenNicSecondaryIpAddress(1L));
@@ -1543,7 +1542,7 @@
         TungstenAnswer addTungstenVmToSecurityGroupAnswer = MockTungstenAnswerFactory.get(true);
         net.juniper.tungsten.api.types.SecurityGroup securityGroup = mock(net.juniper.tungsten.api.types.SecurityGroup.class);
         NicVO nicVO = mock(NicVO.class);
-        SecurityGroupRuleVO securityGroupRuleVO = mock(SecurityGroupRuleVO.class);
+        SecurityGroupRuleVO securityGroupRuleVO = createMockSecurityGroupRuleVO();
         TungstenSecurityGroupRuleVO tungstenSecurityGroupRuleVO = mock(TungstenSecurityGroupRuleVO.class);
         TungstenAnswer addTungstenSecurityGroupRuleAnswer = MockTungstenAnswerFactory.get(true);
 
@@ -1562,7 +1561,6 @@
         when(nicVO.getIPv6Address()).thenReturn("fd00::1");
         when(nicVO.getSecondaryIp()).thenReturn(true);
         when(nicSecIpDao.getSecondaryIpAddressesForNic(anyLong())).thenReturn(List.of("192.168.100.200"));
-        when(securityGroupRuleVO.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
         when(tungstenSecurityGroupRuleDao.persist(any(TungstenSecurityGroupRuleVO.class))).thenReturn(tungstenSecurityGroupRuleVO);
 
         assertTrue(tungstenService.addTungstenVmSecurityGroup(vmInstanceVO));
@@ -1668,4 +1666,10 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static SecurityGroupRuleVO createMockSecurityGroupRuleVO() {
+        SecurityGroupRuleVO securityGroupRuleVO = mock(SecurityGroupRuleVO.class);
+        when(securityGroupRuleVO.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
+        return securityGroupRuleVO;
+    }
 }
