--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -582,7 +582,7 @@
     public void preparePublicNetworkTest() throws ConcurrentOperationException {
         Network network = mock(Network.class);
         NicProfile nicProfile = new NicProfile();
-        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
+        VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.ConsoleProxy);
         DeployDestination deployDestination = mock(DeployDestination.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
@@ -598,7 +598,6 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(createTungstenNetworkPolicyAnswer);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
 
         assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext));
         assertEquals(Nic.ReservationStrategy.Create, nicProfile.getReservationStrategy());
@@ -611,7 +610,6 @@
     public void prepareManagementNetworkTest() throws ConcurrentOperationException {
         Network network = mock(Network.class);
         NicProfile nicProfile = new NicProfile();
-        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
@@ -623,7 +621,7 @@
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
+        VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.SecondaryStorageVm);
 
         assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext));
         assertEquals(Nic.ReservationStrategy.Create, nicProfile.getReservationStrategy());
@@ -636,7 +634,6 @@
     public void prepareWithExceptionTest() throws ConcurrentOperationException {
         Network network = mock(Network.class);
         NicProfile nicProfile = new NicProfile();
-        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
@@ -648,7 +645,7 @@
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
+        VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.SecondaryStorageVm);
 
         tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext);
     }
@@ -657,7 +654,6 @@
     public void releasePublicNetworkTest() throws ConcurrentOperationException, ResourceUnavailableException {
         Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
-        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
@@ -671,7 +667,7 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
+        VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.SecondaryStorageVm);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
@@ -685,7 +681,7 @@
     public void releaseManagementNetworkTest() throws ConcurrentOperationException, ResourceUnavailableException {
         Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
-        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
+        VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.SecondaryStorageVm);
         ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
@@ -696,7 +692,6 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswer);
@@ -708,7 +703,6 @@
     public void releaseWithExceptionTest() throws ConcurrentOperationException, ResourceUnavailableException {
         Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
-        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO host = mock(HostVO.class);
@@ -718,7 +712,7 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
+        VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.SecondaryStorageVm);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenThrow(IllegalArgumentException.class);
@@ -911,14 +905,13 @@
     public void prepareMigrationTest() {
         NicProfile nic = mock(NicProfile.class);
         Network network = mock(Network.class);
-        VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
+        VirtualMachineProfile vm = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.ConsoleProxy);
         DeployDestination dest = mock(DeployDestination.class);
         ReservationContext context = mock(ReservationContext.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         HostVO hostVO = mock(HostVO.class);
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(true);
 
-        when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
@@ -929,6 +922,23 @@
 
     @Test
     public void rollbackMigration() {
+        NicProfile nic = mock(NicProfile.class);
+        Network network = mock(Network.class);
+        ReservationContext src = mock(ReservationContext.class);
+        ReservationContext dest = mock(ReservationContext.class);
+        VirtualMachine virtualMachine = mock(VirtualMachine.class);
+        HostVO hostVO = mock(HostVO.class);
+
+        VirtualMachineProfile vm = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.ConsoleProxy);
+        when(vm.getVirtualMachine()).thenReturn(virtualMachine);
+        when(hostDao.findById(anyLong())).thenReturn(hostVO);
+
+        tungstenElement.rollbackMigration(nic, network, vm, src, dest);
+        verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
+    }
+
+    @Test
+    public void commitMigration() {
         NicProfile nic = mock(NicProfile.class);
         Network network = mock(Network.class);
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
@@ -941,24 +951,6 @@
         when(vm.getVirtualMachine()).thenReturn(virtualMachine);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        tungstenElement.rollbackMigration(nic, network, vm, src, dest);
-        verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
-    }
-
-    @Test
-    public void commitMigration() {
-        NicProfile nic = mock(NicProfile.class);
-        Network network = mock(Network.class);
-        VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
-        ReservationContext src = mock(ReservationContext.class);
-        ReservationContext dest = mock(ReservationContext.class);
-        VirtualMachine virtualMachine = mock(VirtualMachine.class);
-        HostVO hostVO = mock(HostVO.class);
-
-        when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
-        when(vm.getVirtualMachine()).thenReturn(virtualMachine);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
-
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -243,12 +243,11 @@
     public void testDeallocate() {
         final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
-        final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
+        final VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.User);
         final NicVO nicVO = mock(NicVO.class);
 
         when(network.getDataCenterId()).thenReturn(1L);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
         when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
 
         guru.deallocate(network, nicProfile, virtualMachineProfile);
@@ -260,14 +259,14 @@
     public void testDeallocateWithException() {
         final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
-        final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
+        final VirtualMachineProfile virtualMachineProfile = MockVirtualMachineProfile
+            .createMockVirtualMachineProfile(VirtualMachine.Type.User);
         final NicVO nicVO = mock(NicVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenThrow(
             new IllegalArgumentException());
         when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
-        when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
 
         guru.deallocate(network, nicProfile, virtualMachineProfile);
     }
@@ -379,14 +378,13 @@
     public void testReserve() throws InsufficientVirtualNetworkCapacityException, InsufficientAddressCapacityException {
         final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
-        final VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
+        final VirtualMachineProfile vm = MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.User);
         final DeployDestination dest = mock(DeployDestination.class);
         final ReservationContext context = mock(ReservationContext.class);
         final VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         final HostVO host = mock(HostVO.class);
 
         when(nic.getReservationStrategy()).thenReturn(Nic.ReservationStrategy.Start);
-        when(vm.getType()).thenReturn(VirtualMachine.Type.User);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
@@ -473,13 +471,13 @@
     public void testPrepareMigration() {
         final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
-        final VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
+        final VirtualMachineProfile vm = MockVirtualMachineProfile.createMockVirtualMachineProfile(
+                VirtualMachine.Type.User);
         final DeployDestination dest = mock(DeployDestination.class);
         final ReservationContext context = mock(ReservationContext.class);
         final VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         final HostVO hostVO = mock(HostVO.class);
 
-        when(vm.getType()).thenReturn(VirtualMachine.Type.User);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
@@ -495,12 +493,12 @@
     public void testRollbackMigration() {
         final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
-        final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
         final VirtualMachine vm = mock(VirtualMachine.class);
         final HostVO hostVO = mock(HostVO.class);
 
-        when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
+        final VirtualMachineProfile vmProfile =
+            MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.User);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
@@ -513,12 +511,12 @@
     public void testCommitMigration() {
         final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
-        final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
+        final VirtualMachineProfile vmProfile =
+            MockVirtualMachineProfile.createMockVirtualMachineProfile(VirtualMachine.Type.User);
         final ReservationContext context = mock(ReservationContext.class);
         final VirtualMachine vm = mock(VirtualMachine.class);
         final HostVO hostVO = mock(HostVO.class);
 
-        when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/MockVirtualMachineProfile.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/MockVirtualMachineProfile.java
@@ -0,0 +1,16 @@
+package org.apache.cloudstack.network.tungsten.service;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.cloud.vm.VirtualMachine;
+import com.cloud.vm.VirtualMachineProfile;
+
+public class MockVirtualMachineProfile {
+
+    public static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachine.Type type) {
+        VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
+        when(virtualMachineProfile.getType()).thenReturn(type);
+        return virtualMachineProfile;
+    }
+}
