--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1273,8 +1273,8 @@
         when(zoneMock.isLocalStorageEnabled()).thenReturn(false);
         when(diskOfferingMock.isUseLocalStorage()).thenReturn(false);
 
+        networkMock = createMockNetwork();
         Mockito.doReturn(networkMock).when(autoScaleManagerImplSpy).getNetwork(loadBalancerId);
-        when(networkMock.getId()).thenReturn(networkId);
 
         when(zoneMock.getNetworkType()).thenReturn(DataCenter.NetworkType.Basic);
         when(userVmService.createBasicSecurityGroupVirtualMachine(any(), any(), any(), any(), any(), any(), any(),
@@ -1320,8 +1320,8 @@
         when(zoneMock.isLocalStorageEnabled()).thenReturn(false);
         when(diskOfferingMock.isUseLocalStorage()).thenReturn(false);
 
+        networkMock = createMockNetwork();
         Mockito.doReturn(networkMock).when(autoScaleManagerImplSpy).getNetwork(loadBalancerId);
-        when(networkMock.getId()).thenReturn(networkId);
 
         when(zoneMock.getNetworkType()).thenReturn(DataCenter.NetworkType.Advanced);
         when(userVmService.createAdvancedSecurityGroupVirtualMachine(any(), any(), any(), any(), any(), any(), any(),
@@ -1369,8 +1369,8 @@
         when(zoneMock.isLocalStorageEnabled()).thenReturn(false);
         when(diskOfferingMock.isUseLocalStorage()).thenReturn(false);
 
+        networkMock = createMockNetwork();
         Mockito.doReturn(networkMock).when(autoScaleManagerImplSpy).getNetwork(loadBalancerId);
-        when(networkMock.getId()).thenReturn(networkId);
 
         when(zoneMock.getNetworkType()).thenReturn(DataCenter.NetworkType.Advanced);
         when(userVmService.createAdvancedVirtualMachine(any(), any(), any(), any(), any(), any(), any(),
@@ -1394,9 +1394,8 @@
 
     @Test
     public void getVmNetworkIds1() {
-        NetworkVO networkMock = Mockito.mock(NetworkVO.class);
+        NetworkVO networkMock = createMockNetwork();
         when(networkDao.findByUuid(networkUuid)).thenReturn(networkMock);
-        when(networkMock.getId()).thenReturn(networkId);
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("networkids", networkUuid);
@@ -1407,9 +1406,8 @@
 
     @Test
     public void getVmNetworkIds2() {
-        NetworkVO networkMock = Mockito.mock(NetworkVO.class);
+        NetworkVO networkMock = createMockNetwork();
         when(networkDao.findByUuid(networkUuid)).thenReturn(networkMock);
-        when(networkMock.getId()).thenReturn(networkId);
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("networkids", networkUuid);
@@ -2322,8 +2320,8 @@
     public void getNetworkStatsFromVirtualRouterWithoutRouter() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         when(groupTO.getLoadBalancerId()).thenReturn(loadBalancerId);
+        networkMock = createMockNetwork();
         Mockito.doReturn(networkMock).when(autoScaleManagerImplSpy).getNetwork(loadBalancerId);
-        when(networkMock.getId()).thenReturn(networkId);
         Pair<String, Integer> publicIpAddr = new Pair<>(ipAddress, memberPort);
         Mockito.doReturn(publicIpAddr).when(autoScaleManagerImplSpy).getPublicIpAndPort(loadBalancerId);
         when(routerDao.listByNetworkAndRole(networkId, VirtualRouter.Role.VIRTUAL_ROUTER)).thenReturn(new ArrayList<>());
@@ -2338,8 +2336,8 @@
     public void getNetworkStatsFromVirtualRouterWithOneRouter() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         when(groupTO.getLoadBalancerId()).thenReturn(loadBalancerId);
+        networkMock = createMockNetwork();
         Mockito.doReturn(networkMock).when(autoScaleManagerImplSpy).getNetwork(loadBalancerId);
-        when(networkMock.getId()).thenReturn(networkId);
         Pair<String, Integer> publicIpAddr = new Pair<>(ipAddress, memberPort);
         Mockito.doReturn(publicIpAddr).when(autoScaleManagerImplSpy).getPublicIpAndPort(loadBalancerId);
         when(routerDao.listByNetworkAndRole(networkId, VirtualRouter.Role.VIRTUAL_ROUTER)).thenReturn(Arrays.asList(domainRouterMock));
@@ -2512,6 +2510,12 @@
                 "-" + asVmGroupMock.getNextVmSeq() + "-[a-z]{6}";
         Assert.assertTrue(result.first().matches(vmHostNamePattern));
         Assert.assertEquals(result.first(), result.second());
+    }
+
+    private static NetworkVO createMockNetwork() {
+        NetworkVO networkMock = Mockito.mock(NetworkVO.class);
+        when(networkMock.getId()).thenReturn(networkId);
+        return networkMock;
     }
 
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
