--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -2324,7 +2324,6 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk tmplVol = Mockito.mock(KVMPhysicalDisk.class);
         final KVMPhysicalDisk primaryVol = Mockito.mock(KVMPhysicalDisk.class);
 
@@ -2335,8 +2334,8 @@
         final String mountpoint = url.substring(0, index);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool secondaryPool = createMockSecondaryPool(disks);
         when(storagePoolMgr.getStoragePoolByURI(mountpoint)).thenReturn(secondaryPool);
-        when(secondaryPool.listPhysicalDisks()).thenReturn(disks);
         when(storagePoolMgr.getStoragePool(command.getPool().getType(), command.getPoolUuid())).thenReturn(primaryPool);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
@@ -2363,7 +2362,7 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
+        final KVMStoragePool secondaryPool = createMockSecondaryPool(disks);
         final KVMPhysicalDisk tmplVol = Mockito.mock(KVMPhysicalDisk.class);
         final KVMPhysicalDisk primaryVol = Mockito.mock(KVMPhysicalDisk.class);
 
@@ -2372,7 +2371,6 @@
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
         when(storagePoolMgr.getStoragePoolByURI(mountpoint)).thenReturn(secondaryPool);
-        when(secondaryPool.listPhysicalDisks()).thenReturn(disks);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -2399,7 +2397,6 @@
 
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
-        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
         final KVMPhysicalDisk tmplVol = Mockito.mock(KVMPhysicalDisk.class);
         final KVMPhysicalDisk primaryVol = Mockito.mock(KVMPhysicalDisk.class);
 
@@ -2407,8 +2404,8 @@
         final String mountpoint = url.substring(0, index);
 
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
+        final KVMStoragePool secondaryPool = createMockSecondaryPool(spiedDisks);
         when(storagePoolMgr.getStoragePoolByURI(mountpoint)).thenReturn(secondaryPool);
-        when(secondaryPool.listPhysicalDisks()).thenReturn(spiedDisks);
         when(spiedDisks.isEmpty()).thenReturn(false);
 
         final LibvirtRequestWrapper wrapper = LibvirtRequestWrapper.getInstance();
@@ -7698,4 +7695,10 @@
         Pair<Integer, Long> result = libvirtComputingResourceSpy.getPeriodAndQuota(cpuQuotaPercentage);
         Assert.assertEquals(expectedResult, result);
     }
+
+    private static KVMStoragePool createMockSecondaryPool(List<KVMPhysicalDisk> physicalDisks) {
+        final KVMStoragePool secondaryPool = Mockito.mock(KVMStoragePool.class);
+        when(secondaryPool.listPhysicalDisks()).thenReturn(physicalDisks);
+        return secondaryPool;
+    }
 }
