--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
@@ -90,12 +90,7 @@
         }
 
         // Create Virtual=Machine-Interface (VMInterface)
-        NicVO nic = mock(NicVO.class);
-        when(nic.getIPv4Address()).thenReturn("10.1.1.2");
-        when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
-        when(nic.getDeviceId()).thenReturn(100);
-        when(nic.getState()).thenReturn(NicVO.State.Allocated);
-        when(nic.getNetworkId()).thenReturn(10L);
+        NicVO nic = MockNicVO.createMockNic();
 
         when(contrailMgr.getVifNameByVmName(anyString(), anyInt())).thenReturn("testVM1-100");
 
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
@@ -92,12 +92,7 @@
         }
 
         // Create Virtual=Machine-Interface (VMInterface)
-        NicVO nic = mock(NicVO.class);
-        when(nic.getIPv4Address()).thenReturn("10.1.1.2");
-        when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
-        when(nic.getDeviceId()).thenReturn(100);
-        when(nic.getState()).thenReturn(NicVO.State.Allocated);
-        when(nic.getNetworkId()).thenReturn(10L);
+        NicVO nic = MockNicVO.createMockNic();
 
         when(contrailMgr.getVifNameByVmName(anyString(), anyInt())).thenReturn("testVM1-100");
 
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/MockNicVO.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/MockNicVO.java
@@ -0,0 +1,18 @@
+package org.apache.cloudstack.network.contrail.model;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.cloud.vm.NicVO;
+
+public class MockNicVO {
+    public static NicVO createMockNic() {
+        NicVO nic = mock(NicVO.class);
+        when(nic.getIPv4Address()).thenReturn("10.1.1.2");
+        when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
+        when(nic.getDeviceId()).thenReturn(100);
+        when(nic.getState()).thenReturn(NicVO.State.Allocated);
+        when(nic.getNetworkId()).thenReturn(10L);
+        return nic;
+    }
+}
