--- a/server/src/test/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinitionTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinitionTest.java
@@ -82,6 +82,7 @@
     protected NsxProviderDao nsxProviderDao;
     @Mock
     protected NetrisProviderDao netrisProviderDao;
+    private final Account systemAccountOwner = mock(Account.class);
 
     protected RouterDeploymentDefinition deployment;
 
@@ -743,12 +744,11 @@
     public void testSetupAccountOwner() {
         // Prepare
         when(mockNetworkModel.isNetworkSystem(mockNw)).thenReturn(true);
-        final Account newAccountOwner = mock(Account.class);
-        when(mockAccountMgr.getAccount(Account.ACCOUNT_ID_SYSTEM)).thenReturn(newAccountOwner);
+        when(mockAccountMgr.getAccount(Account.ACCOUNT_ID_SYSTEM)).thenReturn(systemAccountOwner);
         //Execute
         deployment.setupAccountOwner();
         // Assert
-        assertEquals("New account owner not properly set", newAccountOwner, deployment.owner);
+        assertEquals("New account owner not properly set", systemAccountOwner, deployment.owner);
     }
 
     @Test
@@ -756,12 +756,11 @@
         // Prepare
         when(mockNetworkModel.isNetworkSystem(mockNw)).thenReturn(false);
         when(mockNw.getGuestType()).thenReturn(Network.GuestType.Shared);
-        final Account newAccountOwner = mock(Account.class);
-        when(mockAccountMgr.getAccount(Account.ACCOUNT_ID_SYSTEM)).thenReturn(newAccountOwner);
+        when(mockAccountMgr.getAccount(Account.ACCOUNT_ID_SYSTEM)).thenReturn(systemAccountOwner);
         //Execute
         deployment.setupAccountOwner();
         // Assert
-        assertEquals("New account owner not properly set", newAccountOwner, deployment.owner);
+        assertEquals("New account owner not properly set", systemAccountOwner, deployment.owner);
     }
 
     @Test
