--- a/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/LoadBalanceRuleHandlerTest.java
+++ b/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/LoadBalanceRuleHandlerTest.java
@@ -129,11 +129,9 @@
 
     @Test
     public void testStartWhenParamsIsNull() throws Exception {
-        DomainRouterVO elbVmMock = mock(DomainRouterVO.class);
         String uuid = "uuid";
-        when(elbVmMock.getUuid()).thenReturn(uuid);
+        DomainRouterVO elbVmMock = createMockElbVm();
         long id = 1L;
-        when(elbVmMock.getId()).thenReturn(id);
         Map<Param, Object> params = null;
 
         loadBalanceRuleHandler.start(elbVmMock, params);
@@ -144,11 +142,9 @@
 
     @Test
     public void testStartWhenParamsIsEmpty() throws Exception {
-        DomainRouterVO elbVmMock = mock(DomainRouterVO.class);
+        DomainRouterVO elbVmMock = createMockElbVm();
         String uuid = "uuid";
-        when(elbVmMock.getUuid()).thenReturn(uuid);
         long id = 1L;
-        when(elbVmMock.getId()).thenReturn(id);
         Map<Param, Object> params = new HashMap<Param, Object>();
 
         loadBalanceRuleHandler.start(elbVmMock, params);
@@ -159,11 +155,9 @@
 
     @Test
     public void testStart() throws Exception {
-        DomainRouterVO elbVmMock = mock(DomainRouterVO.class);
+        DomainRouterVO elbVmMock = createMockElbVm();
         String uuid = "uuid";
-        when(elbVmMock.getUuid()).thenReturn(uuid);
         long id = 1L;
-        when(elbVmMock.getId()).thenReturn(id);
         Map<Param, Object> params = new HashMap<Param, Object>();
         params.put(mock(Param.class), new Object());
 
@@ -212,4 +206,11 @@
         assertEquals(expected, actual);
     }
 
+    private static DomainRouterVO createMockElbVm() {
+        DomainRouterVO elbVmMock = mock(DomainRouterVO.class);
+        when(elbVmMock.getUuid()).thenReturn("uuid");
+        when(elbVmMock.getId()).thenReturn(1L);
+        return elbVmMock;
+    }
+
 }
