--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -170,6 +170,9 @@
     VolumeApiService volumeApiService;
 
     @Mock
+    private VolumeVO volumeVOMock;
+
+    @Mock
     VolumeDao volumeDao;
 
     @Mock
@@ -542,7 +545,6 @@
     @Test
     public void tryRestoreVMTestRestoreSucceeded() throws NoTransitionException {
         BackupOffering offering = mock(BackupOffering.class);
-        VolumeVO volumeVO = mock(VolumeVO.class);
         VMInstanceVO vm = mock(VMInstanceVO.class);
         BackupVO backup = mock(BackupVO.class);
 
@@ -554,9 +556,9 @@
                     Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.anyLong(),
                     Mockito.anyString(), Mockito.eq(0))).thenReturn(2L);
 
-            Mockito.when(volumeDao.findIncludingRemovedByInstanceAndType(1L, null)).thenReturn(Collections.singletonList(volumeVO));
+            Mockito.when(volumeDao.findIncludingRemovedByInstanceAndType(1L, null)).thenReturn(Collections.singletonList(volumeVOMock));
             Mockito.when(virtualMachineManager.stateTransitTo(Mockito.eq(vm), Mockito.eq(VirtualMachine.Event.RestoringRequested), Mockito.any())).thenReturn(true);
-            Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVO), Mockito.eq(Volume.Event.RestoreRequested))).thenReturn(true);
+            Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVOMock), Mockito.eq(Volume.Event.RestoreRequested))).thenReturn(true);
 
             Mockito.when(vm.getId()).thenReturn(1L);
             Mockito.when(offering.getProvider()).thenReturn("veeam");
@@ -570,7 +572,6 @@
     @Test
     public void tryRestoreVMTestRestoreFails() throws NoTransitionException {
         BackupOffering offering = mock(BackupOffering.class);
-        VolumeVO volumeVO = mock(VolumeVO.class);
         VMInstanceVO vm = mock(VMInstanceVO.class);
         BackupVO backup = mock(BackupVO.class);
 
@@ -582,11 +583,11 @@
                     Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.anyLong(),
                     Mockito.anyString(), Mockito.eq(0))).thenReturn(2L);
 
-            Mockito.when(volumeDao.findIncludingRemovedByInstanceAndType(1L, null)).thenReturn(Collections.singletonList(volumeVO));
+            Mockito.when(volumeDao.findIncludingRemovedByInstanceAndType(1L, null)).thenReturn(Collections.singletonList(volumeVOMock));
             Mockito.when(virtualMachineManager.stateTransitTo(Mockito.eq(vm), Mockito.eq(VirtualMachine.Event.RestoringRequested), Mockito.any())).thenReturn(true);
-            Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVO), Mockito.eq(Volume.Event.RestoreRequested))).thenReturn(true);
+            Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVOMock), Mockito.eq(Volume.Event.RestoreRequested))).thenReturn(true);
             Mockito.when(virtualMachineManager.stateTransitTo(Mockito.eq(vm), Mockito.eq(VirtualMachine.Event.RestoringFailed), Mockito.any())).thenReturn(true);
-            Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVO), Mockito.eq(Volume.Event.RestoreFailed))).thenReturn(true);
+            Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVOMock), Mockito.eq(Volume.Event.RestoreFailed))).thenReturn(true);
 
             Mockito.when(vm.getId()).thenReturn(1L);
             Mockito.when(offering.getProvider()).thenReturn("veeam");
@@ -2123,8 +2124,7 @@
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn("testbackupprovider");
 
-        VolumeVO volume = mock(VolumeVO.class);
-        when(volumeDao.findByInstance(vmId)).thenReturn(Collections.singletonList(volume));
+        when(volumeDao.findByInstance(vmId)).thenReturn(Collections.singletonList(volumeVOMock));
 
         BackupProvider backupProvider = mock(BackupProvider.class);
         when(backupProvider.restoreVMFromBackup(vm, backup, false, null)).thenReturn(true);
@@ -2255,8 +2255,7 @@
         when(vm.getRemoved()).thenReturn(null);
         when(vm.getBackupVolumeList()).thenReturn(Collections.emptyList());
 
-        VolumeVO volume = mock(VolumeVO.class);
-        when(volumeDao.findByInstance(vmId)).thenReturn(Collections.singletonList(volume));
+        when(volumeDao.findByInstance(vmId)).thenReturn(Collections.singletonList(volumeVOMock));
 
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(vmInstanceDao.findByIdIncludingRemoved(vmId)).thenReturn(vm);
