--- a/server/src/test/java/org/apache/cloudstack/backup/InternalBackupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/InternalBackupServiceImplTest.java
@@ -124,6 +124,9 @@
     @Mock
     private VMSnapshot vmSnapshotMock;
 
+    @Mock
+    private Command commandMock;
+
     @Spy
     @InjectMocks
     private InternalBackupServiceImpl internalBackupServiceImplSpy;
@@ -140,9 +143,8 @@
     @Test
     public void configureChainInfoTestNonVolumeObjectReturnsImmediately() {
         DataTO dataToMock = mock(DataTO.class);
-        Command cmdMock = mock(Command.class);
-
-        internalBackupServiceImplSpy.configureChainInfo(dataToMock, cmdMock);
+
+        internalBackupServiceImplSpy.configureChainInfo(dataToMock, commandMock);
 
         verify(internalBackupStoragePoolDaoMock, never()).findOneByVolumeId(anyLong());
     }
@@ -164,9 +166,8 @@
         doReturn(internalBackupStoragePoolVoMock).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(VOLUME_ID);
         doReturn("/path/to/parent").when(internalBackupStoragePoolVoMock).getBackupDeltaParentPath();
 
-        Command cmdMock = mock(Command.class);
-
-        internalBackupServiceImplSpy.configureChainInfo(volumeObjectToMock, cmdMock);
+
+        internalBackupServiceImplSpy.configureChainInfo(volumeObjectToMock, commandMock);
 
         verify(internalBackupStoragePoolDaoMock).findOneByVolumeId(VOLUME_ID);
         verify(volumeObjectToMock).setChainInfo("/path/to/parent");
