--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1688,8 +1688,7 @@
 
     @Test
     public void getHostAndVmIdsMap() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        when(groupTO.getId()).thenReturn(vmGroupId);
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO();
         when(autoScaleVmGroupVmMapDao.listByGroup(vmGroupId)).thenReturn(Arrays.asList(autoScaleVmGroupVmMapVOMock));
         when(autoScaleVmGroupVmMapVOMock.getInstanceId()).thenReturn(virtualMachineId);
         when(userVmDao.findById(virtualMachineId)).thenReturn(userVmMock);
@@ -2294,8 +2293,7 @@
 
     @Test
     public void processVmStatsByIdFromHost() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        when(groupTO.getId()).thenReturn(vmGroupId);
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO();
 
         List<Long> vmIds = Arrays.asList(virtualMachineId);
         double cpuUtilization = 6;
@@ -2512,6 +2510,12 @@
                 "-" + asVmGroupMock.getNextVmSeq() + "-[a-z]{6}";
         Assert.assertTrue(result.first().matches(vmHostNamePattern));
         Assert.assertEquals(result.first(), result.second());
+    }
+
+    private static AutoScaleVmGroupTO createMockAutoScaleVmGroupTO() {
+        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
+        when(groupTO.getId()).thenReturn(vmGroupId);
+        return groupTO;
     }
 
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
