--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -202,6 +202,12 @@
         return sharedFS;
     }
 
+    private static NetworkVO createMockNetworkVO() {
+        NetworkVO networkVO = mock(NetworkVO.class);
+        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        return networkVO;
+    }
+
     @Test
     public void testDeploySharedFS() throws ResourceUnavailableException, InsufficientCapacityException, ResourceAllocationException, NoTransitionException, OperationTimedoutException {
         CreateSharedFSCmd cmd = getMockCreateSharedFSCmd();
@@ -253,8 +259,7 @@
         ReflectionTestUtils.setField(sharedFS, "id", s_sharedFSId);
 
         when(cmd.getNetworkId()).thenReturn(s_networkId);
-        NetworkVO networkVO = mock(NetworkVO.class);
-        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        NetworkVO networkVO = createMockNetworkVO();
         when(networkDao.findById(s_networkId)).thenReturn(networkVO);
 
         sharedFSServiceImpl.allocSharedFS(cmd);
@@ -312,8 +317,7 @@
         when(diskOfferingVO.isCustomizedIops()).thenReturn(true);
 
         when(cmd.getNetworkId()).thenReturn(s_networkId);
-        NetworkVO networkVO = mock(NetworkVO.class);
-        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        NetworkVO networkVO = createMockNetworkVO();
         when(networkDao.findById(s_networkId)).thenReturn(networkVO);
 
         when(cmd.getFsFormat()).thenReturn("ext2");
