--- a/plugins/hypervisors/external/src/test/java/org/apache/cloudstack/hypervisor/external/provisioner/ExternalPathPayloadProvisionerTest.java
+++ b/plugins/hypervisors/external/src/test/java/org/apache/cloudstack/hypervisor/external/provisioner/ExternalPathPayloadProvisionerTest.java
@@ -121,6 +121,9 @@
     @Mock
     private ExtensionsManager extensionsManager;
 
+    @Mock
+    private VirtualMachineTO vmTOMock;
+
     private File tempDir;
     private File tempDataDir;
     private Properties testProperties;
@@ -476,8 +479,7 @@
         when(userVmDao.listByLastHostId(anyLong())).thenReturn(new ArrayList<>());
 
         when(hypervisorGuruManager.getGuru(Hypervisor.HypervisorType.External)).thenReturn(hypervisorGuru);
-        VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
-        when(hypervisorGuru.implement(any(VirtualMachineProfile.class))).thenReturn(vmTO);
+        when(hypervisorGuru.implement(any(VirtualMachineProfile.class))).thenReturn(vmTOMock);
 
         doReturn(new Pair<>(true, "PowerOn")).when(provisioner)
             .executeExternalCommand(anyString(), anyString(), anyMap(), anyInt(), anyString(), anyString());
@@ -796,12 +798,11 @@
     @Test
     public void getVirtualMachineTOReturnsValidTOWhenVmIsNotNull() {
         VirtualMachine vm = mock(VirtualMachine.class);
-        VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         when(hypervisorGuruManager.getGuru(Hypervisor.HypervisorType.External)).thenReturn(hypervisorGuru);
-        when(hypervisorGuru.implement(any(VirtualMachineProfile.class))).thenReturn(vmTO);
+        when(hypervisorGuru.implement(any(VirtualMachineProfile.class))).thenReturn(vmTOMock);
         VirtualMachineTO result = provisioner.getVirtualMachineTO(vm);
         assertNotNull(result);
-        assertEquals(vmTO, result);
+        assertEquals(vmTOMock, result);
         Mockito.verify(hypervisorGuruManager).getGuru(Hypervisor.HypervisorType.External);
         Mockito.verify(hypervisorGuru).implement(any(VirtualMachineProfile.class));
     }
