--- a/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
@@ -221,8 +221,7 @@
     public void testValidatePolicyZonesVolumeEdgeZone() {
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getDataCenterId()).thenReturn(1L);
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Edge);
+        DataCenterVO zone = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Edge);
         Mockito.when(dataCenterDao.findById(1L)).thenReturn(zone);
         snapshotManager.validatePolicyZones(List.of(1L), null, volumeVO, Mockito.mock(Account.class));
     }
@@ -231,8 +230,7 @@
     public void testValidatePolicyZonesNullZone() {
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getDataCenterId()).thenReturn(1L);
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Core);
+        DataCenterVO zone = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Core);
         Mockito.when(dataCenterDao.findById(1L)).thenReturn(zone);
         Mockito.when(dataCenterDao.findById(2L)).thenReturn(null);
         snapshotManager.validatePolicyZones(List.of(2L), null, volumeVO, Mockito.mock(Account.class));
@@ -242,8 +240,7 @@
     public void testValidatePolicyZonesDisabledZone() {
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getDataCenterId()).thenReturn(1L);
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Core);
+        DataCenterVO zone = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Core);
         Mockito.when(dataCenterDao.findById(1L)).thenReturn(zone);
         DataCenterVO zone1 = Mockito.mock(DataCenterVO.class);
         Mockito.when(zone1.getAllocationState()).thenReturn(Grouping.AllocationState.Disabled);
@@ -256,11 +253,9 @@
     public void testValidatePolicyZonesEdgeZone() {
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getDataCenterId()).thenReturn(1L);
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Core);
+        DataCenterVO zone = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Core);
         Mockito.when(dataCenterDao.findById(1L)).thenReturn(zone);
-        DataCenterVO zone1 = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone1.getType()).thenReturn(DataCenter.Type.Edge);
+        DataCenterVO zone1 = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Edge);
         Mockito.when(zone1.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
         Mockito.when(dataCenterDao.findById(2L)).thenReturn(zone1);
         snapshotManager.validatePolicyZones(List.of(2L), null, volumeVO, Mockito.mock(Account.class));
@@ -270,11 +265,9 @@
     public void testValidatePolicyZonesValidZone() {
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         Mockito.when(volumeVO.getDataCenterId()).thenReturn(1L);
-        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone.getType()).thenReturn(DataCenter.Type.Core);
+        DataCenterVO zone = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Core);
         Mockito.when(dataCenterDao.findById(1L)).thenReturn(zone);
-        DataCenterVO zone1 = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone1.getType()).thenReturn(DataCenter.Type.Core);
+        DataCenterVO zone1 = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Core);
         Mockito.when(zone1.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
         Mockito.when(dataCenterDao.findById(2L)).thenReturn(zone1);
         snapshotManager.validatePolicyZones(List.of(2L), null, volumeVO, Mockito.mock(Account.class));
@@ -424,9 +417,8 @@
     @Test
     public void testGetCheckedDestinationZoneForSnapshotCopy() {
         long zoneId = 1L;
-        DataCenterVO dataCenterVO = Mockito.mock(DataCenterVO.class);
+        DataCenterVO dataCenterVO = MockDataCenterVO.createMockDataCenterVO(DataCenter.Type.Core);
         Mockito.when(dataCenterVO.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
-        Mockito.when(dataCenterVO.getType()).thenReturn(DataCenter.Type.Core);
         Mockito.when(dataCenterDao.findById(zoneId)).thenReturn(dataCenterVO);
         Assert.assertNotNull(snapshotManager.getCheckedDestinationZoneForSnapshotCopy(zoneId, false));
     }
--- a/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerTest.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerTest.java
@@ -541,8 +541,7 @@
         } catch (NoSuchFieldException | IllegalAccessException e) {
             Assert.fail(String.format("Failed to override config: %s", e.getMessage()));
         }
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
-        Mockito.when(dc.getType()).thenReturn(dcType);
+        DataCenterVO dc = MockDataCenterVO.createMockDataCenterVO(dcType);
         Mockito.when(dataCenterDao.findById(1L)).thenReturn(dc);
     }
     @Test
--- a/server/src/test/java/com/cloud/storage/snapshot/MockDataCenterVO.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/MockDataCenterVO.java
@@ -0,0 +1,13 @@
+package com.cloud.storage.snapshot;
+
+import com.cloud.dc.DataCenter;
+import com.cloud.dc.DataCenterVO;
+import org.mockito.Mockito;
+
+public class MockDataCenterVO {
+    public static DataCenterVO createMockDataCenterVO(DataCenter.Type type) {
+        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
+        Mockito.when(zone.getType()).thenReturn(type);
+        return zone;
+    }
+}
