--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -249,8 +249,7 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
@@ -308,8 +307,7 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
@@ -362,8 +360,7 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
@@ -414,8 +411,7 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
@@ -459,8 +455,7 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         final List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
@@ -495,8 +490,7 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
@@ -534,8 +528,7 @@
         when(mapping.getVlanId()).thenReturn(14);
         when(vcsmapdao.findByNetworkId(anyLong())).thenReturn(mapping);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
 
         final List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
@@ -588,4 +581,10 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    private static BrocadeVcsDeviceVO createMockBrocadeVcsDevice() {
+        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
+        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        return brocadeDevice;
+    }
 }
