--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1623,13 +1623,12 @@
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        CounterTO counterTO = createMockCounterTO(Counter.Source.CPU);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
-        when(counterTO.getSource()).thenReturn(Counter.Source.CPU);
 
         boolean result = autoScaleManagerImplSpy.isNative(groupTO);
         Assert.assertTrue(result);
@@ -1640,13 +1639,12 @@
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        CounterTO counterTO = createMockCounterTO(Counter.Source.VIRTUALROUTER);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
-        when(counterTO.getSource()).thenReturn(Counter.Source.VIRTUALROUTER);
 
         boolean result = autoScaleManagerImplSpy.isNative(groupTO);
         Assert.assertFalse(result);
@@ -1657,13 +1655,12 @@
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        CounterTO counterTO = createMockCounterTO(Counter.Source.VIRTUALROUTER);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
-        when(counterTO.getSource()).thenReturn(Counter.Source.VIRTUALROUTER);
 
         boolean result = autoScaleManagerImplSpy.hasSourceVirtualRouter(groupTO);
         Assert.assertTrue(result);
@@ -1674,13 +1671,12 @@
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        CounterTO counterTO = createMockCounterTO(Counter.Source.CPU);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
         when(conditionTO.getCounter()).thenReturn(counterTO);
-        when(counterTO.getSource()).thenReturn(Counter.Source.CPU);
 
         boolean result = autoScaleManagerImplSpy.hasSourceVirtualRouter(groupTO);
         Assert.assertFalse(result);
@@ -2209,17 +2205,15 @@
 
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO scaleUpConditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO scaleUpCounterTO = Mockito.mock(CounterTO.class);
+        CounterTO scaleUpCounterTO = createMockCounterTO(Counter.Source.CPU);
         when(scaleUpPolicyTO.getConditions()).thenReturn(Arrays.asList(scaleUpConditionTO));
         when(scaleUpConditionTO.getCounter()).thenReturn(scaleUpCounterTO);
-        when(scaleUpCounterTO.getSource()).thenReturn(Counter.Source.CPU);
 
         AutoScalePolicyTO scaleDownPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO scaleDownConditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO scaleDownCounterTO = Mockito.mock(CounterTO.class);
+        CounterTO scaleDownCounterTO = createMockCounterTO(Counter.Source.VIRTUALROUTER);
         when(scaleDownPolicyTO.getConditions()).thenReturn(Arrays.asList(scaleDownConditionTO));
         when(scaleDownConditionTO.getCounter()).thenReturn(scaleDownCounterTO);
-        when(scaleDownCounterTO.getSource()).thenReturn(Counter.Source.VIRTUALROUTER);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
 
@@ -2512,6 +2506,12 @@
                 "-" + asVmGroupMock.getNextVmSeq() + "-[a-z]{6}";
         Assert.assertTrue(result.first().matches(vmHostNamePattern));
         Assert.assertEquals(result.first(), result.second());
+    }
+
+    private static CounterTO createMockCounterTO(Counter.Source source) {
+        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        when(counterTO.getSource()).thenReturn(source);
+        return counterTO;
     }
 
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
