--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
@@ -673,7 +673,6 @@
     public void testExecuteWithValidPath() throws Exception {
         // Setup
         ListDataStoreObjectsCommand cmd = new ListDataStoreObjectsCommand(destDataStoreTO, "valid/path", 0, 10);
-        HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
         HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
         FileInfo fileInfo = mock(FileInfo.class);
         List<FileInfo> fileInfoList = new ArrayList<>();
@@ -682,7 +681,7 @@
 
         doReturn(context).when(vmwareResource).getServiceContext(any());
         doReturn(hyperHost).when(vmwareResource).getHyperHost(context);
-        when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
+        HostDatastoreBrowserMO browserMo = createMockHostDatastoreBrowserMO(results);
         when(results.getFile()).thenReturn(fileInfoList);
         when(fileInfo.getPath()).thenReturn("file.txt");
         when(fileInfo.getFileSize()).thenReturn(1L);
@@ -752,7 +751,6 @@
     public void testExecuteWithRootPath() throws Exception {
         // Setup
         ListDataStoreObjectsCommand cmd = new ListDataStoreObjectsCommand(destDataStoreTO, "/", 0, 10);
-        HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
         HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
         FileInfo fileInfo = mock(FileInfo.class);
         List<FileInfo> fileInfoList = new ArrayList<>();
@@ -761,7 +759,7 @@
 
         doReturn(context).when(vmwareResource).getServiceContext();
         doReturn(hyperHost).when(vmwareResource).getHyperHost(context);
-        when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
+        HostDatastoreBrowserMO browserMo = createMockHostDatastoreBrowserMO(results);
         when(results.getFile()).thenReturn(fileInfoList);
         when(fileInfo.getPath()).thenReturn("file.txt");
         when(fileInfo.getFileSize()).thenReturn(1L);
@@ -797,7 +795,6 @@
     public void testExecuteWithEmptyPath() throws Exception {
         // Setup
         ListDataStoreObjectsCommand cmd = new ListDataStoreObjectsCommand(destDataStoreTO, "", 0, 10);
-        HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
         HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
         FileInfo fileInfo = mock(FileInfo.class);
         List<FileInfo> fileInfoList = new ArrayList<>();
@@ -806,7 +803,7 @@
 
         doReturn(context).when(vmwareResource).getServiceContext();
         doReturn(hyperHost).when(vmwareResource).getHyperHost(context);
-        when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
+        HostDatastoreBrowserMO browserMo = createMockHostDatastoreBrowserMO(results);
         when(results.getFile()).thenReturn(fileInfoList);
         when(fileInfo.getPath()).thenReturn("file.txt");
         when(fileInfo.getFileSize()).thenReturn(1L);
@@ -874,4 +871,10 @@
         Mockito.verify(vmwareResource, Mockito.times(1)).removeVirtualTPMDevice(vmConfigSpec, tpm);
         Mockito.verify(deviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
     }
+
+    private static HostDatastoreBrowserMO createMockHostDatastoreBrowserMO(HostDatastoreBrowserSearchResults results) throws Exception {
+        HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
+        when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
+        return browserMo;
+    }
 }
