--- a/engine/orchestration/src/test/java/com/cloud/agent/manager/AgentManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/agent/manager/AgentManagerImplTest.java
@@ -49,6 +49,7 @@
     private AgentManagerImpl mgr = Mockito.spy(new AgentManagerImpl());
     private HostVO host;
     private StartupCommand[] cmds;
+    private Commands mockedCommands;
 
     @Before
     public void setUp() throws Exception {
@@ -61,6 +62,7 @@
         hostDao = Mockito.mock(HostDao.class);
         hostDetailsDao = Mockito.mock(HostDetailsDao.class);
         storagePoolMonitor = Mockito.mock(Listener.class);
+        mockedCommands = Mockito.mock(Commands.class);
 
         mgr._hostDao = hostDao;
         mgr._hostDetailsDao = hostDetailsDao;
@@ -123,20 +125,18 @@
 
     @Test
     public void testGetTimeoutWithPositiveTimeout() {
-        Commands commands = Mockito.mock(Commands.class);
         int timeout = 30;
-        int result = mgr.getTimeout(commands, timeout);
+        int result = mgr.getTimeout(mockedCommands, timeout);
 
         Assert.assertEquals(30, result);
     }
 
     @Test
     public void testGetTimeoutWithGranularTimeout() {
-        Commands commands = Mockito.mock(Commands.class);
-        Mockito.doReturn(50).when(mgr).getTimeoutFromGranularWaitTime(commands);
+        Mockito.doReturn(50).when(mgr).getTimeoutFromGranularWaitTime(mockedCommands);
 
         int timeout = 0;
-        int result = mgr.getTimeout(commands, timeout);
+        int result = mgr.getTimeout(mockedCommands, timeout);
 
         Assert.assertEquals(50, result);
     }
