--- a/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
+++ b/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
@@ -22,6 +22,7 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.InjectMocks;
@@ -40,16 +41,22 @@
     @InjectMocks
     private ElasticLoadBalancerManagerImpl elasticLoadBalancerManagerImpl;
 
+    private VirtualMachineProfile virtualMachineProfileMock;
+
+    @Before
+    public void setUp() {
+        virtualMachineProfileMock = mock(VirtualMachineProfile.class);
+    }
+
     @Test
     public void testFinalizeStartWhenCmdsAnswerIsNull() throws Exception {
-        VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
         when(cmds.getAnswer("checkSsh")).thenReturn(null);
         ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = false;
-        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, context);
+        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(virtualMachineProfileMock, hostId, cmds, context);
 
         assertEquals(expected, actual);
     }
@@ -58,14 +65,13 @@
     public void testFinalizeStartWhenCmdsAnswerIsNotNullButAnswerResultIsFalse() throws Exception {
         CheckSshAnswer answerMock = mock(CheckSshAnswer.class);
         when(answerMock.getResult()).thenReturn(false);
-        VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
         when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
         ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = false;
-        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, context);
+        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(virtualMachineProfileMock, hostId, cmds, context);
 
         assertEquals(expected, actual);
     }
@@ -74,14 +80,13 @@
     public void testFinalizeStartWhenCmdsAnswerIsNotNullAndAnswerResultIsTrue() throws Exception {
         CheckSshAnswer answerMock = mock(CheckSshAnswer.class);
         when(answerMock.getResult()).thenReturn(true);
-        VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
         when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
         ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = true;
-        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(profileMock, hostId, cmds, context);
+        boolean actual = elasticLoadBalancerManagerImpl.finalizeStart(virtualMachineProfileMock, hostId, cmds, context);
 
         assertEquals(expected, actual);
     }
