--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionManagerImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionManagerImplTest.java
@@ -114,9 +114,8 @@
 
     @Test
     public void testValidateImageStoreForAllZonesWithAllValid() {
-        DataCenterVO zone1 = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone1.getId()).thenReturn(1L);
-        DataCenterVO zone2 = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zone1 = createMockZone();
+        DataCenterVO zone2 = createMockZone();
         Mockito.when(zone2.getId()).thenReturn(2L);
         List<DataCenterVO> zones = Arrays.asList(zone1, zone2);
         Mockito.when(dataCenterDao.listAllZones()).thenReturn(zones);
@@ -130,9 +129,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateImageStoreForAllZonesWithSomeMissingStorage() {
-        DataCenterVO zone1 = Mockito.mock(DataCenterVO.class);
-        Mockito.when(zone1.getId()).thenReturn(1L);
-        DataCenterVO zone2 = Mockito.mock(DataCenterVO.class);
+        DataCenterVO zone1 = createMockZone();
+        DataCenterVO zone2 = createMockZone();
         Mockito.when(zone2.getId()).thenReturn(2L);
         Mockito.when(zone2.getName()).thenReturn("zone-without-storage");
         List<DataCenterVO> zones = Arrays.asList(zone1, zone2);
@@ -144,4 +142,10 @@
 
         ReflectionTestUtils.invokeMethod(kubernetesVersionManager, "validateImageStoreForZone", (Long) null, false);
     }
+
+    private static DataCenterVO createMockZone() {
+        DataCenterVO zone = Mockito.mock(DataCenterVO.class);
+        Mockito.when(zone.getId()).thenReturn(1L);
+        return zone;
+    }
 }
