--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -120,6 +120,12 @@
     }
 
 
+    private static Account createMockAccountWithDomainId() {
+        Account caller = Mockito.mock(Account.class);
+        Mockito.when(caller.getDomainId()).thenReturn(1L);
+        return caller;
+    }
+
     private WebhookJoinVO prepareTestWebhookJoinVO() {
         String name = "webhook";
         String description = "webhook-description";
@@ -860,7 +866,7 @@
         Mockito.when(cmd.isSslVerification()).thenReturn(true);
         Mockito.when(cmd.getScope()).thenReturn("Local");
         Mockito.when(cmd.getState()).thenReturn("Enabled");
-        Mockito.when(caller.getDomainId()).thenReturn(1L);
+        caller = createMockAccountWithDomainId();
         Mockito.when(webhookDao.persist(Mockito.any(WebhookVO.class))).thenReturn(webhook);
         Mockito.when(webhook.getId()).thenReturn(1L);
         Mockito.when(webhookJoinDao.findById(1L)).thenReturn(webhookJoinVO);
