--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -289,9 +289,8 @@
         Mockito.doReturn(storagePoolMgr).when(libvirtComputingResource).getStoragePoolMgr();
         PrimaryDataStoreTO spool = Mockito.mock(PrimaryDataStoreTO.class);
         Mockito.doReturn(spool).when(destVolumeObjectTO).getDataStore();
-        KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
+        KVMStoragePool pool = createMockKVMStoragePool();
         Mockito.doReturn(pool).when(storagePoolMgr).getStoragePool(Mockito.any(), Mockito.any());
-        Mockito.doReturn(true).when(pool).connectPhysicalDisk(Mockito.any(), Mockito.any());
 
         Mockito.doReturn(null).when(destVolumeObjectTO).getPassphrase();
 
@@ -336,9 +335,8 @@
         Mockito.doReturn(storagePoolMgr).when(libvirtComputingResource).getStoragePoolMgr();
         PrimaryDataStoreTO spool = Mockito.mock(PrimaryDataStoreTO.class);
         Mockito.doReturn(spool).when(destVolumeObjectTO).getDataStore();
-        KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
+        KVMStoragePool pool = createMockKVMStoragePool();
         Mockito.doReturn(pool).when(storagePoolMgr).getStoragePool(Mockito.any(), Mockito.any());
-        Mockito.doReturn(true).when(pool).connectPhysicalDisk(Mockito.any(), Mockito.any());
 
         Mockito.doReturn(null).when(destVolumeObjectTO).getPassphrase();
         Mockito.doReturn(domxml).when(dm).getXMLDesc(0);
@@ -385,4 +383,10 @@
         Assert.assertFalse(answer.getResult());
     }
 
+    private static KVMStoragePool createMockKVMStoragePool() {
+        KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
+        Mockito.doReturn(true).when(pool).connectPhysicalDisk(Mockito.any(), Mockito.any());
+        return pool;
+    }
+
 }
