--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLManagerTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLManagerTest.java
@@ -103,6 +103,7 @@
 
     private NetworkACLVO acl;
     private NetworkACLItemVO aclItem;
+    private final NetworkVO networkMock = Mockito.mock(NetworkVO.class);
 
     @Override
     @Before
@@ -131,8 +132,7 @@
     @Test
     @SuppressWarnings("unchecked")
     public void testApplyACL() throws Exception {
-        final NetworkVO network = Mockito.mock(NetworkVO.class);
-        Mockito.when(_networkDao.findById(anyLong())).thenReturn(network);
+        Mockito.when(_networkDao.findById(anyLong())).thenReturn(networkMock);
         Mockito.when(networkOfferingDao.isIpv6Supported(anyLong())).thenReturn(false);
         Mockito.when(_networkModel.isProviderSupportServiceInNetwork(anyLong(), ArgumentMatchers.any(Network.Service.class), ArgumentMatchers.any(Network.Provider.class))).thenReturn(true);
         Mockito.when(_networkAclElements.get(0).applyNetworkACLs(ArgumentMatchers.any(Network.class), ArgumentMatchers.anyList())).thenReturn(true);
@@ -166,13 +166,12 @@
 
         // Make sure it is handled
         final long aclId = 1L;
-        final NetworkVO network = Mockito.mock(NetworkVO.class);
         final List<NetworkVO> networks = new ArrayList<>();
-        networks.add(network);
+        networks.add(networkMock);
 
         when(networkServiceMapDao.canProviderSupportServiceInNetwork(anyLong(), eq(Network.Service.NetworkACL), nullable(Network.Provider.class))).thenReturn(true);
         Mockito.when(_networkDao.listByAclId(anyLong())).thenReturn(networks);
-        Mockito.when(_networkDao.findById(anyLong())).thenReturn(network);
+        Mockito.when(_networkDao.findById(anyLong())).thenReturn(networkMock);
         Mockito.when(networkOfferingDao.isIpv6Supported(anyLong())).thenReturn(false);
         Mockito.when(_networkModel.isProviderSupportServiceInNetwork(anyLong(), any(Network.Service.class), any(Network.Provider.class))).thenReturn(true);
         Mockito.when(_networkAclElements.get(0).getProvider()).thenReturn(Mockito.mock(Network.Provider.class));
