--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -147,10 +147,13 @@
     @Inject
     VMSnapshotDetailsDao vmSnapshotDetailsDao;
 
+    private GuestOSVO guestOSVOMock;
+
     @Override
     @Before
     public void setUp() throws Exception {
         ComponentContext.initComponentsLifeCycle();
+        guestOSVOMock = Mockito.mock(GuestOSVO.class);
     }
 
     @Test
@@ -200,8 +203,7 @@
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
         Mockito.when(userVmDao.findById(ArgumentMatchers.anyLong())).thenReturn(userVmVO);
-        GuestOSVO guestOSVO = Mockito.mock(GuestOSVO.class);
-        Mockito.when(guestOSDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSVO);
+        Mockito.when(guestOSDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSVOMock);
         GuestOSHypervisorVO guestOSHypervisorVO = Mockito.mock(GuestOSHypervisorVO.class);
         Mockito.when(guestOSHypervisorVO.getGuestOsName()).thenReturn(guestOsName);
         Mockito.when(guestOsHypervisorDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSHypervisorVO);
@@ -277,8 +279,7 @@
         Mockito.when(vmSnapshotHelper.pickRunningHost(ArgumentMatchers.anyLong())).thenReturn(hostId);
         Mockito.when(vmSnapshotHelper.getVolumeTOList(ArgumentMatchers.anyLong())).thenReturn(volumeObjectTOs);
         Mockito.when(userVmDao.findById(ArgumentMatchers.anyLong())).thenReturn(userVmVO);
-        GuestOSVO guestOSVO = Mockito.mock(GuestOSVO.class);
-        Mockito.when(guestOSDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSVO);
+        Mockito.when(guestOSDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSVOMock);
         GuestOSHypervisorVO guestOSHypervisorVO = Mockito.mock(GuestOSHypervisorVO.class);
         Mockito.when(guestOSHypervisorVO.getGuestOsName()).thenReturn(guestOsName);
         Mockito.when(guestOsHypervisorDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSHypervisorVO);
