--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -457,11 +457,8 @@
     @Test
     @DB
     public void testNewAssignPublicIpv6ToNetwork() {
-        NicVO nic = Mockito.mock(NicVO.class);
-        Mockito.when(nic.getIPv6Address()).thenReturn(null);
-        Mockito.when(nic.getBroadcastUri()).thenReturn(URI.create(vlan));
+        NicVO nic = createMockNic(URI.create(vlan), nicId);
         Mockito.when(nic.getMacAddress()).thenReturn(macAddress);
-        Mockito.when(nic.getId()).thenReturn(nicId);
         Network network = Mockito.mock(Network.class);
         Mockito.when(network.getId()).thenReturn(networkId);
         prepareMocksForPublicIpv6(false);
@@ -476,10 +473,7 @@
 
     @Test
     public void testFromPlaceholderAssignPublicIpv6ToNetwork() {
-        NicVO nic = Mockito.mock(NicVO.class);
-        Mockito.when(nic.getIPv6Address()).thenReturn(null);
-        Mockito.when(nic.getBroadcastUri()).thenReturn(URI.create(vlan));
-        Mockito.when(nic.getId()).thenReturn(nicId);
+        NicVO nic = createMockNic(URI.create(vlan), nicId);
         Network network = Mockito.mock(Network.class);
         Mockito.when(network.getId()).thenReturn(networkId);
         prepareMocksForPublicIpv6(true);
@@ -801,4 +795,12 @@
         ipv6Service.removePublicIpv6PlaceholderNics(network);
         Assert.assertEquals(0, removedNics.size());
     }
+
+    private static NicVO createMockNic(URI broadcastUri, Long nicId) {
+        NicVO nic = Mockito.mock(NicVO.class);
+        Mockito.when(nic.getIPv6Address()).thenReturn(null);
+        Mockito.when(nic.getBroadcastUri()).thenReturn(broadcastUri);
+        Mockito.when(nic.getId()).thenReturn(nicId);
+        return nic;
+    }
 }
