--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachine vm) {
+        VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
+        when(profile.getVirtualMachine()).thenReturn(vm);
+        return profile;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -404,9 +410,8 @@
     public void testSendStopWithOkAnswer() throws Exception {
         VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
         VirtualMachine vm = mock(VirtualMachine.class);
-        VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
+        VirtualMachineProfile profile = createMockVirtualMachineProfile(vm);
         StopAnswer answer = new StopAnswer(new StopCommand(vm, false, false), "ok", true);
-        when(profile.getVirtualMachine()).thenReturn(vm);
         when(vm.getHostId()).thenReturn(1L);
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(answer);
 
@@ -419,9 +424,8 @@
     public void testSendStopWithFailAnswer() throws Exception {
         VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
         VirtualMachine vm = mock(VirtualMachine.class);
-        VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         StopAnswer answer = new StopAnswer(new StopCommand(vm, false, false), "fail", false);
-        when(profile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile profile = createMockVirtualMachineProfile(vm);
         when(vm.getHostId()).thenReturn(1L);
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(answer);
 
@@ -435,8 +439,7 @@
     public void testSendStopWithNullAnswer() throws Exception {
         VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
         VirtualMachine vm = mock(VirtualMachine.class);
-        VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
-        when(profile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile profile = createMockVirtualMachineProfile(vm);
         when(vm.getHostId()).thenReturn(1L);
         when(vm.getInstanceName()).thenReturn("test-vm");
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(null);
