--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -295,6 +295,8 @@
     private HighAvailabilityManager _haMgr;
     @Mock
     VirtualMachineGuru guru;
+    @Mock
+    private VirtualMachineGuru virtualMachineGuruMock;
 
     private ConfigDepotImpl configDepotImpl;
     private boolean updatedConfigKeyDepot = false;
@@ -402,7 +404,6 @@
 
     @Test
     public void testSendStopWithOkAnswer() throws Exception {
-        VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
         VirtualMachine vm = mock(VirtualMachine.class);
         VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         StopAnswer answer = new StopAnswer(new StopCommand(vm, false, false), "ok", true);
@@ -410,14 +411,13 @@
         when(vm.getHostId()).thenReturn(1L);
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(answer);
 
-        Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(guru, profile, false, false);
+        Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(virtualMachineGuruMock, profile, false, false);
 
         Assert.assertTrue(actual.first());
     }
 
     @Test
     public void testSendStopWithFailAnswer() throws Exception {
-        VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
         VirtualMachine vm = mock(VirtualMachine.class);
         VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         StopAnswer answer = new StopAnswer(new StopCommand(vm, false, false), "fail", false);
@@ -425,7 +425,7 @@
         when(vm.getHostId()).thenReturn(1L);
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(answer);
 
-        Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(guru, profile, false, false);
+        Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(virtualMachineGuruMock, profile, false, false);
 
         assertFalse(actual.first());
         Assert.assertEquals("fail", actual.second());
@@ -433,7 +433,6 @@
 
     @Test
     public void testSendStopWithNullAnswer() throws Exception {
-        VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
         VirtualMachine vm = mock(VirtualMachine.class);
         VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         when(profile.getVirtualMachine()).thenReturn(vm);
@@ -441,7 +440,7 @@
         when(vm.getInstanceName()).thenReturn("test-vm");
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(null);
 
-        Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(guru, profile, false, false);
+        Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(virtualMachineGuruMock, profile, false, false);
 
         assertFalse(actual.first());
         Assert.assertNotNull(actual.second());
@@ -1119,7 +1118,6 @@
         ReflectionTestUtils.setField(vmInstance, "dataCenterId", 1L);
         ReflectionTestUtils.setField(vmInstance, "hypervisorType", HypervisorType.KVM);
 
-        VirtualMachineGuru vmGuru = mock(VirtualMachineGuru.class);
 
         User user = mock(User.class);
 
@@ -1145,10 +1143,10 @@
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
-        doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
+        doReturn(virtualMachineGuruMock).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
-        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
+        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(virtualMachineGuruMock, vmInstance, user, account, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
@@ -1215,7 +1213,6 @@
         ReflectionTestUtils.setField(vmInstance, "dataCenterId", 1L);
         ReflectionTestUtils.setField(vmInstance, "hypervisorType", HypervisorType.KVM);
 
-        VirtualMachineGuru vmGuru = mock(VirtualMachineGuru.class);
 
         User user = mock(User.class);
 
@@ -1241,10 +1238,10 @@
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
-        doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
+        doReturn(virtualMachineGuruMock).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
-        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
+        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(virtualMachineGuruMock, vmInstance, user, account, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
