--- a/server/src/test/java/com/cloud/vm/FirstFitPlannerTest.java
+++ b/server/src/test/java/com/cloud/vm/FirstFitPlannerTest.java
@@ -152,9 +152,11 @@
     int cpuSpeedInOffering = 500;
     int ramInOffering = 512;
     AccountVO acct = new AccountVO(accountId);
+    private ExcludeList excludeList;
 
     @Before
     public void setUp() {
+        excludeList = mock(ExcludeList.class);
         ConfigKey.init(configDepot);
 
         when(configDepot.global()).thenReturn(configDao);
@@ -178,10 +180,9 @@
     public void checkClusterReorderingBasedOnImplicitHostTags() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
-        initializeForTest(vmProfile, plan, avoids);
-
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        initializeForTest(vmProfile, plan, excludeList);
+
+        List<Long> clusterList = planner.orderClusters(vmProfile, plan, excludeList);
         List<Long> reorderedClusterList = new ArrayList<Long>();
         reorderedClusterList.add(4L);
         reorderedClusterList.add(3L);
@@ -197,15 +198,14 @@
     public void checkClusterReorderingForDeployVMWithThresholdCheckDisabled() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
-        initializeForTest(vmProfile, plan, avoids);
+        initializeForTest(vmProfile, plan, excludeList);
         List<Long> clustersCrossingThreshold = initializeForClusterThresholdDisabled();
 
         Map<String, String> details = new HashMap<String, String>();
         details.put("deployvm", "true");
         when(vmDetailsDao.listDetailsKeyPairs(vmProfile.getVirtualMachine().getId())).thenReturn(details);
 
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(vmProfile, plan, excludeList);
         assertTrue("Reordered cluster list have clusters exceeding threshold", (!clusterList.containsAll(clustersCrossingThreshold)));
     }
 
@@ -213,11 +213,10 @@
     public void checkClusterListBasedOnHostTag() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
-        initializeForTest(vmProfile, plan, avoids);
+        initializeForTest(vmProfile, plan, excludeList);
         List<Long> matchingClusters = initializeForClusterListBasedOnHostTag(vmProfile.getServiceOffering());
 
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(vmProfile, plan, excludeList);
 
         assertTrue("Reordered cluster list have clusters which has hosts with specified host tag on offering", (clusterList.containsAll(matchingClusters)));
         assertTrue("Reordered cluster list does not have clusters which don't have hosts with matching host tag on offering", (!clusterList.contains(2L)));
@@ -238,11 +237,10 @@
     public void checkClusterReorderingForStartVMWithThresholdCheckDisabled() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
-        initializeForTest(vmProfile, plan, avoids);
+        initializeForTest(vmProfile, plan, excludeList);
         List<Long> clustersCrossingThreshold = initializeForClusterThresholdDisabled();
 
-        List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
+        List<Long> clusterList = planner.orderClusters(vmProfile, plan, excludeList);
         assertTrue("Reordered cluster list does not have clusters exceeding threshold", (clusterList.containsAll(clustersCrossingThreshold)));
     }
 
