--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -1515,12 +1515,10 @@
                         networkUuid1, ip1, ipv61, mac1, networkUuid2, ip2, ipv62, mac2);
         when(backup2.getDetail(ApiConstants.NICS)).thenReturn(nicsJson);
 
-        NetworkVO network1 = mock(NetworkVO.class);
-        NetworkVO network2 = mock(NetworkVO.class);
+        NetworkVO network1 = createMockNetworkVO(networkId1);
+        NetworkVO network2 = createMockNetworkVO(networkId2);
         when(networkDao.findByUuid(networkUuid1)).thenReturn(network1);
         when(networkDao.findByUuid(networkUuid2)).thenReturn(network2);
-        when(network1.getId()).thenReturn(networkId1);
-        when(network2.getId()).thenReturn(networkId2);
 
         Network.IpAddresses ipAddresses1 = mock(Network.IpAddresses.class);
         Network.IpAddresses ipAddresses2 = mock(Network.IpAddresses.class);
@@ -2623,6 +2621,12 @@
         when(offering.getId()).thenReturn(id);
         when(offering.getName()).thenReturn(name);
         return offering;
+    }
+
+    private static NetworkVO createMockNetworkVO(Long id) {
+        NetworkVO network = mock(NetworkVO.class);
+        when(network.getId()).thenReturn(id);
+        return network;
     }
 
     @Test
