--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -2090,8 +2090,7 @@
 
     @Test
     public void validateOldAndNewAccountsTestBothAreValidDoNothing() {
-        Account newAccount = Mockito.mock(Account.class);
-        Mockito.doReturn(1L).when(newAccount).getAccountId();
+        Account newAccount = createMockAccount();
 
         userVmManagerImpl.validateOldAndNewAccounts(accountMock, newAccount, 1L, "", 1L);
     }
@@ -2147,7 +2146,7 @@
     @Test
     public void validateOldAndNewAccountsTestOldAccountIsNotTheSameAsNewAccountDoesNotThrowInvalidParameterValueException() {
         AccountVO oldAccount = new AccountVO();
-        Mockito.doReturn(1L).when(accountMock).getAccountId();
+        accountMock = createMockAccount();
 
         userVmManagerImpl.validateOldAndNewAccounts(oldAccount, accountMock, 1L, "", 1L);
     }
@@ -4047,6 +4046,12 @@
         leaseDetails.put(VmDetailConstants.INSTANCE_LEASE_EXPIRY_DATE, getLeaseExpiryDate(leaseDuration));
         leaseDetails.put(VmDetailConstants.INSTANCE_LEASE_EXECUTION, leaseExecution);
         return leaseDetails;
+    }
+
+    private static Account createMockAccount() {
+        Account accountMock = Mockito.mock(Account.class);
+        Mockito.doReturn(1L).when(accountMock).getAccountId();
+        return accountMock;
     }
 
     private static DeployVMCmd buildDeployVMCmd(Long volumeId, Long snapshotId) {
