--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -273,6 +273,8 @@
     DataCenterDao dataCenterDao;
     @Mock
     IpAddressManager ipAddressManager;
+    @Mock
+    private VMInstanceVO securityGroupVmInstanceVO;
 
     TungstenServiceImpl tungstenService;
 
@@ -1535,7 +1537,6 @@
 
     @Test
     public void addTungstenVmSecurityGroupTest() {
-        VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
         SecurityGroupVO securityGroupVO = mock(SecurityGroupVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
@@ -1565,12 +1566,11 @@
         when(securityGroupRuleVO.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
         when(tungstenSecurityGroupRuleDao.persist(any(TungstenSecurityGroupRuleVO.class))).thenReturn(tungstenSecurityGroupRuleVO);
 
-        assertTrue(tungstenService.addTungstenVmSecurityGroup(vmInstanceVO));
+        assertTrue(tungstenService.addTungstenVmSecurityGroup(securityGroupVmInstanceVO));
     }
 
     @Test
     public void removeTungstenVmSecurityGroupTest() {
-        VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
         SecurityGroupVO securityGroupVO = mock(SecurityGroupVO.class);
         TungstenAnswer removeTungstenVmFromSecurityGroupAnswer = MockTungstenAnswerFactory.get(true);
@@ -1582,7 +1582,7 @@
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(dataCenterVO.isSecurityGroupEnabled()).thenReturn(true);
 
-        assertTrue(tungstenService.removeTungstenVmSecurityGroup(vmInstanceVO));
+        assertTrue(tungstenService.removeTungstenVmSecurityGroup(securityGroupVmInstanceVO));
     }
 
     @Test
