--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -70,6 +70,8 @@
     @Mock
     protected XenServer610Resource xenServer610Resource;
 
+    private final NicTO nicTO2Mock = Mockito.mock(NicTO.class);
+
     @Test
     public void testCheckNetworkCommandFailure() {
         final XenServer610Resource xenServer610Resource = new XenServer610Resource();
@@ -117,9 +119,8 @@
         volumeToFiler.put(vol2, storage2);
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
-        final NicTO nicTO2 = Mockito.mock(NicTO.class);
         final NicTO nicTO3 = Mockito.mock(NicTO.class);
-        final NicTO [] nicTOs = {nicTO1, nicTO2, nicTO3};
+        final NicTO [] nicTOs = {nicTO1, nicTO2Mock, nicTO3};
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network networkForSm = Mockito.mock(Network.class);
@@ -155,7 +156,7 @@
         try {
             verify(xenServer610Resource, times(1)).prepareISO(conn, vmName, null, null);
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO1);
-            verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO2);
+            verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO2Mock);
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO3);
 
             verify(xenServer610Resource, times(1)).getNativeNetworkForTraffic(conn, TrafficType.Storage, null);
@@ -190,9 +191,8 @@
         volumeToFiler.add(new Pair<>(vol2, storage2.getPath()));
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
-        final NicTO nicTO2 = Mockito.mock(NicTO.class);
         final NicTO nicTO3 = Mockito.mock(NicTO.class);
-        final NicTO [] nicTOs = {nicTO1, nicTO2, nicTO3};
+        final NicTO [] nicTOs = {nicTO1, nicTO2Mock, nicTO3};
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network network = Mockito.mock(Network.class);
@@ -213,7 +213,7 @@
 
         try {
             when(xenServer610Resource.getNetwork(conn, nicTO1)).thenReturn(nw1);
-            when(xenServer610Resource.getNetwork(conn, nicTO2)).thenReturn(nw2);
+            when(xenServer610Resource.getNetwork(conn, nicTO2Mock)).thenReturn(nw2);
             when(xenServer610Resource.getNetwork(conn, nicTO3)).thenReturn(nw3);
 
             when(xenServer610Resource.getNativeNetworkForTraffic(conn, TrafficType.Storage, null)).thenReturn(nativeNetworkForTraffic);
@@ -233,7 +233,7 @@
 
         try {
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO1);
-            verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO2);
+            verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO2Mock);
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO3);
 
             verify(xenServer610Resource, times(1)).getNativeNetworkForTraffic(conn, TrafficType.Storage, null);
@@ -376,7 +376,6 @@
         final VDI vdi2 = Mockito.mock(VDI.class);
 
         final NicTO nic1 = Mockito.mock(NicTO.class);
-        final NicTO nic2 = Mockito.mock(NicTO.class);
 
         final List<Pair<VolumeTO, Object>> volumeToSr = new ArrayList<Pair<VolumeTO, Object>>();
         volumeToSr.add(new Pair<VolumeTO, Object>(volume1, sr1));
@@ -384,7 +383,7 @@
 
         final List<Pair<NicTO, Object>> nicToNetwork = new ArrayList<Pair<NicTO, Object>>();
         nicToNetwork.add(new Pair<NicTO, Object>(nic1, new String("a")));
-        nicToNetwork.add(new Pair<NicTO, Object>(nic2, new String("b")));
+        nicToNetwork.add(new Pair<NicTO, Object>(nicTO2Mock, new String("b")));
 
         final Map<String, String> token = new HashMap<String, String>();
 
