--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,8 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    @Mock
+    private NicTO nicToMock;
 
     @Test
     public void testWrapperInstance() {
@@ -1471,10 +1473,9 @@
 
     @Test
     public void testPlugNicCommand() {
-        final NicTO nicTO = Mockito.mock(NicTO.class);
-        final Connection conn = Mockito.mock(Connection.class);
-
-        final PlugNicCommand plugNic = new PlugNicCommand(nicTO, "Test", VirtualMachine.Type.User);
+        final Connection conn = Mockito.mock(Connection.class);
+
+        final PlugNicCommand plugNic = new PlugNicCommand(nicToMock, "Test", VirtualMachine.Type.User);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
@@ -1490,10 +1491,9 @@
 
     @Test
     public void testUnPlugNicCommand() {
-        final NicTO nicTO = Mockito.mock(NicTO.class);
-        final Connection conn = Mockito.mock(Connection.class);
-
-        final UnPlugNicCommand unplugNic = new UnPlugNicCommand(nicTO, "Test");
+        final Connection conn = Mockito.mock(Connection.class);
+
+        final UnPlugNicCommand unplugNic = new UnPlugNicCommand(nicToMock, "Test");
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
