--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -513,9 +513,7 @@
     @Test
     public void testTrash() {
 
-        final NetworkVO network = mock(NetworkVO.class);
-        when(network.getName()).thenReturn("testnetwork");
-        when(network.getState()).thenReturn(State.Implementing);
+        final NetworkVO network = createMockNetwork();
         when(network.getId()).thenReturn(NETWORK_ID);
         when(network.getDataCenterId()).thenReturn(NETWORK_ID);
 
@@ -588,4 +586,11 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    private static NetworkVO createMockNetwork() {
+        final NetworkVO network = mock(NetworkVO.class);
+        when(network.getName()).thenReturn("testnetwork");
+        when(network.getState()).thenReturn(State.Implementing);
+        return network;
+    }
 }
