--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static ReservationContext createMockReservationContext() {
+        ReservationContext ctx = Mockito.mock(ReservationContext.class);
+        when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
+        return ctx;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1126,7 +1132,7 @@
         Account account = mock(Account.class);
         Account owner = mock(Account.class);
 
-        ReservationContext ctx = mock(ReservationContext.class);
+        ReservationContext ctx = createMockReservationContext();
 
         ItWorkVO work = mock(ItWorkVO.class);
 
@@ -1149,8 +1155,6 @@
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
-
-        when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
         when(serviceOfferingDaoMock.findById(vmInstance.getId(), vmInstance.getServiceOfferingId())).thenReturn(serviceOffering);
 
@@ -1222,7 +1226,7 @@
         Account account = mock(Account.class);
         Account owner = mock(Account.class);
 
-        ReservationContext ctx = mock(ReservationContext.class);
+        ReservationContext ctx = createMockReservationContext();
 
         ItWorkVO work = mock(ItWorkVO.class);
 
@@ -1245,8 +1249,6 @@
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
-
-        when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
         when(serviceOfferingDaoMock.findById(vmInstance.getId(), vmInstance.getServiceOfferingId())).thenReturn(serviceOffering);
 
