--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer620SP1WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer620SP1WrapperTest.java
@@ -47,20 +47,21 @@
     @Mock
     private XenServer620SP1Resource xenServer620SP1Resource;
 
+    private final Connection connection = Mockito.mock(Connection.class);
+
     @Test
     public void testGetGPUStatsCommand() {
         final String guuid = "246a5b75-05ed-4bbc-a171-2d1fe94a1b0e";
 
-        final Connection conn = Mockito.mock(Connection.class);
 
         final GetGPUStatsCommand gpuStats = new GetGPUStatsCommand(guuid, "xen");
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(xenServer620SP1Resource.getConnection()).thenReturn(conn);
+        when(xenServer620SP1Resource.getConnection()).thenReturn(connection);
         try {
-            when(xenServer620SP1Resource.getGPUGroupDetails(conn)).thenReturn(new HashMap<String, HashMap<String, VgpuTypesInfo>>());
+            when(xenServer620SP1Resource.getGPUGroupDetails(connection)).thenReturn(new HashMap<String, HashMap<String, VgpuTypesInfo>>());
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
@@ -70,7 +71,7 @@
         final Answer answer = wrapper.execute(gpuStats, xenServer620SP1Resource);
         verify(xenServer620SP1Resource, times(1)).getConnection();
         try {
-            verify(xenServer620SP1Resource, times(1)).getGPUGroupDetails(conn);
+            verify(xenServer620SP1Resource, times(1)).getGPUGroupDetails(connection);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
@@ -84,16 +85,15 @@
     public void testGetGPUStatsCommandFailure() {
         final String guuid = "246a5b75-05ed-4bbc-a171-2d1fe94a1b0e";
 
-        final Connection conn = Mockito.mock(Connection.class);
 
         final GetGPUStatsCommand gpuStats = new GetGPUStatsCommand(guuid, "xen");
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(xenServer620SP1Resource.getConnection()).thenReturn(conn);
+        when(xenServer620SP1Resource.getConnection()).thenReturn(connection);
         try {
-            when(xenServer620SP1Resource.getGPUGroupDetails(conn)).thenThrow(new CloudRuntimeException("Failed!"));
+            when(xenServer620SP1Resource.getGPUGroupDetails(connection)).thenThrow(new CloudRuntimeException("Failed!"));
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
@@ -103,7 +103,7 @@
         final Answer answer = wrapper.execute(gpuStats, xenServer620SP1Resource);
         verify(xenServer620SP1Resource, times(1)).getConnection();
         try {
-            verify(xenServer620SP1Resource, times(1)).getGPUGroupDetails(conn);
+            verify(xenServer620SP1Resource, times(1)).getGPUGroupDetails(connection);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
