--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -67,6 +67,9 @@
 
     @Mock
     LibvirtUtilitiesHelper libvirtUtilitiesHelper;
+
+    @Mock
+    private Connect connMock;
 
     private String domxml = "<domain type='kvm' id='1'>\n" +
             "  <name>i-2-27-VM</name>\n" +
@@ -276,10 +279,9 @@
         destDetails.put(ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID, "610204d03e3ad60f");
 
         Mockito.doReturn(libvirtUtilitiesHelper).when(libvirtComputingResource).getLibvirtUtilitiesHelper();
-        Connect conn = Mockito.mock(Connect.class);
         Domain dm = Mockito.mock(Domain.class);
-        Mockito.doReturn(conn).when(libvirtUtilitiesHelper).getConnection();
-        Mockito.doReturn(dm).when(libvirtComputingResource).getDomain(conn, vmName);
+        Mockito.doReturn(connMock).when(libvirtUtilitiesHelper).getConnection();
+        Mockito.doReturn(dm).when(libvirtComputingResource).getDomain(connMock, vmName);
 
         DomainInfo domainInfo = Mockito.mock(DomainInfo.class);
         domainInfo.state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
@@ -323,10 +325,9 @@
         destDetails.put(ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID, "610204d03e3ad60f");
 
         Mockito.doReturn(libvirtUtilitiesHelper).when(libvirtComputingResource).getLibvirtUtilitiesHelper();
-        Connect conn = Mockito.mock(Connect.class);
         Domain dm = Mockito.mock(Domain.class);
-        Mockito.doReturn(conn).when(libvirtUtilitiesHelper).getConnection();
-        Mockito.doReturn(dm).when(libvirtComputingResource).getDomain(conn, vmName);
+        Mockito.doReturn(connMock).when(libvirtUtilitiesHelper).getConnection();
+        Mockito.doReturn(dm).when(libvirtComputingResource).getDomain(connMock, vmName);
 
         DomainInfo domainInfo = Mockito.mock(DomainInfo.class);
         domainInfo.state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
@@ -351,7 +352,6 @@
 
     @Test
     public void testCheckBlockJobStatus() throws LibvirtException {
-        Connect conn = Mockito.mock(Connect.class);
         Domain dm = Mockito.mock(Domain.class);
         String destDiskLabel = "vda";
         String srcPath = "bec108c400000018:vol-60-7acb-9e22";
@@ -362,14 +362,13 @@
         blockJobInfo.cur = 100;
         blockJobInfo.end = 100;
 
-        MigrateVolumeAnswer answer = libvirtMigrateVolumeCommandWrapper.checkBlockJobStatus(command, dm, destDiskLabel, srcPath, destPath, libvirtComputingResource, conn, null);
+        MigrateVolumeAnswer answer = libvirtMigrateVolumeCommandWrapper.checkBlockJobStatus(command, dm, destDiskLabel, srcPath, destPath, libvirtComputingResource, connMock, null);
 
         Assert.assertTrue(answer.getResult());
     }
 
     @Test
     public void testCheckBlockJobStatusFailure() throws LibvirtException {
-        Connect conn = Mockito.mock(Connect.class);
         Domain dm = Mockito.mock(Domain.class);
         String destDiskLabel = "vda";
         String srcPath = "bec108c400000018:vol-60-7acb-9e22";
@@ -380,7 +379,7 @@
         blockJobInfo.cur = 10;
         blockJobInfo.end = 100;
 
-        MigrateVolumeAnswer answer = libvirtMigrateVolumeCommandWrapper.checkBlockJobStatus(command, dm, destDiskLabel, srcPath, destPath, libvirtComputingResource, conn, null);
+        MigrateVolumeAnswer answer = libvirtMigrateVolumeCommandWrapper.checkBlockJobStatus(command, dm, destDiskLabel, srcPath, destPath, libvirtComputingResource, connMock, null);
 
         Assert.assertFalse(answer.getResult());
     }
