--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -280,6 +280,7 @@
     Domain domainMock;
     @Mock
     DomainInfo domainInfoMock;
+    private final DomainInfo mockDomainInfo = Mockito.mock(DomainInfo.class);
     @Mock
     DomainInterfaceStats domainInterfaceStatsMock;
     @Mock
@@ -1072,9 +1073,8 @@
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Domain vm = Mockito.mock(Domain.class);
-        final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_SHUTDOWN;
-        info.state = state;
+        mockDomainInfo.state = state;
 
         final String vmName = "Test";
         final StopCommand command = new StopCommand(vmName, false, true);
@@ -1084,7 +1084,7 @@
             when(libvirtUtilitiesHelper.getConnectionByVmName(vmName)).thenReturn(conn);
             when(conn.domainLookupByName(command.getVmName())).thenReturn(vm);
 
-            when(vm.getInfo()).thenReturn(info);
+            when(vm.getInfo()).thenReturn(mockDomainInfo);
 
         } catch (final LibvirtException e) {
             fail(e.getMessage());
@@ -1109,9 +1109,8 @@
     @Test
     public void testStopCommandCheckException1() {
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
-        final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
-        info.state = state;
+        mockDomainInfo.state = state;
 
         final String vmName = "Test";
         final StopCommand command = new StopCommand(vmName, false, true);
@@ -1143,9 +1142,8 @@
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Domain vm = Mockito.mock(Domain.class);
-        final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
-        info.state = state;
+        mockDomainInfo.state = state;
 
         final String vmName = "Test";
         final StopCommand command = new StopCommand(vmName, false, true);
@@ -1155,7 +1153,7 @@
             when(libvirtUtilitiesHelper.getConnectionByVmName(vmName)).thenReturn(conn);
             when(conn.domainLookupByName(command.getVmName())).thenReturn(vm);
 
-            when(vm.getInfo()).thenReturn(info);
+            when(vm.getInfo()).thenReturn(mockDomainInfo);
 
         } catch (final LibvirtException e) {
             fail(e.getMessage());
@@ -1858,7 +1856,6 @@
     public void testCheckVirtualMachineCommand() {
         final Connect conn = Mockito.mock(Connect.class);
         final Domain vm = Mockito.mock(Domain.class);
-        final DomainInfo domainInfo = Mockito.mock(DomainInfo.class);
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
 
         final String vmName = "Test";
@@ -1868,7 +1865,7 @@
         try {
             when(libvirtUtilitiesHelper.getConnectionByVmName(vmName)).thenReturn(conn);
             when(conn.domainLookupByName(vmName)).thenReturn(vm);
-            when(vm.getInfo()).thenReturn(domainInfo);
+            when(vm.getInfo()).thenReturn(mockDomainInfo);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -4115,9 +4112,8 @@
         final KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primaryPool = Mockito.mock(KVMStoragePool.class);
         final Domain vm = Mockito.mock(Domain.class);
-        final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
-        info.state = state;
+        mockDomainInfo.state = state;
 
         final KVMPhysicalDisk disk = Mockito.mock(KVMPhysicalDisk.class);
 
@@ -4127,7 +4123,7 @@
         try {
             when(libvirtUtilitiesHelper.getConnectionByVmName(vmName)).thenReturn(conn);
             when(libvirtComputingResourceMock.getDomain(conn, command.getVmName())).thenReturn(vm);
-            when(vm.getInfo()).thenReturn(info);
+            when(vm.getInfo()).thenReturn(mockDomainInfo);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
@@ -4844,9 +4840,8 @@
         final Connect conn = Mockito.mock(Connect.class);
         final StorageVol v = Mockito.mock(StorageVol.class);
         final Domain vm = Mockito.mock(Domain.class);
-        final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
-        info.state = state;
+        mockDomainInfo.state = state;
 
         when(pool.getType()).thenReturn(StoragePoolType.RBD);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
@@ -4862,7 +4857,7 @@
             when(conn.storageVolLookupByPath(path)).thenReturn(v);
             when(libvirtUtilitiesHelper.getConnectionByVmName(vmInstance)).thenReturn(conn);
             when(conn.domainLookupByName(vmInstance)).thenReturn(vm);
-            when(vm.getInfo()).thenReturn(info);
+            when(vm.getInfo()).thenReturn(mockDomainInfo);
 
             when(conn.getLibVirVersion()).thenReturn(10010l);
         } catch (final LibvirtException e) {
@@ -4903,9 +4898,8 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
         final Domain vm = Mockito.mock(Domain.class);
-        final DomainInfo info = Mockito.mock(DomainInfo.class);
         final DomainState state = DomainInfo.DomainState.VIR_DOMAIN_RUNNING;
-        info.state = state;
+        mockDomainInfo.state = state;
 
         when(pool.getType()).thenReturn(StoragePoolType.Linstor);
         when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
@@ -4918,7 +4912,7 @@
         try {
             when(libvirtUtilitiesHelper.getConnectionByVmName(vmInstance)).thenReturn(conn);
             when(conn.domainLookupByName(vmInstance)).thenReturn(vm);
-            when(vm.getInfo()).thenReturn(info);
+            when(vm.getInfo()).thenReturn(mockDomainInfo);
         } catch (final LibvirtException e) {
             fail(e.getMessage());
         }
