--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -67,6 +67,9 @@
 
     @Mock
     LibvirtUtilitiesHelper libvirtUtilitiesHelper;
+
+    @Mock
+    private DomainBlockJobInfo domainBlockJobInfo;
 
     private String domxml = "<domain type='kvm' id='1'>\n" +
             "  <name>i-2-27-VM</name>\n" +
@@ -357,10 +360,9 @@
         String srcPath = "bec108c400000018:vol-60-7acb-9e22";
         String destPath = "01b381820000001c:vol-60-ec76-b7dc";
         Mockito.doReturn(60).when(command).getWait();
-        DomainBlockJobInfo blockJobInfo = Mockito.mock(DomainBlockJobInfo.class);
-        Mockito.doReturn(blockJobInfo).when(dm).getBlockJobInfo(destDiskLabel, 0);
-        blockJobInfo.cur = 100;
-        blockJobInfo.end = 100;
+        Mockito.doReturn(domainBlockJobInfo).when(dm).getBlockJobInfo(destDiskLabel, 0);
+        domainBlockJobInfo.cur = 100;
+        domainBlockJobInfo.end = 100;
 
         MigrateVolumeAnswer answer = libvirtMigrateVolumeCommandWrapper.checkBlockJobStatus(command, dm, destDiskLabel, srcPath, destPath, libvirtComputingResource, conn, null);
 
@@ -375,10 +377,9 @@
         String srcPath = "bec108c400000018:vol-60-7acb-9e22";
         String destPath = "01b381820000001c:vol-60-ec76-b7dc";
         Mockito.doReturn(1).when(command).getWait();
-        DomainBlockJobInfo blockJobInfo = Mockito.mock(DomainBlockJobInfo.class);
-        Mockito.doReturn(blockJobInfo).when(dm).getBlockJobInfo(destDiskLabel, 0);
-        blockJobInfo.cur = 10;
-        blockJobInfo.end = 100;
+        Mockito.doReturn(domainBlockJobInfo).when(dm).getBlockJobInfo(destDiskLabel, 0);
+        domainBlockJobInfo.cur = 10;
+        domainBlockJobInfo.end = 100;
 
         MigrateVolumeAnswer answer = libvirtMigrateVolumeCommandWrapper.checkBlockJobStatus(command, dm, destDiskLabel, srcPath, destPath, libvirtComputingResource, conn, null);
 
