--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -1289,7 +1289,7 @@
         net.juniper.tungsten.api.types.FirewallRule firewallRule = mock(net.juniper.tungsten.api.types.FirewallRule.class);
         ActionListType actionListType = mock(ActionListType.class);
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
-        FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
+        FirewallRuleEndpointType firewallRuleEndpointType1 = createMockFirewallRuleEndpointType();
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
         FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1303,7 +1303,6 @@
         when(serviceGroup.getReferredName()).thenReturn(List.of("service"));
         when(firewallRule.getEndpoint1()).thenReturn(firewallRuleEndpointType1);
         when(firewallRule.getEndpoint2()).thenReturn(firewallRuleEndpointType2);
-        when(firewallRuleEndpointType1.getTags()).thenReturn(List.of("tag"));
         when(firewallRuleEndpointType2.getTags()).thenReturn(null);
         when(firewallRuleEndpointType2.getAddressGroup()).thenReturn("address:group");
         when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsType);
@@ -1389,7 +1388,7 @@
         net.juniper.tungsten.api.types.FirewallRule firewallRule = mock(net.juniper.tungsten.api.types.FirewallRule.class);
         ActionListType actionListType = mock(ActionListType.class);
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
-        FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
+        FirewallRuleEndpointType firewallRuleEndpointType1 = createMockFirewallRuleEndpointType();
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
         FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1403,7 +1402,6 @@
         when(serviceGroup.getReferredName()).thenReturn(List.of("service"));
         when(firewallRule.getEndpoint1()).thenReturn(firewallRuleEndpointType1);
         when(firewallRule.getEndpoint2()).thenReturn(firewallRuleEndpointType2);
-        when(firewallRuleEndpointType1.getTags()).thenReturn(List.of("tag"));
         when(firewallRuleEndpointType2.getTags()).thenReturn(null);
         when(firewallRuleEndpointType2.getAddressGroup()).thenReturn("address:group");
         when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsType);
@@ -1668,4 +1666,10 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static FirewallRuleEndpointType createMockFirewallRuleEndpointType() {
+        FirewallRuleEndpointType firewallRuleEndpointType = mock(FirewallRuleEndpointType.class);
+        when(firewallRuleEndpointType.getTags()).thenReturn(List.of("tag"));
+        return firewallRuleEndpointType;
+    }
 }
