--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
@@ -62,6 +62,9 @@
     @Mock
     private Connection connectionMock;
 
+    @Mock
+    private Host mockedHost;
+
     private String pathMock = "pathMock";
 
     @Before
@@ -312,11 +315,10 @@
 
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
-        Host hostMock = Mockito.mock(Host.class);
 
         try (MockedStatic<Host> ignored = Mockito.mockStatic(
                 Host.class); MockedStatic<SR> ignored1 = Mockito.mockStatic(SR.class)) {
-            Mockito.when(Host.getByUuid(connectionMock, uuid)).thenReturn(hostMock);
+            Mockito.when(Host.getByUuid(connectionMock, uuid)).thenReturn(mockedHost);
             Mockito.when(SR.introduce(Mockito.eq(connectionMock), Mockito.eq(srUuid), Mockito.eq(pathMock),
                     Mockito.eq(pathMock), Mockito.eq("file"), Mockito.eq("file"), Mockito.eq(false),
                     Mockito.anyMap())).thenThrow(Mockito.mock(exceptionClass));
@@ -339,11 +341,10 @@
         Mockito.doReturn(srMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
-        Host hostMock = Mockito.mock(Host.class);
 
         try (MockedStatic<Host> ignored = Mockito.mockStatic(Host.class);MockedStatic<SR> ignored1 =
                 Mockito.mockStatic(SR.class) ) {
-            Mockito.when(Host.getByUuid(connectionMock, uuid)).thenReturn(hostMock);
+            Mockito.when(Host.getByUuid(connectionMock, uuid)).thenReturn(mockedHost);
 
             InternalError dbUniquenessException = new InternalError(
                     "message: Db_exn.Uniqueness_constraint_violation(\"SR\", \"uuid\", \"fd3edbcf-f142-83d1-3fcb-029ca2446b68\")");
@@ -370,11 +371,10 @@
         SR srMock = Mockito.mock(SR.class);
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
-        Host hostMock = Mockito.mock(Host.class);
 
         try (MockedStatic<Host> ignored = Mockito.mockStatic(Host.class);MockedStatic<SR> ignored1 =
                 Mockito.mockStatic(SR.class);MockedStatic<PBD> pbdMockedStatic = Mockito.mockStatic(PBD.class)) {
-            Mockito.when(Host.getByUuid(connectionMock, uuid)).thenReturn(hostMock);
+            Mockito.when(Host.getByUuid(connectionMock, uuid)).thenReturn(mockedHost);
 
 
             Mockito.when(SR.introduce(Mockito.eq(connectionMock), Mockito.eq(srUuid), Mockito.eq(pathMock),
