--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,7 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    private final Host xenServerHost = Mockito.mock(Host.class);
 
     @Test
     public void testWrapperInstance() {
@@ -1418,7 +1419,6 @@
         final Pool pool = Mockito.mock(Pool.class);
         final Pool.Record poolr = Mockito.mock(Pool.Record.class);
         final Host.Record hostr = Mockito.mock(Host.Record.class);
-        final Host master = Mockito.mock(Host.class);
 
         final ClusterVMMetaDataSyncCommand vmDataSync = new ClusterVMMetaDataSyncCommand(10, 1l);
 
@@ -1428,7 +1428,7 @@
         when(citrixResourceBase.getConnection()).thenReturn(conn);
         try (MockedStatic<Pool.Record> ignored = Mockito.mockStatic(Pool.Record.class)){
             when(citrixResourceBase.getHost()).thenReturn(xsHost);
-            poolr.master = master;
+            poolr.master = xenServerHost;
 
             final Answer answer = wrapper.execute(vmDataSync, citrixResourceBase);
 
@@ -1617,7 +1617,6 @@
         final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
         final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
-        final Host host = Mockito.mock(Host.class);
 
         final ScaleVmCommand scaleVm = new ScaleVmCommand(machineTO);
 
