--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
@@ -47,12 +47,15 @@
 
     MockedStatic<HttpClients> httpClientsMocked;
 
+    private HttpEntity httpEntityMock;
+
     @Before
     public void setup() {
         VRouter vRouter = mock(VRouter.class);
         vRouterApiConnector = new VRouterApiConnectorImpl(vRouter);
         httpClientsMocked = Mockito.mockStatic(HttpClients.class);
         entityUtilsMocked = Mockito.mockStatic(EntityUtils.class);
+        httpEntityMock = mock(HttpEntity.class);
     }
 
     @After
@@ -66,11 +69,10 @@
         Port port = mock(Port.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
         assertTrue(vRouterApiConnector.addPort(port));
@@ -92,11 +94,10 @@
         Port port = mock(Port.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{error:404}");
 
         assertFalse(vRouterApiConnector.addPort(port));
@@ -106,11 +107,10 @@
     public void deletePortTest() throws Exception {
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
         assertTrue(vRouterApiConnector.deletePort("948f421c-edde-4518-a391-09299cc25dc2"));
@@ -130,11 +130,10 @@
     public void enablePortTest() throws Exception {
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
         assertTrue(vRouterApiConnector.enablePort("948f421c-edde-4518-a391-09299cc25dc2"));
@@ -154,11 +153,10 @@
     public void disablePortTest() throws Exception {
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
         assertTrue(vRouterApiConnector.disablePort("948f421c-edde-4518-a391-09299cc25dc2"));
@@ -180,11 +178,10 @@
         Gateway gateway2 = mock(Gateway.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
         assertTrue(vRouterApiConnector.addGateway(Arrays.asList(gateway1, gateway2)));
@@ -208,11 +205,10 @@
         Gateway gateway2 = mock(Gateway.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntityMock);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
         assertTrue(vRouterApiConnector.deleteGateway(Arrays.asList(gateway1, gateway2)));
