--- a/plugins/integrations/cloudian/src/test/java/org/apache/cloudstack/cloudian/client/CloudianClientTest.java
+++ b/plugins/integrations/cloudian/src/test/java/org/apache/cloudstack/cloudian/client/CloudianClientTest.java
@@ -91,6 +91,12 @@
         return group;
     }
 
+    private static HttpResponse createMockHttpResponse(HttpEntity entity) {
+        HttpResponse response = mock(HttpResponse.class);
+        when(response.getEntity()).thenReturn(entity);
+        return response;
+    }
+
     ////////////////////////////////////////////////////////
     //////////////// General API tests /////////////////////
     ////////////////////////////////////////////////////////
@@ -130,8 +136,7 @@
     public void getNonEmptyContentStreamEmpty() {
         InputStream emptyStream = new ByteArrayInputStream(new byte[]{});
         HttpEntity entity = mock(HttpEntity.class);
-        HttpResponse response = mock(HttpResponse.class);
-        when(response.getEntity()).thenReturn(entity);
+        HttpResponse response = createMockHttpResponse(entity);
         try {
                 when(entity.getContent()).thenReturn(emptyStream);
                 Assert.assertNull(client.getNonEmptyContentStream(response));
@@ -144,8 +149,7 @@
     public void getNonEmptyContentStreamWithContent() {
         InputStream nonEmptyStream = new ByteArrayInputStream(new byte[]{9, 8});
         HttpEntity entity = mock(HttpEntity.class);
-        HttpResponse response = mock(HttpResponse.class);
-        when(response.getEntity()).thenReturn(entity);
+        HttpResponse response = createMockHttpResponse(entity);
         try {
                 when(entity.getContent()).thenReturn(nonEmptyStream);
                 InputStream is = client.getNonEmptyContentStream(response);
