--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -260,6 +260,8 @@
     private KVMPhysicalDisk physicalDisk;
     @Mock
     private Map<String, String> details;
+    @Mock
+    private IpPortAndProto ipPortAndProto;
 
     private static final String PHYSICAL_DISK_PATH = "/path/to/disk";
     private static final int DEV_ID = 1;
@@ -3416,7 +3418,7 @@
         final String signature = "signature";
         final Long seqNum = 1l;
         final IpPortAndProto[] ingressRuleSet = new IpPortAndProto[]{Mockito.mock(IpPortAndProto.class)};
-        final IpPortAndProto[] egressRuleSet = new IpPortAndProto[]{Mockito.mock(IpPortAndProto.class)};
+        final IpPortAndProto[] egressRuleSet = new IpPortAndProto[]{ipPortAndProto};
         final List<String> secIps = new Vector<String>();
         final List<String> cidrs = new Vector<String>();
         cidrs.add("0.0.0.0/0");
@@ -3462,8 +3464,8 @@
         final Long vmId = 1l;
         final String signature = "signature";
         final Long seqNum = 1l;
-        final IpPortAndProto[] ingressRuleSet = new IpPortAndProto[]{Mockito.mock(IpPortAndProto.class)};
-        final IpPortAndProto[] egressRuleSet = new IpPortAndProto[]{Mockito.mock(IpPortAndProto.class)};
+        final IpPortAndProto[] ingressRuleSet = new IpPortAndProto[]{ipPortAndProto};
+        final IpPortAndProto[] egressRuleSet = new IpPortAndProto[]{ipPortAndProto};
         final List<String> secIps = new Vector<String>();
         final List<String> cidrs = new Vector<String>();
         cidrs.add("0.0.0.0/0");
@@ -3532,7 +3534,7 @@
         final String signature = "signature";
         final Long seqNum = 1l;
         final IpPortAndProto[] ingressRuleSet = new IpPortAndProto[]{Mockito.mock(IpPortAndProto.class)};
-        final IpPortAndProto[] egressRuleSet = new IpPortAndProto[]{Mockito.mock(IpPortAndProto.class)};
+        final IpPortAndProto[] egressRuleSet = new IpPortAndProto[]{ipPortAndProto};
         final List<String> secIps = new Vector<String>();
 
         final SecurityGroupRulesCmd command = new SecurityGroupRulesCmd(guestIp, guestIp6, guestMac, vmName, vmId, signature, seqNum, ingressRuleSet, egressRuleSet, secIps);
