--- a/engine/schema/src/test/java/com/cloud/upgrade/dao/DatabaseAccessObjectTest.java
+++ b/engine/schema/src/test/java/com/cloud/upgrade/dao/DatabaseAccessObjectTest.java
@@ -410,7 +410,7 @@
 
     @Test
     public void testDropColumn() throws Exception {
-        when(connectionMock.prepareStatement(anyString())).thenReturn(preparedStatementMock);
+        connectionMock = MockConnectionWithPreparedStatement.createMockConnection(preparedStatementMock);
 
         Connection conn = connectionMock;
         String tableName = "tableName";
@@ -495,7 +495,7 @@
     @Test
     public void testDropColumnWhenexecuteUpdateResultsInException() throws Exception {
         SQLException sqlException = new SQLException();
-        when(connectionMock.prepareStatement(anyString())).thenReturn(preparedStatementMock);
+        connectionMock = MockConnectionWithPreparedStatement.createMockConnection(preparedStatementMock);
         when(preparedStatementMock.executeUpdate()).thenThrow(sqlException);
 
         Connection conn = connectionMock;
--- a/engine/schema/src/test/java/com/cloud/upgrade/dao/Upgrade42040to42100Test.java
+++ b/engine/schema/src/test/java/com/cloud/upgrade/dao/Upgrade42040to42100Test.java
@@ -46,6 +46,8 @@
     public void testPerformDataMigration() throws SQLException {
         try (MockedStatic<DbUpgradeUtils> ignored = Mockito.mockStatic(DbUpgradeUtils.class)) {
             DbUpgradeUtils dbUpgradeUtils = Mockito.mock(DbUpgradeUtils.class);
+            PreparedStatement preparedStatement = Mockito.mock(PreparedStatement.class);
+            conn = MockConnectionWithPreparedStatement.createMockConnection(preparedStatement);
             when(dbUpgradeUtils.getTableColumnType(conn, "configuration", "scope")).thenReturn("varchar(255)");
 
             try (MockedStatic<TransactionLegacy> ignored2 = Mockito.mockStatic(TransactionLegacy.class)) {
@@ -67,11 +69,9 @@
                         "     END WHERE scope IS NOT NULL;";
                 when(txn.prepareAutoCloseStatement(sql)).thenReturn(pstmt);
 
-                PreparedStatement preparedStatement = Mockito.mock(PreparedStatement.class);
                 ResultSet resultSet = Mockito.mock(ResultSet.class);
                 Mockito.when(resultSet.next()).thenReturn(false);
                 Mockito.when(preparedStatement.executeQuery()).thenReturn(resultSet);
-                Mockito.when(conn.prepareStatement(anyString())).thenReturn(preparedStatement);
                 upgrade.performDataMigration(conn);
 
                 Mockito.verify(pstmt, Mockito.times(1)).executeUpdate();
--- a/engine/schema/src/test/java/com/cloud/upgrade/dao/Upgrade42100to42200Test.java
+++ b/engine/schema/src/test/java/com/cloud/upgrade/dao/Upgrade42100to42200Test.java
@@ -182,7 +182,7 @@
 
     @Test
     public void testPerformDataMigration() throws SQLException {
-        when(conn.prepareStatement(anyString())).thenReturn(selectStmt);
+        conn = MockConnectionWithPreparedStatement.createMockConnection(selectStmt);
         when(selectStmt.executeQuery()).thenReturn(resultSet);
         when(resultSet.next()).thenReturn(false);
 
--- a/engine/schema/src/test/java/com/cloud/upgrade/dao/MockConnectionWithPreparedStatement.java
+++ b/engine/schema/src/test/java/com/cloud/upgrade/dao/MockConnectionWithPreparedStatement.java
@@ -0,0 +1,15 @@
+package com.cloud.upgrade.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.mockito.Mockito;
+
+public class MockConnectionWithPreparedStatement {
+    public static Connection createMockConnection(PreparedStatement preparedStatement) throws SQLException {
+        Connection connectionMock = Mockito.mock(Connection.class);
+        Mockito.when(connectionMock.prepareStatement(Mockito.anyString())).thenReturn(preparedStatement);
+        return connectionMock;
+    }
+}
