--- a/engine/schema/src/test/java/com/cloud/storage/dao/VolumeDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/storage/dao/VolumeDaoImplTest.java
@@ -54,6 +54,9 @@
     private PreparedStatement preparedStatementMock;
 
     @Mock
+    private ResultSet resultSetMock;
+
+    @Mock
     private TransactionLegacy transactionMock;
 
     private static MockedStatic<TransactionLegacy> mockedTransactionLegacy;
@@ -80,8 +83,7 @@
 
         when(TransactionLegacy.currentTxn()).thenReturn(transactionMock);
         when(transactionMock.prepareAutoCloseStatement(startsWith(ORDER_POOLS_NUMBER_OF_VOLUMES_FOR_ACCOUNT_QUERY_WITH_CLUSTER))).thenReturn(preparedStatementMock);
-        ResultSet rs = Mockito.mock(ResultSet.class);
-        when(preparedStatementMock.executeQuery()).thenReturn(rs, rs);
+        when(preparedStatementMock.executeQuery()).thenReturn(resultSetMock, resultSetMock);
 
         volumeDao.listPoolIdsByVolumeCount(dcId, podId, clusterId, accountId);
 
@@ -102,8 +104,7 @@
 
         when(TransactionLegacy.currentTxn()).thenReturn(transactionMock);
         when(transactionMock.prepareAutoCloseStatement(startsWith(ORDER_POOLS_NUMBER_OF_VOLUMES_FOR_ACCOUNT_QUERY_WITHOUT_CLUSTER))).thenReturn(preparedStatementMock);
-        ResultSet rs = Mockito.mock(ResultSet.class);
-        when(preparedStatementMock.executeQuery()).thenReturn(rs, rs);
+        when(preparedStatementMock.executeQuery()).thenReturn(resultSetMock, resultSetMock);
 
         volumeDao.listPoolIdsByVolumeCount(dcId, null, null, accountId);
 
