--- a/engine/storage/configdrive/src/test/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilderTest.java
+++ b/engine/storage/configdrive/src/test/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilderTest.java
@@ -232,8 +232,7 @@
 
     @Test
     public void linkUserDataTestUserDataFilePathDoesNotExist() {
-        File fileMock = Mockito.mock(File.class);
-        Mockito.doReturn(false).when(fileMock).exists();
+        File fileMock = createMockFile(false);
 
         try (MockedStatic<ConfigDriveBuilder> configDriveBuilderMocked = Mockito.mockStatic(ConfigDriveBuilder.class);
              MockedConstruction<Script> scriptMock = Mockito.mockConstruction(Script.class)
@@ -247,8 +246,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void linkUserDataTestUserDataFilePathExistAndExecutionPresentedSomeError() {
-        File fileMock = Mockito.mock(File.class);
-        Mockito.doReturn(true).when(fileMock).exists();
+        File fileMock = createMockFile(true);
 
         try (MockedStatic<ConfigDriveBuilder> configDriveBuilderMocked = Mockito.mockStatic(ConfigDriveBuilder.class);
              MockedConstruction<Script> ignored = Mockito.mockConstruction(Script.class, (mock, context) -> {
@@ -270,8 +268,7 @@
                  Mockito.doReturn(StringUtils.EMPTY).when(mock).execute();
              })
         ) {
-            File fileMock = Mockito.mock(File.class);
-            Mockito.doReturn(true).when(fileMock).exists();
+            File fileMock = createMockFile(true);
             Mockito.when(ConfigDriveBuilder.getFile(anyString())).thenReturn(fileMock);
 
             configDriveBuilderMocked.when(() -> ConfigDriveBuilder.linkUserData(anyString())).thenCallRealMethod();
@@ -424,8 +421,7 @@
     public void getProgramToGenerateIsoTestGenIsoExistsAndIsExecutable() throws Exception {
         try (MockedStatic<ConfigDriveBuilder> ignored = Mockito.mockStatic(ConfigDriveBuilder.class)) {
 
-            File genIsoFileMock = Mockito.mock(File.class);
-            Mockito.doReturn(true).when(genIsoFileMock).exists();
+            File genIsoFileMock = createMockFile(true);
             Mockito.doReturn(true).when(genIsoFileMock).canExecute();
 
             Mockito.when(ConfigDriveBuilder.getFile("/usr/bin/genisoimage")).thenReturn(genIsoFileMock);
@@ -443,8 +439,7 @@
     public void getProgramToGenerateIsoTestGenIsoExistsbutNotExecutable() throws Exception {
         try (MockedStatic<ConfigDriveBuilder> ignored = Mockito.mockStatic(ConfigDriveBuilder.class)) {
 
-            File genIsoFileMock = Mockito.mock(File.class);
-            Mockito.doReturn(true).when(genIsoFileMock).exists();
+            File genIsoFileMock = createMockFile(true);
             Mockito.doReturn(false).when(genIsoFileMock).canExecute();
 
             Mockito.when(ConfigDriveBuilder.getFile("/usr/bin/genisoimage")).thenReturn(genIsoFileMock);
@@ -459,8 +454,7 @@
     public void getProgramToGenerateIsoTestNotGenIsoMkIsoInLinux() throws Exception {
         try (MockedStatic<ConfigDriveBuilder> ignored = Mockito.mockStatic(ConfigDriveBuilder.class)) {
 
-            File genIsoFileMock = Mockito.mock(File.class);
-            Mockito.doReturn(false).when(genIsoFileMock).exists();
+            File genIsoFileMock = createMockFile(false);
 
             File mkIsoProgramInLinuxFileMock = Mockito.mock(File.class);
             Mockito.doReturn(true).when(mkIsoProgramInLinuxFileMock).exists();
@@ -488,14 +482,11 @@
         try (MockedStatic<ConfigDriveBuilder> ignored = Mockito.mockStatic(ConfigDriveBuilder.class)) {
 
 
-            File genIsoFileMock = Mockito.mock(File.class);
-            Mockito.doReturn(false).when(genIsoFileMock).exists();
-
-            File mkIsoProgramInLinuxFileMock = Mockito.mock(File.class);
-            Mockito.doReturn(false).when(mkIsoProgramInLinuxFileMock).exists();
-
-            File mkIsoProgramInMacOsFileMock = Mockito.mock(File.class);
-            Mockito.doReturn(true).when(mkIsoProgramInMacOsFileMock).exists();
+            File genIsoFileMock = createMockFile(false);
+
+            File mkIsoProgramInLinuxFileMock = createMockFile(false);
+
+            File mkIsoProgramInMacOsFileMock = createMockFile(true);
             Mockito.doReturn(true).when(mkIsoProgramInMacOsFileMock).canExecute();
 
             Mockito.when(ConfigDriveBuilder.getFile("/usr/bin/genisoimage")).thenReturn(genIsoFileMock);
@@ -659,4 +650,10 @@
         Assert.assertEquals(expectedJsonObject, actualJson);
         folder.delete();
     }
+
+    private static File createMockFile(boolean exists) {
+        File genIsoFileMock = Mockito.mock(File.class);
+        Mockito.doReturn(exists).when(genIsoFileMock).exists();
+        return genIsoFileMock;
+    }
 }
