--- a/core/src/test/java/com/cloud/resource/ServerResourceBaseTest.java
+++ b/core/src/test/java/com/cloud/resource/ServerResourceBaseTest.java
@@ -65,8 +65,7 @@
 
     @Test
     public void isValidNicToUseAsPrivateNicTestReturnFalseWhenNicIsVirtual() {
-        NetworkInterface networkInterfaceMock = Mockito.mock(NetworkInterface.class);
-        Mockito.when(networkInterfaceMock.isVirtual()).thenReturn(true);
+        NetworkInterface networkInterfaceMock = createMockNetworkInterface(true);
 
         Assert.assertFalse(serverResourceBaseSpy.isValidNicToUseAsPrivateNic(networkInterfaceMock));
     }
@@ -82,8 +81,7 @@
 
     @Test
     public void isValidNicToUseAsPrivateNicTestReturnFalseWhenNicNameContainsColon() {
-        NetworkInterface networkInterfaceMock = Mockito.mock(NetworkInterface.class);
-        Mockito.when(networkInterfaceMock.isVirtual()).thenReturn(false);
+        NetworkInterface networkInterfaceMock = createMockNetworkInterface(false);
         Mockito.when(networkInterfaceMock.getName()).thenReturn("te:st");
 
         Assert.assertFalse(serverResourceBaseSpy.isValidNicToUseAsPrivateNic(networkInterfaceMock));
@@ -213,6 +211,12 @@
 
             verifyAndAssertNetworkInterfaces(mockedNetUtils, "cloudbr0", "cloudbr1", "cloudbr2", "cloudbr3");
         }
+    }
+
+    private static NetworkInterface createMockNetworkInterface(boolean isVirtual) {
+        NetworkInterface networkInterfaceMock = Mockito.mock(NetworkInterface.class);
+        Mockito.when(networkInterfaceMock.isVirtual()).thenReturn(isVirtual);
+        return networkInterfaceMock;
     }
 
     private Map<String, Object> createParamsMap(String... params) {
