--- a/core/src/test/java/com/cloud/storage/template/QCOW2ProcessorTest.java
+++ b/core/src/test/java/com/cloud/storage/template/QCOW2ProcessorTest.java
@@ -42,6 +42,9 @@
     @Mock
     StorageLayer mockStorageLayer;
 
+    @Mock
+    private File sharedMockFile;
+
     @Before
     public void setUp() throws Exception {
         processor = Mockito.spy(new QCOW2Processor());
@@ -56,9 +59,8 @@
         String templateName = "template";
 
         Mockito.when(mockStorageLayer.exists(Mockito.anyString())).thenReturn(true);
-        File mockFile = Mockito.mock(File.class);
 
-        Mockito.when(mockStorageLayer.getFile(Mockito.anyString())).thenReturn(mockFile);
+        Mockito.when(mockStorageLayer.getFile(Mockito.anyString())).thenReturn(sharedMockFile);
         Mockito.when(mockStorageLayer.getSize(Mockito.anyString())).thenReturn(1000L);
         Mockito.doThrow(new IOException("virtual size calculation failed")).when(processor).getTemplateVirtualSize((File)Mockito.any());
 
@@ -73,9 +75,8 @@
         long actualSize = 1000;
 
         Mockito.when(mockStorageLayer.exists(Mockito.anyString())).thenReturn(true);
-        File mockFile = Mockito.mock(File.class);
 
-        Mockito.when(mockStorageLayer.getFile(Mockito.anyString())).thenReturn(mockFile);
+        Mockito.when(mockStorageLayer.getFile(Mockito.anyString())).thenReturn(sharedMockFile);
         Mockito.when(mockStorageLayer.getSize(Mockito.anyString())).thenReturn(actualSize);
         Mockito.doReturn(virtualSize).when(processor).getTemplateVirtualSize((File)Mockito.any());
 
@@ -100,9 +101,8 @@
     @Test
     public void testGetVirtualSize() throws Exception {
         long virtualSize = 2000;
-        File mockFile = Mockito.mock(File.class);
         Mockito.doReturn(virtualSize).when(processor).getTemplateVirtualSize((File)Mockito.any());
-        Assert.assertEquals(virtualSize, processor.getVirtualSize(mockFile));
-        Mockito.verify(mockFile, Mockito.times(0)).length();
+        Assert.assertEquals(virtualSize, processor.getVirtualSize(sharedMockFile));
+        Mockito.verify(sharedMockFile, Mockito.times(0)).length();
     }
 }
