--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
@@ -158,6 +158,8 @@
     @Mock
     VirtualDevice virtualDevice;
     @Mock
+    private List<VirtualDeviceConfigSpec> virtualDeviceChanges;
+    @Mock
     DataTO srcDataTO;
     @Mock
     NfsTO srcDataNfsTO;
@@ -848,12 +850,11 @@
         details.put(VmDetailConstants.VIRTUAL_TPM_ENABLED, "true");
         when(vmSpec.getDetails()).thenReturn(details);
         when(vmMo.getAllDeviceList()).thenReturn(new ArrayList<>());
-        List<VirtualDeviceConfigSpec> deviceChanges = Mockito.mock(List.class);
-        when(vmConfigSpec.getDeviceChange()).thenReturn(deviceChanges);
+        when(vmConfigSpec.getDeviceChange()).thenReturn(virtualDeviceChanges);
 
         vmwareResource.configureVirtualTPM(vmMo, vmSpec, vmConfigSpec);
         Mockito.verify(vmwareResource, Mockito.times(1)).addVirtualTPMDevice(vmConfigSpec);
-        Mockito.verify(deviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
+        Mockito.verify(virtualDeviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
     }
 
     @Test
@@ -867,11 +868,10 @@
         when(vmSpec.getDetails()).thenReturn(details);
         VirtualTPM tpm = new VirtualTPM();
         when(vmMo.getAllDeviceList()).thenReturn(List.of(tpm));
-        List<VirtualDeviceConfigSpec> deviceChanges = Mockito.mock(List.class);
-        when(vmConfigSpec.getDeviceChange()).thenReturn(deviceChanges);
+        when(vmConfigSpec.getDeviceChange()).thenReturn(virtualDeviceChanges);
 
         vmwareResource.configureVirtualTPM(vmMo, vmSpec, vmConfigSpec);
         Mockito.verify(vmwareResource, Mockito.times(1)).removeVirtualTPMDevice(vmConfigSpec, tpm);
-        Mockito.verify(deviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
+        Mockito.verify(virtualDeviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
     }
 }
