--- a/utils/src/test/java/com/cloud/utils/ScriptTest.java
+++ b/utils/src/test/java/com/cloud/utils/ScriptTest.java
@@ -51,8 +51,7 @@
     @Test
     public void testLogger() {
         Assume.assumeTrue(SystemUtils.IS_OS_LINUX);
-        Logger mock = Mockito.mock(Logger.class);
-        Mockito.doNothing().when(mock).debug((Message) ArgumentMatchers.any());
+        Logger mock = createMockLogger();
         Script script = new Script("/bin/echo", mock);
         script.execute();
     }
@@ -80,8 +79,7 @@
     @Ignore
     public void testExecute() {
         Assume.assumeTrue(SystemUtils.IS_OS_LINUX);
-        Logger mock = Mockito.mock(Logger.class);
-        Mockito.doNothing().when(mock).debug((Message) ArgumentMatchers.any());
+        Logger mock = createMockLogger();
         for (int i = 0; i < 100000; i++) {
             Script script = new Script("/bin/false", mock);
             script.execute();
@@ -147,4 +145,10 @@
         String script = Script.findScript("/bin", "pwd");
         Assert.assertNotNull("/bin/pwd should be there on linux", script);
     }
+
+    private static Logger createMockLogger() {
+        Logger mock = Mockito.mock(Logger.class);
+        Mockito.doNothing().when(mock).debug((Message) ArgumentMatchers.any());
+        return mock;
+    }
 }
