--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -183,6 +183,9 @@
     @Mock
     TungstenApi tungstenApi;
 
+    @Mock
+    private LogicalRouter logicalRouterMock;
+
     TungstenResource tungstenResource;
 
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
@@ -290,13 +293,12 @@
     public void executeRequestCreateTungstenLogicalRouterCommandTest() {
         TungstenCommand command = new CreateTungstenLogicalRouterCommand("logicalRouter", "projectFqn",
             "e8281cd6-9078-4db1-9f47-e52f679e03d1");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
-
-        when(tungstenApi.createTungstenLogicalRouter(anyString(), anyString(), anyString())).thenReturn(logicalRouter);
-
-        TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
-        assertTrue(answer.getResult());
-        assertEquals(logicalRouter, answer.getApiObjectBase());
+
+        when(tungstenApi.createTungstenLogicalRouter(anyString(), anyString(), anyString())).thenReturn(logicalRouterMock);
+
+        TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
+        assertTrue(answer.getResult());
+        assertEquals(logicalRouterMock, answer.getApiObjectBase());
     }
 
     @Test
@@ -330,13 +332,12 @@
     public void executeRequestSetTungstenNetworkGatewayCommandTest() {
         TungstenCommand command = new SetTungstenNetworkGatewayCommand("projectFqn", "router", 1L,
             "b604c7f7-1dbc-42d8-bceb-2c0898034a7a", "192.168.1.1");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         InstanceIp instanceIp = mock(InstanceIp.class);
 
         when(virtualMachineInterface.getUuid()).thenReturn("ac617be6-bf80-4086-9d6a-c05ff78e2264");
         when(tungstenApi.getTungstenObjectByName(eq(LogicalRouter.class), any(), anyString())).thenReturn(
-            logicalRouter);
+            logicalRouterMock);
         when(tungstenApi.createTungstenGatewayVmi(anyString(), anyString(), anyString())).thenReturn(
             virtualMachineInterface);
         when(tungstenApi.createTungstenInstanceIp(anyString(), anyString(), anyString(), anyString())).thenReturn(
@@ -376,10 +377,9 @@
     public void executeRequestClearTungstenNetworkGatewayCommandTest() {
         TungstenCommand command = new ClearTungstenNetworkGatewayCommand("projectFqn", "router", 1L);
 
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         when(tungstenApi.getTungstenObjectByName(eq(LogicalRouter.class), any(), anyString())).thenReturn(
-            logicalRouter);
+            logicalRouterMock);
         when(tungstenApi.deleteTungstenObject(any(LogicalRouter.class))).thenReturn(true);
         when(tungstenApi.getTungstenObjectByName(eq(VirtualMachineInterface.class), any(), anyString())).thenReturn(
             virtualMachineInterface);
@@ -1371,11 +1371,10 @@
     @Test
     public void executeRequestCreateTungstenRoutingLogicalRouterCommandTest() {
         TungstenCommand command = new CreateTungstenRoutingLogicalRouterCommand("projectFqn", "logicalRouterName");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork1 = mock(VirtualNetwork.class);
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         Answer<List<ApiObjectBase>> virtualNetworksAnswer = setupApiObjectBaseListAnswer(virtualNetwork1, virtualNetwork2);
-        when(tungstenApi.createRoutingLogicalRouter(anyString(), anyString(), anyString())).thenReturn(logicalRouter);
+        when(tungstenApi.createRoutingLogicalRouter(anyString(), anyString(), anyString())).thenReturn(logicalRouterMock);
         when(tungstenApi.listConnectedNetworkFromLogicalRouter(any(LogicalRouter.class))).thenAnswer(virtualNetworksAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1386,11 +1385,10 @@
     public void executeRequestAddTungstenNetworkGatewayToLogicalRouterCommandTest() {
         TungstenCommand command = new AddTungstenNetworkGatewayToLogicalRouterCommand("c2f8e1f8-10b6-11ec-82a8-0242ac130003",
                 "439a7efc-113e-11ec-82a8-0242ac130003", "10.0.0.0");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork1 = mock(VirtualNetwork.class);
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         Answer<List<ApiObjectBase>> virtualNetworksAnswer = setupApiObjectBaseListAnswer(virtualNetwork1, virtualNetwork2);
-        when(tungstenApi.addNetworkGatewayToLogicalRouter(anyString(), anyString(), anyString())).thenReturn(logicalRouter);
+        when(tungstenApi.addNetworkGatewayToLogicalRouter(anyString(), anyString(), anyString())).thenReturn(logicalRouterMock);
         when(tungstenApi.listConnectedNetworkFromLogicalRouter(any(LogicalRouter.class))).thenAnswer(virtualNetworksAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1401,11 +1399,10 @@
     public void executeRequestRemoveTungstenNetworkGatewayFromLogicalRouterCommandTest() {
         TungstenCommand command = new RemoveTungstenNetworkGatewayFromLogicalRouterCommand("c2f8e1f8-10b6-11ec-82a8-0242ac130003",
                 "439a7efc-113e-11ec-82a8-0242ac130003");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
         VirtualNetwork virtualNetwork1 = mock(VirtualNetwork.class);
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         List<VirtualNetwork> virtualNetworks = Arrays.asList(virtualNetwork1, virtualNetwork2);
-        when(tungstenApi.removeNetworkGatewayFromLogicalRouter(anyString(), anyString())).thenReturn(logicalRouter);
+        when(tungstenApi.removeNetworkGatewayFromLogicalRouter(anyString(), anyString())).thenReturn(logicalRouterMock);
         when(tungstenApi.listConnectedNetworkFromLogicalRouter(any(LogicalRouter.class))).thenReturn(virtualNetworks);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1415,9 +1412,8 @@
     @Test
     public void executeRequestListTungstenRoutingLogicalRouterCommandTest() {
         TungstenCommand command = new ListTungstenRoutingLogicalRouterCommand(null, "c2f8e1f8-10b6-11ec-82a8-0242ac130003");
-        LogicalRouter logicalRouter1 = mock(LogicalRouter.class);
         LogicalRouter logicalRouter2 = mock(LogicalRouter.class);
-        Answer<List<ApiObjectBase>> logicalRoutersAnswer = setupApiObjectBaseListAnswer(logicalRouter1, logicalRouter2);
+        Answer<List<ApiObjectBase>> logicalRoutersAnswer = setupApiObjectBaseListAnswer(logicalRouterMock, logicalRouter2);
         VirtualNetwork virtualNetwork1 = mock(VirtualNetwork.class);
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         List<VirtualNetwork> virtualNetworks = Arrays.asList(virtualNetwork1, virtualNetwork2);
@@ -1431,9 +1427,8 @@
     @Test
     public void executeRequestDeleteTungstenRoutingLogicalRouterCommandSuccessTest() {
         TungstenCommand command = new DeleteTungstenRoutingLogicalRouterCommand("c2f8e1f8-10b6-11ec-82a8-0242ac130003");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
-
-        when(tungstenApi.getTungstenObject(eq(LogicalRouter.class), anyString())).thenReturn(logicalRouter);
+
+        when(tungstenApi.getTungstenObject(eq(LogicalRouter.class), anyString())).thenReturn(logicalRouterMock);
         when(tungstenApi.deleteTungstenObject(any(LogicalRouter.class))).thenReturn(true);
 
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
@@ -1443,11 +1438,10 @@
     @Test
     public void executeRequestDeleteTungstenRoutingLogicalRouterCommandFailTest() {
         TungstenCommand command = new DeleteTungstenRoutingLogicalRouterCommand("c2f8e1f8-10b6-11ec-82a8-0242ac130003");
-        LogicalRouter logicalRouter = mock(LogicalRouter.class);
         ObjectReference objectReference = mock(ObjectReference.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
 
-        when(tungstenApi.getTungstenObject(eq(LogicalRouter.class), anyString())).thenReturn(logicalRouter);
+        when(tungstenApi.getTungstenObject(eq(LogicalRouter.class), anyString())).thenReturn(logicalRouterMock);
         when(tungstenApi.deleteTungstenObject(any(LogicalRouter.class))).thenReturn(true);
 
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
