--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
@@ -116,6 +116,9 @@
     @Mock
     DiskOfferingDao diskOfferingDao;
 
+    @Mock
+    private ManagedObjectReference morMock;
+
     AutoCloseable closeable;
 
     @Mock
@@ -229,24 +232,23 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
-        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
-
-        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
-            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
-            Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
+
+        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
+            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
+            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(null).when(mockDatacenterMO).findVm(vmName);
         })) {
@@ -266,14 +268,13 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
         DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
         VirtualMachineMO vmMo = Mockito.mock(VirtualMachineMO.class);
         HostMO hostMo = Mockito.mock(HostMO.class);
@@ -287,10 +288,10 @@
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
             Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(vmMo).when(mockDatacenterMO).findVm(vmName);
         })) {
@@ -310,14 +311,13 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
         DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
         VirtualMachineMO vmMo = Mockito.mock(VirtualMachineMO.class);
         HostMO hostMo = Mockito.mock(HostMO.class);
@@ -334,10 +334,10 @@
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
             Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(vmMo).when(mockDatacenterMO).findVm(vmName);
         })) {
@@ -357,26 +357,25 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
         DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
         VirtualMachineMO vmMo = Mockito.mock(VirtualMachineMO.class);
         HostMO hostMo = Mockito.mock(HostMO.class);
         Mockito.doReturn(VirtualMachinePowerState.POWERED_ON).when(vmMo).getPowerState();
         Mockito.doReturn(hostMo).when(vmMo).getRunningHost();
-        Mockito.doReturn(mor).when(hostMo).getHyperHostOwnerResourcePool();
+        Mockito.doReturn(morMock).when(hostMo).getHyperHostOwnerResourcePool();
         UnmanagedInstanceTO instance = Mockito.mock(UnmanagedInstanceTO.class);
         Mockito.doReturn("CentOS").when(instance).getOperatingSystem();
         Mockito.doReturn("test-cluster").when(instance).getClusterName();
         when(VmwareHelper.getUnmanagedInstance(hostMo, vmMo)).thenReturn(instance);
         DatastoreMO datastoreMO = Mockito.mock(DatastoreMO.class);
-        Mockito.doReturn(mor).when(datastoreMO).getMor();
+        Mockito.doReturn(morMock).when(datastoreMO).getMor();
         List<DatastoreMO> datastores = new ArrayList<>();
         datastores.add(datastoreMO);
         Mockito.doReturn(datastores).when(vmMo).getAllDatastores();
@@ -386,14 +385,14 @@
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
             Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(vmMo).when(mockDatacenterMO).findVm(anyString());
-            Mockito.doReturn(mor).when(mockDatacenterMO).getVmFolder();
-            Mockito.doReturn(mor).when(mockDatacenterMO).getMor();
+            Mockito.doReturn(morMock).when(mockDatacenterMO).getVmFolder();
+            Mockito.doReturn(morMock).when(mockDatacenterMO).getMor();
         })) {
             vMwareGuru.getHypervisorVMOutOfBandAndCloneIfRequired(hostIp, vmName, params);
         }
@@ -411,23 +410,22 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
         DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
         VirtualMachineMO vmMo = Mockito.mock(VirtualMachineMO.class);
         HostMO hostMo = Mockito.mock(HostMO.class);
         Mockito.doReturn(VirtualMachinePowerState.POWERED_ON).when(vmMo).getPowerState();
         Mockito.doReturn(hostMo).when(vmMo).getRunningHost();
-        Mockito.doReturn(mor).when(hostMo).getHyperHostOwnerResourcePool();
-        Mockito.doReturn(mor).when(hostMo).getMor();
+        Mockito.doReturn(morMock).when(hostMo).getHyperHostOwnerResourcePool();
+        Mockito.doReturn(morMock).when(hostMo).getMor();
         DatastoreMO datastoreMO = Mockito.mock(DatastoreMO.class);
-        Mockito.doReturn(mor).when(datastoreMO).getMor();
+        Mockito.doReturn(morMock).when(datastoreMO).getMor();
         List<DatastoreMO> datastores = new ArrayList<>();
         datastores.add(datastoreMO);
         Mockito.doReturn(datastores).when(vmMo).getAllDatastores();
@@ -446,14 +444,14 @@
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
             Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(vmMo).when(mockDatacenterMO).findVm(anyString());
-            Mockito.doReturn(mor).when(mockDatacenterMO).getVmFolder();
-            Mockito.doReturn(mor).when(mockDatacenterMO).getMor();
+            Mockito.doReturn(morMock).when(mockDatacenterMO).getVmFolder();
+            Mockito.doReturn(morMock).when(mockDatacenterMO).getMor();
         })) {
             Pair<UnmanagedInstanceTO, Boolean> clonedVm = vMwareGuru.getHypervisorVMOutOfBandAndCloneIfRequired(hostIp, vmName, params);
             assertNotNull(clonedVm);
@@ -473,24 +471,23 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         DataStoreTO dataStore = Mockito.mock(DataStoreTO.class);
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
-        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
-
-        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
-            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
-            Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
+
+        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
+            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
+            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(null).when(mockDatacenterMO).findVm(vmName);
         })) {
@@ -510,14 +507,13 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
         DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
         VirtualMachineMO vmMo = Mockito.mock(VirtualMachineMO.class);
         Mockito.doNothing().when(vmMo).exportVm(anyString(), anyString(), anyBoolean(), anyBoolean(), anyInt());
@@ -528,10 +524,10 @@
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
             Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(vmMo).when(mockDatacenterMO).findVm(vmName);
         })) {
@@ -553,24 +549,23 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
-        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
-
-        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
-            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
-            Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
+
+        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
+            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
+            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(null).when(mockDatacenterMO).findVm(vmName);
         })) {
@@ -591,14 +586,13 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_USERNAME, "username");
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
-        ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(morMock).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(morMock).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
         DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
         VirtualMachineMO vmMo = Mockito.mock(VirtualMachineMO.class);
         Mockito.doReturn(true).when(vmMo).destroy();
@@ -607,10 +601,10 @@
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
             Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
-            Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
-        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
-            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
-            Mockito.doReturn(mor).when(mockVmwareContext).getRootFolder();
+            Mockito.doReturn(morMock).when(mockVmwareClient).getRootFolder();
+        }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
+            Mockito.doReturn(vimClient).when(mockVmwareContext).getVimClient();
+            Mockito.doReturn(morMock).when(mockVmwareContext).getRootFolder();
         }); MockedConstruction<DatacenterMO> ignored3 = Mockito.mockConstruction(DatacenterMO.class, withSettings().spiedInstance(dataCenterMO), (mockDatacenterMO, contextDatacenterMO) -> {
             Mockito.doReturn(vmMo).when(mockDatacenterMO).findVm(vmName);
         })) {
