--- a/utils/src/test/java/org/apache/cloudstack/utils/mailing/SMTPMailSenderTest.java
+++ b/utils/src/test/java/org/apache/cloudstack/utils/mailing/SMTPMailSenderTest.java
@@ -61,6 +61,9 @@
     @Mock
     SMTPMessage messageMock;
 
+    @Mock
+    MimeMessage mimeMessageMock;
+
     private Map<String, String> configsMock = Mockito.mock(Map.class);
     private String namespace = "test";
     private String enabledProtocols = "mail.smtp.ssl.protocols";
@@ -557,7 +560,7 @@
 
     @Test
     public void setMailRecipientsTest() throws UnsupportedEncodingException, MessagingException {
-        SMTPMessage messageMock = new SMTPMessage(Mockito.mock(MimeMessage.class));
+        SMTPMessage messageMock = new SMTPMessage(mimeMessageMock);
 
         Set<MailAddress> recipients = new LinkedHashSet<>();
         recipients.add(new MailAddress(null));
@@ -579,7 +582,7 @@
 
     @Test
     public void setMailRecipientsTestOnlyInvalidEmailSettings() throws UnsupportedEncodingException, MessagingException {
-        SMTPMessage messageMock = new SMTPMessage(Mockito.mock(MimeMessage.class));
+        SMTPMessage messageMock = new SMTPMessage(mimeMessageMock);
 
         messageMock = messageMock = Mockito.spy(messageMock);
         Mockito.doReturn(new Address[0]).when(messageMock).getAllRecipients();
@@ -599,7 +602,7 @@
 
     @Test
     public void setMailRecipientsTestWarnLogging() throws UnsupportedEncodingException, MessagingException  {
-        SMTPMessage messageMock = new SMTPMessage(Mockito.mock(MimeMessage.class));
+        SMTPMessage messageMock = new SMTPMessage(mimeMessageMock);
 
         smtpMailSenderMock.setMailRecipients(messageMock, null, "");
 
