--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,8 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    @Mock
+    private Network networkMock;
 
     @Test
     public void testWrapperInstance() {
@@ -733,7 +735,6 @@
 
         final Connection conn = Mockito.mock(Connection.class);
         final XsLocalNetwork network = Mockito.mock(XsLocalNetwork.class);
-        final Network network2 = Mockito.mock(Network.class);
         final PIF pif = Mockito.mock(PIF.class);
         final PIF.Record pifRec = Mockito.mock(PIF.Record.class);
 
@@ -750,7 +751,7 @@
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
 
         try {
-            when(network.getNetwork()).thenReturn(network2);
+            when(network.getNetwork()).thenReturn(networkMock);
             when(network.getPif(conn)).thenReturn(pif);
             when(network.getPif(conn)).thenReturn(pif);
             when(pif.getRecord(conn)).thenReturn(pifRec);
@@ -1176,7 +1177,6 @@
     @Test
     public void testOvsDestroyBridgeCommand() {
         final Connection conn = Mockito.mock(Connection.class);
-        final Network network = Mockito.mock(Network.class);
 
         final OvsDestroyBridgeCommand destroyBridge = new OvsDestroyBridgeCommand(1l, "bridge", 1l);
 
@@ -1184,19 +1184,19 @@
         assertNotNull(wrapper);
 
         when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.findOrCreateTunnelNetwork(conn, destroyBridge.getBridgeName())).thenReturn(network);
+        when(citrixResourceBase.findOrCreateTunnelNetwork(conn, destroyBridge.getBridgeName())).thenReturn(networkMock);
 
         final Answer answer = wrapper.execute(destroyBridge, citrixResourceBase);
 
         verify(citrixResourceBase, times(1)).getConnection();
         try {
-            verify(citrixResourceBase, times(1)).cleanUpTmpDomVif(conn, network);
+            verify(citrixResourceBase, times(1)).cleanUpTmpDomVif(conn, networkMock);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
             fail(e.getMessage());
         }
-        verify(citrixResourceBase, times(1)).destroyTunnelNetwork(conn, network, destroyBridge.getHostId());
+        verify(citrixResourceBase, times(1)).destroyTunnelNetwork(conn, networkMock, destroyBridge.getHostId());
 
         assertTrue(answer.getResult());
     }
