--- a/plugins/hypervisors/kvm/src/test/java/org/apache/cloudstack/utils/linux/KVMHostInfoTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/org/apache/cloudstack/utils/linux/KVMHostInfoTest.java
@@ -19,6 +19,7 @@
 import org.apache.commons.lang.SystemUtils;
 import org.junit.Assert;
 import org.junit.Assume;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.libvirt.Connect;
@@ -31,15 +32,21 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KVMHostInfoTest {
+    private NodeInfo mockNodeInfo;
+
+    @Before
+    public void setUp() {
+        mockNodeInfo = Mockito.mock(NodeInfo.class);
+    }
+
     @Test
     public void getCpuSpeed() {
         if (!System.getProperty("os.name").equals("Linux")) {
             return;
         }
         Assume.assumeTrue(SystemUtils.IS_OS_LINUX);
-        NodeInfo nodeInfo = Mockito.mock(NodeInfo.class);
-        nodeInfo.mhz = 1000;
-        Assert.assertTrue(KVMHostInfo.getCpuSpeed(null, nodeInfo) > 0L);
+        mockNodeInfo.mhz = 1000;
+        Assert.assertTrue(KVMHostInfo.getCpuSpeed(null, mockNodeInfo) > 0L);
     }
 
     @Test
@@ -63,12 +70,11 @@
         }
         try (MockedStatic<LibvirtConnection> ignored = Mockito.mockStatic(LibvirtConnection.class)) {
             Connect conn = Mockito.mock(Connect.class);
-            NodeInfo nodeInfo = Mockito.mock(NodeInfo.class);
-            nodeInfo.mhz = 1000;
+            mockNodeInfo.mhz = 1000;
             String capabilitiesXml = "<capabilities></capabilities>";
 
             Mockito.when(LibvirtConnection.getConnection()).thenReturn(conn);
-            Mockito.when(conn.nodeInfo()).thenReturn(nodeInfo);
+            Mockito.when(conn.nodeInfo()).thenReturn(mockNodeInfo);
             Mockito.when(conn.getCapabilities()).thenReturn(capabilitiesXml);
             int manualSpeed = 500;
 
@@ -84,12 +90,11 @@
         }
         try (MockedStatic<LibvirtConnection> ignored = Mockito.mockStatic(LibvirtConnection.class)) {
             Connect conn = Mockito.mock(Connect.class);
-            NodeInfo nodeInfo = Mockito.mock(NodeInfo.class);
-            nodeInfo.cpus = 10;
+            mockNodeInfo.cpus = 10;
             String capabilitiesXml = "<capabilities></capabilities>";
 
             Mockito.when(LibvirtConnection.getConnection()).thenReturn(conn);
-            Mockito.when(conn.nodeInfo()).thenReturn(nodeInfo);
+            Mockito.when(conn.nodeInfo()).thenReturn(mockNodeInfo);
             Mockito.when(conn.getCapabilities()).thenReturn(capabilitiesXml);
 
             KVMHostInfo kvmHostInfo = new KVMHostInfo(10, 10, 100, 2);
