--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -702,14 +702,13 @@
         LoadbalancerMember loadbalancerMember1 = mock(LoadbalancerMember.class);
         LoadbalancerMember loadbalancerMember2 = mock(LoadbalancerMember.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
-        ObjectReference<ApiPropertyBase> listerner = mock(ObjectReference.class);
-        ObjectReference<ApiPropertyBase> pool = mock(ObjectReference.class);
-        ObjectReference<ApiPropertyBase> member1 = mock(ObjectReference.class);
-        ObjectReference<ApiPropertyBase> member2 = mock(ObjectReference.class);
+        ObjectReference<ApiPropertyBase> listerner = createMockObjectReference();
+        ObjectReference<ApiPropertyBase> pool = createMockObjectReference();
+        ObjectReference<ApiPropertyBase> member1 = createMockObjectReference();
+        ObjectReference<ApiPropertyBase> member2 = createMockObjectReference();
 
         when(listerner.getUuid()).thenReturn("baf714fa-80a1-454f-9c32-c4d4a6f5c5a4");
         when(pool.getUuid()).thenReturn("6b062909-ba9d-4cf3-bbd3-7db93cf6b4fe");
-        when(member1.getUuid()).thenReturn("7d5575eb-d029-467e-8b78-6056a8c94a71");
         when(member2.getUuid()).thenReturn("88729834-3ebd-413a-adf9-40aff73cf638");
         when(loadbalancer.getLoadbalancerListenerBackRefs()).thenReturn(List.of(listerner));
         when(loadbalancerListener.getLoadbalancerPoolBackRefs()).thenReturn(List.of(pool));
@@ -1461,4 +1460,10 @@
 
         return invocation -> list;
     }
+
+    private static ObjectReference<ApiPropertyBase> createMockObjectReference() {
+        ObjectReference<ApiPropertyBase> objectReference = mock(ObjectReference.class);
+        when(objectReference.getUuid()).thenReturn("7d5575eb-d029-467e-8b78-6056a8c94a71");
+        return objectReference;
+    }
 }
