--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirTakeKbossBackupCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirTakeKbossBackupCommandWrapperTest.java
@@ -368,10 +368,16 @@
         verify(libvirtComputingResourceMock).mergeDeltaForStoppedVm(deltaMergeTreeToMock);
     }
 
+    private static VolumeObjectTO createMockVolumeObjectTO(String uuid) {
+        VolumeObjectTO volumeObjectToMock = Mockito.mock(VolumeObjectTO.class);
+        doReturn(uuid).when(volumeObjectToMock).getUuid();
+        return volumeObjectToMock;
+    }
+
     private void setupKbossTos() {
+        volumeObjectToMock1 = createMockVolumeObjectTO(volUuid1);
         doReturn(volumeObjectToMock1).when(kbossTO1).getVolumeObjectTO();
-        doReturn(volUuid1).when(volumeObjectToMock1).getUuid();
+        volumeObjectToMock2 = createMockVolumeObjectTO(volUuid2);
         doReturn(volumeObjectToMock2).when(kbossTO2).getVolumeObjectTO();
-        doReturn(volUuid2).when(volumeObjectToMock2).getUuid();
     }
 }
