--- a/server/src/test/java/com/cloud/acl/DomainCheckerTest.java
+++ b/server/src/test/java/com/cloud/acl/DomainCheckerTest.java
@@ -53,6 +53,8 @@
     ProjectManager _projectMgr;
     @Mock
     BackupOfferingDetailsDao backupOfferingDetailsDao;
+    @Mock
+    private BackupOfferingVO mockedBackupOfferingVO;
 
     @Spy
     @InjectMocks
@@ -189,10 +191,9 @@
     public void testBackupOfferingAccessRootAdmin() {
         Account rootAdmin = Mockito.mock(Account.class);
         Mockito.when(rootAdmin.getId()).thenReturn(1L);
-        BackupOfferingVO backupOfferingVO = Mockito.mock(BackupOfferingVO.class);
         Mockito.when(_accountService.isRootAdmin(rootAdmin.getId())).thenReturn(true);
 
-        boolean hasAccess = domainChecker.checkAccess(rootAdmin, backupOfferingVO);
+        boolean hasAccess = domainChecker.checkAccess(rootAdmin, mockedBackupOfferingVO);
         Assert.assertTrue(hasAccess);
     }
 
@@ -200,14 +201,13 @@
     public void testBackupOfferingAccessDomainAdmin() {
         Account domainAdmin = Mockito.mock(Account.class);
         Mockito.when(domainAdmin.getId()).thenReturn(2L);
-        BackupOfferingVO backupOfferingVO = Mockito.mock(BackupOfferingVO.class);
         AccountVO owner = Mockito.mock(AccountVO.class);
         Mockito.when(_accountService.isDomainAdmin(domainAdmin.getId())).thenReturn(true);
         Mockito.when(domainAdmin.getDomainId()).thenReturn(10L);
         Mockito.when(_domainDao.isChildDomain(100L, 10L)).thenReturn(true);
-        Mockito.when(backupOfferingDetailsDao.findDomainIds(backupOfferingVO.getId())).thenReturn(Collections.singletonList(100L));
-
-        boolean hasAccess = domainChecker.checkAccess(domainAdmin, backupOfferingVO);
+        Mockito.when(backupOfferingDetailsDao.findDomainIds(mockedBackupOfferingVO.getId())).thenReturn(Collections.singletonList(100L));
+
+        boolean hasAccess = domainChecker.checkAccess(domainAdmin, mockedBackupOfferingVO);
         Assert.assertTrue(hasAccess);
     }
 
@@ -215,11 +215,10 @@
     public void testBackupOfferingAccessNoAccess() {
         Account normalUser = Mockito.mock(Account.class);
         Mockito.when(normalUser.getId()).thenReturn(3L);
-        BackupOfferingVO backupOfferingVO = Mockito.mock(BackupOfferingVO.class);
         Mockito.when(_accountService.isRootAdmin(normalUser.getId())).thenReturn(false);
         Mockito.when(_accountService.isDomainAdmin(normalUser.getId())).thenReturn(false);
 
-        boolean hasAccess = domainChecker.checkAccess(normalUser, backupOfferingVO);
+        boolean hasAccess = domainChecker.checkAccess(normalUser, mockedBackupOfferingVO);
         Assert.assertFalse(hasAccess);
     }
 
