--- a/core/src/test/java/org/apache/cloudstack/storage/command/QuerySnapshotZoneCopyAnswerTest.java
+++ b/core/src/test/java/org/apache/cloudstack/storage/command/QuerySnapshotZoneCopyAnswerTest.java
@@ -19,16 +19,23 @@
 import java.util.List;
 
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
 public class QuerySnapshotZoneCopyAnswerTest {
 
+    private QuerySnapshotZoneCopyCommand querySnapshotZoneCopyCommand;
+
+    @Before
+    public void setUp() {
+        querySnapshotZoneCopyCommand = Mockito.mock(QuerySnapshotZoneCopyCommand.class);
+    }
+
     @Test
     public void testQuerySnapshotZoneCopyAnswerSuccess() {
-        QuerySnapshotZoneCopyCommand cmd = Mockito.mock(QuerySnapshotZoneCopyCommand.class);
         List<String> files = List.of("File1", "File2");
-        QuerySnapshotZoneCopyAnswer answer = new QuerySnapshotZoneCopyAnswer(cmd, files);
+        QuerySnapshotZoneCopyAnswer answer = new QuerySnapshotZoneCopyAnswer(querySnapshotZoneCopyCommand, files);
         Assert.assertTrue(answer.getResult());
         Assert.assertEquals(files.size(), answer.getFiles().size());
         Assert.assertEquals(files.get(0), answer.getFiles().get(0));
@@ -37,9 +44,8 @@
 
     @Test
     public void testQuerySnapshotZoneCopyAnswerFailure() {
-        QuerySnapshotZoneCopyCommand cmd = Mockito.mock(QuerySnapshotZoneCopyCommand.class);
         String err = "SOMEERROR";
-        QuerySnapshotZoneCopyAnswer answer = new QuerySnapshotZoneCopyAnswer(cmd, err);
+        QuerySnapshotZoneCopyAnswer answer = new QuerySnapshotZoneCopyAnswer(querySnapshotZoneCopyCommand, err);
         Assert.assertFalse(answer.getResult());
         Assert.assertEquals(err, answer.getDetails());
     }
