--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -420,14 +420,13 @@
     @Test
     public void testTemplateScheduledForDownloadInOnePool() {
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
-        StoragePoolVO mockPool = mock(StoragePoolVO.class);
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
         VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
 
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
+        StoragePoolVO mockPool = createMockStoragePool(1l);
         when(mockPool.getId()).thenReturn(2l);
         when(mockPool.getStatus()).thenReturn(StoragePoolStatus.Up);
-        when(mockPool.getDataCenterId()).thenReturn(1l);
         when(mockTemplate.getId()).thenReturn(202l);
         when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
         when(vmTemplateDao.findById(anyLong())).thenReturn(mockTemplate);
@@ -472,13 +471,10 @@
         VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
         List<StoragePoolVO> pools = new ArrayList<StoragePoolVO>();
 
-        StoragePoolVO mockPool1 = mock(StoragePoolVO.class);
+        StoragePoolVO mockPool1 = createMockStoragePool(1l);
         when(mockPool1.getId()).thenReturn(2l);
-        when(mockPool1.getDataCenterId()).thenReturn(1l);
-        StoragePoolVO mockPool2 = mock(StoragePoolVO.class);
-        when(mockPool2.getDataCenterId()).thenReturn(1l);
-        StoragePoolVO mockPool3 = mock(StoragePoolVO.class);
-        when(mockPool3.getDataCenterId()).thenReturn(2l);
+        StoragePoolVO mockPool2 = createMockStoragePool(1l);
+        StoragePoolVO mockPool3 = createMockStoragePool(2l);
         pools.add(mockPool1);
         pools.add(mockPool2);
         pools.add(mockPool3);
@@ -669,6 +665,12 @@
         Mockito.verify(dataStoreManager, Mockito.times(0)).getImageStoreWithFreeCapacity(Mockito.anyLong());
     }
 
+    private static StoragePoolVO createMockStoragePool(long dataCenterId) {
+        StoragePoolVO mockPool = mock(StoragePoolVO.class);
+        when(mockPool.getDataCenterId()).thenReturn(dataCenterId);
+        return mockPool;
+    }
+
     private HypervisorTemplateAdapter mockHypervisorTemplateAdapter() {
         HypervisorTemplateAdapter adapter = Mockito.mock(HypervisorTemplateAdapter.class);
         when(adapter.getName()).thenReturn(TemplateAdapter.TemplateAdapterType.Hypervisor.getName());
