--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -890,13 +890,12 @@
 
     @Test
     public void testUpdateGuestOsCategory() {
-        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = new GuestOSCategoryVO("Old name", false);
         long id = 1L;
         String name = "Updated Name";
         Boolean featured = true;
         Integer sortKey = 10;
-        Mockito.when(updateCmd.getId()).thenReturn(id);
+        UpdateGuestOsCategoryCmd updateCmd = createMockUpdateGuestOsCategoryCmd();
         Mockito.when(updateCmd.getName()).thenReturn(name);
         Mockito.when(updateCmd.isFeatured()).thenReturn(featured);
         Mockito.when(updateCmd.getSortKey()).thenReturn(sortKey);
@@ -912,9 +911,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testUpdateGuestOsCategory_ThrowsExceptionWhenCategoryNotFound() {
-        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
         long id = 1L;
-        when(updateCmd.getId()).thenReturn(id);
+        UpdateGuestOsCategoryCmd updateCmd = createMockUpdateGuestOsCategoryCmd();
         when(guestOSCategoryDao.findById(id)).thenReturn(null);
         spy.updateGuestOsCategory(updateCmd);
     }
@@ -991,6 +989,12 @@
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList(1L));
         spy.deleteGuestOsCategory(deleteCmd);
+    }
+
+    private static UpdateGuestOsCategoryCmd createMockUpdateGuestOsCategoryCmd() {
+        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
+        Mockito.when(updateCmd.getId()).thenReturn(1L);
+        return updateCmd;
     }
 
     private void mockGuestOsJoin() {
