--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -171,6 +171,9 @@
     @Mock
     RegisterUserKeysCmd registerCmdMock;
 
+    @Mock
+    private UserAuthenticator userAuthenticatorMock1;
+
     @Before
     public void setUp() throws Exception {
         enableUserTwoFactorAuthenticationMock = Mockito.mock(ConfigKey.class);
@@ -1057,10 +1060,9 @@
 
         accountManagerImpl._userPasswordEncoders = new ArrayList<>();
         UserAuthenticator authenticatorMock1 = Mockito.mock(UserAuthenticator.class);
-        UserAuthenticator authenticatorMock2 = Mockito.mock(UserAuthenticator.class);
 
         accountManagerImpl._userPasswordEncoders.add(authenticatorMock1);
-        accountManagerImpl._userPasswordEncoders.add(authenticatorMock2);
+        accountManagerImpl._userPasswordEncoders.add(userAuthenticatorMock1);
 
         Pair<Boolean, ActionOnFailedAuthentication> authenticationResult = new Pair<Boolean, UserAuthenticator.ActionOnFailedAuthentication>(true,
                 UserAuthenticator.ActionOnFailedAuthentication.INCREMENT_INCORRECT_LOGIN_ATTEMPT_COUNT);
@@ -1071,7 +1073,7 @@
         accountManagerImpl.validateCurrentPassword(userVoMock, currentPassword);
 
         Mockito.verify(authenticatorMock1, Mockito.times(1)).authenticate(username, currentPassword, domainId, null);
-        Mockito.verify(authenticatorMock2, Mockito.times(0)).authenticate(username, currentPassword, domainId, null);
+        Mockito.verify(userAuthenticatorMock1, Mockito.times(0)).authenticate(username, currentPassword, domainId, null);
     }
 
     @Test
@@ -1085,10 +1087,9 @@
         Mockito.doReturn(username).when(userVoMock).getUsername();
 
         accountManagerImpl._userPasswordEncoders = new ArrayList<>();
-        UserAuthenticator authenticatorMock1 = Mockito.mock(UserAuthenticator.class);
         UserAuthenticator authenticatorMock2 = Mockito.mock(UserAuthenticator.class);
 
-        accountManagerImpl._userPasswordEncoders.add(authenticatorMock1);
+        accountManagerImpl._userPasswordEncoders.add(userAuthenticatorMock1);
         accountManagerImpl._userPasswordEncoders.add(authenticatorMock2);
 
         Pair<Boolean, ActionOnFailedAuthentication> authenticationResult = new Pair<Boolean, UserAuthenticator.ActionOnFailedAuthentication>(true,
@@ -1099,7 +1100,7 @@
 
         accountManagerImpl.validateCurrentPassword(userVoMock, currentPassword);
 
-        Mockito.verify(authenticatorMock1, Mockito.times(1)).authenticate(username, currentPassword, domainId, null);
+        Mockito.verify(userAuthenticatorMock1, Mockito.times(1)).authenticate(username, currentPassword, domainId, null);
         Mockito.verify(authenticatorMock2, Mockito.times(1)).authenticate(username, currentPassword, domainId, null);
     }
 
