--- a/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
@@ -271,11 +271,10 @@
         int startIndex = 0;
         int pageSize = 10;
 
-        EndPoint ep = Mockito.mock(EndPoint.class);
+        ListDataStoreObjectsAnswer answer = Mockito.mock(ListDataStoreObjectsAnswer.class);
+        EndPoint ep = createMockEndPoint(answer);
 
         Mockito.when(endPointSelector.select(dataStore)).thenReturn(ep);
-        ListDataStoreObjectsAnswer answer = Mockito.mock(ListDataStoreObjectsAnswer.class);
-        Mockito.when(ep.sendMessage(Mockito.any())).thenReturn(answer);
         Mockito.when(answer.getResult()).thenReturn(true);
         Mockito.when(answer.isPathExists()).thenReturn(false);
         Mockito.when(dataStore.getUuid()).thenReturn("uuid");
@@ -294,11 +293,10 @@
         int startIndex = 0;
         int pageSize = 10;
 
-        EndPoint ep = Mockito.mock(EndPoint.class);
+        ListDataStoreObjectsAnswer answer = Mockito.mock(ListDataStoreObjectsAnswer.class);
+        EndPoint ep = createMockEndPoint(answer);
 
         Mockito.when(endPointSelector.select(dataStore)).thenReturn(ep);
-        ListDataStoreObjectsAnswer answer = Mockito.mock(ListDataStoreObjectsAnswer.class);
-        Mockito.when(ep.sendMessage(Mockito.any())).thenReturn(answer);
         Mockito.when(answer.getResult()).thenReturn(true);
         Mockito.when(answer.isPathExists()).thenReturn(true);
 
@@ -456,4 +454,10 @@
         Assert.assertEquals(200L, response.getResponses().get(1).getSize());
         Assert.assertNotNull(response.getResponses().get(1).getLastUpdated());
     }
+
+    private static EndPoint createMockEndPoint(ListDataStoreObjectsAnswer answer) {
+        EndPoint ep = Mockito.mock(EndPoint.class);
+        Mockito.when(ep.sendMessage(Mockito.any())).thenReturn(answer);
+        return ep;
+    }
 }
