--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -149,6 +149,12 @@
         Assert.assertEquals(webhookJoinVO.getScope().toString(), ReflectionTestUtils.getField(response, "scope"));
     }
 
+    private static WebhookJoinVO createMockWebhookJoinVO() {
+        WebhookJoinVO webhookJoinVO = Mockito.mock(WebhookJoinVO.class);
+        Mockito.when(webhookJoinVO.getId()).thenReturn(1L);
+        return webhookJoinVO;
+    }
+
     @Test
     public void testCreateWebhookResponse() {
         WebhookJoinVO webhookJoinVO = prepareTestWebhookJoinVO();
@@ -186,8 +192,7 @@
         DomainVO domain = Mockito.mock(DomainVO.class);
         Mockito.when(domain.getPath()).thenReturn(domainPath);
         Mockito.when(domainDao.findById(1L)).thenReturn(domain);
-        WebhookJoinVO webhookJoinVO = Mockito.mock(WebhookJoinVO.class);
-        Mockito.when(webhookJoinVO.getId()).thenReturn(1L);
+        WebhookJoinVO webhookJoinVO = createMockWebhookJoinVO();
         Mockito.when(webhookJoinDao.listByAccountOrDomain(accountId, domainPath)).thenReturn(List.of(webhookJoinVO));
         List<Long> result = webhookApiServiceImpl.getIdsOfAccessibleWebhooks(account, false);
         Assert.assertTrue(CollectionUtils.isNotEmpty(result));
@@ -200,8 +205,7 @@
         Account account = Mockito.mock(Account.class);
         Mockito.when(accountManager.isDomainAdmin(accountId)).thenReturn(false);
         Mockito.when(account.getId()).thenReturn(accountId);
-        WebhookJoinVO webhookJoinVO = Mockito.mock(WebhookJoinVO.class);
-        Mockito.when(webhookJoinVO.getId()).thenReturn(1L);
+        WebhookJoinVO webhookJoinVO = createMockWebhookJoinVO();
         Mockito.when(webhookJoinDao.listByAccountOrDomain(accountId, null)).thenReturn(List.of(webhookJoinVO));
         List<Long> result = webhookApiServiceImpl.getIdsOfAccessibleWebhooks(account, false);
         Assert.assertTrue(CollectionUtils.isNotEmpty(result));
