--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -267,6 +267,12 @@
         return webhook;
     }
 
+    private static ExecuteWebhookDeliveryCmd createMockExecuteWebhookDeliveryCmd() {
+        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
+        Mockito.when(cmd.getWebhookId()).thenReturn(1L);
+        return cmd;
+    }
+
     @Test
     public void testValidateWebhookOwnerPayloadUrlUpdateSameExist() {
         WebhookVO webhookVO = mockWebhookVO(1L);
@@ -1238,13 +1244,12 @@
 
     @Test
     public void executeWebhookDeliveryExecutesSuccessfullyForValidWebhookId() {
-        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
+        ExecuteWebhookDeliveryCmd cmd = createMockExecuteWebhookDeliveryCmd();
         WebhookVO webhook = Mockito.mock(WebhookVO.class);
         WebhookDelivery webhookDelivery = Mockito.mock(WebhookDelivery.class);
         WebhookDeliveryResponse response = Mockito.mock(WebhookDeliveryResponse.class);
 
         Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.getWebhookId()).thenReturn(1L);
         Mockito.when(webhookDao.findById(1L)).thenReturn(webhook);
         Mockito.when(webhookService.executeWebhookDelivery(null, webhook, null, null)).thenReturn(webhookDelivery);
         Mockito.when(webhookDelivery.getId()).thenReturn(WebhookDelivery.ID_DUMMY);
