--- a/server/src/test/java/org/apache/cloudstack/dns/DnsProviderManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/dns/DnsProviderManagerImplTest.java
@@ -682,11 +682,16 @@
         manager.associateZoneToNetwork(cmd);
     }
 
-    @Test(expected = InvalidParameterValueException.class)
-    public void testAssociateZoneToNetworkNetworkNotFound() {
+    private static AssociateDnsZoneToNetworkCmd createMockAssociateDnsZoneToNetworkCmd() {
         AssociateDnsZoneToNetworkCmd cmd = mock(AssociateDnsZoneToNetworkCmd.class);
         when(cmd.getDnsZoneId()).thenReturn(ZONE_ID);
         when(cmd.getNetworkId()).thenReturn(NETWORK_ID);
+        return cmd;
+    }
+
+    @Test(expected = InvalidParameterValueException.class)
+    public void testAssociateZoneToNetworkNetworkNotFound() {
+        AssociateDnsZoneToNetworkCmd cmd = createMockAssociateDnsZoneToNetworkCmd();
         when(dnsZoneDao.findById(ZONE_ID)).thenReturn(zoneVO);
         when(dnsServerDao.findById(SERVER_ID)).thenReturn(serverVO);
         when(networkDao.findById(NETWORK_ID)).thenReturn(null);
@@ -695,9 +700,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testAssociateZoneToNetworkNonSharedNetwork() {
-        AssociateDnsZoneToNetworkCmd cmd = mock(AssociateDnsZoneToNetworkCmd.class);
-        when(cmd.getDnsZoneId()).thenReturn(ZONE_ID);
-        when(cmd.getNetworkId()).thenReturn(NETWORK_ID);
+        AssociateDnsZoneToNetworkCmd cmd = createMockAssociateDnsZoneToNetworkCmd();
         when(dnsZoneDao.findById(ZONE_ID)).thenReturn(zoneVO);
         when(dnsServerDao.findById(SERVER_ID)).thenReturn(serverVO);
         NetworkVO network = mock(NetworkVO.class);
@@ -708,9 +711,7 @@
 
     @Test
     public void testAssociateZoneToNetworkSuccess() {
-        AssociateDnsZoneToNetworkCmd cmd = mock(AssociateDnsZoneToNetworkCmd.class);
-        when(cmd.getDnsZoneId()).thenReturn(ZONE_ID);
-        when(cmd.getNetworkId()).thenReturn(NETWORK_ID);
+        AssociateDnsZoneToNetworkCmd cmd = createMockAssociateDnsZoneToNetworkCmd();
 
         when(dnsZoneDao.findById(ZONE_ID)).thenReturn(zoneVO);
         Mockito.doReturn("zone-uuid").when(zoneVO).getUuid();
@@ -728,9 +729,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testAssociateZoneToNetworkAlreadyAssociated() {
-        AssociateDnsZoneToNetworkCmd cmd = mock(AssociateDnsZoneToNetworkCmd.class);
-        when(cmd.getDnsZoneId()).thenReturn(ZONE_ID);
-        when(cmd.getNetworkId()).thenReturn(NETWORK_ID);
+        AssociateDnsZoneToNetworkCmd cmd = createMockAssociateDnsZoneToNetworkCmd();
         when(dnsZoneDao.findById(ZONE_ID)).thenReturn(zoneVO);
         when(dnsServerDao.findById(SERVER_ID)).thenReturn(serverVO);
         NetworkVO network = mock(NetworkVO.class);
