--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricLogicalRouterCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricLogicalRouterCmdTest.java
@@ -45,9 +45,12 @@
 
     AutoCloseable closeable;
 
+    private BaseResponse baseResponseMock;
+
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        baseResponseMock = Mockito.mock(BaseResponse.class);
         listTungstenFabricLogicalRouterCmd = new ListTungstenFabricLogicalRouterCmd();
         listTungstenFabricLogicalRouterCmd.tungstenService = tungstenService;
         ReflectionTestUtils.setField(listTungstenFabricLogicalRouterCmd, "networkUuid", "test");
@@ -65,8 +68,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricLogicalRouterCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         Mockito.when(tungstenService.listRoutingLogicalRouter(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricLogicalRouterCmd.execute();
@@ -77,8 +79,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
