--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -86,6 +86,8 @@
     @Mock
     WebhookDeliveryJoinDao webhookDeliveryJoinDao;
     @Mock
+    private WebhookDeliveryJoinVO webhookDeliveryJoin;
+    @Mock
     WebhookFilterDao webhookFilterDao;
     @Mock
     ManagementServerHostDao managementServerHostDao;
@@ -1058,7 +1060,6 @@
     @Test
     public void listWebhookDeliveriesFiltersByWebhookId() {
         ListWebhookDeliveriesCmd cmd = Mockito.mock(ListWebhookDeliveriesCmd.class);
-        WebhookDeliveryJoinVO delivery = Mockito.mock(WebhookDeliveryJoinVO.class);
         WebhookDeliveryResponse deliveryResponse = Mockito.mock(WebhookDeliveryResponse.class);
 
         Mockito.when(cmd.getId()).thenReturn(null);
@@ -1069,8 +1070,8 @@
         Mockito.when(webhookDeliveryJoinDao.searchAndCountByListApiParameters(Mockito.any(), Mockito.anyList(),
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                         Mockito.anyBoolean()))
-                .thenReturn(new Pair<>(List.of(delivery), 1));
-        Mockito.doReturn(deliveryResponse).when(webhookApiServiceImpl).createWebhookDeliveryResponse(delivery);
+                .thenReturn(new Pair<>(List.of(webhookDeliveryJoin), 1));
+        Mockito.doReturn(deliveryResponse).when(webhookApiServiceImpl).createWebhookDeliveryResponse(webhookDeliveryJoin);
         Mockito.doReturn(null).when(webhookApiServiceImpl)
                 .basicWebhookDeliveryApiCheck(caller, null, 1L, null, null, null, false);
 
@@ -1095,7 +1096,6 @@
     @Test
     public void listWebhookDeliveriesFiltersByEventType() {
         ListWebhookDeliveriesCmd cmd = Mockito.mock(ListWebhookDeliveriesCmd.class);
-        WebhookDeliveryJoinVO delivery = Mockito.mock(WebhookDeliveryJoinVO.class);
         WebhookDeliveryResponse deliveryResponse = Mockito.mock(WebhookDeliveryResponse.class);
 
         Mockito.when(cmd.getId()).thenReturn(null);
@@ -1107,8 +1107,8 @@
         Mockito.when(webhookDeliveryJoinDao.searchAndCountByListApiParameters(Mockito.any(), Mockito.anyList(),
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                         Mockito.anyBoolean()))
-                .thenReturn(new Pair<>(List.of(delivery), 1));
-        Mockito.doReturn(deliveryResponse).when(webhookApiServiceImpl).createWebhookDeliveryResponse(delivery);
+                .thenReturn(new Pair<>(List.of(webhookDeliveryJoin), 1));
+        Mockito.doReturn(deliveryResponse).when(webhookApiServiceImpl).createWebhookDeliveryResponse(webhookDeliveryJoin);
 
         ListResponse<WebhookDeliveryResponse> response = webhookApiServiceImpl.listWebhookDeliveries(cmd);
 
