--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/resource/NsxResourceTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/resource/NsxResourceTest.java
@@ -75,6 +75,8 @@
     AutoCloseable closeable;
     @Mock
     TransportZoneListResult transportZoneListResult;
+    @Mock
+    private CreateNsxSegmentCommand createNsxSegmentCommand;
 
     private static final String transportZone = "Overlay";
     private static final String tier0Gateway = "Tier0-GW01";
@@ -187,8 +189,7 @@
     @Test
     public void testCreateNsxSegmentEmptySites() {
         when(nsxApi.getDefaultSiteId()).thenReturn(null);
-        CreateNsxSegmentCommand command = Mockito.mock(CreateNsxSegmentCommand.class);
-        NsxAnswer answer = (NsxAnswer) nsxResource.executeRequest(command);
+        NsxAnswer answer = (NsxAnswer) nsxResource.executeRequest(createNsxSegmentCommand);
         assertFalse(answer.getResult());
     }
 
@@ -197,8 +198,7 @@
         Site site = mock(Site.class);
         when(nsxApi.getDefaultSiteId()).thenReturn("site1");
         when(nsxApi.getDefaultEnforcementPointPath(anyString())).thenReturn(null);
-        CreateNsxSegmentCommand command = Mockito.mock(CreateNsxSegmentCommand.class);
-        NsxAnswer answer = (NsxAnswer) nsxResource.executeRequest(command);
+        NsxAnswer answer = (NsxAnswer) nsxResource.executeRequest(createNsxSegmentCommand);
         assertFalse(answer.getResult());
     }
 
@@ -206,8 +206,7 @@
     public void testCreateNsxSegmentEmptyTransportZones() {
         Site site = mock(Site.class);
         when(nsxApi.getDefaultSiteId()).thenReturn("site1");
-        CreateNsxSegmentCommand command = Mockito.mock(CreateNsxSegmentCommand.class);
-        NsxAnswer answer = (NsxAnswer) nsxResource.executeRequest(command);
+        NsxAnswer answer = (NsxAnswer) nsxResource.executeRequest(createNsxSegmentCommand);
         assertFalse(answer.getResult());
     }
 
