--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -165,6 +165,9 @@
     protected ObjectStoreDao objectStoreDao;
 
     @Mock
+    private DataStoreProvider dataStoreProvider;
+
+    @Mock
     DataStoreProviderManager dataStoreProviderMgr;
 
     @Mock
@@ -1676,8 +1679,7 @@
         String providerName = "test-provider";
         Map<String, String> details = new HashMap<>();
 
-        DataStoreProvider storeProvider = Mockito.mock(DataStoreProvider.class);
-        Mockito.when(dataStoreProviderMgr.getDataStoreProvider(providerName)).thenReturn(storeProvider);
+        Mockito.when(dataStoreProviderMgr.getDataStoreProvider(providerName)).thenReturn(dataStoreProvider);
 
         storageManagerImpl.discoverObjectStore(name, url, size, providerName, details);
     }
@@ -1690,10 +1692,9 @@
         String providerName = "test-provider";
         Map<String, String> details = new HashMap<>();
 
-        DataStoreProvider storeProvider = Mockito.mock(DataStoreProvider.class);
         ObjectStoreVO existingStore = new ObjectStoreVO();
 
-        Mockito.when(dataStoreProviderMgr.getDataStoreProvider(providerName)).thenReturn(storeProvider);
+        Mockito.when(dataStoreProviderMgr.getDataStoreProvider(providerName)).thenReturn(dataStoreProvider);
         Mockito.when(objectStoreDao.findByUrl(url)).thenReturn(existingStore);
 
         storageManagerImpl.discoverObjectStore(name, url, size, providerName, details);
