--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/cluster/ListClustersCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/cluster/ListClustersCmdTest.java
@@ -62,12 +62,8 @@
 
     @Test
     public void updateClustersExtensions_withClusterResponses_setsExtension() {
-        ClusterResponse cluster1 = mock(ClusterResponse.class);
-        ClusterResponse cluster2 = mock(ClusterResponse.class);
-        when(cluster1.getInternalId()).thenReturn(1L);
-        when(cluster1.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External.name());
-        when(cluster2.getInternalId()).thenReturn(2L);
-        when(cluster2.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External.name());
+        ClusterResponse cluster1 = createMockClusterResponse(1L);
+        ClusterResponse cluster2 = createMockClusterResponse(2L);
         Extension ext1 = mock(Extension.class);
         when(ext1.getUuid()).thenReturn("a");
         Extension ext2 = mock(Extension.class);
@@ -80,4 +76,11 @@
         verify(cluster1).setExtensionId("a");
         verify(cluster2).setExtensionId("b");
     }
+
+    private static ClusterResponse createMockClusterResponse(long internalId) {
+        ClusterResponse clusterResponse = mock(ClusterResponse.class);
+        when(clusterResponse.getInternalId()).thenReturn(internalId);
+        when(clusterResponse.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External.name());
+        return clusterResponse;
+    }
 }
