--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -838,8 +838,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testSearchForConfigurationsMultipleIds() {
-        ListCfgsByCmd cmd = Mockito.mock(ListCfgsByCmd.class);
-        Mockito.when(cmd.getConfigName()).thenReturn("pool.storage.capacity.disablethreshold");
+        ListCfgsByCmd cmd = createMockListCfgsByCmd();
         Mockito.when(cmd.getZoneId()).thenReturn(1L);
         Mockito.when(cmd.getStoragepoolId()).thenReturn(2L);
         spy.searchForConfigurations(cmd);
@@ -848,8 +847,7 @@
     @Test
     public void testSearchForConfigurations() {
         Long poolId = 1L;
-        ListCfgsByCmd cmd = Mockito.mock(ListCfgsByCmd.class);
-        Mockito.when(cmd.getConfigName()).thenReturn("pool.storage.capacity.disablethreshold");
+        ListCfgsByCmd cmd = createMockListCfgsByCmd();
         Mockito.when(cmd.getStoragepoolId()).thenReturn(poolId);
         Mockito.when(cmd.getZoneId()).thenReturn(null);
         Mockito.when(cmd.getClusterId()).thenReturn(null);
@@ -2493,4 +2491,10 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static ListCfgsByCmd createMockListCfgsByCmd() {
+        ListCfgsByCmd cmd = Mockito.mock(ListCfgsByCmd.class);
+        Mockito.when(cmd.getConfigName()).thenReturn("pool.storage.capacity.disablethreshold");
+        return cmd;
+    }
 }
