--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -1082,6 +1082,13 @@
         return cluster;
     }
 
+    private static StoragePoolVO createMockStoragePool(long clusterId) {
+        StoragePoolVO pool = mock(StoragePoolVO.class);
+        when(pool.getScope()).thenReturn(ScopeType.CLUSTER);
+        when(pool.getClusterId()).thenReturn(clusterId);
+        return pool;
+    }
+
     @Test(expected = CloudRuntimeException.class)
     public void testSelectInstanceConversionTemporaryLocationInvalidStorage() {
         ClusterVO cluster = getClusterForTests();
@@ -1095,9 +1102,7 @@
     public void testSelectInstanceConversionTemporaryLocationPoolInvalidScope() {
         ClusterVO cluster = getClusterForTests();
         long poolId = 1L;
-        StoragePoolVO pool = mock(StoragePoolVO.class);
-        when(pool.getScope()).thenReturn(ScopeType.CLUSTER);
-        when(pool.getClusterId()).thenReturn(100L);
+        StoragePoolVO pool = createMockStoragePool(100L);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
         unmanagedVMsManager.selectInstanceConversionTemporaryLocation(cluster, null, null, poolId, false);
     }
@@ -1107,9 +1112,7 @@
     public void testSelectInstanceConversionTemporaryLocationPoolConvertHostDifferentCluster() {
         ClusterVO cluster = getClusterForTests();
         long poolId = 1L;
-        StoragePoolVO pool = mock(StoragePoolVO.class);
-        when(pool.getScope()).thenReturn(ScopeType.CLUSTER);
-        when(pool.getClusterId()).thenReturn(1L);
+        StoragePoolVO pool = createMockStoragePool(1L);
         HostVO host = mock(HostVO.class);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
         when(host.getClusterId()).thenReturn(2L);
@@ -1131,9 +1134,7 @@
     public void testSelectInstanceConversionTemporaryLocationStoragePoolInvalidType() {
         ClusterVO cluster = getClusterForTests();
         long poolId = 1L;
-        StoragePoolVO pool = mock(StoragePoolVO.class);
-        when(pool.getScope()).thenReturn(ScopeType.CLUSTER);
-        when(pool.getClusterId()).thenReturn(1L);
+        StoragePoolVO pool = createMockStoragePool(1L);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
         when(pool.getPoolType()).thenReturn(Storage.StoragePoolType.RBD);
         unmanagedVMsManager.selectInstanceConversionTemporaryLocation(cluster, null, null, poolId, false);
