--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricFirewallPolicyCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricFirewallPolicyCmdTest.java
@@ -41,6 +41,9 @@
     @Mock
     TungstenService tungstenService;
 
+    @Mock
+    private BaseResponse baseResponseMock;
+
     ListTungstenFabricFirewallPolicyCmd listTungstenFabricFirewallPolicyCmd;
 
     AutoCloseable closeable;
@@ -65,8 +68,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricFirewallPolicyCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         Mockito.when(tungstenService.listTungstenFirewallPolicy(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricFirewallPolicyCmd.execute();
@@ -77,8 +79,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
