--- a/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
+++ b/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
@@ -40,15 +40,14 @@
 
     @Test
     public void testSortSnapshotStrategies() {
-        SnapshotStrategy cantHandleStrategy = mock(SnapshotStrategy.class);
-        SnapshotStrategy defaultStrategy = mock(SnapshotStrategy.class);
-        SnapshotStrategy hyperStrategy = mock(SnapshotStrategy.class);
-        SnapshotStrategy highestStrategy = mock(SnapshotStrategy.class);
+        SnapshotStrategy cantHandleStrategy = createMockHighestSnapshotStrategy();
+        SnapshotStrategy defaultStrategy = createMockHighestSnapshotStrategy();
+        SnapshotStrategy hyperStrategy = createMockHighestSnapshotStrategy();
+        SnapshotStrategy highestStrategy = createMockHighestSnapshotStrategy();
 
         doReturn(StrategyPriority.CANT_HANDLE).when(cantHandleStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
         doReturn(StrategyPriority.DEFAULT).when(defaultStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
         doReturn(StrategyPriority.HYPERVISOR).when(hyperStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
-        doReturn(StrategyPriority.HIGHEST).when(highestStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
 
         List<SnapshotStrategy> strategies = new ArrayList<>(5);
         SnapshotStrategy strategy = null;
@@ -143,4 +142,10 @@
         strategy = factory.getDataMotionStrategy(mock(Map.class), mock(Host.class), mock(Host.class));
         assertEquals("Highest strategy was not picked.", highestStrategy, strategy);
     }
+
+    private static SnapshotStrategy createMockHighestSnapshotStrategy() {
+        SnapshotStrategy highestStrategy = mock(SnapshotStrategy.class);
+        doReturn(StrategyPriority.HIGHEST).when(highestStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
+        return highestStrategy;
+    }
 }
