--- a/server/src/test/java/org/apache/cloudstack/schedule/vm/VMScheduleWorkerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/schedule/vm/VMScheduleWorkerTest.java
@@ -134,6 +134,13 @@
         Assert.assertEquals(1L, (long) jobId);
     }
 
+    private static ResourceScheduledJobVO createMockResourceScheduledJob() {
+        ResourceScheduledJobVO job = Mockito.mock(ResourceScheduledJobVO.class);
+        Mockito.when(job.getResourceId()).thenReturn(1L);
+        Mockito.when(job.getActionName()).thenReturn(VMScheduleAction.START.name());
+        return job;
+    }
+
     @Test
     public void testProcessJobInvalidAction() {
         ResourceScheduledJobVO job = Mockito.mock(ResourceScheduledJobVO.class);
@@ -151,11 +158,9 @@
 
     @Test
     public void testProcessJobVMInInvalidState() {
-        ResourceScheduledJobVO job = Mockito.mock(ResourceScheduledJobVO.class);
-        UserVm vm = Mockito.mock(UserVm.class);
-
-        Mockito.when(job.getResourceId()).thenReturn(1L);
-        Mockito.when(job.getActionName()).thenReturn(VMScheduleAction.START.name());
+        ResourceScheduledJobVO job = createMockResourceScheduledJob();
+        UserVm vm = Mockito.mock(UserVm.class);
+
         Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Unknown);
         Mockito.when(userVmManager.getUserVm(1L)).thenReturn(vm);
 
