--- a/api/src/test/java/org/apache/cloudstack/api/command/test/AddIpToVmNicTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/AddIpToVmNicTest.java
@@ -41,6 +41,7 @@
 public class AddIpToVmNicTest extends TestCase {
 
     private AddIpToVmNicCmd addIpToVmNicCmd;
+    private AddIpToVmNicCmd mockedAddIpToVmNicCmd;
     private RemoveIpFromVmNicCmd removeIpFromVmNicCmd;
     private ResponseGenerator responseGenerator;
     private SuccessResponse successResponseGenerator;
@@ -49,43 +50,42 @@
     @Before
     public void setUp() {
 
+        mockedAddIpToVmNicCmd = Mockito.mock(AddIpToVmNicCmd.class);
     }
 
     @Test
     public void testCreateSuccess() throws ResourceAllocationException, ResourceUnavailableException, ConcurrentOperationException, InsufficientCapacityException {
 
         NetworkService networkService = Mockito.mock(NetworkService.class);
-        AddIpToVmNicCmd ipTonicCmd = Mockito.mock(AddIpToVmNicCmd.class);
         NicSecondaryIp secIp = Mockito.mock(NicSecondaryIp.class);
 
         Mockito.when(
             networkService.allocateSecondaryGuestIP(ArgumentMatchers.anyLong(), ArgumentMatchers.any(), ArgumentMatchers.anyString()))
             .thenReturn(secIp);
 
-        ipTonicCmd._networkService = networkService;
+        mockedAddIpToVmNicCmd._networkService = networkService;
         responseGenerator = Mockito.mock(ResponseGenerator.class);
 
         NicSecondaryIpResponse ipres = Mockito.mock(NicSecondaryIpResponse.class);
         Mockito.when(responseGenerator.createSecondaryIPToNicResponse(secIp)).thenReturn(ipres);
 
-        ipTonicCmd._responseGenerator = responseGenerator;
-        ipTonicCmd.execute();
+        mockedAddIpToVmNicCmd._responseGenerator = responseGenerator;
+        mockedAddIpToVmNicCmd.execute();
     }
 
     @Test
     public void testCreateFailure() throws ResourceAllocationException, ResourceUnavailableException, ConcurrentOperationException, InsufficientCapacityException {
 
         NetworkService networkService = Mockito.mock(NetworkService.class);
-        AddIpToVmNicCmd ipTonicCmd = Mockito.mock(AddIpToVmNicCmd.class);
 
         Mockito.when(
             networkService.allocateSecondaryGuestIP(ArgumentMatchers.anyLong(), ArgumentMatchers.any(), ArgumentMatchers.anyString()))
             .thenReturn(null);
 
-        ipTonicCmd._networkService = networkService;
+        mockedAddIpToVmNicCmd._networkService = networkService;
 
         try {
-            ipTonicCmd.execute();
+            mockedAddIpToVmNicCmd.execute();
         } catch (InsufficientAddressCapacityException e) {
             throw new InvalidParameterValueException("Allocating guest ip for nic failed");
         }
