--- a/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
+++ b/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
@@ -366,6 +366,14 @@
         return vpcOffering;
     }
 
+    private static StoragePoolVO createMockStoragePoolVo(String uuid, String name, ScopeType scope) {
+        StoragePoolVO storagePoolVoMock = Mockito.mock(StoragePoolVO.class);
+        Mockito.doReturn(uuid).when(storagePoolVoMock).getUuid();
+        Mockito.doReturn(name).when(storagePoolVoMock).getName();
+        Mockito.doReturn(scope).when(storagePoolVoMock).getScope();
+        return storagePoolVoMock;
+    }
+
     private void mockMethodValidateIfObjectIsNull() {
         Mockito.doNothing().when(presetVariableHelperSpy).validateIfObjectIsNull(Mockito.any(), Mockito.anyLong(), Mockito.anyString());
     }
@@ -886,12 +894,9 @@
 
         List<StoragePoolTagVO> storageTagVOListMock = getStorageTagsForTests();
 
-        StoragePoolVO storagePoolVoMock = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePoolVoMock = createMockStoragePoolVo(expected.getId(), expected.getName(), ScopeType.validateAndGetScopeType(expected.getScope()));
         Mockito.doReturn(storagePoolVoMock).when(primaryStorageDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
 
-        Mockito.doReturn(expected.getId()).when(storagePoolVoMock).getUuid();
-        Mockito.doReturn(expected.getName()).when(storagePoolVoMock).getName();
-        Mockito.doReturn(ScopeType.validateAndGetScopeType(expected.getScope())).when(storagePoolVoMock).getScope();
         Mockito.doReturn(storageTagVOListMock).when(storagePoolTagsDaoMock).findStoragePoolTags(Mockito.anyLong());
 
         Storage result = presetVariableHelperSpy.getPresetVariableValueStorage(1L, 2);
@@ -908,12 +913,9 @@
 
         List<StoragePoolTagVO> storageTagVOListMock = getStorageRuleTagsForTests();
 
-        StoragePoolVO storagePoolVoMock = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePoolVoMock = createMockStoragePoolVo(expected.getId(), expected.getName(), ScopeType.validateAndGetScopeType(expected.getScope()));
         Mockito.doReturn(storagePoolVoMock).when(primaryStorageDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
 
-        Mockito.doReturn(expected.getId()).when(storagePoolVoMock).getUuid();
-        Mockito.doReturn(expected.getName()).when(storagePoolVoMock).getName();
-        Mockito.doReturn(ScopeType.validateAndGetScopeType(expected.getScope())).when(storagePoolVoMock).getScope();
         Mockito.doReturn(storageTagVOListMock).when(storagePoolTagsDaoMock).findStoragePoolTags(Mockito.anyLong());
 
         Storage result = presetVariableHelperSpy.getPresetVariableValueStorage(1L, 2);
