--- a/server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java
@@ -88,10 +88,8 @@
         Mockito.doReturn(callerAccountMock).when(roleManagerImpl).getCurrentAccount();
         Mockito.doReturn(callerAccountRoleMock.getId()).when(callerAccountMock).getRoleId();
 
-        Mockito.when(rolePermission1Mock.getRule()).thenReturn(new Rule("api1"));
-        Mockito.when(rolePermission2Mock.getRule()).thenReturn(new Rule("api2"));
-        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission1Mock).getPermission();
-        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission2Mock).getPermission();
+        rolePermission1Mock = createMockRolePermission(new Rule("api1"));
+        rolePermission2Mock = createMockRolePermission(new Rule("api2"));
         Mockito.doReturn(RolePermissionEntity.Permission.DENY).when(rolePermission2MockWithDeniedPermission).getPermission();
 
         Mockito.doReturn(1L).when(lessPermissionsRoleMock).getId();
@@ -379,4 +377,11 @@
         Mockito.verify(roleDaoMock).listAll();
         Mockito.verify(roleManagerImpl).removeRolesIfNeeded(roles);
     }
+
+    private static RolePermission createMockRolePermission(Rule rule) {
+        RolePermission rolePermissionMock = Mockito.mock(RolePermission.class);
+        Mockito.when(rolePermissionMock.getRule()).thenReturn(rule);
+        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermissionMock).getPermission();
+        return rolePermissionMock;
+    }
 }
