--- a/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategyTest.java
+++ b/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategyTest.java
@@ -107,6 +107,12 @@
         return clvmPoolManager;
     }
 
+    private static DataStore createMockDestDataStore() {
+        DataStore destDataStore = Mockito.mock(DataStore.class);
+        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        return destDataStore;
+    }
+
     @Test
     public void testAddFullCloneFlagOnVMwareDest(){
         strategy.addFullCloneAndDiskprovisiongStrictnessFlagOnVMwareDest(dataTO);
@@ -158,15 +164,13 @@
     public void testCanBypassSecondaryStorageForUnsupportedSrcPoolType() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDestDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.PowerFlex).when(srcVolumeInfo).getStoragePoolType();
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDestDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -181,15 +185,13 @@
     public void testCanBypassSecondaryStorageForUnsupportedDestPoolType() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDestDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(srcVolumeInfo).getStoragePoolType();
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDestDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.Iscsi).when(destVolumeInfo).getStoragePoolType();
 
@@ -204,15 +206,13 @@
     public void testCanBypassSecondaryStorageWithZoneWideNFSPoolsInSameZone() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDestDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(srcVolumeInfo).getStoragePoolType();
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDestDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -283,8 +283,7 @@
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDestDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -309,8 +308,7 @@
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDestDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
