--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/utils/NsxControllerUtilsTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/utils/NsxControllerUtilsTest.java
@@ -53,6 +53,8 @@
 
     @Mock
     private NsxProviderVO nsxProviderVO;
+    @Mock
+    private NsxCommand nsxCommand;
 
     @Before
     public void setup() {
@@ -62,25 +64,22 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testSendCommandAnswerFailure() {
-        NsxCommand cmd = Mockito.mock(NsxCommand.class);
         Mockito.when(nsxProviderDao.findByZoneId(zoneId)).thenReturn(null);
-        nsxControllerUtils.sendNsxCommand(cmd, zoneId);
+        nsxControllerUtils.sendNsxCommand(nsxCommand, zoneId);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void testSendCommandNoNsxProvider() {
-        NsxCommand cmd = Mockito.mock(NsxCommand.class);
-        Mockito.when(agentMgr.easySend(nsxProviderHostId, cmd)).thenReturn(null);
-        nsxControllerUtils.sendNsxCommand(cmd, zoneId);
+        Mockito.when(agentMgr.easySend(nsxProviderHostId, nsxCommand)).thenReturn(null);
+        nsxControllerUtils.sendNsxCommand(nsxCommand, zoneId);
     }
 
     @Test
     public void testSendCommand() {
-        NsxCommand cmd = Mockito.mock(NsxCommand.class);
         NsxAnswer answer = Mockito.mock(NsxAnswer.class);
         Mockito.when(answer.getResult()).thenReturn(true);
-        Mockito.when(agentMgr.easySend(nsxProviderHostId, cmd)).thenReturn(answer);
-        NsxAnswer nsxAnswer = nsxControllerUtils.sendNsxCommand(cmd, zoneId);
+        Mockito.when(agentMgr.easySend(nsxProviderHostId, nsxCommand)).thenReturn(answer);
+        NsxAnswer nsxAnswer = nsxControllerUtils.sendNsxCommand(nsxCommand, zoneId);
         Assert.assertNotNull(nsxAnswer);
     }
 
