--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagTypeCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagTypeCmdTest.java
@@ -45,9 +45,12 @@
 
     AutoCloseable closeable;
 
+    private BaseResponse baseResponseMock;
+
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        baseResponseMock = Mockito.mock(BaseResponse.class);
         listTungstenFabricTagTypeCmd = new ListTungstenFabricTagTypeCmd();
         listTungstenFabricTagTypeCmd.tungstenService = tungstenService;
         ReflectionTestUtils.setField(listTungstenFabricTagTypeCmd, "tagTypeUuid", "test");
@@ -64,8 +67,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricTagTypeCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         Mockito.when(tungstenService.listTungstenTagTypes(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricTagTypeCmd.execute();
@@ -76,8 +78,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
