--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -600,6 +600,14 @@
         }
     }
 
+    private static CreateNetworkCmdByAdmin createMockCreateNetworkCmdByAdmin(List<Long> bgpPeerIds, Long zoneId) {
+        CreateNetworkCmdByAdmin cmd = Mockito.mock(CreateNetworkCmdByAdmin.class);
+        Mockito.when(cmd.getCidrSize()).thenReturn(24);
+        Mockito.when(cmd.getBgpPeerIds()).thenReturn(bgpPeerIds);
+        when(cmd.getZoneId()).thenReturn(zoneId);
+        return cmd;
+    }
+
     @Test(expected = InvalidParameterValueException.class)
     public void testCreateL2NetworkDnsFailure() {
         registerCallContext();
@@ -827,10 +835,8 @@
 
     public void testCreateIpv4RoutedNetworkWithBgpPeersFailure1() {
         registerCallContext();
-        CreateNetworkCmdByAdmin cmd = Mockito.mock(CreateNetworkCmdByAdmin.class);
-        Mockito.when(cmd.getCidrSize()).thenReturn(24);
         List<Long> bgpPeerIds = Arrays.asList(11L, 12L);
-        Mockito.when(cmd.getBgpPeerIds()).thenReturn(bgpPeerIds);
+        CreateNetworkCmdByAdmin cmd = createMockCreateNetworkCmdByAdmin(bgpPeerIds, zoneId);
 
         prepareCreateNetworkDnsMocks(cmd, Network.GuestType.Isolated, false, true, true);
         when(networkOfferingVO.getNetworkMode()).thenReturn(NetworkOffering.NetworkMode.ROUTED);
@@ -839,7 +845,6 @@
         when(routedIpv4Manager.isVirtualRouterGateway(networkOfferingVO)).thenReturn(true);
 
         DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        when(cmd.getZoneId()).thenReturn(zoneId);
         when(_dcDao.findById(zoneId)).thenReturn(zone);
         when(zone.getId()).thenReturn(zoneId);
 
@@ -855,10 +860,8 @@
     @Test
     public void testCreateIpv4RoutedNetworkWithBgpPeersFailure2() {
         registerCallContext();
-        CreateNetworkCmdByAdmin cmd = Mockito.mock(CreateNetworkCmdByAdmin.class);
-        Mockito.when(cmd.getCidrSize()).thenReturn(24);
         List<Long> bgpPeerIds = Arrays.asList(11L, 12L);
-        Mockito.when(cmd.getBgpPeerIds()).thenReturn(bgpPeerIds);
+        CreateNetworkCmdByAdmin cmd = createMockCreateNetworkCmdByAdmin(bgpPeerIds, zoneId);
 
         prepareCreateNetworkDnsMocks(cmd, Network.GuestType.Isolated, false, false, true);
         when(networkOfferingVO.getNetworkMode()).thenReturn(NetworkOffering.NetworkMode.ROUTED);
@@ -867,7 +870,6 @@
         when(routedIpv4Manager.isVirtualRouterGateway(networkOfferingVO)).thenReturn(true);
 
         DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        when(cmd.getZoneId()).thenReturn(zoneId);
         when(_dcDao.findById(zoneId)).thenReturn(zone);
         when(zone.getId()).thenReturn(zoneId);
 
@@ -883,10 +885,8 @@
     @Test
     public void testCreateIpv4RoutedNetworkWithBgpPeersFailure3() {
         registerCallContext();
-        CreateNetworkCmdByAdmin cmd = Mockito.mock(CreateNetworkCmdByAdmin.class);
-        Mockito.when(cmd.getCidrSize()).thenReturn(24);
         List<Long> bgpPeerIds = Arrays.asList(11L, 12L);
-        Mockito.when(cmd.getBgpPeerIds()).thenReturn(bgpPeerIds);
+        CreateNetworkCmdByAdmin cmd = createMockCreateNetworkCmdByAdmin(bgpPeerIds, zoneId);
 
         prepareCreateNetworkDnsMocks(cmd, Network.GuestType.Isolated, false, false, true);
         when(networkOfferingVO.getNetworkMode()).thenReturn(NetworkOffering.NetworkMode.ROUTED);
@@ -897,7 +897,6 @@
         doThrow(new InvalidParameterValueException("validation error")).when(routedIpv4Manager).validateBgpPeers(nullable(Account.class), nullable(Long.class), any(List.class));
 
         DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        when(cmd.getZoneId()).thenReturn(zoneId);
         when(_dcDao.findById(zoneId)).thenReturn(zone);
         when(zone.getId()).thenReturn(zoneId);
 
