--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static ConfigKey createMockConfigKey(Long dataCenterId, boolean valueInReturn) {
+        ConfigKey testConfig = mock(ConfigKey.class);
+        when(testConfig.valueIn(dataCenterId)).thenReturn(valueInReturn);
+        return testConfig;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -377,15 +383,14 @@
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllowsWhenConfigFalse() {
         Host vmHost = mock(Host.class);
-        ConfigKey testConfig = mock(ConfigKey.class);
 
         Long dataCenterId = 5L;
+        ConfigKey testConfig = createMockConfigKey(dataCenterId, false);
         String hostIp = "1.1.1.1";
         String routerIp = "2.2.2.2";
         Map<String, String> ipAddresses = new HashMap<>();
         ipAddresses.put(NetworkElementCommand.ROUTER_IP, routerIp);
 
-        when(testConfig.valueIn(dataCenterId)).thenReturn(false);
         when(vmHost.getDataCenterId()).thenReturn(dataCenterId);
 
         virtualMachineManagerImpl.addHostIpToCertDetailsIfConfigAllows(vmHost, ipAddresses, testConfig);
