--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -374,6 +374,14 @@
         updatedConfigKeyDepot = true;
     }
 
+    private static CreateBackupScheduleCmd createMockCreateBackupScheduleCmd() {
+        CreateBackupScheduleCmd createBackupScheduleCmdMock = Mockito.mock(CreateBackupScheduleCmd.class);
+        when(createBackupScheduleCmdMock.getTimezone()).thenReturn("GMT");
+        when(createBackupScheduleCmdMock.getIntervalType()).thenReturn(DateUtil.IntervalType.DAILY);
+        when(createBackupScheduleCmdMock.getMaxBackups()).thenReturn(8);
+        return createBackupScheduleCmdMock;
+    }
+
     @Test
     public void testExceptionWhenUpdateWithNullId() {
         try {
@@ -609,11 +617,8 @@
         Long domainId = 4L;
         Long backupOfferingId = 5L;
 
-        CreateBackupScheduleCmd cmd = Mockito.mock(CreateBackupScheduleCmd.class);
+        CreateBackupScheduleCmd cmd = createMockCreateBackupScheduleCmd();
         when(cmd.getVmId()).thenReturn(vmId);
-        when(cmd.getTimezone()).thenReturn("GMT");
-        when(cmd.getIntervalType()).thenReturn(DateUtil.IntervalType.DAILY);
-        when(cmd.getMaxBackups()).thenReturn(8);
         when(cmd.getSchedule()).thenReturn("00:00:00");
         when(cmd.getQuiesceVM()).thenReturn(null);
 
@@ -657,10 +662,8 @@
         long domainId = 4L;
         long backupOfferingId = 5L;
 
+        createBackupScheduleCmdMock = createMockCreateBackupScheduleCmd();
         when(createBackupScheduleCmdMock.getVmId()).thenReturn(vmId);
-        when(createBackupScheduleCmdMock.getTimezone()).thenReturn("GMT");
-        when(createBackupScheduleCmdMock.getIntervalType()).thenReturn(DateUtil.IntervalType.DAILY);
-        when(createBackupScheduleCmdMock.getMaxBackups()).thenReturn(8);
 
         when(vmInstanceDao.findById(vmId)).thenReturn(vmInstanceVOMock);
         when(vmInstanceVOMock.getDataCenterId()).thenReturn(zoneId);
