--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -684,6 +684,14 @@
         }
     }
 
+    private static ImportVmCmd createMockImportVmCmd(String importSource, String name) {
+        ImportVmCmd importVmCmd = Mockito.mock(ImportVmCmd.class);
+        when(importVmCmd.getDomainId()).thenReturn(null);
+        when(importVmCmd.getImportSource()).thenReturn(importSource);
+        when(importVmCmd.getName()).thenReturn(name);
+        return importVmCmd;
+    }
+
     private void baseBasicParametersCheckForImportInstance(String name, Long domainId, String accountName) {
         unmanagedVMsManager.basicParametersCheckForImportInstance(name, domainId, accountName);
     }
@@ -767,12 +775,9 @@
         when(cluster.getDataCenterId()).thenReturn(zoneId);
         when(clusterDao.findById(clusterId)).thenReturn(cluster);
 
-        ImportVmCmd importVmCmd = Mockito.mock(ImportVmCmd.class);
-
-        when(importVmCmd.getName()).thenReturn(vmName);
+        ImportVmCmd importVmCmd = createMockImportVmCmd(VmImportService.ImportSource.VMWARE.toString(), vmName);
+
         when(importVmCmd.getClusterId()).thenReturn(clusterId);
-        when(importVmCmd.getDomainId()).thenReturn(null);
-        when(importVmCmd.getImportSource()).thenReturn(VmImportService.ImportSource.VMWARE.toString());
         when(importVmCmd.getHostIp()).thenReturn(host);
         when(importVmCmd.getNicNetworkList()).thenReturn(Map.of("NIC 1", networkId));
         when(importVmCmd.getConvertInstanceHostId()).thenReturn(null);
