--- a/framework/config/src/test/java/org/apache/cloudstack/framework/config/impl/ConfigDepotImplTest.java
+++ b/framework/config/src/test/java/org/apache/cloudstack/framework/config/impl/ConfigDepotImplTest.java
@@ -104,7 +104,7 @@
     private void runTestGetConfigStringValue(String key, String value) {
         ConfigurationVO configurationVO = Mockito.mock(ConfigurationVO.class);
         Mockito.when(configurationVO.getValue()).thenReturn(value);
-        Mockito.when(_configDao.findById(key)).thenReturn(configurationVO);
+        mockConfigurationDaoFindById(key, configurationVO);
         String result = configDepotImpl.getConfigStringValue(key, ConfigKey.Scope.Global, null);
         Assert.assertEquals(value, result);
     }
@@ -193,7 +193,7 @@
         ConfigurationVO configurationVO = new ConfigurationVO(StorageDisableThreshold.category(), "DEFAULT", "component",
                 StorageDisableThreshold.key(), StorageDisableThreshold.defaultValue(), StorageDisableThreshold.description(),
                 StorageDisableThreshold.displayText(), StorageDisableThreshold.parent(), 1L, 10L);
-        Mockito.when(_configDao.findById("pool.storage.capacity.disablethreshold")).thenReturn(configurationVO);
+        mockConfigurationDaoFindById("pool.storage.capacity.disablethreshold", configurationVO);
         configDepotImpl.populateConfigurations();
 
         Mockito.verify(_configDao, Mockito.times(1)).persist(configurationVO);
@@ -217,4 +217,8 @@
         Assert.assertEquals(parentScope, result.first());
         Assert.assertEquals(parentId, result.second());
     }
+
+    private void mockConfigurationDaoFindById(final String key, final ConfigurationVO configurationVO) {
+        Mockito.when(_configDao.findById(key)).thenReturn(configurationVO);
+    }
 }
