--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/UpdateTungstenFabricLBHealthMonitorCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/UpdateTungstenFabricLBHealthMonitorCmdTest.java
@@ -44,6 +44,8 @@
     EntityManager entityManager;
     @Mock
     TungstenService tungstenService;
+    @Mock
+    private TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVOMock;
 
     UpdateTungstenFabricLBHealthMonitorCmd updateTungstenFabricLBHealthMonitorCmd;
 
@@ -72,39 +74,35 @@
 
     @Test
     public void createTest() throws ResourceAllocationException {
-        TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO =
-                Mockito.mock(TungstenFabricLBHealthMonitorVO.class);
         Mockito.when(tungstenService.updateTungstenFabricLBHealthMonitor(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(),
                 ArgumentMatchers.anyInt(), ArgumentMatchers.anyInt(), ArgumentMatchers.anyInt(),
-                ArgumentMatchers.anyString(), ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(tungstenFabricLBHealthMonitorVO);
+                ArgumentMatchers.anyString(), ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(tungstenFabricLBHealthMonitorVOMock);
         updateTungstenFabricLBHealthMonitorCmd.create();
-        Assert.assertEquals(Optional.of(tungstenFabricLBHealthMonitorVO.getId()),
+        Assert.assertEquals(Optional.of(tungstenFabricLBHealthMonitorVOMock.getId()),
                 Optional.ofNullable(updateTungstenFabricLBHealthMonitorCmd.getEntityId()));
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getUuid(),
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getUuid(),
                 updateTungstenFabricLBHealthMonitorCmd.getEntityUuid());
     }
 
     @Test
     public void executeTest() throws Exception {
         updateTungstenFabricLBHealthMonitorCmd.setEntityId(1L);
-        TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO =
-                Mockito.mock(TungstenFabricLBHealthMonitorVO.class);
-        tungstenFabricLBHealthMonitorVO.setType("test");
-        ReflectionTestUtils.setField(tungstenFabricLBHealthMonitorVO, "id", 1L);
-        ReflectionTestUtils.setField(tungstenFabricLBHealthMonitorVO, "uuid", "test");
-        tungstenFabricLBHealthMonitorVO.setRetry(1);
-        tungstenFabricLBHealthMonitorVO.setTimeout(1);
-        tungstenFabricLBHealthMonitorVO.setInterval(1);
-        tungstenFabricLBHealthMonitorVO.setHttpMethod("test");
-        tungstenFabricLBHealthMonitorVO.setExpectedCode("test");
-        tungstenFabricLBHealthMonitorVO.setUrlPath("test");
+        tungstenFabricLBHealthMonitorVOMock.setType("test");
+        ReflectionTestUtils.setField(tungstenFabricLBHealthMonitorVOMock, "id", 1L);
+        ReflectionTestUtils.setField(tungstenFabricLBHealthMonitorVOMock, "uuid", "test");
+        tungstenFabricLBHealthMonitorVOMock.setRetry(1);
+        tungstenFabricLBHealthMonitorVOMock.setTimeout(1);
+        tungstenFabricLBHealthMonitorVOMock.setInterval(1);
+        tungstenFabricLBHealthMonitorVOMock.setHttpMethod("test");
+        tungstenFabricLBHealthMonitorVOMock.setExpectedCode("test");
+        tungstenFabricLBHealthMonitorVOMock.setUrlPath("test");
         LoadBalancer loadBalancer = Mockito.mock(LoadBalancer.class);
         Network network = Mockito.mock(Network.class);
         DataCenter dataCenter = Mockito.mock(DataCenter.class);
 
         Mockito.when(entityManager.findById(ArgumentMatchers.eq(TungstenFabricLBHealthMonitorVO.class),
-                ArgumentMatchers.anyLong())).thenReturn(tungstenFabricLBHealthMonitorVO);
+                ArgumentMatchers.anyLong())).thenReturn(tungstenFabricLBHealthMonitorVOMock);
         Mockito.when(entityManager.findById(ArgumentMatchers.eq(LoadBalancer.class), ArgumentMatchers.anyLong())).thenReturn(loadBalancer);
         Mockito.when(entityManager.findById(ArgumentMatchers.eq(Network.class), ArgumentMatchers.anyLong())).thenReturn(network);
         Mockito.when(entityManager.findById(ArgumentMatchers.eq(DataCenter.class), ArgumentMatchers.anyLong())).thenReturn(dataCenter);
@@ -113,15 +111,15 @@
         TungstenFabricLBHealthMonitorResponse tungstenFabricLBHealthMonitorResponse =
                 (TungstenFabricLBHealthMonitorResponse) updateTungstenFabricLBHealthMonitorCmd.getResponseObject();
 
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getType(), tungstenFabricLBHealthMonitorResponse.getType());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getId(), tungstenFabricLBHealthMonitorResponse.getId());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getUuid(), tungstenFabricLBHealthMonitorResponse.getUuid());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getRetry(), tungstenFabricLBHealthMonitorResponse.getRetry());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getTimeout(), tungstenFabricLBHealthMonitorResponse.getTimeout());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getInterval(), tungstenFabricLBHealthMonitorResponse.getInterval());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getHttpMethod(), tungstenFabricLBHealthMonitorResponse.getHttpMethod());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getExpectedCode(), tungstenFabricLBHealthMonitorResponse.getExpectedCode());
-        Assert.assertEquals(tungstenFabricLBHealthMonitorVO.getUrlPath(), tungstenFabricLBHealthMonitorResponse.getUrlPath());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getType(), tungstenFabricLBHealthMonitorResponse.getType());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getId(), tungstenFabricLBHealthMonitorResponse.getId());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getUuid(), tungstenFabricLBHealthMonitorResponse.getUuid());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getRetry(), tungstenFabricLBHealthMonitorResponse.getRetry());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getTimeout(), tungstenFabricLBHealthMonitorResponse.getTimeout());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getInterval(), tungstenFabricLBHealthMonitorResponse.getInterval());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getHttpMethod(), tungstenFabricLBHealthMonitorResponse.getHttpMethod());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getExpectedCode(), tungstenFabricLBHealthMonitorResponse.getExpectedCode());
+        Assert.assertEquals(tungstenFabricLBHealthMonitorVOMock.getUrlPath(), tungstenFabricLBHealthMonitorResponse.getUrlPath());
 
         Assert.assertEquals(dataCenter.getId(), tungstenFabricLBHealthMonitorResponse.getZoneId());
         Assert.assertEquals(dataCenter.getName(), tungstenFabricLBHealthMonitorResponse.getZoneName());
