--- a/plugins/storage/object/cloudian/src/test/java/org/apache/cloudstack/storage/datastore/driver/CloudianHyperStoreObjectStoreDriverImplTest.java
+++ b/plugins/storage/object/cloudian/src/test/java/org/apache/cloudstack/storage/datastore/driver/CloudianHyperStoreObjectStoreDriverImplTest.java
@@ -414,9 +414,8 @@
         doReturn(cloudianClient).when(cloudianHyperStoreObjectStoreDriverImpl).getCloudianClientByStoreId(TEST_STORE_ID);
 
         // Setup the user/group as existing and active
-        CloudianUser user = mock(CloudianUser.class);
+        CloudianUser user = createMockCloudianUser(true);
         CloudianGroup group = mock(CloudianGroup.class);
-        when(user.getActive()).thenReturn(true);
         when(group.getActive()).thenReturn(true);
         when(cloudianClient.listUser(hsUserId, hsGroupId)).thenReturn(user);
         when(cloudianClient.listGroup(hsGroupId)).thenReturn(group);
@@ -480,8 +479,7 @@
         doReturn(cloudianClient).when(cloudianHyperStoreObjectStoreDriverImpl).getCloudianClientByStoreId(TEST_STORE_ID);
 
         // Setup the user to be found but inactive.
-        CloudianUser user = mock(CloudianUser.class);
-        when(user.getActive()).thenReturn(false);
+        CloudianUser user = createMockCloudianUser(false);
         when(cloudianClient.listUser(hsUserId, hsGroupId)).thenReturn(user);
 
         // Test: user exists but is disabled. This condition requires HyperStore administrator action.
@@ -683,4 +681,10 @@
         CloudRuntimeException thrown = assertThrows(CloudRuntimeException.class, () -> cloudianHyperStoreObjectStoreDriverImpl.getS3ClientByBucketAndStore(bucket, TEST_STORE_ID));
         assertTrue(thrown.getMessage().contains("not found"));
     }
+
+    private static CloudianUser createMockCloudianUser(boolean active) {
+        CloudianUser user = mock(CloudianUser.class);
+        when(user.getActive()).thenReturn(active);
+        return user;
+    }
 }
