--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -196,6 +196,12 @@
         return cmd;
     }
 
+    private static ChangeSharedFSDiskOfferingCmd createMockChangeSharedFSDiskOfferingCmd() {
+        ChangeSharedFSDiskOfferingCmd cmd = mock(ChangeSharedFSDiskOfferingCmd.class);
+        when(cmd.getId()).thenReturn(s_sharedFSId);
+        return cmd;
+    }
+
     private SharedFSVO getMockSharedFS() {
         SharedFSVO sharedFS = new SharedFSVO(s_name, s_description, s_domainId, s_ownerId, s_zoneId,
                 s_providerName, SharedFS.Protocol.NFS, SharedFS.FileSystemType.valueOf(s_fsFormat), s_serviceOfferingId);
@@ -434,8 +440,7 @@
         Long newMinIops = 2000L;
         Long newMaxIops = 4000L;
         Long newDiskOfferingId = 10L;
-        ChangeSharedFSDiskOfferingCmd cmd = mock(ChangeSharedFSDiskOfferingCmd.class);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
+        ChangeSharedFSDiskOfferingCmd cmd = createMockChangeSharedFSDiskOfferingCmd();
         when(cmd.getDiskOfferingId()).thenReturn(newDiskOfferingId);
         when(cmd.getSize()).thenReturn(newSize);
         when(cmd.getMinIops()).thenReturn(newMinIops);
@@ -459,8 +464,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testChangeSharedFSDiskOfferingInvalidState() throws ResourceAllocationException {
-        ChangeSharedFSDiskOfferingCmd cmd = mock(ChangeSharedFSDiskOfferingCmd.class);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
+        ChangeSharedFSDiskOfferingCmd cmd = createMockChangeSharedFSDiskOfferingCmd();
 
         SharedFSVO sharedFS = getMockSharedFS();
         ReflectionTestUtils.setField(sharedFS, "state", SharedFS.State.Destroyed);
