--- a/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
+++ b/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
@@ -31,11 +31,19 @@
 import org.apache.cloudstack.framework.events.EventBusException;
 import org.apache.cloudstack.framework.events.EventSubscriber;
 import org.apache.cloudstack.framework.events.EventTopic;
+import org.junit.Before;
 import org.junit.Test;
 
 import com.cloud.utils.UuidUtils;
 
 public class InMemoryEventBusTest {
+
+    private EventSubscriber eventSubscriber;
+
+    @Before
+    public void setUp() {
+        eventSubscriber = mock(EventSubscriber.class);
+    }
 
     @Test
     public void testConfigure() throws Exception {
@@ -51,28 +59,26 @@
     @Test
     public void testSubscribe() throws Exception {
         EventTopic topic = mock(EventTopic.class);
-        EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        UUID uuid = bus.subscribe(topic, subscriber);
+        UUID uuid = bus.subscribe(topic, eventSubscriber);
         assertNotNull(uuid);
 
         String uuidStr = uuid.toString();
         assertTrue(UuidUtils.isUuid(uuidStr));
         assertTrue(bus.totalSubscribers() == 1);
 
-        bus.unsubscribe(uuid, subscriber);
+        bus.unsubscribe(uuid, eventSubscriber);
         assertTrue(bus.totalSubscribers() == 0);
     }
 
     @Test(expected = EventBusException.class)
     public void testSubscribeFailTopic() throws Exception {
-        EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        bus.subscribe(null, subscriber);
+        bus.subscribe(null, eventSubscriber);
     }
 
     @Test(expected = EventBusException.class)
@@ -87,11 +93,10 @@
     @Test
     public void testUnsubscribe() throws Exception {
         EventTopic topic = mock(EventTopic.class);
-        EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        UUID uuid = bus.subscribe(topic, subscriber);
+        UUID uuid = bus.subscribe(topic, eventSubscriber);
         assertNotNull(uuid);
 
         String uuidStr = uuid.toString();
@@ -99,7 +104,7 @@
         assertTrue(UuidUtils.isUuid(uuidStr));
         assertTrue(bus.totalSubscribers() == 1);
         //
-        bus.unsubscribe(uuid, subscriber);
+        bus.unsubscribe(uuid, eventSubscriber);
         assertTrue(bus.totalSubscribers() == 0);
     }
 
@@ -119,22 +124,20 @@
 
     @Test(expected = EventBusException.class)
     public void testUnsubscribeFailWrongUUID() throws Exception {
-        EventSubscriber subscriber = mock(EventSubscriber.class);
         InMemoryEventBus bus = new InMemoryEventBus();
         UUID uuid = UUID.randomUUID();
 
-        bus.unsubscribe(uuid, subscriber);
+        bus.unsubscribe(uuid, eventSubscriber);
     }
 
     @Test
     public void testPublish() throws Exception {
         EventTopic topic = mock(EventTopic.class);
-        EventSubscriber subscriber = mock(EventSubscriber.class);
         Event event = mock(Event.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        UUID uuid = bus.subscribe(topic, subscriber);
+        UUID uuid = bus.subscribe(topic, eventSubscriber);
         assertNotNull(uuid);
 
         String uuidStr = uuid.toString();
@@ -143,20 +146,19 @@
 
         bus.publish(event);
 
-        verify(subscriber, times(1)).onEvent(event);
+        verify(eventSubscriber, times(1)).onEvent(event);
 
-        bus.unsubscribe(uuid, subscriber);
+        bus.unsubscribe(uuid, eventSubscriber);
         assertTrue(bus.totalSubscribers() == 0);
     }
 
     @Test
     public void testPublishEmpty() throws Exception {
-        EventSubscriber subscriber = mock(EventSubscriber.class);
         Event event = mock(Event.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
         bus.publish(event);
 
-        verify(subscriber, times(0)).onEvent(event);
+        verify(eventSubscriber, times(0)).onEvent(event);
     }
 }
