--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -82,6 +82,12 @@
         return expectedCount;
     }
 
+    private static CancelMaintenanceCmd createMockCancelMaintenanceCmd() {
+        CancelMaintenanceCmd cmd = mock(CancelMaintenanceCmd.class);
+        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        return cmd;
+    }
+
     @Test
     public void countPendingJobs() {
         long expectedCount = prepareCountPendingJobs();
@@ -535,8 +541,7 @@
     @Test
     public void cancelMaintenanceCmdNoMsHost() {
         Mockito.when(msHostDao.findById(1L)).thenReturn(null);
-        CancelMaintenanceCmd cmd = mock(CancelMaintenanceCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        CancelMaintenanceCmd cmd = createMockCancelMaintenanceCmd();
 
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.cancelMaintenance(cmd);
@@ -548,8 +553,7 @@
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getState()).thenReturn(ManagementServerHost.State.Up);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost);
-        CancelMaintenanceCmd cmd = mock(CancelMaintenanceCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        CancelMaintenanceCmd cmd = createMockCancelMaintenanceCmd();
 
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.cancelMaintenance(cmd);
@@ -561,8 +565,7 @@
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getState()).thenReturn(ManagementServerHost.State.Maintenance);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost);
-        CancelMaintenanceCmd cmd = mock(CancelMaintenanceCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        CancelMaintenanceCmd cmd = createMockCancelMaintenanceCmd();
         Mockito.when(clusterManagerMock.execute(anyString(), anyLong(), anyString(), anyBoolean())).thenReturn("Success");
 
         spy.cancelMaintenance(cmd);
