--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -102,6 +102,8 @@
     ASNumberDao asNumberDao;
     @Mock
     BgpPeerDetailsDao bgpPeerDetailsDao;
+    @Mock
+    private BgpPeerVO bgpPeerMock;
 
     @Before
     public void setUp() {
@@ -210,9 +212,8 @@
 
     @Test
     public void testCreateBgpPeersCommandsForNetwork() {
-        BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
         BgpPeerVO bgpPeer2 = Mockito.mock(BgpPeerVO.class);
-        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2);
+        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeerMock, bgpPeer2);
         Commands cmds = new Commands(Command.OnError.Stop);
         VirtualRouter router = Mockito.mock(VirtualRouter.class);
         NetworkVO network = Mockito.mock(NetworkVO.class);
@@ -238,9 +239,8 @@
 
     @Test
     public void testCreateBgpPeersCommandsForVpc() {
-        BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
         BgpPeerVO bgpPeer2 = Mockito.mock(BgpPeerVO.class);
-        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2);
+        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeerMock, bgpPeer2);
         Commands cmds = new Commands(Command.OnError.Stop);
         VirtualRouter router = Mockito.mock(VirtualRouter.class);
         NetworkVO network = Mockito.mock(NetworkVO.class);
