--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -267,6 +267,13 @@
         return webhook;
     }
 
+    private static Webhook createMockWebhook() {
+        Webhook webhook = Mockito.mock(Webhook.class);
+        Mockito.when(webhook.getUuid()).thenReturn("webhookUuid");
+        Mockito.when(webhook.getName()).thenReturn("webhookName");
+        return webhook;
+    }
+
     @Test
     public void testValidateWebhookOwnerPayloadUrlUpdateSameExist() {
         WebhookVO webhookVO = mockWebhookVO(1L);
@@ -564,9 +571,7 @@
         Mockito.when(webhookDelivery.getStartTime()).thenReturn(new Date(1000));
         Mockito.when(webhookDelivery.getEndTime()).thenReturn(new Date(2000));
 
-        Webhook webhook = Mockito.mock(Webhook.class);
-        Mockito.when(webhook.getUuid()).thenReturn("webhookUuid");
-        Mockito.when(webhook.getName()).thenReturn("webhookName");
+        Webhook webhook = createMockWebhook();
 
         ManagementServerHostVO msHost = Mockito.mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getUuid()).thenReturn("managementServerUuid");
@@ -636,9 +641,7 @@
         Mockito.when(webhookDelivery.getStartTime()).thenReturn(new Date(1000));
         Mockito.when(webhookDelivery.getEndTime()).thenReturn(new Date(2000));
 
-        Webhook webhook = Mockito.mock(Webhook.class);
-        Mockito.when(webhook.getUuid()).thenReturn("webhookUuid");
-        Mockito.when(webhook.getName()).thenReturn("webhookName");
+        Webhook webhook = createMockWebhook();
 
         Mockito.when(managementServerHostDao.findByMsid(1L)).thenReturn(null);
 
