--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -19,8 +19,6 @@
 
 package org.apache.cloudstack.affinity;
 
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.ArgumentMatchers.nullable;
 import static org.mockito.Mockito.when;
 
 import java.util.ArrayList;
@@ -79,7 +77,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -88,7 +85,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        configureAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -109,7 +106,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         plan.adjustHostPriority(host2Id, DeploymentPlan.HostPriorityAdjustment.DEFAULT);
@@ -120,7 +116,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id, vm3Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        configureAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -146,7 +142,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -155,7 +150,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        configureAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         when(vm2.getHostId()).thenReturn(null);
@@ -170,4 +165,10 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(1), plan.getHostPriorities().get(host2Id));
     }
+
+    private void configureAffinityGroupVMMapDao(long vmId, List<AffinityGroupVMMapVO> vmGroupMappings,
+            long affinityGroupId, List<Long> groupVMIds) {
+        when(_affinityGroupVMMapDao.findByVmIdType(Mockito.eq(vmId), Mockito.nullable(String.class))).thenReturn(vmGroupMappings);
+        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+    }
 }
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -43,8 +43,6 @@
 import java.util.Date;
 import java.util.List;
 
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.ArgumentMatchers.nullable;
 import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
@@ -78,7 +76,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -87,7 +84,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        configureAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -108,7 +105,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         plan.adjustHostPriority(host2Id, DeploymentPlan.HostPriorityAdjustment.DEFAULT);
@@ -119,7 +115,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id, vm3Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        configureAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -136,6 +132,12 @@
         Assert.assertEquals(Integer.valueOf(0), plan.getHostPriorities().get(host3Id));
     }
 
+    private void configureAffinityGroupVMMapDao(long vmId, List<AffinityGroupVMMapVO> vmGroupMappings,
+            long affinityGroupId, List<Long> groupVMIds) {
+        when(_affinityGroupVMMapDao.findByVmIdType(Mockito.eq(vmId), Mockito.nullable(String.class))).thenReturn(vmGroupMappings);
+        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+    }
+
     @Test
     public void testProcessWithNotRunningVM() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
@@ -145,7 +147,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -154,7 +155,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        configureAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         when(vm2.getHostId()).thenReturn(null);
