--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtRevertSnapshotCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtRevertSnapshotCommandWrapperTest.java
@@ -91,7 +91,7 @@
         String snapshotPath = "test";
         Pair<String, SnapshotObjectTO> expectedResult = new Pair<>(snapshotPath, snapshotObjectToPrimaryMock);
 
-        Mockito.doReturn(snapshotPath).when(snapshotObjectToPrimaryMock).getPath();
+        configureSnapshotPath(snapshotObjectToPrimaryMock, snapshotPath);
 
         try (MockedStatic<Files> ignored = Mockito.mockStatic(Files.class)) {
             Mockito.when(Files.exists(Mockito.any(Path.class))).thenReturn(true);
@@ -115,7 +115,7 @@
             Mockito.when(Paths.get(Mockito.any(), Mockito.any())).thenReturn(pathMock);
             Mockito.when(Files.exists(Mockito.any(Path.class), Mockito.any())).thenReturn(false);
 
-            Mockito.doReturn(snapshotPath).when(snapshotObjectToSecondaryMock).getPath();
+            configureSnapshotPath(snapshotObjectToSecondaryMock, snapshotPath);
             Mockito.doReturn(snapshotPath).when(libvirtRevertSnapshotCommandWrapperSpy).getFullPathAccordingToStorage(
                     Mockito.any(), Mockito.any());
 
@@ -169,4 +169,8 @@
         libvirtRevertSnapshotCommandWrapperSpy.revertVolumeToSnapshot(kvmStoragePoolSecondaryMock, snapshotObjectToPrimaryMock, snapshotObjectToSecondaryMock, kvmStoragePoolPrimaryMock, resourceMock
         , false);
     }
+
+    private static void configureSnapshotPath(SnapshotObjectTO snapshotObjectToMock, String snapshotPath) {
+        Mockito.doReturn(snapshotPath).when(snapshotObjectToMock).getPath();
+    }
 }
