--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -610,12 +610,8 @@
             CallContext callContextMock = Mockito.mock(CallContext.class);
             when(CallContext.current()).thenReturn(callContextMock);;
 
-            ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
+            ListUserDataCmd cmd = createMockListUserDataCmd();
             when(cmd.getId()).thenReturn(1L);
-            when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
@@ -643,12 +639,8 @@
             when(CallContext.current()).thenReturn(callContextMock);
             when(callContextMock.getCallingAccount()).thenReturn(account);
 
-            ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
+            ListUserDataCmd cmd = createMockListUserDataCmd();
             when(cmd.getName()).thenReturn("testSearchUserdataName");
-            when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
@@ -676,12 +668,8 @@
             when(CallContext.current()).thenReturn(callContextMock);
             when(callContextMock.getCallingAccount()).thenReturn(account);
 
-            ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
+            ListUserDataCmd cmd = createMockListUserDataCmd();
             when(cmd.getKeyword()).thenReturn("testSearchUserdataKeyword");
-            when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
@@ -700,6 +688,15 @@
 
             Assert.assertEquals(userdataResultList.first().get(0), userDataList.get(0));
         }
+    }
+
+    private static ListUserDataCmd createMockListUserDataCmd() {
+        ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
+        when(cmd.getAccountName()).thenReturn("testAccountName");
+        when(cmd.getDomainId()).thenReturn(1L);
+        when(cmd.getProjectId()).thenReturn(2L);
+        when(cmd.isRecursive()).thenReturn(false);
+        return cmd;
     }
 
     private UserVmVO mockFilterUefiHostsTestVm(String uefiValue) {
