--- a/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
@@ -258,6 +258,13 @@
         f.set(configKey, o);
     }
 
+    private static BgpPeerNetworkMapVO createMockBgpPeerNetworkMap(Long bgpPeerId, BgpPeer.State state) {
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
+        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId);
+        when(bgpPeerNetworkMap2.getState()).thenReturn(state);
+        return bgpPeerNetworkMap2;
+    }
+
     @Test
     public void testCreateDataCenterIpv4GuestSubnet() {
         CreateIpv4SubnetForZoneCmd cmd = new CreateIpv4SubnetForZoneCmd();
@@ -991,12 +998,8 @@
         when(network.getId()).thenReturn(networkId);
         when(networkDao.findById(networkId)).thenReturn(network);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
-        when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap(bgpPeerId1, BgpPeer.State.Active);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap(bgpPeerId2, BgpPeer.State.Revoke);
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
@@ -1024,12 +1027,8 @@
 
         when(network.getId()).thenReturn(networkId);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
-        when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap(bgpPeerId1, BgpPeer.State.Active);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap(bgpPeerId2, BgpPeer.State.Revoke);
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
@@ -1136,12 +1135,8 @@
         when(vpc.getId()).thenReturn(vpcId);
         when(vpcDao.findById(vpcId)).thenReturn(vpc);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
-        when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap(bgpPeerId1, BgpPeer.State.Active);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap(bgpPeerId2, BgpPeer.State.Revoke);
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
@@ -1169,12 +1164,8 @@
 
         when(vpc.getId()).thenReturn(vpcId);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
-        when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap(bgpPeerId1, BgpPeer.State.Active);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap(bgpPeerId2, BgpPeer.State.Revoke);
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
