--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -129,13 +129,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockSrcPoolSystemIdDetail();
         String srcPoolSystemId = "610204d03e3ad60f";
-        when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
-
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockSrcPoolSystemIdDetail();
         String destPoolSystemId = "610204d03e3ad60f";
-        when(destPoolSystemIdDetail.getValue()).thenReturn(destPoolSystemId);
 
         when(storagePoolDetailsDao.findDetail(1L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcPoolSystemIdDetail);
         when(storagePoolDetailsDao.findDetail(2L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(destPoolSystemIdDetail);
@@ -152,12 +150,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         String srcPoolSystemId = "610204d03e3ad60f";
-        when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
-
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockSrcPoolSystemIdDetail();
+
         String destPoolSystemId = "7332760565f6340f";
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockSrcPoolSystemIdDetail();
         when(destPoolSystemIdDetail.getValue()).thenReturn(destPoolSystemId);
 
         when(storagePoolDetailsDao.findDetail(1L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcPoolSystemIdDetail);
@@ -175,9 +172,8 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         String srcPoolSystemId = "610204d03e3ad60f";
-        when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockSrcPoolSystemIdDetail();
 
         StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         when(destPoolSystemIdDetail.getValue()).thenReturn(null);
@@ -281,6 +277,12 @@
         Answer answer = scaleIOPrimaryDataStoreDriver.liveMigrateVolume(srcData, destData);
 
         Assert.assertFalse(answer.getResult());
+    }
+
+    private static StoragePoolDetailVO createMockSrcPoolSystemIdDetail() {
+        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        when(srcPoolSystemIdDetail.getValue()).thenReturn("610204d03e3ad60f");
+        return srcPoolSystemIdDetail;
     }
 
     private void fillSrcVolumeDetails(VolumeInfo srcData, DataStore srcStore) {
