--- a/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
@@ -96,6 +96,9 @@
     @Mock
     VolumeDataStoreDao volumeDataStoreDao;
 
+    @Mock
+    private ListDataStoreObjectsAnswer listDataStoreObjectsAnswer;
+
     @InjectMocks
     @Spy
     private StorageBrowserImpl storageBrowser;
@@ -185,8 +188,7 @@
         DataStore dataStore = Mockito.mock(DataStore.class);
         Mockito.when(dataStoreMgr.getDataStore(imageStoreId, imageStore.getRole())).thenReturn(dataStore);
 
-        ListDataStoreObjectsAnswer answer = Mockito.mock(ListDataStoreObjectsAnswer.class);
-        Mockito.doReturn(answer).when(storageBrowser).listObjectsInStore(dataStore, path, 0, 10);
+        Mockito.doReturn(listDataStoreObjectsAnswer).when(storageBrowser).listObjectsInStore(dataStore, path, 0, 10);
 
         ListResponse<DataStoreObjectResponse> response = storageBrowser.listImageStoreObjects(cmd);
 
@@ -206,13 +208,12 @@
         DataStore dataStore = Mockito.mock(DataStore.class);
         Mockito.when(dataStoreMgr.getDataStore(storeId, DataStoreRole.Primary)).thenReturn(dataStore);
 
-        ListDataStoreObjectsAnswer answer = Mockito.mock(ListDataStoreObjectsAnswer.class);
-        Mockito.doReturn(answer).when(storageBrowser).listObjectsInStore(dataStore, path, 0, 10);
+        Mockito.doReturn(listDataStoreObjectsAnswer).when(storageBrowser).listObjectsInStore(dataStore, path, 0, 10);
 
         ListResponse<DataStoreObjectResponse> response = storageBrowser.listPrimaryStoreObjects(cmd);
 
         Assert.assertNotNull(response);
-        Assert.assertEquals(answer.getPaths().size(), response.getResponses().size());
+        Assert.assertEquals(listDataStoreObjectsAnswer.getPaths().size(), response.getResponses().size());
     }
 
     @Test
