--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -384,6 +384,12 @@
         resourceManager.getHostCredentials(host);
     }
 
+    private static StoragePoolVO createMockStoragePool(long storagePoolId) {
+        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(storagePool.getId()).thenReturn(storagePoolId);
+        return storagePool;
+    }
+
     private HostVO mockExistingRoutingHost(long dcId, Long podId, Long clusterId) {
         HostVO existing = Mockito.mock(HostVO.class);
         when(existing.getType()).thenReturn(Host.Type.Routing);
@@ -917,10 +923,8 @@
         List<VolumeVO> volumesUsingTheStoragePoolAccessGroups = new ArrayList<>(Arrays.asList(volume1, volume2));
         Mockito.doReturn(volumesUsingTheStoragePoolAccessGroups).when(resourceManager).listOfVolumesUsingTheStorageAccessGroups(sagsToDelete, 1L, null, null, null);
 
-        StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
-        StoragePoolVO pool2 = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(pool1.getId()).thenReturn(10L);
-        Mockito.when(pool2.getId()).thenReturn(12L);
+        StoragePoolVO pool1 = createMockStoragePool(10L);
+        StoragePoolVO pool2 = createMockStoragePool(12L);
         List<StoragePoolVO> poolsToAdd = Arrays.asList(pool1, pool2);
 
         Mockito.doReturn(poolsToAdd)
