--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -239,6 +239,9 @@
     PrimaryDataStoreDao primaryDataStoreDao;
 
     @Mock
+    private StoragePoolVO storagePool;
+
+    @Mock
     HostDao hostDao;
 
     @Mock
@@ -1370,12 +1373,11 @@
         when(rootVolume.getPoolId()).thenReturn(poolId);
         HostVO host = mock(HostVO.class);
         when(hostDao.findById(hostId)).thenReturn(host);
-        StoragePoolVO pool = mock(StoragePoolVO.class);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
-        when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
+        when(primaryDataStoreDao.findById(poolId)).thenReturn(storagePool);
         when(rootVolume.getPoolId()).thenReturn(poolId);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
-        when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
+        when(primaryDataStoreDao.findById(poolId)).thenReturn(storagePool);
         when(backupProvider.restoreBackupToVM(vm, backup, null, null, false)).thenReturn(new Pair<>(true, null));
 
         try (MockedStatic<ActionEventUtils> utils = Mockito.mockStatic(ActionEventUtils.class)) {
@@ -1426,12 +1428,11 @@
         when(rootVolume.getPoolId()).thenReturn(poolId);
         HostVO host = mock(HostVO.class);
         when(hostDao.findById(hostId)).thenReturn(host);
-        StoragePoolVO pool = mock(StoragePoolVO.class);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
-        when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
+        when(primaryDataStoreDao.findById(poolId)).thenReturn(storagePool);
         when(rootVolume.getPoolId()).thenReturn(poolId);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
-        when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
+        when(primaryDataStoreDao.findById(poolId)).thenReturn(storagePool);
         when(backupProvider.restoreBackupToVM(vm, backup, null, null, false)).thenReturn(new Pair<>(false, null));
 
         try (MockedStatic<ActionEventUtils> utils = Mockito.mockStatic(ActionEventUtils.class)) {
