--- a/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
+++ b/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
@@ -111,15 +111,10 @@
     @Test
     @SuppressWarnings("unchecked")
     public void testSortDataMotionStrategies2() {
-        DataMotionStrategy cantHandleStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy defaultStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy hyperStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy highestStrategy = mock(DataMotionStrategy.class);
-
-        doReturn(StrategyPriority.CANT_HANDLE).when(cantHandleStrategy).canHandle(any(Map.class), any(Host.class), any(Host.class));
-        doReturn(StrategyPriority.DEFAULT).when(defaultStrategy).canHandle(any(Map.class), any(Host.class), any(Host.class));
-        doReturn(StrategyPriority.HYPERVISOR).when(hyperStrategy).canHandle(any(Map.class), any(Host.class), any(Host.class));
-        doReturn(StrategyPriority.HIGHEST).when(highestStrategy).canHandle(any(Map.class), any(Host.class), any(Host.class));
+        DataMotionStrategy cantHandleStrategy = createMockDataMotionStrategy(StrategyPriority.CANT_HANDLE);
+        DataMotionStrategy defaultStrategy = createMockDataMotionStrategy(StrategyPriority.DEFAULT);
+        DataMotionStrategy hyperStrategy = createMockDataMotionStrategy(StrategyPriority.HYPERVISOR);
+        DataMotionStrategy highestStrategy = createMockDataMotionStrategy(StrategyPriority.HIGHEST);
 
         List<DataMotionStrategy> strategies = new ArrayList<DataMotionStrategy>(5);
         DataMotionStrategy strategy = null;
@@ -143,4 +138,10 @@
         strategy = factory.getDataMotionStrategy(mock(Map.class), mock(Host.class), mock(Host.class));
         assertEquals("Highest strategy was not picked.", highestStrategy, strategy);
     }
+
+    private static DataMotionStrategy createMockDataMotionStrategy(StrategyPriority priority) {
+        DataMotionStrategy strategy = mock(DataMotionStrategy.class);
+        doReturn(priority).when(strategy).canHandle(any(Map.class), any(Host.class), any(Host.class));
+        return strategy;
+    }
 }
