--- a/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
@@ -310,13 +310,9 @@
         long dataStoreId = 1L;
         List<String> paths = List.of("/path1", "/path2");
 
-        TemplateDataStoreVO templateDataStore1 = Mockito.mock(TemplateDataStoreVO.class);
-        Mockito.when(templateDataStore1.getInstallPath()).thenReturn("/path1");
-        Mockito.when(templateDataStore1.getTemplateId()).thenReturn(1L);
-
-        TemplateDataStoreVO templateDataStore2 = Mockito.mock(TemplateDataStoreVO.class);
-        Mockito.when(templateDataStore2.getInstallPath()).thenReturn("/path2");
-        Mockito.when(templateDataStore2.getTemplateId()).thenReturn(2L);
+        TemplateDataStoreVO templateDataStore1 = createMockTemplateDataStore("/path1", 1L);
+
+        TemplateDataStoreVO templateDataStore2 = createMockTemplateDataStore("/path2", 2L);
 
         List<TemplateDataStoreVO> templateList = List.of(templateDataStore1, templateDataStore2);
 
@@ -456,4 +452,11 @@
         Assert.assertEquals(200L, response.getResponses().get(1).getSize());
         Assert.assertNotNull(response.getResponses().get(1).getLastUpdated());
     }
+
+    private static TemplateDataStoreVO createMockTemplateDataStore(String installPath, long templateId) {
+        TemplateDataStoreVO templateDataStore = Mockito.mock(TemplateDataStoreVO.class);
+        Mockito.when(templateDataStore.getInstallPath()).thenReturn(installPath);
+        Mockito.when(templateDataStore.getTemplateId()).thenReturn(templateId);
+        return templateDataStore;
+    }
 }
