--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -157,6 +157,12 @@
         Assert.assertEquals(expected, accountPSResourceCount);
     }
 
+    private static StoragePoolVO createMockStoragePool(ScopeType scope) {
+        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(pool.getScope()).thenReturn(scope);
+        return pool;
+    }
+
     @Test
     public void testCheckAndUpdateVolumeAccountResourceCountSameSize() {
         runCheckAndUpdateVolumeAccountResourceCountTest(10L, 10L);
@@ -383,8 +389,7 @@
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
 
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(pool.getScope()).thenReturn(ScopeType.CLUSTER); // Shared scope
+        StoragePoolVO pool = createMockStoragePool(ScopeType.CLUSTER); // Shared scope
         Mockito.when(storagePoolDao.findById(MOCK_POOL_ID)).thenReturn(pool);
 
         assertTrue(volumeOrchestrator.volumeOnSharedStoragePool(volume));
@@ -395,8 +400,7 @@
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
 
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(pool.getScope()).thenReturn(ScopeType.HOST); // Local scope
+        StoragePoolVO pool = createMockStoragePool(ScopeType.HOST); // Local scope
         Mockito.when(storagePoolDao.findById(MOCK_POOL_ID)).thenReturn(pool);
 
         Assert.assertFalse(volumeOrchestrator.volumeOnSharedStoragePool(volume));
