--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -267,6 +267,13 @@
         return webhook;
     }
 
+    private static ListWebhooksCmd createMockListWebhooksCmd() {
+        ListWebhooksCmd cmd = Mockito.mock(ListWebhooksCmd.class);
+        Mockito.when(cmd.getStartIndex()).thenReturn(0L);
+        Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
+        return cmd;
+    }
+
     @Test
     public void testValidateWebhookOwnerPayloadUrlUpdateSameExist() {
         WebhookVO webhookVO = mockWebhookVO(1L);
@@ -771,15 +778,13 @@
 
     @Test
     public void listWebhooksReturnsEmptyResponseForNoWebhooks() {
-        ListWebhooksCmd cmd = Mockito.mock(ListWebhooksCmd.class);
+        ListWebhooksCmd cmd = createMockListWebhooksCmd();
 
         Mockito.when(cmd.getId()).thenReturn(null);
         Mockito.when(cmd.getState()).thenReturn(null);
         Mockito.when(cmd.getName()).thenReturn(null);
         Mockito.when(cmd.getKeyword()).thenReturn(null);
         Mockito.when(cmd.getScope()).thenReturn(null);
-        Mockito.when(cmd.getStartIndex()).thenReturn(0L);
-        Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         SearchBuilder<WebhookJoinVO> sb = Mockito.mock(SearchBuilder.class);
         Mockito.when(sb.entity()).thenReturn(Mockito.mock(WebhookJoinVO.class));
         SearchCriteria<WebhookJoinVO> sc = Mockito.mock(SearchCriteria.class);
