--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -157,6 +157,12 @@
         Assert.assertEquals(expected, accountPSResourceCount);
     }
 
+    private static PrimaryDataStore createMockPrimaryDataStore(PrimaryDataStoreDriver driver) {
+        PrimaryDataStore store = Mockito.mock(PrimaryDataStore.class);
+        Mockito.when(store.getDriver()).thenReturn(driver);
+        return store;
+    }
+
     @Test
     public void testCheckAndUpdateVolumeAccountResourceCountSameSize() {
         runCheckAndUpdateVolumeAccountResourceCountTest(10L, 10L);
@@ -180,10 +186,9 @@
 
     @Test
     public void testGrantVolumeAccessToHostIfNeededDriverNoNeed() {
-        PrimaryDataStore store = Mockito.mock(PrimaryDataStore.class);
         PrimaryDataStoreDriver driver = Mockito.mock(PrimaryDataStoreDriver.class);
         Mockito.when(driver.volumesRequireGrantAccessWhenUsed()).thenReturn(false);
-        Mockito.when(store.getDriver()).thenReturn(driver);
+        PrimaryDataStore store = createMockPrimaryDataStore(driver);
         volumeOrchestrator.grantVolumeAccessToHostIfNeeded(store, 1L,
                 Mockito.mock(HostVO.class), "");
         Mockito.verify(volumeService, Mockito.never())
@@ -192,10 +197,9 @@
 
     @Test
     public void testGrantVolumeAccessToHostIfNeededDriverNeeds() {
-        PrimaryDataStore store = Mockito.mock(PrimaryDataStore.class);
         PrimaryDataStoreDriver driver = Mockito.mock(PrimaryDataStoreDriver.class);
         Mockito.when(driver.volumesRequireGrantAccessWhenUsed()).thenReturn(true);
-        Mockito.when(store.getDriver()).thenReturn(driver);
+        PrimaryDataStore store = createMockPrimaryDataStore(driver);
         Mockito.when(volumeDataFactory.getVolume(Mockito.anyLong())).thenReturn(Mockito.mock(VolumeInfo.class));
         Mockito.doReturn(true).when(volumeService)
                 .grantAccess(Mockito.any(DataObject.class), Mockito.any(Host.class), Mockito.any(DataStore.class));
@@ -207,10 +211,9 @@
 
     @Test(expected = StorageAccessException.class)
     public void testGrantVolumeAccessToHostIfNeededDriverNeedsButException() {
-        PrimaryDataStore store = Mockito.mock(PrimaryDataStore.class);
         PrimaryDataStoreDriver driver = Mockito.mock(PrimaryDataStoreDriver.class);
         Mockito.when(driver.volumesRequireGrantAccessWhenUsed()).thenReturn(true);
-        Mockito.when(store.getDriver()).thenReturn(driver);
+        PrimaryDataStore store = createMockPrimaryDataStore(driver);
         Mockito.when(volumeDataFactory.getVolume(Mockito.anyLong())).thenReturn(Mockito.mock(VolumeInfo.class));
         Mockito.doThrow(CloudRuntimeException.class).when(volumeService)
                 .grantAccess(Mockito.any(DataObject.class), Mockito.any(Host.class), Mockito.any(DataStore.class));
