--- a/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
+++ b/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
@@ -76,12 +76,11 @@
     @Test
     public void testSortDataMotionStrategies() {
         DataMotionStrategy cantHandleStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy defaultStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy hyperStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy highestStrategy = mock(DataMotionStrategy.class);
+        DataMotionStrategy defaultStrategy = createMockDefaultDataMotionStrategy();
+        DataMotionStrategy hyperStrategy = createMockDefaultDataMotionStrategy();
+        DataMotionStrategy highestStrategy = createMockDefaultDataMotionStrategy();
 
         doReturn(StrategyPriority.CANT_HANDLE).when(cantHandleStrategy).canHandle(any(DataObject.class), any(DataObject.class));
-        doReturn(StrategyPriority.DEFAULT).when(defaultStrategy).canHandle(any(DataObject.class), any(DataObject.class));
         doReturn(StrategyPriority.HYPERVISOR).when(hyperStrategy).canHandle(any(DataObject.class), any(DataObject.class));
         doReturn(StrategyPriority.HIGHEST).when(highestStrategy).canHandle(any(DataObject.class), any(DataObject.class));
 
@@ -143,4 +142,10 @@
         strategy = factory.getDataMotionStrategy(mock(Map.class), mock(Host.class), mock(Host.class));
         assertEquals("Highest strategy was not picked.", highestStrategy, strategy);
     }
+
+    private static DataMotionStrategy createMockDefaultDataMotionStrategy() {
+        DataMotionStrategy defaultStrategy = mock(DataMotionStrategy.class);
+        doReturn(StrategyPriority.DEFAULT).when(defaultStrategy).canHandle(any(DataObject.class), any(DataObject.class));
+        return defaultStrategy;
+    }
 }
