--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
@@ -47,12 +47,15 @@
 
     MockedStatic<HttpClients> httpClientsMocked;
 
+    private Gateway gateway2Mock;
+
     @Before
     public void setup() {
         VRouter vRouter = mock(VRouter.class);
         vRouterApiConnector = new VRouterApiConnectorImpl(vRouter);
         httpClientsMocked = Mockito.mockStatic(HttpClients.class);
         entityUtilsMocked = Mockito.mockStatic(EntityUtils.class);
+        gateway2Mock = mock(Gateway.class);
     }
 
     @After
@@ -177,56 +180,52 @@
     @Test
     public void addGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
-        entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
-
-        assertTrue(vRouterApiConnector.addGateway(Arrays.asList(gateway1, gateway2)));
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+        entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
+
+        assertTrue(vRouterApiConnector.addGateway(Arrays.asList(gateway1, gateway2Mock)));
     }
 
     @Test
     public void addGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
-
-        assertFalse(vRouterApiConnector.addGateway(Arrays.asList(gateway1, gateway2)));
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+
+        assertFalse(vRouterApiConnector.addGateway(Arrays.asList(gateway1, gateway2Mock)));
     }
 
     @Test
     public void deleteGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
-        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
-        entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
-
-        assertTrue(vRouterApiConnector.deleteGateway(Arrays.asList(gateway1, gateway2)));
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
+        entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
+
+        assertTrue(vRouterApiConnector.deleteGateway(Arrays.asList(gateway1, gateway2Mock)));
     }
 
     @Test
     public void deleteGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
-
-        assertFalse(vRouterApiConnector.deleteGateway(Arrays.asList(gateway1, gateway2)));
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+
+        assertFalse(vRouterApiConnector.deleteGateway(Arrays.asList(gateway1, gateway2Mock)));
     }
 }
