--- a/server/src/test/java/org/apache/cloudstack/kms/KMSManagerImplKeyCreationTest.java
+++ b/server/src/test/java/org/apache/cloudstack/kms/KMSManagerImplKeyCreationTest.java
@@ -76,6 +76,9 @@
     private KMSKekVersionDao kmsKekVersionDao;
 
     @Mock
+    private KMSKekVersionVO kmsKekVersion;
+
+    @Mock
     private HSMProfileDao hsmProfileDao;
 
     @Mock
@@ -131,8 +134,7 @@
         when(mockKey.getId()).thenReturn(1L);
         when(kmsKeyDao.persist(any(KMSKeyVO.class))).thenReturn(mockKey);
 
-        KMSKekVersionVO mockVersion = mock(KMSKekVersionVO.class);
-        when(kmsKekVersionDao.persist(any(KMSKekVersionVO.class))).thenReturn(mockVersion);
+        when(kmsKekVersionDao.persist(any(KMSKekVersionVO.class))).thenReturn(kmsKekVersion);
 
         doReturn(kmsProvider).when(kmsManager).getKMSProvider(testProviderName);
 
@@ -190,8 +192,7 @@
         when(mockKey.getId()).thenReturn(1L);
         when(kmsKeyDao.persist(any(KMSKeyVO.class))).thenReturn(mockKey);
 
-        KMSKekVersionVO mockVersion = mock(KMSKekVersionVO.class);
-        when(kmsKekVersionDao.persist(any(KMSKekVersionVO.class))).thenReturn(mockVersion);
+        when(kmsKekVersionDao.persist(any(KMSKekVersionVO.class))).thenReturn(kmsKekVersion);
 
         doReturn(kmsProvider).when(kmsManager).getKMSProvider(testProviderName);
 
