--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -82,6 +82,12 @@
         return expectedCount;
     }
 
+    private static PrepareForMaintenanceCmd createMockPrepareForMaintenanceCmd() {
+        PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
+        Mockito.when(cmd.getAlgorithm()).thenReturn("test algorithm");
+        return cmd;
+    }
+
     @Test
     public void countPendingJobs() {
         long expectedCount = prepareCountPendingJobs();
@@ -366,8 +372,7 @@
     @Test
     public void prepareForMaintenanceCmdNoOtherMsHostsWithUpState() {
         Mockito.when(msHostDao.listBy(any())).thenReturn(new ArrayList<>());
-        PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
-        Mockito.when(cmd.getAlgorithm()).thenReturn("test algorithm");
+        PrepareForMaintenanceCmd cmd = createMockPrepareForMaintenanceCmd();
         Mockito.doNothing().when(indirectAgentLBMock).checkLBAlgorithmName(anyString());
 
         Assert.assertThrows(CloudRuntimeException.class, () -> {
@@ -381,8 +386,7 @@
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost);
         Mockito.when(msHostDao.listBy(ManagementServerHost.State.Up)).thenReturn(msHostList);
-        PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
-        Mockito.when(cmd.getAlgorithm()).thenReturn("test algorithm");
+        PrepareForMaintenanceCmd cmd = createMockPrepareForMaintenanceCmd();
         Mockito.doNothing().when(indirectAgentLBMock).checkLBAlgorithmName(anyString());
 
         Assert.assertThrows(CloudRuntimeException.class, () -> {
