--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/AddObjectStoragePoolCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/AddObjectStoragePoolCmdTest.java
@@ -92,7 +92,7 @@
         Mockito.doReturn(objectStore).when(storageService).discoverObjectStore(Mockito.anyString(),
                 Mockito.anyString(), Mockito.anyLong(), Mockito.anyString(), any());
         ObjectStoreResponse objectStoreResponse = new ObjectStoreResponse();
-        Mockito.doReturn(objectStoreResponse).when(responseGenerator).createObjectStoreResponse(any());
+        MockResponseGenerator.configureMockResponseGenerator(responseGenerator, objectStoreResponse);
         addObjectStoragePoolCmdSpy.execute();
 
         Mockito.verify(storageService, Mockito.times(1))
--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/UpdateObjectStoragePoolCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/UpdateObjectStoragePoolCmdTest.java
@@ -77,7 +77,7 @@
     public void testUpdateObjectStore() {
         Mockito.doReturn(objectStore).when(storageService).updateObjectStore(1L, updateObjectStoragePoolCmd);
         ObjectStoreResponse objectStoreResponse = new ObjectStoreResponse();
-        Mockito.doReturn(objectStoreResponse).when(responseGenerator).createObjectStoreResponse(any());
+        MockResponseGenerator.configureMockResponseGenerator(responseGenerator, objectStoreResponse);
         updateObjectStoragePoolCmd.execute();
         Mockito.verify(storageService, Mockito.times(1))
                 .updateObjectStore(1L, updateObjectStoragePoolCmd);
--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/MockResponseGenerator.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/MockResponseGenerator.java
@@ -0,0 +1,11 @@
+package org.apache.cloudstack.api.command.admin.storage;
+
+import org.apache.cloudstack.api.ResponseGenerator;
+import org.apache.cloudstack.api.response.ObjectStoreResponse;
+import org.mockito.Mockito;
+
+public class MockResponseGenerator {
+    public static void configureMockResponseGenerator(ResponseGenerator responseGenerator, ObjectStoreResponse objectStoreResponse) {
+        Mockito.doReturn(objectStoreResponse).when(responseGenerator).createObjectStoreResponse(Mockito.any());
+    }
+}
