--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -97,6 +97,8 @@
     HostDao hostDao;
     @Mock
     ConfigurationDao configDao;
+    @Mock
+    private DataStore destinationStore;
 
     static MockedStatic<RemoteHostEndPoint> remoteHostEndPointMock;
 
@@ -194,13 +196,12 @@
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
 
         when(srcData.getDataStore()).thenReturn(srcStore);
-        when(destData.getDataStore()).thenReturn(destStore);
+        when(destData.getDataStore()).thenReturn(destinationStore);
 
         fillSrcVolumeDetails(srcData, srcStore);
-        fillDestVolumeDetails(destData, destStore);
+        fillDestVolumeDetails(destData, destinationStore);
 
         VolumeObjectTO destVolTO = Mockito.mock(VolumeObjectTO.class);
         when(destData.getTO()).thenReturn(destVolTO);
@@ -245,13 +246,12 @@
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
 
         when(srcData.getDataStore()).thenReturn(srcStore);
-        when(destData.getDataStore()).thenReturn(destStore);
+        when(destData.getDataStore()).thenReturn(destinationStore);
 
         fillSrcVolumeDetails(srcData, srcStore);
-        fillDestVolumeDetails(destData, destStore);
+        fillDestVolumeDetails(destData, destinationStore);
 
         VolumeObjectTO destVolTO = Mockito.mock(VolumeObjectTO.class);
         when(destData.getTO()).thenReturn(destVolTO);
