--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirTakeKbossBackupCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirTakeKbossBackupCommandWrapperTest.java
@@ -213,8 +213,7 @@
         String vmName = "ttt";
         doReturn(path1).when(volumeObjectToMock1).getPath();
         doReturn(path2).when(volumeObjectToMock2).getPath();
-        doReturn(deltaPath1).when(kbossTO1).getDeltaPathOnPrimary();
-        doReturn(deltaPath2).when(kbossTO2).getDeltaPathOnPrimary();
+        setupKbossTOsDeltaPathsOnPrimary();
         doReturn(deltaMergeTreeToMock).when(kbossTO1).getDeltaMergeTreeTO();
         doNothing().when(libvirtTakeKbossBackupCommandWrapperSpy).mergeBackupDelta(any(), any(), any(), any(), anyBoolean(), any(), anyBoolean());
         doReturn(true).when(takeKbossBackupCommandMock).isEndChain();
@@ -240,8 +239,7 @@
         setupKbossTos();
         Map<String, String> mapVolumeUUidToNewVolumePath = new HashMap<>();
         String vmName = "ttt";
-        doReturn(deltaPath1).when(kbossTO1).getDeltaPathOnPrimary();
-        doReturn(deltaPath2).when(kbossTO2).getDeltaPathOnPrimary();
+        setupKbossTOsDeltaPathsOnPrimary();
 
         libvirtTakeKbossBackupCommandWrapperSpy.cleanupVm(takeKbossBackupCommandMock, libvirtComputingResourceMock, List.of(kbossTO1, kbossTO2), vmName, true,
                 mapVolumeUUidToNewVolumePath);
@@ -374,4 +372,10 @@
         doReturn(volumeObjectToMock2).when(kbossTO2).getVolumeObjectTO();
         doReturn(volUuid2).when(volumeObjectToMock2).getUuid();
     }
+
+    private void setupKbossTOsDeltaPathsOnPrimary() {
+        doReturn(deltaPath1).when(kbossTO1).getDeltaPathOnPrimary();
+
+        doReturn(deltaPath2).when(kbossTO2).getDeltaPathOnPrimary();
+    }
 }
