--- a/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/browser/StorageBrowserImplTest.java
@@ -345,13 +345,9 @@
         long dataStoreId = 1L;
         List<String> paths = List.of("/path1", "/path2");
 
-        SnapshotDataStoreVO snapshotDataStore1 = Mockito.mock(SnapshotDataStoreVO.class);
-        Mockito.when(snapshotDataStore1.getInstallPath()).thenReturn("/path1");
-        Mockito.when(snapshotDataStore1.getSnapshotId()).thenReturn(1L);
-
-        SnapshotDataStoreVO snapshotDataStore2 = Mockito.mock(SnapshotDataStoreVO.class);
-        Mockito.when(snapshotDataStore2.getInstallPath()).thenReturn("/path2");
-        Mockito.when(snapshotDataStore2.getSnapshotId()).thenReturn(2L);
+        SnapshotDataStoreVO snapshotDataStore1 = createMockSnapshotDataStore("/path1", 1L);
+
+        SnapshotDataStoreVO snapshotDataStore2 = createMockSnapshotDataStore("/path2", 2L);
 
         List<SnapshotDataStoreVO> snapshotDataStoreList = List.of(snapshotDataStore1, snapshotDataStore2);
 
@@ -381,13 +377,9 @@
         List<String> paths = List.of("path1", "path2");
         List<String> absPaths = List.of("/mnt/path1", "/mnt/path2");
 
-        SnapshotDataStoreVO snapshotDataStore1 = Mockito.mock(SnapshotDataStoreVO.class);
-        Mockito.when(snapshotDataStore1.getInstallPath()).thenReturn("/mnt/path1");
-        Mockito.when(snapshotDataStore1.getSnapshotId()).thenReturn(1L);
-
-        SnapshotDataStoreVO snapshotDataStore2 = Mockito.mock(SnapshotDataStoreVO.class);
-        Mockito.when(snapshotDataStore2.getInstallPath()).thenReturn("/mnt/path2");
-        Mockito.when(snapshotDataStore2.getSnapshotId()).thenReturn(2L);
+        SnapshotDataStoreVO snapshotDataStore1 = createMockSnapshotDataStore("/mnt/path1", 1L);
+
+        SnapshotDataStoreVO snapshotDataStore2 = createMockSnapshotDataStore("/mnt/path2", 2L);
 
         List<SnapshotDataStoreVO> snapshotDataStoreList = List.of(snapshotDataStore1, snapshotDataStore2);
 
@@ -456,4 +448,11 @@
         Assert.assertEquals(200L, response.getResponses().get(1).getSize());
         Assert.assertNotNull(response.getResponses().get(1).getLastUpdated());
     }
+
+    private static SnapshotDataStoreVO createMockSnapshotDataStore(String installPath, long snapshotId) {
+        SnapshotDataStoreVO snapshotDataStore = Mockito.mock(SnapshotDataStoreVO.class);
+        Mockito.when(snapshotDataStore.getInstallPath()).thenReturn(installPath);
+        Mockito.when(snapshotDataStore.getSnapshotId()).thenReturn(snapshotId);
+        return snapshotDataStore;
+    }
 }
