--- a/server/src/test/java/org/apache/cloudstack/storage/object/BucketApiServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/object/BucketApiServiceImplTest.java
@@ -140,11 +140,9 @@
         Long objectStoreId = 3L;
         int quota = 1;
 
-        CreateBucketCmd cmd = Mockito.mock(CreateBucketCmd.class);
+        CreateBucketCmd cmd = createMockCreateBucketCmd(poolId, quota);
         Mockito.when(cmd.getBucketName()).thenReturn(bucketName);
         Mockito.when(cmd.getEntityOwnerId()).thenReturn(ACCOUNT_ID);
-        Mockito.when(cmd.getObjectStoragePoolId()).thenReturn(poolId);
-        Mockito.when(cmd.getQuota()).thenReturn(quota);
 
         ObjectStoreVO objectStoreVO = Mockito.mock(ObjectStoreVO.class);
         Mockito.when(objectStoreVO.getId()).thenReturn(objectStoreId);
@@ -176,10 +174,8 @@
         String bucketName = "bucket1";
         int quota = 3;
 
-        CreateBucketCmd cmd = Mockito.mock(CreateBucketCmd.class);
-        Mockito.when(cmd.getObjectStoragePoolId()).thenReturn(poolId);
+        CreateBucketCmd cmd = createMockCreateBucketCmd(poolId, quota);
         Mockito.when(cmd.getEntityId()).thenReturn(bucketId);
-        Mockito.when(cmd.getQuota()).thenReturn(quota);
 
         BucketVO bucket = new BucketVO(bucketName);
         Mockito.when(bucketDao.findById(bucketId)).thenReturn(bucket);
@@ -196,6 +192,13 @@
         bucketApiService.createBucket(cmd);
 
         Assert.assertEquals(Bucket.State.Created, bucket.getState());
+    }
+
+    private static CreateBucketCmd createMockCreateBucketCmd(Long objectStoragePoolId, int quota) {
+        CreateBucketCmd cmd = Mockito.mock(CreateBucketCmd.class);
+        Mockito.when(cmd.getObjectStoragePoolId()).thenReturn(objectStoragePoolId);
+        Mockito.when(cmd.getQuota()).thenReturn(quota);
+        return cmd;
     }
 
     @Test
