--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -267,6 +267,12 @@
         return webhook;
     }
 
+    private static ListWebhooksCmd createMockListWebhooksCmd(String scopeReturn) {
+        ListWebhooksCmd cmd = Mockito.mock(ListWebhooksCmd.class);
+        Mockito.when(cmd.getScope()).thenReturn(scopeReturn);
+        return cmd;
+    }
+
     @Test
     public void testValidateWebhookOwnerPayloadUrlUpdateSameExist() {
         WebhookVO webhookVO = mockWebhookVO(1L);
@@ -795,9 +801,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void listWebhooksThrowsExceptionForInvalidScope() {
-        ListWebhooksCmd cmd = Mockito.mock(ListWebhooksCmd.class);
-
-        Mockito.when(cmd.getScope()).thenReturn("InvalidScope");
+        ListWebhooksCmd cmd = createMockListWebhooksCmd("InvalidScope");
 
         webhookApiServiceImpl.listWebhooks(cmd);
     }
@@ -838,9 +842,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void listWebhooksThrowsExceptionForUnauthorizedScope() {
-        ListWebhooksCmd cmd = Mockito.mock(ListWebhooksCmd.class);
-
-        Mockito.when(cmd.getScope()).thenReturn("Global");
+        ListWebhooksCmd cmd = createMockListWebhooksCmd("Global");
 
         webhookApiServiceImpl.listWebhooks(cmd);
     }
