--- a/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
@@ -74,6 +74,9 @@
     @Mock
     NetworkTopologyContext networkTopologyContext;
 
+    @Mock
+    private AdvancedNetworkTopology advancedNetworkTopologyMock;
+
     @InjectMocks
     VpcVirtualNetworkApplianceManagerImpl _vpcRouterMgr;
 
@@ -147,10 +150,9 @@
     public void testApplyVpnUsersException1() {
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
-        final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
         final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
-        networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
+        networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopologyMock);
         networkTopologyContext.setBasicNetworkTopology(basicNetworkTopology);
         networkTopologyContext.init();
 
@@ -173,10 +175,9 @@
     public void testApplyVpnUsersException2() {
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
-        final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
         final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
-        networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
+        networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopologyMock);
         networkTopologyContext.setBasicNetworkTopology(basicNetworkTopology);
         networkTopologyContext.init();
 
