--- a/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/api/command/ListNetrisProvidersCmdTest.java
+++ b/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/api/command/ListNetrisProvidersCmdTest.java
@@ -45,9 +45,12 @@
 
     private static final long ZONE_ID = 1L;
 
+    private NetrisProviderResponse netrisProviderResponse;
+
     @Before
     public void setup() throws Exception {
         closeable = MockitoAnnotations.openMocks(this);
+        netrisProviderResponse = Mockito.mock(NetrisProviderResponse.class);
         setPrivateField("zoneId", ZONE_ID);
     }
 
@@ -65,8 +68,7 @@
     @Test
     public void testExecuteSuccess() throws ConcurrentOperationException {
         // Setup
-        NetrisProviderResponse providerResponse = Mockito.mock(NetrisProviderResponse.class);
-        List<BaseResponse> providerList = Arrays.asList(providerResponse);
+        List<BaseResponse> providerList = Arrays.asList(netrisProviderResponse);
         Mockito.when(netrisProviderService.listNetrisProviders(ZONE_ID)).thenReturn(providerList);
 
         // Execute
@@ -83,8 +85,7 @@
     public void testExecuteWithoutZoneId() throws ConcurrentOperationException, Exception {
         // Setup
         setPrivateField("zoneId", null);
-        NetrisProviderResponse providerResponse = Mockito.mock(NetrisProviderResponse.class);
-        List<BaseResponse> providerList = Arrays.asList(providerResponse);
+        List<BaseResponse> providerList = Arrays.asList(netrisProviderResponse);
         Mockito.when(netrisProviderService.listNetrisProviders(null)).thenReturn(providerList);
 
         // Execute
