--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -333,6 +333,12 @@
         return host;
     }
 
+    private static DataObject createMockDataObject(DataTO dataTO) {
+        DataObject destData = Mockito.mock(DataObject.class);
+        when(destData.getTO()).thenReturn(dataTO);
+        return destData;
+    }
+
     @Test
     public void updateVolumeAfterCopyVolumeLiveMigrate() {
         VolumeInfo srcData = Mockito.mock(VolumeInfo.class);
@@ -514,12 +520,10 @@
     public void testCopyOfflineVolume() {
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
-        DataObject srcData = Mockito.mock(DataObject.class);
         DataTO srcDataTO = Mockito.mock(DataTO.class);
-        when(srcData.getTO()).thenReturn(srcDataTO);
-        DataObject destData = Mockito.mock(DataObject.class);
+        DataObject srcData = createMockDataObject(srcDataTO);
         DataTO destDataTO = Mockito.mock(DataTO.class);
-        when(destData.getTO()).thenReturn(destDataTO);
+        DataObject destData = createMockDataObject(destDataTO);
         Host destHost = Mockito.mock(Host.class);
 
         doReturn(false).when(scaleIOPrimaryDataStoreDriver).anyVolumeRequiresEncryption(srcData, destData);
@@ -537,12 +541,10 @@
     public void testCopyOfflineVolumeFailureWhenNoEndpointFound() {
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
-        DataObject srcData = Mockito.mock(DataObject.class);
         DataTO srcDataTO = Mockito.mock(DataTO.class);
-        when(srcData.getTO()).thenReturn(srcDataTO);
-        DataObject destData = Mockito.mock(DataObject.class);
+        DataObject srcData = createMockDataObject(srcDataTO);
         DataTO destDataTO = Mockito.mock(DataTO.class);
-        when(destData.getTO()).thenReturn(destDataTO);
+        DataObject destData = createMockDataObject(destDataTO);
         Host destHost = Mockito.mock(Host.class);
 
         doReturn(false).when(scaleIOPrimaryDataStoreDriver).anyVolumeRequiresEncryption(srcData, destData);
