--- a/engine/schema/src/test/java/org/apache/cloudstack/storage/datastore/db/PrimaryDataStoreDaoImplTest.java
+++ b/engine/schema/src/test/java/org/apache/cloudstack/storage/datastore/db/PrimaryDataStoreDaoImplTest.java
@@ -65,6 +65,9 @@
 
     @Mock
     StoragePoolVO storagePoolVO;
+
+    @Mock
+    StoragePoolVO storagePoolEntityVO;
 
     private GenericSearchBuilder<StoragePoolVO, Long> genericSearchBuilder;
 
@@ -168,8 +171,7 @@
     @Test
     public void testListAllIds() {
         GenericSearchBuilder<StoragePoolVO, Long> genericSearchBuilder = mock(SearchBuilder.class);
-        StoragePoolVO entityVO = mock(StoragePoolVO.class);
-        when(genericSearchBuilder.entity()).thenReturn(entityVO);
+        when(genericSearchBuilder.entity()).thenReturn(storagePoolEntityVO);
         doReturn(genericSearchBuilder).when(primaryDataStoreDao).createSearchBuilder(Long.class);
         List<Long> mockIds = Arrays.asList(1L, 2L, 3L);
         doReturn(mockIds).when(primaryDataStoreDao).customSearch(any(), isNull());
@@ -184,8 +186,7 @@
     @Test
     public void testListAllIdsEmptyResult() {
         GenericSearchBuilder<StoragePoolVO, Long> genericSearchBuilder = mock(SearchBuilder.class);
-        StoragePoolVO entityVO = mock(StoragePoolVO.class);
-        when(genericSearchBuilder.entity()).thenReturn(entityVO);
+        when(genericSearchBuilder.entity()).thenReturn(storagePoolEntityVO);
         doReturn(genericSearchBuilder).when(primaryDataStoreDao).createSearchBuilder(Long.class);
         doReturn(Collections.emptyList()).when(primaryDataStoreDao).customSearch(any(), isNull());
         List<Long> result = primaryDataStoreDao.listAllIds();
