--- a/engine/schema/src/test/java/org/apache/cloudstack/resourcedetail/ResourceDetailsDaoBaseTest.java
+++ b/engine/schema/src/test/java/org/apache/cloudstack/resourcedetail/ResourceDetailsDaoBaseTest.java
@@ -73,12 +73,8 @@
     public void testListDetailsKeyPairs() {
         long resourceId = 1L;
         List<String> keys = Arrays.asList("key1", "key2");
-        TestDetailVO result1 = mock(TestDetailVO.class);
-        when(result1.getName()).thenReturn("key1");
-        when(result1.getValue()).thenReturn("value1");
-        TestDetailVO result2 = mock(TestDetailVO.class);
-        when(result2.getName()).thenReturn("key2");
-        when(result2.getValue()).thenReturn("value2");
+        TestDetailVO result1 = createMockTestDetailVO("key1", "value1");
+        TestDetailVO result2 = createMockTestDetailVO("key2", "value2");
         List<TestDetailVO> mockResults = Arrays.asList(result1, result2);
         doReturn(mockResults).when(testDetailsDao).search(any(SearchCriteria.class), isNull());
         Map<String, String> result = testDetailsDao.listDetailsKeyPairs(resourceId, keys);
@@ -106,6 +102,13 @@
         verify(searchCriteria).setParameters("name", keys.toArray());
         verify(testDetailsDao).search(searchCriteria, null);
         assertTrue(result.isEmpty());
+    }
+
+    private static TestDetailVO createMockTestDetailVO(String name, String value) {
+        TestDetailVO result = mock(TestDetailVO.class);
+        when(result.getName()).thenReturn(name);
+        when(result.getValue()).thenReturn(value);
+        return result;
     }
 
     protected static class TestDetailsDao extends ResourceDetailsDaoBase<TestDetailVO> {
