--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1725,8 +1725,7 @@
     public void testValidateRoleChangeUnknownCaller() {
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getRoleId()).thenReturn(1L);
-        Role role = Mockito.mock(Role.class);
-        Mockito.when(role.getRoleType()).thenReturn(RoleType.Unknown);
+        Role role = createMockRole(RoleType.Unknown);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(role);
@@ -1737,10 +1736,8 @@
     public void testValidateRoleChangeUnknownNewRole() {
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getRoleId()).thenReturn(1L);
-        Role newRole = Mockito.mock(Role.class);
-        Mockito.when(newRole.getRoleType()).thenReturn(RoleType.Unknown);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
+        Role newRole = createMockRole(RoleType.Unknown);
+        Role callerRole = createMockRole(RoleType.DomainAdmin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
@@ -1751,13 +1748,10 @@
     public void testValidateRoleNewRoleSameCaller() {
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getRoleId()).thenReturn(1L);
-        Role currentRole = Mockito.mock(Role.class);
-        Mockito.when(currentRole.getRoleType()).thenReturn(RoleType.User);
+        Role currentRole = createMockRole(RoleType.User);
         Mockito.when(roleService.findRole(1L)).thenReturn(currentRole);
-        Role newRole = Mockito.mock(Role.class);
-        Mockito.when(newRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
+        Role newRole = createMockRole(RoleType.DomainAdmin);
+        Role callerRole = createMockRole(RoleType.DomainAdmin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
@@ -1768,12 +1762,10 @@
     public void testValidateRoleCurrentRoleSameCaller() {
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getRoleId()).thenReturn(1L);
-        Role accountRole = Mockito.mock(Role.class);
-        Mockito.when(accountRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
+        Role accountRole = createMockRole(RoleType.DomainAdmin);
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.User);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
+        Role callerRole = createMockRole(RoleType.DomainAdmin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(accountRole);
@@ -1787,8 +1779,7 @@
         Mockito.when(account.getRoleId()).thenReturn(1L);
         Role newRole = Mockito.mock(Role.class);
         Mockito.when(newRole.getRoleType()).thenReturn(RoleType.Admin);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
+        Role callerRole = createMockRole(RoleType.DomainAdmin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(2L)).thenReturn(callerRole);
@@ -1799,12 +1790,10 @@
     public void testValidateRoleNewRoleLowerCaller() {
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getRoleId()).thenReturn(1L);
-        Role newRole = Mockito.mock(Role.class);
-        Mockito.when(newRole.getRoleType()).thenReturn(RoleType.User);
+        Role newRole = createMockRole(RoleType.User);
         Role accountRole = Mockito.mock(Role.class);
         Mockito.when(accountRole.getRoleType()).thenReturn(RoleType.User);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.DomainAdmin);
+        Role callerRole = createMockRole(RoleType.DomainAdmin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(accountRole);
@@ -1817,11 +1806,9 @@
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getRoleId()).thenReturn(1L);
         Mockito.when(account.getDomainId()).thenReturn(2L);
-        Role newRole = Mockito.mock(Role.class);
-        Mockito.when(newRole.getRoleType()).thenReturn(RoleType.Admin);
-        Role accountRole = Mockito.mock(Role.class);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.Admin);
+        Role newRole = createMockRole(RoleType.Admin);
+        Role accountRole = createMockRole(RoleType.User);
+        Role callerRole = createMockRole(RoleType.Admin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(accountRole);
@@ -1834,10 +1821,8 @@
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.isDefault()).thenReturn(true);
         Mockito.when(account.getRoleId()).thenReturn(1L);
-        Role newRole = Mockito.mock(Role.class);
-        Mockito.when(newRole.getRoleType()).thenReturn(RoleType.User);
-        Role callerRole = Mockito.mock(Role.class);
-        Mockito.when(callerRole.getRoleType()).thenReturn(RoleType.Admin);
+        Role newRole = createMockRole(RoleType.User);
+        Role callerRole = createMockRole(RoleType.Admin);
         Account caller = Mockito.mock(Account.class);
         Mockito.when(caller.getRoleId()).thenReturn(2L);
         Mockito.when(roleService.findRole(1L)).thenReturn(Mockito.mock(Role.class));
@@ -2289,4 +2274,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static Role createMockRole(RoleType roleType) {
+        Role callerRole = Mockito.mock(Role.class);
+        Mockito.when(callerRole.getRoleType()).thenReturn(roleType);
+        return callerRole;
+    }
 }
