--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/MockVolumeObjectTOFactory.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/MockVolumeObjectTOFactory.java
@@ -0,0 +1,14 @@
+package org.apache.cloudstack.backup;
+
+import static org.mockito.Mockito.doReturn;
+
+import org.apache.cloudstack.storage.to.VolumeObjectTO;
+import org.mockito.Mockito;
+
+public class MockVolumeObjectTOFactory {
+    public static VolumeObjectTO createMockVolumeObjectTO(long volumeId) {
+        VolumeObjectTO volumeObjectToMock = Mockito.mock(VolumeObjectTO.class);
+        doReturn(volumeId).when(volumeObjectToMock).getVolumeId();
+        return volumeObjectToMock;
+    }
+}
