--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -343,8 +343,7 @@
     public void getActionMessageReturnsCustomSuccessMessage() {
         ExtensionCustomAction action = mock(ExtensionCustomAction.class);
         when(action.getName()).thenReturn("actionName");
-        Extension extension = mock(Extension.class);
-        when(extension.getName()).thenReturn("extension");
+        Extension extension = createMockExtension();
         when(action.getSuccessMessage()).thenReturn("Custom success message");
         String result = extensionsManager.getActionMessage(true, action, extension, ExtensionCustomAction.ResourceType.VirtualMachine, null);
         assertEquals("Custom success message", result);
@@ -365,8 +364,7 @@
     public void getActionMessageReturnsCustomFailureMessage() {
         ExtensionCustomAction action = mock(ExtensionCustomAction.class);
         when(action.getName()).thenReturn("actionName");
-        Extension extension = mock(Extension.class);
-        when(extension.getName()).thenReturn("extension");
+        Extension extension = createMockExtension();
         when(action.getErrorMessage()).thenReturn("Custom failure message");
         String result = extensionsManager.getActionMessage(false, action, extension, ExtensionCustomAction.ResourceType.VirtualMachine, null);
         assertEquals("Custom failure message", result);
@@ -1549,6 +1547,12 @@
         verify(extensionCustomActionDao).remove(actionId);
     }
 
+    private static Extension createMockExtension() {
+        Extension extension = mock(Extension.class);
+        when(extension.getName()).thenReturn("extension");
+        return extension;
+    }
+
     private void mockCallerRole(RoleType roleType) {
         CallContext callContextMock = mock(CallContext.class);
         when(CallContext.current()).thenReturn(callContextMock);
