--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
@@ -147,9 +147,9 @@
     @Before
     public void setup() {
         callContextMocked = Mockito.mockStatic(CallContext.class);
+        callContextMock = createMockCallContext();
         Mockito.when(CallContext.current()).thenReturn(callContextMock);
         Mockito.doReturn(Mockito.mock(User.class)).when(callContextMock).getCallingUser();
-        Mockito.doReturn(Mockito.mock(Account.class)).when(callContextMock).getCallingAccount();
 
         Mockito.when(networkAclDaoMock.findById(networkAclListId)).thenReturn(networkACLVOMock);
         Mockito.when(createNetworkAclCmdMock.getNetworkId()).thenReturn(1L);
@@ -422,8 +422,7 @@
 
     @Test
     public void validateNetworkAclTestAclNotDefaultWithVpc() {
-        CallContext callContextMock = Mockito.mock(CallContext.class);
-        Mockito.doReturn(Mockito.mock(Account.class)).when(callContextMock).getCallingAccount();
+        CallContext callContextMock = createMockCallContext();
 
         Mockito.when(CallContext.current()).thenReturn(callContextMock);
 
@@ -1199,8 +1198,7 @@
         Mockito.when(nextAclRuleMock.getAclId()).thenReturn(networkAclMockId);
         Mockito.when(previousAclRuleMock.getAclId()).thenReturn(networkAclMockId);
 
-        CallContext callContextMock = Mockito.mock(CallContext.class);
-        Mockito.doReturn(Mockito.mock(Account.class)).when(callContextMock).getCallingAccount();
+        CallContext callContextMock = createMockCallContext();
 
         Mockito.doReturn(networkAclMock).when(networkAclDaoMock).findById(networkAclMockId);
         Mockito.when(CallContext.current()).thenReturn(callContextMock);
@@ -1441,6 +1439,12 @@
         validateAclConsistencyTestAclConsistencyHashBlank();
     }
 
+    private static CallContext createMockCallContext() {
+        CallContext callContextMock = Mockito.mock(CallContext.class);
+        Mockito.doReturn(Mockito.mock(Account.class)).when(callContextMock).getCallingAccount();
+        return callContextMock;
+    }
+
     private void validateAclConsistencyTestAclConsistencyHashBlank() {
         ArrayList<NetworkACLItemVO> allAclRules = new ArrayList<>();
         allAclRules.add(networkAclItemVoMock);
