--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -929,18 +929,15 @@
     public void listTungstenPolicyTest() {
         NetworkVO networkVO = mock(NetworkVO.class);
         TungstenAnswer listTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
-        TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
+        TungstenNetworkPolicy tungstenNetworkPolicy = createMockTungstenNetworkPolicy(networkPolicy, virtualNetwork);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(networkDao.findById(anyLong())).thenReturn(networkVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenPolicyCommand.class), anyLong())).thenReturn(listTungstenPolicyAnswer);
         when(listTungstenPolicyAnswer.getTungstenModelList()).thenReturn(List.of(tungstenNetworkPolicy));
-        when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
-        when(tungstenNetworkPolicy.getVirtualNetworkList()).thenReturn(
-                List.of(virtualNetwork));
 
         assertNotNull(tungstenService.listTungstenPolicy(1L, 2L, 3L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
@@ -1055,16 +1052,14 @@
     @Test
     public void removeTungstenPolicyRuleTest() {
         TungstenAnswer removeTungstenPolicyRuleAnswer = MockTungstenAnswerFactory.get(true);
-        TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenPolicyRuleCommand.class), anyLong())).thenReturn(removeTungstenPolicyRuleAnswer);
+        TungstenNetworkPolicy tungstenNetworkPolicy = createMockTungstenNetworkPolicy(networkPolicy, virtualNetwork);
         when(removeTungstenPolicyRuleAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
-        when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
-        when(tungstenNetworkPolicy.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
 
         assertNotNull(tungstenService.removeTungstenPolicyRule(1L, "948f421c-edde-4518-a391-09299cc25dc2", "8b4637b6-5629-46de-8fb2-d0b0502bfa85"));
     }
@@ -1165,16 +1160,14 @@
     @Test
     public void applyTungstenPolicyTest() {
         TungstenAnswer applyTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
-        TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(applyTungstenPolicyAnswer);
+        TungstenNetworkPolicy tungstenNetworkPolicy = createMockTungstenNetworkPolicy(networkPolicy, virtualNetwork);
         when(applyTungstenPolicyAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
-        when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
-        when(tungstenNetworkPolicy.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
 
         assertNotNull(tungstenService.applyTungstenPolicy(1L, "948f421c-edde-4518-a391-09299cc25dc2", "8b4637b6-5629-46de-8fb2-d0b0502bfa85", 1, 1));
     }
@@ -1207,16 +1200,14 @@
     @Test
     public void removeTungstenPolicyTest() {
         TungstenAnswer removeTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
-        TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenPolicyCommand.class), anyLong())).thenReturn(removeTungstenPolicyAnswer);
+        TungstenNetworkPolicy tungstenNetworkPolicy = createMockTungstenNetworkPolicy(networkPolicy, virtualNetwork);
         when(removeTungstenPolicyAnswer.getTungstenModel()).thenReturn(tungstenNetworkPolicy);
-        when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
-        when(tungstenNetworkPolicy.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
 
         assertNotNull(tungstenService.removeTungstenPolicy(1L, "948f421c-edde-4518-a391-09299cc25dc2", "8b4637b6-5629-46de-8fb2-d0b0502bfa85"));
     }
@@ -1668,4 +1659,11 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static TungstenNetworkPolicy createMockTungstenNetworkPolicy(NetworkPolicy networkPolicy, VirtualNetwork virtualNetwork) {
+        TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
+        when(tungstenNetworkPolicy.getNetworkPolicy()).thenReturn(networkPolicy);
+        when(tungstenNetworkPolicy.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
+        return tungstenNetworkPolicy;
+    }
 }
