--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -386,8 +386,7 @@
 
     @Test
     public void sendExtensionPathNotReadyAlertCallsAlertManager() {
-        Extension ext = mock(Extension.class);
-        when(ext.getState()).thenReturn(Extension.State.Enabled);
+        Extension ext = createMockExtension(Extension.State.Enabled);
         extensionsManager.sendExtensionPathNotReadyAlert(ext);
         verify(alertManager, atLeastOnce()).sendAlert(eq(AlertManager.AlertType.ALERT_TYPE_EXTENSION_PATH_NOT_READY),
                 anyLong(), anyLong(), anyString(), anyString());
@@ -395,8 +394,7 @@
 
     @Test
     public void sendExtensionPathNotReadyAlertDoesNotCallsAlertManager() {
-        Extension ext = mock(Extension.class);
-        when(ext.getState()).thenReturn(Extension.State.Disabled);
+        Extension ext = createMockExtension(Extension.State.Disabled);
         extensionsManager.sendExtensionPathNotReadyAlert(ext);
         verify(alertManager, never()).sendAlert(eq(AlertManager.AlertType.ALERT_TYPE_EXTENSION_PATH_NOT_READY),
                 anyLong(), anyLong(), anyString(), anyString());
@@ -1549,6 +1547,12 @@
         verify(extensionCustomActionDao).remove(actionId);
     }
 
+    private static Extension createMockExtension(Extension.State state) {
+        Extension ext = mock(Extension.class);
+        when(ext.getState()).thenReturn(state);
+        return ext;
+    }
+
     private void mockCallerRole(RoleType roleType) {
         CallContext callContextMock = mock(CallContext.class);
         when(CallContext.current()).thenReturn(callContextMock);
