--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -2024,8 +2024,7 @@
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
 
-        BackupProvider backupProvider = mock(BackupProvider.class);
-        when(backupProvider.crossZoneInstanceCreationEnabled(offering)).thenReturn(true);
+        BackupProvider backupProvider = createMockBackupProvider(offering, true);
 
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(offering);
@@ -2052,8 +2051,7 @@
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
 
-        BackupProvider backupProvider = mock(BackupProvider.class);
-        when(backupProvider.crossZoneInstanceCreationEnabled(offering)).thenReturn(false);
+        BackupProvider backupProvider = createMockBackupProvider(offering, false);
 
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(offering);
@@ -2616,6 +2614,12 @@
             assertTrue(result.first().stream().anyMatch(o -> o.getName().equals("Global Offering")));
             assertTrue(result.first().stream().anyMatch(o -> o.getName().equals("User Domain Offering")));
         }
+    }
+
+    private static BackupProvider createMockBackupProvider(BackupOfferingVO offering, boolean crossZoneInstanceCreationEnabled) {
+        BackupProvider backupProvider = mock(BackupProvider.class);
+        when(backupProvider.crossZoneInstanceCreationEnabled(offering)).thenReturn(crossZoneInstanceCreationEnabled);
+        return backupProvider;
     }
 
     private BackupOfferingVO createMockOffering(Long id, String name) {
