--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -202,6 +202,12 @@
         return sharedFS;
     }
 
+    private static ChangeSharedFSServiceOfferingCmd createMockChangeSharedFSServiceOfferingCmd() {
+        ChangeSharedFSServiceOfferingCmd cmd = mock(ChangeSharedFSServiceOfferingCmd.class);
+        when(cmd.getId()).thenReturn(s_sharedFSId);
+        return cmd;
+    }
+
     @Test
     public void testDeploySharedFS() throws ResourceUnavailableException, InsufficientCapacityException, ResourceAllocationException, NoTransitionException, OperationTimedoutException {
         CreateSharedFSCmd cmd = getMockCreateSharedFSCmd();
@@ -470,10 +476,9 @@
 
     @Test
     public void testChangeSharedFSServiceOffering() throws ResourceUnavailableException, InsufficientCapacityException, ManagementServerException, OperationTimedoutException, NoTransitionException, VirtualMachineMigrationException {
-        ChangeSharedFSServiceOfferingCmd cmd = mock(ChangeSharedFSServiceOfferingCmd.class);
         Long newServiceOfferingId = 100L;
+        ChangeSharedFSServiceOfferingCmd cmd = createMockChangeSharedFSServiceOfferingCmd();
         when(cmd.getServiceOfferingId()).thenReturn(newServiceOfferingId);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
 
         SharedFSVO sharedFS = getMockSharedFS();
         ReflectionTestUtils.setField(sharedFS, "id", s_sharedFSId);
@@ -492,9 +497,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testChangeSharedFSServiceOfferingInvalidState() throws ResourceUnavailableException, InsufficientCapacityException, ManagementServerException, OperationTimedoutException, VirtualMachineMigrationException {
-        ChangeSharedFSServiceOfferingCmd cmd = mock(ChangeSharedFSServiceOfferingCmd.class);
+        ChangeSharedFSServiceOfferingCmd cmd = createMockChangeSharedFSServiceOfferingCmd();
         Long newServiceOfferingId = 100L;
-        when(cmd.getId()).thenReturn(s_sharedFSId);
 
         SharedFSVO sharedFS = getMockSharedFS();
         ReflectionTestUtils.setField(sharedFS, "id", s_sharedFSId);
