--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -839,10 +839,9 @@
         DataStore dataStore = mock(DataStore.class);
         when(dataStore.getTO()).thenReturn(dataStoreTO);
 
-        StoragePoolVO destPool = mock(StoragePoolVO.class);
+        StoragePoolVO destPool = createMockStoragePool(Storage.StoragePoolType.NetworkFilesystem);
         when(destPool.getDataCenterId()).thenReturn(zoneId);
         when(destPool.getClusterId()).thenReturn(null);
-        when(destPool.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
         StoragePoolVO zoneDestPool = mock(StoragePoolVO.class);
         if (selectTemporaryStorage) {
             long temporaryStoragePoolId = 1L;
@@ -1082,6 +1081,12 @@
         return cluster;
     }
 
+    private static StoragePoolVO createMockStoragePool(Storage.StoragePoolType poolType) {
+        StoragePoolVO destPool = mock(StoragePoolVO.class);
+        when(destPool.getPoolType()).thenReturn(poolType);
+        return destPool;
+    }
+
     @Test(expected = CloudRuntimeException.class)
     public void testSelectInstanceConversionTemporaryLocationInvalidStorage() {
         ClusterVO cluster = getClusterForTests();
@@ -1424,8 +1429,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testCheckConversionStoragePoolPrimaryStagingPoolTypeNotAllowedForce() {
-        StoragePoolVO destPool = mock(StoragePoolVO.class);
-        Mockito.when(destPool.getPoolType()).thenReturn(Storage.StoragePoolType.RBD);
+        StoragePoolVO destPool = createMockStoragePool(Storage.StoragePoolType.RBD);
         long destPoolId = 1L;
         Mockito.when(primaryDataStoreDao.findById(destPoolId)).thenReturn(destPool);
         unmanagedVMsManager.checkConversionStoragePool(destPoolId, true);
