--- a/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
+++ b/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
@@ -31,11 +31,19 @@
 import org.apache.cloudstack.framework.events.EventBusException;
 import org.apache.cloudstack.framework.events.EventSubscriber;
 import org.apache.cloudstack.framework.events.EventTopic;
+import org.junit.Before;
 import org.junit.Test;
 
 import com.cloud.utils.UuidUtils;
 
 public class InMemoryEventBusTest {
+
+    private EventTopic eventTopic;
+
+    @Before
+    public void setUp() {
+        eventTopic = mock(EventTopic.class);
+    }
 
     @Test
     public void testConfigure() throws Exception {
@@ -50,12 +58,11 @@
 
     @Test
     public void testSubscribe() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
         EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        UUID uuid = bus.subscribe(topic, subscriber);
+        UUID uuid = bus.subscribe(eventTopic, subscriber);
         assertNotNull(uuid);
 
         String uuidStr = uuid.toString();
@@ -77,21 +84,19 @@
 
     @Test(expected = EventBusException.class)
     public void testSubscribeFailSubscriber() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        bus.subscribe(topic, null);
+        bus.subscribe(eventTopic, null);
     }
 
     @Test
     public void testUnsubscribe() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
         EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        UUID uuid = bus.subscribe(topic, subscriber);
+        UUID uuid = bus.subscribe(eventTopic, subscriber);
         assertNotNull(uuid);
 
         String uuidStr = uuid.toString();
@@ -128,13 +133,12 @@
 
     @Test
     public void testPublish() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
         EventSubscriber subscriber = mock(EventSubscriber.class);
         Event event = mock(Event.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
-        UUID uuid = bus.subscribe(topic, subscriber);
+        UUID uuid = bus.subscribe(eventTopic, subscriber);
         assertNotNull(uuid);
 
         String uuidStr = uuid.toString();
