--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -141,6 +141,9 @@
     private StoragePoolVO storagePoolVOMock;
 
     @Mock
+    private StoragePoolVO storagePoolMock;
+
+    @Mock
     private VolumeVO volume1VOMock;
 
     @Mock
@@ -361,26 +364,23 @@
 
     @Test
     public void testStoragePoolHasEnoughIopsNullRequestedIops() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         List<Long> iopsList = Arrays.asList(null, 0L);
         for (Long iops : iopsList) {
-            assertTrue(storageManagerImpl.storagePoolHasEnoughIops(iops, pool));
+            assertTrue(storageManagerImpl.storagePoolHasEnoughIops(iops, storagePoolMock));
         }
     }
 
     @Test
     public void testStoragePoolHasEnoughIopsSuccess1() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.doReturn(true).when(storageManagerImpl).storagePoolHasEnoughIops(
-                Mockito.eq(100L), Mockito.anyList(), Mockito.eq(pool), Mockito.eq(false));
-        assertTrue(storageManagerImpl.storagePoolHasEnoughIops(100L, pool));
+                Mockito.eq(100L), Mockito.anyList(), Mockito.eq(storagePoolMock), Mockito.eq(false));
+        assertTrue(storageManagerImpl.storagePoolHasEnoughIops(100L, storagePoolMock));
     }
 
     @Test
     public void testStoragePoolHasEnoughIopsNoVolumesOrPool() {
         List<Pair<Volume, DiskProfile>> list = new ArrayList<>();
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
-        Assert.assertFalse(storageManagerImpl.storagePoolHasEnoughIops(list, pool));
+        Assert.assertFalse(storageManagerImpl.storagePoolHasEnoughIops(list, storagePoolMock));
         list = List.of(new Pair<>(Mockito.mock(Volume.class), Mockito.mock(DiskProfile.class)));
         Assert.assertFalse(storageManagerImpl.storagePoolHasEnoughIops(list, null));
     }
@@ -402,24 +402,22 @@
         DiskProfile profile = Mockito.mock(DiskProfile.class);
         Mockito.when(profile.getDiskOfferingId()).thenReturn(1L);
         List<Pair<Volume, DiskProfile>> list = List.of(new Pair<>(volume, profile));
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.doReturn(true).when(storageManagerImpl)
-                .storagePoolHasEnoughIops(100L, list, pool, true);
-        assertTrue(storageManagerImpl.storagePoolHasEnoughIops(list, pool));
+                .storagePoolHasEnoughIops(100L, list, storagePoolMock, true);
+        assertTrue(storageManagerImpl.storagePoolHasEnoughIops(list, storagePoolMock));
 
         Mockito.when(profile.getDiskOfferingId()).thenReturn(2L);
         Mockito.when(profile.getMinIops()).thenReturn(200L);
         Mockito.doReturn(false).when(storageManagerImpl)
-                .storagePoolHasEnoughIops(200L, list, pool, true);
-        Assert.assertFalse(storageManagerImpl.storagePoolHasEnoughIops(list, pool));
+                .storagePoolHasEnoughIops(200L, list, storagePoolMock, true);
+        Assert.assertFalse(storageManagerImpl.storagePoolHasEnoughIops(list, storagePoolMock));
     }
 
     @Test
     public void testStoragePoolHasEnoughSpaceNullSize() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         List<Long> sizeList = Arrays.asList(null, 0L);
         for (Long size : sizeList) {
-            assertTrue(storageManagerImpl.storagePoolHasEnoughSpace(size, pool));
+            assertTrue(storageManagerImpl.storagePoolHasEnoughSpace(size, storagePoolMock));
         }
     }
 
@@ -441,21 +439,20 @@
 
     @Test
     public void testIsStoragePoolCompliantWithStoragePolicy() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(diskOfferingDetailsDao.getDetail(1L, ApiConstants.STORAGE_POLICY))
                 .thenReturn("policy");
         try {
             Mockito.doReturn(null)
-                    .when(storageManagerImpl).getCheckDatastorePolicyComplianceAnswer("policy", pool);
-            assertTrue(storageManagerImpl.isStoragePoolCompliantWithStoragePolicy(1L, pool));
+                    .when(storageManagerImpl).getCheckDatastorePolicyComplianceAnswer("policy", storagePoolMock);
+            assertTrue(storageManagerImpl.isStoragePoolCompliantWithStoragePolicy(1L, storagePoolMock));
         } catch (StorageUnavailableException e) {
             Assert.fail(e.getMessage());
         }
         try {
             Mockito.doReturn(new com.cloud.agent.api.Answer(
                             Mockito.mock(CheckDataStoreStoragePolicyComplianceCommand.class)))
-                    .when(storageManagerImpl).getCheckDatastorePolicyComplianceAnswer("policy", pool);
-            assertTrue(storageManagerImpl.isStoragePoolCompliantWithStoragePolicy(1L, pool));
+                    .when(storageManagerImpl).getCheckDatastorePolicyComplianceAnswer("policy", storagePoolMock);
+            assertTrue(storageManagerImpl.isStoragePoolCompliantWithStoragePolicy(1L, storagePoolMock));
         } catch (StorageUnavailableException e) {
             Assert.fail(e.getMessage());
         }
@@ -464,8 +461,8 @@
                     new com.cloud.agent.api.Answer(Mockito.mock(CheckDataStoreStoragePolicyComplianceCommand.class),
                             false, "");
             Mockito.doReturn(answer)
-                    .when(storageManagerImpl).getCheckDatastorePolicyComplianceAnswer("policy", pool);
-            Assert.assertFalse(storageManagerImpl.isStoragePoolCompliantWithStoragePolicy(1L, pool));
+                    .when(storageManagerImpl).getCheckDatastorePolicyComplianceAnswer("policy", storagePoolMock);
+            Assert.assertFalse(storageManagerImpl.isStoragePoolCompliantWithStoragePolicy(1L, storagePoolMock));
         } catch (StorageUnavailableException e) {
             Assert.fail(e.getMessage());
         }
@@ -473,10 +470,9 @@
 
     @Test
     public void testGetCheckDatastorePolicyComplianceAnswerNullAnswer() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         try {
-            Assert.assertNull(storageManagerImpl.getCheckDatastorePolicyComplianceAnswer(null, pool));
-            Assert.assertNull(storageManagerImpl.getCheckDatastorePolicyComplianceAnswer("", pool));
+            Assert.assertNull(storageManagerImpl.getCheckDatastorePolicyComplianceAnswer(null, storagePoolMock));
+            Assert.assertNull(storageManagerImpl.getCheckDatastorePolicyComplianceAnswer("", storagePoolMock));
         } catch (StorageUnavailableException e) {
             Assert.fail(e.getMessage());
         }
@@ -884,51 +880,47 @@
 
     @Test
     public void testCheckPoolforSpaceForResize1() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Long allocatedSizeWithTemplate = 100L * 1024 * 1024 * 1024;
 
-        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(pool, allocatedSizeWithTemplate);
+        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(storagePoolMock, allocatedSizeWithTemplate);
         Long totalAskingSize = maxAskingSize / 2;
 
-        boolean result = storageManagerImpl.checkPoolforSpace(pool, allocatedSizeWithTemplate, totalAskingSize, false);
+        boolean result = storageManagerImpl.checkPoolforSpace(storagePoolMock, allocatedSizeWithTemplate, totalAskingSize, false);
         Assert.assertFalse(result);
     }
 
     @Test
     public void testCheckPoolforSpaceForResize2() {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Long allocatedSizeWithTemplate = 100L * 1024 * 1024 * 1024;
 
-        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(pool, allocatedSizeWithTemplate);
+        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(storagePoolMock, allocatedSizeWithTemplate);
         Long totalAskingSize = maxAskingSize / 2;
 
-        boolean result = storageManagerImpl.checkPoolforSpace(pool, allocatedSizeWithTemplate, totalAskingSize, true);
+        boolean result = storageManagerImpl.checkPoolforSpace(storagePoolMock, allocatedSizeWithTemplate, totalAskingSize, true);
         Assert.assertFalse(result);
     }
 
     @Test
     public void testCheckPoolforSpaceForResize3() throws NoSuchFieldException, IllegalAccessException {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Long allocatedSizeWithTemplate = 100L * 1024 * 1024 * 1024;
 
-        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(pool, allocatedSizeWithTemplate);
+        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(storagePoolMock, allocatedSizeWithTemplate);
         Long totalAskingSize = maxAskingSize + 1;
         overrideDefaultConfigValue(StorageManagerImpl.AllowVolumeReSizeBeyondAllocation, "_defaultValue", "true");
 
-        boolean result = storageManagerImpl.checkPoolforSpace(pool, allocatedSizeWithTemplate, totalAskingSize, true);
+        boolean result = storageManagerImpl.checkPoolforSpace(storagePoolMock, allocatedSizeWithTemplate, totalAskingSize, true);
         Assert.assertFalse(result);
     }
 
     @Test
     public void testCheckPoolforSpaceForResize4() throws NoSuchFieldException, IllegalAccessException {
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Long allocatedSizeWithTemplate = 100L * 1024 * 1024 * 1024;
 
-        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(pool, allocatedSizeWithTemplate);
+        Long maxAskingSize = testCheckPoolforSpaceForResizeSetup(storagePoolMock, allocatedSizeWithTemplate);
         Long totalAskingSize = maxAskingSize / 2;
         overrideDefaultConfigValue(StorageManagerImpl.AllowVolumeReSizeBeyondAllocation, "_defaultValue", "true");
 
-        boolean result = storageManagerImpl.checkPoolforSpace(pool, allocatedSizeWithTemplate, totalAskingSize, true);
+        boolean result = storageManagerImpl.checkPoolforSpace(storagePoolMock, allocatedSizeWithTemplate, totalAskingSize, true);
         assertTrue(result);
     }
 
