--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -423,6 +423,13 @@
         return instanceDisks;
     }
 
+    private static ImportUnmanagedInstanceCmd createMockImportUnmanagedInstanceCmd(String name) {
+        ImportUnmanagedInstanceCmd importUnmanageInstanceCmd = Mockito.mock(ImportUnmanagedInstanceCmd.class);
+        when(importUnmanageInstanceCmd.getName()).thenReturn(name);
+        when(importUnmanageInstanceCmd.getDomainId()).thenReturn(null);
+        return importUnmanageInstanceCmd;
+    }
+
     @After
     public void tearDown() throws Exception {
         closeable.close();
@@ -457,9 +464,7 @@
 
     @Test
     public void importUnmanagedInstanceTest() {
-        ImportUnmanagedInstanceCmd importUnmanageInstanceCmd = Mockito.mock(ImportUnmanagedInstanceCmd.class);
-        when(importUnmanageInstanceCmd.getName()).thenReturn("TestInstance");
-        when(importUnmanageInstanceCmd.getDomainId()).thenReturn(null);
+        ImportUnmanagedInstanceCmd importUnmanageInstanceCmd = createMockImportUnmanagedInstanceCmd("TestInstance");
         when(volumeApiService.doesStoragePoolSupportDiskOffering(any(StoragePool.class), any())).thenReturn(true);
         try (MockedStatic<UsageEventUtils> ignored = Mockito.mockStatic(UsageEventUtils.class);
              MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
@@ -477,9 +482,7 @@
 
     @Test(expected = ServerApiException.class)
     public void importUnmanagedInstanceMissingInstanceTest() {
-        ImportUnmanagedInstanceCmd importUnmanageInstanceCmd = Mockito.mock(ImportUnmanagedInstanceCmd.class);
-        when(importUnmanageInstanceCmd.getName()).thenReturn("SomeInstance");
-        when(importUnmanageInstanceCmd.getDomainId()).thenReturn(null);
+        ImportUnmanagedInstanceCmd importUnmanageInstanceCmd = createMockImportUnmanagedInstanceCmd("SomeInstance");
         try (MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
             unmanagedVMsManager.importUnmanagedInstance(importUnmanageInstanceCmd);
         }
