--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -1915,8 +1915,7 @@
 
         VolumeObjectTO volumeObjectTO2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(22L).when(volumeObjectTO2).getVolumeId();
-        VolumeInfo volumeInfo2 = Mockito.mock(VolumeInfo.class);
-        doReturn(Mockito.mock(VolumeVO.class)).when(volumeInfo2).getVolume();
+        VolumeInfo volumeInfo2 = createMockVolumeInfo();
         doReturn(volumeInfo2).when(volumeDataFactoryMock).getVolume(22L);
 
         doNothing().when(kbossBackupProviderSpy).transitVmState(any(), any(), anyLong());
@@ -1941,8 +1940,7 @@
 
         VolumeObjectTO volumeObjectTO2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(22L).when(volumeObjectTO2).getVolumeId();
-        VolumeInfo volumeInfo2 = Mockito.mock(VolumeInfo.class);
-        doReturn(Mockito.mock(VolumeVO.class)).when(volumeInfo2).getVolume();
+        VolumeInfo volumeInfo2 = createMockVolumeInfo();
         doReturn(volumeInfo2).when(volumeDataFactoryMock).getVolume(22L);
 
         InternalBackupDataStoreVO delta = Mockito.mock(InternalBackupDataStoreVO.class);
@@ -3113,4 +3111,10 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+    private static VolumeInfo createMockVolumeInfo() {
+        VolumeInfo volumeInfo2 = Mockito.mock(VolumeInfo.class);
+        doReturn(Mockito.mock(VolumeVO.class)).when(volumeInfo2).getVolume();
+        return volumeInfo2;
+    }
+
 }
