--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/XenServerGuruTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/XenServerGuruTest.java
@@ -79,8 +79,7 @@
     public void beforeTest() {
         Mockito.when(sourceDataMock.getHypervisorType()).thenReturn(HypervisorType.XenServer);
 
-        Mockito.when(copyCommandMock.getSrcTO()).thenReturn(sourceDataMock);
-        Mockito.when(copyCommandMock.getDestTO()).thenReturn(destinationDataMock);
+        copyCommandMock = createMockCopyCommand(sourceDataMock, destinationDataMock);
 
         Mockito.when(changedHost.getId()).thenReturn(changedHostId);
         Mockito.lenient().when(defaultHost.getId()).thenReturn(defaultHostId);
@@ -97,6 +96,13 @@
         assertPairOfHostToExecuteCommandIsTheDefaultHostId(pairHostToExecuteCommand);
     }
 
+    private static CopyCommand createMockCopyCommand(DataTO sourceData, DataTO destinationData) {
+        CopyCommand copyCommandMock = Mockito.mock(CopyCommand.class);
+        Mockito.when(copyCommandMock.getSrcTO()).thenReturn(sourceData);
+        Mockito.when(copyCommandMock.getDestTO()).thenReturn(destinationData);
+        return copyCommandMock;
+    }
+
     private void assertPairOfHostToExecuteCommandIsTheDefaultHostId(Pair<Boolean, Long> pairHostToExecuteCommand) {
         Assert.assertFalse(pairHostToExecuteCommand.first());
         Assert.assertEquals(defaultHostId, pairHostToExecuteCommand.second());
@@ -104,11 +110,9 @@
 
     @Test
     public void getCommandHostDelegationTestCommandIsStorageSubSystemCommand() {
-        CopyCommand copyCommand = Mockito.mock(CopyCommand.class);
         DataTO srcData = Mockito.mock(DataTO.class);
         DataTO dstData = Mockito.mock(DataTO.class);
-        Mockito.when(copyCommand.getSrcTO()).thenReturn(srcData);
-        Mockito.when(copyCommand.getDestTO()).thenReturn(dstData);
+        CopyCommand copyCommand = createMockCopyCommand(srcData, dstData);
         Mockito.when(srcData.getHypervisorType()).thenReturn(HypervisorType.XenServer);
 
         Pair<Boolean, Long> pairHostToExecuteCommand = xenServerGuru.getCommandHostDelegation(defaultHostId, copyCommand);
