--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -267,6 +267,12 @@
         return webhook;
     }
 
+    private static ExecuteWebhookDeliveryCmd createMockExecuteWebhookDeliveryCmd() {
+        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(1L);
+        return cmd;
+    }
+
     @Test
     public void testValidateWebhookOwnerPayloadUrlUpdateSameExist() {
         WebhookVO webhookVO = mockWebhookVO(1L);
@@ -1189,8 +1195,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void executeWebhookDeliveryThrowsExceptionWhenBothDeliveryIdAndWebhookIdProvided() {
-        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
-        Mockito.when(cmd.getId()).thenReturn(1L);
+        ExecuteWebhookDeliveryCmd cmd = createMockExecuteWebhookDeliveryCmd();
         Mockito.when(cmd.getWebhookId()).thenReturn(2L);
 
         webhookApiServiceImpl.executeWebhookDelivery(cmd);
@@ -1198,8 +1203,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void executeWebhookDeliveryThrowsExceptionForInvalidDeliveryId() {
-        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
-        Mockito.when(cmd.getId()).thenReturn(1L);
+        ExecuteWebhookDeliveryCmd cmd = createMockExecuteWebhookDeliveryCmd();
 
         webhookApiServiceImpl.executeWebhookDelivery(cmd);
     }
