--- a/api/src/test/java/org/apache/cloudstack/api/command/test/RegionCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/RegionCmdTest.java
@@ -38,6 +38,7 @@
 
     private AddRegionCmd addRegionCmd;
     private ResponseGenerator responseGenerator;
+    private Region mockedRegion;
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
@@ -45,6 +46,8 @@
     @Override
     @Before
     public void setUp() {
+
+        mockedRegion = Mockito.mock(Region.class);
 
         addRegionCmd = new AddRegionCmd() {
 
@@ -66,16 +69,15 @@
 
         RegionService regionService = Mockito.mock(RegionService.class);
 
-        Region region = Mockito.mock(Region.class);
 
-        Mockito.when(regionService.addRegion(anyInt(), anyString(), isNull())).thenReturn(region);
+        Mockito.when(regionService.addRegion(anyInt(), anyString(), isNull())).thenReturn(mockedRegion);
 
         addRegionCmd._regionService = regionService;
         responseGenerator = Mockito.mock(ResponseGenerator.class);
 
         RegionResponse regionResponse = Mockito.mock(RegionResponse.class);
 
-        Mockito.when(responseGenerator.createRegionResponse(region)).thenReturn(regionResponse);
+        Mockito.when(responseGenerator.createRegionResponse(mockedRegion)).thenReturn(regionResponse);
 
         addRegionCmd._responseGenerator = responseGenerator;
         addRegionCmd.execute();
@@ -87,7 +89,6 @@
 
         RegionService regionService = Mockito.mock(RegionService.class);
 
-        Region region = Mockito.mock(Region.class);
 
         Mockito.when(regionService.addRegion(anyInt(), anyString(), isNull())).thenReturn(null);
 
