--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -179,6 +179,8 @@
     TungstenFabricLBHealthMonitorDao tungstenFabricLBHealthMonitorDao;
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
+    @Mock
+    private TungstenAnswer deleteVmAnswerMock;
 
     TungstenElement tungstenElement;
 
@@ -664,7 +666,6 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteVmiAnswer = mock(TungstenAnswer.class);
-        TungstenAnswer deleteVmAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteTungstenNetworkPolicyAnswer = mock(TungstenAnswer.class);
 
         when(nicProfile.getIPv4Address()).thenReturn("192.168.100.100");
@@ -675,7 +676,7 @@
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
-        when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswer);
+        when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswerMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(deleteTungstenNetworkPolicyAnswer);
 
         assertTrue(tungstenElement.release(network, nicProfile, virtualMachineProfile, reservationContext));
@@ -691,7 +692,6 @@
         HostVO host = mock(HostVO.class);
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteVmiAnswer = mock(TungstenAnswer.class);
-        TungstenAnswer deleteVmAnswer = mock(TungstenAnswer.class);
 
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
@@ -699,7 +699,7 @@
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
-        when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswer);
+        when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenReturn(deleteVmAnswerMock);
 
         assertTrue(tungstenElement.release(network, nicProfile, virtualMachineProfile, reservationContext));
     }
