--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -196,6 +196,13 @@
         return cmd;
     }
 
+    private static DestroySharedFSCmd createMockDestroySharedFSCmd() {
+        DestroySharedFSCmd cmd = mock(DestroySharedFSCmd.class);
+        when(cmd.getId()).thenReturn(s_sharedFSId);
+        when(cmd.isExpunge()).thenReturn(false);
+        return cmd;
+    }
+
     private SharedFSVO getMockSharedFS() {
         SharedFSVO sharedFS = new SharedFSVO(s_name, s_description, s_domainId, s_ownerId, s_zoneId,
                 s_providerName, SharedFS.Protocol.NFS, SharedFS.FileSystemType.valueOf(s_fsFormat), s_serviceOfferingId);
@@ -506,9 +513,7 @@
 
     @Test
     public void testDestroySharedFS() throws NoTransitionException {
-        DestroySharedFSCmd cmd = mock(DestroySharedFSCmd.class);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
-        when(cmd.isExpunge()).thenReturn(false);
+        DestroySharedFSCmd cmd = createMockDestroySharedFSCmd();
 
         SharedFSVO sharedFS = getMockSharedFS();
         when(sharedFSDao.findById(s_sharedFSId)).thenReturn(sharedFS);
@@ -521,9 +526,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testDestroySharedFSInvalidState() {
-        DestroySharedFSCmd cmd = mock(DestroySharedFSCmd.class);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
-        when(cmd.isExpunge()).thenReturn(false);
+        DestroySharedFSCmd cmd = createMockDestroySharedFSCmd();
         when(cmd.isForced()).thenReturn(false);
 
         SharedFSVO sharedFS = getMockSharedFS();
