--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -356,8 +356,7 @@
 
         StoragePoolVO mockPool = mock(StoragePoolVO.class);
 
-        PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
-        when(mockPrimaryDataStore.getId()).thenReturn(2l);
+        PrimaryDataStore mockPrimaryDataStore = createMockPrimaryDataStore();
 
         VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
         when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
@@ -378,8 +377,7 @@
 
         StoragePoolVO mockPool = mock(StoragePoolVO.class);
 
-        PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
-        when(mockPrimaryDataStore.getId()).thenReturn(2l);
+        PrimaryDataStore mockPrimaryDataStore = createMockPrimaryDataStore();
         when(mockPrimaryDataStore.getDataCenterId()).thenReturn(1l);
 
         when(vmTemplateDao.findById(anyLong(), anyBoolean())).thenReturn(mockTemplate);
@@ -421,10 +419,9 @@
     public void testTemplateScheduledForDownloadInOnePool() {
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         StoragePoolVO mockPool = mock(StoragePoolVO.class);
-        PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
+        PrimaryDataStore mockPrimaryDataStore = createMockPrimaryDataStore();
         VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
 
-        when(mockPrimaryDataStore.getId()).thenReturn(2l);
         when(mockPool.getId()).thenReturn(2l);
         when(mockPool.getStatus()).thenReturn(StoragePoolStatus.Up);
         when(mockPool.getDataCenterId()).thenReturn(1l);
@@ -468,7 +465,7 @@
     @Test
     public void testTemplateScheduledForDownloadInMultiplePool() {
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
-        PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
+        PrimaryDataStore mockPrimaryDataStore = createMockPrimaryDataStore();
         VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
         List<StoragePoolVO> pools = new ArrayList<StoragePoolVO>();
 
@@ -483,7 +480,6 @@
         pools.add(mockPool2);
         pools.add(mockPool3);
 
-        when(mockPrimaryDataStore.getId()).thenReturn(2l);
         when(mockTemplate.getId()).thenReturn(202l);
         when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
         when(vmTemplateDao.findById(anyLong())).thenReturn(mockTemplate);
@@ -669,6 +665,12 @@
         Mockito.verify(dataStoreManager, Mockito.times(0)).getImageStoreWithFreeCapacity(Mockito.anyLong());
     }
 
+    private static PrimaryDataStore createMockPrimaryDataStore() {
+        PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
+        when(mockPrimaryDataStore.getId()).thenReturn(2l);
+        return mockPrimaryDataStore;
+    }
+
     private HypervisorTemplateAdapter mockHypervisorTemplateAdapter() {
         HypervisorTemplateAdapter adapter = Mockito.mock(HypervisorTemplateAdapter.class);
         when(adapter.getName()).thenReturn(TemplateAdapter.TemplateAdapterType.Hypervisor.getName());
