--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -209,6 +209,9 @@
     private VolumeObjectTO volumeObjectToMock;
 
     @Mock
+    private VolumeObjectTO volumeObjectTOForBackup;
+
+    @Mock
     private VirtualMachineManager virtualMachineManagerMock;
 
     @Mock
@@ -780,9 +783,8 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(volumeObjectTOForBackup, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
@@ -805,9 +807,8 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(volumeObjectTOForBackup, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
@@ -838,9 +839,8 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(volumeObjectTOForBackup, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
