--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -267,6 +267,13 @@
         return webhook;
     }
 
+    private static WebhookJoinVO createMockWebhookJoinVO(Webhook.Scope scope) {
+        WebhookJoinVO webhook = Mockito.mock(WebhookJoinVO.class);
+        Mockito.when(webhook.getState()).thenReturn(Webhook.State.Enabled);
+        Mockito.when(webhook.getScope()).thenReturn(scope);
+        return webhook;
+    }
+
     @Test
     public void testValidateWebhookOwnerPayloadUrlUpdateSameExist() {
         WebhookVO webhookVO = mockWebhookVO(1L);
@@ -815,9 +822,7 @@
     public void listWebhooksFiltersByValidScopeAndState() {
         ListWebhooksCmd cmd = Mockito.mock(ListWebhooksCmd.class);
         Mockito.when(accountManager.isAdmin(Mockito.anyLong())).thenReturn(true);
-        WebhookJoinVO webhook = Mockito.mock(WebhookJoinVO.class);
-        Mockito.when(webhook.getState()).thenReturn(Webhook.State.Enabled);
-        Mockito.when(webhook.getScope()).thenReturn(Webhook.Scope.Domain);
+        WebhookJoinVO webhook = createMockWebhookJoinVO(Webhook.Scope.Domain);
 
         Mockito.when(cmd.getScope()).thenReturn("Domain");
         Mockito.when(cmd.getState()).thenReturn("Enabled");
@@ -849,9 +854,7 @@
     public void createWebhookCreatesWebhookSuccessfully() {
         CreateWebhookCmd cmd = Mockito.mock(CreateWebhookCmd.class);
         WebhookVO webhook = Mockito.mock(WebhookVO.class);
-        WebhookJoinVO webhookJoinVO = Mockito.mock(WebhookJoinVO.class);
-        Mockito.when(webhookJoinVO.getState()).thenReturn(Webhook.State.Enabled);
-        Mockito.when(webhookJoinVO.getScope()).thenReturn(Webhook.Scope.Local);
+        WebhookJoinVO webhookJoinVO = createMockWebhookJoinVO(Webhook.Scope.Local);
 
         Mockito.when(cmd.getName()).thenReturn("webhookName");
         Mockito.when(cmd.getDescription()).thenReturn("webhookDescription");
