--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -1042,6 +1042,12 @@
         Mockito.verify(volumeServiceMock, Mockito.times(1)).destroyVolume(volumeMockId);
     }
 
+    private static VolumeInfo createMockVolumeInfo(DataStore dataStore) {
+        VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
+        Mockito.doReturn(dataStore).when(volumeInfo).getDataStore();
+        return volumeInfo;
+    }
+
     private void verifyMocksForTestDestroyVolumeWhenVolumeIsNotInRightState() {
         Mockito.verify(volumeServiceMock, Mockito.times(0)).destroyVolume(volumeMockId);
         Mockito.verify(resourceLimitServiceMock, Mockito.times(0)).decrementVolumeResourceCount(accountMockId, true, volumeSizeMock, newDiskOfferingMock, null);
@@ -1159,13 +1165,10 @@
     @Test
     public void cleanVolumesCacheTest() {
         List<VolumeInfo> volumeInfos = new ArrayList<>();
-        VolumeInfo volumeInfoMock1 = Mockito.mock(VolumeInfo.class);
-        VolumeInfo volumeInfoMock2 = Mockito.mock(VolumeInfo.class);
-
         DataStore dataStoreMock1 = Mockito.mock(DataStore.class);
         DataStore dataStoreMock2 = Mockito.mock(DataStore.class);
-        Mockito.doReturn(dataStoreMock1).when(volumeInfoMock1).getDataStore();
-        Mockito.doReturn(dataStoreMock2).when(volumeInfoMock2).getDataStore();
+        VolumeInfo volumeInfoMock1 = createMockVolumeInfo(dataStoreMock1);
+        VolumeInfo volumeInfoMock2 = createMockVolumeInfo(dataStoreMock2);
 
         volumeInfos.add(volumeInfoMock1);
         volumeInfos.add(volumeInfoMock2);
