--- a/api/src/test/java/org/apache/cloudstack/api/command/test/ListCfgCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/ListCfgCmdTest.java
@@ -44,7 +44,7 @@
     @Override
     @Before
     public void setUp() {
-        responseGenerator = Mockito.mock(ResponseGenerator.class);
+        responseGenerator = MockResponseGenerator.createMockResponseGenerator();
         mgr = Mockito.mock(ManagementService.class);
         listCfgsByCmd = new ListCfgsByCmd();
     }
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/UpdateCfgCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/UpdateCfgCmdTest.java
@@ -42,7 +42,7 @@
     @Override
     @Before
     public void setUp() {
-        responseGenerator = Mockito.mock(ResponseGenerator.class);
+        responseGenerator = MockResponseGenerator.createMockResponseGenerator();
         configService = Mockito.mock(ConfigurationService.class);
         updateCfgCmd = new UpdateCfgCmd();
     }
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/MockResponseGenerator.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/MockResponseGenerator.java
@@ -0,0 +1,10 @@
+package org.apache.cloudstack.api.command.test;
+
+import org.apache.cloudstack.api.ResponseGenerator;
+import org.mockito.Mockito;
+
+public class MockResponseGenerator {
+    public static ResponseGenerator createMockResponseGenerator() {
+        return Mockito.mock(ResponseGenerator.class);
+    }
+}
