--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -222,6 +222,8 @@
 
     @Mock
     private VpcOfferingServiceMapDao vpcOfferingServiceMapDao;
+    @Mock
+    private PrepareExtensionPathCommand prepareExtensionPathCommand;
 
     @Before
     public void setUp() {
@@ -254,7 +256,7 @@
     @Test
     public void getResultFromAnswersStringReturnsSuccess() {
         Extension ext = mock(Extension.class);
-        Answer[] answers = new Answer[]{new Answer(mock(PrepareExtensionPathCommand.class), true, "ok")};
+        Answer[] answers = new Answer[]{new Answer(prepareExtensionPathCommand, true, "ok")};
         String json = GsonHelper.getGson().toJson(answers);
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Pair<Boolean, String> result = extensionsManager.getResultFromAnswersString(json, ext, msHost, "op");
@@ -265,7 +267,7 @@
     @Test
     public void getResultFromAnswersStringReturnsFailure() {
         Extension ext = mock(Extension.class);
-        Answer[] answers = new Answer[]{new Answer(mock(PrepareExtensionPathCommand.class), false, "fail")};
+        Answer[] answers = new Answer[]{new Answer(prepareExtensionPathCommand, false, "fail")};
         String json = GsonHelper.getGson().toJson(answers);
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Pair<Boolean, String> result = extensionsManager.getResultFromAnswersString(json, ext, msHost, "op");
