--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -1267,7 +1267,6 @@
     @Test
     public void createTungstenServiceGroupTest() {
         TungstenAnswer createTungstenServiceGroupAnswer = MockTungstenAnswerFactory.get(true);
-        ServiceGroup serviceGroup = mock(ServiceGroup.class);
         FirewallServiceGroupType firewallServiceGroupType = mock(FirewallServiceGroupType.class);
         FirewallServiceType firewallServiceType = mock(FirewallServiceType.class);
         PortType portType = mock(PortType.class);
@@ -1275,8 +1274,8 @@
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenServiceGroupCommand.class), anyLong())).thenReturn(createTungstenServiceGroupAnswer);
+        ServiceGroup serviceGroup = createMockServiceGroup(firewallServiceGroupType);
         when(createTungstenServiceGroupAnswer.getApiObjectBase()).thenReturn(serviceGroup);
-        when(serviceGroup.getFirewallServiceList()).thenReturn(firewallServiceGroupType);
         when(firewallServiceGroupType.getFirewallService()).thenReturn(List.of(firewallServiceType));
         when(firewallServiceType.getDstPorts()).thenReturn(portType);
 
@@ -1433,7 +1432,6 @@
     @Test
     public void listTungstenServiceGroupTest() {
         TungstenAnswer listTungstenServiceGroupAnswer = MockTungstenAnswerFactory.get(true);
-        ServiceGroup serviceGroup = mock(ServiceGroup.class);
         FirewallServiceGroupType firewallServiceGroupType = mock(FirewallServiceGroupType.class);
         FirewallServiceType firewallServiceType = mock(FirewallServiceType.class);
         PortType portType = mock(PortType.class);
@@ -1441,8 +1439,8 @@
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenServiceGroupCommand.class), anyLong())).thenReturn(listTungstenServiceGroupAnswer);
+        ServiceGroup serviceGroup = createMockServiceGroup(firewallServiceGroupType);
         doReturn(List.of(serviceGroup)).when(listTungstenServiceGroupAnswer).getApiObjectBaseList();
-        when(serviceGroup.getFirewallServiceList()).thenReturn(firewallServiceGroupType);
         when(firewallServiceGroupType.getFirewallService()).thenReturn(List.of(firewallServiceType));
         when(firewallServiceType.getDstPorts()).thenReturn(portType);
 
@@ -1668,4 +1666,10 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static ServiceGroup createMockServiceGroup(FirewallServiceGroupType firewallServiceGroupType) {
+        ServiceGroup serviceGroup = mock(ServiceGroup.class);
+        when(serviceGroup.getFirewallServiceList()).thenReturn(firewallServiceGroupType);
+        return serviceGroup;
+    }
 }
