--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
@@ -116,6 +116,9 @@
     @Mock
     DiskOfferingDao diskOfferingDao;
 
+    @Mock
+    private ServiceContent serviceContent;
+
     AutoCloseable closeable;
 
     @Mock
@@ -230,7 +233,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -267,7 +269,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -311,7 +312,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -358,7 +358,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -412,7 +411,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -474,7 +472,6 @@
 
         DataStoreTO dataStore = Mockito.mock(DataStoreTO.class);
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -511,7 +508,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -554,7 +550,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -592,7 +587,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
