--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
@@ -87,6 +87,8 @@
     private Host hostMock;
     @Mock
     private Record hostRecordMock;
+    @Mock
+    private SR storageRepositoryMock;
 
     private String hostUuidMock = "hostUuidMock";
 
@@ -227,7 +229,6 @@
 
         Connection connectionMock = Mockito.mock(Connection.class);
 
-        SR srExtShared = Mockito.mock(SR.class);
         SR srExtNonShared = Mockito.mock(SR.class);
 
         List<SR> expectedListOfSrs = new ArrayList<>();
@@ -253,7 +254,7 @@
         srExtNonSharedRecord.PBDs = pbds;
 
         Map<SR, SR.Record> mapOfSrsRecords = new HashMap<>();
-        mapOfSrsRecords.put(srExtShared, srExtSharedRecord);
+        mapOfSrsRecords.put(storageRepositoryMock, srExtSharedRecord);
         mapOfSrsRecords.put(srExtNonShared, srExtNonSharedRecord);
         try (MockedStatic<SR> ignored = Mockito.mockStatic(SR.class)) {
             BDDMockito.given(SR.getAllRecords(connectionMock)).willReturn(mapOfSrsRecords);
@@ -275,9 +276,8 @@
     @Test
     public void getAllLocalSrsTest() throws XenAPIException, XmlRpcException {
         Connection connectionMock = Mockito.mock(Connection.class);
-        SR sr1 = Mockito.mock(SR.class);
         List<SR> srsExt = new ArrayList<>();
-        srsExt.add(sr1);
+        srsExt.add(storageRepositoryMock);
 
         SR sr2 = Mockito.mock(SR.class);
         List<SR> srsLvm = new ArrayList<>();
@@ -364,14 +364,13 @@
         citrixResourceBase._dcId = 1;
 
         Connection connectionMock = Mockito.mock(Connection.class);
-        SR srMock = Mockito.mock(SR.class);
 
         StoragePoolInfo storagePoolInfoMock = Mockito.mock(StoragePoolInfo.class);
 
-        Mockito.doNothing().when(citrixResourceBase).configureStorageNameAndDescription(connectionMock, srMock);
-        Mockito.doReturn(storagePoolInfoMock).when(citrixResourceBase).createStoragePoolInfo(connectionMock, srMock);
-
-        StartupStorageCommand startUpStorageCommand = citrixResourceBase.createStartUpStorageCommand(connectionMock, srMock);
+        Mockito.doNothing().when(citrixResourceBase).configureStorageNameAndDescription(connectionMock, storageRepositoryMock);
+        Mockito.doReturn(storagePoolInfoMock).when(citrixResourceBase).createStoragePoolInfo(connectionMock, storageRepositoryMock);
+
+        StartupStorageCommand startUpStorageCommand = citrixResourceBase.createStartUpStorageCommand(connectionMock, storageRepositoryMock);
 
         Assert.assertEquals(hostUuid, startUpStorageCommand.getGuid());
         Assert.assertEquals(storagePoolInfoMock, startUpStorageCommand.getPoolInfo());
@@ -488,7 +487,6 @@
 
     @Test
     public void testListFilesAtPath() throws IOException, XmlRpcException {
-        SR srMock = Mockito.mock(SR.class);
         SFTPv3FileAttributes fileAttributesMock = Mockito.mock(SFTPv3FileAttributes.class);
         SFTPv3DirectoryEntry directoryEntryMock = Mockito.mock(SFTPv3DirectoryEntry.class);
         Vector fileListMock = new Vector();
@@ -505,7 +503,7 @@
         Mockito.when(command.getPath()).thenReturn("/path/to/files");
 
         Mockito.doReturn(connectionMock).when(citrixResourceBase).getConnection();
-        Mockito.doReturn(srMock).when(citrixResourceBase).getStorageRepository(connectionMock, "storeUuid");
+        Mockito.doReturn(storageRepositoryMock).when(citrixResourceBase).getStorageRepository(connectionMock, "storeUuid");
         ReflectionTestUtils.setField(directoryEntryMock, "filename", "file1");
         ReflectionTestUtils.setField(directoryEntryMock, "attributes", fileAttributesMock);
         Mockito.when(fileAttributesMock.isDirectory()).thenReturn(false);
@@ -538,7 +536,6 @@
     @Test
     public void testListFilesAtPathWithNonExistentPath() throws IOException, XmlRpcException {
         Connection connectionMock = Mockito.mock(Connection.class);
-        SR srMock = Mockito.mock(SR.class);
 
         ListDataStoreObjectsCommand command = Mockito.mock(ListDataStoreObjectsCommand.class);
         DataStoreTO store = Mockito.mock(DataStoreTO.class);
@@ -549,7 +546,7 @@
         Mockito.when(command.getPath()).thenReturn("/path/to/non/existent/files");
 
         Mockito.doReturn(connectionMock).when(citrixResourceBase).getConnection();
-        Mockito.doReturn(srMock).when(citrixResourceBase).getStorageRepository(connectionMock, "storeUuid");
+        Mockito.doReturn(storageRepositoryMock).when(citrixResourceBase).getStorageRepository(connectionMock, "storeUuid");
 
         Answer answer;
         try (MockedConstruction<com.trilead.ssh2.Connection> ignored = Mockito.mockConstruction(com.trilead.ssh2.Connection.class, (mock, context) -> {
@@ -576,7 +573,6 @@
     @Test
     public void testListFilesAtPathWithFile() throws IOException, XmlRpcException {
         Connection connectionMock = Mockito.mock(Connection.class);
-        SR srMock = Mockito.mock(SR.class);
         SFTPv3FileAttributes fileAttributesMock = Mockito.mock(SFTPv3FileAttributes.class);
         Vector fileListMock = new Vector();
         fileListMock.add(fileAttributesMock);
@@ -590,7 +586,7 @@
         Mockito.when(command.getPath()).thenReturn("/path/to/file");
 
         Mockito.doReturn(connectionMock).when(citrixResourceBase).getConnection();
-        Mockito.doReturn(srMock).when(citrixResourceBase).getStorageRepository(connectionMock, "storeUuid");
+        Mockito.doReturn(storageRepositoryMock).when(citrixResourceBase).getStorageRepository(connectionMock, "storeUuid");
         ReflectionTestUtils.setField(fileAttributesMock, "size", 1024L);
         ReflectionTestUtils.setField(fileAttributesMock, "mtime", 123456789L);
 
