--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
@@ -301,7 +301,6 @@
     public void createStoragePoolInfoTest() throws XenAPIException, XmlRpcException {
         Connection connectionMock = Mockito.mock(Connection.class);
         Host hostMock = Mockito.mock(Host.class);
-        SR srMock = Mockito.mock(SR.class);
 
         String hostAddress = "hostAddress";
         Mockito.when(hostMock.getAddress(connectionMock)).thenReturn(hostAddress);
@@ -315,10 +314,9 @@
         long srPhysicalSize = 100l;
         long physicalUtilization = 10l;
 
+        SR srMock = createMockSr(connectionMock);
         Mockito.when(srMock.getPhysicalSize(connectionMock)).thenReturn(srPhysicalSize);
-        Mockito.when(srMock.getUuid(connectionMock)).thenReturn(srUuid);
         Mockito.when(srMock.getPhysicalUtilisation(connectionMock)).thenReturn(physicalUtilization);
-        Mockito.when(srMock.getType(connectionMock)).thenReturn(srType);
 
         StoragePoolInfo storagePoolInfo = citrixResourceBase.createStoragePoolInfo(connectionMock, srMock);
 
@@ -339,14 +337,12 @@
         citrixResourceBase._host.setUuid(hostUuid);
 
         Connection connectionMock = Mockito.mock(Connection.class);
-        SR srMock = Mockito.mock(SR.class);
 
         String srUuid = "srUuid";
         String srType = "ext";
         String expectedNameDescription = String.format(nameFormat, srType, hostUuid);
 
-        Mockito.when(srMock.getUuid(connectionMock)).thenReturn(srUuid);
-        Mockito.when(srMock.getType(connectionMock)).thenReturn(srType);
+        SR srMock = createMockSr(connectionMock);
 
         Mockito.doNothing().when(srMock).setNameLabel(connectionMock, srUuid);
         Mockito.doNothing().when(srMock).setNameDescription(connectionMock, expectedNameDescription);
@@ -616,4 +612,11 @@
         Assert.assertEquals(1024L, listAnswer.getSizes().get(0).longValue());
         Assert.assertEquals(123456789000L, listAnswer.getLastModified().get(0).longValue());
     }
+
+    private static SR createMockSr(Connection connectionMock) throws XenAPIException, XmlRpcException {
+        SR srMock = Mockito.mock(SR.class);
+        Mockito.when(srMock.getUuid(connectionMock)).thenReturn("srUuid");
+        Mockito.when(srMock.getType(connectionMock)).thenReturn("ext");
+        return srMock;
+    }
 }
