--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
@@ -62,6 +62,9 @@
     @Mock
     private Connection connectionMock;
 
+    @Mock
+    private SR sharedSrMock;
+
     private String pathMock = "pathMock";
 
     @Before
@@ -107,9 +110,8 @@
 
     @Test
     public void createFileSRTestSrAlreadyConfigured() {
-        SR srMockRetrievedMethod = Mockito.mock(SR.class);
-
-        Mockito.doReturn(srMockRetrievedMethod).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
+
+        Mockito.doReturn(sharedSrMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
 
         SR methodCreateFileSrResult = xenserver625StorageProcessor.createFileSR(connectionMock, pathMock);
 
@@ -117,15 +119,14 @@
         inOrder.verify(xenserver625StorageProcessor, times(1)).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
         inOrder.verify(xenserver625StorageProcessor, times(0)).createNewFileSr(connectionMock, pathMock);
 
-        Assert.assertEquals(srMockRetrievedMethod, methodCreateFileSrResult);
+        Assert.assertEquals(sharedSrMock, methodCreateFileSrResult);
     }
 
     @Test
     public void createFileSRTestSrNotConfiguredAlreadyCreatingSr() {
-        SR srMockCreateMethod = Mockito.mock(SR.class);
 
         Mockito.doReturn(null).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
-        Mockito.doReturn(srMockCreateMethod).when(xenserver625StorageProcessor).createNewFileSr(connectionMock, pathMock);
+        Mockito.doReturn(sharedSrMock).when(xenserver625StorageProcessor).createNewFileSr(connectionMock, pathMock);
 
         SR methodCreateFileSrResult = xenserver625StorageProcessor.createFileSR(connectionMock, pathMock);
 
@@ -133,7 +134,7 @@
         inOrder.verify(xenserver625StorageProcessor, times(1)).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
         inOrder.verify(xenserver625StorageProcessor, times(1)).createNewFileSr(connectionMock, pathMock);
 
-        Assert.assertEquals(srMockCreateMethod, methodCreateFileSrResult);
+        Assert.assertEquals(sharedSrMock, methodCreateFileSrResult);
     }
 
     @Test(expected = CloudRuntimeException.class)
@@ -159,13 +160,12 @@
 
     @Test
     public void retrieveAlreadyConfiguredSrWithoutExceptionMethodWorking() throws XenAPIException, XmlRpcException {
-        SR srMock = Mockito.mock(SR.class);
-        Mockito.doReturn(srMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSr(connectionMock, pathMock);
+        Mockito.doReturn(sharedSrMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSr(connectionMock, pathMock);
 
         SR sr = xenserver625StorageProcessor.retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
 
         Mockito.verify(xenserver625StorageProcessor).retrieveAlreadyConfiguredSr(connectionMock, pathMock);
-        Assert.assertEquals(srMock, sr);
+        Assert.assertEquals(sharedSrMock, sr);
     }
 
     @Test
@@ -335,8 +335,7 @@
         String uuid = "hostUuid";
         Mockito.when(citrixResourceBase._host.getUuid()).thenReturn(uuid);
 
-        SR srMock = Mockito.mock(SR.class);
-        Mockito.doReturn(srMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
+        Mockito.doReturn(sharedSrMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
         Host hostMock = Mockito.mock(Host.class);
@@ -354,7 +353,7 @@
 
             SR sr = xenserver625StorageProcessor.createNewFileSr(connectionMock, pathMock);
 
-            Assert.assertEquals(srMock, sr);
+            Assert.assertEquals(sharedSrMock, sr);
             Mockito.verify(xenserver625StorageProcessor, times(0)).removeSrAndPbdIfPossible(Mockito.eq(connectionMock),
                     Mockito.any(SR.class), Mockito.any(PBD.class));
             Mockito.verify(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock,
@@ -367,7 +366,6 @@
         String uuid = "hostUuid";
         Mockito.when(citrixResourceBase._host.getUuid()).thenReturn(uuid);
 
-        SR srMock = Mockito.mock(SR.class);
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
         Host hostMock = Mockito.mock(Host.class);
@@ -379,20 +377,20 @@
 
             Mockito.when(SR.introduce(Mockito.eq(connectionMock), Mockito.eq(srUuid), Mockito.eq(pathMock),
                     Mockito.eq(pathMock), Mockito.eq("file"), Mockito.eq("file"), Mockito.eq(false),
-                    Mockito.anyMap())).thenReturn(srMock);
+                    Mockito.anyMap())).thenReturn(sharedSrMock);
 
             PBD pbdMock = Mockito.mock(PBD.class);
             Mockito.when(PBD.create(Mockito.eq(connectionMock), Mockito.any(Record.class))).thenReturn(pbdMock);
 
             SR sr = xenserver625StorageProcessor.createNewFileSr(connectionMock, pathMock);
 
-            Assert.assertEquals(srMock, sr);
+            Assert.assertEquals(sharedSrMock, sr);
             Mockito.verify(xenserver625StorageProcessor, times(0)).removeSrAndPbdIfPossible(Mockito.eq(connectionMock),
                     Mockito.any(SR.class), Mockito.any(PBD.class));
             Mockito.verify(xenserver625StorageProcessor, times(0)).retrieveAlreadyConfiguredSrWithoutException(
                     connectionMock, pathMock);
 
-            Mockito.verify(srMock).scan(connectionMock);
+            Mockito.verify(sharedSrMock).scan(connectionMock);
             Mockito.verify(pbdMock).plug(connectionMock);
 
             SR.introduce(Mockito.eq(connectionMock), Mockito.eq(srUuid), Mockito.eq(pathMock), Mockito.eq(pathMock),
@@ -405,27 +403,25 @@
 
     @Test
     public void removeSrAndPbdIfPossibleBothPbdAndSrNotNull() {
-        SR srMock = Mockito.mock(SR.class);
-        Mockito.doNothing().when(xenserver625StorageProcessor).forgetSr(connectionMock, srMock);
+        Mockito.doNothing().when(xenserver625StorageProcessor).forgetSr(connectionMock, sharedSrMock);
 
         PBD pbdMock = Mockito.mock(PBD.class);
         Mockito.doNothing().when(xenserver625StorageProcessor).unplugPbd(connectionMock, pbdMock);
 
-        xenserver625StorageProcessor.removeSrAndPbdIfPossible(connectionMock, srMock, pbdMock);
-
-        Mockito.verify(xenserver625StorageProcessor).forgetSr(connectionMock, srMock);
+        xenserver625StorageProcessor.removeSrAndPbdIfPossible(connectionMock, sharedSrMock, pbdMock);
+
+        Mockito.verify(xenserver625StorageProcessor).forgetSr(connectionMock, sharedSrMock);
         Mockito.verify(xenserver625StorageProcessor).unplugPbd(connectionMock, pbdMock);
 
     }
 
     @Test
     public void removeSrAndPbdIfPossiblePbdNullAndSrNotNull() {
-        SR srMock = Mockito.mock(SR.class);
-        Mockito.doNothing().when(xenserver625StorageProcessor).forgetSr(connectionMock, srMock);
-
-        xenserver625StorageProcessor.removeSrAndPbdIfPossible(connectionMock, srMock, null);
-
-        Mockito.verify(xenserver625StorageProcessor).forgetSr(connectionMock, srMock);
+        Mockito.doNothing().when(xenserver625StorageProcessor).forgetSr(connectionMock, sharedSrMock);
+
+        xenserver625StorageProcessor.removeSrAndPbdIfPossible(connectionMock, sharedSrMock, null);
+
+        Mockito.verify(xenserver625StorageProcessor).forgetSr(connectionMock, sharedSrMock);
         Mockito.verify(xenserver625StorageProcessor, times(0)).unplugPbd(Mockito.eq(connectionMock), Mockito.any(PBD.class));
 
     }
