--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -70,6 +70,8 @@
     @Mock
     protected XenServer610Resource xenServer610Resource;
 
+    private final SR sr2Mock = Mockito.mock(SR.class);
+
     @Test
     public void testCheckNetworkCommandFailure() {
         final XenServer610Resource xenServer610Resource = new XenServer610Resource();
@@ -263,7 +265,6 @@
         final VolumeTO volume2 = MockVolumeTO(path);
 
         final SR sr1 = Mockito.mock(SR.class);
-        final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
         final VDI vdi2 = Mockito.mock(VDI.class);
@@ -276,7 +277,7 @@
 
         final List<Pair<VolumeTO, Object>> volumeToSr = new ArrayList<Pair<VolumeTO, Object>>();
         volumeToSr.add(new Pair<VolumeTO, Object>(volume1, sr1));
-        volumeToSr.add(new Pair<VolumeTO, Object>(volume2, sr2));
+        volumeToSr.add(new Pair<VolumeTO, Object>(volume2, sr2Mock));
 
         final List<Pair<NicTO, Object>> nicToNetwork = new ArrayList<Pair<NicTO, Object>>();
         nicToNetwork.add(new Pair<NicTO, Object>(nic1, network1));
@@ -370,7 +371,6 @@
         final VolumeTO volume2 = MockVolumeTO(path);
 
         final SR sr1 = Mockito.mock(SR.class);
-        final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
         final VDI vdi2 = Mockito.mock(VDI.class);
@@ -380,7 +380,7 @@
 
         final List<Pair<VolumeTO, Object>> volumeToSr = new ArrayList<Pair<VolumeTO, Object>>();
         volumeToSr.add(new Pair<VolumeTO, Object>(volume1, sr1));
-        volumeToSr.add(new Pair<VolumeTO, Object>(volume2, sr2));
+        volumeToSr.add(new Pair<VolumeTO, Object>(volume2, sr2Mock));
 
         final List<Pair<NicTO, Object>> nicToNetwork = new ArrayList<Pair<NicTO, Object>>();
         nicToNetwork.add(new Pair<NicTO, Object>(nic1, new String("a")));
