--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -70,6 +70,9 @@
     @Mock
     protected XenServer610Resource xenServer610Resource;
 
+    @Mock
+    private SR srMock;
+
     @Test
     public void testCheckNetworkCommandFailure() {
         final XenServer610Resource xenServer610Resource = new XenServer610Resource();
@@ -262,7 +265,6 @@
         final VolumeTO volume1 = MockVolumeTO(path);
         final VolumeTO volume2 = MockVolumeTO(path);
 
-        final SR sr1 = Mockito.mock(SR.class);
         final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
@@ -275,7 +277,7 @@
         final Network network2 = Mockito.mock(Network.class);
 
         final List<Pair<VolumeTO, Object>> volumeToSr = new ArrayList<Pair<VolumeTO, Object>>();
-        volumeToSr.add(new Pair<VolumeTO, Object>(volume1, sr1));
+        volumeToSr.add(new Pair<VolumeTO, Object>(volume1, srMock));
         volumeToSr.add(new Pair<VolumeTO, Object>(volume2, sr2));
 
         final List<Pair<NicTO, Object>> nicToNetwork = new ArrayList<Pair<NicTO, Object>>();
@@ -369,7 +371,6 @@
         final VolumeTO volume1 = MockVolumeTO(path);
         final VolumeTO volume2 = MockVolumeTO(path);
 
-        final SR sr1 = Mockito.mock(SR.class);
         final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
@@ -379,7 +380,7 @@
         final NicTO nic2 = Mockito.mock(NicTO.class);
 
         final List<Pair<VolumeTO, Object>> volumeToSr = new ArrayList<Pair<VolumeTO, Object>>();
-        volumeToSr.add(new Pair<VolumeTO, Object>(volume1, sr1));
+        volumeToSr.add(new Pair<VolumeTO, Object>(volume1, srMock));
         volumeToSr.add(new Pair<VolumeTO, Object>(volume2, sr2));
 
         final List<Pair<NicTO, Object>> nicToNetwork = new ArrayList<Pair<NicTO, Object>>();
