--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
@@ -87,6 +87,8 @@
     private Host hostMock;
     @Mock
     private Record hostRecordMock;
+    @Mock
+    private SR sr2Mock;
 
     private String hostUuidMock = "hostUuidMock";
 
@@ -228,10 +230,9 @@
         Connection connectionMock = Mockito.mock(Connection.class);
 
         SR srExtShared = Mockito.mock(SR.class);
-        SR srExtNonShared = Mockito.mock(SR.class);
 
         List<SR> expectedListOfSrs = new ArrayList<>();
-        expectedListOfSrs.add(srExtNonShared);
+        expectedListOfSrs.add(sr2Mock);
 
         Set<PBD> pbds = new HashSet<>();
         PBD pbdMock = Mockito.mock(PBD.class);
@@ -254,7 +255,7 @@
 
         Map<SR, SR.Record> mapOfSrsRecords = new HashMap<>();
         mapOfSrsRecords.put(srExtShared, srExtSharedRecord);
-        mapOfSrsRecords.put(srExtNonShared, srExtNonSharedRecord);
+        mapOfSrsRecords.put(sr2Mock, srExtNonSharedRecord);
         try (MockedStatic<SR> ignored = Mockito.mockStatic(SR.class)) {
             BDDMockito.given(SR.getAllRecords(connectionMock)).willReturn(mapOfSrsRecords);
 
@@ -279,9 +280,8 @@
         List<SR> srsExt = new ArrayList<>();
         srsExt.add(sr1);
 
-        SR sr2 = Mockito.mock(SR.class);
         List<SR> srsLvm = new ArrayList<>();
-        srsLvm.add(sr2);
+        srsLvm.add(sr2Mock);
 
         Mockito.doReturn(srsExt).when(citrixResourceBase).getAllLocalSrForType(connectionMock, SRType.EXT);
         Mockito.doReturn(srsLvm).when(citrixResourceBase).getAllLocalSrForType(connectionMock, SRType.LVM);
