--- a/services/console-proxy/rdpconsole/src/test/java/streamer/BaseElementTest.java
+++ b/services/console-proxy/rdpconsole/src/test/java/streamer/BaseElementTest.java
@@ -37,14 +37,18 @@
             }
 
         };
-        Link testLink1 = mock(Link.class);
-        when(testLink1.pull(anyBoolean())).thenReturn(new ByteBuffer("hello".getBytes()));
-        Link testLink2 = mock(Link.class);
-        when(testLink2.pull(anyBoolean())).thenReturn(new ByteBuffer("hello".getBytes()));
+        Link testLink1 = createMockLink();
+        Link testLink2 = createMockLink();
 
         element.setLink("testpad1", testLink1, streamer.Direction.IN);
         element.setLink("testpad2", testLink2, streamer.Direction.IN);
         element.poll(false);
     }
 
+    private static Link createMockLink() {
+        Link testLink = mock(Link.class);
+        when(testLink.pull(anyBoolean())).thenReturn(new ByteBuffer("hello".getBytes()));
+        return testLink;
+    }
+
 }
