--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxApiClientTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxApiClientTest.java
@@ -164,12 +164,11 @@
 
         LbPools lbPools = Mockito.mock(LbPools.class);
         LbMonitorProfiles lbMonitorProfiles = Mockito.mock(LbMonitorProfiles.class);
-        Structure monitorStructure = Mockito.mock(Structure.class, Mockito.RETURNS_DEEP_STUBS);
+        Structure monitorStructure = createMockMonitorStructure();
 
         Mockito.when(nsxService.apply(LbPools.class)).thenReturn(lbPools);
         Mockito.when(nsxService.apply(LbMonitorProfiles.class)).thenReturn(lbMonitorProfiles);
         Mockito.when(lbMonitorProfiles.get(anyString())).thenThrow(new NotFound(null, null)).thenReturn(monitorStructure);
-        Mockito.when(monitorStructure._getDataValue().getField("path").toString()).thenReturn("/infra/lb-monitor-profiles/test");
         Mockito.when(lbPools.get(lbServerPoolName)).thenThrow(new NotFound(null, null));
 
         client.createNsxLbServerPool(memberList, TIER_1_GATEWAY_NAME, lbServerPoolName, "roundrobin", "80", "TCP");
@@ -185,12 +184,11 @@
 
         LbPools lbPools = Mockito.mock(LbPools.class);
         LbMonitorProfiles lbMonitorProfiles = Mockito.mock(LbMonitorProfiles.class);
-        Structure monitorStructure = Mockito.mock(Structure.class, Mockito.RETURNS_DEEP_STUBS);
+        Structure monitorStructure = createMockMonitorStructure();
 
         Mockito.when(nsxService.apply(LbPools.class)).thenReturn(lbPools);
         Mockito.when(nsxService.apply(LbMonitorProfiles.class)).thenReturn(lbMonitorProfiles);
         Mockito.when(lbMonitorProfiles.get(anyString())).thenThrow(new NotFound(null, null)).thenReturn(monitorStructure);
-        Mockito.when(monitorStructure._getDataValue().getField("path").toString()).thenReturn("/infra/lb-monitor-profiles/test");
         Mockito.when(lbPools.get(lbServerPoolName)).thenThrow(new NotFound(null, null));
 
         client.createNsxLbServerPool(memberList, TIER_1_GATEWAY_NAME, lbServerPoolName, "roundrobin", "80", "UDP");
@@ -391,6 +389,12 @@
         assertTrue(thrownException.getMessage().startsWith("Failed to create NSX LB server pool, due to"));
     }
 
+    private static Structure createMockMonitorStructure() {
+        Structure monitorStructure = Mockito.mock(Structure.class, Mockito.RETURNS_DEEP_STUBS);
+        Mockito.when(monitorStructure._getDataValue().getField("path").toString()).thenReturn("/infra/lb-monitor-profiles/test");
+        return monitorStructure;
+    }
+
     private LbMonitorProfiles mockLbMonitorProfiles() {
         LbMonitorProfiles lbMonitorProfiles = Mockito.mock(LbMonitorProfiles.class);
         Structure monitorStructure = Mockito.mock(Structure.class, Mockito.RETURNS_DEEP_STUBS);
