--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -273,6 +273,8 @@
     DataCenterDao dataCenterDao;
     @Mock
     IpAddressManager ipAddressManager;
+    @Mock
+    private SubnetType subnetTypeMock;
 
     TungstenServiceImpl tungstenService;
 
@@ -895,7 +897,6 @@
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
         AddressType addressType = mock(AddressType.class);
-        SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
@@ -909,7 +910,7 @@
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
+        when(addressType.getSubnet()).thenReturn(subnetTypeMock);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1002,7 +1003,6 @@
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
         AddressType addressType = mock(AddressType.class);
-        SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
@@ -1015,7 +1015,7 @@
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
+        when(addressType.getSubnet()).thenReturn(subnetTypeMock);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1031,7 +1031,6 @@
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
         AddressType addressType = mock(AddressType.class);
-        SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
@@ -1044,7 +1043,7 @@
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
+        when(addressType.getSubnet()).thenReturn(subnetTypeMock);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1252,14 +1251,13 @@
         TungstenAnswer createTungstenAddressGroupAnswer = MockTungstenAnswerFactory.get(true);
         AddressGroup addressGroup = mock(AddressGroup.class);
         SubnetListType subnetListType = mock(SubnetListType.class);
-        SubnetType subnetType = mock(SubnetType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenAddressGroupCommand.class), anyLong())).thenReturn(createTungstenAddressGroupAnswer);
         when(createTungstenAddressGroupAnswer.getApiObjectBase()).thenReturn(addressGroup);
         when(addressGroup.getPrefix()).thenReturn(subnetListType);
-        when(subnetListType.getSubnet()).thenReturn(List.of(subnetType));
+        when(subnetListType.getSubnet()).thenReturn(List.of(subnetTypeMock));
 
         assertNotNull(tungstenService.createTungstenAddressGroup(1L, "test", "192.168.100.0", 24));
     }
@@ -1417,7 +1415,6 @@
         TungstenAnswer listTungstenAddressGroupAnswer = MockTungstenAnswerFactory.get(true);
         AddressGroup addressGroup = mock(AddressGroup.class);
         SubnetListType subnetListType = mock(SubnetListType.class);
-        SubnetType subnetType = mock(SubnetType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
@@ -1425,7 +1422,7 @@
         doReturn(List.of(addressGroup)).when(listTungstenAddressGroupAnswer).getApiObjectBaseList();
         when(addressGroup.getPrefix()).thenReturn(subnetListType);
         when(subnetListType.getSubnet()).thenReturn(
-                List.of(subnetType));
+                List.of(subnetTypeMock));
 
         assertNotNull(tungstenService.listTungstenAddressGroup(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
