--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/resource/BrocadeVcsResourceTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/resource/BrocadeVcsResourceTest.java
@@ -108,8 +108,7 @@
     public void testPingCommandStatusOk() throws ConfigurationException, BrocadeVcsApiException {
         resource.configure("BrocadeVcsResource", parameters);
 
-        final VcsNodeInfo nodeInfo = mock(VcsNodeInfo.class);
-        when(nodeInfo.getNodeState()).thenReturn("Online");
+        final VcsNodeInfo nodeInfo = createMockVcsNodeInfo("Online");
 
         List<VcsNodeInfo> nodes = new ArrayList<VcsNodeInfo>();
         nodes.add(nodeInfo);
@@ -131,8 +130,7 @@
     public void testPingCommandStatusFail() throws ConfigurationException, BrocadeVcsApiException {
         resource.configure("BrocadeVcsResource", parameters);
 
-        final VcsNodeInfo nodeInfo = mock(VcsNodeInfo.class);
-        when(nodeInfo.getNodeState()).thenReturn("Offline");
+        final VcsNodeInfo nodeInfo = createMockVcsNodeInfo("Offline");
 
         List<VcsNodeInfo> nodes = new ArrayList<VcsNodeInfo>();
         nodes.add(nodeInfo);
@@ -258,4 +256,10 @@
         final DisassociateMacFromNetworkAnswer answer = (DisassociateMacFromNetworkAnswer)resource.executeRequest(cmd);
         assertFalse(answer.getResult());
     }
+
+    private static VcsNodeInfo createMockVcsNodeInfo(String nodeState) {
+        final VcsNodeInfo nodeInfo = mock(VcsNodeInfo.class);
+        when(nodeInfo.getNodeState()).thenReturn(nodeState);
+        return nodeInfo;
+    }
 }
