--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
@@ -631,20 +631,23 @@
     @Test(expected = CloudRuntimeException.class)
     public void testCheckSourceVmResourcesAgainstSelectedOfferingResourcesInsufficientMemory() throws Exception {
         VirtualMachineMO virtualMachineMO = Mockito.mock(VirtualMachineMO.class);
-        VirtualMachineConfigSummary configSummary = Mockito.mock(VirtualMachineConfigSummary.class);
+        VirtualMachineConfigSummary configSummary = createMockVirtualMachineConfigSummary(2048);
         Mockito.when(virtualMachineMO.getConfigSummary()).thenReturn(configSummary);
-        Mockito.when(configSummary.getNumCpu()).thenReturn(1);
-        Mockito.when(configSummary.getMemorySizeMB()).thenReturn(2048);
         vMwareGuru.checkSourceVmResourcesAgainstSelectedOfferingResources(virtualMachineMO, 1, 500, 1024);
     }
 
     @Test
     public void testCheckSourceVmResourcesAgainstSelectedOfferingResourcesGreaterOffering() throws Exception {
         VirtualMachineMO virtualMachineMO = Mockito.mock(VirtualMachineMO.class);
+        VirtualMachineConfigSummary configSummary = createMockVirtualMachineConfigSummary(1024);
+        Mockito.when(virtualMachineMO.getConfigSummary()).thenReturn(configSummary);
+        vMwareGuru.checkSourceVmResourcesAgainstSelectedOfferingResources(virtualMachineMO, 2, 1500, 2048);
+    }
+
+    private static VirtualMachineConfigSummary createMockVirtualMachineConfigSummary(int memorySizeMB) {
         VirtualMachineConfigSummary configSummary = Mockito.mock(VirtualMachineConfigSummary.class);
-        Mockito.when(virtualMachineMO.getConfigSummary()).thenReturn(configSummary);
         Mockito.when(configSummary.getNumCpu()).thenReturn(1);
-        Mockito.when(configSummary.getMemorySizeMB()).thenReturn(1024);
-        vMwareGuru.checkSourceVmResourcesAgainstSelectedOfferingResources(virtualMachineMO, 2, 1500, 2048);
+        Mockito.when(configSummary.getMemorySizeMB()).thenReturn(memorySizeMB);
+        return configSummary;
     }
 }
