--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
@@ -49,6 +49,7 @@
 import org.hamcrest.core.StringContains;
 import org.hamcrest.core.StringEndsWith;
 import org.hamcrest.core.StringStartsWith;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.verification.VerificationMode;
@@ -208,6 +209,13 @@
 			new DefaultOAuth2AuthenticatedPrincipal(JWT_CLAIMS, Collections.emptyList()),
 			TestOAuth2AccessTokens.noScopes(), Collections.emptyList());
 
+	private ApplicationContext contextMock;
+
+	@BeforeEach
+	public void setUp() {
+		this.contextMock = mock(ApplicationContext.class);
+	}
+
 	@Autowired(required = false)
 	MockMvc mvc;
 
@@ -815,14 +823,13 @@
 
 	@Test
 	public void getJwtDecoderWhenConfiguredWithDecoderAndJwkSetUriThenLastOneWins() {
-		ApplicationContext context = mock(ApplicationContext.class);
 		JwtDecoder decoder = mock(JwtDecoder.class);
-		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
+		new OAuth2ResourceServerConfigurer<HttpSecurity>(contextMock).jwt((jwt) -> {
 			jwt.jwkSetUri(JWK_SET_URI);
 			jwt.decoder(decoder);
 			assertThat(jwt.getJwtDecoder()).isEqualTo(decoder);
 		});
-		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
+		new OAuth2ResourceServerConfigurer<HttpSecurity>(contextMock).jwt((jwt) -> {
 			jwt.decoder(decoder).jwkSetUri(JWK_SET_URI);
 			assertThat(jwt.getJwtDecoder()).isInstanceOf(NimbusJwtDecoder.class);
 		});
@@ -904,15 +911,13 @@
 
 	@Test
 	public void authenticationEntryPointWhenGivenNullThenThrowsException() {
-		ApplicationContext context = mock(ApplicationContext.class);
-		OAuth2ResourceServerConfigurer configurer = new OAuth2ResourceServerConfigurer(context);
+		OAuth2ResourceServerConfigurer configurer = new OAuth2ResourceServerConfigurer(contextMock);
 		assertThatIllegalArgumentException().isThrownBy(() -> configurer.authenticationEntryPoint(null));
 	}
 
 	@Test
 	public void accessDeniedHandlerWhenGivenNullThenThrowsException() {
-		ApplicationContext context = mock(ApplicationContext.class);
-		OAuth2ResourceServerConfigurer configurer = new OAuth2ResourceServerConfigurer(context);
+		OAuth2ResourceServerConfigurer configurer = new OAuth2ResourceServerConfigurer(contextMock);
 		assertThatIllegalArgumentException().isThrownBy(() -> configurer.accessDeniedHandler(null));
 	}
 
@@ -1157,15 +1162,14 @@
 
 	@Test
 	public void getIntrospectionClientWhenConfiguredWithClientAndIntrospectionUriThenLastOneWins() {
-		ApplicationContext context = mock(ApplicationContext.class);
 		OpaqueTokenIntrospector client = mock(OpaqueTokenIntrospector.class);
-		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).opaqueToken((opaqueToken) -> {
+		new OAuth2ResourceServerConfigurer<HttpSecurity>(contextMock).opaqueToken((opaqueToken) -> {
 			opaqueToken.introspectionUri(INTROSPECTION_URI);
 			opaqueToken.introspectionClientCredentials(CLIENT_ID, CLIENT_SECRET);
 			opaqueToken.introspector(client);
 			assertThat(opaqueToken.getIntrospector()).isEqualTo(client);
 		});
-		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).opaqueToken((opaqueToken) -> {
+		new OAuth2ResourceServerConfigurer<HttpSecurity>(contextMock).opaqueToken((opaqueToken) -> {
 			opaqueToken.introspector(client);
 			opaqueToken.introspectionUri(INTROSPECTION_URI);
 			opaqueToken.introspectionClientCredentials(CLIENT_ID, CLIENT_SECRET);
@@ -1268,14 +1272,13 @@
 
 	@Test
 	public void getAuthenticationManagerWhenConfiguredAuthenticationManagerThenTakesPrecedence() {
-		ApplicationContext context = mock(ApplicationContext.class);
 		OAuth2ResourceServerConfigurer<HttpSecurity> oauth2ResourceServer = new OAuth2ResourceServerConfigurer<>(
-				context);
+				contextMock);
 		AuthenticationManager authenticationManager = mock(AuthenticationManager.class);
 		oauth2ResourceServer
 			.jwt((jwt) -> jwt.authenticationManager(authenticationManager).decoder(mock(JwtDecoder.class)));
 		assertThat(oauth2ResourceServer.getAuthenticationManager(null)).isSameAs(authenticationManager);
-		oauth2ResourceServer = new OAuth2ResourceServerConfigurer<>(context);
+		oauth2ResourceServer = new OAuth2ResourceServerConfigurer<>(contextMock);
 		oauth2ResourceServer.opaqueToken((opaqueToken) -> opaqueToken.authenticationManager(authenticationManager)
 			.introspector(mock(OpaqueTokenIntrospector.class)));
 		assertThat(oauth2ResourceServer.getAuthenticationManager(null)).isSameAs(authenticationManager);
