--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockHeadersConverter.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockHeadersConverter.java
@@ -0,0 +1,22 @@
+package org.springframework.security.oauth2.client.endpoint;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.http.HttpHeaders;
+
+import org.mockito.Mockito;
+
+import static org.mockito.BDDMockito.given;
+
+public final class MockHeadersConverter {
+
+	private MockHeadersConverter() {
+	}
+
+	public static Converter<TokenExchangeGrantRequest, HttpHeaders> createMockHeadersConverter(
+			TokenExchangeGrantRequest grantRequest, HttpHeaders headers) {
+		Converter<TokenExchangeGrantRequest, HttpHeaders> headersConverter = Mockito.mock();
+		given(headersConverter.convert(grantRequest)).willReturn(headers);
+		return headersConverter;
+	}
+
+}
