--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientClientCredentialsTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientClientCredentialsTokenResponseClientTests.java
@@ -382,10 +382,9 @@
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		OAuth2ClientCredentialsGrantRequest grantRequest = new OAuth2ClientCredentialsGrantRequest(clientRegistration);
-		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = mock();
 		MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
 		parameters.add("custom-parameter-name", "custom-parameter-value");
-		given(parametersConverter.convert(grantRequest)).willReturn(parameters);
+		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = MockClientCredentialsParametersConverter.createMockClientCredentialsParametersConverter(grantRequest, parameters);
 		this.tokenResponseClient.setParametersConverter(parametersConverter);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		verify(parametersConverter).convert(grantRequest);
@@ -400,11 +399,10 @@
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		OAuth2ClientCredentialsGrantRequest grantRequest = new OAuth2ClientCredentialsGrantRequest(clientRegistration);
-		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = mock();
 		MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
 		parameters.set(OAuth2ParameterNames.GRANT_TYPE, "custom");
 		parameters.set(OAuth2ParameterNames.SCOPE, "one two");
-		given(parametersConverter.convert(grantRequest)).willReturn(parameters);
+		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = org.springframework.security.oauth2.client.endpoint.MockClientCredentialsParametersConverter.createMockClientCredentialsParametersConverter(grantRequest, parameters);
 		this.tokenResponseClient.setParametersConverter((authorizationGrantRequest) -> parameters);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		RecordedRequest recordedRequest = this.server.takeRequest();
@@ -424,10 +422,9 @@
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		Set<String> scopes = clientRegistration.getScopes();
 		OAuth2ClientCredentialsGrantRequest grantRequest = new OAuth2ClientCredentialsGrantRequest(clientRegistration);
-		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = mock();
 		MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
 		parameters.add("custom-parameter-name", "custom-parameter-value");
-		given(parametersConverter.convert(grantRequest)).willReturn(parameters);
+		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = org.springframework.security.oauth2.client.endpoint.MockClientCredentialsParametersConverter.createMockClientCredentialsParametersConverter(grantRequest, parameters);
 		this.tokenResponseClient.addParametersConverter(parametersConverter);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		verify(parametersConverter).convert(grantRequest);
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveClientCredentialsTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveClientCredentialsTokenResponseClientTests.java
@@ -316,10 +316,9 @@
 	public void getTokenResponseWhenParametersConverterAddedThenCalled() throws Exception {
 		OAuth2ClientCredentialsGrantRequest request = new OAuth2ClientCredentialsGrantRequest(
 				this.clientRegistration.build());
-		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> addedParametersConverter = mock();
 		MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
 		parameters.add("custom-parameter-name", "custom-parameter-value");
-		given(addedParametersConverter.convert(request)).willReturn(parameters);
+		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> addedParametersConverter = org.springframework.security.oauth2.client.endpoint.MockClientCredentialsParametersConverter.createMockClientCredentialsParametersConverter(request, parameters);
 		this.client.addParametersConverter(addedParametersConverter);
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		this.client.getTokenResponse(request).block();
@@ -333,10 +332,9 @@
 	public void getTokenResponseWhenParametersConverterSetThenCalled() throws Exception {
 		OAuth2ClientCredentialsGrantRequest request = new OAuth2ClientCredentialsGrantRequest(
 				this.clientRegistration.build());
-		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = mock();
 		MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
 		parameters.add("custom-parameter-name", "custom-parameter-value");
-		given(parametersConverter.convert(request)).willReturn(parameters);
+		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = MockClientCredentialsParametersConverter.createMockClientCredentialsParametersConverter(request, parameters);
 		this.client.setParametersConverter(parametersConverter);
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		this.client.getTokenResponse(request).block();
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockClientCredentialsParametersConverter.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockClientCredentialsParametersConverter.java
@@ -0,0 +1,19 @@
+package org.springframework.security.oauth2.client.endpoint;
+
+import org.mockito.Mockito;
+
+import org.springframework.util.MultiValueMap;
+import org.springframework.core.convert.converter.Converter;
+
+import static org.mockito.BDDMockito.given;
+
+public class MockClientCredentialsParametersConverter {
+
+	public static Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> createMockClientCredentialsParametersConverter(
+			OAuth2ClientCredentialsGrantRequest request, MultiValueMap<String, String> parameters) {
+		Converter<OAuth2ClientCredentialsGrantRequest, MultiValueMap<String, String>> parametersConverter = Mockito.mock();
+		given(parametersConverter.convert(request)).willReturn(parameters);
+		return parametersConverter;
+	}
+
+}
