--- a/web/src/test/java/org/springframework/security/web/authentication/AuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/AuthenticationFilterTests.java
@@ -85,6 +85,9 @@
 	@Mock
 	private RequestMatcher requestMatcher;
 
+	@Mock
+	private FilterChain filterChain;
+
 	private void givenResolveWillReturnAuthenticationManager() {
 		given(this.authenticationManagerResolver.resolve(any())).willReturn(this.authenticationManager);
 	}
@@ -100,10 +103,9 @@
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verifyNoMoreInteractions(this.authenticationManager);
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
 
@@ -113,10 +115,9 @@
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verifyNoMoreInteractions(this.authenticationManagerResolver);
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
 
@@ -129,10 +130,9 @@
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNotNull();
 		assertThat(request.getAttribute(RequestAttributeSecurityContextRepository.DEFAULT_REQUEST_ATTR_NAME))
 			.isNotNull();
@@ -151,10 +151,9 @@
 		filter.setSecurityContextHolderStrategy(strategy);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		verify(strategy).setContext(any());
 	}
 
@@ -169,10 +168,9 @@
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNotNull();
 		assertThat(request.getAttribute(RequestAttributeSecurityContextRepository.DEFAULT_REQUEST_ATTR_NAME))
 			.isNotNull();
@@ -187,8 +185,7 @@
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
@@ -204,8 +201,7 @@
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
@@ -216,10 +212,9 @@
 		AuthenticationFilter filter = new AuthenticationFilter(this.authenticationManagerResolver,
 				this.authenticationConverter);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, new MockHttpServletResponse(), chain);
+		filter.doFilter(request, new MockHttpServletResponse(), filterChain);
 		verifyNoMoreInteractions(this.authenticationManagerResolver);
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
 
@@ -234,8 +229,7 @@
 		filter.setSuccessHandler(this.successHandler);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verify(this.successHandler).onAuthenticationSuccess(any(), any(), any(), eq(authentication));
 		verifyNoMoreInteractions(this.failureHandler);
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNotNull();
@@ -254,8 +248,7 @@
 		filter.setSuccessHandler(this.successHandler);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		assertThatExceptionOfType(ServletException.class).isThrownBy(() -> filter.doFilter(request, response, chain));
+		assertThatExceptionOfType(ServletException.class).isThrownBy(() -> filter.doFilter(request, response, filterChain));
 		verifyNoMoreInteractions(this.successHandler);
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
@@ -268,8 +261,7 @@
 		filter.setRequestMatcher(this.requestMatcher);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		verifyNoMoreInteractions(this.authenticationConverter, this.authenticationManagerResolver, this.successHandler);
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
