--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2ClientAuthenticationFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2ClientAuthenticationFilterTests.java
@@ -76,6 +76,8 @@
 
 	private OAuth2ClientAuthenticationFilter filter;
 
+	private FilterChain filterChainMock;
+
 	private final HttpMessageConverter<OAuth2Error> errorHttpResponseConverter = new OAuth2ErrorHttpMessageConverter();
 
 	@BeforeEach
@@ -85,6 +87,7 @@
 			.matcher(HttpMethod.POST, this.filterProcessesUrl);
 		this.filter = new OAuth2ClientAuthenticationFilter(this.authenticationManager, this.requestMatcher);
 		this.authenticationConverter = mock(AuthenticationConverter.class);
+		this.filterChainMock = mock(FilterChain.class);
 		this.filter.setAuthenticationConverter(this.authenticationConverter);
 	}
 
@@ -134,11 +137,10 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", requestUri);
 		request.setServletPath(requestUri);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-
-		this.filter.doFilter(request, response, filterChain);
-
-		verify(filterChain).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
+
+		this.filter.doFilter(request, response, filterChainMock);
+
+		verify(filterChainMock).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		verifyNoInteractions(this.authenticationConverter);
 	}
 
@@ -147,11 +149,10 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", this.filterProcessesUrl);
 		request.setServletPath(this.filterProcessesUrl);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-
-		this.filter.doFilter(request, response, filterChain);
-
-		verify(filterChain).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
+
+		this.filter.doFilter(request, response, filterChainMock);
+
+		verify(filterChainMock).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		verifyNoInteractions(this.authenticationManager);
 	}
 
@@ -163,11 +164,10 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", this.filterProcessesUrl);
 		request.setServletPath(this.filterProcessesUrl);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-
-		this.filter.doFilter(request, response, filterChain);
-
-		verifyNoInteractions(filterChain);
+
+		this.filter.doFilter(request, response, filterChainMock);
+
+		verifyNoInteractions(filterChainMock);
 
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.BAD_REQUEST.value());
@@ -228,11 +228,10 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", this.filterProcessesUrl);
 		request.setServletPath(this.filterProcessesUrl);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-
-		this.filter.doFilter(request, response, filterChain);
-
-		verifyNoInteractions(filterChain);
+
+		this.filter.doFilter(request, response, filterChainMock);
+
+		verifyNoInteractions(filterChainMock);
 		verify(this.authenticationManager).authenticate(any());
 
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
@@ -257,11 +256,10 @@
 		request.setServletPath(this.filterProcessesUrl);
 		request.setRemoteAddr(remoteAddress);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-
-		this.filter.doFilter(request, response, filterChain);
-
-		verify(filterChain).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
+
+		this.filter.doFilter(request, response, filterChainMock);
+
+		verify(filterChainMock).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 
 		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
 		assertThat(authentication).isInstanceOf(OAuth2ClientAuthenticationToken.class);
