--- a/web/src/test/java/org/springframework/security/web/authentication/ott/OneTimeTokenAuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/ott/OneTimeTokenAuthenticationFilterTests.java
@@ -64,8 +64,11 @@
 
 	private final HttpServletResponse response = new MockHttpServletResponse();
 
+	private HttpServletResponse responseMock;
+
 	@BeforeEach
 	void setUp() {
+		this.responseMock = mock(HttpServletResponse.class);
 		this.filter.setAuthenticationManager(this.authenticationManager);
 	}
 
@@ -79,10 +82,9 @@
 	@SuppressWarnings("removal")
 	void doFilterWhenUrlDoesNotMatchThenContinues() throws ServletException, IOException {
 		OneTimeTokenAuthenticationConverter converter = mock(OneTimeTokenAuthenticationConverter.class);
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		this.filter.setAuthenticationConverter(converter);
-		this.filter.doFilter(post("/nomatch").buildRequest(new MockServletContext()), response, this.chain);
-		verifyNoInteractions(converter, response);
+		this.filter.doFilter(post("/nomatch").buildRequest(new MockServletContext()), responseMock, this.chain);
+		verifyNoInteractions(converter, responseMock);
 		verify(this.chain).doFilter(any(), any());
 	}
 
@@ -90,10 +92,9 @@
 	@SuppressWarnings("removal")
 	void doFilterWhenMethodDoesNotMatchThenContinues() throws ServletException, IOException {
 		OneTimeTokenAuthenticationConverter converter = mock(OneTimeTokenAuthenticationConverter.class);
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		this.filter.setAuthenticationConverter(converter);
-		this.filter.doFilter(get("/login/ott").buildRequest(new MockServletContext()), response, this.chain);
-		verifyNoInteractions(converter, response);
+		this.filter.doFilter(get("/login/ott").buildRequest(new MockServletContext()), responseMock, this.chain);
+		verifyNoInteractions(converter, responseMock);
 		verify(this.chain).doFilter(any(), any());
 	}
 
