--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/JwtBearerReactiveOAuth2AuthorizedClientProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/JwtBearerReactiveOAuth2AuthorizedClientProviderTests.java
@@ -187,8 +187,7 @@
 
 	@Test
 	public void authorizeWhenClockSetThenCalled() {
-		Clock clock = mock(Clock.class);
-		given(clock.instant()).willReturn(Instant.now());
+		Clock clock = createMockClock();
 		this.authorizedClientProvider.setClock(clock);
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.clientRegistration,
 				this.principal.getName(), TestOAuth2AccessTokens.noScopes());
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeOAuth2AuthorizedClientProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeOAuth2AuthorizedClientProviderTests.java
@@ -381,8 +381,7 @@
 
 	@Test
 	public void authorizeWhenClockSetThenCalled() {
-		Clock clock = mock(Clock.class);
-		given(clock.instant()).willReturn(Instant.now());
+		Clock clock = createMockClock();
 		this.authorizedClientProvider.setClock(clock);
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.clientRegistration,
 				this.principal.getName(), TestOAuth2AccessTokens.noScopes());
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeReactiveOAuth2AuthorizedClientProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeReactiveOAuth2AuthorizedClientProviderTests.java
@@ -386,8 +386,7 @@
 
 	@Test
 	public void authorizeWhenClockSetThenCalled() {
-		Clock clock = mock(Clock.class);
-		given(clock.instant()).willReturn(Instant.now());
+		Clock clock = createMockClock();
 		this.authorizedClientProvider.setClock(clock);
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.clientRegistration,
 				this.principal.getName(), TestOAuth2AccessTokens.noScopes());
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/MockClock.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/MockClock.java
@@ -0,0 +1,19 @@
+package org.springframework.security.oauth2.client;
+
+import java.time.Clock;
+import java.time.Instant;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public final class MockClock {
+
+	private MockClock() {
+	}
+
+	public static Clock createMockClock() {
+		Clock clock = mock(Clock.class);
+		given(clock.instant()).willReturn(Instant.now());
+		return clock;
+	}
+}
