--- a/web/src/test/java/org/springframework/security/web/ObservationFilterChainDecoratorTests.java
+++ b/web/src/test/java/org/springframework/security/web/ObservationFilterChainDecoratorTests.java
@@ -54,8 +54,7 @@
 
 	@Test
 	void decorateWhenDefaultsThenObserves() throws Exception {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createMockObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
@@ -67,8 +66,7 @@
 
 	@Test
 	void decorateWhenNoopThenDoesNotObserve() throws Exception {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createMockObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.NOOP;
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
@@ -80,8 +78,7 @@
 
 	@Test
 	void decorateFiltersWhenDefaultsThenObserves() throws Exception {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createMockObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
@@ -99,8 +96,7 @@
 
 	@Test
 	void decorateFiltersWhenDefaultsThenUsesEventName() throws Exception {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createMockObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
@@ -118,8 +114,7 @@
 	// gh-12787
 	@Test
 	void decorateFiltersWhenErrorsThenClosesObservationOnlyOnce() throws Exception {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createMockObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
@@ -173,6 +168,12 @@
 				Arguments.of(filterWithoutName, "none"));
 	}
 
+	private static ObservationHandler<?> createMockObservationHandler() {
+		ObservationHandler<?> handler = mock(ObservationHandler.class);
+		given(handler.supportsContext(any())).willReturn(true);
+		return handler;
+	}
+
 	private static class BasicAuthenticationFilter implements Filter {
 
 		@Override
