--- a/web/src/test/java/org/springframework/security/web/session/SessionManagementFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/session/SessionManagementFilterTests.java
@@ -55,16 +55,18 @@
  */
 public class SessionManagementFilterTests {
 
+	private SecurityContextRepository contextRepository;
+
 	@BeforeEach
 	@AfterEach
 	public void clearContext() {
+		contextRepository = mock(SecurityContextRepository.class);
 		SecurityContextHolder.clearContext();
 	}
 
 	@Test
 	public void newSessionShouldNotBeCreatedIfSessionExistsAndUserIsNotAuthenticated() throws Exception {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository);
 		HttpServletRequest request = new MockHttpServletRequest();
 		String sessionId = request.getSession().getId();
 		filter.doFilter(request, new MockHttpServletResponse(), new MockFilterChain());
@@ -86,9 +88,8 @@
 
 	@Test
 	public void strategyIsNotInvokedIfAuthenticationIsNull() throws Exception {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo, strategy);
+		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository, strategy);
 		HttpServletRequest request = new MockHttpServletRequest();
 		filter.doFilter(request, new MockHttpServletResponse(), new MockFilterChain());
 		verifyNoMoreInteractions(strategy);
@@ -96,10 +97,9 @@
 
 	@Test
 	public void strategyIsInvokedIfUserIsNewlyAuthenticated() throws Exception {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		// repo will return false to containsContext()
-		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo, strategy);
+		// repo will return false to containsContext()
+		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository, strategy);
 		HttpServletRequest request = new MockHttpServletRequest();
 		authenticateUser();
 		filter.doFilter(request, new MockHttpServletResponse(), new MockFilterChain());
@@ -112,11 +112,10 @@
 
 	@Test
 	public void strategyFailureInvokesFailureHandler() throws Exception {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
 		// repo will return false to containsContext()
 		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
 		AuthenticationFailureHandler failureHandler = mock(AuthenticationFailureHandler.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo, strategy);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository, strategy);
 		filter.setAuthenticationFailureHandler(failureHandler);
 		HttpServletRequest request = new MockHttpServletRequest();
 		HttpServletResponse response = new MockHttpServletResponse();
@@ -132,10 +131,9 @@
 
 	@Test
 	public void responseIsRedirectedToTimeoutUrlIfSetAndSessionIsInvalid() throws Exception {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		// repo will return false to containsContext()
-		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo, strategy);
+		// repo will return false to containsContext()
+		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository, strategy);
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		request.setRequestedSessionId("xxx");
 		request.setRequestedSessionIdValid(false);
@@ -157,10 +155,9 @@
 
 	@Test
 	public void responseIsRedirectedToRequestedUrlIfSetAndSessionIsInvalid() throws Exception {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		// repo will return false to containsContext()
-		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo, strategy);
+		// repo will return false to containsContext()
+		SessionAuthenticationStrategy strategy = mock(SessionAuthenticationStrategy.class);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository, strategy);
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		request.setRequestedSessionId("xxx");
 		request.setRequestedSessionIdValid(false);
@@ -189,9 +186,8 @@
 		RequestedUrlRedirectInvalidSessionStrategy invalidSessionStrategy = new RequestedUrlRedirectInvalidSessionStrategy();
 		invalidSessionStrategy.setCreateNewSession(true);
 		invalidSessionStrategy.setRedirectStrategy(redirectStrategy);
-		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
 		SessionAuthenticationStrategy sessionAuthenticationStrategy = mock(SessionAuthenticationStrategy.class);
-		SessionManagementFilter filter = new SessionManagementFilter(securityContextRepository,
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository,
 				sessionAuthenticationStrategy);
 		filter.setInvalidSessionStrategy(invalidSessionStrategy);
 		MockHttpServletRequest request = new MockHttpServletRequest();
@@ -206,8 +202,8 @@
 		filter.doFilter(request, response, chain);
 
 		// then
-		verify(securityContextRepository).containsContext(request);
-		verifyNoMoreInteractions(securityContextRepository, sessionAuthenticationStrategy, chain);
+		verify(contextRepository).containsContext(request);
+		verifyNoMoreInteractions(contextRepository, sessionAuthenticationStrategy, chain);
 		assertThat(response.isCommitted()).isTrue();
 		assertThat(response.getRedirectedUrl()).isEqualTo("/context/requested");
 		assertThat(response.getStatus()).isEqualTo(302);
@@ -221,9 +217,8 @@
 		RequestedUrlRedirectInvalidSessionStrategy invalidSessionStrategy = new RequestedUrlRedirectInvalidSessionStrategy();
 		invalidSessionStrategy.setCreateNewSession(true);
 		invalidSessionStrategy.setRedirectStrategy(redirectStrategy);
-		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
 		SessionAuthenticationStrategy sessionAuthenticationStrategy = mock(SessionAuthenticationStrategy.class);
-		SessionManagementFilter filter = new SessionManagementFilter(securityContextRepository,
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository,
 				sessionAuthenticationStrategy);
 		filter.setInvalidSessionStrategy(invalidSessionStrategy);
 		MockHttpServletRequest request = new MockHttpServletRequest();
@@ -237,8 +232,8 @@
 		filter.doFilter(request, response, chain);
 
 		// then
-		verify(securityContextRepository).containsContext(request);
-		verifyNoMoreInteractions(securityContextRepository, sessionAuthenticationStrategy, chain);
+		verify(contextRepository).containsContext(request);
+		verifyNoMoreInteractions(contextRepository, sessionAuthenticationStrategy, chain);
 		assertThat(response.isCommitted()).isTrue();
 		assertThat(response.getRedirectedUrl()).isEqualTo("/requested");
 		assertThat(response.getStatus()).isEqualTo(307);
@@ -248,8 +243,7 @@
 	public void customAuthenticationTrustResolver() throws Exception {
 		AuthenticationTrustResolver trustResolver = mock(AuthenticationTrustResolver.class,
 				withSettings().defaultAnswer(Answers.CALLS_REAL_METHODS));
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository);
 		filter.setTrustResolver(trustResolver);
 		HttpServletRequest request = new MockHttpServletRequest();
 		authenticateUser();
@@ -259,8 +253,7 @@
 
 	@Test
 	public void setTrustResolverNull() {
-		SecurityContextRepository repo = mock(SecurityContextRepository.class);
-		SessionManagementFilter filter = new SessionManagementFilter(repo);
+		SessionManagementFilter filter = new SessionManagementFilter(contextRepository);
 		assertThatIllegalArgumentException().isThrownBy(() -> filter.setTrustResolver(null));
 	}
 
