--- a/core/src/test/java/org/springframework/security/authentication/ReactiveUserDetailsServiceAuthenticationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/ReactiveUserDetailsServiceAuthenticationManagerTests.java
@@ -118,8 +118,8 @@
 
 	@Test
 	public void authenticateWhenPasswordEncoderAndSuccessThenSuccess() {
+		this.passwordEncoder = createMockPasswordEncoder(true);
 		this.manager.setPasswordEncoder(this.passwordEncoder);
-		given(this.passwordEncoder.matches(any(), any())).willReturn(true);
 		User user = new User(this.username, this.password, AuthorityUtils.createAuthorityList("ROLE_USER"));
 		given(this.repository.findByUsername(user.getUsername())).willReturn(Mono.just(user));
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated(this.username,
@@ -130,8 +130,8 @@
 
 	@Test
 	public void authenticateWhenPasswordEncoderAndFailThenFail() {
+		this.passwordEncoder = createMockPasswordEncoder(false);
 		this.manager.setPasswordEncoder(this.passwordEncoder);
-		given(this.passwordEncoder.matches(any(), any())).willReturn(false);
 		User user = new User(this.username, this.password, AuthorityUtils.createAuthorityList("ROLE_USER"));
 		given(this.repository.findByUsername(user.getUsername())).willReturn(Mono.just(user));
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated(this.username,
--- a/core/src/test/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManagerTests.java
@@ -103,7 +103,7 @@
 			return Schedulers.immediate().schedule(r);
 		});
 		given(this.userDetailsService.findByUsername(any())).willReturn(Mono.just(this.user));
-		given(this.encoder.matches(any(), any())).willReturn(true);
+		this.encoder = createMockPasswordEncoder(true);
 		this.manager.setScheduler(this.scheduler);
 		this.manager.setPasswordEncoder(this.encoder);
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated(this.user,
@@ -116,7 +116,7 @@
 	public void authenticateWhenPasswordServiceThenUpdated() {
 		String encodedPassword = "encoded";
 		given(this.userDetailsService.findByUsername(any())).willReturn(Mono.just(this.user));
-		given(this.encoder.matches(any(), any())).willReturn(true);
+		this.encoder = createMockPasswordEncoder(true);
 		given(this.encoder.upgradeEncoding(any())).willReturn(true);
 		given(this.encoder.encode(any())).willReturn(encodedPassword);
 		given(this.userDetailsPasswordService.updatePassword(any(), any())).willReturn(Mono.just(this.user));
@@ -132,7 +132,7 @@
 	@Test
 	public void authenticateWhenPasswordServiceAndBadCredentialsThenNotUpdated() {
 		given(this.userDetailsService.findByUsername(any())).willReturn(Mono.just(this.user));
-		given(this.encoder.matches(any(), any())).willReturn(false);
+		this.encoder = createMockPasswordEncoder(false);
 		this.manager.setPasswordEncoder(this.encoder);
 		this.manager.setUserDetailsPasswordService(this.userDetailsPasswordService);
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated(this.user,
@@ -145,7 +145,7 @@
 	@Test
 	public void authenticateWhenPasswordServiceAndUpgradeFalseThenNotUpdated() {
 		given(this.userDetailsService.findByUsername(any())).willReturn(Mono.just(this.user));
-		given(this.encoder.matches(any(), any())).willReturn(true);
+		this.encoder = createMockPasswordEncoder(true);
 		given(this.encoder.upgradeEncoding(any())).willReturn(false);
 		this.manager.setPasswordEncoder(this.encoder);
 		this.manager.setUserDetailsPasswordService(this.userDetailsPasswordService);
@@ -159,7 +159,7 @@
 	public void authenticateWhenPostAuthenticationChecksFail() {
 		given(this.userDetailsService.findByUsername(any())).willReturn(Mono.just(this.user));
 		willThrow(new LockedException("account is locked")).given(this.postAuthenticationChecks).check(any());
-		given(this.encoder.matches(any(), any())).willReturn(true);
+		this.encoder = createMockPasswordEncoder(true);
 		this.manager.setPasswordEncoder(this.encoder);
 		this.manager.setPostAuthenticationChecks(this.postAuthenticationChecks);
 		assertThatExceptionOfType(LockedException.class)
@@ -173,7 +173,7 @@
 	@Test
 	public void authenticateWhenPostAuthenticationChecksNotSet() {
 		given(this.userDetailsService.findByUsername(any())).willReturn(Mono.just(this.user));
-		given(this.encoder.matches(any(), any())).willReturn(true);
+		this.encoder = createMockPasswordEncoder(true);
 		this.manager.setPasswordEncoder(this.encoder);
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated(this.user,
 				this.user.getPassword());
--- a/core/src/test/java/org/springframework/security/authentication/MockPasswordEncoder.java
+++ b/core/src/test/java/org/springframework/security/authentication/MockPasswordEncoder.java
@@ -0,0 +1,20 @@
+package org.springframework.security.authentication;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+public final class MockPasswordEncoder {
+
+	private MockPasswordEncoder() {
+	}
+
+	public static PasswordEncoder createMockPasswordEncoder(boolean matchesResult) {
+		PasswordEncoder encoder = Mockito.mock(PasswordEncoder.class);
+		given(encoder.matches(any(), any())).willReturn(matchesResult);
+		return encoder;
+	}
+
+}
