--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/registration/WebAuthnRegistrationFilterTests.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/registration/WebAuthnRegistrationFilterTests.java
@@ -232,8 +232,7 @@
 
 	@Test
 	void doFilterWhenDeleteAndCustomAuthorizationManagerThenUses() throws Exception {
-		AuthorizationManager<Bytes> authorizationManager = mock(AuthorizationManager.class);
-		given(authorizationManager.authorize(any(), any())).willReturn(new AuthorizationDecision(true));
+		AuthorizationManager<Bytes> authorizationManager = createMockAuthorizationManager(true);
 		this.filter.setDeleteCredentialAuthorizationManager(authorizationManager);
 		MockHttpServletRequest request = MockMvcRequestBuilders.delete("/webauthn/register/123456")
 			.buildRequest(new MockServletContext());
@@ -243,13 +242,18 @@
 
 	@Test
 	void doFilterWhenDeleteAndAuthorizationDeniedThenForbidden() throws Exception {
-		AuthorizationManager<Bytes> authorizationManager = mock(AuthorizationManager.class);
-		given(authorizationManager.authorize(any(), any())).willReturn(new AuthorizationDecision(false));
+		AuthorizationManager<Bytes> authorizationManager = createMockAuthorizationManager(false);
 		this.filter.setDeleteCredentialAuthorizationManager(authorizationManager);
 		MockHttpServletRequest request = MockMvcRequestBuilders.delete("/webauthn/register/123456")
 			.buildRequest(new MockServletContext());
 		this.filter.doFilter(request, this.response, this.chain);
 		assertThat(this.response.getStatus()).isEqualTo(HttpStatus.FORBIDDEN.value());
+	}
+
+	private static AuthorizationManager<Bytes> createMockAuthorizationManager(boolean decision) {
+		AuthorizationManager<Bytes> authorizationManager = mock(AuthorizationManager.class);
+		given(authorizationManager.authorize(any(), any())).willReturn(new AuthorizationDecision(decision));
+		return authorizationManager;
 	}
 
 	private static MockHttpServletRequest registerCredentialRequest(String body) {
