--- a/core/src/test/java/org/springframework/security/authorization/AuthenticatedReactiveAuthorizationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthenticatedReactiveAuthorizationManagerTests.java
@@ -37,8 +37,7 @@
 @ExtendWith(MockitoExtension.class)
 public class AuthenticatedReactiveAuthorizationManagerTests {
 
-	@Mock
-	Authentication authentication;
+	Authentication authentication = MockAuthentication.createMockAuthentication();
 
 	AuthenticatedReactiveAuthorizationManager<Object> manager = AuthenticatedReactiveAuthorizationManager
 		.authenticated();
--- a/core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java
@@ -40,8 +40,7 @@
 @ExtendWith(MockitoExtension.class)
 public class AuthorityReactiveAuthorizationManagerTests {
 
-	@Mock
-	Authentication authentication;
+	Authentication authentication = MockAuthentication.createMockAuthentication();
 
 	AuthorityReactiveAuthorizationManager<Object> manager = AuthorityReactiveAuthorizationManager.hasAuthority("ADMIN");
 
--- a/core/src/test/java/org/springframework/security/authorization/MockAuthentication.java
+++ b/core/src/test/java/org/springframework/security/authorization/MockAuthentication.java
@@ -0,0 +1,16 @@
+package org.springframework.security.authorization;
+
+import org.mockito.Mockito;
+
+import org.springframework.security.core.Authentication;
+
+public final class MockAuthentication {
+
+	private MockAuthentication() {
+	}
+
+	public static Authentication createMockAuthentication() {
+		return Mockito.mock(Authentication.class);
+	}
+
+}
