--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/management/CredentialRecordOwnerAuthorizationManagerTests.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/management/CredentialRecordOwnerAuthorizationManagerTests.java
@@ -116,6 +116,7 @@
 
 	@Test
 	void checkWhenCredentialBelongsToUserThenGranted() {
+		this.userEntities = MockPublicKeyCredentialUserEntityRepository.createMockPublicKeyCredentialUserEntityRepository();
 		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(
 				this.userCredentials, this.userEntities);
 		Bytes credentialId = TestCredentialRecords.userCredential().build().getCredentialId();
@@ -135,6 +136,7 @@
 
 	@Test
 	void checkWhenCredentialBelongsToDifferentUserThenDenied() {
+		this.userEntities = MockPublicKeyCredentialUserEntityRepository.createMockPublicKeyCredentialUserEntityRepository();
 		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(
 				this.userCredentials, this.userEntities);
 		Bytes credentialId = TestCredentialRecords.userCredential().build().getCredentialId();
--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/management/Webauthn4jRelyingPartyOperationsTests.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/management/Webauthn4jRelyingPartyOperationsTests.java
@@ -93,8 +93,7 @@
 @ExtendWith(MockitoExtension.class)
 class Webauthn4jRelyingPartyOperationsTests {
 
-	@Mock
-	private PublicKeyCredentialUserEntityRepository userEntities;
+	private PublicKeyCredentialUserEntityRepository userEntities = MockPublicKeyCredentialUserEntityRepository.createMockPublicKeyCredentialUserEntityRepository();
 
 	@Mock
 	private UserCredentialRepository userCredentials;
@@ -623,6 +622,7 @@
 				user.getAuthorities());
 		PublicKeyCredentialUserEntity userEntity = TestPublicKeyCredentialUserEntities.userEntity().build();
 		CredentialRecord credentialRecord = TestCredentialRecords.userCredential().build();
+		this.userEntities = MockPublicKeyCredentialUserEntityRepository.createMockPublicKeyCredentialUserEntityRepository();
 		given(this.userEntities.findByUsername(user.getUsername())).willReturn(userEntity);
 		given(this.userCredentials.findByUserId(userEntity.getId())).willReturn(Arrays.asList(credentialRecord));
 		PublicKeyCredentialRequestOptionsRequest createRequest = new ImmutablePublicKeyCredentialRequestOptionsRequest(
--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/management/MockPublicKeyCredentialUserEntityRepository.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/management/MockPublicKeyCredentialUserEntityRepository.java
@@ -0,0 +1,14 @@
+package org.springframework.security.web.webauthn.management;
+
+import org.mockito.Mockito;
+
+public final class MockPublicKeyCredentialUserEntityRepository {
+
+	private MockPublicKeyCredentialUserEntityRepository() {
+	}
+
+	public static PublicKeyCredentialUserEntityRepository createMockPublicKeyCredentialUserEntityRepository() {
+		return Mockito.mock(PublicKeyCredentialUserEntityRepository.class);
+	}
+
+}
