--- a/web/src/test/java/org/springframework/security/web/server/WebFilterChainProxyTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/WebFilterChainProxyTests.java
@@ -23,6 +23,7 @@
 import io.micrometer.observation.Observation;
 import io.micrometer.observation.ObservationHandler;
 import io.micrometer.observation.ObservationRegistry;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import reactor.core.publisher.Mono;
@@ -59,6 +60,13 @@
  * @since 5.0
  */
 public class WebFilterChainProxyTests {
+
+	private ServerExchangeRejectedHandler exchangeRejectedHandler;
+
+	@BeforeEach
+	void setup() {
+		this.exchangeRejectedHandler = mock(ServerExchangeRejectedHandler.class);
+	}
 
 	// gh-4668
 	@Test
@@ -169,13 +177,12 @@
 		ServerWebExchangeMatcher notMatch = (exchange) -> MatchResult.notMatch();
 		MatcherSecurityWebFilterChain chain = new MatcherSecurityWebFilterChain(notMatch, filters);
 		WebFilterChainProxy filter = new WebFilterChainProxy(chain);
-		ServerExchangeRejectedHandler handler = mock(ServerExchangeRejectedHandler.class);
 		ServerWebExchangeFirewall firewall = mock(ServerWebExchangeFirewall.class);
 		filter.setFirewall(firewall);
-		filter.setExchangeRejectedHandler(handler);
+		filter.setExchangeRejectedHandler(exchangeRejectedHandler);
 		WebTestClient.bindToController(new Object()).webFilter(filter).build().get().exchange();
 		verify(firewall).getFirewalledExchange(any());
-		verifyNoInteractions(handler);
+		verifyNoInteractions(exchangeRejectedHandler);
 	}
 
 	@Test
@@ -184,15 +191,14 @@
 		ServerWebExchangeMatcher notMatch = (exchange) -> MatchResult.notMatch();
 		MatcherSecurityWebFilterChain chain = new MatcherSecurityWebFilterChain(notMatch, filters);
 		WebFilterChainProxy filter = new WebFilterChainProxy(chain);
-		ServerExchangeRejectedHandler handler = mock(ServerExchangeRejectedHandler.class);
 		ServerWebExchangeFirewall firewall = mock(ServerWebExchangeFirewall.class);
 		given(firewall.getFirewalledExchange(any()))
 			.willReturn(Mono.error(new ServerExchangeRejectedException("Oops")));
 		filter.setFirewall(firewall);
-		filter.setExchangeRejectedHandler(handler);
+		filter.setExchangeRejectedHandler(exchangeRejectedHandler);
 		WebTestClient.bindToController(new Object()).webFilter(filter).build().get().exchange();
 		verify(firewall).getFirewalledExchange(any());
-		verify(handler).handle(any(), any());
+		verify(exchangeRejectedHandler).handle(any(), any());
 	}
 
 	@Test
@@ -207,8 +213,7 @@
 		ServerWebExchangeMatcher match = (exchange) -> MatchResult.match();
 		MatcherSecurityWebFilterChain chain = new MatcherSecurityWebFilterChain(match, filters);
 		WebFilterChainProxy filter = new WebFilterChainProxy(chain);
-		ServerExchangeRejectedHandler handler = mock(ServerExchangeRejectedHandler.class);
-		filter.setExchangeRejectedHandler(handler);
+		filter.setExchangeRejectedHandler(exchangeRejectedHandler);
 		// @formatter:off
 		WebTestClient.bindToController(new Object())
 			.webFilter(filter)
@@ -216,7 +221,7 @@
 			.get()
 			.exchange();
 		// @formatter:on
-		verify(handler).handle(any(), any());
+		verify(exchangeRejectedHandler).handle(any(), any());
 	}
 
 	static void assertFilterChainObservation(Observation.Context context, String filterSection, int chainPosition) {
