--- a/core/src/test/java/org/springframework/security/authorization/AllAuthoritiesAuthorizationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AllAuthoritiesAuthorizationManagerTests.java
@@ -128,7 +128,7 @@
 	@Test
 	void setRoleHierarchyThenUsesResult() {
 		Collection result = AuthorityUtils.createAuthorityList(ROLE_USER, ROLE_ADMIN);
-		given(this.roleHierarchy.getReachableGrantedAuthorities(any())).willReturn(result);
+		this.roleHierarchy = createMockRoleHierarchy(result);
 		AllAuthoritiesAuthorizationManager<Object> manager = AllAuthoritiesAuthorizationManager
 			.hasAllAuthorities(ROLE_USER);
 		manager.setRoleHierarchy(this.roleHierarchy);
--- a/core/src/test/java/org/springframework/security/authorization/AllAuthoritiesReactiveAuthorizationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AllAuthoritiesReactiveAuthorizationManagerTests.java
@@ -133,7 +133,7 @@
 	@Test
 	void setRoleHierarchyThenUsesResult() {
 		Collection result = AuthorityUtils.createAuthorityList(ROLE_USER, ROLE_ADMIN);
-		given(this.roleHierarchy.getReachableGrantedAuthorities(any())).willReturn(result);
+		this.roleHierarchy = createMockRoleHierarchy(result);
 		AllAuthoritiesReactiveAuthorizationManager<Object> manager = AllAuthoritiesReactiveAuthorizationManager
 			.hasAllAuthorities(ROLE_USER);
 		manager.setRoleHierarchy(this.roleHierarchy);
--- a/core/src/test/java/org/springframework/security/authorization/MockRoleHierarchy.java
+++ b/core/src/test/java/org/springframework/security/authorization/MockRoleHierarchy.java
@@ -0,0 +1,23 @@
+package org.springframework.security.authorization;
+
+import java.util.Collection;
+
+import org.mockito.Mockito;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy;
+import org.springframework.security.core.GrantedAuthority;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+public final class MockRoleHierarchy {
+
+	private MockRoleHierarchy() {
+	}
+
+	public static RoleHierarchy createMockRoleHierarchy(Collection<GrantedAuthority> result) {
+		RoleHierarchy roleHierarchy = Mockito.mock(RoleHierarchy.class);
+		given(roleHierarchy.getReachableGrantedAuthorities(any())).willReturn(result);
+		return roleHierarchy;
+	}
+
+}
