--- a/core/src/test/java/org/springframework/security/core/context/InheritableThreadLocalSecurityContextHolderStrategyTests.java
+++ b/core/src/test/java/org/springframework/security/core/context/InheritableThreadLocalSecurityContextHolderStrategyTests.java
@@ -19,6 +19,7 @@
 import java.util.function.Supplier;
 
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.security.core.Authentication;
@@ -31,6 +32,13 @@
 class InheritableThreadLocalSecurityContextHolderStrategyTests {
 
 	InheritableThreadLocalSecurityContextHolderStrategy strategy = new InheritableThreadLocalSecurityContextHolderStrategy();
+
+	private Authentication authenticationMock;
+
+	@BeforeEach
+	void setup() {
+		this.authenticationMock = mock(Authentication.class);
+	}
 
 	@AfterEach
 	void clearContext() {
@@ -46,8 +54,7 @@
 
 	@Test
 	void deferredContext() {
-		Authentication authentication = mock(Authentication.class);
-		Supplier<SecurityContext> deferredContext = () -> new SecurityContextImpl(authentication);
+		Supplier<SecurityContext> deferredContext = () -> new SecurityContextImpl(authenticationMock);
 		this.strategy.setDeferredContext(deferredContext);
 		assertThat(this.strategy.getDeferredContext().get()).isEqualTo(deferredContext.get());
 		assertThat(this.strategy.getContext()).isEqualTo(deferredContext.get());
@@ -62,8 +69,7 @@
 
 	@Test
 	void context() {
-		Authentication authentication = mock(Authentication.class);
-		SecurityContext context = new SecurityContextImpl(authentication);
+		SecurityContext context = new SecurityContextImpl(authenticationMock);
 		this.strategy.setContext(context);
 		assertThat(this.strategy.getContext()).isEqualTo(context);
 		assertThat(this.strategy.getDeferredContext().get()).isEqualTo(context);
@@ -76,9 +82,8 @@
 
 	@Test
 	void getContextWhenEmptyThenReturnsSameInstance() {
-		Authentication authentication = mock(Authentication.class);
-		this.strategy.getContext().setAuthentication(authentication);
-		assertThat(this.strategy.getContext().getAuthentication()).isEqualTo(authentication);
+		this.strategy.getContext().setAuthentication(authenticationMock);
+		assertThat(this.strategy.getContext().getAuthentication()).isEqualTo(authenticationMock);
 	}
 
 }
