--- a/web/src/test/java/org/springframework/security/web/server/authorization/ServerWebExchangeDelegatingServerAccessDeniedHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/authorization/ServerWebExchangeDelegatingServerAccessDeniedHandlerTests.java
@@ -68,10 +68,9 @@
 	@Test
 	public void handleWhenFirstMatchesThenOnlyFirstInvoked() {
 		ServerAccessDeniedHandler firstHandler = mock(ServerAccessDeniedHandler.class);
-		ServerWebExchangeMatcher firstMatcher = mock(ServerWebExchangeMatcher.class);
 		ServerAccessDeniedHandler secondHandler = mock(ServerAccessDeniedHandler.class);
 		ServerWebExchangeMatcher secondMatcher = mock(ServerWebExchangeMatcher.class);
-		given(firstMatcher.matches(this.exchange)).willReturn(MatchResult.match());
+		ServerWebExchangeMatcher firstMatcher = createMockServerWebExchangeMatcher(this.exchange, MatchResult.match());
 		given(firstHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		given(secondHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		this.entries.add(new DelegateEntry(firstMatcher, firstHandler));
@@ -90,9 +89,8 @@
 		ServerAccessDeniedHandler firstHandler = mock(ServerAccessDeniedHandler.class);
 		ServerWebExchangeMatcher firstMatcher = mock(ServerWebExchangeMatcher.class);
 		ServerAccessDeniedHandler secondHandler = mock(ServerAccessDeniedHandler.class);
-		ServerWebExchangeMatcher secondMatcher = mock(ServerWebExchangeMatcher.class);
 		given(firstMatcher.matches(this.exchange)).willReturn(MatchResult.notMatch());
-		given(secondMatcher.matches(this.exchange)).willReturn(MatchResult.match());
+		ServerWebExchangeMatcher secondMatcher = createMockServerWebExchangeMatcher(this.exchange, MatchResult.match());
 		given(firstHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		given(secondHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		this.entries.add(new DelegateEntry(firstMatcher, firstHandler));
@@ -104,4 +102,11 @@
 		verify(this.accessDeniedHandler, never()).handle(this.exchange, null);
 	}
 
+	private static ServerWebExchangeMatcher createMockServerWebExchangeMatcher(ServerWebExchange exchange,
+			MatchResult matchResult) {
+		ServerWebExchangeMatcher matcher = mock(ServerWebExchangeMatcher.class);
+		given(matcher.matches(exchange)).willReturn(matchResult);
+		return matcher;
+	}
+
 }
