--- a/crypto/src/test/java/org/springframework/security/crypto/password/DelegatingPasswordEncoderTests.java
+++ b/crypto/src/test/java/org/springframework/security/crypto/password/DelegatingPasswordEncoderTests.java
@@ -24,6 +24,7 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -43,7 +44,6 @@
 @ExtendWith(MockitoExtension.class)
 public class DelegatingPasswordEncoderTests extends AbstractPasswordEncoderValidationTests {
 
-	@Mock
 	private PasswordEncoder bcrypt;
 
 	@Mock
@@ -76,6 +76,7 @@
 
 	@BeforeEach
 	public void setup() {
+		this.bcrypt = createMockPasswordEncoder(this.rawPassword, this.encodedPassword);
 		this.delegates = new HashMap<>();
 		this.delegates.put(this.bcryptId, this.bcrypt);
 		this.delegates.put("noop", this.noop);
@@ -172,7 +173,6 @@
 
 	@Test
 	public void matchesWhenBCryptThenDelegatesToBCrypt() {
-		given(this.bcrypt.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
 		assertThat(getEncoder().matches(this.rawPassword, this.bcryptEncodedPassword)).isTrue();
 		verify(this.bcrypt).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.noop);
@@ -180,7 +180,6 @@
 
 	@Test
 	public void matchesWhenBCryptBySpecifyDelegatingPasswordEncoderThenDelegatesToBCrypt() {
-		given(this.bcrypt.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
 		assertThat(this.onlySuffixPasswordEncoder.matches(this.rawPassword, "bcrypt$" + this.encodedPassword)).isTrue();
 		verify(this.bcrypt).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.noop);
@@ -188,7 +187,8 @@
 
 	@Test
 	public void matchesWhenNoopThenDelegatesToNoop() {
-		given(this.noop.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
+		this.noop = createMockPasswordEncoder(this.rawPassword, this.encodedPassword);
+		this.delegates.put("noop", this.noop);
 		assertThat(getEncoder().matches(this.rawPassword, this.noopEncodedPassword)).isTrue();
 		verify(this.noop).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.bcrypt);
@@ -247,9 +247,9 @@
 
 	@Test
 	public void matchesWhenNullIdThenDelegatesToInvalidId() {
+		this.invalidId = createMockPasswordEncoder(this.rawPassword, this.encodedPassword);
 		this.delegates.put(null, this.invalidId);
 		setEncoder(new DelegatingPasswordEncoder(this.bcryptId, this.delegates));
-		given(this.invalidId.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
 		assertThat(getEncoder().matches(this.rawPassword, this.encodedPassword)).isTrue();
 		verify(this.invalidId).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.bcrypt, this.noop);
@@ -298,4 +298,10 @@
 		verifyNoMoreInteractions(this.bcrypt, this.noop);
 	}
 
+	private static PasswordEncoder createMockPasswordEncoder(String rawPassword, String encodedPassword) {
+		PasswordEncoder bcrypt = Mockito.mock(PasswordEncoder.class);
+		given(bcrypt.matches(rawPassword, encodedPassword)).willReturn(true);
+		return bcrypt;
+	}
+
 }
