--- a/web/src/test/java/org/springframework/security/web/server/authorization/ServerWebExchangeDelegatingServerAccessDeniedHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/authorization/ServerWebExchangeDelegatingServerAccessDeniedHandlerTests.java
@@ -53,8 +53,7 @@
 	@Test
 	public void handleWhenNothingMatchesThenOnlyDefaultHandlerInvoked() {
 		ServerAccessDeniedHandler handler = mock(ServerAccessDeniedHandler.class);
-		ServerWebExchangeMatcher matcher = mock(ServerWebExchangeMatcher.class);
-		given(matcher.matches(this.exchange)).willReturn(MatchResult.notMatch());
+		ServerWebExchangeMatcher matcher = createMockServerWebExchangeMatcher(this.exchange);
 		given(handler.handle(this.exchange, null)).willReturn(Mono.empty());
 		given(this.accessDeniedHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		this.entries.add(new DelegateEntry(matcher, handler));
@@ -88,10 +87,9 @@
 	@Test
 	public void handleWhenSecondMatchesThenOnlySecondInvoked() {
 		ServerAccessDeniedHandler firstHandler = mock(ServerAccessDeniedHandler.class);
-		ServerWebExchangeMatcher firstMatcher = mock(ServerWebExchangeMatcher.class);
+		ServerWebExchangeMatcher firstMatcher = createMockServerWebExchangeMatcher(this.exchange);
 		ServerAccessDeniedHandler secondHandler = mock(ServerAccessDeniedHandler.class);
 		ServerWebExchangeMatcher secondMatcher = mock(ServerWebExchangeMatcher.class);
-		given(firstMatcher.matches(this.exchange)).willReturn(MatchResult.notMatch());
 		given(secondMatcher.matches(this.exchange)).willReturn(MatchResult.match());
 		given(firstHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		given(secondHandler.handle(this.exchange, null)).willReturn(Mono.empty());
@@ -104,4 +102,10 @@
 		verify(this.accessDeniedHandler, never()).handle(this.exchange, null);
 	}
 
+	private static ServerWebExchangeMatcher createMockServerWebExchangeMatcher(ServerWebExchange exchange) {
+		ServerWebExchangeMatcher matcher = mock(ServerWebExchangeMatcher.class);
+		given(matcher.matches(exchange)).willReturn(MatchResult.notMatch());
+		return matcher;
+	}
+
 }
