--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/AuthenticatedPrincipalOAuth2AuthorizedClientRepositoryTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/AuthenticatedPrincipalOAuth2AuthorizedClientRepositoryTests.java
@@ -53,9 +53,12 @@
 
 	private MockHttpServletResponse response;
 
+	private OAuth2AuthorizedClient authorizedClientMock;
+
 	@BeforeEach
 	public void setup() {
 		this.authorizedClientService = mock(OAuth2AuthorizedClientService.class);
+		this.authorizedClientMock = mock(OAuth2AuthorizedClient.class);
 		this.anonymousAuthorizedClientRepository = mock(OAuth2AuthorizedClientRepository.class);
 		this.authorizedClientRepository = new AuthenticatedPrincipalOAuth2AuthorizedClientRepository(
 				this.authorizedClientService);
@@ -95,19 +98,17 @@
 	@Test
 	public void saveAuthorizedClientWhenAuthenticatedPrincipalThenSaveToService() {
 		Authentication authentication = this.createAuthenticatedPrincipal();
-		OAuth2AuthorizedClient authorizedClient = mock(OAuth2AuthorizedClient.class);
-		this.authorizedClientRepository.saveAuthorizedClient(authorizedClient, authentication, this.request,
+		this.authorizedClientRepository.saveAuthorizedClient(authorizedClientMock, authentication, this.request,
 				this.response);
-		verify(this.authorizedClientService).saveAuthorizedClient(authorizedClient, authentication);
+		verify(this.authorizedClientService).saveAuthorizedClient(authorizedClientMock, authentication);
 	}
 
 	@Test
 	public void saveAuthorizedClientWhenAnonymousPrincipalThenSaveToAnonymousRepository() {
 		Authentication authentication = this.createAnonymousPrincipal();
-		OAuth2AuthorizedClient authorizedClient = mock(OAuth2AuthorizedClient.class);
-		this.authorizedClientRepository.saveAuthorizedClient(authorizedClient, authentication, this.request,
+		this.authorizedClientRepository.saveAuthorizedClient(authorizedClientMock, authentication, this.request,
 				this.response);
-		verify(this.anonymousAuthorizedClientRepository).saveAuthorizedClient(authorizedClient, authentication,
+		verify(this.anonymousAuthorizedClientRepository).saveAuthorizedClient(authorizedClientMock, authentication,
 				this.request, this.response);
 	}
 
