--- a/rsocket/src/test/java/org/springframework/security/rsocket/core/PayloadInterceptorRSocketTests.java
+++ b/rsocket/src/test/java/org/springframework/security/rsocket/core/PayloadInterceptorRSocketTests.java
@@ -33,6 +33,7 @@
 import org.mockito.ArgumentCaptor;
 import org.mockito.Captor;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.mockito.stubbing.Answer;
 import org.reactivestreams.Publisher;
@@ -139,7 +140,7 @@
 	@Test
 	public void fireAndForgetWhenInterceptorErrorsThenDelegateNotSubscribed() {
 		RuntimeException expected = new RuntimeException("Oops");
-		given(this.interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor), this.metadataMimeType, this.dataMimeType);
 		StepVerifier.create(interceptor.fireAndForget(this.payload))
@@ -187,7 +188,7 @@
 	@Test
 	public void requestResponseWhenInterceptorErrorsThenDelegateNotInvoked() {
 		RuntimeException expected = new RuntimeException("Oops");
-		given(this.interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor), this.metadataMimeType, this.dataMimeType);
 		assertThatExceptionOfType(RuntimeException.class)
@@ -239,7 +240,7 @@
 	@Test
 	public void requestStreamWhenInterceptorErrorsThenDelegateNotSubscribed() {
 		RuntimeException expected = new RuntimeException("Oops");
-		given(this.interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor), this.metadataMimeType, this.dataMimeType);
 		StepVerifier.create(interceptor.requestStream(this.payload))
@@ -344,7 +345,7 @@
 	@Test
 	public void requestChannelWhenInterceptorErrorsThenDelegateNotSubscribed() {
 		RuntimeException expected = new RuntimeException("Oops");
-		given(this.interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor), this.metadataMimeType, this.dataMimeType);
 		StepVerifier.create(interceptor.requestChannel(Flux.just(this.payload)))
@@ -394,7 +395,7 @@
 	@Test
 	public void metadataPushWhenInterceptorErrorsThenDelegateNotSubscribed() {
 		RuntimeException expected = new RuntimeException("Oops");
-		given(this.interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor), this.metadataMimeType, this.dataMimeType);
 		StepVerifier.create(interceptor.metadataPush(this.payload))
@@ -456,7 +457,7 @@
 	@Test
 	public void fireAndForgetWhenInterceptor1ErrorsThenInterceptor2AndDelegateNotInvoked() {
 		RuntimeException expected = new RuntimeException("Oops");
-		given(this.interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor, this.interceptor2), this.metadataMimeType, this.dataMimeType);
 		assertThatExceptionOfType(RuntimeException.class)
@@ -472,7 +473,7 @@
 	public void fireAndForgetWhenInterceptor2ErrorsThenInterceptor2AndDelegateNotInvoked() {
 		RuntimeException expected = new RuntimeException("Oops");
 		given(this.interceptor.intercept(any(), any())).willAnswer(withChainNext());
-		given(this.interceptor2.intercept(any(), any())).willReturn(Mono.error(expected));
+		this.interceptor2 = createMockPayloadInterceptor(expected, true);
 		PayloadInterceptorRSocket interceptor = new PayloadInterceptorRSocket(this.delegate,
 				Arrays.asList(this.interceptor, this.interceptor2), this.metadataMimeType, this.dataMimeType);
 		assertThatExceptionOfType(RuntimeException.class)
@@ -482,6 +483,15 @@
 		assertThat(this.exchange.getValue().getPayload()).isEqualTo(this.payload);
 		verify(this.interceptor2).intercept(any(), any());
 		this.voidResult.assertWasNotSubscribed();
+	}
+
+	private static PayloadInterceptor createMockPayloadInterceptor(RuntimeException expected,
+			boolean stubError) {
+		PayloadInterceptor interceptor = Mockito.mock(PayloadInterceptor.class);
+		if (stubError) {
+			given(interceptor.intercept(any(), any())).willReturn(Mono.error(expected));
+		}
+		return interceptor;
 	}
 
 	private Mono<Authentication> assertAuthentication(Authentication authentication) {
