--- a/web/src/test/java/org/springframework/security/web/context/DelegatingSecurityContextRepositoryTests.java
+++ b/web/src/test/java/org/springframework/security/web/context/DelegatingSecurityContextRepositoryTests.java
@@ -55,12 +55,15 @@
 
 	private SecurityContext securityContext;
 
+	private SecurityContext securityContext1Mock;
+
 	@BeforeEach
 	public void setUp() {
 		this.request = new MockHttpServletRequest();
 		this.response = new MockHttpServletResponse();
 		this.strategy = mock(SecurityContextHolderStrategy.class);
 		this.securityContext = mock(SecurityContext.class);
+		this.securityContext1Mock = mock(SecurityContext.class);
 	}
 
 	@ParameterizedTest
@@ -147,10 +150,9 @@
 		SecurityContextRepository delegate1 = mock(SecurityContextRepository.class);
 		SecurityContextRepository delegate2 = mock(SecurityContextRepository.class);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(this.request, this.response);
-		SecurityContext securityContext1 = mock(SecurityContext.class);
 		SecurityContext securityContext2 = mock(SecurityContext.class);
 
-		given(delegate1.loadContext(holder)).willReturn(securityContext1);
+		given(delegate1.loadContext(holder)).willReturn(securityContext1Mock);
 		given(delegate1.containsContext(holder.getRequest())).willReturn(false);
 		given(delegate2.loadContext(holder)).willReturn(securityContext2);
 		given(delegate2.containsContext(holder.getRequest())).willReturn(true);
@@ -167,10 +169,9 @@
 		SecurityContextRepository delegate1 = mock(SecurityContextRepository.class);
 		SecurityContextRepository delegate2 = mock(SecurityContextRepository.class);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(this.request, this.response);
-		SecurityContext securityContext1 = mock(SecurityContext.class);
 		SecurityContext securityContext2 = mock(SecurityContext.class);
 
-		given(delegate1.loadContext(holder)).willReturn(securityContext1);
+		given(delegate1.loadContext(holder)).willReturn(securityContext1Mock);
 		given(delegate1.containsContext(holder.getRequest())).willReturn(true);
 		given(delegate2.loadContext(holder)).willReturn(securityContext2);
 		given(delegate2.containsContext(holder.getRequest())).willReturn(true);
@@ -187,10 +188,9 @@
 		SecurityContextRepository delegate1 = mock(SecurityContextRepository.class);
 		SecurityContextRepository delegate2 = mock(SecurityContextRepository.class);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(this.request, this.response);
-		SecurityContext securityContext1 = mock(SecurityContext.class);
 		SecurityContext securityContext2 = mock(SecurityContext.class);
 
-		given(delegate1.loadContext(holder)).willReturn(securityContext1);
+		given(delegate1.loadContext(holder)).willReturn(securityContext1Mock);
 		given(delegate1.containsContext(holder.getRequest())).willReturn(true);
 		given(delegate2.loadContext(holder)).willReturn(securityContext2);
 		given(delegate2.containsContext(holder.getRequest())).willReturn(false);
@@ -198,7 +198,7 @@
 		DelegatingSecurityContextRepository repository = new DelegatingSecurityContextRepository(delegate1, delegate2);
 		SecurityContext returnedSecurityContext = repository.loadContext(holder);
 
-		assertThat(returnedSecurityContext).isSameAs(securityContext1);
+		assertThat(returnedSecurityContext).isSameAs(securityContext1Mock);
 	}
 
 }
