--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
@@ -118,10 +118,7 @@
 		claims.put("aud", principal.getAudience());
 		claims.put("auth_time", authTime);
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -157,59 +154,61 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", principal.getNonce());
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
+		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
+		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
+		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
+		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
+		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
+		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
+		handler.setUserService(userService);
+		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
+		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
+			.verifyErrorMessage("[invalid_id_token] Invalid issuer");
+	}
+
+	@Test
+	void onAuthorizationSuccessWhenIdTokenSubNotSameThenException() {
+		ClientRegistration clientRegistration = TestClientRegistrations.clientRegistration().build();
+		DefaultOidcUser principal = TestOidcUsers.create();
+		OAuth2AuthenticationToken authenticationToken = new OAuth2AuthenticationToken(principal,
+				principal.getAuthorities(), clientRegistration.getRegistrationId());
+		OAuth2AccessToken accessToken = createAccessToken();
+		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(clientRegistration, principal.getName(),
+				accessToken, null);
+		MockServerWebExchange exchange = MockServerWebExchange.from(MockServerHttpRequest.get("/").build());
+		Map<String, Object> attributes = Map.of(ServerWebExchange.class.getName(), exchange,
+				OidcParameterNames.ID_TOKEN, "id-token-1234");
+		Map<String, Object> claims = new HashMap<>();
+		claims.put("iss", principal.getIssuer());
+		claims.put("sub", "invalid_sub");
+		claims.put("aud", principal.getAudience());
+		claims.put("nonce", principal.getNonce());
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
+		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
+		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
+		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
+		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
+		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
+		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
+		handler.setUserService(userService);
+		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
+		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
+			.verifyErrorMessage("[invalid_id_token] Invalid subject");
+	}
+
+	private static Jwt createMockJwt(Map<String, Object> claims, Instant issuedAt) {
 		Jwt jwt = mock(Jwt.class);
 		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
+		given(jwt.getIssuedAt()).willReturn(issuedAt);
 		given(jwt.getClaims()).willReturn(claims);
-		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
-		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
-		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
-		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
-		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
-		handler.setUserService(userService);
-		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
-		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
-			.verifyErrorMessage("[invalid_id_token] Invalid issuer");
-	}
-
-	@Test
-	void onAuthorizationSuccessWhenIdTokenSubNotSameThenException() {
-		ClientRegistration clientRegistration = TestClientRegistrations.clientRegistration().build();
-		DefaultOidcUser principal = TestOidcUsers.create();
-		OAuth2AuthenticationToken authenticationToken = new OAuth2AuthenticationToken(principal,
-				principal.getAuthorities(), clientRegistration.getRegistrationId());
-		OAuth2AccessToken accessToken = createAccessToken();
-		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(clientRegistration, principal.getName(),
-				accessToken, null);
-		MockServerWebExchange exchange = MockServerWebExchange.from(MockServerHttpRequest.get("/").build());
-		Map<String, Object> attributes = Map.of(ServerWebExchange.class.getName(), exchange,
-				OidcParameterNames.ID_TOKEN, "id-token-1234");
-		Map<String, Object> claims = new HashMap<>();
-		claims.put("iss", principal.getIssuer());
-		claims.put("sub", "invalid_sub");
-		claims.put("aud", principal.getAudience());
-		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
-		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
-		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
-		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
-		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
-		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
-		handler.setUserService(userService);
-		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
-		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
-			.verifyErrorMessage("[invalid_id_token] Invalid subject");
+		return jwt;
 	}
 
 	@Test
@@ -229,10 +228,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt().minus(Duration.ofDays(1)));
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt().minus(Duration.ofDays(1)));
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -265,10 +261,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", Collections.emptyList());
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -301,10 +294,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", List.of("invalid_client-id"));
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -339,10 +329,7 @@
 		claims.put("aud", principal.getAudience());
 		claims.put("auth_time", authTime.minus(5, ChronoUnit.MINUTES));
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -377,10 +364,7 @@
 		claims.put("aud", principal.getAudience());
 		claims.put("auth_time", authTime.plus(5, ChronoUnit.MINUTES));
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -416,23 +400,28 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", "invalid_nonce");
+		Jwt jwt = createMockJwt(claims, principal.getIssuedAt());
+		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
+		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
+		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
+		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
+		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
+		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
+		handler.setUserService(userService);
+		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
+		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
+			.verifyErrorMessage("[invalid_nonce] Invalid nonce");
+	}
+
+	private static Jwt createMockJwt(Map<String, Object> claims, Instant issuedAt) {
 		Jwt jwt = mock(Jwt.class);
 		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
+		given(jwt.getIssuedAt()).willReturn(issuedAt);
 		given(jwt.getClaims()).willReturn(claims);
-		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
-		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
-		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
-		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
-		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
-		handler.setUserService(userService);
-		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
-		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
-			.verifyErrorMessage("[invalid_nonce] Invalid nonce");
+		return jwt;
 	}
 
 	private static OAuth2AccessToken createAccessToken() {
