--- a/access/src/test/java/org/springframework/security/web/access/intercept/FilterSecurityInterceptorTests.java
+++ b/access/src/test/java/org/springframework/security/web/access/intercept/FilterSecurityInterceptorTests.java
@@ -69,6 +69,8 @@
 
 	private AccessDecisionManager adm;
 
+	private AfterInvocationManager afterInvocationManager;
+
 	private FilterInvocationSecurityMetadataSource ods;
 
 	private RunAsManager ram;
@@ -83,6 +85,7 @@
 		this.am = mock(AuthenticationManager.class);
 		this.ods = mock(FilterInvocationSecurityMetadataSource.class);
 		this.adm = mock(AccessDecisionManager.class);
+		this.afterInvocationManager = mock(AfterInvocationManager.class);
 		this.ram = mock(RunAsManager.class);
 		this.publisher = mock(ApplicationEventPublisher.class);
 		this.interceptor.setAuthenticationManager(this.am);
@@ -136,10 +139,9 @@
 		willThrow(new RuntimeException()).given(chain)
 			.doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		given(this.ods.getAttributes(fi)).willReturn(SecurityConfig.createList("MOCK_OK"));
-		AfterInvocationManager aim = mock(AfterInvocationManager.class);
-		this.interceptor.setAfterInvocationManager(aim);
+		this.interceptor.setAfterInvocationManager(afterInvocationManager);
 		assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> this.interceptor.invoke(fi));
-		verifyNoMoreInteractions(aim);
+		verifyNoMoreInteractions(afterInvocationManager);
 	}
 
 	// SEC-1967
@@ -159,8 +161,7 @@
 		willThrow(new RuntimeException()).given(chain)
 			.doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		given(this.ods.getAttributes(fi)).willReturn(SecurityConfig.createList("MOCK_OK"));
-		AfterInvocationManager aim = mock(AfterInvocationManager.class);
-		this.interceptor.setAfterInvocationManager(aim);
+		this.interceptor.setAfterInvocationManager(afterInvocationManager);
 		assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> this.interceptor.invoke(fi));
 		// Check we've changed back
 		assertThat(SecurityContextHolder.getContext()).isSameAs(ctx);
