--- a/acl/src/test/java/org/springframework/security/acls/AclPermissionEvaluatorTests.java
+++ b/acl/src/test/java/org/springframework/security/acls/AclPermissionEvaluatorTests.java
@@ -45,8 +45,7 @@
 		AclService service = mock(AclService.class);
 		AclPermissionEvaluator pe = new AclPermissionEvaluator(service);
 		ObjectIdentity oid = mock(ObjectIdentity.class);
-		ObjectIdentityRetrievalStrategy oidStrategy = mock(ObjectIdentityRetrievalStrategy.class);
-		given(oidStrategy.getObjectIdentity(any(Object.class))).willReturn(oid);
+		ObjectIdentityRetrievalStrategy oidStrategy = createMockObjectIdentityRetrievalStrategy(oid);
 		pe.setObjectIdentityRetrievalStrategy(oidStrategy);
 		pe.setSidRetrievalStrategy(mock(SidRetrievalStrategy.class));
 		Acl acl = mock(Acl.class);
@@ -62,8 +61,7 @@
 		AclService service = mock(AclService.class);
 		AclPermissionEvaluator pe = new AclPermissionEvaluator(service);
 		ObjectIdentity oid = mock(ObjectIdentity.class);
-		ObjectIdentityRetrievalStrategy oidStrategy = mock(ObjectIdentityRetrievalStrategy.class);
-		given(oidStrategy.getObjectIdentity(any(Object.class))).willReturn(oid);
+		ObjectIdentityRetrievalStrategy oidStrategy = createMockObjectIdentityRetrievalStrategy(oid);
 		pe.setObjectIdentityRetrievalStrategy(oidStrategy);
 		pe.setSidRetrievalStrategy(mock(SidRetrievalStrategy.class));
 		Acl acl = mock(Acl.class);
@@ -73,4 +71,10 @@
 		Locale.setDefault(systemLocale);
 	}
 
+	private static ObjectIdentityRetrievalStrategy createMockObjectIdentityRetrievalStrategy(ObjectIdentity oid) {
+		ObjectIdentityRetrievalStrategy oidStrategy = mock(ObjectIdentityRetrievalStrategy.class);
+		given(oidStrategy.getObjectIdentity(any(Object.class))).willReturn(oid);
+		return oidStrategy;
+	}
+
 }
