--- a/core/src/test/java/org/springframework/security/authorization/AuthorizationManagerFactoryTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthorizationManagerFactoryTests.java
@@ -124,7 +124,7 @@
 
 	@Test
 	public void anonymousWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -135,7 +135,7 @@
 
 	@Test
 	public void permitAllWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -146,7 +146,7 @@
 
 	@Test
 	public void denyAllAllWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -157,9 +157,7 @@
 
 	@Test
 	public void hasRoleWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -172,9 +170,7 @@
 
 	@Test
 	public void hasAnyRoleWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -187,9 +183,7 @@
 
 	@Test
 	public void hasAllRolesWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -202,9 +196,7 @@
 
 	@Test
 	public void hasAuthorityWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -217,9 +209,7 @@
 
 	@Test
 	public void hasAnyAuthorityWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -232,9 +222,7 @@
 
 	@Test
 	public void hasAllAuthoritiesWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -246,9 +234,7 @@
 
 	@Test
 	public void authenticatedWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -260,9 +246,7 @@
 
 	@Test
 	public void fullyAuthenticatedWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -274,9 +258,7 @@
 
 	@Test
 	public void rememberMeWhenAdditionalAuthorizationThenInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		given(additional.authorize(any(), any())).willReturn(new AuthorizationDecision(true),
-				new AuthorizationDecision(false));
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -366,6 +348,11 @@
 			.withMessage("condition cannot be null");
 	}
 
+	private static AuthorizationManager<String> createMockAdditionalAuthorization() {
+		AuthorizationManager<String> additional = createMockAdditionalAuthorization();
+		return additional;
+	}
+
 	private void assertUserGranted(AuthorizationManager<String> manager) {
 		assertThat(manager.authorize(() -> TestAuthentication.authenticatedUser(), "").isGranted()).isTrue();
 	}
