--- a/oauth2/oauth2-core/src/test/java/org/springframework/security/oauth2/core/DelegatingOAuth2TokenValidatorTests.java
+++ b/oauth2/oauth2-core/src/test/java/org/springframework/security/oauth2/core/DelegatingOAuth2TokenValidatorTests.java
@@ -19,6 +19,7 @@
 import java.util.Arrays;
 import java.util.Collection;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -38,11 +39,17 @@
 
 	private static final OAuth2Error DETAIL = new OAuth2Error("error", "description", "uri");
 
+	private OAuth2Token tokenMock;
+
+	@BeforeEach
+	public void setup() {
+		tokenMock = mock(OAuth2Token.class);
+	}
+
 	@Test
 	public void validateWhenNoValidatorsConfiguredThenReturnsSuccessfulResult() {
 		DelegatingOAuth2TokenValidator<OAuth2Token> tokenValidator = new DelegatingOAuth2TokenValidator<>();
-		OAuth2Token token = mock(OAuth2Token.class);
-		assertThat(tokenValidator.validate(token).hasErrors()).isFalse();
+		assertThat(tokenValidator.validate(tokenMock).hasErrors()).isFalse();
 	}
 
 	@Test
@@ -53,8 +60,7 @@
 		given(failure.validate(any(OAuth2Token.class))).willReturn(OAuth2TokenValidatorResult.failure(DETAIL));
 		DelegatingOAuth2TokenValidator<OAuth2Token> tokenValidator = new DelegatingOAuth2TokenValidator<>(
 				Arrays.asList(success, failure));
-		OAuth2Token token = mock(OAuth2Token.class);
-		OAuth2TokenValidatorResult result = tokenValidator.validate(token);
+		OAuth2TokenValidatorResult result = tokenValidator.validate(tokenMock);
 		assertThat(result.hasErrors()).isTrue();
 		assertThat(result.getErrors()).containsExactly(DETAIL);
 	}
@@ -69,8 +75,7 @@
 			.willReturn(OAuth2TokenValidatorResult.failure(otherDetail));
 		DelegatingOAuth2TokenValidator<OAuth2Token> tokenValidator = new DelegatingOAuth2TokenValidator<>(firstFailure,
 				secondFailure);
-		OAuth2Token token = mock(OAuth2Token.class);
-		OAuth2TokenValidatorResult result = tokenValidator.validate(token);
+		OAuth2TokenValidatorResult result = tokenValidator.validate(tokenMock);
 		assertThat(result.hasErrors()).isTrue();
 		assertThat(result.getErrors()).containsExactly(DETAIL, otherDetail);
 	}
@@ -83,8 +88,7 @@
 		given(secondSuccess.validate(any(OAuth2Token.class))).willReturn(OAuth2TokenValidatorResult.success());
 		DelegatingOAuth2TokenValidator<OAuth2Token> tokenValidator = new DelegatingOAuth2TokenValidator<>(
 				Arrays.asList(firstSuccess, secondSuccess));
-		OAuth2Token token = mock(OAuth2Token.class);
-		OAuth2TokenValidatorResult result = tokenValidator.validate(token);
+		OAuth2TokenValidatorResult result = tokenValidator.validate(tokenMock);
 		assertThat(result.hasErrors()).isFalse();
 		assertThat(result.getErrors()).isEmpty();
 	}
@@ -105,11 +109,10 @@
 				Arrays.asList(firstSuccess, secondSuccess));
 		DelegatingOAuth2TokenValidator<OAuth2Token> secondValidator = new DelegatingOAuth2TokenValidator<>(firstSuccess,
 				secondSuccess);
-		OAuth2Token token = mock(OAuth2Token.class);
-		firstValidator.validate(token);
-		secondValidator.validate(token);
-		verify(firstSuccess, times(2)).validate(token);
-		verify(secondSuccess, times(2)).validate(token);
+		firstValidator.validate(tokenMock);
+		secondValidator.validate(tokenMock);
+		verify(firstSuccess, times(2)).validate(tokenMock);
+		verify(secondSuccess, times(2)).validate(tokenMock);
 	}
 
 }
