--- a/web/src/test/java/org/springframework/security/web/authentication/DelegatingAuthenticationEntryPointTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/DelegatingAuthenticationEntryPointTests.java
@@ -53,11 +53,14 @@
 
 	private AuthenticationEntryPoint defaultEntryPoint;
 
+	private RequestMatcher requestMatcher;
+
 	private HttpServletRequest request = new MockHttpServletRequest();
 
 	@BeforeEach
 	public void before() {
 		this.defaultEntryPoint = mock(AuthenticationEntryPoint.class);
+		this.requestMatcher = mock(RequestMatcher.class);
 	}
 
 	@Test
@@ -79,15 +82,14 @@
 		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher firstRM = mock(RequestMatcher.class);
 		AuthenticationEntryPoint secondAEP = mock(AuthenticationEntryPoint.class);
-		RequestMatcher secondRM = mock(RequestMatcher.class);
 		given(firstRM.matches(this.request)).willReturn(true);
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint,
-				new RequestMatcherEntry<>(firstRM, firstAEP), new RequestMatcherEntry<>(secondRM, secondAEP));
+				new RequestMatcherEntry<>(firstRM, firstAEP), new RequestMatcherEntry<>(requestMatcher, secondAEP));
 		this.daep.commence(this.request, null, null);
 		verify(firstAEP).commence(this.request, null, null);
 		verify(secondAEP, never()).commence(this.request, null, null);
 		verify(this.defaultEntryPoint, never()).commence(this.request, null, null);
-		verify(secondRM, never()).matches(this.request);
+		verify(requestMatcher, never()).matches(this.request);
 	}
 
 	@Test
@@ -127,9 +129,8 @@
 	@SuppressWarnings("removal")
 	public void constructorAepListWhenNullDefaultEntryPoint() {
 		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
-		RequestMatcher matcher = mock(RequestMatcher.class);
 		List<RequestMatcherEntry<AuthenticationEntryPoint>> entryPoints = List
-			.of(new RequestMatcherEntry<>(matcher, entryPoint));
+			.of(new RequestMatcherEntry<>(requestMatcher, entryPoint));
 		assertThatIllegalArgumentException()
 			.isThrownBy(() -> new DelegatingAuthenticationEntryPoint(null, entryPoints));
 	}
@@ -154,9 +155,8 @@
 	@SuppressWarnings("removal")
 	public void constructorAepVargsWhenNullDefaultEntryPoint() {
 		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
-		RequestMatcher matcher = mock(RequestMatcher.class);
 		RequestMatcherEntry<AuthenticationEntryPoint>[] entryPoints = new RequestMatcherEntry[] {
-				new RequestMatcherEntry<>(matcher, entryPoint) };
+				new RequestMatcherEntry<>(requestMatcher, entryPoint) };
 		assertThatIllegalArgumentException()
 			.isThrownBy(() -> new DelegatingAuthenticationEntryPoint(null, entryPoints));
 	}
