--- a/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
+++ b/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
@@ -118,6 +118,9 @@
 	@Mock
 	private ServerCsrfTokenRepository csrfTokenRepository;
 
+	@Mock
+	private ReactiveClientRegistrationRepository clientRegistrationRepositoryMock;
+
 	private ServerHttpSecurity http;
 
 	@BeforeEach
@@ -604,10 +607,8 @@
 	@SuppressWarnings("unchecked")
 	public void shouldConfigureRequestCacheForOAuth2LoginAuthenticationEntryPointAndSuccessHandler() {
 		ServerRequestCache requestCache = spy(new WebSessionServerRequestCache());
-		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
-				ReactiveClientRegistrationRepository.class);
-		SecurityWebFilterChain securityFilterChain = this.http
-			.oauth2Login((login) -> login.clientRegistrationRepository(clientRegistrationRepository))
+		SecurityWebFilterChain securityFilterChain = this.http
+			.oauth2Login((login) -> login.clientRegistrationRepository(clientRegistrationRepositoryMock))
 			.authorizeExchange((authorize) -> authorize.anyExchange().authenticated())
 			.requestCache((c) -> c.requestCache(requestCache))
 			.build();
@@ -630,13 +631,11 @@
 	public void shouldConfigureAuthorizationRequestRepositoryForOAuth2Login() {
 		ServerAuthorizationRequestRepository<OAuth2AuthorizationRequest> authorizationRequestRepository = mock(
 				ServerAuthorizationRequestRepository.class);
-		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
-				ReactiveClientRegistrationRepository.class);
 		OAuth2AuthorizationRequest authorizationRequest = TestOAuth2AuthorizationRequests.request().build();
 		given(authorizationRequestRepository.removeAuthorizationRequest(any()))
 			.willReturn(Mono.just(authorizationRequest));
 		SecurityWebFilterChain securityFilterChain = this.http
-			.oauth2Login((login) -> login.clientRegistrationRepository(clientRegistrationRepository)
+			.oauth2Login((login) -> login.clientRegistrationRepository(clientRegistrationRepositoryMock)
 				.authorizationRequestRepository(authorizationRequestRepository))
 			.build();
 		WebTestClient client = WebTestClientBuilder.bindToWebFilters(securityFilterChain).build();
