--- a/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLoginTests.java
+++ b/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLoginTests.java
@@ -94,8 +94,7 @@
 	 */
 	@Test
 	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
-		RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
-		given(postProcessor.postProcessRequest(any())).willAnswer((i) -> i.getArgument(0));
+		RequestPostProcessor postProcessor = MockRequestPostProcessor.createMockRequestPostProcessor();
 		MockMvc mockMvc = MockMvcBuilders.standaloneSetup(new Object())
 			.defaultRequest(MockMvcRequestBuilders.get("/").with(postProcessor))
 			.build();
--- a/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLogoutTests.java
+++ b/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLogoutTests.java
@@ -83,8 +83,7 @@
 	 */
 	@Test
 	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
-		RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
-		given(postProcessor.postProcessRequest(any())).willAnswer((i) -> i.getArgument(0));
+		RequestPostProcessor postProcessor = MockRequestPostProcessor.createMockRequestPostProcessor();
 		MockMvc mockMvc = MockMvcBuilders.standaloneSetup(new Object())
 			.defaultRequest(MockMvcRequestBuilders.get("/").with(postProcessor))
 			.build();
--- a/test/src/test/java/org/springframework/security/test/web/servlet/request/MockRequestPostProcessor.java
+++ b/test/src/test/java/org/springframework/security/test/web/servlet/request/MockRequestPostProcessor.java
@@ -0,0 +1,19 @@
+package org.springframework.security.test.web.servlet.request;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+import org.springframework.test.web.servlet.request.RequestPostProcessor;
+
+public final class MockRequestPostProcessor {
+
+	private MockRequestPostProcessor() {
+	}
+
+	public static RequestPostProcessor createMockRequestPostProcessor() {
+		RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
+		given(postProcessor.postProcessRequest(any())).willAnswer((i) -> i.getArgument(0));
+		return postProcessor;
+	}
+}
