--- a/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java
+++ b/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java
@@ -330,11 +330,9 @@
 	@Test
 	public void decodeWhenIssuerLocationThenOk() {
 		String issuer = "https://example.org/issuer";
-		RestOperations restOperations = mock(RestOperations.class);
+		RestOperations restOperations = createMockRestOperations();
 		given(restOperations.exchange(any(RequestEntity.class), any(ParameterizedTypeReference.class)))
 			.willReturn(new ResponseEntity<>(Map.of("issuer", issuer, "jwks_uri", issuer + "/jwks"), HttpStatus.OK));
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
 		NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder.withIssuerLocation(issuer)
 			.restOperations(restOperations)
 			.build();
@@ -345,9 +343,7 @@
 
 	@Test
 	public void decodeWhenDiscoverJwsAlgorithmsThenOk() {
-		RestOperations restOperations = mock(RestOperations.class);
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
+		RestOperations restOperations = createMockRestOperations();
 		JwtDecoder jwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 			.discoverJwsAlgorithms()
 			.restOperations(restOperations)
@@ -359,11 +355,9 @@
 	@Test
 	public void decodeWhenIssuerLocationThenRejectsMismatchingIssuers() {
 		String issuer = "https://example.org/wrong-issuer";
-		RestOperations restOperations = mock(RestOperations.class);
+		RestOperations restOperations = createMockRestOperations();
 		given(restOperations.exchange(any(RequestEntity.class), any(ParameterizedTypeReference.class)))
 			.willReturn(new ResponseEntity<>(Map.of("issuer", issuer, "jwks_uri", issuer + "/jwks"), HttpStatus.OK));
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
 		JwtDecoder jwtDecoder = NimbusJwtDecoder.withIssuerLocation(issuer).restOperations(restOperations).build();
 		assertThatExceptionOfType(JwtValidationException.class).isThrownBy(() -> jwtDecoder.decode(SIGNED_JWT));
 	}
@@ -692,9 +686,7 @@
 	// gh-7290
 	@Test
 	public void decodeWhenJwkSetRequestedThenAcceptHeaderJsonAndJwkSetJson() {
-		RestOperations restOperations = mock(RestOperations.class);
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
+		RestOperations restOperations = createMockRestOperations();
 		// @formatter:off
 		JWTProcessor<SecurityContext> processor = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 				.restOperations(restOperations)
@@ -711,9 +703,7 @@
 	@Test
 	public void decodeWhenCacheThenStoreRetrievedJwkSetToCache() {
 		Cache cache = new ConcurrentMapCache("test-jwk-set-cache");
-		RestOperations restOperations = mock(RestOperations.class);
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
+		RestOperations restOperations = createMockRestOperations();
 		// @formatter:off
 		NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 				.restOperations(restOperations)
@@ -732,9 +722,7 @@
 	@Test
 	public void decodeWhenCacheStoredThenAbleToRetrieveJwkSetFromCache() {
 		Cache cache = new ConcurrentMapCache("test-jwk-set-cache");
-		RestOperations restOperations = mock(RestOperations.class);
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
+		RestOperations restOperations = createMockRestOperations();
 		// @formatter:off
 		NimbusJwtDecoder jwtDecoder1 = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 				.restOperations(restOperations)
@@ -864,9 +852,7 @@
 	// gh-8730
 	@Test
 	public void withJwkSetUriWhenUsingCustomTypeHeaderThenRefuseOmittedType() throws Exception {
-		RestOperations restOperations = mock(RestOperations.class);
-		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
-			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
+		RestOperations restOperations = createMockRestOperations();
 		// @formatter:off
 		NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 				.restOperations(restOperations)
@@ -915,6 +901,13 @@
 		jwtDecoder.decode(jwt.serialize());
 	}
 
+	private static RestOperations createMockRestOperations() {
+		RestOperations restOperations = mock(RestOperations.class);
+		given(restOperations.exchange(any(RequestEntity.class), eq(String.class)))
+			.willReturn(new ResponseEntity<>(JWK_SET, HttpStatus.OK));
+		return restOperations;
+	}
+
 	private RSAPublicKey key() throws InvalidKeySpecException {
 		byte[] decoded = Base64.getDecoder().decode(VERIFY_KEY.getBytes());
 		EncodedKeySpec spec = new X509EncodedKeySpec(decoded);
