--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
@@ -27,6 +27,7 @@
 
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
@@ -34,6 +35,8 @@
 import static org.awaitility.Awaitility.await;
 
 class MigrationRuleListenerTest {
+
+    private MigrationRuleHandler<?> handlerMock;
 
     private String localRule = "key: demo-consumer\n" + "step: APPLICATION_FIRST\n"
             + "threshold: 1.0\n"
@@ -72,6 +75,11 @@
             + "force: false\n"
             + "interfaces:\n";
 
+    @BeforeEach
+    public void setUp() {
+        handlerMock = Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
+    }
+
     @AfterEach
     public void tearDown() {
         ApplicationModel.reset();
@@ -101,9 +109,6 @@
         Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
 
         System.setProperty("dubbo.application.migration.delay", "1");
-        MigrationRuleHandler<?> handler =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
-
         CountDownLatch countDownLatch = new CountDownLatch(1);
         MigrationRuleListener migrationRuleListener =
                 new MigrationRuleListener(ApplicationModel.defaultModel().getDefaultModule()) {
@@ -119,16 +124,16 @@
                 };
 
         MigrationInvoker<?> migrationInvoker = Mockito.mock(MigrationInvoker.class);
-        migrationRuleListener.getHandlers().put(migrationInvoker, handler);
+        migrationRuleListener.getHandlers().put(migrationInvoker, handlerMock);
 
         countDownLatch.countDown();
         await().untilAsserted(() -> {
-            Mockito.verify(handler).doMigrate(Mockito.any());
+            Mockito.verify(handlerMock).doMigrate(Mockito.any());
         });
         //        Mockito.verify(handler, Mockito.timeout(5000)).doMigrate(Mockito.any());
 
         migrationRuleListener.onRefer(null, migrationInvoker, consumerURL, null);
-        Mockito.verify(handler, Mockito.times(2)).doMigrate(Mockito.any());
+        Mockito.verify(handlerMock, Mockito.times(2)).doMigrate(Mockito.any());
     }
 
     /**
@@ -147,16 +152,13 @@
         Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
 
         System.setProperty("dubbo.application.migration.delay", "1000");
-        MigrationRuleHandler<?> handler =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
-
         MigrationRuleListener migrationRuleListener =
                 new MigrationRuleListener(ApplicationModel.defaultModel().getDefaultModule());
         MigrationInvoker<?> migrationInvoker = Mockito.mock(MigrationInvoker.class);
-        migrationRuleListener.getHandlers().put(migrationInvoker, handler);
+        migrationRuleListener.getHandlers().put(migrationInvoker, handlerMock);
         migrationRuleListener.onRefer(null, migrationInvoker, consumerURL, null);
         // check migration happened after invoker referred
-        Mockito.verify(handler, Mockito.times(1)).doMigrate(MigrationRule.getInitRule());
+        Mockito.verify(handlerMock, Mockito.times(1)).doMigrate(MigrationRule.getInitRule());
 
         // check no delay tasks created for there's no local rule and no config center
         Assertions.assertNull(migrationRuleListener.localRuleMigrationFuture);
@@ -192,8 +194,6 @@
         Mockito.when(consumerURL2.getParameter("timestamp")).thenReturn("2");
 
         System.setProperty("dubbo.application.migration.delay", "10");
-        MigrationRuleHandler<?> handler =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
         MigrationRuleHandler<?> handler2 =
                 Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
 
@@ -207,18 +207,18 @@
         MigrationInvoker<?> migrationInvoker2 = Mockito.mock(MigrationInvoker.class);
 
         // Remote rule will be applied when onRefer gets executed
-        migrationRuleListener.getHandlers().put(migrationInvoker, handler);
+        migrationRuleListener.getHandlers().put(migrationInvoker, handlerMock);
         migrationRuleListener.onRefer(null, migrationInvoker, consumerURL, null);
 
         MigrationRule tmpRemoteRule = migrationRuleListener.getRule();
         ArgumentCaptor<MigrationRule> captor = ArgumentCaptor.forClass(MigrationRule.class);
-        Mockito.verify(handler, Mockito.times(1)).doMigrate(captor.capture());
+        Mockito.verify(handlerMock, Mockito.times(1)).doMigrate(captor.capture());
         Assertions.assertEquals(tmpRemoteRule, captor.getValue());
 
         await().until(() -> migrationRuleListener.localRuleMigrationFuture.isDone());
         Assertions.assertNull(migrationRuleListener.ruleMigrationFuture);
         Assertions.assertEquals(tmpRemoteRule, migrationRuleListener.getRule());
-        Mockito.verify(handler, Mockito.times(1)).doMigrate(Mockito.any());
+        Mockito.verify(handlerMock, Mockito.times(1)).doMigrate(Mockito.any());
 
         ArgumentCaptor<MigrationRule> captor2 = ArgumentCaptor.forClass(MigrationRule.class);
         migrationRuleListener.getHandlers().put(migrationInvoker2, handler2);
@@ -230,13 +230,13 @@
 
         await().until(migrationRuleListener.ruleQueue::isEmpty);
         await().untilAsserted(() -> {
-            Mockito.verify(handler, Mockito.times(2)).doMigrate(Mockito.any());
+            Mockito.verify(handlerMock, Mockito.times(2)).doMigrate(Mockito.any());
             Mockito.verify(handler2, Mockito.times(2)).doMigrate(Mockito.any());
         });
 
         Assertions.assertNotNull(migrationRuleListener.ruleMigrationFuture);
         ArgumentCaptor<MigrationRule> captor_event = ArgumentCaptor.forClass(MigrationRule.class);
-        Mockito.verify(handler, Mockito.times(2)).doMigrate(captor_event.capture());
+        Mockito.verify(handlerMock, Mockito.times(2)).doMigrate(captor_event.capture());
         Assertions.assertEquals(
                 "APPLICATION_FIRST", captor_event.getValue().getStep().toString());
         Mockito.verify(handler2, Mockito.times(2)).doMigrate(captor_event.capture());
