--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
@@ -53,9 +53,8 @@
         given(invocation.getArguments()).willReturn(new Object[] {"hello"});
         given(invocation.getObjectAttachments()).willReturn(null);
 
-        invoker = mock(Invoker.class);
+        invoker = MockDemoServiceInvoker.createMockDemoServiceInvoker();
         given(invoker.isAvailable()).willReturn(true);
-        given(invoker.getInterface()).willReturn(DemoService.class);
         AppResponse result = new AppResponse();
         result.setValue("High");
         given(invoker.invoke(invocation)).willReturn(result);
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ExceptionFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ExceptionFilterTest.java
@@ -52,8 +52,7 @@
         ExceptionFilter exceptionFilter = new ExceptionFilter();
         RpcInvocation invocation = new RpcInvocation(
                 "sayHello", DemoService.class.getName(), "", new Class<?>[] {String.class}, new Object[] {"world"});
-        Invoker<DemoService> invoker = mock(Invoker.class);
-        given(invoker.getInterface()).willReturn(DemoService.class);
+        Invoker<DemoService> invoker = MockDemoServiceInvoker.createMockDemoServiceInvoker();
         given(invoker.invoke(eq(invocation))).willThrow(exception);
 
         try {
@@ -86,9 +85,8 @@
         AppResponse appResponse = new AppResponse();
         appResponse.setException(new IllegalArgumentException("java"));
 
-        Invoker<DemoService> invoker = mock(Invoker.class);
+        Invoker<DemoService> invoker = MockDemoServiceInvoker.createMockDemoServiceInvoker();
         when(invoker.invoke(invocation)).thenReturn(appResponse);
-        when(invoker.getInterface()).thenReturn(DemoService.class);
 
         Result newResult = exceptionFilter.invoke(invoker, invocation);
 
@@ -106,9 +104,8 @@
         AppResponse appResponse = new AppResponse();
         appResponse.setException(new LocalException("localException"));
 
-        Invoker<DemoService> invoker = mock(Invoker.class);
+        Invoker<DemoService> invoker = MockDemoServiceInvoker.createMockDemoServiceInvoker();
         when(invoker.invoke(invocation)).thenReturn(appResponse);
-        when(invoker.getInterface()).thenReturn(DemoService.class);
 
         Result newResult = exceptionFilter.invoke(invoker, invocation);
 
@@ -127,9 +124,8 @@
         mockRpcResult.setException(new HessianException("hessian"));
         Result mockAsyncResult = AsyncRpcResult.newDefaultAsyncResult(mockRpcResult, invocation);
 
-        Invoker<DemoService> invoker = mock(Invoker.class);
+        Invoker<DemoService> invoker = MockDemoServiceInvoker.createMockDemoServiceInvoker();
         when(invoker.invoke(invocation)).thenReturn(mockAsyncResult);
-        when(invoker.getInterface()).thenReturn(DemoService.class);
 
         Result asyncResult = exceptionFilter.invoke(invoker, invocation);
 
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/MockDemoServiceInvoker.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/MockDemoServiceInvoker.java
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.rpc.filter;
+
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.support.DemoService;
+
+import org.mockito.Mockito;
+
+public class MockDemoServiceInvoker {
+
+    @SuppressWarnings("unchecked")
+    public static Invoker<DemoService> createMockDemoServiceInvoker() {
+        Invoker<DemoService> invoker = Mockito.mock(Invoker.class);
+        Mockito.when(invoker.getInterface()).thenReturn(DemoService.class);
+        return invoker;
+    }
+}
