--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/BroadCastClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/BroadCastClusterInvokerTest.java
@@ -34,7 +34,6 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.BDDMockito.given;
-import static org.mockito.Mockito.mock;
 
 /**
  * @see BroadcastClusterInvoker
@@ -53,17 +52,13 @@
     @BeforeEach
     public void setUp() throws Exception {
 
-        dic = mock(Directory.class);
-
         invoker1 = new MockInvoker();
         invoker2 = new MockInvoker();
         invoker3 = new MockInvoker();
         invoker4 = new MockInvoker();
 
         url = URL.valueOf("test://127.0.0.1:8080/test");
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
-        given(dic.getInterface()).willReturn(DemoService.class);
+        dic = MockDemoServiceDirectory.createMockDemoServiceDirectory(url);
 
         invocation = new RpcInvocation();
         invocation.setMethodName("test");
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailSafeClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailSafeClusterInvokerTest.java
@@ -57,12 +57,8 @@
     public void setUp() throws Exception {
         RpcContext.removeServiceContext();
 
-        dic = mock(Directory.class);
-
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
+        dic = MockDemoServiceDirectory.createMockDemoServiceDirectory(url);
         given(dic.list(invocation)).willReturn(invokers);
-        given(dic.getInterface()).willReturn(DemoService.class);
         invocation.setMethodName("method1");
 
         invokers.add(invoker);
@@ -100,12 +96,8 @@
 
     @Test
     void testNoInvoke() {
-        dic = mock(Directory.class);
-
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
+        dic = MockDemoServiceDirectory.createMockDemoServiceDirectory(url);
         given(dic.list(invocation)).willReturn(null);
-        given(dic.getInterface()).willReturn(DemoService.class);
 
         invocation.setMethodName("method1");
 
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MockDemoServiceDirectory.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MockDemoServiceDirectory.java
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.rpc.cluster.support;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.rpc.cluster.Directory;
+import org.apache.dubbo.rpc.cluster.filter.DemoService;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public class MockDemoServiceDirectory {
+
+    public static Directory<DemoService> createMockDemoServiceDirectory(URL url) {
+        Directory<DemoService> dic = mock(Directory.class);
+        given(dic.getUrl()).willReturn(url);
+        given(dic.getConsumerUrl()).willReturn(url);
+        given(dic.getInterface()).willReturn(DemoService.class);
+        return dic;
+    }
+}
