--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
@@ -62,26 +62,23 @@
         ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
         ClusterInvoker serviceDiscoveryInvoker = Mockito.mock(ClusterInvoker.class);
 
-        DynamicDirectory directory = Mockito.mock(DynamicDirectory.class);
-        DynamicDirectory serviceDiscoveryDirectory = Mockito.mock(DynamicDirectory.class);
-
-        Mockito.when(invoker.getDirectory()).thenReturn(directory);
-        Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(serviceDiscoveryDirectory);
-
-        Mockito.when(invoker.isAvailable()).thenReturn(true);
-        Mockito.when(serviceDiscoveryInvoker.isAvailable()).thenReturn(true);
-
-        Mockito.when(invoker.hasProxyInvokers()).thenReturn(true);
-        Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
-
         List<Invoker<?>> invokers = new LinkedList<>();
         invokers.add(Mockito.mock(Invoker.class));
         invokers.add(Mockito.mock(Invoker.class));
         List<Invoker<?>> serviceDiscoveryInvokers = new LinkedList<>();
         serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
         serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
-        Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
-        Mockito.when(serviceDiscoveryDirectory.getAllInvokers()).thenReturn(serviceDiscoveryInvokers);
+        DynamicDirectory directory = createMockDynamicDirectory(invokers);
+        DynamicDirectory serviceDiscoveryDirectory = createMockDynamicDirectory(serviceDiscoveryInvokers);
+
+        Mockito.when(invoker.getDirectory()).thenReturn(directory);
+        Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(serviceDiscoveryDirectory);
+
+        Mockito.when(invoker.isAvailable()).thenReturn(true);
+        Mockito.when(serviceDiscoveryInvoker.isAvailable()).thenReturn(true);
+
+        Mockito.when(invoker.hasProxyInvokers()).thenReturn(true);
+        Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
 
         Mockito.when(registryProtocol.getInvoker(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
                 .thenReturn(invoker);
@@ -238,26 +235,23 @@
         ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
         ClusterInvoker serviceDiscoveryInvoker = Mockito.mock(ClusterInvoker.class);
 
-        DynamicDirectory directory = Mockito.mock(DynamicDirectory.class);
-        DynamicDirectory serviceDiscoveryDirectory = Mockito.mock(DynamicDirectory.class);
-
-        Mockito.when(invoker.getDirectory()).thenReturn(directory);
-        Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(serviceDiscoveryDirectory);
-
-        Mockito.when(invoker.isAvailable()).thenReturn(true);
-        Mockito.when(serviceDiscoveryInvoker.isAvailable()).thenReturn(true);
-
-        Mockito.when(invoker.hasProxyInvokers()).thenReturn(true);
-        Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
-
         List<Invoker<?>> invokers = new LinkedList<>();
         invokers.add(Mockito.mock(Invoker.class));
         invokers.add(Mockito.mock(Invoker.class));
         List<Invoker<?>> serviceDiscoveryInvokers = new LinkedList<>();
         serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
         serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
-        Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
-        Mockito.when(serviceDiscoveryDirectory.getAllInvokers()).thenReturn(serviceDiscoveryInvokers);
+        DynamicDirectory directory = createMockDynamicDirectory(invokers);
+        DynamicDirectory serviceDiscoveryDirectory = createMockDynamicDirectory(serviceDiscoveryInvokers);
+
+        Mockito.when(invoker.getDirectory()).thenReturn(directory);
+        Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(serviceDiscoveryDirectory);
+
+        Mockito.when(invoker.isAvailable()).thenReturn(true);
+        Mockito.when(serviceDiscoveryInvoker.isAvailable()).thenReturn(true);
+
+        Mockito.when(invoker.hasProxyInvokers()).thenReturn(true);
+        Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
 
         Mockito.when(registryProtocol.getInvoker(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
                 .thenReturn(invoker);
@@ -302,4 +296,10 @@
         // 独立线程invoker状态切换
 
     }
+
+    private static DynamicDirectory createMockDynamicDirectory(List<Invoker<?>> invokers) {
+        DynamicDirectory directory = Mockito.mock(DynamicDirectory.class);
+        Mockito.when(directory.getAllInvokers()).thenReturn((List<?>) invokers);
+        return directory;
+    }
 }
