--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/FutureFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/FutureFilterTest.java
@@ -50,17 +50,22 @@
         invocation.setArguments(new Object[] {"hello"});
     }
 
-    @Test
-    void testSyncCallback() {
-        @SuppressWarnings("unchecked")
+    @SuppressWarnings("unchecked")
+    private static Invoker<DemoService> createMockInvoker(AppResponse result, URL url) {
         Invoker<DemoService> invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
+        given(invoker.invoke(invocation)).willReturn(result);
+        given(invoker.getUrl()).willReturn(url);
+        return invoker;
+    }
+
+    @Test
+    void testSyncCallback() {
         AppResponse result = new AppResponse();
         result.setValue("High");
-        given(invoker.invoke(invocation)).willReturn(result);
         URL url = URL.valueOf("test://test:11/test?group=dubbo&version=1.1");
-        given(invoker.getUrl()).willReturn(url);
+        Invoker<DemoService> invoker = createMockInvoker(result, url);
 
         Result filterResult = eventFilter.invoke(invoker, invocation);
         assertEquals("High", filterResult.getValue());
@@ -69,15 +74,10 @@
     @Test
     void testSyncCallbackHasException() throws RpcException, Throwable {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            @SuppressWarnings("unchecked")
-            Invoker<DemoService> invoker = mock(Invoker.class);
-            given(invoker.isAvailable()).willReturn(true);
-            given(invoker.getInterface()).willReturn(DemoService.class);
             AppResponse result = new AppResponse();
             result.setException(new RuntimeException());
-            given(invoker.invoke(invocation)).willReturn(result);
             URL url = URL.valueOf("test://test:11/test?group=dubbo&version=1.1&onthrow.method=echo");
-            given(invoker.getUrl()).willReturn(url);
+            Invoker<DemoService> invoker = createMockInvoker(result, url);
 
             eventFilter.invoke(invoker, invocation).recreate();
         });
