--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mesh/route/MeshRuleManagerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mesh/route/MeshRuleManagerTest.java
@@ -68,6 +68,12 @@
     private GovernanceRuleRepository ruleRepository;
     private Set<MeshEnvListenerFactory> envListenerFactories;
 
+    private static MeshEnvListener createMockMeshEnvListener() {
+        MeshEnvListener meshEnvListener = Mockito.mock(MeshEnvListener.class);
+        when(meshEnvListener.isEnable()).thenReturn(false);
+        return meshEnvListener;
+    }
+
     @BeforeEach
     public void setup() {
         originModule = ApplicationModel.defaultModel().getDefaultModule();
@@ -199,9 +205,9 @@
         MeshEnvListenerFactory meshEnvListenerFactory1 = Mockito.mock(MeshEnvListenerFactory.class);
         MeshEnvListenerFactory meshEnvListenerFactory2 = Mockito.mock(MeshEnvListenerFactory.class);
 
-        MeshEnvListener meshEnvListener1 = Mockito.mock(MeshEnvListener.class);
+        MeshEnvListener meshEnvListener1 = createMockMeshEnvListener();
         when(meshEnvListenerFactory1.getListener()).thenReturn(meshEnvListener1);
-        MeshEnvListener meshEnvListener2 = Mockito.mock(MeshEnvListener.class);
+        MeshEnvListener meshEnvListener2 = createMockMeshEnvListener();
         when(meshEnvListenerFactory2.getListener()).thenReturn(meshEnvListener2);
 
         envListenerFactories.add(meshEnvListenerFactory1);
@@ -224,7 +230,6 @@
             }
         };
 
-        when(meshEnvListener1.isEnable()).thenReturn(false);
         when(meshEnvListener2.isEnable()).thenReturn(true);
 
         meshRuleManager.register("dubbo-demo", meshRuleListener1);
