--- a/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
+++ b/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
@@ -477,13 +477,7 @@
                                         .context(ImmutableMap.of("queryId", "test-query-504"))
                                         .build();
 
-    final HttpServletRequest requestMock = Mockito.mock(HttpServletRequest.class);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.avaticaQuery")).thenReturn(null);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.query")).thenReturn(query);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.sqlQuery")).thenReturn(null);
-    Mockito.when(requestMock.getRemoteAddr()).thenReturn("127.0.0.1");
-    Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
+    final HttpServletRequest requestMock = makeRequestMock(query);
 
     final Request proxyRequestMock = Mockito.mock(Request.class);
     final Response responseMock = Mockito.mock(Response.class);
@@ -541,13 +535,7 @@
                                         .context(ImmutableMap.of("queryId", "closed-test"))
                                         .build();
 
-    final HttpServletRequest requestMock = Mockito.mock(HttpServletRequest.class);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.avaticaQuery")).thenReturn(null);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.query")).thenReturn(query);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.sqlQuery")).thenReturn(null);
-    Mockito.when(requestMock.getRemoteAddr()).thenReturn("127.0.0.1");
-    Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
+    final HttpServletRequest requestMock = makeRequestMock(query);
 
     final Request proxyRequestMock = Mockito.mock(Request.class);
     final Response responseMock = Mockito.mock(Response.class);
@@ -612,13 +600,7 @@
                                         .context(ImmutableMap.of("queryId", "zero-status-test"))
                                         .build();
 
-    final HttpServletRequest requestMock = Mockito.mock(HttpServletRequest.class);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.avaticaQuery")).thenReturn(null);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.query")).thenReturn(query);
-    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.sqlQuery")).thenReturn(null);
-    Mockito.when(requestMock.getRemoteAddr()).thenReturn("127.0.0.1");
-    Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
+    final HttpServletRequest requestMock = makeRequestMock(query);
 
     final Response responseMock = Mockito.mock(Response.class);
     Mockito.when(responseMock.getStatus()).thenReturn(0); // Test unassigned http status code case from server
@@ -654,6 +636,18 @@
         stubServiceEmitter.getMetricEvents("query/time").get(0).toMap().get(DruidMetrics.STATUS_CODE)
     );
     Assert.assertEquals("false", stubServiceEmitter.getMetricEvents("query/time").get(0).toMap().get("success"));
+  }
+
+  private HttpServletRequest makeRequestMock(final TimeseriesQuery query)
+  {
+    final HttpServletRequest requestMock = Mockito.mock(HttpServletRequest.class);
+    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.avaticaQuery")).thenReturn(null);
+    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.query")).thenReturn(query);
+    Mockito.when(requestMock.getAttribute("org.apache.druid.proxy.sqlQuery")).thenReturn(null);
+    Mockito.when(requestMock.getRemoteAddr()).thenReturn("127.0.0.1");
+    Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
+           .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
+    return requestMock;
   }
 
 
