--- a/processing/src/test/java/org/apache/druid/java/util/metrics/OshiSysMonitorTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/OshiSysMonitorTest.java
@@ -262,22 +262,10 @@
   public void testDiskStats()
   {
     StubServiceEmitter emitter = new StubServiceEmitter("dev/monitor-test", "localhost:0000");
-    HWDiskStore disk1 = Mockito.mock(HWDiskStore.class);
-    HWDiskStore disk2 = Mockito.mock(HWDiskStore.class);
-    Mockito.when(disk1.getReadBytes()).thenReturn(300L);
-    Mockito.when(disk1.getReads()).thenReturn(200L);
-    Mockito.when(disk1.getWriteBytes()).thenReturn(400L);
-    Mockito.when(disk1.getWrites()).thenReturn(500L);
-    Mockito.when(disk1.getCurrentQueueLength()).thenReturn(100L);
-    Mockito.when(disk1.getTransferTime()).thenReturn(150L);
-    Mockito.when(disk1.getName()).thenReturn("disk1");
-    Mockito.when(disk2.getReadBytes()).thenReturn(2000L);
-    Mockito.when(disk2.getReads()).thenReturn(3000L);
-    Mockito.when(disk2.getWriteBytes()).thenReturn(1000L);
-    Mockito.when(disk2.getWrites()).thenReturn(4000L);
-    Mockito.when(disk2.getCurrentQueueLength()).thenReturn(750L);
-    Mockito.when(disk2.getTransferTime()).thenReturn(800L);
-    Mockito.when(disk2.getName()).thenReturn("disk2");
+    final HWDiskStore disk1 = Mockito.mock(HWDiskStore.class);
+    final HWDiskStore disk2 = Mockito.mock(HWDiskStore.class);
+    configureDiskStore(disk1, "disk1", 300L, 200L, 400L, 500L, 100L, 150L);
+    configureDiskStore(disk2, "disk2", 2000L, 3000L, 1000L, 4000L, 750L, 800L);
     List<HWDiskStore> hwDiskStores = ImmutableList.of(disk1, disk2);
     Mockito.when(hal.getDiskStores()).thenReturn(hwDiskStores);
 
@@ -286,18 +274,8 @@
     m.monitorDiskStats(emitter);
     Assert.assertEquals(0, emitter.getNumEmittedEvents());
 
-    Mockito.when(disk1.getReadBytes()).thenReturn(400L);
-    Mockito.when(disk1.getReads()).thenReturn(220L);
-    Mockito.when(disk1.getWriteBytes()).thenReturn(600L);
-    Mockito.when(disk1.getWrites()).thenReturn(580L);
-    Mockito.when(disk1.getCurrentQueueLength()).thenReturn(300L);
-    Mockito.when(disk1.getTransferTime()).thenReturn(250L);
-    Mockito.when(disk2.getReadBytes()).thenReturn(4500L);
-    Mockito.when(disk2.getReads()).thenReturn(3500L);
-    Mockito.when(disk2.getWriteBytes()).thenReturn(2300L);
-    Mockito.when(disk2.getWrites()).thenReturn(5000L);
-    Mockito.when(disk2.getCurrentQueueLength()).thenReturn(900L);
-    Mockito.when(disk2.getTransferTime()).thenReturn(1100L);
+    configureDiskStore(disk1, "disk1", 400L, 220L, 600L, 580L, 300L, 250L);
+    configureDiskStore(disk2, "disk2", 4500L, 3500L, 2300L, 5000L, 900L, 1100L);
 
     m.monitorDiskStats(emitter);
     Assert.assertEquals(12, emitter.getNumEmittedEvents());
@@ -640,6 +618,26 @@
     }
   }
 
+  private void configureDiskStore(
+      final HWDiskStore disk,
+      final String name,
+      final long readBytes,
+      final long reads,
+      final long writeBytes,
+      final long writes,
+      final long currentQueueLength,
+      final long transferTime
+  )
+  {
+    Mockito.when(disk.getReadBytes()).thenReturn(readBytes);
+    Mockito.when(disk.getReads()).thenReturn(reads);
+    Mockito.when(disk.getWriteBytes()).thenReturn(writeBytes);
+    Mockito.when(disk.getWrites()).thenReturn(writes);
+    Mockito.when(disk.getCurrentQueueLength()).thenReturn(currentQueueLength);
+    Mockito.when(disk.getTransferTime()).thenReturn(transferTime);
+    Mockito.when(disk.getName()).thenReturn(name);
+  }
+
   private OshiSysMonitor createMonitor(SystemInfo si)
   {
     return createMonitor(si, ImmutableList.of());
