--- a/extensions-contrib/cloudfiles-extensions/src/test/java/org/apache/druid/storage/cloudfiles/CloudFilesByteSourceTest.java
+++ b/extensions-contrib/cloudfiles-extensions/src/test/java/org/apache/druid/storage/cloudfiles/CloudFilesByteSourceTest.java
@@ -41,7 +41,7 @@
     CloudFilesObjectApiProxy objectApi = mock(CloudFilesObjectApiProxy.class);
     CloudFilesObject cloudFilesObject = mock(CloudFilesObject.class);
     Payload payload = mock(Payload.class);
-    InputStream stream = mock(InputStream.class);
+    final InputStream stream = createInputStream();
 
     when(objectApi.get(path, 0)).thenReturn(cloudFilesObject);
     when(cloudFilesObject.getPayload()).thenReturn(payload);
@@ -65,7 +65,7 @@
     CloudFilesObjectApiProxy objectApi = mock(CloudFilesObjectApiProxy.class);
     CloudFilesObject cloudFilesObject = mock(CloudFilesObject.class);
     Payload payload = mock(Payload.class);
-    InputStream stream = mock(InputStream.class);
+    final InputStream stream = createInputStream();
 
     when(objectApi.get(path, 0)).thenReturn(cloudFilesObject);
     when(cloudFilesObject.getPayload()).thenReturn(payload);
@@ -88,4 +88,9 @@
     verify(cloudFilesObject).getPayload();
     verify(payload, times(2)).openStream();
   }
+
+  private InputStream createInputStream()
+  {
+    return mock(InputStream.class);
+  }
 }
