--- a/processing/src/test/java/org/apache/druid/query/rowsandcols/serde/WireTransferableContextTest.java
+++ b/processing/src/test/java/org/apache/druid/query/rowsandcols/serde/WireTransferableContextTest.java
@@ -32,6 +32,11 @@
 
 public class WireTransferableContextTest
 {
+  private ObjectMapper mockMapper()
+  {
+    return Mockito.mock(ObjectMapper.class);
+  }
+
   @Test
   public void test_defaultLegacyFrameSerialization()
   {
@@ -41,7 +46,7 @@
   @Test
   public void test_constructor_andGetters()
   {
-    final ObjectMapper mockMapper = Mockito.mock(ObjectMapper.class);
+    final ObjectMapper mockMapper = mockMapper();
     final WireTransferable.ConcreteDeserializer mockDeserializer =
         Mockito.mock(WireTransferable.ConcreteDeserializer.class);
 
@@ -79,7 +84,7 @@
   @Test
   public void test_serializedBytes() throws IOException
   {
-    final ObjectMapper mockMapper = Mockito.mock(ObjectMapper.class);
+    final ObjectMapper mockMapper = mockMapper();
     final WireTransferable mockWireTransferable = Mockito.mock(WireTransferable.class);
     final WireTransferable.ByteArrayOffsetAndLen expectedResult =
         new WireTransferable.ByteArrayOffsetAndLen(new byte[]{1, 2, 3}, 0, 3);
