--- a/sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java
+++ b/sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java
@@ -159,6 +159,13 @@
     return operatorTable.lookupOperatorConversion(round);
   }
 
+  private PlannerContext makePlannerContext()
+  {
+    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
+    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+    return plannerContext;
+  }
+
   @Test
   public void testConcat()
   {
@@ -2721,8 +2728,7 @@
   public void testPresenceOfOverIsInvalid()
   {
     final RexBuilder rexBuilder = new RexBuilder(DruidTypeSystem.TYPE_FACTORY);
-    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
-    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+    final PlannerContext plannerContext = makePlannerContext();
 
     RexNode rexNode = rexBuilder.makeOver(
         testHelper.createSqlType(SqlTypeName.BIGINT),
@@ -2751,8 +2757,7 @@
   public void testCalciteLiteralToDruidLiteral()
   {
     final RexBuilder rexBuilder = new RexBuilder(DruidTypeSystem.TYPE_FACTORY);
-    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
-    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+    final PlannerContext plannerContext = makePlannerContext();
 
     assertDruidLiteral(
         new DruidLiteral(ExpressionType.STRING, null),
