--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -38,7 +38,7 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
+    final CounterTracker counters = mockCounters();
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -74,7 +74,7 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
+    final CounterTracker counters = mockCounters();
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -115,7 +115,7 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
+    final CounterTracker counters = mockCounters();
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -153,7 +153,7 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
+    final CounterTracker counters = mockCounters();
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -185,4 +185,9 @@
     Mockito.verify(frameContext).close();
     Mockito.verify(listener).onFailure(ArgumentMatchers.eq(exception));
   }
+
+  private CounterTracker mockCounters()
+  {
+    return Mockito.mock(CounterTracker.class);
+  }
 }
