--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -32,6 +32,7 @@
 public class RunWorkOrderTest
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
+  private final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
   @Test
   public void test_stop()
@@ -43,8 +44,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
-
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
 
     final RunWorkOrder runWorkOrder =
@@ -79,8 +78,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
-
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
 
     final RunWorkOrder runWorkOrder =
@@ -120,8 +117,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
-
     final ISE exception = new ISE("oops");
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
     Mockito.doThrow(exception).when(exec).cancel(CANCELLATION_ID);
@@ -158,8 +153,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
-
     final ISE exception = new ISE("oops");
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
     Mockito.doThrow(exception).when(frameContext).close();
