--- a/processing/src/test/java/org/apache/druid/query/filter/InDimFilterTest.java
+++ b/processing/src/test/java/org/apache/druid/query/filter/InDimFilterTest.java
@@ -754,7 +754,7 @@
     final ColumnIndexSelector indexSelector = Mockito.mock(ColumnIndexSelector.class);
     final ColumnIndexSupplier indexSupplier = Mockito.mock(ColumnIndexSupplier.class);
     final Utf8ValueSetIndexes valueIndexes = Mockito.mock(Utf8ValueSetIndexes.class);
-    final BitmapColumnIndex bitmapColumnIndex = Mockito.mock(BitmapColumnIndex.class);
+    final BitmapColumnIndex bitmapColumnIndex = mockBitmapColumnIndex();
 
     final InDimFilter.ValuesSet expectedValuesSet = new InDimFilter.ValuesSet();
     expectedValuesSet.addAll(Arrays.asList("v1", "v2"));
@@ -778,7 +778,7 @@
     final ColumnIndexSelector indexSelector = Mockito.mock(ColumnIndexSelector.class);
     final ColumnIndexSupplier indexSupplier = Mockito.mock(ColumnIndexSupplier.class);
     final StringValueSetIndexes valueIndex = Mockito.mock(StringValueSetIndexes.class);
-    final BitmapColumnIndex bitmapColumnIndex = Mockito.mock(BitmapColumnIndex.class);
+    final BitmapColumnIndex bitmapColumnIndex = mockBitmapColumnIndex();
 
     final InDimFilter.ValuesSet expectedValuesSet = new InDimFilter.ValuesSet();
     expectedValuesSet.addAll(Arrays.asList("v1", "v2"));
@@ -791,4 +791,9 @@
     final BitmapColumnIndex retVal = inFilter.getBitmapColumnIndex(indexSelector);
     Assert.assertSame("inFilter returns the intended bitmapColumnIndex", bitmapColumnIndex, retVal);
   }
+
+  private BitmapColumnIndex mockBitmapColumnIndex()
+  {
+    return Mockito.mock(BitmapColumnIndex.class);
+  }
 }
