--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -36,7 +36,7 @@
   @Test
   public void test_stop()
   {
-    final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
+    final FrameProcessorExecutor exec = makeFrameProcessorExecutor();
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
@@ -72,7 +72,7 @@
   @Test
   public void test_stop_error()
   {
-    final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
+    final FrameProcessorExecutor exec = makeFrameProcessorExecutor();
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
@@ -113,7 +113,7 @@
   @Test
   public void test_stop_errorDuringExecCancel()
   {
-    final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
+    final FrameProcessorExecutor exec = makeFrameProcessorExecutor();
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
@@ -151,7 +151,7 @@
   @Test
   public void test_stop_errorDuringFrameContextClose()
   {
-    final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
+    final FrameProcessorExecutor exec = makeFrameProcessorExecutor();
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
@@ -185,4 +185,9 @@
     Mockito.verify(frameContext).close();
     Mockito.verify(listener).onFailure(ArgumentMatchers.eq(exception));
   }
+
+  private FrameProcessorExecutor makeFrameProcessorExecutor()
+  {
+    return Mockito.mock(FrameProcessorExecutor.class);
+  }
 }
