--- a/processing/src/test/java/org/apache/druid/query/aggregation/any/DoubleAnyAggregatorFactoryTest.java
+++ b/processing/src/test/java/org/apache/druid/query/aggregation/any/DoubleAnyAggregatorFactoryTest.java
@@ -80,8 +80,7 @@
   @Test
   public void factorizeVectorForNumericTypeShouldReturnDoubleVectorAggregator()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
-    Mockito.doReturn(true).when(capabilities).isNumeric();
+    mockColumnCapabilities(true);
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertEquals(DoubleAnyVectorAggregator.class, aggregator.getClass());
@@ -90,10 +89,15 @@
   @Test
   public void factorizeVectorForStringTypeShouldReturnDoubleVectorAggregatorWithNilSelector()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
-    Mockito.doReturn(false).when(capabilities).isNumeric();
+    mockColumnCapabilities(false);
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertNull(aggregator.get(BUFFER, POSITION));
   }
+
+  private void mockColumnCapabilities(final boolean isNumeric)
+  {
+    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    Mockito.doReturn(isNumeric).when(capabilities).isNumeric();
+  }
 }
--- a/processing/src/test/java/org/apache/druid/query/aggregation/any/FloatAnyAggregatorFactoryTest.java
+++ b/processing/src/test/java/org/apache/druid/query/aggregation/any/FloatAnyAggregatorFactoryTest.java
@@ -80,8 +80,7 @@
   @Test
   public void factorizeVectorForNumericTypeShouldReturnFloatVectorAggregator()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
-    Mockito.doReturn(true).when(capabilities).isNumeric();
+    mockColumnCapabilities(true);
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertEquals(FloatAnyVectorAggregator.class, aggregator.getClass());
@@ -90,10 +89,15 @@
   @Test
   public void factorizeVectorForStringTypeShouldReturnFloatVectorAggregatorWithNilSelector()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
-    Mockito.doReturn(false).when(capabilities).isNumeric();
+    mockColumnCapabilities(false);
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertNull(aggregator.get(BUFFER, POSITION));
   }
+
+  private void mockColumnCapabilities(final boolean isNumeric)
+  {
+    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    Mockito.doReturn(isNumeric).when(capabilities).isNumeric();
+  }
 }
--- a/processing/src/test/java/org/apache/druid/query/aggregation/any/LongAnyAggregatorFactoryTest.java
+++ b/processing/src/test/java/org/apache/druid/query/aggregation/any/LongAnyAggregatorFactoryTest.java
@@ -80,8 +80,7 @@
   @Test
   public void factorizeVectorWithNumericColumnShouldReturnLongVectorAggregator()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
-    Mockito.doReturn(true).when(capabilities).isNumeric();
+    mockColumnCapabilities(true);
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertEquals(LongAnyVectorAggregator.class, aggregator.getClass());
@@ -90,10 +89,15 @@
   @Test
   public void factorizeVectorForStringTypeShouldReturnLongVectorAggregatorWithNilSelector()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
-    Mockito.doReturn(false).when(capabilities).isNumeric();
+    mockColumnCapabilities(false);
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertNull(aggregator.get(BUFFER, POSITION));
   }
+
+  private void mockColumnCapabilities(final boolean isNumeric)
+  {
+    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    Mockito.doReturn(isNumeric).when(capabilities).isNumeric();
+  }
 }
