--- a/server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java
+++ b/server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java
@@ -769,7 +769,7 @@
   @Test
   public void testCompactWithoutGranularitySpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -800,7 +800,7 @@
   @Test
   public void testCompactWithNotNullIOConfig()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -824,7 +824,7 @@
   @Test
   public void testCompactWithNullIOConfig()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -851,7 +851,7 @@
   @Test
   public void testCompactWithGranularitySpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -888,7 +888,7 @@
   @Test
   public void testCompactWithDimensionSpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -919,7 +919,7 @@
   @Test
   public void testCompactWithoutDimensionSpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -942,7 +942,7 @@
   @Test
   public void testCompactWithProjections()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -988,7 +988,7 @@
   public void testCompactWithCatalogProjections()
   {
     final String dataSource = DATA_SOURCE_PREFIX + 0;
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ObjectMapper mapper = new DefaultObjectMapper((DefaultObjectMapper) JSON_MAPPER);
     final MapMetadataCatalog metadataCatalog = new MapMetadataCatalog(mapper);
 
@@ -1043,7 +1043,7 @@
   @Test
   public void testCompactWithRollupInGranularitySpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1121,7 +1121,7 @@
         )
     );
 
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = ArgumentCaptor.forClass(Object.class);
     Mockito.when(mockClient.runTask(ArgumentMatchers.anyString(), payloadCaptor.capture()))
            .thenReturn(Futures.immediateFuture(null));
@@ -1283,7 +1283,7 @@
   @Test
   public void testCompactWithTransformSpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1313,7 +1313,7 @@
   @Test
   public void testCompactWithoutCustomSpecs()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1338,7 +1338,7 @@
   public void testCompactWithMetricsSpec()
   {
     AggregatorFactory[] aggregatorFactories = new AggregatorFactory[] {new CountAggregatorFactory("cnt")};
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1394,7 +1394,7 @@
     );
     dataSources = DataSourcesSnapshot.fromUsedSegments(segments);
 
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1454,7 +1454,7 @@
     );
     dataSources = DataSourcesSnapshot.fromUsedSegments(segments);
 
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1491,7 +1491,7 @@
   @Test
   public void testCompactWithMetricsSpecShouldSetPreserveExistingMetricsTrue()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -1518,7 +1518,7 @@
   @Test
   public void testCompactWithoutMetricsSpecShouldSetPreserveExistingMetricsFalse()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
+    final OverlordClient mockClient = createMockClient();
     final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
     final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
@@ -2086,6 +2086,11 @@
     }
   }
 
+  private static OverlordClient createMockClient()
+  {
+    return Mockito.mock(OverlordClient.class);
+  }
+
   private static ArgumentCaptor<Object> setUpMockClient(final OverlordClient mockClient)
   {
     final ArgumentCaptor<Object> payloadCaptor = ArgumentCaptor.forClass(Object.class);
