--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
@@ -143,8 +143,8 @@
   @Test
   public void testKillRecursively_delegatesToAllKillers() throws IOException
   {
-    final DataSegmentKiller killerA = Mockito.mock(DataSegmentKiller.class);
-    final DataSegmentKiller killerB = Mockito.mock(DataSegmentKiller.class);
+    final DataSegmentKiller killerA = mockDataSegmentKiller();
+    final DataSegmentKiller killerB = mockDataSegmentKiller();
     final Injector injector = createInjectorFromMap(
         ImmutableMap.of("type_a", killerA, "type_b", killerB)
     );
@@ -160,8 +160,8 @@
   @Test
   public void testKillMultipleSegmentsWithType() throws SegmentLoadingException
   {
-    final DataSegmentKiller killerSane = Mockito.mock(DataSegmentKiller.class);
-    final DataSegmentKiller killerSaneTwo = Mockito.mock(DataSegmentKiller.class);
+    final DataSegmentKiller killerSane = mockDataSegmentKiller();
+    final DataSegmentKiller killerSaneTwo = mockDataSegmentKiller();
     final DataSegment segment1 = Mockito.mock(DataSegment.class);
     final DataSegment segment2 = Mockito.mock(DataSegment.class);
     final DataSegment segment3 = Mockito.mock(DataSegment.class);
@@ -182,6 +182,11 @@
            .kill((List<DataSegment>) argThat(containsInAnyOrder(segment3)));
   }
 
+  private static DataSegmentKiller mockDataSegmentKiller()
+  {
+    return Mockito.mock(DataSegmentKiller.class);
+  }
+
   @LazySingleton
   private static class BadSegmentKiller implements DataSegmentKiller
   {
