--- a/extensions-core/lookups-cached-global/src/test/java/org/apache/druid/query/lookup/NamespaceLookupExtractorFactoryTest.java
+++ b/extensions-core/lookups-cached-global/src/test/java/org/apache/druid/query/lookup/NamespaceLookupExtractorFactoryTest.java
@@ -332,6 +332,11 @@
     }
   }
 
+  private ExtractionNamespace mockExtractionNamespace()
+  {
+    return mock(ExtractionNamespace.class);
+  }
+
 
   @Test
   public void testStartFailsToSchedule() throws Exception
@@ -372,9 +377,8 @@
   @Test
   public void testReplaces()
   {
-    final ExtractionNamespace
-        en1 = mock(ExtractionNamespace.class),
-        en2 = mock(ExtractionNamespace.class);
+    final ExtractionNamespace en1 = mockExtractionNamespace();
+    final ExtractionNamespace en2 = mockExtractionNamespace();
 
     final NamespaceLookupExtractorFactory f1 = new NamespaceLookupExtractorFactory(
         en1,
@@ -499,7 +503,7 @@
   @Test
   public void testExceptionalIntrospectionHandler() throws Exception
   {
-    final ExtractionNamespace extractionNamespace = mock(ExtractionNamespace.class);
+    final ExtractionNamespace extractionNamespace = mockExtractionNamespace();
     when(scheduler.scheduleAndWait(eq(extractionNamespace), anyLong())).thenReturn(entry);
 
     final LookupExtractorFactory lookupExtractorFactory = new NamespaceLookupExtractorFactory(
