--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -41,11 +41,10 @@
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
-    final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    mockStorageParameters(frameContext);
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
@@ -77,11 +76,10 @@
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
-    final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    mockStorageParameters(frameContext);
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
@@ -118,12 +116,11 @@
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
-    final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    mockStorageParameters(frameContext);
     Mockito.doThrow(exception).when(exec).cancel(CANCELLATION_ID);
 
     final RunWorkOrder runWorkOrder =
@@ -156,12 +153,11 @@
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
-    final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    mockStorageParameters(frameContext);
     Mockito.doThrow(exception).when(frameContext).close();
 
     final RunWorkOrder runWorkOrder =
@@ -185,4 +181,10 @@
     Mockito.verify(frameContext).close();
     Mockito.verify(listener).onFailure(ArgumentMatchers.eq(exception));
   }
+
+  private static void mockStorageParameters(final FrameContext frameContext)
+  {
+    final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
+    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+  }
 }
