--- a/processing/src/test/java/org/apache/druid/query/filter/LikeDimFilterTest.java
+++ b/processing/src/test/java/org/apache/druid/query/filter/LikeDimFilterTest.java
@@ -113,7 +113,7 @@
     final ColumnIndexSelector indexSelector = Mockito.mock(ColumnIndexSelector.class);
     final ColumnIndexSupplier indexSupplier = Mockito.mock(ColumnIndexSupplier.class);
     final LexicographicalRangeIndexes rangeIndex = Mockito.mock(LexicographicalRangeIndexes.class);
-    final BitmapColumnIndex bitmapColumnIndex = Mockito.mock(BitmapColumnIndex.class);
+    final BitmapColumnIndex bitmapColumnIndex = makeBitmapColumnIndex();
 
     Mockito.when(indexSelector.getIndexSupplier("dim0")).thenReturn(indexSupplier);
     Mockito.when(indexSupplier.as(LexicographicalRangeIndexes.class)).thenReturn(rangeIndex);
@@ -137,7 +137,7 @@
     final ColumnIndexSelector indexSelector = Mockito.mock(ColumnIndexSelector.class);
     final ColumnIndexSupplier indexSupplier = Mockito.mock(ColumnIndexSupplier.class);
     final StringValueSetIndexes valueIndex = Mockito.mock(StringValueSetIndexes.class);
-    final BitmapColumnIndex bitmapColumnIndex = Mockito.mock(BitmapColumnIndex.class);
+    final BitmapColumnIndex bitmapColumnIndex = makeBitmapColumnIndex();
 
     Mockito.when(indexSelector.getIndexSupplier("dim0")).thenReturn(indexSupplier);
     Mockito.when(indexSupplier.as(StringValueSetIndexes.class)).thenReturn(valueIndex);
@@ -328,6 +328,11 @@
     Assert.assertEquals(pattern + " => " + expected, matcher.describeCompilation());
   }
 
+  private static BitmapColumnIndex makeBitmapColumnIndex()
+  {
+    return Mockito.mock(BitmapColumnIndex.class);
+  }
+
   private void assertMatch(String pattern, String value, DruidPredicateMatch expected)
   {
     LikeDimFilter.LikeMatcher matcher = LikeDimFilter.LikeMatcher.from(pattern, '\\');
