--- a/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
@@ -151,7 +151,7 @@
     }).when(cronScheduler).scheduleAtFixedRate(ArgumentMatchers.anyLong(), ArgumentMatchers.anyLong(),
         ArgumentMatchers.any(), ArgumentMatchers.any(CronTask.class));
 
-    Monitor monitor = Mockito.mock(Monitor.class);
+    final Monitor monitor = createMockMonitor();
 
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
@@ -215,7 +215,7 @@
     }).when(cronScheduler).scheduleAtFixedRate(ArgumentMatchers.anyLong(), ArgumentMatchers.anyLong(),
         ArgumentMatchers.any(), ArgumentMatchers.any(CronTask.class));
 
-    Monitor monitor = Mockito.mock(Monitor.class);
+    final Monitor monitor = createMockMonitor();
 
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
@@ -244,7 +244,7 @@
   public void testStart_UnexpectedExceptionWhileMonitoring() throws InterruptedException
   {
     ExecutorService executor = Mockito.mock(ExecutorService.class);
-    Monitor monitor = Mockito.mock(Monitor.class);
+    final Monitor monitor = createMockMonitor();
     Mockito.when(monitor.monitor(ArgumentMatchers.any(ServiceEmitter.class)))
            .thenThrow(new RuntimeException("Test throwing exception while monitoring"));
 
@@ -309,7 +309,7 @@
   public void testStart_UnexpectedExceptionWhileScheduling() throws InterruptedException
   {
     ExecutorService executor = Mockito.mock(ExecutorService.class);
-    Monitor monitor = Mockito.mock(Monitor.class);
+    final Monitor monitor = createMockMonitor();
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
 
@@ -393,6 +393,11 @@
 
     return future;
   }
+
+  private Monitor createMockMonitor()
+  {
+    return Mockito.mock(Monitor.class);
+  }
   
   
   private static class NoopMonitor implements Monitor
