--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -32,6 +32,7 @@
 public class RunWorkOrderTest
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
+  private final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
 
   @Test
   public void test_stop()
@@ -39,7 +40,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -75,7 +75,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -116,7 +115,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -154,7 +152,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
