--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
@@ -37,10 +37,10 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
-import org.mockito.Mockito;
 
 import javax.annotation.Nullable;
 import java.io.File;
+import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Paths;
 
@@ -65,6 +65,14 @@
     objectMapper = new TestUtils().getTestObjectMapper();
   }
 
+  private TaskActionClient createTaskActionClient(final TaskToolbox toolbox) throws IOException
+  {
+    final TaskActionClient taskActionClient = mock(TaskActionClient.class);
+    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+    when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
+    return taskActionClient;
+  }
+
   @Test
   public void testSetupAndCleanupIsCalledWtihParameter() throws Exception
   {
@@ -88,9 +96,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
-    when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
+    final TaskActionClient taskActionClient = createTaskActionClient(toolbox);
     when(toolbox.getTaskReportFileWriter()).thenReturn(new SingleFileTaskReportFileWriter(temporaryFolder.newFile()));
 
 
@@ -113,7 +119,7 @@
     task.run(toolbox);
 
     // call it 3 times, once to update location in setup, then one for status and location in cleanup
-    Mockito.verify(taskActionClient, times(3)).submit(any());
+    verify(taskActionClient, times(3)).submit(any());
     verify(pusher, times(1)).pushTaskReports(eq("myID"), any());
     verify(pusher, times(1)).pushTaskStatus(eq("myID"), any());
   }
@@ -137,9 +143,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
-    when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
+    final TaskActionClient taskActionClient = createTaskActionClient(toolbox);
 
 
     AbstractTask task = new NoopTask("myID", null, null, 1, 0, null)
@@ -159,7 +163,7 @@
     task.run(toolbox);
 
     // encapsulated task is set to false, should never get called
-    Mockito.verify(taskActionClient, never()).submit(any());
+    verify(taskActionClient, never()).submit(any());
     verify(pusher, never()).pushTaskReports(eq("myID"), any());
   }
 
@@ -182,9 +186,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
-    when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
+    final TaskActionClient taskActionClient = createTaskActionClient(toolbox);
     when(toolbox.getTaskReportFileWriter()).thenReturn(new SingleFileTaskReportFileWriter(temporaryFolder.newFile()));
 
     TaskStatus taskStatus = TaskStatus.failure("myId", "failed");
@@ -221,9 +223,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
-    when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
+    final TaskActionClient taskActionClient = createTaskActionClient(toolbox);
     AbstractTask task = new NoopTask("myID", null, null, 1, 0, null)
     {
       @Nullable
@@ -258,9 +258,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
-    when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
+    final TaskActionClient taskActionClient = createTaskActionClient(toolbox);
     AbstractTask task = new NoopTask("myID", null, null, 1, 0, null)
     {
       @Nullable
