--- a/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorIOConfigTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorIOConfigTest.java
@@ -49,7 +49,7 @@
   @Test
   public void testAllDefaults()
   {
-    LagAggregator lagAggregator = mock(LagAggregator.class);
+    final LagAggregator lagAggregator = makeLagAggregator();
     InputFormat inputFormat = mock(InputFormat.class);
 
     SeekableStreamSupervisorIOConfig config = new SeekableStreamSupervisorIOConfig(
@@ -96,7 +96,7 @@
   @Test
   public void testAutoScalerEnabledPreservesTaskCountWhenNonNull()
   {
-    LagAggregator lagAggregator = mock(LagAggregator.class);
+    final LagAggregator lagAggregator = makeLagAggregator();
 
     // autoScalerEnabled = true
     AutoScalerConfig autoScalerConfig = mock(AutoScalerConfig.class);
@@ -157,7 +157,7 @@
   @Test
   public void testBothLateMessageRejectionPeriodAndStartDateTime()
   {
-    LagAggregator lagAggregator = mock(LagAggregator.class);
+    final LagAggregator lagAggregator = makeLagAggregator();
 
     IAE ex = Assert.assertThrows(
         IAE.class,
@@ -226,7 +226,7 @@
   @Test
   public void testGetMaxAllowedStopsScalingDisabled()
   {
-    LagAggregator lagAggregator = mock(LagAggregator.class);
+    final LagAggregator lagAggregator = makeLagAggregator();
 
     // Autoscaler disabled, stopTaskCount unset
     SeekableStreamSupervisorIOConfig config1 = new SeekableStreamSupervisorIOConfig(
@@ -280,7 +280,7 @@
   @Test
   public void testGetMaxAllowedStopsScalingEnabled()
   {
-    LagAggregator lagAggregator = mock(LagAggregator.class);
+    final LagAggregator lagAggregator = makeLagAggregator();
 
     AutoScalerConfig autoScalerConfig = mock(AutoScalerConfig.class);
 
@@ -437,6 +437,11 @@
     );
   }
 
+  private static LagAggregator makeLagAggregator()
+  {
+    return mock(LagAggregator.class);
+  }
+
   private SeekableStreamSupervisorIOConfig makeSeekableStreamSupervisorIOConfig(@Nullable Integer replicas, @Nullable Map<Integer, Integer> serverPriorityToReplicas)
   {
     return new SeekableStreamSupervisorIOConfig(
@@ -452,7 +457,7 @@
         null,
         null,
         null,
-        mock(LagAggregator.class),
+        makeLagAggregator(),
         null,
         null,
         null,
