--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
@@ -81,10 +81,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = mockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -130,10 +128,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(false);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = mockTaskConfig(false, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -175,10 +171,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = mockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -214,10 +208,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = mockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -251,10 +243,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = mockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -273,6 +263,14 @@
     task.run(toolbox);
     UpdateStatusAction action = new UpdateStatusAction("", TaskStatus.failure(task.getId(), "setup error"));
     verify(taskActionClient).submit(eq(action));
+  }
+
+  private TaskConfig mockTaskConfig(final boolean encapsulatedTask, final File taskDir)
+  {
+    final TaskConfig config = mock(TaskConfig.class);
+    when(config.isEncapsulatedTask()).thenReturn(encapsulatedTask);
+    when(config.getTaskDir(eq("myID"))).thenReturn(taskDir);
+    return config;
   }
 
 
