--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateLocationActionTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateLocationActionTest.java
@@ -43,6 +43,8 @@
 
 public class UpdateLocationActionTest
 {
+  private final TaskRunner runner = mock(TaskRunner.class);
+
   @Test
   public void testFlow() throws UnknownHostException
   {
@@ -52,7 +54,6 @@
     UpdateLocationAction action = new UpdateLocationAction(myLocation);
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
     when(toolbox.getTaskRunner()).thenReturn(Optional.of(runner));
     action.perform(task, toolbox);
     verify(runner, times(1)).updateLocation(eq(task), eq(myLocation));
@@ -67,7 +68,6 @@
     UpdateLocationAction action = new UpdateLocationAction(myLocation);
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
     when(toolbox.getTaskRunner()).thenReturn(Optional.absent());
     action.perform(task, toolbox);
     verify(runner, never()).updateStatus(any(), any());
