--- a/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunnerTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunnerTest.java
@@ -123,7 +123,7 @@
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
-    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
+    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = createStartSequenceNumbers();
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
 
     DateTime now = DateTimes.nowUtc();
@@ -136,7 +136,6 @@
     Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
-    Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
@@ -178,7 +177,7 @@
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
-    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
+    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = createStartSequenceNumbers();
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
 
     DateTime now = DateTimes.nowUtc();
@@ -192,7 +191,6 @@
     Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
-    Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
@@ -233,7 +231,7 @@
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
-    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
+    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = createStartSequenceNumbers();
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
 
     Mockito.when(ioConfig.getRefreshRejectionPeriodsInMinutes()).thenReturn(null);
@@ -244,7 +242,6 @@
     Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
-    Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
@@ -279,7 +276,7 @@
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
-    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
+    SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = createStartSequenceNumbers();
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
 
     Mockito.when(ioConfig.getRefreshRejectionPeriodsInMinutes()).thenReturn(null);
@@ -288,7 +285,6 @@
     Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
-    Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
@@ -335,6 +331,15 @@
     runner.run(createTaskToolbox());
     emitter.verifyValue("ingest/segments/count", 10);
     emitter.verifyValue("ingest/rows/published", 10_000L);
+  }
+
+  private SeekableStreamStartSequenceNumbers<String, String> createStartSequenceNumbers()
+  {
+    final SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(
+        SeekableStreamStartSequenceNumbers.class
+    );
+    Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
+    return sequenceNumbers;
   }
 
   private TaskToolbox createTaskToolbox()
