--- a/indexing-service/src/test/java/org/apache/druid/indexing/overlord/RemoteTaskRunnerFactoryTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/overlord/RemoteTaskRunnerFactoryTest.java
@@ -34,8 +34,7 @@
   @Test
   public void testBuildWithAutoScale()
   {
-    ProvisioningSchedulerConfig provisioningSchedulerConfig = Mockito.mock(ProvisioningSchedulerConfig.class);
-    Mockito.when(provisioningSchedulerConfig.isDoAutoscale()).thenReturn(true);
+    final ProvisioningSchedulerConfig provisioningSchedulerConfig = getProvisioningSchedulerConfig(true);
 
     RemoteTaskRunnerFactory remoteTaskRunnerFactory = getTestRemoteTaskRunnerFactory(provisioningSchedulerConfig);
 
@@ -45,8 +44,7 @@
   @Test
   public void testBuildWithoutAutoScale()
   {
-    ProvisioningSchedulerConfig provisioningSchedulerConfig = Mockito.mock(ProvisioningSchedulerConfig.class);
-    Mockito.when(provisioningSchedulerConfig.isDoAutoscale()).thenReturn(false);
+    final ProvisioningSchedulerConfig provisioningSchedulerConfig = getProvisioningSchedulerConfig(false);
 
     RemoteTaskRunnerFactory remoteTaskRunnerFactory = getTestRemoteTaskRunnerFactory(provisioningSchedulerConfig);
 
@@ -69,4 +67,11 @@
         null
     );
   }
+
+  private ProvisioningSchedulerConfig getProvisioningSchedulerConfig(final boolean doAutoscale)
+  {
+    final ProvisioningSchedulerConfig provisioningSchedulerConfig = Mockito.mock(ProvisioningSchedulerConfig.class);
+    Mockito.when(provisioningSchedulerConfig.isDoAutoscale()).thenReturn(doAutoscale);
+    return provisioningSchedulerConfig;
+  }
 }
