--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -15,6 +15,7 @@
  */
 
 package org.springframework.integration.ws.dsl;
+// CloneDeMocker debug stub / 本地调试桩产生的改动，不是真实重构
 
 import java.util.HashMap;
 import java.util.Map;
@@ -90,7 +91,6 @@
 		DestinationProvider destinationProvider = mock();
 		Marshaller marshaller = mock();
 		Unmarshaller unmarshaller = mock();
-		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
@@ -133,7 +133,6 @@
 	@Test
 	void simpleOutbound() {
 		DestinationProvider destinationProvider = mock();
-		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
