--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyChannelTest.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyChannelTest.java
@@ -81,12 +81,10 @@
         Mockito.when(channel.eventLoop()).thenReturn(Mockito.mock(EventLoop.class));
         Mockito.when(channel.alloc()).thenReturn(PooledByteBufAllocator.DEFAULT);
         NettyChannel nettyChannel = NettyChannel.getOrAddChannel(channel, url, channelHandler);
-        ChannelPromise future = Mockito.mock(ChannelPromise.class);
+        ChannelPromise future = MockChannelPromise.createMockChannelPromise();
         Mockito.when(future.await(1000)).thenReturn(true);
-        Mockito.when(future.cause()).thenReturn(null);
         Mockito.when(channel.writeAndFlush(Mockito.any())).thenReturn(future);
         Mockito.when(channel.newPromise()).thenReturn(future);
-        Mockito.when(future.addListener(Mockito.any())).thenReturn(future);
         nettyChannel.send("msg", true);
 
         NettyChannel finalNettyChannel = nettyChannel;
@@ -138,4 +136,16 @@
         Assertions.assertEquals(nettyChannel, nettyChannel);
         Assertions.assertNotEquals(nettyChannel, nettyChannel2);
     }
+
+    public final class MockChannelPromise {
+        private MockChannelPromise() {
+        }
+
+        public static ChannelPromise createMockChannelPromise() {
+            ChannelPromise future = Mockito.mock(ChannelPromise.class);
+            Mockito.when(future.addListener(Mockito.any())).thenReturn(future);
+            Mockito.when(future.cause()).thenReturn(null);
+            return future;
+        }
+    }
 }
--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandlerTest.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandlerTest.java
@@ -48,11 +48,9 @@
         Mockito.when(channel.eventLoop()).thenReturn(new NioEventLoopGroup().next());
         Mockito.when(channel.alloc()).thenReturn(PooledByteBufAllocator.DEFAULT);
 
-        ChannelPromise future = mock(ChannelPromise.class);
+        ChannelPromise future = MockChannelPromise.createMockChannelPromise();
         when(channel.writeAndFlush(any())).thenReturn(future);
-        when(future.cause()).thenReturn(null);
         when(channel.newPromise()).thenReturn(future);
-        when(future.addListener(Mockito.any())).thenReturn(future);
 
         NettyClientHandler nettyClientHandler = new NettyClientHandler(url, handler);
 
@@ -80,4 +78,16 @@
         Thread.sleep(500);
         Mockito.verify(channel, Mockito.times(1)).writeAndFlush(requestArgumentCaptor.capture());
     }
+
+    public final class MockChannelPromise {
+        private MockChannelPromise() {
+        }
+
+        public static ChannelPromise createMockChannelPromise() {
+            ChannelPromise future = Mockito.mock(ChannelPromise.class);
+            Mockito.when(future.addListener(Mockito.any())).thenReturn(future);
+            Mockito.when(future.cause()).thenReturn(null);
+            return future;
+        }
+    }
 }
