--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/CtrlCHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/CtrlCHandlerTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.qos.server.handler;
 
+import org.mockito.Mock;
 import org.apache.dubbo.qos.common.QosConstants;
 
 import java.nio.charset.StandardCharsets;
@@ -30,13 +31,14 @@
 import static org.mockito.Mockito.verify;
 
 public class CtrlCHandlerTest {
+    @Mock
+    private ChannelHandlerContext context;
     private byte[] CTRLC_BYTES_SEQUENCE = new byte[] {(byte) 0xff, (byte) 0xf4, (byte) 0xff, (byte) 0xfd, (byte) 0x06};
 
     private byte[] RESPONSE_SEQUENCE = new byte[] {(byte) 0xff, (byte) 0xfc, 0x06};
 
     @Test
     void testMatchedExactly() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         CtrlCHandler ctrlCHandler = new CtrlCHandler();
         ctrlCHandler.channelRead(context, Unpooled.wrappedBuffer(CTRLC_BYTES_SEQUENCE));
         verify(context).writeAndFlush(Unpooled.wrappedBuffer(RESPONSE_SEQUENCE));
@@ -47,7 +49,6 @@
 
     @Test
     void testMatchedNotExactly() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         CtrlCHandler ctrlCHandler = new CtrlCHandler();
         // before 'ctrl c', user typed other command like 'help'
         String arbitraryCommand = "help";
@@ -61,7 +62,6 @@
 
     @Test
     void testNotMatched() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         CtrlCHandler ctrlCHandler = new CtrlCHandler();
         String arbitraryCommand = "help" + QosConstants.BR_STR;
         byte[] commandBytes = arbitraryCommand.getBytes(StandardCharsets.UTF_8);
