--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.qos.server.handler;
 
+import org.mockito.Mock;
 import org.apache.dubbo.common.utils.StringUtils;
 import org.apache.dubbo.qos.api.QosConfiguration;
 import org.apache.dubbo.rpc.model.FrameworkModel;
@@ -38,11 +39,12 @@
 import static org.mockito.Mockito.verify;
 
 class QosProcessHandlerTest {
+    @Mock
+    private ChannelPipeline pipeline;
     @Test
     void testDecodeHttp() throws Exception {
         ByteBuf buf = Unpooled.wrappedBuffer(new byte[] {'G'});
         ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
-        ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
         Mockito.when(context.pipeline()).thenReturn(pipeline);
         QosProcessHandler handler = new QosProcessHandler(
                 FrameworkModel.defaultModel(),
@@ -62,7 +64,6 @@
     void testDecodeTelnet() throws Exception {
         ByteBuf buf = Unpooled.wrappedBuffer(new byte[] {'A'});
         ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
-        ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
         Mockito.when(context.pipeline()).thenReturn(pipeline);
         QosProcessHandler handler = new QosProcessHandler(
                 FrameworkModel.defaultModel(),
