--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.registry.nacos;
 
+import org.mockito.Mock;
 import org.apache.dubbo.common.URL;
 
 import java.util.ArrayList;
@@ -35,6 +36,8 @@
 import org.mockito.Mockito;
 
 class NacosNamingServiceWrapperTest {
+    @Mock
+    private EventListener eventListener;
     @Test
     void testSubscribe() throws NacosException {
         NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
@@ -43,7 +46,6 @@
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
-        EventListener eventListener = Mockito.mock(EventListener.class);
         nacosNamingServiceWrapper.subscribe("service_name", "test", eventListener);
         Mockito.verify(namingService, Mockito.times(1)).subscribe("service_name", "test", eventListener);
 
@@ -68,7 +70,6 @@
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
-        EventListener eventListener = Mockito.mock(EventListener.class);
         Mockito.when(connectionManager.getNamingService()).thenReturn(namingService1);
         nacosNamingServiceWrapper.subscribe("service_name", "test", eventListener);
         Mockito.verify(namingService1, Mockito.times(1)).subscribe("service_name", "test", eventListener);
