--- a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/NetUtilsInterfaceDisplayNameHasMetaCharactersTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/NetUtilsInterfaceDisplayNameHasMetaCharactersTest.java
@@ -39,8 +39,8 @@
         String originIgnoredInterfaces = this.getIgnoredInterfaces();
         // mock static methods of final class NetworkInterface
         try (MockedStatic<NetworkInterface> mockedStaticNetif = Mockito.mockStatic(NetworkInterface.class)) {
-            NetworkInterface mockIgnoredNetif = Mockito.mock(NetworkInterface.class);
-            NetworkInterface mockSelectedNetif = Mockito.mock(NetworkInterface.class);
+            NetworkInterface mockIgnoredNetif = createMockNetworkInterface(IGNORED_DISPLAY_NAME_HAS_METACHARACTERS);
+            NetworkInterface mockSelectedNetif = createMockNetworkInterface(SELECTED_DISPLAY_NAME);
             NetworkInterface[] mockNetifs = {mockIgnoredNetif, mockSelectedNetif};
             Enumeration<NetworkInterface> mockEnumIfs = new Enumeration<NetworkInterface>() {
                 private int i = 0;
@@ -85,11 +85,7 @@
                     })
                     .thenReturn(mockEnumIfs);
 
-            Mockito.when(mockIgnoredNetif.isUp()).thenReturn(true);
-            Mockito.when(mockIgnoredNetif.getDisplayName()).thenReturn(IGNORED_DISPLAY_NAME_HAS_METACHARACTERS);
 
-            Mockito.when(mockSelectedNetif.isUp()).thenReturn(true);
-            Mockito.when(mockSelectedNetif.getDisplayName()).thenReturn(SELECTED_DISPLAY_NAME);
             Mockito.when(mockSelectedNetif.getInetAddresses()).thenReturn(mockEnumAddrs);
 
             Mockito.when(mockSelectedAddr.isLoopbackAddress()).thenReturn(false);
@@ -119,4 +115,11 @@
                     CommonConstants.DubboProperty.DUBBO_NETWORK_IGNORED_INTERFACE, "");
         }
     }
+
+    private static NetworkInterface createMockNetworkInterface(String displayName) {
+        NetworkInterface mockNetworkInterface = Mockito.mock(NetworkInterface.class);
+        Mockito.when(mockNetworkInterface.isUp()).thenReturn(true);
+        Mockito.when(mockNetworkInterface.getDisplayName()).thenReturn(displayName);
+        return mockNetworkInterface;
+    }
 }
