--- a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassUtilsTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassUtilsTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.common.utils;
 
+import org.mockito.Mock;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -27,11 +28,12 @@
 import static org.mockito.Mockito.verify;
 
 class ClassUtilsTest {
+    @Mock
+    private ClassLoader classLoader;
     @Test
     void testForNameWithThreadContextClassLoader() throws Exception {
         ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
         try {
-            ClassLoader classLoader = Mockito.mock(ClassLoader.class);
             Thread.currentThread().setContextClassLoader(classLoader);
             ClassUtils.forNameWithThreadContextClassLoader("a.b.c.D");
             verify(classLoader).loadClass("a.b.c.D");
@@ -86,7 +88,6 @@
 
     @Test
     void testForName3() throws Exception {
-        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
         ClassUtils.forName("a.b.c.D", classLoader);
         verify(classLoader).loadClass("a.b.c.D");
     }
