--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.registry.integration;
 
+import org.mockito.Mock;
 import org.apache.dubbo.common.URL;
 import org.apache.dubbo.common.config.CompositeConfiguration;
 import org.apache.dubbo.common.extension.ExtensionLoader;
@@ -64,6 +65,8 @@
 import static org.mockito.Mockito.when;
 
 class RegistryProtocolTest {
+    @Mock
+    private Registry registry;
 
     @AfterEach
     public void tearDown() throws IOException {
@@ -101,7 +104,6 @@
         URL url = serviceConfigURL.addAttributes(attributes);
 
         RegistryFactory registryFactory = mock(RegistryFactory.class);
-        Registry registry = mock(Registry.class);
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
 
@@ -171,7 +173,6 @@
         RegistryFactory registryFactory = mock(RegistryFactory.class);
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
-        Registry registry = mock(Registry.class);
 
         MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
@@ -238,7 +239,6 @@
         URL url = serviceConfigURL.addAttributes(attributes);
 
         RegistryFactory registryFactory = mock(RegistryFactory.class);
-        Registry registry = mock(Registry.class);
 
         MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
@@ -305,7 +305,6 @@
         URL url = serviceConfigURL.addAttributes(attributes);
 
         RegistryFactory registryFactory = mock(RegistryFactory.class);
-        Registry registry = mock(Registry.class);
 
         MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
@@ -505,7 +504,6 @@
         URL url = serviceConfigURL.addAttributes(attributes);
 
         RegistryFactory registryFactory = mock(RegistryFactory.class);
-        Registry registry = mock(Registry.class);
 
         ModuleModel moduleModel = Mockito.spy(ApplicationModel.defaultModel().getDefaultModule());
         moduleModel
