--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
@@ -64,9 +64,7 @@
 
     @Test
     void testSuccessLogger() {
-        Logger successLogger = mock(Logger.class);
-        when(successLogger.isErrorEnabled()).thenReturn(true);
-        when(successLogger.isWarnEnabled()).thenReturn(true);
+        Logger successLogger = MockLogger.createMockLogger();
         FailsafeErrorTypeAwareLogger failsafeLogger = new FailsafeErrorTypeAwareLogger(successLogger);
 
         failsafeLogger.error(REGISTRY_ADDRESS_INVALID, "Registry center", "May be it's offline.", "error");
@@ -112,4 +110,18 @@
                 "error message",
                 new Exception("error"));
     }
+
+    public final class MockLogger {
+
+        private MockLogger() {
+            // Utility class
+        }
+
+        public static Logger createMockLogger() {
+            Logger logger = Mockito.mock(Logger.class);
+            when(logger.isErrorEnabled()).thenReturn(true);
+            when(logger.isWarnEnabled()).thenReturn(true);
+            return logger;
+        }
+    }
 }
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
@@ -67,9 +67,7 @@
 
     @Test
     void testSuccessLogger() {
-        Logger successLogger = mock(Logger.class);
-        Mockito.when(successLogger.isErrorEnabled()).thenReturn(true);
-        Mockito.when(successLogger.isWarnEnabled()).thenReturn(true);
+        Logger successLogger = MockLogger.createMockLogger();
         Mockito.when(successLogger.isInfoEnabled()).thenReturn(true);
         Mockito.when(successLogger.isDebugEnabled()).thenReturn(true);
         Mockito.when(successLogger.isTraceEnabled()).thenReturn(true);
@@ -110,4 +108,18 @@
             failsafeLogger.getLogger().error("should get error");
         });
     }
+
+    public final class MockLogger {
+
+        private MockLogger() {
+            // Utility class
+        }
+
+        public static Logger createMockLogger() {
+            Logger logger = Mockito.mock(Logger.class);
+            when(logger.isErrorEnabled()).thenReturn(true);
+            when(logger.isWarnEnabled()).thenReturn(true);
+            return logger;
+        }
+    }
 }
