--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/CompatibleFilterFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/CompatibleFilterFilterTest.java
@@ -50,11 +50,8 @@
 
     @Test
     void testInvokerGeneric() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("$enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Enum.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
 
+        invocation = MockInvocation.createMockInvocation("$enumlength", new Class<?>[] { Enum.class }, new Object[] { "hello" });
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
@@ -70,11 +67,8 @@
 
     @Test
     void testResultHasException() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Enum.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
 
+        invocation = MockInvocation.createMockInvocation("enumlength", new Class<?>[] { Enum.class }, new Object[] { "hello" });
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
@@ -91,11 +85,8 @@
 
     @Test
     void testInvokerJsonPojoSerialization() throws Exception {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Type[].class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
 
+        invocation = MockInvocation.createMockInvocation("enumlength", new Class<?>[] { Type[].class }, new Object[] { "hello" });
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
@@ -114,11 +105,11 @@
 
     @Test
     void testInvokerNonJsonEnumSerialization() throws Exception {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Type[].class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
 
+        invocation = MockInvocation.createMockInvocation(
+                "enumlength",
+                new Class<?>[] { Type[].class },
+                new Object[] { "hello" });
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
@@ -137,11 +128,8 @@
 
     @Test
     void testInvokerNonJsonNonPojoSerialization() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("echo");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {String.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
 
+        invocation = MockInvocation.createMockInvocation("echo", new Class<?>[] { String.class }, new Object[] { "hello" });
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
@@ -157,11 +145,8 @@
 
     @Test
     void testInvokerNonJsonPojoSerialization() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("echo");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {String.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
 
+        invocation = MockInvocation.createMockInvocation("echo", new Class<?>[] { String.class }, new Object[] { "hello" });
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
         given(invoker.getInterface()).willReturn(DemoService.class);
@@ -174,4 +159,21 @@
         Result filterResult = compatibleFilter.invoke(invoker, invocation);
         assertEquals("hello", filterResult.getValue());
     }
+
+    public final class MockInvocation {
+
+        private MockInvocation() {
+        }
+
+        public static org.apache.dubbo.rpc.Invocation createMockInvocation(
+                String methodName,
+                Class<?>[] parameterTypes,
+                Object[] arguments) {
+            org.apache.dubbo.rpc.Invocation invocation = mock(org.apache.dubbo.rpc.Invocation.class);
+            given(invocation.getMethodName()).willReturn(methodName);
+            given(invocation.getParameterTypes()).willReturn(parameterTypes);
+            given(invocation.getArguments()).willReturn(arguments);
+            return invocation;
+        }
+    }
 }
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
@@ -47,10 +47,7 @@
     @SuppressWarnings("unchecked")
     @Test
     void testSetContext() {
-        invocation = mock(Invocation.class);
-        given(invocation.getMethodName()).willReturn("$enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Enum.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocation.createMockInvocation("$enumlength", new Class<?>[] { Enum.class }, new Object[] { "hello" });
         given(invocation.getObjectAttachments()).willReturn(null);
 
         invoker = mock(Invoker.class);
@@ -74,4 +71,21 @@
         Result result = contextFilter.invoke(invoker, invocation);
         assertNotNull(RpcContext.getServiceContext().getInvoker());
     }
+
+    public final class MockInvocation {
+
+        private MockInvocation() {
+        }
+
+        public static org.apache.dubbo.rpc.Invocation createMockInvocation(
+                String methodName,
+                Class<?>[] parameterTypes,
+                Object[] arguments) {
+            org.apache.dubbo.rpc.Invocation invocation = mock(org.apache.dubbo.rpc.Invocation.class);
+            given(invocation.getMethodName()).willReturn(methodName);
+            given(invocation.getParameterTypes()).willReturn(parameterTypes);
+            given(invocation.getArguments()).willReturn(arguments);
+            return invocation;
+        }
+    }
 }
