--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
@@ -57,7 +57,6 @@
     @SuppressWarnings("all")
     @Test
     void test() {
-        RegistryProtocol registryProtocol = Mockito.mock(RegistryProtocol.class);
 
         ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
         ClusterInvoker serviceDiscoveryInvoker = Mockito.mock(ClusterInvoker.class);
@@ -83,12 +82,8 @@
         Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
         Mockito.when(serviceDiscoveryDirectory.getAllInvokers()).thenReturn(serviceDiscoveryInvokers);
 
-        Mockito.when(registryProtocol.getInvoker(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
-                .thenReturn(invoker);
-        Mockito.when(registryProtocol.getServiceDiscoveryInvoker(
-                        Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
-                .thenReturn(serviceDiscoveryInvoker);
-
+
+        RegistryProtocol registryProtocol = createRegistryProtocol(invoker, serviceDiscoveryInvoker);
         URL consumerURL = Mockito.mock(URL.class);
         Mockito.when(consumerURL.getServiceInterface()).thenReturn("Test");
         Mockito.when(consumerURL.getGroup()).thenReturn("Group");
@@ -233,7 +228,6 @@
     @SuppressWarnings("all")
     @Test
     void testDecide() {
-        RegistryProtocol registryProtocol = Mockito.mock(RegistryProtocol.class);
 
         ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
         ClusterInvoker serviceDiscoveryInvoker = Mockito.mock(ClusterInvoker.class);
@@ -259,12 +253,8 @@
         Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
         Mockito.when(serviceDiscoveryDirectory.getAllInvokers()).thenReturn(serviceDiscoveryInvokers);
 
-        Mockito.when(registryProtocol.getInvoker(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
-                .thenReturn(invoker);
-        Mockito.when(registryProtocol.getServiceDiscoveryInvoker(
-                        Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
-                .thenReturn(serviceDiscoveryInvoker);
-
+
+        RegistryProtocol registryProtocol = createRegistryProtocol(invoker, serviceDiscoveryInvoker);
         URL consumerURL = Mockito.mock(URL.class);
         Mockito.when(consumerURL.getServiceInterface()).thenReturn("Test");
         Mockito.when(consumerURL.getGroup()).thenReturn("Group");
@@ -302,4 +292,23 @@
         // 独立线程invoker状态切换
 
     }
+
+    private static RegistryProtocol createRegistryProtocol(
+            ClusterInvoker invoker,
+            ClusterInvoker serviceDiscoveryInvoker) {
+        RegistryProtocol registryProtocol = Mockito.mock(RegistryProtocol.class);
+        Mockito.when(registryProtocol.getInvoker(
+                Mockito.any(),
+                Mockito.any(),
+                Mockito.any(),
+                Mockito.any()))
+                .thenReturn(invoker);
+        Mockito.when(registryProtocol.getServiceDiscoveryInvoker(
+                Mockito.any(),
+                Mockito.any(),
+                Mockito.any(),
+                Mockito.any()))
+                .thenReturn(serviceDiscoveryInvoker);
+        return registryProtocol;
+    }
 }
