--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mesh/route/MeshRuleManagerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mesh/route/MeshRuleManagerTest.java
@@ -196,13 +196,11 @@
 
     @Test
     void testRegister3() {
-        MeshEnvListenerFactory meshEnvListenerFactory1 = Mockito.mock(MeshEnvListenerFactory.class);
-        MeshEnvListenerFactory meshEnvListenerFactory2 = Mockito.mock(MeshEnvListenerFactory.class);
+        MeshEnvListenerFactory meshEnvListenerFactory2 = createMeshEnvListenerFactory(meshEnvListener2);
 
         MeshEnvListener meshEnvListener1 = Mockito.mock(MeshEnvListener.class);
-        when(meshEnvListenerFactory1.getListener()).thenReturn(meshEnvListener1);
+        MeshEnvListenerFactory meshEnvListenerFactory1 = createMeshEnvListenerFactory(meshEnvListener1);
         MeshEnvListener meshEnvListener2 = Mockito.mock(MeshEnvListener.class);
-        when(meshEnvListenerFactory2.getListener()).thenReturn(meshEnvListener2);
 
         envListenerFactories.add(meshEnvListenerFactory1);
         envListenerFactories.add(meshEnvListenerFactory2);
@@ -270,4 +268,10 @@
         verify(ruleRepository, times(1)).removeListener("dubbo-demo.MESHAPPRULE", "dubbo", meshAppRuleListener);
         verify(meshEnvListener2, times(1)).onUnSubscribe("dubbo-demo");
     }
+
+    private static MeshEnvListenerFactory createMeshEnvListenerFactory(MeshEnvListener listener) {
+        MeshEnvListenerFactory meshEnvListenerFactory = Mockito.mock(MeshEnvListenerFactory.class);
+        when(meshEnvListenerFactory.getListener()).thenReturn(listener);
+        return meshEnvListenerFactory;
+    }
 }
