--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerTest.java
@@ -137,7 +137,6 @@
     @BeforeAll
     public static void setUp() {
 
-        metadataService = Mockito.mock(MetadataService.class);
 
         List<Object> urlsSameRevision = new ArrayList<>();
         urlsSameRevision.add("127.0.0.1:20880?revision=111");
@@ -688,7 +687,7 @@
         ServiceDiscovery serviceDiscovery = Mockito.mock(ServiceDiscovery.class);
         listener = new ServiceInstancesChangedListener(serviceNames, serviceDiscovery);
         ServiceInstancesChangedListener spyListener = Mockito.spy(listener);
-        Mockito.doReturn(null).when(metadataService).getMetadataInfo(eq(null));
+        metadataService = MockMetadataService.createMockMetadataService();
         ServiceInstancesChangedEvent event = new ServiceInstancesChangedEvent("app1", app1InstancesWithNoRevision);
         spyListener.onEvent(event);
         // notification succeeded
@@ -740,4 +739,16 @@
             // ignore
         }
     }
+
+    public final class MockMetadataService {
+
+        private MockMetadataService() {
+        }
+
+        public static MetadataService createMockMetadataService() {
+            MetadataService metadataService = Mockito.mock(MetadataService.class);
+            Mockito.doReturn(null).when(metadataService).getMetadataInfo(eq(null));
+            return metadataService;
+        }
+    }
 }
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerWithoutEmptyProtectTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerWithoutEmptyProtectTest.java
@@ -136,7 +136,6 @@
     @BeforeAll
     public static void setUp() {
 
-        metadataService = Mockito.mock(MetadataService.class);
 
         List<Object> urlsSameRevision = new ArrayList<>();
         urlsSameRevision.add("127.0.0.1:20880?revision=111");
@@ -687,7 +686,7 @@
         ServiceDiscovery serviceDiscovery = Mockito.mock(ServiceDiscovery.class);
         listener = new ServiceInstancesChangedListener(serviceNames, serviceDiscovery);
         ServiceInstancesChangedListener spyListener = Mockito.spy(listener);
-        Mockito.doReturn(null).when(metadataService).getMetadataInfo(eq(null));
+        metadataService = createMockMetadataService();
         ServiceInstancesChangedEvent event = new ServiceInstancesChangedEvent("app1", app1InstancesWithNoRevision);
         spyListener.onEvent(event);
         // notification succeeded
@@ -739,4 +738,16 @@
             // ignore
         }
     }
+
+    public final class MockMetadataService {
+
+        private MockMetadataService() {
+        }
+
+        public static MetadataService createMockMetadataService() {
+            MetadataService metadataService = Mockito.mock(MetadataService.class);
+            Mockito.doReturn(null).when(metadataService).getMetadataInfo(eq(null));
+            return metadataService;
+        }
+    }
 }
