--- a/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanExporterTest.java
+++ b/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanExporterTest.java
@@ -32,12 +32,22 @@
 
     @Test
     void getSpanExporter() {
-        ExporterConfig.ZipkinConfig zipkinConfig = mock(ExporterConfig.ZipkinConfig.class);
-        when(zipkinConfig.getEndpoint()).thenReturn("http://localhost:9411/api/v2/spans");
-        when(zipkinConfig.getConnectTimeout()).thenReturn(Duration.ofSeconds(5));
+        ExporterConfig.ZipkinConfig zipkinConfig = MockZipkinConfig.createZipkinConfig();
         when(zipkinConfig.getReadTimeout()).thenReturn(Duration.ofSeconds(5));
 
         SpanExporter spanExporter = ZipkinSpanExporter.getSpanExporter(ApplicationModel.defaultModel(), zipkinConfig);
         Assertions.assertNotNull(spanExporter);
     }
+
+    public final class MockZipkinConfig {
+        private MockZipkinConfig() {
+        }
+
+        public static org.apache.dubbo.config.nested.ExporterConfig.ZipkinConfig createZipkinConfig() {
+            org.apache.dubbo.config.nested.ExporterConfig.ZipkinConfig zipkinConfig = mock(org.apache.dubbo.config.nested.ExporterConfig.ZipkinConfig.class);
+            when(zipkinConfig.getEndpoint()).thenReturn("http://localhost:9411/api/v2/spans");
+            when(zipkinConfig.getConnectTimeout()).thenReturn(java.time.Duration.ofSeconds(5));
+            return zipkinConfig;
+        }
+    }
 }
--- a/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanHandlerTest.java
+++ b/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanHandlerTest.java
@@ -32,11 +32,21 @@
 
     @Test
     void getSpanHandler() {
-        ExporterConfig.ZipkinConfig zipkinConfig = mock(ExporterConfig.ZipkinConfig.class);
-        when(zipkinConfig.getEndpoint()).thenReturn("http://localhost:9411/api/v2/spans");
-        when(zipkinConfig.getConnectTimeout()).thenReturn(Duration.ofSeconds(5));
 
+        ExporterConfig.ZipkinConfig zipkinConfig = MockZipkinConfig.createZipkinConfig();
         SpanHandler spanHandler = ZipkinSpanHandler.getSpanHandler(ApplicationModel.defaultModel(), zipkinConfig);
         Assertions.assertNotNull(spanHandler);
     }
+
+    public final class MockZipkinConfig {
+        private MockZipkinConfig() {
+        }
+
+        public static org.apache.dubbo.config.nested.ExporterConfig.ZipkinConfig createZipkinConfig() {
+            org.apache.dubbo.config.nested.ExporterConfig.ZipkinConfig zipkinConfig = mock(org.apache.dubbo.config.nested.ExporterConfig.ZipkinConfig.class);
+            when(zipkinConfig.getEndpoint()).thenReturn("http://localhost:9411/api/v2/spans");
+            when(zipkinConfig.getConnectTimeout()).thenReturn(java.time.Duration.ofSeconds(5));
+            return zipkinConfig;
+        }
+    }
 }
