--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/impl/SerializeCheckStatusTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/impl/SerializeCheckStatusTest.java
@@ -33,10 +33,8 @@
 
         SerializeCheckStatus serializeCheckStatus = new SerializeCheckStatus(frameworkModel);
 
-        CommandContext commandContext1 = Mockito.mock(CommandContext.class);
-        Mockito.when(commandContext1.isHttp()).thenReturn(false);
-        CommandContext commandContext2 = Mockito.mock(CommandContext.class);
-        Mockito.when(commandContext2.isHttp()).thenReturn(true);
+        CommandContext commandContext1 = MockCommandContext.createMockCommandContext(false);
+        CommandContext commandContext2 = MockCommandContext.createMockCommandContext(true);
 
         Assertions.assertFalse(
                 serializeCheckStatus.execute(commandContext1, null).contains("Test1234"));
@@ -80,4 +78,15 @@
 
         frameworkModel.destroy();
     }
+
+    public final class MockCommandContext {
+        private MockCommandContext() {
+        }
+
+        public static org.apache.dubbo.qos.api.CommandContext createMockCommandContext(boolean isHttp) {
+            org.apache.dubbo.qos.api.CommandContext commandContext = Mockito.mock(org.apache.dubbo.qos.api.CommandContext.class);
+            Mockito.when(commandContext.isHttp()).thenReturn(isHttp);
+            return commandContext;
+        }
+    }
 }
--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/impl/SerializeWarnedClassesTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/impl/SerializeWarnedClassesTest.java
@@ -33,10 +33,8 @@
 
         SerializeWarnedClasses serializeWarnedClasses = new SerializeWarnedClasses(frameworkModel);
 
-        CommandContext commandContext1 = Mockito.mock(CommandContext.class);
-        Mockito.when(commandContext1.isHttp()).thenReturn(false);
-        CommandContext commandContext2 = Mockito.mock(CommandContext.class);
-        Mockito.when(commandContext2.isHttp()).thenReturn(true);
+        CommandContext commandContext1 = MockCommandContext.createMockCommandContext(false);
+        CommandContext commandContext2 = MockCommandContext.createMockCommandContext(true);
 
         Assertions.assertFalse(
                 serializeWarnedClasses.execute(commandContext1, null).contains("Test1234"));
@@ -60,4 +58,15 @@
 
         frameworkModel.destroy();
     }
+
+    public final class MockCommandContext {
+        private MockCommandContext() {
+        }
+
+        public static org.apache.dubbo.qos.api.CommandContext createMockCommandContext(boolean isHttp) {
+            org.apache.dubbo.qos.api.CommandContext commandContext = Mockito.mock(org.apache.dubbo.qos.api.CommandContext.class);
+            Mockito.when(commandContext.isHttp()).thenReturn(isHttp);
+            return commandContext;
+        }
+    }
 }
