--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
@@ -84,10 +84,9 @@
     @Test
     void testGetLogger() {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            Logger failLogger = mock(Logger.class);
+            Logger failLogger = MockLogger.createMockLogger();
             FailsafeErrorTypeAwareLogger failsafeLogger = new FailsafeErrorTypeAwareLogger(failLogger);
 
-            doThrow(new RuntimeException()).when(failLogger).error(anyString());
             failsafeLogger.getLogger().error("should get error");
         });
     }
@@ -112,4 +111,15 @@
                 "error message",
                 new Exception("error"));
     }
+
+    public final class MockLogger {
+        private MockLogger() {
+        }
+
+        public static org.apache.dubbo.common.logger.Logger createMockLogger() {
+            org.apache.dubbo.common.logger.Logger failLogger = org.mockito.Mockito.mock(org.apache.dubbo.common.logger.Logger.class);
+            org.mockito.Mockito.doThrow(new RuntimeException()).when(failLogger).error(org.mockito.ArgumentMatchers.anyString());
+            return failLogger;
+        }
+    }
 }
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
@@ -103,11 +103,21 @@
     @Test
     void testGetLogger() {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            Logger failLogger = mock(Logger.class);
+            Logger failLogger = MockLogger.createMockLogger();
             FailsafeLogger failsafeLogger = new FailsafeLogger(failLogger);
 
-            doThrow(new RuntimeException()).when(failLogger).error(anyString());
             failsafeLogger.getLogger().error("should get error");
         });
     }
+
+    public final class MockLogger {
+        private MockLogger() {
+        }
+
+        public static org.apache.dubbo.common.logger.Logger createMockLogger() {
+            org.apache.dubbo.common.logger.Logger failLogger = org.mockito.Mockito.mock(org.apache.dubbo.common.logger.Logger.class);
+            org.mockito.Mockito.doThrow(new RuntimeException()).when(failLogger).error(org.mockito.ArgumentMatchers.anyString());
+            return failLogger;
+        }
+    }
 }
