--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.auth.filter;
 
+import org.mockito.Mock;
 import org.apache.dubbo.auth.Constants;
 import org.apache.dubbo.auth.exception.RpcAuthenticationException;
 import org.apache.dubbo.auth.utils.SignatureUtils;
@@ -40,11 +41,12 @@
 import static org.mockito.Mockito.when;
 
 class ProviderAuthFilterTest {
+    @Mock
+    private Invocation invocation;
     @Test
     void testAuthDisabled() {
         URL url = mock(URL.class);
         Invoker invoker = mock(Invoker.class);
-        Invocation invocation = mock(RpcInvocation.class);
         when(invoker.getUrl()).thenReturn(url);
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         providerAuthFilter.invoke(invoker, invocation);
@@ -60,7 +62,6 @@
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
         Invoker invoker = mock(Invoker.class);
-        Invocation invocation = mock(RpcInvocation.class);
         when(invoker.getUrl()).thenReturn(url);
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         providerAuthFilter.invoke(invoker, invocation);
