--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
@@ -16,6 +16,7 @@
  */
 package org.apache.dubbo.auth.filter;
 
+import org.mockito.Mock;
 import org.apache.dubbo.auth.Constants;
 import org.apache.dubbo.common.URL;
 import org.apache.dubbo.common.constants.CommonConstants;
@@ -34,12 +35,13 @@
 import static org.mockito.Mockito.when;
 
 class ConsumerSignFilterTest {
+    @Mock
+    private Invocation invocation;
 
     @Test
     void testAuthDisabled() {
         URL url = mock(URL.class);
         Invoker invoker = mock(Invoker.class);
-        Invocation invocation = mock(Invocation.class);
         when(invoker.getUrl()).thenReturn(url);
         ConsumerSignFilter consumerSignFilter = new ConsumerSignFilter(FrameworkModel.defaultModel());
         consumerSignFilter.invoke(invoker, invocation);
@@ -55,7 +57,6 @@
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
         Invoker invoker = mock(Invoker.class);
-        Invocation invocation = mock(Invocation.class);
         when(invoker.getUrl()).thenReturn(url);
         ConsumerSignFilter consumerSignFilter = new ConsumerSignFilter(FrameworkModel.defaultModel());
         consumerSignFilter.invoke(invoker, invocation);
