--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/ProtocolListenerWrapperTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/ProtocolListenerWrapperTest.java
@@ -47,10 +47,8 @@
                     }
                 };
 
-        Protocol protocolWithoutListener = mock(Protocol.class);
-        when(protocolWithoutListener.refer(DemoService.class, urlWithoutListener))
-                .thenReturn(invokerWithoutListener);
 
+        Protocol protocolWithoutListener = createMockProtocol(urlWithoutListener, invokerWithoutListener);
         ProtocolListenerWrapper protocolListenerWrapperWithoutListener =
                 new ProtocolListenerWrapper(protocolWithoutListener);
 
@@ -71,9 +69,8 @@
                     }
                 };
 
-        Protocol protocol = mock(Protocol.class);
-        when(protocol.refer(DemoService.class, urlWithListener)).thenReturn(invokerWithListener);
 
+        Protocol protocol = createMockProtocol(urlWithListener, invokerWithListener);
         ProtocolListenerWrapper protocolListenerWrapper = new ProtocolListenerWrapper(protocol);
 
         invoker = protocolListenerWrapper.refer(DemoService.class, urlWithListener);
@@ -94,10 +91,8 @@
                     }
                 };
 
-        Protocol protocolWithoutListener = mock(Protocol.class);
-        when(protocolWithoutListener.refer(DemoService.class, urlWithoutListener))
-                .thenReturn(invokerWithoutListener);
 
+        Protocol protocolWithoutListener = createMockProtocol(urlWithoutListener, invokerWithoutListener);
         ProtocolListenerWrapper protocolListenerWrapperWithoutListener =
                 new ProtocolListenerWrapper(protocolWithoutListener);
 
@@ -118,13 +113,18 @@
                     }
                 };
 
-        Protocol protocol = mock(Protocol.class);
-        when(protocol.refer(DemoService.class, urlWithListener)).thenReturn(invokerWithListener);
 
+        Protocol protocol = createMockProtocol(urlWithListener, invokerWithListener);
         ProtocolListenerWrapper protocolListenerWrapper = new ProtocolListenerWrapper(protocol);
 
         invoker = protocolListenerWrapper.refer(DemoService.class, urlWithListener);
         Assertions.assertTrue(invoker instanceof ListenerInvokerWrapper);
         Assertions.assertEquals(1, CountInvokerListener.getCounter());
     }
+
+    private static Protocol createMockProtocol(URL url, AbstractInvoker<DemoService> invoker) {
+        Protocol protocol = Mockito.mock(Protocol.class);
+        when(protocol.refer(DemoService.class, url)).thenReturn(invoker);
+        return protocol;
+    }
 }
