--- a/dubbo-metrics/dubbo-metrics-default/src/test/java/org/apache/dubbo/monitor/support/MonitorFilterTest.java
+++ b/dubbo-metrics/dubbo-metrics-default/src/test/java/org/apache/dubbo/monitor/support/MonitorFilterTest.java
@@ -198,12 +198,11 @@
     @Test
     void testSafeFailForMonitorCollectFail() {
         MonitorFilter monitorFilter = new MonitorFilter();
-        MonitorFactory mockMonitorFactory = mock(MonitorFactory.class);
         Monitor mockMonitor = mock(Monitor.class);
         Mockito.doThrow(new RuntimeException()).when(mockMonitor).collect(any(URL.class));
 
-        monitorFilter.setMonitorFactory(mockMonitorFactory);
-        given(mockMonitorFactory.getMonitor(any(URL.class))).willReturn(mockMonitor);
+        MonitorFactory mockMonitorFactory = createMockMonitorFactory(mockMonitor);
+        monitorFilter.setMonitorFactory(mockMonitorFactory);
         Invocation invocation =
                 new RpcInvocation("aaa", MonitorService.class.getName(), "", new Class<?>[0], new Object[0]);
 
@@ -213,10 +212,9 @@
     @Test
     void testOnResponseWithoutStartTime() {
         MonitorFilter monitorFilter = new MonitorFilter();
-        MonitorFactory mockMonitorFactory = mock(MonitorFactory.class);
         Monitor mockMonitor = mock(Monitor.class);
-        monitorFilter.setMonitorFactory(mockMonitorFactory);
-        given(mockMonitorFactory.getMonitor(any(URL.class))).willReturn(mockMonitor);
+        MonitorFactory mockMonitorFactory = createMockMonitorFactory(mockMonitor);
+        monitorFilter.setMonitorFactory(mockMonitorFactory);
         Invocation invocation =
                 new RpcInvocation("aaa", MonitorService.class.getName(), "", new Class<?>[0], new Object[0]);
 
@@ -229,10 +227,9 @@
     @Test
     void testOnErrorWithoutStartTime() {
         MonitorFilter monitorFilter = new MonitorFilter();
-        MonitorFactory mockMonitorFactory = mock(MonitorFactory.class);
         Monitor mockMonitor = mock(Monitor.class);
-        monitorFilter.setMonitorFactory(mockMonitorFactory);
-        given(mockMonitorFactory.getMonitor(any(URL.class))).willReturn(mockMonitor);
+        MonitorFactory mockMonitorFactory = createMockMonitorFactory(mockMonitor);
+        monitorFilter.setMonitorFactory(mockMonitorFactory);
         Invocation invocation =
                 new RpcInvocation("aaa", MonitorService.class.getName(), "", new Class<?>[0], new Object[0]);
 
@@ -245,4 +242,10 @@
         ApplicationModel applicationModel = ApplicationModel.defaultModel();
         applicationModel.destroy();
     }
+
+    private static MonitorFactory createMockMonitorFactory(Monitor mockMonitor) {
+        MonitorFactory mockMonitorFactory = mock(MonitorFactory.class);
+        given(mockMonitorFactory.getMonitor(any(URL.class))).willReturn(mockMonitor);
+        return mockMonitorFactory;
+    }
 }
