--- a/config/src/test/java/org/springframework/security/config/SecurityNamespaceHandlerTests.java
+++ b/config/src/test/java/org/springframework/security/config/SecurityNamespaceHandlerTests.java
@@ -54,6 +54,8 @@
  */
 @ExtendWith(MockitoExtension.class)
 public class SecurityNamespaceHandlerTests {
+	@Mock
+	private Appender<ILoggingEvent> appender;
 
 	// @formatter:off
 	private static final String XML_AUTHENTICATION_MANAGER = "<authentication-manager>"
@@ -74,7 +76,6 @@
 
 	@Test
 	public void constructionWhenVersionsMatchThenLogsNothing() {
-		Appender<ILoggingEvent> appender = mock(Appender.class);
 		Logger logger = (Logger) LoggerFactory.getLogger(SecurityNamespaceHandler.class);
 		logger.addAppender(appender);
 		assertThat(new SecurityNamespaceHandler()).isNotNull();
@@ -83,7 +84,6 @@
 
 	@Test
 	public void constructorWhenDetectsMismatchingVersionsThenLogsError() {
-		Appender<ILoggingEvent> appender = mock(Appender.class);
 		Logger logger = (Logger) LoggerFactory.getLogger(SecurityNamespaceHandler.class);
 		logger.addAppender(appender);
 		try (MockedStatic<SpringSecurityCoreVersion> core = Mockito.mockStatic(SpringSecurityCoreVersion.class)) {
