--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenEndpointFilterTests.java
@@ -558,10 +558,8 @@
 
 		MockHttpServletRequest request = createAuthorizationCodeTokenRequest(registeredClient);
 
-		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
-				AuthenticationDetailsSource.class);
 		WebAuthenticationDetails webAuthenticationDetails = new WebAuthenticationDetails(request);
-		given(authenticationDetailsSource.buildDetails(any())).willReturn(webAuthenticationDetails);
+		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = MockAuthenticationDetailsSource.createAuthenticationDetailsSource(webAuthenticationDetails);
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
 		OAuth2AccessToken accessToken = new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER, "token",
@@ -761,4 +759,18 @@
 		return request;
 	}
 
+
+	public final class MockAuthenticationDetailsSource {
+
+		private MockAuthenticationDetailsSource() {
+		}
+
+		public static AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> createAuthenticationDetailsSource(
+				WebAuthenticationDetails webAuthenticationDetails) {
+			AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource =
+					mock(AuthenticationDetailsSource.class);
+			given(authenticationDetailsSource.buildDetails(any())).willReturn(webAuthenticationDetails);
+			return authenticationDetailsSource;
+		}
+	}
 }
--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenRevocationEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenRevocationEndpointFilterTests.java
@@ -216,10 +216,8 @@
 
 		MockHttpServletRequest request = createTokenRevocationRequest();
 
-		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
-				AuthenticationDetailsSource.class);
 		WebAuthenticationDetails webAuthenticationDetails = new WebAuthenticationDetails(request);
-		given(authenticationDetailsSource.buildDetails(any())).willReturn(webAuthenticationDetails);
+		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = MockAuthenticationDetailsSource.createAuthenticationDetailsSource(webAuthenticationDetails);
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
 		OAuth2AccessToken accessToken = new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER, "token",
@@ -360,4 +358,18 @@
 		return request;
 	}
 
+
+	public final class MockAuthenticationDetailsSource {
+
+		private MockAuthenticationDetailsSource() {
+		}
+
+		public static AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> createAuthenticationDetailsSource(
+				WebAuthenticationDetails webAuthenticationDetails) {
+			AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource =
+					mock(AuthenticationDetailsSource.class);
+			given(authenticationDetailsSource.buildDetails(any())).willReturn(webAuthenticationDetails);
+			return authenticationDetailsSource;
+		}
+	}
 }
