--- a/web/src/test/java/org/springframework/security/web/authentication/DelegatingAuthenticationEntryPointTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/DelegatingAuthenticationEntryPointTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.authentication;
 
+import org.mockito.Mock;
 import java.io.IOException;
 import java.util.Collections;
 import java.util.List;
@@ -48,6 +49,8 @@
  * @version $Id:$
  */
 public class DelegatingAuthenticationEntryPointTests {
+	@Mock
+	private AuthenticationEntryPoint authenticationEntryPoint;
 
 	private DelegatingAuthenticationEntryPoint daep;
 
@@ -111,16 +114,13 @@
 	@SuppressWarnings("removal")
 	public void constructorAepListWhenNullEntryPoints() {
 		List<RequestMatcherEntry<AuthenticationEntryPoint>> entryPoints = null;
-		assertThatIllegalArgumentException().isThrownBy(
-				() -> new DelegatingAuthenticationEntryPoint(mock(AuthenticationEntryPoint.class), entryPoints));
+		assertThatIllegalArgumentException().isThrownBy(() -> new DelegatingAuthenticationEntryPoint(authenticationEntryPoint, entryPoints));
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	public void constructorAepListWhenEmptyEntryPoints() {
-		assertThatIllegalArgumentException()
-			.isThrownBy(() -> new DelegatingAuthenticationEntryPoint(mock(AuthenticationEntryPoint.class),
-					Collections.emptyList()));
+		assertThatIllegalArgumentException().isThrownBy(() -> new DelegatingAuthenticationEntryPoint(authenticationEntryPoint, Collections.emptyList()));
 	}
 
 	@Test
@@ -138,16 +138,14 @@
 	@SuppressWarnings("removal")
 	public void constructorAepVargsWhenNullEntryPoints() {
 		RequestMatcherEntry<AuthenticationEntryPoint>[] entryPoints = null;
-		assertThatIllegalArgumentException().isThrownBy(
-				() -> new DelegatingAuthenticationEntryPoint(mock(AuthenticationEntryPoint.class), entryPoints));
+		assertThatIllegalArgumentException().isThrownBy(() -> new DelegatingAuthenticationEntryPoint(authenticationEntryPoint, entryPoints));
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	public void constructorAepVargsWhenEmptyEntryPoints() {
 		RequestMatcherEntry<AuthenticationEntryPoint>[] entryPoints = new RequestMatcherEntry[0];
-		assertThatIllegalArgumentException().isThrownBy(
-				() -> new DelegatingAuthenticationEntryPoint(mock(AuthenticationEntryPoint.class), entryPoints));
+		assertThatIllegalArgumentException().isThrownBy(() -> new DelegatingAuthenticationEntryPoint(authenticationEntryPoint, entryPoints));
 	}
 
 	@Test
