--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilterTests.java
@@ -468,10 +468,8 @@
 		RegisteredClient registeredClient = TestRegisteredClients.registeredClient().build();
 		MockHttpServletRequest request = createAuthorizationRequest(registeredClient);
 
-		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
-				AuthenticationDetailsSource.class);
 		WebAuthenticationDetails webAuthenticationDetails = new WebAuthenticationDetails(request);
-		given(authenticationDetailsSource.buildDetails(request)).willReturn(webAuthenticationDetails);
+		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = MockAuthenticationDetailsSource.createMockAuthenticationDetailsSource(request, webAuthenticationDetails);
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
 		OAuth2AuthorizationCodeRequestAuthenticationToken authorizationCodeRequestAuthenticationResult = new OAuth2AuthorizationCodeRequestAuthenticationToken(
@@ -795,4 +793,18 @@
 				scope);
 	}
 
+
+	public final class MockAuthenticationDetailsSource {
+
+		private MockAuthenticationDetailsSource() {
+		}
+
+		public static AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> createMockAuthenticationDetailsSource(
+				HttpServletRequest request,
+				WebAuthenticationDetails webAuthenticationDetails) {
+			AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(AuthenticationDetailsSource.class);
+			given(authenticationDetailsSource.buildDetails(request)).willReturn(webAuthenticationDetails);
+			return authenticationDetailsSource;
+		}
+	}
 }
--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2PushedAuthorizationRequestEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2PushedAuthorizationRequestEndpointFilterTests.java
@@ -366,10 +366,8 @@
 		RegisteredClient registeredClient = TestRegisteredClients.registeredClient().build();
 		MockHttpServletRequest request = createPushedAuthorizationRequest(registeredClient);
 
-		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
-				AuthenticationDetailsSource.class);
 		WebAuthenticationDetails webAuthenticationDetails = new WebAuthenticationDetails(request);
-		given(authenticationDetailsSource.buildDetails(request)).willReturn(webAuthenticationDetails);
+		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = MockAuthenticationDetailsSource.createMockAuthenticationDetailsSource(request, webAuthenticationDetails);
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
 		OAuth2PushedAuthorizationRequestAuthenticationToken pushedAuthorizationRequestAuthenticationResult = new OAuth2PushedAuthorizationRequestAuthenticationToken(
@@ -489,4 +487,18 @@
 		return (Map<String, Object>) this.jsonMessageConverter.read(STRING_OBJECT_MAP.getType(), null, httpResponse);
 	}
 
+
+	public final class MockAuthenticationDetailsSource {
+
+		private MockAuthenticationDetailsSource() {
+		}
+
+		public static AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> createMockAuthenticationDetailsSource(
+				HttpServletRequest request,
+				WebAuthenticationDetails webAuthenticationDetails) {
+			AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(AuthenticationDetailsSource.class);
+			given(authenticationDetailsSource.buildDetails(request)).willReturn(webAuthenticationDetails);
+			return authenticationDetailsSource;
+		}
+	}
 }
