--- a/web/src/test/java/org/springframework/security/web/authentication/AbstractAuthenticationProcessingFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/AbstractAuthenticationProcessingFilterTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.authentication;
 
+import org.mockito.Mock;
 import java.util.ArrayList;
 
 import jakarta.servlet.FilterChain;
@@ -80,6 +81,8 @@
  */
 @SuppressWarnings("deprecation")
 public class AbstractAuthenticationProcessingFilterTests {
+	@Mock
+	private AuthenticationManager authenticationManager;
 
 	SavedRequestAwareAuthenticationSuccessHandler successHandler;
 
@@ -215,8 +218,7 @@
 		MockFilterChain chain = new MockFilterChain(false);
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		// Setup our test object, to grant access
-		MockAuthenticationFilter filter = new MockAuthenticationFilter("/j_mock_post",
-				mock(AuthenticationManager.class));
+		MockAuthenticationFilter filter = new MockAuthenticationFilter("/j_mock_post", authenticationManager);
 		filter.setSessionAuthenticationStrategy(mock(SessionAuthenticationStrategy.class));
 		filter.setAuthenticationSuccessHandler(this.successHandler);
 		filter.setAuthenticationFailureHandler(this.failureHandler);
@@ -246,8 +248,7 @@
 		MockFilterChain chain = new MockFilterChain(false);
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		// Setup our test object, to grant access
-		MockAuthenticationFilter filter = new MockAuthenticationFilter(pathPattern("/j_eradicate_corona_virus"),
-				mock(AuthenticationManager.class));
+		MockAuthenticationFilter filter = new MockAuthenticationFilter(pathPattern("/j_eradicate_corona_virus"), authenticationManager);
 		filter.setSessionAuthenticationStrategy(mock(SessionAuthenticationStrategy.class));
 		filter.setAuthenticationSuccessHandler(this.successHandler);
 		filter.setAuthenticationFailureHandler(this.failureHandler);
