--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2DeviceAuthorizationEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2DeviceAuthorizationEndpointFilterTests.java
@@ -352,11 +352,7 @@
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		FilterChain filterChain = mock(FilterChain.class);
 
-		@SuppressWarnings("unchecked")
-		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
-				AuthenticationDetailsSource.class);
-		given(authenticationDetailsSource.buildDetails(any(HttpServletRequest.class)))
-			.willReturn(new WebAuthenticationDetails(request));
+		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = MockAuthenticationDetailsSource.createMockAuthenticationDetailsSource(request);
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
 		this.filter.doFilter(request, response, filterChain);
@@ -467,4 +463,20 @@
 		return new OAuth2UserCode(USER_CODE, issuedAt, issuedAt.plus(30, ChronoUnit.MINUTES));
 	}
 
+
+	public final class MockAuthenticationDetailsSource {
+
+		private MockAuthenticationDetailsSource() {
+		}
+
+		@SuppressWarnings("unchecked")
+		public static AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> createMockAuthenticationDetailsSource(
+				HttpServletRequest request) {
+			AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource =
+					mock(AuthenticationDetailsSource.class);
+			given(authenticationDetailsSource.buildDetails(any(HttpServletRequest.class)))
+					.willReturn(new WebAuthenticationDetails(request));
+			return authenticationDetailsSource;
+		}
+	}
 }
--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2DeviceVerificationEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2DeviceVerificationEndpointFilterTests.java
@@ -346,11 +346,7 @@
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		FilterChain filterChain = mock(FilterChain.class);
 
-		@SuppressWarnings("unchecked")
-		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
-				AuthenticationDetailsSource.class);
-		given(authenticationDetailsSource.buildDetails(any(HttpServletRequest.class)))
-			.willReturn(new WebAuthenticationDetails(request));
+		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = MockAuthenticationDetailsSource.createMockAuthenticationDetailsSource(request);
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
 		this.filter.doFilter(request, response, filterChain);
@@ -475,4 +471,20 @@
 				scope);
 	}
 
+
+	public final class MockAuthenticationDetailsSource {
+
+		private MockAuthenticationDetailsSource() {
+		}
+
+		@SuppressWarnings("unchecked")
+		public static AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> createMockAuthenticationDetailsSource(
+				HttpServletRequest request) {
+			AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource =
+					mock(AuthenticationDetailsSource.class);
+			given(authenticationDetailsSource.buildDetails(any(HttpServletRequest.class)))
+					.willReturn(new WebAuthenticationDetails(request));
+			return authenticationDetailsSource;
+		}
+	}
 }
