--- a/config/src/test/java/org/springframework/security/config/http/OAuth2ResourceServerBeanDefinitionParserTests.java
+++ b/config/src/test/java/org/springframework/security/config/http/OAuth2ResourceServerBeanDefinitionParserTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.config.http;
 
+import org.mockito.Mock;
 import java.io.BufferedReader;
 import java.io.FileReader;
 import java.io.IOException;
@@ -127,6 +128,8 @@
 @ExtendWith({ SpringExtension.class, SpringTestContextExtension.class })
 @SecurityTestExecutionListeners
 public class OAuth2ResourceServerBeanDefinitionParserTests {
+	@Mock
+	private BeanDefinitionParserDelegate beanDefinitionParserDelegate;
 
 	private static final String CONFIG_LOCATION_PREFIX = "classpath:org/springframework/security/config/http/OAuth2ResourceServerBeanDefinitionParserTests";
 
@@ -860,7 +863,7 @@
 		given(element.hasAttribute(OAuth2ResourceServerBeanDefinitionParser.AUTHENTICATION_MANAGER_RESOLVER_REF))
 			.willReturn(true);
 		Element child = mock(Element.class);
-		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), mock(BeanDefinitionParserDelegate.class));
+		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), beanDefinitionParserDelegate);
 		parser.validateConfiguration(element, child, null, pc);
 		verify(pc.getReaderContext()).error(anyString(), eq(element));
 		reset(pc.getReaderContext());
@@ -875,7 +878,7 @@
 		Element element = mock(Element.class);
 		given(element.hasAttribute(OAuth2ResourceServerBeanDefinitionParser.AUTHENTICATION_MANAGER_RESOLVER_REF))
 			.willReturn(false);
-		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), mock(BeanDefinitionParserDelegate.class));
+		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), beanDefinitionParserDelegate);
 		parser.validateConfiguration(element, null, null, pc);
 		verify(pc.getReaderContext()).error(anyString(), eq(element));
 	}
@@ -886,7 +889,7 @@
 		Element element = mock(Element.class);
 		given(element.hasAttribute(JwtBeanDefinitionParser.JWK_SET_URI)).willReturn(true);
 		given(element.hasAttribute(JwtBeanDefinitionParser.DECODER_REF)).willReturn(true);
-		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), mock(BeanDefinitionParserDelegate.class));
+		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), beanDefinitionParserDelegate);
 		parser.validateConfiguration(element, pc);
 		verify(pc.getReaderContext()).error(anyString(), eq(element));
 	}
@@ -897,7 +900,7 @@
 		Element element = mock(Element.class);
 		given(element.hasAttribute(JwtBeanDefinitionParser.JWK_SET_URI)).willReturn(false);
 		given(element.hasAttribute(JwtBeanDefinitionParser.DECODER_REF)).willReturn(false);
-		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), mock(BeanDefinitionParserDelegate.class));
+		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), beanDefinitionParserDelegate);
 		parser.validateConfiguration(element, pc);
 		verify(pc.getReaderContext()).error(anyString(), eq(element));
 	}
@@ -908,7 +911,7 @@
 		Element element = mock(Element.class);
 		given(element.hasAttribute(OpaqueTokenBeanDefinitionParser.INTROSPECTION_URI)).willReturn(true);
 		given(element.hasAttribute(OpaqueTokenBeanDefinitionParser.INTROSPECTOR_REF)).willReturn(true);
-		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), mock(BeanDefinitionParserDelegate.class));
+		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), beanDefinitionParserDelegate);
 		parser.validateConfiguration(element, pc);
 		verify(pc.getReaderContext()).error(anyString(), eq(element));
 	}
@@ -919,7 +922,7 @@
 		Element element = mock(Element.class);
 		given(element.hasAttribute(OpaqueTokenBeanDefinitionParser.INTROSPECTION_URI)).willReturn(false);
 		given(element.hasAttribute(OpaqueTokenBeanDefinitionParser.INTROSPECTOR_REF)).willReturn(false);
-		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), mock(BeanDefinitionParserDelegate.class));
+		ParserContext pc = new ParserContext(mock(XmlReaderContext.class), beanDefinitionParserDelegate);
 		parser.validateConfiguration(element, pc);
 		verify(pc.getReaderContext()).error(anyString(), eq(element));
 	}
