--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientClientCredentialsTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientClientCredentialsTokenResponseClientTests.java
@@ -347,10 +347,9 @@
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		OAuth2ClientCredentialsGrantRequest grantRequest = new OAuth2ClientCredentialsGrantRequest(clientRegistration);
-		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> headersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.put("custom-header-name", Collections.singletonList("custom-header-value"));
-		given(headersConverter.convert(grantRequest)).willReturn(headers);
+		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> headersConverter = MockOAuth2ClientCredentialsGrantRequestConverter.createMockOAuth2ClientCredentialsGrantRequestConverter(grantRequest, headers);
 		this.tokenResponseClient.addHeadersConverter(headersConverter);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		verify(headersConverter).convert(grantRequest);
@@ -364,11 +363,10 @@
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		OAuth2ClientCredentialsGrantRequest grantRequest = new OAuth2ClientCredentialsGrantRequest(clientRegistration);
-		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> headersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
 		headers.put("custom-header-name", Collections.singletonList("custom-header-value"));
-		given(headersConverter.convert(grantRequest)).willReturn(headers);
+		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> headersConverter = createMockOAuth2ClientCredentialsGrantRequestConverter(grantRequest, headers);
 		this.tokenResponseClient.setHeadersConverter(headersConverter);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		verify(headersConverter).convert(grantRequest);
@@ -472,4 +470,18 @@
 		return "%s=%s".formatted(parameterName, URLEncoder.encode(parameterValue, StandardCharsets.UTF_8));
 	}
 
+
+	public final class MockOAuth2ClientCredentialsGrantRequestConverter {
+
+		private MockOAuth2ClientCredentialsGrantRequestConverter() {
+		}
+
+		public static Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> createMockOAuth2ClientCredentialsGrantRequestConverter(
+				OAuth2ClientCredentialsGrantRequest request,
+				HttpHeaders headers) {
+			Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> converter = mock();
+			given(converter.convert(request)).willReturn(headers);
+			return converter;
+		}
+	}
 }
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveClientCredentialsTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveClientCredentialsTokenResponseClientTests.java
@@ -267,10 +267,9 @@
 	public void convertWhenHeadersConverterAddedThenCalled() throws Exception {
 		OAuth2ClientCredentialsGrantRequest request = new OAuth2ClientCredentialsGrantRequest(
 				this.clientRegistration.build());
-		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> addedHeadersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.put("custom-header-name", Collections.singletonList("custom-header-value"));
-		given(addedHeadersConverter.convert(request)).willReturn(headers);
+		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> addedHeadersConverter = MockOAuth2ClientCredentialsGrantRequestConverter.createMockOAuth2ClientCredentialsGrantRequestConverter(request, headers);
 		this.client.addHeadersConverter(addedHeadersConverter);
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		this.client.getTokenResponse(request).block();
@@ -287,10 +286,9 @@
 		OAuth2ClientCredentialsGrantRequest request = new OAuth2ClientCredentialsGrantRequest(
 				this.clientRegistration.build());
 		ClientRegistration clientRegistration = request.getClientRegistration();
-		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> headersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.setBasicAuth(clientRegistration.getClientId(), clientRegistration.getClientSecret());
-		given(headersConverter.convert(request)).willReturn(headers);
+		Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> headersConverter = MockOAuth2ClientCredentialsGrantConverter.createMockOAuth2ClientCredentialsGrantConverter(request, headers);
 		this.client.setHeadersConverter(headersConverter);
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		this.client.getTokenResponse(request).block();
@@ -427,4 +425,18 @@
 		return "%s=%s".formatted(parameterName, URLEncoder.encode(parameterValue, StandardCharsets.UTF_8));
 	}
 
+
+	public final class MockOAuth2ClientCredentialsGrantRequestConverter {
+
+		private MockOAuth2ClientCredentialsGrantRequestConverter() {
+		}
+
+		public static Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> createMockOAuth2ClientCredentialsGrantRequestConverter(
+				OAuth2ClientCredentialsGrantRequest request,
+				HttpHeaders headers) {
+			Converter<OAuth2ClientCredentialsGrantRequest, HttpHeaders> converter = mock();
+			given(converter.convert(request)).willReturn(headers);
+			return converter;
+		}
+	}
 }
