--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenRevocationEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenRevocationEndpointFilterTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.oauth2.server.authorization.web;
 
+import org.mockito.Mock;
 import java.time.Duration;
 import java.time.Instant;
 import java.util.Arrays;
@@ -71,6 +72,8 @@
  * @author Joe Grandja
  */
 public class OAuth2TokenRevocationEndpointFilterTests {
+	@Mock
+	private FilterChain filterChain;
 
 	private static final String DEFAULT_TOKEN_REVOCATION_ENDPOINT_URI = "/oauth2/revoke";
 
@@ -139,7 +142,6 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", requestUri);
 		request.setServletPath(requestUri);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
@@ -152,7 +154,6 @@
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", requestUri);
 		request.setServletPath(requestUri);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
@@ -198,7 +199,6 @@
 
 		MockHttpServletRequest request = createTokenRevocationRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
@@ -235,7 +235,6 @@
 		SecurityContextHolder.setContext(securityContext);
 
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
@@ -265,7 +264,6 @@
 
 		MockHttpServletRequest request = createTokenRevocationRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
@@ -294,7 +292,6 @@
 
 		MockHttpServletRequest request = createTokenRevocationRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
@@ -318,7 +315,6 @@
 
 		MockHttpServletRequest request = createTokenRevocationRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		this.filter.doFilter(request, response, filterChain);
 
