--- a/saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/web/authentication/Saml2WebSsoAuthenticationFilterTests.java
+++ b/saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/web/authentication/Saml2WebSsoAuthenticationFilterTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.saml2.provider.service.web.authentication;
 
+import org.mockito.Mock;
 import jakarta.servlet.FilterChain;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
@@ -56,6 +57,8 @@
 import static org.springframework.security.web.servlet.util.matcher.PathPatternRequestMatcher.pathPattern;
 
 public class Saml2WebSsoAuthenticationFilterTests {
+	@Mock
+	private FilterChain chain;
 
 	private Saml2WebSsoAuthenticationFilter filter;
 
@@ -129,7 +132,6 @@
 		this.filter.setContinueChainWhenNoRelyingPartyRegistrationFound(true);
 		this.request.setRequestURI("/some/other/path/non-existent-id");
 		this.request.setPathInfo("/some/other/path/non-existent-id");
-		FilterChain chain = mock(FilterChain.class);
 		this.filter.doFilter(this.request, this.response, chain);
 		verify(chain).doFilter(this.request, this.response);
 	}
@@ -142,7 +144,6 @@
 		this.request.setRequestURI("/some/other/path/id");
 		this.request.setPathInfo("/some/other/path/id");
 		this.request.removeParameter(Saml2ParameterNames.SAML_RESPONSE);
-		FilterChain chain = mock(FilterChain.class);
 		this.filter.doFilter(this.request, this.response, chain);
 		verify(chain).doFilter(this.request, this.response);
 	}
