--- a/web/src/test/java/org/springframework/security/web/RequestMatcherRedirectFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/RequestMatcherRedirectFilterTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web;
 
+import org.mockito.Mock;
 import jakarta.servlet.FilterChain;
 import org.junit.jupiter.api.Test;
 
@@ -37,6 +38,8 @@
  * @author Evgeniy Cheban
  */
 public class RequestMatcherRedirectFilterTests {
+	@Mock
+	private FilterChain filterChain;
 
 	@Test
 	public void doFilterWhenRequestMatchThenRedirectToSpecifiedUrl() throws Exception {
@@ -44,7 +47,6 @@
 
 		MockHttpServletRequest request = get("/context").build();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		filter.doFilter(request, response, filterChain);
 
@@ -61,7 +63,6 @@
 		MockHttpServletRequest request = get("/test").build();
 
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		filter.doFilter(request, response, filterChain);
 
