--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/authentication/WebAuthnAuthenticationFilterTests.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/authentication/WebAuthnAuthenticationFilterTests.java
@@ -63,6 +63,8 @@
  */
 @ExtendWith(MockitoExtension.class)
 class WebAuthnAuthenticationFilterTests {
+	@Mock
+	private HttpServletResponse response;
 
 	private static final String VALID_BODY = """
 				{
@@ -116,7 +118,6 @@
 
 	@Test
 	void doFilterWhenUrlDoesNotMatchThenChainContinues() throws Exception {
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		this.filter.setConverter(this.converter);
 		this.filter.setRequestOptionsRepository(this.requestOptionsRepository);
 		this.filter.doFilter(post("/nomatch").buildRequest(new MockServletContext()), response, this.chain);
@@ -126,7 +127,6 @@
 
 	@Test
 	void doFilterWhenMethodDoesNotMatchThenChainContinues() throws Exception {
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		this.filter.setConverter(this.converter);
 		this.filter.setRequestOptionsRepository(this.requestOptionsRepository);
 		this.filter.doFilter(get("/login/webauthn").buildRequest(new MockServletContext()), response, this.chain);
