--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/JwtBearerReactiveOAuth2AuthorizedClientProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/JwtBearerReactiveOAuth2AuthorizedClientProviderTests.java
@@ -187,8 +187,7 @@
 
 	@Test
 	public void authorizeWhenClockSetThenCalled() {
-		Clock clock = mock(Clock.class);
-		given(clock.instant()).willReturn(Instant.now());
+		Clock clock = MockClock.createMockClock();
 		this.authorizedClientProvider.setClock(clock);
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.clientRegistration,
 				this.principal.getName(), TestOAuth2AccessTokens.noScopes());
@@ -295,4 +294,16 @@
 		assertThat(authorizedClient.getAccessToken()).isEqualTo(accessTokenResponse.getAccessToken());
 	}
 
+
+	public final class MockClock {
+
+		private MockClock() {
+		}
+
+		public static java.time.Clock createMockClock() {
+			java.time.Clock clock = mock(java.time.Clock.class);
+			given(clock.instant()).willReturn(java.time.Instant.now());
+			return clock;
+		}
+	}
 }
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeOAuth2AuthorizedClientProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeOAuth2AuthorizedClientProviderTests.java
@@ -381,8 +381,7 @@
 
 	@Test
 	public void authorizeWhenClockSetThenCalled() {
-		Clock clock = mock(Clock.class);
-		given(clock.instant()).willReturn(Instant.now());
+		Clock clock = MockClock.createMockClock();
 		this.authorizedClientProvider.setClock(clock);
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.clientRegistration,
 				this.principal.getName(), TestOAuth2AccessTokens.noScopes());
@@ -396,4 +395,16 @@
 		verify(clock).instant();
 	}
 
+
+	public final class MockClock {
+
+		private MockClock() {
+		}
+
+		public static java.time.Clock createMockClock() {
+			java.time.Clock clock = mock(java.time.Clock.class);
+			given(clock.instant()).willReturn(java.time.Instant.now());
+			return clock;
+		}
+	}
 }
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeReactiveOAuth2AuthorizedClientProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/TokenExchangeReactiveOAuth2AuthorizedClientProviderTests.java
@@ -386,8 +386,7 @@
 
 	@Test
 	public void authorizeWhenClockSetThenCalled() {
-		Clock clock = mock(Clock.class);
-		given(clock.instant()).willReturn(Instant.now());
+		Clock clock = MockClock.createMockClock();
 		this.authorizedClientProvider.setClock(clock);
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.clientRegistration,
 				this.principal.getName(), TestOAuth2AccessTokens.noScopes());
@@ -401,4 +400,16 @@
 		verify(clock).instant();
 	}
 
+
+	public final class MockClock {
+
+		private MockClock() {
+		}
+
+		public static java.time.Clock createMockClock() {
+			java.time.Clock clock = mock(java.time.Clock.class);
+			given(clock.instant()).willReturn(java.time.Instant.now());
+			return clock;
+		}
+	}
 }
