--- a/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationProviderTests.java
+++ b/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationProviderTests.java
@@ -49,6 +49,8 @@
  */
 @ExtendWith(MockitoExtension.class)
 public class JwtAuthenticationProviderTests {
+	@Mock
+	private Object details;
 
 	@Mock
 	Converter<Jwt, AbstractAuthenticationToken> jwtAuthenticationConverter;
@@ -122,7 +124,6 @@
 	@Test
 	public void authenticateWhenConverterDoesNotSetAuthenticationDetailsThenProviderSetsItWithTokenDetails() {
 		BearerTokenAuthenticationToken token = this.authentication();
-		Object details = mock(Object.class);
 		token.setDetails(details);
 		Jwt jwt = TestJwts.jwt().build();
 		JwtAuthenticationToken authentication = new JwtAuthenticationToken(jwt);
@@ -138,7 +139,7 @@
 	@Test
 	public void authenticateWhenConverterSetsAuthenticationDetailsThenProviderDoesNotOverwriteIt() {
 		BearerTokenAuthenticationToken token = this.authentication();
-		Object details = mock(Object.class);
+		// removed local mock; replaced with global field `details`
 		token.setDetails(details);
 		Jwt jwt = TestJwts.jwt().build();
 		JwtAuthenticationToken authentication = new JwtAuthenticationToken(jwt);
