--- a/config/src/test/java/org/springframework/security/config/http/MiscHttpConfigTests.java
+++ b/config/src/test/java/org/springframework/security/config/http/MiscHttpConfigTests.java
@@ -705,8 +705,7 @@
 	@Test
 	public void loginWhenJeeFilterThenExtractsRoles() throws Exception {
 		this.spring.configLocations(xml("JeeFilter")).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("joe");
+		Principal user = createMockPrincipal();
 		// @formatter:off
 		MockHttpServletRequestBuilder rolesRequest = get("/roles")
 				.principal(user)
@@ -724,8 +723,7 @@
 	@Test
 	public void loginWhenJeeFilterCustomSecurityContextHolderStrategyThenUses() throws Exception {
 		this.spring.configLocations(xml("JeeFilterWithSecurityContextHolderStrategy")).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("joe");
+		Principal user = createMockPrincipal();
 		// @formatter:off
 		MockHttpServletRequestBuilder rolesRequest = get("/roles")
 				.principal(user)
@@ -1074,4 +1072,10 @@
 
 	}
 
+
+	private static Principal createMockPrincipal() {
+		Principal user = mock(Principal.class);
+		given(user.getName()).willReturn("joe");
+		return user;
+	}
 }
