--- a/core/src/test/java/org/springframework/security/authorization/method/AuthorizationManagerAfterMethodInterceptorTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/method/AuthorizationManagerAfterMethodInterceptorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.authorization.method;
 
+import org.mockito.Mock;
 import java.util.function.Supplier;
 
 import org.aopalliance.intercept.MethodInvocation;
@@ -52,6 +53,8 @@
  * @author Gengwu Zhao
  */
 public class AuthorizationManagerAfterMethodInterceptorTests {
+	@Mock
+	private MethodInvocation invocation;
 
 	@Test
 	public void instantiateWhenMethodMatcherNullThenException() {
@@ -86,7 +89,6 @@
 		Authentication authentication = TestAuthentication.authenticatedUser();
 		SecurityContextHolderStrategy strategy = mockSecurityContextHolderStrategy(
 				new SecurityContextImpl(authentication));
-		MethodInvocation invocation = mock(MethodInvocation.class);
 		AuthorizationManager<MethodInvocationResult> authorizationManager = AuthenticatedAuthorizationManager
 			.authenticated();
 		AuthorizationManagerAfterMethodInterceptor advice = new AuthorizationManagerAfterMethodInterceptor(
@@ -103,7 +105,6 @@
 				AuthorityUtils.createAuthorityList("authority"));
 		SecurityContextHolderStrategy strategy = mockSecurityContextHolderStrategy(
 				new SecurityContextImpl(authentication));
-		MethodInvocation invocation = mock(MethodInvocation.class);
 		AuthorizationManager<MethodInvocationResult> authorizationManager = AuthenticatedAuthorizationManager
 			.authenticated();
 		AuthorizationManagerAfterMethodInterceptor advice = new AuthorizationManagerAfterMethodInterceptor(
