--- a/web/src/test/java/org/springframework/security/web/server/ObservationWebFilterChainDecoratorTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/ObservationWebFilterChainDecoratorTests.java
@@ -222,8 +222,7 @@
 	@MethodSource("decorateFiltersWhenCompletesThenHasSpringSecurityReachedFilterNameTagArguments")
 	void decorateFiltersWhenCompletesThenHasSpringSecurityReachedFilterNameTag(WebFilter filter,
 			String expectedFilterNameTag) {
-		ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<Observation.Context> handler = MockObservationHandler.createObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationWebFilterChainDecorator decorator = new ObservationWebFilterChainDecorator(registry);
@@ -387,4 +386,16 @@
 
 	}
 
+
+	public final class MockObservationHandler {
+
+		private MockObservationHandler() {
+		}
+
+		public static ObservationHandler<Observation.Context> createObservationHandler() {
+			ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
+			given(handler.supportsContext(any())).willReturn(true);
+			return handler;
+		}
+	}
 }
--- a/web/src/test/java/org/springframework/security/web/server/WebFilterChainProxyTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/WebFilterChainProxyTests.java
@@ -78,8 +78,7 @@
 
 	@Test
 	public void doFilterWhenMatchesThenObservationRegistryObserves() {
-		ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<Observation.Context> handler = MockObservationHandler.createObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		List<WebFilter> filters = Arrays.asList(new PassthroughWebFilter());
@@ -102,8 +101,7 @@
 
 	@Test
 	public void doFilterWhenMismatchesThenObservationRegistryObserves() {
-		ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<Observation.Context> handler = MockObservationHandler.createObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		List<WebFilter> filters = Arrays.asList(new PassthroughWebFilter());
@@ -124,8 +122,7 @@
 
 	@Test
 	public void doFilterWhenFilterExceptionThenObservationRegistryObserves() {
-		ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<Observation.Context> handler = MockObservationHandler.createObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		WebFilter error = mock(WebFilter.class);
@@ -245,4 +242,16 @@
 
 	}
 
+
+	public final class MockObservationHandler {
+
+		private MockObservationHandler() {
+		}
+
+		public static ObservationHandler<Observation.Context> createObservationHandler() {
+			ObservationHandler<Observation.Context> handler = mock(ObservationHandler.class);
+			given(handler.supportsContext(any())).willReturn(true);
+			return handler;
+		}
+	}
 }
