--- a/web/src/test/java/org/springframework/security/web/server/ObservationWebFilterChainDecoratorTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/ObservationWebFilterChainDecoratorTests.java
@@ -52,8 +52,7 @@
 
 	@Test
 	void decorateWhenDefaultsThenObserves() {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationWebFilterChainDecorator decorator = new ObservationWebFilterChainDecorator(registry);
@@ -66,8 +65,7 @@
 
 	@Test
 	void decorateWhenNoopThenDoesNotObserve() {
-		ObservationHandler<?> handler = mock(ObservationHandler.class);
-		given(handler.supportsContext(any())).willReturn(true);
+		ObservationHandler<?> handler = createObservationHandler();
 		ObservationRegistry registry = ObservationRegistry.NOOP;
 		registry.observationConfig().observationHandler(handler);
 		ObservationWebFilterChainDecorator decorator = new ObservationWebFilterChainDecorator(registry);
@@ -387,4 +385,10 @@
 
 	}
 
+
+	private static ObservationHandler<?> createObservationHandler() {
+		ObservationHandler<?> handler = mock(ObservationHandler.class);
+		given(handler.supportsContext(any())).willReturn(true);
+		return handler;
+	}
 }
