--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
@@ -126,8 +126,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -165,8 +164,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -201,8 +199,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -237,8 +234,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -273,8 +269,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -309,8 +304,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -347,8 +341,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -385,8 +378,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -424,8 +416,7 @@
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
 		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
-		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
-		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = createReactiveOAuth2UserService(principal);
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
 		RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler handler = new RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandler();
 		handler.setJwtDecoderFactory(reactiveJwtDecoderFactory);
@@ -459,4 +450,10 @@
 		return new DefaultOidcUser(null, idToken);
 	}
 
+
+	private static ReactiveOAuth2UserService<OidcUserRequest, OidcUser> createReactiveOAuth2UserService(DefaultOidcUser principal) {
+		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
+		given(userService.loadUser(any())).willReturn(Mono.just(principal));
+		return userService;
+	}
 }
