--- a/cas/src/test/java/org/springframework/security/cas/web/CasAuthenticationFilterTests.java
+++ b/cas/src/test/java/org/springframework/security/cas/web/CasAuthenticationFilterTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.cas.web;
 
+import org.mockito.Mock;
 import java.io.IOException;
 
 import jakarta.servlet.FilterChain;
@@ -65,6 +66,8 @@
  * @author Rob Winch
  */
 public class CasAuthenticationFilterTests {
+	@Mock
+	private SecurityContextRepository securityContextRepository;
 
 	@AfterEach
 	public void tearDown() {
@@ -213,7 +216,6 @@
 		serviceProperties.setAuthenticateAllArtifacts(true);
 		filter.setServiceProperties(serviceProperties);
 
-		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
 		ReflectionTestUtils.setField(filter, "securityContextRepository", securityContextRepository);
 
 		filter.successfulAuthentication(request, response, new MockFilterChain(), mock(Authentication.class));
@@ -241,7 +243,6 @@
 
 	@Test
 	void successfulAuthenticationWhenSecurityContextRepositorySetThenUses() throws ServletException, IOException {
-		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
 		CasAuthenticationFilter filter = new CasAuthenticationFilter();
 		filter.setSecurityContextRepository(securityContextRepository);
 		filter.successfulAuthentication(new MockHttpServletRequest(), new MockHttpServletResponse(),
