--- a/acl/src/test/java/org/springframework/security/acls/AclPermissionEvaluatorTests.java
+++ b/acl/src/test/java/org/springframework/security/acls/AclPermissionEvaluatorTests.java
@@ -49,9 +49,8 @@
 		given(oidStrategy.getObjectIdentity(any(Object.class))).willReturn(oid);
 		pe.setObjectIdentityRetrievalStrategy(oidStrategy);
 		pe.setSidRetrievalStrategy(mock(SidRetrievalStrategy.class));
-		Acl acl = mock(Acl.class);
+		Acl acl = createAcl();
 		given(service.readAclById(any(ObjectIdentity.class), anyList())).willReturn(acl);
-		given(acl.isGranted(anyList(), anyList(), eq(false))).willReturn(true);
 		assertThat(pe.hasPermission(mock(Authentication.class), new Object(), "READ")).isTrue();
 	}
 
@@ -66,11 +65,16 @@
 		given(oidStrategy.getObjectIdentity(any(Object.class))).willReturn(oid);
 		pe.setObjectIdentityRetrievalStrategy(oidStrategy);
 		pe.setSidRetrievalStrategy(mock(SidRetrievalStrategy.class));
-		Acl acl = mock(Acl.class);
+		Acl acl = createAcl();
 		given(service.readAclById(any(ObjectIdentity.class), anyList())).willReturn(acl);
-		given(acl.isGranted(anyList(), anyList(), eq(false))).willReturn(true);
 		assertThat(pe.hasPermission(mock(Authentication.class), new Object(), "write")).isTrue();
 		Locale.setDefault(systemLocale);
 	}
 
+
+	private static Acl createAcl() {
+		Acl acl = mock(Acl.class);
+		given(acl.isGranted(anyList(), anyList(), eq(false))).willReturn(true);
+		return acl;
+	}
 }
