--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/WebAuthnConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/WebAuthnConfigurerTests.java
@@ -211,9 +211,8 @@
 		PublicKeyCredentialCreationOptions options = TestPublicKeyCredentialCreationOptions
 			.createPublicKeyCredentialCreationOptions()
 			.build();
-		WebAuthnRelyingPartyOperations rpOperations = mock(WebAuthnRelyingPartyOperations.class);
+		WebAuthnRelyingPartyOperations rpOperations = createMockWebAuthnRelyingPartyOperations(options);
 		ConfigCredentialCreationOptionsRepository.rpOperations = rpOperations;
-		given(rpOperations.createPublicKeyCredentialCreationOptions(any())).willReturn(options);
 		String attrName = "attrName";
 		HttpSessionPublicKeyCredentialCreationOptionsRepository creationOptionsRepository = new HttpSessionPublicKeyCredentialCreationOptionsRepository();
 		creationOptionsRepository.setAttrName(attrName);
@@ -231,9 +230,8 @@
 		PublicKeyCredentialCreationOptions options = TestPublicKeyCredentialCreationOptions
 			.createPublicKeyCredentialCreationOptions()
 			.build();
-		WebAuthnRelyingPartyOperations rpOperations = mock(WebAuthnRelyingPartyOperations.class);
+		WebAuthnRelyingPartyOperations rpOperations = createMockWebAuthnRelyingPartyOperations(options);
 		ConfigCredentialCreationOptionsRepositoryFromBean.rpOperations = rpOperations;
-		given(rpOperations.createPublicKeyCredentialCreationOptions(any())).willReturn(options);
 		String attrName = "attrName";
 		HttpSessionPublicKeyCredentialCreationOptionsRepository creationOptionsRepository = new HttpSessionPublicKeyCredentialCreationOptionsRepository();
 		creationOptionsRepository.setAttrName(attrName);
@@ -251,9 +249,8 @@
 		PublicKeyCredentialCreationOptions options = TestPublicKeyCredentialCreationOptions
 			.createPublicKeyCredentialCreationOptions()
 			.build();
-		WebAuthnRelyingPartyOperations rpOperations = mock(WebAuthnRelyingPartyOperations.class);
+		WebAuthnRelyingPartyOperations rpOperations = createMockWebAuthnRelyingPartyOperations(options);
 		ConfigMessageConverter.rpOperations = rpOperations;
-		given(rpOperations.createPublicKeyCredentialCreationOptions(any())).willReturn(options);
 		HttpMessageConverter<Object> converter = mock(HttpMessageConverter.class);
 		given(converter.canWrite(any(), any())).willReturn(true);
 		String expectedBody = "123";
@@ -607,4 +604,10 @@
 
 	}
 
+
+	private static WebAuthnRelyingPartyOperations createMockWebAuthnRelyingPartyOperations(PublicKeyCredentialCreationOptions options) {
+		WebAuthnRelyingPartyOperations rpOperations = mock(WebAuthnRelyingPartyOperations.class);
+		given(rpOperations.createPublicKeyCredentialCreationOptions(any())).willReturn(options);
+		return rpOperations;
+	}
 }
