--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/server/AuthenticatedPrincipalServerOAuth2AuthorizedClientRepositoryTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/server/AuthenticatedPrincipalServerOAuth2AuthorizedClientRepositoryTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.oauth2.client.web.server;
 
+import org.mockito.Mock;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import reactor.core.publisher.Mono;
@@ -40,6 +41,8 @@
  * @author Rob Winch
  */
 public class AuthenticatedPrincipalServerOAuth2AuthorizedClientRepositoryTests {
+	@Mock
+	private OAuth2AuthorizedClient authorizedClient;
 
 	private String registrationId = "registrationId";
 
@@ -99,7 +102,6 @@
 	public void saveAuthorizedClientWhenAuthenticatedPrincipalThenSaveToService() {
 		given(this.authorizedClientService.saveAuthorizedClient(any(), any())).willReturn(Mono.empty());
 		Authentication authentication = this.createAuthenticatedPrincipal();
-		OAuth2AuthorizedClient authorizedClient = mock(OAuth2AuthorizedClient.class);
 		this.authorizedClientRepository.saveAuthorizedClient(authorizedClient, authentication, this.exchange).block();
 		verify(this.authorizedClientService).saveAuthorizedClient(authorizedClient, authentication);
 	}
@@ -109,7 +111,6 @@
 		given(this.anonymousAuthorizedClientRepository.saveAuthorizedClient(any(), any(), any()))
 			.willReturn(Mono.empty());
 		Authentication authentication = this.createAnonymousPrincipal();
-		OAuth2AuthorizedClient authorizedClient = mock(OAuth2AuthorizedClient.class);
 		this.authorizedClientRepository.saveAuthorizedClient(authorizedClient, authentication, this.exchange).block();
 		verify(this.anonymousAuthorizedClientRepository).saveAuthorizedClient(authorizedClient, authentication,
 				this.exchange);
