--- a/core/src/test/java/org/springframework/security/access/expression/SecurityExpressionRootTests.java
+++ b/core/src/test/java/org/springframework/security/access/expression/SecurityExpressionRootTests.java
@@ -164,16 +164,14 @@
 
 	@Test
 	void isAuthenticatedWhenTrustResolverFalseThenFalse() {
-		AuthenticationTrustResolver atr = mock(AuthenticationTrustResolver.class);
-		given(atr.isAuthenticated(JOE)).willReturn(false);
+		AuthenticationTrustResolver atr = createAuthenticationTrustResolver(false);
 		this.root.setTrustResolver(atr);
 		assertThat(this.root.isAuthenticated()).isFalse();
 	}
 
 	@Test
 	void isAuthenticatedWhenTrustResolverTrueThenTrue() {
-		AuthenticationTrustResolver atr = mock(AuthenticationTrustResolver.class);
-		given(atr.isAuthenticated(JOE)).willReturn(true);
+		AuthenticationTrustResolver atr = createAuthenticationTrustResolver(true);
 		this.root.setTrustResolver(atr);
 		assertThat(this.root.isAuthenticated()).isTrue();
 	}
@@ -322,4 +320,10 @@
 		verify(factory).fullyAuthenticated();
 	}
 
+
+	private static AuthenticationTrustResolver createAuthenticationTrustResolver(boolean isAuthenticated) {
+		AuthenticationTrustResolver atr = mock(AuthenticationTrustResolver.class);
+		given(atr.isAuthenticated(JOE)).willReturn(isAuthenticated);
+		return atr;
+	}
 }
