--- a/web/src/test/java/org/springframework/security/web/authentication/logout/CompositeLogoutHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/logout/CompositeLogoutHandlerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.authentication.logout;
 
+import org.mockito.Mock;
 import java.util.Arrays;
 import java.util.List;
 
@@ -41,6 +42,8 @@
  * @since 4.2.0
  */
 public class CompositeLogoutHandlerTests {
+	@Mock
+	private LogoutHandler csrfLogoutHandler;
 
 	@Test
 	public void buildEmptyCompositeLogoutHandlerThrowsException() {
@@ -51,7 +54,6 @@
 	@Test
 	public void callLogoutHandlersSuccessfullyWithArray() {
 		LogoutHandler securityContextLogoutHandler = mock(SecurityContextLogoutHandler.class);
-		LogoutHandler csrfLogoutHandler = mock(SecurityContextLogoutHandler.class);
 		LogoutHandler handler = new CompositeLogoutHandler(securityContextLogoutHandler, csrfLogoutHandler);
 		handler.logout(mock(HttpServletRequest.class), mock(HttpServletResponse.class), mock(Authentication.class));
 		verify(securityContextLogoutHandler, times(1)).logout(any(HttpServletRequest.class),
@@ -63,7 +65,6 @@
 	@Test
 	public void callLogoutHandlersSuccessfully() {
 		LogoutHandler securityContextLogoutHandler = mock(SecurityContextLogoutHandler.class);
-		LogoutHandler csrfLogoutHandler = mock(SecurityContextLogoutHandler.class);
 		List<LogoutHandler> logoutHandlers = Arrays.asList(securityContextLogoutHandler, csrfLogoutHandler);
 		LogoutHandler handler = new CompositeLogoutHandler(logoutHandlers);
 		handler.logout(mock(HttpServletRequest.class), mock(HttpServletResponse.class), mock(Authentication.class));
