--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcReactiveOAuth2UserServiceTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcReactiveOAuth2UserServiceTests.java
@@ -137,7 +137,8 @@
 	public void loadUserWhenOAuth2UserSubjectNullThenOAuth2AuthenticationException() {
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"),
 				Collections.singletonMap("user", "rob"), "user");
-		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
+		this.oauth2UserService = createMockReactiveOAuth2UserService(oauth2User);
+		this.userService.setOauth2UserService(this.oauth2UserService);
 		assertThatExceptionOfType(OAuth2AuthenticationException.class)
 			.isThrownBy(() -> this.userService.loadUser(userRequest()).block());
 	}
@@ -149,7 +150,8 @@
 		attributes.put("user", "rob");
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"), attributes,
 				"user");
-		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
+		this.oauth2UserService = createMockReactiveOAuth2UserService(oauth2User);
+		this.userService.setOauth2UserService(this.oauth2UserService);
 		assertThatExceptionOfType(OAuth2AuthenticationException.class)
 			.isThrownBy(() -> this.userService.loadUser(userRequest()).block());
 	}
@@ -161,7 +163,7 @@
 		attributes.put("user", "rob");
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"), attributes,
 				"user");
-		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
+		this.userService.setOauth2UserService(createMockReactiveOAuth2UserService(oauth2User));
 		assertThat(this.userService.loadUser(userRequest()).block().getUserInfo()).isNotNull();
 	}
 
@@ -173,7 +175,7 @@
 		attributes.put("user", "rob");
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"), attributes,
 				"user");
-		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
+		this.userService.setOauth2UserService(createMockReactiveOAuth2UserService(oauth2User));
 		assertThat(this.userService.loadUser(userRequest()).block().getName()).isEqualTo("rob");
 	}
 
@@ -184,7 +186,8 @@
 		attributes.put("user", "rob");
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"), attributes,
 				"user");
-		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
+		this.oauth2UserService = createMockReactiveOAuth2UserService(oauth2User);
+		this.userService.setOauth2UserService(this.oauth2UserService);
 		OidcUserRequest userRequest = userRequest();
 		Function<ClientRegistration, Converter<Map<String, Object>, Map<String, Object>>> customClaimTypeConverterFactory = mock(
 				Function.class);
@@ -202,7 +205,8 @@
 		attributes.put("user", "steve");
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"), attributes,
 				"user");
-		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
+		this.oauth2UserService = createMockReactiveOAuth2UserService(oauth2User);
+		this.userService.setOauth2UserService(this.oauth2UserService);
 		Predicate<OidcUserRequest> customRetrieveUserInfo = mock(Predicate.class);
 		this.userService.setRetrieveUserInfo(customRetrieveUserInfo);
 		given(customRetrieveUserInfo.test(any(OidcUserRequest.class))).willReturn(true);
@@ -265,8 +269,7 @@
 				nameAttributeKey);
 		OAuth2User oauth2User = new DefaultOAuth2User(actualUser.getAuthorities(), actualUser.getClaims(),
 				nameAttributeKey);
-		ReactiveOAuth2UserService<OAuth2UserRequest, OAuth2User> oauth2 = mock(ReactiveOAuth2UserService.class);
-		given(oauth2.loadUser(any())).willReturn(Mono.just(oauth2User));
+		ReactiveOAuth2UserService<OAuth2UserRequest, OAuth2User> oauth2 = createMockReactiveOAuth2UserService(oauth2User);
 		given(oidcUserConverter.convert(any())).willReturn(Mono.just(actualUser));
 		this.userService.setOauth2UserService(oauth2);
 		this.userService.setOidcUserConverter(oidcUserConverter);
@@ -340,4 +343,11 @@
 				new MockResponse().setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE).setBody(json));
 	}
 
+
+	private static ReactiveOAuth2UserService<OAuth2UserRequest, OAuth2User> createMockReactiveOAuth2UserService(
+			OAuth2User oauth2User) {
+		ReactiveOAuth2UserService<OAuth2UserRequest, OAuth2User> service = mock(ReactiveOAuth2UserService.class);
+		given(service.loadUser(any())).willReturn(Mono.just(oauth2User));
+		return service;
+	}
 }
