--- a/taglibs/src/test/java/org/springframework/security/taglibs/authz/AbstractAuthorizeTagTests.java
+++ b/taglibs/src/test/java/org/springframework/security/taglibs/authz/AbstractAuthorizeTagTests.java
@@ -91,9 +91,7 @@
 
 	@Test
 	public void privilegeEvaluatorFromRequestUsesSecurityContextHolderStrategy() throws IOException {
-		SecurityContextHolderStrategy strategy = mock(SecurityContextHolderStrategy.class);
-		given(strategy.getContext()).willReturn(new SecurityContextImpl(
-				new TestingAuthenticationToken("user", "password", AuthorityUtils.NO_AUTHORITIES)));
+		SecurityContextHolderStrategy strategy = MockSecurityContextHolderStrategy.createMockSecurityContextHolderStrategy(new SecurityContextImpl(new TestingAuthenticationToken("user", "password", AuthorityUtils.NO_AUTHORITIES)));
 		GenericWebApplicationContext wac = new GenericWebApplicationContext();
 		wac.registerBean(SecurityContextHolderStrategy.class, () -> strategy);
 		wac.refresh();
@@ -176,4 +174,19 @@
 
 	}
 
+
+	public final class MockSecurityContextHolderStrategy {
+
+		private MockSecurityContextHolderStrategy() {
+		}
+
+		public static org.springframework.security.core.context.SecurityContextHolderStrategy createMockSecurityContextHolderStrategy(
+				org.springframework.security.core.context.SecurityContext context) {
+			org.springframework.security.core.context.SecurityContextHolderStrategy strategy =
+					org.mockito.Mockito.mock(
+							org.springframework.security.core.context.SecurityContextHolderStrategy.class);
+			org.mockito.BDDMockito.given(strategy.getContext()).willReturn(context);
+			return strategy;
+		}
+	}
 }
--- a/taglibs/src/test/java/org/springframework/security/taglibs/authz/AccessControlListTagTests.java
+++ b/taglibs/src/test/java/org/springframework/security/taglibs/authz/AccessControlListTagTests.java
@@ -97,8 +97,7 @@
 
 	@Test
 	public void securityContextHolderStrategyIsUsedIfConfigured() throws Exception {
-		SecurityContextHolderStrategy strategy = mock(SecurityContextHolderStrategy.class);
-		given(strategy.getContext()).willReturn(new SecurityContextImpl(this.bob));
+		SecurityContextHolderStrategy strategy = MockSecurityContextHolderStrategy.createMockSecurityContextHolderStrategy(new SecurityContextImpl(this.bob));
 		GenericWebApplicationContext context = new GenericWebApplicationContext();
 		context.registerBean(SecurityContextHolderStrategy.class, () -> strategy);
 		context.registerBean(PermissionEvaluator.class, () -> this.pe);
@@ -201,4 +200,19 @@
 		assertThat((Boolean) this.pageContext.getAttribute("allowed")).isFalse();
 	}
 
+
+	public final class MockSecurityContextHolderStrategy {
+
+		private MockSecurityContextHolderStrategy() {
+		}
+
+		public static org.springframework.security.core.context.SecurityContextHolderStrategy createMockSecurityContextHolderStrategy(
+				org.springframework.security.core.context.SecurityContext context) {
+			org.springframework.security.core.context.SecurityContextHolderStrategy strategy =
+					org.mockito.Mockito.mock(
+							org.springframework.security.core.context.SecurityContextHolderStrategy.class);
+			org.mockito.BDDMockito.given(strategy.getContext()).willReturn(context);
+			return strategy;
+		}
+	}
 }
--- a/taglibs/src/test/java/org/springframework/security/taglibs/authz/AuthenticationTagTests.java
+++ b/taglibs/src/test/java/org/springframework/security/taglibs/authz/AuthenticationTagTests.java
@@ -141,9 +141,7 @@
 
 	@Test
 	public void setSecurityContextHolderStrategyThenUses() throws Exception {
-		SecurityContextHolderStrategy strategy = mock(SecurityContextHolderStrategy.class);
-		given(strategy.getContext()).willReturn(new SecurityContextImpl(
-				new TestingAuthenticationToken("rodAsString", "koala", AuthorityUtils.NO_AUTHORITIES)));
+		SecurityContextHolderStrategy strategy = MockSecurityContextHolderStrategy.createMockSecurityContextHolderStrategy(new SecurityContextImpl(new TestingAuthenticationToken("rodAsString", "koala", AuthorityUtils.NO_AUTHORITIES)));
 		MockServletContext servletContext = new MockServletContext();
 		GenericWebApplicationContext applicationContext = new GenericWebApplicationContext();
 		applicationContext.registerBean(SecurityContextHolderStrategy.class, () -> strategy);
@@ -172,4 +170,19 @@
 
 	}
 
+
+	public final class MockSecurityContextHolderStrategy {
+
+		private MockSecurityContextHolderStrategy() {
+		}
+
+		public static org.springframework.security.core.context.SecurityContextHolderStrategy createMockSecurityContextHolderStrategy(
+				org.springframework.security.core.context.SecurityContext context) {
+			org.springframework.security.core.context.SecurityContextHolderStrategy strategy =
+					org.mockito.Mockito.mock(
+							org.springframework.security.core.context.SecurityContextHolderStrategy.class);
+			org.mockito.BDDMockito.given(strategy.getContext()).willReturn(context);
+			return strategy;
+		}
+	}
 }
