--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
@@ -118,10 +118,7 @@
 		claims.put("aud", principal.getAudience());
 		claims.put("auth_time", authTime);
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -157,10 +154,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -193,10 +187,7 @@
 		claims.put("sub", "invalid_sub");
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -229,10 +220,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt().minus(Duration.ofDays(1)));
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt().minus(Duration.ofDays(1)));
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -265,10 +253,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", Collections.emptyList());
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -301,10 +286,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", List.of("invalid_client-id"));
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -339,10 +321,7 @@
 		claims.put("aud", principal.getAudience());
 		claims.put("auth_time", authTime.minus(5, ChronoUnit.MINUTES));
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -377,10 +356,7 @@
 		claims.put("aud", principal.getAudience());
 		claims.put("auth_time", authTime.plus(5, ChronoUnit.MINUTES));
 		claims.put("nonce", principal.getNonce());
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -416,10 +392,7 @@
 		claims.put("sub", principal.getSubject());
 		claims.put("aud", principal.getAudience());
 		claims.put("nonce", "invalid_nonce");
-		Jwt jwt = mock(Jwt.class);
-		given(jwt.getTokenValue()).willReturn("id-token-1234");
-		given(jwt.getIssuedAt()).willReturn(principal.getIssuedAt());
-		given(jwt.getClaims()).willReturn(claims);
+		Jwt jwt = createJwt(claims, principal.getIssuedAt());
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
 		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
@@ -459,4 +432,12 @@
 		return new DefaultOidcUser(null, idToken);
 	}
 
+
+	private static Jwt createJwt(Map<String, Object> claims, Instant issuedAt) {
+		Jwt jwt = mock(Jwt.class);
+		given(jwt.getTokenValue()).willReturn("id-token-1234");
+		given(jwt.getIssuedAt()).willReturn(issuedAt);
+		given(jwt.getClaims()).willReturn(claims);
+		return jwt;
+	}
 }
