--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/management/CredentialRecordOwnerAuthorizationManagerTests.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/management/CredentialRecordOwnerAuthorizationManagerTests.java
@@ -92,11 +92,9 @@
 
 	@Test
 	void checkWhenCredentialUserEntityUserIdNullThenDenied() {
-		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(
-				this.userCredentials, this.userEntities);
 		Bytes credentialId = TestCredentialRecords.userCredential().build().getCredentialId();
-		given(this.userCredentials.findByCredentialId(credentialId))
-			.willReturn(TestCredentialRecords.userCredential().userEntityUserId(null).build());
+		this.userCredentials = MockUserCredentialRepository.createMockUserCredentialRepository(credentialId, TestCredentialRecords.userCredential().userEntityUserId(null).build());
+		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(this.userCredentials, this.userEntities);
 		TestingAuthenticationToken authentication = new TestingAuthenticationToken("user", "password", "USER");
 		AuthorizationResult decision = manager.authorize(() -> authentication, credentialId);
 		assertThat(decision.isGranted()).isFalse();
@@ -104,11 +102,10 @@
 
 	@Test
 	void checkWhenUserEntityNotFoundThenDenied() {
-		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(
-				this.userCredentials, this.userEntities);
+		CredentialRecordOwnerAuthorizationManager manager;
 		Bytes credentialId = TestCredentialRecords.userCredential().build().getCredentialId();
-		given(this.userCredentials.findByCredentialId(credentialId))
-			.willReturn(TestCredentialRecords.userCredential().build());
+		this.userCredentials = MockUserCredentialRepository.createMockUserCredentialRepository(credentialId, TestCredentialRecords.userCredential().build());
+		manager = new CredentialRecordOwnerAuthorizationManager(this.userCredentials, this.userEntities);
 		TestingAuthenticationToken authentication = new TestingAuthenticationToken("user", "password", "USER");
 		AuthorizationResult decision = manager.authorize(() -> authentication, credentialId);
 		assertThat(decision.isGranted()).isFalse();
@@ -116,12 +113,13 @@
 
 	@Test
 	void checkWhenCredentialBelongsToUserThenGranted() {
+	    this.userCredentials = MockUserCredentialRepository.createMockUserCredentialRepository(
+	            TestCredentialRecords.userCredential().build().getCredentialId(),
+	            TestCredentialRecords.userCredential().build());
 		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(
 				this.userCredentials, this.userEntities);
 		Bytes credentialId = TestCredentialRecords.userCredential().build().getCredentialId();
 		Bytes userId = TestCredentialRecords.userCredential().build().getUserEntityUserId();
-		given(this.userCredentials.findByCredentialId(credentialId))
-			.willReturn(TestCredentialRecords.userCredential().build());
 		PublicKeyCredentialUserEntity userEntity = ImmutablePublicKeyCredentialUserEntity.builder()
 			.name("user")
 			.id(userId)
@@ -135,11 +133,9 @@
 
 	@Test
 	void checkWhenCredentialBelongsToDifferentUserThenDenied() {
-		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(
-				this.userCredentials, this.userEntities);
 		Bytes credentialId = TestCredentialRecords.userCredential().build().getCredentialId();
-		given(this.userCredentials.findByCredentialId(credentialId))
-			.willReturn(TestCredentialRecords.userCredential().build());
+		this.userCredentials = MockUserCredentialRepository.createMockUserCredentialRepository(credentialId, TestCredentialRecords.userCredential().build());
+		CredentialRecordOwnerAuthorizationManager manager = new CredentialRecordOwnerAuthorizationManager(this.userCredentials, this.userEntities);
 		PublicKeyCredentialUserEntity otherUserEntity = ImmutablePublicKeyCredentialUserEntity.builder()
 			.name("user")
 			.id(TestBytes.get())
@@ -151,4 +147,19 @@
 		assertThat(decision.isGranted()).isFalse();
 	}
 
+
+	public final class MockUserCredentialRepository {
+
+		private MockUserCredentialRepository() {
+		}
+
+		public static org.springframework.security.web.webauthn.management.UserCredentialRepository createMockUserCredentialRepository(
+				org.springframework.security.web.webauthn.api.Bytes credentialId,
+				org.springframework.security.web.webauthn.api.ImmutableCredentialRecord credentialRecord) {
+			org.springframework.security.web.webauthn.management.UserCredentialRepository userCredentials =
+					Mockito.mock(org.springframework.security.web.webauthn.management.UserCredentialRepository.class);
+			given(userCredentials.findByCredentialId(credentialId)).willReturn(credentialRecord);
+			return userCredentials;
+		}
+	}
 }
--- a/webauthn/src/test/java/org/springframework/security/web/webauthn/management/Webauthn4jRelyingPartyOperationsTests.java
+++ b/webauthn/src/test/java/org/springframework/security/web/webauthn/management/Webauthn4jRelyingPartyOperationsTests.java
@@ -328,8 +328,8 @@
 
 		ImmutableRelyingPartyRegistrationRequest rpRegistrationRequest = new ImmutableRelyingPartyRegistrationRequest(
 				creationOptions, rpPublicKey);
-		given(this.userCredentials.findByCredentialId(publicKeyCredential.getRawId()))
-			.willReturn(TestCredentialRecords.userCredential().build());
+		this.userCredentials = MockUserCredentialRepository.createMockUserCredentialRepository(
+		        publicKeyCredential.getRawId(), TestCredentialRecords.userCredential().build());
 		assertThatRuntimeException().isThrownBy(() -> this.rpOperations.registerCredential(rpRegistrationRequest));
 	}
 
@@ -648,7 +648,7 @@
 		PublicKeyCredential<AuthenticatorAssertionResponse> publicKey = request.getPublicKey();
 
 		ImmutableCredentialRecord credentialRecord = TestCredentialRecords.fullUserCredential().build();
-		given(this.userCredentials.findByCredentialId(publicKey.getRawId())).willReturn(credentialRecord);
+		this.userCredentials = MockUserCredentialRepository.createMockUserCredentialRepository(publicKey.getRawId(), credentialRecord);
 		JsonMapper json = new JsonMapper();
 		CBORMapper cbor = mock(CBORMapper.class);
 		CBORMapper.Builder builder = mock(CBORMapper.Builder.class, RETURNS_SELF);
@@ -722,4 +722,19 @@
 		return new ObjectMapper(new CBORFactory()).setBase64Variant(Base64Variants.MODIFIED_FOR_URL);
 	}
 
+
+	public final class MockUserCredentialRepository {
+
+		private MockUserCredentialRepository() {
+		}
+
+		public static org.springframework.security.web.webauthn.management.UserCredentialRepository createMockUserCredentialRepository(
+				org.springframework.security.web.webauthn.api.Bytes credentialId,
+				org.springframework.security.web.webauthn.api.ImmutableCredentialRecord credentialRecord) {
+			org.springframework.security.web.webauthn.management.UserCredentialRepository userCredentials =
+					Mockito.mock(org.springframework.security.web.webauthn.management.UserCredentialRepository.class);
+			given(userCredentials.findByCredentialId(credentialId)).willReturn(credentialRecord);
+			return userCredentials;
+		}
+	}
 }
