--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.config.annotation.web.configurers.oauth2.server.resource;
 
+import org.mockito.Mock;
 import java.io.BufferedReader;
 import java.io.FileReader;
 import java.io.IOException;
@@ -184,6 +185,8 @@
  */
 @ExtendWith(SpringTestContextExtension.class)
 public class OAuth2ResourceServerConfigurerTests {
+	@Mock
+	private JwtDecoder decoder;
 
 	private static final String JWT_TOKEN = "token";
 
@@ -831,7 +834,6 @@
 	@Test
 	public void getJwtDecoderWhenConflictingJwtDecodersThenTheDslWiredOneTakesPrecedence() {
 		JwtDecoder decoderBean = mock(JwtDecoder.class);
-		JwtDecoder decoder = mock(JwtDecoder.class);
 		ApplicationContext context = mock(ApplicationContext.class);
 		given(context.getBean(JwtDecoder.class)).willReturn(decoderBean);
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
@@ -855,7 +857,6 @@
 	@Test
 	public void getJwtDecoderWhenTwoJwtDecoderBeansAndAnotherWiredOnDslThenDslWiredOneTakesPrecedence() {
 		JwtDecoder decoderBean = mock(JwtDecoder.class);
-		JwtDecoder decoder = mock(JwtDecoder.class);
 		GenericWebApplicationContext context = new GenericWebApplicationContext();
 		context.registerBean("decoderOne", JwtDecoder.class, () -> decoderBean);
 		context.registerBean("decoderTwo", JwtDecoder.class, () -> decoderBean);
