--- a/access/src/test/java/org/springframework/security/web/access/intercept/FilterSecurityInterceptorTests.java
+++ b/access/src/test/java/org/springframework/security/web/access/intercept/FilterSecurityInterceptorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.access.intercept;
 
+import org.mockito.Mock;
 import jakarta.servlet.FilterChain;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
@@ -64,6 +65,8 @@
  */
 @SuppressWarnings("deprecation")
 public class FilterSecurityInterceptorTests {
+	@Mock
+	private AfterInvocationManager aim;
 
 	private AuthenticationManager am;
 
@@ -136,7 +139,6 @@
 		willThrow(new RuntimeException()).given(chain)
 			.doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		given(this.ods.getAttributes(fi)).willReturn(SecurityConfig.createList("MOCK_OK"));
-		AfterInvocationManager aim = mock(AfterInvocationManager.class);
 		this.interceptor.setAfterInvocationManager(aim);
 		assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> this.interceptor.invoke(fi));
 		verifyNoMoreInteractions(aim);
@@ -159,7 +161,6 @@
 		willThrow(new RuntimeException()).given(chain)
 			.doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		given(this.ods.getAttributes(fi)).willReturn(SecurityConfig.createList("MOCK_OK"));
-		AfterInvocationManager aim = mock(AfterInvocationManager.class);
 		this.interceptor.setAfterInvocationManager(aim);
 		assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> this.interceptor.invoke(fi));
 		// Check we've changed back
