--- a/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
+++ b/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
@@ -108,6 +108,8 @@
  */
 @ExtendWith(MockitoExtension.class)
 public class ServerHttpSecurityTests {
+	@Mock
+	private X509PrincipalExtractor mockExtractor;
 
 	@Mock
 	private ServerSecurityContextRepository contextRepository;
@@ -478,7 +480,7 @@
 	@Test
 	@SuppressWarnings("unchecked")
 	public void addsX509FilterWhenX509AuthenticationIsConfigured() {
-		X509PrincipalExtractor mockExtractor = mock(X509PrincipalExtractor.class);
+		// removed local mock; replaced with global field `mockExtractor`
 		ReactiveAuthenticationManager mockAuthenticationManager = mock(ReactiveAuthenticationManager.class);
 		this.http
 			.x509((x509) -> x509.principalExtractor(mockExtractor).authenticationManager(mockAuthenticationManager));
@@ -489,7 +491,6 @@
 
 	@Test
 	public void x509WhenCustomizedThenAddsX509Filter() {
-		X509PrincipalExtractor mockExtractor = mock(X509PrincipalExtractor.class);
 		ReactiveAuthenticationManager mockAuthenticationManager = mock(ReactiveAuthenticationManager.class);
 		this.http
 			.x509((x509) -> x509.principalExtractor(mockExtractor).authenticationManager(mockAuthenticationManager));
