--- a/access/src/test/java/org/springframework/security/access/intercept/aopalliance/MethodSecurityInterceptorTests.java
+++ b/access/src/test/java/org/springframework/security/access/intercept/aopalliance/MethodSecurityInterceptorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.access.intercept.aopalliance;
 
+import org.mockito.Mock;
 import java.util.List;
 
 import org.aopalliance.intercept.MethodInvocation;
@@ -67,6 +68,8 @@
  */
 @SuppressWarnings({ "unchecked", "deprecation" })
 public class MethodSecurityInterceptorTests {
+	@Mock
+	private AfterInvocationManager aim;
 
 	private TestingAuthenticationToken token;
 
@@ -115,7 +118,6 @@
 	@Test
 	public void gettersReturnExpectedData() {
 		RunAsManager runAs = mock(RunAsManager.class);
-		AfterInvocationManager aim = mock(AfterInvocationManager.class);
 		this.interceptor.setRunAsManager(runAs);
 		this.interceptor.setAfterInvocationManager(aim);
 		assertThat(this.interceptor.getAccessDecisionManager()).isEqualTo(this.adm);
@@ -314,7 +316,6 @@
 		this.token.setAuthenticated(true);
 		SecurityContextHolder.getContext().setAuthentication(this.token);
 		mdsReturnsUserRole();
-		AfterInvocationManager aim = mock(AfterInvocationManager.class);
 		this.interceptor.setAfterInvocationManager(aim);
 		given(mi.proceed()).willThrow(new Throwable());
 		assertThatExceptionOfType(Throwable.class).isThrownBy(() -> this.interceptor.invoke(mi));
