--- a/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
+++ b/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
@@ -646,11 +646,8 @@
 
 	@Test
 	public void shouldUseDefaultAuthorizationRedirectStrategyForOAuth2Login() {
-		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
-				ReactiveClientRegistrationRepository.class);
-		given(clientRegistrationRepository.findByRegistrationId(anyString()))
-			.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
-
+
+		ReactiveClientRegistrationRepository clientRegistrationRepository = createReactiveClientRegistrationRepository();
 		SecurityWebFilterChain securityFilterChain = this.http
 			.oauth2Login((login) -> login.clientRegistrationRepository(clientRegistrationRepository))
 			.build();
@@ -668,10 +665,7 @@
 	@Test
 	public void shouldConfigureAuthorizationRedirectStrategyForOAuth2Login() {
 		ServerRedirectStrategy authorizationRedirectStrategy = mock(ServerRedirectStrategy.class);
-		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
-				ReactiveClientRegistrationRepository.class);
-		given(clientRegistrationRepository.findByRegistrationId(anyString()))
-			.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
+		ReactiveClientRegistrationRepository clientRegistrationRepository = createReactiveClientRegistrationRepository();
 		given(authorizationRedirectStrategy.sendRedirect(any(), any())).willReturn(Mono.empty());
 
 		SecurityWebFilterChain securityFilterChain = this.http
@@ -692,11 +686,8 @@
 
 	@Test
 	public void shouldUseDefaultAuthorizationRedirectStrategyForOAuth2Client() {
-		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
-				ReactiveClientRegistrationRepository.class);
-		given(clientRegistrationRepository.findByRegistrationId(anyString()))
-			.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
-
+
+		ReactiveClientRegistrationRepository clientRegistrationRepository = createReactiveClientRegistrationRepository();
 		SecurityWebFilterChain securityFilterChain = this.http
 			.oauth2Client((client) -> client.clientRegistrationRepository(clientRegistrationRepository))
 			.build();
@@ -714,10 +705,7 @@
 	@Test
 	public void shouldConfigureAuthorizationRedirectStrategyForOAuth2Client() {
 		ServerRedirectStrategy authorizationRedirectStrategy = mock(ServerRedirectStrategy.class);
-		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
-				ReactiveClientRegistrationRepository.class);
-		given(clientRegistrationRepository.findByRegistrationId(anyString()))
-			.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
+		ReactiveClientRegistrationRepository clientRegistrationRepository = createReactiveClientRegistrationRepository();
 		given(authorizationRedirectStrategy.sendRedirect(any(), any())).willReturn(Mono.empty());
 
 		SecurityWebFilterChain securityFilterChain = this.http
@@ -835,4 +823,11 @@
 
 	}
 
+
+	private static ReactiveClientRegistrationRepository createReactiveClientRegistrationRepository() {
+		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(ReactiveClientRegistrationRepository.class);
+		given(clientRegistrationRepository.findByRegistrationId(anyString()))
+				.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
+		return clientRegistrationRepository;
+	}
 }
