--- a/web/src/test/java/org/springframework/security/web/FilterChainProxyTests.java
+++ b/web/src/test/java/org/springframework/security/web/FilterChainProxyTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web;
 
+import org.mockito.Mock;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Collections;
@@ -71,6 +72,8 @@
  * @author Rob Winch
  */
 public class FilterChainProxyTests {
+	@Mock
+	private RequestRejectedHandler rjh;
 
 	private FilterChainProxy fcp;
 
@@ -267,7 +270,6 @@
 	@Test
 	public void requestRejectedHandlerIsCalledIfFirewallThrowsRequestRejectedException() throws Exception {
 		HttpFirewall fw = mock(HttpFirewall.class);
-		RequestRejectedHandler rjh = mock(RequestRejectedHandler.class);
 		this.fcp.setFirewall(fw);
 		this.fcp.setRequestRejectedHandler(rjh);
 		RequestRejectedException requestRejectedException = new RequestRejectedException("Contains illegal chars");
@@ -279,7 +281,6 @@
 	@Test
 	public void requestRejectedHandlerIsCalledIfFirewallThrowsWrappedRequestRejectedException() throws Exception {
 		HttpFirewall fw = mock(HttpFirewall.class);
-		RequestRejectedHandler rjh = mock(RequestRejectedHandler.class);
 		this.fcp.setFirewall(fw);
 		this.fcp.setRequestRejectedHandler(rjh);
 		RequestRejectedException requestRejectedException = new RequestRejectedException("Contains illegal chars");
