--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java
@@ -130,6 +130,8 @@
  */
 @ExtendWith(MockitoExtension.class)
 public class ServletOAuth2AuthorizedClientExchangeFilterFunctionTests {
+	@Mock
+	private OAuth2User user;
 
 	@Mock
 	private OAuth2AuthorizedClientRepository authorizedClientRepository;
@@ -594,7 +596,6 @@
 		this.function.setDefaultOAuth2AuthorizedClient(true);
 		MockHttpServletRequest servletRequest = new MockHttpServletRequest();
 		MockHttpServletResponse servletResponse = new MockHttpServletResponse();
-		OAuth2User user = mock(OAuth2User.class);
 		List<GrantedAuthority> authorities = AuthorityUtils.createAuthorityList("ROLE_USER");
 		OAuth2AuthenticationToken authentication = new OAuth2AuthenticationToken(user, authorities,
 				this.registration.getRegistrationId());
@@ -633,7 +634,6 @@
 		this.function.setDefaultOAuth2AuthorizedClient(true);
 		MockHttpServletRequest servletRequest = new MockHttpServletRequest();
 		MockHttpServletResponse servletResponse = new MockHttpServletResponse();
-		OAuth2User user = mock(OAuth2User.class);
 		List<GrantedAuthority> authorities = AuthorityUtils.createAuthorityList("ROLE_USER");
 		OAuth2AuthenticationToken initialAuthentication = new OAuth2AuthenticationToken(user, authorities,
 				"initial-registration-id");
