--- a/web/src/test/java/org/springframework/security/web/authentication/rememberme/RememberMeAuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/rememberme/RememberMeAuthenticationFilterTests.java
@@ -99,8 +99,7 @@
 
 	@Test
 	public void testOperationWhenNoAuthenticationInContextHolder() throws Exception {
-		AuthenticationManager am = mock(AuthenticationManager.class);
-		given(am.authenticate(this.remembered)).willReturn(this.remembered);
+		AuthenticationManager am = createAuthenticationManager(this.remembered);
 		RememberMeAuthenticationFilter filter = new RememberMeAuthenticationFilter(am,
 				new MockRememberMeServices(this.remembered));
 		filter.afterPropertiesSet();
@@ -139,8 +138,7 @@
 
 	@Test
 	public void authenticationSuccessHandlerIsInvokedOnSuccessfulAuthenticationIfSet() throws Exception {
-		AuthenticationManager am = mock(AuthenticationManager.class);
-		given(am.authenticate(this.remembered)).willReturn(this.remembered);
+		AuthenticationManager am = createAuthenticationManager(this.remembered);
 		RememberMeAuthenticationFilter filter = new RememberMeAuthenticationFilter(am,
 				new MockRememberMeServices(this.remembered));
 		filter.setAuthenticationSuccessHandler(new SimpleUrlAuthenticationSuccessHandler("/target"));
@@ -157,8 +155,7 @@
 	@Test
 	public void securityContextRepositoryInvokedIfSet() throws Exception {
 		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
-		AuthenticationManager am = mock(AuthenticationManager.class);
-		given(am.authenticate(this.remembered)).willReturn(this.remembered);
+		AuthenticationManager am = createAuthenticationManager(this.remembered);
 		RememberMeAuthenticationFilter filter = new RememberMeAuthenticationFilter(am,
 				new MockRememberMeServices(this.remembered));
 		filter.setAuthenticationSuccessHandler(new SimpleUrlAuthenticationSuccessHandler("/target"));
@@ -174,8 +171,7 @@
 	@Test
 	public void sessionAuthenticationStrategyInvokedIfSet() throws Exception {
 		SessionAuthenticationStrategy sessionAuthenticationStrategy = mock(SessionAuthenticationStrategy.class);
-		AuthenticationManager am = mock(AuthenticationManager.class);
-		given(am.authenticate(this.remembered)).willReturn(this.remembered);
+		AuthenticationManager am = createAuthenticationManager(this.remembered);
 		RememberMeAuthenticationFilter filter = new RememberMeAuthenticationFilter(am,
 				new MockRememberMeServices(this.remembered));
 		filter.setAuthenticationSuccessHandler(new SimpleUrlAuthenticationSuccessHandler("/target"));
@@ -212,4 +208,10 @@
 
 	}
 
+
+	private static AuthenticationManager createAuthenticationManager(Authentication remembered) {
+		AuthenticationManager am = mock(AuthenticationManager.class);
+		given(am.authenticate(remembered)).willReturn(remembered);
+		return am;
+	}
 }
