--- a/web/src/test/java/org/springframework/security/web/context/DelegatingSecurityContextRepositoryTests.java
+++ b/web/src/test/java/org/springframework/security/web/context/DelegatingSecurityContextRepositoryTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.context;
 
+import org.mockito.Mock;
 import java.util.ArrayList;
 import java.util.List;
 
@@ -46,6 +47,8 @@
  * @since 5.8
  */
 public class DelegatingSecurityContextRepositoryTests {
+	@Mock
+	private SecurityContext securityContext2;
 
 	private MockHttpServletRequest request;
 
@@ -148,7 +151,6 @@
 		SecurityContextRepository delegate2 = mock(SecurityContextRepository.class);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(this.request, this.response);
 		SecurityContext securityContext1 = mock(SecurityContext.class);
-		SecurityContext securityContext2 = mock(SecurityContext.class);
 
 		given(delegate1.loadContext(holder)).willReturn(securityContext1);
 		given(delegate1.containsContext(holder.getRequest())).willReturn(false);
@@ -168,7 +170,6 @@
 		SecurityContextRepository delegate2 = mock(SecurityContextRepository.class);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(this.request, this.response);
 		SecurityContext securityContext1 = mock(SecurityContext.class);
-		SecurityContext securityContext2 = mock(SecurityContext.class);
 
 		given(delegate1.loadContext(holder)).willReturn(securityContext1);
 		given(delegate1.containsContext(holder.getRequest())).willReturn(true);
@@ -188,7 +189,6 @@
 		SecurityContextRepository delegate2 = mock(SecurityContextRepository.class);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(this.request, this.response);
 		SecurityContext securityContext1 = mock(SecurityContext.class);
-		SecurityContext securityContext2 = mock(SecurityContext.class);
 
 		given(delegate1.loadContext(holder)).willReturn(securityContext1);
 		given(delegate1.containsContext(holder.getRequest())).willReturn(true);
