--- a/acl/src/test/java/org/springframework/security/acls/AclPermissionCacheOptimizerTests.java
+++ b/acl/src/test/java/org/springframework/security/acls/AclPermissionCacheOptimizerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.acls;
 
+import org.mockito.Mock;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
@@ -41,10 +42,11 @@
  */
 @SuppressWarnings({ "unchecked" })
 public class AclPermissionCacheOptimizerTests {
+	@Mock
+	private AclService service;
 
 	@Test
 	public void eagerlyLoadsRequiredAcls() {
-		AclService service = mock(AclService.class);
 		AclPermissionCacheOptimizer pco = new AclPermissionCacheOptimizer(service);
 		ObjectIdentityRetrievalStrategy oidStrat = mock(ObjectIdentityRetrievalStrategy.class);
 		SidRetrievalStrategy sidStrat = mock(SidRetrievalStrategy.class);
@@ -61,7 +63,6 @@
 
 	@Test
 	public void ignoresEmptyCollection() {
-		AclService service = mock(AclService.class);
 		AclPermissionCacheOptimizer pco = new AclPermissionCacheOptimizer(service);
 		ObjectIdentityRetrievalStrategy oids = mock(ObjectIdentityRetrievalStrategy.class);
 		SidRetrievalStrategy sids = mock(SidRetrievalStrategy.class);
