--- a/data/src/test/java/org/springframework/security/data/repository/query/SecurityEvaluationContextExtensionTests.java
+++ b/data/src/test/java/org/springframework/security/data/repository/query/SecurityEvaluationContextExtensionTests.java
@@ -104,8 +104,7 @@
 	public void setTrustResolverWhenNotNullThenVerifyRootObject() {
 		TestingAuthenticationToken explicit = new TestingAuthenticationToken("explicit", "password", "ROLE_EXPLICIT");
 		this.securityExtension = new SecurityEvaluationContextExtension(explicit);
-		AuthenticationTrustResolver trustResolver = mock(AuthenticationTrustResolver.class);
-		given(trustResolver.isAuthenticated(explicit)).willReturn(true);
+		AuthenticationTrustResolver trustResolver = createAuthenticationTrustResolver(explicit);
 		this.securityExtension.setTrustResolver(trustResolver);
 		assertThat(getRoot().isAuthenticated()).isTrue();
 		verify(trustResolver).isAuthenticated(explicit);
@@ -161,8 +160,7 @@
 	public void setAuthorizationManagerFactoryWithTrustResolverThenVerifyRootObject() {
 		TestingAuthenticationToken explicit = new TestingAuthenticationToken("explicit", "password", "ROLE_EXPLICIT");
 		this.securityExtension = new SecurityEvaluationContextExtension(explicit);
-		AuthenticationTrustResolver trustResolver = mock(AuthenticationTrustResolver.class);
-		given(trustResolver.isAuthenticated(explicit)).willReturn(true);
+		AuthenticationTrustResolver trustResolver = createAuthenticationTrustResolver(explicit);
 		DefaultAuthorizationManagerFactory<Object> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setTrustResolver(trustResolver);
 		this.securityExtension.setAuthorizationManagerFactory(factory);
@@ -207,4 +205,10 @@
 		return this.securityExtension.getRootObject();
 	}
 
+
+	private static AuthenticationTrustResolver createAuthenticationTrustResolver(TestingAuthenticationToken explicit) {
+		AuthenticationTrustResolver trustResolver = Mockito.mock(AuthenticationTrustResolver.class);
+		given(trustResolver.isAuthenticated(explicit)).willReturn(true);
+		return trustResolver;
+	}
 }
