--- a/web/src/test/java/org/springframework/security/web/authentication/logout/SecurityContextLogoutHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/logout/SecurityContextLogoutHandlerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.authentication.logout;
 
+import org.mockito.Mock;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
@@ -42,6 +43,8 @@
  * @author Rob Winch
  */
 public class SecurityContextLogoutHandlerTests {
+	@Mock
+	private SecurityContextRepository repository;
 
 	private MockHttpServletRequest request;
 
@@ -85,7 +88,6 @@
 
 	@Test
 	public void logoutWhenSecurityContextRepositoryThenSaveEmptyContext() {
-		SecurityContextRepository repository = mock(SecurityContextRepository.class);
 		this.handler.setSecurityContextRepository(repository);
 		this.handler.logout(this.request, this.response, SecurityContextHolder.getContext().getAuthentication());
 		verify(repository).saveContext(eq(SecurityContextHolder.createEmptyContext()), any(), any());
@@ -93,7 +95,6 @@
 
 	@Test
 	public void logoutWhenClearAuthenticationFalseThenSaveEmptyContext() {
-		SecurityContextRepository repository = mock(SecurityContextRepository.class);
 		this.handler.setSecurityContextRepository(repository);
 		this.handler.setClearAuthentication(false);
 		this.handler.logout(this.request, this.response, SecurityContextHolder.getContext().getAuthentication());
