--- a/web/src/test/java/org/springframework/security/web/authentication/preauth/AbstractPreAuthenticatedProcessingFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/preauth/AbstractPreAuthenticatedProcessingFilterTests.java
@@ -93,8 +93,7 @@
 
 	@Test
 	public void filterChainProceedsOnFailedAuthenticationByDefault() throws Exception {
-		AuthenticationManager am = mock(AuthenticationManager.class);
-		given(am.authenticate(any(Authentication.class))).willThrow(new BadCredentialsException(""));
+		AuthenticationManager am = createAuthenticationManager();
 		this.filter.setAuthenticationManager(am);
 		this.filter.afterPropertiesSet();
 		this.filter.doFilter(new MockHttpServletRequest(), new MockHttpServletResponse(), mock(FilterChain.class));
@@ -105,8 +104,7 @@
 	@Test
 	public void exceptionIsThrownOnFailedAuthenticationIfContinueFilterChainOnUnsuccessfulAuthenticationSetToFalse()
 			throws Exception {
-		AuthenticationManager am = mock(AuthenticationManager.class);
-		given(am.authenticate(any(Authentication.class))).willThrow(new BadCredentialsException(""));
+		AuthenticationManager am = createAuthenticationManager();
 		this.filter.setContinueFilterChainOnUnsuccessfulAuthentication(false);
 		this.filter.setAuthenticationManager(am);
 		this.filter.afterPropertiesSet();
@@ -545,4 +543,11 @@
 
 	}
 
+
+	private static AuthenticationManager createAuthenticationManager() {
+		AuthenticationManager am = mock(AuthenticationManager.class);
+		given(am.authenticate(any(Authentication.class)))
+				.willThrow(new BadCredentialsException(""));
+		return am;
+	}
 }
