--- a/core/src/test/java/org/springframework/security/core/context/ThreadLocalSecurityContextHolderStrategyTests.java
+++ b/core/src/test/java/org/springframework/security/core/context/ThreadLocalSecurityContextHolderStrategyTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.core.context;
 
+import org.mockito.Mock;
 import java.util.function.Supplier;
 
 import org.junit.jupiter.api.AfterEach;
@@ -29,6 +30,8 @@
 import static org.mockito.Mockito.verifyNoInteractions;
 
 class ThreadLocalSecurityContextHolderStrategyTests {
+	@Mock
+	private Authentication authentication;
 
 	ThreadLocalSecurityContextHolderStrategy strategy = new ThreadLocalSecurityContextHolderStrategy();
 
@@ -46,7 +49,6 @@
 
 	@Test
 	void deferredContext() {
-		Authentication authentication = mock(Authentication.class);
 		Supplier<SecurityContext> deferredContext = () -> new SecurityContextImpl(authentication);
 		this.strategy.setDeferredContext(deferredContext);
 		assertThat(this.strategy.getDeferredContext().get()).isEqualTo(deferredContext.get());
@@ -62,7 +64,6 @@
 
 	@Test
 	void context() {
-		Authentication authentication = mock(Authentication.class);
 		SecurityContext context = new SecurityContextImpl(authentication);
 		this.strategy.setContext(context);
 		assertThat(this.strategy.getContext()).isEqualTo(context);
@@ -76,7 +77,6 @@
 
 	@Test
 	void getContextWhenEmptyThenReturnsSameInstance() {
-		Authentication authentication = mock(Authentication.class);
 		this.strategy.getContext().setAuthentication(authentication);
 		assertThat(this.strategy.getContext().getAuthentication()).isEqualTo(authentication);
 	}
