--- a/config/src/test/java/org/springframework/security/config/annotation/web/configuration/OAuth2ClientConfigurationTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configuration/OAuth2ClientConfigurationTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.config.annotation.web.configuration;
 
+import org.mockito.Mock;
 import jakarta.servlet.http.HttpServletRequest;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
@@ -71,6 +72,8 @@
  */
 @ExtendWith(SpringTestContextExtension.class)
 public class OAuth2ClientConfigurationTests {
+	@Mock
+	private OAuth2AuthorizedClientRepository authorizedClientRepository;
 
 	public final SpringTestContext spring = new SpringTestContext(this);
 
@@ -116,7 +119,6 @@
 		String principalName = "user1";
 		TestingAuthenticationToken authentication = new TestingAuthenticationToken(principalName, "password");
 		ClientRegistrationRepository clientRegistrationRepository = mock(ClientRegistrationRepository.class);
-		OAuth2AuthorizedClientRepository authorizedClientRepository = mock(OAuth2AuthorizedClientRepository.class);
 		OAuth2AccessTokenResponseClient accessTokenResponseClient = mock(OAuth2AccessTokenResponseClient.class);
 		ClientRegistration clientRegistration = TestClientRegistrations.clientCredentials()
 			.registrationId(clientRegistrationId)
@@ -197,7 +199,6 @@
 		String principalName = "user1";
 		TestingAuthenticationToken authentication = new TestingAuthenticationToken(principalName, "password");
 		ClientRegistrationRepository clientRegistrationRepository = mock(ClientRegistrationRepository.class);
-		OAuth2AuthorizedClientRepository authorizedClientRepository = mock(OAuth2AuthorizedClientRepository.class);
 		OAuth2AuthorizedClientManager authorizedClientManager = mock(OAuth2AuthorizedClientManager.class);
 		ClientRegistration clientRegistration = TestClientRegistrations.clientRegistration()
 			.registrationId(clientRegistrationId)
