--- a/core/src/test/java/org/springframework/security/authentication/DelegatingReactiveAuthenticationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/DelegatingReactiveAuthenticationManagerTests.java
@@ -51,7 +51,7 @@
 	@Test
 	public void authenticateWhenEmptyAndNotThenReturnsNotEmpty() {
 		given(this.delegate1.authenticate(any())).willReturn(Mono.empty());
-		given(this.delegate2.authenticate(any())).willReturn(Mono.just(this.authentication));
+		this.delegate2 = createReactiveAuthenticationManager(this.authentication);
 		DelegatingReactiveAuthenticationManager manager = new DelegatingReactiveAuthenticationManager(this.delegate1,
 				this.delegate2);
 		assertThat(manager.authenticate(this.authentication).block()).isEqualTo(this.authentication);
@@ -81,8 +81,8 @@
 	@Test
 	public void authenticateWhenContinueOnErrorAndFirstBadCredentialsThenTriesSecond() {
 		given(this.delegate1.authenticate(any())).willReturn(Mono.error(new BadCredentialsException("Test")));
-		given(this.delegate2.authenticate(any())).willReturn(Mono.just(this.authentication));
 
+		this.delegate2 = createReactiveAuthenticationManager(this.authentication);
 		DelegatingReactiveAuthenticationManager manager = managerWithContinueOnError();
 
 		assertThat(manager.authenticate(this.authentication).block()).isEqualTo(this.authentication);
@@ -102,8 +102,8 @@
 
 	@Test
 	public void authenticateWhenContinueOnErrorAndDelegate1NotEmptyThenReturnsNotEmpty() {
-		given(this.delegate1.authenticate(any())).willReturn(Mono.just(this.authentication));
 
+		this.delegate1 = createReactiveAuthenticationManager(this.authentication);
 		DelegatingReactiveAuthenticationManager manager = managerWithContinueOnError();
 
 		assertThat(manager.authenticate(this.authentication).block()).isEqualTo(this.authentication);
@@ -126,4 +126,10 @@
 		return manager;
 	}
 
+
+	private static ReactiveAuthenticationManager createReactiveAuthenticationManager(Authentication authentication) {
+		ReactiveAuthenticationManager mock = Mockito.mock(ReactiveAuthenticationManager.class);
+		given(mock.authenticate(any())).willReturn(Mono.just(authentication));
+		return mock;
+	}
 }
