--- a/core/src/test/java/org/springframework/security/authentication/dao/DaoAuthenticationProviderTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/dao/DaoAuthenticationProviderTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.authentication.dao;
 
+import org.mockito.Mock;
 import java.security.SecureRandom;
 import java.util.ArrayList;
 import java.util.List;
@@ -79,6 +80,8 @@
  * @author Rob Winch
  */
 public class DaoAuthenticationProviderTests {
+	@Mock
+	private UserDetailsPasswordService passwordManager;
 
 	private static final List<GrantedAuthority> ROLES_12 = AuthorityUtils.createAuthorityList("ROLE_ONE", "ROLE_TWO");
 
@@ -285,7 +288,6 @@
 				"password");
 		PasswordEncoder encoder = mock(PasswordEncoder.class);
 		UserDetailsService userDetailsService = mock(UserDetailsService.class);
-		UserDetailsPasswordService passwordManager = mock(UserDetailsPasswordService.class);
 		DaoAuthenticationProvider provider = new DaoAuthenticationProvider(userDetailsService);
 		provider.setPasswordEncoder(encoder);
 		provider.setUserDetailsPasswordService(passwordManager);
@@ -302,7 +304,6 @@
 				"password");
 		PasswordEncoder encoder = mock(PasswordEncoder.class);
 		UserDetailsService userDetailsService = mock(UserDetailsService.class);
-		UserDetailsPasswordService passwordManager = mock(UserDetailsPasswordService.class);
 		DaoAuthenticationProvider provider = new DaoAuthenticationProvider(userDetailsService);
 		provider.setPasswordEncoder(encoder);
 		provider.setUserDetailsPasswordService(passwordManager);
