--- a/core/src/test/java/org/springframework/security/authorization/AuthorizationManagerFactoryTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthorizationManagerFactoryTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.authorization;
 
+import org.mockito.Mock;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.security.authentication.TestAuthentication;
@@ -37,6 +38,8 @@
  * @author Steve Riesenberg
  */
 public class AuthorizationManagerFactoryTests {
+	@Mock
+	private AuthorizationManager<String> additional;
 
 	@Test
 	public void permitAllReturnsSingleResultAuthorizationManagerByDefault() {
@@ -124,7 +127,6 @@
 
 	@Test
 	public void anonymousWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -135,7 +137,6 @@
 
 	@Test
 	public void permitAllWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
@@ -146,7 +147,6 @@
 
 	@Test
 	public void denyAllAllWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
 		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setAdditionalAuthorization(additional);
 
