--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.config.annotation.web.configurers.oauth2.server.resource;
 
+import org.mockito.Mock;
 import java.io.BufferedReader;
 import java.io.FileReader;
 import java.io.IOException;
@@ -184,6 +185,8 @@
  */
 @ExtendWith(SpringTestContextExtension.class)
 public class OAuth2ResourceServerConfigurerTests {
+	@Mock
+	private AuthenticationConverter converter;
 
 	private static final String JWT_TOKEN = "token";
 
@@ -1421,7 +1424,6 @@
 
 	@Test
 	public void getAuthenticationConverterWhenDuplicateConverterBeansAndAnotherOnTheDslThenTheDslOneIsUsed() {
-		AuthenticationConverter converter = mock(AuthenticationConverter.class);
 		AuthenticationConverter converterBean = mock(AuthenticationConverter.class);
 		GenericWebApplicationContext context = new GenericWebApplicationContext();
 		context.registerBean("converterOne", AuthenticationConverter.class, () -> converterBean);
@@ -1434,7 +1436,6 @@
 
 	@Test
 	public void getAuthenticationConverterWhenConverterBeanAndAnotherOnTheDslThenTheDslOneIsUsed() {
-		AuthenticationConverter converter = mock(AuthenticationConverter.class);
 		AuthenticationConverter converterBean = mock(AuthenticationConverter.class);
 		GenericWebApplicationContext context = new GenericWebApplicationContext();
 		context.registerBean(AuthenticationConverter.class, () -> converterBean);
