--- a/taglibs/src/test/java/org/springframework/security/taglibs/authz/AbstractAuthorizeTagTests.java
+++ b/taglibs/src/test/java/org/springframework/security/taglibs/authz/AbstractAuthorizeTagTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.taglibs.authz;
 
+import org.mockito.Mock;
 import java.io.IOException;
 import java.util.Collections;
 
@@ -54,6 +55,8 @@
  *
  */
 public class AbstractAuthorizeTagTests {
+	@Mock
+	private WebInvocationPrivilegeEvaluator expected;
 
 	private AbstractAuthorizeTag tag;
 
@@ -82,7 +85,6 @@
 		this.servletContext.setAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE, wac);
 		given(wac.getBeanNamesForType(SecurityContextHolderStrategy.class)).willReturn(new String[0]);
 		String uri = "/something";
-		WebInvocationPrivilegeEvaluator expected = mock(WebInvocationPrivilegeEvaluator.class);
 		this.tag.setUrl(uri);
 		this.request.setAttribute(WebAttributes.WEB_INVOCATION_PRIVILEGE_EVALUATOR_ATTRIBUTE, expected);
 		this.tag.authorizeUsingUrlCheck();
@@ -99,7 +101,6 @@
 		wac.refresh();
 		this.servletContext.setAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE, wac);
 		String uri = "/something";
-		WebInvocationPrivilegeEvaluator expected = mock(WebInvocationPrivilegeEvaluator.class);
 		this.tag.setUrl(uri);
 		this.request.setAttribute(WebAttributes.WEB_INVOCATION_PRIVILEGE_EVALUATOR_ATTRIBUTE, expected);
 		this.tag.authorizeUsingUrlCheck();
@@ -110,7 +111,6 @@
 	@Test
 	public void privilegeEvaluatorFromChildContext() throws IOException {
 		String uri = "/something";
-		WebInvocationPrivilegeEvaluator expected = mock(WebInvocationPrivilegeEvaluator.class);
 		this.tag.setUrl(uri);
 		WebApplicationContext wac = mock(WebApplicationContext.class);
 		given(wac.getBeansOfType(WebInvocationPrivilegeEvaluator.class))
