--- a/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/authentication/OpaqueTokenReactiveAuthenticationManagerTests.java
+++ b/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/authentication/OpaqueTokenReactiveAuthenticationManagerTests.java
@@ -59,8 +59,7 @@
 	public void authenticateWhenActiveTokenThenOk() throws Exception {
 		OAuth2AuthenticatedPrincipal authority = TestOAuth2AuthenticatedPrincipals
 			.active((attributes) -> attributes.put("extension_field", "twenty-seven"));
-		ReactiveOpaqueTokenIntrospector introspector = mock(ReactiveOpaqueTokenIntrospector.class);
-		given(introspector.introspect(any())).willReturn(Mono.just(authority));
+		ReactiveOpaqueTokenIntrospector introspector = createMockReactiveOpaqueTokenIntrospector(authority);
 		OpaqueTokenReactiveAuthenticationManager provider = new OpaqueTokenReactiveAuthenticationManager(introspector);
 		Authentication result = provider.authenticate(new BearerTokenAuthenticationToken("token")).block();
 		assertThat(result.getPrincipal()).isInstanceOf(OAuth2IntrospectionAuthenticatedPrincipal.class);
@@ -87,8 +86,7 @@
 	public void authenticateWhenMissingScopeAttributeThenNoAuthorities() {
 		OAuth2AuthenticatedPrincipal authority = new OAuth2IntrospectionAuthenticatedPrincipal(
 				Collections.singletonMap("claim", "value"), null);
-		ReactiveOpaqueTokenIntrospector introspector = mock(ReactiveOpaqueTokenIntrospector.class);
-		given(introspector.introspect(any())).willReturn(Mono.just(authority));
+		ReactiveOpaqueTokenIntrospector introspector = createMockReactiveOpaqueTokenIntrospector(authority);
 		OpaqueTokenReactiveAuthenticationManager provider = new OpaqueTokenReactiveAuthenticationManager(introspector);
 		Authentication result = provider.authenticate(new BearerTokenAuthenticationToken("token")).block();
 		assertThat(result.getPrincipal()).isInstanceOf(OAuth2IntrospectionAuthenticatedPrincipal.class);
@@ -149,4 +147,10 @@
 		return (a) -> a.getAuthority().startsWith("SCOPE_");
 	}
 
+
+	private static ReactiveOpaqueTokenIntrospector createMockReactiveOpaqueTokenIntrospector(OAuth2AuthenticatedPrincipal authority) {
+		ReactiveOpaqueTokenIntrospector introspector = mock(ReactiveOpaqueTokenIntrospector.class);
+		given(introspector.introspect(any())).willReturn(Mono.just(authority));
+		return introspector;
+	}
 }
