--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/token/DelegatingOAuth2TokenGeneratorTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/token/DelegatingOAuth2TokenGeneratorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.oauth2.server.authorization.token;
 
+import org.mockito.Mock;
 import java.time.Instant;
 
 import org.junit.jupiter.api.Test;
@@ -35,6 +36,8 @@
  * @author Joe Grandja
  */
 public class DelegatingOAuth2TokenGeneratorTests {
+	@Mock
+	private OAuth2TokenGenerator<OAuth2Token> tokenGenerator2;
 
 	@Test
 	@SuppressWarnings("unchecked")
@@ -56,7 +59,6 @@
 	@SuppressWarnings("unchecked")
 	public void generateWhenTokenGeneratorSupportedThenReturnToken() {
 		OAuth2TokenGenerator<OAuth2Token> tokenGenerator1 = mock(OAuth2TokenGenerator.class);
-		OAuth2TokenGenerator<OAuth2Token> tokenGenerator2 = mock(OAuth2TokenGenerator.class);
 		OAuth2TokenGenerator<OAuth2Token> tokenGenerator3 = mock(OAuth2TokenGenerator.class);
 
 		OAuth2AccessToken accessToken = new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER, "access-token",
@@ -74,7 +76,6 @@
 	@SuppressWarnings("unchecked")
 	public void generateWhenTokenGeneratorNotSupportedThenReturnNull() {
 		OAuth2TokenGenerator<OAuth2Token> tokenGenerator1 = mock(OAuth2TokenGenerator.class);
-		OAuth2TokenGenerator<OAuth2Token> tokenGenerator2 = mock(OAuth2TokenGenerator.class);
 		OAuth2TokenGenerator<OAuth2Token> tokenGenerator3 = mock(OAuth2TokenGenerator.class);
 
 		DelegatingOAuth2TokenGenerator delegatingTokenGenerator = new DelegatingOAuth2TokenGenerator(tokenGenerator1,
