--- a/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospectorTests.java
+++ b/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospectorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.oauth2.server.resource.introspection;
 
+import org.mockito.Mock;
 import java.io.IOException;
 import java.time.Instant;
 import java.util.Arrays;
@@ -58,6 +59,8 @@
  * Tests for {@link SpringOpaqueTokenIntrospector}
  */
 public class SpringOpaqueTokenIntrospectorTests {
+	@Mock
+	private RequestEntity requestEntity;
 
 	private static final ParameterizedTypeReference<Map<String, Object>> STRING_OBJECT_MAP = new ParameterizedTypeReference<Map<String, Object>>() {
 	};
@@ -299,7 +302,6 @@
 	public void setRequestEntityConverterWhenNonNullConverterGivenThenConverterUsed() {
 		RestOperations restOperations = mock(RestOperations.class);
 		Converter<String, RequestEntity<?>> requestEntityConverter = mock(Converter.class);
-		RequestEntity requestEntity = mock(RequestEntity.class);
 		String tokenToIntrospect = "some token";
 		given(requestEntityConverter.convert(tokenToIntrospect)).willReturn(requestEntity);
 		given(restOperations.exchange(requestEntity, STRING_OBJECT_MAP)).willReturn(ACTIVE);
@@ -323,7 +325,6 @@
 	public void setAuthenticationConverterWhenNonNullConverterGivenThenConverterUsed() {
 		RestOperations restOperations = mock(RestOperations.class);
 		Converter<String, RequestEntity<?>> requestEntityConverter = mock(Converter.class);
-		RequestEntity requestEntity = mock(RequestEntity.class);
 		Converter<OAuth2TokenIntrospectionClaimAccessor, OAuth2AuthenticatedPrincipal> authenticationConverter = mock(
 				Converter.class);
 		OAuth2AuthenticatedPrincipal oAuth2AuthenticatedPrincipal = mock(OAuth2AuthenticatedPrincipal.class);
