--- a/kerberos/kerberos-web/src/test/java/org/springframework/security/kerberos/web/SpnegoEntryPointTests.java
+++ b/kerberos/kerberos-web/src/test/java/org/springframework/security/kerberos/web/SpnegoEntryPointTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.kerberos.web;
 
+import org.mockito.Mock;
 import jakarta.servlet.RequestDispatcher;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
@@ -43,6 +44,8 @@
  * @since 1.0
  */
 public class SpnegoEntryPointTests {
+	@Mock
+	private RequestDispatcher requestDispatcher;
 
 	private SpnegoEntryPoint entryPoint = new SpnegoEntryPoint();
 
@@ -62,7 +65,6 @@
 		SpnegoEntryPoint entryPoint = new SpnegoEntryPoint();
 		HttpServletResponse response = mock(HttpServletResponse.class);
 		HttpServletRequest request = mock(HttpServletRequest.class);
-		RequestDispatcher requestDispatcher = mock(RequestDispatcher.class);
 		given(request.getRequestDispatcher(anyString())).willReturn(requestDispatcher);
 		entryPoint.commence(request, response, null);
 		verify(response).addHeader("WWW-Authenticate", "Negotiate");
@@ -75,7 +77,6 @@
 		SpnegoEntryPoint entryPoint = new SpnegoEntryPoint(forwardUrl);
 		HttpServletResponse response = mock(HttpServletResponse.class);
 		HttpServletRequest request = mock(HttpServletRequest.class);
-		RequestDispatcher requestDispatcher = mock(RequestDispatcher.class);
 		given(request.getRequestDispatcher(anyString())).willReturn(requestDispatcher);
 		entryPoint.commence(request, response, null);
 		verify(response).addHeader("WWW-Authenticate", "Negotiate");
@@ -92,7 +93,6 @@
 		HttpServletResponse response = mock(HttpServletResponse.class);
 		HttpServletRequest request = mock(HttpServletRequest.class);
 		given(request.getMethod()).willReturn(RequestMethod.POST.name());
-		RequestDispatcher requestDispatcher = mock(RequestDispatcher.class);
 		given(request.getRequestDispatcher(anyString())).willReturn(requestDispatcher);
 		entryPoint.commence(request, response, null);
 		verify(requestDispatcher).forward(servletRequestCaptor.capture(), eq(response));
@@ -106,7 +106,6 @@
 		SpnegoEntryPoint entryPoint = new SpnegoEntryPoint(forwardUrl, HttpMethod.DELETE);
 		HttpServletResponse response = mock(HttpServletResponse.class);
 		HttpServletRequest request = mock(HttpServletRequest.class);
-		RequestDispatcher requestDispatcher = mock(RequestDispatcher.class);
 		given(request.getRequestDispatcher(anyString())).willReturn(requestDispatcher);
 		entryPoint.commence(request, response, null);
 		verify(requestDispatcher).forward(servletRequestCaptor.capture(), eq(response));
