--- a/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLoginTests.java
+++ b/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLoginTests.java
@@ -94,8 +94,7 @@
 	 */
 	@Test
 	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
-		RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
-		given(postProcessor.postProcessRequest(any())).willAnswer((i) -> i.getArgument(0));
+		RequestPostProcessor postProcessor = MockRequestPostProcessor.createRequestPostProcessor();
 		MockMvc mockMvc = MockMvcBuilders.standaloneSetup(new Object())
 			.defaultRequest(MockMvcRequestBuilders.get("/").with(postProcessor))
 			.build();
@@ -119,4 +118,17 @@
 		assertThat(request.getHeader("Accept")).isEqualTo(MediaType.APPLICATION_FORM_URLENCODED_VALUE);
 	}
 
+
+	public final class MockRequestPostProcessor {
+
+		private MockRequestPostProcessor() {
+		}
+
+		public static RequestPostProcessor createRequestPostProcessor() {
+			RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
+			given(postProcessor.postProcessRequest(any()))
+					.willAnswer((i) -> i.getArgument(0));
+			return postProcessor;
+		}
+	}
 }
--- a/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLogoutTests.java
+++ b/test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLogoutTests.java
@@ -83,8 +83,7 @@
 	 */
 	@Test
 	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
-		RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
-		given(postProcessor.postProcessRequest(any())).willAnswer((i) -> i.getArgument(0));
+		RequestPostProcessor postProcessor = MockRequestPostProcessor.createRequestPostProcessor();
 		MockMvc mockMvc = MockMvcBuilders.standaloneSetup(new Object())
 			.defaultRequest(MockMvcRequestBuilders.get("/").with(postProcessor))
 			.build();
@@ -97,4 +96,17 @@
 		verify(postProcessor).postProcessRequest(any());
 	}
 
+
+	public final class MockRequestPostProcessor {
+
+		private MockRequestPostProcessor() {
+		}
+
+		public static RequestPostProcessor createRequestPostProcessor() {
+			RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
+			given(postProcessor.postProcessRequest(any()))
+					.willAnswer((i) -> i.getArgument(0));
+			return postProcessor;
+		}
+	}
 }
