--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientJwtBearerTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientJwtBearerTokenResponseClientTests.java
@@ -432,8 +432,7 @@
 	@Test
 	public void getTokenResponseWhenRestClientSetThenCalled() {
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
-		RestClient customClient = mock();
-		given(customClient.post()).willReturn(RestClient.builder().build().post());
+		RestClient customClient = MockRestClient.createRestClient();
 		this.tokenResponseClient.setRestClient(customClient);
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest grantRequest = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
@@ -445,4 +444,16 @@
 		return "%s=%s".formatted(parameterName, URLEncoder.encode(parameterValue, StandardCharsets.UTF_8));
 	}
 
+
+	public final class MockRestClient {
+
+		private MockRestClient() {
+		}
+
+		public static RestClient createRestClient() {
+			RestClient customClient = Mockito.mock(RestClient.class);
+			given(customClient.post()).willReturn(RestClient.builder().build().post());
+			return customClient;
+		}
+	}
 }
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientTokenExchangeTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientTokenExchangeTokenResponseClientTests.java
@@ -563,8 +563,7 @@
 	@Test
 	public void getTokenResponseWhenRestClientSetThenCalled() {
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
-		RestClient customClient = mock();
-		given(customClient.post()).willReturn(RestClient.builder().build().post());
+		RestClient customClient = MockRestClient.createRestClient();
 		this.tokenResponseClient.setRestClient(customClient);
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		TokenExchangeGrantRequest grantRequest = new TokenExchangeGrantRequest(clientRegistration, this.subjectToken,
@@ -577,4 +576,16 @@
 		return "%s=%s".formatted(parameterName, URLEncoder.encode(parameterValue, StandardCharsets.UTF_8));
 	}
 
+
+	public final class MockRestClient {
+
+		private MockRestClient() {
+		}
+
+		public static RestClient createRestClient() {
+			RestClient customClient = Mockito.mock(RestClient.class);
+			given(customClient.post()).willReturn(RestClient.builder().build().post());
+			return customClient;
+		}
+	}
 }
