--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/server/WebSessionServerOAuth2AuthorizedClientRepositoryTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/server/WebSessionServerOAuth2AuthorizedClientRepositoryTests.java
@@ -207,8 +207,7 @@
 
 	@Test
 	public void saveAuthorizedClientWhenSessionIsNullThenThrowIllegalArgumentException() {
-		ServerWebExchange exchange = mock(ServerWebExchange.class);
-		given(exchange.getSession()).willReturn(Mono.empty());
+		ServerWebExchange exchange = createServerWebExchange();
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.registration1, this.principalName1,
 				mock(OAuth2AccessToken.class));
 		// @formatter:off
@@ -220,8 +219,7 @@
 
 	@Test
 	public void removeAuthorizedClientWhenSessionIsNullThenThrowIllegalArgumentException() {
-		ServerWebExchange exchange = mock(ServerWebExchange.class);
-		given(exchange.getSession()).willReturn(Mono.empty());
+		ServerWebExchange exchange = createServerWebExchange();
 		// @formatter:off
 		assertThatIllegalArgumentException()
 			.isThrownBy(() -> this.authorizedClientRepository.removeAuthorizedClient(this.registrationId1, null, exchange).block())
@@ -229,4 +227,10 @@
 		// @formatter:on
 	}
 
+
+	private static ServerWebExchange createServerWebExchange() {
+		ServerWebExchange exchange = mock(ServerWebExchange.class);
+		given(exchange.getSession()).willReturn(Mono.empty());
+		return exchange;
+	}
 }
