--- a/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/access/server/BearerTokenServerAccessDeniedHandlerTests.java
+++ b/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/access/server/BearerTokenServerAccessDeniedHandlerTests.java
@@ -48,9 +48,7 @@
 	@Test
 	public void handleWhenNotOAuth2AuthenticatedThenStatus403() {
 		Authentication token = new TestingAuthenticationToken("user", "pass");
-		ServerWebExchange exchange = mock(ServerWebExchange.class);
-		given(exchange.getPrincipal()).willReturn(Mono.just(token));
-		given(exchange.getResponse()).willReturn(new MockServerHttpResponse());
+		ServerWebExchange exchange = createServerWebExchange(token);
 		this.accessDeniedHandler.handle(exchange, null).block();
 		assertThat(exchange.getResponse().getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
 		assertThat(exchange.getResponse().getHeaders().get("WWW-Authenticate")).isEqualTo(Arrays.asList("Bearer"));
@@ -59,9 +57,7 @@
 	@Test
 	public void handleWhenNotOAuth2AuthenticatedAndRealmSetThenStatus403AndAuthHeaderWithRealm() {
 		Authentication token = new TestingAuthenticationToken("user", "pass");
-		ServerWebExchange exchange = mock(ServerWebExchange.class);
-		given(exchange.getPrincipal()).willReturn(Mono.just(token));
-		given(exchange.getResponse()).willReturn(new MockServerHttpResponse());
+		ServerWebExchange exchange = createServerWebExchange(token);
 		this.accessDeniedHandler.setRealmName("test");
 		this.accessDeniedHandler.handle(exchange, null).block();
 		assertThat(exchange.getResponse().getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
@@ -72,9 +68,7 @@
 	@Test
 	public void handleWhenOAuth2AuthenticatedThenStatus403AndAuthHeaderWithInsufficientScopeErrorAttribute() {
 		Authentication token = new TestingOAuth2TokenAuthenticationToken(Collections.emptyMap());
-		ServerWebExchange exchange = mock(ServerWebExchange.class);
-		given(exchange.getPrincipal()).willReturn(Mono.just(token));
-		given(exchange.getResponse()).willReturn(new MockServerHttpResponse());
+		ServerWebExchange exchange = createServerWebExchange(token);
 		this.accessDeniedHandler.handle(exchange, null).block();
 		assertThat(exchange.getResponse().getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
 		// @formatter:off
@@ -115,4 +109,11 @@
 
 	}
 
+
+	private static ServerWebExchange createServerWebExchange(Authentication token) {
+		ServerWebExchange exchange = mock(ServerWebExchange.class);
+		given(exchange.getPrincipal()).willReturn(Mono.just(token));
+		given(exchange.getResponse()).willReturn(new MockServerHttpResponse());
+		return exchange;
+	}
 }
