--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -112,6 +112,7 @@
     private String dhcpProvider = "VirtualRouter";
     private NetworkGuru guru = mock(NetworkGuru.class);
     private NetworkExtensionElement networkExtensionElement;
+    private ReservationContext shutdownReservationContext;
 
     NetworkOfferingVO networkOffering = mock(NetworkOfferingVO.class);
 
@@ -126,6 +127,7 @@
     @Before
     public void setUp() {
         // make class-scope mocks
+        shutdownReservationContext = Mockito.mock(ReservationContext.class);
         testOrchestrator._nicDao = mock(NicDao.class);
         testOrchestrator._networksDao = mock(NetworkDao.class);
         testOrchestrator._networkModel = mock(NetworkModel.class);
@@ -862,12 +864,11 @@
 
     @Test
     public void testShutdownNetworkAcquireLockFailed() {
-        ReservationContext reservationContext = Mockito.mock(ReservationContext.class);
         NetworkVO network = mock(NetworkVO.class);
         long networkId = 1;
         when(testOrchestrator._networksDao.acquireInLockTable(Mockito.anyLong(), Mockito.anyInt())).thenReturn(null);
 
-        boolean shutdownNetworkStatus = testOrchestrator.shutdownNetwork(networkId, reservationContext, false);
+        boolean shutdownNetworkStatus = testOrchestrator.shutdownNetwork(networkId, shutdownReservationContext, false);
         Assert.assertFalse(shutdownNetworkStatus);
 
         verify(testOrchestrator._networksDao, times(1)).acquireInLockTable(networkId, NetworkLockTimeout.value());
@@ -875,14 +876,13 @@
 
     @Test
     public void testShutdownNetworkInAllocatedState() {
-        ReservationContext reservationContext = Mockito.mock(ReservationContext.class);
         NetworkVO network = mock(NetworkVO.class);
         long networkId = 1;
         when(testOrchestrator._networksDao.acquireInLockTable(Mockito.anyLong(), Mockito.anyInt())).thenReturn(network);
         when(network.getId()).thenReturn(networkId);
         when(network.getState()).thenReturn(Network.State.Allocated);
 
-        boolean shutdownNetworkStatus = testOrchestrator.shutdownNetwork(networkId, reservationContext, false);
+        boolean shutdownNetworkStatus = testOrchestrator.shutdownNetwork(networkId, shutdownReservationContext, false);
         Assert.assertTrue(shutdownNetworkStatus);
 
         verify(network, times(1)).getState();
@@ -892,14 +892,13 @@
 
     @Test
     public void testShutdownNetworkInImplementingState() {
-        ReservationContext reservationContext = Mockito.mock(ReservationContext.class);
         NetworkVO network = mock(NetworkVO.class);
         long networkId = 1;
         when(testOrchestrator._networksDao.acquireInLockTable(Mockito.anyLong(), Mockito.anyInt())).thenReturn(network);
         when(network.getId()).thenReturn(networkId);
         when(network.getState()).thenReturn(Network.State.Implementing);
 
-        boolean shutdownNetworkStatus = testOrchestrator.shutdownNetwork(networkId, reservationContext, false);
+        boolean shutdownNetworkStatus = testOrchestrator.shutdownNetwork(networkId, shutdownReservationContext, false);
         Assert.assertFalse(shutdownNetworkStatus);
 
         verify(network).getState();
