--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -89,7 +89,7 @@
     NetworkOfferingServiceMapDao nosd = mock(NetworkOfferingServiceMapDao.class);
     AgentManager agentmgr = mock(AgentManager.class);
     NetworkOrchestrationService netmgr = mock(NetworkOrchestrationService.class);
-    NetworkModel netmodel = mock(NetworkModel.class);
+    NetworkModel netmodel;
 
     HostDao hostdao = mock(HostDao.class);
     NetworkDao netdao = mock(NetworkDao.class);
@@ -164,7 +164,7 @@
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(true);
 
-        when(netmodel.listNetworkOfferingServices(NETWORK_ID)).thenReturn(Arrays.asList(Service.Connectivity));
+        netmodel = MockNetworkModelWithOfferingServices.createMockNetworkModelWithOfferingServices(NETWORK_ID);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -80,7 +80,7 @@
     NetworkOfferingServiceMapDao nosd = mock(NetworkOfferingServiceMapDao.class);
     AgentManager agentmgr = mock(AgentManager.class);
     NetworkOrchestrationService netmgr = mock(NetworkOrchestrationService.class);
-    NetworkModel netmodel = mock(NetworkModel.class);
+    NetworkModel netmodel = MockNetworkModelWithOfferingServices.createMockNetworkModelWithOfferingServices(NETWORK_ID);
 
     HostDao hostdao = mock(HostDao.class);
     NetworkDao netdao = mock(NetworkDao.class);
@@ -162,7 +162,7 @@
 
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(true);
 
-        when(netmodel.listNetworkOfferingServices(NETWORK_ID)).thenReturn(Arrays.asList(Service.Connectivity));
+
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
         final Network network = mock(Network.class);
--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockNetworkModelWithOfferingServices.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockNetworkModelWithOfferingServices.java
@@ -0,0 +1,17 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+
+import com.cloud.network.Network.Service;
+import com.cloud.network.NetworkModel;
+
+public class MockNetworkModelWithOfferingServices {
+    public static NetworkModel createMockNetworkModelWithOfferingServices(final long networkId) {
+        final NetworkModel netmodel = mock(NetworkModel.class);
+        when(netmodel.listNetworkOfferingServices(networkId)).thenReturn(Arrays.asList(Service.Connectivity));
+        return netmodel;
+    }
+}
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockNetworkModelWithOfferingServices.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockNetworkModelWithOfferingServices.java
@@ -0,0 +1,17 @@
+package com.cloud.network.guru;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+
+import com.cloud.network.Network.Service;
+import com.cloud.network.NetworkModel;
+
+public class MockNetworkModelWithOfferingServices {
+    public static NetworkModel createMockNetworkModelWithOfferingServices(final long networkId) {
+        final NetworkModel netmodel = mock(NetworkModel.class);
+        when(netmodel.listNetworkOfferingServices(networkId)).thenReturn(Arrays.asList(Service.Connectivity));
+        return netmodel;
+    }
+}
