--- a/server/src/test/java/com/cloud/network/DedicateGuestVlanRangesTest.java
+++ b/server/src/test/java/com/cloud/network/DedicateGuestVlanRangesTest.java
@@ -188,7 +188,7 @@
         physicalNetwork.addIsolationMethod("VLAN");
         AccountGuestVlanMapVO accountGuestVlanMapVO = new AccountGuestVlanMapVO(1L, 1L);
 
-        when(networkService._physicalNetworkDao.findById(anyLong())).thenReturn(physicalNetwork);
+        networkService._physicalNetworkDao = MockPhysicalNetworkDaoFactory.createMockPhysicalNetworkDao(physicalNetwork);
 
         when(networkService._dcVnetDao.listAllocatedVnetsInRange(anyLong(), anyLong(), anyInt(), anyInt())).thenReturn(null);
 
--- a/server/src/test/java/com/cloud/network/UpdatePhysicalNetworkTest.java
+++ b/server/src/test/java/com/cloud/network/UpdatePhysicalNetworkTest.java
@@ -36,7 +36,7 @@
 import static org.mockito.Mockito.when;
 
 public class UpdatePhysicalNetworkTest {
-    private PhysicalNetworkDao _physicalNetworkDao = mock(PhysicalNetworkDao.class);
+    private PhysicalNetworkDao _physicalNetworkDao;
     private DataCenterVnetDao _datacenterVnetDao = mock(DataCenterVnetDao.class);
     private DataCenterDao _datacenterDao = mock(DataCenterDao.class);
     private DataCenterVO datacentervo = mock(DataCenterVO.class);
@@ -57,7 +57,7 @@
         TransactionLegacy txn = TransactionLegacy.open("updatePhysicalNetworkTest");
         NetworkServiceImpl networkService = setUp();
         existingRange.add("524");
-        when(_physicalNetworkDao.findById(anyLong())).thenReturn(physicalNetworkVO);
+        _physicalNetworkDao = MockPhysicalNetworkDaoFactory.createMockPhysicalNetworkDao(physicalNetworkVO);
         when(_datacenterDao.findById(anyLong())).thenReturn(datacentervo);
         when(_physicalNetworkDao.update(anyLong(), any(physicalNetworkVO.getClass()))).thenReturn(true);
         when(_datacenterVnetDao.listVnetsByPhysicalNetworkAndDataCenter(anyLong(), anyLong())).thenReturn(existingRange);
--- a/server/src/test/java/com/cloud/network/MockPhysicalNetworkDaoFactory.java
+++ b/server/src/test/java/com/cloud/network/MockPhysicalNetworkDaoFactory.java
@@ -0,0 +1,17 @@
+package com.cloud.network;
+
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+import com.cloud.network.dao.PhysicalNetworkDao;
+import com.cloud.network.dao.PhysicalNetworkVO;
+
+public class MockPhysicalNetworkDaoFactory {
+    public static PhysicalNetworkDao createMockPhysicalNetworkDao(PhysicalNetworkVO physicalNetwork) {
+        PhysicalNetworkDao physicalNetworkDao = Mockito.mock(PhysicalNetworkDao.class);
+        when(physicalNetworkDao.findById(anyLong())).thenReturn(physicalNetwork);
+        return physicalNetworkDao;
+    }
+}
