--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
@@ -1109,6 +1109,12 @@
                 Mockito.eq(previousAclRuleMock), Mockito.eq(nextAclRuleMock));
     }
 
+    private static NetworkACLItemVO createMockNetworkAclItemVO() {
+        NetworkACLItemVO networkAclItemVOMock = Mockito.mock(NetworkACLItemVO.class);
+        Mockito.when(networkAclItemVOMock.getAclId()).thenReturn(99L);
+        return networkAclItemVOMock;
+    }
+
     @Test
     public void retrieveAndValidateAclRuleTestUuidNull() {
         NetworkACLItemVO networkACLItemVOReceived = networkAclServiceImpl.retrieveAndValidateAclRule(null);
@@ -1153,14 +1159,14 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void validateMoveAclRulesDataTestPreviousRuleWithDifferentAclId() {
-        Mockito.when(previousAclRuleMock.getAclId()).thenReturn(99L);
+        previousAclRuleMock = createMockNetworkAclItemVO();
 
         networkAclServiceImpl.validateMoveAclRulesData(aclRuleBeingMovedMock, previousAclRuleMock, null);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void validateMoveAclRulesDataTestNextRuleWithDifferentAclId() {
-        Mockito.when(nextAclRuleMock.getAclId()).thenReturn(99L);
+        nextAclRuleMock = createMockNetworkAclItemVO();
 
         networkAclServiceImpl.validateMoveAclRulesData(aclRuleBeingMovedMock, null, nextAclRuleMock);
     }
