--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -82,7 +82,7 @@
 
 public class BrocadeVcsGuestNetworkGuruTest {
     private static final long NETWORK_ID = 42L;
-    PhysicalNetworkDao physnetdao = mock(PhysicalNetworkDao.class);
+    PhysicalNetworkDao physnetdao = MockPhysicalNetworkDao.createMockPhysicalNetworkDao();
     BrocadeVcsDao vcsdao = mock(BrocadeVcsDao.class);
     BrocadeVcsNetworkVlanMappingDao vcsmapdao = mock(BrocadeVcsNetworkVlanMappingDao.class);
     DataCenterDao dcdao = mock(DataCenterDao.class);
@@ -178,6 +178,7 @@
     @Test
     public void testDesignNoIsolationMethodVCS() {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
+        physnetdao = MockPhysicalNetworkDao.createMockPhysicalNetworkDao();
         when(physnetdao.findById((Long)any())).thenReturn(physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
@@ -220,6 +221,7 @@
     @Test
     public void testImplement() throws InsufficientVirtualNetworkCapacityException, URISyntaxException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
+        physnetdao = MockPhysicalNetworkDao.createMockPhysicalNetworkDao();
         when(physnetdao.findById((Long)any())).thenReturn(physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"VCS"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -74,7 +74,7 @@
 
 public class NiciraNvpGuestNetworkGuruTest {
     private static final long NETWORK_ID = 42L;
-    PhysicalNetworkDao physnetdao = mock(PhysicalNetworkDao.class);
+    PhysicalNetworkDao physnetdao = MockPhysicalNetworkDao.createMockPhysicalNetworkDao();
     NiciraNvpDao nvpdao = mock(NiciraNvpDao.class);
     DataCenterDao dcdao = mock(DataCenterDao.class);
     NetworkOfferingServiceMapDao nosd = mock(NetworkOfferingServiceMapDao.class);
@@ -417,6 +417,7 @@
     @Test
     public void testShutdown() throws InsufficientVirtualNetworkCapacityException, URISyntaxException {
         final PhysicalNetworkVO physnet = mock(PhysicalNetworkVO.class);
+        physnetdao = MockPhysicalNetworkDao.createMockPhysicalNetworkDao();
         when(physnetdao.findById((Long)any())).thenReturn(physnet);
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
@@ -0,0 +1,11 @@
+package com.cloud.network.guru;
+
+import com.cloud.network.dao.PhysicalNetworkDao;
+
+import static org.mockito.Mockito.mock;
+
+public class MockPhysicalNetworkDao {
+    public static PhysicalNetworkDao createMockPhysicalNetworkDao() {
+        return mock(PhysicalNetworkDao.class);
+    }
+}
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/MockPhysicalNetworkDao.java
@@ -0,0 +1,11 @@
+package com.cloud.network.guru;
+
+import com.cloud.network.dao.PhysicalNetworkDao;
+
+import static org.mockito.Mockito.mock;
+
+public class MockPhysicalNetworkDao {
+    public static PhysicalNetworkDao createMockPhysicalNetworkDao() {
+        return mock(PhysicalNetworkDao.class);
+    }
+}
