--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
@@ -1109,6 +1109,12 @@
                 Mockito.eq(previousAclRuleMock), Mockito.eq(nextAclRuleMock));
     }
 
+    private static NetworkACLItemVO createMockPreviousAclRule() {
+        NetworkACLItemVO previousAclRuleMock = Mockito.mock(NetworkACLItemVO.class);
+        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
+        return previousAclRuleMock;
+    }
+
     @Test
     public void retrieveAndValidateAclRuleTestUuidNull() {
         NetworkACLItemVO networkACLItemVOReceived = networkAclServiceImpl.retrieveAndValidateAclRule(null);
@@ -1281,7 +1287,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void moveRuleBetweenAclRulesTestThereIsSpaceBetweenPreviousRuleAndNextRuleToAccommodateTheNewRuleWithOtherRuleColliding() {
-        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
+        previousAclRuleMock = createMockPreviousAclRule();
         Mockito.when(nextAclRuleMock.getNumber()).thenReturn(15);
 
         ArrayList<NetworkACLItemVO> allAclRules = new ArrayList<>();
@@ -1297,7 +1303,7 @@
 
     @Test
     public void moveRuleBetweenAclRulesTestThereIsSpaceBetweenPreviousRuleAndNextRuleToAccommodateTheNewRule() {
-        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
+        previousAclRuleMock = createMockPreviousAclRule();
         Mockito.when(nextAclRuleMock.getNumber()).thenReturn(11);
         Mockito.when(aclRuleBeingMovedMock.getId()).thenReturn(1l);
 
@@ -1329,7 +1335,7 @@
 
     @Test
     public void moveRuleBetweenAclRulesTestThereIsNoSpaceBetweenPreviousRuleAndNextRuleToAccommodateTheNewRule() {
-        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
+        previousAclRuleMock = createMockPreviousAclRule();
         Mockito.when(nextAclRuleMock.getNumber()).thenReturn(15);
         Mockito.when(aclRuleBeingMovedMock.getNumber()).thenReturn(50);
         Mockito.when(aclRuleBeingMovedMock.getId()).thenReturn(1l);
