--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -238,6 +238,9 @@
     GuestOSDao guestOSDao;
 
     @Mock
+    GuestOSCategoryVO guestOSCategoryMock;
+
+    @Mock
     ExtensionsManager extensionManager;
 
     @Spy
@@ -960,10 +963,9 @@
     @Test
     public void testDeleteGuestOsCategory_Successful() {
         DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
         Mockito.when(deleteCmd.getId()).thenReturn(id);
-        Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
+        Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategoryMock);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList());
         Mockito.when(guestOSCategoryDao.remove(id)).thenReturn(true);
         boolean result = spy.deleteGuestOsCategory(deleteCmd);
@@ -985,10 +987,9 @@
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteGuestOsCategory_ThrowsExceptionWhenGuestOsExists() {
         DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
         Mockito.when(deleteCmd.getId()).thenReturn(id);
-        Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
+        Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategoryMock);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList(1L));
         spy.deleteGuestOsCategory(deleteCmd);
     }
@@ -1003,7 +1004,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_Success() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Filter filter = Mockito.mock(Filter.class);
         Long id = 1L;
         String name = "Ubuntu";
@@ -1022,12 +1022,12 @@
         Mockito.when(listCmd.isIso()).thenReturn(isIso);
         Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
-        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
+        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategoryMock);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
         Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
         Mockito.when(templateDao.listTemplateIsoByArchVnfAndZone(zoneId, arch, isIso, isVnf)).thenReturn(Arrays.asList(1L, 2L));
-        Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(guestOSCategory), 1);
+        Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(guestOSCategoryMock), 1);
         mockGuestOsJoin();
         Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(searchCriteria), Mockito.any())).thenReturn(mockResult);
         Pair<List<? extends GuestOsCategory>, Integer> result = spy.listGuestOSCategoriesByCriteria(listCmd);
@@ -1042,7 +1042,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_NoResults() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         String name = "CentOS";
         String keyword = "Linux";
@@ -1060,7 +1059,7 @@
         Mockito.when(listCmd.isIso()).thenReturn(isIso);
         Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
-        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
+        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategoryMock);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
         Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
@@ -1080,7 +1079,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_NoGuestOsIdsFound() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         String name = "Ubuntu";
         String keyword = "Linux";
@@ -1098,7 +1096,7 @@
         Mockito.when(listCmd.isIso()).thenReturn(isIso);
         Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
-        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
+        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategoryMock);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
         Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
@@ -1118,18 +1116,17 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_FilterById() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         Mockito.when(listCmd.getId()).thenReturn(id);
         Mockito.when(listCmd.getZoneId()).thenReturn(null);
         Mockito.when(listCmd.isIso()).thenReturn(null);
         Mockito.when(listCmd.isVnf()).thenReturn(null);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
-        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
+        Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategoryMock);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
         Mockito.when(guestOSCategoryDao.createSearchBuilder()).thenReturn(searchBuilder);
         Mockito.when(searchBuilder.create()).thenReturn(searchCriteria);
-        Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(guestOSCategory), 1);
+        Pair<List<GuestOSCategoryVO>, Integer> mockResult = new Pair<>(Arrays.asList(guestOSCategoryMock), 1);
         Mockito.when(guestOSCategoryDao.searchAndCount(Mockito.eq(searchCriteria), Mockito.any())).thenReturn(mockResult);
         mockGuestOsJoin();
         Pair<List<? extends GuestOsCategory>, Integer> result = spy.listGuestOSCategoriesByCriteria(listCmd);
