--- a/server/src/test/java/com/cloud/network/firewall/FirewallManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/firewall/FirewallManagerImplTest.java
@@ -64,6 +64,9 @@
     @Mock
     private NetworkServiceMapDao networkServiceMapDao;
 
+    @Mock
+    private PortForwardingRule portForwardingRule;
+
     @Before
     public void setUp() {
         ReflectionTestUtils.setField(firewallManager, "_firewallElements", Collections.emptyList());
@@ -125,9 +128,8 @@
         Network network = mock(Network.class);
         when(network.getId()).thenReturn(3L);
 
-        PortForwardingRule rule = mock(PortForwardingRule.class);
         @SuppressWarnings("unchecked")
-        List<PortForwardingRule> rules = List.of(rule);
+        List<PortForwardingRule> rules = List.of(portForwardingRule);
 
         String extProviderName = "my-ext-pf-provider";
         when(networkServiceMapDao.getProviderForServiceInNetwork(3L, Network.Service.PortForwarding))
@@ -147,8 +149,7 @@
         Network network = mock(Network.class);
         when(network.getId()).thenReturn(4L);
 
-        PortForwardingRule rule = mock(PortForwardingRule.class);
-        List<? extends FirewallRule> rules = List.of(rule);
+        List<? extends FirewallRule> rules = List.of(portForwardingRule);
 
         when(networkServiceMapDao.getProviderForServiceInNetwork(4L, Network.Service.PortForwarding))
                 .thenReturn(null);
