--- a/server/src/test/java/org/apache/cloudstack/backup/BackupCompressionServiceJobControllerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupCompressionServiceJobControllerTest.java
@@ -106,6 +106,10 @@
     @Mock
     private ConfigKey<Boolean> backupCompressionTaskEnabledMock;
 
+    private static void mockBackupTaskEnabledConfigKey(ConfigKey<Boolean> backupTaskEnabledMock, boolean taskEnabled) {
+        doReturn(taskEnabled).when(backupTaskEnabledMock).value();
+    }
+
     @Spy
     @InjectMocks
     private BackupCompressionServiceJobController backupCompressionServiceJobControllerSpy;
@@ -474,7 +478,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupCompressionTaskEnabledMock).value();
+        mockBackupTaskEnabledConfigKey(backupCompressionTaskEnabledMock, true);
         doReturn(false).when(backupCompressionServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
         backupCompressionServiceJobControllerSpy.searchAndDispatchJobs();
@@ -488,7 +492,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupCompressionTaskEnabledMock).value();
+        mockBackupTaskEnabledConfigKey(backupCompressionTaskEnabledMock, true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupCompressionServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
@@ -506,7 +510,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupCompressionTaskEnabledMock).value();
+        mockBackupTaskEnabledConfigKey(backupCompressionTaskEnabledMock, true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupCompressionServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
         doReturn(List.of(internalBackupServiceJobVoMock)).when(internalBackupServiceJobDaoMock).listWaitingJobsAndScheduledToBeforeNow(eq(datacenterId),
--- a/server/src/test/java/org/apache/cloudstack/backup/BackupValidationServiceJobControllerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupValidationServiceJobControllerTest.java
@@ -63,6 +63,10 @@
     @Mock
     private ConfigKey<Boolean> backupValidationTaskEnabledMock;
 
+    private static void mockBackupTaskEnabledConfigKey(ConfigKey<Boolean> backupTaskEnabledMock, boolean taskEnabled) {
+        doReturn(taskEnabled).when(backupTaskEnabledMock).value();
+    }
+
     @Spy
     @InjectMocks
     private BackupValidationServiceJobController backupValidationServiceJobControllerSpy;
@@ -100,7 +104,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupValidationTaskEnabledMock).value();
+        mockBackupTaskEnabledConfigKey(backupValidationTaskEnabledMock, true);
         doReturn(false).when(backupValidationServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
         backupValidationServiceJobControllerSpy.searchAndDispatchJobs();
@@ -114,7 +118,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupValidationTaskEnabledMock).value();
+        mockBackupTaskEnabledConfigKey(backupValidationTaskEnabledMock, true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupValidationServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
@@ -132,7 +136,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupValidationTaskEnabledMock).value();
+        mockBackupTaskEnabledConfigKey(backupValidationTaskEnabledMock, true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupValidationServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
         doReturn(List.of(internalBackupServiceJobVoMock)).when(backupValidationServiceJobControllerSpy).filterJobsOfDomainsAndAccountsWithDisabledValidationTask(any());
