--- a/access/src/test/java/org/springframework/security/acls/afterinvocation/AclEntryAfterInvocationProviderTests.java
+++ b/access/src/test/java/org/springframework/security/acls/afterinvocation/AclEntryAfterInvocationProviderTests.java
@@ -20,6 +20,7 @@
 import java.util.Collections;
 import java.util.List;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.security.access.AccessDeniedException;
@@ -51,6 +52,13 @@
 @SuppressWarnings({ "unchecked", "deprecation" })
 public class AclEntryAfterInvocationProviderTests {
 
+	private AclService aclService;
+
+	@BeforeEach
+	public void setUp() {
+		aclService = mock(AclService.class);
+	}
+
 	@Test
 	public void rejectsMissingPermissions() {
 		assertThatIllegalArgumentException()
@@ -78,7 +86,7 @@
 
 	@Test
 	public void accessIsGrantedIfNoAttributesDefined() {
-		AclEntryAfterInvocationProvider provider = new AclEntryAfterInvocationProvider(mock(AclService.class),
+		AclEntryAfterInvocationProvider provider = new AclEntryAfterInvocationProvider(aclService,
 				Arrays.asList(mock(Permission.class)));
 		Object returned = new Object();
 		assertThat(returned).isSameAs(provider.decide(mock(Authentication.class), new Object(),
@@ -87,7 +95,7 @@
 
 	@Test
 	public void accessIsGrantedIfObjectTypeNotSupported() {
-		AclEntryAfterInvocationProvider provider = new AclEntryAfterInvocationProvider(mock(AclService.class),
+		AclEntryAfterInvocationProvider provider = new AclEntryAfterInvocationProvider(aclService,
 				Arrays.asList(mock(Permission.class)));
 		provider.setProcessDomainObjectClass(String.class);
 		// Not a String
