--- a/config/src/test/java/org/springframework/security/config/SecurityNamespaceHandlerTests.java
+++ b/config/src/test/java/org/springframework/security/config/SecurityNamespaceHandlerTests.java
@@ -72,25 +72,26 @@
 	@Mock(answer = Answers.CALLS_REAL_METHODS)
 	private MockedStatic<ClassUtils> classUtils;
 
+	@Mock
+	private Appender<ILoggingEvent> loggingAppender;
+
 	@Test
 	public void constructionWhenVersionsMatchThenLogsNothing() {
-		Appender<ILoggingEvent> appender = mock(Appender.class);
 		Logger logger = (Logger) LoggerFactory.getLogger(SecurityNamespaceHandler.class);
-		logger.addAppender(appender);
+		logger.addAppender(loggingAppender);
 		assertThat(new SecurityNamespaceHandler()).isNotNull();
-		verify(appender, never()).doAppend(any(ILoggingEvent.class));
+		verify(loggingAppender, never()).doAppend(any(ILoggingEvent.class));
 	}
 
 	@Test
 	public void constructorWhenDetectsMismatchingVersionsThenLogsError() {
-		Appender<ILoggingEvent> appender = mock(Appender.class);
 		Logger logger = (Logger) LoggerFactory.getLogger(SecurityNamespaceHandler.class);
-		logger.addAppender(appender);
+		logger.addAppender(loggingAppender);
 		try (MockedStatic<SpringSecurityCoreVersion> core = Mockito.mockStatic(SpringSecurityCoreVersion.class)) {
 			core.when(SpringSecurityCoreVersion::getVersion).thenReturn("mismatching");
 			assertThat(new SecurityNamespaceHandler()).isNotNull();
 			ArgumentCaptor<ILoggingEvent> captor = ArgumentCaptor.forClass(ILoggingEvent.class);
-			verify(appender).doAppend(captor.capture());
+			verify(loggingAppender).doAppend(captor.capture());
 			assertThat(captor.getValue().getLevel()).isEqualTo(Level.ERROR);
 		}
 	}
