--- a/web/src/test/java/org/springframework/security/web/session/DefaultSessionAuthenticationStrategyTests.java
+++ b/web/src/test/java/org/springframework/security/web/session/DefaultSessionAuthenticationStrategyTests.java
@@ -18,6 +18,7 @@
 
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpSession;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 
@@ -37,6 +38,13 @@
  * @author Luke Taylor
  */
 public class DefaultSessionAuthenticationStrategyTests {
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	@BeforeEach
+	public void setUp() {
+		this.applicationEventPublisher = mock(ApplicationEventPublisher.class);
+	}
 
 	@Test
 	public void newSessionShouldNotBeCreatedIfNoSessionExistsAndAlwaysCreateIsFalse() {
@@ -64,12 +72,11 @@
 		session.setAttribute("blah", "blah");
 		session.setAttribute("SPRING_SECURITY_SAVED_REQUEST_KEY", "DefaultSavedRequest");
 		String oldSessionId = session.getId();
-		ApplicationEventPublisher eventPublisher = mock(ApplicationEventPublisher.class);
-		strategy.setApplicationEventPublisher(eventPublisher);
+		strategy.setApplicationEventPublisher(applicationEventPublisher);
 		Authentication mockAuthentication = mock(Authentication.class);
 		strategy.onAuthentication(mockAuthentication, request, new MockHttpServletResponse());
 		ArgumentCaptor<ApplicationEvent> eventArgumentCaptor = ArgumentCaptor.forClass(ApplicationEvent.class);
-		verify(eventPublisher).publishEvent(eventArgumentCaptor.capture());
+		verify(applicationEventPublisher).publishEvent(eventArgumentCaptor.capture());
 		assertThat(oldSessionId.equals(request.getSession().getId())).isFalse();
 		assertThat(request.getSession().getAttribute("blah")).isNotNull();
 		assertThat(request.getSession().getAttribute("SPRING_SECURITY_SAVED_REQUEST_KEY")).isNotNull();
@@ -105,12 +112,11 @@
 		session.setAttribute("blah", "blah");
 		session.setAttribute("SPRING_SECURITY_SAVED_REQUEST_KEY", "DefaultSavedRequest");
 		String oldSessionId = session.getId();
-		ApplicationEventPublisher eventPublisher = mock(ApplicationEventPublisher.class);
-		strategy.setApplicationEventPublisher(eventPublisher);
+		strategy.setApplicationEventPublisher(applicationEventPublisher);
 		Authentication mockAuthentication = mock(Authentication.class);
 		strategy.onAuthentication(mockAuthentication, request, new MockHttpServletResponse());
 		ArgumentCaptor<ApplicationEvent> eventArgumentCaptor = ArgumentCaptor.forClass(ApplicationEvent.class);
-		verify(eventPublisher).publishEvent(eventArgumentCaptor.capture());
+		verify(applicationEventPublisher).publishEvent(eventArgumentCaptor.capture());
 		assertThat(request.getSession().getAttribute("blah")).isNull();
 		assertThat(request.getSession().getAttribute("SPRING_SECURITY_SAVED_REQUEST_KEY")).isNotNull();
 		assertThat(eventArgumentCaptor.getValue()).isNotNull();
