--- a/web/src/test/java/org/springframework/security/web/authentication/rememberme/RememberMeAuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/rememberme/RememberMeAuthenticationFilterTests.java
@@ -56,8 +56,11 @@
 
 	Authentication remembered = new TestingAuthenticationToken("remembered", "password", "ROLE_REMEMBERED");
 
+	AuthenticationManager authenticationManager;
+
 	@BeforeEach
 	public void setUp() {
+		authenticationManager = mock(AuthenticationManager.class);
 		SecurityContextHolder.clearContext();
 	}
 
@@ -75,7 +78,7 @@
 	@Test
 	public void testDetectsRememberMeServicesProperty() {
 		assertThatIllegalArgumentException()
-			.isThrownBy(() -> new RememberMeAuthenticationFilter(mock(AuthenticationManager.class), null));
+			.isThrownBy(() -> new RememberMeAuthenticationFilter(authenticationManager, null));
 	}
 
 	@Test
@@ -84,7 +87,7 @@
 		Authentication originalAuth = new TestingAuthenticationToken("user", "password", "ROLE_A");
 		SecurityContextHolder.getContext().setAuthentication(originalAuth);
 		// Setup our filter correctly
-		RememberMeAuthenticationFilter filter = new RememberMeAuthenticationFilter(mock(AuthenticationManager.class),
+		RememberMeAuthenticationFilter filter = new RememberMeAuthenticationFilter(authenticationManager,
 				new MockRememberMeServices(this.remembered));
 		filter.afterPropertiesSet();
 		// Test
