--- a/config/src/test/java/org/springframework/security/config/annotation/web/configuration/RegisterMissingBeanPostProcessorTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configuration/RegisterMissingBeanPostProcessorTests.java
@@ -18,6 +18,7 @@
 
 import java.util.function.Supplier;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 
@@ -41,16 +42,22 @@
 
 	private final RegisterMissingBeanPostProcessor postProcessor = new RegisterMissingBeanPostProcessor();
 
+	private BeanDefinitionRegistry mockedBeanDefinitionRegistry;
+
+	@BeforeEach
+	public void setUp() {
+		this.mockedBeanDefinitionRegistry = mock(BeanDefinitionRegistry.class);
+	}
+
 	@Test
 	public void postProcessBeanDefinitionRegistryWhenClassAddedThenRegisteredWithClass() {
 		this.postProcessor.addBeanDefinition(SimpleBean.class, null);
 		this.postProcessor.setBeanFactory(new DefaultListableBeanFactory());
 
-		BeanDefinitionRegistry beanDefinitionRegistry = mock(BeanDefinitionRegistry.class);
-		this.postProcessor.postProcessBeanDefinitionRegistry(beanDefinitionRegistry);
+		this.postProcessor.postProcessBeanDefinitionRegistry(mockedBeanDefinitionRegistry);
 
 		ArgumentCaptor<BeanDefinition> beanDefinitionCaptor = ArgumentCaptor.forClass(BeanDefinition.class);
-		verify(beanDefinitionRegistry).registerBeanDefinition(endsWith("SimpleBean"), beanDefinitionCaptor.capture());
+		verify(mockedBeanDefinitionRegistry).registerBeanDefinition(endsWith("SimpleBean"), beanDefinitionCaptor.capture());
 
 		RootBeanDefinition beanDefinition = (RootBeanDefinition) beanDefinitionCaptor.getValue();
 		assertThat(beanDefinition.getBeanClass()).isEqualTo(SimpleBean.class);
@@ -63,11 +70,10 @@
 		this.postProcessor.addBeanDefinition(SimpleBean.class, beanSupplier);
 		this.postProcessor.setBeanFactory(new DefaultListableBeanFactory());
 
-		BeanDefinitionRegistry beanDefinitionRegistry = mock(BeanDefinitionRegistry.class);
-		this.postProcessor.postProcessBeanDefinitionRegistry(beanDefinitionRegistry);
+		this.postProcessor.postProcessBeanDefinitionRegistry(mockedBeanDefinitionRegistry);
 
 		ArgumentCaptor<BeanDefinition> beanDefinitionCaptor = ArgumentCaptor.forClass(BeanDefinition.class);
-		verify(beanDefinitionRegistry).registerBeanDefinition(endsWith("SimpleBean"), beanDefinitionCaptor.capture());
+		verify(mockedBeanDefinitionRegistry).registerBeanDefinition(endsWith("SimpleBean"), beanDefinitionCaptor.capture());
 
 		RootBeanDefinition beanDefinition = (RootBeanDefinition) beanDefinitionCaptor.getValue();
 		assertThat(beanDefinition.getBeanClass()).isEqualTo(SimpleBean.class);
@@ -78,9 +84,8 @@
 	public void postProcessBeanDefinitionRegistryWhenNoBeanDefinitionsAddedThenNoneRegistered() {
 		this.postProcessor.setBeanFactory(new DefaultListableBeanFactory());
 
-		BeanDefinitionRegistry beanDefinitionRegistry = mock(BeanDefinitionRegistry.class);
-		this.postProcessor.postProcessBeanDefinitionRegistry(beanDefinitionRegistry);
-		verifyNoInteractions(beanDefinitionRegistry);
+		this.postProcessor.postProcessBeanDefinitionRegistry(mockedBeanDefinitionRegistry);
+		verifyNoInteractions(mockedBeanDefinitionRegistry);
 	}
 
 	@Test
@@ -90,9 +95,8 @@
 		beanFactory.registerBeanDefinition("simpleBean", new RootBeanDefinition(SimpleBean.class));
 		this.postProcessor.setBeanFactory(beanFactory);
 
-		BeanDefinitionRegistry beanDefinitionRegistry = mock(BeanDefinitionRegistry.class);
-		this.postProcessor.postProcessBeanDefinitionRegistry(beanDefinitionRegistry);
-		verifyNoInteractions(beanDefinitionRegistry);
+		this.postProcessor.postProcessBeanDefinitionRegistry(mockedBeanDefinitionRegistry);
+		verifyNoInteractions(mockedBeanDefinitionRegistry);
 	}
 
 	private static final class SimpleBean {
