--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveAuthorizationCodeTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveAuthorizationCodeTokenResponseClientTests.java
@@ -423,9 +423,9 @@
 
 		WebClientReactiveAuthorizationCodeTokenResponseClient customClient = new WebClientReactiveAuthorizationCodeTokenResponseClient();
 
-		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = mock();
 		OAuth2AccessTokenResponse response = TestOAuth2AccessTokenResponses.accessTokenResponse().build();
-		given(extractor.extract(any(), any())).willReturn(Mono.just(response));
+		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = MockOAuth2AccessTokenResponseBodyExtractor
+			.createMockOAuth2AccessTokenResponseBodyExtractor(response);
 
 		customClient.setBodyExtractor(extractor);
 
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveClientCredentialsTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveClientCredentialsTokenResponseClientTests.java
@@ -385,9 +385,9 @@
 
 		WebClientReactiveClientCredentialsTokenResponseClient customClient = new WebClientReactiveClientCredentialsTokenResponseClient();
 
-		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = mock();
 		OAuth2AccessTokenResponse response = TestOAuth2AccessTokenResponses.accessTokenResponse().build();
-		given(extractor.extract(any(), any())).willReturn(Mono.just(response));
+		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = MockOAuth2AccessTokenResponseBodyExtractor
+				.createMockOAuth2AccessTokenResponseBodyExtractor(response);
 
 		customClient.setBodyExtractor(extractor);
 
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveJwtBearerTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveJwtBearerTokenResponseClientTests.java
@@ -303,9 +303,9 @@
 
 	@Test
 	public void getTokenResponseWhenBodyExtractorSetThenCalled() {
-		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> bodyExtractor = mock();
 		OAuth2AccessTokenResponse response = TestOAuth2AccessTokenResponses.accessTokenResponse().build();
-		given(bodyExtractor.extract(any(), any())).willReturn(Mono.just(response));
+		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> bodyExtractor = MockOAuth2AccessTokenResponseBodyExtractor
+				.createMockOAuth2AccessTokenResponseBodyExtractor(response);
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest request = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
 		this.client.setBodyExtractor(bodyExtractor);
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveRefreshTokenTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveRefreshTokenTokenResponseClientTests.java
@@ -396,9 +396,9 @@
 
 		WebClientReactiveRefreshTokenTokenResponseClient customClient = new WebClientReactiveRefreshTokenTokenResponseClient();
 
-		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = mock();
 		OAuth2AccessTokenResponse response = TestOAuth2AccessTokenResponses.accessTokenResponse().build();
-		given(extractor.extract(any(), any())).willReturn(Mono.just(response));
+		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = MockOAuth2AccessTokenResponseBodyExtractor
+				.createMockOAuth2AccessTokenResponseBodyExtractor(response);
 
 		customClient.setBodyExtractor(extractor);
 
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockOAuth2AccessTokenResponseBodyExtractor.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockOAuth2AccessTokenResponseBodyExtractor.java
@@ -0,0 +1,25 @@
+package org.springframework.security.oauth2.client.endpoint;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.http.ReactiveHttpInputMessage;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse;
+import org.springframework.web.reactive.function.BodyExtractor;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public final class MockOAuth2AccessTokenResponseBodyExtractor {
+
+	private MockOAuth2AccessTokenResponseBodyExtractor() {
+	}
+
+	public static BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> createMockOAuth2AccessTokenResponseBodyExtractor(
+			OAuth2AccessTokenResponse response) {
+		BodyExtractor<Mono<OAuth2AccessTokenResponse>, ReactiveHttpInputMessage> extractor = mock();
+		given(extractor.extract(any(), any())).willReturn(Mono.just(response));
+		return extractor;
+	}
+
+}
