--- a/rsocket/src/test/java/org/springframework/security/rsocket/core/PayloadSocketAcceptorInterceptorTests.java
+++ b/rsocket/src/test/java/org/springframework/security/rsocket/core/PayloadSocketAcceptorInterceptorTests.java
@@ -75,7 +75,7 @@
 
 	@Test
 	public void applyWhenDefaultMetadataMimeTypeThenDefaulted() {
-		given(this.setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
+		this.setupPayload = MockConnectionSetupPayload.createMockConnectionSetupPayload();
 		PayloadExchange exchange = captureExchange();
 		assertThat(exchange.getMetadataMimeType().toString())
 			.isEqualTo(WellKnownMimeType.MESSAGE_RSOCKET_COMPOSITE_METADATA.getString());
@@ -85,7 +85,7 @@
 	@Test
 	public void acceptWhenDefaultMetadataMimeTypeOverrideThenDefaulted() {
 		this.acceptorInterceptor.setDefaultMetadataMimeType(MediaType.APPLICATION_JSON);
-		given(this.setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
+		this.setupPayload = MockConnectionSetupPayload.createMockConnectionSetupPayload();
 		PayloadExchange exchange = captureExchange();
 		assertThat(exchange.getMetadataMimeType()).isEqualTo(MediaType.APPLICATION_JSON);
 		assertThat(exchange.getDataMimeType()).isEqualTo(MediaType.APPLICATION_JSON);
--- a/rsocket/src/test/java/org/springframework/security/rsocket/core/PayloadSocketAcceptorTests.java
+++ b/rsocket/src/test/java/org/springframework/security/rsocket/core/PayloadSocketAcceptorTests.java
@@ -109,7 +109,7 @@
 
 	@Test
 	public void acceptWhenDefaultMetadataMimeTypeThenDefaulted() {
-		given(this.setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
+		this.setupPayload = MockConnectionSetupPayload.createMockConnectionSetupPayload();
 		PayloadExchange exchange = captureExchange();
 		assertThat(exchange.getMetadataMimeType().toString())
 			.isEqualTo(WellKnownMimeType.MESSAGE_RSOCKET_COMPOSITE_METADATA.getString());
@@ -119,7 +119,7 @@
 	@Test
 	public void acceptWhenDefaultMetadataMimeTypeOverrideThenDefaulted() {
 		this.acceptor.setDefaultMetadataMimeType(MediaType.APPLICATION_JSON);
-		given(this.setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
+		this.setupPayload = MockConnectionSetupPayload.createMockConnectionSetupPayload();
 		PayloadExchange exchange = captureExchange();
 		assertThat(exchange.getMetadataMimeType()).isEqualTo(MediaType.APPLICATION_JSON);
 		assertThat(exchange.getDataMimeType()).isEqualTo(MediaType.APPLICATION_JSON);
@@ -136,8 +136,8 @@
 
 	@Test
 	public void acceptWhenExplicitMimeTypeThenThenOverrideDefault() {
+		this.setupPayload = MockConnectionSetupPayload.createMockConnectionSetupPayload();
 		given(this.setupPayload.metadataMimeType()).willReturn(MediaType.TEXT_PLAIN_VALUE);
-		given(this.setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
 		PayloadExchange exchange = captureExchange();
 		assertThat(exchange.getMetadataMimeType()).isEqualTo(MediaType.TEXT_PLAIN);
 		assertThat(exchange.getDataMimeType()).isEqualTo(MediaType.APPLICATION_JSON);
@@ -146,8 +146,8 @@
 	@Test
 	// gh-8654
 	public void acceptWhenDelegateAcceptRequiresReactiveSecurityContext() {
+		this.setupPayload = MockConnectionSetupPayload.createMockConnectionSetupPayload();
 		given(this.setupPayload.metadataMimeType()).willReturn(MediaType.TEXT_PLAIN_VALUE);
-		given(this.setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
 		SecurityContext expectedSecurityContext = new SecurityContextImpl(
 				new TestingAuthenticationToken("user", "password", "ROLE_USER"));
 		CaptureSecurityContextSocketAcceptor captureSecurityContext = new CaptureSecurityContextSocketAcceptor(
--- a/rsocket/src/test/java/org/springframework/security/rsocket/core/MockConnectionSetupPayload.java
+++ b/rsocket/src/test/java/org/springframework/security/rsocket/core/MockConnectionSetupPayload.java
@@ -0,0 +1,20 @@
+package org.springframework.security.rsocket.core;
+
+import io.rsocket.ConnectionSetupPayload;
+import org.mockito.Mockito;
+import org.springframework.http.MediaType;
+
+import static org.mockito.BDDMockito.given;
+
+public final class MockConnectionSetupPayload {
+
+	private MockConnectionSetupPayload() {
+	}
+
+	public static ConnectionSetupPayload createMockConnectionSetupPayload() {
+		ConnectionSetupPayload setupPayload = Mockito.mock(ConnectionSetupPayload.class);
+		given(setupPayload.dataMimeType()).willReturn(MediaType.APPLICATION_JSON_VALUE);
+		return setupPayload;
+	}
+
+}
