--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientJwtBearerTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientJwtBearerTokenResponseClientTests.java
@@ -320,10 +320,10 @@
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest grantRequest = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
-		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.put("custom-header-name", Collections.singletonList("custom-header-value"));
-		given(headersConverter.convert(grantRequest)).willReturn(headers);
+		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = createMockJwtBearerGrantRequestHeadersConverter(grantRequest,
+				headers);
 		this.tokenResponseClient.addHeadersConverter(headersConverter);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		verify(headersConverter).convert(grantRequest);
@@ -332,16 +332,23 @@
 		assertThat(recordedRequest.getHeader("custom-header-name")).isEqualTo("custom-header-value");
 	}
 
+	private Converter<JwtBearerGrantRequest, HttpHeaders> createMockJwtBearerGrantRequestHeadersConverter(
+			JwtBearerGrantRequest grantRequest, HttpHeaders headers) {
+		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = mock();
+		given(headersConverter.convert(grantRequest)).willReturn(headers);
+		return headersConverter;
+	}
+
 	@Test
 	public void getTokenResponseWhenHeadersConverterSetThenCalled() throws Exception {
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest grantRequest = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
-		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
 		headers.put("custom-header-name", Collections.singletonList("custom-header-value"));
-		given(headersConverter.convert(grantRequest)).willReturn(headers);
+		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = createMockJwtBearerGrantRequestHeadersConverter(
+				grantRequest, headers);
 		this.tokenResponseClient.setHeadersConverter(headersConverter);
 		this.tokenResponseClient.getTokenResponse(grantRequest);
 		verify(headersConverter).convert(grantRequest);
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveJwtBearerTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/WebClientReactiveJwtBearerTokenResponseClientTests.java
@@ -190,10 +190,10 @@
 	public void getTokenResponseWhenHeadersConverterSetThenCalled() throws Exception {
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest request = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
-		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.setBasicAuth(clientRegistration.getClientId(), clientRegistration.getClientSecret());
-		given(headersConverter.convert(request)).willReturn(headers);
+		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = createMockJwtBearerGrantRequestHeadersConverter(request,
+				headers);
 		this.client.setHeadersConverter(headersConverter);
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		this.client.getTokenResponse(request).block();
@@ -203,14 +203,21 @@
 			.isEqualTo("Basic Y2xpZW50LWlkOmNsaWVudC1zZWNyZXQ=");
 	}
 
+	private Converter<JwtBearerGrantRequest, HttpHeaders> createMockJwtBearerGrantRequestHeadersConverter(
+			JwtBearerGrantRequest request, HttpHeaders headers) {
+		Converter<JwtBearerGrantRequest, HttpHeaders> headersConverter = mock();
+		given(headersConverter.convert(request)).willReturn(headers);
+		return headersConverter;
+	}
+
 	@Test
 	public void getTokenResponseWhenHeadersConverterAddedThenCalled() throws Exception {
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest request = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
-		Converter<JwtBearerGrantRequest, HttpHeaders> addedHeadersConverter = mock();
 		HttpHeaders headers = new HttpHeaders();
 		headers.put("custom-header-name", Collections.singletonList("custom-header-value"));
-		given(addedHeadersConverter.convert(request)).willReturn(headers);
+		Converter<JwtBearerGrantRequest, HttpHeaders> addedHeadersConverter = createMockJwtBearerGrantRequestHeadersConverter(request,
+				headers);
 		this.client.addHeadersConverter(addedHeadersConverter);
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
 		this.client.getTokenResponse(request).block();
