--- a/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospectorTests.java
+++ b/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospectorTests.java
@@ -298,7 +298,7 @@
 	@Test
 	public void setRequestEntityConverterWhenNonNullConverterGivenThenConverterUsed() {
 		RestOperations restOperations = mock(RestOperations.class);
-		Converter<String, RequestEntity<?>> requestEntityConverter = mock(Converter.class);
+		Converter<String, RequestEntity<?>> requestEntityConverter = createMockRequestEntityConverter();
 		RequestEntity requestEntity = mock(RequestEntity.class);
 		String tokenToIntrospect = "some token";
 		given(requestEntityConverter.convert(tokenToIntrospect)).willReturn(requestEntity);
@@ -322,7 +322,7 @@
 	@Test
 	public void setAuthenticationConverterWhenNonNullConverterGivenThenConverterUsed() {
 		RestOperations restOperations = mock(RestOperations.class);
-		Converter<String, RequestEntity<?>> requestEntityConverter = mock(Converter.class);
+		Converter<String, RequestEntity<?>> requestEntityConverter = createMockRequestEntityConverter();
 		RequestEntity requestEntity = mock(RequestEntity.class);
 		Converter<OAuth2TokenIntrospectionClaimAccessor, OAuth2AuthenticatedPrincipal> authenticationConverter = mock(
 				Converter.class);
@@ -409,6 +409,11 @@
 			.isThrownBy(() -> SpringOpaqueTokenIntrospector.withIntrospectionUri(INTROSPECTION_URL).build());
 	}
 
+	@SuppressWarnings("unchecked")
+	private static Converter<String, RequestEntity<?>> createMockRequestEntityConverter() {
+		return mock(Converter.class);
+	}
+
 	private static ResponseEntity<Map<String, Object>> response(String content) {
 		HttpHeaders headers = new HttpHeaders();
 		headers.setContentType(MediaType.APPLICATION_JSON);
