--- a/web/src/test/java/org/springframework/security/web/context/AbstractSecurityWebApplicationInitializerTests.java
+++ b/web/src/test/java/org/springframework/security/web/context/AbstractSecurityWebApplicationInitializerTests.java
@@ -26,6 +26,7 @@
 import jakarta.servlet.FilterRegistration;
 import jakarta.servlet.ServletContext;
 import jakarta.servlet.SessionTrackingMode;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 
@@ -55,6 +56,13 @@
 	private static final EnumSet<DispatcherType> DEFAULT_DISPATCH = EnumSet.of(DispatcherType.REQUEST,
 			DispatcherType.ERROR, DispatcherType.ASYNC, DispatcherType.FORWARD, DispatcherType.INCLUDE);
 
+	private Filter sharedFilter;
+
+	@BeforeEach
+	void setUp() {
+		this.sharedFilter = mock(Filter.class);
+	}
+
 	@Test
 	public void onStartupWhenDefaultContextThenRegistersSpringSecurityFilterChain() {
 		ServletContext context = mock(ServletContext.class);
@@ -152,31 +160,29 @@
 
 	@Test
 	public void onStartupWhenInsertFiltersThenInserted() {
-		Filter filter1 = mock(Filter.class);
 		Filter filter2 = mock(Filter.class);
 		ServletContext context = mock(ServletContext.class);
 		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
 		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
 		given(context.addFilter(eq("springSecurityFilterChain"), proxyCaptor.capture())).willReturn(registration);
-		given(context.addFilter(anyString(), eq(filter1))).willReturn(registration);
+		given(context.addFilter(anyString(), eq(sharedFilter))).willReturn(registration);
 		given(context.addFilter(anyString(), eq(filter2))).willReturn(registration);
 		new AbstractSecurityWebApplicationInitializer() {
 			@Override
 			protected void afterSpringSecurityFilterChain(ServletContext servletContext) {
-				insertFilters(context, filter1, filter2);
+				insertFilters(context, sharedFilter, filter2);
 			}
 		}.onStartup(context);
 		assertProxyDefaults(proxyCaptor.getValue());
 		verify(registration, times(3)).addMappingForUrlPatterns(DEFAULT_DISPATCH, false, "/*");
 		verify(registration, times(3)).setAsyncSupported(true);
 		verifyNoAddListener(context);
-		verify(context).addFilter(anyString(), eq(filter1));
+		verify(context).addFilter(anyString(), eq(sharedFilter));
 		verify(context).addFilter(anyString(), eq(filter2));
 	}
 
 	@Test
 	public void onStartupWhenDuplicateFilterInsertedThenException() {
-		Filter filter1 = mock(Filter.class);
 		ServletContext context = mock(ServletContext.class);
 		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
 		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
@@ -185,7 +191,7 @@
 
 			@Override
 			protected void afterSpringSecurityFilterChain(ServletContext servletContext) {
-				insertFilters(context, filter1);
+				insertFilters(context, sharedFilter);
 			}
 
 		}.onStartup(context))
@@ -193,7 +199,7 @@
 					"Duplicate Filter registration for 'object'. Check to ensure the Filter is only configured once.");
 		assertProxyDefaults(proxyCaptor.getValue());
 		verify(registration).addMappingForUrlPatterns(DEFAULT_DISPATCH, false, "/*");
-		verify(context).addFilter(anyString(), eq(filter1));
+		verify(context).addFilter(anyString(), eq(sharedFilter));
 	}
 
 	@Test
@@ -215,17 +221,16 @@
 
 	@Test
 	public void onStartupWhenNullFilterInsertedThenException() {
-		Filter filter = mock(Filter.class);
-		ServletContext context = mock(ServletContext.class);
-		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
-		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
-		given(context.addFilter(eq("springSecurityFilterChain"), proxyCaptor.capture())).willReturn(registration);
-		given(context.addFilter(anyString(), eq(filter))).willReturn(registration);
+		ServletContext context = mock(ServletContext.class);
+		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
+		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
+		given(context.addFilter(eq("springSecurityFilterChain"), proxyCaptor.capture())).willReturn(registration);
+		given(context.addFilter(anyString(), eq(sharedFilter))).willReturn(registration);
 		assertThatIllegalArgumentException().isThrownBy(() -> new AbstractSecurityWebApplicationInitializer() {
 
 			@Override
 			protected void afterSpringSecurityFilterChain(ServletContext servletContext) {
-				insertFilters(context, filter, null);
+				insertFilters(context, sharedFilter, null);
 			}
 
 		}.onStartup(context)).withMessageContaining("filters cannot contain null values");
@@ -234,18 +239,17 @@
 
 	@Test
 	public void onStartupWhenAppendFiltersThenAppended() {
-		Filter filter1 = mock(Filter.class);
 		Filter filter2 = mock(Filter.class);
 		ServletContext context = mock(ServletContext.class);
 		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
 		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
 		given(context.addFilter(eq("springSecurityFilterChain"), proxyCaptor.capture())).willReturn(registration);
-		given(context.addFilter(anyString(), eq(filter1))).willReturn(registration);
+		given(context.addFilter(anyString(), eq(sharedFilter))).willReturn(registration);
 		given(context.addFilter(anyString(), eq(filter2))).willReturn(registration);
 		new AbstractSecurityWebApplicationInitializer() {
 			@Override
 			protected void afterSpringSecurityFilterChain(ServletContext servletContext) {
-				appendFilters(context, filter1, filter2);
+				appendFilters(context, sharedFilter, filter2);
 			}
 		}.onStartup(context);
 		verify(registration, times(1)).addMappingForUrlPatterns(DEFAULT_DISPATCH, false, "/*");
@@ -257,7 +261,6 @@
 
 	@Test
 	public void onStartupWhenDuplicateFilterAppendedThenException() {
-		Filter filter1 = mock(Filter.class);
 		ServletContext context = mock(ServletContext.class);
 		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
 		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
@@ -266,7 +269,7 @@
 
 			@Override
 			protected void afterSpringSecurityFilterChain(ServletContext servletContext) {
-				appendFilters(context, filter1);
+				appendFilters(context, sharedFilter);
 			}
 
 		}.onStartup(context))
@@ -274,7 +277,7 @@
 					+ "Check to ensure the Filter is only configured once.");
 		assertProxyDefaults(proxyCaptor.getValue());
 		verify(registration).addMappingForUrlPatterns(DEFAULT_DISPATCH, false, "/*");
-		verify(context).addFilter(anyString(), eq(filter1));
+		verify(context).addFilter(anyString(), eq(sharedFilter));
 	}
 
 	@Test
@@ -296,17 +299,16 @@
 
 	@Test
 	public void onStartupWhenNullFilterAppendedThenException() {
-		Filter filter = mock(Filter.class);
-		ServletContext context = mock(ServletContext.class);
-		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
-		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
-		given(context.addFilter(eq("springSecurityFilterChain"), proxyCaptor.capture())).willReturn(registration);
-		given(context.addFilter(anyString(), eq(filter))).willReturn(registration);
+		ServletContext context = mock(ServletContext.class);
+		FilterRegistration.Dynamic registration = mock(FilterRegistration.Dynamic.class);
+		ArgumentCaptor<DelegatingFilterProxy> proxyCaptor = ArgumentCaptor.forClass(DelegatingFilterProxy.class);
+		given(context.addFilter(eq("springSecurityFilterChain"), proxyCaptor.capture())).willReturn(registration);
+		given(context.addFilter(anyString(), eq(sharedFilter))).willReturn(registration);
 		assertThatIllegalArgumentException().isThrownBy(() -> new AbstractSecurityWebApplicationInitializer() {
 
 			@Override
 			protected void afterSpringSecurityFilterChain(ServletContext servletContext) {
-				appendFilters(context, filter, null);
+				appendFilters(context, sharedFilter, null);
 			}
 
 		}.onStartup(context)).withMessageContaining("filters cannot contain null values");
