--- a/access/src/test/java/org/springframework/security/web/access/channel/ChannelDecisionManagerImplTests.java
+++ b/access/src/test/java/org/springframework/security/web/access/channel/ChannelDecisionManagerImplTests.java
@@ -24,6 +24,7 @@
 
 import jakarta.servlet.FilterChain;
 import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.mock.web.MockHttpServletRequest;
@@ -45,6 +46,13 @@
 @SuppressWarnings({ "unchecked", "deprecation" })
 public class ChannelDecisionManagerImplTests {
 
+	private FilterChain filterChain;
+
+	@BeforeEach
+	void setUp() {
+		this.filterChain = mock(FilterChain.class);
+	}
+
 	@Test
 	public void testCannotSetEmptyChannelProcessorsList() throws Exception {
 		ChannelDecisionManagerImpl cdm = new ChannelDecisionManagerImpl();
@@ -83,7 +91,7 @@
 		cdm.afterPropertiesSet();
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterInvocation fi = new FilterInvocation(request, response, mock(FilterChain.class));
+		FilterInvocation fi = new FilterInvocation(request, response, filterChain);
 		List<ConfigAttribute> cad = SecurityConfig.createList("xyz");
 		cdm.decide(fi, cad);
 		Assertions.assertThat(fi.getResponse().isCommitted()).isTrue();
@@ -99,7 +107,7 @@
 		cdm.afterPropertiesSet();
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterInvocation fi = new FilterInvocation(request, response, mock(FilterChain.class));
+		FilterInvocation fi = new FilterInvocation(request, response, filterChain);
 		cdm.decide(fi, SecurityConfig.createList(new String[] { "abc", "ANY_CHANNEL" }));
 		Assertions.assertThat(fi.getResponse().isCommitted()).isFalse();
 	}
@@ -116,7 +124,7 @@
 		cdm.afterPropertiesSet();
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterInvocation fi = new FilterInvocation(request, response, mock(FilterChain.class));
+		FilterInvocation fi = new FilterInvocation(request, response, filterChain);
 		cdm.decide(fi, SecurityConfig.createList("SOME_ATTRIBUTE_NO_PROCESSORS_SUPPORT"));
 		Assertions.assertThat(fi.getResponse().isCommitted()).isFalse();
 	}
