--- a/web/src/test/java/org/springframework/security/web/session/SessionManagementFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/session/SessionManagementFilterTests.java
@@ -55,10 +55,13 @@
  */
 public class SessionManagementFilterTests {
 
+	private FilterChain filterChain;
+
 	@BeforeEach
 	@AfterEach
 	public void clearContext() {
 		SecurityContextHolder.clearContext();
+		filterChain = mock(FilterChain.class);
 	}
 
 	@Test
@@ -120,13 +123,12 @@
 		filter.setAuthenticationFailureHandler(failureHandler);
 		HttpServletRequest request = new MockHttpServletRequest();
 		HttpServletResponse response = new MockHttpServletResponse();
-		FilterChain fc = mock(FilterChain.class);
 		authenticateUser();
 		SessionAuthenticationException exception = new SessionAuthenticationException("Failure");
 		willThrow(exception).given(strategy)
 			.onAuthentication(SecurityContextHolder.getContext().getAuthentication(), request, response);
-		filter.doFilter(request, response, fc);
-		verifyNoMoreInteractions(fc);
+		filter.doFilter(request, response, filterChain);
+		verifyNoMoreInteractions(filterChain);
 		verify(failureHandler).onAuthenticationFailure(request, response, exception);
 	}
 
@@ -149,9 +151,8 @@
 		SimpleRedirectInvalidSessionStrategy iss = new SimpleRedirectInvalidSessionStrategy("/timedOut");
 		iss.setCreateNewSession(true);
 		filter.setInvalidSessionStrategy(iss);
-		FilterChain fc = mock(FilterChain.class);
-		filter.doFilter(request, response, fc);
-		verifyNoMoreInteractions(fc);
+		filter.doFilter(request, response, filterChain);
+		verifyNoMoreInteractions(filterChain);
 		assertThat(response.getRedirectedUrl()).isEqualTo("/timedOut");
 	}
 
@@ -175,9 +176,8 @@
 		RequestedUrlRedirectInvalidSessionStrategy iss = new RequestedUrlRedirectInvalidSessionStrategy();
 		iss.setCreateNewSession(true);
 		filter.setInvalidSessionStrategy(iss);
-		FilterChain fc = mock(FilterChain.class);
-		filter.doFilter(request, response, fc);
-		verifyNoMoreInteractions(fc);
+		filter.doFilter(request, response, filterChain);
+		verifyNoMoreInteractions(filterChain);
 		assertThat(response.getRedirectedUrl()).isEqualTo("/requested");
 	}
 
@@ -200,14 +200,13 @@
 		request.setRequestedSessionIdValid(false);
 		request.setRequestURI("/context/requested");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
 
 		// when
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 
 		// then
 		verify(securityContextRepository).containsContext(request);
-		verifyNoMoreInteractions(securityContextRepository, sessionAuthenticationStrategy, chain);
+		verifyNoMoreInteractions(securityContextRepository, sessionAuthenticationStrategy, filterChain);
 		assertThat(response.isCommitted()).isTrue();
 		assertThat(response.getRedirectedUrl()).isEqualTo("/context/requested");
 		assertThat(response.getStatus()).isEqualTo(302);
@@ -231,14 +230,13 @@
 		request.setRequestedSessionIdValid(false);
 		request.setRequestURI("/requested");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain chain = mock(FilterChain.class);
 
 		// when
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 
 		// then
 		verify(securityContextRepository).containsContext(request);
-		verifyNoMoreInteractions(securityContextRepository, sessionAuthenticationStrategy, chain);
+		verifyNoMoreInteractions(securityContextRepository, sessionAuthenticationStrategy, filterChain);
 		assertThat(response.isCommitted()).isTrue();
 		assertThat(response.getRedirectedUrl()).isEqualTo("/requested");
 		assertThat(response.getStatus()).isEqualTo(307);
