--- a/web/src/test/java/org/springframework/security/web/context/SecurityContextPersistenceFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/context/SecurityContextPersistenceFilterTests.java
@@ -43,6 +43,8 @@
 
 	TestingAuthenticationToken testToken = new TestingAuthenticationToken("someone", "passwd", "ROLE_A");
 
+	private final FilterChain filterChain = mock(FilterChain.class);
+
 	@AfterEach
 	public void clearContext() {
 		SecurityContextHolder.clearContext();
@@ -50,13 +52,12 @@
 
 	@Test
 	public void contextIsClearedAfterChainProceeds() throws Exception {
-		final FilterChain chain = mock(FilterChain.class);
 		final MockHttpServletRequest request = new MockHttpServletRequest();
 		final MockHttpServletResponse response = new MockHttpServletResponse();
 		SecurityContextPersistenceFilter filter = new SecurityContextPersistenceFilter();
 		SecurityContextHolder.getContext().setAuthentication(this.testToken);
-		filter.doFilter(request, response, chain);
-		verify(chain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		filter.doFilter(request, response, filterChain);
+		verify(filterChain).doFilter(any(ServletRequest.class), any(ServletResponse.class));
 		assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
 	}
 
@@ -95,35 +96,32 @@
 
 	@Test
 	public void filterIsNotAppliedAgainIfFilterAppliedAttributeIsSet() throws Exception {
-		final FilterChain chain = mock(FilterChain.class);
 		final MockHttpServletRequest request = new MockHttpServletRequest();
 		final MockHttpServletResponse response = new MockHttpServletResponse();
 		SecurityContextPersistenceFilter filter = new SecurityContextPersistenceFilter(
 				mock(SecurityContextRepository.class));
 		request.setAttribute(SecurityContextPersistenceFilter.FILTER_APPLIED, Boolean.TRUE);
-		filter.doFilter(request, response, chain);
-		verify(chain).doFilter(request, response);
+		filter.doFilter(request, response, filterChain);
+		verify(filterChain).doFilter(request, response);
 	}
 
 	@Test
 	public void sessionIsEagerlyCreatedWhenConfigured() throws Exception {
-		final FilterChain chain = mock(FilterChain.class);
 		final MockHttpServletRequest request = new MockHttpServletRequest();
 		final MockHttpServletResponse response = new MockHttpServletResponse();
 		SecurityContextPersistenceFilter filter = new SecurityContextPersistenceFilter();
 		filter.setForceEagerSessionCreation(true);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		assertThat(request.getSession(false)).isNotNull();
 	}
 
 	@Test
 	public void nullSecurityContextRepoDoesntSaveContextOrCreateSession() throws Exception {
-		final FilterChain chain = mock(FilterChain.class);
 		final MockHttpServletRequest request = new MockHttpServletRequest();
 		final MockHttpServletResponse response = new MockHttpServletResponse();
 		SecurityContextRepository repo = new NullSecurityContextRepository();
 		SecurityContextPersistenceFilter filter = new SecurityContextPersistenceFilter(repo);
-		filter.doFilter(request, response, chain);
+		filter.doFilter(request, response, filterChain);
 		assertThat(repo.containsContext(request)).isFalse();
 		assertThat(request.getSession(false)).isNull();
 	}
