--- a/web/src/test/java/org/springframework/security/web/ObservationFilterChainDecoratorTests.java
+++ b/web/src/test/java/org/springframework/security/web/ObservationFilterChainDecoratorTests.java
@@ -28,6 +28,7 @@
 import jakarta.servlet.ServletException;
 import jakarta.servlet.ServletRequest;
 import jakarta.servlet.ServletResponse;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
@@ -52,6 +53,13 @@
  */
 public class ObservationFilterChainDecoratorTests {
 
+	private FilterChain chainMock;
+
+	@BeforeEach
+	void setUp() {
+		chainMock = mock(FilterChain.class);
+	}
+
 	@Test
 	void decorateWhenDefaultsThenObserves() throws Exception {
 		ObservationHandler<?> handler = mock(ObservationHandler.class);
@@ -59,8 +67,7 @@
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
-		FilterChain chain = mock(FilterChain.class);
-		FilterChain decorated = decorator.decorate(chain);
+		FilterChain decorated = decorator.decorate(chainMock);
 		decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse());
 		verify(handler).onStart(any());
 	}
@@ -72,8 +79,7 @@
 		ObservationRegistry registry = ObservationRegistry.NOOP;
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
-		FilterChain chain = mock(FilterChain.class);
-		FilterChain decorated = decorator.decorate(chain);
+		FilterChain decorated = decorator.decorate(chainMock);
 		decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse());
 		verifyNoInteractions(handler);
 	}
@@ -85,9 +91,8 @@
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
-		FilterChain chain = mock(FilterChain.class);
 		Filter filter = mock(Filter.class);
-		FilterChain decorated = decorator.decorate(chain, List.of(filter));
+		FilterChain decorated = decorator.decorate(chainMock, List.of(filter));
 		decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse());
 		verify(handler, times(2)).onStart(any());
 		ArgumentCaptor<Observation.Event> event = ArgumentCaptor.forClass(Observation.Event.class);
@@ -104,9 +109,8 @@
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
-		FilterChain chain = mock(FilterChain.class);
 		Filter filter = new BasicAuthenticationFilter();
-		FilterChain decorated = decorator.decorate(chain, List.of(filter));
+		FilterChain decorated = decorator.decorate(chainMock, List.of(filter));
 		decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse());
 		ArgumentCaptor<Observation.Event> event = ArgumentCaptor.forClass(Observation.Event.class);
 		verify(handler, times(2)).onEvent(event.capture(), any());
@@ -123,10 +127,9 @@
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
-		FilterChain chain = mock(FilterChain.class);
 		Filter filter = mock(Filter.class);
 		willThrow(IllegalArgumentException.class).given(filter).doFilter(any(), any(), any());
-		FilterChain decorated = decorator.decorate(chain, List.of(filter));
+		FilterChain decorated = decorator.decorate(chainMock, List.of(filter));
 		assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(
 				() -> decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse()));
 		verify(handler).onScopeClosed(any());
@@ -141,8 +144,7 @@
 		ObservationRegistry registry = ObservationRegistry.create();
 		registry.observationConfig().observationHandler(handler);
 		ObservationFilterChainDecorator decorator = new ObservationFilterChainDecorator(registry);
-		FilterChain chain = mock(FilterChain.class);
-		FilterChain decorated = decorator.decorate(chain, List.of(filter));
+		FilterChain decorated = decorator.decorate(chainMock, List.of(filter));
 		decorated.doFilter(new MockHttpServletRequest("GET", "/"), new MockHttpServletResponse());
 		ArgumentCaptor<Observation.Context> context = ArgumentCaptor.forClass(Observation.Context.class);
 		verify(handler, times(3)).onScopeClosed(context.capture());
