--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/OAuth2AuthorizationRequestRedirectFilterTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/OAuth2AuthorizationRequestRedirectFilterTests.java
@@ -220,9 +220,7 @@
 		String requestUri = "/path";
 		MockHttpServletRequest request = get(requestUri).build();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		willThrow(new ClientAuthorizationRequiredException(this.registration1.getRegistrationId())).given(filterChain)
-			.doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		FilterChain filterChain = createMockFilterChain(this.registration1.getRegistrationId());
 		this.filter.doFilter(request, response, filterChain);
 		verify(filterChain).doFilter(any(HttpServletRequest.class), any(HttpServletResponse.class));
 		assertThat(response.getRedirectedUrl()).matches("https://example.com/login/oauth/authorize\\?"
@@ -237,9 +235,7 @@
 		String requestUri = "/path";
 		MockHttpServletRequest request = get(requestUri).build();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		willThrow(new ClientAuthorizationRequiredException(this.registration1.getRegistrationId())).given(filterChain)
-			.doFilter(any(ServletRequest.class), any(ServletResponse.class));
+		FilterChain filterChain = createMockFilterChain(this.registration1.getRegistrationId());
 		OAuth2AuthorizationRequestResolver resolver = mock(OAuth2AuthorizationRequestResolver.class);
 		OAuth2AuthorizationRequestRedirectFilter filter = new OAuth2AuthorizationRequestRedirectFilter(resolver);
 		filter.doFilter(request, response, filterChain);
@@ -351,14 +347,19 @@
 		String requestUri = "/path";
 		MockHttpServletRequest request = get(requestUri).build();
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
+		FilterChain filterChain = createMockFilterChain(this.registration1.getRegistrationId());
 		willAnswer((invocation) -> assertThat((invocation.<HttpServletResponse>getArgument(1)).isCommitted()).isFalse())
 			.given(this.requestCache)
 			.saveRequest(any(HttpServletRequest.class), any(HttpServletResponse.class));
-		willThrow(new ClientAuthorizationRequiredException(this.registration1.getRegistrationId())).given(filterChain)
+		this.filter.doFilter(request, response, filterChain);
+		assertThat(response.isCommitted()).isTrue();
+	}
+
+	private static FilterChain createMockFilterChain(String registrationId) throws Exception {
+		FilterChain filterChain = mock(FilterChain.class);
+		willThrow(new ClientAuthorizationRequiredException(registrationId)).given(filterChain)
 			.doFilter(any(ServletRequest.class), any(ServletResponse.class));
-		this.filter.doFilter(request, response, filterChain);
-		assertThat(response.isCommitted()).isTrue();
+		return filterChain;
 	}
 
 }
