--- a/web/src/test/java/org/springframework/security/web/access/intercept/AuthorizationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/access/intercept/AuthorizationFilterTests.java
@@ -77,10 +77,13 @@
 
 	private final FilterChain chain = new MockFilterChain();
 
+	private FilterChain filterChain;
+
 	@BeforeEach
 	public void setup() {
 		this.authorizationManager = mock(AuthorizationManager.class);
 		this.filter = new AuthorizationFilter(this.authorizationManager);
+		this.filterChain = mock(FilterChain.class);
 	}
 
 	@AfterEach
@@ -102,16 +105,15 @@
 
 		MockHttpServletRequest mockRequest = new MockHttpServletRequest(null, "/path");
 		MockHttpServletResponse mockResponse = new MockHttpServletResponse();
-		FilterChain mockFilterChain = mock(FilterChain.class);
-
-		filter.doFilter(mockRequest, mockResponse, mockFilterChain);
+
+		filter.doFilter(mockRequest, mockResponse, filterChain);
 
 		ArgumentCaptor<Supplier<Authentication>> authenticationCaptor = ArgumentCaptor.forClass(Supplier.class);
 		verify(mockAuthorizationManager).authorize(authenticationCaptor.capture(), eq(mockRequest));
 		Supplier<Authentication> authentication = authenticationCaptor.getValue();
 		assertThat(authentication.get()).isEqualTo(authenticationToken);
 
-		verify(mockFilterChain).doFilter(mockRequest, mockResponse);
+		verify(filterChain).doFilter(mockRequest, mockResponse);
 		verify(strategy).getContext();
 	}
 
@@ -127,13 +129,12 @@
 
 		MockHttpServletRequest mockRequest = new MockHttpServletRequest(null, "/path");
 		MockHttpServletResponse mockResponse = new MockHttpServletResponse();
-		FilterChain mockFilterChain = mock(FilterChain.class);
 
 		willThrow(new AccessDeniedException("Access Denied")).given(mockAuthorizationManager)
 			.authorize(any(), eq(mockRequest));
 
 		assertThatExceptionOfType(AccessDeniedException.class)
-			.isThrownBy(() -> filter.doFilter(mockRequest, mockResponse, mockFilterChain))
+			.isThrownBy(() -> filter.doFilter(mockRequest, mockResponse, filterChain))
 			.withMessage("Access Denied");
 
 		ArgumentCaptor<Supplier<Authentication>> authenticationCaptor = ArgumentCaptor.forClass(Supplier.class);
@@ -141,7 +142,7 @@
 		Supplier<Authentication> authentication = authenticationCaptor.getValue();
 		assertThat(authentication.get()).isEqualTo(authenticationToken);
 
-		verifyNoInteractions(mockFilterChain);
+		verifyNoInteractions(filterChain);
 	}
 
 	@Test
@@ -149,13 +150,12 @@
 		AuthorizationFilter filter = new AuthorizationFilter(AuthenticatedAuthorizationManager.authenticated());
 		MockHttpServletRequest mockRequest = new MockHttpServletRequest(null, "/path");
 		MockHttpServletResponse mockResponse = new MockHttpServletResponse();
-		FilterChain mockFilterChain = mock(FilterChain.class);
 
 		assertThatExceptionOfType(AuthenticationCredentialsNotFoundException.class)
-			.isThrownBy(() -> filter.doFilter(mockRequest, mockResponse, mockFilterChain))
+			.isThrownBy(() -> filter.doFilter(mockRequest, mockResponse, filterChain))
 			.withMessage("An Authentication object was not found in the SecurityContext");
 
-		verifyNoInteractions(mockFilterChain);
+		verifyNoInteractions(filterChain);
 	}
 
 	@Test
@@ -179,7 +179,6 @@
 				AuthenticatedAuthorizationManager.authenticated());
 		MockHttpServletRequest mockRequest = new MockHttpServletRequest(null, "/path");
 		MockHttpServletResponse mockResponse = new MockHttpServletResponse();
-		FilterChain mockFilterChain = mock(FilterChain.class);
 
 		SecurityContext securityContext = new SecurityContextImpl();
 		securityContext.setAuthentication(new TestingAuthenticationToken("user", "password", "ROLE_USER"));
@@ -187,7 +186,7 @@
 
 		AuthorizationEventPublisher eventPublisher = mock(AuthorizationEventPublisher.class);
 		authorizationFilter.setAuthorizationEventPublisher(eventPublisher);
-		authorizationFilter.doFilter(mockRequest, mockResponse, mockFilterChain);
+		authorizationFilter.doFilter(mockRequest, mockResponse, filterChain);
 		verify(eventPublisher).publishAuthorizationEvent(any(Supplier.class), any(HttpServletRequest.class),
 				any(AuthorizationDecision.class));
 	}
@@ -200,9 +199,8 @@
 		mockRequest.setDispatcherType(DispatcherType.ERROR);
 		mockRequest.setAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE, "/error");
 		MockHttpServletResponse mockResponse = new MockHttpServletResponse();
-		FilterChain mockFilterChain = mock(FilterChain.class);
-
-		authorizationFilter.doFilter(mockRequest, mockResponse, mockFilterChain);
+
+		authorizationFilter.doFilter(mockRequest, mockResponse, filterChain);
 		verify(authorizationManager).authorize(any(Supplier.class), eq(mockRequest));
 	}
 
@@ -217,9 +215,8 @@
 		mockRequest.setDispatcherType(DispatcherType.ERROR);
 		mockRequest.setAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE, "/error");
 		MockHttpServletResponse mockResponse = new MockHttpServletResponse();
-		FilterChain mockFilterChain = mock(FilterChain.class);
-
-		authorizationFilter.doFilter(mockRequest, mockResponse, mockFilterChain);
+
+		authorizationFilter.doFilter(mockRequest, mockResponse, filterChain);
 		verifyNoInteractions(authorizationManager);
 	}
 
