--- a/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2DeviceVerificationEndpointFilterTests.java
+++ b/oauth2/oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2DeviceVerificationEndpointFilterTests.java
@@ -87,11 +87,14 @@
 
 	private AuthenticationManager authenticationManager;
 
+	private FilterChain deviceVerificationFilterChain;
+
 	private OAuth2DeviceVerificationEndpointFilter filter;
 
 	@BeforeEach
 	public void setUp() {
 		this.authenticationManager = mock(AuthenticationManager.class);
+		this.deviceVerificationFilterChain = mock(FilterChain.class);
 		this.filter = new OAuth2DeviceVerificationEndpointFilter(this.authenticationManager);
 		mockAuthorizationServerContext();
 	}
@@ -160,9 +163,8 @@
 	public void doFilterWhenNotDeviceVerificationRequestThenNotProcessed() throws Exception {
 		MockHttpServletRequest request = new MockHttpServletRequest(HttpMethod.GET.name(), "/path");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
-		verify(filterChain).doFilter(request, response);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
+		verify(deviceVerificationFilterChain).doFilter(request, response);
 		verifyNoInteractions(this.authenticationManager);
 	}
 
@@ -184,15 +186,14 @@
 		request.addParameter("custom-param-1", "custom-value-1");
 		request.addParameter("custom-param-2", "custom-value-1", "custom-value-2");
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.FOUND.value());
 		assertThat(response.getHeader(HttpHeaders.LOCATION)).isEqualTo("/?success");
 
 		ArgumentCaptor<OAuth2DeviceAuthorizationConsentAuthenticationToken> authenticationCaptor = ArgumentCaptor
 			.forClass(OAuth2DeviceAuthorizationConsentAuthenticationToken.class);
 		verify(this.authenticationManager).authenticate(authenticationCaptor.capture());
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 
 		OAuth2DeviceAuthorizationConsentAuthenticationToken deviceAuthorizationConsentAuthentication = authenticationCaptor
 			.getValue();
@@ -220,15 +221,14 @@
 		request.addParameter("custom-param-1", "custom-value-1");
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.FOUND.value());
 		assertThat(response.getHeader(HttpHeaders.LOCATION)).isEqualTo("/?success");
 
 		ArgumentCaptor<OAuth2DeviceVerificationAuthenticationToken> authenticationCaptor = ArgumentCaptor
 			.forClass(OAuth2DeviceVerificationAuthenticationToken.class);
 		verify(this.authenticationManager).authenticate(authenticationCaptor.capture());
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 
 		OAuth2DeviceVerificationAuthenticationToken deviceVerificationAuthentication = authenticationCaptor.getValue();
 		assertThat(deviceVerificationAuthentication.getPrincipal()).isInstanceOf(TestingAuthenticationToken.class);
@@ -246,8 +246,7 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
 		assertThat(response.getContentType())
 			.isEqualTo(new MediaType("text", "html", StandardCharsets.UTF_8).toString());
@@ -255,7 +254,7 @@
 		assertThat(response.getContentAsString()).contains(scopeCheckbox("scope-2"));
 
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	@Test
@@ -268,8 +267,7 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
 		assertThat(response.getContentType())
 			.isEqualTo(new MediaType("text", "html", StandardCharsets.UTF_8).toString());
@@ -277,7 +275,7 @@
 		assertThat(response.getContentAsString()).contains(scopeCheckbox("scope-2"));
 
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	@Test
@@ -293,9 +291,8 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 		this.filter.setConsentPage("/consent");
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		String redirectUri = UriComponentsBuilder.fromUriString("https://provider.com/consent")
 			.queryParam(OAuth2ParameterNames.SCOPE, "scope-1 scope-2")
 			.queryParam(OAuth2ParameterNames.CLIENT_ID, CLIENT_ID)
@@ -306,7 +303,7 @@
 		assertThat(response.getHeader(HttpHeaders.LOCATION)).isEqualTo(redirectUri);
 
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	@Test
@@ -325,14 +322,13 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.FOUND.value());
 		assertThat(response.getHeader(HttpHeaders.LOCATION)).isEqualTo("/?success");
 
 		verify(authenticationConverter).convert(request);
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	@Test
@@ -344,7 +340,6 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
 
 		@SuppressWarnings("unchecked")
 		AuthenticationDetailsSource<HttpServletRequest, WebAuthenticationDetails> authenticationDetailsSource = mock(
@@ -353,13 +348,13 @@
 			.willReturn(new WebAuthenticationDetails(request));
 		this.filter.setAuthenticationDetailsSource(authenticationDetailsSource);
 
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.FOUND.value());
 		assertThat(response.getHeader(HttpHeaders.LOCATION)).isEqualTo("/?success");
 
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
 		verify(authenticationDetailsSource).buildDetails(request);
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	@Test
@@ -374,13 +369,12 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
 
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
 		verify(authenticationSuccessHandler).onAuthenticationSuccess(request, response, authenticationResult);
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	@Test
@@ -396,13 +390,12 @@
 		request.addParameter(OAuth2ParameterNames.USER_CODE, USER_CODE);
 		updateQueryString(request);
 		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
-		this.filter.doFilter(request, response, filterChain);
+		this.filter.doFilter(request, response, deviceVerificationFilterChain);
 		assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
 
 		verify(this.authenticationManager).authenticate(any(Authentication.class));
 		verify(authenticationFailureHandler).onAuthenticationFailure(request, response, authenticationException);
-		verifyNoInteractions(filterChain);
+		verifyNoInteractions(deviceVerificationFilterChain);
 	}
 
 	private static void mockAuthorizationServerContext() {
