--- a/web/src/test/java/org/springframework/security/web/context/HttpSessionSecurityContextRepositoryTests.java
+++ b/web/src/test/java/org/springframework/security/web/context/HttpSessionSecurityContextRepositoryTests.java
@@ -415,9 +415,8 @@
 		HttpSessionSecurityContextRepository repo = new HttpSessionSecurityContextRepository();
 		repo.setSpringSecurityContextKey("imTheContext");
 		MockHttpServletRequest request = new MockHttpServletRequest();
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		ServletOutputStream outputstream = mock(ServletOutputStream.class);
-		given(response.getOutputStream()).willReturn(outputstream);
+		HttpServletResponse response = createMockHttpServletResponse(outputstream);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(request, response);
 		SecurityContextHolder.setContext(repo.loadContext(holder));
 		SecurityContextHolder.getContext().setAuthentication(this.testToken);
@@ -431,9 +430,8 @@
 		HttpSessionSecurityContextRepository repo = new HttpSessionSecurityContextRepository();
 		repo.setSpringSecurityContextKey("imTheContext");
 		MockHttpServletRequest request = new MockHttpServletRequest();
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		ServletOutputStream outputstream = mock(ServletOutputStream.class);
-		given(response.getOutputStream()).willReturn(outputstream);
+		HttpServletResponse response = createMockHttpServletResponse(outputstream);
 		HttpRequestResponseHolder holder = new HttpRequestResponseHolder(request, response);
 		SecurityContextHolder.setContext(repo.loadContext(holder));
 		SecurityContextHolder.getContext().setAuthentication(this.testToken);
@@ -808,6 +806,12 @@
 		return securityContext;
 	}
 
+	private static HttpServletResponse createMockHttpServletResponse(ServletOutputStream outputstream) {
+		HttpServletResponse response = mock(HttpServletResponse.class);
+		given(response.getOutputStream()).willReturn(outputstream);
+		return response;
+	}
+
 	@Transient
 	private static class SomeTransientAuthentication extends AbstractAuthenticationToken {
 
