--- a/web/src/test/java/org/springframework/security/web/authentication/ott/OneTimeTokenAuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/ott/OneTimeTokenAuthenticationFilterTests.java
@@ -60,6 +60,9 @@
 	@Mock
 	private AuthenticationManager authenticationManager;
 
+	@Mock
+	private HttpServletResponse mockedResponse;
+
 	private final OneTimeTokenAuthenticationFilter filter = new OneTimeTokenAuthenticationFilter();
 
 	private final HttpServletResponse response = new MockHttpServletResponse();
@@ -79,10 +82,9 @@
 	@SuppressWarnings("removal")
 	void doFilterWhenUrlDoesNotMatchThenContinues() throws ServletException, IOException {
 		OneTimeTokenAuthenticationConverter converter = mock(OneTimeTokenAuthenticationConverter.class);
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		this.filter.setAuthenticationConverter(converter);
-		this.filter.doFilter(post("/nomatch").buildRequest(new MockServletContext()), response, this.chain);
-		verifyNoInteractions(converter, response);
+		this.filter.doFilter(post("/nomatch").buildRequest(new MockServletContext()), mockedResponse, this.chain);
+		verifyNoInteractions(converter, mockedResponse);
 		verify(this.chain).doFilter(any(), any());
 	}
 
@@ -90,10 +92,9 @@
 	@SuppressWarnings("removal")
 	void doFilterWhenMethodDoesNotMatchThenContinues() throws ServletException, IOException {
 		OneTimeTokenAuthenticationConverter converter = mock(OneTimeTokenAuthenticationConverter.class);
-		HttpServletResponse response = mock(HttpServletResponse.class);
 		this.filter.setAuthenticationConverter(converter);
-		this.filter.doFilter(get("/login/ott").buildRequest(new MockServletContext()), response, this.chain);
-		verifyNoInteractions(converter, response);
+		this.filter.doFilter(get("/login/ott").buildRequest(new MockServletContext()), mockedResponse, this.chain);
+		verifyNoInteractions(converter, mockedResponse);
 		verify(this.chain).doFilter(any(), any());
 	}
 
