--- a/ldap/src/test/java/org/springframework/security/ldap/ppolicy/PasswordPolicyControlFactoryTests.java
+++ b/ldap/src/test/java/org/springframework/security/ldap/ppolicy/PasswordPolicyControlFactoryTests.java
@@ -32,20 +32,24 @@
 	@Test
 	public void returnsNullForUnrecognisedOID() {
 		PasswordPolicyControlFactory ctrlFactory = new PasswordPolicyControlFactory();
-		Control wrongCtrl = mock(Control.class);
-		given(wrongCtrl.getID()).willReturn("wrongId");
+		Control wrongCtrl = createMockControl("wrongId");
 		assertThat(ctrlFactory.getControlInstance(wrongCtrl)).isNull();
 	}
 
 	@Test
 	public void returnsControlForCorrectOID() {
 		PasswordPolicyControlFactory ctrlFactory = new PasswordPolicyControlFactory();
-		Control control = mock(Control.class);
-		given(control.getID()).willReturn(PasswordPolicyControl.OID);
+		Control control = createMockControl(PasswordPolicyControl.OID);
 		given(control.getEncodedValue()).willReturn(PasswordPolicyResponseControlTests.OPENLDAP_LOCKED_CTRL);
 		Control result = ctrlFactory.getControlInstance(control);
 		assertThat(result).isNotNull();
 		assertThat(PasswordPolicyResponseControlTests.OPENLDAP_LOCKED_CTRL).isEqualTo(result.getEncodedValue());
 	}
 
+	private static Control createMockControl(String id) {
+		Control control = mock(Control.class);
+		given(control.getID()).willReturn(id);
+		return control;
+	}
+
 }
