--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/JeeConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/JeeConfigurerTests.java
@@ -81,8 +81,7 @@
 	@Test
 	public void jeeWhenInvokedTwiceThenUsesOriginalMappableRoles() throws Exception {
 		this.spring.register(InvokeTwiceDoesNotOverride.class).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("user");
+		Principal user = MockPrincipal.createMockPrincipal("user");
 		// @formatter:off
 		MockHttpServletRequestBuilder authRequest = get("/")
 				.principal(user)
@@ -98,8 +97,7 @@
 	@Test
 	public void requestWhenJeeMappableRolesInLambdaThenAuthenticatedWithMappableRoles() throws Exception {
 		this.spring.register(JeeMappableRolesConfig.class).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("user");
+		Principal user = MockPrincipal.createMockPrincipal("user");
 		// @formatter:off
 		MockHttpServletRequestBuilder authRequest = get("/")
 				.principal(user)
@@ -115,8 +113,7 @@
 	@Test
 	public void requestWhenJeeMappableAuthoritiesInLambdaThenAuthenticatedWithMappableAuthorities() throws Exception {
 		this.spring.register(JeeMappableAuthoritiesConfig.class).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("user");
+		Principal user = MockPrincipal.createMockPrincipal("user");
 		// @formatter:off
 		MockHttpServletRequestBuilder authRequest = get("/")
 				.principal(user)
@@ -138,10 +135,9 @@
 		AuthenticationUserDetailsService<PreAuthenticatedAuthenticationToken> userDetailsService = this.spring
 			.getContext()
 			.getBean(AuthenticationUserDetailsService.class);
-		Principal user = mock(Principal.class);
 		User userDetails = new User("user", "N/A", true, true, true, true,
 				AuthorityUtils.createAuthorityList("ROLE_USER"));
-		given(user.getName()).willReturn("user");
+		Principal user = MockPrincipal.createMockPrincipal("user");
 		given(userDetailsService.loadUserDetails(any())).willReturn(userDetails);
 		// @formatter:off
 		MockHttpServletRequestBuilder authRequest = get("/")
--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/NamespaceHttpJeeTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/NamespaceHttpJeeTests.java
@@ -64,8 +64,7 @@
 	@Test
 	public void requestWhenJeeUserThenBehaviorDiffersFromNamespaceForRoleNames() throws Exception {
 		this.spring.register(JeeMappableRolesConfig.class, BaseController.class).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("joe");
+		Principal user = MockPrincipal.createMockPrincipal("joe");
 		this.mvc.perform(get("/roles").principal(user).with((request) -> {
 			request.addUserRole("ROLE_admin");
 			request.addUserRole("ROLE_user");
@@ -77,8 +76,7 @@
 	@Test
 	public void requestWhenCustomAuthenticatedUserDetailsServiceThenBehaviorMatchesNamespace() throws Exception {
 		this.spring.register(JeeUserServiceRefConfig.class, BaseController.class).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("joe");
+		Principal user = MockPrincipal.createMockPrincipal("joe");
 		User result = new User(user.getName(), "N/A", true, true, true, true,
 				AuthorityUtils.createAuthorityList("ROLE_user"));
 		given(bean(AuthenticationUserDetailsService.class).loadUserDetails(any())).willReturn(result);
--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/MockPrincipal.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/MockPrincipal.java
@@ -0,0 +1,20 @@
+package org.springframework.security.config.annotation.web.configurers;
+
+import java.security.Principal;
+
+import org.mockito.Mockito;
+
+import static org.mockito.BDDMockito.given;
+
+public final class MockPrincipal {
+
+	private MockPrincipal() {
+	}
+
+	public static Principal createMockPrincipal(String name) {
+		Principal user = Mockito.mock(Principal.class);
+		given(user.getName()).willReturn(name);
+		return user;
+	}
+
+}
