--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcReactiveOAuth2UserServiceTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcReactiveOAuth2UserServiceTests.java
@@ -203,9 +203,9 @@
 		OAuth2User oauth2User = new DefaultOAuth2User(AuthorityUtils.createAuthorityList("ROLE_USER"), attributes,
 				"user");
 		given(this.oauth2UserService.loadUser(any())).willReturn(Mono.just(oauth2User));
-		Predicate<OidcUserRequest> customRetrieveUserInfo = mock(Predicate.class);
+		Predicate<OidcUserRequest> customRetrieveUserInfo = MockOidcUserRequestRetrieveUserInfoPredicate
+				.createMockOidcUserRequestRetrieveUserInfoPredicate();
 		this.userService.setRetrieveUserInfo(customRetrieveUserInfo);
-		given(customRetrieveUserInfo.test(any(OidcUserRequest.class))).willReturn(true);
 		// @formatter:off
 		OAuth2AccessToken accessToken = new OAuth2AccessToken(
 				this.accessToken.getTokenType(),
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcUserServiceTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcUserServiceTests.java
@@ -172,8 +172,8 @@
 		String userInfoUri = this.server.url("/user").toString();
 		ClientRegistration clientRegistration = this.clientRegistrationBuilder.userInfoUri(userInfoUri).build();
 		this.accessToken = TestOAuth2AccessTokens.noScopes();
-		Predicate<OidcUserRequest> customRetrieveUserInfo = mock(Predicate.class);
-		given(customRetrieveUserInfo.test(any(OidcUserRequest.class))).willReturn(true);
+		Predicate<OidcUserRequest> customRetrieveUserInfo = MockOidcUserRequestRetrieveUserInfoPredicate
+			.createMockOidcUserRequestRetrieveUserInfoPredicate();
 		this.userService.setRetrieveUserInfo(customRetrieveUserInfo);
 		OidcUser user = this.userService
 			.loadUser(new OidcUserRequest(clientRegistration, this.accessToken, this.idToken));
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/MockOidcUserRequestRetrieveUserInfoPredicate.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/userinfo/MockOidcUserRequestRetrieveUserInfoPredicate.java
@@ -0,0 +1,21 @@
+package org.springframework.security.oauth2.client.oidc.userinfo;
+
+import java.util.function.Predicate;
+
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+public final class MockOidcUserRequestRetrieveUserInfoPredicate {
+
+	private MockOidcUserRequestRetrieveUserInfoPredicate() {
+	}
+
+	public static Predicate<OidcUserRequest> createMockOidcUserRequestRetrieveUserInfoPredicate() {
+		Predicate<OidcUserRequest> customRetrieveUserInfo = Mockito.mock(Predicate.class);
+		given(customRetrieveUserInfo.test(any(OidcUserRequest.class))).willReturn(true);
+		return customRetrieveUserInfo;
+	}
+
+}
