--- a/config/src/test/java/org/springframework/security/config/http/MiscHttpConfigTests.java
+++ b/config/src/test/java/org/springframework/security/config/http/MiscHttpConfigTests.java
@@ -705,8 +705,7 @@
 	@Test
 	public void loginWhenJeeFilterThenExtractsRoles() throws Exception {
 		this.spring.configLocations(xml("JeeFilter")).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("joe");
+		Principal user = createMockPrincipal();
 		// @formatter:off
 		MockHttpServletRequestBuilder rolesRequest = get("/roles")
 				.principal(user)
@@ -724,8 +723,7 @@
 	@Test
 	public void loginWhenJeeFilterCustomSecurityContextHolderStrategyThenUses() throws Exception {
 		this.spring.configLocations(xml("JeeFilterWithSecurityContextHolderStrategy")).autowire();
-		Principal user = mock(Principal.class);
-		given(user.getName()).willReturn("joe");
+		Principal user = createMockPrincipal();
 		// @formatter:off
 		MockHttpServletRequestBuilder rolesRequest = get("/roles")
 				.principal(user)
@@ -922,6 +920,12 @@
 		return httpBasic("poster", "password");
 	}
 
+	private static Principal createMockPrincipal() {
+		Principal user = mock(Principal.class);
+		given(user.getName()).willReturn("joe");
+		return user;
+	}
+
 	private static String xml(String configName) {
 		return CONFIG_LOCATION_PREFIX + "-" + configName + ".xml";
 	}
