--- a/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/SupplierReactiveJwtDecoderTests.java
+++ b/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/SupplierReactiveJwtDecoderTests.java
@@ -55,7 +55,7 @@
 	@Test
 	public void decodeWhenInitializedThenCaches() {
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
-		Supplier<ReactiveJwtDecoder> supplier = mock(Supplier.class);
+		Supplier<ReactiveJwtDecoder> supplier = createMockReactiveJwtDecoderSupplier();
 		given(supplier.get()).willReturn(jwtDecoder);
 		given(jwtDecoder.decode("token")).willReturn(Mono.empty());
 		ReactiveJwtDecoder supplierReactiveJwtDecoder = new SupplierReactiveJwtDecoder(supplier);
@@ -68,7 +68,7 @@
 	@Test
 	public void decodeWhenInitializationInitiallyFailsThenRecoverable() {
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
-		Supplier<ReactiveJwtDecoder> broken = mock(Supplier.class);
+		Supplier<ReactiveJwtDecoder> broken = createMockReactiveJwtDecoderSupplier();
 		given(broken.get()).willThrow(RuntimeException.class);
 		given(jwtDecoder.decode("token")).willReturn(Mono.empty());
 		ReactiveJwtDecoder supplierReactiveJwtDecoder = new SupplierReactiveJwtDecoder(broken);
@@ -80,4 +80,8 @@
 		verify(jwtDecoder).decode("token");
 	}
 
+	private static Supplier<ReactiveJwtDecoder> createMockReactiveJwtDecoderSupplier() {
+		return mock(Supplier.class);
+	}
+
 }
