--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/OidcIdTokenDecoderFactoryTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/OidcIdTokenDecoderFactoryTests.java
@@ -148,11 +148,10 @@
 
 	@Test
 	public void createDecoderWhenCustomJwtValidatorFactorySetThenApplied() {
-		Function<ClientRegistration, OAuth2TokenValidator<Jwt>> customJwtValidatorFactory = mock(Function.class);
+		ClientRegistration clientRegistration = this.registration.build();
+		Function<ClientRegistration, OAuth2TokenValidator<Jwt>> customJwtValidatorFactory = MockCustomJwtValidatorFactory
+			.createMockCustomJwtValidatorFactory(clientRegistration);
 		this.idTokenDecoderFactory.setJwtValidatorFactory(customJwtValidatorFactory);
-		ClientRegistration clientRegistration = this.registration.build();
-		given(customJwtValidatorFactory.apply(same(clientRegistration)))
-			.willReturn(new OidcIdTokenValidator(clientRegistration));
 		this.idTokenDecoderFactory.createDecoder(clientRegistration);
 		verify(customJwtValidatorFactory).apply(same(clientRegistration));
 	}
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/ReactiveOidcIdTokenDecoderFactoryTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/ReactiveOidcIdTokenDecoderFactoryTests.java
@@ -148,11 +148,10 @@
 
 	@Test
 	public void createDecoderWhenCustomJwtValidatorFactorySetThenApplied() {
-		Function<ClientRegistration, OAuth2TokenValidator<Jwt>> customJwtValidatorFactory = mock(Function.class);
+		ClientRegistration clientRegistration = this.registration.build();
+		Function<ClientRegistration, OAuth2TokenValidator<Jwt>> customJwtValidatorFactory = MockCustomJwtValidatorFactory
+			.createMockCustomJwtValidatorFactory(clientRegistration);
 		this.idTokenDecoderFactory.setJwtValidatorFactory(customJwtValidatorFactory);
-		ClientRegistration clientRegistration = this.registration.build();
-		given(customJwtValidatorFactory.apply(same(clientRegistration)))
-			.willReturn(new OidcIdTokenValidator(clientRegistration));
 		this.idTokenDecoderFactory.createDecoder(clientRegistration);
 		verify(customJwtValidatorFactory).apply(same(clientRegistration));
 	}
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/MockCustomJwtValidatorFactory.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/MockCustomJwtValidatorFactory.java
@@ -0,0 +1,24 @@
+package org.springframework.security.oauth2.client.oidc.authentication;
+
+import java.util.function.Function;
+
+import org.mockito.BDDMockito;
+import org.mockito.Mockito;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+public final class MockCustomJwtValidatorFactory {
+
+	private MockCustomJwtValidatorFactory() {
+	}
+
+	public static Function<ClientRegistration, OAuth2TokenValidator<Jwt>> createMockCustomJwtValidatorFactory(
+			ClientRegistration clientRegistration) {
+		Function<ClientRegistration, OAuth2TokenValidator<Jwt>> customJwtValidatorFactory = Mockito.mock(Function.class);
+		BDDMockito.given(customJwtValidatorFactory.apply(Mockito.same(clientRegistration)))
+			.willReturn(new OidcIdTokenValidator(clientRegistration));
+		return customJwtValidatorFactory;
+	}
+
+}
