--- a/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/SupplierJwtDecoderTests.java
+++ b/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/SupplierJwtDecoderTests.java
@@ -54,8 +54,7 @@
 	@Test
 	public void decodeWhenInitializedThenCaches() {
 		JwtDecoder jwtDecoder = mock(JwtDecoder.class);
-		Supplier<JwtDecoder> supplier = mock(Supplier.class);
-		given(supplier.get()).willReturn(jwtDecoder);
+		Supplier<JwtDecoder> supplier = createMockJwtDecoderSupplier(jwtDecoder);
 		JwtDecoder supplierJwtDecoder = new SupplierJwtDecoder(supplier);
 		supplierJwtDecoder.decode("token");
 		supplierJwtDecoder.decode("token");
@@ -66,7 +65,7 @@
 	@Test
 	public void decodeWhenInitializationInitiallyFailsThenRecoverable() {
 		JwtDecoder jwtDecoder = mock(JwtDecoder.class);
-		Supplier<JwtDecoder> broken = mock(Supplier.class);
+		Supplier<JwtDecoder> broken = createMockJwtDecoderSupplier(jwtDecoder);
 		given(broken.get()).willThrow(RuntimeException.class);
 		JwtDecoder supplierJwtDecoder = new SupplierJwtDecoder(broken);
 		assertThatExceptionOfType(JwtDecoderInitializationException.class)
@@ -77,4 +76,10 @@
 		verify(jwtDecoder).decode("token");
 	}
 
+	private static Supplier<JwtDecoder> createMockJwtDecoderSupplier(JwtDecoder jwtDecoder) {
+		Supplier<JwtDecoder> supplier = mock(Supplier.class);
+		given(supplier.get()).willReturn(jwtDecoder);
+		return supplier;
+	}
+
 }
