--- a/saml2/saml2-service-provider/src/opensaml5Test/java/org/springframework/security/saml2/provider/service/registration/OpenSaml5AssertingPartyMetadataRepositoryTests.java
+++ b/saml2/saml2-service-provider/src/opensaml5Test/java/org/springframework/security/saml2/provider/service/registration/OpenSaml5AssertingPartyMetadataRepositoryTests.java
@@ -76,6 +76,13 @@
 
 	private static MockWebServer web = new MockWebServer();
 
+	private MetadataResolver metadataResolver(Class<?>... extraInterfaces) {
+		if (extraInterfaces.length == 0) {
+			return mock(MetadataResolver.class);
+		}
+		return mock(MetadataResolver.class, withSettings().extraInterfaces(extraInterfaces));
+	}
+
 	private static String readFile(String fileName) {
 		try {
 			ClassPathResource resource = new ClassPathResource(fileName);
@@ -280,7 +287,7 @@
 
 	@Test
 	public void constructorWhenNoIndexAndNoIteratorThenException() {
-		MetadataResolver resolver = mock(MetadataResolver.class);
+		MetadataResolver resolver = metadataResolver();
 		assertThatExceptionOfType(IllegalArgumentException.class)
 			.isThrownBy(() -> new OpenSaml5AssertingPartyMetadataRepository(resolver));
 	}
@@ -289,8 +296,7 @@
 	public void constructorWhenIterableResolverThenUses() {
 		RelyingPartyRegistration registration = TestRelyingPartyRegistrations.full().build();
 		EntityDescriptor descriptor = TestOpenSamlObjects.entityDescriptor(registration);
-		MetadataResolver resolver = mock(MetadataResolver.class,
-				withSettings().extraInterfaces(IterableMetadataSource.class));
+		MetadataResolver resolver = metadataResolver(IterableMetadataSource.class);
 		given(((IterableMetadataSource) resolver).iterator()).willReturn(List.of(descriptor).iterator());
 		AssertingPartyMetadataRepository parties = new OpenSaml5AssertingPartyMetadataRepository(resolver);
 		parties.iterator()
