--- a/core/src/test/java/org/springframework/security/core/context/ObservationSecurityContextChangedListenerTests.java
+++ b/core/src/test/java/org/springframework/security/core/context/ObservationSecurityContextChangedListenerTests.java
@@ -43,11 +43,14 @@
 
 	private ObservationRegistry observationRegistry;
 
+	private Observation currentObservation;
+
 	private ObservationSecurityContextChangedListener tested;
 
 	@BeforeEach
 	void setup() {
 		this.observationRegistry = mock(ObservationRegistry.class);
+		this.currentObservation = mock(Observation.class);
 		this.tested = new ObservationSecurityContextChangedListener(this.observationRegistry);
 	}
 
@@ -59,13 +62,12 @@
 
 	@Test
 	void securityContextChangedWhenClearedEventThenAddsClearEventToObservation() {
-		Observation observation = mock(Observation.class);
-		given(this.observationRegistry.getCurrentObservation()).willReturn(observation);
+		given(this.observationRegistry.getCurrentObservation()).willReturn(currentObservation);
 		Supplier<SecurityContext> one = mock(Supplier.class);
 		this.tested
 			.securityContextChanged(new SecurityContextChangedEvent(one, SecurityContextChangedEvent.NO_CONTEXT));
 		ArgumentCaptor<Observation.Event> event = ArgumentCaptor.forClass(Observation.Event.class);
-		verify(observation).event(event.capture());
+		verify(currentObservation).event(event.capture());
 		assertThat(event.getValue().getName())
 			.isEqualTo(ObservationSecurityContextChangedListener.SECURITY_CONTEXT_CLEARED);
 		verifyNoInteractions(one);
@@ -73,49 +75,44 @@
 
 	@Test
 	void securityContextChangedWhenNoChangeThenNoEventAddedToObservation() {
-		Observation observation = mock(Observation.class);
-		given(this.observationRegistry.getCurrentObservation()).willReturn(observation);
+		given(this.observationRegistry.getCurrentObservation()).willReturn(currentObservation);
 		this.tested.securityContextChanged(new SecurityContextChangedEvent(this.one, this.one));
-		verifyNoInteractions(observation);
+		verifyNoInteractions(currentObservation);
 	}
 
 	@Test
 	void securityContextChangedWhenChangedEventThenAddsChangeEventToObservation() {
-		Observation observation = mock(Observation.class);
-		given(this.observationRegistry.getCurrentObservation()).willReturn(observation);
+		given(this.observationRegistry.getCurrentObservation()).willReturn(currentObservation);
 		this.tested.securityContextChanged(new SecurityContextChangedEvent(this.one, this.two));
 		ArgumentCaptor<Observation.Event> event = ArgumentCaptor.forClass(Observation.Event.class);
-		verify(observation).event(event.capture());
+		verify(currentObservation).event(event.capture());
 		assertThat(event.getValue().getName())
 			.isEqualTo(ObservationSecurityContextChangedListener.SECURITY_CONTEXT_CHANGED);
 	}
 
 	@Test
 	void securityContextChangedWhenCreatedEventThenAddsCreatedEventToObservation() {
-		Observation observation = mock(Observation.class);
-		given(this.observationRegistry.getCurrentObservation()).willReturn(observation);
+		given(this.observationRegistry.getCurrentObservation()).willReturn(currentObservation);
 		this.tested.securityContextChanged(new SecurityContextChangedEvent(null, this.one));
 		ArgumentCaptor<Observation.Event> event = ArgumentCaptor.forClass(Observation.Event.class);
-		verify(observation).event(event.capture());
+		verify(currentObservation).event(event.capture());
 		assertThat(event.getValue().getName())
 			.isEqualTo(ObservationSecurityContextChangedListener.SECURITY_CONTEXT_CREATED);
 	}
 
 	@Test
 	void securityContextChangedWhenClearedEventThenAddsClearedEventToObservation() {
-		Observation observation = mock(Observation.class);
-		given(this.observationRegistry.getCurrentObservation()).willReturn(observation);
+		given(this.observationRegistry.getCurrentObservation()).willReturn(currentObservation);
 		this.tested.securityContextChanged(new SecurityContextChangedEvent(this.one, new SecurityContextImpl()));
 		ArgumentCaptor<Observation.Event> event = ArgumentCaptor.forClass(Observation.Event.class);
-		verify(observation).event(event.capture());
+		verify(currentObservation).event(event.capture());
 		assertThat(event.getValue().getName())
 			.isEqualTo(ObservationSecurityContextChangedListener.SECURITY_CONTEXT_CLEARED);
 	}
 
 	@Test
 	void securityContextChangedWhenBothNullThenNoNullPointerException() {
-		Observation observation = mock(Observation.class);
-		given(this.observationRegistry.getCurrentObservation()).willReturn(observation);
+		given(this.observationRegistry.getCurrentObservation()).willReturn(currentObservation);
 		this.tested.securityContextChanged(
 				new SecurityContextChangedEvent(new SecurityContextImpl(), new SecurityContextImpl()));
 	}
