--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurerTests.java
@@ -24,6 +24,7 @@
 import io.micrometer.observation.ObservationRegistry;
 import io.micrometer.observation.ObservationTextPublisher;
 import jakarta.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.ArgumentCaptor;
@@ -119,6 +120,13 @@
 	@Autowired
 	MockMvc mvc;
 
+	private AuthorizationManager<RequestAuthorizationContext> authorizationManagerMock;
+
+	@BeforeEach
+	void setUp() {
+		this.authorizationManagerMock = mock();
+	}
+
 	@Test
 	public void configureWhenAuthorizedHttpRequestsAndNoRequestsThenException() {
 		assertThatExceptionOfType(BeanCreationException.class)
@@ -182,9 +190,8 @@
 
 	@Test
 	public void configureWhenCustomAuthorizationManagerFactoryRegisteredThenUsed() {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManagerFactoryConfig.authorizationManagerFactory = authorizationManagerFactory;
 		this.spring.register(AuthorizationManagerFactoryConfig.class).autowire();
 		verify(authorizationManagerFactory).permitAll();
@@ -584,9 +591,8 @@
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndPermitAllThenRespondsWithOk() throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> permitAll = spy(SingleResultAuthorizationManager.permitAll());
 		given(authorizationManagerFactory.permitAll()).willReturn(permitAll);
 		AuthorizationManagerFactoryConfig.authorizationManagerFactory = authorizationManagerFactory;
@@ -596,15 +602,14 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(permitAll).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(permitAll);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndDenyAllThenRespondsWithForbidden()
 			throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> denyAll = spy(SingleResultAuthorizationManager.denyAll());
 		given(authorizationManagerFactory.denyAll()).willReturn(denyAll);
 		AuthorizationManagerFactoryConfig.authorizationManagerFactory = authorizationManagerFactory;
@@ -614,14 +619,13 @@
 		this.mvc.perform(request).andExpect(status().isForbidden());
 		verify(denyAll).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(denyAll);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndHasRoleThenRespondsWithOk() throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> hasRole = spy(AuthorityAuthorizationManager.hasRole("ADMIN"));
 		given(authorizationManagerFactory.hasRole(anyString())).willReturn(hasRole);
 		AuthorizationManagerFactoryConfig.authorizationManagerFactory = authorizationManagerFactory;
@@ -631,14 +635,13 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(hasRole).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(hasRole);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndHasAnyRoleThenRespondsWithOk() throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> hasAnyRole = spy(
 				AuthorityAuthorizationManager.hasAnyRole("USER", "ADMIN"));
 		given(authorizationManagerFactory.hasAnyRole(any(String[].class))).willReturn(hasAnyRole);
@@ -649,15 +652,14 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(hasAnyRole).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(hasAnyRole);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void postWhenCustomAuthorizationManagerFactoryRegisteredAndHasAuthorityThenRespondsWithOk()
 			throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> hasAuthority = spy(
 				AuthorityAuthorizationManager.hasAuthority("write"));
 		given(authorizationManagerFactory.hasAuthority(anyString())).willReturn(hasAuthority);
@@ -670,15 +672,14 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(hasAuthority).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(hasAuthority);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndHasAnyAuthorityThenRespondsWithOk()
 			throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> hasAnyAuthority = spy(
 				AuthorityAuthorizationManager.hasAnyAuthority("resource.read", "read"));
 		given(authorizationManagerFactory.hasAnyAuthority(any(String[].class))).willReturn(hasAnyAuthority);
@@ -690,15 +691,14 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(hasAnyAuthority).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(hasAnyAuthority);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndAuthenticatedThenRespondsWithOk()
 			throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> authenticated = spy(
 				AuthenticatedAuthorizationManager.authenticated());
 		given(authorizationManagerFactory.authenticated()).willReturn(authenticated);
@@ -709,15 +709,14 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(authenticated).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(authenticated);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndFullyAuthenticatedThenRespondsWithOk()
 			throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> fullyAuthenticated = spy(
 				AuthenticatedAuthorizationManager.fullyAuthenticated());
 		given(authorizationManagerFactory.fullyAuthenticated()).willReturn(fullyAuthenticated);
@@ -728,14 +727,13 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(fullyAuthenticated).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(fullyAuthenticated);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndRememberMeThenRespondsWithOk() throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> rememberMe = spy(
 				AuthenticatedAuthorizationManager.rememberMe());
 		given(authorizationManagerFactory.rememberMe()).willReturn(rememberMe);
@@ -747,14 +745,13 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(rememberMe).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(rememberMe);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndAnonymousThenRespondsWithOk() throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactory<RequestAuthorizationContext> authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 		AuthorizationManager<RequestAuthorizationContext> anonymous = spy(
 				AuthenticatedAuthorizationManager.anonymous());
 		given(authorizationManagerFactory.anonymous()).willReturn(anonymous);
@@ -765,20 +762,19 @@
 		this.mvc.perform(request).andExpect(status().isOk());
 		verify(anonymous).authorize(any(), any(RequestAuthorizationContext.class));
 		verifyNoMoreInteractions(anonymous);
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
 	public void getWhenCustomAuthorizationManagerFactoryRegisteredAndAccessThenRespondsWithForbidden()
 			throws Exception {
-		AuthorizationManager<RequestAuthorizationContext> authorizationManager = mock();
 		AuthorizationManagerFactoryConfig.authorizationManagerFactory = mockAuthorizationManagerFactory(
-				authorizationManager);
+				authorizationManagerMock);
 
 		this.spring.register(AuthorizationManagerFactoryConfig.class).autowire();
 		MockHttpServletRequestBuilder request = get("/").with(user("user"));
 		this.mvc.perform(request).andExpect(status().isForbidden());
-		verifyNoInteractions(authorizationManager);
+		verifyNoInteractions(authorizationManagerMock);
 	}
 
 	@Test
