--- a/core/src/test/java/org/springframework/security/authentication/ott/reactive/OneTimeTokenReactiveAuthenticationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/ott/reactive/OneTimeTokenReactiveAuthenticationManagerTests.java
@@ -108,9 +108,7 @@
 	@Test
 	@SuppressWarnings("removal")
 	void authenticateWhenInvalidOneTimeTokenAuthenticationTokenIsPresentThenFail() {
-		ReactiveOneTimeTokenService oneTimeTokenService = mock(ReactiveOneTimeTokenService.class);
-		given(oneTimeTokenService.consume(ArgumentMatchers.any(OneTimeTokenAuthenticationToken.class)))
-			.willReturn(Mono.empty());
+		ReactiveOneTimeTokenService oneTimeTokenService = createMockReactiveOneTimeTokenService();
 		ReactiveUserDetailsService userDetailsService = mock(ReactiveUserDetailsService.class);
 
 		this.authenticationManager = new OneTimeTokenReactiveAuthenticationManager(oneTimeTokenService,
@@ -126,9 +124,7 @@
 	@Test
 	@SuppressWarnings("removal")
 	void authenticateWhenIncorrectTypeOfAuthenticationIsPresentThenFail() {
-		ReactiveOneTimeTokenService oneTimeTokenService = mock(ReactiveOneTimeTokenService.class);
-		given(oneTimeTokenService.consume(ArgumentMatchers.any(OneTimeTokenAuthenticationToken.class)))
-			.willReturn(Mono.empty());
+		ReactiveOneTimeTokenService oneTimeTokenService = createMockReactiveOneTimeTokenService();
 		ReactiveUserDetailsService userDetailsService = mock(ReactiveUserDetailsService.class);
 
 		this.authenticationManager = new OneTimeTokenReactiveAuthenticationManager(oneTimeTokenService,
@@ -142,4 +138,11 @@
 		assertThat(authentication).isNull();
 	}
 
+	private static ReactiveOneTimeTokenService createMockReactiveOneTimeTokenService() {
+		ReactiveOneTimeTokenService oneTimeTokenService = mock(ReactiveOneTimeTokenService.class);
+		given(oneTimeTokenService.consume(ArgumentMatchers.any(OneTimeTokenAuthenticationToken.class)))
+			.willReturn(Mono.empty());
+		return oneTimeTokenService;
+	}
+
 }
