--- a/core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java
@@ -22,6 +22,7 @@
 import java.util.List;
 
 import org.jspecify.annotations.Nullable;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.context.MessageSource;
@@ -48,6 +49,13 @@
  */
 public class ProviderManagerTests {
 
+	private AuthenticationEventPublisher authenticationEventPublisher;
+
+	@BeforeEach
+	void setUp() {
+		authenticationEventPublisher = mock(AuthenticationEventPublisher.class);
+	}
+
 	@Test
 	void authenticationFailsWithUnsupportedToken() {
 		Authentication token = new AbstractAuthenticationToken((Collection<? extends GrantedAuthority>) null) {
@@ -83,11 +91,10 @@
 	void authenticationSucceedsWithSupportedTokenAndReturnsExpectedObject() {
 		Authentication a = new TestingAuthenticationToken("user", "pass", "FACTOR");
 		ProviderManager mgr = new ProviderManager(createProviderWhichReturns(a));
-		AuthenticationEventPublisher publisher = mock(AuthenticationEventPublisher.class);
-		mgr.setAuthenticationEventPublisher(publisher);
+		mgr.setAuthenticationEventPublisher(authenticationEventPublisher);
 		Authentication result = mgr.authenticate(a);
 		assertThat(result.getPrincipal()).isEqualTo(a.getPrincipal());
-		verify(publisher).publishAuthenticationSuccess(result);
+		verify(authenticationEventPublisher).publishAuthenticationSuccess(result);
 	}
 
 	@Test
@@ -95,11 +102,10 @@
 		Authentication a = new TestingAuthenticationToken("user", "pass", "FACTOR");
 		ProviderManager mgr = new ProviderManager(
 				Arrays.asList(createProviderWhichReturns(null), createProviderWhichReturns(a)));
-		AuthenticationEventPublisher publisher = mock(AuthenticationEventPublisher.class);
-		mgr.setAuthenticationEventPublisher(publisher);
+		mgr.setAuthenticationEventPublisher(authenticationEventPublisher);
 		Authentication result = mgr.authenticate(a);
 		assertThat(result.getPrincipal()).isEqualTo(a.getPrincipal());
-		verify(publisher).publishAuthenticationSuccess(result);
+		verify(authenticationEventPublisher).publishAuthenticationSuccess(result);
 	}
 
 	@Test
@@ -231,16 +237,15 @@
 	@Test
 	void providerNotFoundFromParentIsIgnored() {
 		final Authentication authReq = mock(Authentication.class);
-		AuthenticationEventPublisher publisher = mock(AuthenticationEventPublisher.class);
 		AuthenticationManager parent = mock(AuthenticationManager.class);
 		given(parent.authenticate(authReq)).willThrow(new ProviderNotFoundException(""));
 		// Set a provider that throws an exception - this is the exception we expect to be
 		// propagated
 		ProviderManager mgr = new ProviderManager(List.of(createProviderWhichThrows(new BadCredentialsException(""))),
 				parent);
-		mgr.setAuthenticationEventPublisher(publisher);
+		mgr.setAuthenticationEventPublisher(authenticationEventPublisher);
 		assertThatExceptionOfType(BadCredentialsException.class).isThrownBy(() -> mgr.authenticate(authReq))
-			.satisfies((ex) -> verify(publisher).publishAuthenticationFailure(ex, authReq));
+			.satisfies((ex) -> verify(authenticationEventPublisher).publishAuthenticationFailure(ex, authReq));
 	}
 
 	@Test
@@ -249,8 +254,7 @@
 		ProviderManager mgr = new ProviderManager(List.of(createProviderWhichThrows(new BadCredentialsException(""))),
 				parent);
 		Authentication authReq = mock(Authentication.class);
-		AuthenticationEventPublisher publisher = mock(AuthenticationEventPublisher.class);
-		mgr.setAuthenticationEventPublisher(publisher);
+		mgr.setAuthenticationEventPublisher(authenticationEventPublisher);
 		// Set a provider that throws an exception - this is the exception we expect to be
 		// propagated
 		BadCredentialsException expected = new BadCredentialsException("I'm the one from the parent");
@@ -265,10 +269,9 @@
 		LockedException expected = new LockedException("");
 		ProviderManager mgr = new ProviderManager(List.of(createProviderWhichThrows(expected)), parent);
 		Authentication authReq = mock(Authentication.class);
-		AuthenticationEventPublisher publisher = mock(AuthenticationEventPublisher.class);
-		mgr.setAuthenticationEventPublisher(publisher);
+		mgr.setAuthenticationEventPublisher(authenticationEventPublisher);
 		assertThatExceptionOfType(LockedException.class).isThrownBy(() -> mgr.authenticate(authReq));
-		verify(publisher).publishAuthenticationFailure(expected, authReq);
+		verify(authenticationEventPublisher).publishAuthenticationFailure(expected, authReq);
 	}
 
 	@Test
@@ -317,15 +320,14 @@
 		ProviderManager parentMgr = new ProviderManager(createProviderWhichThrows(badCredentialsExParent));
 		ProviderManager childMgr = new ProviderManager(
 				List.of(createProviderWhichThrows(new BadCredentialsException("Bad Credentials in child"))), parentMgr);
-		AuthenticationEventPublisher publisher = mock(AuthenticationEventPublisher.class);
-		parentMgr.setAuthenticationEventPublisher(publisher);
-		childMgr.setAuthenticationEventPublisher(publisher);
+		parentMgr.setAuthenticationEventPublisher(authenticationEventPublisher);
+		childMgr.setAuthenticationEventPublisher(authenticationEventPublisher);
 		Authentication authReq = mock(Authentication.class);
 		assertThatExceptionOfType(BadCredentialsException.class).isThrownBy(() -> childMgr.authenticate(authReq))
 			.isSameAs(badCredentialsExParent);
-		verify(publisher).publishAuthenticationFailure(badCredentialsExParent, authReq); // Parent
+		verify(authenticationEventPublisher).publishAuthenticationFailure(badCredentialsExParent, authReq); // Parent
 																							// publishes
-		verifyNoMoreInteractions(publisher); // Child should not publish (duplicate event)
+		verifyNoMoreInteractions(authenticationEventPublisher); // Child should not publish (duplicate event)
 	}
 
 	private AuthenticationProvider createProviderWhichThrows(final AuthenticationException ex) {
