--- a/taglibs/src/test/java/org/springframework/security/taglibs/authz/AbstractAuthorizeTagTests.java
+++ b/taglibs/src/test/java/org/springframework/security/taglibs/authz/AbstractAuthorizeTagTests.java
@@ -63,12 +63,15 @@
 
 	private MockServletContext servletContext;
 
+	private WebInvocationPrivilegeEvaluator expectedPrivilegeEvaluator;
+
 	@BeforeEach
 	public void setup() {
 		this.tag = new AuthzTag();
 		this.request = new MockHttpServletRequest();
 		this.response = new MockHttpServletResponse();
 		this.servletContext = new MockServletContext();
+		this.expectedPrivilegeEvaluator = mock(WebInvocationPrivilegeEvaluator.class);
 	}
 
 	@AfterEach
@@ -82,11 +85,10 @@
 		this.servletContext.setAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE, wac);
 		given(wac.getBeanNamesForType(SecurityContextHolderStrategy.class)).willReturn(new String[0]);
 		String uri = "/something";
-		WebInvocationPrivilegeEvaluator expected = mock(WebInvocationPrivilegeEvaluator.class);
 		this.tag.setUrl(uri);
-		this.request.setAttribute(WebAttributes.WEB_INVOCATION_PRIVILEGE_EVALUATOR_ATTRIBUTE, expected);
+		this.request.setAttribute(WebAttributes.WEB_INVOCATION_PRIVILEGE_EVALUATOR_ATTRIBUTE, expectedPrivilegeEvaluator);
 		this.tag.authorizeUsingUrlCheck();
-		verify(expected).isAllowed(eq(""), eq(uri), eq("GET"), any());
+		verify(expectedPrivilegeEvaluator).isAllowed(eq(""), eq(uri), eq("GET"), any());
 	}
 
 	@Test
@@ -99,26 +101,24 @@
 		wac.refresh();
 		this.servletContext.setAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE, wac);
 		String uri = "/something";
-		WebInvocationPrivilegeEvaluator expected = mock(WebInvocationPrivilegeEvaluator.class);
 		this.tag.setUrl(uri);
-		this.request.setAttribute(WebAttributes.WEB_INVOCATION_PRIVILEGE_EVALUATOR_ATTRIBUTE, expected);
+		this.request.setAttribute(WebAttributes.WEB_INVOCATION_PRIVILEGE_EVALUATOR_ATTRIBUTE, expectedPrivilegeEvaluator);
 		this.tag.authorizeUsingUrlCheck();
-		verify(expected).isAllowed(eq(""), eq(uri), eq("GET"), any());
+		verify(expectedPrivilegeEvaluator).isAllowed(eq(""), eq(uri), eq("GET"), any());
 		verify(strategy).getContext();
 	}
 
 	@Test
 	public void privilegeEvaluatorFromChildContext() throws IOException {
 		String uri = "/something";
-		WebInvocationPrivilegeEvaluator expected = mock(WebInvocationPrivilegeEvaluator.class);
 		this.tag.setUrl(uri);
 		WebApplicationContext wac = mock(WebApplicationContext.class);
 		given(wac.getBeansOfType(WebInvocationPrivilegeEvaluator.class))
-			.willReturn(Collections.singletonMap("wipe", expected));
+			.willReturn(Collections.singletonMap("wipe", expectedPrivilegeEvaluator));
 		given(wac.getBeanNamesForType(SecurityContextHolderStrategy.class)).willReturn(new String[0]);
 		this.servletContext.setAttribute("org.springframework.web.servlet.FrameworkServlet.CONTEXT.dispatcher", wac);
 		this.tag.authorizeUsingUrlCheck();
-		verify(expected).isAllowed(eq(""), eq(uri), eq("GET"), any());
+		verify(expectedPrivilegeEvaluator).isAllowed(eq(""), eq(uri), eq("GET"), any());
 	}
 
 	@Test
