--- a/web/src/test/java/org/springframework/security/web/authentication/DelegatingAuthenticationEntryPointTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/DelegatingAuthenticationEntryPointTests.java
@@ -51,40 +51,41 @@
 
 	private DelegatingAuthenticationEntryPoint daep;
 
+	private AuthenticationEntryPoint sharedFirstAEP;
+
 	private AuthenticationEntryPoint defaultEntryPoint;
 
 	private HttpServletRequest request = new MockHttpServletRequest();
 
 	@BeforeEach
 	public void before() {
+		this.sharedFirstAEP = mock(AuthenticationEntryPoint.class);
 		this.defaultEntryPoint = mock(AuthenticationEntryPoint.class);
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	public void testDefaultEntryPoint() throws Exception {
-		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher firstRM = mock(RequestMatcher.class);
 		given(firstRM.matches(this.request)).willReturn(false);
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint,
-				new RequestMatcherEntry<>(firstRM, firstAEP));
+				new RequestMatcherEntry<>(firstRM, sharedFirstAEP));
 		this.daep.commence(this.request, null, null);
 		verify(this.defaultEntryPoint).commence(this.request, null, null);
-		verify(firstAEP, never()).commence(this.request, null, null);
+		verify(sharedFirstAEP, never()).commence(this.request, null, null);
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	public void testFirstEntryPoint() throws Exception {
-		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher firstRM = mock(RequestMatcher.class);
 		AuthenticationEntryPoint secondAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher secondRM = mock(RequestMatcher.class);
 		given(firstRM.matches(this.request)).willReturn(true);
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint,
-				new RequestMatcherEntry<>(firstRM, firstAEP), new RequestMatcherEntry<>(secondRM, secondAEP));
-		this.daep.commence(this.request, null, null);
-		verify(firstAEP).commence(this.request, null, null);
+				new RequestMatcherEntry<>(firstRM, sharedFirstAEP), new RequestMatcherEntry<>(secondRM, secondAEP));
+		this.daep.commence(this.request, null, null);
+		verify(sharedFirstAEP).commence(this.request, null, null);
 		verify(secondAEP, never()).commence(this.request, null, null);
 		verify(this.defaultEntryPoint, never()).commence(this.request, null, null);
 		verify(secondRM, never()).matches(this.request);
@@ -93,17 +94,16 @@
 	@Test
 	@SuppressWarnings("removal")
 	public void testSecondEntryPoint() throws Exception {
-		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher firstRM = mock(RequestMatcher.class);
 		AuthenticationEntryPoint secondAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher secondRM = mock(RequestMatcher.class);
 		given(firstRM.matches(this.request)).willReturn(false);
 		given(secondRM.matches(this.request)).willReturn(true);
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint,
-				new RequestMatcherEntry<>(firstRM, firstAEP), new RequestMatcherEntry<>(secondRM, secondAEP));
+				new RequestMatcherEntry<>(firstRM, sharedFirstAEP), new RequestMatcherEntry<>(secondRM, secondAEP));
 		this.daep.commence(this.request, null, null);
 		verify(secondAEP).commence(this.request, null, null);
-		verify(firstAEP, never()).commence(this.request, null, null);
+		verify(sharedFirstAEP, never()).commence(this.request, null, null);
 		verify(this.defaultEntryPoint, never()).commence(this.request, null, null);
 	}
 
@@ -126,10 +126,9 @@
 	@Test
 	@SuppressWarnings("removal")
 	public void constructorAepListWhenNullDefaultEntryPoint() {
-		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
 		RequestMatcher matcher = mock(RequestMatcher.class);
 		List<RequestMatcherEntry<AuthenticationEntryPoint>> entryPoints = List
-			.of(new RequestMatcherEntry<>(matcher, entryPoint));
+			.of(new RequestMatcherEntry<>(matcher, sharedFirstAEP));
 		assertThatIllegalArgumentException()
 			.isThrownBy(() -> new DelegatingAuthenticationEntryPoint(null, entryPoints));
 	}
@@ -153,10 +152,9 @@
 	@Test
 	@SuppressWarnings("removal")
 	public void constructorAepVargsWhenNullDefaultEntryPoint() {
-		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
 		RequestMatcher matcher = mock(RequestMatcher.class);
 		RequestMatcherEntry<AuthenticationEntryPoint>[] entryPoints = new RequestMatcherEntry[] {
-				new RequestMatcherEntry<>(matcher, entryPoint) };
+				new RequestMatcherEntry<>(matcher, sharedFirstAEP) };
 		assertThatIllegalArgumentException()
 			.isThrownBy(() -> new DelegatingAuthenticationEntryPoint(null, entryPoints));
 	}
@@ -164,30 +162,28 @@
 	@Test
 	@SuppressWarnings("removal")
 	public void commenceWhenNoMatchThenDefaultEntryPoint() throws Exception {
-		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
-		RequestMatcher firstRM = mock(RequestMatcher.class);
-		given(firstRM.matches(this.request)).willReturn(false);
-		RequestMatcherEntry<AuthenticationEntryPoint> entry = new RequestMatcherEntry<>(firstRM, firstAEP);
+		RequestMatcher firstRM = mock(RequestMatcher.class);
+		given(firstRM.matches(this.request)).willReturn(false);
+		RequestMatcherEntry<AuthenticationEntryPoint> entry = new RequestMatcherEntry<>(firstRM, sharedFirstAEP);
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint, entry);
 		this.daep.commence(this.request, null, null);
 		verify(this.defaultEntryPoint).commence(this.request, null, null);
-		verify(firstAEP, never()).commence(this.request, null, null);
+		verify(sharedFirstAEP, never()).commence(this.request, null, null);
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	public void commenceWhenMatchFirstEntryPointThenOthersNotInvoked() throws Exception {
-		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher firstRM = mock(RequestMatcher.class);
 		given(firstRM.matches(this.request)).willReturn(true);
-		RequestMatcherEntry<AuthenticationEntryPoint> firstEntry = new RequestMatcherEntry<>(firstRM, firstAEP);
+		RequestMatcherEntry<AuthenticationEntryPoint> firstEntry = new RequestMatcherEntry<>(firstRM, sharedFirstAEP);
 		AuthenticationEntryPoint secondAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher secondRM = mock(RequestMatcher.class);
 		given(secondRM.matches(this.request)).willReturn(false);
-		RequestMatcherEntry<AuthenticationEntryPoint> secondEntry = new RequestMatcherEntry<>(firstRM, firstAEP);
+		RequestMatcherEntry<AuthenticationEntryPoint> secondEntry = new RequestMatcherEntry<>(firstRM, sharedFirstAEP);
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint, firstEntry, secondEntry);
 		this.daep.commence(this.request, null, null);
-		verify(firstAEP).commence(this.request, null, null);
+		verify(sharedFirstAEP).commence(this.request, null, null);
 		verify(secondAEP, never()).commence(this.request, null, null);
 		verify(this.defaultEntryPoint, never()).commence(this.request, null, null);
 		verify(secondRM, never()).matches(this.request);
@@ -196,10 +192,9 @@
 	@Test
 	@SuppressWarnings("removal")
 	public void commenceWhenSecondMatchesThenDefaultNotInvoked() throws Exception {
-		AuthenticationEntryPoint firstAEP = mock(AuthenticationEntryPoint.class);
-		RequestMatcher firstRM = mock(RequestMatcher.class);
-		given(firstRM.matches(this.request)).willReturn(false);
-		RequestMatcherEntry<AuthenticationEntryPoint> firstEntry = new RequestMatcherEntry<>(firstRM, firstAEP);
+		RequestMatcher firstRM = mock(RequestMatcher.class);
+		given(firstRM.matches(this.request)).willReturn(false);
+		RequestMatcherEntry<AuthenticationEntryPoint> firstEntry = new RequestMatcherEntry<>(firstRM, sharedFirstAEP);
 		AuthenticationEntryPoint secondAEP = mock(AuthenticationEntryPoint.class);
 		RequestMatcher secondRM = mock(RequestMatcher.class);
 		given(secondRM.matches(this.request)).willReturn(true);
@@ -207,49 +202,46 @@
 		this.daep = new DelegatingAuthenticationEntryPoint(this.defaultEntryPoint, firstEntry, secondEntry);
 		this.daep.commence(this.request, null, null);
 		verify(secondAEP).commence(this.request, null, null);
-		verify(firstAEP, never()).commence(this.request, null, null);
+		verify(sharedFirstAEP, never()).commence(this.request, null, null);
 		verify(this.defaultEntryPoint, never()).commence(this.request, null, null);
 	}
 
 	@Test
 	void builderWhenDefaultNullAndSingleEntryPointThenReturnsSingle() {
-		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
 
 		AuthenticationEntryPoint result = DelegatingAuthenticationEntryPoint.builder()
-			.addEntryPointFor(entryPoint, mock(RequestMatcher.class))
+			.addEntryPointFor(sharedFirstAEP, mock(RequestMatcher.class))
 			.build();
 
-		assertThat(result).isEqualTo(entryPoint);
+		assertThat(result).isEqualTo(sharedFirstAEP);
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	void builderWhenDefaultNullThenFirstIsDefault() throws ServletException, IOException {
-		AuthenticationEntryPoint firstEntryPoint = mock(AuthenticationEntryPoint.class);
 		AuthenticationEntryPoint secondEntryPoint = mock(AuthenticationEntryPoint.class);
 		RequestMatcher neverMatch = mock(RequestMatcher.class);
 		given(neverMatch.matches(this.request)).willReturn(false);
 		AuthenticationEntryPoint result = DelegatingAuthenticationEntryPoint.builder()
-			.addEntryPointFor(firstEntryPoint, neverMatch)
+			.addEntryPointFor(sharedFirstAEP, neverMatch)
 			.addEntryPointFor(secondEntryPoint, neverMatch)
 			.build();
 
 		result.commence(this.request, null, null);
 
-		verify(firstEntryPoint).commence(any(), any(), any());
+		verify(sharedFirstAEP).commence(any(), any(), any());
 		verifyNoInteractions(secondEntryPoint);
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	void builderWhenDefaultAndEmptyEntryPointsThenReturnsDefault() {
-		AuthenticationEntryPoint defaultEntryPoint = mock(AuthenticationEntryPoint.class);
 
 		AuthenticationEntryPoint result = DelegatingAuthenticationEntryPoint.builder()
-			.defaultEntryPoint(defaultEntryPoint)
+			.defaultEntryPoint(sharedFirstAEP)
 			.build();
 
-		assertThat(result).isEqualTo(defaultEntryPoint);
+		assertThat(result).isEqualTo(sharedFirstAEP);
 	}
 
 	@Test
