--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/RefreshOidcUserReactiveOAuth2AuthorizationSuccessHandlerTests.java
@@ -124,8 +124,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -163,8 +162,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -199,8 +197,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -235,8 +232,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -271,8 +267,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -307,8 +302,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -345,8 +339,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -383,8 +376,7 @@
 		given(jwt.getClaims()).willReturn(claims);
 		ReactiveJwtDecoder jwtDecoder = mock(ReactiveJwtDecoder.class);
 		given(jwtDecoder.decode(any())).willReturn(Mono.just(jwt));
-		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
-		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = createMockReactiveJwtDecoderFactory(jwtDecoder);
 		ReactiveOAuth2UserService<OidcUserRequest, OidcUser> userService = mock(ReactiveOAuth2UserService.class);
 		given(userService.loadUser(any())).willReturn(Mono.just(principal));
 		WebSessionServerSecurityContextRepository serverSecurityContextRepository = new WebSessionServerSecurityContextRepository();
@@ -433,6 +425,13 @@
 		handler.setServerSecurityContextRepository(serverSecurityContextRepository);
 		StepVerifier.create(handler.onAuthorizationSuccess(authorizedClient, authenticationToken, attributes))
 			.verifyErrorMessage("[invalid_nonce] Invalid nonce");
+	}
+
+	private static ReactiveJwtDecoderFactory<ClientRegistration> createMockReactiveJwtDecoderFactory(
+			ReactiveJwtDecoder jwtDecoder) {
+		ReactiveJwtDecoderFactory<ClientRegistration> reactiveJwtDecoderFactory = mock(ReactiveJwtDecoderFactory.class);
+		given(reactiveJwtDecoderFactory.createDecoder(any())).willReturn(jwtDecoder);
+		return reactiveJwtDecoderFactory;
 	}
 
 	private static OAuth2AccessToken createAccessToken() {
