--- a/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
+++ b/config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java
@@ -49,6 +49,7 @@
 import org.hamcrest.core.StringContains;
 import org.hamcrest.core.StringEndsWith;
 import org.hamcrest.core.StringStartsWith;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.verification.VerificationMode;
@@ -216,6 +217,13 @@
 
 	public final SpringTestContext spring = new SpringTestContext(this);
 
+	private JwtDecoder configuredDecoder;
+
+	@BeforeEach
+	void setUp() {
+		this.configuredDecoder = mock(JwtDecoder.class);
+	}
+
 	@Test
 	public void getWhenUsingDefaultsWithValidBearerTokenThenAcceptsRequest() throws Exception {
 		this.spring.register(RestOperationsConfig.class, DefaultConfig.class, BasicController.class).autowire();
@@ -816,24 +824,22 @@
 	@Test
 	public void getJwtDecoderWhenConfiguredWithDecoderAndJwkSetUriThenLastOneWins() {
 		ApplicationContext context = mock(ApplicationContext.class);
-		JwtDecoder decoder = mock(JwtDecoder.class);
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
 			jwt.jwkSetUri(JWK_SET_URI);
-			jwt.decoder(decoder);
-			assertThat(jwt.getJwtDecoder()).isEqualTo(decoder);
+			jwt.decoder(configuredDecoder);
+			assertThat(jwt.getJwtDecoder()).isEqualTo(configuredDecoder);
 		});
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
-			jwt.decoder(decoder).jwkSetUri(JWK_SET_URI);
+			jwt.decoder(configuredDecoder).jwkSetUri(JWK_SET_URI);
 			assertThat(jwt.getJwtDecoder()).isInstanceOf(NimbusJwtDecoder.class);
 		});
 	}
 
 	@Test
 	public void getJwtDecoderWhenConflictingJwtDecodersThenTheDslWiredOneTakesPrecedence() {
-		JwtDecoder decoderBean = mock(JwtDecoder.class);
 		JwtDecoder decoder = mock(JwtDecoder.class);
 		ApplicationContext context = mock(ApplicationContext.class);
-		given(context.getBean(JwtDecoder.class)).willReturn(decoderBean);
+		given(context.getBean(JwtDecoder.class)).willReturn(configuredDecoder);
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
 			jwt.decoder(decoder);
 			assertThat(jwt.getJwtDecoder()).isEqualTo(decoder);
@@ -842,23 +848,21 @@
 
 	@Test
 	public void getJwtDecoderWhenContextHasBeanAndUserConfiguresJwkSetUriThenJwkSetUriTakesPrecedence() {
-		JwtDecoder decoder = mock(JwtDecoder.class);
 		ApplicationContext context = mock(ApplicationContext.class);
-		given(context.getBean(JwtDecoder.class)).willReturn(decoder);
+		given(context.getBean(JwtDecoder.class)).willReturn(configuredDecoder);
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
 			jwt.jwkSetUri(JWK_SET_URI);
-			assertThat(jwt.getJwtDecoder()).isNotEqualTo(decoder);
+			assertThat(jwt.getJwtDecoder()).isNotEqualTo(configuredDecoder);
 			assertThat(jwt.getJwtDecoder()).isInstanceOf(NimbusJwtDecoder.class);
 		});
 	}
 
 	@Test
 	public void getJwtDecoderWhenTwoJwtDecoderBeansAndAnotherWiredOnDslThenDslWiredOneTakesPrecedence() {
-		JwtDecoder decoderBean = mock(JwtDecoder.class);
 		JwtDecoder decoder = mock(JwtDecoder.class);
 		GenericWebApplicationContext context = new GenericWebApplicationContext();
-		context.registerBean("decoderOne", JwtDecoder.class, () -> decoderBean);
-		context.registerBean("decoderTwo", JwtDecoder.class, () -> decoderBean);
+		context.registerBean("decoderOne", JwtDecoder.class, () -> configuredDecoder);
+		context.registerBean("decoderTwo", JwtDecoder.class, () -> configuredDecoder);
 		this.spring.context(context).autowire();
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context).jwt((jwt) -> {
 			jwt.decoder(decoder);
@@ -868,10 +872,9 @@
 
 	@Test
 	public void getJwtDecoderWhenTwoJwtDecoderBeansThenThrowsException() {
-		JwtDecoder decoder = mock(JwtDecoder.class);
 		GenericWebApplicationContext context = new GenericWebApplicationContext();
-		context.registerBean("decoderOne", JwtDecoder.class, () -> decoder);
-		context.registerBean("decoderTwo", JwtDecoder.class, () -> decoder);
+		context.registerBean("decoderOne", JwtDecoder.class, () -> configuredDecoder);
+		context.registerBean("decoderTwo", JwtDecoder.class, () -> configuredDecoder);
 		this.spring.context(context).autowire();
 		new OAuth2ResourceServerConfigurer<HttpSecurity>(context)
 			.jwt((jwt) -> assertThatExceptionOfType(NoUniqueBeanDefinitionException.class)
