--- a/web/src/test/java/org/springframework/security/web/server/authorization/ServerWebExchangeDelegatingServerAccessDeniedHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/authorization/ServerWebExchangeDelegatingServerAccessDeniedHandlerTests.java
@@ -52,10 +52,9 @@
 
 	@Test
 	public void handleWhenNothingMatchesThenOnlyDefaultHandlerInvoked() {
-		ServerAccessDeniedHandler handler = mock(ServerAccessDeniedHandler.class);
 		ServerWebExchangeMatcher matcher = mock(ServerWebExchangeMatcher.class);
 		given(matcher.matches(this.exchange)).willReturn(MatchResult.notMatch());
-		given(handler.handle(this.exchange, null)).willReturn(Mono.empty());
+		ServerAccessDeniedHandler handler = createMockServerAccessDeniedHandler(this.exchange);
 		given(this.accessDeniedHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		this.entries.add(new DelegateEntry(matcher, handler));
 		this.delegator = new ServerWebExchangeDelegatingServerAccessDeniedHandler(this.entries);
@@ -67,13 +66,11 @@
 
 	@Test
 	public void handleWhenFirstMatchesThenOnlyFirstInvoked() {
-		ServerAccessDeniedHandler firstHandler = mock(ServerAccessDeniedHandler.class);
+		ServerAccessDeniedHandler firstHandler = createMockServerAccessDeniedHandler(this.exchange);
 		ServerWebExchangeMatcher firstMatcher = mock(ServerWebExchangeMatcher.class);
-		ServerAccessDeniedHandler secondHandler = mock(ServerAccessDeniedHandler.class);
+		ServerAccessDeniedHandler secondHandler = createMockServerAccessDeniedHandler(this.exchange);
 		ServerWebExchangeMatcher secondMatcher = mock(ServerWebExchangeMatcher.class);
 		given(firstMatcher.matches(this.exchange)).willReturn(MatchResult.match());
-		given(firstHandler.handle(this.exchange, null)).willReturn(Mono.empty());
-		given(secondHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		this.entries.add(new DelegateEntry(firstMatcher, firstHandler));
 		this.entries.add(new DelegateEntry(secondMatcher, secondHandler));
 		this.delegator = new ServerWebExchangeDelegatingServerAccessDeniedHandler(this.entries);
@@ -87,14 +84,12 @@
 
 	@Test
 	public void handleWhenSecondMatchesThenOnlySecondInvoked() {
-		ServerAccessDeniedHandler firstHandler = mock(ServerAccessDeniedHandler.class);
+		ServerAccessDeniedHandler firstHandler = createMockServerAccessDeniedHandler(this.exchange);
 		ServerWebExchangeMatcher firstMatcher = mock(ServerWebExchangeMatcher.class);
-		ServerAccessDeniedHandler secondHandler = mock(ServerAccessDeniedHandler.class);
+		ServerAccessDeniedHandler secondHandler = createMockServerAccessDeniedHandler(this.exchange);
 		ServerWebExchangeMatcher secondMatcher = mock(ServerWebExchangeMatcher.class);
 		given(firstMatcher.matches(this.exchange)).willReturn(MatchResult.notMatch());
 		given(secondMatcher.matches(this.exchange)).willReturn(MatchResult.match());
-		given(firstHandler.handle(this.exchange, null)).willReturn(Mono.empty());
-		given(secondHandler.handle(this.exchange, null)).willReturn(Mono.empty());
 		this.entries.add(new DelegateEntry(firstMatcher, firstHandler));
 		this.entries.add(new DelegateEntry(secondMatcher, secondHandler));
 		this.delegator = new ServerWebExchangeDelegatingServerAccessDeniedHandler(this.entries);
@@ -104,4 +99,10 @@
 		verify(this.accessDeniedHandler, never()).handle(this.exchange, null);
 	}
 
+	private static ServerAccessDeniedHandler createMockServerAccessDeniedHandler(ServerWebExchange exchange) {
+		ServerAccessDeniedHandler handler = mock(ServerAccessDeniedHandler.class);
+		given(handler.handle(exchange, null)).willReturn(Mono.empty());
+		return handler;
+	}
+
 }
