--- a/crypto/src/test/java/org/springframework/security/crypto/password/DelegatingPasswordEncoderTests.java
+++ b/crypto/src/test/java/org/springframework/security/crypto/password/DelegatingPasswordEncoderTests.java
@@ -172,7 +172,7 @@
 
 	@Test
 	public void matchesWhenBCryptThenDelegatesToBCrypt() {
-		given(this.bcrypt.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
+		givenMatchesReturnsTrue(this.bcrypt);
 		assertThat(getEncoder().matches(this.rawPassword, this.bcryptEncodedPassword)).isTrue();
 		verify(this.bcrypt).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.noop);
@@ -180,7 +180,7 @@
 
 	@Test
 	public void matchesWhenBCryptBySpecifyDelegatingPasswordEncoderThenDelegatesToBCrypt() {
-		given(this.bcrypt.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
+		givenMatchesReturnsTrue(this.bcrypt);
 		assertThat(this.onlySuffixPasswordEncoder.matches(this.rawPassword, "bcrypt$" + this.encodedPassword)).isTrue();
 		verify(this.bcrypt).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.noop);
@@ -188,7 +188,7 @@
 
 	@Test
 	public void matchesWhenNoopThenDelegatesToNoop() {
-		given(this.noop.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
+		givenMatchesReturnsTrue(this.noop);
 		assertThat(getEncoder().matches(this.rawPassword, this.noopEncodedPassword)).isTrue();
 		verify(this.noop).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.bcrypt);
@@ -249,7 +249,7 @@
 	public void matchesWhenNullIdThenDelegatesToInvalidId() {
 		this.delegates.put(null, this.invalidId);
 		setEncoder(new DelegatingPasswordEncoder(this.bcryptId, this.delegates));
-		given(this.invalidId.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
+		givenMatchesReturnsTrue(this.invalidId);
 		assertThat(getEncoder().matches(this.rawPassword, this.encodedPassword)).isTrue();
 		verify(this.invalidId).matches(this.rawPassword, this.encodedPassword);
 		verifyNoMoreInteractions(this.bcrypt, this.noop);
@@ -298,4 +298,8 @@
 		verifyNoMoreInteractions(this.bcrypt, this.noop);
 	}
 
+	private void givenMatchesReturnsTrue(PasswordEncoder passwordEncoder) {
+		given(passwordEncoder.matches(this.rawPassword, this.encodedPassword)).willReturn(true);
+	}
+
 }
