--- a/saml2/saml2-service-provider/src/opensaml5Test/java/org/springframework/security/saml2/provider/service/web/authentication/OpenSaml5AuthenticationRequestResolverTests.java
+++ b/saml2/saml2-service-provider/src/opensaml5Test/java/org/springframework/security/saml2/provider/service/web/authentication/OpenSaml5AuthenticationRequestResolverTests.java
@@ -51,8 +51,7 @@
 
 	@Test
 	void resolveWhenRedirectThenSaml2RedirectAuthenticationRequest() {
-		RelyingPartyRegistrationResolver relyingParties = mock(RelyingPartyRegistrationResolver.class);
-		given(relyingParties.resolve(any(), any())).willReturn(this.registration);
+		RelyingPartyRegistrationResolver relyingParties = createMockRelyingPartyRegistrationResolver(this.registration);
 		OpenSaml5AuthenticationRequestResolver resolver = new OpenSaml5AuthenticationRequestResolver(relyingParties);
 		Saml2RedirectAuthenticationRequest authnRequest = resolver.resolve(this.request);
 		assertThat(authnRequest.getBinding()).isEqualTo(Saml2MessageBinding.REDIRECT);
@@ -65,8 +64,7 @@
 		RelyingPartyRegistration registration = TestRelyingPartyRegistrations.full()
 			.assertingPartyMetadata((party) -> party.singleSignOnServiceBinding(Saml2MessageBinding.POST))
 			.build();
-		RelyingPartyRegistrationResolver relyingParties = mock(RelyingPartyRegistrationResolver.class);
-		given(relyingParties.resolve(any(), any())).willReturn(registration);
+		RelyingPartyRegistrationResolver relyingParties = createMockRelyingPartyRegistrationResolver(registration);
 		OpenSaml5AuthenticationRequestResolver resolver = new OpenSaml5AuthenticationRequestResolver(relyingParties);
 		Saml2PostAuthenticationRequest authnRequest = resolver.resolve(this.request);
 		assertThat(authnRequest.getBinding()).isEqualTo(Saml2MessageBinding.POST);
@@ -76,8 +74,7 @@
 
 	@Test
 	void resolveWhenCustomRelayStateThenUses() {
-		RelyingPartyRegistrationResolver relyingParties = mock(RelyingPartyRegistrationResolver.class);
-		given(relyingParties.resolve(any(), any())).willReturn(this.registration);
+		RelyingPartyRegistrationResolver relyingParties = createMockRelyingPartyRegistrationResolver(this.registration);
 		Converter<HttpServletRequest, String> relayState = mock(Converter.class);
 		given(relayState.convert(any())).willReturn("state");
 		OpenSaml5AuthenticationRequestResolver resolver = new OpenSaml5AuthenticationRequestResolver(relyingParties);
@@ -89,8 +86,7 @@
 
 	@Test
 	void resolveWhenCustomAuthenticationUrlTHenUses() {
-		RelyingPartyRegistrationResolver relyingParties = mock(RelyingPartyRegistrationResolver.class);
-		given(relyingParties.resolve(any(), any())).willReturn(this.registration);
+		RelyingPartyRegistrationResolver relyingParties = createMockRelyingPartyRegistrationResolver(this.registration);
 		OpenSaml5AuthenticationRequestResolver resolver = new OpenSaml5AuthenticationRequestResolver(relyingParties);
 		resolver.setRequestMatcher(pathPattern("/custom/authentication/{registrationId}"));
 		Saml2RedirectAuthenticationRequest authnRequest = resolver
@@ -106,4 +102,11 @@
 		return TestMockHttpServletRequests.get(path).build();
 	}
 
+	private static RelyingPartyRegistrationResolver createMockRelyingPartyRegistrationResolver(
+			RelyingPartyRegistration registration) {
+		RelyingPartyRegistrationResolver relyingParties = mock(RelyingPartyRegistrationResolver.class);
+		given(relyingParties.resolve(any(), any())).willReturn(registration);
+		return relyingParties;
+	}
+
 }
