--- a/core/src/test/java/org/springframework/security/authentication/ott/reactive/OneTimeTokenReactiveAuthenticationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/ott/reactive/OneTimeTokenReactiveAuthenticationManagerTests.java
@@ -19,6 +19,7 @@
 import java.time.Instant;
 import java.util.Collection;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentMatchers;
 import reactor.core.publisher.Mono;
@@ -52,6 +53,13 @@
 
 	private ReactiveAuthenticationManager authenticationManager;
 
+	private ReactiveUserDetailsService reactiveUserDetailsService;
+
+	@BeforeEach
+	void setUp() {
+		this.reactiveUserDetailsService = mock(ReactiveUserDetailsService.class);
+	}
+
 	private static final String USERNAME = "user";
 
 	private static final String PASSWORD = "password";
@@ -61,10 +69,9 @@
 	@Test
 	@SuppressWarnings("removal")
 	public void constructorWhenOneTimeTokenServiceNullThenIllegalArgumentException() {
-		ReactiveUserDetailsService userDetailsService = mock(ReactiveUserDetailsService.class);
 		// @formatter:off
 		assertThatIllegalArgumentException()
-				.isThrownBy(() -> new OneTimeTokenReactiveAuthenticationManager(null, userDetailsService));
+				.isThrownBy(() -> new OneTimeTokenReactiveAuthenticationManager(null, reactiveUserDetailsService));
 		// @formatter:on
 	}
 
@@ -111,10 +118,9 @@
 		ReactiveOneTimeTokenService oneTimeTokenService = mock(ReactiveOneTimeTokenService.class);
 		given(oneTimeTokenService.consume(ArgumentMatchers.any(OneTimeTokenAuthenticationToken.class)))
 			.willReturn(Mono.empty());
-		ReactiveUserDetailsService userDetailsService = mock(ReactiveUserDetailsService.class);
 
 		this.authenticationManager = new OneTimeTokenReactiveAuthenticationManager(oneTimeTokenService,
-				userDetailsService);
+				reactiveUserDetailsService);
 
 		// @formatter:off
 		assertThatExceptionOfType(InvalidOneTimeTokenException.class)
@@ -129,10 +135,9 @@
 		ReactiveOneTimeTokenService oneTimeTokenService = mock(ReactiveOneTimeTokenService.class);
 		given(oneTimeTokenService.consume(ArgumentMatchers.any(OneTimeTokenAuthenticationToken.class)))
 			.willReturn(Mono.empty());
-		ReactiveUserDetailsService userDetailsService = mock(ReactiveUserDetailsService.class);
 
 		this.authenticationManager = new OneTimeTokenReactiveAuthenticationManager(oneTimeTokenService,
-				userDetailsService);
+				reactiveUserDetailsService);
 
 		// @formatter:off
 		Authentication authentication = this.authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(USERNAME, PASSWORD))
