--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/authentication/OAuth2LoginAuthenticationProviderTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/authentication/OAuth2LoginAuthenticationProviderTests.java
@@ -186,9 +186,8 @@
 		given(principal.getAuthorities()).willAnswer((Answer<List<GrantedAuthority>>) (invocation) -> authorities);
 		given(this.userService.loadUser(any())).willReturn(principal);
 		List<GrantedAuthority> mappedAuthorities = AuthorityUtils.createAuthorityList("ROLE_OAUTH2_USER");
-		GrantedAuthoritiesMapper authoritiesMapper = mock(GrantedAuthoritiesMapper.class);
-		given(authoritiesMapper.mapAuthorities(anyCollection()))
-			.willAnswer((Answer<List<GrantedAuthority>>) (invocation) -> mappedAuthorities);
+		GrantedAuthoritiesMapper authoritiesMapper = MockGrantedAuthoritiesMapper
+			.createMockGrantedAuthoritiesMapper(mappedAuthorities);
 		this.authenticationProvider.setAuthoritiesMapper(authoritiesMapper);
 		OAuth2LoginAuthenticationToken authentication = (OAuth2LoginAuthenticationToken) this.authenticationProvider
 			.authenticate(new OAuth2LoginAuthenticationToken(this.clientRegistration, this.authorizationExchange));
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/authentication/OAuth2LoginReactiveAuthenticationManagerTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/authentication/OAuth2LoginReactiveAuthenticationManagerTests.java
@@ -196,9 +196,8 @@
 				Collections.singletonMap("user", "rob"), "user");
 		given(this.userService.loadUser(any())).willReturn(Mono.just(user));
 		List<GrantedAuthority> mappedAuthorities = AuthorityUtils.createAuthorityList("ROLE_OAUTH_USER");
-		GrantedAuthoritiesMapper authoritiesMapper = mock(GrantedAuthoritiesMapper.class);
-		given(authoritiesMapper.mapAuthorities(anyCollection()))
-			.willAnswer((Answer<List<GrantedAuthority>>) (invocation) -> mappedAuthorities);
+		GrantedAuthoritiesMapper authoritiesMapper = MockGrantedAuthoritiesMapper
+			.createMockGrantedAuthoritiesMapper(mappedAuthorities);
 		this.manager.setAuthoritiesMapper(authoritiesMapper);
 		OAuth2LoginAuthenticationToken result = (OAuth2LoginAuthenticationToken) this.manager.authenticate(loginToken())
 			.block();
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/authentication/MockGrantedAuthoritiesMapper.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/authentication/MockGrantedAuthoritiesMapper.java
@@ -0,0 +1,25 @@
+package org.springframework.security.oauth2.client.authentication;
+
+import java.util.List;
+
+import org.mockito.stubbing.Answer;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper;
+
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public final class MockGrantedAuthoritiesMapper {
+
+	private MockGrantedAuthoritiesMapper() {
+	}
+
+	public static GrantedAuthoritiesMapper createMockGrantedAuthoritiesMapper(List<GrantedAuthority> mappedAuthorities) {
+		GrantedAuthoritiesMapper authoritiesMapper = mock(GrantedAuthoritiesMapper.class);
+		given(authoritiesMapper.mapAuthorities(anyCollection()))
+			.willAnswer((Answer<List<GrantedAuthority>>) (invocation) -> mappedAuthorities);
+		return authoritiesMapper;
+	}
+
+}
