--- a/web/src/test/java/org/springframework/security/web/csrf/CsrfAuthenticationStrategyTests.java
+++ b/web/src/test/java/org/springframework/security/web/csrf/CsrfAuthenticationStrategyTests.java
@@ -118,10 +118,9 @@
 	// SEC-2872
 	@Test
 	public void delaySavingCsrf() {
+		this.csrfTokenRepository = MockCsrfTokenRepository.createMockCsrfTokenRepository(this.request, this.generatedToken);
+		given(this.csrfTokenRepository.loadToken(this.request)).willReturn(this.existingToken, (CsrfToken) null);
 		this.strategy = new CsrfAuthenticationStrategy(this.csrfTokenRepository);
-		given(this.csrfTokenRepository.loadToken(this.request)).willReturn(this.existingToken, (CsrfToken) null);
-		given(this.csrfTokenRepository.generateToken(this.request)).willReturn(this.generatedToken);
-		given(this.csrfTokenRepository.loadDeferredToken(any(), any())).willCallRealMethod();
 		this.strategy.onAuthentication(new TestingAuthenticationToken("user", "password", "ROLE_USER"), this.request,
 				this.response);
 		verify(this.csrfTokenRepository).saveToken(null, this.request, this.response);
--- a/web/src/test/java/org/springframework/security/web/csrf/CsrfFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/csrf/CsrfFilterTests.java
@@ -108,10 +108,9 @@
 	// SEC-2276
 	@Test
 	public void doFilterDoesNotSaveCsrfTokenUntilAccessed() throws ServletException, IOException {
+		this.tokenRepository = MockCsrfTokenRepository.createMockCsrfTokenRepository(this.request, this.token);
 		this.filter = createCsrfFilter(this.tokenRepository);
 		given(this.requestMatcher.matches(this.request)).willReturn(false);
-		given(this.tokenRepository.generateToken(this.request)).willReturn(this.token);
-		given(this.tokenRepository.loadDeferredToken(any(), any())).willCallRealMethod();
 		this.filter.doFilter(this.request, this.response, this.filterChain);
 		CsrfToken attrToken = (CsrfToken) this.request.getAttribute(this.csrfAttrName);
 		// no CsrfToken should have been saved yet
--- a/web/src/test/java/org/springframework/security/web/csrf/MockCsrfTokenRepository.java
+++ b/web/src/test/java/org/springframework/security/web/csrf/MockCsrfTokenRepository.java
@@ -0,0 +1,22 @@
+package org.springframework.security.web.csrf;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+public final class MockCsrfTokenRepository {
+
+	private MockCsrfTokenRepository() {
+	}
+
+	public static CsrfTokenRepository createMockCsrfTokenRepository(HttpServletRequest request, CsrfToken generatedToken) {
+		CsrfTokenRepository tokenRepository = Mockito.mock(CsrfTokenRepository.class);
+		given(tokenRepository.generateToken(request)).willReturn(generatedToken);
+		given(tokenRepository.loadDeferredToken(any(), any())).willCallRealMethod();
+		return tokenRepository;
+	}
+
+}
