--- a/web/src/test/java/org/springframework/security/web/access/RequestMatcherDelegatingAccessDeniedHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/access/RequestMatcherDelegatingAccessDeniedHandlerTests.java
@@ -41,11 +41,14 @@
 
 	private AccessDeniedHandler accessDeniedHandler;
 
+	private AccessDeniedHandler secondAccessDeniedHandler;
+
 	private HttpServletRequest request;
 
 	@BeforeEach
 	public void setup() {
 		this.accessDeniedHandler = mock(AccessDeniedHandler.class);
+		this.secondAccessDeniedHandler = mock(AccessDeniedHandler.class);
 		this.deniedHandlers = new LinkedHashMap<>();
 		this.request = new MockHttpServletRequest();
 	}
@@ -66,15 +69,14 @@
 	public void handleWhenFirstMatchesThenOnlyFirstInvoked() throws Exception {
 		AccessDeniedHandler firstHandler = mock(AccessDeniedHandler.class);
 		RequestMatcher firstMatcher = mock(RequestMatcher.class);
-		AccessDeniedHandler secondHandler = mock(AccessDeniedHandler.class);
 		RequestMatcher secondMatcher = mock(RequestMatcher.class);
 		given(firstMatcher.matches(this.request)).willReturn(true);
 		this.deniedHandlers.put(firstMatcher, firstHandler);
-		this.deniedHandlers.put(secondMatcher, secondHandler);
+		this.deniedHandlers.put(secondMatcher, secondAccessDeniedHandler);
 		this.delegator = new RequestMatcherDelegatingAccessDeniedHandler(this.deniedHandlers, this.accessDeniedHandler);
 		this.delegator.handle(this.request, null, null);
 		verify(firstHandler).handle(this.request, null, null);
-		verify(secondHandler, never()).handle(this.request, null, null);
+		verify(secondAccessDeniedHandler, never()).handle(this.request, null, null);
 		verify(this.accessDeniedHandler, never()).handle(this.request, null, null);
 		verify(secondMatcher, never()).matches(this.request);
 	}
@@ -83,15 +85,14 @@
 	public void handleWhenSecondMatchesThenOnlySecondInvoked() throws Exception {
 		AccessDeniedHandler firstHandler = mock(AccessDeniedHandler.class);
 		RequestMatcher firstMatcher = mock(RequestMatcher.class);
-		AccessDeniedHandler secondHandler = mock(AccessDeniedHandler.class);
 		RequestMatcher secondMatcher = mock(RequestMatcher.class);
 		given(firstMatcher.matches(this.request)).willReturn(false);
 		given(secondMatcher.matches(this.request)).willReturn(true);
 		this.deniedHandlers.put(firstMatcher, firstHandler);
-		this.deniedHandlers.put(secondMatcher, secondHandler);
+		this.deniedHandlers.put(secondMatcher, secondAccessDeniedHandler);
 		this.delegator = new RequestMatcherDelegatingAccessDeniedHandler(this.deniedHandlers, this.accessDeniedHandler);
 		this.delegator.handle(this.request, null, null);
-		verify(secondHandler).handle(this.request, null, null);
+		verify(secondAccessDeniedHandler).handle(this.request, null, null);
 		verify(firstHandler, never()).handle(this.request, null, null);
 		verify(this.accessDeniedHandler, never()).handle(this.request, null, null);
 	}
