--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/AuthenticatedPrincipalOAuth2AuthorizedClientRepositoryTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/AuthenticatedPrincipalOAuth2AuthorizedClientRepositoryTests.java
@@ -45,6 +45,8 @@
 
 	private OAuth2AuthorizedClientService authorizedClientService;
 
+	private OAuth2AuthorizedClient oauth2AuthorizedClient;
+
 	private OAuth2AuthorizedClientRepository anonymousAuthorizedClientRepository;
 
 	private AuthenticatedPrincipalOAuth2AuthorizedClientRepository authorizedClientRepository;
@@ -56,6 +58,7 @@
 	@BeforeEach
 	public void setup() {
 		this.authorizedClientService = mock(OAuth2AuthorizedClientService.class);
+		this.oauth2AuthorizedClient = mock(OAuth2AuthorizedClient.class);
 		this.anonymousAuthorizedClientRepository = mock(OAuth2AuthorizedClientRepository.class);
 		this.authorizedClientRepository = new AuthenticatedPrincipalOAuth2AuthorizedClientRepository(
 				this.authorizedClientService);
@@ -95,19 +98,17 @@
 	@Test
 	public void saveAuthorizedClientWhenAuthenticatedPrincipalThenSaveToService() {
 		Authentication authentication = this.createAuthenticatedPrincipal();
-		OAuth2AuthorizedClient authorizedClient = mock(OAuth2AuthorizedClient.class);
-		this.authorizedClientRepository.saveAuthorizedClient(authorizedClient, authentication, this.request,
+		this.authorizedClientRepository.saveAuthorizedClient(oauth2AuthorizedClient, authentication, this.request,
 				this.response);
-		verify(this.authorizedClientService).saveAuthorizedClient(authorizedClient, authentication);
+		verify(this.authorizedClientService).saveAuthorizedClient(oauth2AuthorizedClient, authentication);
 	}
 
 	@Test
 	public void saveAuthorizedClientWhenAnonymousPrincipalThenSaveToAnonymousRepository() {
 		Authentication authentication = this.createAnonymousPrincipal();
-		OAuth2AuthorizedClient authorizedClient = mock(OAuth2AuthorizedClient.class);
-		this.authorizedClientRepository.saveAuthorizedClient(authorizedClient, authentication, this.request,
+		this.authorizedClientRepository.saveAuthorizedClient(oauth2AuthorizedClient, authentication, this.request,
 				this.response);
-		verify(this.anonymousAuthorizedClientRepository).saveAuthorizedClient(authorizedClient, authentication,
+		verify(this.anonymousAuthorizedClientRepository).saveAuthorizedClient(oauth2AuthorizedClient, authentication,
 				this.request, this.response);
 	}
 
