--- a/web/src/test/java/org/springframework/security/web/authentication/ott/OneTimeTokenAuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/ott/OneTimeTokenAuthenticationFilterTests.java
@@ -60,6 +60,9 @@
 	@Mock
 	private AuthenticationManager authenticationManager;
 
+	@Mock
+	private OneTimeTokenAuthenticationConverter authenticationConverter;
+
 	private final OneTimeTokenAuthenticationFilter filter = new OneTimeTokenAuthenticationFilter();
 
 	private final HttpServletResponse response = new MockHttpServletResponse();
@@ -78,22 +81,20 @@
 	@Test
 	@SuppressWarnings("removal")
 	void doFilterWhenUrlDoesNotMatchThenContinues() throws ServletException, IOException {
-		OneTimeTokenAuthenticationConverter converter = mock(OneTimeTokenAuthenticationConverter.class);
 		HttpServletResponse response = mock(HttpServletResponse.class);
-		this.filter.setAuthenticationConverter(converter);
+		this.filter.setAuthenticationConverter(authenticationConverter);
 		this.filter.doFilter(post("/nomatch").buildRequest(new MockServletContext()), response, this.chain);
-		verifyNoInteractions(converter, response);
+		verifyNoInteractions(authenticationConverter, response);
 		verify(this.chain).doFilter(any(), any());
 	}
 
 	@Test
 	@SuppressWarnings("removal")
 	void doFilterWhenMethodDoesNotMatchThenContinues() throws ServletException, IOException {
-		OneTimeTokenAuthenticationConverter converter = mock(OneTimeTokenAuthenticationConverter.class);
 		HttpServletResponse response = mock(HttpServletResponse.class);
-		this.filter.setAuthenticationConverter(converter);
+		this.filter.setAuthenticationConverter(authenticationConverter);
 		this.filter.doFilter(get("/login/ott").buildRequest(new MockServletContext()), response, this.chain);
-		verifyNoInteractions(converter, response);
+		verifyNoInteractions(authenticationConverter, response);
 		verify(this.chain).doFilter(any(), any());
 	}
 
