--- a/saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/HttpSessionSaml2AuthenticationRequestRepositoryTests.java
+++ b/saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/HttpSessionSaml2AuthenticationRequestRepositoryTests.java
@@ -44,11 +44,14 @@
 
 	private HttpSessionSaml2AuthenticationRequestRepository authenticationRequestRepository;
 
+	private AbstractSaml2AuthenticationRequest authenticationRequestMock;
+
 	@BeforeEach
 	public void setup() {
 		this.request = new MockHttpServletRequest();
 		this.response = new MockHttpServletResponse();
 		this.authenticationRequestRepository = new HttpSessionSaml2AuthenticationRequestRepository();
+		this.authenticationRequestMock = mock(AbstractSaml2AuthenticationRequest.class);
 	}
 
 	@Test
@@ -80,8 +83,7 @@
 
 	@Test
 	public void saveAuthenticationRequestWhenSessionDontExistsThenCreateAndSave() {
-		AbstractSaml2AuthenticationRequest mockAuthenticationRequest = mock(AbstractSaml2AuthenticationRequest.class);
-		this.authenticationRequestRepository.saveAuthenticationRequest(mockAuthenticationRequest, this.request,
+		this.authenticationRequestRepository.saveAuthenticationRequest(authenticationRequestMock, this.request,
 				this.response);
 		AbstractSaml2AuthenticationRequest authenticationRequest = this.authenticationRequestRepository
 			.loadAuthenticationRequest(this.request);
@@ -90,9 +92,8 @@
 
 	@Test
 	public void saveAuthenticationRequestWhenSessionExistsThenSave() {
-		AbstractSaml2AuthenticationRequest mockAuthenticationRequest = mock(AbstractSaml2AuthenticationRequest.class);
 		this.request.getSession();
-		this.authenticationRequestRepository.saveAuthenticationRequest(mockAuthenticationRequest, this.request,
+		this.authenticationRequestRepository.saveAuthenticationRequest(authenticationRequestMock, this.request,
 				this.response);
 		AbstractSaml2AuthenticationRequest authenticationRequest = this.authenticationRequestRepository
 			.loadAuthenticationRequest(this.request);
