--- a/web/src/test/java/org/springframework/security/web/authentication/logout/DelegatingLogoutSuccessHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/logout/DelegatingLogoutSuccessHandlerTests.java
@@ -23,6 +23,7 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 import org.springframework.mock.web.MockHttpServletResponse;
@@ -78,8 +79,9 @@
 
 	@Test
 	public void onLogoutSuccessFirstMatches() throws Exception {
+		this.matcher = createMockRequestMatcher(this.request);
+		setup();
 		this.delegatingHandler.setDefaultLogoutSuccessHandler(this.defaultHandler);
-		given(this.matcher.matches(this.request)).willReturn(true);
 		this.delegatingHandler.onLogoutSuccess(this.request, this.response, this.authentication);
 		verify(this.handler).onLogoutSuccess(this.request, this.response, this.authentication);
 		verifyNoMoreInteractions(this.matcher2, this.handler2, this.defaultHandler);
@@ -87,8 +89,12 @@
 
 	@Test
 	public void onLogoutSuccessSecondMatches() throws Exception {
+		this.matcher2 = createMockRequestMatcher(this.request);
+		LinkedHashMap<RequestMatcher, LogoutSuccessHandler> matcherToHandler = new LinkedHashMap<>();
+		matcherToHandler.put(this.matcher, this.handler);
+		matcherToHandler.put(this.matcher2, this.handler2);
+		this.delegatingHandler = new DelegatingLogoutSuccessHandler(matcherToHandler);
 		this.delegatingHandler.setDefaultLogoutSuccessHandler(this.defaultHandler);
-		given(this.matcher2.matches(this.request)).willReturn(true);
 		this.delegatingHandler.onLogoutSuccess(this.request, this.response, this.authentication);
 		verify(this.handler2).onLogoutSuccess(this.request, this.response, this.authentication);
 		verifyNoMoreInteractions(this.handler, this.defaultHandler);
@@ -108,4 +114,10 @@
 		verifyNoMoreInteractions(this.handler, this.handler2, this.defaultHandler);
 	}
 
+	private static RequestMatcher createMockRequestMatcher(HttpServletRequest request) {
+		RequestMatcher matcher = Mockito.mock(RequestMatcher.class);
+		given(matcher.matches(request)).willReturn(true);
+		return matcher;
+	}
+
 }
