--- a/core/src/test/java/org/springframework/security/authorization/AuthorizationManagerFactoryTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthorizationManagerFactoryTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.authorization;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.security.authentication.TestAuthentication;
@@ -38,6 +39,13 @@
  */
 public class AuthorizationManagerFactoryTests {
 
+	private AuthorizationManager<String> additionalAuthorizationManager;
+
+	@BeforeEach
+	public void setUp() {
+		this.additionalAuthorizationManager = mock(AuthorizationManager.class);
+	}
+
 	@Test
 	public void permitAllReturnsSingleResultAuthorizationManagerByDefault() {
 		AuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
@@ -124,35 +132,32 @@
 
 	@Test
 	public void anonymousWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
-		factory.setAdditionalAuthorization(additional);
+		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
+		factory.setAdditionalAuthorization(additionalAuthorizationManager);
 
 		factory.anonymous();
 
-		verifyNoInteractions(additional);
+		verifyNoInteractions(additionalAuthorizationManager);
 	}
 
 	@Test
 	public void permitAllWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
-		factory.setAdditionalAuthorization(additional);
+		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
+		factory.setAdditionalAuthorization(additionalAuthorizationManager);
 
 		factory.permitAll();
 
-		verifyNoInteractions(additional);
+		verifyNoInteractions(additionalAuthorizationManager);
 	}
 
 	@Test
 	public void denyAllAllWhenAdditionalAuthorizationThenNotInvoked() {
-		AuthorizationManager<String> additional = mock(AuthorizationManager.class);
-		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
-		factory.setAdditionalAuthorization(additional);
+		DefaultAuthorizationManagerFactory<String> factory = new DefaultAuthorizationManagerFactory<>();
+		factory.setAdditionalAuthorization(additionalAuthorizationManager);
 
 		factory.permitAll();
 
-		verifyNoInteractions(additional);
+		verifyNoInteractions(additionalAuthorizationManager);
 	}
 
 	@Test
