--- a/web/src/test/java/org/springframework/security/web/authentication/AuthenticationEntryPointFailureHandlerTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/AuthenticationEntryPointFailureHandlerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.security.web.authentication;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.security.authentication.AuthenticationServiceException;
@@ -29,18 +30,23 @@
  */
 public class AuthenticationEntryPointFailureHandlerTests {
 
+	private AuthenticationEntryPoint authenticationEntryPoint;
+
+	@BeforeEach
+	void setUp() {
+		this.authenticationEntryPoint = mock(AuthenticationEntryPoint.class);
+	}
+
 	@Test
 	void onAuthenticationFailureWhenRethrowingThenAuthenticationServiceExceptionSwallowed() throws Exception {
-		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
-		AuthenticationEntryPointFailureHandler handler = new AuthenticationEntryPointFailureHandler(entryPoint);
+		AuthenticationEntryPointFailureHandler handler = new AuthenticationEntryPointFailureHandler(authenticationEntryPoint);
 		handler.setRethrowAuthenticationServiceException(false);
 		handler.onAuthenticationFailure(null, null, new AuthenticationServiceException("fail"));
 	}
 
 	@Test
 	void handleWhenDefaultsThenAuthenticationServiceExceptionRethrown() {
-		AuthenticationEntryPoint entryPoint = mock(AuthenticationEntryPoint.class);
-		AuthenticationEntryPointFailureHandler handler = new AuthenticationEntryPointFailureHandler(entryPoint);
+		AuthenticationEntryPointFailureHandler handler = new AuthenticationEntryPointFailureHandler(authenticationEntryPoint);
 		assertThatExceptionOfType(AuthenticationServiceException.class)
 			.isThrownBy(() -> handler.onAuthenticationFailure(null, null, new AuthenticationServiceException("fail")));
 	}
