--- a/core/src/test/java/org/springframework/security/authorization/AuthenticatedReactiveAuthorizationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthenticatedReactiveAuthorizationManagerTests.java
@@ -45,7 +45,7 @@
 
 	@Test
 	public void checkWhenAuthenticatedThenReturnTrue() {
-		given(this.authentication.isAuthenticated()).willReturn(true);
+		this.authentication = MockAuthenticationForReactiveAuthorization.createMockAuthentication();
 		boolean granted = this.manager.authorize(Mono.just(this.authentication), null).block().isGranted();
 		assertThat(granted).isTrue();
 	}
--- a/core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java
+++ b/core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java
@@ -69,7 +69,7 @@
 
 	@Test
 	public void checkWhenHasAuthorityAndAuthenticatedAndNoAuthoritiesThenReturnFalse() {
-		given(this.authentication.isAuthenticated()).willReturn(true);
+		this.authentication = MockAuthenticationForReactiveAuthorization.createMockAuthentication();
 		given(this.authentication.getAuthorities()).willReturn(Collections.emptyList());
 		boolean granted = this.manager.authorize(Mono.just(this.authentication), null).block().isGranted();
 		assertThat(granted).isFalse();
--- a/core/src/test/java/org/springframework/security/authorization/MockAuthenticationForReactiveAuthorization.java
+++ b/core/src/test/java/org/springframework/security/authorization/MockAuthenticationForReactiveAuthorization.java
@@ -0,0 +1,20 @@
+package org.springframework.security.authorization;
+
+import org.mockito.Mockito;
+
+import org.springframework.security.core.Authentication;
+
+import static org.mockito.BDDMockito.given;
+
+public final class MockAuthenticationForReactiveAuthorization {
+
+	private MockAuthenticationForReactiveAuthorization() {
+	}
+
+	public static Authentication createMockAuthentication() {
+		Authentication authentication = Mockito.mock(Authentication.class);
+		given(authentication.isAuthenticated()).willReturn(true);
+		return authentication;
+	}
+
+}
