--- a/core/src/test/java/org/springframework/security/authentication/dao/DaoAuthenticationProviderTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/dao/DaoAuthenticationProviderTests.java
@@ -20,6 +20,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
 
@@ -82,6 +83,13 @@
 
 	private static final List<GrantedAuthority> ROLES_12 = AuthorityUtils.createAuthorityList("ROLE_ONE", "ROLE_TWO");
 
+	private UserDetailsPasswordService userDetailsPasswordService;
+
+	@BeforeEach
+	void setUp() {
+		this.userDetailsPasswordService = mock(UserDetailsPasswordService.class);
+	}
+
 	@Test
 	public void testAuthenticateFailsForIncorrectPasswordCase() {
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated("rod", "KOala");
@@ -285,15 +293,14 @@
 				"password");
 		PasswordEncoder encoder = mock(PasswordEncoder.class);
 		UserDetailsService userDetailsService = mock(UserDetailsService.class);
-		UserDetailsPasswordService passwordManager = mock(UserDetailsPasswordService.class);
 		DaoAuthenticationProvider provider = new DaoAuthenticationProvider(userDetailsService);
 		provider.setPasswordEncoder(encoder);
-		provider.setUserDetailsPasswordService(passwordManager);
+		provider.setUserDetailsPasswordService(userDetailsPasswordService);
 		UserDetails user = PasswordEncodedUser.user();
 		given(encoder.matches(any(), any())).willReturn(false);
 		given(userDetailsService.loadUserByUsername(any())).willReturn(user);
 		assertThatExceptionOfType(BadCredentialsException.class).isThrownBy(() -> provider.authenticate(token));
-		verifyNoMoreInteractions(passwordManager);
+		verifyNoMoreInteractions(userDetailsPasswordService);
 	}
 
 	@Test
@@ -302,16 +309,15 @@
 				"password");
 		PasswordEncoder encoder = mock(PasswordEncoder.class);
 		UserDetailsService userDetailsService = mock(UserDetailsService.class);
-		UserDetailsPasswordService passwordManager = mock(UserDetailsPasswordService.class);
 		DaoAuthenticationProvider provider = new DaoAuthenticationProvider(userDetailsService);
 		provider.setPasswordEncoder(encoder);
-		provider.setUserDetailsPasswordService(passwordManager);
+		provider.setUserDetailsPasswordService(userDetailsPasswordService);
 		UserDetails user = PasswordEncodedUser.user();
 		given(encoder.matches(any(), any())).willReturn(true);
 		given(encoder.upgradeEncoding(any())).willReturn(false);
 		given(userDetailsService.loadUserByUsername(any())).willReturn(user);
 		Authentication result = provider.authenticate(token);
-		verifyNoMoreInteractions(passwordManager);
+		verifyNoMoreInteractions(userDetailsPasswordService);
 	}
 
 	@Test
