--- a/data/src/test/java/org/springframework/security/data/repository/query/SecurityEvaluationContextExtensionTests.java
+++ b/data/src/test/java/org/springframework/security/data/repository/query/SecurityEvaluationContextExtensionTests.java
@@ -104,8 +104,7 @@
 	public void setTrustResolverWhenNotNullThenVerifyRootObject() {
 		TestingAuthenticationToken explicit = new TestingAuthenticationToken("explicit", "password", "ROLE_EXPLICIT");
 		this.securityExtension = new SecurityEvaluationContextExtension(explicit);
-		AuthenticationTrustResolver trustResolver = mock(AuthenticationTrustResolver.class);
-		given(trustResolver.isAuthenticated(explicit)).willReturn(true);
+		AuthenticationTrustResolver trustResolver = createMockAuthenticationTrustResolver(explicit);
 		this.securityExtension.setTrustResolver(trustResolver);
 		assertThat(getRoot().isAuthenticated()).isTrue();
 		verify(trustResolver).isAuthenticated(explicit);
@@ -161,8 +160,7 @@
 	public void setAuthorizationManagerFactoryWithTrustResolverThenVerifyRootObject() {
 		TestingAuthenticationToken explicit = new TestingAuthenticationToken("explicit", "password", "ROLE_EXPLICIT");
 		this.securityExtension = new SecurityEvaluationContextExtension(explicit);
-		AuthenticationTrustResolver trustResolver = mock(AuthenticationTrustResolver.class);
-		given(trustResolver.isAuthenticated(explicit)).willReturn(true);
+		AuthenticationTrustResolver trustResolver = createMockAuthenticationTrustResolver(explicit);
 		DefaultAuthorizationManagerFactory<Object> factory = new DefaultAuthorizationManagerFactory<>();
 		factory.setTrustResolver(trustResolver);
 		this.securityExtension.setAuthorizationManagerFactory(factory);
@@ -203,6 +201,12 @@
 		assertThat(securityExpressionRoot.hasPermission(new Object(), "read")).isFalse();
 	}
 
+	private static AuthenticationTrustResolver createMockAuthenticationTrustResolver(TestingAuthenticationToken explicit) {
+		AuthenticationTrustResolver trustResolver = mock(AuthenticationTrustResolver.class);
+		given(trustResolver.isAuthenticated(explicit)).willReturn(true);
+		return trustResolver;
+	}
+
 	private SecurityExpressionRoot<?> getRoot() {
 		return this.securityExtension.getRootObject();
 	}
