--- a/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
+++ b/config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java
@@ -667,12 +667,11 @@
 
 	@Test
 	public void shouldConfigureAuthorizationRedirectStrategyForOAuth2Login() {
-		ServerRedirectStrategy authorizationRedirectStrategy = mock(ServerRedirectStrategy.class);
 		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
 				ReactiveClientRegistrationRepository.class);
 		given(clientRegistrationRepository.findByRegistrationId(anyString()))
 			.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
-		given(authorizationRedirectStrategy.sendRedirect(any(), any())).willReturn(Mono.empty());
+		ServerRedirectStrategy authorizationRedirectStrategy = createMockAuthorizationRedirectStrategy();
 
 		SecurityWebFilterChain securityFilterChain = this.http
 			.oauth2Login((login) -> login.clientRegistrationRepository(clientRegistrationRepository)
@@ -713,12 +712,11 @@
 
 	@Test
 	public void shouldConfigureAuthorizationRedirectStrategyForOAuth2Client() {
-		ServerRedirectStrategy authorizationRedirectStrategy = mock(ServerRedirectStrategy.class);
+		ServerRedirectStrategy authorizationRedirectStrategy = createMockAuthorizationRedirectStrategy();
 		ReactiveClientRegistrationRepository clientRegistrationRepository = mock(
 				ReactiveClientRegistrationRepository.class);
 		given(clientRegistrationRepository.findByRegistrationId(anyString()))
 			.willReturn(Mono.just(TestClientRegistrations.clientRegistration().build()));
-		given(authorizationRedirectStrategy.sendRedirect(any(), any())).willReturn(Mono.empty());
 
 		SecurityWebFilterChain securityFilterChain = this.http
 			.oauth2Client((client) -> client.clientRegistrationRepository(clientRegistrationRepository)
@@ -769,6 +767,12 @@
 			.isNotFound()
 			.expectBody(String.class)
 			.isEqualTo(null);
+	}
+
+	private static ServerRedirectStrategy createMockAuthorizationRedirectStrategy() {
+		ServerRedirectStrategy authorizationRedirectStrategy = mock(ServerRedirectStrategy.class);
+		given(authorizationRedirectStrategy.sendRedirect(any(), any())).willReturn(Mono.empty());
+		return authorizationRedirectStrategy;
 	}
 
 	private boolean isX509Filter(WebFilter filter) {
