--- a/web/src/test/java/org/springframework/security/web/authentication/AuthenticationFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/AuthenticationFilterTests.java
@@ -145,9 +145,9 @@
 		given(this.authenticationManager.authenticate(any())).willReturn(authentication);
 		AuthenticationFilter filter = new AuthenticationFilter(this.authenticationManager,
 				this.authenticationConverter);
-		SecurityContextHolderStrategy strategy = mock(SecurityContextHolderStrategy.class);
+		SecurityContextHolderStrategy strategy = MockSecurityContextHolderStrategyFactory
+				.createMockSecurityContextHolderStrategy(new SecurityContextImpl());
 		given(strategy.createEmptyContext()).willReturn(new SecurityContextImpl());
-		given(strategy.getContext()).willReturn(new SecurityContextImpl());
 		filter.setSecurityContextHolderStrategy(strategy);
 		MockHttpServletRequest request = new MockHttpServletRequest("GET", "/");
 		MockHttpServletResponse response = new MockHttpServletResponse();
--- a/web/src/test/java/org/springframework/security/web/authentication/DefaultLoginPageGeneratingFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/DefaultLoginPageGeneratingFilterTests.java
@@ -283,7 +283,8 @@
 
 	@Test
 	public void generateWhenAuthenticatedThenReadOnlyUsername() throws Exception {
-		SecurityContextHolderStrategy strategy = mock(SecurityContextHolderStrategy.class);
+		SecurityContextHolderStrategy strategy = MockSecurityContextHolderStrategyFactory
+				.createMockSecurityContextHolderStrategy(new SecurityContextImpl(TestAuthentication.authenticatedUser()));
 		DefaultLoginPageGeneratingFilter filter = new DefaultLoginPageGeneratingFilter();
 		filter.setLoginPageUrl(DefaultLoginPageGeneratingFilter.DEFAULT_LOGIN_PAGE_URL);
 		filter.setFormLoginEnabled(true);
@@ -292,7 +293,6 @@
 		filter.setOneTimeTokenEnabled(true);
 		filter.setOneTimeTokenGenerationUrl("/ott/authenticate");
 		filter.setSecurityContextHolderStrategy(strategy);
-		given(strategy.getContext()).willReturn(new SecurityContextImpl(TestAuthentication.authenticatedUser()));
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		filter.doFilter(TestMockHttpServletRequests.get("/login").build(), response, this.chain);
 		assertThat(response.getContentAsString()).contains("Request a One-Time Token");
--- a/web/src/test/java/org/springframework/security/web/authentication/MockSecurityContextHolderStrategyFactory.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/MockSecurityContextHolderStrategyFactory.java
@@ -0,0 +1,20 @@
+package org.springframework.security.web.authentication;
+
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolderStrategy;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public final class MockSecurityContextHolderStrategyFactory {
+
+	private MockSecurityContextHolderStrategyFactory() {
+	}
+
+	public static SecurityContextHolderStrategy createMockSecurityContextHolderStrategy(SecurityContext securityContext) {
+		SecurityContextHolderStrategy strategy = mock(SecurityContextHolderStrategy.class);
+		given(strategy.getContext()).willReturn(securityContext);
+		return strategy;
+	}
+
+}
