--- a/core/src/test/java/org/springframework/security/authentication/dao/DaoAuthenticationProviderTests.java
+++ b/core/src/test/java/org/springframework/security/authentication/dao/DaoAuthenticationProviderTests.java
@@ -262,15 +262,13 @@
 		String encodedPassword = "encoded";
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated("user",
 				password);
-		PasswordEncoder encoder = mock(PasswordEncoder.class);
+		PasswordEncoder encoder = createMockPasswordEncoder(true);
 		UserDetailsService userDetailsService = mock(UserDetailsService.class);
 		UserDetailsPasswordService passwordManager = mock(UserDetailsPasswordService.class);
 		DaoAuthenticationProvider provider = new DaoAuthenticationProvider(userDetailsService);
 		provider.setPasswordEncoder(encoder);
 		provider.setUserDetailsPasswordService(passwordManager);
 		UserDetails user = PasswordEncodedUser.user();
-		given(encoder.matches(any(), any())).willReturn(true);
-		given(encoder.upgradeEncoding(any())).willReturn(true);
 		given(encoder.encode(any())).willReturn(encodedPassword);
 		given(userDetailsService.loadUserByUsername(any())).willReturn(user);
 		given(passwordManager.updatePassword(any(), any())).willReturn(user);
@@ -300,15 +298,13 @@
 	public void authenticateWhenNotUpgradeAndPasswordManagerThenNoUpdate() {
 		UsernamePasswordAuthenticationToken token = UsernamePasswordAuthenticationToken.unauthenticated("user",
 				"password");
-		PasswordEncoder encoder = mock(PasswordEncoder.class);
+		PasswordEncoder encoder = createMockPasswordEncoder(false);
 		UserDetailsService userDetailsService = mock(UserDetailsService.class);
 		UserDetailsPasswordService passwordManager = mock(UserDetailsPasswordService.class);
 		DaoAuthenticationProvider provider = new DaoAuthenticationProvider(userDetailsService);
 		provider.setPasswordEncoder(encoder);
 		provider.setUserDetailsPasswordService(passwordManager);
 		UserDetails user = PasswordEncodedUser.user();
-		given(encoder.matches(any(), any())).willReturn(true);
-		given(encoder.upgradeEncoding(any())).willReturn(false);
 		given(userDetailsService.loadUserByUsername(any())).willReturn(user);
 		Authentication result = provider.authenticate(token);
 		verifyNoMoreInteractions(passwordManager);
@@ -550,6 +546,13 @@
 		SecurityAssertions.assertThat(result).hasAuthority(FactorGrantedAuthority.PASSWORD_AUTHORITY);
 	}
 
+	private static PasswordEncoder createMockPasswordEncoder(boolean upgradeEncoding) {
+		PasswordEncoder encoder = mock(PasswordEncoder.class);
+		given(encoder.matches(any(), any())).willReturn(true);
+		given(encoder.upgradeEncoding(any())).willReturn(upgradeEncoding);
+		return encoder;
+	}
+
 	private UserDetailsService withUsers(UserDetails... users) {
 		return new InMemoryUserDetailsManager(users);
 	}
