--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java
@@ -149,6 +149,9 @@
 	@Mock
 	private OAuth2AuthorizationFailureHandler authorizationFailureHandler;
 
+	@Mock
+	private OAuth2User oauth2User;
+
 	@Captor
 	private ArgumentCaptor<OAuth2AuthorizationException> authorizationExceptionCaptor;
 
@@ -594,9 +597,8 @@
 		this.function.setDefaultOAuth2AuthorizedClient(true);
 		MockHttpServletRequest servletRequest = new MockHttpServletRequest();
 		MockHttpServletResponse servletResponse = new MockHttpServletResponse();
-		OAuth2User user = mock(OAuth2User.class);
 		List<GrantedAuthority> authorities = AuthorityUtils.createAuthorityList("ROLE_USER");
-		OAuth2AuthenticationToken authentication = new OAuth2AuthenticationToken(user, authorities,
+		OAuth2AuthenticationToken authentication = new OAuth2AuthenticationToken(oauth2User, authorities,
 				this.registration.getRegistrationId());
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.registration, "principalName",
 				this.accessToken);
@@ -633,11 +635,10 @@
 		this.function.setDefaultOAuth2AuthorizedClient(true);
 		MockHttpServletRequest servletRequest = new MockHttpServletRequest();
 		MockHttpServletResponse servletResponse = new MockHttpServletResponse();
-		OAuth2User user = mock(OAuth2User.class);
 		List<GrantedAuthority> authorities = AuthorityUtils.createAuthorityList("ROLE_USER");
-		OAuth2AuthenticationToken initialAuthentication = new OAuth2AuthenticationToken(user, authorities,
+		OAuth2AuthenticationToken initialAuthentication = new OAuth2AuthenticationToken(oauth2User, authorities,
 				"initial-registration-id");
-		OAuth2AuthenticationToken authentication = new OAuth2AuthenticationToken(user, authorities,
+		OAuth2AuthenticationToken authentication = new OAuth2AuthenticationToken(oauth2User, authorities,
 				this.registration.getRegistrationId());
 		OAuth2AuthorizedClient authorizedClient = new OAuth2AuthorizedClient(this.registration, "principalName",
 				this.accessToken);
