--- a/web/src/test/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilterTests.java
@@ -75,8 +75,11 @@
 
 	private static final List<GrantedAuthority> ROLES_12 = AuthorityUtils.createAuthorityList("ROLE_ONE", "ROLE_TWO");
 
+	private SecurityContextRepository mockSecurityContextRepository;
+
 	@BeforeEach
 	public void authenticateCurrentUser() {
+		this.mockSecurityContextRepository = mock(SecurityContextRepository.class);
 		UsernamePasswordAuthenticationToken auth = UsernamePasswordAuthenticationToken.unauthenticated("dano",
 				"hawaii50");
 		SecurityContextHolder.getContext().setAuthentication(auth);
@@ -514,20 +517,19 @@
 
 	@Test
 	void doFilterWhenSwitchUserThenSaveSecurityContext() throws ServletException, IOException {
-		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", "/login/impersonate");
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		MockFilterChain filterChain = new MockFilterChain();
 		request.setParameter(SwitchUserFilter.SPRING_SECURITY_SWITCH_USERNAME_KEY, "jacklord");
 		SwitchUserFilter filter = new SwitchUserFilter();
-		filter.setSecurityContextRepository(securityContextRepository);
+		filter.setSecurityContextRepository(mockSecurityContextRepository);
 		filter.setUserDetailsService(new MockUserDetailsService());
 		filter.setTargetUrl("/target");
 		filter.afterPropertiesSet();
 
 		filter.doFilter(request, response, filterChain);
 
-		verify(securityContextRepository).saveContext(any(), any(), any());
+		verify(mockSecurityContextRepository).saveContext(any(), any(), any());
 	}
 
 	@Test
@@ -540,20 +542,19 @@
 		UsernamePasswordAuthenticationToken admin = UsernamePasswordAuthenticationToken.authenticated("jacklord",
 				"hawaii50", adminAuths);
 		SecurityContextHolder.getContext().setAuthentication(admin);
-		SecurityContextRepository securityContextRepository = mock(SecurityContextRepository.class);
 		MockHttpServletRequest request = new MockHttpServletRequest("POST", "/logout/impersonate");
 		MockHttpServletResponse response = new MockHttpServletResponse();
 		MockFilterChain filterChain = new MockFilterChain();
 		request.setParameter(SwitchUserFilter.SPRING_SECURITY_SWITCH_USERNAME_KEY, "jacklord");
 		SwitchUserFilter filter = new SwitchUserFilter();
-		filter.setSecurityContextRepository(securityContextRepository);
+		filter.setSecurityContextRepository(mockSecurityContextRepository);
 		filter.setUserDetailsService(new MockUserDetailsService());
 		filter.setTargetUrl("/target");
 		filter.afterPropertiesSet();
 
 		filter.doFilter(request, response, filterChain);
 
-		verify(securityContextRepository).saveContext(any(), any(), any());
+		verify(mockSecurityContextRepository).saveContext(any(), any(), any());
 	}
 
 	private class MockUserDetailsService implements UserDetailsService {
