--- a/web/src/test/java/org/springframework/security/web/util/matcher/AndRequestMatcherTests.java
+++ b/web/src/test/java/org/springframework/security/web/util/matcher/AndRequestMatcherTests.java
@@ -129,10 +129,10 @@
 
 	@Test
 	public void matcherWhenMatchersHavePlaceholdersThenPropagatesMatches() {
+		this.delegate = MockRequestMatcher.create(this.request, MatchResult.match(Map.of("param", "value")));
+		this.delegate2 = MockRequestMatcher.create(this.request, MatchResult.match(Map.of("param", "othervalue")));
 		this.matcher = new AndRequestMatcher(this.delegate, this.delegate2);
 
-		given(this.delegate.matcher(this.request)).willReturn(MatchResult.match(Map.of("param", "value")));
-		given(this.delegate2.matcher(this.request)).willReturn(MatchResult.match(Map.of("param", "othervalue")));
 		MatchResult result = this.matcher.matcher(this.request);
 		assertThat(result.getVariables()).containsExactlyEntriesOf(Map.of("param", "othervalue"));
 
--- a/web/src/test/java/org/springframework/security/web/util/matcher/OrRequestMatcherTests.java
+++ b/web/src/test/java/org/springframework/security/web/util/matcher/OrRequestMatcherTests.java
@@ -129,10 +129,9 @@
 
 	@Test
 	public void matcherWhenMatchersHavePlaceholdersThenPropagatesFirstMatch() {
+		this.delegate = MockRequestMatcher.create(this.request, MatchResult.match(Map.of("param", "value")));
+		this.delegate2 = MockRequestMatcher.create(this.request, MatchResult.match(Map.of("param", "othervalue")));
 		this.matcher = new OrRequestMatcher(this.delegate, this.delegate2);
-
-		given(this.delegate.matcher(this.request)).willReturn(MatchResult.match(Map.of("param", "value")));
-		given(this.delegate2.matcher(this.request)).willReturn(MatchResult.match(Map.of("param", "othervalue")));
 		MatchResult result = this.matcher.matcher(this.request);
 		assertThat(result.getVariables()).containsExactlyEntriesOf(Map.of("param", "value"));
 		verifyNoInteractions(this.delegate2);
--- a/web/src/test/java/org/springframework/security/web/util/matcher/MockRequestMatcher.java
+++ b/web/src/test/java/org/springframework/security/web/util/matcher/MockRequestMatcher.java
@@ -0,0 +1,19 @@
+package org.springframework.security.web.util.matcher;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.mockito.BDDMockito;
+import org.mockito.Mockito;
+
+final class MockRequestMatcher {
+
+	private MockRequestMatcher() {
+	}
+
+	static RequestMatcher create(HttpServletRequest request, RequestMatcher.MatchResult matchResult) {
+		RequestMatcher requestMatcher = Mockito.mock(RequestMatcher.class);
+		BDDMockito.given(requestMatcher.matcher(request)).willReturn(matchResult);
+		return requestMatcher;
+	}
+
+}
