--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientJwtBearerTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientJwtBearerTokenResponseClientTests.java
@@ -432,8 +432,7 @@
 	@Test
 	public void getTokenResponseWhenRestClientSetThenCalled() {
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
-		RestClient customClient = mock();
-		given(customClient.post()).willReturn(RestClient.builder().build().post());
+		RestClient customClient = MockRestClient.createMockRestClient();
 		this.tokenResponseClient.setRestClient(customClient);
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		JwtBearerGrantRequest grantRequest = new JwtBearerGrantRequest(clientRegistration, this.jwtAssertion);
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientTokenExchangeTokenResponseClientTests.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/RestClientTokenExchangeTokenResponseClientTests.java
@@ -563,8 +563,7 @@
 	@Test
 	public void getTokenResponseWhenRestClientSetThenCalled() {
 		this.server.enqueue(MockResponses.json("access-token-response.json"));
-		RestClient customClient = mock();
-		given(customClient.post()).willReturn(RestClient.builder().build().post());
+		RestClient customClient = MockRestClient.createMockRestClient();
 		this.tokenResponseClient.setRestClient(customClient);
 		ClientRegistration clientRegistration = this.clientRegistration.build();
 		TokenExchangeGrantRequest grantRequest = new TokenExchangeGrantRequest(clientRegistration, this.subjectToken,
--- a/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockRestClient.java
+++ b/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/endpoint/MockRestClient.java
@@ -0,0 +1,19 @@
+package org.springframework.security.oauth2.client.endpoint;
+
+import org.springframework.web.client.RestClient;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public final class MockRestClient {
+
+	private MockRestClient() {
+	}
+
+	public static RestClient createMockRestClient() {
+		RestClient customClient = mock();
+		given(customClient.post()).willReturn(RestClient.builder().build().post());
+		return customClient;
+	}
+
+}
