--- a/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java
+++ b/oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java
@@ -56,6 +56,7 @@
 import com.nimbusds.jwt.proc.JWTProcessor;
 import okhttp3.mockwebserver.MockWebServer;
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 
@@ -136,6 +137,13 @@
 
 	NimbusJwtDecoder jwtDecoder = new NimbusJwtDecoder(withoutSigning());
 
+	private RestOperations jwkSetRestOperations;
+
+	@BeforeEach
+	public void setup() {
+		this.jwkSetRestOperations = mock(RestOperations.class);
+	}
+
 	@BeforeAll
 	public static void keyFactory() throws NoSuchAlgorithmException {
 		kf = KeyFactory.getInstance("RSA");
@@ -758,18 +766,17 @@
 	// gh-11621
 	@Test
 	public void decodeWhenCacheThenRetrieveFromCache() throws Exception {
-		RestOperations restOperations = mock(RestOperations.class);
 		Cache cache = new ConcurrentMapCache("cache");
 		cache.put(JWK_SET_URI, JWK_SET);
 		// @formatter:off
 		NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 				.cache(cache)
-				.restOperations(restOperations)
+				.restOperations(jwkSetRestOperations)
 				.build();
 		// @formatter:on
 		jwtDecoder.decode(SIGNED_JWT);
 		assertThat(cache.get(JWK_SET_URI, String.class)).isSameAs(JWK_SET);
-		verifyNoInteractions(restOperations);
+		verifyNoInteractions(jwkSetRestOperations);
 	}
 
 	// gh-11621
@@ -846,18 +853,17 @@
 	// gh-11621
 	@Test
 	public void decodeWhenCacheIsConfiguredAndParseFailsOnCachedValueThenExceptionIgnored() {
-		RestOperations restOperations = mock(RestOperations.class);
 		Cache cache = new ConcurrentMapCache("cache");
 		cache.put(JWK_SET_URI, JWK_SET);
 		// @formatter:off
 		NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWK_SET_URI)
 				.cache(cache)
-				.restOperations(restOperations)
+				.restOperations(jwkSetRestOperations)
 				.build();
 		// @formatter:on
 		jwtDecoder.decode(SIGNED_JWT);
 		assertThat(cache.get(JWK_SET_URI, String.class)).isSameAs(JWK_SET);
-		verifyNoInteractions(restOperations);
+		verifyNoInteractions(jwkSetRestOperations);
 
 	}
 
