--- a/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospectorTests.java
+++ b/oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospectorTests.java
@@ -30,6 +30,7 @@
 import okhttp3.mockwebserver.MockResponse;
 import okhttp3.mockwebserver.MockWebServer;
 import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.core.ParameterizedTypeReference;
@@ -125,6 +126,13 @@
 	private static final ResponseEntity<Map<String, Object>> INVALID = response(INVALID_RESPONSE);
 
 	private static final ResponseEntity<Map<String, Object>> MALFORMED_SCOPE = response(MALFORMED_SCOPE_RESPONSE);
+
+	private RestOperations mockRestOperations;
+
+	@BeforeEach
+	public void setUp() {
+		this.mockRestOperations = mock(RestOperations.class);
+	}
 
 	@Test
 	public void introspectWhenActiveTokenThenOk() throws Exception {
@@ -287,9 +295,8 @@
 
 	@Test
 	public void setRequestEntityConverterWhenConverterIsNullThenExceptionIsThrown() {
-		RestOperations restOperations = mock(RestOperations.class);
 		SpringOpaqueTokenIntrospector introspectionClient = new SpringOpaqueTokenIntrospector(INTROSPECTION_URL,
-				restOperations);
+				mockRestOperations);
 		assertThatExceptionOfType(IllegalArgumentException.class)
 			.isThrownBy(() -> introspectionClient.setRequestEntityConverter(null));
 	}
@@ -312,9 +319,8 @@
 
 	@Test
 	public void setAuthenticationConverterWhenConverterIsNullThenExceptionIsThrown() {
-		RestOperations restOperations = mock(RestOperations.class);
 		SpringOpaqueTokenIntrospector introspectionClient = new SpringOpaqueTokenIntrospector(INTROSPECTION_URL,
-				restOperations);
+				mockRestOperations);
 		assertThatExceptionOfType(IllegalArgumentException.class)
 			.isThrownBy(() -> introspectionClient.setAuthenticationConverter(null));
 	}
