--- a/web/src/test/java/org/springframework/security/web/server/authentication/SwitchUserWebFilterTests.java
+++ b/web/src/test/java/org/springframework/security/web/server/authentication/SwitchUserWebFilterTests.java
@@ -78,6 +78,9 @@
 	ServerAuthenticationSuccessHandler successHandler;
 
 	@Mock
+	private ServerAuthenticationSuccessHandler constructorSuccessHandler;
+
+	@Mock
 	private ServerAuthenticationFailureHandler failureHandler;
 
 	@Mock
@@ -290,7 +293,7 @@
 	public void constructorUserDetailsServiceRequired() {
 		assertThatIllegalArgumentException()
 			.isThrownBy(() -> this.switchUserWebFilter = new SwitchUserWebFilter(null,
-					mock(ServerAuthenticationSuccessHandler.class), mock(ServerAuthenticationFailureHandler.class)))
+					this.constructorSuccessHandler, mock(ServerAuthenticationFailureHandler.class)))
 			.withMessage("userDetailsService must be specified");
 	}
 
@@ -313,7 +316,7 @@
 	@Test
 	public void constructorFirstDefaultValues() {
 		this.switchUserWebFilter = new SwitchUserWebFilter(mock(ReactiveUserDetailsService.class),
-				mock(ServerAuthenticationSuccessHandler.class), mock(ServerAuthenticationFailureHandler.class));
+				this.constructorSuccessHandler, mock(ServerAuthenticationFailureHandler.class));
 		final Object securityContextRepository = ReflectionTestUtils.getField(this.switchUserWebFilter,
 				"securityContextRepository");
 		assertThat(securityContextRepository).isInstanceOf(WebSessionServerSecurityContextRepository.class);
