--- a/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
@@ -24,6 +24,8 @@
 import java.util.UUID;
 
 class RequestInformationTest {
+    private final SerializationWriter serializationWriterMock = mock(SerializationWriter.class);
+
     @Test
     void ThrowsInvalidOperationExceptionWhenBaseUrlNotSet() {
         // Arrange as the request builders would
@@ -334,7 +336,6 @@
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
         final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
         requestInfo.setContentFromParsable(
                 requestAdapterMock, "application/json", new TestEntity());
