--- a/plugins/storage/object/cloudian/src/test/java/org/apache/cloudstack/storage/datastore/driver/CloudianHyperStoreObjectStoreDriverImplTest.java
+++ b/plugins/storage/object/cloudian/src/test/java/org/apache/cloudstack/storage/datastore/driver/CloudianHyperStoreObjectStoreDriverImplTest.java
@@ -435,10 +435,8 @@
         doReturn(iamClient).when(cloudianHyperStoreObjectStoreDriverImpl).getIAMClientByStoreId(TEST_STORE_ID, credential);
         ListAccessKeysResult listAccessKeyResult = mock(ListAccessKeysResult.class);
         List<AccessKeyMetadata> listAccessKeyMetadata = new ArrayList<AccessKeyMetadata>();
-        AccessKeyMetadata accessKeyNoMatch = mock(AccessKeyMetadata.class);
-        when(accessKeyNoMatch.getAccessKeyId()).thenReturn("no_match");
-        AccessKeyMetadata accessKeyMatch = mock(AccessKeyMetadata.class);
-        when(accessKeyMatch.getAccessKeyId()).thenReturn(TEST_IAM_AK);
+        AccessKeyMetadata accessKeyNoMatch = createAccessKeyMetadataMock("no_match");
+        AccessKeyMetadata accessKeyMatch = createAccessKeyMetadataMock(TEST_IAM_AK);
         listAccessKeyMetadata.add(accessKeyNoMatch);
         listAccessKeyMetadata.add(accessKeyMatch);
         when(listAccessKeyResult.getAccessKeyMetadata()).thenReturn(listAccessKeyMetadata);
@@ -683,4 +681,10 @@
         CloudRuntimeException thrown = assertThrows(CloudRuntimeException.class, () -> cloudianHyperStoreObjectStoreDriverImpl.getS3ClientByBucketAndStore(bucket, TEST_STORE_ID));
         assertTrue(thrown.getMessage().contains("not found"));
     }
+
+    private static AccessKeyMetadata createAccessKeyMetadataMock(String accessKeyIdReturn) {
+        AccessKeyMetadata accessKeyMetadata = mock(AccessKeyMetadata.class);
+        when(accessKeyMetadata.getAccessKeyId()).thenReturn(accessKeyIdReturn);
+        return accessKeyMetadata;
+    }
 }
