--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -894,7 +894,6 @@
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -908,8 +907,8 @@
         when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
+        AddressType addressType = createMockAddressType(subnetType);
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1001,7 +1000,6 @@
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1014,8 +1012,8 @@
         when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
+         AddressType addressType = createMockAddressType(subnetType);
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1030,7 +1028,6 @@
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1043,8 +1040,8 @@
         when(policyRuleType.getRuleUuid()).thenReturn("8b4637b6-5629-46de-8fb2-d0b0502bfa85");
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
+         AddressType addressType = createMockAddressType(subnetType);
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1668,4 +1665,10 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static AddressType createMockAddressType(SubnetType subnetType) {
+        AddressType addressType = mock(AddressType.class);
+        when(addressType.getSubnet()).thenReturn(subnetType);
+        return addressType;
+    }
 }
