--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -181,6 +181,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenResourceTest {
     @Mock
+    private AddressGroup addressGroup;
+    @Mock
     TungstenApi tungstenApi;
 
     TungstenResource tungstenResource;
@@ -1231,7 +1233,6 @@
     @Test
     public void executeRequestCreateTungstenAddressGroupCommandTest() {
         TungstenCommand command = new CreateTungstenAddressGroupCommand("addressName", "10.0.0.0", 24);
-        AddressGroup addressGroup = mock(AddressGroup.class);
         when(tungstenApi.createTungstenAddressGroup(anyString(), anyString(), anyString(), anyInt())).thenReturn(addressGroup);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1291,9 +1292,8 @@
     @Test
     public void executeRequestListTungstenAddressGroupCommandTest() {
         TungstenCommand command = new ListTungstenAddressGroupCommand("d64e6082-10b3-11ec-82a8-0242ac130003");
-        AddressGroup addressGroup1 = mock(AddressGroup.class);
         AddressGroup addressGroup2 = mock(AddressGroup.class);
-        Answer<List<ApiObjectBase>> addressGroupsAnswer = setupApiObjectBaseListAnswer(addressGroup1, addressGroup2);
+        Answer<List<ApiObjectBase>> addressGroupsAnswer = setupApiObjectBaseListAnswer(addressGroup, addressGroup2);
         when(tungstenApi.listTungstenAddressGroup(anyString())).thenAnswer(addressGroupsAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1343,7 +1343,7 @@
     @Test
     public void executeRequestDeleteTungstenAddressGroupCommandTest() {
         TungstenCommand command = new DeleteTungstenAddressGroupCommand("c2f8e1f8-10b6-11ec-82a8-0242ac130003");
-        AddressGroup addressGroup = mock(AddressGroup.class);
+        // removed local mock; replaced with global field `addressGroup`
         when(tungstenApi.getTungstenObject(eq(AddressGroup.class), anyString())).thenReturn(addressGroup);
         when(tungstenApi.deleteTungstenObject(any(AddressGroup.class))).thenReturn(true);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
