--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -181,6 +181,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenResourceTest {
     @Mock
+    private ApplicationPolicySet applicationPolicySet2;
+    @Mock
     TungstenApi tungstenApi;
 
     TungstenResource tungstenResource;
@@ -1119,7 +1121,6 @@
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
         Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
         when(tungstenApi.applyTungstenPolicyTag(anyString(), anyString())).thenReturn(true);
         when(tungstenApi.getTungstenObject(eq(Tag.class), anyString())).thenReturn(tag);
@@ -1170,7 +1171,6 @@
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
         Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
         when(tungstenApi.removeTungstenTag(anyList(), anyList(), anyList(), anyString(), anyString(), anyString())).thenReturn(tag);
         when(tungstenApi.getBackRefFromVirtualNetwork(eq(VirtualNetwork.class), anyList())).thenAnswer(virtualNetworksAnswer);
@@ -1242,7 +1242,6 @@
     public void executeRequestListTungstenApplicationPolicySetCommandTest() {
         TungstenCommand command = new ListTungstenApplicationPolicySetCommand("d64e6082-10b3-11ec-82a8-0242ac130003");
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
         Answer<List<ApiObjectBase>> applicationPolicySetsAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
         when(tungstenApi.listTungstenApplicationPolicySet(anyString())).thenAnswer(applicationPolicySetsAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
