--- a/utils/src/test/java/com/cloud/utils/rest/RESTServiceConnectorTest.java
+++ b/utils/src/test/java/com/cloud/utils/rest/RESTServiceConnectorTest.java
@@ -71,8 +71,7 @@
         final TestPojo newObject = new TestPojo();
         newObject.setField("newValue");
         final String newObjectJson = gson.toJson(newObject);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -89,8 +88,7 @@
         final TestPojo newObject = new TestPojo();
         newObject.setField("newValue");
         final String newObjectJson = gson.toJson(newObject);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -110,9 +108,8 @@
         final TestPojo newObject = new TestPojo();
         newObject.setField("newValue");
         final String newObjectJson = gson.toJson(newObject);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -132,9 +129,8 @@
         final TestPojo newObject = new TestPojo();
         newObject.setField("newValue");
         final String newObjectJson = gson.toJson(newObject);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -154,9 +150,8 @@
     @Test
     public void testExecuteDeleteObject() throws Exception {
         final HttpEntity entity = mock(HttpEntity.class);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(entity);
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -173,9 +168,8 @@
         final TestPojo existingObject = new TestPojo();
         existingObject.setField("existingValue");
         final String newObjectJson = gson.toJson(existingObject);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -194,9 +188,8 @@
         final TestPojo existingObject = new TestPojo();
         existingObject.setField("existingValue");
         final String newObjectJson = gson.toJson(existingObject);
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
@@ -214,8 +207,7 @@
 
     @Test(expected = JsonParseException.class)
     public void testCustomDeserializerTypeMismatch() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(new StringEntity("[{somethig_not_type : \"WrongType\"}]"));
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
@@ -230,8 +222,7 @@
 
     @Test
     public void testCustomDeserializerForCustomLists() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         when(response.getEntity()).thenReturn(new StringEntity("{results: [{field : \"SomeValue\"}], results_count: 1}"));
         final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
@@ -320,4 +311,10 @@
             }
         }
     }
+
+    private static CloseableHttpResponse createMockCloseableHttpResponse() {
+        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
+        return response;
+    }
 }
