--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -175,6 +175,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VirtualMachineManagerImplTest {
+    @Mock
+    private DeploymentPlanner planner;
 
     @Spy
     @InjectMocks
@@ -1141,7 +1143,6 @@
 
         Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
 
-        DeploymentPlanner planner = mock(DeploymentPlanner.class);
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
@@ -1237,7 +1238,6 @@
 
         Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
 
-        DeploymentPlanner planner = mock(DeploymentPlanner.class);
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
