--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -5310,7 +5310,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
-        final VirtualRoutingResource virtRouterResource = Mockito.mock(VirtualRoutingResource.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5347,9 +5346,8 @@
 
             when(nic.getIp()).thenReturn(controlIp);
             when(nic.getType()).thenReturn(TrafficType.Control);
+             final VirtualRoutingResource virtRouterResource = createVirtualRoutingResource(controlIp, vmName);
             when(libvirtComputingResourceMock.getVirtRouterResource()).thenReturn(virtRouterResource);
-            when(virtRouterResource.connect(controlIp, 1, 5000)).thenReturn(true);
-            when(virtRouterResource.isSystemVMSetup(vmName, controlIp)).thenReturn(true);
         } catch (final InternalErrorException e) {
             fail(e.getMessage());
         } catch (final LibvirtException e) {
@@ -5389,7 +5387,6 @@
         final LibvirtUtilitiesHelper libvirtUtilitiesHelper = Mockito.mock(LibvirtUtilitiesHelper.class);
         final Connect conn = Mockito.mock(Connect.class);
         final LibvirtVMDef vmDef = Mockito.mock(LibvirtVMDef.class);
-        final VirtualRoutingResource virtRouterResource = Mockito.mock(VirtualRoutingResource.class);
 
         final NicTO nic = Mockito.mock(NicTO.class);
         final NicTO[] nics = new NicTO[]{nic};
@@ -5427,9 +5424,8 @@
 
             when(nic.getIp()).thenReturn(controlIp);
             when(nic.getType()).thenReturn(TrafficType.Control);
+            final VirtualRoutingResource virtRouterResource = createVirtualRoutingResource(controlIp, vmName);
             when(libvirtComputingResourceMock.getVirtRouterResource()).thenReturn(virtRouterResource);
-            when(virtRouterResource.connect(controlIp, 1, 5000)).thenReturn(true);
-            when(virtRouterResource.isSystemVMSetup(vmName, controlIp)).thenReturn(true);
         } catch (final InternalErrorException e) {
             fail(e.getMessage());
         } catch (final LibvirtException e) {
@@ -7698,4 +7694,11 @@
         Pair<Integer, Long> result = libvirtComputingResourceSpy.getPeriodAndQuota(cpuQuotaPercentage);
         Assert.assertEquals(expectedResult, result);
     }
+
+    private static VirtualRoutingResource createVirtualRoutingResource(final String controlIp, final String vmName) {
+        final VirtualRoutingResource virtRouterResource = Mockito.mock(VirtualRoutingResource.class);
+        when(virtRouterResource.connect(controlIp, 1, 5000)).thenReturn(true);
+        when(virtRouterResource.isSystemVMSetup(vmName, controlIp)).thenReturn(true);
+        return virtRouterResource;
+    }
 }
