--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
@@ -83,6 +83,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class NsxGuestNetworkGuruTest {
+    @Mock
+    private DataCenter dataCenter;
 
     @Mock
     PhysicalNetworkDao physicalNetworkDao;
@@ -294,7 +296,6 @@
     @Test
     public void testCreateNsxSegmentForVpc() {
         NetworkVO networkVO = Mockito.mock(NetworkVO.class);
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
 
         when(networkVO.getAccountId()).thenReturn(1L);
         when(nsxControllerUtils.sendNsxCommand(any(CreateNsxSegmentCommand.class),
@@ -309,7 +310,6 @@
     public void testCreateNsxSegmentForIsolatedNetwork() {
         NetworkVO networkVO = Mockito.mock(NetworkVO.class);
         NetworkOfferingVO offeringVO = Mockito.mock(NetworkOfferingVO.class);
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
 
         when(networkVO.getAccountId()).thenReturn(1L);
         when(networkVO.getVpcId()).thenReturn(null);
