--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -105,6 +105,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenGuestNetworkGuruTest {
+    @Mock
+    private final NicVO nicVO;
 
     @Mock
     NetworkOfferingServiceMapDao ntwkOfferingSrvcDao;
@@ -244,7 +246,6 @@
         final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        final NicVO nicVO = mock(NicVO.class);
 
         when(network.getDataCenterId()).thenReturn(1L);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
@@ -261,7 +262,6 @@
         final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
-        final NicVO nicVO = mock(NicVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenThrow(
             new IllegalArgumentException());
