--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -580,7 +580,6 @@
 
     @Test
     public void preparePublicNetworkTest() throws ConcurrentOperationException {
-        Network network = mock(Network.class);
         NicProfile nicProfile = new NicProfile();
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
@@ -592,7 +591,7 @@
         TungstenAnswer createTungstenNetworkPolicyAnswer = MockTungstenAnswerFactory.get(true);
 
         nicProfile.setIPv4Address("192.168.100.100");
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
+        network = MockNetwork.createMockNetwork(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
@@ -609,7 +608,6 @@
 
     @Test
     public void prepareManagementNetworkTest() throws ConcurrentOperationException {
-        Network network = mock(Network.class);
         NicProfile nicProfile = new NicProfile();
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
@@ -619,7 +617,7 @@
         TungstenAnswer createTungstenVMAnswer = MockTungstenAnswerFactory.get(true);
 
         nicProfile.setIPv4Address("192.168.100.100");
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
+        Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
@@ -634,7 +632,6 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void prepareWithExceptionTest() throws ConcurrentOperationException {
-        Network network = mock(Network.class);
         NicProfile nicProfile = new NicProfile();
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         DeployDestination deployDestination = mock(DeployDestination.class);
@@ -644,7 +641,7 @@
         TungstenAnswer createTungstenVMAnswer = mock(TungstenAnswer.class);
 
         nicProfile.setIPv4Address("192.168.100.100");
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
+        Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
@@ -655,7 +652,6 @@
 
     @Test
     public void releasePublicNetworkTest() throws ConcurrentOperationException, ResourceUnavailableException {
-        Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
@@ -668,7 +664,7 @@
         TungstenAnswer deleteTungstenNetworkPolicyAnswer = mock(TungstenAnswer.class);
 
         when(nicProfile.getIPv4Address()).thenReturn("192.168.100.100");
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
+        Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
@@ -683,7 +679,6 @@
 
     @Test
     public void releaseManagementNetworkTest() throws ConcurrentOperationException, ResourceUnavailableException {
-        Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
@@ -693,7 +688,7 @@
         TungstenAnswer deleteVmiAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteVmAnswer = mock(TungstenAnswer.class);
 
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
+        Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
@@ -706,7 +701,6 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void releaseWithExceptionTest() throws ConcurrentOperationException, ResourceUnavailableException {
-        Network network = mock(Network.class);
         NicProfile nicProfile = mock(NicProfile.class);
         VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         ReservationContext reservationContext = mock(ReservationContext.class);
@@ -715,7 +709,7 @@
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteVmiAnswer = mock(TungstenAnswer.class);
 
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
+        Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
@@ -910,7 +904,6 @@
     @Test
     public void prepareMigrationTest() {
         NicProfile nic = mock(NicProfile.class);
-        Network network = mock(Network.class);
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         DeployDestination dest = mock(DeployDestination.class);
         ReservationContext context = mock(ReservationContext.class);
@@ -919,7 +912,7 @@
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(true);
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
+        Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(tungstenAnswer);
@@ -962,4 +955,16 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    public final class MockNetwork {
+        private MockNetwork() {
+        }
+
+        public static com.cloud.network.Network createMockNetwork(
+                com.cloud.network.Networks.TrafficType trafficType) {
+            com.cloud.network.Network network = mock(com.cloud.network.Network.class);
+            when(network.getTrafficType()).thenReturn(trafficType);
+            return network;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -241,13 +241,12 @@
 
     @Test
     public void testDeallocate() {
-        final Network network = mock(Network.class);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         final NicVO nicVO = mock(NicVO.class);
 
         when(network.getDataCenterId()).thenReturn(1L);
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
         when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
 
@@ -258,7 +257,6 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testDeallocateWithException() {
-        final Network network = mock(Network.class);
         final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         final NicVO nicVO = mock(NicVO.class);
@@ -266,7 +264,7 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong())).thenThrow(
             new IllegalArgumentException());
         when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
 
         guru.deallocate(network, nicProfile, virtualMachineProfile);
@@ -274,12 +272,11 @@
 
     @Test
     public void testImplementGuestNetwork() {
-        final Network network = mock(Network.class);
         final DeployDestination deployDestination = mock(DeployDestination.class);
         final ReservationContext reservationContext = mock(ReservationContext.class);
         final IPAddressVO ipAddressVO = mock(IPAddressVO.class);
 
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         when(network.getMode()).thenReturn(Networks.Mode.Dhcp);
         when(network.getGateway()).thenReturn("192.168.1.1");
         when(network.getCidr()).thenReturn("192.168.1.0/24");
@@ -322,11 +319,10 @@
 
     @Test
     public void testImplementSharedNetwork() {
-        final Network network = mock(Network.class);
         final DeployDestination deployDestination = mock(DeployDestination.class);
         final ReservationContext reservationContext = mock(ReservationContext.class);
 
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         when(network.getMode()).thenReturn(Networks.Mode.Dhcp);
         when(network.getGateway()).thenReturn("192.168.1.1");
         when(network.getCidr()).thenReturn("192.168.1.0/24");
@@ -355,11 +351,10 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testImplementWithException() {
-        final Network network = mock(Network.class);
         final DeployDestination deployDestination = mock(DeployDestination.class);
         final ReservationContext reservationContext = mock(ReservationContext.class);
 
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         when(network.getMode()).thenReturn(Networks.Mode.Dhcp);
         when(network.getGateway()).thenReturn("192.168.1.1");
         when(network.getCidr()).thenReturn("192.168.1.0/24");
@@ -378,7 +373,6 @@
     @Test
     public void testReserve() throws InsufficientVirtualNetworkCapacityException, InsufficientAddressCapacityException {
         final NicProfile nic = mock(NicProfile.class);
-        final Network network = mock(Network.class);
         final VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         final DeployDestination dest = mock(DeployDestination.class);
         final ReservationContext context = mock(ReservationContext.class);
@@ -387,7 +381,7 @@
 
         when(nic.getReservationStrategy()).thenReturn(Nic.ReservationStrategy.Start);
         when(vm.getType()).thenReturn(VirtualMachine.Type.User);
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
 
@@ -472,7 +466,6 @@
     @Test
     public void testPrepareMigration() {
         final NicProfile nic = mock(NicProfile.class);
-        final Network network = mock(Network.class);
         final VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         final DeployDestination dest = mock(DeployDestination.class);
         final ReservationContext context = mock(ReservationContext.class);
@@ -480,7 +473,7 @@
         final HostVO hostVO = mock(HostVO.class);
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.User);
-        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        final Network network = MockNetwork.createMockNetwork(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
@@ -526,4 +519,16 @@
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
+
+    public final class MockNetwork {
+        private MockNetwork() {
+        }
+
+        public static com.cloud.network.Network createMockNetwork(
+                com.cloud.network.Networks.TrafficType trafficType) {
+            com.cloud.network.Network network = mock(com.cloud.network.Network.class);
+            when(network.getTrafficType()).thenReturn(trafficType);
+            return network;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -413,7 +413,7 @@
     @Test
     public void addManagementNetworkSubnetTest() {
         HostPodVO hostPodVO = mock(HostPodVO.class);
-        Network managementNetwork = mock(Network.class);
+        Network managementNetwork = MockNetwork.createMockNetwork(Networks.TrafficType.Management);
         TungstenAnswer addTungstenNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         DataCenterIpAddressVO dataCenterIpAddressVO = mock(DataCenterIpAddressVO.class);
@@ -425,7 +425,6 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenNetworkDnsCommand.class), anyLong())).thenReturn(getTungstenNetworkDnsAnswer);
         when(getTungstenNetworkDnsAnswer.getDetails()).thenReturn("192.168.100.150");
         when(managementNetwork.getCidr()).thenReturn("192.168.100.0/24");
-        when(managementNetwork.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(dataCenterIpAddressDao.listByPodIdDcIdIpAddress(anyLong(), anyLong(), anyString())).thenReturn(List.of(dataCenterIpAddressVO));
         when(dataCenterIpAddressDao.mark(anyLong(), anyLong(), anyString())).thenReturn(true);
 
@@ -454,7 +453,7 @@
     @Test
     public void removeManagementNetworkSubnetTest() {
         HostPodVO hostPodVO = mock(HostPodVO.class);
-        Network managementNetwork = mock(Network.class);
+        Network managementNetwork = MockNetwork.createMockNetwork(Networks.TrafficType.Management);
         TungstenAnswer removeTungstenNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         DataCenterIpAddressVO dataCenterIpAddressVO = mock(DataCenterIpAddressVO.class);
@@ -464,7 +463,6 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenNetworkSubnetCommand.class), anyLong())).thenReturn(removeTungstenNetworkSubnetAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenNetworkDnsCommand.class), anyLong())).thenReturn(getTungstenNetworkDnsAnswer);
         when(getTungstenNetworkDnsAnswer.getDetails()).thenReturn("192.168.100.150");
-        when(managementNetwork.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(dataCenterIpAddressDao.listByPodIdDcIdIpAddress(anyLong(), anyLong(), anyString())).thenReturn(List.of(dataCenterIpAddressVO));
 
         assertTrue(tungstenService.removeManagementNetworkSubnet(hostPodVO));
@@ -498,7 +496,7 @@
     @Test
     public void addPublicNetworkSubnetTest() {
         VlanVO vlanVO = mock(VlanVO.class);
-        Network publicNetwork = mock(Network.class);
+        Network publicNetwork = MockNetwork.createMockNetwork(Networks.TrafficType.Public);
         TungstenAnswer addTungstenNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer createTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer applyTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
@@ -516,7 +514,6 @@
         when(vlanVO.getVlanGateway()).thenReturn("192.168.100.1");
         when(vlanVO.getVlanNetmask()).thenReturn("255.255.255.0");
         when(publicNetwork.getCidr()).thenReturn("192.168.100.0/24");
-        when(publicNetwork.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(createTungstenPolicyAnswer.getApiObjectBase()).thenReturn(networkPolicy);
         when(ipAddressDao.mark(anyLong(), any(Ip.class))).thenReturn(true);
 
@@ -541,7 +538,7 @@
     @Test
     public void removePublicNetworkSubnetTest() {
         VlanVO vlanVO = mock(VlanVO.class);
-        Network publicNetwork = mock(Network.class);
+        Network publicNetwork = MockNetwork.createMockNetwork(Networks.TrafficType.Public);
         TungstenAnswer deleteTungstenNetworkPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer removeTungstenNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
@@ -552,7 +549,6 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenNetworkSubnetCommand.class), anyLong())).thenReturn(removeTungstenNetworkSubnetAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenNetworkDnsCommand.class), anyLong())).thenReturn(getTungstenNetworkDnsAnswer);
         when(getTungstenNetworkDnsAnswer.getDetails()).thenReturn("192.168.100.150");
-        when(publicNetwork.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
 
         assertTrue(tungstenService.removePublicNetworkSubnet(vlanVO));
@@ -1668,4 +1664,16 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    public final class MockNetwork {
+        private MockNetwork() {
+        }
+
+        public static com.cloud.network.Network createMockNetwork(
+                com.cloud.network.Networks.TrafficType trafficType) {
+            com.cloud.network.Network network = mock(com.cloud.network.Network.class);
+            when(network.getTrafficType()).thenReturn(trafficType);
+            return network;
+        }
+    }
 }
