--- a/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
@@ -524,9 +524,8 @@
     @Test
     public void testCreateVpc() {
         mockVpcDnsResources(true, false);
-        VpcVO vpc = Mockito.mock(VpcVO.class);
+        VpcVO vpc = createMockVpcVO();
         Mockito.when(vpcDao.persist(any(), anyMap())).thenReturn(vpc);
-        Mockito.when(vpc.getUuid()).thenReturn("uuid");
         try (MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
             manager.createVpc(zoneId, vpcOfferingId, vpcOwnerId, vpcName, vpcName, ip4Cidr, vpcDomain,
                     ip4Dns[0], ip4Dns[1], null, null, true, 1500, null, null, null, false, true);
@@ -538,9 +537,8 @@
     @Test
     public void testCreateRoutedVpc() {
         mockVpcDnsResources(true, false);
-        VpcVO vpc = Mockito.mock(VpcVO.class);
+        VpcVO vpc = createMockVpcVO();
         Mockito.when(vpcDao.persist(any(), anyMap())).thenReturn(vpc);
-        Mockito.when(vpc.getUuid()).thenReturn("uuid");
         doReturn(true).when(routedIpv4Manager).isRoutedVpc(any());
         doNothing().when(routedIpv4Manager).getOrCreateIpv4SubnetForVpc(any(), anyString());
         try (MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
@@ -556,9 +554,8 @@
     @Test
     public void testCreateRoutedVpcWithDynamicRouting() {
         mockVpcDnsResources(true, false);
-        VpcVO vpc = Mockito.mock(VpcVO.class);
+        VpcVO vpc = createMockVpcVO();
         Mockito.when(vpcDao.persist(any(), anyMap())).thenReturn(vpc);
-        Mockito.when(vpc.getUuid()).thenReturn("uuid");
         doReturn(true).when(routedIpv4Manager).isRoutedVpc(any());
         doReturn(true).when(routedIpv4Manager).isValidGateway(vpcOfferingVO);
         doReturn(true).when(routedIpv4Manager).isDynamicRoutedVpc(vpcOfferingVO);
@@ -658,4 +655,10 @@
         Assert.assertTrue(result.isEmpty());
     }
 
+
+    private static VpcVO createMockVpcVO() {
+        VpcVO vpc = Mockito.mock(VpcVO.class);
+        Mockito.when(vpc.getUuid()).thenReturn("uuid");
+        return vpc;
+    }
 }
