--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
@@ -71,6 +71,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KVMStorageProcessorTest {
+    @Mock
+    private KVMPhysicalDisk baseFile;
 
     @Mock
     KVMStoragePoolManager storagePoolManager;
@@ -291,7 +293,7 @@
 
     @Test
     public void convertBaseFileToSnapshotFileInPrimaryStorageDirTestFailToConvertWithLibvirtExceptionReturnErrorMessage() {
-        KVMPhysicalDisk baseFile = Mockito.mock(KVMPhysicalDisk.class);
+        // removed local mock; replaced with global field `baseFile`
         String snapshotPath = "snapshotPath";
 
         Mockito.doReturn(true).when(kvmStoragePoolMock).createFolder(Mockito.anyString());
@@ -305,7 +307,6 @@
 
     @Test
     public void convertBaseFileToSnapshotFileInPrimaryStorageDirTestConvertSuccessReturnNull() throws Exception {
-        KVMPhysicalDisk baseFile = Mockito.mock(KVMPhysicalDisk.class);
         String snapshotPath = "snapshotPath";
 
         Mockito.doReturn(true).when(kvmStoragePoolMock).createFolder(Mockito.anyString());
