--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/LibvirtStoragePoolTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/LibvirtStoragePoolTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package com.cloud.hypervisor.kvm.storage;
 
+import org.mockito.Mock;
 import org.apache.cloudstack.utils.qemu.QemuImg.PhysicalDiskFormat;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -29,6 +30,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class LibvirtStoragePoolTest extends TestCase {
+    @Mock
+    private StorageAdaptor adapter;
 
     @Test
     public void testAttributes() {
@@ -37,7 +40,6 @@
 
         StoragePoolType type = StoragePoolType.NetworkFilesystem;
 
-        StorageAdaptor adapter = Mockito.mock(LibvirtStorageAdaptor.class);
         StoragePool storage = Mockito.mock(StoragePool.class);
 
         LibvirtStoragePool pool = new LibvirtStoragePool(uuid, name, type, adapter, storage);
@@ -62,7 +64,6 @@
         String uuid = "f40cbf53-1f37-4c62-8912-801edf398f47";
         String name = "myfirstpool";
 
-        StorageAdaptor adapter = Mockito.mock(LibvirtStorageAdaptor.class);
         StoragePool storage = Mockito.mock(StoragePool.class);
 
         LibvirtStoragePool nfsPool = new LibvirtStoragePool(uuid, name, StoragePoolType.NetworkFilesystem, adapter, storage);
@@ -83,7 +84,7 @@
         String uuid = "60b46738-c5d0-40a9-a79e-9a4fe6295db7";
         String name = "myfirstpool";
 
-        StorageAdaptor adapter = Mockito.mock(LibvirtStorageAdaptor.class);
+        // removed local mock; replaced with global field `adapter`
         StoragePool storage = Mockito.mock(StoragePool.class);
 
         LibvirtStoragePool nfsPool = new LibvirtStoragePool(uuid, name, StoragePoolType.NetworkFilesystem, adapter, storage);
