--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -2125,8 +2125,8 @@
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
 
         when(conditionTO.getId()).thenReturn(conditionId);
-        when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
-
+
+        AutoScalePolicyTO scaleUpPolicyTO = createMockAutoScalePolicyTO(scaleUpPolicyId);
         when(counterDao.findById(counterId)).thenReturn(counterMock);
         when(counterMock.getSource()).thenReturn(Counter.Source.MEMORY);
 
@@ -2157,11 +2157,10 @@
     public void updateCountersMapWithInstantDataForCPU() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
-        AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
+        AutoScalePolicyTO scaleUpPolicyTO = createMockAutoScalePolicyTO(scaleUpPolicyId);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
 
         when(conditionTO.getId()).thenReturn(conditionId);
-        when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
         when(counterDao.findById(counterId)).thenReturn(counterMock);
         when(counterMock.getSource()).thenReturn(Counter.Source.CPU);
 
@@ -2425,10 +2424,9 @@
     @Test
     public void cleanupAsVmGroupStatistics() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
+        AutoScalePolicyTO policyTO = createMockAutoScalePolicyTO(scaleUpPolicyId);
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(groupTO.getId()).thenReturn(vmGroupId);
-        when(policyTO.getId()).thenReturn(scaleUpPolicyId);
 
         autoScaleManagerImplSpy.cleanupAsVmGroupStatistics(groupTO);
 
@@ -2581,4 +2579,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static AutoScalePolicyTO createMockAutoScalePolicyTO(Long scaleUpPolicyId) {
+        AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
+        when(policyTO.getId()).thenReturn(scaleUpPolicyId);
+        return policyTO;
+    }
 }
