--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1286,8 +1286,7 @@
         final Long prefixId = 1L;
         DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
         Mockito.when(cmd.getId()).thenReturn(prefixId);
-        DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
-        Mockito.when(prefixVO.getId()).thenReturn(prefixId);
+        DataCenterGuestIpv6PrefixVO prefixVO = createDataCenterGuestIpv6Prefix(prefixId);
         Mockito.when(dataCenterGuestIpv6PrefixDao.findById(Mockito.anyLong())).thenReturn(prefixVO);
         Mockito.when(ipv6GuestPrefixSubnetNetworkMapDao.listUsedByPrefix(Mockito.anyLong()))
                 .thenReturn(List.of(Mockito.mock(Ipv6GuestPrefixSubnetNetworkMapVO.class)));
@@ -1299,8 +1298,7 @@
         final Long prefixId = 1L;
         DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
         Mockito.when(cmd.getId()).thenReturn(prefixId);
-        DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
-        Mockito.when(prefixVO.getId()).thenReturn(prefixId);
+        DataCenterGuestIpv6PrefixVO prefixVO = createDataCenterGuestIpv6Prefix(prefixId);
         Mockito.when(dataCenterGuestIpv6PrefixDao.findById(Mockito.anyLong())).thenReturn(prefixVO);
         Mockito.when(ipv6GuestPrefixSubnetNetworkMapDao.listUsedByPrefix(Mockito.anyLong())).thenReturn(new ArrayList<>());
         final List<Long> removedPrefix = new ArrayList<>();
@@ -1508,4 +1506,10 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static DataCenterGuestIpv6PrefixVO createDataCenterGuestIpv6Prefix(Long prefixId) {
+        DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
+        Mockito.when(prefixVO.getId()).thenReturn(prefixId);
+        return prefixVO;
+    }
 }
