--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -105,6 +105,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenGuestNetworkGuruTest {
+    @Mock
+    private final VirtualMachine vm;
 
     @Mock
     NetworkOfferingServiceMapDao ntwkOfferingSrvcDao;
@@ -497,7 +499,6 @@
         final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
-        final VirtualMachine vm = mock(VirtualMachine.class);
         final HostVO hostVO = mock(HostVO.class);
 
         when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
@@ -515,7 +516,6 @@
         final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
-        final VirtualMachine vm = mock(VirtualMachine.class);
         final HostVO hostVO = mock(HostVO.class);
 
         when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
