--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
@@ -328,10 +328,9 @@
         resource.configure("NiciraNvpResource", parameters);
 
         final LogicalRouter lrc = mock(LogicalRouter.class);
-        final LogicalRouterPort lrp = mock(LogicalRouterPort.class);
+        final LogicalRouterPort lrp = createLogicalRouterPort();
         final LogicalSwitchPort lsp = mock(LogicalSwitchPort.class);
         when(lrc.getUuid()).thenReturn("ccccc");
-        when(lrp.getUuid()).thenReturn("ddddd").thenReturn("eeeee");
         when(lsp.getUuid()).thenReturn("fffff");
         when(nvpApi.createLogicalRouter((LogicalRouter) any())).thenReturn(lrc);
         when(nvpApi.createLogicalRouterPort(eq("ccccc"), (LogicalRouterPort) any())).thenReturn(lrp);
@@ -375,10 +374,9 @@
         resource.configure("NiciraNvpResource", parameters);
 
         final LogicalRouter lrc = mock(LogicalRouter.class);
-        final LogicalRouterPort lrp = mock(LogicalRouterPort.class);
+        final LogicalRouterPort lrp = createLogicalRouterPort();
         final LogicalSwitchPort lsp = mock(LogicalSwitchPort.class);
         when(lrc.getUuid()).thenReturn("ccccc");
-        when(lrp.getUuid()).thenReturn("ddddd").thenReturn("eeeee");
         when(lsp.getUuid()).thenReturn("fffff");
         when(nvpApi.createLogicalRouter((LogicalRouter) any())).thenReturn(lrc);
         when(nvpApi.createLogicalRouterPort(eq("ccccc"), (LogicalRouterPort) any())).thenReturn(lrp);
@@ -822,4 +820,10 @@
         assertTrue(snr.getMatch().getSourcePort() == 8080);
         assertTrue(snr.getMatch().getEthertype().equals("IPv4") && rules[1].getMatch().getProtocol() == 6);
     }
+
+    private static LogicalRouterPort createLogicalRouterPort() {
+        final LogicalRouterPort lrp = mock(LogicalRouterPort.class);
+        when(lrp.getUuid()).thenReturn("ddddd").thenReturn("eeeee");
+        return lrp;
+    }
 }
