--- a/server/src/test/java/com/cloud/network/IpAddressManagerTest.java
+++ b/server/src/test/java/com/cloud/network/IpAddressManagerTest.java
@@ -71,6 +71,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class IpAddressManagerTest {
+    @Mock
+    private NetworkRuleApplier applier;
 
     @Mock
     IPAddressDao ipAddressDao;
@@ -537,7 +539,6 @@
         when(networkDao.findById(networkId)).thenReturn(network);
 
         FirewallRule rule = makeRule(networkId, null, FirewallRule.Purpose.Firewall, FirewallRule.TrafficType.Ingress);
-        NetworkRuleApplier applier = mock(NetworkRuleApplier.class);
 
         when(ipAddressDao.listByAssociatedNetwork(networkId, null)).thenReturn(new ArrayList<>());
         stubIpAssocHelpers();
@@ -561,7 +562,6 @@
         when(vpcDao.findById(vpcId)).thenReturn(vpc);
 
         FirewallRule rule = makeRule(null, vpcId, FirewallRule.Purpose.Firewall, FirewallRule.TrafficType.Ingress);
-        NetworkRuleApplier applier = mock(NetworkRuleApplier.class);
 
         stubIpAssocHelpers();
 
@@ -584,7 +584,6 @@
         when(vpcDao.findById(vpcId)).thenReturn(vpc);
 
         FirewallRule rule = makeRule(null, vpcId, FirewallRule.Purpose.Firewall, FirewallRule.TrafficType.Egress);
-        NetworkRuleApplier applier = mock(NetworkRuleApplier.class);
 
         stubIpAssocHelpers();
 
@@ -608,7 +607,6 @@
 
         stubIpAssocHelpers();
 
-        NetworkRuleApplier applier = mock(NetworkRuleApplier.class);
         FirewallRule rule = makeRule(null, vpcId, FirewallRule.Purpose.Firewall, FirewallRule.TrafficType.Ingress);
 
         ipAddressManager.applyRules(Collections.singletonList(rule), FirewallRule.Purpose.Firewall, applier, false);
