--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -666,9 +666,8 @@
         StoragePoolVO clvmPool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(clvmPool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
 
-        VMInstanceVO vmInstance = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vmInstance.getInstanceName()).thenReturn(MOCK_VM_NAME);
-
+
+        VMInstanceVO vmInstance = createMockVmInstance();
         ClvmPoolManager clvmPoolManager = Mockito.mock(ClvmPoolManager.class);
         Mockito.when(clvmPoolManager.getClvmLockHostId(Mockito.eq(101L), Mockito.anyString(),
                 Mockito.anyString(), Mockito.any(), Mockito.eq(true))).thenReturn(currentHostId);
@@ -858,9 +857,8 @@
         StoragePoolVO nfsPool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(nfsPool.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
 
-        VMInstanceVO vmInstance = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vmInstance.getInstanceName()).thenReturn(MOCK_VM_NAME);
-
+
+        VMInstanceVO vmInstance = createMockVmInstance();
         ClvmPoolManager clvmPoolManager = Mockito.mock(ClvmPoolManager.class);
         Mockito.when(clvmPoolManager.getClvmLockHostId(Mockito.eq(101L), Mockito.anyString(),
                 Mockito.anyString(), Mockito.any(), Mockito.eq(true))).thenReturn(currentHostId);
@@ -899,9 +897,8 @@
         StoragePoolVO clvmPool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(clvmPool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
 
-        VMInstanceVO vmInstance = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vmInstance.getInstanceName()).thenReturn(MOCK_VM_NAME);
-
+
+        VMInstanceVO vmInstance = createMockVmInstance();
         ClvmPoolManager clvmPoolManager = Mockito.mock(ClvmPoolManager.class);
         Mockito.when(clvmPoolManager.getClvmLockHostId(Mockito.eq(101L), Mockito.anyString(),
                 Mockito.anyString(), Mockito.any(), Mockito.eq(true))).thenReturn(currentHostId);
@@ -945,4 +942,10 @@
         return null;
     }
 
+
+    private static VMInstanceVO createMockVmInstance() {
+        VMInstanceVO vmInstance = Mockito.mock(VMInstanceVO.class);
+        Mockito.when(vmInstance.getInstanceName()).thenReturn(MOCK_VM_NAME);
+        return vmInstance;
+    }
 }
