--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -403,11 +403,10 @@
     @Test
     public void testSendStopWithOkAnswer() throws Exception {
         VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
-        VirtualMachine vm = mock(VirtualMachine.class);
+        VirtualMachine vm = createMockVirtualMachine();
         VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         StopAnswer answer = new StopAnswer(new StopCommand(vm, false, false), "ok", true);
         when(profile.getVirtualMachine()).thenReturn(vm);
-        when(vm.getHostId()).thenReturn(1L);
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(answer);
 
         Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(guru, profile, false, false);
@@ -418,11 +417,10 @@
     @Test
     public void testSendStopWithFailAnswer() throws Exception {
         VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
-        VirtualMachine vm = mock(VirtualMachine.class);
+        VirtualMachine vm = createMockVirtualMachine();
         VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         StopAnswer answer = new StopAnswer(new StopCommand(vm, false, false), "fail", false);
         when(profile.getVirtualMachine()).thenReturn(vm);
-        when(vm.getHostId()).thenReturn(1L);
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(answer);
 
         Pair<Boolean, String> actual = virtualMachineManagerImpl.sendStop(guru, profile, false, false);
@@ -434,10 +432,9 @@
     @Test
     public void testSendStopWithNullAnswer() throws Exception {
         VirtualMachineGuru guru = mock(VirtualMachineGuru.class);
-        VirtualMachine vm = mock(VirtualMachine.class);
+        VirtualMachine vm = createMockVirtualMachine();
         VirtualMachineProfile profile = mock(VirtualMachineProfile.class);
         when(profile.getVirtualMachine()).thenReturn(vm);
-        when(vm.getHostId()).thenReturn(1L);
         when(vm.getInstanceName()).thenReturn("test-vm");
         when(agentManagerMock.send(anyLong(), (Command)any())).thenReturn(null);
 
@@ -2241,4 +2238,10 @@
         verify(clvmPoolManagerMock, times(1)).setClvmLockHostId(3L, destHostId);
     }
 
+
+    private static VirtualMachine createMockVirtualMachine() {
+        VirtualMachine vm = mock(VirtualMachine.class);
+        when(vm.getHostId()).thenReturn(1L);
+        return vm;
+    }
 }
