--- a/plugins/storage/sharedfs/storagevm/src/test/java/org/apache/cloudstack/storage/sharedfs/lifecycle/StorageVmSharedFSLifeCycleTest.java
+++ b/plugins/storage/sharedfs/storagevm/src/test/java/org/apache/cloudstack/storage/sharedfs/lifecycle/StorageVmSharedFSLifeCycleTest.java
@@ -250,8 +250,7 @@
         SharedFS sharedFS = prepareDeploySharedFS();
         when(sharedFS.getAccountId()).thenReturn(s_ownerId);
 
-        Account owner = mock(Account.class);
-        when(owner.getId()).thenReturn(s_ownerId);
+        Account owner = createMockAccount(s_ownerId);
         when(accountMgr.getActiveAccountById(s_ownerId)).thenReturn(owner);
 
         UserVm vm = mock(UserVm.class);
@@ -320,8 +319,7 @@
         when(sharedFS.getFsType()).thenReturn(SharedFS.FileSystemType.valueOf(s_fsFormat));
         when(sharedFS.getAccountId()).thenReturn(s_ownerId);
 
-        Account owner = mock(Account.class);
-        when(owner.getId()).thenReturn(s_ownerId);
+        Account owner = createMockAccount(s_ownerId);
         when(accountMgr.getActiveAccountById(s_ownerId)).thenReturn(owner);
 
         DataCenterVO zone = mock(DataCenterVO.class);
@@ -393,4 +391,10 @@
         boolean result = lifeCycle.reDeploySharedFS(sharedFS);
         Assert.assertEquals(result, true);
     }
+
+    private static Account createMockAccount(long ownerId) {
+        Account owner = mock(Account.class);
+        when(owner.getId()).thenReturn(ownerId);
+        return owner;
+    }
 }
