--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -1738,11 +1738,9 @@
 
         // Mock NICs
         List<NicVO> nics = new ArrayList<>();
-        NicVO nic1 = Mockito.mock(NicVO.class);
-        when(nic1.getDeviceId()).thenReturn(1);
+        NicVO nic1 = createMockNic(1);
         nics.add(nic1);
-        NicVO nic2 = Mockito.mock(NicVO.class);
-        when(nic2.getDeviceId()).thenReturn(0);
+        NicVO nic2 = createMockNic(0);
         nics.add(nic2);
         when(_nicsDao.listByVmId(vmInstanceVoMockId)).thenReturn(nics);
 
@@ -2241,4 +2239,10 @@
         verify(clvmPoolManagerMock, times(1)).setClvmLockHostId(3L, destHostId);
     }
 
+
+    private static NicVO createMockNic(int deviceId) {
+        NicVO nic = Mockito.mock(NicVO.class);
+        when(nic.getDeviceId()).thenReturn(deviceId);
+        return nic;
+    }
 }
