--- a/server/src/test/java/com/cloud/capacity/CapacityManagerTest.java
+++ b/server/src/test/java/com/cloud/capacity/CapacityManagerTest.java
@@ -44,7 +44,7 @@
     private CapacityVO cvoRam = mock(CapacityVO.class);
     private VirtualMachine vm = mock(VirtualMachine.class);
     private ClusterDetailsVO clusterDetailCpu = mock(ClusterDetailsVO.class);
-    private ClusterDetailsVO clusterDetailRam = mock(ClusterDetailsVO.class);
+    private ClusterDetailsVO clusterDetailRam;
 
     public CapacityManagerImpl setUp() {
         CapacityManagerImpl capMgr = new CapacityManagerImpl();
@@ -70,11 +70,17 @@
         when(svo.getRamSize()).thenReturn(512);
         when(cvoCpu.getReservedCapacity()).thenReturn(0l);
         when(cvoRam.getReservedCapacity()).thenReturn(0l);
-        when(clusterDetailRam.getValue()).thenReturn("1.5");
-        when(clusterDetailCpu.getValue()).thenReturn("2");
+        clusterDetailRam = createMockClusterDetailsVO("1.5");
+        clusterDetailCpu = createMockClusterDetailsVO("2");
         when(CDao.update(anyLong(), isA(CapacityVO.class))).thenReturn(true);
         boolean hasCapacity = capMgr.checkIfHostHasCapacity(host, 500, 1024 * 1024 * 1024, false, 2, 2, false);
         Assert.assertTrue(hasCapacity);
 
     }
+
+    private static ClusterDetailsVO createMockClusterDetailsVO(String value) {
+        ClusterDetailsVO clusterDetailsVO = mock(ClusterDetailsVO.class);
+        when(clusterDetailsVO.getValue()).thenReturn(value);
+        return clusterDetailsVO;
+    }
 }
