--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -803,11 +803,9 @@
         long networkId = 1L;
         long dataCenterId = 1L;
         String freeIp = "172.10.10.10";
-        IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
+        IPAddressVO ipAddressVO = createMockIpAddressVO(ip, State.Free);
         Ip ip = mock(Ip.class);
         Mockito.when(ip.addr()).thenReturn(freeIp);
-        Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
-        Mockito.when(ipAddressVO.getState()).thenReturn(State.Free);
         Mockito.when(network.getId()).thenReturn(networkId);
         Mockito.when(dataCenter.getId()).thenReturn(dataCenterId);
         Mockito.when(testOrchestrator._ipAddressDao.findBySourceNetworkIdAndDatacenterIdAndState(networkId, dataCenterId, State.Free)).thenReturn(ipAddressVO);
@@ -825,11 +823,9 @@
         long networkId = 1L;
         long dataCenterId = 1L;
         String requestedIp = "172.10.10.10";
-        IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
         Ip ip = mock(Ip.class);
         Mockito.when(ip.addr()).thenReturn(requestedIp);
-        Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
-        Mockito.when(ipAddressVO.getState()).thenReturn(State.Free);
+        IPAddressVO ipAddressVO = createMockIpAddressVO(ip, State.Free);
         Mockito.when(network.getId()).thenReturn(networkId);
         Mockito.when(dataCenter.getId()).thenReturn(dataCenterId);
         Mockito.when(ipAddresses.getIp4Address()).thenReturn(requestedIp);
@@ -848,11 +844,10 @@
         long networkId = 1L;
         long dataCenterId = 1L;
         String requestedIp = "172.10.10.10";
-        IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
         Ip ip = mock(Ip.class);
         Mockito.when(ip.addr()).thenReturn(requestedIp);
+         IPAddressVO ipAddressVO = createMockIpAddressVO(ip, State.Allocated);
         Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
-        Mockito.when(ipAddressVO.getState()).thenReturn(State.Allocated);
         Mockito.when(network.getId()).thenReturn(networkId);
         Mockito.when(dataCenter.getId()).thenReturn(dataCenterId);
         Mockito.when(ipAddresses.getIp4Address()).thenReturn(requestedIp);
@@ -1073,4 +1068,11 @@
         assertNotNull(result);
         assertEquals(1, result.size());
     }
+
+    private static IPAddressVO createMockIpAddressVO(Ip ip, State stateReturn) {
+        IPAddressVO ipAddressVO = Mockito.mock(IPAddressVO.class);
+        Mockito.when(ipAddressVO.getAddress()).thenReturn(ip);
+        Mockito.when(ipAddressVO.getState()).thenReturn(stateReturn);
+        return ipAddressVO;
+    }
 }
