--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -392,8 +392,8 @@
         Mockito.lenient().when(_accountDao.findById(1L)).thenReturn(callingAccount);
 
         Mockito.lenient().when(accountService.isNormalUser(Mockito.anyLong())).thenReturn(Boolean.TRUE);
-        Mockito.lenient().when(accountMock.getAccountId()).thenReturn(2L);
-
+
+        accountMock = createMockAccount();
         accountManagerImpl.getKeys(_getkeyscmd);
     }
 
@@ -419,8 +419,8 @@
         Mockito.lenient().when(_accountDao.findById(2L)).thenReturn(callingAccount);
 
         Mockito.lenient().when(accountService.isDomainAdmin(Mockito.anyLong())).thenReturn(Boolean.TRUE);
-        Mockito.lenient().when(accountMock.getAccountId()).thenReturn(2L);
-
+
+        accountMock = createMockAccount();
         accountManagerImpl.getKeys(_getkeyscmd);
     }
 
@@ -2003,8 +2003,7 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void testCheckCallerRoleTypeAllowedToUpdateUserLowerAccountRoleType() {
-        Account callingAccount = Mockito.mock(Account.class);
-        Mockito.lenient().when(callingAccount.getAccountId()).thenReturn(2L);
+        Account callingAccount = createMockAccount();
         Mockito.lenient().doReturn(callingAccount).when(accountManagerImpl).getAccount(2L);
         Mockito.lenient().when(accountManagerImpl.getCurrentCallingAccount()).thenReturn(callingAccount);
         Mockito.lenient().when(accountManagerImpl.getRoleType(Mockito.eq(callingAccount))).thenReturn(RoleType.DomainAdmin);
@@ -2289,4 +2288,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static Account createMockAccount() {
+        Account account = Mockito.mock(Account.class);
+        Mockito.lenient().when(account.getAccountId()).thenReturn(2L);
+        return account;
+    }
 }
