--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1688,8 +1688,7 @@
 
     @Test
     public void getHostAndVmIdsMap() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        when(groupTO.getId()).thenReturn(vmGroupId);
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(vmGroupId);
         when(autoScaleVmGroupVmMapDao.listByGroup(vmGroupId)).thenReturn(Arrays.asList(autoScaleVmGroupVmMapVOMock));
         when(autoScaleVmGroupVmMapVOMock.getInstanceId()).thenReturn(virtualMachineId);
         when(userVmDao.findById(virtualMachineId)).thenReturn(userVmMock);
@@ -2294,9 +2293,8 @@
 
     @Test
     public void processVmStatsByIdFromHost() {
-        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        when(groupTO.getId()).thenReturn(vmGroupId);
-
+
+        AutoScaleVmGroupTO groupTO = createMockAutoScaleVmGroupTO(vmGroupId);
         List<Long> vmIds = Arrays.asList(virtualMachineId);
         double cpuUtilization = 6;
         double memoryKBs = 3;
@@ -2581,4 +2579,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static AutoScaleVmGroupTO createMockAutoScaleVmGroupTO(Long id) {
+        AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
+        when(groupTO.getId()).thenReturn(id);
+        return groupTO;
+    }
 }
