--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -1226,9 +1226,8 @@
     public void testSelectKVMHostForImportingInClusterWithImportInstanceIdInvalidResourceState() {
         Long hostId = 1L;
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
-        when(host.getResourceState()).thenReturn(ResourceState.Disabled);
-
+
+        HostVO host = createMockHostVO(ResourceState.Disabled);
         when(hostDao.findById(hostId)).thenReturn(host);
 
         unmanagedVMsManager.selectKVMHostForImportingInCluster(cluster, hostId);
@@ -1360,9 +1359,8 @@
     public void testSelectKVMHostForConversionInClusterWithImportInstanceIdInvalidResourceState() {
         Long hostId = 1L;
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
-        when(host.getResourceState()).thenReturn(ResourceState.Maintenance);
-
+
+        HostVO host = createMockHostVO(ResourceState.Maintenance);
         when(hostDao.findById(hostId)).thenReturn(host);
 
         unmanagedVMsManager.selectKVMHostForConversionInCluster(cluster, hostId, false);
@@ -1609,4 +1607,10 @@
         details.put("key", "not-a-number");
         unmanagedVMsManager.getDetailAsInteger("key", details);
     }
+
+    private static HostVO createMockHostVO(ResourceState resourceState) {
+        HostVO host = Mockito.mock(HostVO.class);
+        when(host.getResourceState()).thenReturn(resourceState);
+        return host;
+    }
 }
