--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -175,6 +175,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VirtualMachineManagerImplTest {
+    @Mock
+    private ServiceOfferingVO serviceOffering;
 
     @Spy
     @InjectMocks
@@ -1130,7 +1132,6 @@
 
         ItWorkVO work = mock(ItWorkVO.class);
 
-        ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
         when(template.isDeployAsIs()).thenReturn(false);
@@ -1226,7 +1227,6 @@
 
         ItWorkVO work = mock(ItWorkVO.class);
 
-        ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
         when(template.isDeployAsIs()).thenReturn(false);
@@ -1720,9 +1720,8 @@
         AccountVO owner = Mockito.mock(AccountVO.class);
         when(_entityMgr.findById(Account.class, accountId)).thenReturn(owner);
 
-        ServiceOfferingVO offering = Mockito.mock(ServiceOfferingVO.class);
-        when(serviceOfferingDaoMock.findById(vmInstanceVoMockId, offeringId)).thenReturn(offering);
-
+
+        when(serviceOfferingDaoMock.findById(vmInstanceVoMockId, offeringId)).thenReturn(serviceOffering);
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
         when(_entityMgr.findByIdIncludingRemoved(VirtualMachineTemplate.class, templateId)).thenReturn(template);
 
