--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -786,14 +786,13 @@
     @Test
     public void processHostAboutToBeRemovedWithSecurityGroupTest() {
         HostVO hostVO = mock(HostVO.class);
-        TungstenProviderVO tungstenProvider = mock(TungstenProviderVO.class);
+        TungstenProviderVO tungstenProvider = MockTungstenProviderVO.createTungstenProviderVO("192.168.100.100");
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
         NetworkVO network = mock(NetworkVO.class);
 
         when(hostVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProvider);
         when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
-        when(tungstenProvider.getGateway()).thenReturn("192.168.100.100");
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
         when(dataCenterVO.isSecurityGroupEnabled()).thenReturn(true);
@@ -806,13 +805,12 @@
     @Test
     public void processHostAboutToBeRemovedWithoutSecurityGroupTest() {
         HostVO hostVO = mock(HostVO.class);
-        TungstenProviderVO tungstenProvider = mock(TungstenProviderVO.class);
+        TungstenProviderVO tungstenProvider = MockTungstenProviderVO.createTungstenProviderVO("192.168.100.100");
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(hostVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProvider);
         when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
-        when(tungstenProvider.getGateway()).thenReturn("192.168.100.100");
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
         when(dataCenterVO.isSecurityGroupEnabled()).thenReturn(false);
@@ -962,4 +960,16 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    public final class MockTungstenProviderVO {
+
+        private MockTungstenProviderVO() {
+        }
+
+        public static com.cloud.network.element.TungstenProviderVO createTungstenProviderVO(String gateway) {
+            com.cloud.network.element.TungstenProviderVO tungstenProvider = mock(com.cloud.network.element.TungstenProviderVO.class);
+            when(tungstenProvider.getGateway()).thenReturn(gateway);
+            return tungstenProvider;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -1503,7 +1503,7 @@
         TungstenAnswer addNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         NetworkDetailVO networkDetailVO = mock(NetworkDetailVO.class);
-        TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
+        TungstenProviderVO tungstenProviderVO = MockTungstenProviderVO.createTungstenProviderVO("192.168.100.1");
         HostVO hostVO = mock(HostVO.class);
         Answer answer = mock(Answer.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
@@ -1528,7 +1528,6 @@
         when(getTungstenNetworkDnsAnswer.getDetails()).thenReturn("192.168.1.150");
         when(createTungstenSharedNetworkAnswer.getApiObjectBase()).thenReturn(apiObjectBase);
         when(apiObjectBase.getQualifiedName()).thenReturn(List.of("network"));
-        when(tungstenProviderVO.getGateway()).thenReturn("192.168.100.1");
 
         assertTrue(tungstenService.createSharedNetwork(network, vlan));
     }
@@ -1668,4 +1667,16 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    public final class MockTungstenProviderVO {
+
+        private MockTungstenProviderVO() {
+        }
+
+        public static com.cloud.network.element.TungstenProviderVO createTungstenProviderVO(String gateway) {
+            com.cloud.network.element.TungstenProviderVO tungstenProvider = mock(com.cloud.network.element.TungstenProviderVO.class);
+            when(tungstenProvider.getGateway()).thenReturn(gateway);
+            return tungstenProvider;
+        }
+    }
 }
