--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -246,9 +246,7 @@
         eventJoins.add(mock(EventJoinVO.class));
         eventJoins.add(mock(EventJoinVO.class));
 
-        NetworkVO network = mock(NetworkVO.class);
-        Mockito.when(network.getId()).thenReturn(1L);
-        Mockito.when(network.getAccountId()).thenReturn(account.getId());
+        NetworkVO network = createMockNetworkVO(account.getId());
         Mockito.when(entityManager.findByUuidIncludingRemoved(Network.class, uuid)).thenReturn(network);
         Mockito.doNothing().when(accountManager).checkAccess(account, SecurityChecker.AccessType.ListEntry, true, network);
         Mockito.when(eventDao.searchAndCount(Mockito.any(), Mockito.any(Filter.class))).thenReturn(pair);
@@ -304,9 +302,7 @@
         String uuid = UUID.randomUUID().toString();
         Mockito.when(cmd.getResourceId()).thenReturn(uuid);
         Mockito.when(cmd.getResourceType()).thenReturn(ApiCommandResourceType.Network.toString());
-        NetworkVO network = mock(NetworkVO.class);
-        Mockito.when(network.getId()).thenReturn(1L);
-        Mockito.when(network.getAccountId()).thenReturn(2L);
+        NetworkVO network = createMockNetworkVO(2L);
         Mockito.when(entityManager.findByUuidIncludingRemoved(Network.class, uuid)).thenReturn(network);
         Mockito.doThrow(new PermissionDeniedException("Denied")).when(accountManager).checkAccess(account, SecurityChecker.AccessType.ListEntry, false, network);
         queryManager.searchForEvents(cmd);
@@ -891,4 +887,11 @@
         assertEquals(1, hostTags.size());
         assertTrue(hostTags.contains(tag1));
     }
+
+    private static NetworkVO createMockNetworkVO(long accountIdReturn) {
+        NetworkVO network = Mockito.mock(NetworkVO.class);
+        Mockito.when(network.getId()).thenReturn(1L);
+        Mockito.when(network.getAccountId()).thenReturn(accountIdReturn);
+        return network;
+    }
 }
