--- a/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
@@ -453,8 +453,7 @@
 
     @Test
     public void testFindCorrectResourceLimitForDomain() {
-        DomainVO domain = Mockito.mock(DomainVO.class);
-        Mockito.when(domain.getId()).thenReturn(1L);
+        DomainVO domain = createMockDomainVO(1L);
         long result = resourceLimitManager.findCorrectResourceLimitForDomain(domain, Resource.ResourceType.cpu, hostTags.get(0));
         Assert.assertEquals(Resource.RESOURCE_UNLIMITED, result);
 
@@ -490,18 +489,16 @@
 
     @Test
     public void testResourceUnlimitedForDomainProjects() {
-            DomainVO domain = Mockito.mock(DomainVO.class);
-            Mockito.when(domain.getId()).thenReturn(1L);
-
+
+            DomainVO domain = createMockDomainVO(1L);
             long result = resourceLimitManager.findCorrectResourceLimitForDomain(domain, Resource.ResourceType.project,
                             hostTags.get(0));
             Assert.assertEquals(Resource.RESOURCE_UNLIMITED, result);
     }
     @Test
     public void testSpecificLimitForDomainProjects() {
-        DomainVO domain = Mockito.mock(DomainVO.class);
-        Mockito.when(domain.getId()).thenReturn(2L);
-
+
+        DomainVO domain = createMockDomainVO(2L);
         ResourceLimitVO limit = new ResourceLimitVO();
         limit.setMax(100L);
         Mockito.when(resourceLimitDao.findByOwnerIdAndTypeAndTag(2L, Resource.ResourceOwnerType.Domain, Resource.ResourceType.project, hostTags.get(0))).thenReturn(limit);
@@ -512,9 +509,8 @@
 
     @Test
     public void testParentDomainLimitForDomainProjects() {
-            DomainVO domain = Mockito.mock(DomainVO.class);
-            Mockito.when(domain.getId()).thenReturn(3L);
-
+
+            DomainVO domain = createMockDomainVO(3L);
             DomainVO parentDomain = Mockito.mock(DomainVO.class);
             Mockito.when(domain.getParent()).thenReturn(5L);
             Mockito.when(domainDao.findById(5L)).thenReturn(parentDomain);
@@ -533,8 +529,7 @@
 
     @Test
     public void testDefaultDomainProjectLimit() {
-            DomainVO domain = Mockito.mock(DomainVO.class);
-            Mockito.when(domain.getId()).thenReturn(4L);
+            DomainVO domain = createMockDomainVO(4L);
             Mockito.when(domain.getParent()).thenReturn(null);
 
             long defaultDomainProjectsMax = 250L;
@@ -1410,4 +1405,10 @@
         Assert.assertEquals(EnumSet.allOf(Resource.ResourceType.class).size(), result.size());
         Assert.assertEquals(25L, result.get(0).getMax().longValue());
     }
+
+    private static DomainVO createMockDomainVO(long id) {
+        DomainVO domain = Mockito.mock(DomainVO.class);
+        Mockito.when(domain.getId()).thenReturn(id);
+        return domain;
+    }
 }
