--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -211,10 +211,8 @@
         Mockito.when(vmSnapshotDao.findById(ArgumentMatchers.anyLong())).thenReturn(vmSnapshot);
         Mockito.when(vmSnapshot.getId()).thenReturn(1L);
         Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
-        HostVO hostVO = Mockito.mock(HostVO.class);
+        HostVO hostVO = MockHostVO.createMockHostVO();
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
-        Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
-        Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
 
         RevertToVMSnapshotAnswer answer = Mockito.mock(RevertToVMSnapshotAnswer.class);
         Mockito.when(answer.getResult()).thenReturn(Boolean.TRUE);
@@ -288,10 +286,8 @@
         Mockito.when(vmSnapshotDao.findById(ArgumentMatchers.anyLong())).thenReturn(vmSnapshot);
         Mockito.when(vmSnapshot.getId()).thenReturn(1L);
         Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
-        HostVO hostVO = Mockito.mock(HostVO.class);
+        HostVO hostVO = MockHostVO.createMockHostVO();
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
-        Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
-        Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
         DeleteVMSnapshotAnswer answer = Mockito.mock(DeleteVMSnapshotAnswer.class);
         Mockito.when(answer.getResult()).thenReturn(true);
 
@@ -449,4 +445,18 @@
             return Mockito.mock(SnapshotDataStoreDao.class);
         }
     }
+
+    public final class MockHostVO {
+        private MockHostVO() {
+        }
+
+        public static com.cloud.host.HostVO createMockHostVO() {
+            com.cloud.host.HostVO hostVO = Mockito.mock(com.cloud.host.HostVO.class);
+            Mockito.when(hostVO.getHypervisorType())
+                    .thenReturn(com.cloud.hypervisor.Hypervisor.HypervisorType.KVM);
+            Mockito.when(hostVO.getHypervisorVersion())
+                    .thenReturn("default");
+            return hostVO;
+        }
+    }
 }
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
@@ -129,7 +129,7 @@
         Mockito.when(guestOsHypervisorDao.findById(ArgumentMatchers.anyLong())).thenReturn(guestOSHypervisorVO);
         Mockito.when(guestOsHypervisorDao.findByOsIdAndHypervisor(ArgumentMatchers.anyLong(), ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(guestOSHypervisorVO);
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(null);
-        HostVO hostVO = Mockito.mock(HostVO.class);
+        HostVO hostVO = MockHostVO.createMockHostVO();
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
         Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
@@ -178,10 +178,9 @@
         Mockito.when(vmSnapshot.getId()).thenReturn(1L);
         Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(null);
-        HostVO hostVO = Mockito.mock(HostVO.class);
+        HostVO hostVO = MockHostVO.createMockHostVO();
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
-        Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
         Exception e = null;
         try {
             vmSnapshotStrategy.revertVMSnapshot(vmSnapshot);
@@ -226,10 +225,9 @@
         Mockito.when(vmSnapshot.getId()).thenReturn(1L);
         Mockito.when(vmSnapshot.getCreated()).thenReturn(new Date());
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(null);
-        HostVO hostVO = Mockito.mock(HostVO.class);
+        HostVO hostVO = MockHostVO.createMockHostVO();
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
-        Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
 
         Exception e = null;
         try {
@@ -342,4 +340,18 @@
             return Mockito.mock(BackupManager.class);
         }
     }
+
+    public final class MockHostVO {
+        private MockHostVO() {
+        }
+
+        public static com.cloud.host.HostVO createMockHostVO() {
+            com.cloud.host.HostVO hostVO = Mockito.mock(com.cloud.host.HostVO.class);
+            Mockito.when(hostVO.getHypervisorType())
+                    .thenReturn(com.cloud.hypervisor.Hypervisor.HypervisorType.KVM);
+            Mockito.when(hostVO.getHypervisorVersion())
+                    .thenReturn("default");
+            return hostVO;
+        }
+    }
 }
