--- a/server/src/test/java/com/cloud/network/IpAddressManagerTest.java
+++ b/server/src/test/java/com/cloud/network/IpAddressManagerTest.java
@@ -360,7 +360,7 @@
 
     @Test
     public void checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocatedTestIpIsNotInQuarantineShouldReturnTrue() {
-        Mockito.when(ipAddressMock.getId()).thenReturn(dummyID);
+        ipAddressMock = MockIpAddress.createMockIpAddress(dummyID);
         Mockito.when(publicIpQuarantineDaoMock.findByPublicIpAddressId(Mockito.anyLong())).thenReturn(null);
 
         boolean result = ipAddressManager.canPublicIpAddressBeAllocated(ipAddressMock, account);
@@ -370,7 +370,7 @@
 
     @Test
     public void checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocatedTestIpIsNoLongerInQuarantineShouldReturnTrue() {
-        Mockito.when(ipAddressMock.getId()).thenReturn(dummyID);
+        ipAddressMock = MockIpAddress.createMockIpAddress(dummyID);
         Mockito.when(publicIpQuarantineDaoMock.findByPublicIpAddressId(Mockito.anyLong())).thenReturn(publicIpQuarantineVOMock);
         Mockito.doReturn(false).when(ipAddressManager).isPublicIpAddressStillInQuarantine(Mockito.any(PublicIpQuarantineVO.class), Mockito.any(Date.class));
         Mockito.doNothing().when(ipAddressManager).removePublicIpAddressFromQuarantine(Mockito.anyLong(), Mockito.anyString());
@@ -382,7 +382,7 @@
 
     @Test
     public void checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocatedTestIpIsInQuarantineAndThePreviousOwnerIsTheSameAsTheNewOwnerShouldReturnTrue() {
-        Mockito.when(ipAddressMock.getId()).thenReturn(dummyID);
+        ipAddressMock = MockIpAddress.createMockIpAddress(dummyID);
         Mockito.when(publicIpQuarantineDaoMock.findByPublicIpAddressId(Mockito.anyLong())).thenReturn(publicIpQuarantineVOMock);
 
         Mockito.doReturn(true).when(ipAddressManager).isPublicIpAddressStillInQuarantine(Mockito.any(PublicIpQuarantineVO.class), Mockito.any(Date.class));
@@ -402,7 +402,7 @@
     public void checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocatedTestIpIsInQuarantineAndThePreviousOwnerIsDifferentFromTheNewOwnerShouldReturnFalse() {
         final String UUID_2 = "uuid_2";
 
-        Mockito.when(ipAddressMock.getId()).thenReturn(dummyID);
+        ipAddressMock = MockIpAddress.createMockIpAddress(dummyID);
         Mockito.when(publicIpQuarantineDaoMock.findByPublicIpAddressId(Mockito.anyLong())).thenReturn(publicIpQuarantineVOMock);
 
         Mockito.doReturn(true).when(ipAddressManager).isPublicIpAddressStillInQuarantine(Mockito.any(PublicIpQuarantineVO.class), Mockito.any(Date.class));
@@ -640,4 +640,17 @@
 
         assertFalse(result);
     }
+
+    public final class MockIpAddress {
+
+        private MockIpAddress() {
+            // Utility class
+        }
+
+        public static com.cloud.network.IpAddress createMockIpAddress(long id) {
+            com.cloud.network.IpAddress ipAddress = org.mockito.Mockito.mock(com.cloud.network.IpAddress.class);
+            org.mockito.Mockito.when(ipAddress.getId()).thenReturn(id);
+            return ipAddress;
+        }
+    }
 }
--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -1206,7 +1206,6 @@
         ReflectionTestUtils.setField(createNetworkCmd, "zoneId", zoneId);
         ReflectionTestUtils.setField(createNetworkCmd, "physicalNetworkId", null);
         NetworkVO networkVO = Mockito.spy(NetworkVO.class);
-        IpAddress ipAddress = Mockito.mock(IPAddressVO.class);
         NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
         Long networkId = 7l;
         when(networkVO.getId()).thenReturn(networkId);
@@ -1214,7 +1213,7 @@
         when(networkDao.findById(networkId)).thenReturn(networkVO);
         when(entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(ntwkOff);
         when(entityMgr.findById(eq(DataCenter.class), anyLong())).thenReturn(dc);
-        when(ipAddress.getId()).thenReturn(5l);
+        IpAddress ipAddress = MockIpAddress.createMockIpAddress(5l);
         when(networkVO.getId()).thenReturn(networkId);
         when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
         try {
@@ -1378,4 +1377,17 @@
 
         Assert.assertFalse(service.getAndValidateSupportForKeepMacAddressOnPublicNicParameter(false, networkOfferingVO));
     }
+
+    public final class MockIpAddress {
+
+        private MockIpAddress() {
+            // Utility class
+        }
+
+        public static com.cloud.network.IpAddress createMockIpAddress(long id) {
+            com.cloud.network.IpAddress ipAddress = org.mockito.Mockito.mock(com.cloud.network.IpAddress.class);
+            org.mockito.Mockito.when(ipAddress.getId()).thenReturn(id);
+            return ipAddress;
+        }
+    }
 }
