--- a/server/src/test/java/com/cloud/vm/FirstFitPlannerTest.java
+++ b/server/src/test/java/com/cloud/vm/FirstFitPlannerTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package com.cloud.vm;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.ArgumentMatchers.anyString;
@@ -104,6 +105,8 @@
 @RunWith(SpringJUnit4ClassRunner.class)
 @ContextConfiguration(loader = AnnotationConfigContextLoader.class)
 public class FirstFitPlannerTest {
+    @Mock
+    private ExcludeList avoids;
 
     @Inject
     FirstFitPlanner planner = new FirstFitPlanner();
@@ -178,7 +181,6 @@
     public void checkClusterReorderingBasedOnImplicitHostTags() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
         initializeForTest(vmProfile, plan, avoids);
 
         List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
@@ -197,7 +199,6 @@
     public void checkClusterReorderingForDeployVMWithThresholdCheckDisabled() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
         initializeForTest(vmProfile, plan, avoids);
         List<Long> clustersCrossingThreshold = initializeForClusterThresholdDisabled();
 
@@ -213,7 +214,6 @@
     public void checkClusterListBasedOnHostTag() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
         initializeForTest(vmProfile, plan, avoids);
         List<Long> matchingClusters = initializeForClusterListBasedOnHostTag(vmProfile.getServiceOffering());
 
@@ -238,7 +238,6 @@
     public void checkClusterReorderingForStartVMWithThresholdCheckDisabled() throws InsufficientServerCapacityException {
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
-        ExcludeList avoids = mock(ExcludeList.class);
         initializeForTest(vmProfile, plan, avoids);
         List<Long> clustersCrossingThreshold = initializeForClusterThresholdDisabled();
 
