--- a/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
@@ -122,10 +122,8 @@
         List<Hypervisor.HypervisorType> hypervisorTypes = List.of(Hypervisor.HypervisorType.KVM);
         List<Long> mockResults = List.of(1001L, 1002L); // Mocked result
         HostVO host = mock(HostVO.class);
-        GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
+        GenericSearchBuilder<HostVO, Long> sb = createMockGenericSearchBuilder(host, sc);
         SearchCriteria<Long> sc = mock(SearchCriteria.class);
-        when(sb.create()).thenReturn(sc);
         when(sb.and()).thenReturn(sb);
         doReturn(sb).when(hostDao).createSearchBuilder(Long.class);
         doReturn(mockResults).when(hostDao).customSearch(any(SearchCriteria.class), any());
@@ -180,10 +178,8 @@
         Long zoneId = 1L, podId = 2L, clusterId = 3L;
         List<Long> mockResults = List.of(1001L, 1002L);
         HostVO host = mock(HostVO.class);
-        GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
-        when(sb.entity()).thenReturn(host);
         SearchCriteria<Long> sc = mock(SearchCriteria.class);
-        when(sb.create()).thenReturn(sc);
+        GenericSearchBuilder<HostVO, Long> sb = createMockGenericSearchBuilder(host, sc);
         doReturn(sb).when(hostDao).createSearchBuilder(Long.class);
         doReturn(mockResults).when(hostDao).customSearch(any(SearchCriteria.class), any());
         List<Long> hostIds = hostDao.listIdsBy(type, status, resourceState, hypervisorType, zoneId, podId, clusterId);
@@ -249,4 +245,13 @@
         Mockito.verify(sc).setParameters("clusterId", clusterId);
         Mockito.verify(sc).setParameters("type", Host.Type.Routing);
     }
+
+    private static GenericSearchBuilder<HostVO, Long> createMockGenericSearchBuilder(
+            HostVO host,
+            SearchCriteria<Long> searchCriteria) {
+        GenericSearchBuilder<HostVO, Long> searchBuilder = mock(GenericSearchBuilder.class);
+        when(searchBuilder.create()).thenReturn(searchCriteria);
+        when(searchBuilder.entity()).thenReturn(host);
+        return searchBuilder;
+    }
 }
