--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -220,6 +220,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenServiceImplTest {
     @Mock
+    private DataCenterIpAddressVO dataCenterIpAddressVO;
+    @Mock
     MessageBus messageBus;
     @Mock
     ProjectDao projectDao;
@@ -416,7 +418,6 @@
         Network managementNetwork = mock(Network.class);
         TungstenAnswer addTungstenNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
-        DataCenterIpAddressVO dataCenterIpAddressVO = mock(DataCenterIpAddressVO.class);
 
         when(hostPodVO.getDescription()).thenReturn("192.168.100.100-192.168.100.200");
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(),
@@ -457,7 +458,6 @@
         Network managementNetwork = mock(Network.class);
         TungstenAnswer removeTungstenNetworkSubnetAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
-        DataCenterIpAddressVO dataCenterIpAddressVO = mock(DataCenterIpAddressVO.class);
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(),
                 eq(Networks.TrafficType.Management))).thenReturn(managementNetwork);
