--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -220,6 +220,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenServiceImplTest {
     @Mock
+    private HostVO hostVO;
+    @Mock
     MessageBus messageBus;
     @Mock
     ProjectDao projectDao;
@@ -643,7 +645,7 @@
         LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
         Network publicNetwork = mock(Network.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        HostVO hostVO = mock(HostVO.class);
+        // removed local mock; replaced with global field `hostVO`
         TungstenAnswer getTungstenLoadBalancerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateLoadBalancerServiceInstanceAnswer = MockTungstenAnswerFactory.get(true);
         Answer updateTungstenLoadbalancerStatsAnswer = mock(Answer.class);
@@ -1504,7 +1506,6 @@
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         NetworkDetailVO networkDetailVO = mock(NetworkDetailVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
-        HostVO hostVO = mock(HostVO.class);
         Answer answer = mock(Answer.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         ApiObjectBase apiObjectBase = mock(ApiObjectBase.class);
