--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
@@ -68,18 +68,14 @@
 
     @Test
     public void testCheckVmCanBeDestroyedNotInCluster() {
-        UserVm vm = Mockito.mock(UserVm.class);
-        Mockito.when(vm.getId()).thenReturn(1L);
-        Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
+        UserVm vm = createMockUserVm();
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(null);
         kubernetesServiceHelper.checkVmCanBeDestroyed(vm);
     }
 
     @Test(expected = CloudRuntimeException.class)
     public void testCheckVmCanBeDestroyedInCloudManagedCluster() {
-        UserVm vm = Mockito.mock(UserVm.class);
-        Mockito.when(vm.getId()).thenReturn(1L);
-        Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
+        UserVm vm = createMockUserVm();
         KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
         Mockito.when(map.getClusterId()).thenReturn(1L);
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(map);
@@ -91,9 +87,7 @@
 
     @Test
     public void testCheckVmCanBeDestroyedInExternalManagedCluster() {
-        UserVm vm = Mockito.mock(UserVm.class);
-        Mockito.when(vm.getId()).thenReturn(1L);
-        Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
+        UserVm vm = createMockUserVm();
         KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
         Mockito.when(map.getClusterId()).thenReturn(1L);
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(map);
@@ -400,4 +394,11 @@
         Assert.assertEquals(1, result.size());
         Assert.assertEquals(Arrays.asList(1L, 2L, 3L), result.get("CONTROL"));
     }
+
+    private static UserVm createMockUserVm() {
+        UserVm vm = Mockito.mock(UserVm.class);
+        Mockito.when(vm.getId()).thenReturn(1L);
+        Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
+        return vm;
+    }
 }
