--- a/server/src/test/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinitionTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinitionTest.java
@@ -73,6 +73,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class RouterDeploymentDefinitionTest extends RouterDeploymentDefinitionTestBase {
+    @Mock
+    private final PublicIp sourceNatIp;
 
     @Mock
     protected NetworkVO mockNw;
@@ -645,7 +647,7 @@
     @Test
     public void testFindSourceNatIPPublicNw() throws InsufficientAddressCapacityException, ConcurrentOperationException {
         // Prepare
-        final PublicIp sourceNatIp = mock(PublicIp.class);
+        // removed local mock; replaced with global field `sourceNatIp`
         when(mockIpAddrMgr.assignSourceNatIpAddressToGuestNetwork(
                 mockOwner, mockNw)).thenReturn(sourceNatIp);
         deployment.isPublicNetwork = true;
@@ -662,7 +664,6 @@
     @Test
     public void testFindSourceNatIPNonPublicNw() throws InsufficientAddressCapacityException, ConcurrentOperationException {
         // Prepare
-        final PublicIp sourceNatIp = mock(PublicIp.class);
         lenient().when(mockIpAddrMgr.assignSourceNatIpAddressToGuestNetwork(
                 mockOwner, mockNw)).thenReturn(sourceNatIp);
         deployment.isPublicNetwork = false;
