--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -617,9 +617,7 @@
     public void testCheckNetworkIpv6UpgradeForNoIpv6Vlan() {
         final long zoneId = 1L;
         Mockito.when(dataCenterGuestIpv6PrefixDao.listByDataCenterId(Mockito.anyLong())).thenReturn(List.of(Mockito.mock(DataCenterGuestIpv6PrefixVO.class)));
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
-        Mockito.when(network.getVpcId()).thenReturn(null);
+        Network network = createMockNetwork(zoneId);
         Mockito.when(ipAddressDao.listByAssociatedNetwork(Mockito.anyLong(), Mockito.anyBoolean())).thenReturn(List.of(Mockito.mock(IPAddressVO.class)));
         VlanVO vlanVO = Mockito.mock(VlanVO.class);
         Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
@@ -638,9 +636,7 @@
     public void testCheckNetworkIpv6UpgradeForNetwork() {
         final long zoneId = 1L;
         Mockito.when(dataCenterGuestIpv6PrefixDao.listByDataCenterId(Mockito.anyLong())).thenReturn(List.of(Mockito.mock(DataCenterGuestIpv6PrefixVO.class)));
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
-        Mockito.when(network.getVpcId()).thenReturn(null);
+        Network network = createMockNetwork(zoneId);
         Mockito.when(ipAddressDao.listByAssociatedNetwork(Mockito.anyLong(), Mockito.anyBoolean())).thenReturn(List.of(Mockito.mock(IPAddressVO.class)));
         VlanVO vlanVO = Mockito.mock(VlanVO.class);
         Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
@@ -801,4 +797,11 @@
         ipv6Service.removePublicIpv6PlaceholderNics(network);
         Assert.assertEquals(0, removedNics.size());
     }
+
+    private static Network createMockNetwork(long zoneId) {
+        Network network = Mockito.mock(Network.class);
+        Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
+        Mockito.when(network.getVpcId()).thenReturn(null);
+        return network;
+    }
 }
