--- a/server/src/test/java/com/cloud/network/guru/ExternalGuestNetworkGuruTest.java
+++ b/server/src/test/java/com/cloud/network/guru/ExternalGuestNetworkGuruTest.java
@@ -44,6 +44,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class ExternalGuestNetworkGuruTest {
     @Mock
+    private Network network;
+    @Mock
     NetworkModel networkModel;
     @Mock
     DataCenterDao dataCenterDao;
@@ -87,7 +89,6 @@
     public void testUpdateNicProfile() {
         NicProfile nicProfile = new NicProfile();
         DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Network network = Mockito.mock(Network.class);
         Mockito.when(dataCenterDao.findById(Mockito.anyLong())).thenReturn(zone);
         Mockito.when(networkModel.getNetworkIp4Dns(network, zone)).thenReturn(new Pair<>(ip4Dns[0], ip4Dns[1]));
         Mockito.when(networkModel.getNetworkIp6Dns(network, zone)).thenReturn(new Pair<>(ip6Dns[0], ip6Dns[1]));
@@ -102,7 +103,7 @@
     @Test
     public void testUpdateNetworkProfile() {
         DataCenterVO zone = Mockito.mock(DataCenterVO.class);
-        Network network = Mockito.mock(Network.class);
+        // removed local mock; replaced with global field `network`
         NetworkProfile profile = new NetworkProfile(network);
         Mockito.when(dataCenterDao.findById(Mockito.anyLong())).thenReturn(zone);
         Mockito.when(networkModel.getNetwork(Mockito.anyLong())).thenReturn(network);
