--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
@@ -63,6 +63,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class NsxPublicNetworkGuruTest {
+    @Mock
+    private Network network;
 
     NetworkOffering offering;
 
@@ -120,7 +122,6 @@
     @Test
     public void testDesign() {
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
-        Network network = Mockito.mock(Network.class);
         Account account = Mockito.mock(Account.class);
 
 //        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
@@ -132,7 +133,6 @@
     @Test
     public void testDesign_whenOfferingIsForGuestTraffic() {
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
-        Network network = Mockito.mock(Network.class);
         Account account = Mockito.mock(Account.class);
 
         when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
@@ -145,7 +145,6 @@
     public void testAllocate() throws InsufficientVirtualNetworkCapacityException, InsufficientAddressCapacityException {
         String publicIpVR = "10.1.12.10";
         String publicIpNSX = "10.1.13.10";
-        Network network = Mockito.mock(Network.class);
         NicProfile profile = Mockito.mock(NicProfile.class);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         IPAddressVO srcNatIpOnVR = new IPAddressVO(new Ip(publicIpVR), 2L , 0xaabbccddeeffL, 2L, true);
