--- a/plugins/storage/sharedfs/storagevm/src/test/java/org/apache/cloudstack/storage/sharedfs/lifecycle/StorageVmSharedFSLifeCycleTest.java
+++ b/plugins/storage/sharedfs/storagevm/src/test/java/org/apache/cloudstack/storage/sharedfs/lifecycle/StorageVmSharedFSLifeCycleTest.java
@@ -254,8 +254,7 @@
         when(owner.getId()).thenReturn(s_ownerId);
         when(accountMgr.getActiveAccountById(s_ownerId)).thenReturn(owner);
 
-        UserVm vm = mock(UserVm.class);
-        when(vm.getId()).thenReturn(s_vmId);
+        UserVm vm = createMockUserVm();
         when(userVmService.createAdvancedVirtualMachine(
                 any(DataCenter.class), any(ServiceOffering.class), any(VirtualMachineTemplate.class), anyList(), any(Account.class), anyString(),
                 anyString(), anyLong(), anyLong(), any(), isNull(), any(Hypervisor.HypervisorType.class), any(BaseCmd.HTTPMethod.class), anyString(),
@@ -341,8 +340,7 @@
         when(templateDao.findSystemVMReadyTemplate(eq(s_zoneId), any(Hypervisor.HypervisorType.class), eq(ResourceManager.SystemVmPreferredArchitecture.defaultValue())))
                 .thenReturn(null, template);
 
-        UserVm vm = mock(UserVm.class);
-        when(vm.getId()).thenReturn(s_vmId);
+        UserVm vm = createMockUserVm();
         when(userVmService.createAdvancedVirtualMachine(
                 any(DataCenter.class), any(ServiceOffering.class), any(VirtualMachineTemplate.class), anyList(), any(Account.class), anyString(),
                 anyString(), anyLong(), anyLong(), any(), isNull(), any(Hypervisor.HypervisorType.class), any(BaseCmd.HTTPMethod.class), anyString(),
@@ -393,4 +391,10 @@
         boolean result = lifeCycle.reDeploySharedFS(sharedFS);
         Assert.assertEquals(result, true);
     }
+
+    private static UserVm createMockUserVm() {
+        UserVm vm = mock(UserVm.class);
+        when(vm.getId()).thenReturn(s_vmId);
+        return vm;
+    }
 }
