--- a/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
+++ b/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
@@ -207,8 +207,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateApiCommandParamsInvalidAccessMethods() {
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        Mockito.when(template.getTemplateType()).thenReturn(Storage.TemplateType.VNF);
+        VirtualMachineTemplate template = createVirtualMachineTemplate(Storage.TemplateType.VNF);
         UpdateVnfTemplateCmd cmd = Mockito.mock(UpdateVnfTemplateCmd.class);
         Map<String, String> vnfDetails = Mockito.spy(new HashMap<>());
         vnfDetails.put("access_methods", "invalid");
@@ -219,8 +218,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateApiCommandParamsInvalidAccessDetails() {
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        Mockito.when(template.getTemplateType()).thenReturn(Storage.TemplateType.VNF);
+        VirtualMachineTemplate template = createVirtualMachineTemplate(Storage.TemplateType.VNF);
         UpdateVnfTemplateCmd cmd = Mockito.mock(UpdateVnfTemplateCmd.class);
         Map<String, String> vnfDetails = Mockito.spy(new HashMap<>());
         vnfDetails.put("invalid", "value");
@@ -231,8 +229,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateApiCommandParamsInvalidTemplateType() {
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        Mockito.when(template.getTemplateType()).thenReturn(Storage.TemplateType.USER);
+        VirtualMachineTemplate template = createVirtualMachineTemplate(Storage.TemplateType.USER);
         UpdateVnfTemplateCmd cmd = Mockito.mock(UpdateVnfTemplateCmd.class);
 
         VnfTemplateUtils.validateApiCommandParams(cmd, template);
@@ -258,4 +255,10 @@
         cidrList.add("10.10.10.0/33");
         VnfTemplateUtils.validateVnfCidrList(cidrList);
     }
+
+    private static VirtualMachineTemplate createVirtualMachineTemplate(Storage.TemplateType templateType) {
+        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
+        Mockito.when(template.getTemplateType()).thenReturn(templateType);
+        return template;
+    }
 }
