--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCheckAndRepairVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCheckAndRepairVolumeCommandWrapperTest.java
@@ -72,9 +72,8 @@
         when(spool.getUuid()).thenReturn("b6be258b-42b8-49a4-ad51-3634ef8ff76a");
         when(storagePoolMgr.getStoragePool(Storage.StoragePoolType.PowerFlex, "b6be258b-42b8-49a4-ad51-3634ef8ff76a")).thenReturn(pool);
 
-        KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
+        KVMPhysicalDisk vol = MockKVMPhysicalDisk.createMockKVMPhysicalDisk();
         when(pool.getPhysicalDisk("cbac516a-0f1f-4559-921c-1a7c6c408ccf")).thenReturn(vol);
-        Mockito.when(vol.getFormat()).thenReturn(QemuImg.PhysicalDiskFormat.QCOW2);
 
         String checkResult = "{\n" +
                 "    \"image-end-offset\": 6442582016,\n" +
@@ -95,4 +94,17 @@
         }
     }
 
+
+    public final class MockKVMPhysicalDisk {
+        private MockKVMPhysicalDisk() {
+        }
+
+        public static com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk createMockKVMPhysicalDisk() {
+            com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk disk =
+                    Mockito.mock(com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk.class);
+            Mockito.when(disk.getFormat())
+                    .thenReturn(org.apache.cloudstack.utils.qemu.QemuImg.PhysicalDiskFormat.QCOW2);
+            return disk;
+        }
+    }
 }
--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVolumesOnStorageCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVolumesOnStorageCommandWrapperTest.java
@@ -126,9 +126,8 @@
 
     @Test
     public void testLibvirtGetVolumesOnStorageCommandWrapperForVolume() {
-        KVMPhysicalDisk disk = Mockito.mock(KVMPhysicalDisk.class);
+        KVMPhysicalDisk disk = MockKVMPhysicalDisk.createMockKVMPhysicalDisk();
         Mockito.when(disk.getPath()).thenReturn(volumePath);
-        Mockito.when(disk.getFormat()).thenReturn(QemuImg.PhysicalDiskFormat.QCOW2);
         Mockito.when(disk.getQemuEncryptFormat()).thenReturn(QemuObject.EncryptFormat.LUKS);
         Mockito.when(storagePool.getPhysicalDisk(volumePath)).thenReturn(disk);
         Mockito.when(storagePool.getType()).thenReturn(poolType);
@@ -155,4 +154,17 @@
         Mockito.verify(volumeOnStorageTO).addDetail(VolumeOnStorageTO.Detail.IS_ENCRYPTED, "true");
         Mockito.verify(volumeOnStorageTO).addDetail(VolumeOnStorageTO.Detail.IS_LOCKED, "false");
     }
+
+    public final class MockKVMPhysicalDisk {
+        private MockKVMPhysicalDisk() {
+        }
+
+        public static com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk createMockKVMPhysicalDisk() {
+            com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk disk =
+                    Mockito.mock(com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk.class);
+            Mockito.when(disk.getFormat())
+                    .thenReturn(org.apache.cloudstack.utils.qemu.QemuImg.PhysicalDiskFormat.QCOW2);
+            return disk;
+        }
+    }
 }
