--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -2259,8 +2259,7 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        StoragePool destPrimaryStorage = Mockito.mock(StoragePool.class);
-        when(destPrimaryStorage.getPoolType()).thenReturn(Storage.StoragePoolType.PowerFlex);
+        StoragePool destPrimaryStorage = createMockStoragePool(Storage.StoragePoolType.PowerFlex);
         Mockito.doReturn(destPrimaryStorage).when(volumeApiServiceImpl)
                 .getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm);
         volumeApiServiceImpl.createVolumeOnPrimaryForAttachIfNeeded(volumeToAttach, vm, null);
@@ -2271,8 +2270,7 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         Mockito.when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        StoragePool destPrimaryStorage = Mockito.mock(StoragePool.class);
-        Mockito.when(destPrimaryStorage.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
+        StoragePool destPrimaryStorage = createMockStoragePool(Storage.StoragePoolType.NetworkFilesystem);
         Mockito.doReturn(destPrimaryStorage).when(volumeApiServiceImpl)
                 .getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm);
         try {
@@ -2896,4 +2894,10 @@
                     e.getMessage() != null && e.getMessage().contains("VM should be in"));
         }
     }
+
+    private static StoragePool createMockStoragePool(Storage.StoragePoolType poolType) {
+        StoragePool storagePool = Mockito.mock(StoragePool.class);
+        when(storagePool.getPoolType()).thenReturn(poolType);
+        return storagePool;
+    }
 }
