--- a/services/secondary-storage/controller/src/test/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerTest.java
+++ b/services/secondary-storage/controller/src/test/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerTest.java
@@ -52,6 +52,8 @@
 
 public class SecondaryStorageManagerTest {
     @Mock
+    private NetworkVO network;
+    @Mock
     DataCenterDao _dcDao;
 
     @Mock
@@ -84,7 +86,6 @@
 
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Public)))
                     .thenReturn(Collections.singletonList(network));
@@ -111,7 +112,6 @@
 
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
@@ -133,7 +133,7 @@
 
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
+        // removed local mock; replaced with global field `network`
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
@@ -156,7 +156,6 @@
 
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
@@ -180,7 +179,6 @@
 
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
@@ -199,7 +197,6 @@
 
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
