--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -577,9 +577,9 @@
     @Test
     public void getStoragePoolNonDestroyedVolumesLogTestNonDestroyedVolumes_VMAttachedLogs() {
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
-        Mockito.doReturn(1L).when(volume1VOMock).getInstanceId();
+        volume1VOMock = createMockVolumeVO();
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO();
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
         Mockito.doReturn("58760044-928f-4c4e-9fef-d0e48423595e").when(vmInstanceVOMock).getUuid();
 
@@ -597,7 +597,7 @@
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
         Mockito.doReturn(null).when(volume1VOMock).getInstanceId();
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO();
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
         Mockito.doReturn("58760044-928f-4c4e-9fef-d0e48423595e").when(vmInstanceVOMock).getUuid();
 
@@ -629,9 +629,9 @@
     @Test
     public void getStoragePoolNonDestroyedVolumesLogTestNonDestroyedVolumes_VMNotExistsLog() {
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
-        Mockito.doReturn(1L).when(volume1VOMock).getInstanceId();
+        volume1VOMock = createMockVolumeVO();
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO();
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
 
         Mockito.when(_volumeDao.findNonDestroyedVolumesByPoolId(storagePoolVOMock.getId(), null)).thenReturn(List.of(volume1VOMock, volume2VOMock));
@@ -1727,4 +1727,10 @@
 
         Assert.assertFalse(result);
     }
+
+    private static VolumeVO createMockVolumeVO() {
+        VolumeVO volumeVOMock = Mockito.mock(VolumeVO.class);
+        Mockito.doReturn(1L).when(volumeVOMock).getInstanceId();
+        return volumeVOMock;
+    }
 }
