--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -110,6 +110,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ApiResponseHelperTest {
+    @Mock
+    private Capacity c4;
 
     @Mock
     UsageService usageService;
@@ -541,7 +543,6 @@
         Mockito.when(c2.getTag()).thenReturn("tag1");
         Capacity c3 = Mockito.mock(Capacity.class);
         Mockito.when(c3.getTag()).thenReturn("tag2");
-        Capacity c4 = Mockito.mock(Capacity.class);
         Assert.assertTrue(apiResponseHelper.capacityListingForSingleTag(List.of(c1, c2)));
         Assert.assertFalse(apiResponseHelper.capacityListingForSingleTag(List.of(c1, c2, c3)));
         Assert.assertFalse(apiResponseHelper.capacityListingForSingleTag(List.of(c4, c2, c3)));
@@ -555,7 +556,6 @@
         Mockito.when(c2.getCapacityType()).thenReturn((short)Resource.ResourceType.user_vm.getOrdinal());
         Capacity c3 = Mockito.mock(Capacity.class);
         Mockito.when(c3.getCapacityType()).thenReturn((short)Resource.ResourceType.volume.getOrdinal());
-        Capacity c4 = Mockito.mock(Capacity.class);
         Assert.assertTrue(apiResponseHelper.capacityListingForSingleNonGpuType(List.of(c1, c2)));
         Assert.assertFalse(apiResponseHelper.capacityListingForSingleNonGpuType(List.of(c1, c2, c3)));
     }
