--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -81,6 +81,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ClusterDrsServiceImplTest {
+    @Mock
+    private ServiceOffering serviceOffering;
 
     @Mock
     ClusterDrsAlgorithm condensedAlgorithm;
@@ -817,7 +819,6 @@
 
         Map<Long, ServiceOffering> vmIdServiceOfferingMap = new HashMap<>();
 
-        ServiceOffering serviceOffering = Mockito.mock(ServiceOffering.class);
         for (VirtualMachine vm : vmList) {
             vmIdServiceOfferingMap.put(vm.getId(), serviceOffering);
         }
@@ -890,7 +891,6 @@
 
         Map<Long, ServiceOffering> vmIdServiceOfferingMap = new HashMap<>();
 
-        ServiceOffering serviceOffering = Mockito.mock(ServiceOffering.class);
         for (VirtualMachine vm : vmList) {
             vmIdServiceOfferingMap.put(vm.getId(), serviceOffering);
         }
