--- a/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
+++ b/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
@@ -56,8 +56,7 @@
 
     @Test
     public void testFinalizeStartWhenCmdsAnswerIsNotNullButAnswerResultIsFalse() throws Exception {
-        CheckSshAnswer answerMock = mock(CheckSshAnswer.class);
-        when(answerMock.getResult()).thenReturn(false);
+        CheckSshAnswer answerMock = createCheckSshAnswer(false);
         VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
@@ -72,8 +71,7 @@
 
     @Test
     public void testFinalizeStartWhenCmdsAnswerIsNotNullAndAnswerResultIsTrue() throws Exception {
-        CheckSshAnswer answerMock = mock(CheckSshAnswer.class);
-        when(answerMock.getResult()).thenReturn(true);
+        CheckSshAnswer answerMock = createCheckSshAnswer(true);
         VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
         Commands cmds = mock(Commands.class);
@@ -98,4 +96,10 @@
             fail();
         }
     }
+
+    private static CheckSshAnswer createCheckSshAnswer(boolean result) {
+        CheckSshAnswer answerMock = mock(CheckSshAnswer.class);
+        when(answerMock.getResult()).thenReturn(result);
+        return answerMock;
+    }
 }
