--- a/server/src/test/java/com/cloud/consoleproxy/ConsoleProxyManagerTest.java
+++ b/server/src/test/java/com/cloud/consoleproxy/ConsoleProxyManagerTest.java
@@ -56,6 +56,8 @@
 import static org.mockito.Mockito.when;
 
 public class ConsoleProxyManagerTest {
+    @Mock
+    private NetworkVO network;
 
     protected Logger logger = LogManager.getLogger(getClass());
 
@@ -142,7 +144,6 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Public)))
                     .thenReturn(Collections.singletonList(network));
@@ -168,7 +169,6 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
@@ -190,7 +190,6 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
@@ -212,7 +211,6 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
@@ -236,7 +234,6 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
@@ -255,7 +252,6 @@
 
         when(dataCenterDaoMock.findById(Mockito.anyLong())).thenReturn(dc);
 
-        NetworkVO network = Mockito.mock(NetworkVO.class);
         NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(networkDaoMock.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
