--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -340,7 +340,7 @@
         User caller = mock(User.class);
         Network network = mock(Network.class);
         Network publicNetwork = mock(Network.class);
-        Ip ip = mock(Ip.class);
+        Ip ip = MockIp.createMockIp("10.10.10.10");
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
@@ -378,7 +378,6 @@
         when(ipAddressVO.getAddress()).thenReturn(ip);
         when(lbVmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(loadBalancerVMMapVOList);
         when(tungstenGuestNetworkIpAddressVO.getGuestIpAddress()).thenReturn(ip);
-        when(ip.addr()).thenReturn("10.10.10.10");
         when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(), anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkLoadbalancerCommand.class), anyLong())).thenReturn(createTungstenNetworkLoadbalancerAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(UpdateTungstenLoadBalancerPoolCommand.class), anyLong())).thenReturn(updateTungstenLoadBalancerPoolAnswer);
@@ -396,7 +395,7 @@
     public void applyLBRulesAddRuleFailTest() {
         Network network = mock(Network.class);
         Network publicNetwork = mock(Network.class);
-        Ip ip = mock(Ip.class);
+        Ip ip = MockIp.createMockIp("10.10.10.10");
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
@@ -427,7 +426,6 @@
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(ipAddressVO.getAddress()).thenReturn(ip);
         when(lbVmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(loadBalancerVMMapVOList);
-        when(ip.addr()).thenReturn("10.10.10.10");
         when(ipAddressMgr.acquireGuestIpAddress(any(), any())).thenReturn("192.168.100.100");
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkLoadbalancerCommand.class), anyLong())).thenReturn(createTungstenNetworkLoadbalancerAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(UpdateTungstenLoadBalancerPoolCommand.class), anyLong())).thenReturn(updateTungstenLoadBalancerPoolAnswer);
@@ -443,7 +441,6 @@
     public void applyLBRulesRevokeRuleFailTest() {
         Network network = mock(Network.class);
         Network publicNetwork = mock(Network.class);
-        Ip ip1 = mock(Ip.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVO loadBalancerVO1 = mock(LoadBalancerVO.class);
@@ -457,8 +454,8 @@
         when(loadBalancingRule1.getSourceIp()).thenReturn(ip1);
         when(loadBalancingRule1.getState()).thenReturn(FirewallRule.State.Revoke);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
+        Ip ip1 = MockIp.createMockIp("10.10.10.10");
         when(ipAddressVO.getAddress()).thenReturn(ip1);
-        when(ip1.addr()).thenReturn("10.10.10.10");
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenLoadBalancerCommand.class), anyLong())).thenReturn(deleteTungstenLoadBalancerCommand);
         when(lbDao.listByIpAddress(anyLong())).thenReturn(loadBalancerVOList1);
         when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(),anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
@@ -471,7 +468,7 @@
     public void applyLBRulesRevokeRuleSuccessTest() {
         Network network = mock(Network.class);
         Network publicNetwork = mock(Network.class);
-        Ip ip = mock(Ip.class);
+        Ip ip = MockIp.createMockIp("10.10.10.10");
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
         LoadBalancerVO loadBalancerVO1 = mock(LoadBalancerVO.class);
@@ -486,7 +483,6 @@
         when(loadBalancingRule.getSourceIp()).thenReturn(ip);
         when(loadBalancingRule.getState()).thenReturn(FirewallRule.State.Revoke);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
-        when(ip.addr()).thenReturn("10.10.10.10");
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenLoadBalancerListenerCommand.class), anyLong())).thenReturn(deleteTungstenLoadBalancerListenerAnswer);
         when(tungstenService.updateLoadBalancer(any(), any())).thenReturn(true);
         when(lbDao.listByIpAddress(anyLong())).thenReturn(loadBalancerVOList);
@@ -962,4 +958,15 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    public final class MockIp {
+        private MockIp() {
+        }
+
+        public static com.cloud.utils.net.Ip createMockIp(String addrReturn) {
+            com.cloud.utils.net.Ip ip = Mockito.mock(com.cloud.utils.net.Ip.class);
+            when(ip.addr()).thenReturn(addrReturn);
+            return ip;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -652,7 +652,7 @@
         LoadBalancerCertMapVO loadBalancerCertMapVO = mock(LoadBalancerCertMapVO.class);
         SslCertVO sslCertVO = mock(SslCertVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
-        Ip ip = mock(Ip.class);
+        Ip ip = MockIp.createMockIp("192.168.100.100");
         AccountVO accountVO = mock(AccountVO.class);
         Loadbalancer loadbalancer = mock(Loadbalancer.class);
 
@@ -672,7 +672,6 @@
         when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(), anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(loadBalancingRule.getSourceIp()).thenReturn(ip);
         when(accountDao.findById(anyLong())).thenReturn(accountVO);
-        when(ip.addr()).thenReturn("192.168.100.100");
         when(getTungstenLoadBalancerAnswer.getApiObjectBase()).thenReturn(loadbalancer);
         when(ipAddressVO.getAddress()).thenReturn(ip);
         when(tungstenGuestNetworkIpAddressVO.getGuestIpAddress()).thenReturn(ip);
@@ -1668,4 +1667,15 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    public final class MockIp {
+        private MockIp() {
+        }
+
+        public static com.cloud.utils.net.Ip createMockIp(String addrReturn) {
+            com.cloud.utils.net.Ip ip = Mockito.mock(com.cloud.utils.net.Ip.class);
+            when(ip.addr()).thenReturn(addrReturn);
+            return ip;
+        }
+    }
 }
