--- a/api/src/test/java/org/apache/cloudstack/api/command/test/CreateAutoScaleVmProfileCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/CreateAutoScaleVmProfileCmdTest.java
@@ -96,7 +96,6 @@
         ReflectionTestUtils.setField(createAutoScaleVmProfileCmd,"projectId", projectId);
         ReflectionTestUtils.setField(createAutoScaleVmProfileCmd,"accountName", accountName);
 
-        autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
 
         response = new AutoScaleVmProfileResponse();
         response.setUserData(userData);
@@ -141,8 +140,7 @@
 
     @Test
     public void testCreateAutoScaleVmProfileSuccess() {
-        when(autoScaleVmProfile.getId()).thenReturn(1L);
-        when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+        autoScaleVmProfile = MockAutoScaleVmProfile.createMockAutoScaleVmProfile();
         when(autoScaleService.createAutoScaleVmProfile(createAutoScaleVmProfileCmd)).thenReturn(autoScaleVmProfile);
 
         try {
@@ -173,4 +171,17 @@
             Assert.fail("Got exception: " + ex.getMessage());
         }
     }
+
+    public final class MockAutoScaleVmProfile {
+
+        private MockAutoScaleVmProfile() {
+        }
+
+        public static AutoScaleVmProfile createMockAutoScaleVmProfile() {
+            AutoScaleVmProfile autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
+            when(autoScaleVmProfile.getId()).thenReturn(1L);
+            when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+            return autoScaleVmProfile;
+        }
+    }
 }
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/UpdateAutoScaleVmProfileCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/UpdateAutoScaleVmProfileCmdTest.java
@@ -78,7 +78,6 @@
         ReflectionTestUtils.setField(updateAutoScaleVmProfileCmd,"otherDeployParams", otherDeployParams);
         ReflectionTestUtils.setField(updateAutoScaleVmProfileCmd,"counterParamList", counterParamList);
 
-        autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
 
         response = new AutoScaleVmProfileResponse();
         response.setUserData(userData);
@@ -105,8 +104,7 @@
 
     @Test
     public void testUpdateAutoScaleVmProfileSuccess() {
-        when(autoScaleVmProfile.getId()).thenReturn(1L);
-        when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+        autoScaleVmProfile = MockAutoScaleVmProfile.createMockAutoScaleVmProfile();
         when(autoScaleService.updateAutoScaleVmProfile(updateAutoScaleVmProfileCmd)).thenReturn(autoScaleVmProfile);
 
         when(entityMgr.findById(AutoScaleVmProfile.class, 1L)).thenReturn(autoScaleVmProfile);
@@ -126,4 +124,17 @@
 
         updateAutoScaleVmProfileCmd.execute();
     }
+
+    public final class MockAutoScaleVmProfile {
+
+        private MockAutoScaleVmProfile() {
+        }
+
+        public static AutoScaleVmProfile createMockAutoScaleVmProfile() {
+            AutoScaleVmProfile autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
+            when(autoScaleVmProfile.getId()).thenReturn(1L);
+            when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+            return autoScaleVmProfile;
+        }
+    }
 }
