--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -258,8 +258,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -317,8 +316,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -371,8 +369,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -423,8 +420,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -588,4 +584,15 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount() {
+            final com.cloud.user.Account acc = org.mockito.Mockito.mock(com.cloud.user.Account.class);
+            org.mockito.Mockito.when(acc.getAccountName()).thenReturn("accountname");
+            return acc;
+        }
+    }
 }
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -281,8 +281,7 @@
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -337,8 +336,7 @@
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -393,8 +391,7 @@
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -450,8 +447,7 @@
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount();
         final ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -469,4 +465,15 @@
         verify(agentmgr, times(1)).easySend(eq(NETWORK_ID), (Command)any());
         verify(implementednetwork, times(1)).setBroadcastUri(null);
     }
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount() {
+            final com.cloud.user.Account acc = org.mockito.Mockito.mock(com.cloud.user.Account.class);
+            org.mockito.Mockito.when(acc.getAccountName()).thenReturn("accountname");
+            return acc;
+        }
+    }
 }
--- a/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/VxlanGuestNetworkGuruTest.java
+++ b/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/VxlanGuestNetworkGuruTest.java
@@ -174,9 +174,8 @@
         Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
 
-        Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
-
+
+        Account acc = MockAccount.createMockAccount();
         ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -219,9 +218,8 @@
         Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
 
-        Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
-
+
+        Account acc = MockAccount.createMockAccount();
         ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -262,9 +260,8 @@
         Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
 
-        Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
-
+
+        Account acc = MockAccount.createMockAccount();
         ReservationContext res = mock(ReservationContext.class);
         when(res.getDomain()).thenReturn(dom);
         when(res.getAccount()).thenReturn(acc);
@@ -277,4 +274,15 @@
         guru.shutdown(implementednetwork, offering);
         verify(implementednetwork, times(1)).setBroadcastUri(null);
     }
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount() {
+            final com.cloud.user.Account acc = org.mockito.Mockito.mock(com.cloud.user.Account.class);
+            org.mockito.Mockito.when(acc.getAccountName()).thenReturn("accountname");
+            return acc;
+        }
+    }
 }
