--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -146,6 +146,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class CitrixRequestWrapperTest {
+    @Mock
+    private final NicTO nicTO;
 
     @Mock
     private CitrixResourceBase citrixResourceBase;
@@ -1471,7 +1473,7 @@
 
     @Test
     public void testPlugNicCommand() {
-        final NicTO nicTO = Mockito.mock(NicTO.class);
+        // removed local mock; replaced with global field `nicTO`
         final Connection conn = Mockito.mock(Connection.class);
 
         final PlugNicCommand plugNic = new PlugNicCommand(nicTO, "Test", VirtualMachine.Type.User);
@@ -1490,7 +1492,6 @@
 
     @Test
     public void testUnPlugNicCommand() {
-        final NicTO nicTO = Mockito.mock(NicTO.class);
         final Connection conn = Mockito.mock(Connection.class);
 
         final UnPlugNicCommand unplugNic = new UnPlugNicCommand(nicTO, "Test");
