--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -551,7 +551,6 @@
         final String uuid = "befc4dcd-f5c6-4015-8791-3c18622b7c7f";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final MaintainCommand maintainCommand = new MaintainCommand();
 
@@ -564,8 +563,8 @@
         assertNotNull(wrapper);
 
         when(citrixResourceBase.getConnection()).thenReturn(conn);
+        final XsHost xsHost = MockXsHost.createMockXsHost(uuid);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
-        when(xsHost.getUuid()).thenReturn(uuid);
         when(conn.getSessionReference()).thenReturn("befc4dcd");
 
         try {
@@ -1929,4 +1928,17 @@
         return false;
     }
 
+
+    public final class MockXsHost {
+
+        private MockXsHost() {
+            // Utility class
+        }
+
+        public static XsHost createMockXsHost(final String uuid) {
+            final XsHost xsHost = Mockito.mock(XsHost.class);
+            when(xsHost.getUuid()).thenReturn(uuid);
+            return xsHost;
+        }
+    }
 }
--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -123,7 +123,6 @@
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network networkForSm = Mockito.mock(Network.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final MigrateWithStorageCommand migrateStorageCommand = new MigrateWithStorageCommand(vmSpec, volumeToFiler);
 
@@ -140,8 +139,8 @@
             when(xenServer610Resource.getNativeNetworkForTraffic(conn, TrafficType.Storage, null)).thenReturn(nativeNetworkForTraffic);
             when(nativeNetworkForTraffic.getNetwork()).thenReturn(networkForSm);
 
+            final XsHost xsHost = MockXsHost.createMockXsHost(uuid);
             when(xenServer610Resource.getHost()).thenReturn(xsHost);
-            when(xsHost.getUuid()).thenReturn(uuid);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
@@ -196,7 +195,6 @@
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network network = Mockito.mock(Network.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final Network nw1 = Mockito.mock(Network.class);
         final Network nw2 = Mockito.mock(Network.class);
@@ -219,8 +217,8 @@
             when(xenServer610Resource.getNativeNetworkForTraffic(conn, TrafficType.Storage, null)).thenReturn(nativeNetworkForTraffic);
             when(nativeNetworkForTraffic.getNetwork()).thenReturn(network);
 
+            final XsHost xsHost = MockXsHost.createMockXsHost(uuid);
             when(xenServer610Resource.getHost()).thenReturn(xsHost);
-            when(xsHost.getUuid()).thenReturn(uuid);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
@@ -412,7 +410,6 @@
         final String uuid = "206b21a7-c6ec-40e2-b5e2-f861b9612f04";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
 
@@ -423,8 +420,8 @@
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
         when(vm.getName()).thenReturn(vmName);
+        final XsHost xsHost = MockXsHost.createMockXsHost(uuid);
         when(xenServer610Resource.getHost()).thenReturn(xsHost);
-        when(xsHost.getUuid()).thenReturn(uuid);
 
         final Answer answer = wrapper.execute(createStorageCommand, xenServer610Resource);
 
@@ -470,4 +467,17 @@
         when(vol.getPath()).thenReturn(path);
         return vol;
     }
+
+    public final class MockXsHost {
+
+        private MockXsHost() {
+            // Utility class
+        }
+
+        public static XsHost createMockXsHost(final String uuid) {
+            final XsHost xsHost = Mockito.mock(XsHost.class);
+            when(xsHost.getUuid()).thenReturn(uuid);
+            return xsHost;
+        }
+    }
 }
