--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -155,6 +155,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VolumeApiServiceImplTest {
+    @Mock
+    private UserVmVO vm;
 
     @Spy
     @InjectMocks
@@ -2214,7 +2216,6 @@
     public void testCreateVolumeOnSecondaryForAttachIfNeeded_ExistingVolumeDeterminesStoragePool() {
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         Mockito.when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         VolumeVO existingVolume = Mockito.mock(VolumeVO.class);
         Mockito.when(existingVolume.getState()).thenReturn(Volume.State.Ready);
         when(existingVolume.getPoolId()).thenReturn(1L);
@@ -2237,7 +2238,6 @@
     public void testCreateVolumeOnPrimaryForAttachIfNeeded_UsesGetPoolForAttach() {
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         Mockito.when(volumeToAttach.getState()).thenReturn(Volume.State.Allocated);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         StoragePool destPrimaryStorage = Mockito.mock(StoragePool.class);
         Mockito.doReturn(destPrimaryStorage).when(volumeApiServiceImpl)
                 .getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm);
@@ -2258,7 +2258,6 @@
     public void testCreateVolumeOnPrimaryForAttachIfNeeded_UnsupportedPoolType_ThrowsException() {
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         StoragePool destPrimaryStorage = Mockito.mock(StoragePool.class);
         when(destPrimaryStorage.getPoolType()).thenReturn(Storage.StoragePoolType.PowerFlex);
         Mockito.doReturn(destPrimaryStorage).when(volumeApiServiceImpl)
@@ -2270,7 +2269,6 @@
     public void testCreateVolumeOnSecondaryForAttachIfNeeded_CreateVolumeFails_ThrowsException() {
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         Mockito.when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         StoragePool destPrimaryStorage = Mockito.mock(StoragePool.class);
         Mockito.when(destPrimaryStorage.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
         Mockito.doReturn(destPrimaryStorage).when(volumeApiServiceImpl)
@@ -2291,7 +2289,6 @@
     public void testCreateVolumeOnSecondaryForAttachIfNeeded_NoSuitablePool_ThrowsException() {
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         Mockito.when(volumeToAttach.getState()).thenReturn(Volume.State.Uploaded);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         Mockito.doReturn(null).when(volumeApiServiceImpl)
                 .getSuitablePoolForAllocatedOrUploadedVolumeForAttach(volumeToAttach, vm);
         CloudRuntimeException exception = Assert.assertThrows(CloudRuntimeException.class, () ->
@@ -2433,7 +2430,6 @@
     public void testIsClvmLightweightMigrationNeeded_SameVG() {
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
@@ -2467,7 +2463,6 @@
     public void testIsClvmLightweightMigrationNeeded_DifferentVG() {
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
@@ -2499,7 +2494,6 @@
     public void testIsClvmLightweightMigrationNeeded_CLVM_NG_SameVG() {
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
         StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
