--- a/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
+++ b/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
@@ -223,9 +223,8 @@
     public void testCreateAttachmentRetryOnce() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        AttachmentData attachmentData = mock(AttachmentData.class);
         AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
-        when(attachmentData.getAttachment()).thenReturn(attachment);
+        AttachmentData attachmentData = createAttachmentData(attachment);
         when(attachment.getId()).thenReturn("eeee");
         when(_bigswitchBcfApi.createAttachment((String)any(), (String)any(), (AttachmentData)any())).thenThrow(new BigSwitchBcfApiException())
         .thenReturn(UUID.randomUUID().toString());
@@ -240,9 +239,8 @@
     public void testCreateAttachmentApiException() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        AttachmentData attachmentData = mock(AttachmentData.class);
         AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
-        when(attachmentData.getAttachment()).thenReturn(attachment);
+        AttachmentData attachmentData = createAttachmentData(attachment);
         when(attachment.getId()).thenReturn("eeee");
         doThrow(new BigSwitchBcfApiException()).when(_bigswitchBcfApi).createAttachment((String)any(), (String)any(), (AttachmentData)any());
 
@@ -428,4 +426,10 @@
         assertTrue(ans.getResult());
     }
 
+
+    private static AttachmentData createAttachmentData(AttachmentData.Attachment attachment) {
+        AttachmentData attachmentData = mock(AttachmentData.class);
+        when(attachmentData.getAttachment()).thenReturn(attachment);
+        return attachmentData;
+    }
 }
