--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -130,6 +130,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenElementTest {
     @Mock
+    private Nic nic;
+    @Mock
     TungstenFabricUtils tungstenFabricUtils;
     @Mock
     NetworkModel networkModel;
@@ -258,7 +260,6 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         TungstenAnswer assignFloatingIpAnswer = MockTungstenAnswerFactory.get(true);
-        Nic nic = mock(Nic.class);
         Network publicNetwork = mock(Network.class);
         List<StaticNatImpl> staticNatList = List.of(staticNat);
 
@@ -279,7 +280,6 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         TungstenAnswer assignFloatingIpAnswer = MockTungstenAnswerFactory.get(false);
-        Nic nic = mock(Nic.class);
         Network publicNetwork = mock(Network.class);
         List<StaticNatImpl> staticNatList = List.of(staticNat);
 
@@ -300,7 +300,6 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         TungstenAnswer releaseFloatingIpAnswer = MockTungstenAnswerFactory.get(true);
-        Nic nic = mock(Nic.class);
         Network publicNetwork = mock(Network.class);
         List<StaticNatImpl> staticNatList = List.of(staticNat);
 
@@ -321,7 +320,6 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         TungstenAnswer releaseFloatingIpAnswer = MockTungstenAnswerFactory.get(false);
-        Nic nic = mock(Nic.class);
         Network publicNetwork = mock(Network.class);
         List<StaticNatImpl> staticNatList = List.of(staticNat);
 
@@ -501,7 +499,6 @@
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         UserVm userVm = mock(UserVm.class);
-        Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(true);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
 
@@ -522,7 +519,6 @@
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         UserVm userVm = mock(UserVm.class);
-        Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(false);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
 
@@ -543,7 +539,6 @@
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         UserVm userVm = mock(UserVm.class);
-        Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(true);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
 
@@ -564,7 +559,6 @@
         PortForwardingRule portForwardingRule = mock(PortForwardingRule.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         UserVm userVm = mock(UserVm.class);
-        Nic nic = mock(Nic.class);
         TungstenAnswer applyTungstenPortForwardingAnswer = MockTungstenAnswerFactory.get(false);
         List<PortForwardingRule> portForwardingRuleList = List.of(portForwardingRule);
 
