--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1047,10 +1047,9 @@
 
     @Test
     public void validateCurrentPasswordTestUserAuthenticatedWithProvidedCurrentPasswordViaFirstAuthenticator() {
-        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
         long domainId = 14L;
-        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
-
+
+        AccountVO accountVoMock = createMockAccountVO(domainId);
         Mockito.doReturn(accountVoMock).when(_accountDao).findById(accountMockId);
         String username = "username";
         Mockito.doReturn(username).when(userVoMock).getUsername();
@@ -1076,10 +1075,9 @@
 
     @Test
     public void validateCurrentPasswordTestUserAuthenticatedWithProvidedCurrentPasswordViaSecondAuthenticator() {
-        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
         long domainId = 14L;
-        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
-
+
+        AccountVO accountVoMock = createMockAccountVO(domainId);
         Mockito.doReturn(accountVoMock).when(_accountDao).findById(accountMockId);
         String username = "username";
         Mockito.doReturn(username).when(userVoMock).getUsername();
@@ -2289,4 +2287,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static AccountVO createMockAccountVO(long domainId) {
+        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
+        return accountVoMock;
+    }
 }
