--- a/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
+++ b/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
@@ -125,8 +125,7 @@
     public void testPingCommandStatusOk() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        ControlClusterStatus ccs = mock(ControlClusterStatus.class);
-        when(ccs.getStatus()).thenReturn(true);
+        ControlClusterStatus ccs = createMockControlClusterStatus(true);
         when(_bigswitchBcfApi.getControlClusterStatus()).thenReturn(ccs);
 
         Capabilities cap = mock(Capabilities.class);
@@ -142,8 +141,7 @@
     public void testPingCommandStatusFail() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        ControlClusterStatus ccs = mock(ControlClusterStatus.class);
-        when(ccs.getStatus()).thenReturn(false);
+        ControlClusterStatus ccs = createMockControlClusterStatus(false);
         when(_bigswitchBcfApi.getControlClusterStatus()).thenReturn(ccs);
 
         PingCommand ping = _resource.getCurrentStatus(42);
@@ -154,8 +152,7 @@
     public void testPingCommandStatusApiException() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        ControlClusterStatus ccs = mock(ControlClusterStatus.class);
-        when(ccs.getStatus()).thenReturn(false);
+        ControlClusterStatus ccs = createMockControlClusterStatus(false);
         when(_bigswitchBcfApi.getControlClusterStatus()).thenThrow(new BigSwitchBcfApiException());
 
         PingCommand ping = _resource.getCurrentStatus(42);
@@ -428,4 +425,10 @@
         assertTrue(ans.getResult());
     }
 
+
+    private static ControlClusterStatus createMockControlClusterStatus(boolean status) {
+        ControlClusterStatus ccs = mock(ControlClusterStatus.class);
+        when(ccs.getStatus()).thenReturn(status);
+        return ccs;
+    }
 }
