--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -180,6 +180,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class UnmanagedVMsManagerImplTest {
+    @Mock
+    private HostVO host;
 
     @Spy
     @InjectMocks
@@ -662,7 +664,6 @@
         when(cmd.getPassword()).thenReturn("pass");
         when(cmd.getImportSource()).thenReturn("external");
         when(cmd.getDomainId()).thenReturn(null);
-        HostVO host = Mockito.mock(HostVO.class);
         DeployDestination mockDest = Mockito.mock(DeployDestination.class);
         when(deploymentPlanningManager.planDeployment(any(), any(), any(), any())).thenReturn(mockDest);
         DiskProfile diskProfile = Mockito.mock(DiskProfile.class);
@@ -1165,7 +1166,6 @@
     @Test
     public void testSelectKVMHostForImportingInClusterWithNullImportInstanceIdSuccess() {
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
         when(hostDao.listByClusterAndHypervisorType(cluster.getId(), cluster.getHypervisorType())).thenReturn(List.of(host));
 
         HostVO returnedHost = unmanagedVMsManager.selectKVMHostForImportingInCluster(cluster, null);
@@ -1279,7 +1279,6 @@
     @Test
     public void testSelectKVMHostForConversionInClusterWithImportInstanceIdSuccessCompatible() {
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
 
         when(hostDao.listByClusterHypervisorTypeAndHostCapability(cluster.getId(),
                 cluster.getHypervisorType(), Host.HOST_INSTANCE_CONVERSION)).thenReturn(List.of(host));
@@ -1291,7 +1290,6 @@
     @Test
     public void testSelectKVMHostForConversionInClusterWithImportInstanceIdSuccessNonCompatible() {
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
 
         when(hostDao.listByClusterHypervisorTypeAndHostCapability(cluster.getId(),
                 cluster.getHypervisorType(), Host.HOST_INSTANCE_CONVERSION)).thenReturn(List.of());
