--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -77,6 +77,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ScaleIOPrimaryDataStoreDriverTest {
+    @Mock
+    private DataTO destDataTO;
 
     @Spy
     @InjectMocks
@@ -518,7 +520,6 @@
         DataTO srcDataTO = Mockito.mock(DataTO.class);
         when(srcData.getTO()).thenReturn(srcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
-        DataTO destDataTO = Mockito.mock(DataTO.class);
         when(destData.getTO()).thenReturn(destDataTO);
         Host destHost = Mockito.mock(Host.class);
 
@@ -541,7 +542,6 @@
         DataTO srcDataTO = Mockito.mock(DataTO.class);
         when(srcData.getTO()).thenReturn(srcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
-        DataTO destDataTO = Mockito.mock(DataTO.class);
         when(destData.getTO()).thenReturn(destDataTO);
         Host destHost = Mockito.mock(Host.class);
 
