--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -180,6 +180,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class UnmanagedVMsManagerImplTest {
+    @Mock
+    private UserVmVO userVmVO;
 
     @Spy
     @InjectMocks
@@ -584,7 +586,6 @@
         when(virtualMachine.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(virtualMachine.getType()).thenReturn(VirtualMachine.Type.User);
         when(virtualMachine.getState()).thenReturn(VirtualMachine.State.Stopped);
-        UserVmVO userVmVO = mock(UserVmVO.class);
         when(userVmDao.findById(anyLong())).thenReturn(userVmVO);
         Mockito.doNothing().when(unmanagedVMsManager).performUnmanageVMInstancePrechecks(any());
         unmanagedVMsManager.unmanageVMInstance(virtualMachineId, null, false);
@@ -596,7 +597,7 @@
         when(virtualMachine.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(virtualMachine.getState()).thenReturn(VirtualMachine.State.Stopped);
         when(virtualMachine.getId()).thenReturn(virtualMachineId);
-        UserVmVO userVmVO = mock(UserVmVO.class);
+        // removed local mock; replaced with global field `userVmVO`
         when(userVmDao.findById(anyLong())).thenReturn(userVmVO);
         when(vmDao.findById(virtualMachineId)).thenReturn(virtualMachine);
         VolumeVO volumeVO = mock(VolumeVO.class);
