--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -535,9 +535,8 @@
             when(cmd.getDomainId()).thenReturn(1L);
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getId()).thenReturn(1L);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
-
-            Mockito.when(userData.getId()).thenReturn(1L);
+
+            UserDataVO userData = createMockUserDataVO();
             when(userDataDao.findById(1L)).thenReturn(userData);
             when(templateDao.findTemplatesLinkedToUserdata(1L)).thenReturn(new ArrayList<VMTemplateVO>());
             when(userVmDao.findByUserDataId(1L)).thenReturn(new ArrayList<UserVmVO>());
@@ -562,8 +561,7 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getId()).thenReturn(1L);
 
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
-            Mockito.when(userData.getId()).thenReturn(1L);
+            UserDataVO userData = createMockUserDataVO();
             when(userDataDao.findById(1L)).thenReturn(userData);
 
             VMTemplateVO vmTemplateVO = Mockito.mock(VMTemplateVO.class);
@@ -589,8 +587,7 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getId()).thenReturn(1L);
 
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
-            Mockito.when(userData.getId()).thenReturn(1L);
+            UserDataVO userData = createMockUserDataVO();
             when(userDataDao.findById(1L)).thenReturn(userData);
 
             when(templateDao.findTemplatesLinkedToUserdata(1L)).thenReturn(new ArrayList<VMTemplateVO>());
@@ -2493,4 +2490,10 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static UserDataVO createMockUserDataVO() {
+        UserDataVO userData = Mockito.mock(UserDataVO.class);
+        Mockito.when(userData.getId()).thenReturn(1L);
+        return userData;
+    }
 }
