--- a/api/src/test/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmdTest.java
@@ -226,12 +226,11 @@
         Long associatedNetworkId = 1L;
         ReflectionTestUtils.setField(cmd, "networkOfferingId", networkOfferingId);
         ReflectionTestUtils.setField(cmd, "associatedNetworkId", associatedNetworkId);
-        NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
         Network network = Mockito.mock(Network.class);
         ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
         Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
         Mockito.when(_entityMgr.findById(Network.class, associatedNetworkId)).thenReturn(network);
-        Mockito.when(networkOffering.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        NetworkOffering networkOffering = createMockNetworkOffering(Network.GuestType.Isolated);
         try {
             cmd.getPhysicalNetworkId();
         } catch (Exception e) {
@@ -244,10 +243,9 @@
         Long physicalNetworkId = 1L;
         Long networkOfferingId = 1L;
         ReflectionTestUtils.setField(cmd, "networkOfferingId", networkOfferingId);
-        NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
-        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
-        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
-        Mockito.when(networkOffering.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
+        NetworkOffering networkOffering = createMockNetworkOffering(Network.GuestType.Isolated);
+        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
         try {
             cmd.getPhysicalNetworkId();
         } catch (Exception e) {
@@ -260,10 +258,9 @@
         Long physicalNetworkId = 1L;
         Long networkOfferingId = 1L;
         ReflectionTestUtils.setField(cmd, "networkOfferingId", networkOfferingId);
-        NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
-        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
-        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
-        Mockito.when(networkOffering.getGuestType()).thenReturn(Network.GuestType.L2);
+        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
+        NetworkOffering networkOffering = createMockNetworkOffering(Network.GuestType.L2);
+        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
         try {
             Assert.assertEquals(cmd.getPhysicalNetworkId(), physicalNetworkId);
         } catch (Exception e) {
@@ -276,10 +273,9 @@
         Long physicalNetworkId = 1L;
         Long networkOfferingId = 1L;
         ReflectionTestUtils.setField(cmd, "networkOfferingId", networkOfferingId);
-        NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
-        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
-        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
-        Mockito.when(networkOffering.getGuestType()).thenReturn(Network.GuestType.Shared);
+        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
+        NetworkOffering networkOffering = createMockNetworkOffering(Network.GuestType.Shared);
+        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
         try {
             Assert.assertEquals(cmd.getPhysicalNetworkId(), physicalNetworkId);
         } catch (Exception e) {
@@ -292,10 +288,9 @@
         Long physicalNetworkId = 1L;
         Long networkOfferingId = 1L;
         ReflectionTestUtils.setField(cmd, "networkOfferingId", networkOfferingId);
-        NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
-        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
-        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
-        Mockito.when(networkOffering.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering networkOffering = createMockNetworkOffering(Network.GuestType.Shared);
+        ReflectionTestUtils.setField(cmd, "physicalNetworkId", physicalNetworkId);
+        Mockito.when(_entityMgr.findById(NetworkOffering.class, networkOfferingId)).thenReturn(networkOffering);
         Long zoneId = 1L;
         ReflectionTestUtils.setField(cmd, "zoneId", zoneId);
         Assert.assertEquals(cmd.getZoneId(), zoneId);
@@ -350,4 +345,10 @@
             Assert.fail("Should successfully create the network");
         }
     }
+
+    private static NetworkOffering createMockNetworkOffering(Network.GuestType guestType) {
+        NetworkOffering networkOffering = Mockito.mock(NetworkOffering.class);
+        Mockito.when(networkOffering.getGuestType()).thenReturn(guestType);
+        return networkOffering;
+    }
 }
