--- a/server/src/test/java/com/cloud/capacity/CapacityManagerImplTest.java
+++ b/server/src/test/java/com/cloud/capacity/CapacityManagerImplTest.java
@@ -99,11 +99,9 @@
     @Test
     public void testGetServiceOfferingsMap() {
         Long offering1Id = 1L;
-        ServiceOfferingVO offering1 = mock(ServiceOfferingVO.class);
-        when(offering1.getId()).thenReturn(offering1Id);
+        ServiceOfferingVO offering1 = createMockServiceOfferingVO(offering1Id);
         Long offering2Id = 2L;
-        ServiceOfferingVO offering2 = mock(ServiceOfferingVO.class);
-        when(offering2.getId()).thenReturn(offering2Id);
+        ServiceOfferingVO offering2 = createMockServiceOfferingVO(offering2Id);
         when(serviceOfferingDao.listAllIncludingRemoved()).thenReturn(List.of(offering1, offering2));
         Map<Long, ServiceOfferingVO> result = capacityManager.getServiceOfferingsMap();
         assertEquals(2, result.size());
@@ -179,4 +177,10 @@
                 eq(ByteScaleUtils.mebibytesToBytes(OFFERING_MEMORY)),
                 eq(false), eq(cpuOvercommit), eq(memoryOvercommit), eq(false));
     }
+
+    private static ServiceOfferingVO createMockServiceOfferingVO(Long id) {
+        ServiceOfferingVO offering = mock(ServiceOfferingVO.class);
+        when(offering.getId()).thenReturn(id);
+        return offering;
+    }
 }
