--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -177,6 +177,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class AutoScaleManagerImplTest {
+    @Mock
+    private AutoScaleVmProfileTO profileTO;
 
     @Spy
     @InjectMocks
@@ -1624,7 +1626,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -1641,7 +1642,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -1658,7 +1658,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -1675,7 +1674,7 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
+        // removed local mock; replaced with global field `profileTO`
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -2156,7 +2155,6 @@
     @Test
     public void updateCountersMapWithInstantDataForCPU() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
 
