--- a/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
+++ b/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
@@ -125,17 +125,11 @@
 
     @Test
     public void testValidateVnfNicsAllGood() {
-        VnfNic nic1 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic1.getDeviceId()).thenReturn(0L);
-        Mockito.when(nic1.isRequired()).thenReturn(true);
-
-        VnfNic nic2 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic2.getDeviceId()).thenReturn(1L);
-        Mockito.when(nic2.isRequired()).thenReturn(true);
-
-        VnfNic nic3 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic3.getDeviceId()).thenReturn(2L);
-        Mockito.when(nic3.isRequired()).thenReturn(false);
+
+        VnfNic nic1 = createMockVnfNic(0L, true);
+        VnfNic nic2 = createMockVnfNic(1L, true);
+
+        VnfNic nic3 = createMockVnfNic(2L, false);
 
         List<VnfNic> nicsList = Arrays.asList(nic1, nic2, nic3);
 
@@ -158,10 +152,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateVnfNicsWithNonConstantDeviceIds() {
-        VnfNic nic1 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic1.getDeviceId()).thenReturn(0L);
-        Mockito.when(nic1.isRequired()).thenReturn(true);
-
+
+        VnfNic nic1 = createMockVnfNic(0L, true);
         VnfNic nic2 = Mockito.mock(VnfNic.class);
         Mockito.when(nic2.getDeviceId()).thenReturn(2L);
 
@@ -174,17 +166,11 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateVnfNicsWithInvalidRequired() {
-        VnfNic nic1 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic1.getDeviceId()).thenReturn(0L);
-        Mockito.when(nic1.isRequired()).thenReturn(true);
-
-        VnfNic nic2 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic2.getDeviceId()).thenReturn(1L);
-        Mockito.when(nic2.isRequired()).thenReturn(false);
-
-        VnfNic nic3 = Mockito.mock(VnfNic.class);
-        Mockito.when(nic3.getDeviceId()).thenReturn(2L);
-        Mockito.when(nic3.isRequired()).thenReturn(true);
+
+        VnfNic nic1 = createMockVnfNic(0L, true);
+        VnfNic nic2 = createMockVnfNic(1L, false);
+
+        VnfNic nic3 = createMockVnfNic(2L, true);
 
         List<VnfNic> nicsList = Arrays.asList(nic1, nic2, nic3);
 
@@ -258,4 +244,11 @@
         cidrList.add("10.10.10.0/33");
         VnfTemplateUtils.validateVnfCidrList(cidrList);
     }
+
+    private static VnfNic createMockVnfNic(long deviceId, boolean required) {
+        VnfNic vnfNic = Mockito.mock(VnfNic.class);
+        Mockito.when(vnfNic.getDeviceId()).thenReturn(deviceId);
+        Mockito.when(vnfNic.isRequired()).thenReturn(required);
+        return vnfNic;
+    }
 }
