--- a/plugins/network-elements/globodns/src/test/java/com/globo/globodns/cloudstack/element/GloboDnsElementTest.java
+++ b/plugins/network-elements/globodns/src/test/java/com/globo/globodns/cloudstack/element/GloboDnsElementTest.java
@@ -161,8 +161,7 @@
         DeployDestination dest = new DeployDestination();
         ReservationContext context = new ReservationContextImpl(null, null, user);
 
-        HostVO hostVO = mock(HostVO.class);
-        when(hostVO.getId()).thenReturn(globoDnsHostId);
+        HostVO hostVO = createMockHostVO(globoDnsHostId);
         when(_hostDao.findByTypeNameAndZoneId(eq(zoneId), eq(Provider.GloboDns.getName()), eq(Type.L2Networking))).thenReturn(hostVO);
 
         when(_agentMgr.easySend(eq(globoDnsHostId), isA(CreateOrUpdateRecordAndReverseCommand.class))).then(new org.mockito.stubbing.Answer<Answer>() {
@@ -193,8 +192,7 @@
         when(_datacenterDao.findById(zoneId)).thenReturn(dataCenterVO);
         ReservationContext context = new ReservationContextImpl(null, null, user);
 
-        HostVO hostVO = mock(HostVO.class);
-        when(hostVO.getId()).thenReturn(globoDnsHostId);
+        HostVO hostVO = createMockHostVO(globoDnsHostId);
         when(_hostDao.findByTypeNameAndZoneId(eq(zoneId), eq(Provider.GloboDns.getName()), eq(Type.L2Networking))).thenReturn(hostVO);
 
         when(_agentMgr.easySend(eq(globoDnsHostId), isA(RemoveRecordCommand.class))).then(new org.mockito.stubbing.Answer<Answer>() {
@@ -263,4 +261,10 @@
             }
         }
     }
+
+    private static HostVO createMockHostVO(long globoDnsHostId) {
+        HostVO hostVO = mock(HostVO.class);
+        when(hostVO.getId()).thenReturn(globoDnsHostId);
+        return hostVO;
+    }
 }
