--- a/server/src/test/java/com/cloud/api/ApiServerTest.java
+++ b/server/src/test/java/com/cloud/api/ApiServerTest.java
@@ -127,12 +127,9 @@
 
     @Test
     public void testForgotPasswordSuccess() {
-        UserAccount userAccount = mock(UserAccount.class);
-        Domain domain = mock(Domain.class);
-
-        Mockito.when(userAccount.getEmail()).thenReturn("test@test.com");
-        Mockito.when(userAccount.getState()).thenReturn("ENABLED");
-        Mockito.when(userAccount.getAccountState()).thenReturn("ENABLED");
+        Domain domain = mock(Domain.class);
+
+        UserAccount userAccount = createMockUserAccount("ENABLED");
         Mockito.when(domain.getState()).thenReturn(Domain.State.Active);
         Mockito.doNothing().when(userPasswordResetManager).setResetTokenAndSend(userAccount);
         Assert.assertTrue(apiServer.forgotPassword(userAccount, domain));
@@ -160,23 +157,17 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testForgotPasswordFailureDisabledAccount() {
-        UserAccount userAccount = mock(UserAccount.class);
-        Domain domain = mock(Domain.class);
-
-        Mockito.when(userAccount.getEmail()).thenReturn("test@test.com");
-        Mockito.when(userAccount.getState()).thenReturn("ENABLED");
-        Mockito.when(userAccount.getAccountState()).thenReturn("DISABLED");
+        Domain domain = mock(Domain.class);
+
+        UserAccount userAccount = createMockUserAccount("DISABLED");
         apiServer.forgotPassword(userAccount, domain);
     }
 
     @Test(expected = CloudRuntimeException.class)
     public void testForgotPasswordFailureInactiveDomain() {
-        UserAccount userAccount = mock(UserAccount.class);
-        Domain domain = mock(Domain.class);
-
-        Mockito.when(userAccount.getEmail()).thenReturn("test@test.com");
-        Mockito.when(userAccount.getState()).thenReturn("ENABLED");
-        Mockito.when(userAccount.getAccountState()).thenReturn("ENABLED");
+        Domain domain = mock(Domain.class);
+
+        UserAccount userAccount = createMockUserAccount("ENABLED");
         Mockito.when(domain.getState()).thenReturn(Domain.State.Inactive);
         apiServer.forgotPassword(userAccount, domain);
     }
@@ -276,4 +267,12 @@
         Mockito.when(accountManager.authenticateUser(anyString(), anyString(), anyLong(), any(), any())).thenReturn(null);
         apiServer.loginUser(session, "user", "pass", 1L, "/", null, null);
     }
+
+    private static UserAccount createMockUserAccount(String accountStateReturn) {
+        UserAccount userAccount = Mockito.mock(UserAccount.class);
+        Mockito.when(userAccount.getEmail()).thenReturn("test@test.com");
+        Mockito.when(userAccount.getState()).thenReturn("ENABLED");
+        Mockito.when(userAccount.getAccountState()).thenReturn(accountStateReturn);
+        return userAccount;
+    }
 }
