--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -594,7 +594,7 @@
         nicProfile.setIPv4Address("192.168.100.100");
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = MockHostDao.createMockHostDao(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(createTungstenNetworkPolicyAnswer);
@@ -621,7 +621,7 @@
         nicProfile.setIPv4Address("192.168.100.100");
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = MockHostDao.createMockHostDao(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
 
@@ -646,7 +646,7 @@
         nicProfile.setIPv4Address("192.168.100.100");
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = createMockHostDao(host);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
 
@@ -670,7 +670,7 @@
         when(nicProfile.getIPv4Address()).thenReturn("192.168.100.100");
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = MockHostDao.createMockHostDao(host);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
         when(ipAddressDao.findByIpAndDcId(anyLong(), anyString())).thenReturn(ipAddressVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
@@ -695,7 +695,7 @@
 
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = MockHostDao.createMockHostDao(host);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
@@ -717,7 +717,7 @@
 
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = createMockHostDao(host);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(deleteTungstenVRouterPortAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong())).thenReturn(deleteVmiAnswer);
@@ -795,7 +795,7 @@
         when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
         when(tungstenProvider.getGateway()).thenReturn("192.168.100.100");
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         when(dataCenterVO.isSecurityGroupEnabled()).thenReturn(true);
         when(networkDao.listByZoneSecurityGroup(anyLong())).thenReturn(List.of(network));
 
@@ -814,7 +814,7 @@
         when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
         when(tungstenProvider.getGateway()).thenReturn("192.168.100.100");
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         when(dataCenterVO.isSecurityGroupEnabled()).thenReturn(false);
 
         tungstenElement.processHostAboutToBeRemoved(1L);
@@ -921,7 +921,7 @@
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(tungstenAnswer);
 
         assertTrue(tungstenElement.prepareMigration(nic, network, vm, dest, context));
@@ -939,8 +939,8 @@
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
         when(vm.getVirtualMachine()).thenReturn(virtualMachine);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
-
+
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         tungstenElement.rollbackMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
@@ -957,9 +957,20 @@
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
         when(vm.getVirtualMachine()).thenReturn(virtualMachine);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
-
+
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    public final class MockHostDao {
+        private MockHostDao() {
+        }
+
+        public static HostDao createMockHostDao(HostVO findByIdReturn) {
+            HostDao hostDao = mock(HostDao.class);
+            when(hostDao.findById(anyLong())).thenReturn(findByIdReturn);
+            return hostDao;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -389,8 +389,8 @@
         when(vm.getType()).thenReturn(VirtualMachine.Type.User);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(host);
-
+
+        hostDao = MockHostDao.createMockHostDao(host);
         guru.reserve(nic, network, vm, dest, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
             anyLong());
@@ -404,7 +404,7 @@
         final VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
         final HostVO host = mock(HostVO.class);
 
-        when(hostDao.findById(anyLong())).thenReturn(host);
+        hostDao = MockHostDao.createMockHostDao(host);
         when(
             tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong())).thenReturn(
             new TungstenAnswer(new TungstenCommand(), true, ""));
@@ -482,7 +482,7 @@
         when(vm.getType()).thenReturn(VirtualMachine.Type.User);
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
             anyLong())).thenReturn(new TungstenAnswer(new TungstenCommand(), true, ""));
 
@@ -502,8 +502,8 @@
 
         when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
-
+
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         guru.rollbackMigration(nic, network, vmProfile, context, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
@@ -520,10 +520,21 @@
 
         when(vmProfile.getType()).thenReturn(VirtualMachine.Type.User);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
-        when(hostDao.findById(anyLong())).thenReturn(hostVO);
-
+
+        hostDao = MockHostDao.createMockHostDao(hostVO);
         guru.commitMigration(nic, network, vmProfile, context, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
+
+    public final class MockHostDao {
+        private MockHostDao() {
+        }
+
+        public static HostDao createMockHostDao(HostVO findByIdReturn) {
+            HostDao hostDao = mock(HostDao.class);
+            when(hostDao.findById(anyLong())).thenReturn(findByIdReturn);
+            return hostDao;
+        }
+    }
 }
