--- a/agent/src/test/java/com/cloud/agent/AgentTest.java
+++ b/agent/src/test/java/com/cloud/agent/AgentTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package com.cloud.agent;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
@@ -49,6 +50,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class AgentTest {
+    @Mock
+    private IAgentControlListener listener;
     Agent agent;
     private AgentShell shell;
     private ServerResource serverResource;
@@ -189,14 +192,12 @@
 
     @Test
     public void testRegisterControlListenerAddsListener() {
-        IAgentControlListener listener = mock(IAgentControlListener.class);
         agent.registerControlListener(listener);
         assertTrue(agent.controlListeners.contains(listener));
     }
 
     @Test
     public void testUnregisterControlListenerRemovesListener() {
-        IAgentControlListener listener = mock(IAgentControlListener.class);
         agent.registerControlListener(listener);
         assertTrue(agent.controlListeners.contains(listener));
         agent.unregisterControlListener(listener);
