--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -1485,14 +1485,12 @@
     @Test
     public void testAddServiceOfferingDetailsToParamsCustomConstrainedOffering() {
         Map<String, String> params = new HashMap<>();
-        ServiceOfferingVO serviceOfferingVO = mock(ServiceOfferingVO.class);
         Map<String, String> details = new HashMap<>();
         details.put(ApiConstants.MIN_CPU_NUMBER, "1");
         details.put(ApiConstants.MIN_MEMORY, "1024");
+        ServiceOfferingVO serviceOfferingVO = createMockServiceOfferingVO();
         Mockito.when(serviceOfferingVO.getDetails()).thenReturn(details);
-        Mockito.when(serviceOfferingVO.getCpu()).thenReturn(null);
         Mockito.when(serviceOfferingVO.getSpeed()).thenReturn(1500);
-        Mockito.when(serviceOfferingVO.getRamSize()).thenReturn(null);
         unmanagedVMsManager.addServiceOfferingDetailsToParams(params, serviceOfferingVO);
         Assert.assertEquals("1", params.get(VmDetailConstants.CPU_NUMBER));
         Assert.assertEquals("1500", params.get(VmDetailConstants.CPU_SPEED));
@@ -1502,10 +1500,8 @@
     @Test
     public void testAddServiceOfferingDetailsToParamsCustomUnconstrainedOffering() {
         Map<String, String> params = new HashMap<>();
-        ServiceOfferingVO serviceOfferingVO = mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOfferingVO.getCpu()).thenReturn(null);
+        ServiceOfferingVO serviceOfferingVO = createMockServiceOfferingVO();
         Mockito.when(serviceOfferingVO.getSpeed()).thenReturn(null);
-        Mockito.when(serviceOfferingVO.getRamSize()).thenReturn(null);
         unmanagedVMsManager.addServiceOfferingDetailsToParams(params, serviceOfferingVO);
         Assert.assertFalse(params.containsKey(VmDetailConstants.CPU_NUMBER));
         Assert.assertFalse(params.containsKey(VmDetailConstants.CPU_SPEED));
@@ -1609,4 +1605,11 @@
         details.put("key", "not-a-number");
         unmanagedVMsManager.getDetailAsInteger("key", details);
     }
+
+    private static ServiceOfferingVO createMockServiceOfferingVO() {
+        ServiceOfferingVO serviceOfferingVO = mock(ServiceOfferingVO.class);
+        Mockito.when(serviceOfferingVO.getCpu()).thenReturn(null);
+        Mockito.when(serviceOfferingVO.getRamSize()).thenReturn(null);
+        return serviceOfferingVO;
+    }
 }
