--- a/server/src/test/java/com/cloud/user/DomainManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/DomainManagerImplTest.java
@@ -74,6 +74,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class DomainManagerImplTest {
     @Mock
+    private SearchCriteria scMock;
+    @Mock
     DomainDao domainDaoMock;
     @Mock
     AccountManager _accountMgr;
@@ -343,7 +345,6 @@
 
     @Test (expected = InvalidParameterValueException.class)
     public void validateUniqueDomainNameTestExistingNameThrowsException() {
-        SearchCriteria scMock = Mockito.mock(SearchCriteria.class);
         ArrayList<DomainVO> listMock = Mockito.mock(ArrayList.class);
         Mockito.doReturn(scMock).when(domainDaoMock).createSearchCriteria();
         Mockito.doReturn(listMock).when(domainDaoMock).search(Mockito.any(SearchCriteria.class), Mockito.any());
@@ -353,7 +354,6 @@
 
     @Test
     public void validateUniqueDomainNameTestUniqueNameDoesNotThrowException() {
-        SearchCriteria scMock = Mockito.mock(SearchCriteria.class);
         ArrayList<DomainVO> listMock = Mockito.mock(ArrayList.class);
         Mockito.doReturn(scMock).when(domainDaoMock).createSearchCriteria();
         Mockito.doReturn(listMock).when(domainDaoMock).search(Mockito.any(SearchCriteria.class), Mockito.any());
