--- a/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
@@ -359,11 +359,9 @@
 
     @Test
     public void testGetVmIdsWithActiveVolumeSnapshots() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        Mockito.when(vol1.getId()).thenReturn(1L);
+        VolumeVO vol1 = createMockVolumeVO(1L);
         Mockito.when(vol1.getInstanceId()).thenReturn(1L);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class);
-        Mockito.when(vol2.getId()).thenReturn(2L);
+        VolumeVO vol2 = createMockVolumeVO(2L);
         Mockito.when(volumeDao.searchRemovedByVms(ids, null)).thenReturn(List.of(vol1, vol2));
         SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
         Mockito.when(snapshotVO.getVolumeId()).thenReturn(1L);
@@ -685,4 +683,10 @@
         resourceCleanupService.parseDateFromConfig(
                 ResourceCleanupService.ExpungedResourcesPurgeStartTime.key(), "ABC");
     }
+
+    private static VolumeVO createMockVolumeVO(long id) {
+        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
+        Mockito.when(volumeVO.getId()).thenReturn(id);
+        return volumeVO;
+    }
 }
