--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -175,6 +175,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VirtualMachineManagerImplTest {
+    @Mock
+    private Account owner;
 
     @Spy
     @InjectMocks
@@ -1124,7 +1126,6 @@
         User user = mock(User.class);
 
         Account account = mock(Account.class);
-        Account owner = mock(Account.class);
 
         ReservationContext ctx = mock(ReservationContext.class);
 
@@ -1220,7 +1221,6 @@
         User user = mock(User.class);
 
         Account account = mock(Account.class);
-        Account owner = mock(Account.class);
 
         ReservationContext ctx = mock(ReservationContext.class);
 
