--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
@@ -497,9 +497,8 @@
         }
 
         ListDataStoreObjectsCommand command = Mockito.mock(ListDataStoreObjectsCommand.class);
-        DataStoreTO store = Mockito.mock(DataStoreTO.class);
+        DataStoreTO store = createDataStoreTO();
         Mockito.when(command.getStore()).thenReturn(store);
-        Mockito.when(store.getUuid()).thenReturn("storeUuid");
         Mockito.when(command.getStartIndex()).thenReturn(0);
         Mockito.when(command.getPageSize()).thenReturn(10);
         Mockito.when(command.getPath()).thenReturn("/path/to/files");
@@ -541,9 +540,8 @@
         SR srMock = Mockito.mock(SR.class);
 
         ListDataStoreObjectsCommand command = Mockito.mock(ListDataStoreObjectsCommand.class);
-        DataStoreTO store = Mockito.mock(DataStoreTO.class);
+        DataStoreTO store = createDataStoreTO();
         Mockito.when(command.getStore()).thenReturn(store);
-        Mockito.when(store.getUuid()).thenReturn("storeUuid");
         Mockito.when(command.getStartIndex()).thenReturn(0);
         Mockito.when(command.getPageSize()).thenReturn(10);
         Mockito.when(command.getPath()).thenReturn("/path/to/non/existent/files");
@@ -582,9 +580,8 @@
         fileListMock.add(fileAttributesMock);
 
         ListDataStoreObjectsCommand command = Mockito.mock(ListDataStoreObjectsCommand.class);
-        DataStoreTO store = Mockito.mock(DataStoreTO.class);
+        DataStoreTO store = createDataStoreTO();
         Mockito.when(command.getStore()).thenReturn(store);
-        Mockito.when(store.getUuid()).thenReturn("storeUuid");
         Mockito.when(command.getStartIndex()).thenReturn(0);
         Mockito.when(command.getPageSize()).thenReturn(10);
         Mockito.when(command.getPath()).thenReturn("/path/to/file");
@@ -616,4 +613,10 @@
         Assert.assertEquals(1024L, listAnswer.getSizes().get(0).longValue());
         Assert.assertEquals(123456789000L, listAnswer.getLastModified().get(0).longValue());
     }
+
+    private static DataStoreTO createDataStoreTO() {
+        DataStoreTO store = Mockito.mock(DataStoreTO.class);
+        Mockito.when(store.getUuid()).thenReturn("storeUuid");
+        return store;
+    }
 }
