--- a/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
@@ -80,6 +80,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class SnapshotManagerImplTest {
+    @Mock
+    private SnapshotPolicyVO policy2;
     @Mock
     AccountDao accountDao;
     @Mock
@@ -445,7 +447,6 @@
         CallContext.register(Mockito.mock(User.class), caller);
 
         SnapshotPolicyVO policy1 = Mockito.mock(SnapshotPolicyVO.class);
-        SnapshotPolicyVO policy2 = Mockito.mock(SnapshotPolicyVO.class);
         List<SnapshotPolicyVO> mockPolicies = List.of(policy1, policy2);
 
         SearchBuilder<SnapshotPolicyVO> mockSearchBuilder = Mockito.mock(SearchBuilder.class);
@@ -480,9 +481,8 @@
         CallContext.register(Mockito.mock(User.class), caller);
 
         SnapshotPolicyVO policy1 = Mockito.mock(SnapshotPolicyVO.class);
-        SnapshotPolicyVO policy2 = Mockito.mock(SnapshotPolicyVO.class);
+
         List<SnapshotPolicyVO> mockPolicies = List.of(policy1, policy2);
-
         SearchBuilder<SnapshotPolicyVO> mockSearchBuilder = Mockito.mock(SearchBuilder.class);
         SearchBuilder<VolumeVO> mockVolumeSearchBuilder = Mockito.mock(SearchBuilder.class);
         SearchCriteria<SnapshotPolicyVO> mockSearchCriteria = Mockito.mock(SearchCriteria.class);
