--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtConvertInstanceCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtConvertInstanceCommandWrapperTest.java
@@ -48,6 +48,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class LibvirtConvertInstanceCommandWrapperTest {
+    @Mock
+    private Script script;
 
     @Spy
     private LibvirtConvertInstanceCommandWrapper convertInstanceCommandWrapper = Mockito.spy(LibvirtConvertInstanceCommandWrapper.class);
@@ -175,7 +177,6 @@
 
     @Test
     public void testAddExtraParamsToScriptSameKeysAndValues() {
-        Script script = Mockito.mock(Script.class);
         String extraParams = "--mac 00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254";
         convertInstanceCommandWrapper.addExtraParamsToScript(extraParams, script);
         Mockito.verify(script).add("--mac", "00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254");
@@ -183,7 +184,6 @@
 
     @Test
     public void testAddExtraParamsToScriptDifferentArgs() {
-        Script script = Mockito.mock(Script.class);
         String extraParams = "--mac 00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254 -x -v";
         convertInstanceCommandWrapper.addExtraParamsToScript(extraParams, script);
         Mockito.verify(script).add("--mac", "00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254");
