--- a/services/secondary-storage/controller/src/test/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerTest.java
+++ b/services/secondary-storage/controller/src/test/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerTest.java
@@ -52,6 +52,8 @@
 
 public class SecondaryStorageManagerTest {
     @Mock
+    private NetworkVO badNetwork;
+    @Mock
     DataCenterDao _dcDao;
 
     @Mock
@@ -85,7 +87,6 @@
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Public)))
                     .thenReturn(Collections.singletonList(network));
 
@@ -112,7 +113,6 @@
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
 
@@ -134,7 +134,6 @@
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
 
@@ -157,7 +156,6 @@
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
 
@@ -181,7 +179,6 @@
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), eq(TrafficType.Guest)))
                     .thenReturn(Collections.singletonList(network));
 
@@ -200,7 +197,6 @@
         when(_dcDao.findById(Mockito.anyLong())).thenReturn(dc);
 
         NetworkVO network = Mockito.mock(NetworkVO.class);
-        NetworkVO badNetwork = Mockito.mock(NetworkVO.class);
         when(_networkDao.listByZoneAndTrafficType(anyLong(), any(TrafficType.class)))
                     .thenReturn(Collections.singletonList(badNetwork));
 
