--- a/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
@@ -119,6 +119,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VpcManagerImplTest {
+    @Mock
+    private Account accountMock;
 
     @Mock
     VpcOfferingServiceMapDao vpcOffSvcMapDao;
@@ -365,7 +367,6 @@
         manager._maxNetworks = 3;
         VpcVO vpcMockVO = Mockito.mock(VpcVO.class);
         Vpc vpcMock = Mockito.mock(Vpc.class);
-        Account accountMock = Mockito.mock(Account.class);
         PhysicalNetwork physicalNetwork = Mockito.mock(PhysicalNetwork.class);
         NetworkOffering offering = Mockito.mock(NetworkOffering.class);
         List<Network.Service> services = new ArrayList<>();
@@ -405,7 +406,6 @@
         long vpcId = 1L;
         Integer publicMtu = 1450;
         String sourceNatIp = "1.2.3.4";
-        Account accountMock = Mockito.mock(Account.class);
         VpcVO vpcVO = new VpcVO();
 
         Answer answer = Mockito.mock(Answer.class);
