--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -733,8 +733,7 @@
 
     @Test
     public void checkIfPodIsDeletableSuccessTest() {
-        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
-        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        HostPodVO hostPodVO = createMockHostPodVO();
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(0);
@@ -748,8 +747,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfPodIsDeletableFailureOnPrivateIpAddressTest() {
-        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
-        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        HostPodVO hostPodVO = createMockHostPodVO();
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(1);
@@ -763,8 +761,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfPodIsDeletableFailureOnVolumeTest() {
-        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
-        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        HostPodVO hostPodVO = createMockHostPodVO();
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
         VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
@@ -781,8 +778,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfPodIsDeletableFailureOnHostTest() {
-        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
-        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        HostPodVO hostPodVO = createMockHostPodVO();
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
         HostVO hostVO = Mockito.mock(HostVO.class);
@@ -799,8 +795,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfPodIsDeletableFailureOnVmInstanceTest() {
-        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
-        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        HostPodVO hostPodVO = createMockHostPodVO();
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
         VMInstanceVO vMInstanceVO = Mockito.mock(VMInstanceVO.class);
@@ -817,8 +812,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfPodIsDeletableFailureOnClusterTest() {
-        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
-        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        HostPodVO hostPodVO = createMockHostPodVO();
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
         ClusterVO clusterVO = Mockito.mock(ClusterVO.class);
@@ -1508,4 +1502,10 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static HostPodVO createMockHostPodVO() {
+        HostPodVO hostPodVO = Mockito.mock(HostPodVO.class);
+        Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
+        return hostPodVO;
+    }
 }
