--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -358,8 +358,7 @@
         //SHARED NETWORKS CASE 2
         when(niciraNvpRouterMappingDao.existsMappingForNetworkId(NETWORK_ID)).thenReturn(false);
 
-        VlanVO vlanVO = mock(VlanVO.class);
-        when(vlanVO.getVlanTag()).thenReturn("111");
+        VlanVO vlanVO = createMockVlanVO();
         when(vlanDao.listVlansByNetworkId(NETWORK_ID)).thenReturn(Arrays.asList(new VlanVO[] {vlanVO}));
 
         when(niciraNvpHost.getDetail("l2gatewayserviceuuid")).thenReturn("bbbb-bbbb-bbbb");
@@ -410,12 +409,17 @@
         //SHARED NETWORKS CASE 2
         when(niciraNvpRouterMappingDao.existsMappingForNetworkId(NETWORK_ID)).thenReturn(false);
 
+        VlanVO vlanVO = createMockVlanVO();
+        when(vlanDao.listVlansByNetworkId(NETWORK_ID)).thenReturn(Arrays.asList(new VlanVO[] {vlanVO}));
+
+        when(niciraNvpHost.getDetail("l2gatewayserviceuuid")).thenReturn(null);
+
+        element.implement(network, offering, dest, context);
+    }
+
+    private static VlanVO createMockVlanVO() {
         VlanVO vlanVO = mock(VlanVO.class);
         when(vlanVO.getVlanTag()).thenReturn("111");
-        when(vlanDao.listVlansByNetworkId(NETWORK_ID)).thenReturn(Arrays.asList(new VlanVO[] {vlanVO}));
-
-        when(niciraNvpHost.getDetail("l2gatewayserviceuuid")).thenReturn(null);
-
-        element.implement(network, offering, dest, context);
+        return vlanVO;
     }
 }
