--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -170,17 +170,16 @@
         // make local mocks
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
 
         // make sure that release dhcp will be called
         when(vm.getType()).thenReturn(Type.User);
+        NetworkVO network = createNetworkVO(guruName);
         when(testOrchestrator._networkModel.areServicesSupportedInNetwork(network.getId(), Service.Dhcp)).thenReturn(true);
         when(network.getTrafficType()).thenReturn(TrafficType.Guest);
         when(network.getGuestType()).thenReturn(GuestType.Shared);
         when(testOrchestrator._nicDao.listByNetworkIdTypeAndGatewayAndBroadcastUri(nic.getNetworkId(), VirtualMachine.Type.User, nic.getIPv4Gateway(), nic.getBroadcastUri()))
                 .thenReturn(new ArrayList<NicVO>());
 
-        when(network.getGuruName()).thenReturn(guruName);
         when(testOrchestrator._networksDao.findById(nic.getNetworkId())).thenReturn(network);
 
         testOrchestrator.removeNic(vm, nic);
@@ -195,12 +194,11 @@
         // make local mocks
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
 
         // make sure that release dhcp won't be called
         when(vm.getType()).thenReturn(Type.DomainRouter);
 
-        when(network.getGuruName()).thenReturn(guruName);
+        NetworkVO network = createNetworkVO(guruName);
         when(testOrchestrator._networksDao.findById(nic.getNetworkId())).thenReturn(network);
 
         testOrchestrator.removeNic(vm, nic);
@@ -215,13 +213,12 @@
         // make local mocks
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
-
+
+        NetworkVO network = createNetworkVO(guruName);
         // make sure that release dhcp will *not* be called
         when(vm.getType()).thenReturn(Type.User);
         when(testOrchestrator._networkModel.areServicesSupportedInNetwork(network.getId(), Service.Dhcp)).thenReturn(false);
 
-        when(network.getGuruName()).thenReturn(guruName);
         when(testOrchestrator._networksDao.findById(nic.getNetworkId())).thenReturn(network);
 
         testOrchestrator.removeNic(vm, nic);
@@ -515,10 +512,9 @@
     public void testDontReleaseNicWhenPreserveNicsSettingEnabled() {
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
 
         when(vm.getType()).thenReturn(Type.User);
-        when(network.getGuruName()).thenReturn(guruName);
+        NetworkVO network = createNetworkVO(guruName);
         when(testOrchestrator._networksDao.findById(nic.getNetworkId())).thenReturn(network);
 
         Long nicId = 1L;
@@ -1073,4 +1069,10 @@
         assertNotNull(result);
         assertEquals(1, result.size());
     }
+
+    private static NetworkVO createNetworkVO(String guruName) {
+        NetworkVO network = mock(NetworkVO.class);
+        when(network.getGuruName()).thenReturn(guruName);
+        return network;
+    }
 }
