--- a/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
@@ -586,8 +586,7 @@
 
     @Test
     public void testExpungedVmResourcesLaterIfNeededFalse() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        VirtualMachine vm = createVirtualMachine();
         Mockito.doReturn(false).when(resourceCleanupService).isVmOfferingPurgeResourcesEnabled(1L);
         resourceCleanupService.purgeExpungedVmResourcesLaterIfNeeded(vm);
         Mockito.verify(resourceCleanupService, Mockito.never()).purgeExpungedResourceLater(Mockito.anyLong(), Mockito.any());
@@ -595,8 +594,7 @@
 
     @Test
     public void testExpungedVmResourcesLaterIfNeeded() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        VirtualMachine vm = createVirtualMachine();
         Mockito.doReturn(true).when(resourceCleanupService).isVmOfferingPurgeResourcesEnabled(1L);
         Mockito.doNothing().when(resourceCleanupService).purgeExpungedResourceLater(Mockito.anyLong(), Mockito.any());
         resourceCleanupService.purgeExpungedVmResourcesLaterIfNeeded(vm);
@@ -685,4 +683,10 @@
         resourceCleanupService.parseDateFromConfig(
                 ResourceCleanupService.ExpungedResourcesPurgeStartTime.key(), "ABC");
     }
+
+    private static VirtualMachine createVirtualMachine() {
+        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
+        Mockito.when(vm.getServiceOfferingId()).thenReturn(1L);
+        return vm;
+    }
 }
