--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -216,6 +216,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class UserVmManagerImplTest {
+    @Mock
+    private DiskOfferingVO currentRootDiskOffering;
 
     @Spy
     @InjectMocks
@@ -906,7 +908,6 @@
     @Test (expected = InvalidParameterValueException.class)
     public void prepareResizeVolumeCmdTestRootVolumeNull() {
         DiskOfferingVO newRootDiskOffering = Mockito.mock(DiskOfferingVO.class);
-        DiskOfferingVO currentRootDiskOffering = Mockito.mock(DiskOfferingVO.class);
         userVmManagerImpl.prepareResizeVolumeCmd(null, currentRootDiskOffering, newRootDiskOffering);
     }
 
@@ -1901,9 +1902,8 @@
         Mockito.when(diskOfferingDao.findById(diskOffId1)).thenReturn(diskOfferingVO1);
         Mockito.when(resourceLimitMgr.getResourceLimitStorageTags(diskOfferingVO1)).thenReturn(List.of("tag1", "tag2"));
         long diskOffId2 = 2L;
-        DiskOfferingVO diskOfferingVO2 = Mockito.mock(DiskOfferingVO.class);
-        Mockito.when(diskOfferingDao.findById(diskOffId2)).thenReturn(diskOfferingVO2);
-        Mockito.when(resourceLimitMgr.getResourceLimitStorageTags(diskOfferingVO2)).thenReturn(List.of("tag2"));
+        Mockito.when(diskOfferingDao.findById(diskOffId2)).thenReturn(currentRootDiskOffering);
+        Mockito.when(resourceLimitMgr.getResourceLimitStorageTags(currentRootDiskOffering)).thenReturn(List.of("tag2"));
         long diskOffId3 = 3L;
         DiskOfferingVO diskOfferingVO3 = Mockito.mock(DiskOfferingVO.class);
         Mockito.when(diskOfferingDao.findById(diskOffId3)).thenReturn(diskOfferingVO3);
