--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -1126,14 +1126,13 @@
         Account account = mock(Account.class);
         Account owner = mock(Account.class);
 
-        ReservationContext ctx = mock(ReservationContext.class);
-
+
+        ReservationContext ctx = createMockReservationContext();
         ItWorkVO work = mock(ItWorkVO.class);
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
-        when(template.isDeployAsIs()).thenReturn(false);
 
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         when(plan.getDataCenterId()).thenReturn(1L);
@@ -1222,8 +1221,8 @@
         Account account = mock(Account.class);
         Account owner = mock(Account.class);
 
-        ReservationContext ctx = mock(ReservationContext.class);
-
+
+        ReservationContext ctx = createMockReservationContext();
         ItWorkVO work = mock(ItWorkVO.class);
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
@@ -1246,7 +1245,6 @@
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
 
-        when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
         when(serviceOfferingDaoMock.findById(vmInstance.getId(), vmInstance.getServiceOfferingId())).thenReturn(serviceOffering);
 
@@ -2241,4 +2239,10 @@
         verify(clvmPoolManagerMock, times(1)).setClvmLockHostId(3L, destHostId);
     }
 
+
+    private static ReservationContext createMockReservationContext() {
+        ReservationContext ctx = mock(ReservationContext.class);
+        when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
+        return ctx;
+    }
 }
