--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -147,10 +147,8 @@
     }
 
     private void runCheckAndUpdateVolumeAccountResourceCountTest(Long originalSize, Long newSize) {
-        VolumeVO v1 = Mockito.mock(VolumeVO.class);
-        Mockito.when(v1.getSize()).thenReturn(originalSize);
-        VolumeVO v2 = Mockito.mock(VolumeVO.class);
-        Mockito.when(v2.getSize()).thenReturn(newSize);
+        VolumeVO v1 = createMockVolumeVO(originalSize);
+        VolumeVO v2 = createMockVolumeVO(newSize);
         volumeOrchestrator.checkAndUpdateVolumeAccountResourceCount(v1, v2);
         Long expected = ObjectUtils.anyNull(originalSize, newSize) ?
                 DEFAULT_ACCOUNT_PS_RESOURCE_COUNT : DEFAULT_ACCOUNT_PS_RESOURCE_COUNT + (newSize - originalSize);
@@ -945,4 +943,10 @@
         return null;
     }
 
+
+    private static VolumeVO createMockVolumeVO(Long size) {
+        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
+        Mockito.when(volumeVO.getSize()).thenReturn(size);
+        return volumeVO;
+    }
 }
