--- a/api/src/test/java/org/apache/cloudstack/api/command/test/AddIpToVmNicTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/AddIpToVmNicTest.java
@@ -95,10 +95,9 @@
     public void testRemoveIpFromVmNicSuccess() throws ResourceAllocationException, ResourceUnavailableException, ConcurrentOperationException,
         InsufficientCapacityException {
 
-        NetworkService networkService = Mockito.mock(NetworkService.class);
+        NetworkService networkService = createMockNetworkService(true);
         RemoveIpFromVmNicCmd removeIpFromNic = Mockito.mock(RemoveIpFromVmNicCmd.class);
 
-        Mockito.when(networkService.releaseSecondaryIpFromNic(ArgumentMatchers.anyInt())).thenReturn(true);
 
         removeIpFromNic._networkService = networkService;
         removeIpFromNic.execute();
@@ -106,10 +105,9 @@
 
     @Test
     public void testRemoveIpFromVmNicFailure() throws InsufficientAddressCapacityException {
-        NetworkService networkService = Mockito.mock(NetworkService.class);
+        NetworkService networkService = createMockNetworkService(false);
         RemoveIpFromVmNicCmd removeIpFromNic = Mockito.mock(RemoveIpFromVmNicCmd.class);
 
-        Mockito.when(networkService.releaseSecondaryIpFromNic(ArgumentMatchers.anyInt())).thenReturn(false);
 
         removeIpFromNic._networkService = networkService;
         successResponseGenerator = Mockito.mock(SuccessResponse.class);
@@ -120,4 +118,11 @@
             Assert.assertEquals("Failed to remove secondary  ip address for the nic", exception.getLocalizedMessage());
         }
     }
+
+    private static NetworkService createMockNetworkService(boolean releaseSecondaryIpFromNicResult) {
+        NetworkService networkService = Mockito.mock(NetworkService.class);
+        Mockito.when(networkService.releaseSecondaryIpFromNic(ArgumentMatchers.anyInt()))
+                .thenReturn(releaseSecondaryIpFromNicResult);
+        return networkService;
+    }
 }
