--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -220,6 +220,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenServiceImplTest {
     @Mock
+    private SecurityGroup securityGroup;
+    @Mock
     MessageBus messageBus;
     @Mock
     ProjectDao projectDao;
@@ -682,7 +684,6 @@
 
     @Test
     public void createTungstenSecurityGroupTest() {
-        SecurityGroup securityGroup = mock(SecurityGroup.class);
         ProjectVO projectVO = mock(ProjectVO.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         DomainVO domainVO = mock(DomainVO.class);
@@ -704,7 +705,6 @@
 
     @Test
     public void deleteTungstenSecurityGroupTest() {
-        SecurityGroup securityGroup = mock(SecurityGroup.class);
         TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenAnswer deleteTungstenSecurityGroupAnswer = MockTungstenAnswerFactory.get(true);
 
