--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -72,8 +72,7 @@
 
     @Test
     public void testProcessWithEmptyPlan() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = MockVirtualMachine.createMockVirtualMachine(vmId);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
@@ -102,8 +101,7 @@
 
     @Test
     public void testProcessWithPlan() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = MockVirtualMachine.createMockVirtualMachine(vmId);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
@@ -139,8 +137,7 @@
 
     @Test
     public void testProcessWithNotRunningVM() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = MockVirtualMachine.createMockVirtualMachine(vmId);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
@@ -170,4 +167,15 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(1), plan.getHostPriorities().get(host2Id));
     }
+
+    public final class MockVirtualMachine {
+        private MockVirtualMachine() {
+        }
+
+        public static VirtualMachine createMockVirtualMachine(long vmId) {
+            VirtualMachine vm = Mockito.mock(VirtualMachine.class);
+            when(vm.getId()).thenReturn(vmId);
+            return vm;
+        }
+    }
 }
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -71,8 +71,7 @@
 
     @Test
     public void testProcessWithEmptyPlan() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = MockVirtualMachine.createMockVirtualMachine(vmId);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
@@ -101,8 +100,7 @@
 
     @Test
     public void testProcessWithPlan() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = MockVirtualMachine.createMockVirtualMachine(vmId);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
@@ -138,8 +136,7 @@
 
     @Test
     public void testProcessWithNotRunningVM() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = MockVirtualMachine.createMockVirtualMachine(vmId);
         VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
@@ -169,4 +166,15 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(-1), plan.getHostPriorities().get(host2Id));
     }
+
+    public final class MockVirtualMachine {
+        private MockVirtualMachine() {
+        }
+
+        public static VirtualMachine createMockVirtualMachine(long vmId) {
+            VirtualMachine vm = Mockito.mock(VirtualMachine.class);
+            when(vm.getId()).thenReturn(vmId);
+            return vm;
+        }
+    }
 }
