--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -862,11 +862,10 @@
         String tag1 = "tag1";
         String tag2 = "tag2";
         VMInstanceVO vmInstance = mock(VMInstanceVO.class);
-        HostVO host = mock(HostVO.class);
+        HostVO host = createHostVO();
         when(vmInstance.getHostId()).thenReturn(null);
         when(vmInstance.getLastHostId()).thenReturn(1L);
         when(hostDao.findById(1L)).thenReturn(host);
-        when(host.getClusterId()).thenReturn(1L);
         when(hostTagsDao.listByClusterId(1L)).thenReturn(Arrays.asList(tag1, tag2));
 
         List<String> hostTags = new ArrayList<>(Collections.singleton(tag1));
@@ -879,11 +878,10 @@
     public void testAddVmCurrentClusterHostTagsEmptyHostTagsInCluster() {
         String tag1 = "tag1";
         VMInstanceVO vmInstance = mock(VMInstanceVO.class);
-        HostVO host = mock(HostVO.class);
+        HostVO host = createHostVO();
         when(vmInstance.getHostId()).thenReturn(null);
         when(vmInstance.getLastHostId()).thenReturn(1L);
         when(hostDao.findById(1L)).thenReturn(host);
-        when(host.getClusterId()).thenReturn(1L);
         when(hostTagsDao.listByClusterId(1L)).thenReturn(null);
 
         List<String> hostTags = new ArrayList<>(Collections.singleton(tag1));
@@ -891,4 +889,10 @@
         assertEquals(1, hostTags.size());
         assertTrue(hostTags.contains(tag1));
     }
+
+    private static HostVO createHostVO() {
+        HostVO host = mock(HostVO.class);
+        when(host.getClusterId()).thenReturn(1L);
+        return host;
+    }
 }
