--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -1786,8 +1786,7 @@
         RunCustomActionAnswer answer = mock(RunCustomActionAnswer.class);
         when(answer.getResult()).thenReturn(true);
 
-        VirtualMachine vm = mock(VirtualMachine.class);
-        when(vm.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
+        VirtualMachine vm = createVirtualMachineMock();
         when(entityManager.findByUuid(eq(VirtualMachine.class), anyString())).thenReturn(vm);
         when(virtualMachineManager.findClusterAndHostIdForVm(vm, false)).thenReturn(new Pair<>(1L, 1L));
 
@@ -1885,8 +1884,7 @@
         when(extensionDao.findById(anyLong())).thenReturn(extensionVO);
         when(extensionVO.getState()).thenReturn(Extension.State.Enabled);
 
-        VirtualMachine vm = mock(VirtualMachine.class);
-        when(vm.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
+        VirtualMachine vm = createVirtualMachineMock();
         when(entityManager.findByUuid(eq(VirtualMachine.class), anyString())).thenReturn(vm);
         when(virtualMachineManager.findClusterAndHostIdForVm(vm, false)).thenReturn(new Pair<>(1L, 1L));
 
@@ -3062,4 +3060,10 @@
         assertTrue(resp.getResult().get(ApiConstants.DETAILS).contains("Action failed"));
     }
 
+
+    private static VirtualMachine createVirtualMachineMock() {
+        VirtualMachine vm = mock(VirtualMachine.class);
+        when(vm.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
+        return vm;
+    }
 }
