--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -140,6 +140,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ManagementServerImplTest {
+    @Mock
+    private UserDataVO userData;
 
     @Mock
     SearchCriteria<IPAddressVO> sc;
@@ -494,7 +496,6 @@
             when(cmd.getUserData()).thenReturn("testUserdata");
             when(cmd.getName()).thenReturn("testName");
 
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
             when(userDataDao.findByName(account.getAccountId(), account.getDomainId(), "testName")).thenReturn(null);
             when(userDataDao.findByUserData(account.getAccountId(), account.getDomainId(), "testUserdata")).thenReturn(userData);
 
@@ -515,7 +516,6 @@
             RegisterUserDataCmd cmd = Mockito.mock(RegisterUserDataCmd.class);
             when(cmd.getName()).thenReturn("testName");
 
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
             when(userDataDao.findByName(account.getAccountId(), account.getDomainId(), "testName")).thenReturn(userData);
 
             spy.registerUserData(cmd);
@@ -616,7 +616,6 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getId()).thenReturn(1L);
             when(cmd.isRecursive()).thenReturn(false);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
@@ -649,8 +648,8 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getName()).thenReturn("testSearchUserdataName");
             when(cmd.isRecursive()).thenReturn(false);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
-
+
+            // removed local mock; replaced with global field `userData`
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
             when(sb.entity()).thenReturn(userData);
@@ -682,7 +681,6 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getKeyword()).thenReturn("testSearchUserdataKeyword");
             when(cmd.isRecursive()).thenReturn(false);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
