--- a/plugins/backup/nas/src/test/java/org/apache/cloudstack/backup/NASBackupProviderTest.java
+++ b/plugins/backup/nas/src/test/java/org/apache/cloudstack/backup/NASBackupProviderTest.java
@@ -235,11 +235,9 @@
         Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         Mockito.when(hostDao.findById(hostId)).thenReturn(host);
 
-        VolumeVO volume1 = mock(VolumeVO.class);
-        Mockito.when(volume1.getState()).thenReturn(Volume.State.Ready);
+        VolumeVO volume1 = MockVolumeVO.createMockVolume();
         Mockito.when(volume1.getSize()).thenReturn(100L);
-        VolumeVO volume2 = mock(VolumeVO.class);
-        Mockito.when(volume2.getState()).thenReturn(Volume.State.Ready);
+        VolumeVO volume2 = MockVolumeVO.createMockVolume();
         Mockito.when(volume2.getSize()).thenReturn(200L);
         Mockito.when(volumeDao.findByInstance(vmId)).thenReturn(List.of(volume1, volume2));
 
@@ -1034,4 +1032,16 @@
         Mockito.verify(backupDao, Mockito.never()).remove(51L);
         Mockito.verify(backupDao).remove(50L);
     }
+
+    public final class MockVolumeVO {
+
+        private MockVolumeVO() {
+        }
+
+        public static VolumeVO createMockVolume() {
+            VolumeVO volume = Mockito.mock(VolumeVO.class);
+            Mockito.when(volume.getState()).thenReturn(Volume.State.Ready);
+            return volume;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -715,9 +715,8 @@
         when(backupScheduleDao.findById(scheduleId)).thenReturn(schedule);
         when(schedule.getMaxBackups()).thenReturn(2);
 
-        VolumeVO volume = mock(VolumeVO.class);
+        VolumeVO volume = MockVolumeVO.createMockVolume();
         when(volumeDao.findByInstance(vmId)).thenReturn(List.of(volume));
-        when(volume.getState()).thenReturn(Volume.State.Ready);
         when(volumeApiService.getVolumePhysicalSize(null, null, null)).thenReturn(newBackupSize);
 
         BackupProvider backupProvider = mock(BackupProvider.class);
@@ -820,9 +819,8 @@
         when(vm.getBackupOfferingId()).thenReturn(backupOfferingId);
         when(vm.getAccountId()).thenReturn(accountId);
 
-        VolumeVO volume = mock(VolumeVO.class);
+        VolumeVO volume = MockVolumeVO.createMockVolume();
         when(volumeDao.findByInstance(vmId)).thenReturn(List.of(volume));
-        when(volume.getState()).thenReturn(Volume.State.Ready);
         when(volumeApiService.getVolumePhysicalSize(null, null, null)).thenReturn(size);
 
         overrideBackupFrameworkConfigValue();
@@ -2874,4 +2872,16 @@
         verify(backupOfferingDao).persist(any());
         verify(backupOfferingDetailsDao, never()).saveDetails(any());
     }
+
+    public final class MockVolumeVO {
+
+        private MockVolumeVO() {
+        }
+
+        public static VolumeVO createMockVolume() {
+            VolumeVO volume = Mockito.mock(VolumeVO.class);
+            Mockito.when(volume.getState()).thenReturn(Volume.State.Ready);
+            return volume;
+        }
+    }
 }
