--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -621,8 +621,7 @@
         Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
         Mockito.when(network.getVpcId()).thenReturn(null);
         Mockito.when(ipAddressDao.listByAssociatedNetwork(Mockito.anyLong(), Mockito.anyBoolean())).thenReturn(List.of(Mockito.mock(IPAddressVO.class)));
-        VlanVO vlanVO = Mockito.mock(VlanVO.class);
-        Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
+        VlanVO vlanVO = createVlanVO(vlan);
         Mockito.when(vlanDao.findById(Mockito.anyLong())).thenReturn(vlanVO);
         Mockito.when(vlanDao.listIpv6RangeByZoneIdAndVlanId(Mockito.anyLong(), Mockito.anyString())).thenReturn(new ArrayList<>());
         DataCenterVO zoneMock = Mockito.mock(DataCenterVO.class);
@@ -642,8 +641,7 @@
         Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
         Mockito.when(network.getVpcId()).thenReturn(null);
         Mockito.when(ipAddressDao.listByAssociatedNetwork(Mockito.anyLong(), Mockito.anyBoolean())).thenReturn(List.of(Mockito.mock(IPAddressVO.class)));
-        VlanVO vlanVO = Mockito.mock(VlanVO.class);
-        Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
+        VlanVO vlanVO = createVlanVO(vlan);
         Mockito.when(vlanDao.findById(Mockito.anyLong())).thenReturn(vlanVO);
         Mockito.when(vlanDao.listIpv6RangeByZoneIdAndVlanId(zoneId, vlan)).thenReturn(List.of(vlanVO));
         try {
@@ -801,4 +799,10 @@
         ipv6Service.removePublicIpv6PlaceholderNics(network);
         Assert.assertEquals(0, removedNics.size());
     }
+
+    private static VlanVO createVlanVO(String vlan) {
+        VlanVO vlanVO = Mockito.mock(VlanVO.class);
+        Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
+        return vlanVO;
+    }
 }
