--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -220,6 +220,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenServiceImplTest {
     @Mock
+    private NetworkVO networkVO;
+    @Mock
     MessageBus messageBus;
     @Mock
     ProjectDao projectDao;
@@ -319,7 +321,6 @@
     @Test
     public void createTungstenFloatingIpTest() throws Exception {
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        NetworkVO networkVO = mock(NetworkVO.class);
         TungstenAnswer createTungstenFloatingIpAnswer = MockTungstenAnswerFactory.get(true);
         Ip ip = mock(Ip.class);
 
@@ -333,7 +334,7 @@
     @Test
     public void deleteTungstenFloatingIpWithIpAddressTest() throws Exception {
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        NetworkVO networkVO = mock(NetworkVO.class);
+        // removed local mock; replaced with global field `networkVO`
         TungstenAnswer deleteTungstenFloatingIpAnswer = MockTungstenAnswerFactory.get(true);
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(networkVO);
@@ -927,7 +928,6 @@
 
     @Test
     public void listTungstenPolicyTest() {
-        NetworkVO networkVO = mock(NetworkVO.class);
         TungstenAnswer listTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenNetworkPolicy tungstenNetworkPolicy = mock(TungstenNetworkPolicy.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
@@ -1604,7 +1604,6 @@
 
     @Test
     public void addNetworkGatewayToLogicalRouterTest() {
-        NetworkVO networkVO = mock(NetworkVO.class);
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(true);
         TungstenLogicalRouter tungstenLogicalRouter = mock(TungstenLogicalRouter.class);
         LogicalRouter logicalRouter = mock(LogicalRouter.class);
@@ -1641,7 +1640,6 @@
 
     @Test
     public void removeNetworkGatewayFromLogicalRouterTest() {
-        NetworkVO networkVO = mock(NetworkVO.class);
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(true);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenLogicalRouter tungstenLogicalRouter = mock(TungstenLogicalRouter.class);
