--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
@@ -160,8 +160,7 @@
 
         Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("d1");
-        Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("a1");
+        Account acc = MockAccount.createMockAccount("a1");
         ReservationContext context = mock(ReservationContext.class);
         when(context.getDomain()).thenReturn(dom);
         when(context.getAccount()).thenReturn(acc);
@@ -406,4 +405,15 @@
 
         assertTrue(_element.applyStaticNats(network, rules));
     }
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount(String accountName) {
+            com.cloud.user.Account account = Mockito.mock(com.cloud.user.Account.class);
+            when(account.getAccountName()).thenReturn(accountName);
+            return account;
+        }
+    }
 }
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -180,8 +180,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount("accountname");
         final ReservationContext context = mock(ReservationContext.class);
         when(context.getDomain()).thenReturn(dom);
         when(context.getAccount()).thenReturn(acc);
@@ -296,8 +295,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount("accountname");
         final ReservationContext context = mock(ReservationContext.class);
         when(context.getDomain()).thenReturn(dom);
         when(context.getAccount()).thenReturn(acc);
@@ -349,8 +347,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount("accountname");
         final ReservationContext context = mock(ReservationContext.class);
         when(context.getDomain()).thenReturn(dom);
         when(context.getAccount()).thenReturn(acc);
@@ -401,8 +398,7 @@
 
         final Domain dom = mock(Domain.class);
         when(dom.getName()).thenReturn("domain");
-        final Account acc = mock(Account.class);
-        when(acc.getAccountName()).thenReturn("accountname");
+        final Account acc = MockAccount.createMockAccount("accountname");
         final ReservationContext context = mock(ReservationContext.class);
         when(context.getDomain()).thenReturn(dom);
         when(context.getAccount()).thenReturn(acc);
@@ -418,4 +414,15 @@
 
         element.implement(network, offering, dest, context);
     }
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount(String accountName) {
+            com.cloud.user.Account account = Mockito.mock(com.cloud.user.Account.class);
+            when(account.getAccountName()).thenReturn(accountName);
+            return account;
+        }
+    }
 }
