--- a/plugins/storage/object/ceph/src/test/java/org/apache/cloudstack/storage/datastore/driver/CephObjectStoreDriverImplTest.java
+++ b/plugins/storage/object/ceph/src/test/java/org/apache/cloudstack/storage/datastore/driver/CephObjectStoreDriverImplTest.java
@@ -91,8 +91,7 @@
     @Test
     public void testCreateBucket() throws Exception {
         doReturn(rgwClient).when(cephObjectStoreDriverImpl).getS3Client(anyLong(), anyLong());
-        when(accountDetailsDao.findDetail(anyLong(),anyString())).
-                thenReturn(new AccountDetailVO(1L, "abc","def"));
+        accountDetailsDao = MockAccountDetailsDao.createMockAccountDetailsDao();
         when(bucketDao.findById(anyLong())).thenReturn(new BucketVO(bucket.getName()));
         Bucket bucketRet = cephObjectStoreDriverImpl.createBucket(bucket, false);
         assertEquals(bucketRet.getName(), bucket.getName());
@@ -109,4 +108,18 @@
         assertTrue(success);
         verify(rgwAdmin, times(1)).removeBucket(anyString());
     }
+
+    public final class MockAccountDetailsDao {
+
+        private MockAccountDetailsDao() {
+            // Utility class
+        }
+
+        public static AccountDetailsDao createMockAccountDetailsDao() {
+            AccountDetailsDao accountDetailsDao = Mockito.mock(AccountDetailsDao.class);
+            when(accountDetailsDao.findDetail(anyLong(), anyString()))
+                    .thenReturn(new AccountDetailVO(1L, "abc", "def"));
+            return accountDetailsDao;
+        }
+    }
 }
--- a/plugins/storage/object/minio/src/test/java/org/apache/cloudstack/storage/datastore/driver/MinIOObjectStoreDriverImplTest.java
+++ b/plugins/storage/object/minio/src/test/java/org/apache/cloudstack/storage/datastore/driver/MinIOObjectStoreDriverImplTest.java
@@ -117,8 +117,7 @@
         when(bucketDao.listByObjectStoreIdAndAccountId(anyLong(), anyLong())).thenReturn(new ArrayList<BucketVO>());
         when(account.getUuid()).thenReturn(UUID.randomUUID().toString());
         when(accountDao.findById(anyLong())).thenReturn(account);
-        when(accountDetailsDao.findDetail(anyLong(),anyString())).
-                thenReturn(new AccountDetailVO(1L, "abc","def"));
+        accountDetailsDao = MockAccountDetailsDao.createMockAccountDetailsDao();
         when(bucketDao.findById(anyLong())).thenReturn(new BucketVO());
         Bucket bucketRet = minioObjectStoreDriverImpl.createBucket(bucket, false);
         assertEquals(bucketRet.getName(), bucket.getName());
@@ -166,4 +165,18 @@
         assertEquals(accessKey, persistedMap.get(MinIOObjectStoreDriverImpl.MINIO_ACCESS_KEY));
         assertEquals(secretKey, persistedMap.get(MinIOObjectStoreDriverImpl.MINIO_SECRET_KEY));
     }
+
+    public final class MockAccountDetailsDao {
+
+        private MockAccountDetailsDao() {
+            // Utility class
+        }
+
+        public static AccountDetailsDao createMockAccountDetailsDao() {
+            AccountDetailsDao accountDetailsDao = Mockito.mock(AccountDetailsDao.class);
+            when(accountDetailsDao.findDetail(anyLong(), anyString()))
+                    .thenReturn(new AccountDetailVO(1L, "abc", "def"));
+            return accountDetailsDao;
+        }
+    }
 }
