--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -292,9 +292,8 @@
         Mockito.when(hostDao.findById(ArgumentMatchers.anyLong())).thenReturn(hostVO);
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
         Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
-        DeleteVMSnapshotAnswer answer = Mockito.mock(DeleteVMSnapshotAnswer.class);
-        Mockito.when(answer.getResult()).thenReturn(true);
-
+
+        DeleteVMSnapshotAnswer answer = MockDeleteVMSnapshotAnswer.createMockDeleteVMSnapshotAnswer();
         boolean result = vmStrategy.deleteVMSnapshot(vmSnapshot);
         assertTrue(result);
     }
@@ -449,4 +448,17 @@
             return Mockito.mock(SnapshotDataStoreDao.class);
         }
     }
+
+    public final class MockDeleteVMSnapshotAnswer {
+
+        private MockDeleteVMSnapshotAnswer() {
+            // Utility class
+        }
+
+        public static DeleteVMSnapshotAnswer createMockDeleteVMSnapshotAnswer() {
+            DeleteVMSnapshotAnswer answer = Mockito.mock(DeleteVMSnapshotAnswer.class);
+            Mockito.when(answer.getResult()).thenReturn(true);
+            return answer;
+        }
+    }
 }
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
@@ -240,8 +240,7 @@
 
         assertNotNull(e);
 
-        DeleteVMSnapshotAnswer answer = Mockito.mock(DeleteVMSnapshotAnswer.class);
-        Mockito.when(answer.getResult()).thenReturn(true);
+        DeleteVMSnapshotAnswer answer = MockDeleteVMSnapshotAnswer.createMockDeleteVMSnapshotAnswer();
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(answer);
 
         boolean result = vmSnapshotStrategy.deleteVMSnapshot(vmSnapshot);
@@ -342,4 +341,17 @@
             return Mockito.mock(BackupManager.class);
         }
     }
+
+    public final class MockDeleteVMSnapshotAnswer {
+
+        private MockDeleteVMSnapshotAnswer() {
+            // Utility class
+        }
+
+        public static DeleteVMSnapshotAnswer createMockDeleteVMSnapshotAnswer() {
+            DeleteVMSnapshotAnswer answer = Mockito.mock(DeleteVMSnapshotAnswer.class);
+            Mockito.when(answer.getResult()).thenReturn(true);
+            return answer;
+        }
+    }
 }
