--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -183,12 +183,11 @@
         DataCenterVO mockDc = mock(DataCenterVO.class);
         ExcludeList avoids = mock(ExcludeList.class);
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
-        VMInstanceVO vm = mock(VMInstanceVO.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
 
         when(avoids.shouldAvoid(mockDc)).thenReturn(true);
+        VMInstanceVO vm = createMockVmInstanceVo(1L);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
-        when(vm.getDataCenterId()).thenReturn(1L);
         when(dcDao.findById(1L)).thenReturn(mockDc);
 
         List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
@@ -325,7 +324,7 @@
 
     private void initializeForTest(VirtualMachineProfileImpl vmProfile, DataCenterDeployment plan) {
         DataCenterVO mockDc = mock(DataCenterVO.class);
-        VMInstanceVO vm = mock(VMInstanceVO.class);
+        VMInstanceVO vm = createMockVmInstanceVo(dataCenterId);
         UserVmVO userVm = mock(UserVmVO.class);
         ServiceOfferingVO offering = mock(ServiceOfferingVO.class);
 
@@ -338,7 +337,6 @@
         when(vmDao.findById(12L)).thenReturn(userVm);
         when(userVm.getAccountId()).thenReturn(accountId);
 
-        when(vm.getDataCenterId()).thenReturn(dataCenterId);
         when(dcDao.findById(1L)).thenReturn(mockDc);
         when(plan.getDataCenterId()).thenReturn(dataCenterId);
         when(plan.getClusterId()).thenReturn(null);
@@ -570,4 +568,10 @@
             }
         }
     }
+
+    private static VMInstanceVO createMockVmInstanceVo(long dataCenterId) {
+        VMInstanceVO vm = mock(VMInstanceVO.class);
+        when(vm.getDataCenterId()).thenReturn(dataCenterId);
+        return vm;
+    }
 }
