--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -209,10 +209,9 @@
         remoteHostEndPointMock.when(() -> RemoteHostEndPoint.getHypervisorHostEndPoint(host)).thenReturn(ep);
 
         long volumeVirtualSize = 68673077248L;
-        DataTO dataTO = Mockito.mock(DataTO.class);
+        DataTO dataTO = createMockDataTO("bec0ba7700000007:vol-11-6aef-10ee");
         CreateObjectAnswer createAnswer = new CreateObjectAnswer(dataTO);
         doReturn(createAnswer).when(scaleIOPrimaryDataStoreDriver).createVolume(destData, 2L, true, volumeVirtualSize);
-        when(dataTO.getPath()).thenReturn("bec0ba7700000007:vol-11-6aef-10ee");
         doReturn(true).when(scaleIOPrimaryDataStoreDriver)
                 .grantAccess(any(), any(), any());
 
@@ -451,10 +450,9 @@
         String srcVolumePath = "bec0ba7700000007:vol-11-6aef-10ee";
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataTO volumeTO = Mockito.mock(DataTO.class);
+        DataTO volumeTO = createMockDataTO(srcVolumePath);
         when(srcData.getDataStore()).thenReturn(srcStore);
         when(srcData.getTO()).thenReturn(volumeTO);
-        when(volumeTO.getPath()).thenReturn(srcVolumePath);
         doNothing().when(scaleIOPrimaryDataStoreDriver).revokeVolumeAccess(any(), any(), any());
 
         ScaleIOGatewayClient client = Mockito.mock(ScaleIOGatewayClient.class);
@@ -474,10 +472,9 @@
         String srcVolumePath = "bec0ba7700000007:vol-11-6aef-10ee";
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataTO volumeTO = Mockito.mock(DataTO.class);
+        DataTO volumeTO = createMockDataTO(srcVolumePath);
         when(srcData.getDataStore()).thenReturn(srcStore);
         when(srcData.getTO()).thenReturn(volumeTO);
-        when(volumeTO.getPath()).thenReturn(srcVolumePath);
         String sdcId = "7332760565f6340f";
         doReturn(sdcId).when(scaleIOPrimaryDataStoreDriver).getConnectedSdc(srcStore, host);
 
@@ -500,10 +497,9 @@
         String srcVolumePath = "bec0ba7700000007:vol-11-6aef-10ee";
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataTO volumeTO = Mockito.mock(DataTO.class);
         when(srcData.getDataStore()).thenReturn(srcStore);
+        DataTO volumeTO = createMockDataTO(srcVolumePath);
         when(srcData.getTO()).thenReturn(volumeTO);
-        when(volumeTO.getPath()).thenReturn(srcVolumePath);
         String sdcId = "7332760565f6340f";
         doReturn(null).when(scaleIOPrimaryDataStoreDriver).getConnectedSdc(srcStore, host);
 
@@ -597,4 +593,10 @@
                         16L * (1024 * 1024 * 1024),
                         true));
     }
+
+    private static DataTO createMockDataTO(String path) {
+        DataTO dataTO = Mockito.mock(DataTO.class);
+        when(dataTO.getPath()).thenReturn(path);
+        return dataTO;
+    }
 }
