--- a/server/src/test/java/com/cloud/ha/KVMFencerTest.java
+++ b/server/src/test/java/com/cloud/ha/KVMFencerTest.java
@@ -105,10 +105,8 @@
         Mockito.lenient().when(host.getPodId()).thenReturn(1l);
         Mockito.when(host.getId()).thenReturn(1l);
 
-        HostVO secondHost = Mockito.mock(HostVO.class);
+        HostVO secondHost = createMockHostVo();
         Mockito.lenient().when(secondHost.getClusterId()).thenReturn(1l);
-        Mockito.when(secondHost.getHypervisorType()).thenReturn(HypervisorType.KVM);
-        Mockito.when(secondHost.getStatus()).thenReturn(Status.Up);
         Mockito.lenient().when(secondHost.getDataCenterId()).thenReturn(1l);
         Mockito.lenient().when(secondHost.getPodId()).thenReturn(1l);
         Mockito.when(host.getId()).thenReturn(2l);
@@ -133,10 +131,8 @@
         Mockito.when(host.getPodId()).thenReturn(1l);
         Mockito.when(host.getId()).thenReturn(1l);
 
-        HostVO secondHost = Mockito.mock(HostVO.class);
+        HostVO secondHost = createMockHostVo();
         Mockito.lenient().when(secondHost.getClusterId()).thenReturn(1l);
-        Mockito.when(secondHost.getHypervisorType()).thenReturn(HypervisorType.KVM);
-        Mockito.when(secondHost.getStatus()).thenReturn(Status.Up);
         Mockito.lenient().when(secondHost.getDataCenterId()).thenReturn(1l);
         Mockito.lenient().when(secondHost.getPodId()).thenReturn(1l);
         Mockito.when(host.getId()).thenReturn(2l);
@@ -160,10 +156,8 @@
         Mockito.when(host.getPodId()).thenReturn(1l);
         Mockito.when(host.getId()).thenReturn(1l);
 
-        HostVO secondHost = Mockito.mock(HostVO.class);
+        HostVO secondHost = createMockHostVo();
         Mockito.lenient().when(secondHost.getClusterId()).thenReturn(1l);
-        Mockito.when(secondHost.getHypervisorType()).thenReturn(HypervisorType.KVM);
-        Mockito.when(secondHost.getStatus()).thenReturn(Status.Up);
         Mockito.lenient().when(secondHost.getDataCenterId()).thenReturn(1l);
         Mockito.lenient().when(secondHost.getPodId()).thenReturn(1l);
         Mockito.when(host.getId()).thenReturn(2l);
@@ -192,4 +186,11 @@
         Assert.assertNull(fencer.fenceOff(virtualMachine, host));
     }
 
+
+    private static HostVO createMockHostVo() {
+        HostVO host = Mockito.mock(HostVO.class);
+        Mockito.when(host.getHypervisorType()).thenReturn(HypervisorType.KVM);
+        Mockito.when(host.getStatus()).thenReturn(Status.Up);
+        return host;
+    }
 }
