--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -995,10 +995,8 @@
 
     @Test
     public void testValidateAutoScaleCounters() {
-        Counter counterCpuMock = Mockito.mock(Counter.class);
-        when(counterCpuMock.getSource()).thenReturn(Counter.Source.CPU);
-        Counter counterMemoryMock = Mockito.mock(Counter.class);
-        when(counterMemoryMock.getSource()).thenReturn(Counter.Source.MEMORY);
+        Counter counterCpuMock = createMockCounter(Counter.Source.CPU);
+        Counter counterMemoryMock = createMockCounter(Counter.Source.MEMORY);
 
         List<Counter> counters = Arrays.asList(counterCpuMock, counterMemoryMock);
 
@@ -1014,10 +1012,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateAutoScaleCountersFail() {
-        Counter counterCpuMock = Mockito.mock(Counter.class);
-        when(counterCpuMock.getSource()).thenReturn(Counter.Source.CPU);
-        Counter counterMemoryMock = Mockito.mock(Counter.class);
-        when(counterMemoryMock.getSource()).thenReturn(Counter.Source.MEMORY);
+        Counter counterCpuMock = createMockCounter(Counter.Source.CPU);
+        Counter counterMemoryMock = createMockCounter(Counter.Source.MEMORY);
 
         List<Counter> counters = Arrays.asList(counterCpuMock, counterMemoryMock);
 
@@ -2581,4 +2577,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static Counter createMockCounter(Counter.Source source) {
+        Counter counterMock = Mockito.mock(Counter.class);
+        when(counterMock.getSource()).thenReturn(source);
+        return counterMock;
+    }
 }
