--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1458,11 +1458,9 @@
 
     @Test
     public void getVmSshKeyPairs1() {
-        SSHKeyPairVO keypair1 = Mockito.mock(SSHKeyPairVO.class);
-        SSHKeyPairVO keypair2 = Mockito.mock(SSHKeyPairVO.class);
+        SSHKeyPairVO keypair2 = createMockSshKeyPairVO("name2");
+        SSHKeyPairVO keypair1 = createMockSshKeyPairVO("name1");
         when(sshKeyPairDao.findByName(anyLong(), anyLong(), anyString())).thenReturn(keypair1).thenReturn(keypair2);
-        when(keypair1.getName()).thenReturn("name1");
-        when(keypair2.getName()).thenReturn("name2");
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("keypairs", "keypair1,keypair2");
@@ -2581,4 +2579,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static SSHKeyPairVO createMockSshKeyPairVO(String name) {
+        SSHKeyPairVO keyPair = Mockito.mock(SSHKeyPairVO.class);
+        when(keyPair.getName()).thenReturn(name);
+        return keyPair;
+    }
 }
