--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -2367,21 +2367,17 @@
 
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO scaleUpConditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO scaleUpCounterTO = Mockito.mock(CounterTO.class);
         when(scaleUpPolicyTO.getId()).thenReturn(scaleUpPolicyId);
         when(scaleUpPolicyTO.getConditions()).thenReturn(Arrays.asList(scaleUpConditionTO));
+         CounterTO scaleUpCounterTO = createCounterTO(VirtualRouterAutoScaleCounter.NETWORK_RECEIVED_AVERAGE_MBPS.toString());
         when(scaleUpConditionTO.getCounter()).thenReturn(scaleUpCounterTO);
-        when(scaleUpCounterTO.getProvider()).thenReturn(Network.Provider.VirtualRouter.getName());
-        when(scaleUpCounterTO.getValue()).thenReturn(VirtualRouterAutoScaleCounter.NETWORK_RECEIVED_AVERAGE_MBPS.toString());
 
         AutoScalePolicyTO scaleDownPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO scaleDownConditionTO = Mockito.mock(ConditionTO.class);
-        CounterTO scaleDownCounterTO = Mockito.mock(CounterTO.class);
         when(scaleDownPolicyTO.getId()).thenReturn(scaleDownPolicyId);
         when(scaleDownPolicyTO.getConditions()).thenReturn(Arrays.asList(scaleDownConditionTO));
+         CounterTO scaleDownCounterTO = createCounterTO(VirtualRouterAutoScaleCounter.NETWORK_TRANSMIT_AVERAGE_MBPS.toString());
         when(scaleDownConditionTO.getCounter()).thenReturn(scaleDownCounterTO);
-        when(scaleDownCounterTO.getProvider()).thenReturn(Network.Provider.VirtualRouter.getName());
-        when(scaleDownCounterTO.getValue()).thenReturn(VirtualRouterAutoScaleCounter.NETWORK_TRANSMIT_AVERAGE_MBPS.toString());
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(scaleUpPolicyTO, scaleDownPolicyTO));
 
@@ -2581,4 +2577,11 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static CounterTO createCounterTO(String value) {
+        CounterTO counterTO = Mockito.mock(CounterTO.class);
+        when(counterTO.getProvider()).thenReturn(Network.Provider.VirtualRouter.getName());
+        when(counterTO.getValue()).thenReturn(value);
+        return counterTO;
+    }
 }
