--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -74,9 +74,8 @@
     public void testProcessWithEmptyPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
         when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
@@ -104,9 +103,8 @@
     public void testProcessWithPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
         when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
@@ -141,9 +139,8 @@
     public void testProcessWithNotRunningVM() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
         when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
@@ -170,4 +167,15 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(1), plan.getHostPriorities().get(host2Id));
     }
+
+    public final class MockVirtualMachineProfile {
+        private MockVirtualMachineProfile() {
+        }
+
+        public static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachine vm) {
+            VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
+            when(vmProfile.getVirtualMachine()).thenReturn(vm);
+            return vmProfile;
+        }
+    }
 }
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -73,9 +73,8 @@
     public void testProcessWithEmptyPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
         when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
@@ -103,9 +102,8 @@
     public void testProcessWithPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
         when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
@@ -140,9 +138,8 @@
     public void testProcessWithNotRunningVM() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
 
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
         when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
@@ -169,4 +166,15 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(-1), plan.getHostPriorities().get(host2Id));
     }
+
+    public final class MockVirtualMachineProfile {
+        private MockVirtualMachineProfile() {
+        }
+
+        public static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachine vm) {
+            VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
+            when(vmProfile.getVirtualMachine()).thenReturn(vm);
+            return vmProfile;
+        }
+    }
 }
