--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -249,9 +249,8 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -308,9 +307,8 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -362,9 +360,8 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -414,9 +411,8 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -459,9 +455,8 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDevice();
         final List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -495,9 +490,8 @@
 
         when(netmodel.findPhysicalNetworkId(anyLong(), (String)any(), (TrafficType)any())).thenReturn(NETWORK_ID);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -534,9 +528,8 @@
         when(mapping.getVlanId()).thenReturn(14);
         when(vcsmapdao.findByNetworkId(anyLong())).thenReturn(mapping);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         final List<BrocadeVcsDeviceVO> devices = new ArrayList();
         devices.add(brocadeDevice);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -577,9 +570,8 @@
         when(vcsmapdao.findByNetworkId(anyLong())).thenReturn(mapping);
         when(vcsmapdao.remove(anyLong())).thenReturn(true);
 
-        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
-        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
-
+
+        final BrocadeVcsDeviceVO brocadeDevice = createMockBrocadeVcsDeviceVO();
         final List<BrocadeVcsDeviceVO> devices = mock(List.class);
         when(devices.isEmpty()).thenReturn(true);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
@@ -588,4 +580,10 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    private static BrocadeVcsDeviceVO createMockBrocadeVcsDeviceVO() {
+        final BrocadeVcsDeviceVO brocadeDevice = mock(BrocadeVcsDeviceVO.class);
+        when(brocadeDevice.getHostId()).thenReturn(NETWORK_ID);
+        return brocadeDevice;
+    }
 }
