--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -126,6 +126,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class NetworkServiceImplTest {
     @Mock
+    private NetworkVO networkVO;
+    @Mock
     Object job;
     @Mock
     Object _responseObject;
@@ -687,7 +689,6 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testCheckAndUpdateNetworkDnsL2NetworkFailure() {
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
         NetworkOffering offering = Mockito.mock(NetworkOffering.class);
         Mockito.when(offering.getGuestType()).thenReturn(Network.GuestType.L2);
         service.checkAndUpdateNetworkDns(networkVO, offering, ip4Dns[0], null, null, null);
@@ -1259,7 +1260,6 @@
     @Test(expected = CloudRuntimeException.class)
     public void testGetNicVlanValueForExternalVm_Nsx_NoBean() {
         NicTO nic = mock(NicTO.class);
-        NetworkVO networkVO = mock(NetworkVO.class);
         String broadcastUri = "nsx://segment";
         when(nic.getBroadcastUri()).thenReturn(URI.create(broadcastUri));
         when(nic.getNetworkId()).thenReturn(42L);
