--- a/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/OAuth2UserAuthenticatorTest.java
+++ b/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/OAuth2UserAuthenticatorTest.java
@@ -51,6 +51,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class OAuth2UserAuthenticatorTest {
+    @Mock
+    private UserAccount userAccount;
 
     @Mock
     private UserAccountDao userAccountDao;
@@ -86,7 +88,6 @@
         String[] email = {"testemail"};
         String[] secretCode = {"testsecretcode"};
 
-        UserAccount userAccount = mock(UserAccount.class);
         UserVO user = mock(UserVO.class);
         UserOAuth2Authenticator userOAuth2Authenticator = mock(UserOAuth2Authenticator.class);
 
@@ -119,7 +120,6 @@
         String[] email = {"testemail"};
         String[] secretCode = {"testsecretcode"};
 
-        UserAccount userAccount = mock(UserAccount.class);
         UserVO user = mock(UserVO.class);
         UserOAuth2Authenticator userOAuth2Authenticator = mock(UserOAuth2Authenticator.class);
 
@@ -196,7 +196,6 @@
         String username = "testuser";
         Long domainId = 1L;
 
-        UserAccount userAccount = mock(UserAccount.class);
         when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccount);
         when(userDao.getUser(userAccount.getId())).thenReturn(mock(UserVO.class));
 
