--- a/server/src/test/java/org/apache/cloudstack/backup/InternalBackupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/InternalBackupServiceImplTest.java
@@ -63,6 +63,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class InternalBackupServiceImplTest {
+    @Mock
+    private Command cmdMock;
 
     @Mock
     private InternalBackupStoragePoolDao internalBackupStoragePoolDaoMock;
@@ -140,7 +142,6 @@
     @Test
     public void configureChainInfoTestNonVolumeObjectReturnsImmediately() {
         DataTO dataToMock = mock(DataTO.class);
-        Command cmdMock = mock(Command.class);
 
         internalBackupServiceImplSpy.configureChainInfo(dataToMock, cmdMock);
 
@@ -164,7 +165,6 @@
         doReturn(internalBackupStoragePoolVoMock).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(VOLUME_ID);
         doReturn("/path/to/parent").when(internalBackupStoragePoolVoMock).getBackupDeltaParentPath();
 
-        Command cmdMock = mock(Command.class);
 
         internalBackupServiceImplSpy.configureChainInfo(volumeObjectToMock, cmdMock);
 
