--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.engine.orchestration;
 
+import org.mockito.Mock;
 import static org.apache.cloudstack.engine.orchestration.service.NetworkOrchestrationService.NetworkLockTimeout;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
@@ -105,6 +106,8 @@
  */
 @RunWith(JUnit4.class)
 public class NetworkOrchestratorTest extends TestCase {
+    @Mock
+    private DataCenter dataCenter;
 
     NetworkOrchestrator testOrchestrator = Mockito.spy(new NetworkOrchestrator());
 
@@ -773,7 +776,6 @@
     @Test
     public void testGetGuestIpForNicImportL2Network() {
         Network network = Mockito.mock(Network.class);
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
         Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(network.getGuestType()).thenReturn(GuestType.L2);
         Assert.assertNull(testOrchestrator.getSelectedIpForNicImport(network, dataCenter, ipAddresses));
@@ -909,7 +911,6 @@
 
     @Test(expected = InsufficientVirtualNetworkCapacityException.class)
     public void testImportNicAcquireGuestIPFailed() throws Exception {
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
         VirtualMachine vm = mock(VirtualMachine.class);
         Network network = Mockito.mock(Network.class);
         Mockito.when(network.getGuestType()).thenReturn(GuestType.Isolated);
@@ -928,7 +929,6 @@
 
     @Test(expected = InsufficientVirtualNetworkCapacityException.class)
     public void testImportNicAutoAcquireGuestIPFailed() throws Exception {
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
         VirtualMachine vm = mock(VirtualMachine.class);
         Network network = Mockito.mock(Network.class);
         Mockito.when(network.getGuestType()).thenReturn(GuestType.Isolated);
@@ -947,7 +947,6 @@
 
     @Test
     public void testImportNicNoIP4Address() throws Exception {
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
         Long vmId = 1L;
         Hypervisor.HypervisorType hypervisorType = Hypervisor.HypervisorType.KVM;
         VirtualMachine vm = mock(VirtualMachine.class);
@@ -984,7 +983,6 @@
 
     @Test
     public void testImportNicWithIP4Address() throws Exception {
-        DataCenter dataCenter = Mockito.mock(DataCenter.class);
         Long vmId = 1L;
         Hypervisor.HypervisorType hypervisorType = Hypervisor.HypervisorType.KVM;
         VirtualMachine vm = mock(VirtualMachine.class);
