--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLManagerTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLManagerTest.java
@@ -15,6 +15,7 @@
 
 package com.cloud.network.vpc;
 
+import org.mockito.Mock;
 import com.cloud.configuration.ConfigurationManager;
 import com.cloud.network.Network;
 import com.cloud.network.NetworkModel;
@@ -73,6 +74,8 @@
 @RunWith(SpringJUnit4ClassRunner.class)
 @ContextConfiguration(loader = AnnotationConfigContextLoader.class)
 public class NetworkACLManagerTest extends TestCase {
+    @Mock
+    private final NetworkVO network;
     @Inject
     NetworkACLManagerImpl _aclMgr;
 
@@ -131,7 +134,6 @@
     @Test
     @SuppressWarnings("unchecked")
     public void testApplyACL() throws Exception {
-        final NetworkVO network = Mockito.mock(NetworkVO.class);
         Mockito.when(_networkDao.findById(anyLong())).thenReturn(network);
         Mockito.when(networkOfferingDao.isIpv6Supported(anyLong())).thenReturn(false);
         Mockito.when(_networkModel.isProviderSupportServiceInNetwork(anyLong(), ArgumentMatchers.any(Network.Service.class), ArgumentMatchers.any(Network.Provider.class))).thenReturn(true);
@@ -166,7 +168,6 @@
 
         // Make sure it is handled
         final long aclId = 1L;
-        final NetworkVO network = Mockito.mock(NetworkVO.class);
         final List<NetworkVO> networks = new ArrayList<>();
         networks.add(network);
 
