--- a/server/src/test/java/com/cloud/api/ApiServerTest.java
+++ b/server/src/test/java/com/cloud/api/ApiServerTest.java
@@ -64,6 +64,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ApiServerTest {
+    @Mock
+    private Domain domain;
 
     @InjectMocks
     ApiServer apiServer = new ApiServer();
@@ -142,7 +144,6 @@
     @Test(expected = CloudRuntimeException.class)
     public void testForgotPasswordFailureNoEmail() {
         UserAccount userAccount = mock(UserAccount.class);
-        Domain domain = mock(Domain.class);
 
         Mockito.when(userAccount.getEmail()).thenReturn("");
         apiServer.forgotPassword(userAccount, domain);
@@ -151,7 +152,6 @@
     @Test(expected = CloudRuntimeException.class)
     public void testForgotPasswordFailureDisabledUser() {
         UserAccount userAccount = mock(UserAccount.class);
-        Domain domain = mock(Domain.class);
 
         Mockito.when(userAccount.getEmail()).thenReturn("test@test.com");
         Mockito.when(userAccount.getState()).thenReturn("DISABLED");
@@ -161,7 +161,6 @@
     @Test(expected = CloudRuntimeException.class)
     public void testForgotPasswordFailureDisabledAccount() {
         UserAccount userAccount = mock(UserAccount.class);
-        Domain domain = mock(Domain.class);
 
         Mockito.when(userAccount.getEmail()).thenReturn("test@test.com");
         Mockito.when(userAccount.getState()).thenReturn("ENABLED");
