--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -392,16 +392,10 @@
         // Mock vms on each host.
         long offeringIdForVmsOfThisAccount = 15L;
         long offeringIdForVmsOfOtherAccount = 16L;
-        UserVmVO vm1 = mock(UserVmVO.class);
-        when(vm1.getAccountId()).thenReturn(accountId);
-        when(vm1.getServiceOfferingId()).thenReturn(offeringIdForVmsOfThisAccount);
-        UserVmVO vm2 = mock(UserVmVO.class);
-        when(vm2.getAccountId()).thenReturn(accountId);
-        when(vm2.getServiceOfferingId()).thenReturn(offeringIdForVmsOfThisAccount);
+        UserVmVO vm1 = createMockUserVmVO(accountId, offeringIdForVmsOfThisAccount);
+        UserVmVO vm2 = createMockUserVmVO(accountId, offeringIdForVmsOfThisAccount);
         // Vm from different account
-        UserVmVO vm3 = mock(UserVmVO.class);
-        when(vm3.getAccountId()).thenReturn(201L);
-        when(vm3.getServiceOfferingId()).thenReturn(offeringIdForVmsOfOtherAccount);
+        UserVmVO vm3 = createMockUserVmVO(201L, offeringIdForVmsOfOtherAccount);
         List<VMInstanceVO> vmsForHost1 = new ArrayList<VMInstanceVO>();
         List<VMInstanceVO> vmsForHost2 = new ArrayList<VMInstanceVO>();
         List<VMInstanceVO> vmsForHost3 = new ArrayList<VMInstanceVO>();
@@ -570,4 +564,11 @@
             }
         }
     }
+
+    private static UserVmVO createMockUserVmVO(long accountId, long serviceOfferingId) {
+        UserVmVO vm = mock(UserVmVO.class);
+        when(vm.getAccountId()).thenReturn(accountId);
+        when(vm.getServiceOfferingId()).thenReturn(serviceOfferingId);
+        return vm;
+    }
 }
