--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -124,6 +124,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class QueryManagerImplTest {
+    @Mock
+    private DomainVO domain;
     public static final long USER_ID = 1;
     public static final long ACCOUNT_ID = 1;
 
@@ -542,7 +544,6 @@
         Mockito.when(cmd.getUserSource()).thenReturn(userSource);
 
         UserAccountJoinVO user = new UserAccountJoinVO();
-        DomainVO domain = mock(DomainVO.class);
         SearchBuilder<UserAccountJoinVO> sb = mock(SearchBuilder.class);
         SearchCriteria<UserAccountJoinVO> sc = mock(SearchCriteria.class);
         List<UserAccountJoinVO> users = new ArrayList<>();
@@ -795,7 +796,6 @@
         Mockito.when(cmd.getAccountType()).thenReturn(accountType);
         Mockito.when(cmd.getApiKeyAccess()).thenReturn(apiKeyAccess);
 
-        DomainVO domain = mock(DomainVO.class);
         SearchBuilder<AccountVO> sb = mock(SearchBuilder.class);
         SearchCriteria<AccountVO> sc = mock(SearchCriteria.class);
         Pair<List<AccountVO>, Integer> uniqueAccountPair = new Pair<>(new ArrayList<>(), 0);
