--- a/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/OAuth2UserAuthenticatorTest.java
+++ b/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/OAuth2UserAuthenticatorTest.java
@@ -51,6 +51,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class OAuth2UserAuthenticatorTest {
+    @Mock
+    private UserVO user;
 
     @Mock
     private UserAccountDao userAccountDao;
@@ -87,7 +89,6 @@
         String[] secretCode = {"testsecretcode"};
 
         UserAccount userAccount = mock(UserAccount.class);
-        UserVO user = mock(UserVO.class);
         UserOAuth2Authenticator userOAuth2Authenticator = mock(UserOAuth2Authenticator.class);
 
         when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccount);
@@ -120,7 +121,7 @@
         String[] secretCode = {"testsecretcode"};
 
         UserAccount userAccount = mock(UserAccount.class);
-        UserVO user = mock(UserVO.class);
+        // removed local mock; replaced with global field `user`
         UserOAuth2Authenticator userOAuth2Authenticator = mock(UserOAuth2Authenticator.class);
 
         when(userAccountDao.getUserAccount(username, domainId)).thenReturn(userAccount);
