--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -71,6 +71,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class CommandSetupHelperTest {
+    @Mock
+    private ASNumberVO asNumberVO;
 
     @InjectMocks
     protected CommandSetupHelper commandSetupHelper = new CommandSetupHelper();
@@ -222,7 +224,6 @@
         when(router.getDataCenterId()).thenReturn(zoneId);
         when(router.getVpcId()).thenReturn(null);
         when(network.getId()).thenReturn(networkId);
-        ASNumberVO asNumberVO = Mockito.mock(ASNumberVO.class);
         when(asNumberDao.findByZoneAndNetworkId(zoneId, networkId)).thenReturn(asNumberVO);
         DataCenterVO dc = Mockito.mock(DataCenterVO.class);
         when(dcDao.findById(zoneId)).thenReturn(dc);
@@ -249,7 +250,6 @@
         long vpcId = 11L;
         when(router.getDataCenterId()).thenReturn(zoneId);
         when(router.getVpcId()).thenReturn(vpcId);
-        ASNumberVO asNumberVO = Mockito.mock(ASNumberVO.class);
         when(asNumberDao.findByZoneAndVpcId(zoneId, vpcId)).thenReturn(asNumberVO);
 
         long networkOfferingId = 12L;
