--- a/engine/schema/src/test/java/com/cloud/gpu/dao/GpuDeviceDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/gpu/dao/GpuDeviceDaoImplTest.java
@@ -202,12 +202,9 @@
 
     @Test
     public void getDistinctGpuCardIds() {
-        GpuDeviceVO device1 = mock(GpuDeviceVO.class);
-        GpuDeviceVO device2 = mock(GpuDeviceVO.class);
-        GpuDeviceVO device3 = mock(GpuDeviceVO.class);
-        when(device1.getCardId()).thenReturn(1L);
-        when(device2.getCardId()).thenReturn(2L);
-        when(device3.getCardId()).thenReturn(1L);
+        GpuDeviceVO device2 = createMockGpuDeviceVO(2L);
+        GpuDeviceVO device3 = createMockGpuDeviceVO(1L);
+        GpuDeviceVO device1 = createMockGpuDeviceVO(1L);
 
         doReturn(List.of(device1, device2, device3)).when(gpuDeviceDao).listBy(any(SearchCriteria.class));
 
@@ -274,4 +271,10 @@
         Assert.assertEquals("Expected correct where clause", "gpu_device.parent_gpu_device_id = ?",
                 scCaptor.getValue().getWhereClause().trim());
     }
+
+    private static GpuDeviceVO createMockGpuDeviceVO(long cardId) {
+        GpuDeviceVO device = mock(GpuDeviceVO.class);
+        when(device.getCardId()).thenReturn(cardId);
+        return device;
+    }
 }
