--- a/server/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateManagerImplTest.java
@@ -257,8 +257,7 @@
         VnfTemplateNicVO vnfNic3 = new VnfTemplateNicVO(templateId, 2L, "eth2", false, false, "third");
         Mockito.doReturn(Arrays.asList(vnfNic1, vnfNic2, vnfNic3)).when(vnfTemplateNicDao).listByTemplateId(templateId);
 
-        UserVm vm = Mockito.mock(UserVm.class);
-        when(vm.getId()).thenReturn(vmId);
+        UserVm vm = createMockUserVm(vmId);
         NicVO nic1 = Mockito.mock(NicVO.class);
         NicVO nic2 = Mockito.mock(NicVO.class);
         NicVO nic3 = Mockito.mock(NicVO.class);
@@ -350,8 +349,7 @@
         DataCenter zone = Mockito.mock(DataCenter.class);
         when(zone.getId()).thenReturn(zoneId);
         Account owner = Mockito.mock(Account.class);
-        UserVm vm = Mockito.mock(UserVm.class);
-        when(vm.getId()).thenReturn(vmId);
+        UserVm vm = createMockUserVm(vmId);
         DeployVnfApplianceCmd cmd = Mockito.mock(DeployVnfApplianceCmd.class);
 
         Map<Network, String> networkAndIpMap = new HashMap<>();
@@ -386,4 +384,10 @@
         Mockito.verify(firewallRulesDao, Mockito.times(3)).persist(any());
         Mockito.verify(firewallService, Mockito.times(1)).applyIngressFwRules(publicIpId, owner);
     }
+
+    private static UserVm createMockUserVm(long vmId) {
+        UserVm vm = Mockito.mock(UserVm.class);
+        when(vm.getId()).thenReturn(vmId);
+        return vm;
+    }
 }
