--- a/server/src/test/java/org/apache/cloudstack/schedule/ResourceScheduleManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/schedule/ResourceScheduleManagerImplTest.java
@@ -62,6 +62,8 @@
 import static org.junit.Assert.assertNotNull;
 
 public class ResourceScheduleManagerImplTest {
+    @Mock
+    private Account ownerAccount;
 
     @Spy
     @InjectMocks
@@ -127,7 +129,6 @@
     public void createSchedule() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         ResourceScheduleVO schedule = Mockito.mock(ResourceScheduleVO.class);
-        Account ownerAccount = Mockito.mock(Account.class);
 
         Mockito.when(vm.getUuid()).thenReturn(UUID.randomUUID().toString());
         Mockito.when(vm.getId()).thenReturn(1L);
@@ -283,7 +284,6 @@
     public void createScheduleAutoScale() {
         AutoScaleVmGroup group = Mockito.mock(AutoScaleVmGroup.class);
         ResourceScheduleVO schedule = Mockito.mock(ResourceScheduleVO.class);
-        Account ownerAccount = Mockito.mock(Account.class);
         Map<String, String> details = new HashMap<>();
         details.put("minmembers", "2");
         details.put("maxmembers", "5");
