--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -791,9 +791,8 @@
 
     @Test
     public void testGetBestMigration() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-
+
+        ClusterVO cluster = createMockClusterVO();
         HostVO destHost = Mockito.mock(HostVO.class);
         Mockito.when(destHost.getClusterId()).thenReturn(1L);
 
@@ -864,9 +863,8 @@
 
     @Test
     public void testGetBestMigrationDifferentCluster() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-
+
+        ClusterVO cluster = createMockClusterVO();
         HostVO destHost = Mockito.mock(HostVO.class);
         Mockito.when(destHost.getClusterId()).thenReturn(2L);
 
@@ -951,4 +949,10 @@
 
         Mockito.verify(clusterDrsService, Mockito.times(2)).executeDrsPlan(Mockito.any(ClusterDrsPlanVO.class));
     }
+
+    private static ClusterVO createMockClusterVO() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.when(cluster.getId()).thenReturn(1L);
+        return cluster;
+    }
 }
