--- a/server/src/test/java/com/cloud/network/element/VirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VirtualRouterElementTest.java
@@ -535,8 +535,7 @@
         List<BgpPeerVO> bgpPeers = Mockito.mock(List.class);
         VpcVO vpc = Mockito.mock(VpcVO.class);
 
-        DomainRouterVO router = Mockito.mock(DomainRouterVO.class);
-        when(router.getState()).thenReturn(VirtualMachine.State.Running);
+        DomainRouterVO router = createMockDomainRouter();
         long zoneId = 10L;
         long vpcId = 11L;
         when(vpc.getId()).thenReturn(vpcId);
@@ -559,8 +558,7 @@
         List<BgpPeerVO> bgpPeers = Mockito.mock(List.class);
         NetworkVO network = Mockito.mock(NetworkVO.class);
 
-        DomainRouterVO router = Mockito.mock(DomainRouterVO.class);
-        when(router.getState()).thenReturn(VirtualMachine.State.Running);
+        DomainRouterVO router = createMockDomainRouter();
         long zoneId = 10L;
         long networkId = 11L;
         when(network.getId()).thenReturn(networkId);
@@ -577,4 +575,10 @@
         Assert.assertTrue(result);
         verify(networkTopology).applyBgpPeers(any(), any(), any());
     }
+
+    private static DomainRouterVO createMockDomainRouter() {
+        DomainRouterVO router = Mockito.mock(DomainRouterVO.class);
+        when(router.getState()).thenReturn(VirtualMachine.State.Running);
+        return router;
+    }
 }
