--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -140,6 +140,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ManagementServerImplTest {
+    @Mock
+    private GuestOSCategoryVO guestOSCategory;
 
     @Mock
     SearchCriteria<IPAddressVO> sc;
@@ -960,7 +962,6 @@
     @Test
     public void testDeleteGuestOsCategory_Successful() {
         DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
         Mockito.when(deleteCmd.getId()).thenReturn(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
@@ -985,7 +986,6 @@
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteGuestOsCategory_ThrowsExceptionWhenGuestOsExists() {
         DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
         Mockito.when(deleteCmd.getId()).thenReturn(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
@@ -1003,7 +1003,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_Success() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Filter filter = Mockito.mock(Filter.class);
         Long id = 1L;
         String name = "Ubuntu";
@@ -1042,7 +1041,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_NoResults() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         String name = "CentOS";
         String keyword = "Linux";
@@ -1080,7 +1078,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_NoGuestOsIdsFound() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         String name = "Ubuntu";
         String keyword = "Linux";
@@ -1118,7 +1115,6 @@
     @Test
     public void testListGuestOSCategoriesByCriteria_FilterById() {
         ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
-        GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         Mockito.when(listCmd.getId()).thenReturn(id);
         Mockito.when(listCmd.getZoneId()).thenReturn(null);
