--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -778,9 +778,8 @@
         when(entityManager.findById(VirtualMachineTemplate.class, templateId)).thenReturn(templateMock);
         when(serviceOfferingMock.isDynamic()).thenReturn(false);
 
-        DispatchChain dispatchChainMock = Mockito.mock(DispatchChain.class);
+        DispatchChain dispatchChainMock = createDispatchChain();
         when(dispatchChainFactory.getStandardDispatchChain()).thenReturn(dispatchChainMock);
-        Mockito.doNothing().when(dispatchChainMock).dispatch(any());
         try (MockedStatic<ComponentContext> ignored = Mockito.mockStatic(ComponentContext.class)) {
             when(ComponentContext.inject(DeployVMCmd.class)).thenReturn(Mockito.mock(DeployVMCmd.class));
 
@@ -815,9 +814,8 @@
         when(entityManager.findById(VirtualMachineTemplate.class, templateId)).thenReturn(templateMock);
         Mockito.doThrow(InvalidParameterValueException.class).when(userVmMgr).finalizeUserData(any(), any(), any());
 
-        DispatchChain dispatchChainMock = Mockito.mock(DispatchChain.class);
+        DispatchChain dispatchChainMock = createDispatchChain();
         when(dispatchChainFactory.getStandardDispatchChain()).thenReturn(dispatchChainMock);
-        Mockito.doNothing().when(dispatchChainMock).dispatch(any());
         try (MockedStatic<ComponentContext> ignored = Mockito.mockStatic(ComponentContext.class)) {
             when(ComponentContext.inject(DeployVMCmd.class)).thenReturn(Mockito.mock(DeployVMCmd.class));
 
@@ -2581,4 +2579,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static DispatchChain createDispatchChain() {
+        DispatchChain dispatchChainMock = Mockito.mock(DispatchChain.class);
+        Mockito.doNothing().when(dispatchChainMock).dispatch(any());
+        return dispatchChainMock;
+    }
 }
