--- a/server/src/test/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocatorTest.java
+++ b/server/src/test/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocatorTest.java
@@ -241,7 +241,7 @@
     String hostVmTag = "haVmTag";
 
     Mockito.doReturn(hostVmTag).when(virtualMachineProfile).getParameter(Mockito.any(VirtualMachineProfile.Param.class));
-    Mockito.doReturn(allUpAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+    resourceManagerMock = MockResourceManager.createMockResourceManager(allUpAndEnabledHosts);
     Mockito.doReturn(hostsWithHaTag).when(hostDaoMock).listByHostTag(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString());
     Mockito.doNothing().when(firstFitAllocatorSpy).filterHostsWithUefiEnabled(Mockito.any(Host.Type.class), Mockito.any(VirtualMachineProfile.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyList());
     Mockito.doNothing().when(firstFitAllocatorSpy).addHostsBasedOnTagRules(Mockito.anyString(), Mockito.anyList());
@@ -276,7 +276,7 @@
     List<HostVO> allUpAndEnabledHosts = new ArrayList<>(Arrays.asList(host1, host2, host3));
     List<HostVO> upAndEnabledHostsWithNoHa = new ArrayList<>(Arrays.asList(host1, host2));
 
-    Mockito.doReturn(allUpAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+    resourceManagerMock = MockResourceManager.createMockResourceManager(allUpAndEnabledHosts);
     Mockito.doReturn(upAndEnabledHostsWithNoHa).when(resourceManagerMock).listAllUpAndEnabledNonHAHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
     Mockito.doNothing().when(firstFitAllocatorSpy).filterHostsWithUefiEnabled(Mockito.any(Host.Type.class), Mockito.any(VirtualMachineProfile.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyList());
     Mockito.doNothing().when(firstFitAllocatorSpy).addHostsBasedOnTagRules(Mockito.nullable(String.class), Mockito.anyList());
@@ -292,7 +292,7 @@
   public void retrieveHostsTestHostsToFilterIsNullAndNoHaTagWithHostTagShouldCallRetainHostsMatchingServiceOfferingAndTemplateTags() {
     List<HostVO> allUpAndEnabledHosts = new ArrayList<>(Arrays.asList(host1, host2, host3));
 
-    Mockito.doReturn(allUpAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+    resourceManagerMock = MockResourceManager.createMockResourceManager(allUpAndEnabledHosts);
     Mockito.doNothing().when(firstFitAllocatorSpy).retainHostsMatchingServiceOfferingAndTemplateTags(Mockito.anyList(), Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString(), Mockito.anyString());
     Mockito.doNothing().when(firstFitAllocatorSpy).filterHostsWithUefiEnabled(Mockito.any(Host.Type.class), Mockito.any(VirtualMachineProfile.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyList());
     Mockito.doNothing().when(firstFitAllocatorSpy).addHostsBasedOnTagRules(Mockito.anyString(), Mockito.anyList());
@@ -614,4 +614,22 @@
     when(memCapacity1.getCapacityType()).thenReturn(CapacityVO.CAPACITY_TYPE_MEMORY);
     return Arrays.asList(cpuCapacity1, memCapacity1, cpuCapacity2, memCapacity2);
   }
+
+    public final class MockResourceManager {
+
+        private MockResourceManager() {
+        }
+
+        public static ResourceManager createMockResourceManager(List<HostVO> hostsToReturn) {
+            ResourceManager resourceManagerMock = Mockito.mock(ResourceManager.class);
+            Mockito.doReturn(hostsToReturn)
+                    .when(resourceManagerMock)
+                    .listAllUpAndEnabledHosts(
+                            Mockito.any(Host.Type.class),
+                            Mockito.anyLong(),
+                            Mockito.anyLong(),
+                            Mockito.anyLong());
+            return resourceManagerMock;
+        }
+    }
 }
--- a/server/src/test/java/com/cloud/agent/manager/allocator/impl/RandomAllocatorTest.java
+++ b/server/src/test/java/com/cloud/agent/manager/allocator/impl/RandomAllocatorTest.java
@@ -219,7 +219,7 @@
         List<HostVO> upAndEnabledHosts = new ArrayList<>(Arrays.asList(host1, host2));
         List<HostVO> hostsWithNoRuleTagsAndHostTags = new ArrayList<>(Arrays.asList(host1));
 
-        Mockito.doReturn(upAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+        resourceManagerMock = MockResourceManager.createMockResourceManager(upAndEnabledHosts);
         Mockito.doReturn(hostsWithNoRuleTagsAndHostTags).when(hostDao).listAllHostsThatHaveNoRuleTag(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
         Mockito.doReturn(emptyList).when(hostDao).findHostsWithTagRuleThatMatchComputeOfferingTags(Mockito.nullable(String.class));
         Mockito.doNothing().when(randomAllocator).filterHostsBasedOnGuestOsRules(Mockito.any(VMTemplateVO.class), Mockito.anyList());
@@ -235,7 +235,7 @@
         List<HostVO> hostsWithNoRuleTagsAndHostTags = new ArrayList<>(Arrays.asList(host1));
         List<HostVO> hostsMatchingRuleTags = new ArrayList<>(Arrays.asList(host2));
 
-        Mockito.doReturn(upAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+        resourceManagerMock = MockResourceManager.createMockResourceManager(upAndEnabledHosts);
         Mockito.doReturn(hostsWithNoRuleTagsAndHostTags).when(hostDao).listAllHostsThatHaveNoRuleTag(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
         Mockito.doReturn(hostsMatchingRuleTags).when(hostDao).findHostsWithTagRuleThatMatchComputeOfferingTags(Mockito.nullable(String.class));
         Mockito.doNothing().when(randomAllocator).filterHostsBasedOnGuestOsRules(Mockito.any(VMTemplateVO.class), Mockito.anyList());
@@ -251,7 +251,7 @@
         List<HostVO> upAndEnabledHosts = new ArrayList<>(Arrays.asList(host1, host2));
         List<HostVO> hostsWithMatchingTags = new ArrayList<>(Arrays.asList(host1));
 
-        Mockito.doReturn(upAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+        resourceManagerMock = MockResourceManager.createMockResourceManager(upAndEnabledHosts);
         Mockito.doReturn(hostsWithMatchingTags).when(hostDao).listByHostTag(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString());
         Mockito.doReturn(emptyList).when(hostDao).findHostsWithTagRuleThatMatchComputeOfferingTags(Mockito.nullable(String.class));
         Mockito.doNothing().when(randomAllocator).filterHostsBasedOnGuestOsRules(Mockito.any(VMTemplateVO.class), Mockito.anyList());
@@ -267,7 +267,7 @@
         List<HostVO> hostsWithMatchingTags = new ArrayList<>(Arrays.asList(host1));
         List<HostVO> hostsMatchingRuleTags = new ArrayList<>(Arrays.asList(host3));
 
-        Mockito.doReturn(upAndEnabledHosts).when(resourceManagerMock).listAllUpAndEnabledHosts(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong());
+        resourceManagerMock = MockResourceManager.createMockResourceManager(upAndEnabledHosts);
         Mockito.doReturn(hostsWithMatchingTags).when(hostDao).listByHostTag(Mockito.any(Host.Type.class), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString());
         Mockito.doReturn(hostsMatchingRuleTags).when(hostDao).findHostsWithTagRuleThatMatchComputeOfferingTags(Mockito.nullable(String.class));
         Mockito.doNothing().when(randomAllocator).filterHostsBasedOnGuestOsRules(Mockito.any(VMTemplateVO.class), Mockito.anyList());
@@ -337,4 +337,22 @@
         Assert.assertEquals(host1, availableHosts.get(0));
         Assert.assertEquals(host3, availableHosts.get(1));
     }
+
+    public final class MockResourceManager {
+
+        private MockResourceManager() {
+        }
+
+        public static ResourceManager createMockResourceManager(List<HostVO> hostsToReturn) {
+            ResourceManager resourceManagerMock = Mockito.mock(ResourceManager.class);
+            Mockito.doReturn(hostsToReturn)
+                    .when(resourceManagerMock)
+                    .listAllUpAndEnabledHosts(
+                            Mockito.any(Host.Type.class),
+                            Mockito.anyLong(),
+                            Mockito.anyLong(),
+                            Mockito.anyLong());
+            return resourceManagerMock;
+        }
+    }
 }
