--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -207,11 +207,8 @@
         vmList.add(vm1);
         vmList.add(vm2);
 
-        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOffering.getCpu()).thenReturn(1);
-        Mockito.when(serviceOffering.getRamSize()).thenReturn(1024);
-        Mockito.when(serviceOffering.getSpeed()).thenReturn(1000);
-
+
+        ServiceOfferingVO serviceOffering = createServiceOfferingVO();
         Mockito.when(hostDao.findByClusterId(1L)).thenReturn(hostList);
         Mockito.when(vmInstanceDao.listByClusterId(1L)).thenReturn(vmList);
         Mockito.when(balancedAlgorithm.needsDrs(Mockito.any(), Mockito.anyList(), Mockito.anyList())).thenReturn(
@@ -593,11 +590,8 @@
         Mockito.when(hostJoin2.getMemReservedCapacity()).thenReturn(0L);
         Mockito.when(hostJoin2.getTotalMemory()).thenReturn(8192L);
 
-        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOffering.getCpu()).thenReturn(1);
-        Mockito.when(serviceOffering.getRamSize()).thenReturn(1024);
-        Mockito.when(serviceOffering.getSpeed()).thenReturn(1000);
-
+
+        ServiceOfferingVO serviceOffering = createServiceOfferingVO();
         Mockito.when(hostDao.findByClusterId(1L)).thenReturn(hostList);
         Mockito.when(vmInstanceDao.listByClusterId(1L)).thenReturn(vmList);
         Mockito.when(balancedAlgorithm.needsDrs(Mockito.any(), Mockito.anyList(), Mockito.anyList())).thenReturn(
@@ -951,4 +945,12 @@
 
         Mockito.verify(clusterDrsService, Mockito.times(2)).executeDrsPlan(Mockito.any(ClusterDrsPlanVO.class));
     }
+
+    private static ServiceOfferingVO createServiceOfferingVO() {
+        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
+        Mockito.when(serviceOffering.getCpu()).thenReturn(1);
+        Mockito.when(serviceOffering.getRamSize()).thenReturn(1024);
+        Mockito.when(serviceOffering.getSpeed()).thenReturn(1000);
+        return serviceOffering;
+    }
 }
