--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -95,6 +95,8 @@
 import com.cloud.vm.snapshot.VMSnapshotVO;
 
 public class AccountManagerImplTest extends AccountManagentImplTestBase {
+    @Mock
+    private User otherUserMock;
 
     @Mock
     private UserVmManagerImpl _vmMgr;
@@ -596,7 +598,6 @@
         Mockito.doReturn(1L).when(userVoMock).getId();
         Mockito.doReturn(apiKeyPairVOMock).when(apiKeyPairDaoMock).getLastApiKeyCreatedByUser(1L);
 
-        User otherUserMock = Mockito.mock(User.class);
 
         Ternary<User, Account, ApiKeyPair> pairUserAccountMock = new Ternary<User, Account, ApiKeyPair>(otherUserMock, Mockito.mock(Account.class), apiKeyPairVOMock);
         Mockito.doReturn(pairUserAccountMock).when(accountManagerImpl).findUserByApiKey(apiKey);
@@ -2128,9 +2129,8 @@
         Account rootAdminAccount = Mockito.mock(Account.class);
         Mockito.when(rootAdminAccount.getId()).thenReturn(1L);
         Mockito.when(accountManagerImpl.isRootAdmin(1L)).thenReturn(true);
-        User callingUser = Mockito.mock(User.class);
-        CallContext.register(callingUser, rootAdminAccount);
-
+
+        CallContext.register(otherUserMock, rootAdminAccount);
         String newPassword = "newPassword";
         configureUserMockAuthenticators(newPassword);
         Mockito.doNothing().when(accountManagerImpl).checkAccess(any(Account.class), any(Domain.class));
