--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -253,8 +253,7 @@
         ReflectionTestUtils.setField(sharedFS, "id", s_sharedFSId);
 
         when(cmd.getNetworkId()).thenReturn(s_networkId);
-        NetworkVO networkVO = mock(NetworkVO.class);
-        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        NetworkVO networkVO = createNetworkVO();
         when(networkDao.findById(s_networkId)).thenReturn(networkVO);
 
         sharedFSServiceImpl.allocSharedFS(cmd);
@@ -312,8 +311,7 @@
         when(diskOfferingVO.isCustomizedIops()).thenReturn(true);
 
         when(cmd.getNetworkId()).thenReturn(s_networkId);
-        NetworkVO networkVO = mock(NetworkVO.class);
-        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        NetworkVO networkVO = createNetworkVO();
         when(networkDao.findById(s_networkId)).thenReturn(networkVO);
 
         when(cmd.getFsFormat()).thenReturn("ext2");
@@ -662,4 +660,10 @@
             verify(_stateMachine, times(1)).transitTo(sharedFS, SharedFS.Event.OperationFailed, null, sharedFSDao);
         }
     }
+
+    private static NetworkVO createNetworkVO() {
+        NetworkVO networkVO = mock(NetworkVO.class);
+        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        return networkVO;
+    }
 }
