--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
@@ -55,8 +55,7 @@
 
         // Create Virtual-Network (VN)
         NetworkVO network = MockNetworkVO.getNetwork(Network.State.Implemented);
-        NetworkDao networkDao = mock(NetworkDao.class);
-        when(networkDao.findById(anyLong())).thenReturn(network);
+        NetworkDao networkDao = MockNetworkDao.createMockNetworkDao(network);
         when(controller.getNetworkDao()).thenReturn(networkDao);
 
         when(contrailMgr.getCanonicalName(network)).thenReturn("testnetwork");
@@ -125,4 +124,15 @@
             fail("ipInstance update failed ");
         }
     }
+
+    public final class MockNetworkDao {
+        private MockNetworkDao() {
+        }
+
+        public static NetworkDao createMockNetworkDao(NetworkVO network) {
+            NetworkDao networkDao = mock(NetworkDao.class);
+            when(networkDao.findById(anyLong())).thenReturn(network);
+            return networkDao;
+        }
+    }
 }
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
@@ -57,8 +57,7 @@
 
         // Create Virtual-Network (VN)
         NetworkVO network = MockNetworkVO.getNetwork(Network.State.Implemented);
-        NetworkDao networkDao = mock(NetworkDao.class);
-        when(networkDao.findById(anyLong())).thenReturn(network);
+        NetworkDao networkDao = MockNetworkDao.createMockNetworkDao(network);
         when(controller.getNetworkDao()).thenReturn(networkDao);
 
         when(contrailMgr.getCanonicalName(network)).thenReturn("testnetwork");
@@ -130,4 +129,15 @@
             fail("vm-interface verify failed ");
         }
     }
+
+    public final class MockNetworkDao {
+        private MockNetworkDao() {
+        }
+
+        public static NetworkDao createMockNetworkDao(NetworkVO network) {
+            NetworkDao networkDao = mock(NetworkDao.class);
+            when(networkDao.findById(anyLong())).thenReturn(network);
+            return networkDao;
+        }
+    }
 }
