--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
@@ -134,8 +134,7 @@
     public void testPingCommandStatusOk() throws ConfigurationException, NiciraNvpApiException {
         resource.configure("NiciraNvpResource", parameters);
 
-        final ControlClusterStatus ccs = mock(ControlClusterStatus.class);
-        when(ccs.getClusterStatus()).thenReturn("stable");
+        final ControlClusterStatus ccs = createControlClusterStatus("stable");
         when(nvpApi.getControlClusterStatus()).thenReturn(ccs);
 
         final PingCommand ping = resource.getCurrentStatus(42);
@@ -148,8 +147,7 @@
     public void testPingCommandStatusFail() throws ConfigurationException, NiciraNvpApiException {
         resource.configure("NiciraNvpResource", parameters);
 
-        final ControlClusterStatus ccs = mock(ControlClusterStatus.class);
-        when(ccs.getClusterStatus()).thenReturn("unstable");
+        final ControlClusterStatus ccs = createControlClusterStatus("unstable");
         when(nvpApi.getControlClusterStatus()).thenReturn(ccs);
 
         final PingCommand ping = resource.getCurrentStatus(42);
@@ -160,8 +158,7 @@
     public void testPingCommandStatusApiException() throws ConfigurationException, NiciraNvpApiException {
         resource.configure("NiciraNvpResource", parameters);
 
-        final ControlClusterStatus ccs = mock(ControlClusterStatus.class);
-        when(ccs.getClusterStatus()).thenReturn("unstable");
+        final ControlClusterStatus ccs = createControlClusterStatus("unstable");
         when(nvpApi.getControlClusterStatus()).thenThrow(new NiciraNvpApiException());
 
         final PingCommand ping = resource.getCurrentStatus(42);
@@ -822,4 +819,10 @@
         assertTrue(snr.getMatch().getSourcePort() == 8080);
         assertTrue(snr.getMatch().getEthertype().equals("IPv4") && rules[1].getMatch().getProtocol() == 6);
     }
+
+    private static ControlClusterStatus createControlClusterStatus(String clusterStatus) {
+        final ControlClusterStatus ccs = mock(ControlClusterStatus.class);
+        when(ccs.getClusterStatus()).thenReturn(clusterStatus);
+        return ccs;
+    }
 }
