--- a/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
@@ -142,11 +142,10 @@
 
     @Test
     public void testGetLoadBalancerServiceProvider() {
-        LoadBalancerVO loadBalancerMock = Mockito.mock(LoadBalancerVO.class);
         NetworkVO networkMock = Mockito.mock(NetworkVO.class);
         List<Network.Provider> providers = Arrays.asList(Network.Provider.VirtualRouter);
 
-        when(loadBalancerMock.getNetworkId()).thenReturn(10L);
+        LoadBalancerVO loadBalancerMock = createLoadBalancerMock(10L);
         when(_networkDao.findById(anyLong())).thenReturn(networkMock);
         when(_networkMgr.getProvidersForServiceInNetwork(networkMock, Network.Service.Lb)).thenReturn(providers);
 
@@ -324,13 +323,12 @@
     @Test
     public void testGetVmNicInLoadBalancerDefaultCase() {
         UserVm userVm = Mockito.mock(UserVm.class);
-        LoadBalancerVO loadBalancer = Mockito.mock(LoadBalancerVO.class);
         Network loadBalancerNetwork = Mockito.mock(Network.class);
         Account owner = Mockito.mock(Account.class);
 
         when(vpcManager.isNetworkOnVpcEnabledConserveMode(Mockito.eq(loadBalancerNetwork))).thenReturn(false);
 
-        when(loadBalancer.getNetworkId()).thenReturn(networkId);
+        LoadBalancerVO loadBalancer = createLoadBalancerMock(networkId);
         Nic nic = Mockito.mock(Nic.class);
         when(nic.getNetworkId()).thenReturn(networkId);
         List<? extends Nic> nics = Collections.singletonList(nic);
@@ -378,4 +376,10 @@
         lbr.createPublicLoadBalancerRule("xid", "name", "desc", 53, 53, 53, 53,
                 null, "tcp", "roundrobin", networkId, lbOwnerId, false, "tcp", null, null);
     }
+
+    private static LoadBalancerVO createLoadBalancerMock(long networkId) {
+        LoadBalancerVO loadBalancerMock = Mockito.mock(LoadBalancerVO.class);
+        when(loadBalancerMock.getNetworkId()).thenReturn(networkId);
+        return loadBalancerMock;
+    }
 }
