--- a/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
@@ -500,13 +500,10 @@
 
     @Test
     public void getConsoleConnectionDetailsReturnsDetailsForExternalHypervisor() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = Mockito.mock(ConsoleConnectionDetails.class);
 
-        Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
-        Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
-        Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
+        VirtualMachine vm = createMockVirtualMachine("vm-uuid", "vm-hostname", "vnc-password");
         Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
 
@@ -520,7 +517,6 @@
 
     @Test
     public void getConsoleConnectionDetailsReturnsDetailsForKVMHypervisor() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         int port = 5900;
@@ -530,9 +526,7 @@
 
         Pair<String, Integer> hostPortInfo = new Pair<>(hostAddress, port);
 
-        Mockito.when(vm.getUuid()).thenReturn(vmUuid);
-        Mockito.when(vm.getHostName()).thenReturn(vmHostName);
-        Mockito.when(vm.getVncPassword()).thenReturn(vncPassword);
+        VirtualMachine vm = createMockVirtualMachine(vmUuid, vmHostName, vncPassword);
         Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(hostPortInfo);
@@ -547,14 +541,11 @@
 
     @Test
     public void getConsoleConnectionDetailsReturnsDetailsWithRDPForHyperV() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         Pair<String, Integer> hostPortInfo = new Pair<>(hostAddress, -9);
 
-        Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
-        Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
-        Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
+        VirtualMachine vm = createMockVirtualMachine("vm-uuid", "vm-hostname", "vnc-password");
         Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.Hyperv);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(hostPortInfo);
@@ -573,12 +564,9 @@
 
     @Test
     public void getConsoleConnectionDetailsReturnsNullHostInvalidPortWhenVncPortInfoIsMissing() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        HostVO host = Mockito.mock(HostVO.class);
-
-        Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
-        Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
-        Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
+        HostVO host = Mockito.mock(HostVO.class);
+
+        VirtualMachine vm = createMockVirtualMachine("vm-uuid", "vm-hostname", "vnc-password");
         Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of());
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(new Pair<>(null, -1));
@@ -591,14 +579,11 @@
 
     @Test
     public void getConsoleConnectionDetailsSetsLocaleWhenKeyboardDetailIsPresent() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         Pair<String, Integer> hostPortInfo = new Pair<>(hostAddress, 5900);
 
-        Mockito.when(vm.getUuid()).thenReturn("vm-uuid");
-        Mockito.when(vm.getHostName()).thenReturn("vm-hostname");
-        Mockito.when(vm.getVncPassword()).thenReturn("vnc-password");
+        VirtualMachine vm = createMockVirtualMachine("vm-uuid", "vm-hostname", "vnc-password");
         Mockito.when(host.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         Mockito.when(vmInstanceDetailsDao.listDetailsKeyPairs(Mockito.anyLong(), Mockito.anyList())).thenReturn(Map.of(VmDetailConstants.KEYBOARD, "en-us"));
         Mockito.when(managementServer.getVncPort(vm)).thenReturn(hostPortInfo);
@@ -787,4 +772,12 @@
         Assert.assertNull(endpoint.getWebsocketExtra());
         Assert.assertNull(endpoint.getWebsocketHost());
     }
+
+    private static VirtualMachine createMockVirtualMachine(String vmUuid, String vmHostName, String vncPassword) {
+        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
+        Mockito.when(vm.getUuid()).thenReturn(vmUuid);
+        Mockito.when(vm.getHostName()).thenReturn(vmHostName);
+        Mockito.when(vm.getVncPassword()).thenReturn(vncPassword);
+        return vm;
+    }
 }
