--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -146,6 +146,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class CitrixRequestWrapperTest {
+    @Mock
+    private final VirtualRoutingResource routingResource;
 
     @Mock
     private CitrixResourceBase citrixResourceBase;
@@ -1835,7 +1837,7 @@
 
     @Test
     public void testIpAssocVpcCommand() {
-        final VirtualRoutingResource routingResource = Mockito.mock(VirtualRoutingResource.class);
+        // removed local mock; replaced with global field `routingResource`
         final IpAddressTO[] ips = new IpAddressTO[0];
 
         final IpAssocVpcCommand ipAssociation = new IpAssocVpcCommand(ips);
@@ -1855,7 +1857,6 @@
 
     @Test
     public void testIpAssocCommand() {
-        final VirtualRoutingResource routingResource = Mockito.mock(VirtualRoutingResource.class);
         final IpAddressTO[] ips = new IpAddressTO[0];
 
         final IpAssocCommand ipAssociation = new IpAssocCommand(ips);
