--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenProviderServiceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenProviderServiceTest.java
@@ -52,6 +52,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenProviderServiceTest {
+    @Mock
+    private TungstenProviderVO tungstenProviderVO;
 
     @Mock
     DataCenterDao dcDao;
@@ -121,7 +123,6 @@
 
     @Test
     public void listTungstenProviderWithZoneIdTest() {
-        TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
 
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
 
@@ -131,11 +132,11 @@
 
     @Test
     public void listTungstenProviderWithoutZoneIdTest() {
-        TungstenProviderVO tungstenProviderVO1 = Mockito.mock(TungstenProviderVO.class);
+        // removed local mock; replaced with global field `tungstenProviderVO`
         TungstenProviderVO tungstenProviderVO2 = Mockito.mock(TungstenProviderVO.class);
 
-        when(tungstenProviderDao.listAll()).thenReturn(Arrays.asList(tungstenProviderVO1, tungstenProviderVO2));
 
+        when(tungstenProviderDao.listAll()).thenReturn(Arrays.asList(tungstenProviderVO, tungstenProviderVO2));
         List<BaseResponse> baseResponseList = tungstenProviderService.listTungstenProvider(null);
         assertEquals(2, baseResponseList.size());
     }
