--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -716,7 +716,6 @@
 
     @Test
     public void addTungstenSecurityGroupEgressRuleTest() {
-        SecurityRule securityRule = mock(SecurityRule.class);
         SecurityGroupVO securityGroupVO = mock(SecurityGroupVO.class);
         TungstenAnswer getTungstenSecurityGroupAnswer = mock(TungstenAnswer.class);
         TungstenAnswer removeTungstenSecurityGroupRuleAnswer = MockTungstenAnswerFactory.get(true);
@@ -732,10 +731,8 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenSecurityGroupRuleCommand.class), anyLong())).thenReturn(removeTungstenSecurityGroupRuleAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(AddTungstenSecurityGroupRuleCommand.class), anyLong())).thenReturn(addTungstenSecurityGroupRuleAnswer);
         when(getTungstenSecurityGroupAnswer.getApiObjectBase()).thenReturn(securityGroup);
-        when(securityRule.getRuleType()).thenReturn(SecurityRule.SecurityRuleType.EgressRule);
+        SecurityRule securityRule = createSecurityRule(SecurityRule.SecurityRuleType.EgressRule);
         when(tungstenSecurityGroupRuleDao.findDefaultSecurityRule(anyLong(), anyString(), anyString())).thenReturn(tungstenSecurityGroupRuleVO);
-        when(securityRule.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
-        when(securityRule.getEndPort()).thenReturn(NetUtils.PORT_RANGE_MIN);
         when(securityGroupVMMapDao.listVmIdsBySecurityGroup(anyLong())).thenReturn(List.of(1L));
         when(nicDao.findDefaultNicForVM(anyLong())).thenReturn(nicVO);
         when(nicVO.getIPv4Address()).thenReturn("192.168.100.100");
@@ -750,7 +747,6 @@
 
     @Test
     public void addTungstenSecurityGroupIngressRuleTest() {
-        SecurityRule securityRule = mock(SecurityRule.class);
         SecurityGroupVO securityGroupVO = mock(SecurityGroupVO.class);
         TungstenAnswer getTungstenSecurityGroupAnswer = mock(TungstenAnswer.class);
         TungstenAnswer addTungstenSecurityGroupRuleAnswer = MockTungstenAnswerFactory.get(true);
@@ -762,12 +758,10 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenSecurityGroupCommand.class), anyLong())).thenReturn(getTungstenSecurityGroupAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(AddTungstenSecurityGroupRuleCommand.class), anyLong())).thenReturn(addTungstenSecurityGroupRuleAnswer);
         when(getTungstenSecurityGroupAnswer.getApiObjectBase()).thenReturn(securityGroup);
-        when(securityRule.getRuleType()).thenReturn(SecurityRule.SecurityRuleType.IngressRule);
-        when(securityRule.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
-        when(securityRule.getEndPort()).thenReturn(NetUtils.PORT_RANGE_MIN);
         when(securityRule.getAllowedNetworkId()).thenReturn(null);
 
         assertTrue(tungstenService.addTungstenSecurityGroupRule(List.of(securityRule)));
+        SecurityRule securityRule = createSecurityRule(SecurityRule.SecurityRuleType.IngressRule);
     }
 
     @Test
@@ -1668,4 +1662,12 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static SecurityRule createSecurityRule(SecurityRule.SecurityRuleType ruleType) {
+        SecurityRule securityRule = mock(SecurityRule.class);
+        when(securityRule.getRuleType()).thenReturn(ruleType);
+        when(securityRule.getProtocol()).thenReturn(NetUtils.ALL_PROTO);
+        when(securityRule.getEndPort()).thenReturn(NetUtils.PORT_RANGE_MIN);
+        return securityRule;
+    }
 }
