--- a/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
@@ -69,6 +69,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ConsoleAccessManagerImplTest {
+    @Mock
+    private VirtualMachine vm;
 
     @Mock
     private AccountManager accountManager;
@@ -333,7 +335,6 @@
 
     @Test
     public void getConsoleConnectionDetailsForExternalVmReturnsNullWhenAnswerIsNull() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("sid", "en", "tag", "displayName");
 
@@ -346,7 +347,6 @@
 
     @Test
     public void getConsoleConnectionDetailsForExternalVmReturnsNullWhenAnswerResultIsFalse() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("sid", "en", "tag", "displayName");
         Answer answer = Mockito.mock(Answer.class);
@@ -362,7 +362,6 @@
 
     @Test
     public void getConsoleConnectionDetailsForExternalVmReturnsNullWhenAnswerIsNotOfTypeGetExternalConsoleAnswer() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("sid", "en", "tag", "displayName");
         Answer answer = Mockito.mock(Answer.class);
@@ -377,7 +376,6 @@
 
     @Test
     public void getConsoleConnectionDetailsForExternalVmSetsDetailsWhenAnswerIsValid() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("sid", "en", "tag", "displayName");
         GetExternalConsoleAnswer answer = Mockito.mock(GetExternalConsoleAnswer.class);
@@ -404,7 +402,6 @@
 
     @Test
     public void getConsoleConnectionDetailsForExternalVmSetsDetailsWhenAnswerIsValidDirect() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("sid", "en", "tag", "displayName");
         GetExternalConsoleAnswer answer = Mockito.mock(GetExternalConsoleAnswer.class);
@@ -425,7 +422,6 @@
 
     @Test
     public void getConsoleConnectionDetailsForExternalVmDoesNotSetSidWhenPasswordIsBlank() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("sid", "en", "tag", "displayName");
         GetExternalConsoleAnswer answer = Mockito.mock(GetExternalConsoleAnswer.class);
@@ -612,7 +608,6 @@
 
     @Test
     public void generateConsoleProxyClientParamSetsBasicDetailsCorrectly() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         String hostAddress = "192.168.1.100";
         int port = 5902;
@@ -642,7 +637,6 @@
 
     @Test
     public void generateConsoleProxyClientParamSetsExtraSecurityTokenWhenProvided() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("password", null, null, null);
 
@@ -654,7 +648,6 @@
     @Test
     public void generateConsoleProxyClientParamSetsLocaleWhenProvided() {
         HostVO host = Mockito.mock(HostVO.class);
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails(null, "fr-fr", null, null);
 
         ConsoleProxyClientParam param = consoleAccessManager.generateConsoleProxyClientParam(details, "ticket", "sessionUuid", "127.0.0.1", null, vm, host);
@@ -669,7 +662,6 @@
         String password = "adminPass";
         HostVO host = Mockito.mock(HostVO.class);
         Mockito.when(host.getId()).thenReturn(hostId);
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails(null, null, null, null);
         details.setUsingRDP(true);
         String ip = "10.0.0.1";
@@ -687,7 +679,6 @@
     @Test
     public void generateConsoleProxyClientParamSetsTunnelDetailsWhenProvided() {
         HostVO host = Mockito.mock(HostVO.class);
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         ConsoleConnectionDetails details = new ConsoleConnectionDetails(null, null, null, null);
         details.setTunnelUrl("tunnelUrl");
         details.setTunnelSession("tunnelSession");
@@ -700,7 +691,6 @@
 
     @Test
     public void returnsNullWhenConsoleConnectionDetailsAreNull() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         HostVO host = Mockito.mock(HostVO.class);
         Mockito.doReturn(null).when(consoleAccessManager).getConsoleConnectionDetails(vm, host);
 
