--- a/api/src/test/java/org/apache/cloudstack/api/command/user/vm/ListVMsCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/vm/ListVMsCmdTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.api.command.user.vm;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
@@ -51,6 +52,8 @@
 import com.cloud.utils.db.EntityManager;
 
 public class ListVMsCmdTest {
+    @Mock
+    private ResourceIcon icon1;
 
     EntityManager _entityMgr;
     ResourceIconManager resourceIconManager;
@@ -78,7 +81,6 @@
         UserVmResponse vm2 = mock(UserVmResponse.class);
         when(vm2.getId()).thenReturn(vm2Uuid);
         List<UserVmResponse> responses = Arrays.asList(vm1, vm2);
-        ResourceIcon icon1 = mock(ResourceIcon.class);
         ResourceIcon icon2 = mock(ResourceIcon.class);
         Map<String, ResourceIcon> initialIcons = new HashMap<>();
         initialIcons.put(vm1Uuid, icon1);
@@ -174,7 +176,6 @@
         when(guestOS2.getCategoryId()).thenReturn(20L);
         when(_entityMgr.listByUuids(eq(GuestOS.class), anySet()))
                 .thenReturn(Arrays.asList(guestOS1, guestOS2));
-        ResourceIcon icon1 = mock(ResourceIcon.class);
         ResourceIcon icon2 = mock(ResourceIcon.class);
         Map<Long, ResourceIcon> categoryIcons = new HashMap<>();
         categoryIcons.put(10L, icon1);
