--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionManagerImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionManagerImplTest.java
@@ -58,8 +58,7 @@
 
     @Test
     public void testUpdateTemplateDetailsInKubernetesSupportedVersionResponseNullTemplate() {
-        KubernetesSupportedVersion kubernetesSupportedVersion = Mockito.mock(KubernetesSupportedVersion.class);
-        Mockito.when(kubernetesSupportedVersion.getIsoId()).thenReturn(1L);
+        KubernetesSupportedVersion kubernetesSupportedVersion = createMockKubernetesSupportedVersion();
         KubernetesSupportedVersionResponse response = new KubernetesSupportedVersionResponse();
         kubernetesVersionManager.updateTemplateDetailsInKubernetesSupportedVersionResponse(kubernetesSupportedVersion,
                 response, true);
@@ -68,8 +67,7 @@
 
     @Test
     public void testUpdateTemplateDetailsInKubernetesSupportedVersionResponseValidTemplate() {
-        KubernetesSupportedVersion kubernetesSupportedVersion = Mockito.mock(KubernetesSupportedVersion.class);
-        Mockito.when(kubernetesSupportedVersion.getIsoId()).thenReturn(1L);
+        KubernetesSupportedVersion kubernetesSupportedVersion = createMockKubernetesSupportedVersion();
         KubernetesSupportedVersionResponse response = new KubernetesSupportedVersionResponse();
         TemplateJoinVO templateJoinVO = Mockito.mock(TemplateJoinVO.class);
         Mockito.when(templateJoinVO.getArch()).thenReturn(CPU.CPUArch.getDefault());
@@ -144,4 +142,10 @@
 
         ReflectionTestUtils.invokeMethod(kubernetesVersionManager, "validateImageStoreForZone", (Long) null, false);
     }
+
+    private static KubernetesSupportedVersion createMockKubernetesSupportedVersion() {
+        KubernetesSupportedVersion kubernetesSupportedVersion = Mockito.mock(KubernetesSupportedVersion.class);
+        Mockito.when(kubernetesSupportedVersion.getIsoId()).thenReturn(1L);
+        return kubernetesSupportedVersion;
+    }
 }
