--- a/server/src/test/java/com/cloud/network/IpAddressManagerTest.java
+++ b/server/src/test/java/com/cloud/network/IpAddressManagerTest.java
@@ -390,9 +390,9 @@
 
         Mockito.when(publicIpQuarantineVOMock.getPreviousOwnerId()).thenReturn(dummyID);
         Mockito.when(accountManagerMock.getAccount(Mockito.anyLong())).thenReturn(previousOwnerMock);
-        Mockito.when(previousOwnerMock.getUuid()).thenReturn(UUID);
-        Mockito.when(newOwnerMock.getUuid()).thenReturn(UUID);
-
+        previousOwnerMock = createMockAccountVO(UUID);
+
+        newOwnerMock = createMockAccountVO(UUID);
         boolean result = ipAddressManager.canPublicIpAddressBeAllocated(ipAddressMock, newOwnerMock);
 
         Assert.assertTrue(result);
@@ -409,9 +409,9 @@
 
         Mockito.when(publicIpQuarantineVOMock.getPreviousOwnerId()).thenReturn(dummyID);
         Mockito.when(accountManagerMock.getAccount(Mockito.anyLong())).thenReturn(previousOwnerMock);
-        Mockito.when(previousOwnerMock.getUuid()).thenReturn(UUID);
-        Mockito.when(newOwnerMock.getUuid()).thenReturn(UUID_2);
-
+        previousOwnerMock = createMockAccountVO(UUID);
+
+        newOwnerMock = createMockAccountVO(UUID_2);
         boolean result = ipAddressManager.canPublicIpAddressBeAllocated(ipAddressMock, newOwnerMock);
 
         Assert.assertFalse(result);
@@ -640,4 +640,10 @@
 
         assertFalse(result);
     }
+
+    private static AccountVO createMockAccountVO(String uuidReturn) {
+        AccountVO accountVOMock = Mockito.mock(AccountVO.class);
+        Mockito.when(accountVOMock.getUuid()).thenReturn(uuidReturn);
+        return accountVOMock;
+    }
 }
