--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -144,6 +144,8 @@
 import static org.mockito.Mockito.when;
 
 public class ConfigurationManagerTest {
+    @Mock
+    private VolumeVO volumeVO;
 
     private Logger logger = LogManager.getLogger(ConfigurationManagerTest.class);
 
@@ -767,7 +769,6 @@
         Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         ArrayList<VolumeVO> arrayList = new ArrayList<VolumeVO>();
         arrayList.add(volumeVO);
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(0);
@@ -927,7 +928,6 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfZoneIsDeletableFailureOnVolumeTest() {
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         ArrayList<VolumeVO> arrayList = new ArrayList<VolumeVO>();
         arrayList.add(volumeVO);
 
