--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -172,6 +172,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class TemplateManagerImplTest extends TestCase {
+    @Mock
+    private VMTemplateVO mockTemplate;
 
     @Spy
     @InjectMocks
@@ -448,7 +450,6 @@
 
     @Test
     public void testTemplateScheduledForDownloadInDisabledPool() {
-        VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         StoragePoolVO mockPool = mock(StoragePoolVO.class);
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
         VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
@@ -561,14 +562,13 @@
         when(cmd.getUserdataId()).thenReturn(2L);
         when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
-        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
         when(vmTemplateDao.findById(anyLong())).thenReturn(template);
 
         when(userDataDaoMock.findById(anyLong())).thenReturn(userDataMock);
 
         VirtualMachineTemplate resultTemplate = templateManager.linkUserDataToTemplate(cmd);
 
-        Assert.assertEquals(template, resultTemplate);
+        Assert.assertEquals(mockTemplate, resultTemplate);
     }
 
     @Test(expected = InvalidParameterValueException.class)
@@ -579,7 +579,6 @@
         when(cmd.getUserdataId()).thenReturn(2L);
         when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
-        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
 
         templateManager.linkUserDataToTemplate(cmd);
     }
@@ -592,7 +591,6 @@
         when(cmd.getUserdataId()).thenReturn(2L);
         when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
-        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
 
         templateManager.linkUserDataToTemplate(cmd);
     }
