--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -216,8 +216,7 @@
         Mockito.when(hostVO.getHypervisorType()).thenReturn(hypervisorType);
         Mockito.when(hostVO.getHypervisorVersion()).thenReturn(hypervisorVersion);
 
-        RevertToVMSnapshotAnswer answer = Mockito.mock(RevertToVMSnapshotAnswer.class);
-        Mockito.when(answer.getResult()).thenReturn(Boolean.TRUE);
+        RevertToVMSnapshotAnswer answer = createRevertToVMSnapshotAnswer();
         boolean result = vmStrategy.revertVMSnapshot(vmSnapshot);
         assertTrue(result);
     }
@@ -449,4 +448,16 @@
             return Mockito.mock(SnapshotDataStoreDao.class);
         }
     }
+
+    public final class MockRevertToVMSnapshotAnswer {
+
+        private MockRevertToVMSnapshotAnswer() {
+        }
+
+        public static com.cloud.agent.api.RevertToVMSnapshotAnswer createRevertToVMSnapshotAnswer() {
+            com.cloud.agent.api.RevertToVMSnapshotAnswer answer = Mockito.mock(com.cloud.agent.api.RevertToVMSnapshotAnswer.class);
+            Mockito.when(answer.getResult()).thenReturn(Boolean.TRUE);
+            return answer;
+        }
+    }
 }
--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
@@ -191,8 +191,7 @@
 
         assertNotNull(e);
 
-        RevertToVMSnapshotAnswer answer = Mockito.mock(RevertToVMSnapshotAnswer.class);
-        Mockito.when(answer.getResult()).thenReturn(Boolean.TRUE);
+        RevertToVMSnapshotAnswer answer = MockRevertToVMSnapshotAnswer.createRevertToVMSnapshotAnswer();
         Mockito.when(agentMgr.send(ArgumentMatchers.anyLong(), ArgumentMatchers.any(Command.class))).thenReturn(answer);
         boolean result = vmSnapshotStrategy.revertVMSnapshot(vmSnapshot);
         assertTrue(result);
@@ -342,4 +341,16 @@
             return Mockito.mock(BackupManager.class);
         }
     }
+
+    public final class MockRevertToVMSnapshotAnswer {
+
+        private MockRevertToVMSnapshotAnswer() {
+        }
+
+        public static com.cloud.agent.api.RevertToVMSnapshotAnswer createRevertToVMSnapshotAnswer() {
+            com.cloud.agent.api.RevertToVMSnapshotAnswer answer = Mockito.mock(com.cloud.agent.api.RevertToVMSnapshotAnswer.class);
+            Mockito.when(answer.getResult()).thenReturn(Boolean.TRUE);
+            return answer;
+        }
+    }
 }
