--- a/api/src/test/java/org/apache/cloudstack/api/command/test/AddVpnUserCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/AddVpnUserCmdTest.java
@@ -74,11 +74,10 @@
     @Test
     public void testCreateSuccess() {
 
-        AccountService accountService = Mockito.mock(AccountService.class);
 
         Account account = Mockito.mock(Account.class);
-        Mockito.when(accountService.getAccount(nullable(Long.class))).thenReturn(account);
 
+        AccountService accountService = createMockAccountService(account);
         addVpnUserCmd._accountService = accountService;
 
         RemoteAccessVpnService ravService = Mockito.mock(RemoteAccessVpnService.class);
@@ -96,10 +95,9 @@
     @Test
     public void testCreateFailure() {
 
-        AccountService accountService = Mockito.mock(AccountService.class);
         Account account = Mockito.mock(Account.class);
-        Mockito.when(accountService.getAccount(nullable(Long.class))).thenReturn(account);
 
+        AccountService accountService = createMockAccountService(account);
         addVpnUserCmd._accountService = accountService;
 
         RemoteAccessVpnService ravService = Mockito.mock(RemoteAccessVpnService.class);
@@ -116,4 +114,10 @@
 
     }
 
+
+    private static AccountService createMockAccountService(Account account) {
+        AccountService accountService = Mockito.mock(AccountService.class);
+        Mockito.when(accountService.getAccount(nullable(Long.class))).thenReturn(account);
+        return accountService;
+    }
 }
