--- a/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedNASStrategyTest.java
+++ b/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedNASStrategyTest.java
@@ -79,6 +79,8 @@
 @ExtendWith(MockitoExtension.class)
 @MockitoSettings(strictness = Strictness.LENIENT)
 public class UnifiedNASStrategyTest {
+    @Mock
+    private VolumeVO volumeVO;
 
     @Mock
     private NASFeignClient nasFeignClient;
@@ -173,7 +175,6 @@
         // Setup CloudStackVolume
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
         VolumeObject volumeObject = mock(VolumeObject.class);
-        VolumeVO volumeVO = mock(VolumeVO.class);
         EndPoint endPoint = mock(EndPoint.class);
         Answer answer = new Answer(null, true, "Success");
 
@@ -217,7 +218,6 @@
     public void testCreateCloudStackVolume_KVMHostFailed() {
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
         VolumeObject volumeObject = mock(VolumeObject.class);
-        VolumeVO volumeVO = mock(VolumeVO.class);
         EndPoint endPoint = mock(EndPoint.class);
         Answer answer = new Answer(null, false, "Failed to create volume");
 
@@ -240,7 +240,6 @@
     public void testCreateCloudStackVolume_NoEndpoint() {
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
         VolumeObject volumeObject = mock(VolumeObject.class);
-        VolumeVO volumeVO = mock(VolumeVO.class);
 
         when(cloudStackVolume.getDatastoreId()).thenReturn("1");
         when(cloudStackVolume.getVolumeInfo()).thenReturn(volumeObject);
