--- a/server/src/test/java/org/apache/cloudstack/schedule/vm/VMScheduleWorkerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/schedule/vm/VMScheduleWorkerTest.java
@@ -62,6 +62,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VMScheduleWorkerTest {
+    @Mock
+    private VirtualMachine vm;
     @Spy
     @InjectMocks
     private VMScheduleWorker vmScheduleWorker = new VMScheduleWorker();
@@ -347,7 +349,6 @@
 
     @Test
     public void testSubmitStopVMJob() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(asyncJobManager.submitAsyncJob(Mockito.any(AsyncJobVO.class))).thenReturn(1L);
         Mockito.when(asyncJobDispatcher.getName()).thenReturn("ApiAsyncJobDispatcher");
         try (MockedStatic<ComponentContext> ignored = Mockito.mockStatic(ComponentContext.class)) {
@@ -360,7 +361,6 @@
 
     @Test
     public void testSubmitRebootVMJob() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(asyncJobManager.submitAsyncJob(Mockito.any(AsyncJobVO.class))).thenReturn(1L);
         Mockito.when(asyncJobDispatcher.getName()).thenReturn("ApiAsyncJobDispatcher");
         try (MockedStatic<ComponentContext> ignored = Mockito.mockStatic(ComponentContext.class)) {
@@ -373,7 +373,6 @@
 
     @Test
     public void testSubmitStartVMJob() {
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(asyncJobManager.submitAsyncJob(Mockito.any(AsyncJobVO.class))).thenReturn(1L);
         Mockito.when(asyncJobDispatcher.getName()).thenReturn("ApiAsyncJobDispatcher");
         try (MockedStatic<ComponentContext> ignored = Mockito.mockStatic(ComponentContext.class)) {
