--- a/server/src/test/java/com/cloud/resource/RollingMaintenanceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/RollingMaintenanceManagerImplTest.java
@@ -189,8 +189,7 @@
 
     @Test
     public void testGetComputeResourcesCpuSpeedAndRamSize_ForNormalOffering() {
-        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOffering.isDynamic()).thenReturn(false);
+        ServiceOfferingVO serviceOffering = createMockServiceOffering(false);
         Mockito.when(serviceOffering.getCpu()).thenReturn(1);
         Mockito.when(serviceOffering.getSpeed()).thenReturn(500);
         Mockito.when(serviceOffering.getRamSize()).thenReturn(512);
@@ -207,8 +206,7 @@
 
     @Test
     public void testGetComputeResourcesCpuSpeedAndRamSize_ForCustomOffering() {
-        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
-        Mockito.when(serviceOffering.isDynamic()).thenReturn(true);
+        ServiceOfferingVO serviceOffering = createMockServiceOffering(true);
         Mockito.when(serviceOffering.getCpu()).thenReturn(null);
         Mockito.when(serviceOffering.getSpeed()).thenReturn(null);
         Mockito.when(serviceOffering.getRamSize()).thenReturn(null);
@@ -232,4 +230,10 @@
         Assert.assertEquals(1000, cpuSpeedAndRamSize.second().intValue());
         Assert.assertEquals(1024, cpuSpeedAndRamSize.third().intValue());
     }
+
+    private static ServiceOfferingVO createMockServiceOffering(boolean dynamic) {
+        ServiceOfferingVO serviceOffering = Mockito.mock(ServiceOfferingVO.class);
+        Mockito.when(serviceOffering.isDynamic()).thenReturn(dynamic);
+        return serviceOffering;
+    }
 }
