--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -66,6 +66,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class XenServer610WrapperTest {
+    @Mock
+    private final NicTO nicTO3;
 
     @Mock
     protected XenServer610Resource xenServer610Resource;
@@ -118,7 +120,6 @@
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
         final NicTO nicTO2 = Mockito.mock(NicTO.class);
-        final NicTO nicTO3 = Mockito.mock(NicTO.class);
         final NicTO [] nicTOs = {nicTO1, nicTO2, nicTO3};
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
@@ -191,7 +192,6 @@
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
         final NicTO nicTO2 = Mockito.mock(NicTO.class);
-        final NicTO nicTO3 = Mockito.mock(NicTO.class);
         final NicTO [] nicTOs = {nicTO1, nicTO2, nicTO3};
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
