--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -1811,9 +1811,8 @@
 
     @Test
     public void cleanupValidationTestStartedVmFalseAndValidationNotPreparedAndFailedToDestroyVolume() throws ResourceUnavailableException {
-        VolumeVO dataVolume = Mockito.mock(VolumeVO.class);
+        VolumeVO dataVolume = createMockVolumeVO(52L);
         doReturn(Volume.Type.DATADISK).when(dataVolume).getVolumeType();
-        doReturn(52L).when(dataVolume).getId();
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
@@ -1879,9 +1878,8 @@
         VolumeVO rootVolume = Mockito.mock(VolumeVO.class);
         doReturn(Volume.Type.ROOT).when(rootVolume).getVolumeType();
 
-        VolumeVO dataVolume = Mockito.mock(VolumeVO.class);
+        VolumeVO dataVolume = createMockVolumeVO(52L);
         doReturn(Volume.Type.DATADISK).when(dataVolume).getVolumeType();
-        doReturn(52L).when(dataVolume).getId();
 
         doReturn(77L).when(userVmVOMock).getId();
         doReturn(88L).when(userVmVOMock).getHostId();
@@ -2030,7 +2028,7 @@
 
     @Test
     public void mergeCurrentDeltaIntoVolumeTestNoDeltaDoesNothing() {
-        doReturn(volumeId).when(volumeVoMock).getId();
+        volumeVoMock = createMockVolumeVO(volumeId);
         doReturn(null).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(volumeId);
 
         kbossBackupProviderSpy.mergeCurrentDeltaIntoVolume(volumeVoMock, virtualMachineMock, "detach", true);
@@ -2041,7 +2039,7 @@
 
     @Test (expected = CloudRuntimeException.class)
     public void mergeCurrentDeltaIntoVolumeTestNullAnswer() {
-        doReturn(volumeId).when(volumeVoMock).getId();
+        volumeVoMock = createMockVolumeVO(volumeId);
         doReturn(internalBackupStoragePoolVoMock).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(volumeId);
         doReturn(backupId).when(internalBackupStoragePoolVoMock).getBackupId();
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(backupId);
@@ -2060,7 +2058,7 @@
 
     @Test
     public void mergeCurrentDeltaIntoVolumeTestNoSucceedingSnapshot() {
-        doReturn(volumeId).when(volumeVoMock).getId();
+        volumeVoMock = createMockVolumeVO(volumeId);
         doReturn(internalBackupStoragePoolVoMock).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(volumeId);
         doReturn(backupId).when(internalBackupStoragePoolVoMock).getBackupId();
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(backupId);
@@ -2088,7 +2086,7 @@
 
     @Test
     public void mergeCurrentDeltaIntoVolumeTestWithSucceedingSnapshotWithMoreDeltas() {
-        doReturn(volumeId).when(volumeVoMock).getId();
+        volumeVoMock = createMockVolumeVO(volumeId);
         doReturn(internalBackupStoragePoolVoMock).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(volumeId);
         doReturn(backupId).when(internalBackupStoragePoolVoMock).getBackupId();
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(backupId);
@@ -3113,4 +3111,10 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+
+    private static VolumeVO createMockVolumeVO(long volumeId) {
+        VolumeVO volume = Mockito.mock(VolumeVO.class);
+        doReturn(volumeId).when(volume).getId();
+        return volume;
+    }
 }
