--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -561,13 +561,11 @@
 
     @Test
     public void allocateDnsIpAddressTest() {
-        NetworkVO networkVO = mock(NetworkVO.class);
+        NetworkVO networkVO = createMockNetworkVO();
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenNetworkDnsCommand.class), anyLong())).thenReturn(getTungstenNetworkDnsAnswer);
-        when(networkVO.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
-        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
         when(tungstenGuestNetworkIpAddressDao.persist(any(TungstenGuestNetworkIpAddressVO.class))).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(networkVO.getCidr()).thenReturn("192.168.100.0/24");
         when(getTungstenNetworkDnsAnswer.getDetails()).thenReturn("192.168.100.100");
@@ -577,13 +575,11 @@
 
     @Test
     public void deallocateDnsIpAddressTest() {
-        NetworkVO networkVO = mock(NetworkVO.class);
         TungstenAnswer getTungstenNetworkDnsAnswer = MockTungstenAnswerFactory.get(true);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenNetworkDnsCommand.class), anyLong())).thenReturn(getTungstenNetworkDnsAnswer);
-        when(networkVO.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
-        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        NetworkVO networkVO = createMockNetworkVO();
         when(tungstenGuestNetworkIpAddressDao.findByNetworkAndGuestIpAddress(anyLong(), anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
         when(getTungstenNetworkDnsAnswer.getDetails()).thenReturn("192.168.100.100");
 
@@ -1668,4 +1664,11 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static NetworkVO createMockNetworkVO() {
+        NetworkVO networkVO = mock(NetworkVO.class);
+        when(networkVO.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
+        return networkVO;
+    }
 }
