--- a/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
@@ -73,6 +73,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class LoadBalancingRulesManagerImplTest{
+    @Mock
+    private LoadBalancerVO loadBalancerMock;
 
     @Mock
     NetworkDao _networkDao;
@@ -157,7 +159,6 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testGetLoadBalancerServiceProviderFail() {
-        LoadBalancerVO loadBalancerMock = Mockito.mock(LoadBalancerVO.class);
         NetworkVO networkMock = Mockito.mock(NetworkVO.class);
 
         when(_networkDao.findById(Mockito.any())).thenReturn(networkMock);
@@ -344,7 +345,7 @@
         long vmId = 30L;
         UserVm userVm = Mockito.mock(UserVm.class);
         when(userVm.getId()).thenReturn(vmId);
-        LoadBalancerVO loadBalancer = Mockito.mock(LoadBalancerVO.class);
+        // removed local mock; replaced with global field `loadBalancerMock`
         Network loadBalancerNetwork = Mockito.mock(Network.class);
         Account owner = Mockito.mock(Account.class);
 
@@ -360,7 +361,7 @@
         Nic nic = Mockito.mock(Nic.class);
         when(_networkModel.getNicInNetwork(Mockito.eq(vmId), Mockito.eq(networkTier2Id))).thenReturn(nic);
 
-        Nic nicInLb = lbr.getVmNicInLoadBalancer(userVm, loadBalancer, loadBalancerNetwork, vmIdNetworkIdMap, owner);
+        Nic nicInLb = lbr.getVmNicInLoadBalancer(userVm, loadBalancerMock, loadBalancerNetwork, vmIdNetworkIdMap, owner);
         Assert.assertEquals(nic, nicInLb);
     }
 
