--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -177,6 +177,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class AutoScaleManagerImplTest {
+    @Mock
+    private CounterTO counterTO2;
 
     @Spy
     @InjectMocks
@@ -1710,7 +1712,6 @@
         ConditionTO conditionTO1 = Mockito.mock(ConditionTO.class);
         CounterTO counterTO1 = Mockito.mock(CounterTO.class);
         ConditionTO conditionTO2 = Mockito.mock(ConditionTO.class);
-        CounterTO counterTO2 = Mockito.mock(CounterTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO1, conditionTO2));
@@ -2245,11 +2246,10 @@
         hostAndVmIdsMap.put(4L, Arrays.asList(5L, 6L));
 
         CounterTO scaleUpCounter = Mockito.mock(CounterTO.class);
-        CounterTO scaleDownCounter = Mockito.mock(CounterTO.class);
         Map<Long, List<CounterTO>> policyCountersMap = new HashMap<>();
         policyCountersMap.put(scaleUpPolicyId, Arrays.asList(scaleUpCounter));
-        policyCountersMap.put(scaleDownPolicyId, Arrays.asList(scaleDownCounter));
-
+
+        policyCountersMap.put(scaleDownPolicyId, Arrays.asList(counterTO2));
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
         Mockito.doReturn(hostAndVmIdsMap).when(autoScaleManagerImplSpy).getHostAndVmIdsMap(groupTO);
         Mockito.doReturn(policyCountersMap).when(autoScaleManagerImplSpy).getPolicyCounters(groupTO);
