--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -130,6 +130,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenElementTest {
     @Mock
+    private VirtualMachine virtualMachine;
+    @Mock
     TungstenFabricUtils tungstenFabricUtils;
     @Mock
     NetworkModel networkModel;
@@ -934,7 +936,6 @@
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         ReservationContext src = mock(ReservationContext.class);
         ReservationContext dest = mock(ReservationContext.class);
-        VirtualMachine virtualMachine = mock(VirtualMachine.class);
         HostVO hostVO = mock(HostVO.class);
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
@@ -952,7 +953,6 @@
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         ReservationContext src = mock(ReservationContext.class);
         ReservationContext dest = mock(ReservationContext.class);
-        VirtualMachine virtualMachine = mock(VirtualMachine.class);
         HostVO hostVO = mock(HostVO.class);
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
