--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
@@ -90,13 +90,8 @@
         }
 
         // Create Virtual=Machine-Interface (VMInterface)
-        NicVO nic = mock(NicVO.class);
-        when(nic.getIPv4Address()).thenReturn("10.1.1.2");
-        when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
-        when(nic.getDeviceId()).thenReturn(100);
-        when(nic.getState()).thenReturn(NicVO.State.Allocated);
-        when(nic.getNetworkId()).thenReturn(10L);
 
+        NicVO nic = MockNicVO.createNicVO();
         when(contrailMgr.getVifNameByVmName(anyString(), anyInt())).thenReturn("testVM1-100");
 
         VMInterfaceModel vmiModel = new VMInterfaceModel(UUID.randomUUID().toString());
@@ -125,4 +120,19 @@
             fail("ipInstance update failed ");
         }
     }
+
+    public final class MockNicVO {
+        private MockNicVO() {
+        }
+
+        public static com.cloud.vm.NicVO createNicVO() {
+            com.cloud.vm.NicVO nic = mock(com.cloud.vm.NicVO.class);
+            when(nic.getIPv4Address()).thenReturn("10.1.1.2");
+            when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
+            when(nic.getDeviceId()).thenReturn(100);
+            when(nic.getState()).thenReturn(com.cloud.vm.NicVO.State.Allocated);
+            when(nic.getNetworkId()).thenReturn(10L);
+            return nic;
+        }
+    }
 }
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
@@ -92,13 +92,8 @@
         }
 
         // Create Virtual=Machine-Interface (VMInterface)
-        NicVO nic = mock(NicVO.class);
-        when(nic.getIPv4Address()).thenReturn("10.1.1.2");
-        when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
-        when(nic.getDeviceId()).thenReturn(100);
-        when(nic.getState()).thenReturn(NicVO.State.Allocated);
-        when(nic.getNetworkId()).thenReturn(10L);
 
+        NicVO nic = MockNicVO.createNicVO();
         when(contrailMgr.getVifNameByVmName(anyString(), anyInt())).thenReturn("testVM1-100");
 
         uuid = UUID.randomUUID().toString();
@@ -130,4 +125,19 @@
             fail("vm-interface verify failed ");
         }
     }
+
+    public final class MockNicVO {
+        private MockNicVO() {
+        }
+
+        public static com.cloud.vm.NicVO createNicVO() {
+            com.cloud.vm.NicVO nic = mock(com.cloud.vm.NicVO.class);
+            when(nic.getIPv4Address()).thenReturn("10.1.1.2");
+            when(nic.getMacAddress()).thenReturn("00:01:02:03:04:05");
+            when(nic.getDeviceId()).thenReturn(100);
+            when(nic.getState()).thenReturn(com.cloud.vm.NicVO.State.Allocated);
+            when(nic.getNetworkId()).thenReturn(10L);
+            return nic;
+        }
+    }
 }
