--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -175,6 +175,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VirtualMachineManagerImplTest {
+    @Mock
+    private VMTemplateVO template;
 
     @Spy
     @InjectMocks
@@ -1011,7 +1013,6 @@
         Mockito.when(vm.getTemplateId()).thenReturn(templateId);
         Mockito.when(vm.getDataCenterId()).thenReturn(dcId);
         Mockito.when(volumeDaoMock.findReadyRootVolumesByInstance(vmId)).thenReturn(null);
-        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
         Mockito.when(vm.getId()).thenReturn(templateId);
         Mockito.when(templateDao.findById(templateId)).thenReturn(template);
         virtualMachineManagerImpl.checkIfTemplateNeededForCreatingVmVolumes(vm);
@@ -1723,7 +1724,6 @@
         ServiceOfferingVO offering = Mockito.mock(ServiceOfferingVO.class);
         when(serviceOfferingDaoMock.findById(vmInstanceVoMockId, offeringId)).thenReturn(offering);
 
-        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
         when(_entityMgr.findByIdIncludingRemoved(VirtualMachineTemplate.class, templateId)).thenReturn(template);
 
         when(hostMock.getClusterId()).thenReturn(clusterMockId);
