--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/KvmFileBasedStorageVmSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/KvmFileBasedStorageVmSnapshotStrategyTest.java
@@ -370,9 +370,8 @@
         long hostId = 40L;
 
         UserVmVO userVm = mock(UserVmVO.class);
-        VMSnapshotVO vmSnapshot = mock(VMSnapshotVO.class);
-
-        when(vmSnapshot.getVmId()).thenReturn(vmId);
+
+        VMSnapshotVO vmSnapshot = createMockVMSnapshot(vmId);
         when(userVm.getId()).thenReturn(vmId);
         when(userVm.getUuid()).thenReturn("vm-uuid");
         when(userVm.getState()).thenReturn(VirtualMachine.State.Running);
@@ -393,9 +392,8 @@
         long hostId = 40L;
 
         UserVmVO userVm = mock(UserVmVO.class);
-        VMSnapshotVO vmSnapshot = mock(VMSnapshotVO.class);
-
-        when(vmSnapshot.getVmId()).thenReturn(vmId);
+
+        VMSnapshotVO vmSnapshot = createMockVMSnapshot(vmId);
         when(userVm.getId()).thenReturn(vmId);
         when(userVm.getUuid()).thenReturn("vm-uuid");
         when(userVm.getState()).thenReturn(VirtualMachine.State.Running);
@@ -445,9 +443,8 @@
         long hostId = 40L;
 
         UserVmVO userVm = mock(UserVmVO.class);
-        VMSnapshotVO vmSnapshot = mock(VMSnapshotVO.class);
-
-        when(vmSnapshot.getVmId()).thenReturn(vmId);
+
+        VMSnapshotVO vmSnapshot = createMockVMSnapshot(vmId);
         when(userVm.getId()).thenReturn(vmId);
         when(userVm.getUuid()).thenReturn("vm-uuid");
         when(userVm.getState()).thenReturn(VirtualMachine.State.Running);
@@ -468,9 +465,8 @@
         long hostId = 40L;
 
         UserVmVO userVm = mock(UserVmVO.class);
-        VMSnapshotVO vmSnapshot = mock(VMSnapshotVO.class);
-
-        when(vmSnapshot.getVmId()).thenReturn(vmId);
+
+        VMSnapshotVO vmSnapshot = createMockVMSnapshot(vmId);
         when(userVm.getId()).thenReturn(vmId);
         when(userVm.getUuid()).thenReturn("vm-uuid");
         when(userVm.getState()).thenReturn(VirtualMachine.State.Stopped);
@@ -492,9 +488,8 @@
         long hostId = 40L;
 
         UserVmVO userVm = mock(UserVmVO.class);
-        VMSnapshotVO vmSnapshot = mock(VMSnapshotVO.class);
-
-        when(vmSnapshot.getVmId()).thenReturn(vmId);
+
+        VMSnapshotVO vmSnapshot = createMockVMSnapshot(vmId);
         when(userVm.getId()).thenReturn(vmId);
         when(userVm.getUuid()).thenReturn("vm-uuid");
         when(userVm.getState()).thenReturn(VirtualMachine.State.Stopped);
@@ -547,4 +542,10 @@
         when(host.getId()).thenReturn(hostId);
         when(hostDao.listAllUpAndEnabledNonHAHosts(Host.Type.Routing, clusterId, podId, dataCenterId, null)).thenReturn(List.of(host));
     }
+
+    private static VMSnapshotVO createMockVMSnapshot(long vmId) {
+        VMSnapshotVO vmSnapshot = mock(VMSnapshotVO.class);
+        when(vmSnapshot.getVmId()).thenReturn(vmId);
+        return vmSnapshot;
+    }
 }
