--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -95,6 +95,8 @@
 import com.cloud.vm.snapshot.VMSnapshotVO;
 
 public class AccountManagerImplTest extends AccountManagentImplTestBase {
+    @Mock
+    private UserVO userVO;
 
     @Mock
     private UserVmManagerImpl _vmMgr;
@@ -1136,7 +1138,6 @@
         Long userId = 1L;
 
         UserAccountVO userAccount = Mockito.mock(UserAccountVO.class);
-        UserVO userVO = Mockito.mock(UserVO.class);
 
         Mockito.when(userAccountDao.findById(userId)).thenReturn(userAccount);
         Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
@@ -1155,8 +1156,8 @@
         Long userId = 1L;
 
         UserAccountVO userAccount = Mockito.mock(UserAccountVO.class);
-        UserVO userVO = Mockito.mock(UserVO.class);
-
+
+        // removed local mock; replaced with global field `userVO`
         Mockito.when(userAccountDao.findById(userId)).thenReturn(userAccount);
         Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
@@ -1184,7 +1185,6 @@
         Long userId = 1L;
 
         UserAccountVO userAccount = Mockito.mock(UserAccountVO.class);
-        UserVO userVO = Mockito.mock(UserVO.class);
 
         Mockito.when(userAccountDao.findById(userId)).thenReturn(userAccount);
         Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
