--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1179,11 +1179,9 @@
         List<StoragePoolVO> pools = new ArrayList<>(Arrays.asList(storagePoolVO));
         List<VolumeVO> volumes = new ArrayList<>(Arrays.asList(volumeVO));
 
-        StoragePoolTagVO poolTagMock1 = Mockito.mock(StoragePoolTagVO.class);
-        StoragePoolTagVO poolTagMock2 = Mockito.mock(StoragePoolTagVO.class);
+        StoragePoolTagVO poolTagMock1 = createStoragePoolTagMock("tag1");
+        StoragePoolTagVO poolTagMock2 = createStoragePoolTagMock("tag2");
         List<StoragePoolTagVO> poolTags = List.of(poolTagMock1, poolTagMock2);
-        Mockito.doReturn("tag1").when(poolTagMock1).getTag();
-        Mockito.doReturn("tag2").when(poolTagMock2).getTag();
 
         Mockito.when(primaryDataStoreDao.listStoragePoolsWithActiveVolumesByOfferingId(anyLong())).thenReturn(pools);
         Mockito.when(storagePoolTagsDao.findStoragePoolTags(anyLong())).thenReturn(poolTags);
@@ -1508,4 +1506,10 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static StoragePoolTagVO createStoragePoolTagMock(String tag) {
+        StoragePoolTagVO poolTagMock = Mockito.mock(StoragePoolTagVO.class);
+        Mockito.doReturn(tag).when(poolTagMock).getTag();
+        return poolTagMock;
+    }
 }
