--- a/engine/storage/volume/src/test/java/org/apache/cloudstack/storage/volume/VolumeServiceTest.java
+++ b/engine/storage/volume/src/test/java/org/apache/cloudstack/storage/volume/VolumeServiceTest.java
@@ -61,6 +61,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VolumeServiceTest extends TestCase{
+    @Mock
+    private StoragePool pool;
 
     @Spy
     VolumeServiceImpl volumeServiceImplSpy;
@@ -257,7 +259,6 @@
     public void testCheckAndRepairVolume() throws StorageUnavailableException {
         VolumeInfo volume = Mockito.mock(VolumeInfo.class);
         Mockito.when(volume.getPoolId()).thenReturn(1L);
-        StoragePool pool = Mockito.mock(StoragePool.class);
         Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(pool);
         List<Long> hostIds = new ArrayList<>();
         hostIds.add(1L);
@@ -299,7 +300,7 @@
     public void testCheckAndRepairVolumeWhenFailure() throws StorageUnavailableException {
         VolumeInfo volume = Mockito.mock(VolumeInfo.class);
         Mockito.when(volume.getPoolId()).thenReturn(1L);
-        StoragePool pool = Mockito.mock(StoragePool.class);
+        // removed local mock; replaced with global field `pool`
         Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(pool);
         List<Long> hostIds = new ArrayList<>();
         hostIds.add(1L);
