--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -460,8 +460,7 @@
         long templateId = 1L;
         Account account = mock(Account.class);
         Mockito.when(account.getType()).thenReturn(Account.Type.ADMIN);
-        VMTemplateVO template = mock(VMTemplateVO.class);
-        Mockito.when(template.getTemplateTag()).thenReturn("tag");
+        VMTemplateVO template = createMockVmTemplate();
         Mockito.when(templateDao.findByIdIncludingRemoved(templateId)).thenReturn(template);
         Mockito.lenient().doThrow(PermissionDeniedException.class).when(accountManager).checkAccess(account, null, false, template);
         List<String> result = queryManager.getHostTagsFromTemplateForServiceOfferingsListing(account, templateId);
@@ -473,12 +472,11 @@
         long templateId = 1L;
         Account account = mock(Account.class);
         Mockito.when(account.getType()).thenReturn(Account.Type.NORMAL);
-        VMTemplateVO template = mock(VMTemplateVO.class);
+        VMTemplateVO template = createMockVmTemplate();
         Mockito.when(templateDao.findByIdIncludingRemoved(templateId)).thenReturn(template);
         Mockito.lenient().doNothing().when(accountManager).checkAccess(account, null, false, template);
         List<String> result = queryManager.getHostTagsFromTemplateForServiceOfferingsListing(account, templateId);
         Assert.assertTrue(CollectionUtils.isEmpty(result));
-        Mockito.when(template.getTemplateTag()).thenReturn("tag");
         result = queryManager.getHostTagsFromTemplateForServiceOfferingsListing(account, templateId);
         Assert.assertTrue(CollectionUtils.isNotEmpty(result));
     }
@@ -891,4 +889,10 @@
         assertEquals(1, hostTags.size());
         assertTrue(hostTags.contains(tag1));
     }
+
+    private static VMTemplateVO createMockVmTemplate() {
+        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
+        Mockito.when(template.getTemplateTag()).thenReturn("tag");
+        return template;
+    }
 }
