--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -155,6 +155,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VolumeApiServiceImplTest {
+    @Mock
+    private HostPodVO pod;
 
     @Spy
     @InjectMocks
@@ -2143,7 +2145,6 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
         ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        HostPodVO pod = Mockito.mock(HostPodVO.class);
         DataCenterVO zone = mockZone();
         mockDiskOffering();
         StoragePool pool = Mockito.mock(StoragePool.class);
@@ -2167,7 +2168,6 @@
         DataCenterVO zone = mockZone();
         Pair<Long, Long> clusterHostId = new Pair<>(1L, 2L);
         ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        HostPodVO pod = Mockito.mock(HostPodVO.class);
         mockDiskOffering();
         when(vm.getDataCenterId()).thenReturn(1L);
         when(clusterDao.findById(1L)).thenReturn(cluster);
@@ -2184,7 +2184,6 @@
         VolumeInfo volumeToAttach = Mockito.mock(VolumeInfo.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
         DataCenterVO zone = mockZone();
-        HostPodVO pod = Mockito.mock(HostPodVO.class);
         mockDiskOffering();
         StoragePool pool = Mockito.mock(StoragePool.class);
         when(vm.getDataCenterId()).thenReturn(1L);
