--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -963,10 +963,8 @@
 
         SearchBuilder<AutoScalePolicyVO> searchBuilderMock = Mockito.mock(SearchBuilder.class);
         SearchCriteria<AutoScalePolicyVO> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
-        when(asScaleUpPolicyMock.getDuration()).thenReturn(scaleUpPolicyDuration);
-        when(asScaleUpPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEUP);
-        when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
-        when(asScaleDownPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEDOWN);
+        asScaleUpPolicyMock = createAutoScalePolicyMock(scaleUpPolicyDuration, AutoScalePolicy.Action.SCALEUP);
+        asScaleDownPolicyMock = createAutoScalePolicyMock(scaleDownPolicyDuration, AutoScalePolicy.Action.SCALEDOWN);
 
         Mockito.doReturn(searchBuilderMock).when(asPolicyDao).createSearchBuilder();
         when(searchBuilderMock.entity()).thenReturn(asScaleUpPolicyMock);
@@ -1050,10 +1048,8 @@
 
         SearchBuilder<AutoScalePolicyVO> searchBuilderMock = Mockito.mock(SearchBuilder.class);
         SearchCriteria<AutoScalePolicyVO> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
-        when(asScaleUpPolicyMock.getDuration()).thenReturn(scaleUpPolicyDuration);
-        when(asScaleUpPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEUP);
-        when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
-        when(asScaleDownPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEDOWN);
+        asScaleUpPolicyMock = createAutoScalePolicyMock(scaleUpPolicyDuration, AutoScalePolicy.Action.SCALEUP);
+        asScaleDownPolicyMock = createAutoScalePolicyMock(scaleDownPolicyDuration, AutoScalePolicy.Action.SCALEDOWN);
 
         Mockito.doReturn(searchBuilderMock).when(asPolicyDao).createSearchBuilder();
         when(searchBuilderMock.entity()).thenReturn(asScaleUpPolicyMock);
@@ -2581,4 +2577,11 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static AutoScalePolicyVO createAutoScalePolicyMock(int duration, AutoScalePolicy.Action action) {
+        AutoScalePolicyVO policyMock = Mockito.mock(AutoScalePolicyVO.class);
+        when(policyMock.getAction()).thenReturn(action);
+        when(policyMock.getDuration()).thenReturn(duration);
+        return policyMock;
+    }
 }
