--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -151,9 +151,8 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDeviceVO();
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
 
         final NetworkOffering offering = mock(NetworkOffering.class);
         when(offering.getId()).thenReturn(NETWORK_ID);
@@ -224,9 +223,8 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDeviceVO();
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
 
         final NetworkOffering offering = mock(NetworkOffering.class);
         when(offering.getId()).thenReturn(NETWORK_ID);
@@ -250,9 +248,8 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDeviceVO();
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
 
         final NetworkOffering offering = mock(NetworkOffering.class);
         when(offering.getId()).thenReturn(NETWORK_ID);
@@ -304,10 +301,9 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
-        when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
-
+
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDeviceVO();
+        when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] { device }));
         final NetworkOffering offering = mock(NetworkOffering.class);
         when(offering.getId()).thenReturn(NETWORK_ID);
         when(offering.getTrafficType()).thenReturn(TrafficType.Guest);
@@ -362,9 +358,8 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDeviceVO();
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
 
         final NetworkOffering offering = mock(NetworkOffering.class);
         when(offering.getId()).thenReturn(NETWORK_ID);
@@ -416,9 +411,8 @@
         when(physnet.getIsolationMethods()).thenReturn(Arrays.asList(new String[] {"STT", "VXLAN"}));
         when(physnet.getId()).thenReturn(NETWORK_ID);
 
-        final NiciraNvpDeviceVO device = mock(NiciraNvpDeviceVO.class);
+        final NiciraNvpDeviceVO device = createMockNiciraNvpDeviceVO();
         when(nvpdao.listByPhysicalNetwork(NETWORK_ID)).thenReturn(Arrays.asList(new NiciraNvpDeviceVO[] {device}));
-        when(device.getId()).thenReturn(1L);
 
         final NetworkOffering offering = mock(NetworkOffering.class);
         when(offering.getId()).thenReturn(NETWORK_ID);
@@ -469,4 +463,10 @@
         verify(agentmgr, times(1)).easySend(eq(NETWORK_ID), (Command)any());
         verify(implementednetwork, times(1)).setBroadcastUri(null);
     }
+
+    private static NiciraNvpDeviceVO createMockNiciraNvpDeviceVO() {
+        final NiciraNvpDeviceVO device = Mockito.mock(NiciraNvpDeviceVO.class);
+        when(device.getId()).thenReturn(1L);
+        return device;
+    }
 }
