--- a/plugins/network-elements/globodns/src/test/java/com/globo/globodns/cloudstack/element/GloboDnsElementTest.java
+++ b/plugins/network-elements/globodns/src/test/java/com/globo/globodns/cloudstack/element/GloboDnsElementTest.java
@@ -155,8 +155,7 @@
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         when(vm.getHostName()).thenReturn("vm-name");
         when(vm.getType()).thenReturn(VirtualMachine.Type.User);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-        when(dataCenterVO.getId()).thenReturn(zoneId);
+        DataCenterVO dataCenterVO = createMockDataCenterVO(zoneId);
         when(_datacenterDao.findById(zoneId)).thenReturn(dataCenterVO);
         DeployDestination dest = new DeployDestination();
         ReservationContext context = new ReservationContextImpl(null, null, user);
@@ -188,8 +187,7 @@
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         when(vm.getHostName()).thenReturn("vm-name");
         when(vm.getType()).thenReturn(VirtualMachine.Type.User);
-        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
-        when(dataCenterVO.getId()).thenReturn(zoneId);
+        DataCenterVO dataCenterVO = createMockDataCenterVO(zoneId);
         when(_datacenterDao.findById(zoneId)).thenReturn(dataCenterVO);
         ReservationContext context = new ReservationContextImpl(null, null, user);
 
@@ -263,4 +261,10 @@
             }
         }
     }
+
+    private static DataCenterVO createMockDataCenterVO(long zoneId) {
+        DataCenterVO dataCenterVO = mock(DataCenterVO.class);
+        when(dataCenterVO.getId()).thenReturn(zoneId);
+        return dataCenterVO;
+    }
 }
