--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -496,8 +496,7 @@
     public void testGetCheckDatastorePolicyComplianceAnswerAgentException() throws StorageUnavailableException {
         StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(pool.getId()).thenReturn(1L);
-        VsphereStoragePolicyVO policy = Mockito.mock(VsphereStoragePolicyVO.class);
-        Mockito.when(policy.getPolicyId()).thenReturn("some");
+        VsphereStoragePolicyVO policy = createVsphereStoragePolicyVO();
         Mockito.when(vsphereStoragePolicyDao.findById(Mockito.anyLong()))
                 .thenReturn(policy);
         Mockito.doReturn(new ArrayList<>(List.of(1L, 2L))).when(storageManagerImpl).getUpHostsInPool(Mockito.anyLong());
@@ -523,8 +522,7 @@
     public void testGetCheckDatastorePolicyComplianceAnswerSuccess() throws StorageUnavailableException {
         StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(pool.getId()).thenReturn(1L);
-        VsphereStoragePolicyVO policy = Mockito.mock(VsphereStoragePolicyVO.class);
-        Mockito.when(policy.getPolicyId()).thenReturn("some");
+        VsphereStoragePolicyVO policy = createVsphereStoragePolicyVO();
         Mockito.when(vsphereStoragePolicyDao.findById(Mockito.anyLong()))
                 .thenReturn(policy);
         Mockito.doReturn(new ArrayList<>(List.of(1L, 2L))).when(storageManagerImpl).getUpHostsInPool(Mockito.anyLong());
@@ -1727,4 +1725,10 @@
 
         Assert.assertFalse(result);
     }
+
+    private static VsphereStoragePolicyVO createVsphereStoragePolicyVO() {
+        VsphereStoragePolicyVO policy = Mockito.mock(VsphereStoragePolicyVO.class);
+        Mockito.when(policy.getPolicyId()).thenReturn("some");
+        return policy;
+    }
 }
