--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -1271,7 +1271,6 @@
     @Test
     public void testUpdateHostPasswordCommand() {
         final XenServerUtilitiesHelper xenServerUtilitiesHelper = Mockito.mock(XenServerUtilitiesHelper.class);
-        final Pair<Boolean, String> result = Mockito.mock(Pair.class);
 
         final UpdateHostPasswordCommand updatePwd = new UpdateHostPasswordCommand("test", "123", "127.0.0.1");
 
@@ -1290,9 +1289,8 @@
                         cmdLine.toString());
 
         try {
+            final Pair<Boolean, String> result = createPairResult(true);
             when(xenServerUtilitiesHelper.executeSshWrapper(hostIp, 22, username, null, hostPasswd, cmdLine.toString())).thenReturn(result);
-            when(result.first()).thenReturn(true);
-            when(result.second()).thenReturn("");
         } catch (final Exception e) {
             fail(e.getMessage());
         }
@@ -1320,7 +1318,6 @@
     @Test
     public void testUpdateHostPasswordCommandFail() {
         final XenServerUtilitiesHelper xenServerUtilitiesHelper = Mockito.mock(XenServerUtilitiesHelper.class);
-        final Pair<Boolean, String> result = Mockito.mock(Pair.class);
 
         final UpdateHostPasswordCommand updatePwd = new UpdateHostPasswordCommand("test", "123", "127.0.0.1");
 
@@ -1338,10 +1335,9 @@
         when(xenServerUtilitiesHelper.buildCommandLine(XenServerUtilitiesHelper.SCRIPT_CMD_PATH, VRScripts.UPDATE_HOST_PASSWD, username, newPassword)).thenReturn(
                         cmdLine.toString());
 
+        final Pair<Boolean, String> result = createPairResult(false);
         try {
             when(xenServerUtilitiesHelper.executeSshWrapper(hostIp, 22, username, null, hostPasswd, cmdLine.toString())).thenReturn(result);
-            when(result.first()).thenReturn(false);
-            when(result.second()).thenReturn("");
         } catch (final Exception e) {
             fail(e.getMessage());
         }
@@ -1929,4 +1925,12 @@
         return false;
     }
 
+
+    @SuppressWarnings("unchecked")
+    private static Pair<Boolean, String> createPairResult(final boolean firstReturn) {
+        final Pair<Boolean, String> result = Mockito.mock(Pair.class);
+        when(result.first()).thenReturn(firstReturn);
+        when(result.second()).thenReturn("");
+        return result;
+    }
 }
