--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
@@ -131,17 +131,13 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateVpcTierDefaultDenyRule() {
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getState()).thenReturn(Network.State.Implemented);
-        Mockito.when(network.getNetworkACLId()).thenReturn(NetworkACL.DEFAULT_DENY);
+        Network network = createMockNetwork(NetworkACL.DEFAULT_DENY);
         kubernetesClusterManager.validateVpcTier(network);
     }
 
     @Test
     public void testValidateVpcTierValid() {
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getState()).thenReturn(Network.State.Implemented);
-        Mockito.when(network.getNetworkACLId()).thenReturn(NetworkACL.DEFAULT_ALLOW);
+        Network network = createMockNetwork(NetworkACL.DEFAULT_ALLOW);
         kubernetesClusterManager.validateVpcTier(network);
     }
 
@@ -914,4 +910,11 @@
         Mockito.verify(kubernetesClusterAffinityGroupMapDao).listAffinityGroupIdsByClusterIdAndNodeType(1L, WORKER.name());
     }
 
+
+    private static Network createMockNetwork(long networkAclId) {
+        Network network = Mockito.mock(Network.class);
+        Mockito.when(network.getState()).thenReturn(Network.State.Implemented);
+        Mockito.when(network.getNetworkACLId()).thenReturn(networkAclId);
+        return network;
+    }
 }
