--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesServiceHelperImplTest.java
@@ -80,8 +80,7 @@
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getId()).thenReturn(1L);
         Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
-        KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(map.getClusterId()).thenReturn(1L);
+        KubernetesClusterVmMapVO map = createKubernetesClusterVmMapVO();
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(map);
         KubernetesClusterVO kubernetesCluster = Mockito.mock(KubernetesClusterVO.class);
         Mockito.when(kubernetesClusterDao.findById(1L)).thenReturn(kubernetesCluster);
@@ -94,8 +93,7 @@
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getId()).thenReturn(1L);
         Mockito.when(vm.getUserVmType()).thenReturn(UserVmManager.CKS_NODE);
-        KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
-        Mockito.when(map.getClusterId()).thenReturn(1L);
+        KubernetesClusterVmMapVO map = createKubernetesClusterVmMapVO();
         Mockito.when(kubernetesClusterVmMapDao.findByVmId(1L)).thenReturn(map);
         KubernetesClusterVO kubernetesCluster = Mockito.mock(KubernetesClusterVO.class);
         Mockito.when(kubernetesClusterDao.findById(1L)).thenReturn(kubernetesCluster);
@@ -400,4 +398,10 @@
         Assert.assertEquals(1, result.size());
         Assert.assertEquals(Arrays.asList(1L, 2L, 3L), result.get("CONTROL"));
     }
+
+    private static KubernetesClusterVmMapVO createKubernetesClusterVmMapVO() {
+        KubernetesClusterVmMapVO map = Mockito.mock(KubernetesClusterVmMapVO.class);
+        Mockito.when(map.getClusterId()).thenReturn(1L);
+        return map;
+    }
 }
