--- a/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
@@ -121,10 +121,8 @@
 
     @Test
     public void testGetSortedTemplatesListWithPreferredArch_NoPreferred() {
-        VMTemplateVO template1 = Mockito.mock(VMTemplateVO.class);
-        when(template1.getId()).thenReturn(1L);
-        VMTemplateVO template2 = Mockito.mock(VMTemplateVO.class);
-        when(template2.getId()).thenReturn(2L);
+        VMTemplateVO template1 = createMockVmTemplate(1L);
+        VMTemplateVO template2 = createMockVmTemplate(2L);
         Map<Pair<Hypervisor.HypervisorType, CPU.CPUArch>, VMTemplateVO> uniqueTemplates = new HashMap<>();
         uniqueTemplates.put(new Pair<>(Hypervisor.HypervisorType.KVM, CPU.CPUArch.amd64), template1);
         uniqueTemplates.put(new Pair<>(Hypervisor.HypervisorType.KVM, CPU.CPUArch.arm64), template2);
@@ -418,4 +416,10 @@
 
         assertNull(result);
     }
+
+    private static VMTemplateVO createMockVmTemplate(long id) {
+        VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
+        when(template.getId()).thenReturn(id);
+        return template;
+    }
 }
