--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
@@ -442,8 +442,7 @@
     @Test
     public void testGetCksClusterPreferredArchDifferentArchsPreferCKSIsoArch() {
         String systemVMArch = "x86_64";
-        VMTemplateVO cksIso = Mockito.mock(VMTemplateVO.class);
-        Mockito.when(cksIso.getArch()).thenReturn(CPU.CPUArch.arm64);
+        VMTemplateVO cksIso = createMockVmTemplateVo(CPU.CPUArch.arm64);
         String cksClusterPreferredArch = kubernetesClusterManager.getCksClusterPreferredArch(systemVMArch, cksIso);
         Assert.assertEquals(CPU.CPUArch.arm64.getType(), cksClusterPreferredArch);
     }
@@ -451,8 +450,7 @@
     @Test
     public void testGetCksClusterPreferredArchSameArch() {
         String systemVMArch = "x86_64";
-        VMTemplateVO cksIso = Mockito.mock(VMTemplateVO.class);
-        Mockito.when(cksIso.getArch()).thenReturn(CPU.CPUArch.amd64);
+        VMTemplateVO cksIso = createMockVmTemplateVo(CPU.CPUArch.amd64);
         String cksClusterPreferredArch = kubernetesClusterManager.getCksClusterPreferredArch(systemVMArch, cksIso);
         Assert.assertEquals(CPU.CPUArch.amd64.getType(), cksClusterPreferredArch);
     }
@@ -914,4 +912,10 @@
         Mockito.verify(kubernetesClusterAffinityGroupMapDao).listAffinityGroupIdsByClusterIdAndNodeType(1L, WORKER.name());
     }
 
+
+    private static VMTemplateVO createMockVmTemplateVo(CPU.CPUArch arch) {
+        VMTemplateVO vmTemplate = Mockito.mock(VMTemplateVO.class);
+        Mockito.when(vmTemplate.getArch()).thenReturn(arch);
+        return vmTemplate;
+    }
 }
