--- a/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/LoadBalanceRuleHandlerTest.java
+++ b/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/LoadBalanceRuleHandlerTest.java
@@ -76,8 +76,7 @@
 
     @Test
     public void testAddCandidateVmIsPodIpMatchesWhenIdsHaveSameValue() throws Exception {
-        DomainRouterVO candidateVmMock = mock(DomainRouterVO.class);
-        when(candidateVmMock.getPodIdToDeployIn()).thenReturn(new Long(1));
+        DomainRouterVO candidateVmMock = createMockDomainRouterVO(new Long(1));
         Long podIdForDirectIp = new Long(1);
         List<DomainRouterVO> candidateVms = new ArrayList<DomainRouterVO>();
 
@@ -88,8 +87,7 @@
 
     @Test
     public void testAddCandidateVmIsPodIpMatchesWhenPodIdForDirectIpIsNull() throws Exception {
-        DomainRouterVO candidateVmMock = mock(DomainRouterVO.class);
-        when(candidateVmMock.getPodIdToDeployIn()).thenReturn(new Long(1));
+        DomainRouterVO candidateVmMock = createMockDomainRouterVO(new Long(1));
         Long podIdForDirectIp = null;
         List<DomainRouterVO> candidateVms = new ArrayList<DomainRouterVO>();
 
@@ -111,8 +109,7 @@
 
     @Test(expected = NullPointerException.class)
     public void testAddCandidateVmIsPodIpMatchesWhenCandidateVmsIsNull() throws Exception {
-        DomainRouterVO candidateVmMock = mock(DomainRouterVO.class);
-        when(candidateVmMock.getPodIdToDeployIn()).thenReturn(new Long(1));
+        DomainRouterVO candidateVmMock = createMockDomainRouterVO(new Long(1));
         Long podIdForDirectIp = new Long(1);
         List<DomainRouterVO> candidateVms = null;
 
@@ -200,8 +197,7 @@
         PodVlanMapVO podVlanMapVoMock = mock(PodVlanMapVO.class);
         when(podVlanMapVoMock.getPodId()).thenReturn(podId);
         when(podVlanMapDao.listPodVlanMapsByVlan(podId)).thenReturn(podVlanMapVoMock);
-        DomainRouterVO unusedElbVmThatMatchesPodId = mock(DomainRouterVO.class);
-        when(unusedElbVmThatMatchesPodId.getPodIdToDeployIn()).thenReturn(podId);
+        DomainRouterVO unusedElbVmThatMatchesPodId = createMockDomainRouterVO(podId);
         List<DomainRouterVO> unusedElbVms = Arrays.asList(new DomainRouterVO[] {unusedElbVmThatMatchesPodId, mock(DomainRouterVO.class)});
         when(this.elasticLbVmMapDao.listUnusedElbVms()).thenReturn(unusedElbVms);
 
@@ -212,4 +208,10 @@
         assertEquals(expected, actual);
     }
 
+
+    private static DomainRouterVO createMockDomainRouterVO(Long podId) {
+        DomainRouterVO candidateVmMock = mock(DomainRouterVO.class);
+        when(candidateVmMock.getPodIdToDeployIn()).thenReturn(podId);
+        return candidateVmMock;
+    }
 }
