--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -352,16 +352,13 @@
         Mockito.when(accountMock.getType()).thenReturn(Account.Type.NORMAL);
 
         Mockito.when(accountMock.getDomainId()).thenReturn(callerDomainId);
-        TemplateJoinVO templateMock1 = mock(TemplateJoinVO.class);
-        Mockito.when(templateMock1.getDomainId()).thenReturn(callerDomainId);
+        TemplateJoinVO templateMock1 = createMockTemplateJoinVO(callerDomainId);
         Mockito.lenient().doReturn(false).when(queryManagerImplSpy).checkIfDomainSharesTemplates(callerDomainId);
 
-        TemplateJoinVO templateMock2 = mock(TemplateJoinVO.class);
-        Mockito.when(templateMock2.getDomainId()).thenReturn(sharableDomainId);
+        TemplateJoinVO templateMock2 = createMockTemplateJoinVO(sharableDomainId);
         Mockito.doReturn(true).when(queryManagerImplSpy).checkIfDomainSharesTemplates(sharableDomainId);
 
-        TemplateJoinVO templateMock3 = mock(TemplateJoinVO.class);
-        Mockito.when(templateMock3.getDomainId()).thenReturn(unsharableDomainId);
+        TemplateJoinVO templateMock3 = createMockTemplateJoinVO(unsharableDomainId);
         Mockito.doReturn(false).when(queryManagerImplSpy).checkIfDomainSharesTemplates(unsharableDomainId);
 
         List<TemplateJoinVO> publicTemplates = List.of(templateMock1, templateMock2, templateMock3);
@@ -891,4 +888,10 @@
         assertEquals(1, hostTags.size());
         assertTrue(hostTags.contains(tag1));
     }
+
+    private static TemplateJoinVO createMockTemplateJoinVO(long domainId) {
+        TemplateJoinVO templateMock = mock(TemplateJoinVO.class);
+        Mockito.when(templateMock.getDomainId()).thenReturn(domainId);
+        return templateMock;
+    }
 }
