--- a/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/lb/LoadBalancingRulesManagerImplTest.java
@@ -73,6 +73,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class LoadBalancingRulesManagerImplTest{
+    @Mock
+    private NetworkVO networkMock;
 
     @Mock
     NetworkDao _networkDao;
@@ -143,7 +145,6 @@
     @Test
     public void testGetLoadBalancerServiceProvider() {
         LoadBalancerVO loadBalancerMock = Mockito.mock(LoadBalancerVO.class);
-        NetworkVO networkMock = Mockito.mock(NetworkVO.class);
         List<Network.Provider> providers = Arrays.asList(Network.Provider.VirtualRouter);
 
         when(loadBalancerMock.getNetworkId()).thenReturn(10L);
@@ -158,7 +159,6 @@
     @Test(expected = CloudRuntimeException.class)
     public void testGetLoadBalancerServiceProviderFail() {
         LoadBalancerVO loadBalancerMock = Mockito.mock(LoadBalancerVO.class);
-        NetworkVO networkMock = Mockito.mock(NetworkVO.class);
 
         when(_networkDao.findById(Mockito.any())).thenReturn(networkMock);
         when(_networkMgr.getProvidersForServiceInNetwork(networkMock, Network.Service.Lb)).thenReturn(new ArrayList<>());
