--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
@@ -63,6 +63,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class NsxPublicNetworkGuruTest {
+    @Mock
+    private Account account;
 
     NetworkOffering offering;
 
@@ -121,8 +123,8 @@
     public void testDesign() {
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
         Network network = Mockito.mock(Network.class);
-        Account account = Mockito.mock(Account.class);
 
+        // removed local mock; replaced with global field `account`
 //        when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
 
         Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, account);
@@ -133,8 +135,8 @@
     public void testDesign_whenOfferingIsForGuestTraffic() {
         DeploymentPlan plan = Mockito.mock(DeploymentPlan.class);
         Network network = Mockito.mock(Network.class);
-        Account account = Mockito.mock(Account.class);
 
+        // removed local mock; replaced with global field `account`
         when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
 
         Network designedNetwork = guru.design(offering, plan, network, "net1", 1L, account);
