--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -66,6 +66,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class XenServer610WrapperTest {
+    @Mock
+    private final StorageFilerTO storage1;
 
     @Mock
     protected XenServer610Resource xenServer610Resource;
@@ -109,7 +111,6 @@
 
         final VolumeTO vol1 = Mockito.mock(VolumeTO.class);
         final VolumeTO vol2 = Mockito.mock(VolumeTO.class);
-        final StorageFilerTO storage1 = Mockito.mock(StorageFilerTO.class);
         final StorageFilerTO storage2 = Mockito.mock(StorageFilerTO.class);
 
         final Map<VolumeTO, StorageFilerTO> volumeToFiler = new  HashMap<VolumeTO, StorageFilerTO>();
@@ -182,7 +183,7 @@
 
         final VolumeTO vol1 = Mockito.mock(VolumeTO.class);
         final VolumeTO vol2 = Mockito.mock(VolumeTO.class);
-        final StorageFilerTO storage1 = Mockito.mock(StorageFilerTO.class);
+        // removed local mock; replaced with global field `storage1`
         final StorageFilerTO storage2 = Mockito.mock(StorageFilerTO.class);
 
         final List<Pair<VolumeTO, String>> volumeToFiler = new ArrayList<>();
