--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -987,9 +987,8 @@
         when(template.getUserDataId()).thenReturn(2L);
         when(template.getUserDataOverridePolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
-        UserDataVO apiUserDataVO = Mockito.mock(UserDataVO.class);
+        UserDataVO apiUserDataVO = createMockUserDataVO(templateUserData);
         doReturn(apiUserDataVO).when(userDataDao).findById(userDataId);
-        when(apiUserDataVO.getUserData()).thenReturn(templateUserData);
 
         String finalUserdata = userVmManagerImpl.finalizeUserData(null, userDataId, template);
 
@@ -1001,9 +1000,8 @@
         String userData = "testUserdata";
         Long userDataId = 1L;
 
-        UserDataVO apiUserDataVO = Mockito.mock(UserDataVO.class);
+        UserDataVO apiUserDataVO = createMockUserDataVO(userData);
         doReturn(apiUserDataVO).when(userDataDao).findById(userDataId);
-        when(apiUserDataVO.getUserData()).thenReturn(userData);
 
         VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
         when(template.getUserDataId()).thenReturn(null);
@@ -1020,9 +1018,8 @@
         VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
         when(template.getUserDataId()).thenReturn(2L);
         when(template.getUserDataOverridePolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
-        UserDataVO templateUserDataVO = Mockito.mock(UserDataVO.class);
+        UserDataVO templateUserDataVO = createMockUserDataVO(templateUserData);
         doReturn(templateUserDataVO).when(userDataDao).findById(2L);
-        when(templateUserDataVO.getUserData()).thenReturn(templateUserData);
 
         String finalUserdata = userVmManagerImpl.finalizeUserData(null, null, template);
 
@@ -1146,9 +1143,8 @@
 
         String testUserData = "testUserdata";
         when(cmd.getUserdataId()).thenReturn(1L);
-        UserDataVO apiUserDataVO = Mockito.mock(UserDataVO.class);
+        UserDataVO apiUserDataVO = createMockUserDataVO(testUserData);
         when(userDataDao.findById(1L)).thenReturn(apiUserDataVO);
-        when(apiUserDataVO.getUserData()).thenReturn(testUserData);
         when(cmd.getHttpMethod()).thenReturn(HTTPMethod.GET);
 
         when(userDataManager.validateUserData(testUserData, HTTPMethod.GET)).thenReturn(testUserData);
@@ -4740,4 +4736,10 @@
         Assert.assertEquals(Long.valueOf(500L), volume.getMinIops());
         Assert.assertEquals(Long.valueOf(2000L), volume.getMaxIops());
     }
+
+    private static UserDataVO createMockUserDataVO(String userData) {
+        UserDataVO userDataVO = Mockito.mock(UserDataVO.class);
+        when(userDataVO.getUserData()).thenReturn(userData);
+        return userDataVO;
+    }
 }
