--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -180,6 +180,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class UnmanagedVMsManagerImplTest {
+    @Mock
+    private DiskOfferingVO dataDiskOffering;
 
     @Spy
     @InjectMocks
@@ -1043,7 +1045,6 @@
         StoragePoolVO selectedPool = mock(StoragePoolVO.class);
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
         DiskOfferingVO rootDiskOffering = mock(DiskOfferingVO.class);
-        DiskOfferingVO dataDiskOffering = mock(DiskOfferingVO.class);
 
         when(serviceOffering.getDiskOfferingId()).thenReturn(21L);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(selectedPool);
@@ -1062,7 +1063,7 @@
         StoragePoolVO selectedPool = mock(StoragePoolVO.class);
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
         DiskOfferingVO rootDiskOffering = mock(DiskOfferingVO.class);
-        DiskOfferingVO dataDiskOffering = mock(DiskOfferingVO.class);
+        // removed local mock; replaced with global field `dataDiskOffering`
 
         when(serviceOffering.getDiskOfferingId()).thenReturn(31L);
         when(primaryDataStoreDao.findById(poolId)).thenReturn(selectedPool);
