--- a/server/src/test/java/com/cloud/vm/UserVmManagerTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerTest.java
@@ -786,9 +786,8 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testApplyUserDataInNetworkWithoutElement() throws Exception {
-        UserVm userVm = mock(UserVm.class);
-        lenient().when(userVm.getId()).thenReturn(1L);
-
+
+        UserVm userVm = createMockUserVm();
         when(_nicMock.getNetworkId()).thenReturn(2L);
         when(_networkMock.getNetworkOfferingId()).thenReturn(3L);
         when(_networkDao.findById(2L)).thenReturn(_networkMock);
@@ -803,9 +802,8 @@
 
     @Test
     public void testApplyUserDataSuccessful() throws Exception {
-        UserVm userVm = mock(UserVm.class);
-        lenient().when(userVm.getId()).thenReturn(1L);
-
+
+        UserVm userVm = createMockUserVm();
         when(_nicMock.getNetworkId()).thenReturn(2L);
         when(_networkMock.getNetworkOfferingId()).thenReturn(3L);
         when(_networkDao.findById(2L)).thenReturn(_networkMock);
@@ -839,4 +837,10 @@
         _userVmMgr.persistDeviceBusInfo(_vmMock, "lsilogic");
         verify(_vmDao, times(1)).saveDetails(any(UserVmVO.class));
     }
+
+    private static UserVm createMockUserVm() {
+        UserVm userVm = mock(UserVm.class);
+        lenient().when(userVm.getId()).thenReturn(1L);
+        return userVm;
+    }
 }
