--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -77,6 +77,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ScaleIOPrimaryDataStoreDriverTest {
+    @Mock
+    private DataTO srcDataTO;
 
     @Spy
     @InjectMocks
@@ -515,7 +517,7 @@
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
         DataObject srcData = Mockito.mock(DataObject.class);
-        DataTO srcDataTO = Mockito.mock(DataTO.class);
+        // removed local mock; replaced with global field `srcDataTO`
         when(srcData.getTO()).thenReturn(srcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
         DataTO destDataTO = Mockito.mock(DataTO.class);
@@ -538,7 +540,7 @@
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
         DataObject srcData = Mockito.mock(DataObject.class);
-        DataTO srcDataTO = Mockito.mock(DataTO.class);
+        // removed local mock; replaced with global field `srcDataTO`
         when(srcData.getTO()).thenReturn(srcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
         DataTO destDataTO = Mockito.mock(DataTO.class);
