--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterUpgradeWorkerTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterUpgradeWorkerTest.java
@@ -60,12 +60,9 @@
         long controlNodeId = 1L;
         long workerNode1Id = 2L;
         long workerNode2Id = 3L;
-        UserVm controlNode = Mockito.mock(UserVm.class);
-        Mockito.when(controlNode.getId()).thenReturn(controlNodeId);
-        UserVm workerNode1 = Mockito.mock(UserVm.class);
-        Mockito.when(workerNode1.getId()).thenReturn(workerNode1Id);
-        UserVm workerNode2 = Mockito.mock(UserVm.class);
-        Mockito.when(workerNode2.getId()).thenReturn(workerNode2Id);
+        UserVm controlNode = createMockUserVm(controlNodeId);
+        UserVm workerNode1 = createMockUserVm(workerNode1Id);
+        UserVm workerNode2 = createMockUserVm(workerNode2Id);
         KubernetesClusterVmMapVO controlNodeMap = Mockito.mock(KubernetesClusterVmMapVO.class);
         KubernetesClusterVmMapVO workerNode1Map = Mockito.mock(KubernetesClusterVmMapVO.class);
         KubernetesClusterVmMapVO workerNode2Map = Mockito.mock(KubernetesClusterVmMapVO.class);
@@ -80,4 +77,10 @@
         Assert.assertTrue(ids.contains(controlNodeId) && ids.contains(workerNode1Id));
         Assert.assertFalse(ids.contains(workerNode2Id));
     }
+
+    private static UserVm createMockUserVm(long id) {
+        UserVm userVm = Mockito.mock(UserVm.class);
+        Mockito.when(userVm.getId()).thenReturn(id);
+        return userVm;
+    }
 }
