--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVmIpAddressCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVmIpAddressCommandWrapperTest.java
@@ -260,13 +260,11 @@
     @Test
     public void testIpFromDomIfCommandExecutionFailure() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-            when(getVmIpAddressCommand.getVmName()).thenReturn("testVm");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.1.0/24");
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
+            GetVmIpAddressCommand getVmIpAddressCommand = createGetVmIpAddressCommand();
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(null);
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -282,13 +280,11 @@
     @Test
     public void testIpFromDhcpLeaseFileCommandExecutionFailure() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-            when(getVmIpAddressCommand.getVmName()).thenReturn("testVm");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.1.0/24");
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
+            GetVmIpAddressCommand getVmIpAddressCommand = createGetVmIpAddressCommand();
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(null);
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -353,4 +349,11 @@
                 scriptMock.close();
         }
     }
+
+    private static GetVmIpAddressCommand createGetVmIpAddressCommand() {
+        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
+        when(getVmIpAddressCommand.getVmName()).thenReturn("testVm");
+        when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.1.0/24");
+        return getVmIpAddressCommand;
+    }
 }
