--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -549,12 +549,9 @@
 
     @Test
     public void testCapacityListingForSingleNonGpuType() {
-        Capacity c1 = Mockito.mock(Capacity.class);
-        Mockito.when(c1.getCapacityType()).thenReturn((short)Resource.ResourceType.user_vm.getOrdinal());
-        Capacity c2 = Mockito.mock(Capacity.class);
-        Mockito.when(c2.getCapacityType()).thenReturn((short)Resource.ResourceType.user_vm.getOrdinal());
-        Capacity c3 = Mockito.mock(Capacity.class);
-        Mockito.when(c3.getCapacityType()).thenReturn((short)Resource.ResourceType.volume.getOrdinal());
+        Capacity c1 = createMockCapacity((short) Resource.ResourceType.user_vm.getOrdinal());
+        Capacity c2 = createMockCapacity((short) Resource.ResourceType.user_vm.getOrdinal());
+        Capacity c3 = createMockCapacity((short) Resource.ResourceType.volume.getOrdinal());
         Capacity c4 = Mockito.mock(Capacity.class);
         Assert.assertTrue(apiResponseHelper.capacityListingForSingleNonGpuType(List.of(c1, c2)));
         Assert.assertFalse(apiResponseHelper.capacityListingForSingleNonGpuType(List.of(c1, c2, c3)));
@@ -800,4 +797,10 @@
             Assert.assertEquals(expected.getVmName(), response.getVmName());
         }
     }
+
+    private static Capacity createMockCapacity(short capacityType) {
+        Capacity capacity = Mockito.mock(Capacity.class);
+        Mockito.when(capacity.getCapacityType()).thenReturn(capacityType);
+        return capacity;
+    }
 }
