--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -785,14 +785,12 @@
 
     @Test
     public void processHostAboutToBeRemovedWithSecurityGroupTest() {
-        HostVO hostVO = mock(HostVO.class);
+        HostVO hostVO = createMockHostVO();
         TungstenProviderVO tungstenProvider = mock(TungstenProviderVO.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
         NetworkVO network = mock(NetworkVO.class);
 
-        when(hostVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProvider);
-        when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
         when(tungstenProvider.getGateway()).thenReturn("192.168.100.100");
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
@@ -805,13 +803,11 @@
 
     @Test
     public void processHostAboutToBeRemovedWithoutSecurityGroupTest() {
-        HostVO hostVO = mock(HostVO.class);
         TungstenProviderVO tungstenProvider = mock(TungstenProviderVO.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
-        when(hostVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
+        HostVO hostVO = createMockHostVO();
         when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProvider);
-        when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
         when(tungstenProvider.getGateway()).thenReturn("192.168.100.100");
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
@@ -962,4 +958,11 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    private static HostVO createMockHostVO() {
+        HostVO hostVO = mock(HostVO.class);
+        when(hostVO.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.KVM);
+        when(hostVO.getPublicIpAddress()).thenReturn("192.168.100.100");
+        return hostVO;
+    }
 }
