--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -954,8 +954,7 @@
         Mockito.when(vm.getId()).thenReturn(vmId);
         Mockito.when(vm.getHypervisorType()).thenReturn(hypervisorType);
         Long networkId = 1L;
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getId()).thenReturn(networkId);
+        Network network = createMockNetwork(networkId);
         Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(ipAddresses.getIp4Address()).thenReturn(null);
         URI broadcastUri = URI.create("vlan://123");
@@ -991,8 +990,7 @@
         Mockito.when(vm.getId()).thenReturn(vmId);
         Mockito.when(vm.getHypervisorType()).thenReturn(hypervisorType);
         Long networkId = 1L;
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getId()).thenReturn(networkId);
+        Network network = createMockNetwork(networkId);
         String ipAddress = "10.1.10.10";
         Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(ipAddresses.getIp4Address()).thenReturn(ipAddress);
@@ -1073,4 +1071,10 @@
         assertNotNull(result);
         assertEquals(1, result.size());
     }
+
+    private static Network createMockNetwork(Long networkId) {
+        Network network = Mockito.mock(Network.class);
+        Mockito.when(network.getId()).thenReturn(networkId);
+        return network;
+    }
 }
