--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -130,6 +130,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenElementTest {
     @Mock
+    private LoadBalancerVMMapVO loadBalancerVMMapVO;
+    @Mock
     TungstenFabricUtils tungstenFabricUtils;
     @Mock
     NetworkModel networkModel;
@@ -342,7 +344,6 @@
         Network publicNetwork = mock(Network.class);
         Ip ip = mock(Ip.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
-        LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
         LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
@@ -398,7 +399,6 @@
         Network publicNetwork = mock(Network.class);
         Ip ip = mock(Ip.class);
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
-        LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
         LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
