--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVmIpAddressCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVmIpAddressCommandWrapperTest.java
@@ -66,14 +66,11 @@
     @Test
     public void testExecuteWithValidVmName() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
         LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
         MockedStatic<Script> scriptMock = mockStatic(Script.class);
 
-        when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-        when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
+        GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", false);
         when(getVmIpAddressCommand.getMacAddress()).thenReturn("02:0c:02:f9:00:80");
-        when(getVmIpAddressCommand.isWindows()).thenReturn(false);
         when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, VIRSH_DOMIF_OUTPUT));
 
         Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -89,14 +86,14 @@
     @Test
     public void testExecuteWithInvalidVmName() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
         LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
         MockedStatic<Script> scriptMock = mockStatic(Script.class);
 
-        when(getVmIpAddressCommand.getVmName()).thenReturn("invalidVmName!");
-        when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
         when(getVmIpAddressCommand.getMacAddress()).thenReturn("02:0c:02:f9:00:80");
-        when(getVmIpAddressCommand.isWindows()).thenReturn(false);
+        GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand(
+                "invalidVmName!",
+                "192.168.0.0/24",
+                false);
         when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, VIRSH_DOMIF_OUTPUT));
 
         Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -113,17 +110,14 @@
     @Test
     public void testExecuteWithWindowsVm() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-
-        try {
-            scriptMock = mockStatic(Script.class);
-
-            when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+
+        try {
+            scriptMock = mockStatic(Script.class);
+
+            GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", true);
             when(getVmIpAddressCommand.getMacAddress()).thenReturn("02:0c:02:f9:00:80");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(true);
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, VIRSH_DOMIF_OUTPUT_WINDOWS));
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -140,17 +134,14 @@
     @Test
     public void testExecuteWithWindowsVm2() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-
-        try {
-            scriptMock = mockStatic(Script.class);
-
-            when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+
+        try {
+            scriptMock = mockStatic(Script.class);
+
+            GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", true);
             when(getVmIpAddressCommand.getMacAddress()).thenReturn("02:0c:02:f9:00:80");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(true);
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, "")).thenReturn(new Pair<>(0, "192.168.0.10"));
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -166,15 +157,12 @@
     @Test
     public void testExecuteWithNoIpFound() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-
-            when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(false);
+        GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", false);
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
+
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, ""));
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -190,15 +178,12 @@
     @Test
     public void testExecuteWithValidVmNameAndNoIpFound() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-
-            when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(false);
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
+
+            GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", false);
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, ""));
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -214,14 +199,11 @@
     @Test
     public void testExecuteWithValidVmNameAndIpFromDhcpLeaseFile() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-            when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(false);
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
+            GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", false);
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, "192.168.0.10"));
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -237,14 +219,11 @@
     @Test
     public void testExecuteWithValidVmNameAndIpFromWindowsRegistry() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-            when(getVmIpAddressCommand.getVmName()).thenReturn("validVmName");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.0.0/24");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(true);
+        GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("validVmName", "192.168.0.0/24", true);
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(new Pair<>(0, "\"192.168.0.10\""));
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -304,14 +283,11 @@
     @Test
     public void testIpFromWindowsRegistryCommandExecutionFailure() {
         LibvirtComputingResource libvirtComputingResource = mock(LibvirtComputingResource.class);
-        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
-        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
-        MockedStatic<Script> scriptMock = null;
-        try {
-            scriptMock = mockStatic(Script.class);
-            when(getVmIpAddressCommand.getVmName()).thenReturn("testVm");
-            when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn("192.168.1.0/24");
-            when(getVmIpAddressCommand.isWindows()).thenReturn(true);
+        LibvirtGetVmIpAddressCommandWrapper commandWrapper = new LibvirtGetVmIpAddressCommandWrapper();
+        MockedStatic<Script> scriptMock = null;
+        try {
+            scriptMock = mockStatic(Script.class);
+            GetVmIpAddressCommand getVmIpAddressCommand = createMockGetVmIpAddressCommand("testVm", "192.168.1.0/24", true);
             when(Script.executePipedCommands(anyList(), anyLong())).thenReturn(null);
 
             Answer answer = commandWrapper.execute(getVmIpAddressCommand, libvirtComputingResource);
@@ -353,4 +329,15 @@
                 scriptMock.close();
         }
     }
+
+    private static GetVmIpAddressCommand createMockGetVmIpAddressCommand(
+            String vmName,
+            String vmNetworkCidr,
+            boolean windows) {
+        GetVmIpAddressCommand getVmIpAddressCommand = mock(GetVmIpAddressCommand.class);
+        when(getVmIpAddressCommand.getVmName()).thenReturn(vmName);
+        when(getVmIpAddressCommand.getVmNetworkCidr()).thenReturn(vmNetworkCidr);
+        when(getVmIpAddressCommand.isWindows()).thenReturn(windows);
+        return getVmIpAddressCommand;
+    }
 }
