--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -1046,10 +1046,8 @@
         List<VolumeVO> volumesInPool = Arrays.asList(volume1, volume2);
         Mockito.doReturn(volumesInPool).when(volumeDao).findNonDestroyedVolumesByPoolId(poolId);
 
-        VMInstanceVO vmInstance1 = Mockito.mock(VMInstanceVO.class);
-        VMInstanceVO vmInstance2 = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vmInstance1.getHostId()).thenReturn(2L);
-        Mockito.when(vmInstance2.getHostId()).thenReturn(3L);
+        VMInstanceVO vmInstance2 = createMockVmInstance(3L);
+        VMInstanceVO vmInstance1 = createMockVmInstance(2L);
 
         Mockito.doReturn(vmInstance1).when(vmInstanceDao).findById(100L);
         Mockito.doReturn(vmInstance2).when(vmInstanceDao).findById(101L);
@@ -1393,4 +1391,10 @@
         Mockito.verify(hostDao).findByClusterId(clusterId, Host.Type.Routing);
         Mockito.verify(hostDao).findByPodId(podId, Host.Type.Routing);
     }
+
+    private static VMInstanceVO createMockVmInstance(long hostId) {
+        VMInstanceVO vmInstance = Mockito.mock(VMInstanceVO.class);
+        Mockito.when(vmInstance.getHostId()).thenReturn(hostId);
+        return vmInstance;
+    }
 }
