--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -19,6 +19,7 @@
 
 package com.cloud.network.guru;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.anyInt;
@@ -81,6 +82,8 @@
 import com.cloud.vm.VirtualMachineProfile;
 
 public class BrocadeVcsGuestNetworkGuruTest {
+    @Mock
+    private final Network network;
     private static final long NETWORK_ID = 42L;
     PhysicalNetworkDao physnetdao = mock(PhysicalNetworkDao.class);
     BrocadeVcsDao vcsdao = mock(BrocadeVcsDao.class);
@@ -167,7 +170,6 @@
         when(netmodel.listNetworkOfferingServices(NETWORK_ID)).thenReturn(Arrays.asList(Service.Connectivity));
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
         final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
@@ -188,7 +190,6 @@
         when(offering.getGuestType()).thenReturn(GuestType.Isolated);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
         final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
@@ -210,7 +211,6 @@
         when(nosd.areServicesSupportedByNetworkOffering(NETWORK_ID, Service.Connectivity)).thenReturn(false);
 
         final DeploymentPlan plan = mock(DeploymentPlan.class);
-        final Network network = mock(Network.class);
         final Account account = mock(Account.class);
 
         final Network designednetwork = guru.design(offering, plan, network, "", 1L, account);
