--- a/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
@@ -51,6 +51,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class HostDaoImplTest {
+    @Mock
+    private SearchCriteria<String> sc;
 
     @Spy
     HostDaoImpl hostDao = new HostDaoImpl();
@@ -147,7 +149,6 @@
         HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, String> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
-        SearchCriteria<String> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         doReturn(sb).when(hostDao).createSearchBuilder(String.class);
         doReturn(mockResults.stream().map(h -> h.name()).collect(Collectors.toList())).when(hostDao)
@@ -239,7 +240,6 @@
         HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, String> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
-        SearchCriteria<String> sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         doReturn(sb).when(hostDao).createSearchBuilder(String.class);
         doReturn(mockResults.stream().map(h -> h.getType()).collect(Collectors.toList())).when(hostDao)
