--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -1420,9 +1420,8 @@
 
     @Test
     public void getVmOverrideDiskOfferingId() {
-        DiskOfferingVO diskOfferingMock = Mockito.mock(DiskOfferingVO.class);
+        DiskOfferingVO diskOfferingMock = createMockDiskOffering(overrideDiskOfferingId);
         when(diskOfferingDao.findByUuid(overrideDiskOfferingUuid)).thenReturn(diskOfferingMock);
-        when(diskOfferingMock.getId()).thenReturn(overrideDiskOfferingId);
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("overridediskofferingid", overrideDiskOfferingUuid);
@@ -1434,9 +1433,8 @@
 
     @Test
     public void getVmDiskOfferingId() {
-        DiskOfferingVO diskOfferingMock = Mockito.mock(DiskOfferingVO.class);
+        DiskOfferingVO diskOfferingMock = createMockDiskOffering(diskOfferingId);
         when(diskOfferingDao.findByUuid(diskOfferingUuid)).thenReturn(diskOfferingMock);
-        when(diskOfferingMock.getId()).thenReturn(diskOfferingId);
 
         Map<String, String> deployParams = new HashMap<>();
         deployParams.put("diskofferingid", diskOfferingUuid);
@@ -2581,4 +2579,10 @@
     public void testValidateMinMaxMembersInvalidRange() {
         autoScaleManagerImplSpy.validateMinMaxMembers(5, 1);
     }
+
+    private static DiskOfferingVO createMockDiskOffering(Long diskOfferingId) {
+        DiskOfferingVO diskOfferingMock = Mockito.mock(DiskOfferingVO.class);
+        when(diskOfferingMock.getId()).thenReturn(diskOfferingId);
+        return diskOfferingMock;
+    }
 }
