--- a/server/src/test/java/com/cloud/network/element/ConfigDriveNetworkElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/ConfigDriveNetworkElementTest.java
@@ -198,7 +198,7 @@
         when(deployDestination.getDataCenter()).thenReturn(dataCenter);
         when(hostVO.getId()).thenReturn(HOSTID);
         when(nic.getNetworkId()).thenReturn(NETWORK_ID);
-        when(network.getId()).thenReturn(NETWORK_ID);
+        network = MockNetworkVO.createMockNetworkVO(NETWORK_ID);
         when(_networkModel.getNetwork(NETWORK_ID)).thenReturn(network);
         //when(_networkModel.getUserDataUpdateProvider(network)).thenReturn(_configDrivesNetworkElement);
 
@@ -324,4 +324,15 @@
             assertTrue(createCommand.getIsoFile().equals(ConfigDrive.createConfigDrivePath(profile.getInstanceName())));
         }
     }
+
+    public final class MockNetworkVO {
+        private MockNetworkVO() {
+        }
+
+        public static NetworkVO createMockNetworkVO(long networkId) {
+            NetworkVO network = Mockito.mock(NetworkVO.class);
+            when(network.getId()).thenReturn(networkId);
+            return network;
+        }
+    }
 }
--- a/server/src/test/java/com/cloud/network/element/VirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VirtualRouterElementTest.java
@@ -557,13 +557,12 @@
     @Test
     public void testApplyBgpPeersForNetwork() throws ResourceUnavailableException {
         List<BgpPeerVO> bgpPeers = Mockito.mock(List.class);
-        NetworkVO network = Mockito.mock(NetworkVO.class);
 
         DomainRouterVO router = Mockito.mock(DomainRouterVO.class);
         when(router.getState()).thenReturn(VirtualMachine.State.Running);
         long zoneId = 10L;
         long networkId = 11L;
-        when(network.getId()).thenReturn(networkId);
+        NetworkVO network = MockNetworkVO.createMockNetworkVO(networkId);
         when(network.getDataCenterId()).thenReturn(zoneId);
         when(_routerDao.listByNetworkAndRole(networkId, VirtualRouter.Role.VIRTUAL_ROUTER)).thenReturn(Arrays.asList(router));
         DataCenterVO dc = Mockito.mock(DataCenterVO.class);
@@ -577,4 +576,15 @@
         Assert.assertTrue(result);
         verify(networkTopology).applyBgpPeers(any(), any(), any());
     }
+
+    public final class MockNetworkVO {
+        private MockNetworkVO() {
+        }
+
+        public static NetworkVO createMockNetworkVO(long networkId) {
+            NetworkVO network = Mockito.mock(NetworkVO.class);
+            when(network.getId()).thenReturn(networkId);
+            return network;
+        }
+    }
 }
