--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -260,10 +260,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         when(guestGuru.implement(network, offering, dest, res)).thenReturn(network);
 
         final CreateNetworkAnswer answer = mock(CreateNetworkAnswer.class);
@@ -319,10 +317,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         when(guestGuru.implement(network, offering, dest, res)).thenReturn(network);
 
         final CreateNetworkAnswer answer = mock(CreateNetworkAnswer.class);
@@ -373,10 +369,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         final AssociateMacToNetworkAnswer answer = mock(AssociateMacToNetworkAnswer.class);
         when(answer.getResult()).thenReturn(true);
 
@@ -425,10 +419,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         final AssociateMacToNetworkAnswer answer = mock(AssociateMacToNetworkAnswer.class);
         when(answer.getResult()).thenReturn(true);
 
@@ -588,4 +580,17 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    public final class MockReservationContext {
+
+        private MockReservationContext() {
+        }
+
+        public static ReservationContext createMockReservationContext(Account account, Domain domain) {
+            final ReservationContext res = mock(ReservationContext.class);
+            when(res.getAccount()).thenReturn(account);
+            when(res.getDomain()).thenReturn(domain);
+            return res;
+        }
+    }
 }
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuruTest.java
@@ -283,10 +283,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         final CreateLogicalSwitchAnswer answer = mock(CreateLogicalSwitchAnswer.class);
         when(answer.getResult()).thenReturn(true);
         when(answer.getLogicalSwitchUuid()).thenReturn("aaaaa");
@@ -339,10 +337,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = createMockReservationContext(acc, dom);
         final CreateLogicalSwitchAnswer answer = mock(CreateLogicalSwitchAnswer.class);
         when(answer.getResult()).thenReturn(true);
         when(answer.getLogicalSwitchUuid()).thenReturn("aaaaa");
@@ -395,10 +391,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         final CreateLogicalSwitchAnswer answer = mock(CreateLogicalSwitchAnswer.class);
         when(answer.getResult()).thenReturn(true);
         //when(answer.getLogicalSwitchUuid()).thenReturn("aaaaa");
@@ -452,10 +446,8 @@
         when(dom.getName()).thenReturn("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
-        final ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        final ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         final DeleteLogicalSwitchAnswer answer = mock(DeleteLogicalSwitchAnswer.class);
         when(answer.getResult()).thenReturn(true);
         when(agentmgr.easySend(eq(NETWORK_ID), (Command)any())).thenReturn(answer);
@@ -469,4 +461,17 @@
         verify(agentmgr, times(1)).easySend(eq(NETWORK_ID), (Command)any());
         verify(implementednetwork, times(1)).setBroadcastUri(null);
     }
+
+    public final class MockReservationContext {
+
+        private MockReservationContext() {
+        }
+
+        public static ReservationContext createMockReservationContext(Account account, Domain domain) {
+            final ReservationContext res = mock(ReservationContext.class);
+            when(res.getAccount()).thenReturn(account);
+            when(res.getDomain()).thenReturn(domain);
+            return res;
+        }
+    }
 }
--- a/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/VxlanGuestNetworkGuruTest.java
+++ b/plugins/network-elements/vxlan/src/test/java/com/cloud/network/guru/VxlanGuestNetworkGuruTest.java
@@ -177,10 +177,8 @@
         Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
 
-        ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         Network implementednetwork = guru.implement(network, offering, dest, res);
         assertTrue(implementednetwork != null);
     }
@@ -222,10 +220,8 @@
         Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
 
-        ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         Network implementednetwork = guru.implement(network, offering, dest, res);
         assertTrue(implementednetwork != null);
         assertTrue(implementednetwork.getCidr().equals("10.1.1.0/24"));
@@ -265,10 +261,8 @@
         Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
 
-        ReservationContext res = mock(ReservationContext.class);
-        when(res.getDomain()).thenReturn(dom);
-        when(res.getAccount()).thenReturn(acc);
-
+
+        ReservationContext res = MockReservationContext.createMockReservationContext(acc, dom);
         NetworkProfile implementednetwork = mock(NetworkProfile.class);
         when(implementednetwork.getId()).thenReturn(42L);
         when(implementednetwork.getBroadcastUri()).thenReturn(new URI("vxlan:12345"));
@@ -277,4 +271,17 @@
         guru.shutdown(implementednetwork, offering);
         verify(implementednetwork, times(1)).setBroadcastUri(null);
     }
+
+    public final class MockReservationContext {
+
+        private MockReservationContext() {
+        }
+
+        public static ReservationContext createMockReservationContext(Account account, Domain domain) {
+            final ReservationContext res = mock(ReservationContext.class);
+            when(res.getAccount()).thenReturn(account);
+            when(res.getDomain()).thenReturn(domain);
+            return res;
+        }
+    }
 }
