--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -661,8 +661,7 @@
         NetworkVO network1 = Mockito.mock(NetworkVO.class);
         Mockito.when(network1.getDns1()).thenReturn(ip4Dns[0]);
         Long offeringId = 1L;
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getId()).thenReturn(offeringId);
+        NetworkOffering offering = createNetworkOffering(offeringId);
         Mockito.when(networkOfferingServiceMapDao.areServicesSupportedByNetworkOffering(offeringId, Network.Service.Dns)).thenReturn(true);
         Assert.assertFalse(service.checkAndUpdateNetworkDns(network1, offering, null, null, null, null));
         Assert.assertFalse(service.checkAndUpdateNetworkDns(network1, Mockito.mock(NetworkOffering.class), ip4Dns[0], null, null, null));
@@ -675,8 +674,7 @@
         NetworkVO networkVO = new NetworkVO();
         networkVO.setDns1(ip4Dns[0]);
         Long offeringId = 1L;
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getId()).thenReturn(offeringId);
+        NetworkOffering offering = createNetworkOffering(offeringId);
         Mockito.when(networkOfferingServiceMapDao.areServicesSupportedByNetworkOffering(offeringId, Network.Service.Dns)).thenReturn(false);
         Assert.assertTrue(service.checkAndUpdateNetworkDns(networkVO, offering, null, null, null, null));
         Assert.assertNull(networkVO.getDns1());
@@ -721,8 +719,7 @@
         Mockito.when(networkVO.getIp6Cidr()).thenReturn(null);
         Mockito.when(networkVO.getGuestType()).thenReturn(Network.GuestType.Shared);
         Long offeringId = 1L;
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getId()).thenReturn(offeringId);
+        NetworkOffering offering = createNetworkOffering(offeringId);
         Mockito.when(networkOfferingServiceMapDao.areServicesSupportedByNetworkOffering(offeringId, Network.Service.Dns)).thenReturn(true);
         service.checkAndUpdateNetworkDns(networkVO, offering, null, null, ip6Dns[0], null);
     }
@@ -733,8 +730,7 @@
         Mockito.when(networkVO.getVpcId()).thenReturn(null);
         Mockito.when(networkVO.getGuestType()).thenReturn(Network.GuestType.Isolated);
         Long offeringId = 1L;
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getId()).thenReturn(offeringId);
+        NetworkOffering offering = createNetworkOffering(offeringId);
         Mockito.when(networkOfferingDao.isIpv6Supported(offeringId)).thenReturn(false);
         Mockito.when(networkOfferingServiceMapDao.areServicesSupportedByNetworkOffering(offeringId, Network.Service.Dns)).thenReturn(true);
         service.checkAndUpdateNetworkDns(networkVO, offering, null, null, ip6Dns[0], null);
@@ -753,8 +749,7 @@
         }
         networkVO.setIp6Cidr("cidr");
         Long offeringId = 1L;
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getId()).thenReturn(offeringId);
+        NetworkOffering offering = createNetworkOffering(offeringId);
         Mockito.when(networkOfferingServiceMapDao.areServicesSupportedByNetworkOffering(offeringId, Network.Service.Dns)).thenReturn(true);
         boolean updated = service.checkAndUpdateNetworkDns(networkVO, offering, ip4Dns[0], null, ip6Dns[0], null);
         Assert.assertTrue(updated);
@@ -1378,4 +1373,10 @@
 
         Assert.assertFalse(service.getAndValidateSupportForKeepMacAddressOnPublicNicParameter(false, networkOfferingVO));
     }
+
+    private static NetworkOffering createNetworkOffering(Long offeringId) {
+        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
+        Mockito.when(offering.getId()).thenReturn(offeringId);
+        return offering;
+    }
 }
