--- a/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/api/command/AddNetrisProviderCmdTest.java
+++ b/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/api/command/AddNetrisProviderCmdTest.java
@@ -35,6 +35,8 @@
 import java.lang.reflect.Field;
 
 public class AddNetrisProviderCmdTest {
+    @Mock
+    private NetrisProvider provider;
 
     @Mock
     private NetrisProviderService netrisProviderService;
@@ -92,7 +94,6 @@
 
     @Test
     public void testExecuteSuccess() throws ConcurrentOperationException {
-        NetrisProvider provider = Mockito.mock(NetrisProvider.class);
         NetrisProviderResponse response = Mockito.mock(NetrisProviderResponse.class);
         Mockito.when(netrisProviderService.addProvider(cmd)).thenReturn(provider);
         Mockito.when(netrisProviderService.createNetrisProviderResponse(provider)).thenReturn(response);
@@ -105,7 +106,6 @@
 
     @Test(expected = ServerApiException.class)
     public void testExecuteFailure() throws ConcurrentOperationException {
-        NetrisProvider provider = Mockito.mock(NetrisProvider.class);
         Mockito.when(netrisProviderService.addProvider(cmd)).thenReturn(provider);
         Mockito.when(netrisProviderService.createNetrisProviderResponse(provider)).thenReturn(null);
         cmd.execute();
