--- a/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/host/dao/HostDaoImplTest.java
@@ -51,6 +51,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class HostDaoImplTest {
+    @Mock
+    private HostVO host;
 
     @Spy
     HostDaoImpl hostDao = new HostDaoImpl();
@@ -82,7 +84,6 @@
         GenericDaoBase.SumCount mockSumCount = new GenericDaoBase.SumCount();
         mockSumCount.count = 10;
         mockSumCount.sum = 20;
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, GenericDaoBase.SumCount> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
         doReturn(sb).when(hostDao).createSearchBuilder(GenericDaoBase.SumCount.class);
@@ -99,7 +100,6 @@
     public void testIsHostUp() {
         long testHostId = 101L;
         List<Status> statuses = List.of(Status.Up);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Status> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
         SearchCriteria<Status> sc = mock(SearchCriteria.class);
@@ -121,7 +121,6 @@
         List<Host.Type> types = List.of(Host.Type.Routing);
         List<Hypervisor.HypervisorType> hypervisorTypes = List.of(Hypervisor.HypervisorType.KVM);
         List<Long> mockResults = List.of(1001L, 1002L); // Mocked result
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
         SearchCriteria<Long> sc = mock(SearchCriteria.class);
@@ -144,7 +143,6 @@
     public void testListDistinctHypervisorTypes() {
         Long zoneId = 1L;
         List<Hypervisor.HypervisorType> mockResults = List.of(Hypervisor.HypervisorType.KVM, Hypervisor.HypervisorType.XenServer);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, String> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
         SearchCriteria<String> sc = mock(SearchCriteria.class);
@@ -179,7 +177,6 @@
         Hypervisor.HypervisorType hypervisorType = Hypervisor.HypervisorType.KVM;
         Long zoneId = 1L, podId = 2L, clusterId = 3L;
         List<Long> mockResults = List.of(1001L, 1002L);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, Long> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
         SearchCriteria<Long> sc = mock(SearchCriteria.class);
@@ -236,7 +233,6 @@
     public void testListDistinctArchTypes() {
         Long clusterId = 1L;
         List<CPU.CPUArch> mockResults = List.of(CPU.CPUArch.amd64, CPU.CPUArch.arm64);
-        HostVO host = mock(HostVO.class);
         GenericSearchBuilder<HostVO, String> sb = mock(GenericSearchBuilder.class);
         when(sb.entity()).thenReturn(host);
         SearchCriteria<String> sc = mock(SearchCriteria.class);
