--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -1037,11 +1037,9 @@
         List<HostVO> clusterHosts = Arrays.asList(host1, host2);
         Mockito.doReturn(clusterHosts).when(resourceManager).listAllUpHosts(Host.Type.Routing, 1L, 1L, 1L);
 
-        VolumeVO volume1 = Mockito.mock(VolumeVO.class);
-        VolumeVO volume2 = Mockito.mock(VolumeVO.class);
-
-        Mockito.when(volume1.getInstanceId()).thenReturn(100L);
-        Mockito.when(volume2.getInstanceId()).thenReturn(101L);
+        VolumeVO volume2 = createMockVolumeVO(101L);
+
+        VolumeVO volume1 = createMockVolumeVO(100L);
 
         List<VolumeVO> volumesInPool = Arrays.asList(volume1, volume2);
         Mockito.doReturn(volumesInPool).when(volumeDao).findNonDestroyedVolumesByPoolId(poolId);
@@ -1393,4 +1391,10 @@
         Mockito.verify(hostDao).findByClusterId(clusterId, Host.Type.Routing);
         Mockito.verify(hostDao).findByPodId(podId, Host.Type.Routing);
     }
+
+    private static VolumeVO createMockVolumeVO(long instanceId) {
+        VolumeVO volume = Mockito.mock(VolumeVO.class);
+        Mockito.when(volume.getInstanceId()).thenReturn(instanceId);
+        return volume;
+    }
 }
