--- a/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
+++ b/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
@@ -1210,10 +1210,9 @@
         Long domainId = 2L;
         Double value = 10D;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
         BigDecimal currentBalance = BigDecimal.ZERO;
 
-        Mockito.doReturn(accountId).when(account).getId();
+        AccountVO account = createMockAccount(accountId);
         Mockito.doReturn(domainId).when(account).getDomainId();
         Mockito.doReturn(null).when(cmd).getQuotaEnforce();
         Mockito.doReturn(null).when(cmd).getMinBalance();
@@ -1238,9 +1237,8 @@
         Long accountId = 1L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
-
-        Mockito.doReturn(accountId).when(account).getId();
+
+        AccountVO account = createMockAccount(accountId);
         Mockito.when(cmd.getQuotaEnforce()).thenReturn(Boolean.TRUE);
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
 
@@ -1255,10 +1253,9 @@
         Long accountId = 1L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
-
+
+        AccountVO account = createMockAccount(accountId);
         Mockito.when(cmd.getMinBalance()).thenReturn(50.0);
-        Mockito.doReturn(accountId).when(account).getId();
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
 
         quotaResponseBuilderSpy.persistQuotaCredits(cmd, value, depositedOn, account, false);
@@ -1273,10 +1270,9 @@
         Long domainId = 2L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
         BigDecimal currentBalance = BigDecimal.ZERO;
 
-        Mockito.doReturn(accountId).when(account).getId();
+        AccountVO account = createMockAccount(accountId);
         Mockito.doReturn(domainId).when(account).getDomainId();
         Mockito.when(cmd.getMinBalance()).thenReturn(50.0);
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
@@ -1368,4 +1364,10 @@
         Assert.assertEquals(record2.getStartDate(), secondHistory.getStartDate());
         Assert.assertEquals(record2.getEndDate(), secondHistory.getEndDate());
     }
+
+    private static AccountVO createMockAccount(Long accountId) {
+        AccountVO account = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(accountId).when(account).getId();
+        return account;
+    }
 }
