--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -146,6 +146,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class CitrixRequestWrapperTest {
+    @Mock
+    private final VirtualMachineTO machineTO;
 
     @Mock
     private CitrixResourceBase citrixResourceBase;
@@ -327,7 +329,6 @@
 
     @Test
     public void testPrepareForMigrationCommand() {
-        final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
         final PrepareForMigrationCommand prepareCommand = new PrepareForMigrationCommand(machineTO);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
@@ -341,7 +342,6 @@
 
     @Test
     public void testMigrateCommand() {
-        final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
         final MigrateCommand migrateCommand = new MigrateCommand("Test", "127.0.0.1", false, machineTO, false);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
@@ -635,11 +635,10 @@
 
     @Test
     public void testStartCommand() {
-        final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
         final com.cloud.host.Host host = Mockito.mock(com.cloud.host.Host.class);
 
-        final StartCommand startCommand = new StartCommand(vm, host, false);
-
+
+        final StartCommand startCommand = new StartCommand(machineTO, host, false);
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
@@ -1614,7 +1613,6 @@
     public void testScaleVmCommand() {
         final String uuid = "6172d8b7-ba10-4a70-93f9-ecaf41f51d53";
 
-        final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
         final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
         final Host host = Mockito.mock(Host.class);
