--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -98,6 +98,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class Ipv6ServiceImplTest {
+    @Mock
+    private Vpc vpc;
 
     @Mock
     NetworkOfferingDao networkOfferingDao;
@@ -574,7 +576,6 @@
     @Test
     public void testEmptyUpdateIpv6RoutesForVpcResponse() {
         VpcResponse response = new VpcResponse();
-        Vpc vpc = Mockito.mock(Vpc.class);
         List<NetworkVO> networks = new ArrayList<>();
         Mockito.doReturn(networks).when(networkModel).listNetworksByVpc(Mockito.anyLong());
         ipv6Service.updateIpv6RoutesForVpcResponse(vpc, response);
@@ -584,7 +585,6 @@
     @Test
     public void testUpdateIpv6RoutesForVpcResponse() {
         VpcResponse response = new VpcResponse();
-        Vpc vpc = Mockito.mock(Vpc.class);
         List<NetworkVO> networks = new ArrayList<>();
         NetworkVO network = Mockito.mock(NetworkVO.class);
         Mockito.when(network.getIp6Cidr()).thenReturn(cidr);
