--- a/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
@@ -378,8 +378,7 @@
     public void testGetFilteredVmIdsForSnapshots() {
         List<Long> skippedVmIds = ids.subList(0, 2);
         Long notSkippedVmIds = ids.get(2);
-        VMSnapshotVO vmSnapshotVO = Mockito.mock(VMSnapshotVO.class);
-        Mockito.when(vmSnapshotVO.getVmId()).thenReturn(1L);
+        VMSnapshotVO vmSnapshotVO = createMockVMSnapshotVO();
         Mockito.when(vmSnapshotDao.searchByVms(Mockito.anyList())).thenReturn(List.of(vmSnapshotVO));
         HashSet<Long> set = new HashSet<>();
         set.add(1L);
@@ -406,8 +405,7 @@
                         Mockito.anyLong(), Mockito.anyList())).thenReturn(List.of(vm1, vm2, vm3));
         List<Long> skippedVmIds = ids.subList(0, 2);
         Long notSkippedVmIds = ids.get(2);
-        VMSnapshotVO vmSnapshotVO = Mockito.mock(VMSnapshotVO.class);
-        Mockito.when(vmSnapshotVO.getVmId()).thenReturn(1L);
+        VMSnapshotVO vmSnapshotVO = createMockVMSnapshotVO();
         Mockito.when(vmSnapshotDao.searchByVms(Mockito.anyList())).thenReturn(List.of(vmSnapshotVO));
         HashSet<Long> set = new HashSet<>();
         set.add(1L);
@@ -685,4 +683,10 @@
         resourceCleanupService.parseDateFromConfig(
                 ResourceCleanupService.ExpungedResourcesPurgeStartTime.key(), "ABC");
     }
+
+    private static VMSnapshotVO createMockVMSnapshotVO() {
+        VMSnapshotVO vmSnapshotVO = Mockito.mock(VMSnapshotVO.class);
+        Mockito.when(vmSnapshotVO.getVmId()).thenReturn(1L);
+        return vmSnapshotVO;
+    }
 }
