--- a/server/src/test/java/com/cloud/api/query/dao/SnapshotJoinDaoImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/dao/SnapshotJoinDaoImplTest.java
@@ -86,10 +86,8 @@
     public void testFindById_WithValidResults() {
         Long zoneId = 1L;
         long id = 1L;
-        SnapshotJoinVO snapshot1 = mock(SnapshotJoinVO.class);
-        when(snapshot1.getSnapshotStorePair()).thenReturn("Primary_1");
-        SnapshotJoinVO snapshot2 = mock(SnapshotJoinVO.class);
-        when(snapshot2.getSnapshotStorePair()).thenReturn("Image_1");
+        SnapshotJoinVO snapshot1 = createMockSnapshotJoinVO("Primary_1");
+        SnapshotJoinVO snapshot2 = createMockSnapshotJoinVO("Image_1");
         doReturn(Arrays.asList(snapshot1, snapshot2)).when(snapshotJoinDao).search(mockSearchCriteria, null);
         List<SnapshotJoinVO> result = snapshotJoinDao.findById(zoneId, id);
         assertNotNull(result);
@@ -106,4 +104,10 @@
         List<SnapshotJoinVO> result = snapshotJoinDao.findById(null, id);
         assertNull(result);
     }
+
+    private static SnapshotJoinVO createMockSnapshotJoinVO(String snapshotStorePair) {
+        SnapshotJoinVO snapshotJoinVO = mock(SnapshotJoinVO.class);
+        when(snapshotJoinVO.getSnapshotStorePair()).thenReturn(snapshotStorePair);
+        return snapshotJoinVO;
+    }
 }
