--- a/api/src/test/java/org/apache/cloudstack/api/command/test/ActivateProjectCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/ActivateProjectCmdTest.java
@@ -72,8 +72,7 @@
         Project project = Mockito.mock(Project.class);
         Mockito.when(project.getId()).thenReturn(2L);
         ProjectService projectService = Mockito.mock(ProjectService.class);
-        Account account = Mockito.mock(Account.class);
-        Mockito.when(account.getId()).thenReturn(2L);
+        Account account = MockAccount.createMockAccount();
         Mockito.when(projectService.getProject(ArgumentMatchers.anyLong())).thenReturn(project);
 
         Mockito.when(projectService.getProjectOwner(ArgumentMatchers.anyLong())).thenReturn(account);
@@ -83,4 +82,15 @@
 
     }
 
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount() {
+            com.cloud.user.Account account = org.mockito.Mockito.mock(com.cloud.user.Account.class);
+            org.mockito.Mockito.when(account.getId()).thenReturn(2L);
+            return account;
+        }
+    }
 }
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/AddAccountToProjectCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/AddAccountToProjectCmdTest.java
@@ -99,9 +99,8 @@
         Mockito.when(project.getId()).thenReturn(2L);
 
         ProjectService projectService = Mockito.mock(ProjectService.class);
-        Account account = Mockito.mock(Account.class);
 
-        Mockito.when(account.getId()).thenReturn(2L);
+        Account account = MockAccount.createMockAccount();
         Mockito.when(projectService.getProject(ArgumentMatchers.anyLong())).thenReturn(project);
 
         Mockito.when(projectService.getProjectOwner(ArgumentMatchers.anyLong())).thenReturn(account);
@@ -175,4 +174,15 @@
      * }
      */
 
+
+    public final class MockAccount {
+        private MockAccount() {
+        }
+
+        public static com.cloud.user.Account createMockAccount() {
+            com.cloud.user.Account account = org.mockito.Mockito.mock(com.cloud.user.Account.class);
+            org.mockito.Mockito.when(account.getId()).thenReturn(2L);
+            return account;
+        }
+    }
 }
