--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -144,6 +144,8 @@
 import static org.mockito.Mockito.when;
 
 public class ConfigurationManagerTest {
+    @Mock
+    private VMInstanceVO vMInstanceVO;
 
     private Logger logger = LogManager.getLogger(ConfigurationManagerTest.class);
 
@@ -803,7 +805,6 @@
         Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
-        VMInstanceVO vMInstanceVO = Mockito.mock(VMInstanceVO.class);
         ArrayList<VMInstanceVO> arrayList = new ArrayList<VMInstanceVO>();
         arrayList.add(vMInstanceVO);
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(0);
@@ -909,7 +910,6 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfZoneIsDeletableFailureOnVmInstanceTest() {
-        VMInstanceVO vMInstanceVO = Mockito.mock(VMInstanceVO.class);
         ArrayList<VMInstanceVO> arrayList = new ArrayList<VMInstanceVO>();
         arrayList.add(vMInstanceVO);
 
