--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -150,6 +150,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class BackupManagerTest {
+    @Mock
+    private DomainVO domain;
     @Spy
     @InjectMocks
     BackupManagerImpl backupManager = new BackupManagerImpl();
@@ -2324,7 +2326,6 @@
         when(cmd.getId()).thenReturn(id);
         when(cmd.getDomainIds()).thenReturn(List.of(domainId));
 
-        DomainVO domain = Mockito.mock(DomainVO.class);
         when(domainDao.findById(domainId)).thenReturn(domain);
 
         when(domainHelper.filterChildSubDomains(List.of(domainId))).thenReturn(new ArrayList<>(List.of(domainId)));
@@ -2659,7 +2660,6 @@
         when(savedOffering.getId()).thenReturn(savedOfferingId);
         when(backupOfferingDao.persist(any(BackupOfferingVO.class))).thenReturn(savedOffering);
 
-        DomainVO domain = Mockito.mock(DomainVO.class);
         when(domainDao.findById(11L)).thenReturn(domain);
 
         overrideBackupFrameworkConfigValue();
@@ -2708,9 +2708,8 @@
         when(backupOfferingDao.persist(any(BackupOfferingVO.class))).thenReturn(savedOffering);
 
         // domain handling
-        DomainVO domain21 = Mockito.mock(DomainVO.class);
         DomainVO domain22 = Mockito.mock(DomainVO.class);
-        when(domainDao.findById(21L)).thenReturn(domain21);
+        when(domainDao.findById(21L)).thenReturn(domain);
         when(domainDao.findById(22L)).thenReturn(domain22);
         when(domainHelper.filterChildSubDomains(sourceDomainIds)).thenReturn(new ArrayList<>(sourceDomainIds));
 
