--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -240,9 +240,8 @@
         when(dataCenterDao.findById(s_zoneId)).thenReturn(null);
         Assert.assertThrows(InvalidParameterValueException.class, () -> sharedFSServiceImpl.allocSharedFS(cmd));
 
-        DataCenterVO zone = mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenter(Grouping.AllocationState.Enabled);
         when(dataCenterDao.findById(s_zoneId)).thenReturn(zone);
-        when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
 
         DiskOfferingVO diskOfferingVO = mock(DiskOfferingVO.class);
         when(diskOfferingDao.findById(s_diskOfferingId)).thenReturn(diskOfferingVO);
@@ -270,9 +269,8 @@
         when(dataCenterDao.findById(s_zoneId)).thenReturn(null);
         Assert.assertThrows(InvalidParameterValueException.class, () -> sharedFSServiceImpl.allocSharedFS(cmd));
 
-        DataCenterVO zone = mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenter(Grouping.AllocationState.Disabled);
         when(dataCenterDao.findById(s_zoneId)).thenReturn(zone);
-        when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Disabled);
         Assert.assertThrows(PermissionDeniedException.class, () -> sharedFSServiceImpl.allocSharedFS(cmd));
 
         when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
@@ -284,9 +282,8 @@
     public void tesAllocSharedFSInvalidDiskOffering() {
         CreateSharedFSCmd cmd = getMockCreateSharedFSCmd();
 
-        DataCenterVO zone = mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenter(Grouping.AllocationState.Enabled);
         when(dataCenterDao.findById(s_zoneId)).thenReturn(zone);
-        when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
 
         DiskOfferingVO diskOfferingVO = mock(DiskOfferingVO.class);
         when(diskOfferingDao.findById(s_diskOfferingId)).thenReturn(diskOfferingVO);
@@ -302,9 +299,8 @@
     public void testAllocSharedFSInvalidFsFormat() {
         CreateSharedFSCmd cmd = getMockCreateSharedFSCmd();
 
-        DataCenterVO zone = mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenter(Grouping.AllocationState.Enabled);
         when(dataCenterDao.findById(s_zoneId)).thenReturn(zone);
-        when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
 
         DiskOfferingVO diskOfferingVO = mock(DiskOfferingVO.class);
         when(diskOfferingDao.findById(s_diskOfferingId)).thenReturn(diskOfferingVO);
@@ -445,9 +441,8 @@
         ReflectionTestUtils.setField(sharedFS, "state", SharedFS.State.Ready);
         when(sharedFSDao.findById(s_sharedFSId)).thenReturn(sharedFS);
 
-        DataCenterVO zone = mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenter(Grouping.AllocationState.Enabled);
         when(dataCenterDao.findById(s_zoneId)).thenReturn(zone);
-        when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
 
         DiskOfferingVO diskOfferingVO = mock(DiskOfferingVO.class);
         when(diskOfferingDao.findById(newDiskOfferingId)).thenReturn(diskOfferingVO);
@@ -480,9 +475,8 @@
         ReflectionTestUtils.setField(sharedFS, "state", SharedFS.State.Stopped);
         when(sharedFSDao.findById(s_sharedFSId)).thenReturn(sharedFS);
 
-        DataCenterVO zone = mock(DataCenterVO.class);
+        DataCenterVO zone = createMockDataCenter(Grouping.AllocationState.Enabled);
         when(dataCenterDao.findById(s_zoneId)).thenReturn(zone);
-        when(zone.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
 
         when(lifeCycle.changeSharedFSServiceOffering(sharedFS, newServiceOfferingId)).thenReturn(true);
 
@@ -662,4 +656,10 @@
             verify(_stateMachine, times(1)).transitTo(sharedFS, SharedFS.Event.OperationFailed, null, sharedFSDao);
         }
     }
+
+    private static DataCenterVO createMockDataCenter(Grouping.AllocationState allocationState) {
+        DataCenterVO zone = mock(DataCenterVO.class);
+        when(zone.getAllocationState()).thenReturn(allocationState);
+        return zone;
+    }
 }
