--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessorTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessorTest.java
@@ -18,6 +18,7 @@
  */
 package com.cloud.hypervisor.xenserver.resource;
 
+import org.mockito.Mock;
 import static org.mockito.Mockito.when;
 
 import java.util.List;
@@ -29,9 +30,10 @@
 import com.cloud.agent.api.to.SwiftTO;
 
 public class XenServerStorageProcessorTest {
+    @Mock
+    private CitrixResourceBase resource;
     @Test
     public void testOrderOfSwiftUplodScriptParamsWithoutStoragePolicy() {
-        CitrixResourceBase resource = Mockito.mock(CitrixResourceBase.class);
         XenServerStorageProcessor mock = new XenServerStorageProcessor(resource);
 
         SwiftTO swift = Mockito.mock(SwiftTO.class);
@@ -71,7 +73,6 @@
 
     @Test
     public void testOrderOfSwiftUplodScriptParamsWithStoragePolicy() {
-        CitrixResourceBase resource = Mockito.mock(CitrixResourceBase.class);
         XenServerStorageProcessor mock = new XenServerStorageProcessor(resource);
 
         SwiftTO swift = Mockito.mock(SwiftTO.class);
