--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer620SP1WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer620SP1WrapperTest.java
@@ -43,6 +43,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class XenServer620SP1WrapperTest {
+    @Mock
+    private final Connection conn;
 
     @Mock
     private XenServer620SP1Resource xenServer620SP1Resource;
@@ -51,7 +53,6 @@
     public void testGetGPUStatsCommand() {
         final String guuid = "246a5b75-05ed-4bbc-a171-2d1fe94a1b0e";
 
-        final Connection conn = Mockito.mock(Connection.class);
 
         final GetGPUStatsCommand gpuStats = new GetGPUStatsCommand(guuid, "xen");
 
@@ -84,7 +85,6 @@
     public void testGetGPUStatsCommandFailure() {
         final String guuid = "246a5b75-05ed-4bbc-a171-2d1fe94a1b0e";
 
-        final Connection conn = Mockito.mock(Connection.class);
 
         final GetGPUStatsCommand gpuStats = new GetGPUStatsCommand(guuid, "xen");
 
