--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XcpServerWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XcpServerWrapperTest.java
@@ -43,6 +43,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class XcpServerWrapperTest {
+    @Mock
+    private final Connection conn;
 
     @Mock
     protected XcpServerResource XcpServerResource;
@@ -54,7 +56,6 @@
 
     @Test
     public void testNetworkUsageCommandCreate() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final String privateIP = "192.168.0.10";
         final String domRName = "dom";
@@ -78,7 +79,6 @@
 
     @Test
     public void testNetworkUsageCommandGet() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final String privateIP = "192.168.0.10";
         final String domRName = "dom";
@@ -102,7 +102,6 @@
 
     @Test
     public void testNetworkUsageCommandExceptiopn() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final String privateIP = "192.168.0.10";
         final String domRName = "dom";
@@ -170,7 +169,6 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        final Connection conn = Mockito.mock(Connection.class);
         when(XcpServerResource.getConnection()).thenReturn(conn);
         when(XcpServerResource.getNetworkStats(conn, getAutoScaleMetricsCommand.getPrivateIP(), getAutoScaleMetricsCommand.getPublicIP())).thenReturn(networkStats);
         when(XcpServerResource.getNetworkLbStats(getAutoScaleMetricsCommand.getPrivateIP(),  getAutoScaleMetricsCommand.getPublicIP(), getAutoScaleMetricsCommand.getPort())).thenReturn(lbStats);
@@ -202,7 +200,6 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        final Connection conn = Mockito.mock(Connection.class);
         when(XcpServerResource.getConnection()).thenReturn(conn);
         when(XcpServerResource.getNetworkStats(conn, getAutoScaleMetricsCommand.getPrivateIP(), getAutoScaleMetricsCommand.getPublicIP())).thenThrow(NumberFormatException.class);
 
