--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCheckVirtualMachineCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCheckVirtualMachineCommandWrapperTest.java
@@ -171,8 +171,9 @@
         when(libvirtException.getMessage()).thenReturn("Domain not found");
         when(libvirtComputingResource.getVmState(conn, VM_NAME)).thenReturn(PowerState.PowerOn);
         when(libvirtComputingResource.getVncPort(conn, VM_NAME)).thenReturn(5900);
-        when(conn.domainLookupByName(VM_NAME)).thenThrow(libvirtException);
-
+
+        conn = MockConnect.createMockConnect(VM_NAME, libvirtException);
+        when(libvirtUtilitiesHelper.getConnectionByVmName(VM_NAME)).thenReturn(conn);
         CheckVirtualMachineAnswer answer = (CheckVirtualMachineAnswer) wrapper.execute(command, libvirtComputingResource);
 
         assertFalse(answer.getResult());
@@ -188,4 +189,16 @@
         verify(libvirtComputingResource).getLibvirtUtilitiesHelper();
         verify(libvirtUtilitiesHelper).getConnectionByVmName(VM_NAME);
     }
+
+    public final class MockConnect {
+        private MockConnect() {
+        }
+
+        public static Connect createMockConnect(String vmName, LibvirtException libvirtException)
+                throws LibvirtException {
+            Connect connect = Mockito.mock(Connect.class);
+            when(connect.domainLookupByName(vmName)).thenThrow(libvirtException);
+            return connect;
+        }
+    }
 }
--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtPreMigrationCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtPreMigrationCommandWrapperTest.java
@@ -172,8 +172,8 @@
     public void testExecute_LibvirtException_ReturnsFailure() throws LibvirtException {
         LibvirtException libvirtException = Mockito.mock(LibvirtException.class);
         when(libvirtException.getMessage()).thenReturn("Connection failed");
-        when(libvirtUtilitiesHelper.getConnectionByVmName(VM_NAME)).thenReturn(connect);
-        when(connect.domainLookupByName(VM_NAME)).thenThrow(libvirtException);
+        connect = MockConnect.createMockConnect(VM_NAME, libvirtException);
+        when(libvirtUtilitiesHelper.getConnectionByVmName(VM_NAME)).thenReturn(connect);
 
         Answer answer = wrapper.execute(preMigrationCommand, libvirtComputingResource);
 
@@ -279,4 +279,16 @@
 
         return disks;
     }
+
+    public final class MockConnect {
+        private MockConnect() {
+        }
+
+        public static Connect createMockConnect(String vmName, LibvirtException libvirtException)
+                throws LibvirtException {
+            Connect connect = Mockito.mock(Connect.class);
+            when(connect.domainLookupByName(vmName)).thenThrow(libvirtException);
+            return connect;
+        }
+    }
 }
