--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -54,6 +54,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class LibvirtMigrateVolumeCommandWrapperTest {
+    @Mock
+    private DomainBlockJobInfo blockJobInfo;
 
     @Spy
     @InjectMocks
@@ -357,7 +359,6 @@
         String srcPath = "bec108c400000018:vol-60-7acb-9e22";
         String destPath = "01b381820000001c:vol-60-ec76-b7dc";
         Mockito.doReturn(60).when(command).getWait();
-        DomainBlockJobInfo blockJobInfo = Mockito.mock(DomainBlockJobInfo.class);
         Mockito.doReturn(blockJobInfo).when(dm).getBlockJobInfo(destDiskLabel, 0);
         blockJobInfo.cur = 100;
         blockJobInfo.end = 100;
@@ -375,7 +376,6 @@
         String srcPath = "bec108c400000018:vol-60-7acb-9e22";
         String destPath = "01b381820000001c:vol-60-ec76-b7dc";
         Mockito.doReturn(1).when(command).getWait();
-        DomainBlockJobInfo blockJobInfo = Mockito.mock(DomainBlockJobInfo.class);
         Mockito.doReturn(blockJobInfo).when(dm).getBlockJobInfo(destDiskLabel, 0);
         blockJobInfo.cur = 10;
         blockJobInfo.end = 100;
