--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -1290,7 +1290,7 @@
         ActionListType actionListType = mock(ActionListType.class);
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
-        FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
+        FirewallRuleEndpointType firewallRuleEndpointType2 = createMockFirewallRuleEndpointType();
         FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
@@ -1304,8 +1304,6 @@
         when(firewallRule.getEndpoint1()).thenReturn(firewallRuleEndpointType1);
         when(firewallRule.getEndpoint2()).thenReturn(firewallRuleEndpointType2);
         when(firewallRuleEndpointType1.getTags()).thenReturn(List.of("tag"));
-        when(firewallRuleEndpointType2.getTags()).thenReturn(null);
-        when(firewallRuleEndpointType2.getAddressGroup()).thenReturn("address:group");
         when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsType);
 
         assertNotNull(tungstenService.createTungstenFirewallRule(1L, "f5ba12c8-d4c5-4c20-a57d-67a9b6fca652", "test", "pass", "948f421c-edde-4518-a391-09299cc25dc2",
@@ -1390,7 +1388,7 @@
         ActionListType actionListType = mock(ActionListType.class);
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
-        FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
+        FirewallRuleEndpointType firewallRuleEndpointType2 = createMockFirewallRuleEndpointType();
         FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
@@ -1404,8 +1402,6 @@
         when(firewallRule.getEndpoint1()).thenReturn(firewallRuleEndpointType1);
         when(firewallRule.getEndpoint2()).thenReturn(firewallRuleEndpointType2);
         when(firewallRuleEndpointType1.getTags()).thenReturn(List.of("tag"));
-        when(firewallRuleEndpointType2.getTags()).thenReturn(null);
-        when(firewallRuleEndpointType2.getAddressGroup()).thenReturn("address:group");
         when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsType);
 
         assertNotNull(tungstenService.listTungstenFirewallRule(1L, "948f421c-edde-4518-a391-09299cc25dc2",
@@ -1668,4 +1664,11 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static FirewallRuleEndpointType createMockFirewallRuleEndpointType() {
+        FirewallRuleEndpointType firewallRuleEndpointType = mock(FirewallRuleEndpointType.class);
+        when(firewallRuleEndpointType.getAddressGroup()).thenReturn("address:group");
+        when(firewallRuleEndpointType.getTags()).thenReturn(null);
+        return firewallRuleEndpointType;
+    }
 }
