--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -181,6 +181,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenResourceTest {
     @Mock
+    private FirewallPolicy firewallPolicy;
+    @Mock
     TungstenApi tungstenApi;
 
     TungstenResource tungstenResource;
@@ -1196,7 +1198,7 @@
     @Test
     public void executeRequestCreateTungstenFirewallPolicyCommandTest() {
         TungstenCommand command = new CreateTungstenFirewallPolicyCommand("firewallPolicyName", "b30ff54a-10a9-11ec-82a8-0242ac130003", 1);
-        FirewallPolicy firewallPolicy = mock(FirewallPolicy.class);
+        // removed local mock; replaced with global field `firewallPolicy`
         when(tungstenApi.createTungstenFirewallPolicy(anyString(), anyString(), anyString(), anyInt())).thenReturn(firewallPolicy);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1254,9 +1256,8 @@
     public void executeRequestListTungstenFirewallPolicyCommandTest() {
         TungstenCommand command = new ListTungstenFirewallPolicyCommand("d64e6082-10b3-11ec-82a8-0242ac130003",
                 "db1f89d8-10b3-11ec-82a8-0242ac130003");
-        FirewallPolicy firewallPolicy1 = mock(FirewallPolicy.class);
         FirewallPolicy firewallPolicy2 = mock(FirewallPolicy.class);
-        Answer<List<ApiObjectBase>> firewallPoliciesAnswer = setupApiObjectBaseListAnswer(firewallPolicy1, firewallPolicy2);
+        Answer<List<ApiObjectBase>> firewallPoliciesAnswer = setupApiObjectBaseListAnswer(firewallPolicy, firewallPolicy2);
         when(tungstenApi.listTungstenFirewallPolicy(anyString(), anyString())).thenAnswer(firewallPoliciesAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1313,7 +1314,6 @@
     @Test
     public void executeRequestDeleteTungstenFirewallPolicyCommandTest() {
         TungstenCommand command = new DeleteTungstenFirewallPolicyCommand("0b27c38c-10b6-11ec-82a8-0242ac130003");
-        FirewallPolicy firewallPolicy = mock(FirewallPolicy.class);
         when(tungstenApi.getTungstenObject(eq(FirewallPolicy.class), anyString())).thenReturn(firewallPolicy);
         when(tungstenApi.deleteTungstenObject(any(FirewallPolicy.class))).thenReturn(true);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
