--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -220,6 +220,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenServiceImplTest {
     @Mock
+    private FirewallRuleMatchTagsType firewallRuleMatchTagsType;
+    @Mock
     MessageBus messageBus;
     @Mock
     ProjectDao projectDao;
@@ -1291,7 +1293,6 @@
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
-        FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
@@ -1391,7 +1392,6 @@
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
-        FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
