--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -181,6 +181,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenResourceTest {
     @Mock
+    private FloatingIp floatingIp;
+    @Mock
     TungstenApi tungstenApi;
 
     TungstenResource tungstenResource;
@@ -406,8 +408,8 @@
     public void executeRequestCreateTungstenFloatingIpCommandTest() {
         TungstenCommand command = new CreateTungstenFloatingIpCommand("projectFqn",
             "ac617be6-bf80-4086-9d6a-c05ff78e2264", "fip", "fi", "192.168.1.100");
-        FloatingIp floatingIp = mock(FloatingIp.class);
-
+
+        // removed local mock; replaced with global field `floatingIp`
         when(tungstenApi.createTungstenFloatingIp(anyString(), anyString(), anyString(), anyString(),
             anyString())).thenReturn(floatingIp);
 
@@ -470,7 +472,6 @@
             "fi");
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         FloatingIpPool floatingIpPool = mock(FloatingIpPool.class);
-        FloatingIp floatingIp = mock(FloatingIp.class);
 
         when(tungstenApi.getTungstenObject(eq(VirtualNetwork.class), anyString())).thenReturn(virtualNetwork);
         when(tungstenApi.getTungstenObjectByName(eq(FloatingIpPool.class), any(), anyString())).thenReturn(
@@ -532,9 +533,8 @@
     @Test
     public void executeRequestGetTungstenFloatingIpsCommandTest() {
         TungstenCommand command = new GetTungstenFloatingIpsCommand("b604c7f7-1dbc-42d8-bceb-2c0898034a7a", "fip");
-        FloatingIp floatingIp1 = mock(FloatingIp.class);
         FloatingIp floatingIp2 = mock(FloatingIp.class);
-        List<? extends ApiObjectBase> floatingIpList = Arrays.asList(floatingIp1, floatingIp2);
+        List<? extends ApiObjectBase> floatingIpList = Arrays.asList(floatingIp, floatingIp2);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         FloatingIpPool floatingIpPool = mock(FloatingIpPool.class);
 
