--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
@@ -628,9 +628,7 @@
         GetHypervisorGuestOsNamesCommand cmd = mock(GetHypervisorGuestOsNamesCommand.class);
         when(cmd.getKeyword()).thenReturn("CentOS");
         when(_resource.getHyperHost(context, null)).thenReturn(hyperHost);
-        GuestOsDescriptor guestOsDescriptor = mock(GuestOsDescriptor.class);
-        when(guestOsDescriptor.getFullName()).thenReturn("centos64Guest");
-        when(guestOsDescriptor.getId()).thenReturn("centos64Guest");
+        GuestOsDescriptor guestOsDescriptor = createGuestOsDescriptor();
         List<GuestOsDescriptor> guestOsDescriptors = new ArrayList<>();
         guestOsDescriptors.add(guestOsDescriptor);
         when(hyperHost.getGuestOsDescriptors()).thenReturn(guestOsDescriptors);
@@ -645,9 +643,7 @@
     public void testGetHypervisorGuestOsNamesCommandSuccessWithoutKeyword() throws Exception {
         GetHypervisorGuestOsNamesCommand cmd = mock(GetHypervisorGuestOsNamesCommand.class);
         when(_resource.getHyperHost(context, null)).thenReturn(hyperHost);
-        GuestOsDescriptor guestOsDescriptor = mock(GuestOsDescriptor.class);
-        when(guestOsDescriptor.getFullName()).thenReturn("centos64Guest");
-        when(guestOsDescriptor.getId()).thenReturn("centos64Guest");
+        GuestOsDescriptor guestOsDescriptor = createGuestOsDescriptor();
         List<GuestOsDescriptor> guestOsDescriptors = new ArrayList<>();
         guestOsDescriptors.add(guestOsDescriptor);
         when(hyperHost.getGuestOsDescriptors()).thenReturn(guestOsDescriptors);
@@ -874,4 +870,11 @@
         Mockito.verify(vmwareResource, Mockito.times(1)).removeVirtualTPMDevice(vmConfigSpec, tpm);
         Mockito.verify(deviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
     }
+
+    private static GuestOsDescriptor createGuestOsDescriptor() {
+        GuestOsDescriptor guestOsDescriptor = mock(GuestOsDescriptor.class);
+        when(guestOsDescriptor.getFullName()).thenReturn("centos64Guest");
+        when(guestOsDescriptor.getId()).thenReturn("centos64Guest");
+        return guestOsDescriptor;
+    }
 }
