--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
@@ -52,6 +52,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class Xenserver625StorageProcessorTest {
+    @Mock
+    private Host hostMock;
 
     @InjectMocks
     private Xenserver625StorageProcessor xenserver625StorageProcessor;
@@ -312,7 +314,6 @@
 
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
-        Host hostMock = Mockito.mock(Host.class);
 
         try (MockedStatic<Host> ignored = Mockito.mockStatic(
                 Host.class); MockedStatic<SR> ignored1 = Mockito.mockStatic(SR.class)) {
@@ -339,7 +340,7 @@
         Mockito.doReturn(srMock).when(xenserver625StorageProcessor).retrieveAlreadyConfiguredSrWithoutException(connectionMock, pathMock);
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
-        Host hostMock = Mockito.mock(Host.class);
+        // removed local mock; replaced with global field `hostMock`
 
         try (MockedStatic<Host> ignored = Mockito.mockStatic(Host.class);MockedStatic<SR> ignored1 =
                 Mockito.mockStatic(SR.class) ) {
@@ -370,7 +371,6 @@
         SR srMock = Mockito.mock(SR.class);
         String srUuid = UuidUtils.nameUUIDFromBytes(pathMock.getBytes()).toString();
 
-        Host hostMock = Mockito.mock(Host.class);
 
         try (MockedStatic<Host> ignored = Mockito.mockStatic(Host.class);MockedStatic<SR> ignored1 =
                 Mockito.mockStatic(SR.class);MockedStatic<PBD> pbdMockedStatic = Mockito.mockStatic(PBD.class)) {
