--- a/plugins/integrations/cloudian/src/test/java/org/apache/cloudstack/cloudian/client/CloudianClientTest.java
+++ b/plugins/integrations/cloudian/src/test/java/org/apache/cloudstack/cloudian/client/CloudianClientTest.java
@@ -129,11 +129,10 @@
     @Test
     public void getNonEmptyContentStreamEmpty() {
         InputStream emptyStream = new ByteArrayInputStream(new byte[]{});
-        HttpEntity entity = mock(HttpEntity.class);
+        HttpEntity entity = createMockHttpEntity(emptyStream);
         HttpResponse response = mock(HttpResponse.class);
         when(response.getEntity()).thenReturn(entity);
         try {
-                when(entity.getContent()).thenReturn(emptyStream);
                 Assert.assertNull(client.getNonEmptyContentStream(response));
         } catch (IOException e) {
                 Assert.fail("Should not be any exception here");
@@ -143,11 +142,10 @@
     @Test
     public void getNonEmptyContentStreamWithContent() {
         InputStream nonEmptyStream = new ByteArrayInputStream(new byte[]{9, 8});
-        HttpEntity entity = mock(HttpEntity.class);
+        HttpEntity entity = createMockHttpEntity(nonEmptyStream);
         HttpResponse response = mock(HttpResponse.class);
         when(response.getEntity()).thenReturn(entity);
         try {
-                when(entity.getContent()).thenReturn(nonEmptyStream);
                 InputStream is = client.getNonEmptyContentStream(response);
                 Assert.assertNotNull(is);
                 Assert.assertEquals(9, is.read());
@@ -787,4 +785,10 @@
         boolean result = client.removeGroup(group.getGroupId());
         Assert.assertFalse(result);
     }
+
+    private static HttpEntity createMockHttpEntity(InputStream contentStream) {
+        HttpEntity entity = mock(HttpEntity.class);
+        when(entity.getContent()).thenReturn(contentStream);
+        return entity;
+    }
 }
