--- a/engine/schema/src/test/java/com/cloud/storage/dao/VolumeDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/storage/dao/VolumeDaoImplTest.java
@@ -51,6 +51,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class VolumeDaoImplTest {
     @Mock
+    private ResultSet rs;
+    @Mock
     private PreparedStatement preparedStatementMock;
 
     @Mock
@@ -80,7 +82,6 @@
 
         when(TransactionLegacy.currentTxn()).thenReturn(transactionMock);
         when(transactionMock.prepareAutoCloseStatement(startsWith(ORDER_POOLS_NUMBER_OF_VOLUMES_FOR_ACCOUNT_QUERY_WITH_CLUSTER))).thenReturn(preparedStatementMock);
-        ResultSet rs = Mockito.mock(ResultSet.class);
         when(preparedStatementMock.executeQuery()).thenReturn(rs, rs);
 
         volumeDao.listPoolIdsByVolumeCount(dcId, podId, clusterId, accountId);
@@ -102,7 +103,6 @@
 
         when(TransactionLegacy.currentTxn()).thenReturn(transactionMock);
         when(transactionMock.prepareAutoCloseStatement(startsWith(ORDER_POOLS_NUMBER_OF_VOLUMES_FOR_ACCOUNT_QUERY_WITHOUT_CLUSTER))).thenReturn(preparedStatementMock);
-        ResultSet rs = Mockito.mock(ResultSet.class);
         when(preparedStatementMock.executeQuery()).thenReturn(rs, rs);
 
         volumeDao.listPoolIdsByVolumeCount(dcId, null, null, accountId);
