--- a/plugins/metrics/src/test/java/org/apache/cloudstack/metrics/MetricsServiceImplTest.java
+++ b/plugins/metrics/src/test/java/org/apache/cloudstack/metrics/MetricsServiceImplTest.java
@@ -61,6 +61,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class MetricsServiceImplTest {
+    @Mock
+    private Date endDate;
 
     @Spy
     @InjectMocks
@@ -248,7 +250,6 @@
         Map<Long,List<VmStatsVO>> expected = new HashMap<Long,List<VmStatsVO>>();
         expected.put(fakeVmId1, new ArrayList<VmStatsVO>());
         Date startDate = Mockito.mock(Date.class);
-        Date endDate = Mockito.mock(Date.class);
 
         Map<Long,List<VmStatsVO>> result = spy.searchForVmMetricsStatsInternal(
                 startDate, endDate, Arrays.asList(userVmVOMock));
@@ -264,7 +265,6 @@
         Mockito.doNothing().when(spy).validateDateParams(Mockito.any(), Mockito.any());
         Map<Long,List<VmStatsVO>> expected = new HashMap<Long,List<VmStatsVO>>();
         Date startDate = Mockito.mock(Date.class);
-        Date endDate = Mockito.mock(Date.class);
         Map<Long,List<VmStatsVO>> result = spy.searchForVmMetricsStatsInternal(
                 startDate, endDate, new ArrayList<UserVmVO>());
 
