--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
@@ -73,15 +73,14 @@
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
         InputStream inputStream = mock(InputStream.class);
-        DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
         DocumentBuilder documentBuilder = mock(DocumentBuilder.class);
 
+        DocumentBuilderFactory documentBuilderFactory = createMockDocumentBuilderFactory(documentBuilder);
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         when(httpEntity.getContent()).thenReturn(inputStream);
         documentBuilderFactoryMocked.when(DocumentBuilderFactory::newInstance).thenReturn(documentBuilderFactory);
-        when(documentBuilderFactory.newDocumentBuilder()).thenReturn(documentBuilder);
         when(documentBuilder.parse(any(InputStream.class))).thenReturn(document);
 
         assertEquals(document, introspectApiConnector.getSnhItfReq("948f421c-edde-4518-a391-09299cc25dc2"));
@@ -119,17 +118,22 @@
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
         InputStream inputStream = mock(InputStream.class);
-        DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
         DocumentBuilder documentBuilder = mock(DocumentBuilder.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         when(httpEntity.getContent()).thenReturn(inputStream);
+        DocumentBuilderFactory documentBuilderFactory = createMockDocumentBuilderFactory(documentBuilder);
         documentBuilderFactoryMocked.when(DocumentBuilderFactory::newInstance).thenReturn(documentBuilderFactory);
-        when(documentBuilderFactory.newDocumentBuilder()).thenReturn(documentBuilder);
         when(documentBuilder.parse(any(InputStream.class))).thenThrow(SAXException.class);
 
         assertNull(introspectApiConnector.getSnhItfReq("948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static DocumentBuilderFactory createMockDocumentBuilderFactory(DocumentBuilder documentBuilder) {
+        DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
+        when(documentBuilderFactory.newDocumentBuilder()).thenReturn(documentBuilder);
+        return documentBuilderFactory;
+    }
 }
