--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.network.tungsten.vrouter;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.mockito.ArgumentMatchers.any;
@@ -46,6 +47,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class IntrospectApiConnectorImplTest {
+    @Mock
+    private InputStream inputStream;
     IntrospectApiConnector introspectApiConnector;
 
     MockedStatic<HttpClients> httpClientsMocked;
@@ -72,7 +75,6 @@
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
-        InputStream inputStream = mock(InputStream.class);
         DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
         DocumentBuilder documentBuilder = mock(DocumentBuilder.class);
 
@@ -102,7 +104,6 @@
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
-        InputStream inputStream = mock(InputStream.class);
         DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
@@ -118,7 +119,6 @@
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
-        InputStream inputStream = mock(InputStream.class);
         DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
         DocumentBuilder documentBuilder = mock(DocumentBuilder.class);
 
