--- a/core/src/test/java/com/cloud/storage/template/QCOW2ProcessorTest.java
+++ b/core/src/test/java/com/cloud/storage/template/QCOW2ProcessorTest.java
@@ -90,8 +90,7 @@
     public void testGetVirtualSizeWhenVirtualSizeThrowsException() throws Exception {
         long virtualSize = 2000;
         long actualSize = 1000;
-        File mockFile = Mockito.mock(File.class);
-        Mockito.when(mockFile.length()).thenReturn(actualSize);
+        File mockFile = MockFile.createMockFile(actualSize);
         Mockito.doThrow(new IOException("virtual size calculation failed")).when(processor).getTemplateVirtualSize((File)Mockito.any());
         Assert.assertEquals(actualSize, processor.getVirtualSize(mockFile));
         Mockito.verify(mockFile, Mockito.times(1)).length();
@@ -105,4 +104,15 @@
         Assert.assertEquals(virtualSize, processor.getVirtualSize(mockFile));
         Mockito.verify(mockFile, Mockito.times(0)).length();
     }
+
+    public final class MockFile {
+        private MockFile() {
+        }
+
+        public static java.io.File createMockFile(long actualSize) {
+            java.io.File mockFile = org.mockito.Mockito.mock(java.io.File.class);
+            org.mockito.Mockito.when(mockFile.length()).thenReturn(actualSize);
+            return mockFile;
+        }
+    }
 }
--- a/core/src/test/java/com/cloud/storage/template/VhdProcessorTest.java
+++ b/core/src/test/java/com/cloud/storage/template/VhdProcessorTest.java
@@ -94,8 +94,7 @@
     public void testGetVirtualSizeWhenVirtualSizeThrowsException() throws Exception {
         long virtualSize = 2000;
         long actualSize = 1000;
-        File mockFile = Mockito.mock(File.class);
-        Mockito.when(mockFile.length()).thenReturn(actualSize);
+        File mockFile = MockFile.createMockFile(actualSize);
         Mockito.doThrow(new IOException("virtual size calculation failed")).when(processor).getTemplateVirtualSize((File) Mockito.any());
         Assert.assertEquals(actualSize, processor.getVirtualSize(mockFile));
         Mockito.verify(mockFile,Mockito.times(1)).length();
@@ -142,4 +141,15 @@
         long actualVirtualSizeBytes = processor.getVirtualSize(new File(zipVhdPath));
         Assert.assertEquals(expectedVirtualSizeBytes, actualVirtualSizeBytes);
     }
+
+    public final class MockFile {
+        private MockFile() {
+        }
+
+        public static java.io.File createMockFile(long actualSize) {
+            java.io.File mockFile = org.mockito.Mockito.mock(java.io.File.class);
+            org.mockito.Mockito.when(mockFile.length()).thenReturn(actualSize);
+            return mockFile;
+        }
+    }
 }
