--- a/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
+++ b/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
@@ -59,9 +59,8 @@
                 Map.entry("description", "The first NIC of VNF appliance")
         )));
 
-        Map<String, Object> vnfNicsMock = Mockito.mock(Map.class);
-        Mockito.when(vnfNicsMock.values()).thenReturn(vnfNics.values());
-
+
+        Map<String, Object> vnfNicsMock = createMockMap(vnfNics.values());
         List<VnfNic> nicsList = VnfTemplateUtils.getVnfNicsList(vnfNicsMock);
         Mockito.verify(vnfNicsMock).values();
 
@@ -86,9 +85,8 @@
                 Map.entry("description", "The second NIC of VNF appliance")
         )));
 
-        Map<String, Object> vnfNicsMock = Mockito.mock(Map.class);
-        Mockito.when(vnfNicsMock.values()).thenReturn(vnfNics.values());
-
+
+        Map<String, Object> vnfNicsMock = createMockMap(vnfNics.values());
         List<VnfNic> nicsList = VnfTemplateUtils.getVnfNicsList(vnfNicsMock);
     }
 
@@ -101,9 +99,8 @@
                 Map.entry("description", "The second NIC of VNF appliance")
         )));
 
-        Map<String, Object> vnfNicsMock = Mockito.mock(Map.class);
-        Mockito.when(vnfNicsMock.values()).thenReturn(vnfNics.values());
-
+
+        Map<String, Object> vnfNicsMock = createMockMap(vnfNics.values());
         List<VnfNic> nicsList = VnfTemplateUtils.getVnfNicsList(vnfNicsMock);
     }
 
@@ -117,9 +114,8 @@
                 Map.entry("description", "The second NIC of VNF appliance")
         )));
 
-        Map<String, Object> vnfNicsMock = Mockito.mock(Map.class);
-        Mockito.when(vnfNicsMock.values()).thenReturn(vnfNics.values());
-
+
+        Map<String, Object> vnfNicsMock = createMockMap(vnfNics.values());
         List<VnfNic> nicsList = VnfTemplateUtils.getVnfNicsList(vnfNicsMock);
     }
 
@@ -258,4 +254,10 @@
         cidrList.add("10.10.10.0/33");
         VnfTemplateUtils.validateVnfCidrList(cidrList);
     }
+
+    private static Map<String, Object> createMockMap(Collection<Object> values) {
+        Map<String, Object> vnfNicsMock = Mockito.mock(Map.class);
+        Mockito.when(vnfNicsMock.values()).thenReturn(values);
+        return vnfNicsMock;
+    }
 }
