--- a/server/src/test/java/org/apache/cloudstack/jsinterpreter/JsInterpreterHelperTest.java
+++ b/server/src/test/java/org/apache/cloudstack/jsinterpreter/JsInterpreterHelperTest.java
@@ -192,12 +192,9 @@
         fields.put("field", jsonNodeMock);
 
         JsonNode root = Mockito.mock(JsonNode.class);
-        JsonNode node1 = Mockito.mock(JsonNode.class);
-        JsonNode node2 = Mockito.mock(JsonNode.class);
-        JsonNode node3 = Mockito.mock(JsonNode.class);
-        Mockito.doReturn(fields.entrySet().iterator()).when(node1).fields();
-        Mockito.doReturn(fields.entrySet().iterator()).when(node2).fields();
-        Mockito.doReturn(fields.entrySet().iterator()).when(node3).fields();
+        JsonNode node2 = createMockJsonNode(fields);
+        JsonNode node3 = createMockJsonNode(fields);
+        JsonNode node1 = createMockJsonNode(fields);
 
         Map<String, JsonNode> childrenMap = new HashMap<>();
         childrenMap.put("node1", node1);
@@ -225,4 +222,10 @@
 
         Assert.assertEquals("value.osName", jsInterpreterHelperSpy.getVariable().toString());
     }
+
+    private static JsonNode createMockJsonNode(Map<String, JsonNode> fields) {
+        JsonNode node = Mockito.mock(JsonNode.class);
+        Mockito.doReturn(fields.entrySet().iterator()).when(node).fields();
+        return node;
+    }
 }
