--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
@@ -109,6 +109,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VmwareResourceTest {
+    @Mock
+    private List<VirtualDeviceConfigSpec> deviceChanges;
 
     private static final String VOLUME_PATH = "XXXXXXXXXXXX";
 
@@ -848,7 +850,6 @@
         details.put(VmDetailConstants.VIRTUAL_TPM_ENABLED, "true");
         when(vmSpec.getDetails()).thenReturn(details);
         when(vmMo.getAllDeviceList()).thenReturn(new ArrayList<>());
-        List<VirtualDeviceConfigSpec> deviceChanges = Mockito.mock(List.class);
         when(vmConfigSpec.getDeviceChange()).thenReturn(deviceChanges);
 
         vmwareResource.configureVirtualTPM(vmMo, vmSpec, vmConfigSpec);
@@ -867,7 +868,6 @@
         when(vmSpec.getDetails()).thenReturn(details);
         VirtualTPM tpm = new VirtualTPM();
         when(vmMo.getAllDeviceList()).thenReturn(List.of(tpm));
-        List<VirtualDeviceConfigSpec> deviceChanges = Mockito.mock(List.class);
         when(vmConfigSpec.getDeviceChange()).thenReturn(deviceChanges);
 
         vmwareResource.configureVirtualTPM(vmMo, vmSpec, vmConfigSpec);
