--- a/utils/src/test/java/org/apache/cloudstack/utils/mailing/SMTPMailSenderTest.java
+++ b/utils/src/test/java/org/apache/cloudstack/utils/mailing/SMTPMailSenderTest.java
@@ -48,6 +48,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class SMTPMailSenderTest extends TestCase {
+    @Mock
+    private MimeMessage mimeMessage;
 
     private SMTPMailSender smtpMailSender;
 
@@ -557,8 +559,8 @@
 
     @Test
     public void setMailRecipientsTest() throws UnsupportedEncodingException, MessagingException {
-        SMTPMessage messageMock = new SMTPMessage(Mockito.mock(MimeMessage.class));
-
+
+        SMTPMessage messageMock = new SMTPMessage(mimeMessage);
         Set<MailAddress> recipients = new LinkedHashSet<>();
         recipients.add(new MailAddress(null));
         recipients.add(new MailAddress(""));
@@ -579,8 +581,8 @@
 
     @Test
     public void setMailRecipientsTestOnlyInvalidEmailSettings() throws UnsupportedEncodingException, MessagingException {
-        SMTPMessage messageMock = new SMTPMessage(Mockito.mock(MimeMessage.class));
-
+
+        SMTPMessage messageMock = new SMTPMessage(mimeMessage);
         messageMock = messageMock = Mockito.spy(messageMock);
         Mockito.doReturn(new Address[0]).when(messageMock).getAllRecipients();
 
@@ -599,8 +601,8 @@
 
     @Test
     public void setMailRecipientsTestWarnLogging() throws UnsupportedEncodingException, MessagingException  {
-        SMTPMessage messageMock = new SMTPMessage(Mockito.mock(MimeMessage.class));
-
+
+        SMTPMessage messageMock = new SMTPMessage(mimeMessage);
         smtpMailSenderMock.setMailRecipients(messageMock, null, "");
 
         Mockito.verify(smtpMailSenderMock.logger).error(Mockito.anyString());
