--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -175,6 +175,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VirtualMachineManagerImplTest {
+    @Mock
+    private NicTO nicTO;
 
     @Spy
     @InjectMocks
@@ -1632,7 +1634,7 @@
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
 
-        when(vmTO.getNics()).thenReturn(new NicTO[]{mock(NicTO.class)});
+        when(vmTO.getNics()).thenReturn(new NicTO[] { nicTO });
         when(updatedTO.getNics()).thenReturn(null);
 
         virtualMachineManagerImpl.updateExternalVmNicsFromPrepareAnswer(vmTO, updatedTO);
@@ -1646,8 +1648,8 @@
         VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
 
         when(vmTO.getNics()).thenReturn(null);
-        when(updatedTO.getNics()).thenReturn(new NicTO[]{mock(NicTO.class)});
-
+
+        when(updatedTO.getNics()).thenReturn(new NicTO[] { nicTO });
         virtualMachineManagerImpl.updateExternalVmNicsFromPrepareAnswer(vmTO, updatedTO);
 
         verify(_nicsDao, never()).findByUuid(anyString());
@@ -1683,7 +1685,7 @@
         VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
         when(vmProfile.getTemplate()).thenReturn(template);
-        NicTO[] nics = new NicTO[]{mock(NicTO.class)};
+        NicTO[] nics = new NicTO[] { nicTO };
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         when(vmTO.getNics()).thenReturn(nics);
         doReturn(vmTO).when(virtualMachineManagerImpl).toVmTO(vmProfile);
