--- a/plugins/hypervisors/kvm/src/test/java/org/apache/cloudstack/utils/linux/KVMHostInfoTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/org/apache/cloudstack/utils/linux/KVMHostInfoTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.utils.linux;
 
+import org.mockito.Mock;
 import org.apache.commons.lang.SystemUtils;
 import org.junit.Assert;
 import org.junit.Assume;
@@ -31,13 +32,14 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KVMHostInfoTest {
+    @Mock
+    private NodeInfo nodeInfo;
     @Test
     public void getCpuSpeed() {
         if (!System.getProperty("os.name").equals("Linux")) {
             return;
         }
         Assume.assumeTrue(SystemUtils.IS_OS_LINUX);
-        NodeInfo nodeInfo = Mockito.mock(NodeInfo.class);
         nodeInfo.mhz = 1000;
         Assert.assertTrue(KVMHostInfo.getCpuSpeed(null, nodeInfo) > 0L);
     }
@@ -63,7 +65,6 @@
         }
         try (MockedStatic<LibvirtConnection> ignored = Mockito.mockStatic(LibvirtConnection.class)) {
             Connect conn = Mockito.mock(Connect.class);
-            NodeInfo nodeInfo = Mockito.mock(NodeInfo.class);
             nodeInfo.mhz = 1000;
             String capabilitiesXml = "<capabilities></capabilities>";
 
@@ -84,7 +85,6 @@
         }
         try (MockedStatic<LibvirtConnection> ignored = Mockito.mockStatic(LibvirtConnection.class)) {
             Connect conn = Mockito.mock(Connect.class);
-            NodeInfo nodeInfo = Mockito.mock(NodeInfo.class);
             nodeInfo.cpus = 10;
             String capabilitiesXml = "<capabilities></capabilities>";
 
