--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -366,8 +366,7 @@
     @Test
     public void prepareForMaintenanceCmdNoOtherMsHostsWithUpState() {
         Mockito.when(msHostDao.listBy(any())).thenReturn(new ArrayList<>());
-        PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
-        Mockito.when(cmd.getAlgorithm()).thenReturn("test algorithm");
+        PrepareForMaintenanceCmd cmd = createMockPrepareForMaintenanceCmd();
         Mockito.doNothing().when(indirectAgentLBMock).checkLBAlgorithmName(anyString());
 
         Assert.assertThrows(CloudRuntimeException.class, () -> {
@@ -381,8 +380,7 @@
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost);
         Mockito.when(msHostDao.listBy(ManagementServerHost.State.Up)).thenReturn(msHostList);
-        PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
-        Mockito.when(cmd.getAlgorithm()).thenReturn("test algorithm");
+        PrepareForMaintenanceCmd cmd = createMockPrepareForMaintenanceCmd();
         Mockito.doNothing().when(indirectAgentLBMock).checkLBAlgorithmName(anyString());
 
         Assert.assertThrows(CloudRuntimeException.class, () -> {
@@ -568,4 +566,10 @@
         spy.cancelMaintenance(cmd);
         Mockito.verify(clusterManagerMock, Mockito.times(1)).execute(anyString(), anyLong(), anyString(), anyBoolean());
     }
+
+    private static PrepareForMaintenanceCmd createMockPrepareForMaintenanceCmd() {
+        PrepareForMaintenanceCmd cmd = mock(PrepareForMaintenanceCmd.class);
+        Mockito.when(cmd.getAlgorithm()).thenReturn("test algorithm");
+        return cmd;
+    }
 }
