--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -921,13 +921,10 @@
 
     @Test
     public void testUpdateGuestOsCategory_NoChanges() {
-        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = new GuestOSCategoryVO("Old name", false);
         long id = 1L;
-        when(updateCmd.getId()).thenReturn(id);
+        UpdateGuestOsCategoryCmd updateCmd = createMockUpdateGuestOsCategoryCmd(id);
         when(updateCmd.getName()).thenReturn(null);
-        when(updateCmd.isFeatured()).thenReturn(null);
-        when(updateCmd.getSortKey()).thenReturn(null);
         when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         GuestOsCategory result = spy.updateGuestOsCategory(updateCmd);
         Assert.assertNotNull(result);
@@ -939,14 +936,11 @@
 
     @Test
     public void testUpdateGuestOsCategory_UpdateNameOnly() {
-        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = new GuestOSCategoryVO("Old name", false);
         long id = 1L;
         String name = "Updated Name";
-        Mockito.when(updateCmd.getId()).thenReturn(id);
+        UpdateGuestOsCategoryCmd updateCmd = createMockUpdateGuestOsCategoryCmd(id);
         Mockito.when(updateCmd.getName()).thenReturn(name);
-        Mockito.when(updateCmd.isFeatured()).thenReturn(null);
-        Mockito.when(updateCmd.getSortKey()).thenReturn(null);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         Mockito.when(guestOSCategoryDao.update(Mockito.eq(id), any(GuestOSCategoryVO.class))).thenReturn(true);
         GuestOsCategory result = spy.updateGuestOsCategory(updateCmd);
@@ -2493,4 +2487,12 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static UpdateGuestOsCategoryCmd createMockUpdateGuestOsCategoryCmd(long id) {
+        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
+        when(updateCmd.getId()).thenReturn(id);
+        when(updateCmd.getSortKey()).thenReturn(null);
+        when(updateCmd.isFeatured()).thenReturn(null);
+        return updateCmd;
+    }
 }
