--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricAddressGroupCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricAddressGroupCmdTest.java
@@ -80,7 +80,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenAddressGroup(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricAddressGroupCmd.execute();
@@ -88,4 +88,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricApplicationPolicySetCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricApplicationPolicySetCmdTest.java
@@ -80,7 +80,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenApplicationPolicySet(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricApplictionPolicySetCmd.execute();
@@ -88,4 +88,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricFirewallPolicyCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricFirewallPolicyCmdTest.java
@@ -80,8 +80,7 @@
         BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
-        List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenFirewallPolicy(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricFirewallPolicyCmd.execute();
@@ -89,4 +88,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricFirewallRuleCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricFirewallRuleCmdTest.java
@@ -81,7 +81,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenFirewallRule(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricFirewallRuleCmd.execute();
@@ -89,4 +89,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricLogicalRouterCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricLogicalRouterCmdTest.java
@@ -81,7 +81,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listRoutingLogicalRouter(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricLogicalRouterCmd.execute();
@@ -89,4 +89,18 @@
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
         Assert.assertEquals(baseResponseList, responseList.getResponses());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNetworkCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNetworkCmdTest.java
@@ -83,7 +83,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenNetwork(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyBoolean())).thenReturn(baseResponseList);
         listTungstenFabricNetworkCmd.execute();
@@ -92,4 +92,18 @@
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
         Assert.assertEquals(baseResponseList, responseList.getResponses());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNicCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNicCmdTest.java
@@ -82,7 +82,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenNic(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricNicCmd.execute();
@@ -90,4 +90,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricPolicyCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricPolicyCmdTest.java
@@ -83,7 +83,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenPolicy(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyLong(),
@@ -93,4 +93,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricPolicyRuleCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricPolicyRuleCmdTest.java
@@ -84,7 +84,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenPolicyRule(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
@@ -93,4 +93,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricServiceGroupCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricServiceGroupCmdTest.java
@@ -81,7 +81,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenServiceGroup(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricServiceGroupCmd.execute();
@@ -89,4 +89,18 @@
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
         Assert.assertEquals(baseResponseList, responseList.getResponses());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagCmdTest.java
@@ -91,7 +91,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenTags(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(),
                 ArgumentMatchers.anyString(),
@@ -104,4 +104,18 @@
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
         Assert.assertEquals(baseResponseList, responseList.getResponses());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagTypeCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagTypeCmdTest.java
@@ -80,7 +80,7 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenTagTypes(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricTagTypeCmd.execute();
@@ -88,4 +88,18 @@
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricVmCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricVmCmdTest.java
@@ -79,11 +79,25 @@
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
-        Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
+        tungstenService = MockTungstenService.createMockTungstenService(tungstenProviderVO);
         Mockito.when(tungstenService.listTungstenVm(ArgumentMatchers.anyLong(), ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricVmCmd.execute();
         ListResponse<BaseResponse> responseList = (ListResponse<BaseResponse>) listTungstenFabricVmCmd.getResponseObject();
         Assert.assertEquals(baseResponseList, responseList.getResponses());
         Assert.assertEquals(Integer.valueOf(1), responseList.getCount());
     }
+
+    public final class MockTungstenService {
+        private MockTungstenService() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenService createMockTungstenService(
+                com.cloud.network.element.TungstenProviderVO tungstenProviderVO) {
+            org.apache.cloudstack.network.tungsten.service.TungstenService tungstenService =
+                    org.mockito.Mockito.mock(org.apache.cloudstack.network.tungsten.service.TungstenService.class);
+            org.mockito.Mockito.when(tungstenService.getTungstenProviders())
+                    .thenReturn(java.util.Arrays.asList(tungstenProviderVO));
+            return tungstenService;
+        }
+    }
 }
