--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1284,8 +1284,7 @@
     @Test(expected = CloudRuntimeException.class)
     public void testUsedDeleteDataCenterGuestIpv6Prefix() {
         final Long prefixId = 1L;
-        DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
-        Mockito.when(cmd.getId()).thenReturn(prefixId);
+        DeleteGuestNetworkIpv6PrefixCmd cmd = createMockDeleteGuestNetworkIpv6PrefixCmd(prefixId);
         DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
         Mockito.when(prefixVO.getId()).thenReturn(prefixId);
         Mockito.when(dataCenterGuestIpv6PrefixDao.findById(Mockito.anyLong())).thenReturn(prefixVO);
@@ -1297,8 +1296,7 @@
     @Test
     public void testDeleteDataCenterGuestIpv6Prefix() {
         final Long prefixId = 1L;
-        DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
-        Mockito.when(cmd.getId()).thenReturn(prefixId);
+        DeleteGuestNetworkIpv6PrefixCmd cmd = createMockDeleteGuestNetworkIpv6PrefixCmd(prefixId);
         DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
         Mockito.when(prefixVO.getId()).thenReturn(prefixId);
         Mockito.when(dataCenterGuestIpv6PrefixDao.findById(Mockito.anyLong())).thenReturn(prefixVO);
@@ -1508,4 +1506,10 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static DeleteGuestNetworkIpv6PrefixCmd createMockDeleteGuestNetworkIpv6PrefixCmd(Long prefixId) {
+        DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(prefixId);
+        return cmd;
+    }
 }
