--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -258,9 +258,8 @@
     @Test
     public void triggerShutdownCmdNoMsHost() {
         Mockito.when(msHostDao.findById(1L)).thenReturn(null);
-        TriggerShutdownCmd cmd = mock(TriggerShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
-
+
+        TriggerShutdownCmd cmd = createMockTriggerShutdownCmd();
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.triggerShutdown(cmd);
         });
@@ -271,9 +270,8 @@
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getState()).thenReturn(ManagementServerHost.State.PreparingForMaintenance);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost);
-        TriggerShutdownCmd cmd = mock(TriggerShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
-
+
+        TriggerShutdownCmd cmd = createMockTriggerShutdownCmd();
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.triggerShutdown(cmd);
         });
@@ -288,9 +286,8 @@
         msHostList.add(msHost2);
         Mockito.lenient().when(msHostDao.listBy(any())).thenReturn(msHostList);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost1);
-        TriggerShutdownCmd cmd = mock(TriggerShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
-
+
+        TriggerShutdownCmd cmd = createMockTriggerShutdownCmd();
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.triggerShutdown(cmd);
         });
@@ -301,8 +298,7 @@
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getState()).thenReturn(ManagementServerHost.State.ReadyToShutDown);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost);
-        TriggerShutdownCmd cmd = mock(TriggerShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        TriggerShutdownCmd cmd = createMockTriggerShutdownCmd();
         Mockito.when(clusterManagerMock.execute(anyString(), anyLong(), anyString(), anyBoolean())).thenReturn("Success");
 
         spy.triggerShutdown(cmd);
@@ -568,4 +564,10 @@
         spy.cancelMaintenance(cmd);
         Mockito.verify(clusterManagerMock, Mockito.times(1)).execute(anyString(), anyLong(), anyString(), anyBoolean());
     }
+
+    private static TriggerShutdownCmd createMockTriggerShutdownCmd() {
+        TriggerShutdownCmd cmd = mock(TriggerShutdownCmd.class);
+        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        return cmd;
+    }
 }
