--- a/server/src/test/java/org/apache/cloudstack/backup/BackupCompressionServiceJobControllerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupCompressionServiceJobControllerTest.java
@@ -460,8 +460,8 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(false).when(backupCompressionTaskEnabledMock).value();
-
+
+        backupCompressionTaskEnabledMock = MockConfigKey.createMockConfigKey(false);
         backupCompressionServiceJobControllerSpy.searchAndDispatchJobs();
 
         verify(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
@@ -474,7 +474,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupCompressionTaskEnabledMock).value();
+        backupCompressionTaskEnabledMock = MockConfigKey.createMockConfigKey(true);
         doReturn(false).when(backupCompressionServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
         backupCompressionServiceJobControllerSpy.searchAndDispatchJobs();
@@ -488,7 +488,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupCompressionTaskEnabledMock).value();
+        backupCompressionTaskEnabledMock = MockConfigKey.createMockConfigKey(true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupCompressionServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
@@ -506,7 +506,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("compression_lock", 300);
         doNothing().when(backupCompressionServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupCompressionTaskEnabledMock).value();
+        backupCompressionTaskEnabledMock = MockConfigKey.createMockConfigKey(true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupCompressionServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
         doReturn(List.of(internalBackupServiceJobVoMock)).when(internalBackupServiceJobDaoMock).listWaitingJobsAndScheduledToBeforeNow(eq(datacenterId),
@@ -525,4 +525,15 @@
         verify(internalBackupServiceJobDaoMock).unlockFromLockTable("compression_lock");
         verify(internalBackupServiceJobDaoMock).listExecutingJobsByZoneIdAndJobType(eq(datacenterId), eq(InternalBackupServiceJobType.StartCompression));
     }
+
+    public final class MockConfigKey {
+        private MockConfigKey() {
+        }
+
+        public static ConfigKey<Boolean> createMockConfigKey(boolean value) {
+            ConfigKey<Boolean> configKeyMock = Mockito.mock(ConfigKey.class);
+            doReturn(value).when(configKeyMock).value();
+            return configKeyMock;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/cloudstack/backup/BackupValidationServiceJobControllerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupValidationServiceJobControllerTest.java
@@ -86,8 +86,8 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(false).when(backupValidationTaskEnabledMock).value();
 
+        backupValidationTaskEnabledMock = MockConfigKey.createMockConfigKey(false);
         backupValidationServiceJobControllerSpy.searchAndDispatchJobs();
 
         verify(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
@@ -100,7 +100,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupValidationTaskEnabledMock).value();
+        backupValidationTaskEnabledMock = MockConfigKey.createMockConfigKey(true);
         doReturn(false).when(backupValidationServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
         backupValidationServiceJobControllerSpy.searchAndDispatchJobs();
@@ -114,7 +114,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupValidationTaskEnabledMock).value();
+        backupValidationTaskEnabledMock = MockConfigKey.createMockConfigKey(true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupValidationServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
 
@@ -132,7 +132,7 @@
         doReturn(List.of(dataCenterVoMock)).when(dataCenterDaoMock).listEnabledZones();
         doReturn(true).when(internalBackupServiceJobDaoMock).lockInLockTable("validation_lock", 300);
         doNothing().when(backupValidationServiceJobControllerSpy).rescheduleLostJobs();
-        doReturn(true).when(backupValidationTaskEnabledMock).value();
+        backupValidationTaskEnabledMock = MockConfigKey.createMockConfigKey(true);
         doReturn(datacenterId).when(dataCenterVoMock).getId();
         doReturn(true).when(backupValidationServiceJobControllerSpy).isFrameworkEnabledForZone(dataCenterVoMock);
         doReturn(List.of(internalBackupServiceJobVoMock)).when(backupValidationServiceJobControllerSpy).filterJobsOfDomainsAndAccountsWithDisabledValidationTask(any());
@@ -147,4 +147,15 @@
         verify(backupValidationServiceJobControllerSpy).submitQueuedJobsForExecution(any(), any(), any(), any(), eq(datacenterId));
         verify(internalBackupServiceJobDaoMock).unlockFromLockTable("validation_lock");
     }
+
+    public final class MockConfigKey {
+        private MockConfigKey() {
+        }
+
+        public static ConfigKey<Boolean> createMockConfigKey(boolean value) {
+            ConfigKey<Boolean> configKeyMock = Mockito.mock(ConfigKey.class);
+            doReturn(value).when(configKeyMock).value();
+            return configKeyMock;
+        }
+    }
 }
