--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -1767,11 +1767,8 @@
 
     @Test
     public void runCustomAction_SuccessfulExecution_ReturnsExpectedResult() throws Exception {
-        RunCustomActionCmd cmd = mock(RunCustomActionCmd.class);
-        when(cmd.getCustomActionId()).thenReturn(1L);
-        when(cmd.getResourceId()).thenReturn("vm-123");
-        when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
-
+
+        RunCustomActionCmd cmd = createRunCustomActionCmd();
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
         when(extensionCustomActionDao.findById(1L)).thenReturn(actionVO);
         when(actionVO.isEnabled()).thenReturn(true);
@@ -1871,11 +1868,8 @@
 
     @Test
     public void runCustomAction_ExecutionThrowsException() throws Exception {
-        RunCustomActionCmd cmd = mock(RunCustomActionCmd.class);
-        when(cmd.getCustomActionId()).thenReturn(1L);
-        when(cmd.getResourceId()).thenReturn("vm-123");
-        when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
-
+
+        RunCustomActionCmd cmd = createRunCustomActionCmd();
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
         when(extensionCustomActionDao.findById(1L)).thenReturn(actionVO);
         when(actionVO.isEnabled()).thenReturn(true);
@@ -1905,11 +1899,8 @@
 
     @Test(expected = PermissionDeniedException.class)
     public void runCustomAction_CheckAccessThrowsException() throws Exception {
-        RunCustomActionCmd cmd = mock(RunCustomActionCmd.class);
-        when(cmd.getCustomActionId()).thenReturn(1L);
-        when(cmd.getResourceId()).thenReturn("vm-123");
-        when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
-
+
+        RunCustomActionCmd cmd = createRunCustomActionCmd();
         ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
         when(extensionCustomActionDao.findById(1L)).thenReturn(actionVO);
         when(actionVO.isEnabled()).thenReturn(true);
@@ -3062,4 +3053,12 @@
         assertTrue(resp.getResult().get(ApiConstants.DETAILS).contains("Action failed"));
     }
 
+
+    private static RunCustomActionCmd createRunCustomActionCmd() {
+        RunCustomActionCmd cmd = mock(RunCustomActionCmd.class);
+        when(cmd.getCustomActionId()).thenReturn(1L);
+        when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
+        when(cmd.getResourceId()).thenReturn("vm-123");
+        return cmd;
+    }
 }
