--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionServiceTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionServiceTest.java
@@ -219,48 +219,36 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void addKubernetesSupportedVersionLowerUnsupportedTest() {
-        AddKubernetesSupportedVersionCmd cmd = Mockito.mock(AddKubernetesSupportedVersionCmd.class);
-        when(cmd.getMinimumCpu()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU);
-        when(cmd.getMinimumRamSize()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE);
-        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
-        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
-        CallContext.register(user, account);
-        when(cmd.getSemanticVersion()).thenReturn("1.1.1");
+        AddKubernetesSupportedVersionCmd cmd = createMockAddKubernetesSupportedVersionCmd(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU, KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE, "1.1.1");
+        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
+        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
+        CallContext.register(user, account);
         kubernetesVersionService.addKubernetesSupportedVersion(cmd);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void addKubernetesSupportedVersionInvalidCpuTest() {
-        AddKubernetesSupportedVersionCmd cmd = Mockito.mock(AddKubernetesSupportedVersionCmd.class);
-        when(cmd.getMinimumCpu()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU-1);
-        when(cmd.getMinimumRamSize()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE);
-        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
-        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
-        when(cmd.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AddKubernetesSupportedVersionCmd cmd = createMockAddKubernetesSupportedVersionCmd(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU - 1, KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE, KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
+        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
         CallContext.register(user, account);
         kubernetesVersionService.addKubernetesSupportedVersion(cmd);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void addKubernetesSupportedVersionInvalidRamSizeTest() {
-        AddKubernetesSupportedVersionCmd cmd = Mockito.mock(AddKubernetesSupportedVersionCmd.class);
-        when(cmd.getMinimumCpu()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU);
-        when(cmd.getMinimumRamSize()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE-10);
-        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
-        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
-        when(cmd.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AddKubernetesSupportedVersionCmd cmd = createMockAddKubernetesSupportedVersionCmd(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU, KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE - 10, KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
+        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
         CallContext.register(user, account);
         kubernetesVersionService.addKubernetesSupportedVersion(cmd);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void addKubernetesSupportedVersionEmptyUrlTest() {
-        AddKubernetesSupportedVersionCmd cmd = Mockito.mock(AddKubernetesSupportedVersionCmd.class);
-        when(cmd.getMinimumCpu()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU);
-        when(cmd.getMinimumRamSize()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE);
-        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
-        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
-        when(cmd.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AddKubernetesSupportedVersionCmd cmd = createMockAddKubernetesSupportedVersionCmd(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU, KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE, KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
+        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
         CallContext.register(user, account);
         when(cmd.getUrl()).thenReturn("");
         kubernetesVersionService.addKubernetesSupportedVersion(cmd);
@@ -268,15 +256,12 @@
 
     @Test
     public void addKubernetesSupportedVersionIsoUrlTest() throws ResourceAllocationException, NoSuchFieldException {
-        AddKubernetesSupportedVersionCmd cmd = Mockito.mock(AddKubernetesSupportedVersionCmd.class);
-        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
-        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
-        CallContext.register(user, account);
-        when(cmd.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
+        AccountVO account = new AccountVO("admin", 1L, "", Account.Type.ADMIN, "uuid");
+        UserVO user = new UserVO(1, "adminuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
+        CallContext.register(user, account);
+        AddKubernetesSupportedVersionCmd cmd = createMockAddKubernetesSupportedVersionCmd(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU, KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE, KubernetesVersionService.MIN_KUBERNETES_VERSION);
         when(cmd.getUrl()).thenReturn("https://download.cloudstack.com");
         when(cmd.getChecksum()).thenReturn(null);
-        when(cmd.getMinimumCpu()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_CPU);
-        when(cmd.getMinimumRamSize()).thenReturn(KubernetesClusterService.MIN_KUBERNETES_CLUSTER_NODE_RAM_SIZE);
         when(cmd.getArch()).thenReturn(CPU.CPUArch.amd64);
         Account systemAccount =  new AccountVO("system", 1L, "", Account.Type.ADMIN, "uuid");
         when(accountManager.getSystemAccount()).thenReturn(systemAccount);
@@ -343,4 +328,12 @@
         Assert.assertEquals(KubernetesSupportedVersion.State.Disabled.toString(), response.getState());
         Assert.assertEquals(KubernetesVersionService.MIN_KUBERNETES_VERSION, response.getSemanticVersion());
     }
+
+    private static AddKubernetesSupportedVersionCmd createMockAddKubernetesSupportedVersionCmd(int minimumCpu, int minimumRamSize, String semanticVersion) {
+        AddKubernetesSupportedVersionCmd cmd = Mockito.mock(AddKubernetesSupportedVersionCmd.class);
+        when(cmd.getMinimumCpu()).thenReturn(minimumCpu);
+        when(cmd.getMinimumRamSize()).thenReturn(minimumRamSize);
+        when(cmd.getSemanticVersion()).thenReturn(semanticVersion);
+        return cmd;
+    }
 }
