--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -1182,7 +1182,6 @@
     @Test
     public void applyTungstenTagTest() {
         TungstenAnswer applyTungstenTagAnswer = MockTungstenAnswerFactory.get(true);
-        TungstenTag tungstenTag = mock(TungstenTag.class);
         Tag tag = mock(Tag.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
@@ -1191,13 +1190,9 @@
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
+        TungstenTag tungstenTag = createMockTungstenTag(tag, networkPolicy, virtualNetwork, virtualMachine, virtualMachineInterface);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenTagCommand.class), anyLong())).thenReturn(applyTungstenTagAnswer);
         when(applyTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
-        when(tungstenTag.getTag()).thenReturn(tag);
-        when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
-        when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
-        when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
-        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterface));
 
         assertNotNull(tungstenService.applyTungstenTag(1L, List.of("948f421c-edde-4518-a391-09299cc25dc2"), List.of("8b4637b6-5629-46de-8fb2-d0b0502bfa85")
                 , List.of("8d097a79-a38d-4db4-8a41-16f15d9c5afa"), "a329662e-1805-4a89-9b05-2b818ea35978", "d5e3f5c5-97ed-41b6-9b6f-7f696b9eddeb"
@@ -1224,7 +1219,6 @@
     @Test
     public void removeTungstenTagTest() {
         TungstenAnswer removeTungstenTagAnswer = MockTungstenAnswerFactory.get(true);
-        TungstenTag tungstenTag = mock(TungstenTag.class);
         Tag tag = mock(Tag.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
@@ -1234,12 +1228,8 @@
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenTagCommand.class), anyLong())).thenReturn(removeTungstenTagAnswer);
+        TungstenTag tungstenTag = createMockTungstenTag(tag, networkPolicy, virtualNetwork, virtualMachine, virtualMachineInterface);
         when(removeTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
-        when(tungstenTag.getTag()).thenReturn(tag);
-        when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
-        when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
-        when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
-        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterface));
 
         assertNotNull(tungstenService.removeTungstenTag(1L, List.of("948f421c-edde-4518-a391-09299cc25dc2"),
                 List.of("8b4637b6-5629-46de-8fb2-d0b0502bfa85"),
@@ -1668,4 +1658,18 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static TungstenTag createMockTungstenTag(Tag tag,
+            NetworkPolicy networkPolicy,
+            VirtualNetwork virtualNetwork,
+            VirtualMachine virtualMachine,
+            VirtualMachineInterface virtualMachineInterface) {
+        TungstenTag tungstenTag = mock(TungstenTag.class);
+        when(tungstenTag.getTag()).thenReturn(tag);
+        when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
+        when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
+        when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
+        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterface));
+        return tungstenTag;
+    }
 }
