--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -54,6 +54,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class LibvirtMigrateVolumeCommandWrapperTest {
+    @Mock
+    private PrimaryDataStoreTO spool;
 
     @Spy
     @InjectMocks
@@ -287,7 +289,6 @@
 
         KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         Mockito.doReturn(storagePoolMgr).when(libvirtComputingResource).getStoragePoolMgr();
-        PrimaryDataStoreTO spool = Mockito.mock(PrimaryDataStoreTO.class);
         Mockito.doReturn(spool).when(destVolumeObjectTO).getDataStore();
         KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
         Mockito.doReturn(pool).when(storagePoolMgr).getStoragePool(Mockito.any(), Mockito.any());
@@ -334,7 +335,6 @@
 
         KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         Mockito.doReturn(storagePoolMgr).when(libvirtComputingResource).getStoragePoolMgr();
-        PrimaryDataStoreTO spool = Mockito.mock(PrimaryDataStoreTO.class);
         Mockito.doReturn(spool).when(destVolumeObjectTO).getDataStore();
         KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
         Mockito.doReturn(pool).when(storagePoolMgr).getStoragePool(Mockito.any(), Mockito.any());
