--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -530,11 +530,7 @@
             when(callContextMock.getCallingAccount()).thenReturn(account);
             when(accountManager.finalizeOwner(nullable(Account.class), nullable(String.class), nullable(Long.class), nullable(Long.class))).thenReturn(account);
 
-            DeleteUserDataCmd cmd = Mockito.mock(DeleteUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
-            when(cmd.getId()).thenReturn(1L);
+            DeleteUserDataCmd cmd = createMockDeleteUserDataCmd();
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             Mockito.when(userData.getId()).thenReturn(1L);
@@ -556,12 +552,8 @@
             when(callContextMock.getCallingAccount()).thenReturn(account);
             when(accountManager.finalizeOwner(nullable(Account.class), nullable(String.class), nullable(Long.class), nullable(Long.class))).thenReturn(account);
 
-            DeleteUserDataCmd cmd = Mockito.mock(DeleteUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
-            when(cmd.getId()).thenReturn(1L);
-
+
+            DeleteUserDataCmd cmd = createMockDeleteUserDataCmd();
             UserDataVO userData = Mockito.mock(UserDataVO.class);
             Mockito.when(userData.getId()).thenReturn(1L);
             when(userDataDao.findById(1L)).thenReturn(userData);
@@ -583,12 +575,8 @@
             when(callContextMock.getCallingAccount()).thenReturn(account);
             when(accountManager.finalizeOwner(nullable(Account.class), nullable(String.class), nullable(Long.class), nullable(Long.class))).thenReturn(account);
 
-            DeleteUserDataCmd cmd = Mockito.mock(DeleteUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
-            when(cmd.getId()).thenReturn(1L);
-
+
+            DeleteUserDataCmd cmd = createMockDeleteUserDataCmd();
             UserDataVO userData = Mockito.mock(UserDataVO.class);
             Mockito.when(userData.getId()).thenReturn(1L);
             when(userDataDao.findById(1L)).thenReturn(userData);
@@ -2493,4 +2481,13 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static DeleteUserDataCmd createMockDeleteUserDataCmd() {
+        DeleteUserDataCmd cmd = Mockito.mock(DeleteUserDataCmd.class);
+        when(cmd.getAccountName()).thenReturn("testAccountName");
+        when(cmd.getDomainId()).thenReturn(1L);
+        when(cmd.getId()).thenReturn(1L);
+        when(cmd.getProjectId()).thenReturn(2L);
+        return cmd;
+    }
 }
