--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -130,6 +130,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenElementTest {
     @Mock
+    private TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO;
+    @Mock
     TungstenFabricUtils tungstenFabricUtils;
     @Mock
     NetworkModel networkModel;
@@ -349,7 +351,6 @@
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
-        TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer createTungstenNetworkLoadbalancerAnswer = MockTungstenAnswerFactory.get(true);
@@ -412,7 +413,6 @@
         TungstenAnswer updateTungstenLoadBalancerMemberAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateTungstenHealthMonitorAnswer = MockTungstenAnswerFactory.get(true);
         List<Pair<String, String>> pairList = List.of(new Pair<>("cookieName", "cookieValue"));
-        TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
 
         when(lbStickinessPolicy.getMethodName()).thenReturn("AppCookie");
         when(lbStickinessPolicy.getParams()).thenReturn(pairList);
