--- a/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
+++ b/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
@@ -886,12 +886,9 @@
 
         List<StoragePoolTagVO> storageTagVOListMock = getStorageTagsForTests();
 
-        StoragePoolVO storagePoolVoMock = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePoolVoMock = createMockStoragePoolVO(expected);
         Mockito.doReturn(storagePoolVoMock).when(primaryStorageDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
 
-        Mockito.doReturn(expected.getId()).when(storagePoolVoMock).getUuid();
-        Mockito.doReturn(expected.getName()).when(storagePoolVoMock).getName();
-        Mockito.doReturn(ScopeType.validateAndGetScopeType(expected.getScope())).when(storagePoolVoMock).getScope();
         Mockito.doReturn(storageTagVOListMock).when(storagePoolTagsDaoMock).findStoragePoolTags(Mockito.anyLong());
 
         Storage result = presetVariableHelperSpy.getPresetVariableValueStorage(1L, 2);
@@ -908,12 +905,9 @@
 
         List<StoragePoolTagVO> storageTagVOListMock = getStorageRuleTagsForTests();
 
-        StoragePoolVO storagePoolVoMock = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePoolVoMock = createMockStoragePoolVO(expected);
         Mockito.doReturn(storagePoolVoMock).when(primaryStorageDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
 
-        Mockito.doReturn(expected.getId()).when(storagePoolVoMock).getUuid();
-        Mockito.doReturn(expected.getName()).when(storagePoolVoMock).getName();
-        Mockito.doReturn(ScopeType.validateAndGetScopeType(expected.getScope())).when(storagePoolVoMock).getScope();
         Mockito.doReturn(storageTagVOListMock).when(storagePoolTagsDaoMock).findStoragePoolTags(Mockito.anyLong());
 
         Storage result = presetVariableHelperSpy.getPresetVariableValueStorage(1L, 2);
@@ -1447,4 +1441,12 @@
 
         assertPresetVariableIdAndName(expected, result);
     }
+
+    private static StoragePoolVO createMockStoragePoolVO(Storage expected) {
+        StoragePoolVO storagePoolVoMock = Mockito.mock(StoragePoolVO.class);
+        Mockito.doReturn(expected.getId()).when(storagePoolVoMock).getUuid();
+        Mockito.doReturn(expected.getName()).when(storagePoolVoMock).getName();
+        Mockito.doReturn(ScopeType.validateAndGetScopeType(expected.getScope())).when(storagePoolVoMock).getScope();
+        return storagePoolVoMock;
+    }
 }
