--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -1513,9 +1513,7 @@
     @Test
     public void isResetPasswordOnRestoreFromBackupTestNoCmdOverrideFallsBackToZoneSettingTrue() throws IllegalAccessException, NoSuchFieldException {
         overrideDefaultConfigValue(UserVmManager.ResetPasswordOnRestoreFromBackup, "true");
-        CreateVMFromBackupCmd cmd = mock(CreateVMFromBackupCmd.class);
-        when(cmd.getResetPassword()).thenReturn(null);
-        when(cmd.getEntityId()).thenReturn(vmId);
+        CreateVMFromBackupCmd cmd = createMockCreateVMFromBackupCmd(vmId);
         when(userVmDao.findById(vmId)).thenReturn(userVmVoMock);
         Mockito.doReturn(1L).when(userVmVoMock).getDataCenterId();
 
@@ -1527,9 +1525,7 @@
     @Test
     public void isResetPasswordOnRestoreFromBackupTestNoCmdOverrideFallsBackToZoneSettingFalse() throws IllegalAccessException, NoSuchFieldException {
         overrideDefaultConfigValue(UserVmManager.ResetPasswordOnRestoreFromBackup, "false");
-        CreateVMFromBackupCmd cmd = mock(CreateVMFromBackupCmd.class);
-        when(cmd.getResetPassword()).thenReturn(null);
-        when(cmd.getEntityId()).thenReturn(vmId);
+        CreateVMFromBackupCmd cmd = createMockCreateVMFromBackupCmd(vmId);
         when(userVmDao.findById(vmId)).thenReturn(userVmVoMock);
         Mockito.doReturn(1L).when(userVmVoMock).getDataCenterId();
 
@@ -4740,4 +4736,11 @@
         Assert.assertEquals(Long.valueOf(500L), volume.getMinIops());
         Assert.assertEquals(Long.valueOf(2000L), volume.getMaxIops());
     }
+
+    private static CreateVMFromBackupCmd createMockCreateVMFromBackupCmd(long vmId) {
+        CreateVMFromBackupCmd cmd = mock(CreateVMFromBackupCmd.class);
+        when(cmd.getEntityId()).thenReturn(vmId);
+        when(cmd.getResetPassword()).thenReturn(null);
+        return cmd;
+    }
 }
