--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -959,10 +959,9 @@
 
     @Test
     public void testDeleteGuestOsCategory_Successful() {
-        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
-        Mockito.when(deleteCmd.getId()).thenReturn(id);
+        DeleteGuestOsCategoryCmd deleteCmd = createDeleteGuestOsCategoryCmd(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList());
         Mockito.when(guestOSCategoryDao.remove(id)).thenReturn(true);
@@ -975,19 +974,17 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteGuestOsCategory_ThrowsExceptionWhenCategoryNotFound() {
-        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
         long id = 1L;
-        Mockito.when(deleteCmd.getId()).thenReturn(id);
+        DeleteGuestOsCategoryCmd deleteCmd = createDeleteGuestOsCategoryCmd(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(null);
         spy.deleteGuestOsCategory(deleteCmd);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteGuestOsCategory_ThrowsExceptionWhenGuestOsExists() {
-        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
-        Mockito.when(deleteCmd.getId()).thenReturn(id);
+        DeleteGuestOsCategoryCmd deleteCmd = createDeleteGuestOsCategoryCmd(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList(1L));
         spy.deleteGuestOsCategory(deleteCmd);
@@ -2493,4 +2490,10 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static DeleteGuestOsCategoryCmd createDeleteGuestOsCategoryCmd(long id) {
+        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
+        Mockito.when(deleteCmd.getId()).thenReturn(id);
+        return deleteCmd;
+    }
 }
