--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -77,6 +77,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ScaleIOPrimaryDataStoreDriverTest {
+    @Mock
+    private DataStore destStore;
 
     @Spy
     @InjectMocks
@@ -194,7 +196,6 @@
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
 
         when(srcData.getDataStore()).thenReturn(srcStore);
         when(destData.getDataStore()).thenReturn(destStore);
@@ -245,7 +246,6 @@
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
 
         when(srcData.getDataStore()).thenReturn(srcStore);
         when(destData.getDataStore()).thenReturn(destStore);
