--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -77,6 +77,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ScaleIOPrimaryDataStoreDriverTest {
+    @Mock
+    private VolumeObjectTO destVolTO;
 
     @Spy
     @InjectMocks
@@ -202,7 +204,6 @@
         fillSrcVolumeDetails(srcData, srcStore);
         fillDestVolumeDetails(destData, destStore);
 
-        VolumeObjectTO destVolTO = Mockito.mock(VolumeObjectTO.class);
         when(destData.getTO()).thenReturn(destVolTO);
         Host host = prepareEndpointForVolumeOperation(srcData);
         RemoteHostEndPoint ep = Mockito.mock(RemoteHostEndPoint.class);
@@ -253,7 +254,6 @@
         fillSrcVolumeDetails(srcData, srcStore);
         fillDestVolumeDetails(destData, destStore);
 
-        VolumeObjectTO destVolTO = Mockito.mock(VolumeObjectTO.class);
         when(destData.getTO()).thenReturn(destVolTO);
         Host host = prepareEndpointForVolumeOperation(srcData);
         RemoteHostEndPoint ep = Mockito.mock(RemoteHostEndPoint.class);
