--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -221,9 +221,8 @@
     @Test
     public void cancelShutdownCmdNoMsHost() {
         Mockito.when(msHostDao.findById(1L)).thenReturn(null);
-        CancelShutdownCmd cmd = mock(CancelShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
-
+
+        CancelShutdownCmd cmd = createCancelShutdownCmd();
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.cancelShutdown(cmd);
         });
@@ -234,9 +233,8 @@
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getState()).thenReturn(ManagementServerHost.State.Up);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost);
-        CancelShutdownCmd cmd = mock(CancelShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
-
+
+        CancelShutdownCmd cmd = createCancelShutdownCmd();
         Assert.assertThrows(CloudRuntimeException.class, () -> {
             spy.cancelShutdown(cmd);
         });
@@ -247,8 +245,7 @@
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Mockito.when(msHost.getState()).thenReturn(ManagementServerHost.State.ReadyToShutDown);
         Mockito.when(msHostDao.findById(1L)).thenReturn(msHost);
-        CancelShutdownCmd cmd = mock(CancelShutdownCmd.class);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        CancelShutdownCmd cmd = createCancelShutdownCmd();
         Mockito.when(clusterManagerMock.execute(anyString(), anyLong(), anyString(), anyBoolean())).thenReturn("Success");
 
         spy.cancelShutdown(cmd);
@@ -568,4 +565,10 @@
         spy.cancelMaintenance(cmd);
         Mockito.verify(clusterManagerMock, Mockito.times(1)).execute(anyString(), anyLong(), anyString(), anyBoolean());
     }
+
+    private static CancelShutdownCmd createCancelShutdownCmd() {
+        CancelShutdownCmd cmd = Mockito.mock(CancelShutdownCmd.class);
+        Mockito.when(cmd.getManagementServerId()).thenReturn(1L);
+        return cmd;
+    }
 }
