--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1142,10 +1142,9 @@
         Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
 
-        ConfigKey<Boolean> enableUserTwoFactorAuthentication = Mockito.mock(ConfigKey.class);
+        ConfigKey<Boolean> enableUserTwoFactorAuthentication = createMockConfigKey(false);
         AccountManagerImpl.enableUserTwoFactorAuthentication = enableUserTwoFactorAuthentication;
 
-        Mockito.when(enableUserTwoFactorAuthentication.valueIn(1L)).thenReturn(false);
 
         accountManagerImpl.enableTwoFactorAuthentication(userId, "totp");
     }
@@ -1161,9 +1160,8 @@
         Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
 
-        ConfigKey<Boolean> enableUserTwoFactorAuthentication = Mockito.mock(ConfigKey.class);
+        ConfigKey<Boolean> enableUserTwoFactorAuthentication = createMockConfigKey(true);
         AccountManagerImpl.enableUserTwoFactorAuthentication = enableUserTwoFactorAuthentication;
-        Mockito.when(enableUserTwoFactorAuthentication.valueIn(1L)).thenReturn(true);
 
         UserTwoFactorAuthenticator totpProvider = Mockito.mock(UserTwoFactorAuthenticator.class);
         Map<String, UserTwoFactorAuthenticator> userTwoFactorAuthenticationProvidersMap = Mockito.mock(HashMap.class);
@@ -1190,9 +1188,8 @@
         Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
 
-        ConfigKey<Boolean> enableUserTwoFactorAuthentication = Mockito.mock(ConfigKey.class);
+        ConfigKey<Boolean> enableUserTwoFactorAuthentication = createMockConfigKey(true);
         AccountManagerImpl.enableUserTwoFactorAuthentication = enableUserTwoFactorAuthentication;
-        Mockito.when(enableUserTwoFactorAuthentication.valueIn(1L)).thenReturn(true);
 
         UserTwoFactorAuthenticator totpProvider = Mockito.mock(UserTwoFactorAuthenticator.class);
         Map<String, UserTwoFactorAuthenticator> userTwoFactorAuthenticationProvidersMap = Mockito.mock(HashMap.class);
@@ -1324,7 +1321,8 @@
         Mockito.when(userAccountDao.findById(1L)).thenReturn(userAccountVO);
         Mockito.when(userDaoMock.findById(1L)).thenReturn(userVoMock);
         Mockito.when(userAccountVO.getDomainId()).thenReturn(1L);
-        Mockito.when(enableUserTwoFactorAuthenticationMock.valueIn(1L)).thenReturn(true);
+        enableUserTwoFactorAuthenticationMock = createMockConfigKey(true);
+        accountManagerImpl.enableUserTwoFactorAuthentication = enableUserTwoFactorAuthenticationMock;
         Mockito.when(cmd.getEnable()).thenReturn(true);
 
         UserTwoFactorAuthenticator staticpinProvider = Mockito.mock(UserTwoFactorAuthenticator.class);
@@ -2289,4 +2287,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static ConfigKey<Boolean> createMockConfigKey(boolean valueInReturn) {
+        ConfigKey<Boolean> configKey = Mockito.mock(ConfigKey.class);
+        Mockito.when(configKey.valueIn(1L)).thenReturn(valueInReturn);
+        return configKey;
+    }
 }
