--- a/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
@@ -384,8 +384,7 @@
         HashSet<Long> set = new HashSet<>();
         set.add(1L);
         Mockito.doReturn(set).when(resourceCleanupService).getVmIdsWithActiveVolumeSnapshots(ids);
-        BackupVO backupVO = Mockito.mock(BackupVO.class);
-        Mockito.when(backupVO.getVmId()).thenReturn(2L);
+        BackupVO backupVO = createBackupVO();
         Mockito.when(backupDao.searchByVmIds(Mockito.anyList())).thenReturn(List.of(backupVO));
         Pair<List<Long>, List<Long>> result = resourceCleanupService.getFilteredVmIdsForSnapshotsAndBackups(new ArrayList<>(ids));
         Assert.assertEquals(1, result.first().size());
@@ -412,8 +411,7 @@
         HashSet<Long> set = new HashSet<>();
         set.add(1L);
         Mockito.doReturn(set).when(resourceCleanupService).getVmIdsWithActiveVolumeSnapshots(Mockito.anyList());
-        BackupVO backupVO = Mockito.mock(BackupVO.class);
-        Mockito.when(backupVO.getVmId()).thenReturn(2L);
+        BackupVO backupVO = createBackupVO();
         Mockito.when(backupDao.searchByVmIds(Mockito.anyList())).thenReturn(List.of(backupVO));
 
         Pair<List<Long>, List<Long>> result =
@@ -685,4 +683,10 @@
         resourceCleanupService.parseDateFromConfig(
                 ResourceCleanupService.ExpungedResourcesPurgeStartTime.key(), "ABC");
     }
+
+    private static BackupVO createBackupVO() {
+        BackupVO backupVO = Mockito.mock(BackupVO.class);
+        Mockito.when(backupVO.getVmId()).thenReturn(2L);
+        return backupVO;
+    }
 }
