--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -132,6 +132,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KbossBackupProviderTest {
+    @Mock
+    private VolumeObjectTO vol1;
 
     @Mock
     private VirtualMachine virtualMachineMock;
@@ -780,7 +782,6 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
@@ -805,7 +806,6 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
@@ -838,7 +838,6 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
