--- a/engine/storage/volume/src/test/java/org/apache/cloudstack/storage/volume/VolumeServiceTest.java
+++ b/engine/storage/volume/src/test/java/org/apache/cloudstack/storage/volume/VolumeServiceTest.java
@@ -255,7 +255,7 @@
 
     @Test
     public void testCheckAndRepairVolume() throws StorageUnavailableException {
-        VolumeInfo volume = Mockito.mock(VolumeInfo.class);
+        VolumeInfo volume = createMockVolumeInfo();
         Mockito.when(volume.getPoolId()).thenReturn(1L);
         StoragePool pool = Mockito.mock(StoragePool.class);
         Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(pool);
@@ -267,7 +267,6 @@
 
         CheckAndRepairVolumePayload payload = new CheckAndRepairVolumePayload(null);
         Mockito.when(volume.getpayload()).thenReturn(payload);
-        Mockito.when(volume.getPath()).thenReturn("cbac516a-0f1f-4559-921c-1a7c6c408ccf");
         Mockito.when(volume.getPassphrase()).thenReturn(new byte[] {3, 1, 2, 3});
         Mockito.when(volume.getEncryptFormat()).thenReturn("LUKS");
 
@@ -297,7 +296,7 @@
 
     @Test
     public void testCheckAndRepairVolumeWhenFailure() throws StorageUnavailableException {
-        VolumeInfo volume = Mockito.mock(VolumeInfo.class);
+        VolumeInfo volume = createMockVolumeInfo();
         Mockito.when(volume.getPoolId()).thenReturn(1L);
         StoragePool pool = Mockito.mock(StoragePool.class);
         Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(pool);
@@ -309,7 +308,6 @@
 
         CheckAndRepairVolumePayload payload = new CheckAndRepairVolumePayload(null);
         Mockito.when(volume.getpayload()).thenReturn(payload);
-        Mockito.when(volume.getPath()).thenReturn("cbac516a-0f1f-4559-921c-1a7c6c408ccf");
         Mockito.when(volume.getPassphrase()).thenReturn(new byte[] {3, 1, 2, 3});
         Mockito.when(volume.getEncryptFormat()).thenReturn("LUKS");
 
@@ -359,4 +357,10 @@
         Mockito.when(diskOfferingDaoMock.findByIdIncludingRemoved(diskOfferingId)).thenReturn(diskOffering);
         Mockito.when(diskOfferingDaoMock.findById(diskOfferingId)).thenReturn(diskOffering);
     }
+
+    private static VolumeInfo createMockVolumeInfo() {
+        VolumeInfo volume = Mockito.mock(VolumeInfo.class);
+        Mockito.when(volume.getPath()).thenReturn("cbac516a-0f1f-4559-921c-1a7c6c408ccf");
+        return volume;
+    }
 }
