--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1261,13 +1261,11 @@
                         Mockito.mock(DataCenterGuestIpv6PrefixVO.class)));
         List<? extends DataCenterGuestIpv6Prefix> prefixes = configurationMgr.listDataCenterGuestIpv6Prefixes(cmd);
         assertEquals(1, prefixes.size());
-        ListGuestNetworkIpv6PrefixesCmd cmd1 = Mockito.mock(ListGuestNetworkIpv6PrefixesCmd.class);
-        Mockito.when(cmd1.getId()).thenReturn(null);
+        ListGuestNetworkIpv6PrefixesCmd cmd1 = createMockListGuestNetworkIpv6PrefixesCmd();
         Mockito.when(cmd1.getZoneId()).thenReturn(1L);
         prefixes = configurationMgr.listDataCenterGuestIpv6Prefixes(cmd1);
         assertEquals(2, prefixes.size());
-        ListGuestNetworkIpv6PrefixesCmd cmd2 = Mockito.mock(ListGuestNetworkIpv6PrefixesCmd.class);
-        Mockito.when(cmd2.getId()).thenReturn(null);
+        ListGuestNetworkIpv6PrefixesCmd cmd2 = createMockListGuestNetworkIpv6PrefixesCmd();
         Mockito.when(cmd2.getZoneId()).thenReturn(null);
         prefixes = configurationMgr.listDataCenterGuestIpv6Prefixes(cmd2);
         assertEquals(3, prefixes.size());
@@ -1508,4 +1506,10 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static ListGuestNetworkIpv6PrefixesCmd createMockListGuestNetworkIpv6PrefixesCmd() {
+        ListGuestNetworkIpv6PrefixesCmd cmd = Mockito.mock(ListGuestNetworkIpv6PrefixesCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(null);
+        return cmd;
+    }
 }
