--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -18,6 +18,7 @@
  */
 package org.apache.cloudstack.storage.vmsnapshot;
 
+import org.mockito.Mock;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.when;
 
@@ -103,6 +104,8 @@
 @RunWith(SpringJUnit4ClassRunner.class)
 @ContextConfiguration(loader = AnnotationConfigContextLoader.class)
 public class VMSnapshotStrategyKVMTest extends TestCase{
+    @Mock
+    private DataStore dataStore;
     List<StoragePoolVO> storage;
     @Inject
     VMSnapshotHelper vmSnapshotHelper;
@@ -160,7 +163,6 @@
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
         SnapshotInfo snapshotInfo = Mockito.mock(SnapshotInfo.class);
         SnapshotStrategy strategy = Mockito.mock(SnapshotStrategy.class);
-        DataStore dataStore = Mockito.mock(DataStore.class);
         String volUuid = UUID.randomUUID().toString();
         String vmUuid = UUID.randomUUID().toString();
         SnapshotVO snapshot = new SnapshotVO(vol.getDataCenterId(), vol.getAccountId(), vol.getDomainId(),
@@ -228,7 +230,6 @@
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
         SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
         Snapshot snap = Mockito.mock(Snapshot.class);
-        DataStore dataStore = Mockito.mock(DataStore.class);
 
         String volUuid = UUID.randomUUID().toString();
         String vmUuid = UUID.randomUUID().toString();
