--- a/server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java
@@ -88,10 +88,8 @@
         Mockito.doReturn(callerAccountMock).when(roleManagerImpl).getCurrentAccount();
         Mockito.doReturn(callerAccountRoleMock.getId()).when(callerAccountMock).getRoleId();
 
-        Mockito.when(rolePermission1Mock.getRule()).thenReturn(new Rule("api1"));
-        Mockito.when(rolePermission2Mock.getRule()).thenReturn(new Rule("api2"));
-        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission1Mock).getPermission();
-        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission2Mock).getPermission();
+        rolePermission2Mock = createRolePermission("api2");
+        rolePermission1Mock = createRolePermission("api1");
         Mockito.doReturn(RolePermissionEntity.Permission.DENY).when(rolePermission2MockWithDeniedPermission).getPermission();
 
         Mockito.doReturn(1L).when(lessPermissionsRoleMock).getId();
@@ -379,4 +377,13 @@
         Mockito.verify(roleDaoMock).listAll();
         Mockito.verify(roleManagerImpl).removeRolesIfNeeded(roles);
     }
+
+    private static RolePermission createRolePermission(String ruleName) {
+        RolePermission rolePermissionMock = Mockito.mock(RolePermission.class);
+        Mockito.when(rolePermissionMock.getRule()).thenReturn(new Rule(ruleName));
+        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW)
+                .when(rolePermissionMock)
+                .getPermission();
+        return rolePermissionMock;
+    }
 }
