--- a/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
+++ b/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
@@ -1170,8 +1170,7 @@
 
     @Test
     public void addQuotaCreditsTestValidParameters() {
-        QuotaCreditsCmd cmd = Mockito.mock(QuotaCreditsCmd.class);
-        Mockito.doReturn(10D).when(cmd).getValue();
+        QuotaCreditsCmd cmd = createQuotaCreditsCmd(10D);
         Mockito.doReturn(BigDecimal.TEN).when(quotaCreditsVoMock).getCredit();
         Mockito.doReturn(accountMock).when(accountDaoMock).findById(Mockito.anyLong());
         Mockito.doReturn(null).when(quotaBalanceDaoMock).findLaterBalanceEntry(Mockito.anyLong(), Mockito.anyLong(),
@@ -1195,8 +1194,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void addQuotaCreditsTestThrowsExceptionWhenDepositDateIsIncorrect() {
-        QuotaCreditsCmd cmd = Mockito.mock(QuotaCreditsCmd.class);
-        Mockito.doReturn(100.0).when(cmd).getValue();
+        QuotaCreditsCmd cmd = createQuotaCreditsCmd(100.0);
         Mockito.doReturn(accountMock).when(accountDaoMock).findById(Mockito.anyLong());
         Mockito.doReturn(quotaBalanceVoMock).when(quotaBalanceDaoMock).findLaterBalanceEntry(Mockito.anyLong(), Mockito.anyLong(), Mockito.any());
 
@@ -1368,4 +1366,10 @@
         Assert.assertEquals(record2.getStartDate(), secondHistory.getStartDate());
         Assert.assertEquals(record2.getEndDate(), secondHistory.getEndDate());
     }
+
+    private static QuotaCreditsCmd createQuotaCreditsCmd(double value) {
+        QuotaCreditsCmd cmd = Mockito.mock(QuotaCreditsCmd.class);
+        Mockito.doReturn(value).when(cmd).getValue();
+        return cmd;
+    }
 }
