--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -940,12 +940,9 @@
         Long poolId = 1L;
         List<String> storageAccessGroups = Arrays.asList("sag1", "sag2");
 
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(storagePool.getId()).thenReturn(poolId);
-        Mockito.when(storagePool.getScope()).thenReturn(ScopeType.CLUSTER);
+        StoragePoolVO storagePool = createStoragePool(poolId, ScopeType.CLUSTER);
         Mockito.when(storagePool.getClusterId()).thenReturn(1L);
         Mockito.when(storagePool.getPodId()).thenReturn(1L);
-        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
 
         Mockito.when(storagePoolDao.findById(poolId)).thenReturn(storagePool);
 
@@ -983,11 +980,8 @@
         Long poolId = 1L;
         List<String> storageAccessGroups = new ArrayList<>();
 
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(storagePool.getId()).thenReturn(poolId);
-        Mockito.when(storagePool.getScope()).thenReturn(ScopeType.ZONE);
-        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
-
+
+        StoragePoolVO storagePool = createStoragePool(poolId, ScopeType.ZONE);
         Mockito.when(storagePoolDao.findById(poolId)).thenReturn(storagePool);
 
         HostVO host1 = Mockito.mock(HostVO.class);
@@ -1020,12 +1014,9 @@
         Long poolId = 1L;
         List<String> storageAccessGroups = Arrays.asList("sag1", "sag2");
 
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(storagePool.getId()).thenReturn(poolId);
-        Mockito.when(storagePool.getScope()).thenReturn(ScopeType.CLUSTER);
+        StoragePoolVO storagePool = createStoragePool(poolId, ScopeType.CLUSTER);
         Mockito.when(storagePool.getClusterId()).thenReturn(1L);
         Mockito.when(storagePool.getPodId()).thenReturn(1L);
-        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
 
         Mockito.when(storagePoolDao.findById(poolId)).thenReturn(storagePool);
 
@@ -1393,4 +1384,12 @@
         Mockito.verify(hostDao).findByClusterId(clusterId, Host.Type.Routing);
         Mockito.verify(hostDao).findByPodId(podId, Host.Type.Routing);
     }
+
+    private static StoragePoolVO createStoragePool(Long poolId, ScopeType scope) {
+        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
+        Mockito.when(storagePool.getId()).thenReturn(poolId);
+        Mockito.when(storagePool.getScope()).thenReturn(scope);
+        return storagePool;
+    }
 }
