--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -132,6 +132,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KbossBackupProviderTest {
+    @Mock
+    private VolumeObjectTO vol2;
 
     @Mock
     private VirtualMachine virtualMachineMock;
@@ -781,7 +783,6 @@
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
         VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
-        VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
@@ -806,7 +807,6 @@
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
         VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
-        VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
@@ -839,7 +839,6 @@
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
         VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
-        VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
         doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
