--- a/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
+++ b/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
@@ -75,15 +75,11 @@
 
     @Test
     public void testSortDataMotionStrategies() {
-        DataMotionStrategy cantHandleStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy defaultStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy hyperStrategy = mock(DataMotionStrategy.class);
-        DataMotionStrategy highestStrategy = mock(DataMotionStrategy.class);
+        DataMotionStrategy defaultStrategy = createMockDataMotionStrategy(StrategyPriority.DEFAULT);
+        DataMotionStrategy hyperStrategy = createMockDataMotionStrategy(StrategyPriority.HYPERVISOR);
 
-        doReturn(StrategyPriority.CANT_HANDLE).when(cantHandleStrategy).canHandle(any(DataObject.class), any(DataObject.class));
-        doReturn(StrategyPriority.DEFAULT).when(defaultStrategy).canHandle(any(DataObject.class), any(DataObject.class));
-        doReturn(StrategyPriority.HYPERVISOR).when(hyperStrategy).canHandle(any(DataObject.class), any(DataObject.class));
-        doReturn(StrategyPriority.HIGHEST).when(highestStrategy).canHandle(any(DataObject.class), any(DataObject.class));
+        DataMotionStrategy cantHandleStrategy = createMockDataMotionStrategy(StrategyPriority.CANT_HANDLE);
+        DataMotionStrategy highestStrategy = createMockDataMotionStrategy(StrategyPriority.HIGHEST);
 
         List<DataMotionStrategy> strategies = new ArrayList<DataMotionStrategy>(5);
         DataMotionStrategy strategy = null;
@@ -143,4 +139,10 @@
         strategy = factory.getDataMotionStrategy(mock(Map.class), mock(Host.class), mock(Host.class));
         assertEquals("Highest strategy was not picked.", highestStrategy, strategy);
     }
+
+    private static DataMotionStrategy createMockDataMotionStrategy(StrategyPriority strategyPriority) {
+        DataMotionStrategy strategy = mock(DataMotionStrategy.class);
+        doReturn(strategyPriority).when(strategy).canHandle(any(DataObject.class), any(DataObject.class));
+        return strategy;
+    }
 }
