--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -132,6 +132,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KbossBackupProviderTest {
+    @Mock
+    private CallContext callContextMock;
 
     @Mock
     private VirtualMachine virtualMachineMock;
@@ -1817,7 +1819,6 @@
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
 
             kbossBackupProviderSpy.cleanupValidation(false, userVmVOMock, backupVoMock, List.of(dataVolume));
@@ -1841,7 +1842,7 @@
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
+            // removed local mock; replaced with global field `callContextMock`
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
 
             kbossBackupProviderSpy.cleanupValidation(true, userVmVOMock, backupVoMock, List.of(dataVolume));
@@ -1863,7 +1864,6 @@
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
 
             kbossBackupProviderSpy.cleanupValidation(true, userVmVOMock, backupVoMock, List.of(dataVolume));
@@ -1891,7 +1891,6 @@
         doReturn(volumeVoMock).when(volumeApiServiceMock).destroyVolume(Mockito.eq(52L), any(), Mockito.eq(true), Mockito.eq(true), Mockito.isNull());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
 
             kbossBackupProviderSpy.cleanupValidation(true, userVmVOMock, backupVoMock, List.of(rootVolume, dataVolume));
