--- a/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedNASStrategyTest.java
+++ b/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedNASStrategyTest.java
@@ -79,6 +79,8 @@
 @ExtendWith(MockitoExtension.class)
 @MockitoSettings(strictness = Strictness.LENIENT)
 public class UnifiedNASStrategyTest {
+    @Mock
+    private VolumeInfo volumeInfo;
 
     @Mock
     private NASFeignClient nasFeignClient;
@@ -517,7 +519,6 @@
     @Test
     public void testDeleteCloudStackVolume_Success() throws Exception {
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
-        VolumeInfo volumeInfo = mock(VolumeInfo.class);
         EndPoint endpoint = mock(EndPoint.class);
         Answer answer = mock(Answer.class);
 
@@ -538,7 +539,6 @@
     @Test
     public void testDeleteCloudStackVolume_EndpointNotFound() {
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
-        VolumeInfo volumeInfo = mock(VolumeInfo.class);
 
         when(cloudStackVolume.getVolumeInfo()).thenReturn(volumeInfo);
         when(epSelector.select(volumeInfo)).thenReturn(null);
@@ -552,7 +552,6 @@
     @Test
     public void testDeleteCloudStackVolume_AnswerResultFalse() throws Exception {
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
-        VolumeInfo volumeInfo = mock(VolumeInfo.class);
         EndPoint endpoint = mock(EndPoint.class);
         Answer answer = mock(Answer.class);
 
@@ -571,7 +570,6 @@
     @Test
     public void testDeleteCloudStackVolume_AnswerNull() throws Exception {
         CloudStackVolume cloudStackVolume = mock(CloudStackVolume.class);
-        VolumeInfo volumeInfo = mock(VolumeInfo.class);
         EndPoint endpoint = mock(EndPoint.class);
 
         when(cloudStackVolume.getVolumeInfo()).thenReturn(volumeInfo);
