--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -84,9 +84,8 @@
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
 
-        AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+        AffinityGroupVO affinityGroupVO = MockAffinityGroupVO.createMockAffinityGroupVO(affinityGroupId);
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
-        when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
@@ -116,9 +115,8 @@
         plan.adjustHostPriority(host3Id, DeploymentPlan.HostPriorityAdjustment.LOWER);
         ExcludeList avoid = new ExcludeList();
 
-        AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+        AffinityGroupVO affinityGroupVO = MockAffinityGroupVO.createMockAffinityGroupVO(affinityGroupId);
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
-        when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id, vm3Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
@@ -151,9 +149,8 @@
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
 
-        AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+        AffinityGroupVO affinityGroupVO = MockAffinityGroupVO.createMockAffinityGroupVO(affinityGroupId);
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
-        when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
         VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
@@ -170,4 +167,16 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(1), plan.getHostPriorities().get(host2Id));
     }
+
+    public final class MockAffinityGroupVO {
+
+        private MockAffinityGroupVO() {
+        }
+
+        public static AffinityGroupVO createMockAffinityGroupVO(long affinityGroupId) {
+            AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+            when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
+            return affinityGroupVO;
+        }
+    }
 }
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -83,9 +83,8 @@
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
 
-        AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+        AffinityGroupVO affinityGroupVO = MockAffinityGroupVO.createMockAffinityGroupVO(affinityGroupId);
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
-        when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
@@ -115,9 +114,8 @@
         plan.adjustHostPriority(host3Id, DeploymentPlan.HostPriorityAdjustment.HIGHER);
         ExcludeList avoid = new ExcludeList();
 
-        AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+        AffinityGroupVO affinityGroupVO = createMockAffinityGroupVO(affinityGroupId);
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
-        when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id, vm3Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
@@ -150,9 +148,8 @@
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
 
-        AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+        AffinityGroupVO affinityGroupVO = MockAffinityGroupVO.createMockAffinityGroupVO(affinityGroupId);
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
-        when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
         when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
         VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
@@ -169,4 +166,16 @@
         Assert.assertNotNull(plan.getHostPriorities().get(host2Id));
         Assert.assertEquals(Integer.valueOf(-1), plan.getHostPriorities().get(host2Id));
     }
+
+    public final class MockAffinityGroupVO {
+
+        private MockAffinityGroupVO() {
+        }
+
+        public static AffinityGroupVO createMockAffinityGroupVO(long affinityGroupId) {
+            AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+            when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
+            return affinityGroupVO;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/cloudstack/affinity/AffinityGroupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/affinity/AffinityGroupServiceImplTest.java
@@ -213,8 +213,7 @@
 
     @Test
     public void shouldDeleteDomainLevelAffinityGroup() {
-        AffinityGroupVO mockGroup = Mockito.mock(AffinityGroupVO.class);
-        when(mockGroup.getId()).thenReturn(2L);
+        AffinityGroupVO mockGroup = MockAffinityGroupVO.createMockAffinityGroupVO(2L);
         when(_affinityGroupDao.findById(ArgumentMatchers.anyLong())).thenReturn(mockGroup);
         _affinityService.deleteAffinityGroup(2L, null, null, DOMAIN_ID, null);
         Mockito.verify(_affinityGroupDao).expunge(2L);
@@ -222,8 +221,7 @@
 
     @Test
     public void shouldDeleteAffintyGroupById() {
-        AffinityGroupVO mockGroup = Mockito.mock(AffinityGroupVO.class);
-        when(mockGroup.getId()).thenReturn(1L);
+        AffinityGroupVO mockGroup = MockAffinityGroupVO.createMockAffinityGroupVO(1L);
         when(_affinityGroupDao.findById(ArgumentMatchers.anyLong())).thenReturn(mockGroup);
         _affinityService.deleteAffinityGroup(1L, ACCOUNT_NAME, null, DOMAIN_ID, null);
         Mockito.verify(_affinityGroupDao).expunge(1L);
@@ -375,4 +373,16 @@
             }
         }
     }
+
+    public final class MockAffinityGroupVO {
+
+        private MockAffinityGroupVO() {
+        }
+
+        public static AffinityGroupVO createMockAffinityGroupVO(long affinityGroupId) {
+            AffinityGroupVO affinityGroupVO = Mockito.mock(AffinityGroupVO.class);
+            when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
+            return affinityGroupVO;
+        }
+    }
 }
