--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -150,6 +150,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class BackupManagerTest {
+    @Mock
+    private BackupOfferingVO offering;
     @Spy
     @InjectMocks
     BackupManagerImpl backupManager = new BackupManagerImpl();
@@ -1598,7 +1600,6 @@
         when(vmInstanceDao.findByIdIncludingRemoved(vmId)).thenReturn(vm);
         when(backupDao.listByVmIdAndOffering(zoneId, vmId, backupOfferingId)).thenReturn(new ArrayList<>());
 
-        BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(offering);
 
         when(backupProvider.deleteBackup(backup, false)).thenReturn(true);
@@ -2291,7 +2292,6 @@
     @Test
     public void getBackupOfferingDomainsTestReturnsDomains() {
         Long offeringId = 1L;
-        BackupOfferingVO offering = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.findById(offeringId)).thenReturn(offering);
         when(backupOfferingDetailsDao.findDomainIds(offeringId)).thenReturn(List.of(10L, 20L));
 
@@ -2330,7 +2330,6 @@
         when(domainHelper.filterChildSubDomains(List.of(domainId))).thenReturn(new ArrayList<>(List.of(domainId)));
         when(backupOfferingDetailsDao.findDomainIds(id)).thenReturn(new ArrayList<>());
 
-        BackupOfferingVO offering = Mockito.mock(BackupOfferingVO.class);
         BackupOfferingVO offeringUpdate = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.findById(id)).thenReturn(offering);
         when(backupOfferingDao.createForUpdate(id)).thenReturn(offeringUpdate);
@@ -2359,9 +2358,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(offering);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2403,9 +2401,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(offering);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2448,9 +2445,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(offering);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2494,9 +2490,7 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2538,9 +2532,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(offering);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2587,9 +2580,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(offering);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
