--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -737,15 +737,13 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testRegisterTemplateWithSystemTemplateTypeByUser() {
-        RegisterVnfTemplateCmd cmd = Mockito.mock(RegisterVnfTemplateCmd.class);
-        when(cmd.getTemplateType()).thenReturn(Storage.TemplateType.SYSTEM.toString());
+        RegisterVnfTemplateCmd cmd = createMockRegisterVnfTemplateCmd();
         Storage.TemplateType type = templateManager.validateTemplateType(cmd, false, true, Hypervisor.HypervisorType.KVM);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void testRegisterVnfTemplateWithTemplateType() {
-        RegisterVnfTemplateCmd cmd = Mockito.mock(RegisterVnfTemplateCmd.class);
-        when(cmd.getTemplateType()).thenReturn(Storage.TemplateType.SYSTEM.toString());
+        RegisterVnfTemplateCmd cmd = createMockRegisterVnfTemplateCmd();
         Storage.TemplateType type = templateManager.validateTemplateType(cmd, false, true, Hypervisor.HypervisorType.KVM);
         Assert.assertEquals(Storage.TemplateType.VNF, type);
     }
@@ -1312,4 +1310,10 @@
             }
         }
     }
+
+    private static RegisterVnfTemplateCmd createMockRegisterVnfTemplateCmd() {
+        RegisterVnfTemplateCmd cmd = Mockito.mock(RegisterVnfTemplateCmd.class);
+        when(cmd.getTemplateType()).thenReturn(Storage.TemplateType.SYSTEM.toString());
+        return cmd;
+    }
 }
