--- a/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
@@ -997,9 +997,8 @@
         BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
         when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
-
+
+        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = createMockBgpPeerNetworkMapVO();
         when(bgpPeerNetworkMap2.getId()).thenReturn(bgpPeerNetworkMapId2);
 
         when(bgpPeerNetworkMapDao.listByNetworkId(networkId)).thenReturn(Arrays.asList(bgpPeerNetworkMap1, bgpPeerNetworkMap2))
@@ -1030,9 +1029,8 @@
         BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
         when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
-
+
+        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = createMockBgpPeerNetworkMapVO();
         when(bgpPeerNetworkMap3.getId()).thenReturn(bgpPeerNetworkMapId3);
 
         when(bgpPeerNetworkMapDao.listByNetworkId(networkId)).thenReturn(Arrays.asList(bgpPeerNetworkMap1, bgpPeerNetworkMap2))
@@ -1142,9 +1140,8 @@
         BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
         when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
-
+
+        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = createMockBgpPeerNetworkMapVO();
         when(bgpPeerNetworkMap2.getId()).thenReturn(bgpPeerNetworkMapId2);
 
         when(bgpPeerNetworkMapDao.listByVpcId(vpcId)).thenReturn(Arrays.asList(bgpPeerNetworkMap1, bgpPeerNetworkMap2))
@@ -1175,8 +1172,7 @@
         BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
         when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap3 = createMockBgpPeerNetworkMapVO();
 
         when(bgpPeerNetworkMap3.getId()).thenReturn(bgpPeerNetworkMapId3);
 
@@ -1214,4 +1210,10 @@
         verify(ipv4GuestSubnetNetworkMapDao).deleteByParentId(zoneSubnetId);
         verify(dataCenterIpv4GuestSubnetDao).remove(zoneSubnetId);
     }
+
+    private static BgpPeerNetworkMapVO createMockBgpPeerNetworkMapVO() {
+        BgpPeerNetworkMapVO bgpPeerNetworkMap = Mockito.mock(BgpPeerNetworkMapVO.class);
+        when(bgpPeerNetworkMap.getState()).thenReturn(BgpPeer.State.Add);
+        return bgpPeerNetworkMap;
+    }
 }
