--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -335,11 +335,9 @@
 
     @Test
     public void updateVolumeAfterCopyVolumeLiveMigrate() {
-        VolumeInfo srcData = Mockito.mock(VolumeInfo.class);
-        VolumeInfo destData = Mockito.mock(VolumeInfo.class);
-
-        when(srcData.getId()).thenReturn(1L);
-        when(destData.getId()).thenReturn(1L);
+        VolumeInfo destData = createMockVolumeInfo(1L);
+
+        VolumeInfo srcData = createMockVolumeInfo(1L);
 
         VolumeVO volume = new VolumeVO("root", 1L, 1L, 1L, 1L, 1L, "root", "root", Storage.ProvisioningType.THIN, 1, null, null, "root", Volume.Type.ROOT);
         volume.setPoolId(2L);
@@ -353,11 +351,9 @@
 
     @Test
     public void updateVolumeAfterCopyVolumeOffline() {
-        VolumeInfo srcData = Mockito.mock(VolumeInfo.class);
-        VolumeInfo destData = Mockito.mock(VolumeInfo.class);
-
-        when(srcData.getId()).thenReturn(1L);
-        when(destData.getId()).thenReturn(2L);
+        VolumeInfo destData = createMockVolumeInfo(2L);
+
+        VolumeInfo srcData = createMockVolumeInfo(1L);
 
         VolumeVO volume = new VolumeVO("root", 1L, 1L, 1L, 1L, 1L, "root", "root", Storage.ProvisioningType.THIN, 1, null, null, "root", Volume.Type.ROOT);
         when(volumeDao.findById(1L)).thenReturn(volume);
@@ -374,12 +370,11 @@
     public void revertBlockCopyVolumeOperationsOnDeleteSuccess() throws Exception{
         //Either destination volume delete success or failure, DB operations should get revert
 
-        VolumeInfo srcData = Mockito.mock(VolumeInfo.class);
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
         Host host = Mockito.mock(Host.class);
         String destVolumePath = "01b332b300000007:vol-11-b9e2-10ee";
 
-        when(srcData.getId()).thenReturn(1L);
+        VolumeInfo srcData = createMockVolumeInfo(1L);
         when(srcData.getPoolId()).thenReturn(1L);
         when(destData.getId()).thenReturn(1L);
 
@@ -410,13 +405,12 @@
     public void revertBlockCopyVolumeOperationsOnDeleteFailure() throws Exception{
         //Either destination volume delete success or failure, DB operations should get revert
 
-        VolumeInfo srcData = Mockito.mock(VolumeInfo.class);
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
         Host host = Mockito.mock(Host.class);
         String srcVolumePath = "bec0ba7700000007:vol-11-6aef-10ee";
         String destVolumePath = "01b332b300000007:vol-11-b9e2-10ee";
 
-        when(srcData.getId()).thenReturn(1L);
+        VolumeInfo srcData = createMockVolumeInfo(1L);
         when(srcData.getPoolId()).thenReturn(1L);
         when(destData.getId()).thenReturn(1L);
 
@@ -597,4 +591,10 @@
                         16L * (1024 * 1024 * 1024),
                         true));
     }
+
+    private static VolumeInfo createMockVolumeInfo(long id) {
+        VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
+        when(volumeInfo.getId()).thenReturn(id);
+        return volumeInfo;
+    }
 }
