--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -365,16 +365,7 @@
     public void setParametersTestWhenStateIsFreeAndSystemVmPublicIsTrue() throws IllegalArgumentException, IllegalAccessException, NoSuchFieldException, SecurityException {
         overrideDefaultConfigValue(ipAddressManagerImpl.SystemVmPublicIpReservationModeStrictness, "_defaultValue", "true");
 
-        ListPublicIpAddressesCmd cmd = Mockito.mock(ListPublicIpAddressesCmd.class);
-        Mockito.when(cmd.getNetworkId()).thenReturn(10L);
-        Mockito.when(cmd.getZoneId()).thenReturn(null);
-        Mockito.when(cmd.getIpAddress()).thenReturn(null);
-        Mockito.when(cmd.getPhysicalNetworkId()).thenReturn(null);
-        Mockito.when(cmd.getVlanId()).thenReturn(null);
-        Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.isSourceNat()).thenReturn(null);
-        Mockito.when(cmd.isStaticNat()).thenReturn(null);
-        Mockito.when(cmd.getTags()).thenReturn(null);
+        ListPublicIpAddressesCmd cmd = createMockListPublicIpAddressesCmd();
         List<IpAddress.State> states = Collections.singletonList(IpAddress.State.Free);
         spy.setParameters(sc, cmd, VlanType.VirtualNetwork, Boolean.FALSE, states);
 
@@ -388,16 +379,7 @@
     @Test
     public void setParametersTestWhenStateIsFreeAndSystemVmPublicIsFalse() throws NoSuchFieldException, IllegalAccessException {
         overrideDefaultConfigValue(ipAddressManagerImpl.SystemVmPublicIpReservationModeStrictness, "_defaultValue", "false");
-        ListPublicIpAddressesCmd cmd = Mockito.mock(ListPublicIpAddressesCmd.class);
-        Mockito.when(cmd.getNetworkId()).thenReturn(10L);
-        Mockito.when(cmd.getZoneId()).thenReturn(null);
-        Mockito.when(cmd.getIpAddress()).thenReturn(null);
-        Mockito.when(cmd.getPhysicalNetworkId()).thenReturn(null);
-        Mockito.when(cmd.getVlanId()).thenReturn(null);
-        Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.isSourceNat()).thenReturn(null);
-        Mockito.when(cmd.isStaticNat()).thenReturn(null);
-        Mockito.when(cmd.getTags()).thenReturn(null);
+        ListPublicIpAddressesCmd cmd = createMockListPublicIpAddressesCmd();
         List<IpAddress.State> states = Collections.singletonList(IpAddress.State.Free);
         spy.setParameters(sc, cmd, VlanType.VirtualNetwork, Boolean.FALSE, states);
 
@@ -411,16 +393,7 @@
     @Test
     public void setParametersTestWhenStateIsNullAndSystemVmPublicIsFalse() throws NoSuchFieldException, IllegalAccessException {
         overrideDefaultConfigValue(ipAddressManagerImpl.SystemVmPublicIpReservationModeStrictness, "_defaultValue", "false");
-        ListPublicIpAddressesCmd cmd = Mockito.mock(ListPublicIpAddressesCmd.class);
-        Mockito.when(cmd.getNetworkId()).thenReturn(10L);
-        Mockito.when(cmd.getZoneId()).thenReturn(null);
-        Mockito.when(cmd.getIpAddress()).thenReturn(null);
-        Mockito.when(cmd.getPhysicalNetworkId()).thenReturn(null);
-        Mockito.when(cmd.getVlanId()).thenReturn(null);
-        Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.isSourceNat()).thenReturn(null);
-        Mockito.when(cmd.isStaticNat()).thenReturn(null);
-        Mockito.when(cmd.getTags()).thenReturn(null);
+        ListPublicIpAddressesCmd cmd = createMockListPublicIpAddressesCmd();
         spy.setParameters(sc, cmd, VlanType.VirtualNetwork, Boolean.TRUE, Collections.emptyList());
 
         Mockito.verify(sc, Mockito.times(1)).setJoinParameters("vlanSearch", "vlanType", VlanType.VirtualNetwork);
@@ -433,16 +406,7 @@
     @Test
     public void setParametersTestWhenStateIsNullAndSystemVmPublicIsTrue() throws NoSuchFieldException, IllegalAccessException {
         overrideDefaultConfigValue(ipAddressManagerImpl.SystemVmPublicIpReservationModeStrictness, "_defaultValue", "true");
-        ListPublicIpAddressesCmd cmd = Mockito.mock(ListPublicIpAddressesCmd.class);
-        Mockito.when(cmd.getNetworkId()).thenReturn(10L);
-        Mockito.when(cmd.getZoneId()).thenReturn(null);
-        Mockito.when(cmd.getIpAddress()).thenReturn(null);
-        Mockito.when(cmd.getPhysicalNetworkId()).thenReturn(null);
-        Mockito.when(cmd.getVlanId()).thenReturn(null);
-        Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.isSourceNat()).thenReturn(null);
-        Mockito.when(cmd.isStaticNat()).thenReturn(null);
-        Mockito.when(cmd.getTags()).thenReturn(null);
+        ListPublicIpAddressesCmd cmd = createMockListPublicIpAddressesCmd();
         spy.setParameters(sc, cmd, VlanType.VirtualNetwork, Boolean.TRUE, Collections.emptyList());
 
         Mockito.verify(sc, Mockito.times(1)).setJoinParameters("vlanSearch", "vlanType", VlanType.VirtualNetwork);
@@ -2493,4 +2457,18 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static ListPublicIpAddressesCmd createMockListPublicIpAddressesCmd() {
+        ListPublicIpAddressesCmd cmd = Mockito.mock(ListPublicIpAddressesCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(null);
+        Mockito.when(cmd.getIpAddress()).thenReturn(null);
+        Mockito.when(cmd.getNetworkId()).thenReturn(10L);
+        Mockito.when(cmd.getPhysicalNetworkId()).thenReturn(null);
+        Mockito.when(cmd.getTags()).thenReturn(null);
+        Mockito.when(cmd.getVlanId()).thenReturn(null);
+        Mockito.when(cmd.getZoneId()).thenReturn(null);
+        Mockito.when(cmd.isSourceNat()).thenReturn(null);
+        Mockito.when(cmd.isStaticNat()).thenReturn(null);
+        return cmd;
+    }
 }
