--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -1206,9 +1206,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void executeWebhookDeliveryThrowsExceptionForInvalidWebhookId() {
-        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
-        Mockito.when(cmd.getWebhookId()).thenReturn(1L);
-
+
+        ExecuteWebhookDeliveryCmd cmd = createExecuteWebhookDeliveryCmd();
         webhookApiServiceImpl.executeWebhookDelivery(cmd);
     }
 
@@ -1238,13 +1237,12 @@
 
     @Test
     public void executeWebhookDeliveryExecutesSuccessfullyForValidWebhookId() {
-        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
+        ExecuteWebhookDeliveryCmd cmd = createExecuteWebhookDeliveryCmd();
         WebhookVO webhook = Mockito.mock(WebhookVO.class);
         WebhookDelivery webhookDelivery = Mockito.mock(WebhookDelivery.class);
         WebhookDeliveryResponse response = Mockito.mock(WebhookDeliveryResponse.class);
 
         Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.getWebhookId()).thenReturn(1L);
         Mockito.when(webhookDao.findById(1L)).thenReturn(webhook);
         Mockito.when(webhookService.executeWebhookDelivery(null, webhook, null, null)).thenReturn(webhookDelivery);
         Mockito.when(webhookDelivery.getId()).thenReturn(WebhookDelivery.ID_DUMMY);
@@ -1491,4 +1489,10 @@
         Assert.assertNotNull(result);
         Assert.assertEquals(managementServerHostVO, result);
     }
+
+    private static ExecuteWebhookDeliveryCmd createExecuteWebhookDeliveryCmd() {
+        ExecuteWebhookDeliveryCmd cmd = Mockito.mock(ExecuteWebhookDeliveryCmd.class);
+        Mockito.when(cmd.getWebhookId()).thenReturn(1L);
+        return cmd;
+    }
 }
