--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/config/UpdateCfgCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/config/UpdateCfgCmdTest.java
@@ -50,9 +50,7 @@
     @Test
     public void setResponseValueSetsEncryptedValueWhenConfigurationIsEncrypted() {
         ConfigurationResponse response = new ConfigurationResponse();
-        Configuration cfg = Mockito.mock(Configuration.class);
-        Mockito.when(cfg.isEncrypted()).thenReturn(true);
-        Mockito.when(cfg.getValue()).thenReturn("testValue");
+        Configuration cfg = createMockConfiguration(true);
         Mockito.when(DBEncryptionUtil.encrypt("testValue")).thenReturn("encryptedValue");
         updateCfgCmd.setResponseValue(response, cfg);
         Assert.assertEquals("encryptedValue", response.getValue());
@@ -61,9 +59,7 @@
     @Test
     public void setResponseValueSetsPlainValueWhenConfigurationIsNotEncrypted() {
         ConfigurationResponse response = new ConfigurationResponse();
-        Configuration cfg = Mockito.mock(Configuration.class);
-        Mockito.when(cfg.isEncrypted()).thenReturn(false);
-        Mockito.when(cfg.getValue()).thenReturn("testValue");
+        Configuration cfg = createMockConfiguration(false);
         updateCfgCmd.setResponseValue(response, cfg);
         Assert.assertEquals("testValue", response.getValue());
     }
@@ -78,4 +74,11 @@
         Assert.assertNull(response.getValue());
     }
 
+
+    private static Configuration createMockConfiguration(boolean encrypted) {
+        Configuration cfg = Mockito.mock(Configuration.class);
+        Mockito.when(cfg.isEncrypted()).thenReturn(encrypted);
+        Mockito.when(cfg.getValue()).thenReturn("testValue");
+        return cfg;
+    }
 }
