--- a/api/src/test/java/org/apache/cloudstack/api/command/test/RegionCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/RegionCmdTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.api.command.test;
 
+import org.mockito.Mock;
 import static org.mockito.ArgumentMatchers.anyInt;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.ArgumentMatchers.isNull;
@@ -35,6 +36,8 @@
 import junit.framework.TestCase;
 
 public class RegionCmdTest extends TestCase {
+    @Mock
+    private Region region;
 
     private AddRegionCmd addRegionCmd;
     private ResponseGenerator responseGenerator;
@@ -66,7 +69,6 @@
 
         RegionService regionService = Mockito.mock(RegionService.class);
 
-        Region region = Mockito.mock(Region.class);
 
         Mockito.when(regionService.addRegion(anyInt(), anyString(), isNull())).thenReturn(region);
 
@@ -87,7 +89,6 @@
 
         RegionService regionService = Mockito.mock(RegionService.class);
 
-        Region region = Mockito.mock(Region.class);
 
         Mockito.when(regionService.addRegion(anyInt(), anyString(), isNull())).thenReturn(null);
 
