--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -71,6 +71,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class CommandSetupHelperTest {
+    @Mock
+    private BgpPeerVO bgpPeer2;
 
     @InjectMocks
     protected CommandSetupHelper commandSetupHelper = new CommandSetupHelper();
@@ -211,7 +213,6 @@
     @Test
     public void testCreateBgpPeersCommandsForNetwork() {
         BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
-        BgpPeerVO bgpPeer2 = Mockito.mock(BgpPeerVO.class);
         List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2);
         Commands cmds = new Commands(Command.OnError.Stop);
         VirtualRouter router = Mockito.mock(VirtualRouter.class);
@@ -239,7 +240,6 @@
     @Test
     public void testCreateBgpPeersCommandsForVpc() {
         BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
-        BgpPeerVO bgpPeer2 = Mockito.mock(BgpPeerVO.class);
         List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2);
         Commands cmds = new Commands(Command.OnError.Stop);
         VirtualRouter router = Mockito.mock(VirtualRouter.class);
