--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -132,6 +132,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class KbossBackupProviderTest {
+    @Mock
+    private InternalBackupDataStoreVO parentDelta;
 
     @Mock
     private VirtualMachine virtualMachineMock;
@@ -1239,7 +1241,6 @@
         InternalBackupJoinVO parentVo = Mockito.mock(InternalBackupJoinVO.class);
         doReturn(parentVo).when(internalBackupJoinDaoMock).findById(parentId);
         doReturn(List.of(internalBackupDataStoreVoMock)).when(internalBackupDataStoreDaoMock).listByBackupId(backupId);
-        InternalBackupDataStoreVO parentDelta = mock(InternalBackupDataStoreVO.class);
         doReturn(List.of(parentDelta)).when(internalBackupDataStoreDaoMock).listByBackupId(parentId);
         doReturn(dataStoreMock).when(dataStoreManagerMock).getDataStore(anyLong(), any());
         doReturn(hostVOMock).when(hostDaoMock).findById(anyLong());
@@ -1266,7 +1267,6 @@
         InternalBackupJoinVO parentVo = Mockito.mock(InternalBackupJoinVO.class);
         doReturn(parentVo).when(internalBackupJoinDaoMock).findById(parentId);
         doReturn(List.of(internalBackupDataStoreVoMock)).when(internalBackupDataStoreDaoMock).listByBackupId(backupId);
-        InternalBackupDataStoreVO parentDelta = mock(InternalBackupDataStoreVO.class);
         doReturn(List.of(parentDelta)).when(internalBackupDataStoreDaoMock).listByBackupId(parentId);
         doReturn(dataStoreMock).when(dataStoreManagerMock).getDataStore(anyLong(), any());
         doReturn(hostVOMock).when(hostDaoMock).findById(anyLong());
