--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.network.tungsten.vrouter;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
@@ -41,6 +42,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class VRouterApiConnectorImplTest {
+    @Mock
+    private Gateway gateway2;
     VRouterApiConnector vRouterApiConnector;
 
     MockedStatic<EntityUtils> entityUtilsMocked;
@@ -177,7 +180,6 @@
     @Test
     public void addGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
@@ -193,7 +195,6 @@
     @Test
     public void addGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
@@ -205,7 +206,6 @@
     @Test
     public void deleteGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
@@ -221,7 +221,6 @@
     @Test
     public void deleteGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
