--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -150,6 +150,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class BackupManagerTest {
+    @Mock
+    private StoragePoolVO pool;
     @Spy
     @InjectMocks
     BackupManagerImpl backupManager = new BackupManagerImpl();
@@ -1370,7 +1372,6 @@
         when(rootVolume.getPoolId()).thenReturn(poolId);
         HostVO host = mock(HostVO.class);
         when(hostDao.findById(hostId)).thenReturn(host);
-        StoragePoolVO pool = mock(StoragePoolVO.class);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
         when(rootVolume.getPoolId()).thenReturn(poolId);
@@ -1426,7 +1427,6 @@
         when(rootVolume.getPoolId()).thenReturn(poolId);
         HostVO host = mock(HostVO.class);
         when(hostDao.findById(hostId)).thenReturn(host);
-        StoragePoolVO pool = mock(StoragePoolVO.class);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
         when(rootVolume.getPoolId()).thenReturn(poolId);
