--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/AddObjectStoragePoolCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/AddObjectStoragePoolCmdTest.java
@@ -91,11 +91,28 @@
     public void testAddObjectStore() throws DiscoveryException {
         Mockito.doReturn(objectStore).when(storageService).discoverObjectStore(Mockito.anyString(),
                 Mockito.anyString(), Mockito.anyLong(), Mockito.anyString(), any());
-        ObjectStoreResponse objectStoreResponse = new ObjectStoreResponse();
-        Mockito.doReturn(objectStoreResponse).when(responseGenerator).createObjectStoreResponse(any());
+        responseGenerator = MockResponseGenerator.createMockResponseGenerator();
         addObjectStoragePoolCmdSpy.execute();
 
         Mockito.verify(storageService, Mockito.times(1))
                 .discoverObjectStore(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any());
     }
+
+    public final class MockResponseGenerator {
+        private MockResponseGenerator() {
+        }
+
+        public static org.apache.cloudstack.api.ResponseGenerator createMockResponseGenerator() {
+            org.apache.cloudstack.api.ResponseGenerator responseGenerator =
+                    Mockito.mock(org.apache.cloudstack.api.ResponseGenerator.class);
+            org.apache.cloudstack.api.response.ObjectStoreResponse objectStoreResponse =
+                    new org.apache.cloudstack.api.response.ObjectStoreResponse();
+
+            Mockito.doReturn(objectStoreResponse)
+                    .when(responseGenerator)
+                    .createObjectStoreResponse(any());
+
+            return responseGenerator;
+        }
+    }
 }
--- a/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/UpdateObjectStoragePoolCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/admin/storage/UpdateObjectStoragePoolCmdTest.java
@@ -76,11 +76,28 @@
     @Test
     public void testUpdateObjectStore() {
         Mockito.doReturn(objectStore).when(storageService).updateObjectStore(1L, updateObjectStoragePoolCmd);
-        ObjectStoreResponse objectStoreResponse = new ObjectStoreResponse();
-        Mockito.doReturn(objectStoreResponse).when(responseGenerator).createObjectStoreResponse(any());
+        responseGenerator = MockResponseGenerator.createMockResponseGenerator();
         updateObjectStoragePoolCmd.execute();
         Mockito.verify(storageService, Mockito.times(1))
                 .updateObjectStore(1L, updateObjectStoragePoolCmd);
     }
 
+
+    public final class MockResponseGenerator {
+        private MockResponseGenerator() {
+        }
+
+        public static org.apache.cloudstack.api.ResponseGenerator createMockResponseGenerator() {
+            org.apache.cloudstack.api.ResponseGenerator responseGenerator =
+                    Mockito.mock(org.apache.cloudstack.api.ResponseGenerator.class);
+            org.apache.cloudstack.api.response.ObjectStoreResponse objectStoreResponse =
+                    new org.apache.cloudstack.api.response.ObjectStoreResponse();
+
+            Mockito.doReturn(objectStoreResponse)
+                    .when(responseGenerator)
+                    .createObjectStoreResponse(any());
+
+            return responseGenerator;
+        }
+    }
 }
