--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -833,13 +833,9 @@
 
     @Test
     public void updateHostsExtensions_withHostResponses_setsExtension() {
-        HostResponse host1 = mock(HostResponse.class);
-        HostResponse host2 = mock(HostResponse.class);
-
-        when(host1.getClusterInternalId()).thenReturn(1L);
-        when(host1.getHypervisor()).thenReturn(Hypervisor.HypervisorType.External.name());
-        when(host2.getClusterInternalId()).thenReturn(2L);
-        when(host2.getHypervisor()).thenReturn(Hypervisor.HypervisorType.External.name());
+        HostResponse host2 = createMockHostResponse(2L);
+
+        HostResponse host1 = createMockHostResponse(1L);
 
         Extension ext1 = mock(Extension.class);
         when(ext1.getUuid()).thenReturn("a");
@@ -891,4 +887,11 @@
         assertEquals(1, hostTags.size());
         assertTrue(hostTags.contains(tag1));
     }
+
+    private static HostResponse createMockHostResponse(long clusterInternalId) {
+        HostResponse hostResponse = Mockito.mock(HostResponse.class);
+        when(hostResponse.getClusterInternalId()).thenReturn(clusterInternalId);
+        when(hostResponse.getHypervisor()).thenReturn(Hypervisor.HypervisorType.External.name());
+        return hostResponse;
+    }
 }
