--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -2024,9 +2024,8 @@
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
 
-        BackupProvider backupProvider = mock(BackupProvider.class);
-        when(backupProvider.crossZoneInstanceCreationEnabled(offering)).thenReturn(true);
-
+
+        BackupProvider backupProvider = createMockBackupProvider(offering, true);
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(offering);
         when(backupManager.getBackupProvider(providerName)).thenReturn(backupProvider);
@@ -2052,9 +2051,8 @@
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
 
-        BackupProvider backupProvider = mock(BackupProvider.class);
-        when(backupProvider.crossZoneInstanceCreationEnabled(offering)).thenReturn(false);
-
+
+        BackupProvider backupProvider = createMockBackupProvider(offering, false);
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(offering);
         when(backupManager.getBackupProvider(providerName)).thenReturn(backupProvider);
@@ -2874,4 +2872,10 @@
         verify(backupOfferingDao).persist(any());
         verify(backupOfferingDetailsDao, never()).saveDetails(any());
     }
+
+    private static BackupProvider createMockBackupProvider(BackupOfferingVO offering, boolean crossZoneInstanceCreationEnabled) {
+        BackupProvider backupProvider = mock(BackupProvider.class);
+        when(backupProvider.crossZoneInstanceCreationEnabled(offering)).thenReturn(crossZoneInstanceCreationEnabled);
+        return backupProvider;
+    }
 }
