--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterActionWorkerTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterActionWorkerTest.java
@@ -173,8 +173,7 @@
         Mockito.when(kubernetesClusterAffinityGroupMapDao.listAffinityGroupIdsByClusterIdAndNodeType(DEFAULT_ID, "CONTROL"))
             .thenReturn(new ArrayList<>(Arrays.asList(1L)));
 
-        AffinityGroupVO explicitGroup = Mockito.mock(AffinityGroupVO.class);
-        Mockito.when(explicitGroup.getId()).thenReturn(99L);
+        AffinityGroupVO explicitGroup = createMockAffinityGroup();
         Mockito.when(affinityGroupDao.findByAccountAndType(Mockito.anyLong(), Mockito.eq("ExplicitDedication")))
             .thenReturn(explicitGroup);
 
@@ -218,15 +217,20 @@
         Mockito.when(kubernetesClusterAffinityGroupMapDao.listAffinityGroupIdsByClusterIdAndNodeType(DEFAULT_ID, "CONTROL"))
             .thenReturn(new ArrayList<>(Arrays.asList(99L, 2L)));
 
+        AffinityGroupVO explicitGroup = createMockAffinityGroup();
+        Mockito.when(affinityGroupDao.findByAccountAndType(Mockito.anyLong(), Mockito.eq("ExplicitDedication")))
+            .thenReturn(explicitGroup);
+
+        List<Long> result = actionWorker.getMergedAffinityGroupIds(KubernetesClusterNodeType.CONTROL, 1L, 1L);
+
+        Assert.assertEquals(2, result.size());
+        Assert.assertTrue(result.contains(99L));
+        Assert.assertTrue(result.contains(2L));
+    }
+
+    private static AffinityGroupVO createMockAffinityGroup() {
         AffinityGroupVO explicitGroup = Mockito.mock(AffinityGroupVO.class);
         Mockito.when(explicitGroup.getId()).thenReturn(99L);
-        Mockito.when(affinityGroupDao.findByAccountAndType(Mockito.anyLong(), Mockito.eq("ExplicitDedication")))
-            .thenReturn(explicitGroup);
-
-        List<Long> result = actionWorker.getMergedAffinityGroupIds(KubernetesClusterNodeType.CONTROL, 1L, 1L);
-
-        Assert.assertEquals(2, result.size());
-        Assert.assertTrue(result.contains(99L));
-        Assert.assertTrue(result.contains(2L));
+        return explicitGroup;
     }
 }
