--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -1445,11 +1445,10 @@
 
     @Test
     public void deleteWebhookFilterDeletesFilterSuccessfully() {
-        DeleteWebhookFilterCmd cmd = Mockito.mock(DeleteWebhookFilterCmd.class);
+        DeleteWebhookFilterCmd cmd = createDeleteWebhookFilterCmd();
         WebhookFilterVO filter = Mockito.mock(WebhookFilterVO.class);
         WebhookVO webhook = Mockito.mock(WebhookVO.class);
 
-        Mockito.when(cmd.getId()).thenReturn(1L);
         Mockito.when(webhookFilterDao.searchBy(Mockito.any(), Mockito.any(), Mockito.anyLong(), Mockito.anyLong()))
                 .thenReturn(new Pair<>(List.of(filter), 1));
         Mockito.when(webhookDao.findById(Mockito.anyLong())).thenReturn(webhook);
@@ -1462,9 +1461,8 @@
 
     @Test
     public void deleteWebhookFilterHandlesNoFiltersToDelete() {
-        DeleteWebhookFilterCmd cmd = Mockito.mock(DeleteWebhookFilterCmd.class);
-
-        Mockito.when(cmd.getId()).thenReturn(1L);
+
+        DeleteWebhookFilterCmd cmd = createDeleteWebhookFilterCmd();
         Mockito.when(webhookFilterDao.searchBy(Mockito.any(), Mockito.any(), Mockito.anyLong(), Mockito.anyLong()))
                 .thenReturn(new Pair<>(List.of(), 0));
 
@@ -1491,4 +1489,10 @@
         Assert.assertNotNull(result);
         Assert.assertEquals(managementServerHostVO, result);
     }
+
+    private static DeleteWebhookFilterCmd createDeleteWebhookFilterCmd() {
+        DeleteWebhookFilterCmd cmd = Mockito.mock(DeleteWebhookFilterCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(1L);
+        return cmd;
+    }
 }
