--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -2607,10 +2607,8 @@
     public void processValidationAnswerTestBootNotValidated() {
         doNothing().when(kbossBackupProviderSpy).setBackupAsInvalidAndSendAlert(eq(backupVoMock), any());
 
-        ValidateKbossVmAnswer answer = Mockito.mock(ValidateKbossVmAnswer.class);
-        doReturn(true).when(answer).getResult();
-        doReturn(false).when(answer).isBootValidated();
-
+
+        ValidateKbossVmAnswer answer = createValidateKbossVmAnswer(false);
         ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
         doReturn(true).when(cmd).isWaitForBoot();
 
@@ -2624,9 +2622,7 @@
     public void processValidationAnswerTestBootNotValidatedScriptResultFalse() {
         doNothing().when(kbossBackupProviderSpy).setBackupAsInvalidAndSendAlert(eq(backupVoMock), any());
 
-        ValidateKbossVmAnswer answer = Mockito.mock(ValidateKbossVmAnswer.class);
-        doReturn(true).when(answer).getResult();
-        doReturn(false).when(answer).isBootValidated();
+        ValidateKbossVmAnswer answer = createValidateKbossVmAnswer(false);
         doReturn("false").when(answer).getScriptResult();
 
         ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
@@ -2644,12 +2640,10 @@
     public void processValidationAnswerTestBootNotValidatedScriptResultFalseScreenshotPathNull() {
         doNothing().when(kbossBackupProviderSpy).setBackupAsInvalidAndSendAlert(eq(backupVoMock), any());
 
-        ValidateKbossVmAnswer answer = Mockito.mock(ValidateKbossVmAnswer.class);
-        doReturn(true).when(answer).getResult();
-        doReturn(false).when(answer).isBootValidated();
+
+        ValidateKbossVmAnswer answer = createValidateKbossVmAnswer(false);
         doReturn("false").when(answer).getScriptResult();
         doReturn(null).when(answer).getScreenshotPath();
-
         ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
         doReturn(true).when(cmd).isWaitForBoot();
         doReturn(true).when(cmd).isExecuteScript();
@@ -2665,9 +2659,7 @@
 
     @Test
     public void processValidationAnswerTestBootValidatedScriptResultTrueScreenshotPathNotNull() {
-        ValidateKbossVmAnswer answer = Mockito.mock(ValidateKbossVmAnswer.class);
-        doReturn(true).when(answer).getResult();
-        doReturn(true).when(answer).isBootValidated();
+        ValidateKbossVmAnswer answer = createValidateKbossVmAnswer(true);
         doReturn(null).when(answer).getScriptResult();
         doReturn("snap-path").when(answer).getScreenshotPath();
 
@@ -3113,4 +3105,11 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+
+    private static ValidateKbossVmAnswer createValidateKbossVmAnswer(boolean bootValidated) {
+        ValidateKbossVmAnswer answer = Mockito.mock(ValidateKbossVmAnswer.class);
+        doReturn(true).when(answer).getResult();
+        doReturn(bootValidated).when(answer).isBootValidated();
+        return answer;
+    }
 }
