--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -101,6 +101,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class StorageManagerImplTest {
+    @Mock
+    private DataStoreProvider storeProvider;
 
     @Mock
     VolumeDao _volumeDao;
@@ -1676,7 +1678,6 @@
         String providerName = "test-provider";
         Map<String, String> details = new HashMap<>();
 
-        DataStoreProvider storeProvider = Mockito.mock(DataStoreProvider.class);
         Mockito.when(dataStoreProviderMgr.getDataStoreProvider(providerName)).thenReturn(storeProvider);
 
         storageManagerImpl.discoverObjectStore(name, url, size, providerName, details);
@@ -1690,7 +1691,6 @@
         String providerName = "test-provider";
         Map<String, String> details = new HashMap<>();
 
-        DataStoreProvider storeProvider = Mockito.mock(DataStoreProvider.class);
         ObjectStoreVO existingStore = new ObjectStoreVO();
 
         Mockito.when(dataStoreProviderMgr.getDataStoreProvider(providerName)).thenReturn(storeProvider);
