--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -130,6 +130,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenElementTest {
     @Mock
+    private TungstenAnswer deleteVmAnswer;
+    @Mock
     TungstenFabricUtils tungstenFabricUtils;
     @Mock
     NetworkModel networkModel;
@@ -664,7 +666,6 @@
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteVmiAnswer = mock(TungstenAnswer.class);
-        TungstenAnswer deleteVmAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteTungstenNetworkPolicyAnswer = mock(TungstenAnswer.class);
 
         when(nicProfile.getIPv4Address()).thenReturn("192.168.100.100");
@@ -691,7 +692,6 @@
         HostVO host = mock(HostVO.class);
         TungstenAnswer deleteTungstenVRouterPortAnswer = mock(TungstenAnswer.class);
         TungstenAnswer deleteVmiAnswer = mock(TungstenAnswer.class);
-        TungstenAnswer deleteVmAnswer = mock(TungstenAnswer.class);
 
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
