--- a/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
+++ b/api/src/test/java/org/apache/cloudstack/storage/template/VnfTemplateUtilsTest.java
@@ -209,11 +209,10 @@
     public void testValidateApiCommandParamsInvalidAccessMethods() {
         VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
         Mockito.when(template.getTemplateType()).thenReturn(Storage.TemplateType.VNF);
-        UpdateVnfTemplateCmd cmd = Mockito.mock(UpdateVnfTemplateCmd.class);
         Map<String, String> vnfDetails = Mockito.spy(new HashMap<>());
         vnfDetails.put("access_methods", "invalid");
-        Mockito.when(cmd.getVnfDetails()).thenReturn(vnfDetails);
-
+
+        UpdateVnfTemplateCmd cmd = createMockUpdateVnfTemplateCmd(vnfDetails);
         VnfTemplateUtils.validateApiCommandParams(cmd, template);
     }
 
@@ -221,11 +220,10 @@
     public void testValidateApiCommandParamsInvalidAccessDetails() {
         VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
         Mockito.when(template.getTemplateType()).thenReturn(Storage.TemplateType.VNF);
-        UpdateVnfTemplateCmd cmd = Mockito.mock(UpdateVnfTemplateCmd.class);
         Map<String, String> vnfDetails = Mockito.spy(new HashMap<>());
         vnfDetails.put("invalid", "value");
-        Mockito.when(cmd.getVnfDetails()).thenReturn(vnfDetails);
-
+
+        UpdateVnfTemplateCmd cmd = createMockUpdateVnfTemplateCmd(vnfDetails);
         VnfTemplateUtils.validateApiCommandParams(cmd, template);
     }
 
@@ -258,4 +256,10 @@
         cidrList.add("10.10.10.0/33");
         VnfTemplateUtils.validateVnfCidrList(cidrList);
     }
+
+    private static UpdateVnfTemplateCmd createMockUpdateVnfTemplateCmd(Map<String, String> vnfDetails) {
+        UpdateVnfTemplateCmd cmd = Mockito.mock(UpdateVnfTemplateCmd.class);
+        Mockito.when(cmd.getVnfDetails()).thenReturn(vnfDetails);
+        return cmd;
+    }
 }
