--- a/server/src/test/java/com/cloud/alert/AlertManagerImplTest.java
+++ b/server/src/test/java/com/cloud/alert/AlertManagerImplTest.java
@@ -231,12 +231,10 @@
     public void testRecalculateStorageCapacities() {
         List<Long> mockPoolIds = List.of(101L, 102L, 103L);
         Mockito.when(primaryDataStoreDao.listAllIds()).thenReturn(mockPoolIds);
-        StoragePoolVO sharedPool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(sharedPool.isShared()).thenReturn(true);
+        StoragePoolVO sharedPool = createMockStoragePool(true);
         Mockito.when(primaryDataStoreDao.findById(mockPoolIds.get(0))).thenReturn(sharedPool);
         Mockito.when(primaryDataStoreDao.findById(mockPoolIds.get(1))).thenReturn(sharedPool);
-        StoragePoolVO nonSharedPool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(nonSharedPool.isShared()).thenReturn(false);
+        StoragePoolVO nonSharedPool = createMockStoragePool(false);
         Mockito.when(primaryDataStoreDao.findById(mockPoolIds.get(2))).thenReturn(nonSharedPool);
         Mockito.when(capacityManager.getAllocatedPoolCapacity(sharedPool, null)).thenReturn(10L);
         Mockito.when(capacityManager.getAllocatedPoolCapacity(nonSharedPool, null)).thenReturn(20L);
@@ -354,4 +352,10 @@
         Assert.assertNotNull(expectedTypes);
         Assert.assertTrue(expectedTypes.isEmpty());
     }
+
+    private static StoragePoolVO createMockStoragePool(boolean shared) {
+        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(storagePool.isShared()).thenReturn(shared);
+        return storagePool;
+    }
 }
