--- a/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
+++ b/engine/storage/src/test/java/org/apache/cloudstack/engine/subsystem/api/storage/StrategyPriorityTest.java
@@ -40,15 +40,11 @@
 
     @Test
     public void testSortSnapshotStrategies() {
-        SnapshotStrategy cantHandleStrategy = mock(SnapshotStrategy.class);
-        SnapshotStrategy defaultStrategy = mock(SnapshotStrategy.class);
-        SnapshotStrategy hyperStrategy = mock(SnapshotStrategy.class);
-        SnapshotStrategy highestStrategy = mock(SnapshotStrategy.class);
+        SnapshotStrategy highestStrategy = createSnapshotStrategy(StrategyPriority.HIGHEST);
 
-        doReturn(StrategyPriority.CANT_HANDLE).when(cantHandleStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
-        doReturn(StrategyPriority.DEFAULT).when(defaultStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
-        doReturn(StrategyPriority.HYPERVISOR).when(hyperStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
-        doReturn(StrategyPriority.HIGHEST).when(highestStrategy).canHandle(any(Snapshot.class), Mockito.nullable(Long.class), any(SnapshotStrategy.SnapshotOperation.class));
+        SnapshotStrategy cantHandleStrategy = createSnapshotStrategy(StrategyPriority.CANT_HANDLE);
+        SnapshotStrategy defaultStrategy = createSnapshotStrategy(StrategyPriority.DEFAULT);
+        SnapshotStrategy hyperStrategy = createSnapshotStrategy(StrategyPriority.HYPERVISOR);
 
         List<SnapshotStrategy> strategies = new ArrayList<>(5);
         SnapshotStrategy strategy = null;
@@ -143,4 +139,13 @@
         strategy = factory.getDataMotionStrategy(mock(Map.class), mock(Host.class), mock(Host.class));
         assertEquals("Highest strategy was not picked.", highestStrategy, strategy);
     }
+
+    private static SnapshotStrategy createSnapshotStrategy(StrategyPriority strategyPriority) {
+        SnapshotStrategy strategy = mock(SnapshotStrategy.class);
+        doReturn(strategyPriority).when(strategy).canHandle(
+                any(Snapshot.class),
+                Mockito.nullable(Long.class),
+                any(SnapshotStrategy.SnapshotOperation.class));
+        return strategy;
+    }
 }
