--- a/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
@@ -56,6 +56,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class VpcVirtualRouterElementTest {
     @Mock
+    private final AdvancedNetworkTopology advancedNetworkTopology;
+    @Mock
     DataCenterDao _dcDao;
     @Mock
     private DomainRouterDao _routerDao;
@@ -147,7 +149,6 @@
     public void testApplyVpnUsersException1() {
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
-        final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
         final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
         networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
@@ -173,7 +174,6 @@
     public void testApplyVpnUsersException2() {
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
-        final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
         final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
         networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
