--- a/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/sharedfs/SharedFSServiceImplTest.java
@@ -434,8 +434,7 @@
         Long newMinIops = 2000L;
         Long newMaxIops = 4000L;
         Long newDiskOfferingId = 10L;
-        ChangeSharedFSDiskOfferingCmd cmd = mock(ChangeSharedFSDiskOfferingCmd.class);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
+        ChangeSharedFSDiskOfferingCmd cmd = createChangeSharedFSDiskOfferingCmd(s_sharedFSId);
         when(cmd.getDiskOfferingId()).thenReturn(newDiskOfferingId);
         when(cmd.getSize()).thenReturn(newSize);
         when(cmd.getMinIops()).thenReturn(newMinIops);
@@ -459,9 +458,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testChangeSharedFSDiskOfferingInvalidState() throws ResourceAllocationException {
-        ChangeSharedFSDiskOfferingCmd cmd = mock(ChangeSharedFSDiskOfferingCmd.class);
-        when(cmd.getId()).thenReturn(s_sharedFSId);
-
+
+        ChangeSharedFSDiskOfferingCmd cmd = createChangeSharedFSDiskOfferingCmd(s_sharedFSId);
         SharedFSVO sharedFS = getMockSharedFS();
         ReflectionTestUtils.setField(sharedFS, "state", SharedFS.State.Destroyed);
         when(sharedFSDao.findById(s_sharedFSId)).thenReturn(sharedFS);
@@ -662,4 +660,10 @@
             verify(_stateMachine, times(1)).transitTo(sharedFS, SharedFS.Event.OperationFailed, null, sharedFSDao);
         }
     }
+
+    private static ChangeSharedFSDiskOfferingCmd createChangeSharedFSDiskOfferingCmd(Long sharedFSId) {
+        ChangeSharedFSDiskOfferingCmd cmd = mock(ChangeSharedFSDiskOfferingCmd.class);
+        when(cmd.getId()).thenReturn(sharedFSId);
+        return cmd;
+    }
 }
