--- a/engine/schema/src/test/java/org/apache/cloudstack/resourcedetail/ResourceDetailsDaoBaseTest.java
+++ b/engine/schema/src/test/java/org/apache/cloudstack/resourcedetail/ResourceDetailsDaoBaseTest.java
@@ -73,12 +73,8 @@
     public void testListDetailsKeyPairs() {
         long resourceId = 1L;
         List<String> keys = Arrays.asList("key1", "key2");
-        TestDetailVO result1 = mock(TestDetailVO.class);
-        when(result1.getName()).thenReturn("key1");
-        when(result1.getValue()).thenReturn("value1");
-        TestDetailVO result2 = mock(TestDetailVO.class);
-        when(result2.getName()).thenReturn("key2");
-        when(result2.getValue()).thenReturn("value2");
+        TestDetailVO result1 = createTestDetailVO("key1", "value1");
+        TestDetailVO result2 = createTestDetailVO("key2", "value2");
         List<TestDetailVO> mockResults = Arrays.asList(result1, result2);
         doReturn(mockResults).when(testDetailsDao).search(any(SearchCriteria.class), isNull());
         Map<String, String> result = testDetailsDao.listDetailsKeyPairs(resourceId, keys);
@@ -178,4 +174,11 @@
             this.value = value;
         }
     }
+
+    private static TestDetailVO createTestDetailVO(String name, String value) {
+        TestDetailVO testDetailVO = mock(TestDetailVO.class);
+        when(testDetailVO.getName()).thenReturn(name);
+        when(testDetailVO.getValue()).thenReturn(value);
+        return testDetailVO;
+    }
 }
