--- a/server/src/test/java/org/apache/cloudstack/gui/theme/GuiThemeServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/gui/theme/GuiThemeServiceImplTest.java
@@ -75,9 +75,8 @@
         Long accountId = Account.ACCOUNT_ID_SYSTEM;
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
+            CallContext callContextMock = createCallContext(accountId);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
-            Mockito.doReturn(accountId).when(callContextMock).getCallingAccountId();
             Mockito.doReturn(emptyPair).when(guiThemeServiceSpy).listGuiThemesWithNoAuthentication(Mockito.nullable(ListGuiThemesCmd.class));
             Mockito.when(listGuiThemesCmdMock.getListOnlyDefaultTheme()).thenReturn(false);
             guiThemeServiceSpy.listGuiThemes(listGuiThemesCmdMock);
@@ -92,9 +91,8 @@
         Long accountId = 3L;
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
+            CallContext callContextMock = createCallContext(accountId);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
-            Mockito.doReturn(accountId).when(callContextMock).getCallingAccountId();
             Mockito.doReturn(emptyPair).when(guiThemeServiceSpy).listGuiThemesInternal(Mockito.nullable(ListGuiThemesCmd.class));
             Mockito.when(listGuiThemesCmdMock.getListOnlyDefaultTheme()).thenReturn(false);
             guiThemeServiceSpy.listGuiThemes(listGuiThemesCmdMock);
@@ -233,4 +231,10 @@
 
         guiThemeServiceSpy.checkIfDefaultThemeIsAllowed(BLANK_STRING, BLANK_STRING, BLANK_STRING, 1L);
     }
+
+    private static CallContext createCallContext(Long accountId) {
+        CallContext callContextMock = Mockito.mock(CallContext.class);
+        Mockito.doReturn(accountId).when(callContextMock).getCallingAccountId();
+        return callContextMock;
+    }
 }
