--- a/api/src/test/java/org/apache/cloudstack/api/command/user/backup/ListBackupScheduleCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/backup/ListBackupScheduleCmdTest.java
@@ -40,6 +40,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ListBackupScheduleCmdTest {
+    @Mock
+    private CallContext callContext;
 
     @Mock
     private BackupManager backupManager;
@@ -67,7 +69,6 @@
         Mockito.when(responseGenerator.createBackupScheduleResponse(schedule)).thenReturn(scheduleResponse);
 
         Account mockAccount = Mockito.mock(Account.class);
-        CallContext callContext = Mockito.mock(CallContext.class);
         try (org.mockito.MockedStatic<CallContext> mocked = Mockito.mockStatic(CallContext.class)) {
             cmd.execute();
         }
@@ -81,7 +82,6 @@
     @Test
     public void testExecuteWithNoSchedules() {
         Mockito.when(backupManager.listBackupSchedules(cmd)).thenReturn(new ArrayList<>());
-        CallContext callContext = Mockito.mock(CallContext.class);
 
         try (org.mockito.MockedStatic<CallContext> mocked = Mockito.mockStatic(CallContext.class)) {
             mocked.when(CallContext::current).thenReturn(callContext);
