--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -622,7 +622,7 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
-        when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
+        tungstenFabricUtils = MockTungstenFabricUtils.createTungstenFabricUtils(createTungstenVMAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
 
         assertTrue(tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext));
@@ -647,7 +647,7 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Management);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(host);
-        when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(createTungstenVMAnswer);
+        tungstenFabricUtils = MockTungstenFabricUtils.createTungstenFabricUtils(createTungstenVMAnswer);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.SecondaryStorageVm);
 
         tungstenElement.prepare(network, nicProfile, virtualMachineProfile, deployDestination, reservationContext);
@@ -922,8 +922,8 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Public);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
-        when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class), anyLong())).thenReturn(tungstenAnswer);
-
+
+        tungstenFabricUtils = MockTungstenFabricUtils.createTungstenFabricUtils(tungstenAnswer);
         assertTrue(tungstenElement.prepareMigration(nic, network, vm, dest, context));
     }
 
@@ -962,4 +962,19 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    public final class MockTungstenFabricUtils {
+        private MockTungstenFabricUtils() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenFabricUtils createTungstenFabricUtils(
+                org.apache.cloudstack.network.tungsten.agent.api.TungstenAnswer tungstenAnswer) {
+            org.apache.cloudstack.network.tungsten.service.TungstenFabricUtils tungstenFabricUtils =
+                    mock(org.apache.cloudstack.network.tungsten.service.TungstenFabricUtils.class);
+            when(tungstenFabricUtils.sendTungstenCommand(
+                    any(org.apache.cloudstack.network.tungsten.agent.api.CreateTungstenVirtualMachineCommand.class),
+                    anyLong())).thenReturn(tungstenAnswer);
+            return tungstenFabricUtils;
+        }
+    }
 }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -483,9 +483,8 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
-        when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
-            anyLong())).thenReturn(new TungstenAnswer(new TungstenCommand(), true, ""));
-
+
+        tungstenFabricUtils = MockTungstenFabricUtils.createTungstenFabricUtils(new TungstenAnswer(new TungstenCommand(), true, ""));
         guru.prepareMigration(nic, network, vm, dest, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
             anyLong());
@@ -526,4 +525,19 @@
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
+
+    public final class MockTungstenFabricUtils {
+        private MockTungstenFabricUtils() {
+        }
+
+        public static org.apache.cloudstack.network.tungsten.service.TungstenFabricUtils createTungstenFabricUtils(
+                org.apache.cloudstack.network.tungsten.agent.api.TungstenAnswer tungstenAnswer) {
+            org.apache.cloudstack.network.tungsten.service.TungstenFabricUtils tungstenFabricUtils =
+                    mock(org.apache.cloudstack.network.tungsten.service.TungstenFabricUtils.class);
+            when(tungstenFabricUtils.sendTungstenCommand(
+                    any(org.apache.cloudstack.network.tungsten.agent.api.CreateTungstenVirtualMachineCommand.class),
+                    anyLong())).thenReturn(tungstenAnswer);
+            return tungstenFabricUtils;
+        }
+    }
 }
