--- a/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
@@ -813,13 +813,9 @@
 
     @Test
     public void testCreateBgpPeerResponse() {
-        BgpPeerVO bgpPeer = Mockito.mock(BgpPeerVO.class);
-        when(bgpPeer.getDataCenterId()).thenReturn(zoneId);
-        when(bgpPeer.getDomainId()).thenReturn(domainId);
-        when(bgpPeer.getAccountId()).thenReturn(accountId);
+        BgpPeerVO bgpPeer = createMockBgpPeerVO(accountId, zoneId, domainId, bgpPeerUuid);
         when(bgpPeer.getCreated()).thenReturn(created);
         when(bgpPeer.getAsNumber()).thenReturn(asNumber);
-        when(bgpPeer.getUuid()).thenReturn(bgpPeerUuid);
         when(bgpPeer.getIp4Address()).thenReturn(ip4Address);
         when(bgpPeer.getIp6Address()).thenReturn(ip6Address);
 
@@ -1060,8 +1056,8 @@
 
         when(account.getDomainId()).thenReturn(domainId);
         when(account.getAccountId()).thenReturn(accountId);
+         bgpPeer = createMockBgpPeerVO(accountId, zoneId, domainId, bgpPeerUuid);
         when(bgpPeerDao.findById(bgpPeerId)).thenReturn(bgpPeer);
-        when(bgpPeer.getUuid()).thenReturn(bgpPeerUuid);
         try {
             when(bgpPeer.getDataCenterId()).thenReturn(zoneId + 1);
             routedIpv4Manager.validateBgpPeers(account, zoneId, bgpPeerIds);
@@ -1214,4 +1210,13 @@
         verify(ipv4GuestSubnetNetworkMapDao).deleteByParentId(zoneSubnetId);
         verify(dataCenterIpv4GuestSubnetDao).remove(zoneSubnetId);
     }
+
+    private static BgpPeerVO createMockBgpPeerVO(Long accountId, Long dataCenterId, Long domainId, String uuid) {
+        BgpPeerVO bgpPeer = Mockito.mock(BgpPeerVO.class);
+        when(bgpPeer.getAccountId()).thenReturn(accountId);
+        when(bgpPeer.getDataCenterId()).thenReturn(dataCenterId);
+        when(bgpPeer.getDomainId()).thenReturn(domainId);
+        when(bgpPeer.getUuid()).thenReturn(uuid);
+        return bgpPeer;
+    }
 }
