--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricTagCmdTest.java
@@ -37,6 +37,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class ListTungstenFabricTagCmdTest {
+    @Mock
+    private BaseResponse baseResponse;
 
     @Mock
     TungstenService tungstenService;
@@ -69,7 +71,6 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricTagCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         Mockito.when(tungstenService.listTungstenTags(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(),
@@ -87,7 +88,6 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
         List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
