--- a/engine/schema/src/test/java/org/apache/cloudstack/storage/datastore/db/PrimaryDataStoreDaoImplTest.java
+++ b/engine/schema/src/test/java/org/apache/cloudstack/storage/datastore/db/PrimaryDataStoreDaoImplTest.java
@@ -51,6 +51,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class PrimaryDataStoreDaoImplTest extends TestCase {
+    @Mock
+    private StoragePoolVO entityVO;
 
     @Mock
     StoragePoolDetailsDao _detailsDao;
@@ -168,7 +170,7 @@
     @Test
     public void testListAllIds() {
         GenericSearchBuilder<StoragePoolVO, Long> genericSearchBuilder = mock(SearchBuilder.class);
-        StoragePoolVO entityVO = mock(StoragePoolVO.class);
+        // removed local mock; replaced with global field `entityVO`
         when(genericSearchBuilder.entity()).thenReturn(entityVO);
         doReturn(genericSearchBuilder).when(primaryDataStoreDao).createSearchBuilder(Long.class);
         List<Long> mockIds = Arrays.asList(1L, 2L, 3L);
@@ -184,7 +186,6 @@
     @Test
     public void testListAllIdsEmptyResult() {
         GenericSearchBuilder<StoragePoolVO, Long> genericSearchBuilder = mock(SearchBuilder.class);
-        StoragePoolVO entityVO = mock(StoragePoolVO.class);
         when(genericSearchBuilder.entity()).thenReturn(entityVO);
         doReturn(genericSearchBuilder).when(primaryDataStoreDao).createSearchBuilder(Long.class);
         doReturn(Collections.emptyList()).when(primaryDataStoreDao).customSearch(any(), isNull());
