--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -917,10 +917,8 @@
         List<VolumeVO> volumesUsingTheStoragePoolAccessGroups = new ArrayList<>(Arrays.asList(volume1, volume2));
         Mockito.doReturn(volumesUsingTheStoragePoolAccessGroups).when(resourceManager).listOfVolumesUsingTheStorageAccessGroups(sagsToDelete, 1L, null, null, null);
 
-        StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
-        StoragePoolVO pool2 = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(pool1.getId()).thenReturn(10L);
-        Mockito.when(pool2.getId()).thenReturn(12L);
+        StoragePoolVO pool2 = createMockStoragePool(12L);
+        StoragePoolVO pool1 = createMockStoragePool(10L);
         List<StoragePoolVO> poolsToAdd = Arrays.asList(pool1, pool2);
 
         Mockito.doReturn(poolsToAdd)
@@ -1393,4 +1391,10 @@
         Mockito.verify(hostDao).findByClusterId(clusterId, Host.Type.Routing);
         Mockito.verify(hostDao).findByPodId(podId, Host.Type.Routing);
     }
+
+    private static StoragePoolVO createMockStoragePool(long id) {
+        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(pool.getId()).thenReturn(id);
+        return pool;
+    }
 }
