--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -1032,17 +1032,13 @@
 
     @Test
     public void listWebhookDeliveriesReturnsEmptyResponseForNoDeliveries() {
-        ListWebhookDeliveriesCmd cmd = Mockito.mock(ListWebhookDeliveriesCmd.class);
-
-        Mockito.when(cmd.getId()).thenReturn(null);
+
+        ListWebhookDeliveriesCmd cmd = createMockListWebhookDeliveriesCmd();
         Mockito.when(cmd.getWebhookId()).thenReturn(null);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(null);
         Mockito.when(cmd.getKeyword()).thenReturn(null);
         Mockito.when(cmd.getStartDate()).thenReturn(null);
         Mockito.when(cmd.getEndDate()).thenReturn(null);
         Mockito.when(cmd.getEventType()).thenReturn(null);
-        Mockito.when(cmd.getStartIndex()).thenReturn(0L);
-        Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         Mockito.when(webhookDeliveryJoinDao.searchAndCountByListApiParameters(Mockito.any(), Mockito.anyList(),
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                         Mockito.anyBoolean()))
@@ -1057,15 +1053,11 @@
 
     @Test
     public void listWebhookDeliveriesFiltersByWebhookId() {
-        ListWebhookDeliveriesCmd cmd = Mockito.mock(ListWebhookDeliveriesCmd.class);
+        ListWebhookDeliveriesCmd cmd = createMockListWebhookDeliveriesCmd();
         WebhookDeliveryJoinVO delivery = Mockito.mock(WebhookDeliveryJoinVO.class);
         WebhookDeliveryResponse deliveryResponse = Mockito.mock(WebhookDeliveryResponse.class);
 
-        Mockito.when(cmd.getId()).thenReturn(null);
         Mockito.when(cmd.getWebhookId()).thenReturn(1L);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(null);
-        Mockito.when(cmd.getStartIndex()).thenReturn(0L);
-        Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         Mockito.when(webhookDeliveryJoinDao.searchAndCountByListApiParameters(Mockito.any(), Mockito.anyList(),
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                         Mockito.anyBoolean()))
@@ -1094,16 +1086,12 @@
 
     @Test
     public void listWebhookDeliveriesFiltersByEventType() {
-        ListWebhookDeliveriesCmd cmd = Mockito.mock(ListWebhookDeliveriesCmd.class);
+        ListWebhookDeliveriesCmd cmd = createMockListWebhookDeliveriesCmd();
         WebhookDeliveryJoinVO delivery = Mockito.mock(WebhookDeliveryJoinVO.class);
         WebhookDeliveryResponse deliveryResponse = Mockito.mock(WebhookDeliveryResponse.class);
 
-        Mockito.when(cmd.getId()).thenReturn(null);
         Mockito.when(cmd.getWebhookId()).thenReturn(null);
-        Mockito.when(cmd.getManagementServerId()).thenReturn(null);
         Mockito.when(cmd.getEventType()).thenReturn("EventType");
-        Mockito.when(cmd.getStartIndex()).thenReturn(0L);
-        Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         Mockito.when(webhookDeliveryJoinDao.searchAndCountByListApiParameters(Mockito.any(), Mockito.anyList(),
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                         Mockito.anyBoolean()))
@@ -1491,4 +1479,13 @@
         Assert.assertNotNull(result);
         Assert.assertEquals(managementServerHostVO, result);
     }
+
+    private static ListWebhookDeliveriesCmd createMockListWebhookDeliveriesCmd() {
+        ListWebhookDeliveriesCmd cmd = Mockito.mock(ListWebhookDeliveriesCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(null);
+        Mockito.when(cmd.getManagementServerId()).thenReturn(null);
+        Mockito.when(cmd.getStartIndex()).thenReturn(0L);
+        Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
+        return cmd;
+    }
 }
