--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -541,7 +541,7 @@
 
     @Test
     public void tryRestoreVMTestRestoreSucceeded() throws NoTransitionException {
-        BackupOffering offering = mock(BackupOffering.class);
+        BackupOffering offering = createMockBackupOffering();
         VolumeVO volumeVO = mock(VolumeVO.class);
         VMInstanceVO vm = mock(VMInstanceVO.class);
         BackupVO backup = mock(BackupVO.class);
@@ -559,7 +559,6 @@
             Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVO), Mockito.eq(Volume.Event.RestoreRequested))).thenReturn(true);
 
             Mockito.when(vm.getId()).thenReturn(1L);
-            Mockito.when(offering.getProvider()).thenReturn("veeam");
             Mockito.doReturn(backupProvider).when(backupManager).getBackupProvider("veeam");
             Mockito.when(backupProvider.restoreVMFromBackup(vm, backup, false, null)).thenReturn(true);
 
@@ -569,7 +568,7 @@
 
     @Test
     public void tryRestoreVMTestRestoreFails() throws NoTransitionException {
-        BackupOffering offering = mock(BackupOffering.class);
+        BackupOffering offering = createMockBackupOffering();
         VolumeVO volumeVO = mock(VolumeVO.class);
         VMInstanceVO vm = mock(VMInstanceVO.class);
         BackupVO backup = mock(BackupVO.class);
@@ -589,7 +588,6 @@
             Mockito.when(volumeApiService.stateTransitTo(Mockito.eq(volumeVO), Mockito.eq(Volume.Event.RestoreFailed))).thenReturn(true);
 
             Mockito.when(vm.getId()).thenReturn(1L);
-            Mockito.when(offering.getProvider()).thenReturn("veeam");
             Mockito.doReturn(backupProvider).when(backupManager).getBackupProvider("veeam");
             Mockito.when(backupProvider.restoreVMFromBackup(vm, backup, false, null)).thenReturn(false);
             try {
@@ -2874,4 +2872,10 @@
         verify(backupOfferingDao).persist(any());
         verify(backupOfferingDetailsDao, never()).saveDetails(any());
     }
+
+    private static BackupOffering createMockBackupOffering() {
+        BackupOffering offering = mock(BackupOffering.class);
+        Mockito.when(offering.getProvider()).thenReturn("veeam");
+        return offering;
+    }
 }
