--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -697,10 +697,8 @@
         Mockito.when(webhookFilter.getValue()).thenReturn("value");
         Mockito.when(webhookFilter.getCreated()).thenReturn(new Date(1000));
 
-        WebhookVO webhookVO = Mockito.mock(WebhookVO.class);
-        Mockito.when(webhookVO.getUuid()).thenReturn("webhookUuid");
-        Mockito.when(webhookVO.getName()).thenReturn("webhookName");
-
+
+        WebhookVO webhookVO = createMockWebhookVO();
         WebhookFilterResponse response = webhookApiServiceImpl.createWebhookFilterResponse(webhookFilter, webhookVO);
 
         Assert.assertEquals("webhookfilter", ReflectionTestUtils.getField(response, "objectName"));
@@ -725,9 +723,7 @@
         Mockito.when(webhookFilter.getValue()).thenReturn("value");
         Mockito.when(webhookFilter.getCreated()).thenReturn(new Date(1000));
 
-        WebhookVO webhookVO = Mockito.mock(WebhookVO.class);
-        Mockito.when(webhookVO.getUuid()).thenReturn("webhookUuid");
-        Mockito.when(webhookVO.getName()).thenReturn("webhookName");
+        WebhookVO webhookVO = createMockWebhookVO();
         Mockito.when(webhookDao.findById(1L)).thenReturn(webhookVO);
 
         WebhookFilterResponse response = webhookApiServiceImpl.createWebhookFilterResponse(webhookFilter, null);
@@ -1491,4 +1487,11 @@
         Assert.assertNotNull(result);
         Assert.assertEquals(managementServerHostVO, result);
     }
+
+    private static org.apache.cloudstack.mom.webhook.vo.WebhookVO createMockWebhookVO() {
+        org.apache.cloudstack.mom.webhook.vo.WebhookVO webhookVO = Mockito.mock(org.apache.cloudstack.mom.webhook.vo.WebhookVO.class);
+        Mockito.when(webhookVO.getUuid()).thenReturn("webhookUuid");
+        Mockito.when(webhookVO.getName()).thenReturn("webhookName");
+        return webhookVO;
+    }
 }
