--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -1772,10 +1772,8 @@
         when(cmd.getResourceId()).thenReturn("vm-123");
         when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
 
-        ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
+        ExtensionCustomActionVO actionVO = createMockExtensionCustomActionVO();
         when(extensionCustomActionDao.findById(1L)).thenReturn(actionVO);
-        when(actionVO.isEnabled()).thenReturn(true);
-        when(actionVO.getResourceType()).thenReturn(ExtensionCustomAction.ResourceType.VirtualMachine);
         when(actionVO.getAllowedRoleTypes()).thenReturn(
                 RoleType.toCombinedMask(List.of(RoleType.Admin, RoleType.DomainAdmin, RoleType.User)));
 
@@ -1876,10 +1874,8 @@
         when(cmd.getResourceId()).thenReturn("vm-123");
         when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
 
-        ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
+        ExtensionCustomActionVO actionVO = createMockExtensionCustomActionVO();
         when(extensionCustomActionDao.findById(1L)).thenReturn(actionVO);
-        when(actionVO.isEnabled()).thenReturn(true);
-        when(actionVO.getResourceType()).thenReturn(ExtensionCustomAction.ResourceType.VirtualMachine);
 
         ExtensionVO extensionVO = mock(ExtensionVO.class);
         when(extensionDao.findById(anyLong())).thenReturn(extensionVO);
@@ -1910,10 +1906,8 @@
         when(cmd.getResourceId()).thenReturn("vm-123");
         when(cmd.getParameters()).thenReturn(Map.of("param1", "value1"));
 
-        ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
+        ExtensionCustomActionVO actionVO = createMockExtensionCustomActionVO();
         when(extensionCustomActionDao.findById(1L)).thenReturn(actionVO);
-        when(actionVO.isEnabled()).thenReturn(true);
-        when(actionVO.getResourceType()).thenReturn(ExtensionCustomAction.ResourceType.VirtualMachine);
         when(actionVO.getAllowedRoleTypes()).thenReturn(RoleType.toCombinedMask(List.of(RoleType.Admin, RoleType.DomainAdmin, RoleType.User)));
 
         ExtensionVO extensionVO = mock(ExtensionVO.class);
@@ -3062,4 +3056,11 @@
         assertTrue(resp.getResult().get(ApiConstants.DETAILS).contains("Action failed"));
     }
 
+
+    private static ExtensionCustomActionVO createMockExtensionCustomActionVO() {
+        ExtensionCustomActionVO actionVO = mock(ExtensionCustomActionVO.class);
+        when(actionVO.getResourceType()).thenReturn(ExtensionCustomAction.ResourceType.VirtualMachine);
+        when(actionVO.isEnabled()).thenReturn(true);
+        return actionVO;
+    }
 }
