--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -129,13 +129,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         String srcPoolSystemId = "610204d03e3ad60f";
-        when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
-
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockStoragePoolDetailVO(srcPoolSystemId);
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockStoragePoolDetailVO(destPoolSystemId);
         String destPoolSystemId = "610204d03e3ad60f";
-        when(destPoolSystemIdDetail.getValue()).thenReturn(destPoolSystemId);
 
         when(storagePoolDetailsDao.findDetail(1L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcPoolSystemIdDetail);
         when(storagePoolDetailsDao.findDetail(2L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(destPoolSystemIdDetail);
@@ -152,13 +150,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         String srcPoolSystemId = "610204d03e3ad60f";
-        when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
-
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockStoragePoolDetailVO(srcPoolSystemId);
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockStoragePoolDetailVO(destPoolSystemId);
         String destPoolSystemId = "7332760565f6340f";
-        when(destPoolSystemIdDetail.getValue()).thenReturn(destPoolSystemId);
 
         when(storagePoolDetailsDao.findDetail(1L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcPoolSystemIdDetail);
         when(storagePoolDetailsDao.findDetail(2L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(destPoolSystemIdDetail);
@@ -175,10 +171,9 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         String srcPoolSystemId = "610204d03e3ad60f";
-        when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
-
+
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockStoragePoolDetailVO(srcPoolSystemId);
         StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
         when(destPoolSystemIdDetail.getValue()).thenReturn(null);
 
@@ -597,4 +592,10 @@
                         16L * (1024 * 1024 * 1024),
                         true));
     }
+
+    private static StoragePoolDetailVO createMockStoragePoolDetailVO(String value) {
+        StoragePoolDetailVO storagePoolDetailVO = Mockito.mock(StoragePoolDetailVO.class);
+        when(storagePoolDetailVO.getValue()).thenReturn(value);
+        return storagePoolDetailVO;
+    }
 }
