--- a/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/api/command/ListNetrisProvidersCmdTest.java
+++ b/plugins/network-elements/netris/src/test/java/org/apache/cloudstack/api/command/ListNetrisProvidersCmdTest.java
@@ -34,6 +34,8 @@
 import java.util.List;
 
 public class ListNetrisProvidersCmdTest {
+    @Mock
+    private NetrisProviderResponse providerResponse;
 
     @Mock
     private NetrisProviderService netrisProviderService;
@@ -65,7 +67,6 @@
     @Test
     public void testExecuteSuccess() throws ConcurrentOperationException {
         // Setup
-        NetrisProviderResponse providerResponse = Mockito.mock(NetrisProviderResponse.class);
         List<BaseResponse> providerList = Arrays.asList(providerResponse);
         Mockito.when(netrisProviderService.listNetrisProviders(ZONE_ID)).thenReturn(providerList);
 
@@ -83,7 +84,6 @@
     public void testExecuteWithoutZoneId() throws ConcurrentOperationException, Exception {
         // Setup
         setPrivateField("zoneId", null);
-        NetrisProviderResponse providerResponse = Mockito.mock(NetrisProviderResponse.class);
         List<BaseResponse> providerList = Arrays.asList(providerResponse);
         Mockito.when(netrisProviderService.listNetrisProviders(null)).thenReturn(providerList);
 
