--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/dao/WebhookDeliveryDaoImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/dao/WebhookDeliveryDaoImplTest.java
@@ -96,10 +96,8 @@
     public void removeOlderDeliveriesWhenParametersMatch() {
         long webhookId = 2L;
 
-        WebhookDeliveryVO d1 = mock(WebhookDeliveryVO.class);
-        when(d1.getId()).thenReturn(1L);
-        WebhookDeliveryVO d2 = mock(WebhookDeliveryVO.class);
-        when(d2.getId()).thenReturn(2L);
+        WebhookDeliveryVO d1 = createMockWebhookDeliveryVO(1L);
+        WebhookDeliveryVO d2 = createMockWebhookDeliveryVO(2L);
         List<WebhookDeliveryVO> list = List.of(d1, d2);
         doReturn(list).when(webhookDeliveryDao).listBy(any(SearchCriteria.class), any());
         doReturn(10).when(webhookDeliveryDao).remove(any(SearchCriteria.class));
@@ -123,4 +121,10 @@
         verify(mockSearchCriteria).setParameters("webhookId", webhookId);
         verify(mockSearchBuilder, never()).and(eq("id"), any(), eq(SearchCriteria.Op.NOTIN));
     }
+
+    private static WebhookDeliveryVO createMockWebhookDeliveryVO(long id) {
+        WebhookDeliveryVO delivery = mock(WebhookDeliveryVO.class);
+        when(delivery.getId()).thenReturn(id);
+        return delivery;
+    }
 }
