--- a/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategyTest.java
+++ b/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategyTest.java
@@ -158,15 +158,13 @@
     public void testCanBypassSecondaryStorageForUnsupportedSrcPoolType() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.PowerFlex).when(srcVolumeInfo).getStoragePoolType();
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -181,15 +179,13 @@
     public void testCanBypassSecondaryStorageForUnsupportedDestPoolType() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(srcVolumeInfo).getStoragePoolType();
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.Iscsi).when(destVolumeInfo).getStoragePoolType();
 
@@ -204,15 +200,13 @@
     public void testCanBypassSecondaryStorageWithZoneWideNFSPoolsInSameZone() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(srcVolumeInfo).getStoragePoolType();
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -283,8 +277,7 @@
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -309,8 +302,7 @@
 
         VolumeObject destVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(destVolumeInfo).getHypervisorType();
-        DataStore destDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new ZoneScope(1L)).when(destDataStore).getScope();
+        DataStore destDataStore = createMockDataStore();
         Mockito.doReturn(destDataStore).when(destVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(destVolumeInfo).getStoragePoolType();
 
@@ -504,4 +496,10 @@
         Mockito.verify(clvmPoolManager, never()).getClvmLockHostId(any(Long.class), any(String.class),
                 any(String.class), any(StoragePool.class), Mockito.anyBoolean());
     }
+
+    private static DataStore createMockDataStore() {
+        DataStore dataStore = Mockito.mock(DataStore.class);
+        Mockito.doReturn(new ZoneScope(1L)).when(dataStore).getScope();
+        return dataStore;
+    }
 }
