--- a/core/src/test/java/com/cloud/agent/resource/virtualnetwork/model/BgpPeersTest.java
+++ b/core/src/test/java/com/cloud/agent/resource/virtualnetwork/model/BgpPeersTest.java
@@ -16,6 +16,7 @@
 // under the License.
 package com.cloud.agent.resource.virtualnetwork.model;
 
+import org.mockito.Mock;
 import org.apache.cloudstack.network.BgpPeerTO;
 import org.junit.Assert;
 import org.junit.Test;
@@ -25,10 +26,11 @@
 import java.util.List;
 
 public class BgpPeersTest {
+    @Mock
+    private BgpPeerTO bgpPeerTO;
 
     @Test
     public void testBgpPeers() {
-        BgpPeerTO bgpPeerTO = Mockito.mock(BgpPeerTO.class);
         List<BgpPeerTO> bgpPeerTOs = new ArrayList<>();
         bgpPeerTOs.add(bgpPeerTO);
 
@@ -44,7 +46,6 @@
         BgpPeers bgpPeers = new BgpPeers();
         Assert.assertEquals(ConfigBase.BGP_PEERS, bgpPeers.getType());
 
-        BgpPeerTO bgpPeerTO = Mockito.mock(BgpPeerTO.class);
         List<BgpPeerTO> bgpPeerTOs = new ArrayList<>();
         bgpPeerTOs.add(bgpPeerTO);
         bgpPeers.setPeers(bgpPeerTOs);
