--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -2018,9 +2018,8 @@
         Long backupOfferingId = 2L;
         String providerName = "testbackupprovider";
 
-        BackupVO backup = mock(BackupVO.class);
-        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
-
+
+        BackupVO backup = createMockBackup(backupOfferingId);
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
 
@@ -2046,9 +2045,8 @@
         Long backupOfferingId = 2L;
         String providerName = "testbackupprovider";
 
-        BackupVO backup = mock(BackupVO.class);
-        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
-
+
+        BackupVO backup = createMockBackup(backupOfferingId);
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
 
@@ -2073,9 +2071,8 @@
         Long backupId = 1L;
         Long backupOfferingId = 2L;
 
-        BackupVO backup = mock(BackupVO.class);
-        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
-
+
+        BackupVO backup = createMockBackup(backupOfferingId);
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(null);
 
@@ -2874,4 +2871,10 @@
         verify(backupOfferingDao).persist(any());
         verify(backupOfferingDetailsDao, never()).saveDetails(any());
     }
+
+    private static BackupVO createMockBackup(Long backupOfferingId) {
+        BackupVO backup = mock(BackupVO.class);
+        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
+        return backup;
+    }
 }
