--- a/server/src/test/java/com/cloud/alert/AlertManagerImplTest.java
+++ b/server/src/test/java/com/cloud/alert/AlertManagerImplTest.java
@@ -71,6 +71,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class AlertManagerImplTest {
+    @Mock
+    private MessageBus messageBusMock;
 
     @Spy
     @InjectMocks
@@ -280,7 +282,6 @@
 
     @Test
     public void initMessageBusListenerSubscribesToConfigurationEditEvent() {
-        MessageBus messageBusMock = Mockito.mock(MessageBus.class);
         alertManagerImplMock.messageBus = messageBusMock;
         alertManagerImplMock.initMessageBusListener();
         Mockito.verify(messageBusMock).subscribe(Mockito.eq(EventTypes.EVENT_CONFIGURATION_VALUE_EDIT), Mockito.any());
@@ -288,7 +289,6 @@
 
     @Test
     public void initMessageBusListenerTriggersSetupRepetitiveAlertTypesOnAllowedKeyEdit() {
-        MessageBus messageBusMock = Mockito.mock(MessageBus.class);
         alertManagerImplMock.messageBus = messageBusMock;
         alertManagerImplMock.initMessageBusListener();
         ArgumentCaptor<MessageSubscriber> captor = ArgumentCaptor.forClass(MessageSubscriber.class);
@@ -300,7 +300,6 @@
 
     @Test
     public void initMessageBusListenerDoesNotTriggerSetupRepetitiveAlertTypesOnOtherKeyEdit() {
-        MessageBus messageBusMock = Mockito.mock(MessageBus.class);
         alertManagerImplMock.messageBus = messageBusMock;
         alertManagerImplMock.initMessageBusListener();
         ArgumentCaptor<MessageSubscriber> captor = ArgumentCaptor.forClass(MessageSubscriber.class);
