--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1355,10 +1355,7 @@
 
     @Test
     public void testCreateEdgeZone() {
-        CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
-        Mockito.when(cmd.isEdge()).thenReturn(true);
-        Mockito.when(cmd.getNetworkType()).thenReturn(NetworkType.Advanced.toString());
-        Mockito.when(cmd.getDomainId()).thenReturn(null);
+        CreateZoneCmd cmd = createMockCreateZoneCmd(true, NetworkType.Advanced.toString());
         mockPersistDatacenterForCreateZone();
         DataCenter zone = configurationMgr.createZone(cmd);
         Assert.assertNotNull(zone);
@@ -1368,10 +1365,7 @@
 
     @Test
     public void testCreateCoreZone() {
-        CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
-        Mockito.when(cmd.isEdge()).thenReturn(false);
-        Mockito.when(cmd.getNetworkType()).thenReturn(NetworkType.Advanced.toString());
-        Mockito.when(cmd.getDomainId()).thenReturn(null);
+        CreateZoneCmd cmd = createMockCreateZoneCmd(false, NetworkType.Advanced.toString());
         mockPersistDatacenterForCreateZone();
         DataCenter zone = configurationMgr.createZone(cmd);
         Assert.assertNotNull(zone);
@@ -1381,10 +1375,7 @@
 
     @Test
     public void testCreateBasicZone() {
-        CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
-        Mockito.when(cmd.isEdge()).thenReturn(false);
-        Mockito.when(cmd.getNetworkType()).thenReturn(NetworkType.Basic.toString());
-        Mockito.when(cmd.getDomainId()).thenReturn(null);
+        CreateZoneCmd cmd = createMockCreateZoneCmd(false, NetworkType.Basic.toString());
         mockPersistDatacenterForCreateZone();
         DataCenter zone = configurationMgr.createZone(cmd);
         Assert.assertNotNull(zone);
@@ -1393,10 +1384,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testCreateBasicEdgeZoneFailure() {
-        CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
-        Mockito.when(cmd.isEdge()).thenReturn(true);
-        Mockito.when(cmd.getNetworkType()).thenReturn(NetworkType.Basic.toString());
-        Mockito.when(cmd.getDomainId()).thenReturn(null);
+        CreateZoneCmd cmd = createMockCreateZoneCmd(true, NetworkType.Basic.toString());
         configurationMgr.createZone(cmd);
     }
 
@@ -1508,4 +1496,12 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static CreateZoneCmd createMockCreateZoneCmd(boolean isEdge, String networkType) {
+        CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
+        Mockito.when(cmd.isEdge()).thenReturn(isEdge);
+        Mockito.when(cmd.getNetworkType()).thenReturn(networkType);
+        Mockito.when(cmd.getDomainId()).thenReturn(null);
+        return cmd;
+    }
 }
