--- a/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
+++ b/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.cloudstack.mom.inmemory;
 
+import org.mockito.Mock;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.Mockito.mock;
@@ -36,6 +37,8 @@
 import com.cloud.utils.UuidUtils;
 
 public class InMemoryEventBusTest {
+    @Mock
+    private EventTopic topic;
 
     @Test
     public void testConfigure() throws Exception {
@@ -50,7 +53,6 @@
 
     @Test
     public void testSubscribe() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
         EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
@@ -77,7 +79,6 @@
 
     @Test(expected = EventBusException.class)
     public void testSubscribeFailSubscriber() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
@@ -86,7 +87,6 @@
 
     @Test
     public void testUnsubscribe() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
         EventSubscriber subscriber = mock(EventSubscriber.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
@@ -128,7 +128,6 @@
 
     @Test
     public void testPublish() throws Exception {
-        EventTopic topic = mock(EventTopic.class);
         EventSubscriber subscriber = mock(EventSubscriber.class);
         Event event = mock(Event.class);
 
