--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessorTest.java
@@ -52,6 +52,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class Xenserver625StorageProcessorTest {
+    @Mock
+    private PBD pbdMock;
 
     @InjectMocks
     private Xenserver625StorageProcessor xenserver625StorageProcessor;
@@ -245,7 +247,6 @@
 
     @Test
     public void forgetSrTest() throws XenAPIException, XmlRpcException {
-        PBD pbdMock = Mockito.mock(PBD.class);
         Set<PBD> pbds = new HashSet<>();
         pbds.add(pbdMock);
 
@@ -262,7 +263,6 @@
 
     @Test
     public void unplugPbdTest() throws XenAPIException, XmlRpcException {
-        PBD pbdMock = Mockito.mock(PBD.class);
 
         xenserver625StorageProcessor.unplugPbd(connectionMock, pbdMock);
 
@@ -381,7 +381,6 @@
                     Mockito.eq(pathMock), Mockito.eq("file"), Mockito.eq("file"), Mockito.eq(false),
                     Mockito.anyMap())).thenReturn(srMock);
 
-            PBD pbdMock = Mockito.mock(PBD.class);
             Mockito.when(PBD.create(Mockito.eq(connectionMock), Mockito.any(Record.class))).thenReturn(pbdMock);
 
             SR sr = xenserver625StorageProcessor.createNewFileSr(connectionMock, pathMock);
@@ -408,7 +407,6 @@
         SR srMock = Mockito.mock(SR.class);
         Mockito.doNothing().when(xenserver625StorageProcessor).forgetSr(connectionMock, srMock);
 
-        PBD pbdMock = Mockito.mock(PBD.class);
         Mockito.doNothing().when(xenserver625StorageProcessor).unplugPbd(connectionMock, pbdMock);
 
         xenserver625StorageProcessor.removeSrAndPbdIfPossible(connectionMock, srMock, pbdMock);
@@ -432,7 +430,6 @@
 
     @Test
     public void removeSrAndPbdIfPossiblePbdNotNullButSrNull() {
-        PBD pbdMock = Mockito.mock(PBD.class);
         Mockito.doNothing().when(xenserver625StorageProcessor).unplugPbd(connectionMock, pbdMock);
 
         xenserver625StorageProcessor.removeSrAndPbdIfPossible(connectionMock, null, pbdMock);
