--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -66,6 +66,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class XenServer610WrapperTest {
+    @Mock
+    private final SR sr1;
 
     @Mock
     protected XenServer610Resource xenServer610Resource;
@@ -262,7 +264,6 @@
         final VolumeTO volume1 = MockVolumeTO(path);
         final VolumeTO volume2 = MockVolumeTO(path);
 
-        final SR sr1 = Mockito.mock(SR.class);
         final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
@@ -369,7 +370,6 @@
         final VolumeTO volume1 = MockVolumeTO(path);
         final VolumeTO volume2 = MockVolumeTO(path);
 
-        final SR sr1 = Mockito.mock(SR.class);
         final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
