--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBaseTest.java
@@ -384,11 +384,9 @@
         Connection connectionMock = Mockito.mock(Connection.class);
 
         List<SR> srsMocks = new ArrayList<>();
-        SR srMock1 = Mockito.mock(SR.class);
-        SR srMock2 = Mockito.mock(SR.class);
-
-        Mockito.when(srMock1.getPhysicalSize(connectionMock)).thenReturn(0l);
-        Mockito.when(srMock2.getPhysicalSize(connectionMock)).thenReturn(100l);
+        SR srMock1 = createMockSr(connectionMock, 0l);
+        SR srMock2 = createMockSr(connectionMock, 100l);
+
         srsMocks.add(srMock1);
         srsMocks.add(srMock2);
 
@@ -616,4 +614,10 @@
         Assert.assertEquals(1024L, listAnswer.getSizes().get(0).longValue());
         Assert.assertEquals(123456789000L, listAnswer.getLastModified().get(0).longValue());
     }
+
+    private static SR createMockSr(Connection connectionMock, long physicalSize) {
+        SR srMock = Mockito.mock(SR.class);
+        Mockito.when(srMock.getPhysicalSize(connectionMock)).thenReturn(physicalSize);
+        return srMock;
+    }
 }
