--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/brocade/BrocadeVcsApiTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/brocade/BrocadeVcsApiTest.java
@@ -109,8 +109,7 @@
         method = mock(HttpPatch.class);
 
         response = mock(HttpResponse.class);
-        final StatusLine statusLine = mock(StatusLine.class);
-        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_NO_CONTENT);
+        final StatusLine statusLine = createMockStatusLine();
         when(response.getStatusLine()).thenReturn(statusLine);
 
         // Execute
@@ -129,8 +128,7 @@
         method = mock(HttpPatch.class);
 
         response = mock(HttpResponse.class);
-        final StatusLine statusLine = mock(StatusLine.class);
-        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_NO_CONTENT);
+        final StatusLine statusLine = createMockStatusLine();
         when(response.getStatusLine()).thenReturn(statusLine);
 
         // Execute
@@ -149,8 +147,7 @@
         method = mock(HttpPatch.class);
 
         response = mock(HttpResponse.class);
-        final StatusLine statusLine = mock(StatusLine.class);
-        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_NO_CONTENT);
+        final StatusLine statusLine = createMockStatusLine();
         when(response.getStatusLine()).thenReturn(statusLine);
 
         // Execute
@@ -169,8 +166,7 @@
         method = mock(HttpPatch.class);
 
         response = mock(HttpResponse.class);
-        final StatusLine statusLine = mock(StatusLine.class);
-        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_NO_CONTENT);
+        final StatusLine statusLine = createMockStatusLine();
         when(response.getStatusLine()).thenReturn(statusLine);
 
         // Execute
@@ -182,4 +178,10 @@
         assertEquals("Wrong HTTP method for Network creation REST service", "patch", type);
     }
 
+
+    private static StatusLine createMockStatusLine() {
+        final StatusLine statusLine = mock(StatusLine.class);
+        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_NO_CONTENT);
+        return statusLine;
+    }
 }
