--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -220,6 +220,8 @@
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenServiceImplTest {
     @Mock
+    private TagType tagtype;
+    @Mock
     MessageBus messageBus;
     @Mock
     ProjectDao projectDao;
@@ -1095,7 +1097,6 @@
     @Test
     public void createTungstenTagTypeTest() {
         TungstenAnswer createTungstenTagTypeAnswer = MockTungstenAnswerFactory.get(true);
-        TagType tagtype = mock(TagType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenTagTypeCommand.class), anyLong())).thenReturn(createTungstenTagTypeAnswer);
@@ -1134,7 +1135,6 @@
     @Test
     public void listTungstenTagTypesTest() {
         TungstenAnswer listTungstenTagTypeAnswer = MockTungstenAnswerFactory.get(true);
-        TagType tagtype = mock(TagType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
