--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/resource/BrocadeVcsResourceTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/resource/BrocadeVcsResourceTest.java
@@ -114,10 +114,9 @@
         List<VcsNodeInfo> nodes = new ArrayList<VcsNodeInfo>();
         nodes.add(nodeInfo);
 
-        final VcsNodes vcsNodes = mock(VcsNodes.class);
         final Output output = mock(Output.class);
+         final VcsNodes vcsNodes = createMockVcsNodes(nodes);
         when(output.getVcsNodes()).thenReturn(vcsNodes);
-        when(vcsNodes.getVcsNodeInfo()).thenReturn(nodes);
         when(api.getSwitchStatus()).thenReturn(output);
 
         final PingCommand ping = resource.getCurrentStatus(42);
@@ -137,10 +136,9 @@
         List<VcsNodeInfo> nodes = new ArrayList<VcsNodeInfo>();
         nodes.add(nodeInfo);
 
-        final VcsNodes vcsNodes = mock(VcsNodes.class);
+        final VcsNodes vcsNodes = createMockVcsNodes(nodes);
         final Output output = mock(Output.class);
         when(output.getVcsNodes()).thenReturn(vcsNodes);
-        when(vcsNodes.getVcsNodeInfo()).thenReturn(nodes);
         when(api.getSwitchStatus()).thenReturn(output);
 
         final PingCommand ping = resource.getCurrentStatus(42);
@@ -258,4 +256,10 @@
         final DisassociateMacFromNetworkAnswer answer = (DisassociateMacFromNetworkAnswer)resource.executeRequest(cmd);
         assertFalse(answer.getResult());
     }
+
+    private static VcsNodes createMockVcsNodes(List<VcsNodeInfo> nodes) {
+        final VcsNodes vcsNodes = mock(VcsNodes.class);
+        when(vcsNodes.getVcsNodeInfo()).thenReturn(nodes);
+        return vcsNodes;
+    }
 }
