--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -66,6 +66,8 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class XenServer610WrapperTest {
+    @Mock
+    private final VDI vdi2;
 
     @Mock
     protected XenServer610Resource xenServer610Resource;
@@ -266,7 +268,6 @@
         final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
-        final VDI vdi2 = Mockito.mock(VDI.class);
 
         final NicTO nic1 = Mockito.mock(NicTO.class);
         final NicTO nic2 = Mockito.mock(NicTO.class);
@@ -373,7 +374,6 @@
         final SR sr2 = Mockito.mock(SR.class);
 
         final VDI vdi1 = Mockito.mock(VDI.class);
-        final VDI vdi2 = Mockito.mock(VDI.class);
 
         final NicTO nic1 = Mockito.mock(NicTO.class);
         final NicTO nic2 = Mockito.mock(NicTO.class);
