--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/codec/ExchangeCodecTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/codec/ExchangeCodecTest.java
@@ -79,6 +79,10 @@
         return serialization;
     }
 
+    private Channel newMockChannel() {
+        return Mockito.mock(Channel.class);
+    }
+
     private Object decode(byte[] request) throws IOException {
         ChannelBuffer buffer = ChannelBuffers.wrappedBuffer(request);
         AbstractMockChannel channel = getServerSideChannel(url);
@@ -145,7 +149,7 @@
 
     @Test
     void test_Decode_Error_Length() throws IOException {
-        DefaultFuture future = DefaultFuture.newFuture(Mockito.mock(Channel.class), new Request(0), 100000, null);
+        DefaultFuture future = DefaultFuture.newFuture(newMockChannel(), new Request(0), 100000, null);
 
         byte[] header = new byte[] {MAGIC_HIGH, MAGIC_LOW, SERIALIZATION_BYTE, 20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
         Person person = new Person();
@@ -240,7 +244,7 @@
 
     @Test
     void test_Decode_Return_Response_Person() throws IOException {
-        DefaultFuture future = DefaultFuture.newFuture(Mockito.mock(Channel.class), new Request(0), 100000, null);
+        DefaultFuture future = DefaultFuture.newFuture(newMockChannel(), new Request(0), 100000, null);
 
         // 00000010-response/oneway/heartbeat=false/hessian |20-stats=ok|id=0|length=0
         byte[] header = new byte[] {MAGIC_HIGH, MAGIC_LOW, SERIALIZATION_BYTE, 20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
@@ -351,7 +355,7 @@
 
     @Test
     void test_Header_Response_NoSerializationFlag() throws IOException {
-        DefaultFuture future = DefaultFuture.newFuture(Mockito.mock(Channel.class), new Request(0), 100000, null);
+        DefaultFuture future = DefaultFuture.newFuture(newMockChannel(), new Request(0), 100000, null);
 
         // 00000010-response/oneway/heartbeat=false/noset |20-stats=ok|id=0|length=0
         byte[] header = new byte[] {MAGIC_HIGH, MAGIC_LOW, SERIALIZATION_BYTE, 20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
@@ -367,7 +371,7 @@
 
     @Test
     void test_Header_Response_Heartbeat() throws IOException {
-        DefaultFuture future = DefaultFuture.newFuture(Mockito.mock(Channel.class), new Request(0), 100000, null);
+        DefaultFuture future = DefaultFuture.newFuture(newMockChannel(), new Request(0), 100000, null);
 
         // 00000010-response/oneway/heartbeat=true |20-stats=ok|id=0|length=0
         byte[] header = new byte[] {MAGIC_HIGH, MAGIC_LOW, SERIALIZATION_BYTE, 20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
@@ -405,7 +409,7 @@
     @Test
     @Disabled("Event should not be object.")
     void test_Encode_Response() throws IOException {
-        DefaultFuture future = DefaultFuture.newFuture(Mockito.mock(Channel.class), new Request(1001), 100000, null);
+        DefaultFuture future = DefaultFuture.newFuture(newMockChannel(), new Request(1001), 100000, null);
 
         ChannelBuffer encodeBuffer = ChannelBuffers.dynamicBuffer(1024);
         Channel channel = getClientSideChannel(url);
