--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
@@ -175,9 +175,7 @@
 
             List<Instance> instances = new ArrayList<>();
             instances.add(instance);
-            when(namingService.getAllInstances(
-                            serviceName, this.registryUrl.getParameter(GROUP_KEY, Constants.DEFAULT_GROUP)))
-                    .thenReturn(instances);
+            stubAllInstances(namingService, serviceName, instances);
         } catch (NacosException e) {
             // ignore
         }
@@ -214,9 +212,7 @@
 
             List<Instance> instances = new ArrayList<>();
             instances.add(instance);
-            when(namingService.getAllInstances(
-                            serviceName, this.registryUrl.getParameter(GROUP_KEY, Constants.DEFAULT_GROUP)))
-                    .thenReturn(instances);
+            stubAllInstances(namingService, serviceName, instances);
 
         } catch (NacosException e) {
             // ignore
@@ -259,9 +255,7 @@
 
             List<Instance> instances = new ArrayList<>();
             instances.add(instance);
-            when(namingService.getAllInstances(
-                            serviceName, this.registryUrl.getParameter(GROUP_KEY, Constants.DEFAULT_GROUP)))
-                    .thenReturn(instances);
+            stubAllInstances(namingService, serviceName, instances);
 
             String serviceNameWithoutVersion = "providers:org.apache.dubbo.registry.nacos.NacosService:default";
             String serviceName1 = "providers:org.apache.dubbo.registry.nacos.NacosService:1.0.0:default";
@@ -309,4 +303,11 @@
         Assertions.assertEquals(2, registered.size());
         Assertions.assertEquals(1, subscribed.get(serviceUrlWithWildcard).size());
     }
+
+    private void stubAllInstances(NamingService namingService, String serviceName, List<Instance> instances)
+            throws NacosException {
+        when(namingService.getAllInstances(
+                        serviceName, this.registryUrl.getParameter(GROUP_KEY, Constants.DEFAULT_GROUP)))
+                .thenReturn(instances);
+    }
 }
