--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
@@ -39,7 +39,7 @@
     void testSubscribe() throws NacosException {
         NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
         NamingService namingService = Mockito.mock(NamingService.class);
-        Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
+        stubNamingService(connectionManager, namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
@@ -560,7 +560,7 @@
         CountDownLatch startLatch = new CountDownLatch(1);
         CountDownLatch stopLatch = new CountDownLatch(1);
         NamingService namingService = Mockito.mock(NamingService.class);
-        Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
+        stubNamingService(connectionManager, namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper =
                 new NacosNamingServiceWrapper(connectionManager, false, 0, 0);
@@ -748,4 +748,8 @@
             Assertions.fail(e);
         }
     }
+
+    private void stubNamingService(NacosConnectionManager connectionManager, NamingService namingService) {
+        Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
+    }
 }
