--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
@@ -186,7 +186,7 @@
                 new NacosNamingServiceWrapper(new NacosConnectionManager(namingService), 0, 0);
         nacosRegistry = new NacosRegistry(this.registryUrl, nacosNamingServiceWrapper);
 
-        NotifyListener listener = mock(NotifyListener.class);
+        NotifyListener listener = createNotifyListener();
         nacosRegistry.subscribe(serviceUrl, listener);
 
         Map<URL, Set<NotifyListener>> subscribed = nacosRegistry.getSubscribed();
@@ -226,7 +226,7 @@
                 new NacosNamingServiceWrapper(new NacosConnectionManager(namingService), 0, 0);
         nacosRegistry = new NacosRegistry(this.registryUrl, nacosNamingServiceWrapper);
 
-        NotifyListener listener = mock(NotifyListener.class);
+        NotifyListener listener = createNotifyListener();
         nacosRegistry.subscribe(serviceUrl, listener);
 
         Map<URL, Set<NotifyListener>> subscribed = nacosRegistry.getSubscribed();
@@ -297,7 +297,7 @@
                 + "?interface=org.apache.dubbo.registry.nacos.NacosService"
                 + "&notify=false&methods=test1,test2&category=providers&version=1.0.0&group=default");
 
-        NotifyListener listener = mock(NotifyListener.class);
+        NotifyListener listener = createNotifyListener();
         nacosRegistry.subscribe(serviceUrlWithWildcard, listener);
         nacosRegistry.subscribe(serviceUrlWithOutWildcard, listener);
 
@@ -309,4 +309,8 @@
         Assertions.assertEquals(2, registered.size());
         Assertions.assertEquals(1, subscribed.get(serviceUrlWithWildcard).size());
     }
+
+    private NotifyListener createNotifyListener() {
+        return mock(NotifyListener.class);
+    }
 }
