--- a/processing/src/test/java/org/apache/druid/java/util/metrics/OshiSysMonitorTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/OshiSysMonitorTest.java
@@ -93,9 +93,7 @@
   public void testMemStats()
   {
     StubServiceEmitter emitter = new StubServiceEmitter("dev/monitor-test", "localhost:0000");
-    GlobalMemory mem = Mockito.mock(GlobalMemory.class);
-    Mockito.when(mem.getTotal()).thenReturn(64L);
-    Mockito.when(mem.getAvailable()).thenReturn(16L);
+    GlobalMemory mem = createMockGlobalMemory();
     Mockito.when(hal.getMemory()).thenReturn(mem);
 
     OshiSysMonitor m = createMonitor(si);
@@ -115,9 +113,7 @@
   public void testMemStatsSkipOthers()
   {
     StubServiceEmitter emitter = new StubServiceEmitter("dev/monitor-test", "localhost:0000");
-    GlobalMemory mem = Mockito.mock(GlobalMemory.class);
-    Mockito.when(mem.getTotal()).thenReturn(64L);
-    Mockito.when(mem.getAvailable()).thenReturn(16L);
+    GlobalMemory mem = createMockGlobalMemory();
     Mockito.when(hal.getMemory()).thenReturn(mem);
 
     OshiSysMonitor m = createMonitor(si, ImmutableList.of("mem"));
@@ -649,4 +645,11 @@
   {
     return new OshiSysMonitor(new OshiSysMonitorConfig(categories), si);
   }
+
+    private static GlobalMemory createMockGlobalMemory() {
+        GlobalMemory mem = Mockito.mock(GlobalMemory.class);
+        Mockito.when(mem.getAvailable()).thenReturn(16L);
+        Mockito.when(mem.getTotal()).thenReturn(64L);
+        return mem;
+    }
 }
