--- a/multi-stage-query/src/test/java/org/apache/druid/msq/dart/controller/DartControllerRegistryTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/dart/controller/DartControllerRegistryTest.java
@@ -152,10 +152,9 @@
   {
     final DartControllerRegistry registry = new DartControllerRegistry(makeConfig(10, Period.hours(1)));
     final Controller controller = Mockito.mock(Controller.class);
-    final ControllerContext controllerContext = Mockito.mock(ControllerContext.class);
     Mockito.when(controller.queryId()).thenReturn("dart1");
+    final ControllerContext controllerContext = createMockControllerContext();
     Mockito.when(controller.getControllerContext()).thenReturn(controllerContext);
-    Mockito.when(controllerContext.selfNode()).thenReturn(Mockito.mock(DruidNode.class));
 
     // Set up live reports
     final TaskReport.ReportMap liveReportMap = new TaskReport.ReportMap();
@@ -189,10 +188,9 @@
   {
     final DartControllerRegistry registry = new DartControllerRegistry(makeConfig(10, Period.hours(1)));
     final Controller controller = Mockito.mock(Controller.class);
-    final ControllerContext controllerContext = Mockito.mock(ControllerContext.class);
+    final ControllerContext controllerContext = createMockControllerContext();
     Mockito.when(controller.queryId()).thenReturn("dart1");
     Mockito.when(controller.getControllerContext()).thenReturn(controllerContext);
-    Mockito.when(controllerContext.selfNode()).thenReturn(Mockito.mock(DruidNode.class));
     Mockito.when(controller.liveReports()).thenReturn(null);
 
     final ControllerHolder holder = new ControllerHolder(
@@ -310,4 +308,10 @@
       }
     };
   }
+
+    private static ControllerContext createMockControllerContext() {
+        final ControllerContext controllerContext = Mockito.mock(ControllerContext.class);
+        Mockito.when(controllerContext.selfNode()).thenReturn(Mockito.mock(DruidNode.class));
+        return controllerContext;
+    }
 }
