--- a/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
+++ b/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.server.initialization.jetty;
 
+import org.mockito.Mock;
 import com.google.inject.Binding;
 import com.google.inject.Injector;
 import com.google.inject.Provider;
@@ -41,6 +42,8 @@
 
 public class JettyServerModuleTest
 {
+    @Mock
+    private JettyServerInitializer initializer;
   @Test
   public void testJettyServerModule()
   {
@@ -101,7 +104,6 @@
     Injector injector = Mockito.mock(Injector.class);
     TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
-    JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
 
     Server server = JettyServerModule.makeAndInitializeServer(
@@ -143,7 +145,6 @@
     Injector injector = Mockito.mock(Injector.class);
     TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
-    JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
 
     // Create a custom SslContextFactory via binding
@@ -207,7 +208,6 @@
     Injector injector = Mockito.mock(Injector.class);
     TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
-    JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
 
     // Create a custom SslContextFactory via binding
