--- a/processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/HashVectorGrouperTest.java
+++ b/processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/HashVectorGrouperTest.java
@@ -97,9 +97,8 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
-
+
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
     int startingNumBuckets = 4;
     int maxBuckets = 16;
     final int bufferSize = (keySize + aggSize) * maxBuckets;
@@ -138,9 +137,8 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
-
+
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
     int startingNumBuckets = 4;
     int maxBuckets = 32;
     final int bufferSize = (keySize + aggSize) * maxBuckets;
@@ -185,9 +183,8 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
-
+
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
     int startingNumBuckets = 4;
     int maxBuckets = 64;
     final int bufferSize = (keySize + aggSize) * maxBuckets;
@@ -239,9 +236,8 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
-
+
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
     int startingNumBuckets = 4;
     int maxBuckets = 128;
     final int bufferSize = (keySize + aggSize) * maxBuckets;
@@ -299,4 +295,10 @@
       keySpace.putInt(((long) Integer.BYTES * i), bucket);
     }
   }
+
+    private static AggregatorAdapters createMockAggregatorAdapters(int spaceNeeded) {
+        final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
+        Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(spaceNeeded);
+        return aggregatorAdapters;
+    }
 }
