--- a/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
+++ b/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.server.initialization.jetty;
 
+import org.mockito.Mock;
 import com.google.inject.Binding;
 import com.google.inject.Injector;
 import com.google.inject.Provider;
@@ -41,6 +42,8 @@
 
 public class JettyServerModuleTest
 {
+    @Mock
+    private TLSCertificateChecker certificateChecker;
   @Test
   public void testJettyServerModule()
   {
@@ -99,7 +102,6 @@
     ServerConfig serverConfig = new ServerConfig();
     Lifecycle lifecycle = new Lifecycle();
     Injector injector = Mockito.mock(Injector.class);
-    TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
     JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
@@ -141,7 +143,6 @@
     ServerConfig serverConfig = new ServerConfig();
     Lifecycle lifecycle = new Lifecycle();
     Injector injector = Mockito.mock(Injector.class);
-    TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
     JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
@@ -205,7 +206,6 @@
     ServerConfig serverConfig = new ServerConfig();
     Lifecycle lifecycle = new Lifecycle();
     Injector injector = Mockito.mock(Injector.class);
-    TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
     JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
