--- a/processing/src/test/java/org/apache/druid/query/aggregation/any/DoubleAnyAggregatorFactoryTest.java
+++ b/processing/src/test/java/org/apache/druid/query/aggregation/any/DoubleAnyAggregatorFactoryTest.java
@@ -80,7 +80,7 @@
   @Test
   public void factorizeVectorForNumericTypeShouldReturnDoubleVectorAggregator()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    selectorFactory = MockVectorColumnSelectorFactory.createMockVectorColumnSelectorFactory(FIELD_NAME, capabilities, valueSelector);
     Mockito.doReturn(true).when(capabilities).isNumeric();
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
@@ -90,10 +90,29 @@
   @Test
   public void factorizeVectorForStringTypeShouldReturnDoubleVectorAggregatorWithNilSelector()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    selectorFactory = MockVectorColumnSelectorFactory.createMockVectorColumnSelectorFactory(FIELD_NAME, capabilities, valueSelector);
     Mockito.doReturn(false).when(capabilities).isNumeric();
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertNull(aggregator.get(BUFFER, POSITION));
   }
+
+    public final class MockVectorColumnSelectorFactory {
+        private MockVectorColumnSelectorFactory() {
+        }
+
+        public static org.apache.druid.segment.vector.VectorColumnSelectorFactory createMockVectorColumnSelectorFactory(
+                String fieldName,
+                org.apache.druid.segment.column.ColumnCapabilities capabilities,
+                org.apache.druid.segment.vector.VectorValueSelector valueSelector) {
+            org.apache.druid.segment.vector.VectorColumnSelectorFactory selectorFactory =
+                    Mockito.mock(org.apache.druid.segment.vector.VectorColumnSelectorFactory.class);
+
+            Mockito.doReturn(null).when(selectorFactory).getColumnCapabilities(fieldName);
+            Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(fieldName);
+            Mockito.doReturn(valueSelector).when(selectorFactory).makeValueSelector(fieldName);
+
+            return selectorFactory;
+        }
+    }
 }
--- a/processing/src/test/java/org/apache/druid/query/aggregation/any/FloatAnyAggregatorFactoryTest.java
+++ b/processing/src/test/java/org/apache/druid/query/aggregation/any/FloatAnyAggregatorFactoryTest.java
@@ -80,7 +80,7 @@
   @Test
   public void factorizeVectorForNumericTypeShouldReturnFloatVectorAggregator()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    selectorFactory = MockVectorColumnSelectorFactory.createMockVectorColumnSelectorFactory(FIELD_NAME, capabilities, valueSelector);
     Mockito.doReturn(true).when(capabilities).isNumeric();
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
@@ -92,8 +92,27 @@
   {
     Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
     Mockito.doReturn(false).when(capabilities).isNumeric();
-    VectorAggregator aggregator = target.factorizeVector(selectorFactory);
+    selectorFactory = MockVectorColumnSelectorFactory.createMockVectorColumnSelectorFactory(FIELD_NAME, capabilities, valueSelector);
     Assert.assertNotNull(aggregator);
     Assert.assertNull(aggregator.get(BUFFER, POSITION));
   }
+
+    public final class MockVectorColumnSelectorFactory {
+        private MockVectorColumnSelectorFactory() {
+        }
+
+        public static org.apache.druid.segment.vector.VectorColumnSelectorFactory createMockVectorColumnSelectorFactory(
+                String fieldName,
+                org.apache.druid.segment.column.ColumnCapabilities capabilities,
+                org.apache.druid.segment.vector.VectorValueSelector valueSelector) {
+            org.apache.druid.segment.vector.VectorColumnSelectorFactory selectorFactory =
+                    Mockito.mock(org.apache.druid.segment.vector.VectorColumnSelectorFactory.class);
+
+            Mockito.doReturn(null).when(selectorFactory).getColumnCapabilities(fieldName);
+            Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(fieldName);
+            Mockito.doReturn(valueSelector).when(selectorFactory).makeValueSelector(fieldName);
+
+            return selectorFactory;
+        }
+    }
 }
--- a/processing/src/test/java/org/apache/druid/query/aggregation/any/LongAnyAggregatorFactoryTest.java
+++ b/processing/src/test/java/org/apache/druid/query/aggregation/any/LongAnyAggregatorFactoryTest.java
@@ -80,7 +80,7 @@
   @Test
   public void factorizeVectorWithNumericColumnShouldReturnLongVectorAggregator()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    selectorFactory = MockVectorColumnSelectorFactory.createMockVectorColumnSelectorFactory(FIELD_NAME, capabilities, valueSelector);
     Mockito.doReturn(true).when(capabilities).isNumeric();
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
@@ -90,10 +90,29 @@
   @Test
   public void factorizeVectorForStringTypeShouldReturnLongVectorAggregatorWithNilSelector()
   {
-    Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(FIELD_NAME);
+    selectorFactory = MockVectorColumnSelectorFactory.createMockVectorColumnSelectorFactory(FIELD_NAME, capabilities, valueSelector);
     Mockito.doReturn(false).when(capabilities).isNumeric();
     VectorAggregator aggregator = target.factorizeVector(selectorFactory);
     Assert.assertNotNull(aggregator);
     Assert.assertNull(aggregator.get(BUFFER, POSITION));
   }
+
+    public final class MockVectorColumnSelectorFactory {
+        private MockVectorColumnSelectorFactory() {
+        }
+
+        public static org.apache.druid.segment.vector.VectorColumnSelectorFactory createMockVectorColumnSelectorFactory(
+                String fieldName,
+                org.apache.druid.segment.column.ColumnCapabilities capabilities,
+                org.apache.druid.segment.vector.VectorValueSelector valueSelector) {
+            org.apache.druid.segment.vector.VectorColumnSelectorFactory selectorFactory =
+                    Mockito.mock(org.apache.druid.segment.vector.VectorColumnSelectorFactory.class);
+
+            Mockito.doReturn(null).when(selectorFactory).getColumnCapabilities(fieldName);
+            Mockito.doReturn(capabilities).when(selectorFactory).getColumnCapabilities(fieldName);
+            Mockito.doReturn(valueSelector).when(selectorFactory).makeValueSelector(fieldName);
+
+            return selectorFactory;
+        }
+    }
 }
