--- a/processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/HashVectorGrouperTest.java
+++ b/processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/HashVectorGrouperTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.query.groupby.epinephelinae;
 
+import org.mockito.Mock;
 import com.google.common.base.Suppliers;
 import org.apache.datasketches.memory.WritableMemory;
 import org.apache.druid.query.aggregation.AggregatorAdapters;
@@ -30,11 +31,12 @@
 
 public class HashVectorGrouperTest
 {
+    @Mock
+    private final AggregatorAdapters aggregatorAdapters;
   @Test
   public void testCloseAggregatorAdaptorsShouldBeClosed()
   {
     final ByteBuffer buffer = ByteBuffer.wrap(new byte[4096]);
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
     final HashVectorGrouper grouper = new HashVectorGrouper(
         Suppliers.ofInstance(buffer),
         1024,
@@ -56,7 +58,6 @@
     final int bufferSize = 100 * 1024;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
     final ByteBuffer buffer = ByteBuffer.wrap(new byte[bufferSize]);
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
     final HashVectorGrouper grouper = new HashVectorGrouper(
         Suppliers.ofInstance(buffer),
         keySize,
@@ -76,7 +77,6 @@
     final int keySize = 10000;
     final int bufferSize = 100 * 1024;
     final ByteBuffer buffer = ByteBuffer.wrap(new byte[bufferSize]);
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
     final HashVectorGrouper grouper = new HashVectorGrouper(
         Suppliers.ofInstance(buffer),
         keySize,
