--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
@@ -52,9 +52,8 @@
   @Test
   public void testArchiveSegmentUnknowType()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
 
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(null);
     final OmniDataSegmentArchiver segmentArchiver = injector.getInstance(OmniDataSegmentArchiver.class);
     Assert.assertThrows(
@@ -119,4 +118,15 @@
       return null;
     }
   }
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static DataSegment createMockDataSegment() {
+            final DataSegment segment = Mockito.mock(DataSegment.class);
+            Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+            return segment;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
@@ -63,9 +63,8 @@
   @Test
   public void testKillSegmentUnknowType()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
-
+
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(null);
     final OmniDataSegmentKiller segmentKiller = injector.getInstance(OmniDataSegmentKiller.class);
     Assert.assertThrows(
@@ -204,4 +203,15 @@
     }
   }
 
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static DataSegment createMockDataSegment() {
+            final DataSegment segment = Mockito.mock(DataSegment.class);
+            Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+            return segment;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
@@ -54,9 +54,8 @@
   @Test
   public void testMoveSegmentUnknownType()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
 
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(null);
     final OmniDataSegmentMover segmentMover = injector.getInstance(OmniDataSegmentMover.class);
     Assert.assertThrows(
@@ -115,4 +114,15 @@
       return null;
     }
   }
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static DataSegment createMockDataSegment() {
+            final DataSegment segment = Mockito.mock(DataSegment.class);
+            Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+            return segment;
+        }
+    }
 }
