--- a/multi-stage-query/src/test/java/org/apache/druid/msq/dart/controller/DartControllerRegistryTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/dart/controller/DartControllerRegistryTest.java
@@ -151,10 +151,8 @@
   public void test_getQueryInfoAndReport_runningQuery()
   {
     final DartControllerRegistry registry = new DartControllerRegistry(makeConfig(10, Period.hours(1)));
-    final Controller controller = Mockito.mock(Controller.class);
     final ControllerContext controllerContext = Mockito.mock(ControllerContext.class);
-    Mockito.when(controller.queryId()).thenReturn("dart1");
-    Mockito.when(controller.getControllerContext()).thenReturn(controllerContext);
+    final Controller controller = createMockController(controllerContext);
     Mockito.when(controllerContext.selfNode()).thenReturn(Mockito.mock(DruidNode.class));
 
     // Set up live reports
@@ -188,10 +186,8 @@
   public void test_getQueryInfoAndReport_runningQuery_noLiveReports()
   {
     final DartControllerRegistry registry = new DartControllerRegistry(makeConfig(10, Period.hours(1)));
-    final Controller controller = Mockito.mock(Controller.class);
     final ControllerContext controllerContext = Mockito.mock(ControllerContext.class);
-    Mockito.when(controller.queryId()).thenReturn("dart1");
-    Mockito.when(controller.getControllerContext()).thenReturn(controllerContext);
+    final Controller controller = createMockController(controllerContext);
     Mockito.when(controllerContext.selfNode()).thenReturn(Mockito.mock(DruidNode.class));
     Mockito.when(controller.liveReports()).thenReturn(null);
 
@@ -310,4 +306,11 @@
       }
     };
   }
+
+    private static Controller createMockController(final ControllerContext controllerContext) {
+        final Controller controller = Mockito.mock(Controller.class);
+        Mockito.when(controller.queryId()).thenReturn("dart1");
+        Mockito.when(controller.getControllerContext()).thenReturn(controllerContext);
+        return controller;
+    }
 }
