--- a/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/CostBasedAutoScalerTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/CostBasedAutoScalerTest.java
@@ -442,12 +442,11 @@
     SupervisorSpec spec = Mockito.mock(SupervisorSpec.class);
     SeekableStreamSupervisor supervisor = Mockito.mock(SeekableStreamSupervisor.class);
     ServiceEmitter emitter = Mockito.mock(ServiceEmitter.class);
-    SeekableStreamSupervisorIOConfig ioConfig = Mockito.mock(SeekableStreamSupervisorIOConfig.class);
 
     when(spec.getId()).thenReturn("s-up");
     when(spec.getDataSources()).thenReturn(List.of("test-datasource"));
+    SeekableStreamSupervisorIOConfig ioConfig = createMockSeekableStreamSupervisorIOConfig("stream");
     when(supervisor.getIoConfig()).thenReturn(ioConfig);
-    when(ioConfig.getStream()).thenReturn("stream");
 
     CostBasedAutoScalerConfig cfgWithDefaults = CostBasedAutoScalerConfig.builder()
                                                                          .taskCountMax(10)
@@ -487,13 +486,12 @@
     SupervisorSpec spec = Mockito.mock(SupervisorSpec.class);
     SeekableStreamSupervisor supervisor = Mockito.mock(SeekableStreamSupervisor.class);
     ServiceEmitter emitter = Mockito.mock(ServiceEmitter.class);
-    SeekableStreamSupervisorIOConfig ioConfig = Mockito.mock(SeekableStreamSupervisorIOConfig.class);
 
     when(spec.getId()).thenReturn("test-supervisor");
     when(spec.getDataSources()).thenReturn(List.of("test-datasource"));
     when(spec.isSuspended()).thenReturn(false);
+     SeekableStreamSupervisorIOConfig ioConfig = createMockSeekableStreamSupervisorIOConfig("test-stream");
     when(supervisor.getIoConfig()).thenReturn(ioConfig);
-    when(ioConfig.getStream()).thenReturn("test-stream");
     when(ioConfig.getTaskDuration()).thenReturn(Duration.standardHours(1));
     when(supervisor.computeLagStats()).thenReturn(new LagStats(100, 100, 100));
     // No task stats means the moving average rate is unavailable
@@ -627,4 +625,10 @@
     taskStats.put("movingAverages", movingAverages);
     return taskStats;
   }
+
+    private static SeekableStreamSupervisorIOConfig createMockSeekableStreamSupervisorIOConfig(String streamReturn) {
+        SeekableStreamSupervisorIOConfig ioConfig = Mockito.mock(SeekableStreamSupervisorIOConfig.class);
+        when(ioConfig.getStream()).thenReturn(streamReturn);
+        return ioConfig;
+    }
 }
