--- a/indexing-service/src/test/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfoTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfoTest.java
@@ -239,10 +239,8 @@
     // Parallel index task
     TaskResource taskResource0 = mock(TaskResource.class);
     when(taskResource0.getRequiredCapacity()).thenReturn(3);
-    Task parallelIndexTask = mock(ParallelIndexSupervisorTask.class);
-    when(parallelIndexTask.getType()).thenReturn(ParallelIndexSupervisorTask.TYPE);
-    when(parallelIndexTask.getTaskResource()).thenReturn(taskResource0);
-
+
+    Task parallelIndexTask = createMockTask(ParallelIndexSupervisorTask.TYPE, taskResource0);
     // Since task satisifies parallel and total slot constraints, can run
     Assert.assertTrue(workerInfo.canRunTask(parallelIndexTask, 0.5));
 
@@ -253,10 +251,8 @@
     // Some other indexing task
     TaskResource taskResource1 = mock(TaskResource.class);
     when(taskResource1.getRequiredCapacity()).thenReturn(5);
-    Task anyOtherTask = mock(IndexTask.class);
-    when(anyOtherTask.getType()).thenReturn("index");
-    when(anyOtherTask.getTaskResource()).thenReturn(taskResource1);
-
+
+    Task anyOtherTask = createMockTask("index", taskResource1);
     // Not a parallel index task ->  satisfies parallel index constraint
     // But does not satisfy the total slot constraint and cannot run (11 > 10)
     Assert.assertFalse(workerInfo.canRunTask(anyOtherTask, 0.5));
@@ -266,10 +262,8 @@
     TaskResource taskResource2 = mock(TaskResource.class);
     when(taskResource2.getRequiredCapacity()).thenReturn(1);
     when(taskResource2.getAvailabilityGroup()).thenReturn("grp1");
-    Task grp1Task = mock(IndexTask.class);
-    when(grp1Task.getType()).thenReturn("blah");
-    when(grp1Task.getTaskResource()).thenReturn(taskResource2);
-
+
+    Task grp1Task = createMockTask("blah", taskResource2);
     // Satisifies parallel index and total index slot constraints but cannot run due availability
     Assert.assertFalse(workerInfo.canRunTask(grp1Task, 0.3));
   }
@@ -284,4 +278,12 @@
       Assert.assertNotEquals(o1.hashCode(), o2.hashCode());
     }
   }
+
+    private static Task createMockTask(String taskType, TaskResource taskResource)
+    {
+        Task task = mock(Task.class);
+        when(task.getTaskResource()).thenReturn(taskResource);
+        when(task.getType()).thenReturn(taskType);
+        return task;
+    }
 }
