--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
@@ -81,10 +81,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = createMockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -130,10 +128,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(false);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = createMockTaskConfig(false, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -175,10 +171,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = createMockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -214,10 +208,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = createMockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -251,10 +243,8 @@
     TaskLogPusher pusher = mock(TaskLogPusher.class);
     when(toolbox.getTaskLogPusher()).thenReturn(pusher);
 
-    TaskConfig config = mock(TaskConfig.class);
-    when(config.isEncapsulatedTask()).thenReturn(true);
-    File folder = temporaryFolder.newFolder();
-    when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+    File folder = temporaryFolder.newFolder();
+    TaskConfig config = createMockTaskConfig(true, folder);
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
@@ -324,4 +314,11 @@
 
     Assert.assertNotSame(builder1, builder2);
   }
+
+    private static TaskConfig createMockTaskConfig(boolean encapsulatedTask, File folder) {
+        TaskConfig config = mock(TaskConfig.class);
+        when(config.isEncapsulatedTask()).thenReturn(encapsulatedTask);
+        when(config.getTaskDir(eq("myID"))).thenReturn(folder);
+        return config;
+    }
 }
