--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/ProcessingBuffersSetTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/ProcessingBuffersSetTest.java
@@ -106,11 +106,10 @@
   {
     // Create a mock StageDefinition and StageProcessor
     final StageDefinition stageDef = Mockito.mock(StageDefinition.class);
-    final StageProcessor<?, ?> stageProcessor = Mockito.mock(StageProcessor.class);
 
     // Configure mocks: processor factory does not use processing buffers
+    final StageProcessor<?, ?> stageProcessor = createStageProcessor(false);
     Mockito.when(stageDef.getProcessor()).thenReturn(stageProcessor);
-    Mockito.when(stageProcessor.usesProcessingBuffers()).thenReturn(false);
 
     // Create a ProcessingBuffersSet
     final ProcessingBuffersSet buffersSet =
@@ -128,11 +127,10 @@
   {
     // Create a mock StageDefinition and StageProcessor
     final StageDefinition stageDef = Mockito.mock(StageDefinition.class);
-    final StageProcessor<?, ?> stageProcessor = Mockito.mock(StageProcessor.class);
 
     // Configure mocks: processor factory does use processing buffers
+    final StageProcessor<?, ?> stageProcessor = createStageProcessor(true);
     Mockito.when(stageDef.getProcessor()).thenReturn(stageProcessor);
-    Mockito.when(stageProcessor.usesProcessingBuffers()).thenReturn(true);
 
     // Create a ProcessingBuffersSet
     final ProcessingBuffersSet buffersSet =
@@ -146,4 +144,10 @@
     Assert.assertEquals(1024, buffers.getBufferPool().take().get().capacity());
     Assert.assertThrows(NoSuchElementException.class, () -> buffers.getBufferPool().take());
   }
+
+    private static StageProcessor<?, ?> createStageProcessor(boolean usesProcessingBuffers) {
+        final StageProcessor<?, ?> stageProcessor = Mockito.mock(StageProcessor.class);
+        Mockito.when(stageProcessor.usesProcessingBuffers()).thenReturn(usesProcessingBuffers);
+        return stageProcessor;
+    }
 }
