--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
@@ -40,9 +40,8 @@
   public void testArchiveSegmentWithType() throws SegmentLoadingException
   {
     final DataSegmentArchiver archiver = Mockito.mock(DataSegmentArchiver.class);
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "sane"));
 
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(archiver);
     final DataSegmentArchiver segmentArchiver = injector.getInstance(OmniDataSegmentArchiver.class);
     segmentArchiver.archive(segment);
@@ -119,4 +118,17 @@
       return null;
     }
   }
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static org.apache.druid.timeline.DataSegment createMockDataSegment() {
+            final org.apache.druid.timeline.DataSegment segment =
+                    org.mockito.Mockito.mock(org.apache.druid.timeline.DataSegment.class);
+            org.mockito.Mockito.when(segment.getLoadSpec())
+                    .thenReturn(com.google.common.collect.ImmutableMap.of("type", "sane"));
+            return segment;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
@@ -42,9 +42,8 @@
   public void testMoveSegmentWithType() throws SegmentLoadingException
   {
     final DataSegmentMover mover = Mockito.mock(DataSegmentMover.class);
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "sane"));
 
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(mover);
     final OmniDataSegmentMover segmentMover = injector.getInstance(OmniDataSegmentMover.class);
     segmentMover.move(segment, ImmutableMap.of());
@@ -115,4 +114,17 @@
       return null;
     }
   }
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static org.apache.druid.timeline.DataSegment createMockDataSegment() {
+            final org.apache.druid.timeline.DataSegment segment =
+                    org.mockito.Mockito.mock(org.apache.druid.timeline.DataSegment.class);
+            org.mockito.Mockito.when(segment.getLoadSpec())
+                    .thenReturn(com.google.common.collect.ImmutableMap.of("type", "sane"));
+            return segment;
+        }
+    }
 }
