--- a/processing/src/test/java/org/apache/druid/query/filter/InDimFilterTest.java
+++ b/processing/src/test/java/org/apache/druid/query/filter/InDimFilterTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.query.filter;
 
+import org.mockito.Mock;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
@@ -58,6 +59,8 @@
 
 public class InDimFilterTest extends InitializedNullHandlingTest
 {
+    @Mock
+    private final BitmapColumnIndex bitmapColumnIndex;
   private ObjectMapper mapper = new DefaultObjectMapper();
 
   private final String serializedFilter =
@@ -754,7 +757,6 @@
     final ColumnIndexSelector indexSelector = Mockito.mock(ColumnIndexSelector.class);
     final ColumnIndexSupplier indexSupplier = Mockito.mock(ColumnIndexSupplier.class);
     final Utf8ValueSetIndexes valueIndexes = Mockito.mock(Utf8ValueSetIndexes.class);
-    final BitmapColumnIndex bitmapColumnIndex = Mockito.mock(BitmapColumnIndex.class);
 
     final InDimFilter.ValuesSet expectedValuesSet = new InDimFilter.ValuesSet();
     expectedValuesSet.addAll(Arrays.asList("v1", "v2"));
@@ -778,7 +780,6 @@
     final ColumnIndexSelector indexSelector = Mockito.mock(ColumnIndexSelector.class);
     final ColumnIndexSupplier indexSupplier = Mockito.mock(ColumnIndexSupplier.class);
     final StringValueSetIndexes valueIndex = Mockito.mock(StringValueSetIndexes.class);
-    final BitmapColumnIndex bitmapColumnIndex = Mockito.mock(BitmapColumnIndex.class);
 
     final InDimFilter.ValuesSet expectedValuesSet = new InDimFilter.ValuesSet();
     expectedValuesSet.addAll(Arrays.asList("v1", "v2"));
