--- a/indexing-service/src/test/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfoTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfoTest.java
@@ -237,8 +237,7 @@
 
 
     // Parallel index task
-    TaskResource taskResource0 = mock(TaskResource.class);
-    when(taskResource0.getRequiredCapacity()).thenReturn(3);
+    TaskResource taskResource0 = createMockTaskResource(3);
     Task parallelIndexTask = mock(ParallelIndexSupervisorTask.class);
     when(parallelIndexTask.getType()).thenReturn(ParallelIndexSupervisorTask.TYPE);
     when(parallelIndexTask.getTaskResource()).thenReturn(taskResource0);
@@ -251,8 +250,7 @@
 
 
     // Some other indexing task
-    TaskResource taskResource1 = mock(TaskResource.class);
-    when(taskResource1.getRequiredCapacity()).thenReturn(5);
+    TaskResource taskResource1 = createMockTaskResource(5);
     Task anyOtherTask = mock(IndexTask.class);
     when(anyOtherTask.getType()).thenReturn("index");
     when(anyOtherTask.getTaskResource()).thenReturn(taskResource1);
@@ -263,8 +261,7 @@
 
 
     // Task has an availability conflict ("grp1")
-    TaskResource taskResource2 = mock(TaskResource.class);
-    when(taskResource2.getRequiredCapacity()).thenReturn(1);
+    TaskResource taskResource2 = createMockTaskResource(1);
     when(taskResource2.getAvailabilityGroup()).thenReturn("grp1");
     Task grp1Task = mock(IndexTask.class);
     when(grp1Task.getType()).thenReturn("blah");
@@ -284,4 +281,10 @@
       Assert.assertNotEquals(o1.hashCode(), o2.hashCode());
     }
   }
+
+    private static TaskResource createMockTaskResource(int requiredCapacity) {
+        TaskResource taskResource = mock(TaskResource.class);
+        when(taskResource.getRequiredCapacity()).thenReturn(requiredCapacity);
+        return taskResource;
+    }
 }
