--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
@@ -88,8 +88,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+    TaskActionClient taskActionClient = createTaskActionClient();
     when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
     when(toolbox.getTaskReportFileWriter()).thenReturn(new SingleFileTaskReportFileWriter(temporaryFolder.newFile()));
 
@@ -137,8 +136,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+    TaskActionClient taskActionClient = createTaskActionClient();
     when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
 
 
@@ -182,8 +180,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+    TaskActionClient taskActionClient = createTaskActionClient();
     when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
     when(toolbox.getTaskReportFileWriter()).thenReturn(new SingleFileTaskReportFileWriter(temporaryFolder.newFile()));
 
@@ -221,8 +218,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+    TaskActionClient taskActionClient = createTaskActionClient();
     when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
     AbstractTask task = new NoopTask("myID", null, null, 1, 0, null)
     {
@@ -258,8 +254,7 @@
     when(toolbox.getConfig()).thenReturn(config);
     when(toolbox.getJsonMapper()).thenReturn(objectMapper);
 
-    TaskActionClient taskActionClient = mock(TaskActionClient.class);
-    when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+    TaskActionClient taskActionClient = createTaskActionClient();
     when(toolbox.getTaskActionClient()).thenReturn(taskActionClient);
     AbstractTask task = new NoopTask("myID", null, null, 1, 0, null)
     {
@@ -324,4 +319,10 @@
 
     Assert.assertNotSame(builder1, builder2);
   }
+
+    private static TaskActionClient createTaskActionClient() {
+        TaskActionClient taskActionClient = mock(TaskActionClient.class);
+        when(taskActionClient.submit(any())).thenReturn(TaskConfig.class);
+        return taskActionClient;
+    }
 }
