--- a/sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java
+++ b/sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java
@@ -2721,9 +2721,8 @@
   public void testPresenceOfOverIsInvalid()
   {
     final RexBuilder rexBuilder = new RexBuilder(DruidTypeSystem.TYPE_FACTORY);
-    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
-    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
-
+
+    final PlannerContext plannerContext = createPlannerContext();
     RexNode rexNode = rexBuilder.makeOver(
         testHelper.createSqlType(SqlTypeName.BIGINT),
         SqlStdOperatorTable.SUM,
@@ -2751,9 +2750,8 @@
   public void testCalciteLiteralToDruidLiteral()
   {
     final RexBuilder rexBuilder = new RexBuilder(DruidTypeSystem.TYPE_FACTORY);
-    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
-    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
-
+
+    final PlannerContext plannerContext = createPlannerContext();
     assertDruidLiteral(
         new DruidLiteral(ExpressionType.STRING, null),
         Expressions.calciteLiteralToDruidLiteral(
@@ -2915,4 +2913,10 @@
         StringUtils.format("%s: %s", actual.type(), actual.value())
     );
   }
+
+    private static PlannerContext createPlannerContext() {
+        final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
+        Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+        return plannerContext;
+    }
 }
