--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
@@ -67,9 +67,8 @@
   @Test
   public void testBadSegmentArchiverAccessException()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "bad"));
 
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(null);
     final OmniDataSegmentArchiver segmentArchiver = injector.getInstance(OmniDataSegmentArchiver.class);
     Assert.assertThrows(
@@ -119,4 +118,17 @@
       return null;
     }
   }
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static org.apache.druid.timeline.DataSegment createMockDataSegment() {
+            final org.apache.druid.timeline.DataSegment segment =
+                    org.mockito.Mockito.mock(org.apache.druid.timeline.DataSegment.class);
+            org.mockito.Mockito.when(segment.getLoadSpec())
+                    .thenReturn(com.google.common.collect.ImmutableMap.of("type", "bad"));
+            return segment;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
@@ -78,9 +78,8 @@
   @Test
   public void testBadSegmentKillerAccessException()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "bad"));
-
+
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(null);
     final OmniDataSegmentKiller segmentKiller = injector.getInstance(OmniDataSegmentKiller.class);
     Assert.assertThrows(
@@ -204,4 +203,17 @@
     }
   }
 
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static org.apache.druid.timeline.DataSegment createMockDataSegment() {
+            final org.apache.druid.timeline.DataSegment segment =
+                    org.mockito.Mockito.mock(org.apache.druid.timeline.DataSegment.class);
+            org.mockito.Mockito.when(segment.getLoadSpec())
+                    .thenReturn(com.google.common.collect.ImmutableMap.of("type", "bad"));
+            return segment;
+        }
+    }
 }
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
@@ -69,9 +69,8 @@
   @Test
   public void testBadSegmentMoverAccessException()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "bad"));
 
+    final DataSegment segment = MockDataSegment.createMockDataSegment();
     final Injector injector = createInjector(null);
     final OmniDataSegmentMover segmentMover = injector.getInstance(OmniDataSegmentMover.class);
     Assert.assertThrows(
@@ -115,4 +114,17 @@
       return null;
     }
   }
+
+    public final class MockDataSegment {
+        private MockDataSegment() {
+        }
+
+        public static org.apache.druid.timeline.DataSegment createMockDataSegment() {
+            final org.apache.druid.timeline.DataSegment segment =
+                    org.mockito.Mockito.mock(org.apache.druid.timeline.DataSegment.class);
+            org.mockito.Mockito.when(segment.getLoadSpec())
+                    .thenReturn(com.google.common.collect.ImmutableMap.of("type", "bad"));
+            return segment;
+        }
+    }
 }
