--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.msq.exec;
 
+import org.mockito.Mock;
 import org.apache.druid.frame.processor.FrameProcessorExecutor;
 import org.apache.druid.java.util.common.ISE;
 import org.apache.druid.msq.counters.CounterTracker;
@@ -31,6 +32,8 @@
 
 public class RunWorkOrderTest
 {
+    @Mock
+    private final RunWorkOrderListener listener;
   private static final String CANCELLATION_ID = "my-cancellation-id";
 
   @Test
@@ -43,7 +46,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
 
@@ -79,7 +81,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
 
@@ -120,8 +121,8 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
+    // removed local mock; replaced with global field `listener`
     final ISE exception = new ISE("oops");
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
     Mockito.doThrow(exception).when(exec).cancel(CANCELLATION_ID);
@@ -158,7 +159,6 @@
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
-    final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
