--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateLocationActionTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateLocationActionTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.indexing.common.actions;
 
+import org.mockito.Mock;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.google.common.base.Optional;
 import org.apache.druid.error.DruidException;
@@ -43,6 +44,8 @@
 
 public class UpdateLocationActionTest
 {
+    @Mock
+    private TaskRunner runner;
   @Test
   public void testFlow() throws UnknownHostException
   {
@@ -52,7 +55,6 @@
     UpdateLocationAction action = new UpdateLocationAction(myLocation);
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
     when(toolbox.getTaskRunner()).thenReturn(Optional.of(runner));
     action.perform(task, toolbox);
     verify(runner, times(1)).updateLocation(eq(task), eq(myLocation));
@@ -67,7 +69,6 @@
     UpdateLocationAction action = new UpdateLocationAction(myLocation);
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
     when(toolbox.getTaskRunner()).thenReturn(Optional.absent());
     action.perform(task, toolbox);
     verify(runner, never()).updateStatus(any(), any());
