--- a/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
@@ -153,9 +153,8 @@
 
     Monitor monitor = Mockito.mock(Monitor.class);
 
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
-
+
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
         Mockito.mock(ServiceEmitter.class),
@@ -217,9 +216,8 @@
 
     Monitor monitor = Mockito.mock(Monitor.class);
 
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
-
+
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
         Mockito.mock(ServiceEmitter.class),
@@ -248,9 +246,8 @@
     Mockito.when(monitor.monitor(ArgumentMatchers.any(ServiceEmitter.class)))
            .thenThrow(new RuntimeException("Test throwing exception while monitoring"));
 
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
-
+
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
     CountDownLatch latch = new CountDownLatch(1);
     AtomicBoolean monitorResultHolder = new AtomicBoolean(false);
     Mockito.doAnswer(new Answer<Future<?>>()
@@ -310,9 +307,8 @@
   {
     ExecutorService executor = Mockito.mock(ExecutorService.class);
     Monitor monitor = Mockito.mock(Monitor.class);
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
-
+
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
     CountDownLatch latch = new CountDownLatch(1);
     Mockito.doAnswer(new Answer<Future<?>>()
     {
@@ -415,4 +411,10 @@
       return true;
     }
   }
+
+    private static DruidMonitorSchedulerConfig createMockDruidMonitorSchedulerConfig() {
+        DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
+        Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
+        return config;
+    }
 }
