--- a/server/src/test/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCacheTest.java
+++ b/server/src/test/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCacheTest.java
@@ -1899,10 +1899,8 @@
         segmentsMetadataManagerConfigSupplier
     );
 
-    SegmentReplicaCount zeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
-    SegmentReplicaCount nonZeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
-    Mockito.when(zeroSegmentReplicaCount.required()).thenReturn(0);
-    Mockito.when(nonZeroSegmentReplicaCount.required()).thenReturn(1);
+    SegmentReplicaCount nonZeroSegmentReplicaCount = createMockSegmentReplicaCount(1);
+    SegmentReplicaCount zeroSegmentReplicaCount = createMockSegmentReplicaCount(0);
     SegmentReplicationStatus segmentReplicationStatus = Mockito.mock(SegmentReplicationStatus.class);
     Mockito.when(segmentReplicationStatus.getReplicaCountsInCluster(ArgumentMatchers.eq(coldSegment.getId())))
            .thenReturn(zeroSegmentReplicaCount);
@@ -2118,10 +2116,8 @@
         segmentsMetadataManagerConfigSupplier
     );
 
-    SegmentReplicaCount zeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
-    SegmentReplicaCount nonZeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
-    Mockito.when(zeroSegmentReplicaCount.required()).thenReturn(0);
-    Mockito.when(nonZeroSegmentReplicaCount.required()).thenReturn(1);
+    SegmentReplicaCount zeroSegmentReplicaCount = createMockSegmentReplicaCount(0);
+    SegmentReplicaCount nonZeroSegmentReplicaCount = createMockSegmentReplicaCount(1);
     SegmentReplicationStatus segmentReplicationStatus = Mockito.mock(SegmentReplicationStatus.class);
     Mockito.when(segmentReplicationStatus.getReplicaCountsInCluster(ArgumentMatchers.eq(coldSegmentAlpha.getId())))
            .thenReturn(zeroSegmentReplicaCount);
@@ -2455,4 +2451,10 @@
     Assert.assertEquals("m1", columnNames.get(2));
     Assert.assertEquals(ColumnType.LONG, fooRowSignature.getColumnType(columnNames.get(2)).get());
   }
+
+    private static SegmentReplicaCount createMockSegmentReplicaCount(int required) {
+        SegmentReplicaCount segmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
+        Mockito.when(segmentReplicaCount.required()).thenReturn(required);
+        return segmentReplicaCount;
+    }
 }
