--- a/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
+++ b/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
@@ -19,6 +19,7 @@
 
 package org.apache.druid.server;
 
+import org.mockito.Mock;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.InjectableValues;
 import com.fasterxml.jackson.databind.ObjectMapper;
@@ -127,6 +128,8 @@
 
 public class AsyncQueryForwardingServletTest extends BaseJettyTest
 {
+    @Mock
+    private final Request proxyRequestMock;
   private static int port1;
   private static int port2;
 
@@ -485,7 +488,6 @@
     Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
-    final Request proxyRequestMock = Mockito.mock(Request.class);
     final Response responseMock = Mockito.mock(Response.class);
     Mockito.when(responseMock.getStatus()).thenReturn(504); // Gateway Timeout
     Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
@@ -549,7 +551,6 @@
     Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
-    final Request proxyRequestMock = Mockito.mock(Request.class);
     final Response responseMock = Mockito.mock(Response.class);
     Mockito.when(responseMock.getStatus()).thenReturn(200); // Status OK
     Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
