--- a/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
@@ -48,6 +48,8 @@
 
 public class ClockDriftSafeMonitorSchedulerTest
 {
+    @Mock
+    private ServiceEmitter serviceEmitter;
   // A real executor service to execute CronTask asynchronously.
   // Many tests in this class use mocks to easily control the behavior of CronScheduler and the ExecutorService
   // used by MonitorScheduler. However, as MonitorScheduler uses two differnt threads in production, one for
@@ -92,13 +94,7 @@
     
     ExecutorService executor = Mockito.mock(ExecutorService.class);
 
-    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
-        Mockito.mock(DruidMonitorSchedulerConfig.class),
-        Mockito.mock(ServiceEmitter.class),
-        ImmutableList.of(monitor1, monitor2),
-        CronScheduler.newBuilder(Duration.ofSeconds(1L)).setThreadName("monitor-scheduler-test").build(),
-        executor
-    );
+    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(Mockito.mock(DruidMonitorSchedulerConfig.class), serviceEmitter, ImmutableList.of(monitor1, monitor2), CronScheduler.newBuilder(Duration.ofSeconds(1L)).setThreadName("monitor-scheduler-test").build(), executor);
 
     final Optional<Monitor1> maybeFound1 = scheduler.findMonitor(Monitor1.class);
     final Optional<Monitor2> maybeFound2 = scheduler.findMonitor(Monitor2.class);
@@ -156,13 +152,7 @@
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
 
-    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
-        config,
-        Mockito.mock(ServiceEmitter.class),
-        ImmutableList.of(monitor),
-        cronScheduler,
-        executor
-    );
+    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(config, serviceEmitter, ImmutableList.of(monitor), cronScheduler, executor);
     scheduler.start();
     latch.await(5, TimeUnit.SECONDS);
 
@@ -220,13 +210,7 @@
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
 
-    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
-        config,
-        Mockito.mock(ServiceEmitter.class),
-        ImmutableList.of(monitor),
-        cronScheduler,
-        executor
-    );
+    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(config, serviceEmitter, ImmutableList.of(monitor), cronScheduler, executor);
     scheduler.start();
     latch.await(5, TimeUnit.SECONDS);
 
@@ -285,13 +269,7 @@
         ArgumentMatchers.any(), ArgumentMatchers.any(CronTask.class));
 
     
-    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
-        config,
-        Mockito.mock(ServiceEmitter.class),
-        ImmutableList.of(monitor),
-        cronScheduler,
-        executor
-    );
+    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(config, serviceEmitter, ImmutableList.of(monitor), cronScheduler, executor);
     scheduler.start();
     latch.await(5, TimeUnit.SECONDS);
 
@@ -336,13 +314,7 @@
         ArgumentMatchers.any(), ArgumentMatchers.any(CronTask.class));
 
     
-    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
-        config,
-        Mockito.mock(ServiceEmitter.class),
-        ImmutableList.of(monitor),
-        cronScheduler,
-        executor
-    );
+    final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(config, serviceEmitter, ImmutableList.of(monitor), cronScheduler, executor);
     scheduler.start();
     latch.await(5, TimeUnit.SECONDS);
 
