--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/DimensionCardinalityReportTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/DimensionCardinalityReportTest.java
@@ -311,8 +311,7 @@
   public void testSupervisorDetermineNegativeNumShardsFromCardinalityReport()
   {
     logger.clearLogEvents();
-    Union negativeUnion = mock(Union.class);
-    Mockito.when(negativeUnion.getEstimate()).thenReturn(-1.0);
+    Union negativeUnion = createUnion(-1.0);
     Interval interval = Intervals.of("2001-01-01/P1D");
     Map<Interval, Union> intervalToUnion = ImmutableMap.of(interval, negativeUnion);
     Map<Interval, Integer> intervalToNumShards =
@@ -337,8 +336,7 @@
   @Test
   public void testSupervisorDeterminePositiveNumShardsFromCardinalityReport()
   {
-    Union union = mock(Union.class);
-    Mockito.when(union.getEstimate()).thenReturn(24.0);
+    Union union = createUnion(24.0);
     Interval interval = Intervals.of("2001-01-01/P1D");
     Map<Interval, Union> intervalToUnion = ImmutableMap.of(interval, union);
     Map<Interval, Integer> intervalToNumShards =
@@ -350,8 +348,7 @@
   public void testSupervisorDeterminePositiveOneShardFromCardinalityReport()
   {
     logger.clearLogEvents();
-    Union union = mock(Union.class);
-    Mockito.when(union.getEstimate()).thenReturn(24.0);
+    Union union = createUnion(24.0);
     Interval interval = Intervals.of("2001-01-01/P1D");
     Map<Interval, Union> intervalToUnion = ImmutableMap.of(interval, union);
     Map<Interval, Integer> intervalToNumShards =
@@ -373,4 +370,10 @@
     );
   }
 
+
+    private static Union createUnion(double estimate) {
+        Union union = mock(Union.class);
+        Mockito.when(union.getEstimate()).thenReturn(estimate);
+        return union;
+    }
 }
