diff --git a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/StickyTest.java b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/StickyTest.java
index 47cf58b..dd9de90 100644
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/StickyTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/StickyTest.java
@@ -108,15 +108,8 @@ class StickyTest {
             url = url.addParameter(methodName + "." + CLUSTER_STICKY_KEY, String.valueOf(check));
         }
 
-        given(invoker1.invoke(invocation)).willReturn(result);
-        given(invoker1.isAvailable()).willReturn(true);
-        given(invoker1.getUrl()).willReturn(url.setPort(1));
-        given(invoker1.getInterface()).willReturn(StickyTest.class);
-
-        given(invoker2.invoke(invocation)).willReturn(result);
-        given(invoker2.isAvailable()).willReturn(true);
-        given(invoker2.getUrl()).willReturn(url.setPort(2));
-        given(invoker2.getInterface()).willReturn(StickyTest.class);
+        mockInvoker(invoker1, 1);
+        mockInvoker(invoker2, 2);
 
         invocation.setMethodName(methodName);
 
@@ -130,6 +123,13 @@ class StickyTest {
         return count;
     }
 
+    private void mockInvoker(Invoker<StickyTest> invoker, int port) {
+        given(invoker.invoke(invocation)).willReturn(result);
+        given(invoker.isAvailable()).willReturn(true);
+        given(invoker.getUrl()).willReturn(url.setPort(port));
+        given(invoker.getInterface()).willReturn(StickyTest.class);
+    }
+
     static class StickyClusterInvoker<T> extends AbstractClusterInvoker<T> {
         private Invoker<T> selectedInvoker;
 
