diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
index 4168851..aff495c 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -114,11 +114,10 @@ class RegistryProtocolTest {
                 .getApplicationModel()
                 .getApplicationConfigManager()
                 .setApplication(new ApplicationConfig("application1"));
-        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock = mock(ExtensionLoader.class);
+        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock =
+                createMockExtensionLoader(url, registryProtocolListeners);
         Mockito.when(moduleModel.getExtensionLoader(RegistryProtocolListener.class))
                 .thenReturn(extensionLoaderMock);
-        Mockito.when(extensionLoaderMock.getActivateExtension(url, REGISTRY_PROTOCOL_LISTENER_KEY))
-                .thenReturn(registryProtocolListeners);
         url = url.setScopeModel(moduleModel);
 
         when(registryFactory.getRegistry(registryProtocol.getRegistryUrl(url))).thenReturn(registry);
@@ -182,11 +181,10 @@ class RegistryProtocolTest {
                 .getApplicationModel()
                 .getApplicationConfigManager()
                 .setApplication(new ApplicationConfig("application1"));
-        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock = mock(ExtensionLoader.class);
+        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock =
+                createMockExtensionLoader(url, registryProtocolListeners);
         Mockito.when(moduleModel.getExtensionLoader(RegistryProtocolListener.class))
                 .thenReturn(extensionLoaderMock);
-        Mockito.when(extensionLoaderMock.getActivateExtension(url, REGISTRY_PROTOCOL_LISTENER_KEY))
-                .thenReturn(registryProtocolListeners);
         url = url.setScopeModel(moduleModel);
 
         when(registryFactory.getRegistry(registryProtocol.getRegistryUrl(url))).thenReturn(registry);
@@ -395,11 +393,10 @@ class RegistryProtocolTest {
                 .getApplicationModel()
                 .getApplicationConfigManager()
                 .setApplication(new ApplicationConfig("application1"));
-        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock = mock(ExtensionLoader.class);
+        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock =
+                createMockExtensionLoader(url, registryProtocolListeners);
         Mockito.when(moduleModel.getExtensionLoader(RegistryProtocolListener.class))
                 .thenReturn(extensionLoaderMock);
-        Mockito.when(extensionLoaderMock.getActivateExtension(url, REGISTRY_PROTOCOL_LISTENER_KEY))
-                .thenReturn(registryProtocolListeners);
         url = url.setScopeModel(moduleModel);
 
         registryProtocol.interceptInvoker(clusterInvoker, url, consumerUrl);
@@ -462,11 +459,10 @@ class RegistryProtocolTest {
                 .getApplicationModel()
                 .getApplicationConfigManager()
                 .setApplication(new ApplicationConfig("application1"));
-        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock = mock(ExtensionLoader.class);
+        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock =
+                createMockExtensionLoader(url, registryProtocolListeners);
         Mockito.when(moduleModel.getExtensionLoader(RegistryProtocolListener.class))
                 .thenReturn(extensionLoaderMock);
-        Mockito.when(extensionLoaderMock.getActivateExtension(url, REGISTRY_PROTOCOL_LISTENER_KEY))
-                .thenReturn(registryProtocolListeners);
         url = url.setScopeModel(moduleModel);
 
         registryProtocol.interceptInvoker(clusterInvoker, url, consumerUrl);
@@ -544,4 +540,12 @@ class RegistryProtocolTest {
 
         verify(registry, times(1)).register(registeredConsumerUrl);
     }
+
+    private static ExtensionLoader<RegistryProtocolListener> createMockExtensionLoader(
+            URL url, List<RegistryProtocolListener> registryProtocolListeners) {
+        ExtensionLoader<RegistryProtocolListener> extensionLoaderMock = mock(ExtensionLoader.class);
+        Mockito.when(extensionLoaderMock.getActivateExtension(url, REGISTRY_PROTOCOL_LISTENER_KEY))
+                .thenReturn(registryProtocolListeners);
+        return extensionLoaderMock;
+    }
 }
