diff --git a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
index 110cff9..9461ee9 100644
--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosRegistryTest.java
@@ -96,7 +96,7 @@ class NacosRegistryTest {
             instance.setIp(ip);
             instance.setPort(port);
             instance.setMetadata(new HashMap<>(newUrl.getParameters()));
-            doNothing().when(namingService).registerInstance(serviceName, Constants.DEFAULT_GROUP, instance);
+            mockRegisterInstance(namingService, serviceName, instance);
         } catch (NacosException e) {
             // ignore
         }
@@ -133,7 +133,7 @@ class NacosRegistryTest {
             instance.setIp(ip);
             instance.setPort(port);
             instance.setMetadata(new HashMap<>(newUrl.getParameters()));
-            doNothing().when(namingService).registerInstance(serviceName, Constants.DEFAULT_GROUP, instance);
+            mockRegisterInstance(namingService, serviceName, instance);
 
             doNothing().when(namingService).deregisterInstance(serviceName, Constants.DEFAULT_GROUP, ip, port);
         } catch (NacosException e) {
@@ -309,4 +309,9 @@ class NacosRegistryTest {
         Assertions.assertEquals(2, registered.size());
         Assertions.assertEquals(1, subscribed.get(serviceUrlWithWildcard).size());
     }
+
+    private void mockRegisterInstance(NamingService namingService, String serviceName, Instance instance)
+            throws NacosException {
+        doNothing().when(namingService).registerInstance(serviceName, Constants.DEFAULT_GROUP, instance);
+    }
 }
