diff --git a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
index 2aca73f..467eb36 100644
--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
@@ -46,14 +46,10 @@ class AccessKeyAuthenticatorTest {
                 .addParameter(Constants.SECRET_ACCESS_KEY_KEY, "sk");
         Invocation invocation = new RpcInvocation();
 
-        AccessKeyAuthenticator helper = mock(AccessKeyAuthenticator.class);
-        doCallRealMethod().when(helper).sign(invocation, url);
-        when(helper.getSignature(eq(url), eq(invocation), eq("sk"), anyString()))
-                .thenReturn("dubbo");
-
         AccessKeyPair accessKeyPair = mock(AccessKeyPair.class);
         when(accessKeyPair.getSecretKey()).thenReturn("sk");
-        when(helper.getAccessKeyPair(invocation, url)).thenReturn(accessKeyPair);
+        AccessKeyAuthenticator helper = createMockAccessKeyAuthenticator(url, invocation, accessKeyPair);
+        doCallRealMethod().when(helper).sign(invocation, url);
 
         helper.sign(invocation, url);
         assertEquals(String.valueOf(invocation.getAttachment(CommonConstants.CONSUMER)), url.getApplication());
@@ -73,14 +69,10 @@ class AccessKeyAuthenticatorTest {
         invocation.setAttachment(Constants.REQUEST_TIMESTAMP_KEY, String.valueOf(System.currentTimeMillis()));
         invocation.setAttachment(CommonConstants.CONSUMER, "test");
 
-        AccessKeyAuthenticator helper = mock(AccessKeyAuthenticator.class);
-        doCallRealMethod().when(helper).authenticate(invocation, url);
-        when(helper.getSignature(eq(url), eq(invocation), eq("sk"), anyString()))
-                .thenReturn("dubbo");
-
         AccessKeyPair accessKeyPair = mock(AccessKeyPair.class);
         when(accessKeyPair.getSecretKey()).thenReturn("sk");
-        when(helper.getAccessKeyPair(invocation, url)).thenReturn(accessKeyPair);
+        AccessKeyAuthenticator helper = createMockAccessKeyAuthenticator(url, invocation, accessKeyPair);
+        doCallRealMethod().when(helper).authenticate(invocation, url);
 
         assertDoesNotThrow(() -> helper.authenticate(invocation, url));
     }
@@ -113,4 +105,13 @@ class AccessKeyAuthenticatorTest {
         String signature = helper.getSignature(url, invocation, secretKey, String.valueOf(System.currentTimeMillis()));
         assertNotNull(signature);
     }
+
+    private static AccessKeyAuthenticator createMockAccessKeyAuthenticator(
+            URL url, Invocation invocation, AccessKeyPair accessKeyPair) {
+        AccessKeyAuthenticator helper = mock(AccessKeyAuthenticator.class);
+        when(helper.getSignature(eq(url), eq(invocation), eq("sk"), anyString()))
+                .thenReturn("dubbo");
+        when(helper.getAccessKeyPair(invocation, url)).thenReturn(accessKeyPair);
+        return helper;
+    }
 }
