diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassUtilsTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassUtilsTest.java
index 36ae28a..44193a1 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassUtilsTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassUtilsTest.java
@@ -31,7 +31,7 @@ class ClassUtilsTest {
     void testForNameWithThreadContextClassLoader() throws Exception {
         ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
         try {
-            ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+            ClassLoader classLoader = mockClassLoader();
             Thread.currentThread().setContextClassLoader(classLoader);
             ClassUtils.forNameWithThreadContextClassLoader("a.b.c.D");
             verify(classLoader).loadClass("a.b.c.D");
@@ -86,7 +86,7 @@ class ClassUtilsTest {
 
     @Test
     void testForName3() throws Exception {
-        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+        ClassLoader classLoader = mockClassLoader();
         ClassUtils.forName("a.b.c.D", classLoader);
         verify(classLoader).loadClass("a.b.c.D");
     }
@@ -152,4 +152,8 @@ class ClassUtilsTest {
         assertThat(ClassUtils.convertPrimitive(double.class, null), equalTo(null));
         assertThat(ClassUtils.convertPrimitive(double.class, "10.1"), equalTo(Double.valueOf(10.1)));
     }
+
+    private ClassLoader mockClassLoader() {
+        return Mockito.mock(ClassLoader.class);
+    }
 }
