diff --git a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/TelnetProcessHandlerTest.java b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/TelnetProcessHandlerTest.java
index e3bd8b8..ac4ff2b 100644
--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/TelnetProcessHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/TelnetProcessHandlerTest.java
@@ -37,7 +37,7 @@ import static org.mockito.Mockito.when;
 class TelnetProcessHandlerTest {
     @Test
     void testPrompt() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        ChannelHandlerContext context = mockContext();
         TelnetProcessHandler handler = new TelnetProcessHandler(
                 FrameworkModel.defaultModel(),
                 QosConfiguration.builder()
@@ -49,7 +49,7 @@ class TelnetProcessHandlerTest {
 
     @Test
     void testBye() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        ChannelHandlerContext context = mockContext();
         TelnetProcessHandler handler = new TelnetProcessHandler(
                 FrameworkModel.defaultModel(), QosConfiguration.builder().build());
         ChannelFuture future = mock(ChannelFuture.class);
@@ -60,7 +60,7 @@ class TelnetProcessHandlerTest {
 
     @Test
     void testUnknownCommand() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        ChannelHandlerContext context = mockContext();
         TelnetProcessHandler handler = new TelnetProcessHandler(
                 FrameworkModel.defaultModel(), QosConfiguration.builder().build());
         handler.channelRead0(context, "unknown");
@@ -71,7 +71,7 @@ class TelnetProcessHandlerTest {
 
     @Test
     void testGreeting() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        ChannelHandlerContext context = mockContext();
         TelnetProcessHandler handler = new TelnetProcessHandler(
                 FrameworkModel.defaultModel(),
                 QosConfiguration.builder()
@@ -83,4 +83,8 @@ class TelnetProcessHandlerTest {
         assertThat(captor.getValue(), containsString("greeting"));
         assertThat(captor.getValue(), containsString("dubbo>"));
     }
+
+    private ChannelHandlerContext mockContext() {
+        return mock(ChannelHandlerContext.class);
+    }
 }
