diff --git a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
index 9a27ea8..777fb7a 100644
--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
@@ -42,8 +42,7 @@ class QosProcessHandlerTest {
     void testDecodeHttp() throws Exception {
         ByteBuf buf = Unpooled.wrappedBuffer(new byte[] {'G'});
         ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
-        ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
-        Mockito.when(context.pipeline()).thenReturn(pipeline);
+        ChannelPipeline pipeline = mockPipeline(context);
         QosProcessHandler handler = new QosProcessHandler(
                 FrameworkModel.defaultModel(),
                 QosConfiguration.builder()
@@ -62,8 +61,7 @@ class QosProcessHandlerTest {
     void testDecodeTelnet() throws Exception {
         ByteBuf buf = Unpooled.wrappedBuffer(new byte[] {'A'});
         ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
-        ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
-        Mockito.when(context.pipeline()).thenReturn(pipeline);
+        ChannelPipeline pipeline = mockPipeline(context);
         QosProcessHandler handler = new QosProcessHandler(
                 FrameworkModel.defaultModel(),
                 QosConfiguration.builder()
@@ -78,4 +76,10 @@ class QosProcessHandlerTest {
         verify(pipeline).addLast(any(TelnetProcessHandler.class));
         verify(pipeline).remove(handler);
     }
+
+    private ChannelPipeline mockPipeline(ChannelHandlerContext context) {
+        ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
+        Mockito.when(context.pipeline()).thenReturn(pipeline);
+        return pipeline;
+    }
 }
