diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
index a1ef5cb..a129418 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
@@ -89,13 +89,7 @@ class MigrationInvokerTest {
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
                 .thenReturn(serviceDiscoveryInvoker);
 
-        URL consumerURL = Mockito.mock(URL.class);
-        Mockito.when(consumerURL.getServiceInterface()).thenReturn("Test");
-        Mockito.when(consumerURL.getGroup()).thenReturn("Group");
-        Mockito.when(consumerURL.getVersion()).thenReturn("0.0.0");
-        Mockito.when(consumerURL.getServiceKey()).thenReturn("Group/Test:0.0.0");
-        Mockito.when(consumerURL.getDisplayServiceKey()).thenReturn("Test:0.0.0");
-        Mockito.when(consumerURL.getOrDefaultApplicationModel()).thenReturn(ApplicationModel.defaultModel());
+        URL consumerURL = createMockConsumerUrl();
 
         Mockito.when(invoker.getUrl()).thenReturn(consumerURL);
         Mockito.when(serviceDiscoveryInvoker.getUrl()).thenReturn(consumerURL);
@@ -265,13 +259,7 @@ class MigrationInvokerTest {
                         Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
                 .thenReturn(serviceDiscoveryInvoker);
 
-        URL consumerURL = Mockito.mock(URL.class);
-        Mockito.when(consumerURL.getServiceInterface()).thenReturn("Test");
-        Mockito.when(consumerURL.getGroup()).thenReturn("Group");
-        Mockito.when(consumerURL.getVersion()).thenReturn("0.0.0");
-        Mockito.when(consumerURL.getServiceKey()).thenReturn("Group/Test:0.0.0");
-        Mockito.when(consumerURL.getDisplayServiceKey()).thenReturn("Test:0.0.0");
-        Mockito.when(consumerURL.getOrDefaultApplicationModel()).thenReturn(ApplicationModel.defaultModel());
+        URL consumerURL = createMockConsumerUrl();
 
         Mockito.when(invoker.getUrl()).thenReturn(consumerURL);
         Mockito.when(serviceDiscoveryInvoker.getUrl()).thenReturn(consumerURL);
@@ -302,4 +290,15 @@ class MigrationInvokerTest {
         // 独立线程invoker状态切换
 
     }
+
+    private static URL createMockConsumerUrl() {
+        URL consumerURL = Mockito.mock(URL.class);
+        Mockito.when(consumerURL.getServiceInterface()).thenReturn("Test");
+        Mockito.when(consumerURL.getGroup()).thenReturn("Group");
+        Mockito.when(consumerURL.getVersion()).thenReturn("0.0.0");
+        Mockito.when(consumerURL.getServiceKey()).thenReturn("Group/Test:0.0.0");
+        Mockito.when(consumerURL.getDisplayServiceKey()).thenReturn("Test:0.0.0");
+        Mockito.when(consumerURL.getOrDefaultApplicationModel()).thenReturn(ApplicationModel.defaultModel());
+        return consumerURL;
+    }
 }
