diff --git a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosConnectionsManagerTest.java b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosConnectionsManagerTest.java
index ca17d3c..2383076 100644
--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosConnectionsManagerTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosConnectionsManagerTest.java
@@ -39,9 +39,13 @@ import static com.alibaba.nacos.client.constant.Constants.HealthCheck.UP;
 import static org.mockito.ArgumentMatchers.any;
 
 public class NacosConnectionsManagerTest {
+    private static NamingService createMockNamingService() {
+        return Mockito.mock(NamingService.class);
+    }
+
     @Test
     public void testGet() {
-        NamingService namingService = Mockito.mock(NamingService.class);
+        NamingService namingService = createMockNamingService();
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(namingService);
         Assertions.assertEquals(namingService, nacosConnectionManager.getNamingService());
         Assertions.assertEquals(namingService, nacosConnectionManager.getNamingService());
@@ -54,7 +58,7 @@ public class NacosConnectionsManagerTest {
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = createMockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
