diff --git a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
index 59dbef5..d5e90e1 100644
--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
@@ -31,16 +31,14 @@ import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
 class ConsumerSignFilterTest {
 
     @Test
     void testAuthDisabled() {
         URL url = mock(URL.class);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(Invocation.class);
-        when(invoker.getUrl()).thenReturn(url);
         ConsumerSignFilter consumerSignFilter = new ConsumerSignFilter(FrameworkModel.defaultModel());
         consumerSignFilter.invoke(invoker, invocation);
         verify(invocation, never()).setAttachment(eq(Constants.REQUEST_SIGNATURE_KEY), anyString());
@@ -54,9 +52,8 @@ class ConsumerSignFilterTest {
                 .addParameter(CommonConstants.APPLICATION_KEY, "test")
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(Invocation.class);
-        when(invoker.getUrl()).thenReturn(url);
         ConsumerSignFilter consumerSignFilter = new ConsumerSignFilter(FrameworkModel.defaultModel());
         consumerSignFilter.invoke(invoker, invocation);
         verify(invocation, times(1)).setAttachment(eq(Constants.REQUEST_SIGNATURE_KEY), anyString());
diff --git a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
index 725b0e4..afcd9f1 100644
--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
@@ -43,9 +43,8 @@ class ProviderAuthFilterTest {
     @Test
     void testAuthDisabled() {
         URL url = mock(URL.class);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
-        when(invoker.getUrl()).thenReturn(url);
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         providerAuthFilter.invoke(invoker, invocation);
         verify(url, never()).getParameter(eq(Constants.AUTHENTICATOR_KEY), eq(Constants.DEFAULT_AUTHENTICATOR));
@@ -59,9 +58,8 @@ class ProviderAuthFilterTest {
                 .addParameter(CommonConstants.APPLICATION_KEY, "test")
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
-        when(invoker.getUrl()).thenReturn(url);
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         providerAuthFilter.invoke(invoker, invocation);
         verify(invocation, atLeastOnce()).getAttachment(anyString());
@@ -75,10 +73,9 @@ class ProviderAuthFilterTest {
                 .addParameter(CommonConstants.APPLICATION_KEY, "test")
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
         when(invocation.getAttachment(Constants.REQUEST_SIGNATURE_KEY)).thenReturn(null);
-        when(invoker.getUrl()).thenReturn(url);
 
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         Result result = providerAuthFilter.invoke(invoker, invocation);
@@ -93,10 +90,9 @@ class ProviderAuthFilterTest {
                 .addParameter(CommonConstants.APPLICATION_KEY, "test")
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
         when(invocation.getAttachment(Constants.REQUEST_SIGNATURE_KEY)).thenReturn(null);
-        when(invoker.getUrl()).thenReturn(url);
 
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         Result result = providerAuthFilter.invoke(invoker, invocation);
@@ -109,13 +105,12 @@ class ProviderAuthFilterTest {
                 .addParameter(CommonConstants.APPLICATION_KEY, "test-provider")
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
         when(invocation.getObjectAttachment(Constants.REQUEST_SIGNATURE_KEY)).thenReturn("dubbo");
         when(invocation.getObjectAttachment(Constants.AK_KEY)).thenReturn("ak");
         when(invocation.getObjectAttachment(CommonConstants.CONSUMER)).thenReturn("test-consumer");
         when(invocation.getObjectAttachment(Constants.REQUEST_TIMESTAMP_KEY)).thenReturn(System.currentTimeMillis());
-        when(invoker.getUrl()).thenReturn(url);
 
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         Result result = providerAuthFilter.invoke(invoker, invocation);
@@ -138,7 +133,7 @@ class ProviderAuthFilterTest {
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
 
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
         when(invocation.getObjectAttachment(Constants.AK_KEY)).thenReturn("ak");
         when(invocation.getObjectAttachment(CommonConstants.CONSUMER)).thenReturn("test-consumer");
@@ -146,7 +141,6 @@ class ProviderAuthFilterTest {
         when(invocation.getMethodName()).thenReturn(method);
         Object[] fakeParams = new Object[] {"dubbo1", "dubbo3"};
         when(invocation.getArguments()).thenReturn(fakeParams);
-        when(invoker.getUrl()).thenReturn(url);
 
         String requestString = String.format(
                 Constants.SIGNATURE_STRING_FORMAT,
@@ -175,13 +169,12 @@ class ProviderAuthFilterTest {
                 .addParameter(CommonConstants.APPLICATION_KEY, "test-provider")
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
-        Invoker invoker = mock(Invoker.class);
+        Invoker invoker = MockInvoker.createMockInvoker(url);
         Invocation invocation = mock(RpcInvocation.class);
         when(invocation.getAttachment(Constants.AK_KEY)).thenReturn("ak");
         when(invocation.getAttachment(CommonConstants.CONSUMER)).thenReturn("test-consumer");
         when(invocation.getAttachment(Constants.REQUEST_TIMESTAMP_KEY)).thenReturn(String.valueOf(currentTimeMillis));
         when(invocation.getMethodName()).thenReturn(method);
-        when(invoker.getUrl()).thenReturn(url);
 
         String requestString = String.format(
                 Constants.SIGNATURE_STRING_FORMAT,
