diff --git a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java
index a2f6b14..4126027 100644
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java
@@ -73,21 +73,21 @@ class ConnectivityValidationTest {
     @BeforeEach
     public void setup() {
         ApplicationModel.defaultModel().getBeanFactory().registerBean(MetricsDispatcher.class);
-        invoker1 = Mockito.mock(Invoker.class);
-        invoker2 = Mockito.mock(Invoker.class);
-        invoker3 = Mockito.mock(Invoker.class);
-        invoker4 = Mockito.mock(Invoker.class);
-        invoker5 = Mockito.mock(Invoker.class);
-        invoker6 = Mockito.mock(Invoker.class);
-        invoker7 = Mockito.mock(Invoker.class);
-        invoker8 = Mockito.mock(Invoker.class);
-        invoker9 = Mockito.mock(Invoker.class);
-        invoker10 = Mockito.mock(Invoker.class);
-        invoker11 = Mockito.mock(Invoker.class);
-        invoker12 = Mockito.mock(Invoker.class);
-        invoker13 = Mockito.mock(Invoker.class);
-        invoker14 = Mockito.mock(Invoker.class);
-        invoker15 = Mockito.mock(Invoker.class);
+        invoker1 = createMockInvoker();
+        invoker2 = createMockInvoker();
+        invoker3 = createMockInvoker();
+        invoker4 = createMockInvoker();
+        invoker5 = createMockInvoker();
+        invoker6 = createMockInvoker();
+        invoker7 = createMockInvoker();
+        invoker8 = createMockInvoker();
+        invoker9 = createMockInvoker();
+        invoker10 = createMockInvoker();
+        invoker11 = createMockInvoker();
+        invoker12 = createMockInvoker();
+        invoker13 = createMockInvoker();
+        invoker14 = createMockInvoker();
+        invoker15 = createMockInvoker();
 
         configInvoker(invoker1);
         configInvoker(invoker2);
@@ -126,6 +126,10 @@ class ConnectivityValidationTest {
         when(invoker.isAvailable()).thenReturn(true);
     }
 
+    private Invoker createMockInvoker() {
+        return Mockito.<Invoker>mock(Invoker.class);
+    }
+
     @BeforeAll
     public static void setupClass() {
         System.setProperty(CommonConstants.RECONNECT_TASK_PERIOD, "1");
