diff --git a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java
index 7ffb0cc..e01274b 100644
--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java
@@ -31,6 +31,7 @@ import io.netty.handler.codec.ByteToMessageDecoder;
 import io.netty.handler.codec.DecoderException;
 import io.netty.handler.codec.MessageToByteEncoder;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -44,12 +45,18 @@ import static org.mockito.Mockito.doThrow;
  */
 class NettyCodecAdapterTest {
 
+    private ChannelHandler channelHandler;
+
+    @BeforeEach
+    void setUp() {
+        channelHandler = Mockito.mock(ChannelHandler.class);
+    }
+
     @Test
     void test() {
         Codec2 codec2 = Mockito.mock(Codec2.class);
         URL url = Mockito.mock(URL.class);
-        ChannelHandler handler = Mockito.mock(ChannelHandler.class);
-        NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, handler);
+        NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, channelHandler);
         io.netty.channel.ChannelHandler decoder = nettyCodecAdapter.getDecoder();
         io.netty.channel.ChannelHandler encoder = nettyCodecAdapter.getEncoder();
         Assertions.assertTrue(decoder instanceof ByteToMessageDecoder);
@@ -64,8 +71,7 @@ class NettyCodecAdapterTest {
         URL url = Mockito.mock(URL.class);
         doReturn("default").when(url).getParameter(eq(Constants.CODEC_KEY));
 
-        ChannelHandler handler = Mockito.mock(ChannelHandler.class);
-        NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, handler);
+        NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, channelHandler);
         io.netty.channel.ChannelHandler decoder = nettyCodecAdapter.getDecoder();
         EmbeddedChannel embeddedChannel = new EmbeddedChannel();
         embeddedChannel.pipeline().addLast(decoder);
