diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
index 4168851..20121d7 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -71,6 +71,10 @@ class RegistryProtocolTest {
         ApplicationModel.defaultModel().destroy();
     }
 
+    private Cluster createClusterMock() {
+        return mock(Cluster.class);
+    }
+
     /**
      * verify the generated consumer url information
      */
@@ -123,7 +127,7 @@ class RegistryProtocolTest {
 
         when(registryFactory.getRegistry(registryProtocol.getRegistryUrl(url))).thenReturn(registry);
 
-        Cluster cluster = mock(Cluster.class);
+        Cluster cluster = createClusterMock();
 
         Invoker<?> invoker = registryProtocol.doRefer(cluster, registry, DemoService.class, url, parameters);
 
@@ -191,7 +195,7 @@ class RegistryProtocolTest {
 
         when(registryFactory.getRegistry(registryProtocol.getRegistryUrl(url))).thenReturn(registry);
 
-        Cluster cluster = mock(Cluster.class);
+        Cluster cluster = createClusterMock();
 
         Invoker<?> invoker = registryProtocol.doRefer(cluster, registry, DemoService.class, url, parameters);
 
@@ -521,7 +525,7 @@ class RegistryProtocolTest {
 
         when(registryFactory.getRegistry(registryProtocol.getRegistryUrl(url))).thenReturn(registry);
 
-        Cluster cluster = mock(Cluster.class);
+        Cluster cluster = createClusterMock();
 
         Invoker<?> invoker = registryProtocol.doRefer(cluster, registry, DemoService.class, url, parameters);
 
