diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
index be23099..0aaf805 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
@@ -84,10 +84,9 @@ class FailsafeErrorTypeAwareLoggerTest {
     @Test
     void testGetLogger() {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            Logger failLogger = mock(Logger.class);
+            Logger failLogger = MockLogger.createMockLoggerThatThrowsOnError();
             FailsafeErrorTypeAwareLogger failsafeLogger = new FailsafeErrorTypeAwareLogger(failLogger);
 
-            doThrow(new RuntimeException()).when(failLogger).error(anyString());
             failsafeLogger.getLogger().error("should get error");
         });
     }
diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
index 65de0c9..5f45efe 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
@@ -103,10 +103,9 @@ class FailsafeLoggerTest {
     @Test
     void testGetLogger() {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            Logger failLogger = mock(Logger.class);
+            Logger failLogger = MockLogger.createMockLoggerThatThrowsOnError();
             FailsafeLogger failsafeLogger = new FailsafeLogger(failLogger);
 
-            doThrow(new RuntimeException()).when(failLogger).error(anyString());
             failsafeLogger.getLogger().error("should get error");
         });
     }
