diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/ProtocolListenerWrapperTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/ProtocolListenerWrapperTest.java
index 56491e0..9f9ab05 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/ProtocolListenerWrapperTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/ProtocolListenerWrapperTest.java
@@ -47,9 +47,7 @@ class ProtocolListenerWrapperTest {
                     }
                 };
 
-        Protocol protocolWithoutListener = mock(Protocol.class);
-        when(protocolWithoutListener.refer(DemoService.class, urlWithoutListener))
-                .thenReturn(invokerWithoutListener);
+        Protocol protocolWithoutListener = createMockProtocol(urlWithoutListener, invokerWithoutListener);
 
         ProtocolListenerWrapper protocolListenerWrapperWithoutListener =
                 new ProtocolListenerWrapper(protocolWithoutListener);
@@ -71,8 +69,7 @@ class ProtocolListenerWrapperTest {
                     }
                 };
 
-        Protocol protocol = mock(Protocol.class);
-        when(protocol.refer(DemoService.class, urlWithListener)).thenReturn(invokerWithListener);
+        Protocol protocol = createMockProtocol(urlWithListener, invokerWithListener);
 
         ProtocolListenerWrapper protocolListenerWrapper = new ProtocolListenerWrapper(protocol);
 
@@ -94,9 +91,7 @@ class ProtocolListenerWrapperTest {
                     }
                 };
 
-        Protocol protocolWithoutListener = mock(Protocol.class);
-        when(protocolWithoutListener.refer(DemoService.class, urlWithoutListener))
-                .thenReturn(invokerWithoutListener);
+        Protocol protocolWithoutListener = createMockProtocol(urlWithoutListener, invokerWithoutListener);
 
         ProtocolListenerWrapper protocolListenerWrapperWithoutListener =
                 new ProtocolListenerWrapper(protocolWithoutListener);
@@ -118,8 +113,7 @@ class ProtocolListenerWrapperTest {
                     }
                 };
 
-        Protocol protocol = mock(Protocol.class);
-        when(protocol.refer(DemoService.class, urlWithListener)).thenReturn(invokerWithListener);
+        Protocol protocol = createMockProtocol(urlWithListener, invokerWithListener);
 
         ProtocolListenerWrapper protocolListenerWrapper = new ProtocolListenerWrapper(protocol);
 
@@ -127,4 +121,10 @@ class ProtocolListenerWrapperTest {
         Assertions.assertTrue(invoker instanceof ListenerInvokerWrapper);
         Assertions.assertEquals(1, CountInvokerListener.getCounter());
     }
+
+    private static Protocol createMockProtocol(URL url, AbstractInvoker<DemoService> invoker) {
+        Protocol protocol = mock(Protocol.class);
+        when(protocol.refer(DemoService.class, url)).thenReturn(invoker);
+        return protocol;
+    }
 }
