diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
index 4168851..ea78f8b 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -79,8 +79,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -147,8 +146,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -218,8 +216,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -284,8 +281,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -351,8 +347,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -417,8 +412,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -483,8 +477,7 @@ class RegistryProtocolTest {
         ApplicationConfig applicationConfig = new ApplicationConfig();
         applicationConfig.setName("application1");
 
-        ConfigManager configManager = mock(ConfigManager.class);
-        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        ConfigManager configManager = createMockConfigManager(applicationConfig);
 
         CompositeConfiguration compositeConfiguration = mock(CompositeConfiguration.class);
         when(compositeConfiguration.convert(Boolean.class, ENABLE_CONFIGURATION_LISTEN, true))
@@ -544,4 +537,10 @@ class RegistryProtocolTest {
 
         verify(registry, times(1)).register(registeredConsumerUrl);
     }
+
+    private static ConfigManager createMockConfigManager(ApplicationConfig applicationConfig) {
+        ConfigManager configManager = mock(ConfigManager.class);
+        when(configManager.getApplicationOrElseThrow()).thenReturn(applicationConfig);
+        return configManager;
+    }
 }
