diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
index a1ef5cb..b7de040 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
@@ -54,6 +54,12 @@ class MigrationInvokerTest {
         FrameworkModel.destroyAll();
     }
 
+    private static MigrationRule createMockMigrationRule() {
+        MigrationRule migrationRule = Mockito.mock(MigrationRule.class);
+        Mockito.when(migrationRule.getForce(Mockito.any())).thenReturn(true);
+        return migrationRule;
+    }
+
     @SuppressWarnings("all")
     @Test
     void test() {
@@ -103,8 +109,7 @@ class MigrationInvokerTest {
         MigrationInvoker<?> migrationInvoker =
                 new MigrationInvoker<>(registryProtocol, null, null, DemoService.class, null, consumerURL);
 
-        MigrationRule migrationRule = Mockito.mock(MigrationRule.class);
-        Mockito.when(migrationRule.getForce(Mockito.any())).thenReturn(true);
+        MigrationRule migrationRule = createMockMigrationRule();
         migrationInvoker.migrateToForceInterfaceInvoker(migrationRule);
         migrationInvoker.setMigrationStep(MigrationStep.FORCE_INTERFACE);
         migrationInvoker.invoke(null);
@@ -279,8 +284,7 @@ class MigrationInvokerTest {
         MigrationInvoker<?> migrationInvoker =
                 new MigrationInvoker<>(registryProtocol, null, null, DemoService.class, null, consumerURL);
 
-        MigrationRule migrationRule = Mockito.mock(MigrationRule.class);
-        Mockito.when(migrationRule.getForce(Mockito.any())).thenReturn(true);
+        MigrationRule migrationRule = createMockMigrationRule();
         migrationInvoker.migrateToApplicationFirstInvoker(migrationRule);
         migrationInvoker.setMigrationStep(MigrationStep.APPLICATION_FIRST);
         migrationInvoker.invoke(null);
