diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
index 4168851..c27a0d9 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -71,6 +71,10 @@ class RegistryProtocolTest {
         ApplicationModel.defaultModel().destroy();
     }
 
+    private MigrationInvoker<?> createMigrationInvokerMock() {
+        return mock(MigrationInvoker.class);
+    }
+
     /**
      * verify the generated consumer url information
      */
@@ -371,7 +375,7 @@ class RegistryProtocolTest {
         attributes.put(REFER_KEY, refer);
         URL url = serviceConfigURL.addAttributes(attributes);
 
-        MigrationInvoker<?> clusterInvoker = mock(MigrationInvoker.class);
+        MigrationInvoker<?> clusterInvoker = createMigrationInvokerMock();
 
         Map<String, Object> consumerAttribute = new HashMap<>(url.getAttributes());
         consumerAttribute.remove(REFER_KEY);
@@ -439,7 +443,7 @@ class RegistryProtocolTest {
         URL url = serviceConfigURL.addAttributes(attributes);
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
-        MigrationInvoker<?> clusterInvoker = mock(MigrationInvoker.class);
+        MigrationInvoker<?> clusterInvoker = createMigrationInvokerMock();
 
         Map<String, Object> consumerAttribute = new HashMap<>(url.getAttributes());
         consumerAttribute.remove(REFER_KEY);
