diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
index c6718e9..c15a2ca 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
@@ -38,7 +38,7 @@ class StubSuppliersTest {
 
     @Test
     void addSupplier() {
-        Invoker<?> invoker = Mockito.mock(Invoker.class);
+        Invoker<?> invoker = mockInvoker();
         ServiceDescriptor descriptor = Mockito.mock(ServiceDescriptor.class);
         StubSuppliers.addSupplier(serviceName, i -> invoker);
         Assertions.assertEquals(invoker, StubSuppliers.createStub(serviceName, invoker));
@@ -46,7 +46,7 @@ class StubSuppliersTest {
 
     @Test
     void createStub() {
-        Invoker<?> invoker = Mockito.mock(Invoker.class);
+        Invoker<?> invoker = mockInvoker();
         try {
             StubSuppliers.createStub(serviceName + 1, invoker);
             fail();
@@ -64,4 +64,8 @@ class StubSuppliersTest {
             // pass
         }
     }
+
+    private Invoker<?> mockInvoker() {
+        return Mockito.mock(Invoker.class);
+    }
 }
