diff --git a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
index 2aca73f..025a33c 100644
--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
@@ -51,8 +51,7 @@ class AccessKeyAuthenticatorTest {
         when(helper.getSignature(eq(url), eq(invocation), eq("sk"), anyString()))
                 .thenReturn("dubbo");
 
-        AccessKeyPair accessKeyPair = mock(AccessKeyPair.class);
-        when(accessKeyPair.getSecretKey()).thenReturn("sk");
+        AccessKeyPair accessKeyPair = mockAccessKeyPair();
         when(helper.getAccessKeyPair(invocation, url)).thenReturn(accessKeyPair);
 
         helper.sign(invocation, url);
@@ -78,8 +77,7 @@ class AccessKeyAuthenticatorTest {
         when(helper.getSignature(eq(url), eq(invocation), eq("sk"), anyString()))
                 .thenReturn("dubbo");
 
-        AccessKeyPair accessKeyPair = mock(AccessKeyPair.class);
-        when(accessKeyPair.getSecretKey()).thenReturn("sk");
+        AccessKeyPair accessKeyPair = mockAccessKeyPair();
         when(helper.getAccessKeyPair(invocation, url)).thenReturn(accessKeyPair);
 
         assertDoesNotThrow(() -> helper.authenticate(invocation, url));
@@ -113,4 +111,10 @@ class AccessKeyAuthenticatorTest {
         String signature = helper.getSignature(url, invocation, secretKey, String.valueOf(System.currentTimeMillis()));
         assertNotNull(signature);
     }
+
+    private AccessKeyPair mockAccessKeyPair() {
+        AccessKeyPair accessKeyPair = mock(AccessKeyPair.class);
+        when(accessKeyPair.getSecretKey()).thenReturn("sk");
+        return accessKeyPair;
+    }
 }
