diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
index bdb39c6..0dd5725 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
@@ -30,6 +30,7 @@ import org.apache.dubbo.rpc.support.MyInvoker;
 
 import org.junit.jupiter.api.Test;
 
+import static org.apache.dubbo.rpc.filter.DemoServiceInvokerTestUtils.mockInvoker;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.mock;
@@ -53,9 +54,8 @@ class ContextFilterTest {
         given(invocation.getArguments()).willReturn(new Object[] {"hello"});
         given(invocation.getObjectAttachments()).willReturn(null);
 
-        invoker = mock(Invoker.class);
+        invoker = mockInvoker();
         given(invoker.isAvailable()).willReturn(true);
-        given(invoker.getInterface()).willReturn(DemoService.class);
         AppResponse result = new AppResponse();
         result.setValue("High");
         given(invoker.invoke(invocation)).willReturn(result);
diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ExceptionFilterTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ExceptionFilterTest.java
index 909e88b..b3e424d 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ExceptionFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ExceptionFilterTest.java
@@ -33,6 +33,7 @@ import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 import static org.apache.dubbo.common.constants.LoggerCodeConstants.CONFIG_FILTER_VALIDATION_EXCEPTION;
+import static org.apache.dubbo.rpc.filter.DemoServiceInvokerTestUtils.mockInvoker;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.BDDMockito.given;
@@ -52,8 +53,7 @@ class ExceptionFilterTest {
         ExceptionFilter exceptionFilter = new ExceptionFilter();
         RpcInvocation invocation = new RpcInvocation(
                 "sayHello", DemoService.class.getName(), "", new Class<?>[] {String.class}, new Object[] {"world"});
-        Invoker<DemoService> invoker = mock(Invoker.class);
-        given(invoker.getInterface()).willReturn(DemoService.class);
+        Invoker<DemoService> invoker = mockInvoker();
         given(invoker.invoke(eq(invocation))).willThrow(exception);
 
         try {
@@ -86,9 +86,8 @@ class ExceptionFilterTest {
         AppResponse appResponse = new AppResponse();
         appResponse.setException(new IllegalArgumentException("java"));
 
-        Invoker<DemoService> invoker = mock(Invoker.class);
+        Invoker<DemoService> invoker = mockInvoker();
         when(invoker.invoke(invocation)).thenReturn(appResponse);
-        when(invoker.getInterface()).thenReturn(DemoService.class);
 
         Result newResult = exceptionFilter.invoke(invoker, invocation);
 
@@ -106,9 +105,8 @@ class ExceptionFilterTest {
         AppResponse appResponse = new AppResponse();
         appResponse.setException(new LocalException("localException"));
 
-        Invoker<DemoService> invoker = mock(Invoker.class);
+        Invoker<DemoService> invoker = mockInvoker();
         when(invoker.invoke(invocation)).thenReturn(appResponse);
-        when(invoker.getInterface()).thenReturn(DemoService.class);
 
         Result newResult = exceptionFilter.invoke(invoker, invocation);
 
@@ -127,9 +125,8 @@ class ExceptionFilterTest {
         mockRpcResult.setException(new HessianException("hessian"));
         Result mockAsyncResult = AsyncRpcResult.newDefaultAsyncResult(mockRpcResult, invocation);
 
-        Invoker<DemoService> invoker = mock(Invoker.class);
+        Invoker<DemoService> invoker = mockInvoker();
         when(invoker.invoke(invocation)).thenReturn(mockAsyncResult);
-        when(invoker.getInterface()).thenReturn(DemoService.class);
 
         Result asyncResult = exceptionFilter.invoke(invoker, invocation);
 
