diff --git a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java
index a2f6b14..ad70db7 100644
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/ConnectivityValidationTest.java
@@ -73,21 +73,21 @@ class ConnectivityValidationTest {
     @BeforeEach
     public void setup() {
         ApplicationModel.defaultModel().getBeanFactory().registerBean(MetricsDispatcher.class);
-        invoker1 = Mockito.mock(Invoker.class);
-        invoker2 = Mockito.mock(Invoker.class);
-        invoker3 = Mockito.mock(Invoker.class);
-        invoker4 = Mockito.mock(Invoker.class);
-        invoker5 = Mockito.mock(Invoker.class);
-        invoker6 = Mockito.mock(Invoker.class);
-        invoker7 = Mockito.mock(Invoker.class);
-        invoker8 = Mockito.mock(Invoker.class);
-        invoker9 = Mockito.mock(Invoker.class);
-        invoker10 = Mockito.mock(Invoker.class);
-        invoker11 = Mockito.mock(Invoker.class);
-        invoker12 = Mockito.mock(Invoker.class);
-        invoker13 = Mockito.mock(Invoker.class);
-        invoker14 = Mockito.mock(Invoker.class);
-        invoker15 = Mockito.mock(Invoker.class);
+        invoker1 = createInvoker();
+        invoker2 = createInvoker();
+        invoker3 = createInvoker();
+        invoker4 = createInvoker();
+        invoker5 = createInvoker();
+        invoker6 = createInvoker();
+        invoker7 = createInvoker();
+        invoker8 = createInvoker();
+        invoker9 = createInvoker();
+        invoker10 = createInvoker();
+        invoker11 = createInvoker();
+        invoker12 = createInvoker();
+        invoker13 = createInvoker();
+        invoker14 = createInvoker();
+        invoker15 = createInvoker();
 
         configInvoker(invoker1);
         configInvoker(invoker2);
@@ -123,7 +123,15 @@ class ConnectivityValidationTest {
 
     private void configInvoker(Invoker invoker) {
         when(invoker.getUrl()).thenReturn(URL.valueOf(""));
-        when(invoker.isAvailable()).thenReturn(true);
+        setAvailable(invoker, true);
+    }
+
+    private Invoker createInvoker() {
+        return Mockito.mock(Invoker.class);
+    }
+
+    private void setAvailable(Invoker invoker, boolean available) {
+        when(invoker.isAvailable()).thenReturn(available);
     }
 
     @BeforeAll
@@ -146,16 +154,16 @@ class ConnectivityValidationTest {
         Assertions.assertNotNull(
                 clusterInvoker.select(loadBalance, invocation, directory.list(invocation), Collections.emptyList()));
 
-        when(invoker1.isAvailable()).thenReturn(false);
-        when(invoker2.isAvailable()).thenReturn(false);
-        when(invoker3.isAvailable()).thenReturn(false);
-        when(invoker4.isAvailable()).thenReturn(false);
-        when(invoker5.isAvailable()).thenReturn(false);
+        setAvailable(invoker1, false);
+        setAvailable(invoker2, false);
+        setAvailable(invoker3, false);
+        setAvailable(invoker4, false);
+        setAvailable(invoker5, false);
 
         clusterInvoker.select(loadBalance, invocation, directory.list(invocation), Collections.emptyList());
         Assertions.assertEquals(0, directory.list(invocation).size());
 
-        when(invoker1.isAvailable()).thenReturn(true);
+        setAvailable(invoker1, true);
         Set<Invoker> invokerSet = new HashSet<>();
         invokerSet.add(invoker1);
         waitRefresh(invokerSet);
@@ -163,7 +171,7 @@ class ConnectivityValidationTest {
         Assertions.assertNotNull(
                 clusterInvoker.select(loadBalance, invocation, directory.list(invocation), Collections.emptyList()));
 
-        when(invoker2.isAvailable()).thenReturn(true);
+        setAvailable(invoker2, true);
         invokerSet.add(invoker2);
         waitRefresh(invokerSet);
         Assertions.assertEquals(2, directory.list(invocation).size());
@@ -172,14 +180,14 @@ class ConnectivityValidationTest {
 
         invokerList.remove(invoker5);
         directory.notify(invokerList);
-        when(invoker2.isAvailable()).thenReturn(true);
+        setAvailable(invoker2, true);
         waitRefresh(invokerSet);
         Assertions.assertEquals(2, directory.list(invocation).size());
         Assertions.assertNotNull(
                 clusterInvoker.select(loadBalance, invocation, directory.list(invocation), Collections.emptyList()));
 
-        when(invoker3.isAvailable()).thenReturn(true);
-        when(invoker4.isAvailable()).thenReturn(true);
+        setAvailable(invoker3, true);
+        setAvailable(invoker4, true);
         invokerSet.add(invoker3);
         invokerSet.add(invoker4);
         waitRefresh(invokerSet);
@@ -200,14 +208,14 @@ class ConnectivityValidationTest {
 
         Assertions.assertEquals(2, directory.list(invocation).size());
 
-        when(invoker1.isAvailable()).thenReturn(false);
+        setAvailable(invoker1, false);
         Assertions.assertEquals(
                 invoker2,
                 clusterInvoker.select(
                         loadBalance, invocation, directory.list(invocation), Collections.singletonList(invoker2)));
         Assertions.assertEquals(1, directory.list(invocation).size());
 
-        when(invoker1.isAvailable()).thenReturn(true);
+        setAvailable(invoker1, true);
         Set<Invoker> invokerSet = new HashSet<>();
         invokerSet.add(invoker1);
         waitRefresh(invokerSet);
@@ -234,20 +242,20 @@ class ConnectivityValidationTest {
 
         Assertions.assertEquals(15, directory.list(invocation).size());
 
-        when(invoker2.isAvailable()).thenReturn(false);
-        when(invoker3.isAvailable()).thenReturn(false);
-        when(invoker4.isAvailable()).thenReturn(false);
-        when(invoker5.isAvailable()).thenReturn(false);
-        when(invoker6.isAvailable()).thenReturn(false);
-        when(invoker7.isAvailable()).thenReturn(false);
-        when(invoker8.isAvailable()).thenReturn(false);
-        when(invoker9.isAvailable()).thenReturn(false);
-        when(invoker10.isAvailable()).thenReturn(false);
-        when(invoker11.isAvailable()).thenReturn(false);
-        when(invoker12.isAvailable()).thenReturn(false);
-        when(invoker13.isAvailable()).thenReturn(false);
-        when(invoker14.isAvailable()).thenReturn(false);
-        when(invoker15.isAvailable()).thenReturn(false);
+        setAvailable(invoker2, false);
+        setAvailable(invoker3, false);
+        setAvailable(invoker4, false);
+        setAvailable(invoker5, false);
+        setAvailable(invoker6, false);
+        setAvailable(invoker7, false);
+        setAvailable(invoker8, false);
+        setAvailable(invoker9, false);
+        setAvailable(invoker10, false);
+        setAvailable(invoker11, false);
+        setAvailable(invoker12, false);
+        setAvailable(invoker13, false);
+        setAvailable(invoker14, false);
+        setAvailable(invoker15, false);
         for (int i = 0; i < 15; i++) {
             clusterInvoker.select(loadBalance, invocation, directory.list(invocation), Collections.emptyList());
         }
@@ -258,25 +266,25 @@ class ConnectivityValidationTest {
                             loadBalance, invocation, directory.list(invocation), Collections.emptyList()));
         }
 
-        when(invoker1.isAvailable()).thenReturn(false);
+        setAvailable(invoker1, false);
         clusterInvoker.select(loadBalance, invocation, directory.list(invocation), Collections.emptyList());
         Assertions.assertEquals(0, directory.list(invocation).size());
 
-        when(invoker1.isAvailable()).thenReturn(true);
-        when(invoker2.isAvailable()).thenReturn(true);
-        when(invoker3.isAvailable()).thenReturn(true);
-        when(invoker4.isAvailable()).thenReturn(true);
-        when(invoker5.isAvailable()).thenReturn(true);
-        when(invoker6.isAvailable()).thenReturn(true);
-        when(invoker7.isAvailable()).thenReturn(true);
-        when(invoker8.isAvailable()).thenReturn(true);
-        when(invoker9.isAvailable()).thenReturn(true);
-        when(invoker10.isAvailable()).thenReturn(true);
-        when(invoker11.isAvailable()).thenReturn(true);
-        when(invoker12.isAvailable()).thenReturn(true);
-        when(invoker13.isAvailable()).thenReturn(true);
-        when(invoker14.isAvailable()).thenReturn(true);
-        when(invoker15.isAvailable()).thenReturn(true);
+        setAvailable(invoker1, true);
+        setAvailable(invoker2, true);
+        setAvailable(invoker3, true);
+        setAvailable(invoker4, true);
+        setAvailable(invoker5, true);
+        setAvailable(invoker6, true);
+        setAvailable(invoker7, true);
+        setAvailable(invoker8, true);
+        setAvailable(invoker9, true);
+        setAvailable(invoker10, true);
+        setAvailable(invoker11, true);
+        setAvailable(invoker12, true);
+        setAvailable(invoker13, true);
+        setAvailable(invoker14, true);
+        setAvailable(invoker15, true);
 
         Set<Invoker> invokerSet = new HashSet<>();
         invokerSet.add(invoker1);
