diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
index c6718e9..0c82bfd 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
@@ -20,6 +20,7 @@ import org.apache.dubbo.rpc.Invoker;
 import org.apache.dubbo.rpc.model.ServiceDescriptor;
 
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -28,6 +29,12 @@ import static org.junit.jupiter.api.Assertions.fail;
 class StubSuppliersTest {
 
     private final String serviceName = "service";
+    private Invoker<?> invokerMock;
+
+    @BeforeEach
+    void setUp() {
+        invokerMock = Mockito.mock(Invoker.class);
+    }
 
     @Test
     void addDescriptor() {
@@ -38,17 +45,15 @@ class StubSuppliersTest {
 
     @Test
     void addSupplier() {
-        Invoker<?> invoker = Mockito.mock(Invoker.class);
         ServiceDescriptor descriptor = Mockito.mock(ServiceDescriptor.class);
-        StubSuppliers.addSupplier(serviceName, i -> invoker);
-        Assertions.assertEquals(invoker, StubSuppliers.createStub(serviceName, invoker));
+        StubSuppliers.addSupplier(serviceName, i -> invokerMock);
+        Assertions.assertEquals(invokerMock, StubSuppliers.createStub(serviceName, invokerMock));
     }
 
     @Test
     void createStub() {
-        Invoker<?> invoker = Mockito.mock(Invoker.class);
         try {
-            StubSuppliers.createStub(serviceName + 1, invoker);
+            StubSuppliers.createStub(serviceName + 1, invokerMock);
             fail();
         } catch (IllegalStateException e) {
             // pass
