diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
index be23099..632de9c 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
@@ -30,7 +30,6 @@ import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
 /**
  * Tests for FailsafeErrorTypeAwareLogger to test whether it 'ignores' exceptions thrown by logger or not.
@@ -64,9 +63,7 @@ class FailsafeErrorTypeAwareLoggerTest {
 
     @Test
     void testSuccessLogger() {
-        Logger successLogger = mock(Logger.class);
-        when(successLogger.isErrorEnabled()).thenReturn(true);
-        when(successLogger.isWarnEnabled()).thenReturn(true);
+        Logger successLogger = MockLoggerWithEnabledLevels.createMockSuccessLogger();
         FailsafeErrorTypeAwareLogger failsafeLogger = new FailsafeErrorTypeAwareLogger(successLogger);
 
         failsafeLogger.error(REGISTRY_ADDRESS_INVALID, "Registry center", "May be it's offline.", "error");
diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
index 65de0c9..3c28499 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
@@ -67,9 +67,7 @@ class FailsafeLoggerTest {
 
     @Test
     void testSuccessLogger() {
-        Logger successLogger = mock(Logger.class);
-        Mockito.when(successLogger.isErrorEnabled()).thenReturn(true);
-        Mockito.when(successLogger.isWarnEnabled()).thenReturn(true);
+        Logger successLogger = MockLoggerWithEnabledLevels.createMockSuccessLogger();
         Mockito.when(successLogger.isInfoEnabled()).thenReturn(true);
         Mockito.when(successLogger.isDebugEnabled()).thenReturn(true);
         Mockito.when(successLogger.isTraceEnabled()).thenReturn(true);
