diff --git a/dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/status/DataSourceStatusCheckerTest.java b/dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/status/DataSourceStatusCheckerTest.java
index 7edf411..5dcc8ce 100644
--- a/dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/status/DataSourceStatusCheckerTest.java
+++ b/dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/status/DataSourceStatusCheckerTest.java
@@ -60,6 +60,12 @@ class DataSourceStatusCheckerTest {
         Mockito.reset(applicationContext);
     }
 
+    private static DataSource createMockDataSource(Connection connection) throws SQLException {
+        DataSource dataSource = mock(DataSource.class);
+        given(dataSource.getConnection()).willReturn(connection);
+        return dataSource;
+    }
+
     @Test
     void testWithoutApplicationContext() {
         Status status = dataSourceStatusChecker.check();
@@ -81,9 +87,8 @@ class DataSourceStatusCheckerTest {
     @Test
     void testWithDatasourceHasNextResult() throws SQLException {
         Map<String, DataSource> map = new HashMap<String, DataSource>();
-        DataSource dataSource = mock(DataSource.class);
         Connection connection = mock(Connection.class, Answers.RETURNS_DEEP_STUBS);
-        given(dataSource.getConnection()).willReturn(connection);
+        DataSource dataSource = createMockDataSource(connection);
         given(connection.getMetaData().getTypeInfo().next()).willReturn(true);
 
         map.put("mockDatabase", dataSource);
@@ -97,9 +102,8 @@ class DataSourceStatusCheckerTest {
     @Test
     void testWithDatasourceNotHasNextResult() throws SQLException {
         Map<String, DataSource> map = new HashMap<String, DataSource>();
-        DataSource dataSource = mock(DataSource.class);
         Connection connection = mock(Connection.class, Answers.RETURNS_DEEP_STUBS);
-        given(dataSource.getConnection()).willReturn(connection);
+        DataSource dataSource = createMockDataSource(connection);
         given(connection.getMetaData().getTypeInfo().next()).willReturn(false);
 
         map.put("mockDatabase", dataSource);
