diff --git a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyChannelTest.java b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyChannelTest.java
index 3d51c14..c893739 100644
--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyChannelTest.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyChannelTest.java
@@ -81,12 +81,10 @@ class NettyChannelTest {
         Mockito.when(channel.eventLoop()).thenReturn(Mockito.mock(EventLoop.class));
         Mockito.when(channel.alloc()).thenReturn(PooledByteBufAllocator.DEFAULT);
         NettyChannel nettyChannel = NettyChannel.getOrAddChannel(channel, url, channelHandler);
-        ChannelPromise future = Mockito.mock(ChannelPromise.class);
+        ChannelPromise future = NettyTestUtils.mockChannelPromise();
         Mockito.when(future.await(1000)).thenReturn(true);
-        Mockito.when(future.cause()).thenReturn(null);
         Mockito.when(channel.writeAndFlush(Mockito.any())).thenReturn(future);
         Mockito.when(channel.newPromise()).thenReturn(future);
-        Mockito.when(future.addListener(Mockito.any())).thenReturn(future);
         nettyChannel.send("msg", true);
 
         NettyChannel finalNettyChannel = nettyChannel;
@@ -109,11 +107,9 @@ class NettyChannelTest {
                 },
                 "in timeout(1000ms) limit");
 
-        ChannelPromise channelPromise = Mockito.mock(ChannelPromise.class);
+        ChannelPromise channelPromise = NettyTestUtils.mockChannelPromise();
         Mockito.when(channel.newPromise()).thenReturn(channelPromise);
         Mockito.when(channelPromise.await(1000)).thenReturn(true);
-        Mockito.when(channelPromise.cause()).thenReturn(null);
-        Mockito.when(channelPromise.addListener(Mockito.any())).thenReturn(channelPromise);
         finalNettyChannel.send("msg", true);
         ArgumentCaptor<GenericFutureListener> listenerArgumentCaptor =
                 ArgumentCaptor.forClass(GenericFutureListener.class);
diff --git a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandlerTest.java b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandlerTest.java
index 8db75f0..a8bff19 100644
--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandlerTest.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyClientHandlerTest.java
@@ -32,8 +32,6 @@ import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 
 class NettyClientHandlerTest {
 
@@ -48,11 +46,9 @@ class NettyClientHandlerTest {
         Mockito.when(channel.eventLoop()).thenReturn(new NioEventLoopGroup().next());
         Mockito.when(channel.alloc()).thenReturn(PooledByteBufAllocator.DEFAULT);
 
-        ChannelPromise future = mock(ChannelPromise.class);
-        when(channel.writeAndFlush(any())).thenReturn(future);
-        when(future.cause()).thenReturn(null);
-        when(channel.newPromise()).thenReturn(future);
-        when(future.addListener(Mockito.any())).thenReturn(future);
+        ChannelPromise future = NettyTestUtils.mockChannelPromise();
+        Mockito.when(channel.writeAndFlush(any())).thenReturn(future);
+        Mockito.when(channel.newPromise()).thenReturn(future);
 
         NettyClientHandler nettyClientHandler = new NettyClientHandler(url, handler);
 
