diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerTest.java
index 66a0376..b746eed 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerTest.java
@@ -180,14 +180,8 @@ class ServiceInstancesChangedListenerTest {
         metadataInfo_444 = JsonUtils.toJavaObject(metadata_444, MetadataInfo.class);
         metadataInfo_555_tri = JsonUtils.toJavaObject(metadata_555_triple, MetadataInfo.class);
 
-        serviceDiscovery = Mockito.mock(ServiceDiscovery.class);
-        when(serviceDiscovery.getUrl()).thenReturn(registryURL);
-
-        when(serviceDiscovery.getRemoteMetadata(eq("111"), anyList())).thenReturn(metadataInfo_111);
-        when(serviceDiscovery.getRemoteMetadata(eq("222"), anyList())).thenReturn(metadataInfo_222);
-        when(serviceDiscovery.getRemoteMetadata(eq("333"), anyList())).thenReturn(metadataInfo_333);
-        when(serviceDiscovery.getRemoteMetadata(eq("444"), anyList())).thenReturn(MetadataInfo.EMPTY);
-        when(serviceDiscovery.getRemoteMetadata(eq("555"), anyList())).thenReturn(metadataInfo_555_tri);
+        serviceDiscovery = MockServiceDiscoveryWithMetadata.createMockServiceDiscovery(
+                registryURL, metadataInfo_111, metadataInfo_222, metadataInfo_333, metadataInfo_555_tri);
     }
 
     @BeforeEach
diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerWithoutEmptyProtectTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerWithoutEmptyProtectTest.java
index 942eaca..5973d10 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerWithoutEmptyProtectTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListenerWithoutEmptyProtectTest.java
@@ -179,14 +179,8 @@ public class ServiceInstancesChangedListenerWithoutEmptyProtectTest {
         metadataInfo_444 = JsonUtils.toJavaObject(metadata_444, MetadataInfo.class);
         metadataInfo_555_tri = JsonUtils.toJavaObject(metadata_555_triple, MetadataInfo.class);
 
-        serviceDiscovery = Mockito.mock(ServiceDiscovery.class);
-        when(serviceDiscovery.getUrl()).thenReturn(registryURL);
-
-        when(serviceDiscovery.getRemoteMetadata(eq("111"), anyList())).thenReturn(metadataInfo_111);
-        when(serviceDiscovery.getRemoteMetadata(eq("222"), anyList())).thenReturn(metadataInfo_222);
-        when(serviceDiscovery.getRemoteMetadata(eq("333"), anyList())).thenReturn(metadataInfo_333);
-        when(serviceDiscovery.getRemoteMetadata(eq("444"), anyList())).thenReturn(MetadataInfo.EMPTY);
-        when(serviceDiscovery.getRemoteMetadata(eq("555"), anyList())).thenReturn(metadataInfo_555_tri);
+        serviceDiscovery = MockServiceDiscoveryWithMetadata.createMockServiceDiscovery(
+                registryURL, metadataInfo_111, metadataInfo_222, metadataInfo_333, metadataInfo_555_tri);
     }
 
     @BeforeEach
