diff --git a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
index a1ef5cb..966cf15 100644
--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
@@ -103,8 +103,7 @@ class MigrationInvokerTest {
         MigrationInvoker<?> migrationInvoker =
                 new MigrationInvoker<>(registryProtocol, null, null, DemoService.class, null, consumerURL);
 
-        MigrationRule migrationRule = Mockito.mock(MigrationRule.class);
-        Mockito.when(migrationRule.getForce(Mockito.any())).thenReturn(true);
+        MigrationRule migrationRule = mockMigrationRule(true);
         migrationInvoker.migrateToForceInterfaceInvoker(migrationRule);
         migrationInvoker.setMigrationStep(MigrationStep.FORCE_INTERFACE);
         migrationInvoker.invoke(null);
@@ -279,8 +278,7 @@ class MigrationInvokerTest {
         MigrationInvoker<?> migrationInvoker =
                 new MigrationInvoker<>(registryProtocol, null, null, DemoService.class, null, consumerURL);
 
-        MigrationRule migrationRule = Mockito.mock(MigrationRule.class);
-        Mockito.when(migrationRule.getForce(Mockito.any())).thenReturn(true);
+        MigrationRule migrationRule = mockMigrationRule(true);
         migrationInvoker.migrateToApplicationFirstInvoker(migrationRule);
         migrationInvoker.setMigrationStep(MigrationStep.APPLICATION_FIRST);
         migrationInvoker.invoke(null);
@@ -302,4 +300,10 @@ class MigrationInvokerTest {
         // 独立线程invoker状态切换
 
     }
+
+    private MigrationRule mockMigrationRule(boolean force) {
+        MigrationRule migrationRule = Mockito.mock(MigrationRule.class);
+        Mockito.when(migrationRule.getForce(Mockito.any())).thenReturn(force);
+        return migrationRule;
+    }
 }
