diff --git a/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanExporterTest.java b/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanExporterTest.java
index effc700..b083ca7 100644
--- a/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanExporterTest.java
+++ b/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanExporterTest.java
@@ -25,16 +25,13 @@ import io.opentelemetry.sdk.trace.export.SpanExporter;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
-import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 class ZipkinSpanExporterTest {
 
     @Test
     void getSpanExporter() {
-        ExporterConfig.ZipkinConfig zipkinConfig = mock(ExporterConfig.ZipkinConfig.class);
-        when(zipkinConfig.getEndpoint()).thenReturn("http://localhost:9411/api/v2/spans");
-        when(zipkinConfig.getConnectTimeout()).thenReturn(Duration.ofSeconds(5));
+        ExporterConfig.ZipkinConfig zipkinConfig = ZipkinTestUtils.mockZipkinConfig();
         when(zipkinConfig.getReadTimeout()).thenReturn(Duration.ofSeconds(5));
 
         SpanExporter spanExporter = ZipkinSpanExporter.getSpanExporter(ApplicationModel.defaultModel(), zipkinConfig);
diff --git a/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanHandlerTest.java b/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanHandlerTest.java
index 0ebbe7b..3c3e155 100644
--- a/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanHandlerTest.java
+++ b/dubbo-metrics/dubbo-tracing/src/test/java/org/apache/dubbo/tracing/exporter/zipkin/ZipkinSpanHandlerTest.java
@@ -19,22 +19,15 @@ package org.apache.dubbo.tracing.exporter.zipkin;
 import org.apache.dubbo.config.nested.ExporterConfig;
 import org.apache.dubbo.rpc.model.ApplicationModel;
 
-import java.time.Duration;
-
 import brave.handler.SpanHandler;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
 class ZipkinSpanHandlerTest {
 
     @Test
     void getSpanHandler() {
-        ExporterConfig.ZipkinConfig zipkinConfig = mock(ExporterConfig.ZipkinConfig.class);
-        when(zipkinConfig.getEndpoint()).thenReturn("http://localhost:9411/api/v2/spans");
-        when(zipkinConfig.getConnectTimeout()).thenReturn(Duration.ofSeconds(5));
+        ExporterConfig.ZipkinConfig zipkinConfig = ZipkinTestUtils.mockZipkinConfig();
 
         SpanHandler spanHandler = ZipkinSpanHandler.getSpanHandler(ApplicationModel.defaultModel(), zipkinConfig);
         Assertions.assertNotNull(spanHandler);
