diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/CompatibleFilterFilterTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/CompatibleFilterFilterTest.java
index d5931e7..a7e8364 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/CompatibleFilterFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/CompatibleFilterFilterTest.java
@@ -50,10 +50,8 @@ class CompatibleFilterFilterTest {
 
     @Test
     void testInvokerGeneric() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("$enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Enum.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                RpcInvocation.class, "$enumlength", new Class<?>[] {Enum.class});
 
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
@@ -70,10 +68,8 @@ class CompatibleFilterFilterTest {
 
     @Test
     void testResultHasException() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Enum.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                RpcInvocation.class, "enumlength", new Class<?>[] {Enum.class});
 
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
@@ -91,10 +87,8 @@ class CompatibleFilterFilterTest {
 
     @Test
     void testInvokerJsonPojoSerialization() throws Exception {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Type[].class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                RpcInvocation.class, "enumlength", new Class<?>[] {Type[].class});
 
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
@@ -114,10 +108,8 @@ class CompatibleFilterFilterTest {
 
     @Test
     void testInvokerNonJsonEnumSerialization() throws Exception {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Type[].class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                RpcInvocation.class, "enumlength", new Class<?>[] {Type[].class});
 
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
@@ -137,10 +129,8 @@ class CompatibleFilterFilterTest {
 
     @Test
     void testInvokerNonJsonNonPojoSerialization() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("echo");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {String.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                RpcInvocation.class, "echo", new Class<?>[] {String.class});
 
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
@@ -157,10 +147,8 @@ class CompatibleFilterFilterTest {
 
     @Test
     void testInvokerNonJsonPojoSerialization() {
-        invocation = mock(RpcInvocation.class);
-        given(invocation.getMethodName()).willReturn("echo");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {String.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                RpcInvocation.class, "echo", new Class<?>[] {String.class});
 
         invoker = mock(Invoker.class);
         given(invoker.isAvailable()).willReturn(true);
diff --git a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
index bdb39c6..746b19c 100644
--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/filter/ContextFilterTest.java
@@ -47,10 +47,8 @@ class ContextFilterTest {
     @SuppressWarnings("unchecked")
     @Test
     void testSetContext() {
-        invocation = mock(Invocation.class);
-        given(invocation.getMethodName()).willReturn("$enumlength");
-        given(invocation.getParameterTypes()).willReturn(new Class<?>[] {Enum.class});
-        given(invocation.getArguments()).willReturn(new Object[] {"hello"});
+        invocation = MockInvocationWithMethodDetails.createMockInvocation(
+                Invocation.class, "$enumlength", new Class<?>[] {Enum.class});
         given(invocation.getObjectAttachments()).willReturn(null);
 
         invoker = mock(Invoker.class);
