diff --git a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
index f6f1f17..edf779f 100644
--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
@@ -38,7 +38,7 @@ class NacosNamingServiceWrapperTest {
     @Test
     void testSubscribe() throws NacosException {
         NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
-        NamingService namingService = Mockito.mock(NamingService.class);
+        NamingService namingService = mockNamingService();
         Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
@@ -63,8 +63,8 @@ class NacosNamingServiceWrapperTest {
     @Test
     void testSubscribeMultiManager() throws NacosException {
         NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
-        NamingService namingService1 = Mockito.mock(NamingService.class);
-        NamingService namingService2 = Mockito.mock(NamingService.class);
+        NamingService namingService1 = mockNamingService();
+        NamingService namingService2 = mockNamingService();
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
@@ -94,7 +94,7 @@ class NacosNamingServiceWrapperTest {
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -221,7 +221,7 @@ class NacosNamingServiceWrapperTest {
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -391,7 +391,7 @@ class NacosNamingServiceWrapperTest {
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -559,7 +559,7 @@ class NacosNamingServiceWrapperTest {
 
         CountDownLatch startLatch = new CountDownLatch(1);
         CountDownLatch stopLatch = new CountDownLatch(1);
-        NamingService namingService = Mockito.mock(NamingService.class);
+        NamingService namingService = mockNamingService();
         Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper =
@@ -654,6 +654,10 @@ class NacosNamingServiceWrapperTest {
         Assertions.assertNotEquals(instancesInfo, instancesInfoNew);
     }
 
+    private NamingService mockNamingService() {
+        return Mockito.mock(NamingService.class);
+    }
+
     @Test
     void testSuccess() {
         NamingService namingService = new MockNamingService() {
