diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ExecutorUtilTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ExecutorUtilTest.java
index 8924f78..26364cb 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ExecutorUtilTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ExecutorUtilTest.java
@@ -47,9 +47,8 @@ class ExecutorUtilTest {
 
     @Test
     void testGracefulShutdown1() throws Exception {
-        ExecutorService executor = Mockito.mock(ExecutorService.class);
+        ExecutorService executor = createMockExecutor();
         when(executor.isTerminated()).thenReturn(false, true);
-        when(executor.awaitTermination(20, TimeUnit.MILLISECONDS)).thenReturn(false);
         ExecutorUtil.gracefulShutdown(executor, 20);
         verify(executor).shutdown();
         verify(executor).shutdownNow();
@@ -57,9 +56,8 @@ class ExecutorUtilTest {
 
     @Test
     void testGracefulShutdown2() throws Exception {
-        ExecutorService executor = Mockito.mock(ExecutorService.class);
+        ExecutorService executor = createMockExecutor();
         when(executor.isTerminated()).thenReturn(false, false, false);
-        when(executor.awaitTermination(20, TimeUnit.MILLISECONDS)).thenReturn(false);
         when(executor.awaitTermination(10, TimeUnit.MILLISECONDS)).thenReturn(false, true);
         ExecutorUtil.gracefulShutdown(executor, 20);
 
@@ -84,4 +82,10 @@ class ExecutorUtilTest {
         url = ExecutorUtil.setThreadName(url, "default-name");
         assertThat(url.getParameter(THREAD_NAME_KEY), equalTo("custom-thread-localhost:1234"));
     }
+
+    private static ExecutorService createMockExecutor() throws InterruptedException {
+        ExecutorService executor = Mockito.mock(ExecutorService.class);
+        when(executor.awaitTermination(20, TimeUnit.MILLISECONDS)).thenReturn(false);
+        return executor;
+    }
 }
