diff --git a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/BroadCastClusterInvokerTest.java b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/BroadCastClusterInvokerTest.java
index 5e63e97..2017a13 100644
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/BroadCastClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/BroadCastClusterInvokerTest.java
@@ -34,7 +34,6 @@ import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.BDDMockito.given;
-import static org.mockito.Mockito.mock;
 
 /**
  * @see BroadcastClusterInvoker
@@ -53,17 +52,13 @@ class BroadCastClusterInvokerTest {
     @BeforeEach
     public void setUp() throws Exception {
 
-        dic = mock(Directory.class);
-
         invoker1 = new MockInvoker();
         invoker2 = new MockInvoker();
         invoker3 = new MockInvoker();
         invoker4 = new MockInvoker();
 
         url = URL.valueOf("test://127.0.0.1:8080/test");
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
-        given(dic.getInterface()).willReturn(DemoService.class);
+        dic = MockDemoServiceDirectory.createMockDemoServiceDirectory(url);
 
         invocation = new RpcInvocation();
         invocation.setMethodName("test");
diff --git a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailSafeClusterInvokerTest.java b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailSafeClusterInvokerTest.java
index 54f4f84..d000ecc 100644
--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailSafeClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailSafeClusterInvokerTest.java
@@ -57,12 +57,8 @@ class FailSafeClusterInvokerTest {
     public void setUp() throws Exception {
         RpcContext.removeServiceContext();
 
-        dic = mock(Directory.class);
-
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
+        dic = MockDemoServiceDirectory.createMockDemoServiceDirectory(url);
         given(dic.list(invocation)).willReturn(invokers);
-        given(dic.getInterface()).willReturn(DemoService.class);
         invocation.setMethodName("method1");
 
         invokers.add(invoker);
@@ -100,12 +96,8 @@ class FailSafeClusterInvokerTest {
 
     @Test
     void testNoInvoke() {
-        dic = mock(Directory.class);
-
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
+        dic = MockDemoServiceDirectory.createMockDemoServiceDirectory(url);
         given(dic.list(invocation)).willReturn(null);
-        given(dic.getInterface()).willReturn(DemoService.class);
 
         invocation.setMethodName("method1");
 
