diff --git a/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java b/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
index 526d5fa..7797e81 100644
--- a/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
@@ -24,6 +24,8 @@ import java.util.Map;
 import java.util.UUID;
 
 class RequestInformationTest {
+    private final SerializationWriter serializationWriterMock = mock(SerializationWriter.class);
+
     @Test
     void ThrowsInvalidOperationExceptionWhenBaseUrlNotSet() {
         // Arrange as the request builders would
@@ -334,13 +336,12 @@ class RequestInformationTest {
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
-        final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
+        final RequestAdapter requestAdapterMock = createMockRequestAdapter(serializationWriterMock);
         requestInfo.setContentFromParsable(
                 requestAdapterMock, "application/json", new TestEntity());
 
-        verify(writerMock, times(1)).writeObjectValue(any(), any(TestEntity.class));
-        verify(writerMock, never())
+        verify(serializationWriterMock, times(1)).writeObjectValue(any(), any(TestEntity.class));
+        verify(serializationWriterMock, never())
                 .writeCollectionOfObjectValues(anyString(), any(ArrayList.class));
     }
 
@@ -350,13 +351,12 @@ class RequestInformationTest {
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
-        final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
+        final RequestAdapter requestAdapterMock = createMockRequestAdapter(serializationWriterMock);
         requestInfo.setContentFromParsable(
                 requestAdapterMock, "application/json", new TestEntity[] {new TestEntity()});
 
-        verify(writerMock, never()).writeObjectValue(any(), any(TestEntity.class));
-        verify(writerMock, times(1)).writeCollectionOfObjectValues(any(), any(Iterable.class));
+        verify(serializationWriterMock, never()).writeObjectValue(any(), any(TestEntity.class));
+        verify(serializationWriterMock, times(1)).writeCollectionOfObjectValues(any(), any(Iterable.class));
     }
 
     @Test
@@ -365,13 +365,12 @@ class RequestInformationTest {
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
-        final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
+        final RequestAdapter requestAdapterMock = createMockRequestAdapter(serializationWriterMock);
         requestInfo.setContentFromScalarCollection(
                 requestAdapterMock, "application/json", new String[] {"foo"});
 
-        verify(writerMock, never()).writeStringValue(any(), anyString());
-        verify(writerMock, times(1)).writeCollectionOfPrimitiveValues(any(), any(Iterable.class));
+        verify(serializationWriterMock, never()).writeStringValue(any(), anyString());
+        verify(serializationWriterMock, times(1)).writeCollectionOfPrimitiveValues(any(), any(Iterable.class));
     }
 
     @Test
@@ -380,12 +379,11 @@ class RequestInformationTest {
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
-        final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
+        final RequestAdapter requestAdapterMock = createMockRequestAdapter(serializationWriterMock);
         requestInfo.setContentFromScalar(requestAdapterMock, "application/json", "foo");
 
-        verify(writerMock, times(1)).writeStringValue(any(), anyString());
-        verify(writerMock, never()).writeCollectionOfPrimitiveValues(any(), any(Iterable.class));
+        verify(serializationWriterMock, times(1)).writeStringValue(any(), anyString());
+        verify(serializationWriterMock, never()).writeCollectionOfPrimitiveValues(any(), any(Iterable.class));
     }
 
     @Test
@@ -394,8 +392,7 @@ class RequestInformationTest {
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate =
                 "http://localhost/{URITemplate}/ParameterMapping?IsCaseSensitive={IsCaseSensitive}";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
-        final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
+        final RequestAdapter requestAdapterMock = createMockRequestAdapter(serializationWriterMock);
 
         final MultipartBody multipartBody = new MultipartBody();
         multipartBody.requestAdapter = requestAdapterMock;
