diff --git a/components/abstractions/src/test/java/com/microsoft/kiota/serialization/DeserializationHelpersTest.java b/components/abstractions/src/test/java/com/microsoft/kiota/serialization/DeserializationHelpersTest.java
index a47f9be..eb65e7d 100644
--- a/components/abstractions/src/test/java/com/microsoft/kiota/serialization/DeserializationHelpersTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/serialization/DeserializationHelpersTest.java
@@ -35,6 +35,18 @@ class DeserializationHelpersTest {
         return nullValue();
     }
 
+    private static ParseNode createMockParseNode() {
+        final ParseNode mockParseNode = mock(ParseNode.class);
+        when(mockParseNode.getObjectValue(any()))
+                .thenReturn(
+                        new TestEntity() {
+                            {
+                                setId("123");
+                            }
+                        });
+        return mockParseNode;
+    }
+
     @Test
     void defensive() {
         assertThrows(
@@ -88,14 +100,7 @@ class DeserializationHelpersTest {
     @Test
     void deserializesObjectWithoutReflection() throws IOException {
         final String strValue = "{'id':'123'}";
-        final ParseNode mockParseNode = mock(ParseNode.class);
-        when(mockParseNode.getObjectValue(any()))
-                .thenReturn(
-                        new TestEntity() {
-                            {
-                                setId("123");
-                            }
-                        });
+        final ParseNode mockParseNode = createMockParseNode();
         final ParseNodeFactory mockParseNodeFactory = mock(ParseNodeFactory.class);
         when(mockParseNodeFactory.getParseNode(any(), any())).thenReturn(mockParseNode);
         ParseNodeFactoryRegistry.defaultInstance.contentTypeAssociatedFactories.put(
@@ -110,14 +115,7 @@ class DeserializationHelpersTest {
     @Test
     void deserializesObjectWithReflection() throws IOException {
         final String strValue = "{'id':'123'}";
-        final ParseNode mockParseNode = mock(ParseNode.class);
-        when(mockParseNode.getObjectValue(any()))
-                .thenReturn(
-                        new TestEntity() {
-                            {
-                                setId("123");
-                            }
-                        });
+        final ParseNode mockParseNode = createMockParseNode();
         final ParseNodeFactory mockParseNodeFactory = mock(ParseNodeFactory.class);
         when(mockParseNodeFactory.getParseNode(any(), any())).thenReturn(mockParseNode);
         ParseNodeFactoryRegistry.defaultInstance.contentTypeAssociatedFactories.put(
