diff --git a/components/abstractions/src/test/java/com/microsoft/kiota/MultiPartBodyTest.java b/components/abstractions/src/test/java/com/microsoft/kiota/MultiPartBodyTest.java
index 3549ef2..2d91b2c 100644
--- a/components/abstractions/src/test/java/com/microsoft/kiota/MultiPartBodyTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/MultiPartBodyTest.java
@@ -16,6 +16,10 @@ class MultiPartBodyTest {
         return java.util.Collections.<String, T>emptyMap().get("missing");
     }
 
+    private SerializationWriter serializationWriter() {
+        return mock(SerializationWriter.class);
+    }
+
     @Test
     void defensive() {
         final MultipartBody multipartBody = new MultipartBody();
@@ -38,14 +42,14 @@ class MultiPartBodyTest {
     @Test
     void requiresRequestAdapter() {
         final MultipartBody multipartBody = new MultipartBody();
-        final SerializationWriter writer = mock(SerializationWriter.class);
+        final SerializationWriter writer = serializationWriter();
         assertThrows(IllegalStateException.class, () -> multipartBody.serialize(writer));
     }
 
     @Test
     void requiresPartsForSerialization() {
         final MultipartBody multipartBody = new MultipartBody();
-        final SerializationWriter writer = mock(SerializationWriter.class);
+        final SerializationWriter writer = serializationWriter();
         final RequestAdapter requestAdapter = mock(RequestAdapter.class);
         multipartBody.requestAdapter = requestAdapter;
         assertThrows(IllegalStateException.class, () -> multipartBody.serialize(writer));
@@ -76,7 +80,7 @@ class MultiPartBodyTest {
     @Test
     void notAddFilename() {
         final MultipartBody multipartBody = new MultipartBody();
-        final SerializationWriter writer = mock(SerializationWriter.class);
+        final SerializationWriter writer = serializationWriter();
         multipartBody.requestAdapter = mock(RequestAdapter.class);
         multipartBody.addOrReplacePart("foo", "bar", "baz");
         multipartBody.serialize(writer);
@@ -86,7 +90,7 @@ class MultiPartBodyTest {
     @Test
     void addFilename() {
         final MultipartBody multipartBody = new MultipartBody();
-        final SerializationWriter writer = mock(SerializationWriter.class);
+        final SerializationWriter writer = serializationWriter();
         multipartBody.requestAdapter = mock(RequestAdapter.class);
         multipartBody.addOrReplacePart("foo", "bar", "baz", "image.png");
         multipartBody.serialize(writer);
