diff --git a/components/authentication/azure/src/test/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProviderTest.java b/components/authentication/azure/src/test/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProviderTest.java
index 081fe16..6808f98 100644
--- a/components/authentication/azure/src/test/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProviderTest.java
+++ b/components/authentication/azure/src/test/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProviderTest.java
@@ -42,7 +42,7 @@ public class AzureIdentityAccessTokenProviderTest {
     @ParameterizedTest
     @ValueSource(strings = {"http://graph.microsoft.com/me"})
     void testNonLocalhostHttpUrlIsInvalid(String urlString) {
-        var tokenCredential = mock(TokenCredential.class);
+        var tokenCredential = createTokenCredential();
         var accessTokenProvider =
                 new AzureIdentityAccessTokenProvider(tokenCredential, new String[] {}, "");
         assertThrows(
@@ -54,7 +54,7 @@ public class AzureIdentityAccessTokenProviderTest {
 
     @Test
     void testKeepUserProvidedScopes() throws URISyntaxException {
-        var tokenCredential = mock(TokenCredential.class);
+        var tokenCredential = createTokenCredential();
         String[] userProvidedScopes = {
             "https://graph.microsoft.com/User.Read", "https://graph.microsoft.com/Application.Read"
         };
@@ -66,7 +66,7 @@ public class AzureIdentityAccessTokenProviderTest {
 
     @Test
     void testConfigureDefaultScopeWhenScopesNotProvided() throws URISyntaxException {
-        var tokenCredential = mock(TokenCredential.class);
+        var tokenCredential = createTokenCredential();
         var accessTokenProvider =
                 new AzureIdentityAccessTokenProvider(tokenCredential, new String[] {});
         assertScopes(
@@ -79,7 +79,7 @@ public class AzureIdentityAccessTokenProviderTest {
     @NullAndEmptySource
     void testConfigureDefaultScopeWhenScopesNullOrEmpty(String[] nullOrEmptyUserProvidedScopes)
             throws URISyntaxException {
-        var tokenCredential = mock(TokenCredential.class);
+        var tokenCredential = createTokenCredential();
         var accessTokenProvider =
                 new AzureIdentityAccessTokenProvider(
                         tokenCredential, new String[] {}, nullOrEmptyUserProvidedScopes);
@@ -104,4 +104,8 @@ public class AzureIdentityAccessTokenProviderTest {
         List<String> actualScopes = tokenRequestContextArgumentCaptor.getValue().getScopes();
         assertLinesMatch(Arrays.asList(expectedScopes), actualScopes);
     }
+
+    private static TokenCredential createTokenCredential() {
+        return mock(TokenCredential.class);
+    }
 }
