--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
@@ -43,7 +43,7 @@
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
-        EventListener eventListener = Mockito.mock(EventListener.class);
+        EventListener eventListener = mockEventListener();
         nacosNamingServiceWrapper.subscribe("service_name", "test", eventListener);
         Mockito.verify(namingService, Mockito.times(1)).subscribe("service_name", "test", eventListener);
 
@@ -68,7 +68,7 @@
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
-        EventListener eventListener = Mockito.mock(EventListener.class);
+        EventListener eventListener = mockEventListener();
         Mockito.when(connectionManager.getNamingService()).thenReturn(namingService1);
         nacosNamingServiceWrapper.subscribe("service_name", "test", eventListener);
         Mockito.verify(namingService1, Mockito.times(1)).subscribe("service_name", "test", eventListener);
@@ -86,6 +86,10 @@
         nacosNamingServiceWrapper.unsubscribe("service_name", "mock", eventListener);
         Mockito.verify(namingService1, Mockito.times(0)).unsubscribe("service_name", "mock", eventListener);
         Mockito.verify(namingService2, Mockito.times(0)).unsubscribe("service_name", "mock", eventListener);
+    }
+
+    private EventListener mockEventListener() {
+        return Mockito.mock(EventListener.class);
     }
 
     @Test
