--- a/dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/status/DataSourceStatusCheckerTest.java
+++ b/dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/status/DataSourceStatusCheckerTest.java
@@ -82,9 +82,8 @@
     void testWithDatasourceHasNextResult() throws SQLException {
         Map<String, DataSource> map = new HashMap<String, DataSource>();
         DataSource dataSource = mock(DataSource.class);
-        Connection connection = mock(Connection.class, Answers.RETURNS_DEEP_STUBS);
+        Connection connection = mockConnectionWithTypeInfoNextResult(true);
         given(dataSource.getConnection()).willReturn(connection);
-        given(connection.getMetaData().getTypeInfo().next()).willReturn(true);
 
         map.put("mockDatabase", dataSource);
         given(applicationContext.getBeansOfType(eq(DataSource.class), anyBoolean(), anyBoolean()))
@@ -98,9 +97,8 @@
     void testWithDatasourceNotHasNextResult() throws SQLException {
         Map<String, DataSource> map = new HashMap<String, DataSource>();
         DataSource dataSource = mock(DataSource.class);
-        Connection connection = mock(Connection.class, Answers.RETURNS_DEEP_STUBS);
+        Connection connection = mockConnectionWithTypeInfoNextResult(false);
         given(dataSource.getConnection()).willReturn(connection);
-        given(connection.getMetaData().getTypeInfo().next()).willReturn(false);
 
         map.put("mockDatabase", dataSource);
         given(applicationContext.getBeansOfType(eq(DataSource.class), anyBoolean(), anyBoolean()))
@@ -109,4 +107,10 @@
 
         assertThat(status.getLevel(), is(Status.Level.ERROR));
     }
+
+    private Connection mockConnectionWithTypeInfoNextResult(boolean hasNextResult) throws SQLException {
+        Connection connection = mock(Connection.class, Answers.RETURNS_DEEP_STUBS);
+        given(connection.getMetaData().getTypeInfo().next()).willReturn(hasNextResult);
+        return connection;
+    }
 }
