--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvokerTest.java
@@ -91,6 +91,12 @@
         }
     }
 
+    private void mockInvoker(Invoker invoker, String group) {
+        given(invoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, group));
+        given(invoker.getInterface()).willReturn(MenuService.class);
+        given(invoker.invoke(invocation)).willReturn(new AppResponse());
+    }
+
     @BeforeEach
     public void setUp() throws Exception {
 
@@ -190,14 +196,10 @@
         given(invocation.getObjectAttachments()).willReturn(new HashMap<>());
         given(invocation.getInvoker()).willReturn(firstInvoker);
 
-        given(firstInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "first"));
-        given(firstInvoker.getInterface()).willReturn(MenuService.class);
-        given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(firstInvoker, "first");
         given(firstInvoker.isAvailable()).willReturn(true);
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(secondInvoker, "second");
         given(secondInvoker.isAvailable()).willReturn(true);
 
         given(directory.list(invocation)).willReturn(new ArrayList() {
@@ -300,16 +302,12 @@
         given(invocation.getObjectAttachments()).willReturn(new HashMap<>());
         given(invocation.getInvoker()).willReturn(firstInvoker);
 
-        given(firstInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "first"));
-        given(firstInvoker.getInterface()).willReturn(MenuService.class);
-        given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(firstInvoker, "first");
         given(firstInvoker.isAvailable()).willReturn(true);
         given(firstInvoker.invoke(invocation))
                 .willThrow(new RpcException(RpcException.NO_INVOKER_AVAILABLE_AFTER_FILTER));
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(secondInvoker, "second");
         given(secondInvoker.isAvailable()).willReturn(true);
         given(secondInvoker.invoke(invocation))
                 .willThrow(new RpcException(RpcException.NO_INVOKER_AVAILABLE_AFTER_FILTER));
@@ -357,15 +355,11 @@
         given(invocation.getObjectAttachments()).willReturn(new HashMap<>());
         given(invocation.getInvoker()).willReturn(firstInvoker);
 
-        given(firstInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "first"));
-        given(firstInvoker.getInterface()).willReturn(MenuService.class);
-        given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(firstInvoker, "first");
         given(firstInvoker.isAvailable()).willReturn(true);
         given(firstInvoker.invoke(invocation)).willThrow(new RpcException(RpcException.NETWORK_EXCEPTION));
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(secondInvoker, "second");
         given(secondInvoker.isAvailable()).willReturn(true);
         given(secondInvoker.invoke(invocation)).willThrow(new RpcException(RpcException.NETWORK_EXCEPTION));
 
@@ -405,14 +399,10 @@
         given(invocation.getObjectAttachments()).willReturn(new HashMap<>());
         given(invocation.getInvoker()).willReturn(firstInvoker);
 
-        given(firstInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "first"));
-        given(firstInvoker.getInterface()).willReturn(MenuService.class);
-        given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(firstInvoker, "first");
         given(firstInvoker.isAvailable()).willReturn(true);
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(secondInvoker, "second");
         given(secondInvoker.isAvailable()).willReturn(true);
 
         given(directory.list(invocation)).willReturn(new ArrayList() {
@@ -512,13 +502,8 @@
         given(invocation.getObjectAttachments()).willReturn(new HashMap<>());
         given(invocation.getInvoker()).willReturn(firstInvoker);
 
-        given(firstInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "first"));
-        given(firstInvoker.getInterface()).willReturn(MenuService.class);
-        given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
-
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
+        mockInvoker(firstInvoker, "first");
+        mockInvoker(secondInvoker, "second");
 
         given(directory.list(invocation)).willReturn(new ArrayList() {
 
