--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
@@ -37,9 +37,8 @@
 class NacosNamingServiceWrapperTest {
     @Test
     void testSubscribe() throws NacosException {
-        NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
         NamingService namingService = Mockito.mock(NamingService.class);
-        Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
+        NacosConnectionManager connectionManager = mockConnectionManager(namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
@@ -555,12 +554,10 @@
 
     @Test
     void testConcurrency() throws NacosException, InterruptedException {
-        NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
-
         CountDownLatch startLatch = new CountDownLatch(1);
         CountDownLatch stopLatch = new CountDownLatch(1);
         NamingService namingService = Mockito.mock(NamingService.class);
-        Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
+        NacosConnectionManager connectionManager = mockConnectionManager(namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper =
                 new NacosNamingServiceWrapper(connectionManager, false, 0, 0);
@@ -748,4 +745,10 @@
             Assertions.fail(e);
         }
     }
+
+    private NacosConnectionManager mockConnectionManager(NamingService namingService) {
+        NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
+        Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
+        return connectionManager;
+    }
 }
