--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
@@ -118,7 +118,7 @@
                     }
                 };
 
-        MigrationInvoker<?> migrationInvoker = Mockito.mock(MigrationInvoker.class);
+        MigrationInvoker<?> migrationInvoker = createMigrationInvoker();
         migrationRuleListener.getHandlers().put(migrationInvoker, handler);
 
         countDownLatch.countDown();
@@ -152,7 +152,7 @@
 
         MigrationRuleListener migrationRuleListener =
                 new MigrationRuleListener(ApplicationModel.defaultModel().getDefaultModule());
-        MigrationInvoker<?> migrationInvoker = Mockito.mock(MigrationInvoker.class);
+        MigrationInvoker<?> migrationInvoker = createMigrationInvoker();
         migrationRuleListener.getHandlers().put(migrationInvoker, handler);
         migrationRuleListener.onRefer(null, migrationInvoker, consumerURL, null);
         // check migration happened after invoker referred
@@ -203,7 +203,7 @@
                 new MigrationRuleListener(ApplicationModel.defaultModel().getDefaultModule());
         Assertions.assertNotNull(migrationRuleListener.localRuleMigrationFuture);
         Assertions.assertNull(migrationRuleListener.ruleMigrationFuture);
-        MigrationInvoker<?> migrationInvoker = Mockito.mock(MigrationInvoker.class);
+        MigrationInvoker<?> migrationInvoker = createMigrationInvoker();
         MigrationInvoker<?> migrationInvoker2 = Mockito.mock(MigrationInvoker.class);
 
         // Remote rule will be applied when onRefer gets executed
@@ -243,4 +243,8 @@
         Assertions.assertEquals(
                 "APPLICATION_FIRST", captor_event.getValue().getStep().toString());
     }
+
+    private MigrationInvoker<?> createMigrationInvoker() {
+        return Mockito.mock(MigrationInvoker.class);
+    }
 }
