--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeServerTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeServerTest.java
@@ -51,10 +51,8 @@
         Channel channel2 = Mockito.mock(Channel.class);
         Channel exchangeChannel1 = new HeaderExchangeChannel(channel1);
         Channel exchangeChannel2 = new HeaderExchangeChannel(channel2);
-        Mockito.when(channel1.getAttribute(HeaderExchangeChannel.class.getName() + ".CHANNEL"))
-                .thenReturn(exchangeChannel1);
-        Mockito.when(channel2.getAttribute(HeaderExchangeChannel.class.getName() + ".CHANNEL"))
-                .thenReturn(exchangeChannel2);
+        mockExchangeChannel(channel1, exchangeChannel1);
+        mockExchangeChannel(channel2, exchangeChannel2);
         Collection<Channel> exChannels = Arrays.asList(exchangeChannel1, exchangeChannel2);
         Mockito.when(server.getChannels()).thenReturn(Arrays.asList(channel1, channel2));
         Assertions.assertEquals(headerExchangeServer.getChannels(), exChannels);
@@ -89,4 +87,9 @@
         Assertions.assertThrows(RemotingException.class, () -> headerExchangeServer.send("test"));
         Assertions.assertThrows(RemotingException.class, () -> headerExchangeServer.send("test", true));
     }
+
+    private void mockExchangeChannel(Channel channel, Channel exchangeChannel) {
+        Mockito.when(channel.getAttribute(HeaderExchangeChannel.class.getName() + ".CHANNEL"))
+                .thenReturn(exchangeChannel);
+    }
 }
