--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/AccessKeyAuthenticatorTest.java
@@ -100,17 +100,21 @@
     void testGetAccessKeyPairFailed() {
         URL url = URL.valueOf("dubbo://10.10.10.10:2181").addParameter(Constants.ACCESS_KEY_ID_KEY, "ak");
         AccessKeyAuthenticator helper = new AccessKeyAuthenticator(FrameworkModel.defaultModel());
-        Invocation invocation = mock(Invocation.class);
+        Invocation invocation = mockInvocation();
         assertThrows(RuntimeException.class, () -> helper.getAccessKeyPair(invocation, url));
     }
 
     @Test
     void testGetSignatureNoParameter() {
         URL url = mock(URL.class);
-        Invocation invocation = mock(Invocation.class);
+        Invocation invocation = mockInvocation();
         String secretKey = "123456";
         AccessKeyAuthenticator helper = new AccessKeyAuthenticator(FrameworkModel.defaultModel());
         String signature = helper.getSignature(url, invocation, secretKey, String.valueOf(System.currentTimeMillis()));
         assertNotNull(signature);
     }
+
+    private Invocation mockInvocation() {
+        return mock(Invocation.class);
+    }
 }
