--- a/components/abstractions/src/test/java/com/microsoft/kiota/MultiPartBodyTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/MultiPartBodyTest.java
@@ -1,5 +1,6 @@
 package com.microsoft.kiota;
 
+import org.mockito.Mock;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
@@ -12,6 +13,8 @@
 import org.junit.jupiter.api.Test;
 
 class MultiPartBodyTest {
+    @Mock
+    private final RequestAdapter requestAdapter;
     private static <T> T nullValue() {
         return java.util.Collections.<String, T>emptyMap().get("missing");
     }
@@ -46,7 +49,6 @@
     void requiresPartsForSerialization() {
         final MultipartBody multipartBody = new MultipartBody();
         final SerializationWriter writer = mock(SerializationWriter.class);
-        final RequestAdapter requestAdapter = mock(RequestAdapter.class);
         multipartBody.requestAdapter = requestAdapter;
         assertThrows(IllegalStateException.class, () -> multipartBody.serialize(writer));
     }
@@ -54,7 +56,6 @@
     @Test
     void addsPart() {
         final MultipartBody multipartBody = new MultipartBody();
-        final RequestAdapter requestAdapter = mock(RequestAdapter.class);
         multipartBody.requestAdapter = requestAdapter;
         multipartBody.addOrReplacePart("foo", "bar", "baz");
         final Object result = multipartBody.getPartValue("foo");
@@ -65,7 +66,6 @@
     @Test
     void removesPart() {
         final MultipartBody multipartBody = new MultipartBody();
-        final RequestAdapter requestAdapter = mock(RequestAdapter.class);
         multipartBody.requestAdapter = requestAdapter;
         multipartBody.addOrReplacePart("foo", "bar", "baz");
         multipartBody.removePart("FOO");
