--- a/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/RequestInformationTest.java
@@ -1,5 +1,6 @@
 package com.microsoft.kiota;
 
+import org.mockito.Mock;
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
@@ -24,6 +25,8 @@
 import java.util.UUID;
 
 class RequestInformationTest {
+    @Mock
+    private final SerializationWriter writerMock;
     @Test
     void ThrowsInvalidOperationExceptionWhenBaseUrlNotSet() {
         // Arrange as the request builders would
@@ -334,7 +337,6 @@
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
         final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
         requestInfo.setContentFromParsable(
                 requestAdapterMock, "application/json", new TestEntity());
@@ -350,7 +352,6 @@
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
         final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
         requestInfo.setContentFromParsable(
                 requestAdapterMock, "application/json", new TestEntity[] {new TestEntity()});
@@ -365,7 +366,6 @@
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
         final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
         requestInfo.setContentFromScalarCollection(
                 requestAdapterMock, "application/json", new String[] {"foo"});
@@ -380,7 +380,6 @@
         final RequestInformation requestInfo = new RequestInformation();
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate = "http://localhost/users";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
         final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
         requestInfo.setContentFromScalar(requestAdapterMock, "application/json", "foo");
 
@@ -394,7 +393,6 @@
         requestInfo.httpMethod = HttpMethod.POST;
         requestInfo.urlTemplate =
                 "http://localhost/{URITemplate}/ParameterMapping?IsCaseSensitive={IsCaseSensitive}";
-        final SerializationWriter writerMock = mock(SerializationWriter.class);
         final RequestAdapter requestAdapterMock = createMockRequestAdapter(writerMock);
 
         final MultipartBody multipartBody = new MultipartBody();
