--- a/components/authentication/azure/src/test/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProviderTest.java
+++ b/components/authentication/azure/src/test/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProviderTest.java
@@ -1,5 +1,6 @@
 package com.microsoft.kiota.authentication;
 
+import org.mockito.Mock;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertLinesMatch;
 import static org.junit.jupiter.api.Assertions.assertThrows;
@@ -24,6 +25,8 @@
 import java.util.List;
 
 public class AzureIdentityAccessTokenProviderTest {
+    @Mock
+    private TokenCredential tokenCredential;
 
     @ParameterizedTest
     @ValueSource(
@@ -42,7 +45,6 @@
     @ParameterizedTest
     @ValueSource(strings = {"http://graph.microsoft.com/me"})
     void testNonLocalhostHttpUrlIsInvalid(String urlString) {
-        var tokenCredential = mock(TokenCredential.class);
         var accessTokenProvider =
                 new AzureIdentityAccessTokenProvider(tokenCredential, new String[] {}, "");
         assertThrows(
@@ -54,7 +56,6 @@
 
     @Test
     void testKeepUserProvidedScopes() throws URISyntaxException {
-        var tokenCredential = mock(TokenCredential.class);
         String[] userProvidedScopes = {
             "https://graph.microsoft.com/User.Read", "https://graph.microsoft.com/Application.Read"
         };
@@ -66,7 +67,6 @@
 
     @Test
     void testConfigureDefaultScopeWhenScopesNotProvided() throws URISyntaxException {
-        var tokenCredential = mock(TokenCredential.class);
         var accessTokenProvider =
                 new AzureIdentityAccessTokenProvider(tokenCredential, new String[] {});
         assertScopes(
@@ -79,7 +79,6 @@
     @NullAndEmptySource
     void testConfigureDefaultScopeWhenScopesNullOrEmpty(String[] nullOrEmptyUserProvidedScopes)
             throws URISyntaxException {
-        var tokenCredential = mock(TokenCredential.class);
         var accessTokenProvider =
                 new AzureIdentityAccessTokenProvider(
                         tokenCredential, new String[] {}, nullOrEmptyUserProvidedScopes);
