--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -894,7 +894,7 @@
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
+        AddressType addressType = createMockAddressType(subnetType);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -909,7 +909,6 @@
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1001,7 +1000,7 @@
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
+        AddressType addressType = createMockAddressType(subnetType);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1015,7 +1014,6 @@
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1030,7 +1028,7 @@
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
         PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
         ActionListType actionListType = mock(ActionListType.class);
-        AddressType addressType = mock(AddressType.class);
+        AddressType addressType = createMockAddressType(subnetType);
         SubnetType subnetType = mock(SubnetType.class);
         PortType portType = mock(PortType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1044,7 +1042,6 @@
         when(policyRuleType.getActionList()).thenReturn(actionListType);
         when(actionListType.getSimpleAction()).thenReturn("pass");
         when(policyRuleType.getSrcAddresses()).thenReturn(List.of(addressType));
-        when(addressType.getSubnet()).thenReturn(subnetType);
         when(policyRuleType.getSrcPorts()).thenReturn(List.of(portType));
         when(policyRuleType.getDstAddresses()).thenReturn(List.of(addressType));
         when(policyRuleType.getDstPorts()).thenReturn(List.of(portType));
@@ -1668,4 +1665,9 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static AddressType createMockAddressType(SubnetType subnetType) {
+        AddressType addressType = createMockAddressType(subnetType);
+        return addressType;
+    }
 }
