--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -183,6 +183,8 @@
     @Mock
     TungstenApi tungstenApi;
 
+    private ApplicationPolicySet applicationPolicySet2Mock;
+
     TungstenResource tungstenResource;
 
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
@@ -192,6 +194,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        applicationPolicySet2Mock = mock(ApplicationPolicySet.class);
         tungstenResource = new TungstenResource();
         tungstenResource.tungstenApi = tungstenApi;
         ReflectionTestUtils.setField(tungstenResource, "vrouterPort", "9091");
@@ -1119,8 +1122,7 @@
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
-        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
+        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2Mock);
         when(tungstenApi.applyTungstenPolicyTag(anyString(), anyString())).thenReturn(true);
         when(tungstenApi.getTungstenObject(eq(Tag.class), anyString())).thenReturn(tag);
         when(tungstenApi.getBackRefFromVirtualNetwork(eq(VirtualNetwork.class), anyList())).thenAnswer(virtualNetworksAnswer);
@@ -1170,8 +1172,7 @@
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
-        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
+        Answer<List<ApiObjectBase>> applicationPoliciesAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2Mock);
         when(tungstenApi.removeTungstenTag(anyList(), anyList(), anyList(), anyString(), anyString(), anyString())).thenReturn(tag);
         when(tungstenApi.getBackRefFromVirtualNetwork(eq(VirtualNetwork.class), anyList())).thenAnswer(virtualNetworksAnswer);
         when(tungstenApi.getBackRefFromVirtualMachine(eq(VirtualMachine.class), anyList())).thenAnswer(virtualMachinesAnswer);
@@ -1242,8 +1243,7 @@
     public void executeRequestListTungstenApplicationPolicySetCommandTest() {
         TungstenCommand command = new ListTungstenApplicationPolicySetCommand("d64e6082-10b3-11ec-82a8-0242ac130003");
         ApplicationPolicySet applicationPolicySet1 = mock(ApplicationPolicySet.class);
-        ApplicationPolicySet applicationPolicySet2 = mock(ApplicationPolicySet.class);
-        Answer<List<ApiObjectBase>> applicationPolicySetsAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2);
+        Answer<List<ApiObjectBase>> applicationPolicySetsAnswer = setupApiObjectBaseListAnswer(applicationPolicySet1, applicationPolicySet2Mock);
         when(tungstenApi.listTungstenApplicationPolicySet(anyString())).thenAnswer(applicationPolicySetsAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
