--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -1332,17 +1332,15 @@
     @Test
     public void createTungstenApplicationPolicySetTest() {
         TungstenAnswer createTungstenApplicationPolicySetAnswer = MockTungstenAnswerFactory.get(true);
-        ApplicationPolicySet applicationPolicySet = mock(ApplicationPolicySet.class);
         ObjectReference<ApiPropertyBase> objectReference = mock(ObjectReference.class);
         ObjectReference<FirewallSequence> firewallSequenceObjectReference = mock(ObjectReference.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenApplicationPolicySetCommand.class), anyLong())).thenReturn(createTungstenApplicationPolicySetAnswer);
+        ApplicationPolicySet applicationPolicySet = createMockApplicationPolicySet(objectReference, firewallSequenceObjectReference);
         when(createTungstenApplicationPolicySetAnswer.getApiObjectBase()).thenReturn(applicationPolicySet);
-        when(applicationPolicySet.getTag()).thenReturn(List.of(objectReference));
         when(objectReference.getReferredName()).thenReturn(List.of("tag"));
-        when(applicationPolicySet.getFirewallPolicy()).thenReturn(List.of(firewallSequenceObjectReference));
         when(firewallSequenceObjectReference.getReferredName()).thenReturn(List.of("firewallrule"));
 
         assertNotNull(tungstenService.createTungstenApplicationPolicySet(1L, "test"));
@@ -1351,17 +1349,15 @@
     @Test
     public void listTungstenApplicationPolicySetTest() {
         TungstenAnswer listTungstenApplicationPolicySetAnswer = MockTungstenAnswerFactory.get(true);
-        ApplicationPolicySet applicationPolicySet = mock(ApplicationPolicySet.class);
         ObjectReference<ApiPropertyBase> objectReference = mock(ObjectReference.class);
         ObjectReference<FirewallSequence> firewallSequenceObjectReference = mock(ObjectReference.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenApplicationPolicySetCommand.class), anyLong())).thenReturn(listTungstenApplicationPolicySetAnswer);
+        ApplicationPolicySet applicationPolicySet = createMockApplicationPolicySet(objectReference, firewallSequenceObjectReference);
         doReturn(List.of(applicationPolicySet)).when(listTungstenApplicationPolicySetAnswer).getApiObjectBaseList();
-        when(applicationPolicySet.getTag()).thenReturn(List.of(objectReference));
         when(objectReference.getReferredName()).thenReturn(List.of("tag"));
-        when(applicationPolicySet.getFirewallPolicy()).thenReturn(List.of(firewallSequenceObjectReference));
         when(firewallSequenceObjectReference.getReferredName()).thenReturn(List.of("firewallrule"));
 
         assertNotNull(tungstenService.listTungstenApplicationPolicySet(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
@@ -1668,4 +1664,12 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static ApplicationPolicySet createMockApplicationPolicySet(ObjectReference<ApiPropertyBase> objectReference,
+            ObjectReference<FirewallSequence> firewallSequenceObjectReference) {
+        ApplicationPolicySet applicationPolicySet = mock(ApplicationPolicySet.class);
+        when(applicationPolicySet.getTag()).thenReturn(List.of(objectReference));
+        when(applicationPolicySet.getFirewallPolicy()).thenReturn(List.of(firewallSequenceObjectReference));
+        return applicationPolicySet;
+    }
 }
