--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/nicira/NiciraNvpApiTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/nicira/NiciraNvpApiTest.java
@@ -87,12 +87,18 @@
             .build();
     }
 
+    private static CloseableHttpResponse createMockCloseableHttpResponse() throws Exception {
+        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+
+        when(response.getStatusLine()).thenReturn(HTTP_200_REPSONSE);
+        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_LIST_JSON_RESPONSE));
+        return response;
+    }
+
     @Test
     @SuppressWarnings("unchecked")
     public void testFindSecurityProfile() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_200_REPSONSE);
-        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_LIST_JSON_RESPONSE));
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = spy(HttpClientHelper.createHttpClient(2));
         doReturn(response).when(httpClient).execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class));
         final NiciraNvpApi api = buildApi(httpClient);
@@ -118,9 +124,7 @@
     @Test
     @SuppressWarnings("unchecked")
     public void testFindSecurityProfileByUuid() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_200_REPSONSE);
-        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_LIST_JSON_RESPONSE));
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = spy(HttpClientHelper.createHttpClient(2));
         doReturn(response).when(httpClient).execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class));
         final NiciraNvpApi api = buildApi(httpClient);
