--- a/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/keycloak/KeycloakOAuth2ProviderTest.java
+++ b/plugins/user-authenticators/oauth2/src/test/java/org/apache/cloudstack/oauth2/keycloak/KeycloakOAuth2ProviderTest.java
@@ -88,15 +88,13 @@
     public void testVerifyCodeAndFetchEmailHttpError() throws IOException {
         when(oauthProviderDao.findByProviderAndDomainWithGlobalFallback("keycloak", null)).thenReturn(mockProviderVO);
 
-        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         StatusLine statusLine = mock(StatusLine.class);
 
         when(statusLine.getStatusCode()).thenReturn(400);
-        when(response.getStatusLine()).thenReturn(statusLine);
 
         HttpEntity entity = mock(HttpEntity.class);
         when(entity.getContent()).thenReturn(new ByteArrayInputStream("error".getBytes()));
-        when(response.getEntity()).thenReturn(entity);
+        CloseableHttpResponse response = createMockResponse(entity, statusLine);
 
         when(httpClient.execute(any(HttpPost.class))).thenReturn(response);
 
@@ -130,16 +128,14 @@
         String encodedPayload = Base64.getUrlEncoder().withoutPadding().encodeToString(payload.getBytes());
         String fakeJwt = encodedHeader + "." + encodedPayload + ".not-checked-signature";
 
-        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         StatusLine statusLine = mock(StatusLine.class);
         HttpEntity entity = mock(HttpEntity.class);
 
         when(statusLine.getStatusCode()).thenReturn(200);
-        when(response.getStatusLine()).thenReturn(statusLine);
+        CloseableHttpResponse response = createMockResponse(entity, statusLine);
 
         String jsonResponseBody = "{\"id_token\":\"" + fakeJwt + "\", \"access_token\":\"acc-123\"}";
         when(entity.getContent()).thenReturn(new ByteArrayInputStream(jsonResponseBody.getBytes(StandardCharsets.UTF_8)));
-        when(response.getEntity()).thenReturn(entity);
 
         when(httpClient.execute(any(HttpPost.class))).thenReturn(response);
 
@@ -165,16 +161,14 @@
         String encodedPayload = Base64.getUrlEncoder().withoutPadding().encodeToString(payload.getBytes());
         String fakeJwt = encodedHeader + "." + encodedPayload + ".not-checked-signature";
 
-        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         StatusLine statusLine = mock(StatusLine.class);
         HttpEntity entity = mock(HttpEntity.class);
 
         when(statusLine.getStatusCode()).thenReturn(200);
-        when(response.getStatusLine()).thenReturn(statusLine);
+        CloseableHttpResponse response = createMockResponse(entity, statusLine);
 
         String jsonResponseBody = "{\"id_token\":\"" + fakeJwt + "\", \"access_token\":\"acc-123\"}";
         when(entity.getContent()).thenReturn(new ByteArrayInputStream(jsonResponseBody.getBytes(StandardCharsets.UTF_8)));
-        when(response.getEntity()).thenReturn(entity);
 
         when(httpClient.execute(any(HttpPost.class))).thenReturn(response);
 
@@ -200,16 +194,14 @@
         String encodedPayload = Base64.getUrlEncoder().withoutPadding().encodeToString(payload.getBytes());
         String fakeJwt = encodedHeader + "." + encodedPayload + ".not-checked-signature";
 
-        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         StatusLine statusLine = mock(StatusLine.class);
         HttpEntity entity = mock(HttpEntity.class);
 
         when(statusLine.getStatusCode()).thenReturn(200);
-        when(response.getStatusLine()).thenReturn(statusLine);
+        CloseableHttpResponse response = createMockResponse(entity, statusLine);
 
         String jsonResponseBody = "{\"id_token\":\"" + fakeJwt + "\", \"access_token\":\"acc-123\"}";
         when(entity.getContent()).thenReturn(new ByteArrayInputStream(jsonResponseBody.getBytes(StandardCharsets.UTF_8)));
-        when(response.getEntity()).thenReturn(entity);
 
         when(httpClient.execute(any(HttpPost.class))).thenReturn(response);
 
@@ -222,4 +214,11 @@
     public void testGetDescription() {
         assertEquals("Keycloak OAuth2 Provider Plugin", provider.getDescription());
     }
+
+    private static CloseableHttpResponse createMockResponse(HttpEntity entity, StatusLine statusLine) {
+        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        when(response.getStatusLine()).thenReturn(statusLine);
+        when(response.getEntity()).thenReturn(entity);
+        return response;
+    }
 }
