--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
@@ -89,10 +89,9 @@
 
     @Test
     public void getSnhItfReqWithIOExceptionTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
 
         assertNull(introspectApiConnector.getSnhItfReq("948f421c-edde-4518-a391-09299cc25dc2"));
     }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
@@ -79,10 +79,9 @@
     @Test
     public void addPortWithExceptionTest() throws Exception {
         Port port = mock(Port.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
 
         assertFalse(vRouterApiConnector.addPort(port));
     }
@@ -118,10 +117,9 @@
 
     @Test
     public void deletePortWithExceptionTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
 
         assertFalse(vRouterApiConnector.deletePort("948f421c-edde-4518-a391-09299cc25dc2"));
     }
@@ -142,10 +140,9 @@
 
     @Test
     public void enablePortWithExceptionTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
 
         assertFalse(vRouterApiConnector.enablePort("948f421c-edde-4518-a391-09299cc25dc2"));
     }
@@ -166,10 +163,9 @@
 
     @Test
     public void disablePortWithExceptionTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
 
         assertFalse(vRouterApiConnector.disablePort("948f421c-edde-4518-a391-09299cc25dc2"));
     }
@@ -194,10 +190,9 @@
     public void addGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
         Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
 
         assertFalse(vRouterApiConnector.addGateway(Arrays.asList(gateway1, gateway2)));
     }
@@ -222,10 +217,9 @@
     public void deleteGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
         Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient();
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
 
         assertFalse(vRouterApiConnector.deleteGateway(Arrays.asList(gateway1, gateway2)));
     }
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/MockCloseableHttpClient.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/MockCloseableHttpClient.java
@@ -0,0 +1,19 @@
+package org.apache.cloudstack.network.tungsten.vrouter;
+
+import java.io.IOException;
+
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class MockCloseableHttpClient {
+
+    public static CloseableHttpClient createMockCloseableHttpClient() throws IOException {
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        when(httpClient.execute(any(HttpUriRequest.class))).thenThrow(IOException.class);
+        return httpClient;
+    }
+}
