--- a/utils/src/test/java/com/cloud/utils/rest/RESTServiceConnectorTest.java
+++ b/utils/src/test/java/com/cloud/utils/rest/RESTServiceConnectorTest.java
@@ -73,8 +73,7 @@
         final String newObjectJson = gson.toJson(newObject);
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
 
@@ -91,8 +90,7 @@
         final String newObjectJson = gson.toJson(newObject);
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
 
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
@@ -113,8 +111,7 @@
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
 
@@ -135,8 +132,7 @@
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
 
@@ -157,8 +153,7 @@
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getEntity()).thenReturn(entity);
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
 
@@ -176,8 +171,7 @@
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
 
@@ -197,8 +191,7 @@
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getEntity()).thenReturn(new StringEntity(newObjectJson));
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final RESTServiceConnector connector = new RESTServiceConnector.Builder().client(restClient).build();
 
@@ -233,8 +226,7 @@
         final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
         when(response.getStatusLine()).thenReturn(HTTP_200_STATUS_LINE);
         when(response.getEntity()).thenReturn(new StringEntity("{results: [{field : \"SomeValue\"}], results_count: 1}"));
-        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        final CloseableHttpClient httpClient = createMockCloseableHttpClient(response);
         final RestClient restClient = new BasicRestClient(httpClient, HttpClientContext.create(), "localhost");
         final Class<? extends CollectionType> clazzListOfTestPojo = new ObjectMapper().getTypeFactory().constructCollectionType(List.class, TestPojo.class).getClass();
         final RESTServiceConnector connector = new RESTServiceConnector.Builder()
@@ -243,6 +235,12 @@
             .build();
 
         connector.executeRetrieveObject(TestPojo.class, "/somepath");
+    }
+
+    private static CloseableHttpClient createMockCloseableHttpClient(final CloseableHttpResponse response) {
+        final CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class))).thenReturn(response);
+        return httpClient;
     }
 
     class NiciraList<T> {
