--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/nicira/NiciraNvpApiTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/nicira/NiciraNvpApiTest.java
@@ -87,6 +87,13 @@
             .build();
     }
 
+    private static CloseableHttpResponse createMockCloseableHttpResponse() throws Exception {
+        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        when(response.getStatusLine()).thenReturn(HTTP_201_REPSONSE);
+        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_JSON_RESPONSE));
+        return response;
+    }
+
     @Test
     @SuppressWarnings("unchecked")
     public void testFindSecurityProfile() throws Exception {
@@ -146,9 +153,7 @@
 
     @Test
     public void testCreateSecurityProfile() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_201_REPSONSE);
-        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_JSON_RESPONSE));
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = spy(HttpClientHelper.createHttpClient(2));
         doReturn(response).when(httpClient).execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class));
         final NiciraNvpApi api = buildApi(httpClient);
@@ -165,9 +170,7 @@
 
     @Test
     public void testUpdateSecurityProfile() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_201_REPSONSE);
-        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_JSON_RESPONSE));
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = spy(HttpClientHelper.createHttpClient(2));
         doReturn(response).when(httpClient).execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class));
         final NiciraNvpApi api = buildApi(httpClient);
@@ -181,9 +184,7 @@
 
     @Test
     public void testDeleteSecurityProfile() throws Exception {
-        final CloseableHttpResponse response = mock(CloseableHttpResponse.class);
-        when(response.getStatusLine()).thenReturn(HTTP_201_REPSONSE);
-        when(response.getEntity()).thenReturn(new StringEntity(SEC_PROFILE_JSON_RESPONSE));
+        final CloseableHttpResponse response = createMockCloseableHttpResponse();
         final CloseableHttpClient httpClient = spy(HttpClientHelper.createHttpClient(2));
         doReturn(response).when(httpClient).execute(any(HttpHost.class), any(HttpRequest.class), any(HttpClientContext.class));
         final NiciraNvpApi api = buildApi(httpClient);
