--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/IntrospectApiConnectorImplTest.java
@@ -69,15 +69,14 @@
     @Test
     public void getSnhItfReqTest() throws Exception {
         Document document = mock(Document.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
         HttpEntity httpEntity = mock(HttpEntity.class);
         InputStream inputStream = mock(InputStream.class);
         DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
         DocumentBuilder documentBuilder = mock(DocumentBuilder.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         when(httpEntity.getContent()).thenReturn(inputStream);
         documentBuilderFactoryMocked.when(DocumentBuilderFactory::newInstance).thenReturn(documentBuilderFactory);
@@ -99,14 +98,13 @@
 
     @Test
     public void getSnhItfReqWithParserConfigurationExceptionTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
         HttpEntity httpEntity = mock(HttpEntity.class);
         InputStream inputStream = mock(InputStream.class);
         DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
         documentBuilderFactoryMocked.when(DocumentBuilderFactory::newInstance).thenReturn(documentBuilderFactory);
         when(documentBuilderFactory.newDocumentBuilder()).thenThrow(ParserConfigurationException.class);
 
@@ -115,15 +113,14 @@
 
     @Test
     public void getSnhItfReqWithSAXExceptionTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
         HttpEntity httpEntity = mock(HttpEntity.class);
         InputStream inputStream = mock(InputStream.class);
         DocumentBuilderFactory documentBuilderFactory = mock(DocumentBuilderFactory.class);
         DocumentBuilder documentBuilder = mock(DocumentBuilder.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         when(httpEntity.getContent()).thenReturn(inputStream);
         documentBuilderFactoryMocked.when(DocumentBuilderFactory::newInstance).thenReturn(documentBuilderFactory);
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
@@ -64,12 +64,11 @@
     @Test
     public void addPortTest() throws Exception {
         Port port = mock(Port.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
@@ -90,12 +89,11 @@
     @Test
     public void addPortWithFailTest() throws Exception {
         Port port = mock(Port.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{error:404}");
 
@@ -104,12 +102,11 @@
 
     @Test
     public void deletePortTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
@@ -128,12 +125,11 @@
 
     @Test
     public void enablePortTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
@@ -152,12 +148,11 @@
 
     @Test
     public void disablePortTest() throws Exception {
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
@@ -178,12 +173,11 @@
     public void addGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
         Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
@@ -206,12 +200,11 @@
     public void deleteGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
         Gateway gateway2 = mock(Gateway.class);
-        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
-        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
-        HttpEntity httpEntity = mock(HttpEntity.class);
-
-        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
-        when(httpClient.execute(any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        CloseableHttpClient httpClient = MockCloseableHttpClient.createMockCloseableHttpClient(closeableHttpResponse);
+        HttpEntity httpEntity = mock(HttpEntity.class);
+
+        httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
         when(closeableHttpResponse.getEntity()).thenReturn(httpEntity);
         entityUtilsMocked.when(()-> EntityUtils.toString(any(HttpEntity.class))).thenReturn("{}");
 
--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/MockCloseableHttpClient.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/MockCloseableHttpClient.java
@@ -0,0 +1,14 @@
+package org.apache.cloudstack.network.tungsten.vrouter;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.mockito.Mockito;
+
+public class MockCloseableHttpClient {
+    public static CloseableHttpClient createMockCloseableHttpClient(CloseableHttpResponse closeableHttpResponse) throws java.io.IOException {
+        CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+        Mockito.when(httpClient.execute(Mockito.any(HttpUriRequest.class))).thenReturn(closeableHttpResponse);
+        return httpClient;
+    }
+}
