--- a/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resourcelimit/ResourceLimitManagerImplTest.java
@@ -137,6 +137,8 @@
     EntityManager entityManager;
     @Mock
     SnapshotDataStoreDao snapshotDataStoreDao;
+    @Mock
+    private Account accountMock;
 
     private CallContext callContext;
     private List<String> hostTags = List.of("htag1", "htag2", "htag3");
@@ -291,9 +293,8 @@
         Mockito.when(serviceOffering.getCpu()).thenReturn(2);
         Mockito.when(serviceOffering.getRamSize()).thenReturn(256);
         Mockito.when(template.getTemplateTag()).thenReturn(hostTags.get(0));
-        Account account = Mockito.mock(Account.class);
         try (MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
-            resourceLimitManager.checkVmResourceLimit(account, true, serviceOffering, template, reservations);
+            resourceLimitManager.checkVmResourceLimit(accountMock, true, serviceOffering, template, reservations);
             List<String> tags = new ArrayList<>();
             tags.add(null);
             tags.add(hostTags.get(0));
@@ -310,9 +311,8 @@
         DiskOffering diskOffering = Mockito.mock(DiskOffering.class);
         Mockito.when(diskOffering.getTags()).thenReturn(checkTag);
         Mockito.when(diskOffering.getTagsArray()).thenReturn(new String[]{checkTag});
-        Account account = Mockito.mock(Account.class);
         try (MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
-            resourceLimitManager.checkVolumeResourceLimit(account, true, 100L, diskOffering, reservations);
+            resourceLimitManager.checkVolumeResourceLimit(accountMock, true, 100L, diskOffering, reservations);
             Assert.assertEquals(2, reservations.size());
         } catch (ResourceAllocationException e) {
             Assert.fail("Exception encountered: " + e.getMessage());
@@ -690,8 +690,7 @@
         Long domainId = 2L;
         Resource.ResourceType type = Resource.ResourceType.secondary_storage;
         Mockito.when(domainDao.findById(domainId)).thenReturn(Mockito.mock(DomainVO.class));
-        Account removedAccount = Mockito.mock(Account.class);
-        Mockito.when(entityManager.findByIdIncludingRemoved(Account.class, accountId)).thenReturn(removedAccount);
+        Mockito.when(entityManager.findByIdIncludingRemoved(Account.class, accountId)).thenReturn(accountMock);
         Mockito.doNothing().when(resourceLimitManager).removeResourceLimitAndCountForNonMatchingTags(Mockito.anyLong(),
                 Mockito.any(), Mockito.anyList(), Mockito.anyList());
         Mockito.doReturn(1L).when(resourceLimitManager).recalculateAccountResourceCount(accountId, type, null);
@@ -701,7 +700,7 @@
         List<? extends ResourceCount> result = resourceLimitManager.recalculateResourceCount(accountId, domainId, type.getOrdinal(), null);
 
         Assert.assertEquals(1, result.size());
-        Mockito.verify(accountManager).verifyCallerPrivilegeForUserOrAccountOperations(removedAccount);
+        Mockito.verify(accountManager).verifyCallerPrivilegeForUserOrAccountOperations(accountMock);
     }
 
     @Test(expected = InvalidParameterValueException.class)
@@ -979,7 +978,6 @@
     @Test
     public void testCheckVolumeResourceCount() throws ResourceAllocationException {
         List<Reserver> reservations = new ArrayList<>();
-        Account account = Mockito.mock(Account.class);
         String tag = "tag";
         long delta = 10L;
         Mockito.doReturn(new ArrayList<>()).when(resourceLimitManager)
@@ -994,7 +992,7 @@
                 .getResourceLimitStorageTagsForResourceCountOperation(Mockito.anyBoolean(), Mockito.any(DiskOffering.class), Mockito.any());
 
         try (MockedConstruction<CheckedReservation> mockCheckedReservation = Mockito.mockConstruction(CheckedReservation.class)) {
-            resourceLimitManager.checkVolumeResourceLimit(account, false, delta, Mockito.mock(DiskOffering.class), reservations);
+            resourceLimitManager.checkVolumeResourceLimit(accountMock, false, delta, Mockito.mock(DiskOffering.class), reservations);
             Assert.assertEquals(2,  reservations.size());
         }
     }
@@ -1309,8 +1307,7 @@
         Long maxGB = 10L;
         Long maxBytes = maxGB * Resource.ResourceType.bytesToGiB;
 
-        Account account = mock(Account.class);
-        when(entityManager.findById(Account.class, accountId)).thenReturn(account);
+        when(entityManager.findById(Account.class, accountId)).thenReturn(accountMock);
         ResourceLimitVO resourceLimitVO = mock(ResourceLimitVO.class);
         when(resourceLimitVO.getId()).thenReturn(resourceLimitId);
         when(resourceLimitDao.findByOwnerIdAndTypeAndTag(accountId, Resource.ResourceOwnerType.Account, Resource.ResourceType.backup_storage, null)).thenReturn(resourceLimitVO);
