--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -432,16 +432,14 @@
 
     @Test
     public void testVolumeInactiveVmNotFound() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        VolumeVO volume = createMockVolume();
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(null);
         assertTrue(volumeOrchestrator.volumeInactive(volume));
     }
 
     @Test
     public void testVolumeInactiveVmStopped() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        VolumeVO volume = createMockVolume();
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Stopped);
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(vm);
@@ -450,8 +448,7 @@
 
     @Test
     public void testVolumeInactiveVmDestroyed() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        VolumeVO volume = createMockVolume();
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Destroyed);
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(vm);
@@ -460,8 +457,7 @@
 
     @Test
     public void testVolumeInactiveVmRunning() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        VolumeVO volume = createMockVolume();
         UserVm vm = Mockito.mock(UserVm.class);
         Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Running); // Active state
         Mockito.when(entityMgr.findById(UserVm.class, MOCK_VM_ID)).thenReturn(vm);
@@ -478,16 +474,14 @@
 
     @Test
     public void testGetVmNameOnVolumeVmNotFound() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        VolumeVO volume = createMockVolume();
         Mockito.when(entityMgr.findById(VirtualMachine.class, MOCK_VM_ID)).thenReturn(null);
         Assert.assertNull(volumeOrchestrator.getVmNameOnVolume(volume));
     }
 
     @Test
     public void testGetVmNameOnVolumeSuccess() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        VolumeVO volume = createMockVolume();
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(vm.getInstanceName()).thenReturn(MOCK_VM_NAME);
         Mockito.when(entityMgr.findById(VirtualMachine.class, MOCK_VM_ID)).thenReturn(vm);
@@ -581,6 +575,12 @@
 
         assertTrue("VM restart should be true if local disk is recreatable",
                 volumeOrchestrator.canVmRestartOnAnotherServer(MOCK_VM_ID));
+    }
+
+    private static VolumeVO createMockVolume() {
+        VolumeVO volume = Mockito.mock(VolumeVO.class);
+        Mockito.when(volume.getInstanceId()).thenReturn(MOCK_VM_ID);
+        return volume;
     }
 
     private void overrideDefaultConfigValue(final ConfigKey configKey, final String value) throws IllegalAccessException, NoSuchFieldException {
