--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -959,8 +959,7 @@
         Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(ipAddresses.getIp4Address()).thenReturn(null);
         URI broadcastUri = URI.create("vlan://123");
-        NicVO nic = mock(NicVO.class);
-        Mockito.when(nic.getBroadcastUri()).thenReturn(broadcastUri);
+        NicVO nic = createMockNic();
         String macAddress = "02:01:01:82:00:01";
         int deviceId = 1;
         Integer networkRate = 200;
@@ -997,7 +996,7 @@
         Network.IpAddresses ipAddresses = Mockito.mock(Network.IpAddresses.class);
         Mockito.when(ipAddresses.getIp4Address()).thenReturn(ipAddress);
         URI broadcastUri = URI.create("vlan://123");
-        NicVO nic = mock(NicVO.class);
+        NicVO nic = createMockNic();
         Mockito.when(nic.getBroadcastUri()).thenReturn(broadcastUri);
         String macAddress = "02:01:01:82:00:01";
         int deviceId = 1;
@@ -1073,4 +1072,9 @@
         assertNotNull(result);
         assertEquals(1, result.size());
     }
+    private static NicVO createMockNic() {
+        NicVO nic = mock(NicVO.class);
+        Mockito.when(nic.getBroadcastUri()).thenReturn(URI.create("vlan://123"));
+        return nic;
+    }
 }
