--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -1005,8 +1005,7 @@
         doReturn(apiUserDataVO).when(userDataDao).findById(userDataId);
         when(apiUserDataVO.getUserData()).thenReturn(userData);
 
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        when(template.getUserDataId()).thenReturn(null);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate();
 
         String finalUserdata = userVmManagerImpl.finalizeUserData(null, userDataId, template);
 
@@ -1032,8 +1031,7 @@
     @Test
     public void testUserDataAllowOverrideWithUserdataText() {
         String userData = "testUserdata";
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        when(template.getUserDataId()).thenReturn(null);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate();
 
         String finalUserdata = userVmManagerImpl.finalizeUserData(userData, null, template);
 
@@ -4740,4 +4738,10 @@
         Assert.assertEquals(Long.valueOf(500L), volume.getMinIops());
         Assert.assertEquals(Long.valueOf(2000L), volume.getMaxIops());
     }
+
+    private static VirtualMachineTemplate createMockVirtualMachineTemplate() {
+        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
+        when(template.getUserDataId()).thenReturn(null);
+        return template;
+    }
 }
