--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -502,8 +502,7 @@
         when(devices.isEmpty()).thenReturn(true);
         when(vcsdao.listByPhysicalNetwork(anyLong())).thenReturn(devices);
 
-        final DisassociateMacFromNetworkAnswer answer = mock(DisassociateMacFromNetworkAnswer.class);
-        when(answer.getResult()).thenReturn(true);
+        final DisassociateMacFromNetworkAnswer answer = createMockDisassociateMacFromNetworkAnswer();
         when(agentmgr.easySend(eq(NETWORK_ID), (Command)any())).thenReturn(answer);
 
         guru.deallocate(network, nic, vmProfile);
@@ -588,4 +587,10 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+
+    private static DisassociateMacFromNetworkAnswer createMockDisassociateMacFromNetworkAnswer() {
+        final DisassociateMacFromNetworkAnswer answer = mock(DisassociateMacFromNetworkAnswer.class);
+        when(answer.getResult()).thenReturn(true);
+        return answer;
+    }
 }
