--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImplTest.java
@@ -582,8 +582,7 @@
 
     @Test
     public void testValidateNodeAffinityGroupsNoAffinityGroups() {
-        KubernetesCluster cluster = Mockito.mock(KubernetesCluster.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        KubernetesCluster cluster = createMockKubernetesCluster();
         List<Long> nodeIds = Arrays.asList(100L, 101L);
 
         Mockito.when(kubernetesClusterAffinityGroupMapDao.listAffinityGroupIdsByClusterIdAndNodeType(1L, WORKER.name()))
@@ -596,8 +595,7 @@
 
     @Test
     public void testValidateNodeAffinityGroupsNullAffinityGroups() {
-        KubernetesCluster cluster = Mockito.mock(KubernetesCluster.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        KubernetesCluster cluster = createMockKubernetesCluster();
         List<Long> nodeIds = Arrays.asList(100L, 101L);
 
         Mockito.when(kubernetesClusterAffinityGroupMapDao.listAffinityGroupIdsByClusterIdAndNodeType(1L, WORKER.name()))
@@ -610,8 +608,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateNodeAffinityGroupsAntiAffinityNewNodeOnExistingHost() {
-        KubernetesCluster cluster = Mockito.mock(KubernetesCluster.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        KubernetesCluster cluster = createMockKubernetesCluster();
         Mockito.when(cluster.getName()).thenReturn("test-cluster");
 
         Long newNodeId = 100L;
@@ -722,8 +719,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateNodeAffinityGroupsAffinityNewNodeOnDifferentHost() {
-        KubernetesCluster cluster = Mockito.mock(KubernetesCluster.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        KubernetesCluster cluster = createMockKubernetesCluster();
         Mockito.when(cluster.getName()).thenReturn("test-cluster");
 
         Long newNodeId = 100L;
@@ -766,8 +762,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testValidateNodeAffinityGroupsAntiAffinityMultipleNewNodesOnSameHost() {
-        KubernetesCluster cluster = Mockito.mock(KubernetesCluster.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        KubernetesCluster cluster = createMockKubernetesCluster();
         Mockito.when(cluster.getName()).thenReturn("test-cluster");
 
         Long newNodeId1 = 100L;
@@ -914,4 +909,10 @@
         Mockito.verify(kubernetesClusterAffinityGroupMapDao).listAffinityGroupIdsByClusterIdAndNodeType(1L, WORKER.name());
     }
 
+    private static KubernetesCluster createMockKubernetesCluster() {
+        KubernetesCluster cluster = Mockito.mock(KubernetesCluster.class);
+        Mockito.when(cluster.getId()).thenReturn(1L);
+        return cluster;
+    }
+
 }
