--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -811,10 +811,8 @@
     public void testZoneWideVolumeRequiresStorageMotionSameClusterHost() {
         PrimaryDataStore dataStore = Mockito.mock(PrimaryDataStore.class);
         Mockito.when(dataStore.isManaged()).thenReturn(true);
-        Host host1 = Mockito.mock(Host.class);
-        Mockito.when(host1.getClusterId()).thenReturn(1L);
-        Host host2 = Mockito.mock(Host.class);
-        Mockito.when(host2.getClusterId()).thenReturn(1L);
+        Host host1 = createMockHost(1L);
+        Host host2 = createMockHost(1L);
         Assert.assertFalse(spy.zoneWideVolumeRequiresStorageMotion(dataStore, host1, host2));
     }
 
@@ -824,10 +822,8 @@
         Mockito.when(dataStore.isManaged()).thenReturn(true);
         PrimaryDataStoreDriver driver = Mockito.mock(PrimaryDataStoreDriver.class);
         Mockito.when(dataStore.getDriver()).thenReturn(driver);
-        Host host1 = Mockito.mock(Host.class);
-        Mockito.when(host1.getClusterId()).thenReturn(1L);
-        Host host2 = Mockito.mock(Host.class);
-        Mockito.when(host2.getClusterId()).thenReturn(2L);
+        Host host1 = createMockHost(1L);
+        Host host2 = createMockHost(2L);
 
         Mockito.when(driver.zoneWideVolumesAvailableWithoutClusterMotion()).thenReturn(true);
         Assert.assertFalse(spy.zoneWideVolumeRequiresStorageMotion(dataStore, host1, host2));
@@ -2493,4 +2489,10 @@
         Assert.assertEquals(5L, (long) deploymentPlan.getPodId());
         Assert.assertEquals(4L, (long) deploymentPlan.getClusterId());
     }
+
+    private static Host createMockHost(Long clusterId) {
+        Host host2 = Mockito.mock(Host.class);
+        Mockito.when(host2.getClusterId()).thenReturn(clusterId);
+        return host2;
+    }
 }
