--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -100,11 +100,13 @@
     DataCenterDao dcDao;
     @Mock
     ASNumberDao asNumberDao;
+    private ASNumberVO asNumberMock;
     @Mock
     BgpPeerDetailsDao bgpPeerDetailsDao;
 
     @Before
     public void setUp() {
+        asNumberMock = Mockito.mock(ASNumberVO.class);
         ReflectionTestUtils.setField(commandSetupHelper, "_nicDao", nicDao);
         ReflectionTestUtils.setField(commandSetupHelper, "_networkDao", networkDao);
         ReflectionTestUtils.setField(commandSetupHelper, "_ipAddressDao", ipAddressDao);
@@ -222,8 +224,7 @@
         when(router.getDataCenterId()).thenReturn(zoneId);
         when(router.getVpcId()).thenReturn(null);
         when(network.getId()).thenReturn(networkId);
-        ASNumberVO asNumberVO = Mockito.mock(ASNumberVO.class);
-        when(asNumberDao.findByZoneAndNetworkId(zoneId, networkId)).thenReturn(asNumberVO);
+        when(asNumberDao.findByZoneAndNetworkId(zoneId, networkId)).thenReturn(asNumberMock);
         DataCenterVO dc = Mockito.mock(DataCenterVO.class);
         when(dcDao.findById(zoneId)).thenReturn(dc);
         when(dc.getNetworkType()).thenReturn(DataCenter.NetworkType.Advanced);
@@ -249,8 +250,7 @@
         long vpcId = 11L;
         when(router.getDataCenterId()).thenReturn(zoneId);
         when(router.getVpcId()).thenReturn(vpcId);
-        ASNumberVO asNumberVO = Mockito.mock(ASNumberVO.class);
-        when(asNumberDao.findByZoneAndVpcId(zoneId, vpcId)).thenReturn(asNumberVO);
+        when(asNumberDao.findByZoneAndVpcId(zoneId, vpcId)).thenReturn(asNumberMock);
 
         long networkOfferingId = 12L;
         NetworkOfferingVO offering = Mockito.mock(NetworkOfferingVO.class);
