--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestratorTest.java
@@ -170,7 +170,7 @@
         // make local mocks
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
+        NetworkVO network = createMockNetwork();
 
         // make sure that release dhcp will be called
         when(vm.getType()).thenReturn(Type.User);
@@ -195,7 +195,7 @@
         // make local mocks
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
+        NetworkVO network = createMockNetwork();
 
         // make sure that release dhcp won't be called
         when(vm.getType()).thenReturn(Type.DomainRouter);
@@ -215,7 +215,7 @@
         // make local mocks
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
+        NetworkVO network = createMockNetwork();
 
         // make sure that release dhcp will *not* be called
         when(vm.getType()).thenReturn(Type.User);
@@ -515,7 +515,7 @@
     public void testDontReleaseNicWhenPreserveNicsSettingEnabled() {
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         NicVO nic = mock(NicVO.class);
-        NetworkVO network = mock(NetworkVO.class);
+        NetworkVO network = createMockNetwork();
 
         when(vm.getType()).thenReturn(Type.User);
         when(network.getGuruName()).thenReturn(guruName);
@@ -1021,6 +1021,12 @@
         }
     }
 
+    private static NetworkVO createMockNetwork() {
+        NetworkVO network = Mockito.mock(NetworkVO.class);
+        when(network.getGuruName()).thenReturn("GuestNetworkGuru");
+        return network;
+    }
+
     // -----------------------------------------------------------------------
     // Tests for getNetworkElementsIncludingExtensions
     // -----------------------------------------------------------------------
