--- a/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
@@ -170,8 +170,7 @@
         when(vr.getHostId()).thenReturn(HOST_ID);
         doReturn(true).when(nwHelperUT).checkRouterVersion(vr);
 
-        Commands commands = mock(Commands.class);
-        when(commands.size()).thenReturn(3);
+        Commands commands = createMockCommands();
         Answer answer1 = mock(Answer.class);
         Answer answer2 = mock(Answer.class);
         Answer answer3 = mock(Answer.class);
@@ -208,8 +207,7 @@
         when(vr.getHostId()).thenReturn(HOST_ID);
         doReturn(true).when(nwHelperUT).checkRouterVersion(vr);
 
-        Commands commands = mock(Commands.class);
-        when(commands.size()).thenReturn(3);
+        Commands commands = createMockCommands();
         Answer answer1 = mock(Answer.class);
         Answer answer2 = mock(Answer.class);
         Answer answer3 = mock(Answer.class);
@@ -246,8 +244,7 @@
         when(vr.getHostId()).thenReturn(HOST_ID);
         doReturn(true).when(nwHelperUT).checkRouterVersion(vr);
 
-        Commands commands = mock(Commands.class);
-        when(commands.size()).thenReturn(3);
+        Commands commands = createMockCommands();
         Answer answer1 = mock(Answer.class);
         Answer answer2 = mock(Answer.class);
         // In the second iteration it should match and return, without invoking the third
@@ -313,6 +310,12 @@
     }
 
 
+    private static Commands createMockCommands() {
+        Commands commands = mock(Commands.class);
+        when(commands.size()).thenReturn(3);
+        return commands;
+    }
+
     private NicProfile getExpectedNicProfile(boolean vxlan, String vlanTag) {
         NicProfile nic = new NicProfile();
         nic.setDefaultNic(true);
