--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -252,6 +252,12 @@
         ApiResponseHelper.setResponseIpAddress(result, response);
 
         assertTrue(response.getIpAddr().equals("ipv4"));
+    }
+
+    private static Capacity createMockCapacity() {
+        Capacity c2 = Mockito.mock(Capacity.class);
+        Mockito.when(c2.getCapacityType()).thenReturn((short)Resource.ResourceType.user_vm.getOrdinal());
+        return c2;
     }
 
     private void setResult(NicSecondaryIp result, String ipv4, String ipv6) {
@@ -549,11 +555,9 @@
 
     @Test
     public void testCapacityListingForSingleNonGpuType() {
-        Capacity c1 = Mockito.mock(Capacity.class);
-        Mockito.when(c1.getCapacityType()).thenReturn((short)Resource.ResourceType.user_vm.getOrdinal());
-        Capacity c2 = Mockito.mock(Capacity.class);
-        Mockito.when(c2.getCapacityType()).thenReturn((short)Resource.ResourceType.user_vm.getOrdinal());
-        Capacity c3 = Mockito.mock(Capacity.class);
+        Capacity c1 = createMockCapacity();
+        Capacity c2 = createMockCapacity();
+        Capacity c3 = createMockCapacity();
         Mockito.when(c3.getCapacityType()).thenReturn((short)Resource.ResourceType.volume.getOrdinal());
         Capacity c4 = Mockito.mock(Capacity.class);
         Assert.assertTrue(apiResponseHelper.capacityListingForSingleNonGpuType(List.of(c1, c2)));
