--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenFabricUtilsTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenFabricUtilsTest.java
@@ -42,6 +42,8 @@
     AgentManager agentMgr;
     @Mock
     TungstenProviderDao tungstenProviderDao;
+    @Mock
+    private TungstenProviderVO tungstenProviderVOMock;
 
     TungstenFabricUtils tungstenFabricUtils;
 
@@ -62,11 +64,10 @@
 
     @Test
     public void sendTungstenCommandSuccessTest() {
-        TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(true);
         TungstenCommand tungstenCommand = mock(TungstenCommand.class);
 
-        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
+        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVOMock);
         when(agentMgr.easySend(anyLong(), any(TungstenCommand.class))).thenReturn(tungstenAnswer);
 
         assertEquals(tungstenAnswer, tungstenFabricUtils.sendTungstenCommand(tungstenCommand, anyLong()));
@@ -81,21 +82,19 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void sendTungstenCommandWithNullAnswer() {
-        TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenCommand tungstenCommand = mock(TungstenCommand.class);
 
-        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
+        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVOMock);
 
         tungstenFabricUtils.sendTungstenCommand(tungstenCommand, anyLong());
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void sendTungstenCommandWithFalseAnswer() {
-        TungstenProviderVO tungstenProviderVO = mock(TungstenProviderVO.class);
         TungstenAnswer tungstenAnswer = MockTungstenAnswerFactory.get(false);
         TungstenCommand tungstenCommand = mock(TungstenCommand.class);
 
-        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVO);
+        when(tungstenProviderDao.findByZoneId(anyLong())).thenReturn(tungstenProviderVOMock);
         when(agentMgr.easySend(anyLong(), any(TungstenCommand.class))).thenReturn(tungstenAnswer);
 
         tungstenFabricUtils.sendTungstenCommand(tungstenCommand, anyLong());
