--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -500,6 +500,13 @@
         updateUserPwdChange();
     }
 
+    private static UserAccount createMockUserAccount() {
+        UserAccount user = Mockito.mock(UserAccount.class);
+        Mockito.when(user.getId()).thenReturn(1L);
+        Mockito.when(user.isUser2faEnabled()).thenReturn(true);
+        return user;
+    }
+
     private void updateUserPwdChange() {
         Mockito.doReturn(true).when(UpdateUserCmdMock).isPasswordChangeRequired();
         Mockito.when(userVoMock.getAccountId()).thenReturn(10L);
@@ -1889,9 +1896,7 @@
 
     @Test
     public void testClearUser2FA_When2FAInSetupState_Disable2FA() {
-        UserAccount user = Mockito.mock(UserAccount.class);
-        Mockito.when(user.getId()).thenReturn(1L);
-        Mockito.when(user.isUser2faEnabled()).thenReturn(true);
+        UserAccount user = createMockUserAccount();
         UserDetailVO userDetail = new UserDetailVO();
         userDetail.setValue(UserAccountVO.Setup2FAstatus.ENABLED.name());
         UserAccountVO userAccountVO = new UserAccountVO();
