--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -344,7 +344,7 @@
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
-        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
+        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = createMockLbStickinessPolicy();
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
@@ -358,14 +358,12 @@
         TungstenAnswer updateTungstenLoadBalancerListenerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateTungstenHealthMonitorAnswer = MockTungstenAnswerFactory.get(true);
         LoadBalancingRule.LbSslCert lbSslCert = mock(LoadBalancingRule.LbSslCert.class);
-        when(lbStickinessPolicy.getMethodName()).thenReturn("AppCookie");
         List<Pair<String, String>> pairList = List.of(new Pair<>("cookieName", "cookieValue"));
 
         ApiKeyPairVO latest = new ApiKeyPairVO();
         latest.setApiKey("apikey");
         latest.setSecretKey("secretkey");
         when(ApiDBUtils.searchForLatestUserKeyPair(Mockito.anyLong())).thenReturn(latest);
-        when(lbStickinessPolicy.getParams()).thenReturn(pairList);
         when(loadBalancingRule1.getId()).thenReturn(1L);
         when(loadBalancingRule1.getState()).thenReturn(FirewallRule.State.Add);
         when(loadBalancingRule1.getAlgorithm()).thenReturn("roundrobin");
@@ -400,7 +398,7 @@
         LoadBalancingRule loadBalancingRule1 = mock(LoadBalancingRule.class);
         LoadBalancerVMMapVO loadBalancerVMMapVO = mock(LoadBalancerVMMapVO.class);
         LoadBalancerVO loadBalancerVO = mock(LoadBalancerVO.class);
-        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = mock(LoadBalancingRule.LbStickinessPolicy.class);
+        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = createMockLbStickinessPolicy();
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVMMapVO> loadBalancerVMMapVOList = List.of(loadBalancerVMMapVO);
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
@@ -414,8 +412,7 @@
         List<Pair<String, String>> pairList = List.of(new Pair<>("cookieName", "cookieValue"));
         TungstenFabricLBHealthMonitorVO tungstenFabricLBHealthMonitorVO = mock(TungstenFabricLBHealthMonitorVO.class);
 
-        when(lbStickinessPolicy.getMethodName()).thenReturn("AppCookie");
-        when(lbStickinessPolicy.getParams()).thenReturn(pairList);
+
         when(loadBalancingRule1.getId()).thenReturn(1L);
         when(loadBalancingRule1.getState()).thenReturn(FirewallRule.State.Add);
         when(loadBalancingRule1.getAlgorithm()).thenReturn("roundrobin");
@@ -962,4 +959,11 @@
         tungstenElement.commitMigration(nic, network, vm, src, dest);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class), anyLong());
     }
+
+    private static LoadBalancingRule.LbStickinessPolicy createMockLbStickinessPolicy() {
+        LoadBalancingRule.LbStickinessPolicy lbStickinessPolicy = Mockito.mock(LoadBalancingRule.LbStickinessPolicy.class);
+        when(lbStickinessPolicy.getMethodName()).thenReturn("AppCookie");
+        when(lbStickinessPolicy.getParams()).thenReturn(List.of(new Pair<>("cookieName", "cookieValue")));
+        return lbStickinessPolicy;
+    }
 }
