--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -380,8 +380,7 @@
 
     @Test
     public void testVolumeOnSharedStoragePoolTrue() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
+        VolumeVO volume = createMockVolume();
 
         StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(pool.getScope()).thenReturn(ScopeType.CLUSTER); // Shared scope
@@ -392,8 +391,7 @@
 
     @Test
     public void testVolumeOnSharedStoragePoolFalseHostScope() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
+        VolumeVO volume = createMockVolume();
 
         StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
         Mockito.when(pool.getScope()).thenReturn(ScopeType.HOST); // Local scope
@@ -413,8 +411,7 @@
 
     @Test
     public void testVolumeOnSharedStoragePoolFalsePoolNotFound() {
-        VolumeVO volume = Mockito.mock(VolumeVO.class);
-        Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
+        VolumeVO volume = createMockVolume();
 
         Mockito.when(storagePoolDao.findById(MOCK_POOL_ID)).thenReturn(null); // Pool not found in DB
 
@@ -922,6 +919,12 @@
         } catch (InvocationTargetException e) {
             throw e.getCause();
         }
+    }
+
+    private static VolumeVO createMockVolume() {
+        VolumeVO volume = Mockito.mock(VolumeVO.class);
+        Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
+        return volume;
     }
 
     private void setField(Object target, String fieldName, Object value) throws Exception {
