--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -129,6 +129,7 @@
     @Mock
     NetworkOrchestrationService networkOrchestrationService;
     FirewallManager firewallManager = Mockito.mock(FirewallManager.class);
+    private Vpc vpcMock;
 
     @InjectMocks
     private Ipv6ServiceImpl ipv6Service = new Ipv6ServiceImpl();
@@ -164,6 +165,7 @@
         updatedPrefixSubnetMap = new ArrayList<>();
         persistedPrefixSubnetMap = new ArrayList<>();
         closeable = MockitoAnnotations.openMocks(this);
+        vpcMock = Mockito.mock(Vpc.class);
         ipv6Service.firewallManager = firewallManager;
         Mockito.when(ipv6GuestPrefixSubnetNetworkMapDao.update(Mockito.anyLong(), Mockito.any(Ipv6GuestPrefixSubnetNetworkMapVO.class))).thenAnswer((Answer<Boolean>) invocation -> {
             Ipv6GuestPrefixSubnetNetworkMapVO map = (Ipv6GuestPrefixSubnetNetworkMapVO)invocation.getArguments()[1];
@@ -574,17 +576,15 @@
     @Test
     public void testEmptyUpdateIpv6RoutesForVpcResponse() {
         VpcResponse response = new VpcResponse();
-        Vpc vpc = Mockito.mock(Vpc.class);
         List<NetworkVO> networks = new ArrayList<>();
         Mockito.doReturn(networks).when(networkModel).listNetworksByVpc(Mockito.anyLong());
-        ipv6Service.updateIpv6RoutesForVpcResponse(vpc, response);
+        ipv6Service.updateIpv6RoutesForVpcResponse(vpcMock, response);
         Assert.assertTrue(CollectionUtils.isEmpty(response.getIpv6Routes()));
     }
 
     @Test
     public void testUpdateIpv6RoutesForVpcResponse() {
         VpcResponse response = new VpcResponse();
-        Vpc vpc = Mockito.mock(Vpc.class);
         List<NetworkVO> networks = new ArrayList<>();
         NetworkVO network = Mockito.mock(NetworkVO.class);
         Mockito.when(network.getIp6Cidr()).thenReturn(cidr);
@@ -597,7 +597,7 @@
         Mockito.when(nicDao.listByVmId(Mockito.anyLong())).thenReturn(List.of(nic));
         Mockito.doReturn(networks).when(networkModel).listNetworksByVpc(Mockito.anyLong());
         Mockito.when(networkOfferingDao.isIpv6Supported(Mockito.anyLong())).thenReturn(true);
-        ipv6Service.updateIpv6RoutesForVpcResponse(vpc, response);
+        ipv6Service.updateIpv6RoutesForVpcResponse(vpcMock, response);
         Assert.assertEquals(1, response.getIpv6Routes().size());
         Ipv6RouteResponse routeResponse = new ArrayList<>(response.getIpv6Routes()).get(0);
         Assert.assertEquals(ipv6Address, routeResponse.getGateway());
