--- a/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
+++ b/plugins/network-elements/bigswitch/src/test/java/com/cloud/network/resource/BigSwitchBcfResourceTest.java
@@ -223,7 +223,7 @@
     public void testCreateAttachmentRetryOnce() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        AttachmentData attachmentData = mock(AttachmentData.class);
+        AttachmentData attachmentData = createMockAttachmentData();
         AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
         when(attachmentData.getAttachment()).thenReturn(attachment);
         when(attachment.getId()).thenReturn("eeee");
@@ -240,7 +240,7 @@
     public void testCreateAttachmentApiException() throws ConfigurationException, BigSwitchBcfApiException {
         _resource.configure("BigSwitchBcfResource", _parameters);
 
-        AttachmentData attachmentData = mock(AttachmentData.class);
+        AttachmentData attachmentData = createMockAttachmentData();
         AttachmentData.Attachment attachment = mock(AttachmentData.Attachment.class);
         when(attachmentData.getAttachment()).thenReturn(attachment);
         when(attachment.getId()).thenReturn("eeee");
@@ -428,4 +428,8 @@
         assertTrue(ans.getResult());
     }
 
+    private static AttachmentData createMockAttachmentData() {
+        return mock(AttachmentData.class);
+    }
+
 }
