--- a/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/NetworkServiceImplTest.java
@@ -348,6 +348,12 @@
         f.set(configKey, o);
     }
 
+    private static NetworkOffering createMockNetworkOffering(Network.GuestType guestType) {
+        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
+        when(ntwkOff.getGuestType()).thenReturn(guestType);
+        return ntwkOff;
+    }
+
     @Test
     public void testGetPrivateVlanPairNoVlans() {
         Pair<String, Network.PVlanType> pair = service.getPrivateVlanPair(null, null, null);
@@ -688,8 +694,7 @@
     @Test(expected = InvalidParameterValueException.class)
     public void testCheckAndUpdateNetworkDnsL2NetworkFailure() {
         NetworkVO networkVO = Mockito.mock(NetworkVO.class);
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getGuestType()).thenReturn(Network.GuestType.L2);
+        NetworkOffering offering = createMockNetworkOffering(Network.GuestType.L2);
         service.checkAndUpdateNetworkDns(networkVO, offering, ip4Dns[0], null, null, null);
     }
 
@@ -697,8 +702,7 @@
     public void testCheckAndUpdateNetworkDnsVpcTierFailure() {
         NetworkVO networkVO = Mockito.mock(NetworkVO.class);
         Mockito.when(networkVO.getVpcId()).thenReturn(1L);
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-        Mockito.when(offering.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering offering = createMockNetworkOffering(Network.GuestType.Shared);
         service.checkAndUpdateNetworkDns(networkVO, offering, ip4Dns[0], null, null, null);
     }
 
@@ -1088,8 +1092,7 @@
 
     @Test
     public void validateNotSharedNetworkRouterIPv4() {
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.L2);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.L2);
         service.validateSharedNetworkRouterIPs(null, null, null, null, null, null, null, null, null, ntwkOff);
     }
 
@@ -1101,8 +1104,7 @@
         String routerPv6 = "fd17:ac56:1234:2000::fb";
         String startIPv6 = "fd17:ac56:1234:2000::1";
         String endIPv6 = "fd17:ac56:1234:2000::fc";
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.Shared);
         service.validateSharedNetworkRouterIPs(IP4_GATEWAY, startIP, endIP, IP4_NETMASK, routerIPv4, routerPv6, startIPv6, endIPv6, IP6_CIDR, ntwkOff);
     }
 
@@ -1114,8 +1116,7 @@
         String routerPv6 = "fd17:ac56:1234:2000::fb";
         String startIPv6 = "fd17:ac56:1234:2000::1";
         String endIPv6 = "fd17:ac56:1234:2000::fc";
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.Shared);
         boolean passing = false;
         try {
             service.validateSharedNetworkRouterIPs(IP4_GATEWAY, startIP, endIP, IP4_NETMASK, routerIPv4, routerPv6, startIPv6, endIPv6, IP6_CIDR, ntwkOff);
@@ -1134,8 +1135,7 @@
         String routerPv6 = "fd17:ac56:1234:2000::1";
         String startIPv6 = "fd17:ac56:1234:2000::1";
         String endIPv6 = null;
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.Shared);
         service.validateSharedNetworkRouterIPs(IP4_GATEWAY, startIP, endIP, IP4_NETMASK, routerIPv4, routerPv6, startIPv6, endIPv6, IP6_CIDR, ntwkOff);
     }
 
@@ -1143,8 +1143,7 @@
     public void validateSharedNetworkIPv6RouterNotInRange() {
         String routerIPv4 = null;
         String routerIPv6 = "fd17:ac56:1234:2001::1";
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.Shared);
         boolean passing = true;
         try {
             service.validateSharedNetworkRouterIPs(IP4_GATEWAY, null, null, IP4_NETMASK, routerIPv4, routerIPv6, null, null, IP6_CIDR, ntwkOff);
@@ -1158,8 +1157,7 @@
     @Test
     public void invalidateSharedNetworkIPv6RouterAddress() {
         String routerIPv6 = "fd17:ac56:1234:2000::fg";
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.Shared);
         boolean passing = false;
         try {
             service.validateSharedNetworkRouterIPs(IP4_GATEWAY, null, null, IP4_NETMASK, null, routerIPv6, null, null, IP6_CIDR, ntwkOff);
@@ -1173,8 +1171,7 @@
     @Test
     public void invalidateSharedNetworkIPv4RouterAddress() {
         String routerIPv4 = "10.100.1000.1";
-        NetworkOffering ntwkOff = Mockito.mock(NetworkOffering.class);
-        when(ntwkOff.getGuestType()).thenReturn(Network.GuestType.Shared);
+        NetworkOffering ntwkOff = createMockNetworkOffering(Network.GuestType.Shared);
         boolean passing = false;
         try {
             service.validateSharedNetworkRouterIPs(IP4_GATEWAY, null, null, IP4_NETMASK, routerIPv4, null, null, null, IP6_CIDR, ntwkOff);
