--- a/server/src/test/java/com/cloud/acl/DomainCheckerTest.java
+++ b/server/src/test/java/com/cloud/acl/DomainCheckerTest.java
@@ -65,6 +65,12 @@
         return entity;
     }
 
+    private static AccountVO createMockAccountVO() {
+        AccountVO accountVO = Mockito.mock(AccountVO.class);
+        Mockito.when(accountVO.getDomainId()).thenReturn(101L);
+        return accountVO;
+    }
+
     @Test
     public void testRootAdminHasAccess() {
         Account rootAdmin = Mockito.mock(Account.class);
@@ -101,8 +107,7 @@
         Mockito.when(caller.getDomainId()).thenReturn(100L);
         Mockito.when(caller.getType()).thenReturn(Account.Type.DOMAIN_ADMIN);
         ControlledEntity entity = getMockedEntity(2L);
-        AccountVO owner = Mockito.mock(AccountVO.class);
-        Mockito.when(owner.getDomainId()).thenReturn(101L);
+        AccountVO owner = createMockAccountVO();
         Mockito.when(_accountDao.findByIdIncludingRemoved(entity.getAccountId())).thenReturn(owner);
         Mockito.when(_domainDao.isChildDomain(100L, 101L)).thenReturn(true);
 
@@ -116,8 +121,7 @@
         Mockito.when(caller.getDomainId()).thenReturn(100L);
         Mockito.when(caller.getType()).thenReturn(Account.Type.DOMAIN_ADMIN);
         ControlledEntity entity = getMockedEntity(2L);
-        AccountVO removedOwner = Mockito.mock(AccountVO.class);
-        Mockito.when(removedOwner.getDomainId()).thenReturn(101L);
+        AccountVO removedOwner = createMockAccountVO();
         Mockito.when(_accountDao.findByIdIncludingRemoved(entity.getAccountId())).thenReturn(removedOwner);
         Mockito.when(_domainDao.isChildDomain(100L, 101L)).thenReturn(true);
 
