--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -202,6 +202,8 @@
     @Mock
     private VMInstanceVO vmInstanceVOMock;
 
+    private VMInstanceVO vmMock;
+
     @Mock
     private CreateBackupScheduleCmd createBackupScheduleCmdMock;
 
@@ -291,6 +293,7 @@
     @Before
     public void setup() throws Exception {
         gson = new Gson();
+        vmMock = mock(VMInstanceVO.class);
 
         closeable = MockitoAnnotations.openMocks(this);
         when(backupOfferingDao.findById(null)).thenReturn(null);
@@ -1323,8 +1326,7 @@
 
         overrideBackupFrameworkConfigValue();
 
-        VMInstanceVO vm = mock(VMInstanceVO.class);
-        when(vmInstanceDao.findById(vmId)).thenReturn(vm);
+        when(vmInstanceDao.findById(vmId)).thenReturn(vmMock);
         BackupScheduleVO schedule = mock(BackupScheduleVO.class);
         when(schedule.getId()).thenReturn(scheduleId);
         when(backupScheduleDao.listByVM(vmId)).thenReturn(List.of(schedule));
@@ -1629,8 +1631,7 @@
         when(backup.getZoneId()).thenReturn(1L);
         when(backupDao.findByIdIncludingRemoved(backupId)).thenReturn(backup);
 
-        VMInstanceVO vm = mock(VMInstanceVO.class);
-        when(vmInstanceDao.findByIdIncludingRemoved(vmId)).thenReturn(vm);
+        when(vmInstanceDao.findByIdIncludingRemoved(vmId)).thenReturn(vmMock);
 
         overrideBackupFrameworkConfigValue();
 
