--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -256,10 +256,8 @@
         NetworkOfferingVO offering = Mockito.mock(NetworkOfferingVO.class);
         when(networkOfferingDao.findByIdIncludingRemoved(networkOfferingId)).thenReturn(offering);
         when(offering.getRoutingMode()).thenReturn(NetworkOffering.RoutingMode.Dynamic);
-        NetworkVO network1 = Mockito.mock(NetworkVO.class);
-        when(network1.getNetworkOfferingId()).thenReturn(networkOfferingId);
-        NetworkVO network2 = Mockito.mock(NetworkVO.class);
-        when(network2.getNetworkOfferingId()).thenReturn(networkOfferingId);
+        NetworkVO network1 = createMockNetwork(networkOfferingId);
+        NetworkVO network2 = createMockNetwork(networkOfferingId);
         when(networkDao.listByVpc(vpcId)).thenReturn(Arrays.asList(network1, network2));
 
         DataCenterVO dc = Mockito.mock(DataCenterVO.class);
@@ -311,4 +309,10 @@
         ConfigKey<Integer> configKey = NetworkOrchestrationService.DhcpLeaseTimeout;
         Assert.assertTrue("ConfigKey should be dynamic", configKey.isDynamic());
     }
+
+    private static NetworkVO createMockNetwork(long networkOfferingId) {
+        NetworkVO network = Mockito.mock(NetworkVO.class);
+        when(network.getNetworkOfferingId()).thenReturn(networkOfferingId);
+        return network;
+    }
 }
