--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -134,8 +134,7 @@
 
     private final long accountMockId = 100L;
 
-    @Mock
-    private Account accountMock;
+        private Account accountMock = createMockAccount();
 
     @Mock
     private DomainVO domainVoMock;
@@ -500,6 +499,12 @@
         updateUserPwdChange();
     }
 
+    private static AccountVO createMockAccount() {
+        AccountVO accountMock = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(22L).when(accountMock).getDomainId();
+        return accountMock;
+    }
+
     private void updateUserPwdChange() {
         Mockito.doReturn(true).when(UpdateUserCmdMock).isPasswordChangeRequired();
         Mockito.when(userVoMock.getAccountId()).thenReturn(10L);
@@ -785,7 +790,7 @@
         String userName = "username";
         Mockito.doReturn(userName).when(UpdateUserCmdMock).getUsername();
         Mockito.lenient().doReturn(userName).when(userVoMock).getUsername();
-        Mockito.doReturn(domanIdCurrentUser).when(accountMock).getDomainId();
+        accountMock = createMockAccount();
 
         long userVoDuplicatedMockId = 67L;
         UserVO userVoDuplicatedMock = Mockito.mock(UserVO.class);
@@ -795,9 +800,8 @@
 
         Mockito.doReturn(accountIdUserDuplicated).when(userVoDuplicatedMock).getAccountId();
 
-        Account accountUserDuplicatedMock = Mockito.mock(AccountVO.class);
+        Account accountUserDuplicatedMock = createMockAccount();
         Mockito.lenient().doReturn(accountIdUserDuplicated).when(accountUserDuplicatedMock).getId();
-        Mockito.doReturn(domanIdCurrentUser).when(accountUserDuplicatedMock).getDomainId();
 
         List<UserVO> usersWithSameUserName = new ArrayList<>();
         usersWithSameUserName.add(userVoMock);
@@ -820,7 +824,7 @@
         String userName = "username";
         Mockito.doReturn(userName).when(UpdateUserCmdMock).getUsername();
         Mockito.lenient().doReturn(userName).when(userVoMock).getUsername();
-        Mockito.doReturn(domanIdCurrentUser).when(accountMock).getDomainId();
+        accountMock = createMockAccount();
 
         long userVoDuplicatedMockId = 67L;
         UserVO userVoDuplicatedMock = Mockito.mock(UserVO.class);
@@ -830,7 +834,7 @@
         long accountIdUserDuplicated = 98L;
         Mockito.doReturn(accountIdUserDuplicated).when(userVoDuplicatedMock).getAccountId();
 
-        Account accountUserDuplicatedMock = Mockito.mock(AccountVO.class);
+        Account accountUserDuplicatedMock = createMockAccount();
         Mockito.lenient().doReturn(accountIdUserDuplicated).when(accountUserDuplicatedMock).getId();
         Mockito.doReturn(45L).when(accountUserDuplicatedMock).getDomainId();
 
