--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLManagerTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLManagerTest.java
@@ -189,12 +189,11 @@
         // Create 4 rules to test all 4 scenarios: only revoke should
         // be deleted, only add should update
         final List<NetworkACLItemVO> rules = new ArrayList<>();
-        final NetworkACLItemVO ruleActive = Mockito.mock(NetworkACLItemVO.class);
-        final NetworkACLItemVO ruleStaged = Mockito.mock(NetworkACLItemVO.class);
+        final NetworkACLItemVO ruleActive = createMockNetworkACLItemVO();
+        final NetworkACLItemVO ruleStaged = createMockNetworkACLItemVO();
         final NetworkACLItemVO rule2Revoke = Mockito.mock(NetworkACLItemVO.class);
         final NetworkACLItemVO rule2Add = Mockito.mock(NetworkACLItemVO.class);
         Mockito.when(ruleActive.getState()).thenReturn(NetworkACLItem.State.Active);
-        Mockito.when(ruleStaged.getState()).thenReturn(NetworkACLItem.State.Staged);
         Mockito.when(rule2Add.getState()).thenReturn(NetworkACLItem.State.Add);
         Mockito.when(rule2Revoke.getState()).thenReturn(NetworkACLItem.State.Revoke);
         rules.add(ruleActive);
@@ -257,6 +256,12 @@
         assertTrue("Operation should be successful!", result);
     }
 
+    private static NetworkACLItemVO createMockNetworkACLItemVO() {
+        final NetworkACLItemVO ruleStaged = Mockito.mock(NetworkACLItemVO.class);
+        Mockito.when(ruleStaged.getState()).thenReturn(NetworkACLItem.State.Staged);
+        return ruleStaged;
+    }
+
     @Configuration
     @ComponentScan(basePackageClasses = {NetworkACLManagerImpl.class}, includeFilters = {
             @ComponentScan.Filter(value = NetworkACLTestConfiguration.Library.class, type = FilterType.CUSTOM)}, useDefaultFilters = false)
