--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -268,12 +268,15 @@
     @Mock
     Host hostMock;
 
+    private UserDataVO userDataMock;
+
     private AutoCloseable closeable;
     private MockedStatic<ApiDBUtils> apiDBUtilsMock;
 
     @Before
     public void setup() throws IllegalAccessException, NoSuchFieldException {
         closeable = MockitoAnnotations.openMocks(this);
+        userDataMock = Mockito.mock(UserDataVO.class);
         CallContext.register(Mockito.mock(User.class), Mockito.mock(Account.class));
         spy._accountMgr = accountManager;
         spy.userDataDao = userDataDao;
@@ -494,9 +497,8 @@
             when(cmd.getUserData()).thenReturn("testUserdata");
             when(cmd.getName()).thenReturn("testName");
 
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
             when(userDataDao.findByName(account.getAccountId(), account.getDomainId(), "testName")).thenReturn(null);
-            when(userDataDao.findByUserData(account.getAccountId(), account.getDomainId(), "testUserdata")).thenReturn(userData);
+            when(userDataDao.findByUserData(account.getAccountId(), account.getDomainId(), "testUserdata")).thenReturn(userDataMock);
 
             spy.registerUserData(cmd);
         }
@@ -515,8 +517,7 @@
             RegisterUserDataCmd cmd = Mockito.mock(RegisterUserDataCmd.class);
             when(cmd.getName()).thenReturn("testName");
 
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
-            when(userDataDao.findByName(account.getAccountId(), account.getDomainId(), "testName")).thenReturn(userData);
+            when(userDataDao.findByName(account.getAccountId(), account.getDomainId(), "testName")).thenReturn(userDataMock);
 
             spy.registerUserData(cmd);
         }
@@ -616,17 +617,16 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getId()).thenReturn(1L);
             when(cmd.isRecursive()).thenReturn(false);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
-            when(sb.entity()).thenReturn(userData);
+            when(sb.entity()).thenReturn(userDataMock);
 
             SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
             when(sb.create()).thenReturn(sc);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
-            userDataList.add(userData);
+            userDataList.add(userDataMock);
             Pair<List<UserDataVO>, Integer> result = new Pair(userDataList, 1);
             when(userDataDao.searchAndCount(nullable(SearchCriteria.class), nullable(Filter.class))).thenReturn(result);
 
@@ -649,17 +649,16 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getName()).thenReturn("testSearchUserdataName");
             when(cmd.isRecursive()).thenReturn(false);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
-            when(sb.entity()).thenReturn(userData);
+            when(sb.entity()).thenReturn(userDataMock);
 
             SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
             when(sb.create()).thenReturn(sc);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
-            userDataList.add(userData);
+            userDataList.add(userDataMock);
             Pair<List<UserDataVO>, Integer> result = new Pair(userDataList, 1);
             when(userDataDao.searchAndCount(nullable(SearchCriteria.class), nullable(Filter.class))).thenReturn(result);
 
@@ -682,17 +681,16 @@
             when(cmd.getProjectId()).thenReturn(2L);
             when(cmd.getKeyword()).thenReturn("testSearchUserdataKeyword");
             when(cmd.isRecursive()).thenReturn(false);
-            UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
-            when(sb.entity()).thenReturn(userData);
+            when(sb.entity()).thenReturn(userDataMock);
 
             SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
             when(sb.create()).thenReturn(sc);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
-            userDataList.add(userData);
+            userDataList.add(userDataMock);
             Pair<List<UserDataVO>, Integer> result = new Pair(userDataList, 1);
             when(userDataDao.searchAndCount(nullable(SearchCriteria.class), nullable(Filter.class))).thenReturn(result);
 
