--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -682,6 +682,12 @@
                 hostStatus, "1.0", null, null, 1L, null, 0, 0, null, 0, null);
     }
 
+    private static VMInstanceVO createMockVmInstance1() {
+        VMInstanceVO vmInstance1 = Mockito.mock(VMInstanceVO.class);
+        Mockito.when(vmInstance1.getHostId()).thenReturn(2L);
+        return vmInstance1;
+    }
+
     @Test
     public void testDestroyLocalStoragePoolVolumesBothRootDisksAndDataDisks() {
         resourceManager.destroyLocalStoragePoolVolumes(poolId);
@@ -1046,9 +1052,8 @@
         List<VolumeVO> volumesInPool = Arrays.asList(volume1, volume2);
         Mockito.doReturn(volumesInPool).when(volumeDao).findNonDestroyedVolumesByPoolId(poolId);
 
-        VMInstanceVO vmInstance1 = Mockito.mock(VMInstanceVO.class);
-        VMInstanceVO vmInstance2 = Mockito.mock(VMInstanceVO.class);
-        Mockito.when(vmInstance1.getHostId()).thenReturn(2L);
+        VMInstanceVO vmInstance1 = createMockVmInstance1();
+        VMInstanceVO vmInstance2 = createMockVmInstance1();
         Mockito.when(vmInstance2.getHostId()).thenReturn(3L);
 
         Mockito.doReturn(vmInstance1).when(vmInstanceDao).findById(100L);
