--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxGuestNetworkGuruTest.java
@@ -147,7 +147,7 @@
 
         when(physicalNetwork.getIsolationMethods()).thenReturn(List.of("NSX"));
 
-        when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        offering = MockNetworkOffering.createMockNetworkOffering(Networks.TrafficType.Guest);
         when(offering.getGuestType()).thenReturn(Network.GuestType.Isolated);
         when(offering.getNetworkMode()).thenReturn(NetworkOffering.NetworkMode.NATTED);
         when(offering.getId()).thenReturn(1L);
--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/NsxPublicNetworkGuruTest.java
@@ -97,8 +97,7 @@
         ReflectionTestUtils.setField(guru, "nsxControllerUtils", nsxControllerUtils);
         ReflectionTestUtils.setField(guru, "networkModel", networkModel);
 
-        offering = Mockito.mock(NetworkOffering.class);
-        when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Public);
+        offering = MockNetworkOffering.createMockNetworkOffering(Networks.TrafficType.Public);
         when(offering.isSystemOnly()).thenReturn(true);
         when(networkModel.isProviderForNetworkOffering(eq(Network.Provider.Nsx), anyLong())).thenReturn(true);
     }
@@ -110,9 +109,7 @@
 
     @Test
     public void testCannotHandle() {
-        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
-
-        when(offering.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
+        NetworkOffering offering = MockNetworkOffering.createMockNetworkOffering(Networks.TrafficType.Guest);
 
         Assert.assertFalse(guru.canHandle(offering));
     }
--- a/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/MockNetworkOffering.java
+++ b/plugins/network-elements/nsx/src/test/java/org/apache/cloudstack/service/MockNetworkOffering.java
@@ -0,0 +1,16 @@
+package org.apache.cloudstack.service;
+
+import com.cloud.network.Networks;
+import com.cloud.offering.NetworkOffering;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockNetworkOffering {
+
+    public static NetworkOffering createMockNetworkOffering(Networks.TrafficType trafficType) {
+        NetworkOffering offering = Mockito.mock(NetworkOffering.class);
+        when(offering.getTrafficType()).thenReturn(trafficType);
+        return offering;
+    }
+}
