--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -2902,10 +2902,8 @@
 
             Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
 
-            Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
-
             Mockito.doReturn(callerUser).when(userDao).findById(Mockito.anyLong());
-            Mockito.doReturn(networkMock).when(implementedNetwork).first();
+            Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = createMockImplementedNetwork(networkMock);
             Mockito.doReturn(null).when(implementedNetwork).second();
             Mockito.doReturn(implementedNetwork).when(_networkMgr).implementNetwork(Mockito.anyLong(), Mockito.any(), Mockito.any());
 
@@ -2925,10 +2923,9 @@
 
             Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
 
-            Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
+            Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = createMockImplementedNetwork(networkMock);
 
             Mockito.doReturn(callerUser).when(userDao).findById(Mockito.anyLong());
-            Mockito.doReturn(networkMock).when(implementedNetwork).first();
             Mockito.doReturn(networkMock).when(implementedNetwork).second();
             Mockito.doReturn(implementedNetwork).when(_networkMgr).implementNetwork(Mockito.anyLong(), Mockito.any(), Mockito.any());
 
@@ -4740,4 +4737,10 @@
         Assert.assertEquals(Long.valueOf(500L), volume.getMinIops());
         Assert.assertEquals(Long.valueOf(2000L), volume.getMaxIops());
     }
+
+    private static Pair<? extends NetworkGuru, ? extends Network> createMockImplementedNetwork(NetworkVO networkMock) {
+        Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
+        Mockito.doReturn(networkMock).when(implementedNetwork).first();
+        return implementedNetwork;
+    }
 }
