--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static Host createMockHost() {
+        Host vmHost = mock(Host.class);
+        when(vmHost.getDataCenterId()).thenReturn(5L);
+        return vmHost;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -376,7 +382,7 @@
 
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllowsWhenConfigFalse() {
-        Host vmHost = mock(Host.class);
+        Host vmHost = createMockHost();
         ConfigKey testConfig = mock(ConfigKey.class);
 
         Long dataCenterId = 5L;
@@ -386,7 +392,6 @@
         ipAddresses.put(NetworkElementCommand.ROUTER_IP, routerIp);
 
         when(testConfig.valueIn(dataCenterId)).thenReturn(false);
-        when(vmHost.getDataCenterId()).thenReturn(dataCenterId);
 
         virtualMachineManagerImpl.addHostIpToCertDetailsIfConfigAllows(vmHost, ipAddresses, testConfig);
         assertFalse(ipAddresses.containsKey(NetworkElementCommand.HYPERVISOR_HOST_PRIVATE_IP));
