--- a/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
@@ -94,7 +94,6 @@
 
         final Vpc vpc = Mockito.mock(Vpc.class);
         final DataCenterVO dataCenterVO = Mockito.mock(DataCenterVO.class);
-        final RemoteAccessVpn remoteAccessVpn = Mockito.mock(RemoteAccessVpn.class);
         final DomainRouterVO domainRouterVO1 = Mockito.mock(DomainRouterVO.class);
         final DomainRouterVO domainRouterVO2 = Mockito.mock(DomainRouterVO.class);
         final VpnUser vpnUser1 = Mockito.mock(VpnUser.class);
@@ -111,7 +110,7 @@
         final Long vpcId = new Long(1l);
         final Long zoneId = new Long(1l);
 
-        when(remoteAccessVpn.getVpcId()).thenReturn(vpcId);
+        final RemoteAccessVpn remoteAccessVpn = createMockRemoteAccessVpn(vpcId);
         when(_vpcRouterMgr.getVpcRouters(vpcId)).thenReturn(routers);
         when(_entityMgr.findById(Vpc.class, vpcId)).thenReturn(vpc);
         when(vpc.getZoneId()).thenReturn(zoneId);
@@ -180,13 +179,11 @@
         networkTopologyContext.setBasicNetworkTopology(basicNetworkTopology);
         networkTopologyContext.init();
 
-        final RemoteAccessVpn remoteAccessVpn = Mockito.mock(RemoteAccessVpn.class);
-
         final List<VpnUser> users = new ArrayList<VpnUser>();
 
         final Long vpcId = new Long(1l);
 
-        when(remoteAccessVpn.getVpcId()).thenReturn(vpcId);
+        final RemoteAccessVpn remoteAccessVpn = createMockRemoteAccessVpn(vpcId);
         when(_vpcRouterMgr.getVpcRouters(vpcId)).thenReturn(null);
 
         try {
@@ -214,4 +211,10 @@
         verify(_networkMdl).isProviderSupportServiceInNetwork(200L, Network.Service.Firewall, Network.Provider.VPCVirtualRouter);
         verify(_vpcMgr, never()).isProviderSupportServiceInVpc(100L, Network.Service.Firewall, Network.Provider.VPCVirtualRouter);
     }
+
+    private static RemoteAccessVpn createMockRemoteAccessVpn(final Long vpcId) {
+        final RemoteAccessVpn remoteAccessVpn = Mockito.mock(RemoteAccessVpn.class);
+        when(remoteAccessVpn.getVpcId()).thenReturn(vpcId);
+        return remoteAccessVpn;
+    }
 }
