--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -123,6 +123,8 @@
     @Mock
     private UserVO userVoMock;
 
+    private UserVO userVOMock;
+
     @Mock
     private ApiKeyPairService apiKeyPairService;
 
@@ -178,6 +180,7 @@
 
         allowOperationsOnUsersInSameAccountMock = Mockito.mock(ConfigKey.class);
         accountManagerImpl.allowOperationsOnUsersInSameAccount = allowOperationsOnUsersInSameAccountMock;
+        userVOMock = Mockito.mock(UserVO.class);
     }
 
     @Before
@@ -1136,10 +1139,9 @@
         Long userId = 1L;
 
         UserAccountVO userAccount = Mockito.mock(UserAccountVO.class);
-        UserVO userVO = Mockito.mock(UserVO.class);
 
         Mockito.when(userAccountDao.findById(userId)).thenReturn(userAccount);
-        Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
+        Mockito.when(userDaoMock.findById(userId)).thenReturn(userVOMock);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
 
         ConfigKey<Boolean> enableUserTwoFactorAuthentication = Mockito.mock(ConfigKey.class);
@@ -1155,10 +1157,9 @@
         Long userId = 1L;
 
         UserAccountVO userAccount = Mockito.mock(UserAccountVO.class);
-        UserVO userVO = Mockito.mock(UserVO.class);
 
         Mockito.when(userAccountDao.findById(userId)).thenReturn(userAccount);
-        Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
+        Mockito.when(userDaoMock.findById(userId)).thenReturn(userVOMock);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
 
         ConfigKey<Boolean> enableUserTwoFactorAuthentication = Mockito.mock(ConfigKey.class);
@@ -1184,10 +1185,9 @@
         Long userId = 1L;
 
         UserAccountVO userAccount = Mockito.mock(UserAccountVO.class);
-        UserVO userVO = Mockito.mock(UserVO.class);
 
         Mockito.when(userAccountDao.findById(userId)).thenReturn(userAccount);
-        Mockito.when(userDaoMock.findById(userId)).thenReturn(userVO);
+        Mockito.when(userDaoMock.findById(userId)).thenReturn(userVOMock);
         Mockito.when(userAccount.getDomainId()).thenReturn(1L);
 
         ConfigKey<Boolean> enableUserTwoFactorAuthentication = Mockito.mock(ConfigKey.class);
