--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -392,14 +392,13 @@
         // Mock vms on each host.
         long offeringIdForVmsOfThisAccount = 15L;
         long offeringIdForVmsOfOtherAccount = 16L;
-        UserVmVO vm1 = mock(UserVmVO.class);
+        UserVmVO vm1 = createMockUserVm();
         when(vm1.getAccountId()).thenReturn(accountId);
-        when(vm1.getServiceOfferingId()).thenReturn(offeringIdForVmsOfThisAccount);
-        UserVmVO vm2 = mock(UserVmVO.class);
+        UserVmVO vm2 = createMockUserVm();
         when(vm2.getAccountId()).thenReturn(accountId);
         when(vm2.getServiceOfferingId()).thenReturn(offeringIdForVmsOfThisAccount);
         // Vm from different account
-        UserVmVO vm3 = mock(UserVmVO.class);
+        UserVmVO vm3 = createMockUserVm();
         when(vm3.getAccountId()).thenReturn(201L);
         when(vm3.getServiceOfferingId()).thenReturn(offeringIdForVmsOfOtherAccount);
         List<VMInstanceVO> vmsForHost1 = new ArrayList<VMInstanceVO>();
@@ -427,6 +426,13 @@
         when(offeringForVMOfOtherAccount.getDeploymentPlanner()).thenReturn("FirstFitPlanner");
     }
 
+    private static UserVmVO createMockUserVm() {
+        UserVmVO vm = mock(UserVmVO.class);
+        when(vm.getAccountId()).thenReturn(200L);
+        when(vm.getServiceOfferingId()).thenReturn(15L);
+        return vm;
+    }
+
     @Configuration
     @ComponentScan(basePackageClasses = {ImplicitDedicationPlanner.class},
                    includeFilters = {@Filter(value = TestConfiguration.Library.class, type = FilterType.CUSTOM)},
