--- a/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
+++ b/plugins/maintenance/src/test/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImplTest.java
@@ -76,6 +76,10 @@
     @Mock
     ManagementServerHostDao msHostDao;
 
+    private ManagementServerHostVO createManagementServerHostMock() {
+        return mock(ManagementServerHostVO.class);
+    }
+
     private long prepareCountPendingJobs() {
         long expectedCount = 1L;
         Mockito.doReturn(expectedCount).when(jobManagerMock).countPendingNonPseudoJobs(1L);
@@ -158,7 +162,7 @@
     public void prepareForShutdownCmdOtherMsHostsInPreparingState() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost2);
         Mockito.lenient().when(msHostDao.listBy(any())).thenReturn(msHostList);
@@ -283,7 +287,7 @@
     public void triggerShutdownCmdMsInUpStateAndOtherMsHostsInPreparingState() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost2);
         Mockito.lenient().when(msHostDao.listBy(any())).thenReturn(msHostList);
@@ -393,7 +397,7 @@
     @Test
     public void prepareForMaintenanceCmdNoMsHost() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
         msHostList.add(msHost2);
@@ -411,7 +415,7 @@
     public void prepareForMaintenanceCmdMsHostWithNonUpState() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Maintenance);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
         msHostList.add(msHost2);
@@ -429,7 +433,7 @@
     public void prepareForMaintenanceCmdOtherMsHostsInPreparingState() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList1 = new ArrayList<>();
         msHostList1.add(msHost1);
         msHostList1.add(msHost2);
@@ -451,7 +455,7 @@
     public void prepareForMaintenanceCmdNoIndirectMsHosts() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
         msHostList.add(msHost2);
@@ -473,7 +477,7 @@
     public void prepareForMaintenanceCmdNullResponseFromClusterManager() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
         msHostList.add(msHost2);
@@ -494,7 +498,7 @@
     public void prepareForMaintenanceCmdFailedResponseFromClusterManager() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
         msHostList.add(msHost2);
@@ -515,7 +519,7 @@
     public void prepareForMaintenanceCmdSuccessResponseFromClusterManager() {
         ManagementServerHostVO msHost1 = mock(ManagementServerHostVO.class);
         Mockito.when(msHost1.getState()).thenReturn(ManagementServerHost.State.Up);
-        ManagementServerHostVO msHost2 = mock(ManagementServerHostVO.class);
+        ManagementServerHostVO msHost2 = createManagementServerHostMock();
         List<ManagementServerHostVO> msHostList = new ArrayList<>();
         msHostList.add(msHost1);
         msHostList.add(msHost2);
