--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -1159,10 +1159,7 @@
 
         Host destHost = mock(Host.class);
         Pod destPod = mock(Pod.class);
-        DeployDestination dest = mock(DeployDestination.class);
-        when(dest.getHost()).thenReturn(destHost);
-        when(dest.getPod()).thenReturn(destPod);
-        when(dest.getCluster()).thenReturn(mock(Cluster.class));
+        DeployDestination dest = createMockDeployDestination(destHost, destPod);
         when(destHost.getId()).thenReturn(1L);
         when(destPod.getId()).thenReturn(2L);
         when(_dpMgr.planDeployment(any(VirtualMachineProfileImpl.class), any(DataCenterDeployment.class), any(ExcludeList.class), any(DeploymentPlanner.class))).thenReturn(dest);
@@ -1255,10 +1252,7 @@
 
         Host destHost = mock(Host.class);
         Pod destPod = mock(Pod.class);
-        DeployDestination dest = mock(DeployDestination.class);
-        when(dest.getHost()).thenReturn(destHost);
-        when(dest.getPod()).thenReturn(destPod);
-        when(dest.getCluster()).thenReturn(mock(Cluster.class));
+        DeployDestination dest = createMockDeployDestination(destHost, destPod);
         when(destHost.getId()).thenReturn(1L);
         when(destPod.getId()).thenReturn(2L);
         when(_dpMgr.planDeployment(any(VirtualMachineProfileImpl.class), any(DataCenterDeployment.class), any(ExcludeList.class), any(DeploymentPlanner.class))).thenReturn(dest);
@@ -2241,4 +2235,12 @@
         verify(clvmPoolManagerMock, times(1)).setClvmLockHostId(3L, destHostId);
     }
 
+    private static DeployDestination createMockDeployDestination(Host destHost, Pod destPod) {
+        DeployDestination dest = mock(DeployDestination.class);
+        when(dest.getHost()).thenReturn(destHost);
+        when(dest.getPod()).thenReturn(destPod);
+        when(dest.getCluster()).thenReturn(mock(Cluster.class));
+        return dest;
+    }
+
 }
