--- a/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
+++ b/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
@@ -138,13 +138,12 @@
 
         Map<Long, List<String>> emptyMap = new HashMap<Long, List<String>>();
 
-        LoadBalancerDao lbdao = Mockito.mock(LoadBalancerDao.class);
+        LoadBalancerDao lbdao = createMockLoadBalancerDao();
         _lbMgr._lbDao =  lbdao;
 
         AutoScaleVmGroupDao autoScaleVmGroupDao = Mockito.mock(AutoScaleVmGroupDao.class);
         _lbMgr._autoScaleVmGroupDao = autoScaleVmGroupDao;
 
-        when(lbdao.findById(anyLong())).thenReturn(Mockito.mock(LoadBalancerVO.class));
         when(autoScaleVmGroupDao.isAutoScaleLoadBalancer(anyLong())).thenReturn(Boolean.FALSE);
 
         _lbMgr.assignToLoadBalancer(1L, null, emptyMap, null, false);
@@ -305,4 +304,10 @@
         CallContext.unregister();
     }
 
+    private static LoadBalancerDao createMockLoadBalancerDao() {
+        LoadBalancerDao lbdao = Mockito.mock(LoadBalancerDao.class);
+        when(lbdao.findById(anyLong())).thenReturn(Mockito.mock(LoadBalancerVO.class));
+        return lbdao;
+    }
+
 }
