--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -179,6 +179,7 @@
     @Spy
     @InjectMocks
     private VirtualMachineManagerImpl virtualMachineManagerImpl = new VirtualMachineManagerImpl();
+    private Account ownerMock;
     @Mock
     private AgentManager agentManagerMock;
     @Mock
@@ -301,6 +302,7 @@
 
     @Before
     public void setup() {
+        ownerMock = mock(Account.class);
         ReflectionTestUtils.getField(VirtualMachineManager.VmMetadataManufacturer, "s_depot");
         virtualMachineManagerImpl.setHostAllocators(new ArrayList<>());
 
@@ -1124,7 +1126,6 @@
         User user = mock(User.class);
 
         Account account = mock(Account.class);
-        Account owner = mock(Account.class);
 
         ReservationContext ctx = mock(ReservationContext.class);
 
@@ -1148,13 +1149,13 @@
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
-        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
+        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, ownerMock, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
         when(serviceOfferingDaoMock.findById(vmInstance.getId(), vmInstance.getServiceOfferingId())).thenReturn(serviceOffering);
 
-        when(_entityMgr.findById(Account.class, vmInstance.getAccountId())).thenReturn(owner);
+        when(_entityMgr.findById(Account.class, vmInstance.getAccountId())).thenReturn(ownerMock);
         when(_entityMgr.findByIdIncludingRemoved(VirtualMachineTemplate.class, vmInstance.getTemplateId())).thenReturn(template);
 
         Host destHost = mock(Host.class);
@@ -1220,7 +1221,6 @@
         User user = mock(User.class);
 
         Account account = mock(Account.class);
-        Account owner = mock(Account.class);
 
         ReservationContext ctx = mock(ReservationContext.class);
 
@@ -1244,13 +1244,13 @@
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
         Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
-        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
+        Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, ownerMock, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
 
         when(serviceOfferingDaoMock.findById(vmInstance.getId(), vmInstance.getServiceOfferingId())).thenReturn(serviceOffering);
 
-        when(_entityMgr.findById(Account.class, vmInstance.getAccountId())).thenReturn(owner);
+        when(_entityMgr.findById(Account.class, vmInstance.getAccountId())).thenReturn(ownerMock);
         when(_entityMgr.findByIdIncludingRemoved(VirtualMachineTemplate.class, vmInstance.getTemplateId())).thenReturn(template);
 
         Host destHost = mock(Host.class);
