--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/InstanceIpModelTest.java
@@ -77,8 +77,7 @@
         when(vm.getState()).thenReturn(VirtualMachine.State.Running);
         when(vm.getDomainId()).thenReturn(10L);
         when(vm.getAccountId()).thenReturn(42L);
-        UserVmDao VmDao = mock(UserVmDao.class);
-        when(VmDao.findById(anyLong())).thenReturn(null);
+        UserVmDao VmDao = MockUserVmDao.createMockUserVmDao();
         when(controller.getVmDao()).thenReturn(VmDao);
 
         VirtualMachineModel vmModel = new VirtualMachineModel(vm, UUID.randomUUID().toString());
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VirtualMachineModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VirtualMachineModelTest.java
@@ -88,8 +88,7 @@
         when(vm.getDomainId()).thenReturn(10L);
         when(vm.getAccountId()).thenReturn(42L);
 
-        UserVmDao VmDao = mock(UserVmDao.class);
-        when(VmDao.findById(anyLong())).thenReturn(null);
+        UserVmDao VmDao = MockUserVmDao.createMockUserVmDao();
         when(controller.getVmDao()).thenReturn(VmDao);
 
         VirtualMachineModel vmModel = new VirtualMachineModel(vm, uuid);
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/VMInterfaceModelTest.java
@@ -79,8 +79,7 @@
         when(vm.getState()).thenReturn(VirtualMachine.State.Running);
         when(vm.getDomainId()).thenReturn(10L);
         when(vm.getAccountId()).thenReturn(42L);
-        UserVmDao VmDao = mock(UserVmDao.class);
-        when(VmDao.findById(anyLong())).thenReturn(null);
+        UserVmDao VmDao = MockUserVmDao.createMockUserVmDao();
         when(controller.getVmDao()).thenReturn(VmDao);
 
         VirtualMachineModel vmModel = new VirtualMachineModel(vm, UUID.randomUUID().toString());
--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/MockUserVmDao.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/MockUserVmDao.java
@@ -0,0 +1,15 @@
+package org.apache.cloudstack.network.contrail.model;
+
+import com.cloud.vm.dao.UserVmDao;
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+
+public class MockUserVmDao {
+    public static UserVmDao createMockUserVmDao() {
+        UserVmDao VmDao = Mockito.mock(UserVmDao.class);
+        when(VmDao.findById(anyLong())).thenReturn(null);
+        return VmDao;
+    }
+}
