--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -116,6 +116,8 @@
 
     ApiResponseHelper helper;
 
+    private NicSecondaryIp nicSecondaryIpMock;
+
     @Mock
     AccountManager accountManagerMock;
 
@@ -180,6 +182,7 @@
 
     @Before
     public void setup() {
+        nicSecondaryIpMock = Mockito.mock(NicSecondaryIp.class);
         AccountVO account = new AccountVO("testaccount", 1L, "networkdomain", Account.Type.NORMAL, "uuid");
         account.setId(1);
         UserVO user = new UserVO(1, "testuser", "password", "firstname", "lastName", "email", "timezone",
@@ -245,11 +248,10 @@
 
     @Test
     public void setResponseIpAddressTestIpv4() {
-        NicSecondaryIp result = Mockito.mock(NicSecondaryIp.class);
         NicSecondaryIpResponse response = new NicSecondaryIpResponse();
-        setResult(result, "ipv4", "ipv6");
-
-        ApiResponseHelper.setResponseIpAddress(result, response);
+        setResult(nicSecondaryIpMock, "ipv4", "ipv6");
+
+        ApiResponseHelper.setResponseIpAddress(nicSecondaryIpMock, response);
 
         assertTrue(response.getIpAddr().equals("ipv4"));
     }
@@ -261,11 +263,10 @@
 
     @Test
     public void setResponseIpAddressTestIpv6() {
-        NicSecondaryIp result = Mockito.mock(NicSecondaryIp.class);
         NicSecondaryIpResponse response = new NicSecondaryIpResponse();
-        setResult(result, null, "ipv6");
-
-        ApiResponseHelper.setResponseIpAddress(result, response);
+        setResult(nicSecondaryIpMock, null, "ipv6");
+
+        ApiResponseHelper.setResponseIpAddress(nicSecondaryIpMock, response);
 
         assertTrue(response.getIpAddr().equals("ipv6"));
     }
