--- a/engine/storage/volume/src/test/java/org/apache/cloudstack/storage/volume/VolumeServiceTest.java
+++ b/engine/storage/volume/src/test/java/org/apache/cloudstack/storage/volume/VolumeServiceTest.java
@@ -101,6 +101,9 @@
     @Mock
     DiskOfferingDao diskOfferingDaoMock;
 
+    @Mock
+    private StoragePool storagePoolMock;
+
     @Before
     public void setup(){
         volumeServiceImplSpy = Mockito.spy(new VolumeServiceImpl());
@@ -257,8 +260,7 @@
     public void testCheckAndRepairVolume() throws StorageUnavailableException {
         VolumeInfo volume = Mockito.mock(VolumeInfo.class);
         Mockito.when(volume.getPoolId()).thenReturn(1L);
-        StoragePool pool = Mockito.mock(StoragePool.class);
-        Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(pool);
+        Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(storagePoolMock);
         List<Long> hostIds = new ArrayList<>();
         hostIds.add(1L);
         Mockito.when(storageManagerMock.getUpHostsInPool(1L)).thenReturn(hostIds);
@@ -282,12 +284,12 @@
                 "    \"fragmented-clusters\": 96135\n" +
                 "}";
 
-        CheckAndRepairVolumeCommand command = new CheckAndRepairVolumeCommand(volume.getPath(), new StorageFilerTO(pool), payload.getRepair(),
+        CheckAndRepairVolumeCommand command = new CheckAndRepairVolumeCommand(volume.getPath(), new StorageFilerTO(storagePoolMock), payload.getRepair(),
                 volume.getPassphrase(), volume.getEncryptFormat());
 
         CheckAndRepairVolumeAnswer answer = new CheckAndRepairVolumeAnswer(command, true, checkResult);
         answer.setVolumeCheckExecutionResult(checkResult);
-        Mockito.when(storageManagerMock.sendToPool(pool, new long[]{1L}, command)).thenReturn(answer);
+        Mockito.when(storageManagerMock.sendToPool(storagePoolMock, new long[]{1L}, command)).thenReturn(answer);
 
         Pair<String, String> result = volumeServiceImplSpy.checkAndRepairVolume(volume);
 
@@ -299,8 +301,7 @@
     public void testCheckAndRepairVolumeWhenFailure() throws StorageUnavailableException {
         VolumeInfo volume = Mockito.mock(VolumeInfo.class);
         Mockito.when(volume.getPoolId()).thenReturn(1L);
-        StoragePool pool = Mockito.mock(StoragePool.class);
-        Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(pool);
+        Mockito.when(storageManagerMock.getStoragePool(1L)).thenReturn(storagePoolMock);
         List<Long> hostIds = new ArrayList<>();
         hostIds.add(1L);
         Mockito.when(storageManagerMock.getUpHostsInPool(1L)).thenReturn(hostIds);
@@ -313,11 +314,11 @@
         Mockito.when(volume.getPassphrase()).thenReturn(new byte[] {3, 1, 2, 3});
         Mockito.when(volume.getEncryptFormat()).thenReturn("LUKS");
 
-        CheckAndRepairVolumeCommand command = new CheckAndRepairVolumeCommand(volume.getPath(), new StorageFilerTO(pool), payload.getRepair(),
+        CheckAndRepairVolumeCommand command = new CheckAndRepairVolumeCommand(volume.getPath(), new StorageFilerTO(storagePoolMock), payload.getRepair(),
                 volume.getPassphrase(), volume.getEncryptFormat());
 
         CheckAndRepairVolumeAnswer answer = new CheckAndRepairVolumeAnswer(command, false, "Unable to execute qemu command");
-        Mockito.when(storageManagerMock.sendToPool(pool, new long[]{1L}, command)).thenReturn(answer);
+        Mockito.when(storageManagerMock.sendToPool(storagePoolMock, new long[]{1L}, command)).thenReturn(answer);
 
         Pair<String, String> result = volumeServiceImplSpy.checkAndRepairVolume(volume);
 
