--- a/plugins/storage/sharedfs/storagevm/src/test/java/org/apache/cloudstack/storage/sharedfs/lifecycle/StorageVmSharedFSLifeCycleTest.java
+++ b/plugins/storage/sharedfs/storagevm/src/test/java/org/apache/cloudstack/storage/sharedfs/lifecycle/StorageVmSharedFSLifeCycleTest.java
@@ -182,9 +182,7 @@
     public void testCheckPrerequisites() {
         DataCenterVO zone = mock(DataCenterVO.class);
         when(zone.getId()).thenReturn(s_zoneId);
-        ServiceOfferingVO serviceOfferingVO = mock(ServiceOfferingVO.class);
-        when(serviceOfferingVO.getCpu()).thenReturn(4);
-        when(serviceOfferingVO.getRamSize()).thenReturn(1024);
+        ServiceOfferingVO serviceOfferingVO = createMockServiceOfferingVO();
         when(serviceOfferingVO.isOfferHA()).thenReturn(true);
         when(serviceOfferingDao.findById(s_serviceOfferingId)).thenReturn(serviceOfferingVO);
         lifeCycle.checkPrerequisites(zone, s_serviceOfferingId);
@@ -215,10 +213,8 @@
     public void testCheckPrerequisitesHAException() {
         DataCenterVO zone = mock(DataCenterVO.class);
         when(zone.getId()).thenReturn(s_zoneId);
-        ServiceOfferingVO serviceOfferingVO = mock(ServiceOfferingVO.class);
+        ServiceOfferingVO serviceOfferingVO = createMockServiceOfferingVO();
         when(serviceOfferingDao.findById(s_serviceOfferingId)).thenReturn(serviceOfferingVO);
-        when(serviceOfferingVO.getCpu()).thenReturn(4);
-        when(serviceOfferingVO.getRamSize()).thenReturn(1024);
         InvalidParameterValueException exception = Assert.assertThrows(InvalidParameterValueException.class, () -> lifeCycle.checkPrerequisites(zone, s_serviceOfferingId));
         Assert.assertEquals(exception.getMessage(), "Service offering's should be HA enabled");
     }
@@ -393,4 +389,11 @@
         boolean result = lifeCycle.reDeploySharedFS(sharedFS);
         Assert.assertEquals(result, true);
     }
+
+    private static ServiceOfferingVO createMockServiceOfferingVO() {
+        ServiceOfferingVO serviceOfferingVO = mock(ServiceOfferingVO.class);
+        when(serviceOfferingVO.getCpu()).thenReturn(4);
+        when(serviceOfferingVO.getRamSize()).thenReturn(1024);
+        return serviceOfferingVO;
+    }
 }
