--- a/api/src/test/java/org/apache/cloudstack/api/command/user/vm/ListVMsCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/vm/ListVMsCmdTest.java
@@ -166,12 +166,8 @@
         when(vm2.getGuestOsId()).thenReturn(os2Uuid);
         when(vm2.getId()).thenReturn(vm2Uuid);
         List<UserVmResponse> responses = Arrays.asList(vm1, vm2);
-        GuestOS guestOS1 = mock(GuestOS.class);
-        when(guestOS1.getUuid()).thenReturn(os1Uuid);
-        when(guestOS1.getCategoryId()).thenReturn(10L);
-        GuestOS guestOS2 = mock(GuestOS.class);
-        when(guestOS2.getUuid()).thenReturn(os2Uuid);
-        when(guestOS2.getCategoryId()).thenReturn(20L);
+        GuestOS guestOS1 = createMockGuestOS(os1Uuid, 10L);
+        GuestOS guestOS2 = createMockGuestOS(os2Uuid, 20L);
         when(_entityMgr.listByUuids(eq(GuestOS.class), anySet()))
                 .thenReturn(Arrays.asList(guestOS1, guestOS2));
         ResourceIcon icon1 = mock(ResourceIcon.class);
@@ -209,9 +205,7 @@
         when(vm1.getGuestOsId()).thenReturn(osUuid);
         when(vm1.getId()).thenReturn(vmUuid);
         List<UserVmResponse> responses = Collections.singletonList(vm1);
-        GuestOS guestOS1 = mock(GuestOS.class);
-        when(guestOS1.getCategoryId()).thenReturn(10L);
-        when(guestOS1.getUuid()).thenReturn(osUuid);
+        GuestOS guestOS1 = createMockGuestOS(osUuid, 10L);
         when(_entityMgr.listByUuids(eq(GuestOS.class), anySet()))
                 .thenReturn(Collections.singletonList(guestOS1));
         when(resourceIconManager.getByResourceTypeAndIds(eq(ResourceTag.ResourceObjectType.GuestOsCategory), anySet()))
@@ -220,4 +214,11 @@
         assertTrue(result.containsKey(vmUuid));
         assertNull(result.get(vmUuid));
     }
+
+    private static GuestOS createMockGuestOS(String uuid, long categoryId) {
+        GuestOS guestOS1 = mock(GuestOS.class);
+        when(guestOS1.getUuid()).thenReturn(uuid);
+        when(guestOS1.getCategoryId()).thenReturn(categoryId);
+        return guestOS1;
+    }
 }
