--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,8 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    @Mock
+    private VirtualRoutingResource routingResourceMock;
 
     @Test
     public void testWrapperInstance() {
@@ -1835,7 +1837,6 @@
 
     @Test
     public void testIpAssocVpcCommand() {
-        final VirtualRoutingResource routingResource = Mockito.mock(VirtualRoutingResource.class);
         final IpAddressTO[] ips = new IpAddressTO[0];
 
         final IpAssocVpcCommand ipAssociation = new IpAssocVpcCommand(ips);
@@ -1843,11 +1844,11 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getVirtualRoutingResource()).thenReturn(routingResource);
+        when(citrixResourceBase.getVirtualRoutingResource()).thenReturn(routingResourceMock);
 
         final Answer answer = wrapper.execute(ipAssociation, citrixResourceBase);
 
-        verify(routingResource, times(1)).executeRequest(ipAssociation);
+        verify(routingResourceMock, times(1)).executeRequest(ipAssociation);
 
         // Requires more testing, but the VirtualResourceRouting is quite big.
         assertNull(answer);
@@ -1855,7 +1856,6 @@
 
     @Test
     public void testIpAssocCommand() {
-        final VirtualRoutingResource routingResource = Mockito.mock(VirtualRoutingResource.class);
         final IpAddressTO[] ips = new IpAddressTO[0];
 
         final IpAssocCommand ipAssociation = new IpAssocCommand(ips);
@@ -1863,11 +1863,11 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getVirtualRoutingResource()).thenReturn(routingResource);
+        when(citrixResourceBase.getVirtualRoutingResource()).thenReturn(routingResourceMock);
 
         final Answer answer = wrapper.execute(ipAssociation, citrixResourceBase);
 
-        verify(routingResource, times(1)).executeRequest(ipAssociation);
+        verify(routingResourceMock, times(1)).executeRequest(ipAssociation);
 
         // Requires more testing, but the VirtualResourceRouting is quite big.
         assertNull(answer);
