--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -458,6 +458,12 @@
         }
     }
 
+    private static Capacity createMockCapacity(String tag) {
+        Capacity capacity = Mockito.mock(Capacity.class);
+        Mockito.when(capacity.getTag()).thenReturn(tag);
+        return capacity;
+    }
+
     private UnmanagedInstanceTO getUnmanagedInstanceForTests() {
         UnmanagedInstanceTO instance = Mockito.mock(UnmanagedInstanceTO.class);
         Mockito.when(instance.getPowerState()).thenReturn(UnmanagedInstanceTO.PowerState.PowerOff);
@@ -535,12 +541,9 @@
 
     @Test
     public void testCapacityListingForSingleTag() {
-        Capacity c1 = Mockito.mock(Capacity.class);
-        Mockito.when(c1.getTag()).thenReturn("tag1");
-        Capacity c2 = Mockito.mock(Capacity.class);
-        Mockito.when(c2.getTag()).thenReturn("tag1");
-        Capacity c3 = Mockito.mock(Capacity.class);
-        Mockito.when(c3.getTag()).thenReturn("tag2");
+        Capacity c1 = createMockCapacity("tag1");
+        Capacity c2 = createMockCapacity("tag1");
+        Capacity c3 = createMockCapacity("tag2");
         Capacity c4 = Mockito.mock(Capacity.class);
         Assert.assertTrue(apiResponseHelper.capacityListingForSingleTag(List.of(c1, c2)));
         Assert.assertFalse(apiResponseHelper.capacityListingForSingleTag(List.of(c1, c2, c3)));
