--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCheckAndRepairVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtCheckAndRepairVolumeCommandWrapperTest.java
@@ -72,9 +72,8 @@
         when(spool.getUuid()).thenReturn("b6be258b-42b8-49a4-ad51-3634ef8ff76a");
         when(storagePoolMgr.getStoragePool(Storage.StoragePoolType.PowerFlex, "b6be258b-42b8-49a4-ad51-3634ef8ff76a")).thenReturn(pool);
 
-        KVMPhysicalDisk vol = Mockito.mock(KVMPhysicalDisk.class);
+        KVMPhysicalDisk vol = createMockKvmPhysicalDisk();
         when(pool.getPhysicalDisk("cbac516a-0f1f-4559-921c-1a7c6c408ccf")).thenReturn(vol);
-        Mockito.when(vol.getFormat()).thenReturn(QemuImg.PhysicalDiskFormat.QCOW2);
 
         String checkResult = "{\n" +
                 "    \"image-end-offset\": 6442582016,\n" +
--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVolumesOnStorageCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetVolumesOnStorageCommandWrapperTest.java
@@ -126,9 +126,8 @@
 
     @Test
     public void testLibvirtGetVolumesOnStorageCommandWrapperForVolume() {
-        KVMPhysicalDisk disk = Mockito.mock(KVMPhysicalDisk.class);
+        KVMPhysicalDisk disk = createMockKvmPhysicalDisk();
         Mockito.when(disk.getPath()).thenReturn(volumePath);
-        Mockito.when(disk.getFormat()).thenReturn(QemuImg.PhysicalDiskFormat.QCOW2);
         Mockito.when(disk.getQemuEncryptFormat()).thenReturn(QemuObject.EncryptFormat.LUKS);
         Mockito.when(storagePool.getPhysicalDisk(volumePath)).thenReturn(disk);
         Mockito.when(storagePool.getType()).thenReturn(poolType);
--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/MockKVMPhysicalDisk.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/MockKVMPhysicalDisk.java
@@ -0,0 +1,14 @@
+package com.cloud.hypervisor.kvm.resource.wrapper;
+
+import com.cloud.hypervisor.kvm.storage.KVMPhysicalDisk;
+import org.apache.cloudstack.utils.qemu.QemuImg;
+import org.mockito.Mockito;
+
+public class MockKVMPhysicalDisk {
+
+    public static KVMPhysicalDisk createMockKvmPhysicalDisk() {
+        KVMPhysicalDisk disk = Mockito.mock(KVMPhysicalDisk.class);
+        Mockito.when(disk.getFormat()).thenReturn(QemuImg.PhysicalDiskFormat.QCOW2);
+        return disk;
+    }
+}
