--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -70,6 +70,8 @@
     @Mock
     protected XenServer610Resource xenServer610Resource;
 
+    private final NicTO sharedNicTO3 = Mockito.mock(NicTO.class);
+
     @Test
     public void testCheckNetworkCommandFailure() {
         final XenServer610Resource xenServer610Resource = new XenServer610Resource();
@@ -118,8 +120,7 @@
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
         final NicTO nicTO2 = Mockito.mock(NicTO.class);
-        final NicTO nicTO3 = Mockito.mock(NicTO.class);
-        final NicTO [] nicTOs = {nicTO1, nicTO2, nicTO3};
+        final NicTO [] nicTOs = {nicTO1, nicTO2, sharedNicTO3};
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network networkForSm = Mockito.mock(Network.class);
@@ -156,7 +157,7 @@
             verify(xenServer610Resource, times(1)).prepareISO(conn, vmName, null, null);
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO1);
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO2);
-            verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO3);
+            verify(xenServer610Resource, times(1)).getNetwork(conn, sharedNicTO3);
 
             verify(xenServer610Resource, times(1)).getNativeNetworkForTraffic(conn, TrafficType.Storage, null);
             verify(nativeNetworkForTraffic, times(1)).getNetwork();
@@ -191,8 +192,7 @@
 
         final NicTO nicTO1 = Mockito.mock(NicTO.class);
         final NicTO nicTO2 = Mockito.mock(NicTO.class);
-        final NicTO nicTO3 = Mockito.mock(NicTO.class);
-        final NicTO [] nicTOs = {nicTO1, nicTO2, nicTO3};
+        final NicTO [] nicTOs = {nicTO1, nicTO2, sharedNicTO3};
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network network = Mockito.mock(Network.class);
@@ -214,7 +214,7 @@
         try {
             when(xenServer610Resource.getNetwork(conn, nicTO1)).thenReturn(nw1);
             when(xenServer610Resource.getNetwork(conn, nicTO2)).thenReturn(nw2);
-            when(xenServer610Resource.getNetwork(conn, nicTO3)).thenReturn(nw3);
+            when(xenServer610Resource.getNetwork(conn, sharedNicTO3)).thenReturn(nw3);
 
             when(xenServer610Resource.getNativeNetworkForTraffic(conn, TrafficType.Storage, null)).thenReturn(nativeNetworkForTraffic);
             when(nativeNetworkForTraffic.getNetwork()).thenReturn(network);
@@ -234,7 +234,7 @@
         try {
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO1);
             verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO2);
-            verify(xenServer610Resource, times(1)).getNetwork(conn, nicTO3);
+            verify(xenServer610Resource, times(1)).getNetwork(conn, sharedNicTO3);
 
             verify(xenServer610Resource, times(1)).getNativeNetworkForTraffic(conn, TrafficType.Storage, null);
             verify(nativeNetworkForTraffic, times(1)).getNetwork();
