--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -1418,8 +1418,7 @@
     @Test
     public void validateIfVmHaveBackupsTestExceptionWhenTryToDetachVolumeFromVMWhichBackupOffering() {
         try {
-            UserVmVO vm = Mockito.mock(UserVmVO.class);
-            when(vm.getBackupOfferingId()).thenReturn(1l);
+            UserVmVO vm = createMockUserVm();
             volumeApiServiceImpl.validateIfVmHasBackups(vm, false);
         } catch (Exception e) {
             Assert.assertEquals("Unable to detach volume, cannot detach volume from a VM that has backups. First remove the VM from the backup offering or set the global configuration 'backup.enable.attach.detach.of.volumes' to true.", e.getMessage());
@@ -1429,8 +1428,7 @@
     @Test
     public void validateIfVmHaveBackupsTestExceptionWhenTryToAttachVolumeFromVMWhichBackupOffering() {
         try {
-            UserVmVO vm = Mockito.mock(UserVmVO.class);
-            when(vm.getBackupOfferingId()).thenReturn(1l);
+            UserVmVO vm = createMockUserVm();
             volumeApiServiceImpl.validateIfVmHasBackups(vm, true);
         } catch (Exception e) {
             Assert.assertEquals("Unable to attach volume, please specify a VM that does not have any backups or set the global configuration 'backup.enable.attach.detach.of.volumes' to true.", e.getMessage());
@@ -2054,6 +2052,12 @@
             verify(volumeServiceMock, times(0)).resize(any(VolumeInfo.class));
         }
     }
+    private static UserVmVO createMockUserVm() {
+        UserVmVO vm = Mockito.mock(UserVmVO.class);
+        when(vm.getBackupOfferingId()).thenReturn(1l);
+        return vm;
+    }
+
     private UserVmVO getMockedVm() {
         UserVmVO vm = Mockito.mock(UserVmVO.class);
         Mockito.when(vm.getId()).thenReturn(1L);
