--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -183,12 +183,11 @@
         DataCenterVO mockDc = mock(DataCenterVO.class);
         ExcludeList avoids = mock(ExcludeList.class);
         VirtualMachineProfileImpl vmProfile = mock(VirtualMachineProfileImpl.class);
-        VMInstanceVO vm = mock(VMInstanceVO.class);
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
 
         when(avoids.shouldAvoid(mockDc)).thenReturn(true);
+        VMInstanceVO vm = createMockVm(1L);
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
-        when(vm.getDataCenterId()).thenReturn(1L);
         when(dcDao.findById(1L)).thenReturn(mockDc);
 
         List<Long> clusterList = planner.orderClusters(vmProfile, plan, avoids);
@@ -321,6 +320,12 @@
         // Validations.
         // Check cluster list is empty.
         assertTrue("Cluster list should not be null/empty", (clusterList == null || clusterList.isEmpty()));
+    }
+
+    private static VMInstanceVO createMockVm(long dataCenterIdReturn) {
+        VMInstanceVO vm = mock(VMInstanceVO.class);
+        when(vm.getDataCenterId()).thenReturn(dataCenterIdReturn);
+        return vm;
     }
 
     private void initializeForTest(VirtualMachineProfileImpl vmProfile, DataCenterDeployment plan) {
