--- a/api/src/test/java/org/apache/cloudstack/api/command/test/CreateAutoScaleVmProfileCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/CreateAutoScaleVmProfileCmdTest.java
@@ -96,7 +96,7 @@
         ReflectionTestUtils.setField(createAutoScaleVmProfileCmd,"projectId", projectId);
         ReflectionTestUtils.setField(createAutoScaleVmProfileCmd,"accountName", accountName);
 
-        autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
+        autoScaleVmProfile = MockAutoScaleVmProfile.createMockAutoScaleVmProfile();
 
         response = new AutoScaleVmProfileResponse();
         response.setUserData(userData);
@@ -141,8 +141,7 @@
 
     @Test
     public void testCreateAutoScaleVmProfileSuccess() {
-        when(autoScaleVmProfile.getId()).thenReturn(1L);
-        when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+
         when(autoScaleService.createAutoScaleVmProfile(createAutoScaleVmProfileCmd)).thenReturn(autoScaleVmProfile);
 
         try {
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/UpdateAutoScaleVmProfileCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/UpdateAutoScaleVmProfileCmdTest.java
@@ -78,7 +78,7 @@
         ReflectionTestUtils.setField(updateAutoScaleVmProfileCmd,"otherDeployParams", otherDeployParams);
         ReflectionTestUtils.setField(updateAutoScaleVmProfileCmd,"counterParamList", counterParamList);
 
-        autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
+        autoScaleVmProfile = MockAutoScaleVmProfile.createMockAutoScaleVmProfile();
 
         response = new AutoScaleVmProfileResponse();
         response.setUserData(userData);
@@ -105,8 +105,7 @@
 
     @Test
     public void testUpdateAutoScaleVmProfileSuccess() {
-        when(autoScaleVmProfile.getId()).thenReturn(1L);
-        when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+
         when(autoScaleService.updateAutoScaleVmProfile(updateAutoScaleVmProfileCmd)).thenReturn(autoScaleVmProfile);
 
         when(entityMgr.findById(AutoScaleVmProfile.class, 1L)).thenReturn(autoScaleVmProfile);
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/MockAutoScaleVmProfile.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/MockAutoScaleVmProfile.java
@@ -0,0 +1,15 @@
+package org.apache.cloudstack.api.command.test;
+
+import com.cloud.network.as.AutoScaleVmProfile;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockAutoScaleVmProfile {
+    public static AutoScaleVmProfile createMockAutoScaleVmProfile() {
+        AutoScaleVmProfile autoScaleVmProfile = Mockito.mock(AutoScaleVmProfile.class);
+        when(autoScaleVmProfile.getId()).thenReturn(1L);
+        when(autoScaleVmProfile.getUuid()).thenReturn("1111-2222-3333-4444");
+        return autoScaleVmProfile;
+    }
+}
