--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -359,13 +359,10 @@
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
 
-        VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
-        when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
+        VMTemplateStoragePoolVO mockTemplateStore = createMockTemplateStore();
 
         when(vmTemplateDao.findById(anyLong(), anyBoolean())).thenReturn(mockTemplate);
         when(vmTemplatePoolDao.findByPoolTemplate(anyLong(), anyLong(), nullable(String.class))).thenReturn(mockTemplateStore);
-
-        doNothing().when(mockTemplateStore).setMarkedForGC(anyBoolean());
 
         VMTemplateStoragePoolVO returnObject = templateManager.prepareTemplateForCreate(mockTemplate, (StoragePool) mockPrimaryDataStore);
         assertTrue("Test template is already seeded", returnObject == mockTemplateStore);
@@ -422,22 +419,19 @@
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         StoragePoolVO mockPool = mock(StoragePoolVO.class);
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
-        VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
+        VMTemplateStoragePoolVO mockTemplateStore = createMockTemplateStore();
 
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
         when(mockPool.getId()).thenReturn(2l);
         when(mockPool.getStatus()).thenReturn(StoragePoolStatus.Up);
         when(mockPool.getDataCenterId()).thenReturn(1l);
         when(mockTemplate.getId()).thenReturn(202l);
-        when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
         when(vmTemplateDao.findById(anyLong())).thenReturn(mockTemplate);
         when(dataStoreManager.getPrimaryDataStore(anyLong())).thenReturn(mockPrimaryDataStore);
         when(vmTemplateDao.findById(anyLong(), anyBoolean())).thenReturn(mockTemplate);
         when(vmTemplatePoolDao.findByPoolTemplate(anyLong(), anyLong(), nullable(String.class))).thenReturn(mockTemplateStore);
         when(primaryDataStoreDao.findById(anyLong())).thenReturn(mockPool);
 
-        doNothing().when(mockTemplateStore).setMarkedForGC(anyBoolean());
-
         ExecutorService preloadExecutor = new CustomThreadPoolExecutor(8, 8, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue(),
                 new NamedThreadFactory("Template-Preloader"));
         templateManager._preloadExecutor = preloadExecutor;
@@ -469,7 +463,6 @@
     public void testTemplateScheduledForDownloadInMultiplePool() {
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
-        VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
         List<StoragePoolVO> pools = new ArrayList<StoragePoolVO>();
 
         StoragePoolVO mockPool1 = mock(StoragePoolVO.class);
@@ -485,14 +478,12 @@
 
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
         when(mockTemplate.getId()).thenReturn(202l);
-        when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
+        VMTemplateStoragePoolVO mockTemplateStore = createMockTemplateStore();
         when(vmTemplateDao.findById(anyLong())).thenReturn(mockTemplate);
         when(dataStoreManager.getPrimaryDataStore(anyLong())).thenReturn(mockPrimaryDataStore);
         when(vmTemplateDao.findById(anyLong(), anyBoolean())).thenReturn(mockTemplate);
         when(vmTemplatePoolDao.findByPoolTemplate(anyLong(), anyLong(), nullable(String.class))).thenReturn(mockTemplateStore);
         when(primaryDataStoreDao.listByStatus(StoragePoolStatus.Up)).thenReturn(pools);
-
-        doNothing().when(mockTemplateStore).setMarkedForGC(anyBoolean());
 
         ExecutorService preloadExecutor = new CustomThreadPoolExecutor(8, 8, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue(),
                 new NamedThreadFactory("Template-Preloader"));
@@ -669,6 +660,13 @@
         Mockito.verify(dataStoreManager, Mockito.times(0)).getImageStoreWithFreeCapacity(Mockito.anyLong());
     }
 
+    private static VMTemplateStoragePoolVO createMockTemplateStore() {
+        VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
+        when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
+        doNothing().when(mockTemplateStore).setMarkedForGC(anyBoolean());
+        return mockTemplateStore;
+    }
+
     private HypervisorTemplateAdapter mockHypervisorTemplateAdapter() {
         HypervisorTemplateAdapter adapter = Mockito.mock(HypervisorTemplateAdapter.class);
         when(adapter.getName()).thenReturn(TemplateAdapter.TemplateAdapterType.Hypervisor.getName());
