--- a/api/src/test/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmdTest.java
@@ -35,6 +35,7 @@
 import org.apache.cloudstack.api.ApiConstants.BootType;
 import org.apache.cloudstack.api.ApiConstants.IoDriverPolicy;
 import org.apache.cloudstack.vm.lease.VMLeaseManager;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Spy;
@@ -58,6 +59,13 @@
     @Spy
     private DeployVMCmd cmd = new DeployVMCmd();
 
+    private NetworkService networkServiceMockField;
+
+    @Before
+    public void setUp() {
+        networkServiceMockField = mock(NetworkService.class);
+    }
+
     @Test
     public void testGetBootType_ValidUEFI() {
         ReflectionTestUtils.setField(cmd, "bootType", "UEFI");
@@ -352,8 +360,7 @@
         ReflectionTestUtils.setField(cmd, "ipAddress", null);
 
         Network mockNetwork = mock(Network.class);
-        NetworkService networkServiceMock = mock(NetworkService.class);
-        ReflectionTestUtils.setField(cmd, "_networkService", networkServiceMock);
+        ReflectionTestUtils.setField(cmd, "_networkService", networkServiceMockField);
 
         Map<Long, IpAddresses> result = cmd.getIpToNetworkMap();
 
@@ -377,8 +384,7 @@
         ReflectionTestUtils.setField(cmd, "ipAddress", null);
 
         Network mockNetwork = mock(Network.class);
-        NetworkService networkServiceMock = mock(NetworkService.class);
-        ReflectionTestUtils.setField(cmd, "_networkService", networkServiceMock);
+        ReflectionTestUtils.setField(cmd, "_networkService", networkServiceMockField);
 
         InvalidParameterValueException thrownException = assertThrows(InvalidParameterValueException.class, () -> {
             cmd.getIpToNetworkMap();
