--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -100,6 +100,8 @@
     @Mock
     Account caller;
 
+    private SearchCriteria<WebhookJoinVO> searchCriteria;
+
     @Spy
     @InjectMocks
     WebhookApiServiceImpl webhookApiServiceImpl = new WebhookApiServiceImpl();
@@ -108,6 +110,7 @@
 
     @Before
     public void setup() {
+        searchCriteria = Mockito.mock(SearchCriteria.class);
         callContextMocked = Mockito.mockStatic(CallContext.class);
         CallContext callContextMock = Mockito.mock(CallContext.class);
         callContextMocked.when(CallContext::current).thenReturn(callContextMock);
@@ -782,8 +785,7 @@
         Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         SearchBuilder<WebhookJoinVO> sb = Mockito.mock(SearchBuilder.class);
         Mockito.when(sb.entity()).thenReturn(Mockito.mock(WebhookJoinVO.class));
-        SearchCriteria<WebhookJoinVO> sc = Mockito.mock(SearchCriteria.class);
-        Mockito.when(sb.create()).thenReturn(sc);
+        Mockito.when(sb.create()).thenReturn(searchCriteria);
         Mockito.when(webhookJoinDao.createSearchBuilder()).thenReturn(sb);
         Mockito.when(webhookJoinDao.searchAndCount(Mockito.any(), Mockito.any())).thenReturn(new Pair<>(List.of(), 0));
 
@@ -825,8 +827,7 @@
         Mockito.when(cmd.getPageSizeVal()).thenReturn(10L);
         SearchBuilder<WebhookJoinVO> sb = Mockito.mock(SearchBuilder.class);
         Mockito.when(sb.entity()).thenReturn(Mockito.mock(WebhookJoinVO.class));
-        SearchCriteria<WebhookJoinVO> sc = Mockito.mock(SearchCriteria.class);
-        Mockito.when(sb.create()).thenReturn(sc);
+        Mockito.when(sb.create()).thenReturn(searchCriteria);
         Mockito.when(webhookJoinDao.createSearchBuilder()).thenReturn(sb);
         Mockito.when(webhookJoinDao.searchAndCount(Mockito.any(), Mockito.any())).thenReturn(new Pair<>(List.of(webhook), 1));
 
