--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -189,6 +189,12 @@
         closeable.close();
     }
 
+    private static FirewallRuleVO createMockIngressFirewallRule() {
+        FirewallRuleVO ingressFirewallRule = Mockito.mock(FirewallRuleVO.class);
+        Mockito.when(ingressFirewallRule.getTrafficType()).thenReturn(FirewallRule.TrafficType.Ingress);
+        return ingressFirewallRule;
+    }
+
     private DataCenterGuestIpv6PrefixVO prepareMocksForIpv6Subnet() {
         final long prefixId = 1L;
         DataCenterGuestIpv6PrefixVO prefix = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
@@ -663,8 +669,7 @@
             ipv6Service.updateIpv6FirewallRule(cmd);
             Assert.fail("No InvalidParameterValueException");
         } catch (InvalidParameterValueException ignored) {}
-        FirewallRuleVO ingressFirewallRule = Mockito.mock(FirewallRuleVO.class);
-        Mockito.when(ingressFirewallRule.getTrafficType()).thenReturn(FirewallRule.TrafficType.Ingress);
+        FirewallRuleVO ingressFirewallRule = createMockIngressFirewallRule();
         Mockito.when(firewallDao.findById(firewallRuleId)).thenReturn(ingressFirewallRule);
         try {
             ipv6Service.updateIpv6FirewallRule(cmd);
@@ -681,8 +686,7 @@
             ipv6Service.revokeIpv6FirewallRule(firewallRuleId);
             Assert.fail("No InvalidParameterValueException");
         } catch (InvalidParameterValueException ignored) {}
-        FirewallRuleVO ingressFirewallRule = Mockito.mock(FirewallRuleVO.class);
-        Mockito.when(ingressFirewallRule.getTrafficType()).thenReturn(FirewallRule.TrafficType.Ingress);
+        FirewallRuleVO ingressFirewallRule = createMockIngressFirewallRule();
         Mockito.when(firewallDao.findById(firewallRuleId)).thenReturn(ingressFirewallRule);
         try {
             ipv6Service.revokeIpv6FirewallRule(firewallRuleId);
