--- a/server/src/test/java/com/cloud/server/StatsCollectorTest.java
+++ b/server/src/test/java/com/cloud/server/StatsCollectorTest.java
@@ -130,6 +130,9 @@
 
     @Mock
     private NetworkDao networkDao;
+
+    @Mock
+    private NetworkVO networkVOMock;
 
     @Mock
     private VlanDao vlanDao;
@@ -773,10 +776,9 @@
     @Test
     public void isNetworkEligibleForNetworkStats_RoutedNetwork_ReturnsTrue() {
         Long networkId = 1L;
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
-        Mockito.when(networkDao.findById(networkId)).thenReturn(networkVO);
+        Mockito.when(networkDao.findById(networkId)).thenReturn(networkVOMock);
         Mockito.when(vlanDao.listVlansByNetworkId(networkId)).thenReturn(Collections.emptyList());
-        Mockito.when(routedIpv4Manager.isRoutedNetwork(networkVO)).thenReturn(true);
+        Mockito.when(routedIpv4Manager.isRoutedNetwork(networkVOMock)).thenReturn(true);
 
         Assert.assertTrue(statsCollector.isNetworkEligibleForNetworkStats(networkId));
     }
@@ -784,9 +786,8 @@
     @Test
     public void isNetworkEligibleForNetworkStats_DirectAttachedNetwork_ReturnsTrue() {
         Long networkId = 1L;
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
-        Mockito.when(networkDao.findById(networkId)).thenReturn(networkVO);
-        Mockito.when(routedIpv4Manager.isRoutedNetwork(networkVO)).thenReturn(false);
+        Mockito.when(networkDao.findById(networkId)).thenReturn(networkVOMock);
+        Mockito.when(routedIpv4Manager.isRoutedNetwork(networkVOMock)).thenReturn(false);
         List<VlanVO> vlans = Collections.singletonList(buildVlan(VlanType.DirectAttached));
         Mockito.when(vlanDao.listVlansByNetworkId(networkId)).thenReturn(vlans);
 
@@ -796,9 +797,8 @@
     @Test
     public void isNetworkEligibleForNetworkStats_NeitherRoutedNorDirectAttached_ReturnsFalse() {
         Long networkId = 1L;
-        NetworkVO networkVO = Mockito.mock(NetworkVO.class);
-        Mockito.when(networkDao.findById(networkId)).thenReturn(networkVO);
-        Mockito.when(routedIpv4Manager.isRoutedNetwork(networkVO)).thenReturn(false);
+        Mockito.when(networkDao.findById(networkId)).thenReturn(networkVOMock);
+        Mockito.when(routedIpv4Manager.isRoutedNetwork(networkVOMock)).thenReturn(false);
         List<VlanVO> vlans = Collections.singletonList(buildVlan(VlanType.VirtualNetwork));
         Mockito.when(vlanDao.listVlansByNetworkId(networkId)).thenReturn(vlans);
 
