--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -1132,6 +1132,12 @@
         Assert.assertFalse("Host2 should not be included as it does not match any storage access group.", hostsToConnect.contains(host2));
     }
 
+    private static ClusterVO createMockCluster() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.when(cluster.getId()).thenReturn(3L);
+        return cluster;
+    }
+
     @Test
     public void testUpdateZoneStorageAccessGroups() {
         long zoneId = 1L;
@@ -1152,9 +1158,9 @@
         Mockito.when(host2.getId()).thenReturn(host2Id);
 
         HostPodVO pod1 = Mockito.mock(HostPodVO.class);
-        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
+        ClusterVO cluster1 = createMockCluster();
         Mockito.when(pod1.getId()).thenReturn(podId);
-        Mockito.when(cluster1.getId()).thenReturn(clusterId);
+
         Mockito.when(podDao.findById(podId)).thenReturn(pod1);
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(cluster1);
         Mockito.when(podDao.listByDataCenterId(zoneId)).thenReturn(Collections.singletonList(pod1));
@@ -1189,9 +1195,9 @@
         Mockito.when(host2.getId()).thenReturn(host2Id);
 
         HostPodVO pod1 = Mockito.mock(HostPodVO.class);
-        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
+        ClusterVO cluster1 = createMockCluster();
         Mockito.when(pod1.getStorageAccessGroups()).thenReturn("group1,group3");
-        Mockito.when(cluster1.getId()).thenReturn(clusterId);
+
         Mockito.when(podDao.findById(podId)).thenReturn(pod1);
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(cluster1);
         Mockito.when(clusterDao.listByPodId(podId)).thenReturn(Collections.singletonList(cluster1));
@@ -1223,9 +1229,9 @@
         Mockito.when(host1.getId()).thenReturn(host1Id);
         Mockito.when(host2.getId()).thenReturn(host2Id);
 
-        ClusterVO cluster1 = Mockito.mock(ClusterVO.class);
+        ClusterVO cluster1 = createMockCluster();
         Mockito.when(cluster1.getStorageAccessGroups()).thenReturn("group1,group3");
-        Mockito.when(cluster1.getId()).thenReturn(clusterId);
+
         Mockito.when(clusterDao.findById(clusterId)).thenReturn(cluster1);
         Mockito.when(hostDao.findHypervisorHostInCluster(clusterId)).thenReturn(Arrays.asList(host1, host2));
         Mockito.when(hostDao.findByClusterId(clusterId)).thenReturn(Arrays.asList(host1, host2));
