--- a/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/NetworkHelperImplTest.java
@@ -166,8 +166,7 @@
             throws AgentUnavailableException, OperationTimedoutException, ResourceUnavailableException {
         // Prepare
         NetworkHelperImpl nwHelperUT = networkHelperSpy;
-        VirtualRouter vr = mock(VirtualRouter.class);
-        when(vr.getHostId()).thenReturn(HOST_ID);
+        VirtualRouter vr = createMockVirtualRouter();
         doReturn(true).when(nwHelperUT).checkRouterVersion(vr);
 
         Commands commands = mock(Commands.class);
@@ -193,6 +192,12 @@
         assertFalse(result);
     }
 
+    private static VirtualRouter createMockVirtualRouter() {
+        VirtualRouter vr = mock(VirtualRouter.class);
+        when(vr.getHostId()).thenReturn(HOST_ID);
+        return vr;
+    }
+
     /**
      * The only way result can be true is if each and every command receive a true result
      *
@@ -204,8 +209,7 @@
             throws AgentUnavailableException, OperationTimedoutException, ResourceUnavailableException {
         // Prepare
         NetworkHelperImpl nwHelperUT = networkHelperSpy;
-        VirtualRouter vr = mock(VirtualRouter.class);
-        when(vr.getHostId()).thenReturn(HOST_ID);
+        VirtualRouter vr = createMockVirtualRouter();
         doReturn(true).when(nwHelperUT).checkRouterVersion(vr);
 
         Commands commands = mock(Commands.class);
@@ -231,6 +235,12 @@
         assertTrue(result);
     }
 
+    private static VirtualRouter createMockVirtualRouter() {
+        VirtualRouter vr = mock(VirtualRouter.class);
+        when(vr.getHostId()).thenReturn(HOST_ID);
+        return vr;
+    }
+
     /**
      * If the number of answers is different to the number of commands the result is false
      *
@@ -242,8 +252,7 @@
             throws AgentUnavailableException, OperationTimedoutException, ResourceUnavailableException {
         // Prepare
         NetworkHelperImpl nwHelperUT = networkHelperSpy;
-        VirtualRouter vr = mock(VirtualRouter.class);
-        when(vr.getHostId()).thenReturn(HOST_ID);
+        VirtualRouter vr = createMockVirtualRouter();
         doReturn(true).when(nwHelperUT).checkRouterVersion(vr);
 
         Commands commands = mock(Commands.class);
@@ -534,4 +543,10 @@
         Map<Network, List<? extends NicProfile>> nic = networkHelperSpy.configurePublicNic(routerDeploymentDefinitionMock, false);
         Mockito.verify(ipv6ServiceMock).updateNicIpv6(Mockito.eq(nic.get(networkMock).get(0)), Mockito.any(), Mockito.eq(networkMock));
     }
+
+    private static VirtualRouter createMockVirtualRouter() {
+        VirtualRouter vr = mock(VirtualRouter.class);
+        when(vr.getHostId()).thenReturn(HOST_ID);
+        return vr;
+    }
 }
