--- a/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImplTest.java
@@ -423,6 +423,12 @@
         return instanceDisks;
     }
 
+    private static HostVO createMockHost(ResourceState resourceState) {
+        HostVO host = Mockito.mock(HostVO.class);
+        when(host.getResourceState()).thenReturn(resourceState);
+        return host;
+    }
+
     @After
     public void tearDown() throws Exception {
         closeable.close();
@@ -1226,8 +1232,7 @@
     public void testSelectKVMHostForImportingInClusterWithImportInstanceIdInvalidResourceState() {
         Long hostId = 1L;
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
-        when(host.getResourceState()).thenReturn(ResourceState.Disabled);
+        HostVO host = createMockHost(ResourceState.Disabled);
 
         when(hostDao.findById(hostId)).thenReturn(host);
 
@@ -1360,8 +1365,7 @@
     public void testSelectKVMHostForConversionInClusterWithImportInstanceIdInvalidResourceState() {
         Long hostId = 1L;
         ClusterVO cluster = getClusterForTests();
-        HostVO host = Mockito.mock(HostVO.class);
-        when(host.getResourceState()).thenReturn(ResourceState.Maintenance);
+        HostVO host = createMockHost(ResourceState.Maintenance);
 
         when(hostDao.findById(hostId)).thenReturn(host);
 
