--- a/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
+++ b/framework/quota/src/test/java/org/apache/cloudstack/quota/activationrule/presetvariables/PresetVariableHelperTest.java
@@ -370,6 +370,13 @@
         Mockito.doNothing().when(presetVariableHelperSpy).validateIfObjectIsNull(Mockito.any(), Mockito.anyLong(), Mockito.anyString());
     }
 
+    private static HostVO createMockHostVo() {
+        HostVO hostVoMock = Mockito.mock(HostVO.class);
+        Mockito.doReturn("host_id").when(hostVoMock).getUuid();
+        Mockito.doReturn("host_name").when(hostVoMock).getName();
+        return hostVoMock;
+    }
+
     private GenericPresetVariable getGenericPresetVariableForTests() {
         GenericPresetVariable gpv = new GenericPresetVariable();
         gpv.setId("test_id");
@@ -652,7 +659,7 @@
     @Test
     public void getPresetVariableValueHostTestSetFieldsAndReturnObject() {
         Host expected = getHostForTests();
-        HostVO hostVoMock = Mockito.mock(HostVO.class);
+        HostVO hostVoMock = createMockHostVo();
         List<HostTagVO> hostTagVOListMock = getHostTagsForTests();
 
         Mockito.doReturn(hostVoMock).when(hostDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
@@ -670,7 +677,7 @@
     @Test
     public void getPresetVariableValueHostTestSetFieldsWithRuleTagAndReturnObject() {
         Host expected = getHostForTests();
-        HostVO hostVoMock = Mockito.mock(HostVO.class);
+        HostVO hostVoMock = createMockHostVo();
         List<HostTagVO> hostTagVOListMock = getHostRuleTagsForTests();
 
         Mockito.doReturn(hostVoMock).when(hostDaoMock).findByIdIncludingRemoved(Mockito.anyLong());
