--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static NicVO createMockNicVO() {
+        NicVO nic1 = Mockito.mock(NicVO.class);
+        when(nic1.getDeviceId()).thenReturn(1);
+        return nic1;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1738,10 +1744,9 @@
 
         // Mock NICs
         List<NicVO> nics = new ArrayList<>();
-        NicVO nic1 = Mockito.mock(NicVO.class);
-        when(nic1.getDeviceId()).thenReturn(1);
+        NicVO nic1 = createMockNicVO();
         nics.add(nic1);
-        NicVO nic2 = Mockito.mock(NicVO.class);
+        NicVO nic2 = createMockNicVO();
         when(nic2.getDeviceId()).thenReturn(0);
         nics.add(nic2);
         when(_nicsDao.listByVmId(vmInstanceVoMockId)).thenReturn(nics);
