--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/resource/NiciraNvpResourceTest.java
@@ -172,8 +172,7 @@
     public void testRetries() throws ConfigurationException, NiciraNvpApiException {
         resource.configure("NiciraNvpResource", parameters);
 
-        final LogicalSwitch ls = mock(LogicalSwitch.class);
-        when(ls.getUuid()).thenReturn("cccc").thenReturn("cccc");
+        final LogicalSwitch ls = createMockLogicalSwitch();
         when(nvpApi.createLogicalSwitch((LogicalSwitch) any())).thenThrow(new NiciraNvpApiException()).thenThrow(new NiciraNvpApiException()).thenReturn(ls);
 
         final CreateLogicalSwitchCommand clsc = new CreateLogicalSwitchCommand((String) parameters.get("guid"), "stt", "loigicalswitch", "owner");
@@ -185,8 +184,7 @@
     public void testCreateLogicalSwitch() throws ConfigurationException, NiciraNvpApiException {
         resource.configure("NiciraNvpResource", parameters);
 
-        final LogicalSwitch ls = mock(LogicalSwitch.class);
-        when(ls.getUuid()).thenReturn("cccc").thenReturn("cccc");
+        final LogicalSwitch ls = createMockLogicalSwitch();
         when(nvpApi.createLogicalSwitch((LogicalSwitch) any())).thenReturn(ls);
 
         final CreateLogicalSwitchCommand clsc = new CreateLogicalSwitchCommand((String) parameters.get("guid"), "stt", "loigicalswitch", "owner");
@@ -199,8 +197,7 @@
     public void testCreateLogicalSwitchApiException() throws ConfigurationException, NiciraNvpApiException {
         resource.configure("NiciraNvpResource", parameters);
 
-        final LogicalSwitch ls = mock(LogicalSwitch.class);
-        when(ls.getUuid()).thenReturn("cccc").thenReturn("cccc");
+        final LogicalSwitch ls = createMockLogicalSwitch();
         when(nvpApi.createLogicalSwitch((LogicalSwitch) any())).thenThrow(new NiciraNvpApiException());
 
         final CreateLogicalSwitchCommand clsc = new CreateLogicalSwitchCommand((String) parameters.get("guid"), "stt", "loigicalswitch", "owner");
@@ -822,4 +819,10 @@
         assertTrue(snr.getMatch().getSourcePort() == 8080);
         assertTrue(snr.getMatch().getEthertype().equals("IPv4") && rules[1].getMatch().getProtocol() == 6);
     }
+
+    private static LogicalSwitch createMockLogicalSwitch() {
+        final LogicalSwitch ls = mock(LogicalSwitch.class);
+        when(ls.getUuid()).thenReturn("cccc").thenReturn("cccc");
+        return ls;
+    }
 }
