--- a/server/src/test/java/com/cloud/template/HypervisorTemplateAdapterTest.java
+++ b/server/src/test/java/com/cloud/template/HypervisorTemplateAdapterTest.java
@@ -130,6 +130,8 @@
     @Mock
     DataStoreManager dataStoreManagerMock;
 
+    private TemplateProfile templateProfileSharedMock;
+
     @Mock
     HeuristicRuleHelper heuristicRuleHelperMock;
 
@@ -156,6 +158,7 @@
     @Before
     public void before() {
         closeable = MockitoAnnotations.openMocks(this);
+        templateProfileSharedMock = Mockito.mock(TemplateProfile.class);
     }
 
     @After
@@ -380,32 +383,29 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void getImageStoresThrowsExceptionIfNotFoundTestNullImageStoreShouldThrowCloudRuntimeException() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
 
         Mockito.when(dataStoreManagerMock.getImageStoresByZoneIds(Mockito.anyLong())).thenReturn(null);
 
-        _adapter.getImageStoresThrowsExceptionIfNotFound(zoneId, templateProfileMock);
+        _adapter.getImageStoresThrowsExceptionIfNotFound(zoneId, templateProfileSharedMock);
     }
 
     @Test(expected = CloudRuntimeException.class)
     public void getImageStoresThrowsExceptionIfNotFoundTestEmptyImageStoreShouldThrowCloudRuntimeException() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
         List<DataStore> imageStoresList = new ArrayList<>();
 
         Mockito.when(dataStoreManagerMock.getImageStoresByZoneIds(Mockito.anyLong())).thenReturn(imageStoresList);
 
-        _adapter.getImageStoresThrowsExceptionIfNotFound(zoneId, templateProfileMock);
+        _adapter.getImageStoresThrowsExceptionIfNotFound(zoneId, templateProfileSharedMock);
     }
 
     @Test
     public void getImageStoresThrowsExceptionIfNotFoundTestNonEmptyImageStoreShouldNotThrowCloudRuntimeException() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
         DataStore dataStoreMock = Mockito.mock(DataStore.class);
         List<DataStore> imageStoresList = List.of(dataStoreMock);
 
         Mockito.when(dataStoreManagerMock.getImageStoresByZoneIds(Mockito.anyLong())).thenReturn(imageStoresList);
 
-        _adapter.getImageStoresThrowsExceptionIfNotFound(zoneId, templateProfileMock);
+        _adapter.getImageStoresThrowsExceptionIfNotFound(zoneId, templateProfileSharedMock);
     }
 
     @Test
