--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -151,6 +151,8 @@
     @Mock
     AgentManager agentMgr;
 
+    private final NicProfile nicProfileMock = mock(NicProfile.class);
+
     TungstenGuestNetworkGuru guru;
 
     AutoCloseable closeable;
@@ -242,7 +244,6 @@
     @Test
     public void testDeallocate() {
         final Network network = mock(Network.class);
-        final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         final NicVO nicVO = mock(NicVO.class);
 
@@ -251,7 +252,7 @@
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
         when(nicDao.listByVmId(anyLong())).thenReturn(List.of(nicVO));
 
-        guru.deallocate(network, nicProfile, virtualMachineProfile);
+        guru.deallocate(network, nicProfileMock, virtualMachineProfile);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVmCommand.class), anyLong());
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVmInterfaceCommand.class), anyLong());
     }
@@ -259,7 +260,6 @@
     @Test(expected = CloudRuntimeException.class)
     public void testDeallocateWithException() {
         final Network network = mock(Network.class);
-        final NicProfile nicProfile = mock(NicProfile.class);
         final VirtualMachineProfile virtualMachineProfile = mock(VirtualMachineProfile.class);
         final NicVO nicVO = mock(NicVO.class);
 
@@ -269,7 +269,7 @@
         when(network.getTrafficType()).thenReturn(Networks.TrafficType.Guest);
         when(virtualMachineProfile.getType()).thenReturn(VirtualMachine.Type.User);
 
-        guru.deallocate(network, nicProfile, virtualMachineProfile);
+        guru.deallocate(network, nicProfileMock, virtualMachineProfile);
     }
 
     @Test
@@ -398,7 +398,6 @@
 
     @Test
     public void testRelease() {
-        final NicProfile nic = mock(NicProfile.class);
         final VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         final IPAddressVO ipAddressVO = mock(IPAddressVO.class);
         final VMInstanceVO vmInstanceVO = mock(VMInstanceVO.class);
@@ -410,7 +409,7 @@
             new TungstenAnswer(new TungstenCommand(), true, ""));
         when(vmInstanceDao.findById(anyLong())).thenReturn(vmInstanceVO);
 
-        guru.release(nic, vm, "test");
+        guru.release(nicProfileMock, vm, "test");
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
@@ -471,7 +470,6 @@
 
     @Test
     public void testPrepareMigration() {
-        final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
         final VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         final DeployDestination dest = mock(DeployDestination.class);
@@ -486,14 +484,13 @@
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
             anyLong())).thenReturn(new TungstenAnswer(new TungstenCommand(), true, ""));
 
-        guru.prepareMigration(nic, network, vm, dest, context);
+        guru.prepareMigration(nicProfileMock, network, vm, dest, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(CreateTungstenVirtualMachineCommand.class),
             anyLong());
     }
 
     @Test
     public void testRollbackMigration() {
-        final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
@@ -504,14 +501,13 @@
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        guru.rollbackMigration(nic, network, vmProfile, context, context);
+        guru.rollbackMigration(nicProfileMock, network, vmProfile, context, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
 
     @Test
     public void testCommitMigration() {
-        final NicProfile nic = mock(NicProfile.class);
         final Network network = mock(Network.class);
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         final ReservationContext context = mock(ReservationContext.class);
@@ -522,7 +518,7 @@
         when(vmProfile.getVirtualMachine()).thenReturn(vm);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        guru.commitMigration(nic, network, vmProfile, context, context);
+        guru.commitMigration(nicProfileMock, network, vmProfile, context, context);
         verify(tungstenFabricUtils, times(1)).sendTungstenCommand(any(DeleteTungstenVRouterPortCommand.class),
             anyLong());
     }
