--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1179,11 +1179,10 @@
         List<StoragePoolVO> pools = new ArrayList<>(Arrays.asList(storagePoolVO));
         List<VolumeVO> volumes = new ArrayList<>(Arrays.asList(volumeVO));
 
-        StoragePoolTagVO poolTagMock1 = Mockito.mock(StoragePoolTagVO.class);
-        StoragePoolTagVO poolTagMock2 = Mockito.mock(StoragePoolTagVO.class);
+        StoragePoolTagVO poolTagMock1 = createMockStoragePoolTag();
+        StoragePoolTagVO poolTagMock2 = createMockStoragePoolTag();
         List<StoragePoolTagVO> poolTags = List.of(poolTagMock1, poolTagMock2);
         Mockito.doReturn("tag1").when(poolTagMock1).getTag();
-        Mockito.doReturn("tag2").when(poolTagMock2).getTag();
 
         Mockito.when(primaryDataStoreDao.listStoragePoolsWithActiveVolumesByOfferingId(anyLong())).thenReturn(pools);
         Mockito.when(storagePoolTagsDao.findStoragePoolTags(anyLong())).thenReturn(poolTags);
@@ -1353,6 +1352,12 @@
         });
     }
 
+    private static StoragePoolTagVO createMockStoragePoolTag() {
+        StoragePoolTagVO poolTagMock2 = Mockito.mock(StoragePoolTagVO.class);
+        Mockito.doReturn("tag2").when(poolTagMock2).getTag();
+        return poolTagMock2;
+    }
+
     @Test
     public void testCreateEdgeZone() {
         CreateZoneCmd cmd = Mockito.mock(CreateZoneCmd.class);
