--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -235,6 +235,7 @@
     EntityManager entityMgr;
     @Mock
     NetworkModel networkModel;
+    private Network publicNetworkMock;
     @Mock
     DomainDao domainDao;
     @Mock
@@ -281,6 +282,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        publicNetworkMock = mock(Network.class);
         tungstenService = new TungstenServiceImpl();
         tungstenService.projectDao = projectDao;
         tungstenService.tungstenProviderDao = tungstenProviderDao;
@@ -383,7 +385,6 @@
 
     @Test
     public void createManagementNetworkTest() {
-        Network managementNetwork = mock(Network.class);
         VirtualNetwork managementVirtualNetwork = mock(VirtualNetwork.class);
         VirtualNetwork fabricVirtualNetwork = mock(VirtualNetwork.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
@@ -395,7 +396,7 @@
         TungstenAnswer applyTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(),
-                eq(Networks.TrafficType.Management))).thenReturn(managementNetwork);
+                eq(Networks.TrafficType.Management))).thenReturn(publicNetworkMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkCommand.class), anyLong())).thenReturn(createTungstenNetworkAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(UpdateTungstenDefaultSecurityGroupCommand.class), anyLong())).thenReturn(updateTungstenDefaultSecurityGroupAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(UpdateTungstenVrouterConfigCommand.class), anyLong())).thenReturn(updateTungstenGlobalVrouterConfigAnswer);
@@ -434,7 +435,6 @@
 
     @Test
     public void deleteManagementNetworkTest() {
-        Network managementNetwork = mock(Network.class);
         TungstenAnswer deleteTungstenManagementPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer getTungstenFabricNetworkAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenFabricPolicyAnswer = MockTungstenAnswerFactory.get(true);
@@ -442,7 +442,7 @@
         VirtualNetwork fabricVirtualNetwork = mock(VirtualNetwork.class);
 
         when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(),
-                eq(Networks.TrafficType.Management))).thenReturn(managementNetwork);
+                eq(Networks.TrafficType.Management))).thenReturn(publicNetworkMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(GetTungstenFabricNetworkCommand.class), anyLong())).thenReturn(getTungstenFabricNetworkAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(deleteTungstenFabricPolicyAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkCommand.class), anyLong())).thenReturn(deleteTungstenNetworkAnswer);
@@ -473,7 +473,6 @@
 
     @Test
     public void createPublicNetworkTest() {
-        Network publicNetwork = mock(Network.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         NetworkDetailVO networkDetailVO = mock(NetworkDetailVO.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
@@ -482,7 +481,7 @@
         TungstenAnswer createTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer applyTungstenPolicyAnswer = MockTungstenAnswerFactory.get(true);
 
-        when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(publicNetwork);
+        when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(publicNetworkMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkCommand.class), anyLong())).thenReturn(createPublicNetworkAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenFloatingIpPoolCommand.class), anyLong())).thenReturn(createFloatingIpPoolAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(createTungstenPolicyAnswer);
@@ -525,12 +524,11 @@
 
     @Test
     public void deletePublicNetworkTest() {
-        Network publicNetwork = mock(Network.class);
         TungstenAnswer deleteTungstenNetworkPolicyAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenFloatingIpPoolAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenNetworkAnswer = MockTungstenAnswerFactory.get(true);
 
-        when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(publicNetwork);
+        when(networkModel.getSystemNetworkByZoneAndTrafficType(anyLong(), eq(Networks.TrafficType.Public))).thenReturn(publicNetworkMock);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkPolicyCommand.class), anyLong())).thenReturn(deleteTungstenNetworkPolicyAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenFloatingIpPoolCommand.class), anyLong())).thenReturn(deleteTungstenFloatingIpPoolAnswer);
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenNetworkCommand.class), anyLong())).thenReturn(deleteTungstenNetworkAnswer);
@@ -639,7 +637,6 @@
 
     @Test
     public void updateLoadBalancerTest() {
-        Network network = mock(Network.class);
         LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
         Network publicNetwork = mock(Network.class);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
@@ -677,7 +674,7 @@
         when(ipAddressVO.getAddress()).thenReturn(ip);
         when(tungstenGuestNetworkIpAddressVO.getGuestIpAddress()).thenReturn(ip);
 
-        assertTrue(tungstenService.updateLoadBalancer(network, loadBalancingRule));
+        assertTrue(tungstenService.updateLoadBalancer(publicNetworkMock, loadBalancingRule));
     }
 
     @Test
