--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -265,6 +265,7 @@
     private ResourceManager _resourceMgr;
     @Mock
     private DeploymentPlanningManager _dpMgr;
+    private DeploymentPlanner deploymentPlanner;
     @Mock
     private HypervisorGuruManager _hvGuruMgr;
     @Mock
@@ -301,6 +302,7 @@
 
     @Before
     public void setup() {
+        deploymentPlanner = mock(DeploymentPlanner.class);
         ReflectionTestUtils.getField(VirtualMachineManager.VmMetadataManufacturer, "s_depot");
         virtualMachineManagerImpl.setHostAllocators(new ArrayList<>());
 
@@ -1141,7 +1143,6 @@
 
         Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
 
-        DeploymentPlanner planner = mock(DeploymentPlanner.class);
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
@@ -1193,7 +1194,7 @@
             when(CallContext.current()).thenReturn(callContext);
 
             try {
-                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, planner);
+                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, deploymentPlanner);
             } catch (CloudRuntimeException e) {
                 assertEquals(e.getMessage(), "Error while transitioning");
             }
@@ -1237,7 +1238,6 @@
 
         Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
 
-        DeploymentPlanner planner = mock(DeploymentPlanner.class);
 
         when(vmInstanceDaoMock.findByUuid("vm-uuid")).thenReturn(vmInstance);
 
@@ -1289,7 +1289,7 @@
             when(CallContext.current()).thenReturn(callContext);
 
             try {
-                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, planner);
+                virtualMachineManagerImpl.orchestrateStart("vm-uuid", params, plan, deploymentPlanner);
             } catch (CloudRuntimeException e) {
                 assertEquals(e.getMessage(), "Error while transitioning");
             }
