--- a/server/src/test/java/com/cloud/network/element/VirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VirtualRouterElementTest.java
@@ -190,6 +190,7 @@
     @Mock private UserVmManager _userVmMgr;
     @Mock private VirtualMachineManager _itMgr;
     @Mock private NetworkTopologyContext networkTopologyContext;
+    @Mock private DataCenterVO dataCenter;
 
     @InjectMocks
     private RouterDeploymentDefinitionBuilder routerDeploymentDefinitionBuilder;
@@ -542,10 +543,9 @@
         when(vpc.getId()).thenReturn(vpcId);
         when(vpc.getZoneId()).thenReturn(zoneId);
         when(_routerDao.listByVpcId(vpcId)).thenReturn(Arrays.asList(router));
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
-        when(_dcDao.findById(zoneId)).thenReturn(dc);
+        when(_dcDao.findById(zoneId)).thenReturn(dataCenter);
         NetworkTopology networkTopology = Mockito.mock(NetworkTopology.class);
-        when(networkTopologyContext.retrieveNetworkTopology(dc)).thenReturn(networkTopology);
+        when(networkTopologyContext.retrieveNetworkTopology(dataCenter)).thenReturn(networkTopology);
         doReturn(true).when(networkTopology).applyBgpPeers(any(), any(), any());
 
         boolean result = virtualRouterElement.applyBgpPeers(vpc, null, bgpPeers);
@@ -566,10 +566,9 @@
         when(network.getId()).thenReturn(networkId);
         when(network.getDataCenterId()).thenReturn(zoneId);
         when(_routerDao.listByNetworkAndRole(networkId, VirtualRouter.Role.VIRTUAL_ROUTER)).thenReturn(Arrays.asList(router));
-        DataCenterVO dc = Mockito.mock(DataCenterVO.class);
-        when(_dcDao.findById(zoneId)).thenReturn(dc);
+        when(_dcDao.findById(zoneId)).thenReturn(dataCenter);
         NetworkTopology networkTopology = Mockito.mock(NetworkTopology.class);
-        when(networkTopologyContext.retrieveNetworkTopology(dc)).thenReturn(networkTopology);
+        when(networkTopologyContext.retrieveNetworkTopology(dataCenter)).thenReturn(networkTopology);
         doReturn(true).when(networkTopology).applyBgpPeers(any(), any(), any());
 
         boolean result = virtualRouterElement.applyBgpPeers(null, network, bgpPeers);
