--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -454,6 +454,14 @@
         Mockito.when(ActionEventUtils.onCompletedActionEvent(Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.anyLong(), Mockito.anyString(), Mockito.anyLong())).thenReturn(1L);
     }
 
+    private static NicVO createMockNic() {
+        NicVO nic = Mockito.mock(NicVO.class);
+        Mockito.when(nic.getIPv6Address()).thenReturn(null);
+        Mockito.when(nic.getBroadcastUri()).thenReturn(URI.create("vlan"));
+        Mockito.when(nic.getId()).thenReturn(100L);
+        return nic;
+    }
+
     @Test
     @DB
     public void testNewAssignPublicIpv6ToNetwork() {
@@ -476,8 +484,7 @@
 
     @Test
     public void testFromPlaceholderAssignPublicIpv6ToNetwork() {
-        NicVO nic = Mockito.mock(NicVO.class);
-        Mockito.when(nic.getIPv6Address()).thenReturn(null);
+        NicVO nic = createMockNic();
         Mockito.when(nic.getBroadcastUri()).thenReturn(URI.create(vlan));
         Mockito.when(nic.getId()).thenReturn(nicId);
         Network network = Mockito.mock(Network.class);
