--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -118,6 +118,9 @@
     private ServiceOfferingDao serviceOfferingDao;
 
     @Mock
+    private ServiceOffering serviceOfferingMock;
+
+    @Mock
     private VMInstanceDao vmInstanceDao;
 
     @Mock
@@ -817,9 +820,8 @@
 
         Map<Long, ServiceOffering> vmIdServiceOfferingMap = new HashMap<>();
 
-        ServiceOffering serviceOffering = Mockito.mock(ServiceOffering.class);
         for (VirtualMachine vm : vmList) {
-            vmIdServiceOfferingMap.put(vm.getId(), serviceOffering);
+            vmIdServiceOfferingMap.put(vm.getId(), serviceOfferingMock);
         }
 
         // Create caches for the new method signature
@@ -890,9 +892,8 @@
 
         Map<Long, ServiceOffering> vmIdServiceOfferingMap = new HashMap<>();
 
-        ServiceOffering serviceOffering = Mockito.mock(ServiceOffering.class);
         for (VirtualMachine vm : vmList) {
-            vmIdServiceOfferingMap.put(vm.getId(), serviceOffering);
+            vmIdServiceOfferingMap.put(vm.getId(), serviceOfferingMock);
         }
 
         // Create caches for the new method signature
