--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -97,6 +97,7 @@
     HostDao hostDao;
     @Mock
     ConfigurationDao configDao;
+    private DataTO sharedDestDataTO;
 
     static MockedStatic<RemoteHostEndPoint> remoteHostEndPointMock;
 
@@ -115,6 +116,7 @@
     @Before
     public void initMocks() {
         closeable = MockitoAnnotations.openMocks(this);
+        sharedDestDataTO = Mockito.mock(DataTO.class);
     }
 
     @After
@@ -518,8 +520,7 @@
         DataTO srcDataTO = Mockito.mock(DataTO.class);
         when(srcData.getTO()).thenReturn(srcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
-        DataTO destDataTO = Mockito.mock(DataTO.class);
-        when(destData.getTO()).thenReturn(destDataTO);
+        when(destData.getTO()).thenReturn(sharedDestDataTO);
         Host destHost = Mockito.mock(Host.class);
 
         doReturn(false).when(scaleIOPrimaryDataStoreDriver).anyVolumeRequiresEncryption(srcData, destData);
@@ -541,8 +542,7 @@
         DataTO srcDataTO = Mockito.mock(DataTO.class);
         when(srcData.getTO()).thenReturn(srcDataTO);
         DataObject destData = Mockito.mock(DataObject.class);
-        DataTO destDataTO = Mockito.mock(DataTO.class);
-        when(destData.getTO()).thenReturn(destDataTO);
+        when(destData.getTO()).thenReturn(sharedDestDataTO);
         Host destHost = Mockito.mock(Host.class);
 
         doReturn(false).when(scaleIOPrimaryDataStoreDriver).anyVolumeRequiresEncryption(srcData, destData);
