--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
@@ -158,8 +158,7 @@
 
         DeployDestination dest = mock(DeployDestination.class);
 
-        Domain dom = mock(Domain.class);
-        when(dom.getName()).thenReturn("d1");
+        Domain dom = MockDomain.createMockDomain("d1");
         Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("a1");
         ReservationContext context = mock(ReservationContext.class);
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/NiciraNvpElementTest.java
@@ -178,8 +178,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final Domain dom = mock(Domain.class);
-        when(dom.getName()).thenReturn("domain");
+        final Domain dom = MockDomain.createMockDomain("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
         final ReservationContext context = mock(ReservationContext.class);
@@ -294,8 +293,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final Domain dom = mock(Domain.class);
-        when(dom.getName()).thenReturn("domain");
+        final Domain dom = MockDomain.createMockDomain("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
         final ReservationContext context = mock(ReservationContext.class);
@@ -347,8 +345,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final Domain dom = mock(Domain.class);
-        when(dom.getName()).thenReturn("domain");
+        final Domain dom = MockDomain.createMockDomain("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
         final ReservationContext context = mock(ReservationContext.class);
@@ -399,8 +396,7 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final Domain dom = mock(Domain.class);
-        when(dom.getName()).thenReturn("domain");
+        final Domain dom = MockDomain.createMockDomain("domain");
         final Account acc = mock(Account.class);
         when(acc.getAccountName()).thenReturn("accountname");
         final ReservationContext context = mock(ReservationContext.class);
--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/MockDomain.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/MockDomain.java
@@ -0,0 +1,14 @@
+package com.cloud.network.element;
+
+import com.cloud.domain.Domain;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockDomain {
+    public static Domain createMockDomain(String name) {
+        Domain domain = Mockito.mock(Domain.class);
+        when(domain.getName()).thenReturn(name);
+        return domain;
+    }
+}
--- a/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/MockDomain.java
+++ b/plugins/network-elements/nicira-nvp/src/test/java/com/cloud/network/element/MockDomain.java
@@ -0,0 +1,14 @@
+package com.cloud.network.element;
+
+import com.cloud.domain.Domain;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockDomain {
+    public static Domain createMockDomain(String name) {
+        Domain domain = Mockito.mock(Domain.class);
+        when(domain.getName()).thenReturn(name);
+        return domain;
+    }
+}
