--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -519,12 +519,9 @@
 
     @Test
     public void testCanVmRestartOnAnotherServerAllShared() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class);
-        Mockito.when(vol1.getPoolId()).thenReturn(10L);
+        VolumeVO vol1 = createMockVolumeVO();
+        VolumeVO vol2 = createMockVolumeVO();
         Mockito.when(vol2.getPoolId()).thenReturn(20L);
-        Mockito.when(vol1.isRecreatable()).thenReturn(false);
-        Mockito.when(vol2.isRecreatable()).thenReturn(false);
 
 
         StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
@@ -542,12 +539,9 @@
 
     @Test
     public void testCanVmRestartOnAnotherServerOneLocalNotRecreatable() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class); // Local, not recreatable
-        Mockito.when(vol1.getPoolId()).thenReturn(10L);
+        VolumeVO vol1 = createMockVolumeVO();
+        VolumeVO vol2 = createMockVolumeVO(); // Local, not recreatable
         Mockito.when(vol2.getPoolId()).thenReturn(30L);
-        Mockito.when(vol1.isRecreatable()).thenReturn(false);
-        Mockito.when(vol2.isRecreatable()).thenReturn(false); // Not recreatable
 
         StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO pool2 = Mockito.mock(StoragePoolVO.class);
@@ -564,11 +558,9 @@
 
     @Test
     public void testCanVmRestartOnAnotherServerOneLocalRecreatable() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class); // Local, but recreatable
-        Mockito.when(vol1.getPoolId()).thenReturn(10L);
+        VolumeVO vol1 = createMockVolumeVO();
+        VolumeVO vol2 = createMockVolumeVO();
         Mockito.when(vol2.getPoolId()).thenReturn(30L);
-        Mockito.when(vol1.isRecreatable()).thenReturn(false);
         Mockito.when(vol2.isRecreatable()).thenReturn(true); // Recreatable
 
         StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
@@ -945,4 +937,11 @@
         return null;
     }
 
+    private static VolumeVO createMockVolumeVO() {
+        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
+        Mockito.when(vol1.getPoolId()).thenReturn(10L);
+        Mockito.when(vol1.isRecreatable()).thenReturn(false);
+        return vol1;
+    }
+
 }
