--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -61,6 +61,9 @@
 
     @Mock
     MigrateVolumeCommand command;
+
+    @Mock
+    private MigrateVolumeAnswer regularVolumeAnswerMock;
 
     @Mock
     LibvirtComputingResource libvirtComputingResource;
@@ -229,7 +232,6 @@
         Mockito.doReturn(Storage.StoragePoolType.PowerFlex).when(srcPrimaryDataStore).getPoolType();
 
         MigrateVolumeAnswer powerFlexAnswer = Mockito.mock(MigrateVolumeAnswer.class);
-        MigrateVolumeAnswer regularVolumeAnswer = Mockito.mock(MigrateVolumeAnswer.class);
 
         Mockito.doReturn(true).when(powerFlexAnswer).getResult();
         Mockito.doReturn(powerFlexAnswer).when(libvirtMigrateVolumeCommandWrapper).migratePowerFlexVolume(command, libvirtComputingResource);
@@ -248,7 +250,6 @@
         Mockito.doReturn(srcPrimaryDataStore).when(srcVolumeObjectTO).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.NetworkFilesystem).when(srcPrimaryDataStore).getPoolType();
 
-        MigrateVolumeAnswer powerFlexAnswer = Mockito.mock(MigrateVolumeAnswer.class);
         MigrateVolumeAnswer regularVolumeAnswer = Mockito.mock(MigrateVolumeAnswer.class);
 
         Mockito.doReturn(false).when(regularVolumeAnswer).getResult();
