--- a/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
@@ -84,6 +84,7 @@
     AccountDao accountDao;
     @Mock
     SnapshotDao snapshotDao;
+    private SnapshotVO mockSnapshotVO;
     @Mock
     AccountManager accountManager;
     @Mock
@@ -115,6 +116,7 @@
 
     @Before
     public void setUp() {
+        mockSnapshotVO = Mockito.mock(SnapshotVO.class);
         snapshotManager._snapshotPolicyDao = snapshotPolicyDao;
         snapshotManager._volsDao = volumeDao;
         snapshotManager._accountMgr = accountManager;
@@ -345,14 +347,12 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testGetCheckedSnapshotForCopyNoSnapshot() {
-        SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
-        snapshotManager.getCheckedSnapshotForCopy(snapshotVO, List.of(100L), null, false);
+        snapshotManager.getCheckedSnapshotForCopy(mockSnapshotVO, List.of(100L), null, false);
     }
 
     @Test(expected = InvalidParameterValueException.class)
     public void testGetCheckedSnapshotForCopyNoSnapshotBackup() {
-        SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
-        snapshotManager.getCheckedSnapshotForCopy(snapshotVO, List.of(100L), null, false);
+        snapshotManager.getCheckedSnapshotForCopy(mockSnapshotVO, List.of(100L), null, false);
     }
 
     @Test(expected = InvalidParameterValueException.class)
