--- a/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/storage/dao/VMTemplateDaoImplTest.java
@@ -37,6 +37,7 @@
 import java.util.Map;
 
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.InjectMocks;
@@ -70,6 +71,13 @@
     @Spy
     @InjectMocks
     VMTemplateDaoImpl templateDao = new VMTemplateDaoImpl();
+
+    private SearchCriteria<VMTemplateVO> sc;
+
+    @Before
+    public void setUp() {
+        sc = mock(SearchCriteria.class);
+    }
 
     @Test
     public void testFindLatestTemplateByName_ReturnsTemplate() {
@@ -346,7 +354,6 @@
         VMTemplateVO expectedTemplate = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(expectedTemplate);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         List<VMTemplateVO> templates = Collections.singletonList(expectedTemplate);
@@ -364,7 +371,6 @@
         VMTemplateVO template = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(template);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         doReturn(Collections.emptyList()).when(templateDao).listBy(any(SearchCriteria.class), any(Filter.class));
@@ -380,7 +386,6 @@
         VMTemplateVO expectedTemplate = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(expectedTemplate);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         List<VMTemplateVO> templates = Collections.singletonList(expectedTemplate);
@@ -398,7 +403,6 @@
         VMTemplateVO expectedTemplate = mock(VMTemplateVO.class);
         SearchBuilder<VMTemplateVO> sb = mock(SearchBuilder.class);
         when(sb.entity()).thenReturn(expectedTemplate);
-        SearchCriteria<VMTemplateVO>sc = mock(SearchCriteria.class);
         when(sb.create()).thenReturn(sc);
         when(templateDao.createSearchBuilder()).thenReturn(sb);
         List<VMTemplateVO> templates = Collections.singletonList(expectedTemplate);
