--- a/engine/schema/src/test/java/com/cloud/host/HostVOTest.java
+++ b/engine/schema/src/test/java/com/cloud/host/HostVOTest.java
@@ -113,8 +113,8 @@
 
         host.setHostTags(Arrays.asList("tag1", "tag2", "tag3"), false);
         offering.setHostTag("tag2,tag1");
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        Mockito.when(template.getTemplateTag()).thenReturn("tag3");
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate("tag3");
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate("tag2");
         assertTrue(host.checkHostServiceOfferingAndTemplateTags(offering, template, Set.of("tag2", "tag3")));
         actualMissingTags = host.getHostServiceOfferingAndTemplateMissingTags(offering, template, Set.of("tag2", "tag3"));
         assertTrue(actualMissingTags.isEmpty());
@@ -133,17 +133,21 @@
     public void testWrongOfferingTag() {
         host.setHostTags(Arrays.asList("tag1", "tag2"), false);
         offering.setHostTag("tag2,tag4");
-        VirtualMachineTemplate template = Mockito.mock(VirtualMachineTemplate.class);
-        Mockito.when(template.getTemplateTag()).thenReturn("tag1");
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate("tag1");
         assertFalse(host.checkHostServiceOfferingAndTemplateTags(offering, template, Set.of("tag1", "tag2", "tag3", "tag4")));
         Set<String> actualMissingTags = host.getHostServiceOfferingAndTemplateMissingTags(offering, template, Set.of("tag1", "tag2", "tag3", "tag4"));
         assertEquals(Set.of("tag4"), actualMissingTags);
 
         offering.setHostTag("tag1,tag2");
-        template = Mockito.mock(VirtualMachineTemplate.class);
-        Mockito.when(template.getTemplateTag()).thenReturn("tag3");
+        template = createMockVirtualMachineTemplate("tag3");
         actualMissingTags = host.getHostServiceOfferingAndTemplateMissingTags(offering, template, Set.of("tag1", "tag2", "tag3", "tag4"));
         assertFalse(host.checkHostServiceOfferingAndTemplateTags(offering, template, Set.of("tag1", "tag2", "tag3", "tag4")));
         assertEquals(Set.of("tag3"), actualMissingTags);
     }
+
+    private static VirtualMachineTemplate createMockVirtualMachineTemplate(String templateTag) {
+        
+        Mockito.when(template.getTemplateTag()).thenReturn(templateTag);
+        return template;
+    }
 }
