--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -965,8 +965,7 @@
         SearchCriteria<AutoScalePolicyVO> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
         when(asScaleUpPolicyMock.getDuration()).thenReturn(scaleUpPolicyDuration);
         when(asScaleUpPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEUP);
-        when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
-        when(asScaleDownPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEDOWN);
+        AutoScalePolicyVO asScaleDownPolicyMock = createMockAutoScaleDownPolicy();
 
         Mockito.doReturn(searchBuilderMock).when(asPolicyDao).createSearchBuilder();
         when(searchBuilderMock.entity()).thenReturn(asScaleUpPolicyMock);
@@ -1052,8 +1051,7 @@
         SearchCriteria<AutoScalePolicyVO> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
         when(asScaleUpPolicyMock.getDuration()).thenReturn(scaleUpPolicyDuration);
         when(asScaleUpPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEUP);
-        when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
-        when(asScaleDownPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEDOWN);
+        AutoScalePolicyVO asScaleDownPolicyMock = createMockAutoScaleDownPolicy();
 
         Mockito.doReturn(searchBuilderMock).when(asPolicyDao).createSearchBuilder();
         when(searchBuilderMock.entity()).thenReturn(asScaleUpPolicyMock);
@@ -2514,6 +2512,13 @@
         Assert.assertEquals(result.first(), result.second());
     }
 
+    private static AutoScalePolicyVO createMockAutoScaleDownPolicy() {
+        AutoScalePolicyVO asScaleDownPolicyMock = Mockito.mock(AutoScalePolicyVO.class);
+        when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
+        when(asScaleDownPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEDOWN);
+        return asScaleDownPolicyMock;
+    }
+
     private void runGetNextVmHostAndDisplayNameGeneratesCorrectHostAndDisplayNameForWindowsTemplate() {
         when(asVmGroupMock.getName()).thenReturn(vmGroupName);
         when(asVmGroupMock.getNextVmSeq()).thenReturn(1L);
