--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -180,6 +180,8 @@
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
 
+    private VirtualMachine virtualMachineMock;
+
     TungstenElement tungstenElement;
 
     MockedStatic<ApiDBUtils> apiDBUtilsMocked;
@@ -191,6 +193,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        virtualMachineMock = mock(VirtualMachine.class);
         tungstenElement = new TungstenElement();
         tungstenElement.tungstenFabricUtils = tungstenFabricUtils;
         tungstenElement.networkModel = networkModel;
@@ -934,11 +937,10 @@
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         ReservationContext src = mock(ReservationContext.class);
         ReservationContext dest = mock(ReservationContext.class);
-        VirtualMachine virtualMachine = mock(VirtualMachine.class);
         HostVO hostVO = mock(HostVO.class);
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
-        when(vm.getVirtualMachine()).thenReturn(virtualMachine);
+        when(vm.getVirtualMachine()).thenReturn(virtualMachineMock);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
         tungstenElement.rollbackMigration(nic, network, vm, src, dest);
@@ -952,11 +954,10 @@
         VirtualMachineProfile vm = mock(VirtualMachineProfile.class);
         ReservationContext src = mock(ReservationContext.class);
         ReservationContext dest = mock(ReservationContext.class);
-        VirtualMachine virtualMachine = mock(VirtualMachine.class);
         HostVO hostVO = mock(HostVO.class);
 
         when(vm.getType()).thenReturn(VirtualMachine.Type.ConsoleProxy);
-        when(vm.getVirtualMachine()).thenReturn(virtualMachine);
+        when(vm.getVirtualMachine()).thenReturn(virtualMachineMock);
         when(hostDao.findById(anyLong())).thenReturn(hostVO);
 
         tungstenElement.commitMigration(nic, network, vm, src, dest);
