--- a/agent/src/test/java/com/cloud/agent/AgentTest.java
+++ b/agent/src/test/java/com/cloud/agent/AgentTest.java
@@ -53,6 +53,7 @@
     private AgentShell shell;
     private ServerResource serverResource;
     private Logger logger;
+    private Link linkMock;
 
     @Before
     public void setUp() throws ConfigurationException {
@@ -63,6 +64,7 @@
         doReturn(1).when(shell).getPingRetries();
         agent = new Agent(shell, 1, serverResource);
         logger = mock(Logger.class);
+        linkMock = mock(Link.class);
         ReflectionTestUtils.setField(agent, "logger", logger);
     }
 
@@ -150,9 +152,8 @@
 
     @Test
     public void testSetLinkAssignsLink() {
-        Link mockLink = mock(Link.class);
-        agent.setLink(mockLink);
-        assertEquals(mockLink, agent.link);
+        agent.setLink(linkMock);
+        assertEquals(linkMock, agent.link);
     }
 
     @Test
@@ -210,10 +211,9 @@
 
     @Test
     public void testCloseAndTerminateLinkValidLinkCallsCloseAndTerminate() {
-        Link mockLink = mock(Link.class);
-        agent.closeAndTerminateLink(mockLink);
-        verify(mockLink).close();
-        verify(mockLink).terminated();
+        agent.closeAndTerminateLink(linkMock);
+        verify(linkMock).close();
+        verify(linkMock).terminated();
     }
 
     @Test
