--- a/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
+++ b/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
@@ -225,6 +225,12 @@
         return tariffVO;
     }
 
+    private static AccountVO createMockAccount() {
+        AccountVO account = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(1L).when(account).getId();
+        return account;
+    }
+
     @Test
     public void testQuotaResponse() {
         QuotaTariffVO tariffVO = makeTariffTestData();
@@ -1238,9 +1244,8 @@
         Long accountId = 1L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
-
-        Mockito.doReturn(accountId).when(account).getId();
+        AccountVO account = createMockAccount();
+
         Mockito.when(cmd.getQuotaEnforce()).thenReturn(Boolean.TRUE);
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
 
@@ -1255,9 +1260,9 @@
         Long accountId = 1L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
 
         Mockito.when(cmd.getMinBalance()).thenReturn(50.0);
+        AccountVO account = createMockAccount();
         Mockito.doReturn(accountId).when(account).getId();
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
 
@@ -1273,10 +1278,9 @@
         Long domainId = 2L;
         Double value = 100.0;
         Date depositedOn = new Date();
-        AccountVO account = Mockito.mock(AccountVO.class);
+        AccountVO account = createMockAccount();
         BigDecimal currentBalance = BigDecimal.ZERO;
 
-        Mockito.doReturn(accountId).when(account).getId();
         Mockito.doReturn(domainId).when(account).getDomainId();
         Mockito.when(cmd.getMinBalance()).thenReturn(50.0);
         Mockito.when(quotaCreditsDaoMock.saveCredits(Mockito.any(QuotaCreditsVO.class))).thenAnswer(invocation -> invocation.getArgument(0));
