--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenElementTest.java
@@ -180,6 +180,8 @@
     @Mock
     LoadBalancerCertMapDao loadBalancerCertMapDao;
 
+    private TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVOMock;
+
     TungstenElement tungstenElement;
 
     MockedStatic<ApiDBUtils> apiDBUtilsMocked;
@@ -191,6 +193,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        tungstenGuestNetworkIpAddressVOMock = mock(TungstenGuestNetworkIpAddressVO.class);
         tungstenElement = new TungstenElement();
         tungstenElement.tungstenFabricUtils = tungstenFabricUtils;
         tungstenElement.networkModel = networkModel;
@@ -406,7 +409,6 @@
         List<LoadBalancingRule.LbStickinessPolicy> lbStickinessPolicyList = List.of(lbStickinessPolicy);
         List<LoadBalancerVO> loadBalancerVOList = List.of(loadBalancerVO);
         IPAddressVO ipAddressVO = mock(IPAddressVO.class);
-        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer createTungstenNetworkLoadbalancerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateTungstenLoadBalancerPoolAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer updateTungstenLoadBalancerMemberAnswer = MockTungstenAnswerFactory.get(true);
@@ -449,7 +451,6 @@
         LoadBalancerVO loadBalancerVO1 = mock(LoadBalancerVO.class);
         List<LoadBalancingRule> loadBalancingRuleList1 = List.of(loadBalancingRule1);
         List<LoadBalancerVO> loadBalancerVOList1 = List.of(loadBalancerVO1);
-        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer deleteTungstenLoadBalancerListenerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenLoadBalancerCommand = MockTungstenAnswerFactory.get(true);
 
@@ -461,7 +462,7 @@
         when(ip1.addr()).thenReturn("10.10.10.10");
         when(tungstenFabricUtils.sendTungstenCommand(any(DeleteTungstenLoadBalancerCommand.class), anyLong())).thenReturn(deleteTungstenLoadBalancerCommand);
         when(lbDao.listByIpAddress(anyLong())).thenReturn(loadBalancerVOList1);
-        when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(),anyString())).thenReturn(tungstenGuestNetworkIpAddressVO);
+        when(tungstenGuestNetworkIpAddressDao.findByNetworkIdAndPublicIp(anyLong(),anyString())).thenReturn(tungstenGuestNetworkIpAddressVOMock);
         when(tungstenGuestNetworkIpAddressDao.remove(anyLong())).thenReturn(false);
 
         assertFalse(tungstenElement.applyLBRules(network, loadBalancingRuleList1));
@@ -476,7 +477,6 @@
         LoadBalancingRule loadBalancingRule = mock(LoadBalancingRule.class);
         LoadBalancerVO loadBalancerVO1 = mock(LoadBalancerVO.class);
         LoadBalancerVO loadBalancerVO2 = mock(LoadBalancerVO.class);
-        TungstenGuestNetworkIpAddressVO tungstenGuestNetworkIpAddressVO = mock(TungstenGuestNetworkIpAddressVO.class);
         TungstenAnswer deleteTungstenLoadBalancerListenerAnswer = MockTungstenAnswerFactory.get(true);
         TungstenAnswer deleteTungstenLoadBalancerCommand = MockTungstenAnswerFactory.get(true);
         List<LoadBalancingRule> loadBalancingRuleList = List.of(loadBalancingRule);
