--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/KvmFileBasedStorageVmSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/KvmFileBasedStorageVmSnapshotStrategyTest.java
@@ -215,8 +215,6 @@
         UserVmVO userVm = mock(UserVmVO.class);
         VolumeVO rootVolume = mock(VolumeVO.class);
         StoragePoolVO storagePool = mock(StoragePoolVO.class);
-        HostVO defaultHost = mock(HostVO.class);
-        HostVO capableHost = mock(HostVO.class);
 
         when(userVm.getId()).thenReturn(vmId);
         when(userVm.getUuid()).thenReturn("vm-uuid");
@@ -231,8 +229,8 @@
         when(storagePool.getClusterId()).thenReturn(clusterId);
         when(storagePool.getPodId()).thenReturn(podId);
         when(storagePool.getDataCenterId()).thenReturn(dataCenterId);
-        when(defaultHost.getId()).thenReturn(defaultHostId);
-        when(capableHost.getId()).thenReturn(capableHostId);
+        HostVO defaultHost = createMockCapableHost(defaultHostId);
+        HostVO capableHost = createMockCapableHost(capableHostId);
         when(hostDao.listAllUpAndEnabledNonHAHosts(Host.Type.Routing, clusterId, podId, dataCenterId, null))
                 .thenReturn(List.of(defaultHost, capableHost));
         when(hostDetailsDao.findDetail(defaultHostId, Host.HOST_UEFI_ENABLE)).thenReturn(null);
@@ -547,4 +545,10 @@
         when(host.getId()).thenReturn(hostId);
         when(hostDao.listAllUpAndEnabledNonHAHosts(Host.Type.Routing, clusterId, podId, dataCenterId, null)).thenReturn(List.of(host));
     }
+
+    private static HostVO createMockCapableHost(long capableHostId) {
+        HostVO capableHost = mock(HostVO.class);
+        when(capableHost.getId()).thenReturn(capableHostId);
+        return capableHost;
+    }
 }
