--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenGuestNetworkGuruTest.java
@@ -151,6 +151,8 @@
     @Mock
     AgentManager agentMgr;
 
+    private final VirtualMachine virtualMachine = mock(VirtualMachine.class);
+
     TungstenGuestNetworkGuru guru;
 
     AutoCloseable closeable;
