--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtConvertInstanceCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtConvertInstanceCommandWrapperTest.java
@@ -55,6 +55,8 @@
     @Mock
     private LibvirtComputingResource libvirtComputingResourceMock;
 
+    private Script scriptField;
+
     @Mock
     private KVMStoragePool temporaryPool;
     @Mock
@@ -71,6 +73,7 @@
 
     @Before
     public void setUp() {
+        scriptField = Mockito.mock(Script.class);
         Mockito.when(secondaryDataStore.getUrl()).thenReturn(secondaryPoolUrl);
         Mockito.when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolManager);
         Mockito.when(storagePoolManager.getStoragePoolByURI(secondaryPoolUrl)).thenReturn(temporaryPool);
@@ -175,20 +178,18 @@
 
     @Test
     public void testAddExtraParamsToScriptSameKeysAndValues() {
-        Script script = Mockito.mock(Script.class);
         String extraParams = "--mac 00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254";
-        convertInstanceCommandWrapper.addExtraParamsToScript(extraParams, script);
-        Mockito.verify(script).add("--mac", "00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254");
+        convertInstanceCommandWrapper.addExtraParamsToScript(extraParams, scriptField);
+        Mockito.verify(scriptField).add("--mac", "00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254");
     }
 
     @Test
     public void testAddExtraParamsToScriptDifferentArgs() {
-        Script script = Mockito.mock(Script.class);
         String extraParams = "--mac 00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254 -x -v";
-        convertInstanceCommandWrapper.addExtraParamsToScript(extraParams, script);
-        Mockito.verify(script).add("--mac", "00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254");
-        Mockito.verify(script).add("-x");
-        Mockito.verify(script).add("-v");
+        convertInstanceCommandWrapper.addExtraParamsToScript(extraParams, scriptField);
+        Mockito.verify(scriptField).add("--mac", "00:0c:29:e6:3d:9d:ip:192.168.0.89,192.168.0.1,24,192.168.0.254");
+        Mockito.verify(scriptField).add("-x");
+        Mockito.verify(scriptField).add("-v");
     }
 
     @Test
