--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
@@ -338,8 +338,7 @@
         when(hostVO.getId()).thenReturn(1L);
         when(_hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        VlanVO vlanVO = mock(VlanVO.class);
-        when(vlanVO.getVlanTag()).thenReturn(null);
+        VlanVO vlanVO = createMockVlanVO();
         when(_vlanDao.findById(anyLong())).thenReturn(vlanVO);
 
         PortForwardingRule rule = mock(PortForwardingRule.class);
@@ -388,8 +387,7 @@
         when(hostVO.getId()).thenReturn(1L);
         when(_hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        VlanVO vlanVO = mock(VlanVO.class);
-        when(vlanVO.getVlanTag()).thenReturn(null);
+        VlanVO vlanVO = createMockVlanVO();
         when(_vlanDao.findById(anyLong())).thenReturn(vlanVO);
 
         StaticNat rule = mock(StaticNat.class);
@@ -406,4 +404,10 @@
 
         assertTrue(_element.applyStaticNats(network, rules));
     }
+
+    private static VlanVO createMockVlanVO() {
+        VlanVO vlanVO = mock(VlanVO.class);
+        when(vlanVO.getVlanTag()).thenReturn(null);
+        return vlanVO;
+    }
 }
