--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -844,9 +844,8 @@
     public void validateConditionsToReplaceDiskOfferingOfVolumeTestRootVolume() {
         Mockito.lenient().when(volumeVoMock.getVolumeType()).thenReturn(Type.ROOT);
         Mockito.doReturn(vmInstanceMockId).when(volumeVoMock).getInstanceId();
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
+        UserVmVO vm = createMockUserVmWithServiceOfferingId();
         when(_vmInstanceDao.findById(anyLong())).thenReturn(vm);
-        when(vm.getServiceOfferingId()).thenReturn(1L);
         ServiceOfferingVO serviceOfferingVO = Mockito.mock(ServiceOfferingVO.class);
         serviceOfferingVO.setDiskOfferingStrictness(false);
         when(serviceOfferingDao.findById(anyLong())).thenReturn(serviceOfferingVO);
@@ -858,9 +857,8 @@
     public void validateConditionsToReplaceDiskOfferingOfVolumeTestRootVolumeWithDiskOfferingStrictnessTrue() {
         Mockito.lenient().when(volumeVoMock.getVolumeType()).thenReturn(Type.ROOT);
         Mockito.doReturn(vmInstanceMockId).when(volumeVoMock).getInstanceId();
-        UserVmVO vm = Mockito.mock(UserVmVO.class);
+        UserVmVO vm = createMockUserVmWithServiceOfferingId();
         when(_vmInstanceDao.findById(anyLong())).thenReturn(vm);
-        when(vm.getServiceOfferingId()).thenReturn(1L);
         ServiceOfferingVO serviceOfferingVO = Mockito.mock(ServiceOfferingVO.class);
         when(serviceOfferingDao.findById(anyLong())).thenReturn(serviceOfferingVO);
         when(serviceOfferingVO.getDiskOfferingStrictness()).thenReturn(true);
@@ -2054,6 +2052,12 @@
             verify(volumeServiceMock, times(0)).resize(any(VolumeInfo.class));
         }
     }
+    private static UserVmVO createMockUserVmWithServiceOfferingId() {
+        UserVmVO vm = Mockito.mock(UserVmVO.class);
+        when(vm.getServiceOfferingId()).thenReturn(1L);
+        return vm;
+    }
+
     private UserVmVO getMockedVm() {
         UserVmVO vm = Mockito.mock(UserVmVO.class);
         Mockito.when(vm.getId()).thenReturn(1L);
