--- a/server/src/test/java/com/cloud/bgp/BGPServiceImplTest.java
+++ b/server/src/test/java/com/cloud/bgp/BGPServiceImplTest.java
@@ -89,8 +89,7 @@
     @Test
     public void testApplyBgpPeersForIsolatedNetwork() throws ResourceUnavailableException {
         Long networkId = 11L;
-        Network network = Mockito.mock(Network.class);
-        when(network.getId()).thenReturn(networkId);
+        Network network = createMockNetwork();
         when(network.getVpcId()).thenReturn(null);
 
         when(routedIpv4Manager.isDynamicRoutedNetwork(network)).thenReturn(true);
@@ -114,7 +113,7 @@
         Long accountId = 12L;
         Long vpcId = 13L;
         Long zoneId = 1L;
-        Network network = Mockito.mock(Network.class);
+        Network network = createMockNetwork();
         when(network.getId()).thenReturn(networkId);
         when(network.getVpcId()).thenReturn(vpcId);
         when(network.getAccountId()).thenReturn(accountId);
@@ -173,4 +172,10 @@
 
         verify(vpcVirtualRouterElement).applyBgpPeers(eq(vpc), eq(null), any());
     }
+
+    private static Network createMockNetwork() {
+        Network network = Mockito.mock(Network.class);
+        when(network.getId()).thenReturn(11L);
+        return network;
+    }
 }
