--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyTest.java
@@ -99,11 +99,13 @@
     HostDao hostDao;
     @Inject
     PrimaryDataStoreDao primaryDataStoreDao;
+    private VMSnapshotTO vmSnapshotTOMock;
 
     @Override
     @Before
     public void setUp() {
         ComponentContext.initComponentsLifeCycle();
+        vmSnapshotTOMock = Mockito.mock(VMSnapshotTO.class);
     }
 
     @Test
