--- a/server/src/test/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinitionTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinitionTest.java
@@ -594,10 +594,9 @@
         // Prepare
         when(mockNetworkModel.getPhysicalNetworkId(deployment.guestNetwork)).thenReturn(PHYSICAL_NW_ID);
         final Type type = Type.VirtualRouter;
-        final PhysicalNetworkServiceProviderVO physicalNwSrvProvider = mock(PhysicalNetworkServiceProviderVO.class);
+        final PhysicalNetworkServiceProviderVO physicalNwSrvProvider = createMockPhysicalNetworkServiceProvider();
         when(physicalProviderDao.findByServiceProvider(PHYSICAL_NW_ID, type.toString()))
         .thenReturn(physicalNwSrvProvider);
-        when(physicalNwSrvProvider.getId()).thenReturn(PROVIDER_ID);
 
         final VirtualRouterProviderVO vrProvider = mock(VirtualRouterProviderVO.class);
         when(mockVrProviderDao.findByNspIdAndType(PROVIDER_ID, type))
@@ -630,10 +629,9 @@
         // Prepare
         when(mockNetworkModel.getPhysicalNetworkId(deployment.guestNetwork)).thenReturn(PHYSICAL_NW_ID);
         final Type type = Type.VirtualRouter;
-        final PhysicalNetworkServiceProviderVO physicalNwSrvProvider = mock(PhysicalNetworkServiceProviderVO.class);
+        final PhysicalNetworkServiceProviderVO physicalNwSrvProvider = createMockPhysicalNetworkServiceProvider();
         when(physicalProviderDao.findByServiceProvider(PHYSICAL_NW_ID, type.toString()))
         .thenReturn(physicalNwSrvProvider);
-        when(physicalNwSrvProvider.getId()).thenReturn(PROVIDER_ID);
 
         when(mockVrProviderDao.findByNspIdAndType(PROVIDER_ID, type))
         .thenReturn(null);
@@ -859,4 +857,10 @@
             throws ConcurrentOperationException, InsufficientCapacityException, ResourceUnavailableException {
         driveTestExecuteDeployment(2, true);
     }
+
+    private static PhysicalNetworkServiceProviderVO createMockPhysicalNetworkServiceProvider() {
+        final PhysicalNetworkServiceProviderVO physicalNwSrvProvider = mock(PhysicalNetworkServiceProviderVO.class);
+        when(physicalNwSrvProvider.getId()).thenReturn(PROVIDER_ID);
+        return physicalNwSrvProvider;
+    }
 }
