--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtUnprepareStorageClientCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtUnprepareStorageClientCommandWrapperTest.java
@@ -46,12 +46,7 @@
 
     @Test
     public void testUnprepareStorageClientSuccess() {
-        UnprepareStorageClientCommand cmd = Mockito.mock(UnprepareStorageClientCommand.class);
-        Mockito.when(cmd.getPoolType()).thenReturn(Storage.StoragePoolType.PowerFlex);
-        Mockito.when(cmd.getPoolUuid()).thenReturn(poolUuid);
-        Map<String, String> details = new HashMap<>();
-        details.put(ScaleIOGatewayClient.STORAGE_POOL_MDMS, "1.1.1.1,2.2.2.2");
-        Mockito.when(cmd.getDetails()).thenReturn(details);
+        UnprepareStorageClientCommand cmd = createMockUnprepareStorageClientCommand();
 
         KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         Mockito.when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
@@ -64,12 +59,7 @@
 
     @Test
     public void testUnprepareStorageClientFailure() {
-        UnprepareStorageClientCommand cmd = Mockito.mock(UnprepareStorageClientCommand.class);
-        Mockito.when(cmd.getPoolType()).thenReturn(Storage.StoragePoolType.PowerFlex);
-        Mockito.when(cmd.getPoolUuid()).thenReturn(poolUuid);
-        Map<String, String> details = new HashMap<>();
-        details.put(ScaleIOGatewayClient.STORAGE_POOL_MDMS, "1.1.1.1,2.2.2.2");
-        Mockito.when(cmd.getDetails()).thenReturn(details);
+        UnprepareStorageClientCommand cmd = createMockUnprepareStorageClientCommand();
 
         KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         Mockito.when(libvirtComputingResourceMock.getStoragePoolMgr()).thenReturn(storagePoolMgr);
@@ -80,4 +70,14 @@
         Assert.assertFalse(result.getResult());
         Assert.assertEquals("Unprepare storage client failed", result.getDetails());
     }
+
+    private static UnprepareStorageClientCommand createMockUnprepareStorageClientCommand() {
+        UnprepareStorageClientCommand cmd = Mockito.mock(UnprepareStorageClientCommand.class);
+        Mockito.when(cmd.getPoolType()).thenReturn(Storage.StoragePoolType.PowerFlex);
+        Mockito.when(cmd.getPoolUuid()).thenReturn(poolUuid);
+        Map<String, String> details = new HashMap<>();
+        details.put(ScaleIOGatewayClient.STORAGE_POOL_MDMS, "1.1.1.1,2.2.2.2");
+        Mockito.when(cmd.getDetails()).thenReturn(details);
+        return cmd;
+    }
 }
