--- a/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
+++ b/server/src/test/java/com/cloud/api/ApiResponseHelperTest.java
@@ -501,7 +501,7 @@
         PublicIpQuarantine quarantinedIpMock = Mockito.mock(PublicIpQuarantine.class);
         IPAddressVO ipAddressVoMock = Mockito.mock(IPAddressVO.class);
         Account previousOwner = Mockito.mock(Account.class);
-        Account removerAccount = Mockito.mock(Account.class);
+        Account removerAccount = createMockRemoverAccount();
 
         Mockito.when(quarantinedIpMock.getUuid()).thenReturn(quarantinedIpUuid);
         Mockito.when(quarantinedIpMock.getPreviousOwnerId()).thenReturn(previousOwnerId);
@@ -516,7 +516,6 @@
         Mockito.when(quarantinedIpMock.getEndDate()).thenReturn(endDate);
         Mockito.when(quarantinedIpMock.getRemovalReason()).thenReturn(removalReason);
         Mockito.when(quarantinedIpMock.getRemoverAccountId()).thenReturn(removerAccountId);
-        Mockito.when(removerAccount.getUuid()).thenReturn(removerAccountUuid);
         Mockito.doReturn(removerAccount).when(accountManagerMock).getAccount(removerAccountId);
 
         IpQuarantineResponse result = apiResponseHelper.createQuarantinedIpsResponse(quarantinedIpMock);
@@ -531,6 +530,12 @@
         Assert.assertEquals(removalReason, result.getRemovalReason());
         Assert.assertEquals(removerAccountUuid, result.getRemoverAccountId());
         Assert.assertEquals("quarantinedip", result.getResponseName());
+    }
+
+    private static Account createMockRemoverAccount() {
+        Account removerAccount = Mockito.mock(Account.class);
+        Mockito.when(removerAccount.getUuid()).thenReturn("remover_account_uuid");
+        return removerAccount;
     }
 
     @Test
