--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -165,9 +165,7 @@
 
     @Test
     public void testGetDrsPlan() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -264,9 +262,7 @@
 
     @Test
     public void testGetDrsPlanWithSystemVMs() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -305,9 +301,7 @@
 
     @Test
     public void testGetDrsPlanWithNonRunningVMs() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -347,9 +341,7 @@
 
     @Test
     public void testGetDrsPlanWithSkipDrsFlag() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -395,9 +387,7 @@
 
     @Test
     public void testGetDrsPlanWithNoCompatibleHosts() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -443,9 +433,7 @@
 
     @Test
     public void testGetDrsPlanWithExceptionInCompatibilityCheck() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -492,9 +480,7 @@
 
     @Test
     public void testGetDrsPlanWithNoBestMigration() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -543,9 +529,7 @@
 
     @Test
     public void testGetDrsPlanWithMultipleIterations() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -619,9 +603,7 @@
 
     @Test
     public void testGetDrsPlanWithMigrationToOriginalHost() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         HostVO host1 = Mockito.mock(HostVO.class);
         Mockito.when(host1.getId()).thenReturn(1L);
@@ -703,9 +685,7 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void testGenerateDrsPlanConfigurationException() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
         Mockito.when(clusterDao.findById(1L)).thenReturn(cluster);
         Mockito.when(clusterDrsService.getDrsPlan(cluster, 5)).thenThrow(new ConfigurationException("test"));
         Mockito.when(cmd.getMaxMigrations()).thenReturn(5);
@@ -715,9 +695,7 @@
 
     @Test
     public void testGenerateDrsPlan() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
-        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        ClusterVO cluster = createMockCluster();
 
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(vm.getId()).thenReturn(1L);
@@ -951,4 +929,11 @@
 
         Mockito.verify(clusterDrsService, Mockito.times(2)).executeDrsPlan(Mockito.any(ClusterDrsPlanVO.class));
     }
+
+    private static ClusterVO createMockCluster() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.when(cluster.getId()).thenReturn(1L);
+        Mockito.when(cluster.getAllocationState()).thenReturn(Grouping.AllocationState.Enabled);
+        return cluster;
+    }
 }
