--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -275,6 +275,7 @@
     private VMInstanceDetailsDao vmInstanceDetailsDao;
     @Mock
     private ItWorkDao _workDao;
+    private ItWorkVO workMock;
     @Mock
     protected StateMachine2<State, VirtualMachine.Event, VirtualMachine> _stateMachine;
     @Mock
@@ -303,6 +304,7 @@
     public void setup() {
         ReflectionTestUtils.getField(VirtualMachineManager.VmMetadataManufacturer, "s_depot");
         virtualMachineManagerImpl.setHostAllocators(new ArrayList<>());
+        workMock = mock(ItWorkVO.class);
 
         when(vmInstanceMock.getName()).thenReturn(vmName);
         when(vmInstanceMock.getId()).thenReturn(vmInstanceVoMockId);
@@ -1128,7 +1130,6 @@
 
         ReservationContext ctx = mock(ReservationContext.class);
 
-        ItWorkVO work = mock(ItWorkVO.class);
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
@@ -1147,7 +1148,7 @@
 
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
-        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
+        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, workMock);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
@@ -1224,7 +1225,6 @@
 
         ReservationContext ctx = mock(ReservationContext.class);
 
-        ItWorkVO work = mock(ItWorkVO.class);
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
@@ -1243,7 +1243,7 @@
 
         doReturn(vmGuru).when(virtualMachineManagerImpl).getVmGuru(vmInstance);
 
-        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, work);
+        Ternary<VMInstanceVO, ReservationContext, ItWorkVO> start = new Ternary<>(vmInstance, ctx, workMock);
         Mockito.doReturn(start).when(virtualMachineManagerImpl).changeToStartState(vmGuru, vmInstance, user, account, owner, serviceOffering, template);
 
         when(ctx.getJournal()).thenReturn(Mockito.mock(Journal.class));
