--- a/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
+++ b/server/src/test/java/com/cloud/network/lb/AssignLoadBalancerTest.java
@@ -166,8 +166,8 @@
         UserVmVO vm = new UserVmVO(2L, "test", "test", 101L, Hypervisor.HypervisorType.Any, 21L, false, false, domainId, 200L, 1, 5L, "", 1L, null, "test");
 
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
-        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
+        LoadBalancerVMMapDao lb2VmMapDao;
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
@@ -175,12 +175,13 @@
         VpcManager vpcMgr = Mockito.mock(VpcManager.class);
 
         _lbMgr._lbDao = lbDao;
-        _lbMgr._lb2VmMapDao = lb2VmMapDao;
         _lbMgr._vmDao = userVmDao;
         _lbMgr._accountDao = accountDao;
         _lbMgr._accountMgr = accountMgr;
         _lbMgr._networkDao = networkDao;
         _lbvmMapList = new ArrayList<>();
+        lb2VmMapDao = createMockLoadBalancerVMMapDao(_lbvmMapList);
+        _lbMgr._lb2VmMapDao = lb2VmMapDao;
         _lbMgr._rulesMgr = _rulesMgr;
         _lbMgr._networkModel = _networkModel;
         _lbMgr._autoScaleVmGroupDao = autoScaleVmGroupDao;
@@ -188,7 +189,6 @@
 
         when(lbDao.findById(anyLong())).thenReturn(Mockito.mock(LoadBalancerVO.class));
         when(userVmDao.findById(anyLong())).thenReturn(vm);
-        when(lb2VmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(_lbvmMapList);
         when(accountDao.findById(anyLong())).thenReturn(Mockito.mock(AccountVO.class));
         Mockito.doNothing().when(accountMgr).checkAccess(any(Account.class), any(SecurityChecker.AccessType.class), any(Boolean.class), any(Network.class));
         when(autoScaleVmGroupDao.isAutoScaleLoadBalancer(anyLong())).thenReturn(Boolean.FALSE);
@@ -215,7 +215,7 @@
         UserVmVO vm = new UserVmVO(2L, "test", "test", 101L, Hypervisor.HypervisorType.Any, 21L, false, false, domainId, 200L, 1, 5L, "", 1L, null, "test");
 
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
-        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
+        LoadBalancerVMMapDao lb2VmMapDao = createMockLoadBalancerVMMapDao(_lbvmMapList);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         NetworkDao networkDao = Mockito.mock(NetworkDao.class);
@@ -266,7 +266,6 @@
         UserVmVO vm = new UserVmVO(2L, "test", "test", 101L, Hypervisor.HypervisorType.Any, 21L, false, false, domainId, 200L, 1, 5L, "", 1L, null, "test");
 
         LoadBalancerDao lbDao = Mockito.mock(LoadBalancerDao.class);
-        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
         UserVmDao userVmDao = Mockito.mock(UserVmDao.class);
         AccountDao accountDao = Mockito.mock(AccountDao.class);
         AccountManager accountMgr = Mockito.mock(AccountManager.class);
@@ -277,7 +276,6 @@
         VpcManager vpcMgr = Mockito.mock(VpcManager.class);
 
         _lbMgr._lbDao = lbDao;
-        _lbMgr._lb2VmMapDao = lb2VmMapDao;
         _lbMgr._vmDao = userVmDao;
         _lbMgr._accountDao = accountDao;
         _lbMgr._accountMgr = accountMgr;
@@ -285,6 +283,8 @@
         _lbMgr._nicSecondaryIpDao = nicSecIpDao;
         _lbvmMapList = new ArrayList<>();
         _lbvmMapList.add(lbVmMapVO);
+        LoadBalancerVMMapDao lb2VmMapDao = createMockLoadBalancerVMMapDao(_lbvmMapList);
+        _lbMgr._lb2VmMapDao = lb2VmMapDao;
         _lbMgr._rulesMgr = _rulesMgr;
         _lbMgr._networkModel = _networkModel;
         _lbMgr._autoScaleVmGroupDao = autoScaleVmGroupDao;
@@ -292,7 +292,6 @@
 
         when(lbDao.findById(anyLong())).thenReturn(lbVO);
         when(userVmDao.findById(anyLong())).thenReturn(vm);
-        when(lb2VmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(_lbvmMapList);
         when (nicSecIpDao.findByIp4AddressAndNicId(anyString(), anyLong())).thenReturn(null);
         when(autoScaleVmGroupDao.isAutoScaleLoadBalancer(anyLong())).thenReturn(Boolean.FALSE);
         when(vpcMgr.isNetworkOnVpcEnabledConserveMode(any())).thenReturn(false);
@@ -300,6 +299,12 @@
         _lbMgr.assignToLoadBalancer(1L, null, vmIdIpMap, null, false);
     }
 
+    private static LoadBalancerVMMapDao createMockLoadBalancerVMMapDao(List<LoadBalancerVMMapVO> lbvmMapList) {
+        LoadBalancerVMMapDao lb2VmMapDao = Mockito.mock(LoadBalancerVMMapDao.class);
+        when(lb2VmMapDao.listByLoadBalancerId(anyLong(), anyBoolean())).thenReturn(lbvmMapList);
+        return lb2VmMapDao;
+    }
+
     @After
     public void tearDown() {
         CallContext.unregister();
