--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1047,9 +1047,8 @@
 
     @Test
     public void validateCurrentPasswordTestUserAuthenticatedWithProvidedCurrentPasswordViaFirstAuthenticator() {
-        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
+        AccountVO accountVoMock = createMockAccountVo();
         long domainId = 14L;
-        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
 
         Mockito.doReturn(accountVoMock).when(_accountDao).findById(accountMockId);
         String username = "username";
@@ -1076,9 +1075,8 @@
 
     @Test
     public void validateCurrentPasswordTestUserAuthenticatedWithProvidedCurrentPasswordViaSecondAuthenticator() {
-        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
+        AccountVO accountVoMock = createMockAccountVo();
         long domainId = 14L;
-        Mockito.doReturn(domainId).when(accountVoMock).getDomainId();
 
         Mockito.doReturn(accountVoMock).when(_accountDao).findById(accountMockId);
         String username = "username";
@@ -2289,4 +2287,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static AccountVO createMockAccountVo() {
+        AccountVO accountVoMock = Mockito.mock(AccountVO.class);
+        Mockito.doReturn(14L).when(accountVoMock).getDomainId();
+        return accountVoMock;
+    }
 }
