--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer56WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer56WrapperTest.java
@@ -57,6 +57,8 @@
     @Mock
     private XenServer56Resource xenServer56Resource;
 
+    private final Host hostMock = Mockito.mock(Host.class);
+
     final static long[] vpcStats = { 1000L, 2000L };
     final static long[] networkStats = { 3000L, 4000L };
     final static long[] lbStats = { 5L };
