--- a/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImplTest.java
@@ -728,8 +728,7 @@
         long hostId = 1L;
         String url = "url";
         String consoleAddress = "127.0.0.1";
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        Mockito.when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = createMockVirtualMachine();
         HostVO host = Mockito.mock(HostVO.class);
         Mockito.when(host.getId()).thenReturn(hostId);
         String tag = UUID.randomUUID().toString();
@@ -771,8 +770,7 @@
         ConsoleConnectionDetails details = new ConsoleConnectionDetails("password", "en", "tag", null);
         details.setDirectUrl(url);
         details.setModeFromExternalProtocol("direct");
-        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
-        Mockito.when(vm.getId()).thenReturn(vmId);
+        VirtualMachine vm = createMockVirtualMachine();
         HostVO host = Mockito.mock(HostVO.class);
         Mockito.when(host.getId()).thenReturn(hostId);
         Mockito.doReturn(details).when(consoleAccessManager).getConsoleConnectionDetails(vm, host);
@@ -787,4 +785,9 @@
         Assert.assertNull(endpoint.getWebsocketExtra());
         Assert.assertNull(endpoint.getWebsocketHost());
     }
+    private static VirtualMachine createMockVirtualMachine() {
+        VirtualMachine vm = Mockito.mock(VirtualMachine.class);
+        Mockito.when(vm.getId()).thenReturn(100L);
+        return vm;
+    }
 }
