--- a/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/StorageManagerImplTest.java
@@ -197,6 +197,12 @@
         assertEquals(expectedLocalStorageName, localStoragePoolName);
     }
 
+    private static VolumeVO createMockVolumeVO(Long instanceId) {
+        VolumeVO volume2VOMock = Mockito.mock(VolumeVO.class);
+        Mockito.doReturn(instanceId).when(volume2VOMock).getInstanceId();
+        return volume2VOMock;
+    }
+
     private VolumeVO mockVolumeForIsVolumeSuspectedDestroyDuplicateTest() {
         VolumeVO volumeVO = new VolumeVO("data", 1L, 1L, 1L, 1L, 1L, "data", "data", Storage.ProvisioningType.THIN, 1, null, null, "data", Volume.Type.DATADISK);
         volumeVO.setPoolId(1L);
@@ -577,9 +583,9 @@
     @Test
     public void getStoragePoolNonDestroyedVolumesLogTestNonDestroyedVolumes_VMAttachedLogs() {
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
-        Mockito.doReturn(1L).when(volume1VOMock).getInstanceId();
+        volume1VOMock = createMockVolumeVO(1L);
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO(1L);
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
         Mockito.doReturn("58760044-928f-4c4e-9fef-d0e48423595e").when(vmInstanceVOMock).getUuid();
 
@@ -597,7 +603,7 @@
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
         Mockito.doReturn(null).when(volume1VOMock).getInstanceId();
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO(1L);
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
         Mockito.doReturn("58760044-928f-4c4e-9fef-d0e48423595e").when(vmInstanceVOMock).getUuid();
 
@@ -629,9 +635,9 @@
     @Test
     public void getStoragePoolNonDestroyedVolumesLogTestNonDestroyedVolumes_VMNotExistsLog() {
         Mockito.doReturn(1L).when(storagePoolVOMock).getId();
-        Mockito.doReturn(1L).when(volume1VOMock).getInstanceId();
+        volume1VOMock = createMockVolumeVO(1L);
         Mockito.doReturn("786633d1-a942-4374-9d56-322dd4b0d202").when(volume1VOMock).getUuid();
-        Mockito.doReturn(1L).when(volume2VOMock).getInstanceId();
+        volume2VOMock = createMockVolumeVO(1L);
         Mockito.doReturn("ffb46333-e983-4c21-b5f0-51c5877a3805").when(volume2VOMock).getUuid();
 
         Mockito.when(_volumeDao.findNonDestroyedVolumesByPoolId(storagePoolVOMock.getId(), null)).thenReturn(List.of(volume1VOMock, volume2VOMock));
