--- a/server/src/test/java/com/cloud/network/firewall/FirewallManagerTest.java
+++ b/server/src/test/java/com/cloud/network/firewall/FirewallManagerTest.java
@@ -110,6 +110,8 @@
     FirewallRule fwRule100to200;
     FirewallRule fwRule151to200;
 
+    FirewallRule nonVpcRuleMock;
+
     FirewallRule pfRule50to150;
     FirewallRule pfRule100to200;
     FirewallRule pfRule151to200;
@@ -118,6 +120,7 @@
     @Before
     public void initMocks() {
         closeable = MockitoAnnotations.openMocks(this);
+        nonVpcRuleMock = Mockito.mock(FirewallRule.class);
 
         fwRule50to150 = createFirewallRule(50, 150, Purpose.Firewall);
         fwRule100to200 = createFirewallRule(100, 150, Purpose.Firewall);
@@ -608,9 +611,8 @@
     @Test
     public void testCreateFirewallRuleRoutesToVpcWhenVpcIdProvided() throws NetworkRuleConflictException {
         Account caller = Mockito.mock(Account.class);
-        FirewallRule vpcRule = Mockito.mock(FirewallRule.class);
-
-        doReturn(vpcRule).when(_firewallMgr).createFirewallRuleForVpc(
+
+        doReturn(nonVpcRuleMock).when(_firewallMgr).createFirewallRuleForVpc(
                 Mockito.anyLong(), Mockito.eq(caller), Mockito.any(), Mockito.anyInt(), Mockito.anyInt(),
                 Mockito.anyString(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                 Mockito.any(), Mockito.any(FirewallRuleType.class), Mockito.anyLong(),
@@ -635,9 +637,8 @@
     @Test
     public void testCreateFirewallRuleRoutesToNonVpcWhenVpcIdNull() throws NetworkRuleConflictException {
         Account caller = Mockito.mock(Account.class);
-        FirewallRule nonVpcRule = Mockito.mock(FirewallRule.class);
-
-        doReturn(nonVpcRule).when(_firewallMgr).createFirewallRuleForNonVPC(
+
+        doReturn(nonVpcRuleMock).when(_firewallMgr).createFirewallRuleForNonVPC(
                 Mockito.any(), Mockito.eq(caller), Mockito.any(), Mockito.any(), Mockito.any(),
                 Mockito.anyString(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
                 Mockito.any(), Mockito.any(FirewallRuleType.class), Mockito.anyLong(),
