--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
@@ -283,6 +283,12 @@
         Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSameZoneStoreForQCOW2Volume(Mockito.mock(Snapshot.class), volumeVO));
     }
 
+    private static VolumeVO createMockVolumeVO() {
+        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
+        Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
+        return volumeVO;
+    }
+
     private void prepareMocksForIsSnapshotStoredOnSameZoneStoreForQCOW2VolumeTest(Long matchingZoneId) {
         SnapshotDataStoreVO ref1 = Mockito.mock(SnapshotDataStoreVO.class);
         Mockito.when(ref1.getDataStoreId()).thenReturn(201L);
@@ -333,7 +339,7 @@
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_NullPoolId() {
         Snapshot snapshot = Mockito.mock(Snapshot.class);
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
+        VolumeVO volumeVO = createMockVolumeVO();
         Mockito.when(volumeVO.getPoolId()).thenReturn(null);
 
         Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, volumeVO));
@@ -342,8 +348,7 @@
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_NullPool() {
         Snapshot snapshot = Mockito.mock(Snapshot.class);
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
+        VolumeVO volumeVO = createMockVolumeVO();
         Mockito.when(dataStoreManager.getDataStore(10L, DataStoreRole.Primary)).thenReturn(null);
 
         Assert.assertFalse(defaultSnapshotStrategySpy.isSnapshotStoredOnSecondaryForCLVMVolume(snapshot, volumeVO));
@@ -365,8 +370,7 @@
     @Test
     public void testIsSnapshotStoredOnSecondaryForCLVMVolume_RBDPool() {
         Snapshot snapshot = Mockito.mock(Snapshot.class);
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
+        VolumeVO volumeVO = createMockVolumeVO();
 
         StoragePool pool = Mockito.mock(StoragePool.class, Mockito.withSettings().extraInterfaces(DataStore.class));
         Mockito.when(pool.getPoolType()).thenReturn(Storage.StoragePoolType.RBD);
@@ -380,8 +384,7 @@
         Snapshot snapshot = Mockito.mock(Snapshot.class);
         Mockito.when(snapshot.getId()).thenReturn(1L);
 
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeVO.getPoolId()).thenReturn(10L);
+        VolumeVO volumeVO = createMockVolumeVO();
 
         StoragePool pool = Mockito.mock(StoragePool.class, Mockito.withSettings().extraInterfaces(DataStore.class));
         Mockito.when(pool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
