--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -189,6 +189,13 @@
         closeable.close();
     }
 
+    private static Network createMockNetwork() {
+        Network network = Mockito.mock(Network.class);
+        Mockito.when(network.getDataCenterId()).thenReturn(1L);
+        Mockito.when(network.getVpcId()).thenReturn(null);
+        return network;
+    }
+
     private DataCenterGuestIpv6PrefixVO prepareMocksForIpv6Subnet() {
         final long prefixId = 1L;
         DataCenterGuestIpv6PrefixVO prefix = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
@@ -617,9 +624,7 @@
     public void testCheckNetworkIpv6UpgradeForNoIpv6Vlan() {
         final long zoneId = 1L;
         Mockito.when(dataCenterGuestIpv6PrefixDao.listByDataCenterId(Mockito.anyLong())).thenReturn(List.of(Mockito.mock(DataCenterGuestIpv6PrefixVO.class)));
-        Network network = Mockito.mock(Network.class);
-        Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
-        Mockito.when(network.getVpcId()).thenReturn(null);
+        Network network = createMockNetwork();
         Mockito.when(ipAddressDao.listByAssociatedNetwork(Mockito.anyLong(), Mockito.anyBoolean())).thenReturn(List.of(Mockito.mock(IPAddressVO.class)));
         VlanVO vlanVO = Mockito.mock(VlanVO.class);
         Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
@@ -638,9 +643,8 @@
     public void testCheckNetworkIpv6UpgradeForNetwork() {
         final long zoneId = 1L;
         Mockito.when(dataCenterGuestIpv6PrefixDao.listByDataCenterId(Mockito.anyLong())).thenReturn(List.of(Mockito.mock(DataCenterGuestIpv6PrefixVO.class)));
-        Network network = Mockito.mock(Network.class);
+        Network network = createMockNetwork();
         Mockito.when(network.getDataCenterId()).thenReturn(zoneId);
-        Mockito.when(network.getVpcId()).thenReturn(null);
         Mockito.when(ipAddressDao.listByAssociatedNetwork(Mockito.anyLong(), Mockito.anyBoolean())).thenReturn(List.of(Mockito.mock(IPAddressVO.class)));
         VlanVO vlanVO = Mockito.mock(VlanVO.class);
         Mockito.when(vlanVO.getVlanTag()).thenReturn(vlan);
