--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -413,8 +413,11 @@
     @Mock
     LoadBalancerVMMapVO loadBalancerVMMapMock;
 
+    private AutoScaleVmProfileTO profileTOMock;
+
     @Before
     public void setUp() {
+        profileTOMock = Mockito.mock(AutoScaleVmProfileTO.class);
 
         account = new AccountVO("testaccount", 1L, "networkdomain", Account.Type.NORMAL, "uuid");
         account.setId(5L);
@@ -1624,7 +1627,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -1641,7 +1643,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -1658,7 +1659,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -1675,7 +1675,6 @@
         AutoScalePolicyTO policyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
         CounterTO counterTO = Mockito.mock(CounterTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
 
         when(groupTO.getPolicies()).thenReturn(Arrays.asList(policyTO));
         when(policyTO.getConditions()).thenReturn(Arrays.asList(conditionTO));
@@ -2156,7 +2155,6 @@
     @Test
     public void updateCountersMapWithInstantDataForCPU() {
         AutoScaleVmGroupTO groupTO = Mockito.mock(AutoScaleVmGroupTO.class);
-        AutoScaleVmProfileTO profileTO = Mockito.mock(AutoScaleVmProfileTO.class);
         AutoScalePolicyTO scaleUpPolicyTO = Mockito.mock(AutoScalePolicyTO.class);
         ConditionTO conditionTO = Mockito.mock(ConditionTO.class);
 
