--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -265,6 +265,8 @@
     @Mock
     private BackupOfferingDetailsDao backupOfferingDetailsDao;
 
+    private SearchCriteria<BackupOfferingVO> backupOfferingSearchCriteria;
+
     @Mock
     DomainHelper domainHelper;
 
@@ -291,6 +293,7 @@
     @Before
     public void setup() throws Exception {
         gson = new Gson();
+        backupOfferingSearchCriteria = Mockito.mock(SearchCriteria.class);
 
         closeable = MockitoAnnotations.openMocks(this);
         when(backupOfferingDao.findById(null)).thenReturn(null);
@@ -2358,12 +2361,11 @@
         List<BackupOfferingVO> allOfferings = List.of(globalOffering, domainOffering);
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
         BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
         when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(backupOfferingSearchCriteria);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
             .thenReturn(new Pair<>(allOfferings, allOfferings.size()));
 
@@ -2402,12 +2404,11 @@
         List<BackupOfferingVO> allOfferings = List.of(directDomainOffering, otherDomainOffering);
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
         BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
         when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(backupOfferingSearchCriteria);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
             .thenReturn(new Pair<>(allOfferings, allOfferings.size()));
 
@@ -2447,12 +2448,11 @@
         List<BackupOfferingVO> allOfferings = List.of(parentDomainOffering, siblingDomainOffering);
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
         BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
         when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(backupOfferingSearchCriteria);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
             .thenReturn(new Pair<>(allOfferings, allOfferings.size()));
 
@@ -2493,12 +2493,11 @@
         List<BackupOfferingVO> allOfferings = List.of(siblingOffering);
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
         BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
         when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(backupOfferingSearchCriteria);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
             .thenReturn(new Pair<>(allOfferings, allOfferings.size()));
 
@@ -2537,12 +2536,11 @@
         List<BackupOfferingVO> allOfferings = List.of(multiDomainOffering);
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
         BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
         when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(backupOfferingSearchCriteria);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
             .thenReturn(new Pair<>(allOfferings, allOfferings.size()));
 
@@ -2586,12 +2584,11 @@
         List<BackupOfferingVO> allOfferings = List.of(globalOffering, userDomainOffering, otherDomainOffering);
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
         BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
         when(sb.entity()).thenReturn(entityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
-        when(sb.create()).thenReturn(sc);
+        when(sb.create()).thenReturn(backupOfferingSearchCriteria);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
             .thenReturn(new Pair<>(allOfferings, allOfferings.size()));
 
