--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -215,6 +215,7 @@
     DiskOfferingDao diskOfferingDao;
     @Mock
     VolumeVO volumeVO;
+    private VolumeVO volumeVOMock;
     @Mock
     StoragePoolVO storagePoolVO;
     @Mock
@@ -237,6 +238,7 @@
     @Before
     public void setup() throws Exception {
         closeable = MockitoAnnotations.openMocks(this);
+        volumeVOMock = Mockito.mock(VolumeVO.class);
 
         Account account = new AccountVO("testaccount", 1, "networkdomain", Account.Type.NORMAL, UUID.randomUUID().toString());
         when(configurationMgr._accountMgr.getAccount(anyLong())).thenReturn(account);
@@ -767,9 +769,8 @@
         Mockito.when(hostPodVO.getDataCenterId()).thenReturn(new Random().nextLong());
         Mockito.when(_podDao.findById(anyLong())).thenReturn(hostPodVO);
 
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         ArrayList<VolumeVO> arrayList = new ArrayList<VolumeVO>();
-        arrayList.add(volumeVO);
+        arrayList.add(volumeVOMock);
         Mockito.when(_privateIpAddressDao.countIPs(anyLong(), anyLong(), anyBoolean())).thenReturn(0);
         Mockito.when(_volumeDao.findByPod(anyLong())).thenReturn(arrayList);
         Mockito.when(_hostDao.findByPodId(anyLong())).thenReturn(new ArrayList<HostVO>());
@@ -927,9 +928,8 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void checkIfZoneIsDeletableFailureOnVolumeTest() {
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
         ArrayList<VolumeVO> arrayList = new ArrayList<VolumeVO>();
-        arrayList.add(volumeVO);
+        arrayList.add(volumeVOMock);
 
         Mockito.when(_hostDao.listEnabledIdsByDataCenterId(anyLong())).thenReturn(new ArrayList<>());
         Mockito.when(_podDao.listByDataCenterId(anyLong())).thenReturn(new ArrayList<HostPodVO>());
