--- a/api/src/test/java/org/apache/cloudstack/api/command/test/ActivateProjectCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/ActivateProjectCmdTest.java
@@ -69,8 +69,7 @@
 
     @Test
     public void testGetEntityOwnerIdForProject() {
-        Project project = Mockito.mock(Project.class);
-        Mockito.when(project.getId()).thenReturn(2L);
+        Project project = MockProject.createMockProject();
         ProjectService projectService = Mockito.mock(ProjectService.class);
         Account account = Mockito.mock(Account.class);
         Mockito.when(account.getId()).thenReturn(2L);
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/AddAccountToProjectCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/AddAccountToProjectCmdTest.java
@@ -95,8 +95,7 @@
     @Test
     public void testGetEntityOwnerIdForProject() {
 
-        Project project = Mockito.mock(Project.class);
-        Mockito.when(project.getId()).thenReturn(2L);
+        Project project = MockProject.createMockProject();
 
         ProjectService projectService = Mockito.mock(ProjectService.class);
         Account account = Mockito.mock(Account.class);
--- a/api/src/test/java/org/apache/cloudstack/api/command/test/MockProject.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/test/MockProject.java
@@ -0,0 +1,12 @@
+package org.apache.cloudstack.api.command.test;
+
+import com.cloud.projects.Project;
+import org.mockito.Mockito;
+
+public class MockProject {
+    public static Project createMockProject() {
+        Project project = Mockito.mock(Project.class);
+        Mockito.when(project.getId()).thenReturn(2L);
+        return project;
+    }
+}
