--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -145,6 +145,9 @@
     SearchCriteria<IPAddressVO> sc;
 
     @Mock
+    private SearchCriteria<UserDataVO> userDataSearchCriteria;
+
+    @Mock
     RegisterSSHKeyPairCmd regCmd;
 
     @Mock
@@ -622,8 +625,7 @@
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
             when(sb.entity()).thenReturn(userData);
 
-            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
-            when(sb.create()).thenReturn(sc);
+            when(sb.create()).thenReturn(userDataSearchCriteria);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
             userDataList.add(userData);
@@ -655,8 +657,7 @@
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
             when(sb.entity()).thenReturn(userData);
 
-            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
-            when(sb.create()).thenReturn(sc);
+            when(sb.create()).thenReturn(userDataSearchCriteria);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
             userDataList.add(userData);
@@ -688,8 +689,7 @@
             when(userDataDao.createSearchBuilder()).thenReturn(sb);
             when(sb.entity()).thenReturn(userData);
 
-            SearchCriteria<UserDataVO> sc = Mockito.mock(SearchCriteria.class);
-            when(sb.create()).thenReturn(sc);
+            when(sb.create()).thenReturn(userDataSearchCriteria);
 
             List<UserDataVO> userDataList = new ArrayList<UserDataVO>();
             userDataList.add(userData);
