--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorkerTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorkerTest.java
@@ -46,6 +46,8 @@
 
     @Mock
     private KubernetesCluster kubernetesCluster;
+    @Mock
+    private KubernetesCluster kubernetesClusterMock;
     @Mock
     private KubernetesClusterManagerImpl clusterManager;
     @Mock
@@ -131,9 +133,8 @@
 
     @Test
     public void testGetWorkerNodesToRemoveForDownsize_singleRemoval() {
-        KubernetesCluster kubernetesCluster = Mockito.mock(KubernetesCluster.class);
         KubernetesClusterManagerImpl clusterManager = Mockito.mock(KubernetesClusterManagerImpl.class);
-        KubernetesClusterScaleWorker worker = new KubernetesClusterScaleWorker(kubernetesCluster, new java.util.HashMap<>(), 2L, null, false, null, null, clusterManager);
+        KubernetesClusterScaleWorker worker = new KubernetesClusterScaleWorker(kubernetesClusterMock, new java.util.HashMap<>(), 2L, null, false, null, null, clusterManager);
         KubernetesClusterScaleWorker spyWorker = Mockito.spy(worker);
 
         KubernetesClusterVmMapVO vm1 = Mockito.mock(KubernetesClusterVmMapVO.class);
@@ -161,9 +162,8 @@
 
     @Test
     public void testGetWorkerNodesToRemoveForDownsize_noRemoval() {
-        KubernetesCluster kubernetesCluster = Mockito.mock(KubernetesCluster.class);
 
-        KubernetesClusterScaleWorker worker = new KubernetesClusterScaleWorker(kubernetesCluster, new java.util.HashMap<>(), 3L, null, false, null, null, clusterManager);
+        KubernetesClusterScaleWorker worker = new KubernetesClusterScaleWorker(kubernetesClusterMock, new java.util.HashMap<>(), 3L, null, false, null, null, clusterManager);
         KubernetesClusterScaleWorker spyWorker = Mockito.spy(worker);
 
         KubernetesClusterVmMapVO vm1 = Mockito.mock(KubernetesClusterVmMapVO.class);
