--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -74,8 +74,7 @@
     public void testProcessWithEmptyPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
@@ -104,8 +103,7 @@
     public void testProcessWithPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
@@ -141,8 +139,7 @@
     public void testProcessWithNotRunningVM() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -73,8 +73,7 @@
     public void testProcessWithEmptyPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile vmProfile = org.apache.cloudstack.affinity.MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
@@ -103,8 +102,7 @@
     public void testProcessWithPlan() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile vmProfile = org.apache.cloudstack.affinity.MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
@@ -140,8 +138,7 @@
     public void testProcessWithNotRunningVM() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         when(vm.getId()).thenReturn(vmId);
-        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
-        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        VirtualMachineProfile vmProfile = MockVirtualMachineProfile.createMockVirtualMachineProfile(vm);
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/MockVirtualMachineProfile.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/MockVirtualMachineProfile.java
@@ -0,0 +1,16 @@
+package org.apache.cloudstack.affinity;
+
+import com.cloud.vm.VirtualMachine;
+import com.cloud.vm.VirtualMachineProfile;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+public class MockVirtualMachineProfile {
+    public static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachine vm) {
+        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
+        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        return vmProfile;
+    }
+}
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/MockVirtualMachineProfile.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/MockVirtualMachineProfile.java
@@ -0,0 +1,16 @@
+package org.apache.cloudstack.affinity;
+
+import com.cloud.vm.VirtualMachine;
+import com.cloud.vm.VirtualMachineProfile;
+
+import static org.mockito.Mockito.when;
+
+import org.mockito.Mockito;
+
+public class MockVirtualMachineProfile {
+    public static VirtualMachineProfile createMockVirtualMachineProfile(VirtualMachine vm) {
+        VirtualMachineProfile vmProfile = Mockito.mock(VirtualMachineProfile.class);
+        when(vmProfile.getVirtualMachine()).thenReturn(vm);
+        return vmProfile;
+    }
+}
