--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -352,6 +352,12 @@
         return method;
     }
 
+    private static Pod createMockPod() {
+        Pod destPod = mock(Pod.class);
+        when(destPod.getId()).thenReturn(2L);
+        return destPod;
+    }
+
     @Test
     public void testaddHostIpToCertDetailsIfConfigAllows() {
         Host vmHost = mock(Host.class);
@@ -1158,13 +1164,12 @@
         when(_entityMgr.findByIdIncludingRemoved(VirtualMachineTemplate.class, vmInstance.getTemplateId())).thenReturn(template);
 
         Host destHost = mock(Host.class);
-        Pod destPod = mock(Pod.class);
+        Pod destPod = createMockPod();
         DeployDestination dest = mock(DeployDestination.class);
         when(dest.getHost()).thenReturn(destHost);
         when(dest.getPod()).thenReturn(destPod);
         when(dest.getCluster()).thenReturn(mock(Cluster.class));
         when(destHost.getId()).thenReturn(1L);
-        when(destPod.getId()).thenReturn(2L);
         when(_dpMgr.planDeployment(any(VirtualMachineProfileImpl.class), any(DataCenterDeployment.class), any(ExcludeList.class), any(DeploymentPlanner.class))).thenReturn(dest);
 
         doNothing().when(virtualMachineManagerImpl).checkIfTemplateNeededForCreatingVmVolumes(vmInstance);
@@ -1254,13 +1259,12 @@
         when(_entityMgr.findByIdIncludingRemoved(VirtualMachineTemplate.class, vmInstance.getTemplateId())).thenReturn(template);
 
         Host destHost = mock(Host.class);
-        Pod destPod = mock(Pod.class);
+        Pod destPod = createMockPod();
         DeployDestination dest = mock(DeployDestination.class);
         when(dest.getHost()).thenReturn(destHost);
         when(dest.getPod()).thenReturn(destPod);
         when(dest.getCluster()).thenReturn(mock(Cluster.class));
         when(destHost.getId()).thenReturn(1L);
-        when(destPod.getId()).thenReturn(2L);
         when(_dpMgr.planDeployment(any(VirtualMachineProfileImpl.class), any(DataCenterDeployment.class), any(ExcludeList.class), any(DeploymentPlanner.class))).thenReturn(dest);
 
         doNothing().when(virtualMachineManagerImpl).checkIfTemplateNeededForCreatingVmVolumes(vmInstance);
