--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -350,10 +350,8 @@
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
-        final DeployDestination dest = mock(DeployDestination.class);
-
         final DataCenterVO dc = mock(DataCenterVO.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        final DeployDestination dest = createMockDeployDestination(dc);
         when(dcdao.findById((long)anyInt())).thenReturn(dc);
 
         final HostVO brocadeHost = mock(HostVO.class);
@@ -402,10 +400,8 @@
 
         final VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
 
-        final DeployDestination dest = mock(DeployDestination.class);
-
         final DataCenterVO dc = mock(DataCenterVO.class);
-        when(dest.getDataCenter()).thenReturn(dc);
+        final DeployDestination dest = createMockDeployDestination(dc);
         when(dcdao.findById((long)anyInt())).thenReturn(dc);
 
         final HostVO brocadeHost = mock(HostVO.class);
@@ -588,4 +584,9 @@
         assertTrue(result == false);
         verify(agentmgr, times(0)).easySend(eq(NETWORK_ID), (Command)any());
     }
+    private static DeployDestination createMockDeployDestination(DataCenterVO dataCenter) {
+        final DeployDestination dest = mock(DeployDestination.class);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
+        return dest;
+    }
 }
