--- a/engine/schema/src/test/java/com/cloud/gpu/dao/GpuDeviceDaoImplTest.java
+++ b/engine/schema/src/test/java/com/cloud/gpu/dao/GpuDeviceDaoImplTest.java
@@ -239,12 +239,10 @@
 
     @Test
     public void getDistinctVgpuProfileIds() {
-        GpuDeviceVO device1 = mock(GpuDeviceVO.class);
-        GpuDeviceVO device2 = mock(GpuDeviceVO.class);
-        GpuDeviceVO device3 = mock(GpuDeviceVO.class);
-        when(device1.getVgpuProfileId()).thenReturn(1L);
+        GpuDeviceVO device1 = createMockGpuDevice();
+        GpuDeviceVO device2 = createMockGpuDevice();
+        GpuDeviceVO device3 = createMockGpuDevice();
         when(device2.getVgpuProfileId()).thenReturn(2L);
-        when(device3.getVgpuProfileId()).thenReturn(1L);
 
         doReturn(List.of(device1, device2, device3)).when(gpuDeviceDao).listBy(any(SearchCriteria.class));
 
@@ -274,4 +272,10 @@
         Assert.assertEquals("Expected correct where clause", "gpu_device.parent_gpu_device_id = ?",
                 scCaptor.getValue().getWhereClause().trim());
     }
+
+    private static GpuDeviceVO createMockGpuDevice() {
+        GpuDeviceVO device1 = mock(GpuDeviceVO.class);
+        when(device1.getVgpuProfileId()).thenReturn(1L);
+        return device1;
+    }
 }
