--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -98,6 +98,8 @@
     @Mock
     ConfigurationDao configDao;
 
+    private DataTO srcDataTOMock;
+
     static MockedStatic<RemoteHostEndPoint> remoteHostEndPointMock;
 
     private AutoCloseable closeable;
@@ -115,6 +117,7 @@
     @Before
     public void initMocks() {
         closeable = MockitoAnnotations.openMocks(this);
+        srcDataTOMock = Mockito.mock(DataTO.class);
     }
 
     @After
@@ -515,8 +518,7 @@
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
         DataObject srcData = Mockito.mock(DataObject.class);
-        DataTO srcDataTO = Mockito.mock(DataTO.class);
-        when(srcData.getTO()).thenReturn(srcDataTO);
+        when(srcData.getTO()).thenReturn(srcDataTOMock);
         DataObject destData = Mockito.mock(DataObject.class);
         DataTO destDataTO = Mockito.mock(DataTO.class);
         when(destData.getTO()).thenReturn(destDataTO);
@@ -538,8 +540,7 @@
         when(configDao.getValue(Config.CopyVolumeWait.key())).thenReturn("3600");
 
         DataObject srcData = Mockito.mock(DataObject.class);
-        DataTO srcDataTO = Mockito.mock(DataTO.class);
-        when(srcData.getTO()).thenReturn(srcDataTO);
+        when(srcData.getTO()).thenReturn(srcDataTOMock);
         DataObject destData = Mockito.mock(DataObject.class);
         DataTO destDataTO = Mockito.mock(DataTO.class);
         when(destData.getTO()).thenReturn(destDataTO);
