--- a/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/as/AutoScaleManagerImplTest.java
@@ -413,8 +413,12 @@
     @Mock
     LoadBalancerVMMapVO loadBalancerVMMapMock;
 
+    private SearchCriteria<AutoScalePolicyVO> searchCriteriaPolicyMock;
+
     @Before
     public void setUp() {
+
+        searchCriteriaPolicyMock = Mockito.mock(SearchCriteria.class);
 
         account = new AccountVO("testaccount", 1L, "networkdomain", Account.Type.NORMAL, "uuid");
         account.setId(5L);
@@ -962,7 +966,6 @@
         when(lbVmMapDao.isVmAttachedToLoadBalancer(loadBalancerId)).thenReturn(false);
 
         SearchBuilder<AutoScalePolicyVO> searchBuilderMock = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<AutoScalePolicyVO> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
         when(asScaleUpPolicyMock.getDuration()).thenReturn(scaleUpPolicyDuration);
         when(asScaleUpPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEUP);
         when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
@@ -970,8 +973,8 @@
 
         Mockito.doReturn(searchBuilderMock).when(asPolicyDao).createSearchBuilder();
         when(searchBuilderMock.entity()).thenReturn(asScaleUpPolicyMock);
-        Mockito.doReturn(searchCriteriaMock).when(searchBuilderMock).create();
-        when(asPolicyDao.search(searchCriteriaMock, null)).thenReturn(Arrays.asList(asScaleUpPolicyMock)).thenReturn(Arrays.asList(asScaleDownPolicyMock));
+        Mockito.doReturn(searchCriteriaPolicyMock).when(searchBuilderMock).create();
+        when(asPolicyDao.search(searchCriteriaPolicyMock, null)).thenReturn(Arrays.asList(asScaleUpPolicyMock)).thenReturn(Arrays.asList(asScaleDownPolicyMock));
 
         when(autoScaleVmProfileDao.findById(vmProfileId)).thenReturn(asVmProfileMock);
         Mockito.doReturn(Network.Provider.VirtualRouter).when(autoScaleManagerImplSpy).getLoadBalancerServiceProvider(loadBalancerId);
@@ -1049,7 +1052,6 @@
         when(asVmGroupMock.getLoadBalancerId()).thenReturn(loadBalancerId);
 
         SearchBuilder<AutoScalePolicyVO> searchBuilderMock = Mockito.mock(SearchBuilder.class);
-        SearchCriteria<AutoScalePolicyVO> searchCriteriaMock = Mockito.mock(SearchCriteria.class);
         when(asScaleUpPolicyMock.getDuration()).thenReturn(scaleUpPolicyDuration);
         when(asScaleUpPolicyMock.getAction()).thenReturn(AutoScalePolicy.Action.SCALEUP);
         when(asScaleDownPolicyMock.getDuration()).thenReturn(scaleDownPolicyDuration);
@@ -1057,8 +1059,8 @@
 
         Mockito.doReturn(searchBuilderMock).when(asPolicyDao).createSearchBuilder();
         when(searchBuilderMock.entity()).thenReturn(asScaleUpPolicyMock);
-        Mockito.doReturn(searchCriteriaMock).when(searchBuilderMock).create();
-        when(asPolicyDao.search(searchCriteriaMock, null)).thenReturn(Arrays.asList(asScaleUpPolicyMock)).thenReturn(Arrays.asList(asScaleDownPolicyMock));
+        Mockito.doReturn(searchCriteriaPolicyMock).when(searchBuilderMock).create();
+        when(asPolicyDao.search(searchCriteriaPolicyMock, null)).thenReturn(Arrays.asList(asScaleUpPolicyMock)).thenReturn(Arrays.asList(asScaleDownPolicyMock));
 
         when(lbDao.findById(loadBalancerId)).thenReturn(loadBalancerMock);
         when(autoScaleVmProfileDao.findById(vmProfileId)).thenReturn(asVmProfileMock);
