--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -2399,11 +2399,10 @@
         String expectedMessage = String.format("Required network offering ID [%s] is not in [%s] state.", 1L, NetworkOffering.State.Enabled);
 
         Mockito.doReturn(networkOfferingVoListMock).when(networkOfferingDaoMock).listByAvailability(NetworkOffering.Availability.Required, false);
+        networkOfferingVoMock = createMockNetworkOfferingVO(1L);
         Mockito.doReturn(networkOfferingVoMock).when(networkOfferingVoListMock).get(0);
 
         Mockito.doReturn(NetworkOffering.State.Disabled).when(networkOfferingVoMock).getState();
-
-        Mockito.doReturn(1L).when(networkOfferingVoMock).getId();
 
         InvalidParameterValueException assertThrows = Assert.assertThrows(expectedInvalidParameterValueException, () -> {
             userVmManagerImpl.getOfferingWithRequiredAvailabilityForNetworkCreation();
--- a/server/src/test/java/com/cloud/vm/UserVmManagerTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerTest.java
@@ -554,8 +554,8 @@
         when(_vmDao.findById(anyLong())).thenReturn(_vmMock);
         when(_networkDao.findById(anyLong())).thenReturn(_networkMock);
         doReturn(9L).when(_networkMock).getNetworkOfferingId();
+        _networkOfferingMock = createMockNetworkOfferingVO(10L);
         when(_networkOfferingDao.findByIdIncludingRemoved(anyLong())).thenReturn(_networkOfferingMock);
-        doReturn(10L).when(_networkOfferingMock).getId();
 
         List<Service> services = new ArrayList<Service>();
         services.add(Service.Dhcp);
@@ -643,7 +643,7 @@
     // vm is running in network with dhcp support
     @Test(expected = InvalidParameterValueException.class)
     public void testUpdateVmNicIpFailure1() throws Exception {
-        UpdateVmNicIpCmd cmd = new UpdateVmNicIpCmd();
+        UpdateNicIpCmd cmd = new UpdateVmNicIpCmd();
         Class<?> _class = cmd.getClass();
 
         Field virtualmachineIdField = _class.getDeclaredField("nicId");
@@ -660,8 +660,8 @@
         when(_networkDao.findById(anyLong())).thenReturn(_networkMock);
         when(_networkMock.getState()).thenReturn(Network.State.Implemented);
         doReturn(9L).when(_networkMock).getNetworkOfferingId();
+        _networkOfferingMock = createMockNetworkOfferingVO(10L);
         when(_networkOfferingDao.findByIdIncludingRemoved(anyLong())).thenReturn(_networkOfferingMock);
-        doReturn(10L).when(_networkOfferingMock).getId();
 
         List<Service> services = new ArrayList<Service>();
         services.add(Service.Dhcp);
@@ -697,8 +697,8 @@
         when(_vmDao.findById(anyLong())).thenReturn(_vmMock);
         when(_networkDao.findById(anyLong())).thenReturn(_networkMock);
         doReturn(9L).when(_networkMock).getNetworkOfferingId();
+        _networkOfferingMock = createMockNetworkOfferingVO(10L);
         when(_networkOfferingDao.findByIdIncludingRemoved(anyLong())).thenReturn(_networkOfferingMock);
-        doReturn(10L).when(_networkOfferingMock).getId();
 
         List<Service> services = new ArrayList<Service>();
         services.add(Service.Dhcp);
@@ -744,8 +744,8 @@
         when(_vmDao.findById(anyLong())).thenReturn(_vmMock);
         when(_networkDao.findById(anyLong())).thenReturn(_networkMock);
         doReturn(9L).when(_networkMock).getNetworkOfferingId();
+        _networkOfferingMock = createMockNetworkOfferingVO(10L);
         when(_networkOfferingDao.findByIdIncludingRemoved(anyLong())).thenReturn(_networkOfferingMock);
-        doReturn(10L).when(_networkOfferingMock).getId();
 
         List<Service> services = new ArrayList<Service>();
         services.add(Service.Dhcp);
--- a/server/src/test/java/com/cloud/vm/MockNetworkOfferingVOWithId.java
+++ b/server/src/test/java/com/cloud/vm/MockNetworkOfferingVOWithId.java
@@ -0,0 +1,12 @@
+package com.cloud.vm;
+
+import com.cloud.offerings.NetworkOfferingVO;
+import org.mockito.Mockito;
+
+public class MockNetworkOfferingVOWithId {
+    public static NetworkOfferingVO createMockNetworkOfferingVO(long id) {
+        NetworkOfferingVO networkOfferingVO = Mockito.mock(NetworkOfferingVO.class);
+        Mockito.doReturn(id).when(networkOfferingVO).getId();
+        return networkOfferingVO;
+    }
+}
