--- a/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
+++ b/plugins/network-elements/brocade-vcs/src/test/java/com/cloud/network/guru/BrocadeVcsGuestNetworkGuruTest.java
@@ -81,6 +81,7 @@
 import com.cloud.vm.VirtualMachineProfile;
 
 public class BrocadeVcsGuestNetworkGuruTest {
+    private final DataCenterVO dataCenter = mock(DataCenterVO.class);
     private static final long NETWORK_ID = 42L;
     PhysicalNetworkDao physnetdao = mock(PhysicalNetworkDao.class);
     BrocadeVcsDao vcsdao = mock(BrocadeVcsDao.class);
@@ -352,9 +353,8 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenterVO dc = mock(DataCenterVO.class);
-        when(dest.getDataCenter()).thenReturn(dc);
-        when(dcdao.findById((long)anyInt())).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
+        when(dcdao.findById((long)anyInt())).thenReturn(dataCenter);
 
         final HostVO brocadeHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
@@ -404,9 +404,8 @@
 
         final DeployDestination dest = mock(DeployDestination.class);
 
-        final DataCenterVO dc = mock(DataCenterVO.class);
-        when(dest.getDataCenter()).thenReturn(dc);
-        when(dcdao.findById((long)anyInt())).thenReturn(dc);
+        when(dest.getDataCenter()).thenReturn(dataCenter);
+        when(dcdao.findById((long)anyInt())).thenReturn(dataCenter);
 
         final HostVO brocadeHost = mock(HostVO.class);
         when(hostdao.findById(anyLong())).thenReturn(brocadeHost);
