--- a/server/src/test/java/com/cloud/vm/UserVmManagerTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerTest.java
@@ -455,6 +455,12 @@
         return serviceOffering;
     }
 
+    private static UserVm createMockUserVm() {
+        UserVm userVm = mock(UserVm.class);
+        lenient().when(userVm.getId()).thenReturn(1L);
+        return userVm;
+    }
+
     // Test Move VM b/w accounts where caller is not ROOT/Domain admin
     @Test(expected = InvalidParameterValueException.class)
     public void testMoveVmToUser1() throws Exception {
@@ -774,7 +780,7 @@
 
     @Test
     public void testApplyUserDataInNetworkWithoutUserDataSupport() throws Exception {
-        UserVm userVm = mock(UserVm.class);
+        UserVm userVm = createMockUserVm();
 
         when(_nicMock.getNetworkId()).thenReturn(2L);
         when(_networkMock.getNetworkOfferingId()).thenReturn(3L);
@@ -803,8 +809,7 @@
 
     @Test
     public void testApplyUserDataSuccessful() throws Exception {
-        UserVm userVm = mock(UserVm.class);
-        lenient().when(userVm.getId()).thenReturn(1L);
+        UserVm userVm = createMockUserVm();
 
         when(_nicMock.getNetworkId()).thenReturn(2L);
         when(_networkMock.getNetworkOfferingId()).thenReturn(3L);
