--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -373,7 +373,7 @@
     @Test
     public void testCreateStoragePoolCommand() {
         final StoragePoolVO poolVO = Mockito.mock(StoragePoolVO.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final CreateStoragePoolCommand createStorageCommand = new CreateStoragePoolCommand(false, poolVO);
 
@@ -391,7 +391,7 @@
     @Test
     public void testModifyStoragePoolCommand() {
         final StoragePoolVO poolVO = Mockito.mock(StoragePoolVO.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final ModifyStoragePoolCommand modifyStorageCommand = new ModifyStoragePoolCommand(false, poolVO);
 
@@ -482,7 +482,7 @@
 
     @Test
     public void testGetStorageStatsCommand() {
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
         final DataStoreTO store = Mockito.mock(DataStoreTO.class);
 
         final GetStorageStatsCommand storageStatsCommand = new GetStorageStatsCommand(store);
@@ -500,7 +500,7 @@
 
     @Test
     public void testPrimaryStorageDownloadCommand() {
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
         final StoragePoolVO poolVO = Mockito.mock(StoragePoolVO.class);
 
         final PrimaryStorageDownloadCommand storageDownloadCommand = new PrimaryStorageDownloadCommand("Test", "http://127.0.0.1", ImageFormat.VHD, 1l, poolVO, 200);
@@ -551,7 +551,7 @@
         final String uuid = "befc4dcd-f5c6-4015-8791-3c18622b7c7f";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final MaintainCommand maintainCommand = new MaintainCommand();
 
@@ -698,7 +698,7 @@
     @Test
     public void testSecurityGroupRulesCommand() {
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final String guestIp = "127.0.0.1";
         final String guestIp6 = "2001:db8::cad:40ff:fefd:75c4";
@@ -737,7 +737,7 @@
         final PIF pif = Mockito.mock(PIF.class);
         final PIF.Record pifRec = Mockito.mock(PIF.Record.class);
 
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final OvsFetchInterfaceCommand fetchInterCommand = new OvsFetchInterfaceCommand(label);
 
@@ -774,7 +774,7 @@
         final String bridge = "gre";
         final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final OvsCreateGreTunnelCommand createGreCommand = new OvsCreateGreTunnelCommand("127.0.0.1", "KEY", 1l, 2l);
 
@@ -945,7 +945,7 @@
     @Test
     public void testCleanupNetworkRulesCmd() {
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final CleanupNetworkRulesCmd cleanupNets = new CleanupNetworkRulesCmd(20);
 
@@ -968,7 +968,7 @@
     @Test
     public void testCleanupNetworkRulesCmdLTZ() {
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final CleanupNetworkRulesCmd cleanupNets = new CleanupNetworkRulesCmd(20);
 
@@ -1413,7 +1413,7 @@
         final String uuid = "6172d8b7-ba10-4a70-93f9-ecaf41f51d53";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final Pool pool = Mockito.mock(Pool.class);
         final Pool.Record poolr = Mockito.mock(Pool.Record.class);
@@ -1616,7 +1616,7 @@
 
         final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
         final Host host = Mockito.mock(Host.class);
 
         final ScaleVmCommand scaleVm = new ScaleVmCommand(machineTO);
--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -123,7 +123,7 @@
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network networkForSm = Mockito.mock(Network.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final MigrateWithStorageCommand migrateStorageCommand = new MigrateWithStorageCommand(vmSpec, volumeToFiler);
 
@@ -196,7 +196,7 @@
 
         final XsLocalNetwork nativeNetworkForTraffic = Mockito.mock(XsLocalNetwork.class);
         final Network network = Mockito.mock(Network.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final Network nw1 = Mockito.mock(Network.class);
         final Network nw2 = Mockito.mock(Network.class);
@@ -412,7 +412,7 @@
         final String uuid = "206b21a7-c6ec-40e2-b5e2-f861b9612f04";
 
         final Connection conn = Mockito.mock(Connection.class);
-        final XsHost xsHost = Mockito.mock(XsHost.class);
+        final XsHost xsHost = MockXsHost.createMockXsHost();
 
         final VirtualMachineTO vm = Mockito.mock(VirtualMachineTO.class);
 
--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/MockXsHost.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/MockXsHost.java
@@ -0,0 +1,13 @@
+package com.cloud.hypervisor.xenserver.resource.wrapper.xenbase;
+
+import com.cloud.hypervisor.xenserver.resource.XsHost;
+import org.mockito.Mockito;
+
+public final class MockXsHost {
+    private MockXsHost() {
+    }
+
+    public static XsHost createMockXsHost() {
+        return Mockito.mock(XsHost.class);
+    }
+}
