--- a/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/NetworkACLServiceImplTest.java
@@ -1097,6 +1097,10 @@
                 Mockito.anyList());
     }
 
+    private static void configureNetworkACLItemVO(NetworkACLItemVO mock, int number) {
+        Mockito.when(mock.getNumber()).thenReturn(number);
+    }
+
     private void configureMoveMethodsToDoNothing() {
         Mockito.doReturn(networkACLVOMock).when(networkAclDaoMock).acquireInLockTable(Mockito.anyLong());
         Mockito.doReturn(true).when(networkAclDaoMock).releaseFromLockTable(Mockito.anyLong());
@@ -1281,8 +1285,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void moveRuleBetweenAclRulesTestThereIsSpaceBetweenPreviousRuleAndNextRuleToAccommodateTheNewRuleWithOtherRuleColliding() {
-        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
-        Mockito.when(nextAclRuleMock.getNumber()).thenReturn(15);
+        configureNetworkACLItemVO(previousAclRuleMock, 10);
+        configureNetworkACLItemVO(nextAclRuleMock, 15);
 
         ArrayList<NetworkACLItemVO> allAclRules = new ArrayList<>();
         NetworkACLItemVO networkACLItemVO1 = new NetworkACLItemVO();
@@ -1297,8 +1301,8 @@
 
     @Test
     public void moveRuleBetweenAclRulesTestThereIsSpaceBetweenPreviousRuleAndNextRuleToAccommodateTheNewRule() {
-        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
-        Mockito.when(nextAclRuleMock.getNumber()).thenReturn(11);
+        configureNetworkACLItemVO(previousAclRuleMock, 10);
+        configureNetworkACLItemVO(nextAclRuleMock, 11);
         Mockito.when(aclRuleBeingMovedMock.getId()).thenReturn(1l);
 
         ArrayList<NetworkACLItemVO> allAclRules = new ArrayList<>();
@@ -1329,8 +1333,8 @@
 
     @Test
     public void moveRuleBetweenAclRulesTestThereIsNoSpaceBetweenPreviousRuleAndNextRuleToAccommodateTheNewRule() {
-        Mockito.when(previousAclRuleMock.getNumber()).thenReturn(10);
-        Mockito.when(nextAclRuleMock.getNumber()).thenReturn(15);
+        configureNetworkACLItemVO(previousAclRuleMock, 10);
+        configureNetworkACLItemVO(nextAclRuleMock, 15);
         Mockito.when(aclRuleBeingMovedMock.getNumber()).thenReturn(50);
         Mockito.when(aclRuleBeingMovedMock.getId()).thenReturn(1l);
 
@@ -1354,7 +1358,7 @@
     @Test
     public void updateAclRuleToNewPositionAndExecuteShiftIfNecessaryTest() {
 
-        Mockito.when(nextAclRuleMock.getNumber()).thenReturn(11);
+        configureNetworkACLItemVO(nextAclRuleMock, 11);
         Mockito.when(nextAclRuleMock.getId()).thenReturn(50l);
 
         Mockito.when(aclRuleBeingMovedMock.getNumber()).thenReturn(50);
