--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResourceTest.java
@@ -272,6 +272,7 @@
     @Mock
     private VolumeObjectTO volumeObjectToMock;
 
+    private final Volume volumeMock = Mockito.mock(Volume.class);
 
     @Spy
     private LibvirtComputingResource libvirtComputingResourceSpy = Mockito.spy(new LibvirtComputingResource());
@@ -2257,10 +2258,9 @@
     @Test
     public void testDestroyCommand() {
         final StoragePool pool = Mockito.mock(StoragePool.class);
-        final Volume volume = Mockito.mock(Volume.class);
         final String vmName = "Test";
 
-        final DestroyCommand command = new DestroyCommand(pool, volume, vmName);
+        final DestroyCommand command = new DestroyCommand(pool, volumeMock, vmName);
 
         final KVMStoragePoolManager poolManager = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primary = Mockito.mock(KVMStoragePool.class);
@@ -2284,10 +2284,9 @@
     @Test
     public void testDestroyCommandError() {
         final StoragePool pool = Mockito.mock(StoragePool.class);
-        final Volume volume = Mockito.mock(Volume.class);
         final String vmName = "Test";
 
-        final DestroyCommand command = new DestroyCommand(pool, volume, vmName);
+        final DestroyCommand command = new DestroyCommand(pool, volumeMock, vmName);
 
         final KVMStoragePoolManager poolManager = Mockito.mock(KVMStoragePoolManager.class);
         final KVMStoragePool primary = Mockito.mock(KVMStoragePool.class);
