--- a/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/resource/ResourceCleanupServiceImplTest.java
@@ -160,6 +160,12 @@
         }
     }
 
+    private static VolumeVO createMockVolumeVO() {
+        VolumeVO vol2 = Mockito.mock(VolumeVO.class);
+        Mockito.when(vol2.getId()).thenReturn(2L);
+        return vol2;
+    }
+
 
     @Test
     public void testPurgeLinkedSnapshotEntitiesNoSnapshots() {
@@ -362,8 +368,7 @@
         VolumeVO vol1 = Mockito.mock(VolumeVO.class);
         Mockito.when(vol1.getId()).thenReturn(1L);
         Mockito.when(vol1.getInstanceId()).thenReturn(1L);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class);
-        Mockito.when(vol2.getId()).thenReturn(2L);
+        VolumeVO vol2 = createMockVolumeVO();
         Mockito.when(volumeDao.searchRemovedByVms(ids, null)).thenReturn(List.of(vol1, vol2));
         SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
         Mockito.when(snapshotVO.getVolumeId()).thenReturn(1L);
