--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -236,6 +236,8 @@
     @Mock
     private DataCenterVO _dcMock;
 
+    private DataCenterVO zoneMock;
+
     @Mock
     protected NicDao nicDao;
 
@@ -527,6 +529,7 @@
 
     @Before
     public void beforeTest() {
+        zoneMock = mock(DataCenterVO.class);
         userVmManagerImpl.resourceLimitService = resourceLimitMgr;
 
         Mockito.when(updateVmCommand.getId()).thenReturn(vmId);
@@ -3574,8 +3577,7 @@
         Account owner = mock(Account.class);
         when(accountService.getActiveAccountById(accountId)).thenReturn(owner);
 
-        DataCenterVO zone = mock(DataCenterVO.class);
-        when(_dcDao.findById(zoneId)).thenReturn(zone);
+        when(_dcDao.findById(zoneId)).thenReturn(zoneMock);
 
         BackupVO backup = mock(BackupVO.class);
         when(backup.getZoneId()).thenReturn(zoneId);
@@ -3750,8 +3752,7 @@
         Account owner = mock(Account.class);
         when(accountService.getActiveAccountById(accountId)).thenReturn(owner);
 
-        DataCenterVO zone = mock(DataCenterVO.class);
-        when(_dcDao.findById(zoneId)).thenReturn(zone);
+        when(_dcDao.findById(zoneId)).thenReturn(zoneMock);
 
         BackupVO backup = mock(BackupVO.class);
         when(backup.getZoneId()).thenReturn(zoneId);
@@ -4251,8 +4252,7 @@
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
         when(_serviceOfferingDao.findById(serviceOfferingId)).thenReturn(serviceOffering);
 
-        DataCenterVO zone = mock(DataCenterVO.class);
-        when(_dcDao.findById(zoneId)).thenReturn(zone);
+        when(_dcDao.findById(zoneId)).thenReturn(zoneMock);
 
         BackupVO backup = mock(BackupVO.class);
         when(backup.getZoneId()).thenReturn(zoneId);
@@ -4318,8 +4318,7 @@
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
         when(_serviceOfferingDao.findById(serviceOfferingId)).thenReturn(serviceOffering);
 
-        DataCenterVO zone = mock(DataCenterVO.class);
-        when(_dcDao.findById(zoneId)).thenReturn(zone);
+        when(_dcDao.findById(zoneId)).thenReturn(zoneMock);
 
         BackupVO backup = mock(BackupVO.class);
         when(backup.getZoneId()).thenReturn(zoneId);
