--- a/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
+++ b/plugins/network-elements/elastic-loadbalancer/src/test/java/com/cloud/network/lb/ElasticLoadBalancerManagerImplTest.java
@@ -60,8 +60,7 @@
         when(answerMock.getResult()).thenReturn(false);
         VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
-        Commands cmds = mock(Commands.class);
-        when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
+        Commands cmds = createMockCommands(answerMock);
         ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = false;
@@ -76,8 +75,7 @@
         when(answerMock.getResult()).thenReturn(true);
         VirtualMachineProfile profileMock = mock(VirtualMachineProfile.class);
         long hostId = 1L;
-        Commands cmds = mock(Commands.class);
-        when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
+        Commands cmds = createMockCommands(answerMock);
         ReservationContext context = mock(ReservationContext.class);
 
         boolean expected = true;
@@ -98,4 +96,10 @@
             fail();
         }
     }
+
+    private static Commands createMockCommands(CheckSshAnswer answerMock) {
+        Commands cmds = mock(Commands.class);
+        when(cmds.getAnswer("checkSsh")).thenReturn(answerMock);
+        return cmds;
+    }
 }
