--- a/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/vpc/VpcManagerImplTest.java
@@ -174,6 +174,7 @@
 
     public static final long ACCOUNT_ID = 1;
     private AccountVO account;
+    private Account sharedAccountMock;
     private UserVO user;
 
     private static final String IP4_GATEWAY = "10.0.16.1";
@@ -208,6 +209,7 @@
     @Before
     public void setup() throws NoSuchFieldException, IllegalAccessException {
         closeable = MockitoAnnotations.openMocks(this);
+        sharedAccountMock = Mockito.mock(Account.class);
         manager = new VpcManagerImpl();
         manager._vpcOffSvcMapDao = vpcOfferingServiceMapDao;
         manager.vpcDao = vpcDao;
@@ -365,7 +367,6 @@
         manager._maxNetworks = 3;
         VpcVO vpcMockVO = Mockito.mock(VpcVO.class);
         Vpc vpcMock = Mockito.mock(Vpc.class);
-        Account accountMock = Mockito.mock(Account.class);
         PhysicalNetwork physicalNetwork = Mockito.mock(PhysicalNetwork.class);
         NetworkOffering offering = Mockito.mock(NetworkOffering.class);
         List<Network.Service> services = new ArrayList<>();
@@ -385,16 +386,16 @@
         Mockito.when(vpcMock.getCidr()).thenReturn("10.0.0.0/8");
         Mockito.when(vpcMock.getNetworkDomain()).thenReturn("cs1cloud.internal");
 
-        manager.validateNewVpcGuestNetwork("10.10.10.0/24", "10.10.10.1", accountMock, vpcMock, "cs1cloud.internal");
+        manager.validateNewVpcGuestNetwork("10.10.10.0/24", "10.10.10.1", sharedAccountMock, vpcMock, "cs1cloud.internal");
         manager.validateNtwkOffForNtwkInVpc(2L, 1, "10.10.10.0/24", "111-", vpcMock, "10.1.1.1", new AccountVO(), null);
         manager.validateNtwkOffForVpc(offering, services);
         manager.createVpcGuestNetwork(1L, "vpcNet1", "vpc tier 1", null,
-                "10.10.10.0/24", null, null, accountMock, null, physicalNetwork,
-                1L, null, null, 1L, null, accountMock,
+                "10.10.10.0/24", null, null, sharedAccountMock, null, physicalNetwork,
+                1L, null, null, 1L, null, sharedAccountMock,
                 true, null, null, null, null, null, null, null, new Pair<>(1000, 1000), null);
 
         Mockito.verify(networkMgr, times(1)).createGuestNetwork(1L, "vpcNet1", "vpc tier 1", null,
-                "10.10.10.0/24", null, false, "cs1cloud.internal", accountMock, null,
+                "10.10.10.0/24", null, false, "cs1cloud.internal", sharedAccountMock, null,
                 physicalNetwork, zoneId, null, null, 1L, null, null,
                 true, null, null, null, null,
                 null, null, null, null, null, new Pair<>(1000, 1000), null);
@@ -405,7 +406,6 @@
         long vpcId = 1L;
         Integer publicMtu = 1450;
         String sourceNatIp = "1.2.3.4";
-        Account accountMock = Mockito.mock(Account.class);
         VpcVO vpcVO = new VpcVO();
 
         Answer answer = Mockito.mock(Answer.class);
