--- a/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
+++ b/plugins/network-elements/cisco-vnmc/src/test/java/com/cloud/network/element/CiscoVnmcElementTest.java
@@ -221,8 +221,7 @@
         List<VlanVO> vlanVOList = new ArrayList<VlanVO>();
         when(_vlanDao.listVlansByPhysicalNetworkId(network.getPhysicalNetworkId())).thenReturn(vlanVOList);
 
-        Answer answer = mock(Answer.class);
-        when(answer.getResult()).thenReturn(true);
+        Answer answer = createMockAnswer();
 
         when(_agentMgr.easySend(anyLong(), any(CreateLogicalEdgeFirewallCommand.class))).thenReturn(answer);
         when(_agentMgr.easySend(anyLong(), any(ConfigureNexusVsmForAsaCommand.class))).thenReturn(answer);
@@ -254,8 +253,7 @@
         when(hostVO.getId()).thenReturn(1L);
         when(_hostDao.findById(anyLong())).thenReturn(hostVO);
 
-        Answer answer = mock(Answer.class);
-        when(answer.getResult()).thenReturn(true);
+        Answer answer = createMockAnswer();
 
         when(_agentMgr.easySend(anyLong(), any(CleanupLogicalEdgeFirewallCommand.class))).thenReturn(answer);
 
@@ -298,8 +296,7 @@
         List<FirewallRule> rules = new ArrayList<FirewallRule>();
         rules.add(rule);
 
-        Answer answer = mock(Answer.class);
-        when(answer.getResult()).thenReturn(true);
+        Answer answer = createMockAnswer();
 
         when(_agentMgr.easySend(anyLong(), any(SetFirewallRulesCommand.class))).thenReturn(answer);
 
@@ -348,8 +345,7 @@
         List<PortForwardingRule> rules = new ArrayList<PortForwardingRule>();
         rules.add(rule);
 
-        Answer answer = mock(Answer.class);
-        when(answer.getResult()).thenReturn(true);
+        Answer answer = createMockAnswer();
 
         when(_agentMgr.easySend(anyLong(), any(SetPortForwardingRulesCommand.class))).thenReturn(answer);
 
@@ -406,4 +402,10 @@
 
         assertTrue(_element.applyStaticNats(network, rules));
     }
+
+    private static Answer createMockAnswer() {
+        Answer answer = mock(Answer.class);
+        when(answer.getResult()).thenReturn(true);
+        return answer;
+    }
 }
