--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -1074,7 +1074,7 @@
         };
         UserVmJoinVO userVm = new UserVmJoinVO();
         NetworkVO networkVO = mock(NetworkVO.class);
-        AccountVO accountVO = mock(AccountVO.class);
+        AccountVO accountVO = MockAccountVO.createMockAccountVO(2L);
         DomainVO domainVO = mock(DomainVO.class);
         domainVO.setName("testDomain");
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1097,7 +1097,6 @@
         Mockito.when(dcDao.findById(anyLong())).thenReturn(dataCenterVO);
         Mockito.when(vpcDao.findById(anyLong())).thenReturn(vpcVO);
         Mockito.when(dataCenterVO.getId()).thenReturn(1L);
-        when(accountVO.getId()).thenReturn(2L);
         Mockito.when(domainVO.getId()).thenReturn(3L);
         Mockito.when(vpcVO.getId()).thenReturn(4L);
         Mockito.when(networkVO.getId()).thenReturn(5L);
--- a/engine/orchestration/src/test/java/com/cloud/vm/MockAccountVO.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/MockAccountVO.java
@@ -0,0 +1,14 @@
+package com.cloud.vm;
+
+import com.cloud.user.AccountVO;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockAccountVO {
+    public static AccountVO createMockAccountVO(long accountId) {
+        AccountVO account = Mockito.mock(AccountVO.class);
+        when(account.getId()).thenReturn(accountId);
+        return account;
+    }
+}
--- a/server/src/test/java/com/cloud/vm/MockAccountVO.java
+++ b/server/src/test/java/com/cloud/vm/MockAccountVO.java
@@ -0,0 +1,14 @@
+package com.cloud.vm;
+
+import com.cloud.user.AccountVO;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockAccountVO {
+    public static AccountVO createMockAccountVO(long accountId) {
+        AccountVO account = Mockito.mock(AccountVO.class);
+        when(account.getId()).thenReturn(accountId);
+        return account;
+    }
+}
