--- a/server/src/test/java/com/cloud/template/HypervisorTemplateAdapterTest.java
+++ b/server/src/test/java/com/cloud/template/HypervisorTemplateAdapterTest.java
@@ -274,9 +274,8 @@
         when(template.getAccountId()).thenReturn(1l);
         when(template.getUuid()).thenReturn("Test UUID"); //TODO possibly return this from method for comparison, if things work how i want
 
-        TemplateProfile profile = mock(TemplateProfile.class);
+        TemplateProfile profile = createMockTemplateProfile();
         when(profile.getTemplate()).thenReturn(template);
-        when(profile.getZoneIdList()).thenReturn(null);
 
         TemplateApiResult result = mock(TemplateApiResult.class);
         when(result.isSuccess()).thenReturn(true);
@@ -320,10 +319,8 @@
 
     @Test
     public void createTemplateWithinZonesTestZoneIdsNullShouldCallListAllZones() {
-        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
+        TemplateProfile templateProfileMock = createMockTemplateProfile();
         VMTemplateVO vmTemplateVOMock = Mockito.mock(VMTemplateVO.class);
-
-        Mockito.when(templateProfileMock.getZoneIdList()).thenReturn(null);
 
         _adapter.createTemplateWithinZones(templateProfileMock, vmTemplateVOMock);
 
@@ -555,4 +552,10 @@
         Mockito.when(dataStoreManagerMock.getImageStoresByScope(Mockito.any())).thenReturn(new ArrayList<>());
         _adapter.checkZoneImageStores(templateVO, List.of(1L));
     }
+
+    private static TemplateProfile createMockTemplateProfile() {
+        TemplateProfile templateProfileMock = Mockito.mock(TemplateProfile.class);
+        Mockito.when(templateProfileMock.getZoneIdList()).thenReturn(null);
+        return templateProfileMock;
+    }
 }
