--- a/engine/storage/image/src/test/java/org/apache/cloudstack/storage/image/TemplateServiceImplTest.java
+++ b/engine/storage/image/src/test/java/org/apache/cloudstack/storage/image/TemplateServiceImplTest.java
@@ -104,10 +104,13 @@
     @Mock
     DataCenterDao _dcDao;
 
+    private DataCenterVO destinationZoneMock;
+
     Map<String, TemplateProp> templatesInSourceStore = new HashMap<>();
 
     @Before
     public void setUp() {
+        destinationZoneMock = Mockito.mock(DataCenterVO.class);
         Long zoneId = 1L;
         Mockito.doReturn(2L).when(tmpltMock).getId();
         Mockito.doReturn("unique-name").when(tmpltMock).getUniqueName();
@@ -248,8 +251,7 @@
         Mockito.doReturn(sourceTmplMock).when(templateDataFactoryMock).getTemplate(100L, otherZoneStoreMock);
         Mockito.doReturn("/mnt/secondary/template.qcow2").when(sourceTmplMock).getInstallPath();
 
-        DataCenterVO dstZoneMock = Mockito.mock(DataCenterVO.class);
-        Mockito.doReturn(dstZoneMock).when(_dcDao).findById(1L);
+        Mockito.doReturn(destinationZoneMock).when(_dcDao).findById(1L);
         Mockito.doReturn(true).when(templateManagerMock).copy(Mockito.anyLong(), Mockito.any(), Mockito.any(), Mockito.any());
 
         boolean result = templateService.tryCopyingTemplateToImageStore(tmpltMock, destStoreMock);
@@ -310,8 +312,7 @@
         Scope sourceScopeMock = Mockito.mock(Scope.class);
         Mockito.doReturn(sourceStoreMock).when(sourceTmplMock).getDataStore();
 
-        DataCenterVO dstZoneMock = Mockito.mock(DataCenterVO.class);
-        Mockito.doReturn(dstZoneMock).when(_dcDao).findById(1L);
+        Mockito.doReturn(destinationZoneMock).when(_dcDao).findById(1L);
         VMTemplateVO templateVoMock = Mockito.mock(VMTemplateVO.class);
         Mockito.doReturn(templateVoMock).when(templateDao).findById(100L);
 
