--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -118,6 +118,7 @@
     @InjectMocks
     private VolumeOrchestrator volumeOrchestrator = new VolumeOrchestrator();
 
+    private VolumeVO persistedVolume;
     private static final Long DEFAULT_ACCOUNT_PS_RESOURCE_COUNT = 100L;
     private Long accountPSResourceCount;
     private static final long MOCK_VM_ID = 202L;
@@ -127,6 +128,7 @@
     @Before
     public void setUp() throws Exception {
         accountPSResourceCount = DEFAULT_ACCOUNT_PS_RESOURCE_COUNT;
+        persistedVolume = Mockito.mock(VolumeVO.class);
         Mockito.when(resourceLimitMgr.recalculateResourceCount(Mockito.anyLong(), Mockito.anyLong(), Mockito.anyInt())).thenReturn(new ArrayList<>());
         Mockito.doAnswer((Answer<Void>) invocation -> {
             Resource.ResourceType type = (Resource.ResourceType)invocation.getArguments()[1];
@@ -241,8 +243,7 @@
         MockedConstruction<VolumeVO> volumeVOMockedConstructionConstruction = Mockito.mockConstruction(VolumeVO.class, (mock, context) -> {
         });
 
-        VolumeVO volumeVO = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeDao.persist(Mockito.any(VolumeVO.class))).thenReturn(volumeVO);
+        Mockito.when(volumeDao.persist(Mockito.any(VolumeVO.class))).thenReturn(persistedVolume);
 
         volumeOrchestrator.importVolume(volumeType, name, diskOffering, sizeInBytes, null, null,
                 zoneId, hypervisorType, null, null, owner,
@@ -282,8 +283,7 @@
         Mockito.when(oldVol.getPassphraseId()).thenReturn(null); // no encryption
         Mockito.when(oldVol.getKmsKeyId()).thenReturn(null); // no encryption
 
-        VolumeVO persistedVol = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeDao.persist(Mockito.any(VolumeVO.class))).thenReturn(persistedVol);
+        Mockito.when(volumeDao.persist(Mockito.any(VolumeVO.class))).thenReturn(persistedVolume);
 
         VolumeVO result = volumeOrchestrator.allocateDuplicateVolumeVO(oldVol, null, null);
         assertNotNull(result);
@@ -316,8 +316,7 @@
         Mockito.when(passphrase.getId()).thenReturn(999L);
         Mockito.when(passphraseDao.persist(Mockito.any())).thenReturn(passphrase);
 
-        VolumeVO persistedVol = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeDao.persist(Mockito.any())).thenReturn(persistedVol);
+        Mockito.when(volumeDao.persist(Mockito.any())).thenReturn(persistedVolume);
 
         VolumeVO result = volumeOrchestrator.allocateDuplicateVolumeVO(oldVol, null, null);
         assertNotNull(result);
@@ -340,8 +339,7 @@
         Mockito.when(oldVol.getMaxIops()).thenReturn(250L);
         Mockito.when(oldVol.get_iScsiName()).thenReturn("iqn.tmpl");
 
-        VolumeVO persistedVol = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeDao.persist(Mockito.any())).thenReturn(persistedVol);
+        Mockito.when(volumeDao.persist(Mockito.any())).thenReturn(persistedVolume);
 
         VolumeVO result = volumeOrchestrator.allocateDuplicateVolumeVO(oldVol, null, 222L);
         assertNotNull(result);
@@ -370,8 +368,7 @@
         DiskOffering diskOffering = Mockito.mock(DiskOffering.class);
         Mockito.when(diskOffering.getEncrypt()).thenReturn(false); // explicitly disables encryption
 
-        VolumeVO persistedVol = Mockito.mock(VolumeVO.class);
-        Mockito.when(volumeDao.persist(Mockito.any())).thenReturn(persistedVol);
+        Mockito.when(volumeDao.persist(Mockito.any())).thenReturn(persistedVolume);
 
         VolumeVO result = volumeOrchestrator.allocateDuplicateVolumeVO(oldVol, diskOffering, null);
         assertNotNull(result);
