--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -519,12 +519,8 @@
 
     @Test
     public void testCanVmRestartOnAnotherServerAllShared() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class);
-        Mockito.when(vol1.getPoolId()).thenReturn(10L);
-        Mockito.when(vol2.getPoolId()).thenReturn(20L);
-        Mockito.when(vol1.isRecreatable()).thenReturn(false);
-        Mockito.when(vol2.isRecreatable()).thenReturn(false);
+        VolumeVO vol1 = createMockVolumeVO(10L, false);
+        VolumeVO vol2 = createMockVolumeVO(20L, false);
 
 
         StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
@@ -542,12 +538,8 @@
 
     @Test
     public void testCanVmRestartOnAnotherServerOneLocalNotRecreatable() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class); // Local, not recreatable
-        Mockito.when(vol1.getPoolId()).thenReturn(10L);
-        Mockito.when(vol2.getPoolId()).thenReturn(30L);
-        Mockito.when(vol1.isRecreatable()).thenReturn(false);
-        Mockito.when(vol2.isRecreatable()).thenReturn(false); // Not recreatable
+        VolumeVO vol1 = createMockVolumeVO(10L, false);
+        VolumeVO vol2 = createMockVolumeVO(30L, false); // Local, not recreatable
 
         StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO pool2 = Mockito.mock(StoragePoolVO.class);
@@ -564,12 +556,8 @@
 
     @Test
     public void testCanVmRestartOnAnotherServerOneLocalRecreatable() {
-        VolumeVO vol1 = Mockito.mock(VolumeVO.class);
-        VolumeVO vol2 = Mockito.mock(VolumeVO.class); // Local, but recreatable
-        Mockito.when(vol1.getPoolId()).thenReturn(10L);
-        Mockito.when(vol2.getPoolId()).thenReturn(30L);
-        Mockito.when(vol1.isRecreatable()).thenReturn(false);
-        Mockito.when(vol2.isRecreatable()).thenReturn(true); // Recreatable
+        VolumeVO vol1 = createMockVolumeVO(10L, false);
+        VolumeVO vol2 = createMockVolumeVO(30L, true); // Local, but recreatable
 
         StoragePoolVO pool1 = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO pool2 = Mockito.mock(StoragePoolVO.class);
@@ -581,6 +569,13 @@
 
         assertTrue("VM restart should be true if local disk is recreatable",
                 volumeOrchestrator.canVmRestartOnAnotherServer(MOCK_VM_ID));
+    }
+
+    private static VolumeVO createMockVolumeVO(Long poolId, boolean recreatable) {
+        VolumeVO vol2 = Mockito.mock(VolumeVO.class);
+        Mockito.when(vol2.getPoolId()).thenReturn(poolId);
+        Mockito.when(vol2.isRecreatable()).thenReturn(recreatable);
+        return vol2;
     }
 
     private void overrideDefaultConfigValue(final ConfigKey configKey, final String value) throws IllegalAccessException, NoSuchFieldException {
