--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -339,6 +339,12 @@
         }
     }
 
+    private static VirtualMachineTemplate createMockVirtualMachineTemplate(boolean isDeployAsIs) {
+        VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
+        when(template.isDeployAsIs()).thenReturn(isDeployAsIs);
+        return template;
+    }
+
     private ClvmPoolManager injectMockedClvmPoolManager() {
         ClvmPoolManager clvmPoolManagerMock = mock(ClvmPoolManager.class);
         ReflectionTestUtils.setField(virtualMachineManagerImpl, "clvmPoolManager", clvmPoolManagerMock);
@@ -1132,8 +1138,7 @@
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
-        VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
-        when(template.isDeployAsIs()).thenReturn(false);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate(false);
 
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         when(plan.getDataCenterId()).thenReturn(1L);
@@ -1228,8 +1233,7 @@
 
         ServiceOfferingVO serviceOffering = mock(ServiceOfferingVO.class);
 
-        VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
-        when(template.isDeployAsIs()).thenReturn(false);
+        VirtualMachineTemplate template = createMockVirtualMachineTemplate(false);
 
         DataCenterDeployment plan = mock(DataCenterDeployment.class);
         when(plan.getDataCenterId()).thenReturn(1L);
