--- a/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/cluster/ClusterDrsServiceImplTest.java
@@ -791,8 +791,7 @@
 
     @Test
     public void testGetBestMigration() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        ClusterVO cluster = createMockCluster();
 
         HostVO destHost = Mockito.mock(HostVO.class);
         Mockito.when(destHost.getClusterId()).thenReturn(1L);
@@ -864,8 +863,7 @@
 
     @Test
     public void testGetBestMigrationDifferentCluster() throws ConfigurationException {
-        ClusterVO cluster = Mockito.mock(ClusterVO.class);
-        Mockito.when(cluster.getId()).thenReturn(1L);
+        ClusterVO cluster = createMockCluster();
 
         HostVO destHost = Mockito.mock(HostVO.class);
         Mockito.when(destHost.getClusterId()).thenReturn(2L);
@@ -920,6 +918,12 @@
 
         assertNull(bestMigration.second());
         assertNull(bestMigration.first());
+    }
+
+    private static ClusterVO createMockCluster() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.when(cluster.getId()).thenReturn(1L);
+        return cluster;
     }
 
     @Test
