--- a/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
+++ b/plugins/deployment-planners/implicit-dedication/src/test/java/org/apache/cloudstack/implicitplanner/ImplicitPlannerTest.java
@@ -418,13 +418,16 @@
         when(vmInstanceDao.listByLastHostId(7L)).thenReturn(stoppedVmsForHost);
 
         // Mock the offering with which the vm was created.
-        ServiceOfferingVO offeringForVmOfThisAccount = mock(ServiceOfferingVO.class);
+        ServiceOfferingVO offeringForVmOfThisAccount = createMockServiceOfferingVO(planner.getName());
         when(serviceOfferingDao.findByIdIncludingRemoved(offeringIdForVmsOfThisAccount)).thenReturn(offeringForVmOfThisAccount);
-        when(offeringForVmOfThisAccount.getDeploymentPlanner()).thenReturn(planner.getName());
-
-        ServiceOfferingVO offeringForVMOfOtherAccount = mock(ServiceOfferingVO.class);
-        when(serviceOfferingDao.findByIdIncludingRemoved(offeringIdForVmsOfOtherAccount)).thenReturn(offeringForVMOfOtherAccount);
-        when(offeringForVMOfOtherAccount.getDeploymentPlanner()).thenReturn("FirstFitPlanner");
+
+        when(serviceOfferingDao.findByIdIncludingRemoved(offeringIdForVmsOfOtherAccount)).thenReturn(createMockServiceOfferingVO("FirstFitPlanner"));
+    }
+
+    private static ServiceOfferingVO createMockServiceOfferingVO(String deploymentPlanner) {
+        ServiceOfferingVO offering = mock(ServiceOfferingVO.class);
+        when(offering.getDeploymentPlanner()).thenReturn(deploymentPlanner);
+        return offering;
     }
 
     @Configuration
