--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java
@@ -153,6 +153,7 @@
     private RebootAnswer rebootAnswer;
     @Mock
     private CreateAnswer createAnswer;
+    private final Connection connection = Mockito.mock(Connection.class);
 
     @Test
     public void testWrapperInstance() {
@@ -653,17 +654,16 @@
     @Test
     public void testOvsSetTagAndFlowCommand() {
         final Network network = Mockito.mock(Network.class);
-        final Connection conn = Mockito.mock(Connection.class);
 
         final OvsSetTagAndFlowCommand tagAndFlowCommand = new OvsSetTagAndFlowCommand("Test", "tag", "vlan://1", "123", 1l);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.setupvSwitchNetwork(conn)).thenReturn(network);
-        try {
-            when(network.getBridge(conn)).thenReturn("br0");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.setupvSwitchNetwork(connection)).thenReturn(network);
+        try {
+            when(network.getBridge(connection)).thenReturn("br0");
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
         } catch (final XenAPIException e) {
@@ -675,7 +675,7 @@
         final Answer answer = wrapper.execute(tagAndFlowCommand, citrixResourceBase);
 
         verify(citrixResourceBase, times(1)).getConnection();
-        verify(citrixResourceBase, times(1)).setupvSwitchNetwork(conn);
+        verify(citrixResourceBase, times(1)).setupvSwitchNetwork(connection);
         verify(citrixResourceBase, times(1)).setIsOvs(true);
 
         assertFalse(answer.getResult());
@@ -697,7 +697,6 @@
 
     @Test
     public void testSecurityGroupRulesCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final String guestIp = "127.0.0.1";
@@ -716,7 +715,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
 
         final Answer answer = wrapper.execute(sshCommand, citrixResourceBase);
@@ -731,7 +730,6 @@
         final String label = "[abc]";
         final String uuid = "befc4dcd-f5c6-4015-8791-3c18622b7c7f";
 
-        final Connection conn = Mockito.mock(Connection.class);
         final XsLocalNetwork network = Mockito.mock(XsLocalNetwork.class);
         final Network network2 = Mockito.mock(Network.class);
         final PIF pif = Mockito.mock(PIF.class);
@@ -746,16 +744,16 @@
 
         when(citrixResourceBase.isXcp()).thenReturn(true);
         when(citrixResourceBase.getLabel()).thenReturn("[abc]");
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
 
         try {
             when(network.getNetwork()).thenReturn(network2);
-            when(network.getPif(conn)).thenReturn(pif);
-            when(network.getPif(conn)).thenReturn(pif);
-            when(pif.getRecord(conn)).thenReturn(pifRec);
-            when(network.getNetwork().getUuid(conn)).thenReturn(uuid);
-            when(citrixResourceBase.getNetworkByName(conn, label)).thenReturn(network);
+            when(network.getPif(connection)).thenReturn(pif);
+            when(network.getPif(connection)).thenReturn(pif);
+            when(pif.getRecord(connection)).thenReturn(pifRec);
+            when(network.getNetwork().getUuid(connection)).thenReturn(uuid);
+            when(citrixResourceBase.getNetworkByName(connection, label)).thenReturn(network);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
@@ -772,7 +770,6 @@
     @Test
     public void testOvsCreateGreTunnelCommand() {
         final String bridge = "gre";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
 
@@ -781,13 +778,13 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
-        when(citrixResourceBase.setupvSwitchNetwork(conn)).thenReturn(network);
-        try {
-            when(network.getBridge(conn)).thenReturn(bridge);
+        when(citrixResourceBase.setupvSwitchNetwork(connection)).thenReturn(network);
+        try {
+            when(network.getBridge(connection)).thenReturn(bridge);
             when(
-                            citrixResourceBase.callHostPlugin(conn, "ovsgre", "ovs_create_gre", "bridge", bridge, "remoteIP", createGreCommand.getRemoteIp(), "greKey",
+                            citrixResourceBase.callHostPlugin(connection, "ovsgre", "ovs_create_gre", "bridge", bridge, "remoteIP", createGreCommand.getRemoteIp(), "greKey",
                                             createGreCommand.getKey(), "from", Long.toString(createGreCommand.getFrom()), "to", Long.toString(createGreCommand.getTo()))).thenReturn(
                             "1:2");
 
@@ -810,7 +807,6 @@
     @Test
     public void testOvsDeleteFlowCommandSuccess() {
         final String bridge = "gre";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsDeleteFlowCommand deleteFlowCommand = new OvsDeleteFlowCommand("Test");
@@ -818,11 +814,11 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.setupvSwitchNetwork(conn)).thenReturn(network);
-        try {
-            when(network.getBridge(conn)).thenReturn(bridge);
-            when(citrixResourceBase.callHostPlugin(conn, "ovsgre", "ovs_delete_flow", "bridge", bridge, "vmName", deleteFlowCommand.getVmName())).thenReturn("SUCCESS");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.setupvSwitchNetwork(connection)).thenReturn(network);
+        try {
+            when(network.getBridge(connection)).thenReturn(bridge);
+            when(citrixResourceBase.callHostPlugin(connection, "ovsgre", "ovs_delete_flow", "bridge", bridge, "vmName", deleteFlowCommand.getVmName())).thenReturn("SUCCESS");
 
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
@@ -843,7 +839,6 @@
     @Test
     public void testOvsDeleteFlowCommandFailure() {
         final String bridge = "gre";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsDeleteFlowCommand deleteFlowCommand = new OvsDeleteFlowCommand("Test");
@@ -851,11 +846,11 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.setupvSwitchNetwork(conn)).thenReturn(network);
-        try {
-            when(network.getBridge(conn)).thenReturn(bridge);
-            when(citrixResourceBase.callHostPlugin(conn, "ovsgre", "ovs_delete_flow", "bridge", bridge, "vmName", deleteFlowCommand.getVmName())).thenReturn("FAILED");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.setupvSwitchNetwork(connection)).thenReturn(network);
+        try {
+            when(network.getBridge(connection)).thenReturn(bridge);
+            when(citrixResourceBase.callHostPlugin(connection, "ovsgre", "ovs_delete_flow", "bridge", bridge, "vmName", deleteFlowCommand.getVmName())).thenReturn("FAILED");
 
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
@@ -876,7 +871,6 @@
     @Test
     public void testOvsVpcPhysicalTopologyConfigCommand() {
         final String bridge = "gre";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsVpcPhysicalTopologyConfigCommand.Host[] hosts = new OvsVpcPhysicalTopologyConfigCommand.Host[0];
@@ -888,10 +882,10 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.findOrCreateTunnelNetwork(conn, physicalTopology.getBridgeName())).thenReturn(network);
-            when(network.getBridge(conn)).thenReturn(bridge);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.findOrCreateTunnelNetwork(connection, physicalTopology.getBridgeName())).thenReturn(network);
+            when(network.getBridge(connection)).thenReturn(bridge);
 
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
@@ -911,7 +905,6 @@
     @Test
     public void testOvsVpcRoutingPolicyConfigCommand() {
         final String bridge = "gre";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsVpcRoutingPolicyConfigCommand.Acl[] acls = new OvsVpcRoutingPolicyConfigCommand.Acl[0];
@@ -922,10 +915,10 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.findOrCreateTunnelNetwork(conn, routingPolicy.getBridgeName())).thenReturn(network);
-            when(network.getBridge(conn)).thenReturn(bridge);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.findOrCreateTunnelNetwork(connection, routingPolicy.getBridgeName())).thenReturn(network);
+            when(network.getBridge(connection)).thenReturn(bridge);
 
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
@@ -944,7 +937,6 @@
 
     @Test
     public void testCleanupNetworkRulesCmd() {
-        final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final CleanupNetworkRulesCmd cleanupNets = new CleanupNetworkRulesCmd(20);
@@ -953,10 +945,10 @@
         assertNotNull(wrapper);
 
         when(citrixResourceBase.canBridgeFirewall()).thenReturn(true);
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
         when(citrixResourceBase.getVMInstanceName()).thenReturn("VM");
-        when(citrixResourceBase.callHostPlugin(conn, "vmops", "cleanup_rules", "instance", citrixResourceBase.getVMInstanceName())).thenReturn("1");
+        when(citrixResourceBase.callHostPlugin(connection, "vmops", "cleanup_rules", "instance", citrixResourceBase.getVMInstanceName())).thenReturn("1");
 
         final Answer answer = wrapper.execute(cleanupNets, citrixResourceBase);
 
@@ -967,7 +959,6 @@
 
     @Test
     public void testCleanupNetworkRulesCmdLTZ() {
-        final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final CleanupNetworkRulesCmd cleanupNets = new CleanupNetworkRulesCmd(20);
@@ -976,10 +967,10 @@
         assertNotNull(wrapper);
 
         when(citrixResourceBase.canBridgeFirewall()).thenReturn(true);
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
         when(citrixResourceBase.getVMInstanceName()).thenReturn("VM");
-        when(citrixResourceBase.callHostPlugin(conn, "vmops", "cleanup_rules", "instance", citrixResourceBase.getVMInstanceName())).thenReturn("-1");
+        when(citrixResourceBase.callHostPlugin(connection, "vmops", "cleanup_rules", "instance", citrixResourceBase.getVMInstanceName())).thenReturn("-1");
 
         final Answer answer = wrapper.execute(cleanupNets, citrixResourceBase);
 
@@ -1006,14 +997,13 @@
 
     @Test
     public void testNetworkRulesSystemVmCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final NetworkRulesSystemVmCommand netRules = new NetworkRulesSystemVmCommand("Test", VirtualMachine.Type.User);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(netRules, citrixResourceBase);
 
@@ -1024,15 +1014,14 @@
 
     @Test
     public void testNetworkRulesSystemVmCommandNonUser() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final NetworkRulesSystemVmCommand netRules = new NetworkRulesSystemVmCommand("Test", VirtualMachine.Type.DomainRouter);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.callHostPlugin(conn, "vmops", "default_network_rules_systemvm", "vmName", netRules.getVmName())).thenReturn("true");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.callHostPlugin(connection, "vmops", "default_network_rules_systemvm", "vmName", netRules.getVmName())).thenReturn("true");
 
         final Answer answer = wrapper.execute(netRules, citrixResourceBase);
 
@@ -1043,15 +1032,14 @@
 
     @Test
     public void testNetworkRulesSystemVmCommandNonUserFalse() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final NetworkRulesSystemVmCommand netRules = new NetworkRulesSystemVmCommand("Test", VirtualMachine.Type.DomainRouter);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.callHostPlugin(conn, "vmops", "default_network_rules_systemvm", "vmName", netRules.getVmName())).thenReturn("false");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.callHostPlugin(connection, "vmops", "default_network_rules_systemvm", "vmName", netRules.getVmName())).thenReturn("false");
 
         final Answer answer = wrapper.execute(netRules, citrixResourceBase);
 
@@ -1063,7 +1051,6 @@
     @Test
     public void testOvsCreateTunnelCommandSuccess() {
         final String bridge = "tunnel";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsCreateTunnelCommand createTunnel = new OvsCreateTunnelCommand("127.0.0.1", 1, 1l, 2l, 1l, "127.0.1.1", "net01", "cd84c713-f448-48c9-ba25-e6740d4a9003");
@@ -1071,12 +1058,12 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.findOrCreateTunnelNetwork(conn, createTunnel.getNetworkName())).thenReturn(network);
-            when(network.getBridge(conn)).thenReturn(bridge);
-
-            when(citrixResourceBase.callHostPlugin(conn, "ovstunnel", "create_tunnel", "bridge", bridge, "remote_ip", createTunnel.getRemoteIp(),
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.findOrCreateTunnelNetwork(connection, createTunnel.getNetworkName())).thenReturn(network);
+            when(network.getBridge(connection)).thenReturn(bridge);
+
+            when(citrixResourceBase.callHostPlugin(connection, "ovstunnel", "create_tunnel", "bridge", bridge, "remote_ip", createTunnel.getRemoteIp(),
                             "key", createTunnel.getKey().toString(), "from",
                             createTunnel.getFrom().toString(), "to", createTunnel.getTo().toString(), "cloudstack-network-id",
                             createTunnel.getNetworkUuid())).thenReturn("SUCCESS:0");
@@ -1092,7 +1079,7 @@
         final Answer answer = wrapper.execute(createTunnel, citrixResourceBase);
 
         verify(citrixResourceBase, times(1)).getConnection();
-        verify(citrixResourceBase, times(1)).configureTunnelNetwork(conn, createTunnel.getNetworkId(), createTunnel.getFrom(), createTunnel.getNetworkName());
+        verify(citrixResourceBase, times(1)).configureTunnelNetwork(connection, createTunnel.getNetworkId(), createTunnel.getFrom(), createTunnel.getNetworkName());
 
         assertTrue(answer.getResult());
     }
@@ -1100,7 +1087,6 @@
     @Test
     public void testOvsCreateTunnelCommandFail() {
         final String bridge = "tunnel";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsCreateTunnelCommand createTunnel = new OvsCreateTunnelCommand("127.0.0.1", 1, 1l, 2l, 1l, "127.0.1.1", "net01", "cd84c713-f448-48c9-ba25-e6740d4a9003");
@@ -1108,12 +1094,12 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.findOrCreateTunnelNetwork(conn, createTunnel.getNetworkName())).thenReturn(network);
-            when(network.getBridge(conn)).thenReturn(bridge);
-
-            when(citrixResourceBase.callHostPlugin(conn, "ovstunnel", "create_tunnel", "bridge", bridge, "remote_ip", createTunnel.getRemoteIp(),
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.findOrCreateTunnelNetwork(connection, createTunnel.getNetworkName())).thenReturn(network);
+            when(network.getBridge(connection)).thenReturn(bridge);
+
+            when(citrixResourceBase.callHostPlugin(connection, "ovstunnel", "create_tunnel", "bridge", bridge, "remote_ip", createTunnel.getRemoteIp(),
                             "key", createTunnel.getKey().toString(), "from",
                             createTunnel.getFrom().toString(), "to", createTunnel.getTo().toString(), "cloudstack-network-id",
                             createTunnel.getNetworkUuid())).thenReturn("FAIL:1");
@@ -1129,22 +1115,21 @@
         final Answer answer = wrapper.execute(createTunnel, citrixResourceBase);
 
         verify(citrixResourceBase, times(1)).getConnection();
-        verify(citrixResourceBase, times(1)).configureTunnelNetwork(conn, createTunnel.getNetworkId(), createTunnel.getFrom(), createTunnel.getNetworkName());
+        verify(citrixResourceBase, times(1)).configureTunnelNetwork(connection, createTunnel.getNetworkId(), createTunnel.getFrom(), createTunnel.getNetworkName());
 
         assertFalse(answer.getResult());
     }
 
     @Test
     public void testOvsCreateTunnelCommandNoNet() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final OvsCreateTunnelCommand createTunnel = new OvsCreateTunnelCommand("127.0.0.1", 1, 1l, 2l, 1l, "127.0.1.1", "net01", "cd84c713-f448-48c9-ba25-e6740d4a9003");
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.findOrCreateTunnelNetwork(conn, createTunnel.getNetworkName())).thenReturn(null);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.findOrCreateTunnelNetwork(connection, createTunnel.getNetworkName())).thenReturn(null);
 
         final Answer answer = wrapper.execute(createTunnel, citrixResourceBase);
 
@@ -1155,27 +1140,25 @@
 
     @Test
     public void testOvsSetupBridgeCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final OvsSetupBridgeCommand setupBridge = new OvsSetupBridgeCommand("Test", 1l, 1l);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(setupBridge, citrixResourceBase);
 
         verify(citrixResourceBase, times(1)).getConnection();
-        verify(citrixResourceBase, times(1)).findOrCreateTunnelNetwork(conn, setupBridge.getBridgeName());
-        verify(citrixResourceBase, times(1)).configureTunnelNetwork(conn, setupBridge.getNetworkId(), setupBridge.getHostId(), setupBridge.getBridgeName());
+        verify(citrixResourceBase, times(1)).findOrCreateTunnelNetwork(connection, setupBridge.getBridgeName());
+        verify(citrixResourceBase, times(1)).configureTunnelNetwork(connection, setupBridge.getNetworkId(), setupBridge.getHostId(), setupBridge.getBridgeName());
 
         assertTrue(answer.getResult());
     }
 
     @Test
     public void testOvsDestroyBridgeCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsDestroyBridgeCommand destroyBridge = new OvsDestroyBridgeCommand(1l, "bridge", 1l);
@@ -1183,20 +1166,20 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.findOrCreateTunnelNetwork(conn, destroyBridge.getBridgeName())).thenReturn(network);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.findOrCreateTunnelNetwork(connection, destroyBridge.getBridgeName())).thenReturn(network);
 
         final Answer answer = wrapper.execute(destroyBridge, citrixResourceBase);
 
         verify(citrixResourceBase, times(1)).getConnection();
         try {
-            verify(citrixResourceBase, times(1)).cleanUpTmpDomVif(conn, network);
+            verify(citrixResourceBase, times(1)).cleanUpTmpDomVif(connection, network);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
             fail(e.getMessage());
         }
-        verify(citrixResourceBase, times(1)).destroyTunnelNetwork(conn, network, destroyBridge.getHostId());
+        verify(citrixResourceBase, times(1)).destroyTunnelNetwork(connection, network, destroyBridge.getHostId());
 
         assertTrue(answer.getResult());
     }
@@ -1204,7 +1187,6 @@
     @Test
     public void testOvsDestroyTunnelCommandSuccess() {
         final String bridge = "tunnel";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsDestroyTunnelCommand destroyTunnel = new OvsDestroyTunnelCommand(1l, "net01", "port11");
@@ -1212,12 +1194,12 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.findOrCreateTunnelNetwork(conn, destroyTunnel.getBridgeName())).thenReturn(network);
-            when(network.getBridge(conn)).thenReturn(bridge);
-
-            when(citrixResourceBase.callHostPlugin(conn, "ovstunnel", "destroy_tunnel", "bridge", bridge, "in_port", destroyTunnel.getInPortName())).thenReturn("SUCCESS");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.findOrCreateTunnelNetwork(connection, destroyTunnel.getBridgeName())).thenReturn(network);
+            when(network.getBridge(connection)).thenReturn(bridge);
+
+            when(citrixResourceBase.callHostPlugin(connection, "ovstunnel", "destroy_tunnel", "bridge", bridge, "in_port", destroyTunnel.getInPortName())).thenReturn("SUCCESS");
 
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
@@ -1237,7 +1219,6 @@
     @Test
     public void testOvsDestroyTunnelCommandFailed() {
         final String bridge = "tunnel";
-        final Connection conn = Mockito.mock(Connection.class);
         final Network network = Mockito.mock(Network.class);
 
         final OvsDestroyTunnelCommand destroyTunnel = new OvsDestroyTunnelCommand(1l, "net01", "port11");
@@ -1245,12 +1226,12 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.findOrCreateTunnelNetwork(conn, destroyTunnel.getBridgeName())).thenReturn(network);
-            when(network.getBridge(conn)).thenReturn(bridge);
-
-            when(citrixResourceBase.callHostPlugin(conn, "ovstunnel", "destroy_tunnel", "bridge", bridge, "in_port", destroyTunnel.getInPortName())).thenReturn("FAILED");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.findOrCreateTunnelNetwork(connection, destroyTunnel.getBridgeName())).thenReturn(network);
+            when(network.getBridge(connection)).thenReturn(bridge);
+
+            when(citrixResourceBase.callHostPlugin(connection, "ovstunnel", "destroy_tunnel", "bridge", bridge, "in_port", destroyTunnel.getInPortName())).thenReturn("FAILED");
 
         } catch (final BadServerResponse e) {
             fail(e.getMessage());
@@ -1412,7 +1393,6 @@
     public void testClusterVMMetaDataSyncCommand() {
         final String uuid = "6172d8b7-ba10-4a70-93f9-ecaf41f51d53";
 
-        final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
 
         final Pool pool = Mockito.mock(Pool.class);
@@ -1425,7 +1405,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         try (MockedStatic<Pool.Record> ignored = Mockito.mockStatic(Pool.Record.class)){
             when(citrixResourceBase.getHost()).thenReturn(xsHost);
             poolr.master = master;
@@ -1472,14 +1452,13 @@
     @Test
     public void testPlugNicCommand() {
         final NicTO nicTO = Mockito.mock(NicTO.class);
-        final Connection conn = Mockito.mock(Connection.class);
 
         final PlugNicCommand plugNic = new PlugNicCommand(nicTO, "Test", VirtualMachine.Type.User);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(plugNic, citrixResourceBase);
 
@@ -1491,14 +1470,13 @@
     @Test
     public void testUnPlugNicCommand() {
         final NicTO nicTO = Mockito.mock(NicTO.class);
-        final Connection conn = Mockito.mock(Connection.class);
 
         final UnPlugNicCommand unplugNic = new UnPlugNicCommand(nicTO, "Test");
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(unplugNic, citrixResourceBase);
 
@@ -1509,7 +1487,6 @@
 
     @Test
     public void testCreateVMSnapshotCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final VMSnapshotTO snapshotTO = Mockito.mock(VMSnapshotTO.class);
         final List<VolumeObjectTO> volumeTOs = new ArrayList<VolumeObjectTO>();
@@ -1519,7 +1496,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(vmSnapshot, citrixResourceBase);
 
@@ -1530,7 +1507,6 @@
 
     @Test
     public void testDeleteVMSnapshotCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final VMSnapshotTO snapshotTO = Mockito.mock(VMSnapshotTO.class);
         final List<VolumeObjectTO> volumeTOs = new ArrayList<VolumeObjectTO>();
@@ -1540,7 +1516,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(vmSnapshot, citrixResourceBase);
 
@@ -1551,7 +1527,6 @@
 
     @Test
     public void testRevertToVMSnapshotCommand() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final VMSnapshotTO snapshotTO = Mockito.mock(VMSnapshotTO.class);
         final List<VolumeObjectTO> volumeTOs = new ArrayList<VolumeObjectTO>();
@@ -1561,7 +1536,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(vmSnapshot, citrixResourceBase);
 
@@ -1572,15 +1547,14 @@
 
     @Test
     public void testNetworkRulesVmSecondaryIpCommandSuccess() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final NetworkRulesVmSecondaryIpCommand rulesVm = new NetworkRulesVmSecondaryIpCommand("Test", VirtualMachine.Type.User);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.callHostPlugin(conn, "vmops", "network_rules_vmSecondaryIp", "vmName", rulesVm.getVmName(), "vmMac", rulesVm.getVmMac(),
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.callHostPlugin(connection, "vmops", "network_rules_vmSecondaryIp", "vmName", rulesVm.getVmName(), "vmMac", rulesVm.getVmMac(),
                         "vmSecIp", rulesVm.getVmSecIp(), "action", rulesVm.getAction())).thenReturn("true");
 
         final Answer answer = wrapper.execute(rulesVm, citrixResourceBase);
@@ -1592,15 +1566,14 @@
 
     @Test
     public void testNetworkRulesVmSecondaryIpCommandFailure() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final NetworkRulesVmSecondaryIpCommand rulesVm = new NetworkRulesVmSecondaryIpCommand("Test", VirtualMachine.Type.User);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.callHostPlugin(conn, "vmops", "network_rules_vmSecondaryIp", "vmName", rulesVm.getVmName(), "vmMac", rulesVm.getVmMac(),
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.callHostPlugin(connection, "vmops", "network_rules_vmSecondaryIp", "vmName", rulesVm.getVmName(), "vmMac", rulesVm.getVmMac(),
                         "vmSecIp", rulesVm.getVmSecIp(), "action", rulesVm.getAction())).thenReturn("false");
 
         final Answer answer = wrapper.execute(rulesVm, citrixResourceBase);
@@ -1615,7 +1588,6 @@
         final String uuid = "6172d8b7-ba10-4a70-93f9-ecaf41f51d53";
 
         final VirtualMachineTO machineTO = Mockito.mock(VirtualMachineTO.class);
-        final Connection conn = Mockito.mock(Connection.class);
         final XsHost xsHost = Mockito.mock(XsHost.class);
         final Host host = Mockito.mock(Host.class);
 
@@ -1624,7 +1596,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
         when(citrixResourceBase.getHost()).thenReturn(xsHost);
         when(citrixResourceBase.getHost().getUuid()).thenReturn(uuid);
 
@@ -1639,7 +1611,6 @@
     public void testPvlanSetupCommandDhcpSuccess() {
         final String label = "net";
 
-        final Connection conn = Mockito.mock(Connection.class);
         final XsLocalNetwork network = Mockito.mock(XsLocalNetwork.class);
         final Network network2 = Mockito.mock(Network.class);
 
@@ -1655,18 +1626,18 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.getNativeNetworkForTraffic(conn, TrafficType.Guest, "tag")).thenReturn(network);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.getNativeNetworkForTraffic(connection, TrafficType.Guest, "tag")).thenReturn(network);
             when(network.getNetwork()).thenReturn(network2);
-            when(network2.getNameLabel(conn)).thenReturn(label);
+            when(network2.getNameLabel(connection)).thenReturn(label);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
             fail(e.getMessage());
         }
 
-        when(citrixResourceBase.callHostPlugin(conn, "ovs-pvlan", "setup-pvlan-dhcp", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
+        when(citrixResourceBase.callHostPlugin(connection, "ovs-pvlan", "setup-pvlan-dhcp", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
                         isolatedPvlan, "dhcp-name", dhcpName, "dhcp-ip", dhcpIp, "dhcp-mac", dhcpMac)).thenReturn("true");
 
         final Answer answer = wrapper.execute(lanSetup, citrixResourceBase);
@@ -1680,7 +1651,6 @@
     public void testPvlanSetupCommandDhcpFailure() {
         final String label = "net";
 
-        final Connection conn = Mockito.mock(Connection.class);
         final XsLocalNetwork network = Mockito.mock(XsLocalNetwork.class);
         final Network network2 = Mockito.mock(Network.class);
 
@@ -1696,18 +1666,18 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.getNativeNetworkForTraffic(conn, TrafficType.Guest, "tag")).thenReturn(network);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.getNativeNetworkForTraffic(connection, TrafficType.Guest, "tag")).thenReturn(network);
             when(network.getNetwork()).thenReturn(network2);
-            when(network2.getNameLabel(conn)).thenReturn(label);
+            when(network2.getNameLabel(connection)).thenReturn(label);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
             fail(e.getMessage());
         }
 
-        when(citrixResourceBase.callHostPlugin(conn, "ovs-pvlan", "setup-pvlan-dhcp", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
+        when(citrixResourceBase.callHostPlugin(connection, "ovs-pvlan", "setup-pvlan-dhcp", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
                         isolatedPvlan, "dhcp-name", dhcpName, "dhcp-ip", dhcpIp, "dhcp-mac", dhcpMac)).thenReturn("false");
 
         final Answer answer = wrapper.execute(lanSetup, citrixResourceBase);
@@ -1721,7 +1691,6 @@
     public void testPvlanSetupCommandVmSuccess() {
         final String label = "net";
 
-        final Connection conn = Mockito.mock(Connection.class);
         final XsLocalNetwork network = Mockito.mock(XsLocalNetwork.class);
         final Network network2 = Mockito.mock(Network.class);
 
@@ -1735,18 +1704,18 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.getNativeNetworkForTraffic(conn, TrafficType.Guest, "tag")).thenReturn(network);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.getNativeNetworkForTraffic(connection, TrafficType.Guest, "tag")).thenReturn(network);
             when(network.getNetwork()).thenReturn(network2);
-            when(network2.getNameLabel(conn)).thenReturn(label);
+            when(network2.getNameLabel(connection)).thenReturn(label);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
             fail(e.getMessage());
         }
 
-        when(citrixResourceBase.callHostPlugin(conn, "ovs-pvlan", "setup-pvlan-vm", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
+        when(citrixResourceBase.callHostPlugin(connection, "ovs-pvlan", "setup-pvlan-vm", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
                         isolatedPvlan, "vm-mac", vmMac)).thenReturn("true");
 
         final Answer answer = wrapper.execute(lanSetup, citrixResourceBase);
@@ -1760,7 +1729,6 @@
     public void testPvlanSetupCommandVmFailure() {
         final String label = "net";
 
-        final Connection conn = Mockito.mock(Connection.class);
         final XsLocalNetwork network = Mockito.mock(XsLocalNetwork.class);
         final Network network2 = Mockito.mock(Network.class);
 
@@ -1774,18 +1742,18 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        try {
-            when(citrixResourceBase.getNativeNetworkForTraffic(conn, TrafficType.Guest, "tag")).thenReturn(network);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        try {
+            when(citrixResourceBase.getNativeNetworkForTraffic(connection, TrafficType.Guest, "tag")).thenReturn(network);
             when(network.getNetwork()).thenReturn(network2);
-            when(network2.getNameLabel(conn)).thenReturn(label);
+            when(network2.getNameLabel(connection)).thenReturn(label);
         } catch (final XenAPIException e) {
             fail(e.getMessage());
         } catch (final XmlRpcException e) {
             fail(e.getMessage());
         }
 
-        when(citrixResourceBase.callHostPlugin(conn, "ovs-pvlan", "setup-pvlan-vm", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
+        when(citrixResourceBase.callHostPlugin(connection, "ovs-pvlan", "setup-pvlan-vm", "op", op, "nw-label", label, "primary-pvlan", primaryPvlan, "isolated-pvlan",
                         isolatedPvlan, "vm-mac", vmMac)).thenReturn("false");
 
         final Answer answer = wrapper.execute(lanSetup, citrixResourceBase);
@@ -1797,15 +1765,14 @@
 
     @Test
     public void testPerformanceMonitorCommandSuccess() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final PerformanceMonitorCommand performanceMonitor = new PerformanceMonitorCommand(new Hashtable<String, String>(), 200);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.getPerfMon(conn, performanceMonitor.getParams(), performanceMonitor.getWait())).thenReturn("performance");
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.getPerfMon(connection, performanceMonitor.getParams(), performanceMonitor.getWait())).thenReturn("performance");
 
         final Answer answer = wrapper.execute(performanceMonitor, citrixResourceBase);
 
@@ -1816,15 +1783,14 @@
 
     @Test
     public void testPerformanceMonitorCommandFailure() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final PerformanceMonitorCommand performanceMonitor = new PerformanceMonitorCommand(new Hashtable<String, String>(), 200);
 
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.getPerfMon(conn, performanceMonitor.getParams(), performanceMonitor.getWait())).thenReturn(null);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.getPerfMon(connection, performanceMonitor.getParams(), performanceMonitor.getWait())).thenReturn(null);
 
         final Answer answer = wrapper.execute(performanceMonitor, citrixResourceBase);
 
@@ -1894,7 +1860,6 @@
     @Test
     public void testGetVmIpAddressCommand() throws XenAPIException, XmlRpcException {
 
-        final Connection conn = Mockito.mock(Connection.class);
         final VM vm = Mockito.mock(VM.class);
         final VMGuestMetrics mtr = Mockito.mock(VMGuestMetrics.class);
         final VMGuestMetrics.Record rec = Mockito.mock(VMGuestMetrics.Record.class);
@@ -1908,10 +1873,10 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(citrixResourceBase.getConnection()).thenReturn(conn);
-        when(citrixResourceBase.getVM(conn, getVmIpAddrCmd.getVmName())).thenReturn(vm);
-        when(vm.getGuestMetrics(conn)).thenReturn(mtr);
-        when(mtr.getRecord(conn)).thenReturn(rec);
+        when(citrixResourceBase.getConnection()).thenReturn(connection);
+        when(citrixResourceBase.getVM(connection, getVmIpAddrCmd.getVmName())).thenReturn(vm);
+        when(vm.getGuestMetrics(connection)).thenReturn(mtr);
+        when(mtr.getRecord(connection)).thenReturn(rec);
 
         final Answer answer = wrapper.execute(getVmIpAddrCmd, citrixResourceBase);
 
