--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XcpServerWrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XcpServerWrapperTest.java
@@ -47,6 +47,8 @@
     @Mock
     protected XcpServerResource XcpServerResource;
 
+    private final Connection connection = Mockito.mock(Connection.class);
+
 
     final static long[] vpcStats = { 1000L, 2000L };
     final static long[] networkStats = { 3000L, 4000L };
@@ -54,7 +56,6 @@
 
     @Test
     public void testNetworkUsageCommandCreate() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final String privateIP = "192.168.0.10";
         final String domRName = "dom";
@@ -66,8 +67,8 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(XcpServerResource.getConnection()).thenReturn(conn);
-        when(XcpServerResource.networkUsage(conn, usageCommand.getPrivateIP(), "create", null)).thenReturn("success");
+        when(XcpServerResource.getConnection()).thenReturn(connection);
+        when(XcpServerResource.networkUsage(connection, usageCommand.getPrivateIP(), "create", null)).thenReturn("success");
 
         final Answer answer = wrapper.execute(usageCommand, XcpServerResource);
 
@@ -78,7 +79,6 @@
 
     @Test
     public void testNetworkUsageCommandGet() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final String privateIP = "192.168.0.10";
         final String domRName = "dom";
@@ -90,8 +90,8 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(XcpServerResource.getConnection()).thenReturn(conn);
-        when(XcpServerResource.getNetworkStats(conn, usageCommand.getPrivateIP(), null)).thenReturn(new long[]{1l, 1l});
+        when(XcpServerResource.getConnection()).thenReturn(connection);
+        when(XcpServerResource.getNetworkStats(connection, usageCommand.getPrivateIP(), null)).thenReturn(new long[]{1l, 1l});
 
         final Answer answer = wrapper.execute(usageCommand, XcpServerResource);
 
@@ -102,7 +102,6 @@
 
     @Test
     public void testNetworkUsageCommandExceptiopn() {
-        final Connection conn = Mockito.mock(Connection.class);
 
         final String privateIP = "192.168.0.10";
         final String domRName = "dom";
@@ -114,7 +113,7 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        when(XcpServerResource.getConnection()).thenReturn(conn);
+        when(XcpServerResource.getConnection()).thenReturn(connection);
 
         final Answer answer = wrapper.execute(usageCommand, XcpServerResource);
 
@@ -170,9 +169,8 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        final Connection conn = Mockito.mock(Connection.class);
-        when(XcpServerResource.getConnection()).thenReturn(conn);
-        when(XcpServerResource.getNetworkStats(conn, getAutoScaleMetricsCommand.getPrivateIP(), getAutoScaleMetricsCommand.getPublicIP())).thenReturn(networkStats);
+        when(XcpServerResource.getConnection()).thenReturn(connection);
+        when(XcpServerResource.getNetworkStats(connection, getAutoScaleMetricsCommand.getPrivateIP(), getAutoScaleMetricsCommand.getPublicIP())).thenReturn(networkStats);
         when(XcpServerResource.getNetworkLbStats(getAutoScaleMetricsCommand.getPrivateIP(),  getAutoScaleMetricsCommand.getPublicIP(), getAutoScaleMetricsCommand.getPort())).thenReturn(lbStats);
 
         final Answer answer = wrapper.execute(getAutoScaleMetricsCommand, XcpServerResource);
@@ -202,9 +200,8 @@
         final CitrixRequestWrapper wrapper = CitrixRequestWrapper.getInstance();
         assertNotNull(wrapper);
 
-        final Connection conn = Mockito.mock(Connection.class);
-        when(XcpServerResource.getConnection()).thenReturn(conn);
-        when(XcpServerResource.getNetworkStats(conn, getAutoScaleMetricsCommand.getPrivateIP(), getAutoScaleMetricsCommand.getPublicIP())).thenThrow(NumberFormatException.class);
+        when(XcpServerResource.getConnection()).thenReturn(connection);
+        when(XcpServerResource.getNetworkStats(connection, getAutoScaleMetricsCommand.getPrivateIP(), getAutoScaleMetricsCommand.getPublicIP())).thenThrow(NumberFormatException.class);
 
         final Answer answer = wrapper.execute(getAutoScaleMetricsCommand, XcpServerResource);
         assertFalse(answer.getResult());
