--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -274,6 +274,8 @@
     @Mock
     IpAddressManager ipAddressManager;
 
+    private FirewallRuleMatchTagsType firewallRuleMatchTagsTypeMock;
+
     TungstenServiceImpl tungstenService;
 
     AutoCloseable closeable;
@@ -281,6 +283,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        firewallRuleMatchTagsTypeMock = mock(FirewallRuleMatchTagsType.class);
         tungstenService = new TungstenServiceImpl();
         tungstenService.projectDao = projectDao;
         tungstenService.tungstenProviderDao = tungstenProviderDao;
@@ -1291,7 +1294,6 @@
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
-        FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
@@ -1306,7 +1308,7 @@
         when(firewallRuleEndpointType1.getTags()).thenReturn(List.of("tag"));
         when(firewallRuleEndpointType2.getTags()).thenReturn(null);
         when(firewallRuleEndpointType2.getAddressGroup()).thenReturn("address:group");
-        when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsType);
+        when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsTypeMock);
 
         assertNotNull(tungstenService.createTungstenFirewallRule(1L, "f5ba12c8-d4c5-4c20-a57d-67a9b6fca652", "test", "pass", "948f421c-edde-4518-a391-09299cc25dc2",
                 "8b4637b6-5629-46de-8fb2-d0b0502bfa85", "8d097a79-a38d-4db4-8a41-16f15d9c5afa", null, "<>", "a329662e-1805-4a89-9b05-2b818ea35978"
@@ -1391,7 +1393,6 @@
         ObjectReference<ApiPropertyBase> serviceGroup = mock(ObjectReference.class);
         FirewallRuleEndpointType firewallRuleEndpointType1 = mock(FirewallRuleEndpointType.class);
         FirewallRuleEndpointType firewallRuleEndpointType2 = mock(FirewallRuleEndpointType.class);
-        FirewallRuleMatchTagsType firewallRuleMatchTagsType = mock(FirewallRuleMatchTagsType.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
@@ -1406,7 +1407,7 @@
         when(firewallRuleEndpointType1.getTags()).thenReturn(List.of("tag"));
         when(firewallRuleEndpointType2.getTags()).thenReturn(null);
         when(firewallRuleEndpointType2.getAddressGroup()).thenReturn("address:group");
-        when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsType);
+        when(firewallRule.getMatchTags()).thenReturn(firewallRuleMatchTagsTypeMock);
 
         assertNotNull(tungstenService.listTungstenFirewallRule(1L, "948f421c-edde-4518-a391-09299cc25dc2",
                 "8b4637b6-5629-46de-8fb2-d0b0502bfa85"));
