--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -183,6 +183,8 @@
     @Mock
     TungstenApi tungstenApi;
 
+    private FloatingIp floatingIpMock;
+
     TungstenResource tungstenResource;
 
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
@@ -194,6 +196,7 @@
         closeable = MockitoAnnotations.openMocks(this);
         tungstenResource = new TungstenResource();
         tungstenResource.tungstenApi = tungstenApi;
+        floatingIpMock = mock(FloatingIp.class);
         ReflectionTestUtils.setField(tungstenResource, "vrouterPort", "9091");
         tungstenVRouterApiMocked = Mockito.mockStatic(TungstenVRouterApi.class);
 
@@ -406,14 +409,13 @@
     public void executeRequestCreateTungstenFloatingIpCommandTest() {
         TungstenCommand command = new CreateTungstenFloatingIpCommand("projectFqn",
             "ac617be6-bf80-4086-9d6a-c05ff78e2264", "fip", "fi", "192.168.1.100");
-        FloatingIp floatingIp = mock(FloatingIp.class);
 
         when(tungstenApi.createTungstenFloatingIp(anyString(), anyString(), anyString(), anyString(),
-            anyString())).thenReturn(floatingIp);
-
-        TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
-        assertTrue(answer.getResult());
-        assertEquals(floatingIp, answer.getApiObjectBase());
+            anyString())).thenReturn(floatingIpMock);
+
+        TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
+        assertTrue(answer.getResult());
+        assertEquals(floatingIpMock, answer.getApiObjectBase());
     }
 
     @Test
@@ -470,12 +472,11 @@
             "fi");
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         FloatingIpPool floatingIpPool = mock(FloatingIpPool.class);
-        FloatingIp floatingIp = mock(FloatingIp.class);
 
         when(tungstenApi.getTungstenObject(eq(VirtualNetwork.class), anyString())).thenReturn(virtualNetwork);
         when(tungstenApi.getTungstenObjectByName(eq(FloatingIpPool.class), any(), anyString())).thenReturn(
             floatingIpPool);
-        when(tungstenApi.getTungstenObjectByName(eq(FloatingIp.class), any(), anyString())).thenReturn(floatingIp);
+        when(tungstenApi.getTungstenObjectByName(eq(FloatingIp.class), any(), anyString())).thenReturn(floatingIpMock);
         when(tungstenApi.deleteTungstenObject(any(FloatingIp.class))).thenReturn(true);
 
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
@@ -532,9 +533,8 @@
     @Test
     public void executeRequestGetTungstenFloatingIpsCommandTest() {
         TungstenCommand command = new GetTungstenFloatingIpsCommand("b604c7f7-1dbc-42d8-bceb-2c0898034a7a", "fip");
-        FloatingIp floatingIp1 = mock(FloatingIp.class);
         FloatingIp floatingIp2 = mock(FloatingIp.class);
-        List<? extends ApiObjectBase> floatingIpList = Arrays.asList(floatingIp1, floatingIp2);
+        List<? extends ApiObjectBase> floatingIpList = Arrays.asList(floatingIpMock, floatingIp2);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         FloatingIpPool floatingIpPool = mock(FloatingIpPool.class);
 
