--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java
@@ -674,7 +674,6 @@
         // Setup
         ListDataStoreObjectsCommand cmd = new ListDataStoreObjectsCommand(destDataStoreTO, "valid/path", 0, 10);
         HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
-        HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
         FileInfo fileInfo = mock(FileInfo.class);
         List<FileInfo> fileInfoList = new ArrayList<>();
         fileInfoList.add(fileInfo);
@@ -682,8 +681,8 @@
 
         doReturn(context).when(vmwareResource).getServiceContext(any());
         doReturn(hyperHost).when(vmwareResource).getHyperHost(context);
+        HostDatastoreBrowserSearchResults results = createMockHostDatastoreBrowserSearchResults(fileInfoList);
         when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
-        when(results.getFile()).thenReturn(fileInfoList);
         when(fileInfo.getPath()).thenReturn("file.txt");
         when(fileInfo.getFileSize()).thenReturn(1L);
         when(fileInfo.getModification()).thenReturn(VmwareHelper.getXMLGregorianCalendar(date, 0));
@@ -753,7 +752,6 @@
         // Setup
         ListDataStoreObjectsCommand cmd = new ListDataStoreObjectsCommand(destDataStoreTO, "/", 0, 10);
         HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
-        HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
         FileInfo fileInfo = mock(FileInfo.class);
         List<FileInfo> fileInfoList = new ArrayList<>();
         fileInfoList.add(fileInfo);
@@ -761,8 +759,8 @@
 
         doReturn(context).when(vmwareResource).getServiceContext();
         doReturn(hyperHost).when(vmwareResource).getHyperHost(context);
+        HostDatastoreBrowserSearchResults results = createMockHostDatastoreBrowserSearchResults(fileInfoList);
         when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
-        when(results.getFile()).thenReturn(fileInfoList);
         when(fileInfo.getPath()).thenReturn("file.txt");
         when(fileInfo.getFileSize()).thenReturn(1L);
         when(fileInfo.getModification()).thenReturn(VmwareHelper.getXMLGregorianCalendar(date, 0));
@@ -798,7 +796,6 @@
         // Setup
         ListDataStoreObjectsCommand cmd = new ListDataStoreObjectsCommand(destDataStoreTO, "", 0, 10);
         HostDatastoreBrowserMO browserMo = mock(HostDatastoreBrowserMO.class);
-        HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
         FileInfo fileInfo = mock(FileInfo.class);
         List<FileInfo> fileInfoList = new ArrayList<>();
         fileInfoList.add(fileInfo);
@@ -806,8 +803,8 @@
 
         doReturn(context).when(vmwareResource).getServiceContext();
         doReturn(hyperHost).when(vmwareResource).getHyperHost(context);
+        HostDatastoreBrowserSearchResults results = createMockHostDatastoreBrowserSearchResults(fileInfoList);
         when(browserMo.searchDatastore(anyString(), any(HostDatastoreBrowserSearchSpec.class))).thenReturn(results);
-        when(results.getFile()).thenReturn(fileInfoList);
         when(fileInfo.getPath()).thenReturn("file.txt");
         when(fileInfo.getFileSize()).thenReturn(1L);
         when(fileInfo.getModification()).thenReturn(VmwareHelper.getXMLGregorianCalendar(date, 0));
@@ -874,4 +871,9 @@
         Mockito.verify(vmwareResource, Mockito.times(1)).removeVirtualTPMDevice(vmConfigSpec, tpm);
         Mockito.verify(deviceChanges, Mockito.times(1)).add(any(VirtualDeviceConfigSpec.class));
     }
+    private static HostDatastoreBrowserSearchResults createMockHostDatastoreBrowserSearchResults(List<FileInfo> fileInfoList) {
+        HostDatastoreBrowserSearchResults results = mock(HostDatastoreBrowserSearchResults.class);
+        when(results.getFile()).thenReturn(new ArrayList<>(fileInfoList));
+        return results;
+    }
 }
