--- a/server/src/test/java/com/cloud/network/router/VirtualNetworkApplianceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/network/router/VirtualNetworkApplianceManagerImplTest.java
@@ -85,6 +85,7 @@
 import org.apache.cloudstack.network.RoutedIpv4Manager;
 import org.apache.cloudstack.utils.identity.ManagementServerNode;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.InjectMocks;
@@ -262,6 +263,13 @@
 
     @Mock
     private BGPService bgpService;
+
+    private List<? extends BgpPeer> bgpPeers;
+
+    @Before
+    public void setUp() {
+        bgpPeers = Mockito.mock(List.class);
+    }
 
     @Mock
     private LoadBalancingRulesManager _lbMgr;
@@ -371,7 +379,6 @@
         when(_networkDao.findById(guestNetworkId)).thenReturn(network);
         when(network.getVpcId()).thenReturn(null);
         when(routedIpv4Manager.isDynamicRoutedNetwork(network)).thenReturn(true);
-        List<? extends BgpPeer> bgpPeers = Mockito.mock(List.class);
         doReturn(bgpPeers).when(bgpService).getBgpPeersForNetwork(network);
         virtualNetworkApplianceManagerImpl.finalizeNetworkRulesForNetwork(cmds, router, Network.Provider.VirtualRouter, guestNetworkId);
 
@@ -391,7 +398,6 @@
         VpcVO vpc = Mockito.mock(VpcVO.class);
         when(_vpcDao.findById(vpcId)).thenReturn(vpc);
         when(routedIpv4Manager.isDynamicRoutedVpc(vpc)).thenReturn(true);
-        List<? extends BgpPeer> bgpPeers = Mockito.mock(List.class);
         doReturn(bgpPeers).when(bgpService).getBgpPeersForVpc(vpc);
 
         virtualNetworkApplianceManagerImpl.finalizeNetworkRulesForNetwork(cmds, router, Network.Provider.VirtualRouter, guestNetworkId);
