--- a/engine/storage/configdrive/src/test/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilderTest.java
+++ b/engine/storage/configdrive/src/test/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilderTest.java
@@ -173,16 +173,14 @@
 
     @Test(expected = CloudRuntimeException.class)
     public void writeVendorDataJsonFileTestCannotCreateOpenStackFolder() {
-        File folderFileMock = Mockito.mock(File.class);
-        Mockito.doReturn(false).when(folderFileMock).mkdirs();
+        File folderFileMock = createMockFolderFile();
 
         ConfigDriveBuilder.writeVendorDataJsonFile(folderFileMock);
     }
 
     @Test(expected = CloudRuntimeException.class)
     public void writeVendorDataJsonFileTest() {
-        File folderFileMock = Mockito.mock(File.class);
-        Mockito.doReturn(false).when(folderFileMock).mkdirs();
+        File folderFileMock = createMockFolderFile();
 
         ConfigDriveBuilder.writeVendorDataJsonFile(folderFileMock);
     }
@@ -659,4 +657,10 @@
         Assert.assertEquals(expectedJsonObject, actualJson);
         folder.delete();
     }
+
+    private static File createMockFolderFile() {
+        File folderFileMock = Mockito.mock(File.class);
+        Mockito.doReturn(false).when(folderFileMock).mkdirs();
+        return folderFileMock;
+    }
 }
