--- a/core/src/test/java/com/cloud/storage/template/OVAProcessorTest.java
+++ b/core/src/test/java/com/cloud/storage/template/OVAProcessorTest.java
@@ -109,10 +109,8 @@
         long actualSize = 1000;
         String templatePath = "/tmp";
         String templateName = "template";
-        File mockFile = Mockito.mock(File.class);
+        File mockFile = createMockFile(templatePath, templateName);
         Mockito.when(mockFile.length()).thenReturn(actualSize);
-        Mockito.when(mockFile.getParent()).thenReturn(templatePath);
-        Mockito.when(mockFile.getName()).thenReturn(templateName);
         Mockito.doThrow(new InternalErrorException("virtual size calculation failed")).when(processor).getTemplateVirtualSize(templatePath, templateName);
         Assert.assertEquals(actualSize, processor.getVirtualSize(mockFile));
         Mockito.verify(mockFile, Mockito.times(1)).length();
@@ -124,9 +122,7 @@
         long actualSize = 1000;
         String templatePath = "/tmp";
         String templateName = "template";
-        File mockFile = Mockito.mock(File.class);
-        Mockito.when(mockFile.getParent()).thenReturn(templatePath);
-        Mockito.when(mockFile.getName()).thenReturn(templateName);
+        File mockFile = createMockFile(templatePath, templateName);
         Mockito.doReturn(virtualSize).when(processor).getTemplateVirtualSize(templatePath, templateName);
         Assert.assertEquals(virtualSize, processor.getVirtualSize(mockFile));
         Mockito.verify(mockFile, Mockito.times(0)).length();
@@ -152,4 +148,11 @@
             Assert.assertEquals("template name:", templateName + ".ova", info.filename);
         }
     }
+
+    private static File createMockFile(String templatePath, String templateName) {
+        File mockFile = Mockito.mock(File.class);
+        Mockito.when(mockFile.getParent()).thenReturn(templatePath);
+        Mockito.when(mockFile.getName()).thenReturn(templateName);
+        return mockFile;
+    }
 }
