--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtPreMigrationCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtPreMigrationCommandWrapperTest.java
@@ -68,11 +68,14 @@
 
     private LibvirtPreMigrationCommandWrapper wrapper;
 
+    private LibvirtException libvirtExceptionMock;
+
     private static final String VM_NAME = "test-vm";
 
     @Before
     public void setUp() {
         wrapper = new LibvirtPreMigrationCommandWrapper();
+        libvirtExceptionMock = Mockito.mock(LibvirtException.class);
         when(preMigrationCommand.getVmName()).thenReturn(VM_NAME);
         when(preMigrationCommand.getVirtualMachine()).thenReturn(virtualMachineTO);
         when(libvirtComputingResource.getLibvirtUtilitiesHelper()).thenReturn(libvirtUtilitiesHelper);
@@ -185,11 +188,10 @@
     @Test
     public void testExecute_DomainFreeFails_StillReturnsSuccess() throws LibvirtException {
         List<DiskDef> disks = createNonClvmDisks();
-        LibvirtException libvirtException = Mockito.mock(LibvirtException.class);
-        when(libvirtUtilitiesHelper.getConnectionByVmName(VM_NAME)).thenReturn(connect);
-        when(connect.domainLookupByName(VM_NAME)).thenReturn(domain);
-        when(libvirtComputingResource.getDisks(connect, VM_NAME)).thenReturn(disks);
-        Mockito.doThrow(libvirtException).when(domain).free();
+        when(libvirtUtilitiesHelper.getConnectionByVmName(VM_NAME)).thenReturn(connect);
+        when(connect.domainLookupByName(VM_NAME)).thenReturn(domain);
+        when(libvirtComputingResource.getDisks(connect, VM_NAME)).thenReturn(disks);
+        Mockito.doThrow(libvirtExceptionMock).when(domain).free();
 
         Answer answer = wrapper.execute(preMigrationCommand, libvirtComputingResource);
 
@@ -198,9 +200,8 @@
 
     @Test
     public void testExecute_NullVmName_ReturnsFailure() throws LibvirtException {
-        LibvirtException libvirtException = Mockito.mock(LibvirtException.class);
         when(preMigrationCommand.getVmName()).thenReturn(null);
-        when(libvirtUtilitiesHelper.getConnectionByVmName(null)).thenThrow(libvirtException);
+        when(libvirtUtilitiesHelper.getConnectionByVmName(null)).thenThrow(libvirtExceptionMock);
 
         Answer answer = wrapper.execute(preMigrationCommand, libvirtComputingResource);
 
