--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -185,6 +185,8 @@
 
     TungstenResource tungstenResource;
 
+    private LoadbalancerMember loadbalancerMember;
+
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
 
     AutoCloseable closeable;
@@ -193,6 +195,7 @@
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
         tungstenResource = new TungstenResource();
+        loadbalancerMember = mock(LoadbalancerMember.class);
         tungstenResource.tungstenApi = tungstenApi;
         ReflectionTestUtils.setField(tungstenResource, "vrouterPort", "9091");
         tungstenVRouterApiMocked = Mockito.mockStatic(TungstenVRouterApi.class);
@@ -648,7 +651,6 @@
         LoadbalancerListener loadbalancerListener = mock(LoadbalancerListener.class);
         LoadbalancerHealthmonitor loadbalancerHealthmonitor = mock(LoadbalancerHealthmonitor.class);
         LoadbalancerPool loadbalancerPool = mock(LoadbalancerPool.class);
-        LoadbalancerMember member1 = mock(LoadbalancerMember.class);
         LoadbalancerMember member2 = mock(LoadbalancerMember.class);
 
         when(virtualNetwork.getUuid()).thenReturn("ac617be6-bf80-4086-9d6a-c05ff78e2264");
@@ -674,7 +676,7 @@
         when(tungstenApi.createTungstenLoadbalancerPool(anyString(), anyString(), anyString(), anyString(), anyString(),
             anyString())).thenReturn(loadbalancerPool);
         when(tungstenApi.createTungstenLoadbalancerMember(anyString(), anyString(), anyString(), anyString(), anyInt(),
-            anyInt())).thenReturn(member1).thenReturn(member2);
+            anyInt())).thenReturn(loadbalancerMember).thenReturn(member2);
 
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -699,7 +701,6 @@
         Loadbalancer loadbalancer = mock(Loadbalancer.class);
         LoadbalancerListener loadbalancerListener = mock(LoadbalancerListener.class);
         LoadbalancerPool loadbalancerPool = mock(LoadbalancerPool.class);
-        LoadbalancerMember loadbalancerMember1 = mock(LoadbalancerMember.class);
         LoadbalancerMember loadbalancerMember2 = mock(LoadbalancerMember.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         ObjectReference<ApiPropertyBase> listerner = mock(ObjectReference.class);
@@ -720,7 +721,7 @@
         when(tungstenApi.getTungstenObject(eq(LoadbalancerListener.class), anyString())).thenReturn(
             loadbalancerListener);
         when(tungstenApi.getTungstenObject(eq(LoadbalancerPool.class), anyString())).thenReturn(loadbalancerPool);
-        when(tungstenApi.getTungstenObject(eq(LoadbalancerMember.class), anyString())).thenReturn(loadbalancerMember1)
+        when(tungstenApi.getTungstenObject(eq(LoadbalancerMember.class), anyString())).thenReturn(loadbalancerMember)
             .thenReturn(loadbalancerMember2);
         when(tungstenApi.deleteTungstenObject(any())).thenReturn(true);
         when(tungstenApi.releaseTungstenFloatingIp(anyString(), anyString(), anyString())).thenReturn(true);
