--- a/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
+++ b/engine/orchestration/src/test/java/com/cloud/vm/VirtualMachineManagerImplTest.java
@@ -299,8 +299,11 @@
     private ConfigDepotImpl configDepotImpl;
     private boolean updatedConfigKeyDepot = false;
 
+    private NicTO nicTOMock;
+
     @Before
     public void setup() {
+        nicTOMock = mock(NicTO.class);
         ReflectionTestUtils.getField(VirtualMachineManager.VmMetadataManufacturer, "s_depot");
         virtualMachineManagerImpl.setHostAllocators(new ArrayList<>());
 
@@ -1632,7 +1635,7 @@
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
 
-        when(vmTO.getNics()).thenReturn(new NicTO[]{mock(NicTO.class)});
+        when(vmTO.getNics()).thenReturn(new NicTO[]{nicTOMock});
         when(updatedTO.getNics()).thenReturn(null);
 
         virtualMachineManagerImpl.updateExternalVmNicsFromPrepareAnswer(vmTO, updatedTO);
@@ -1646,7 +1649,7 @@
         VirtualMachineTO updatedTO = mock(VirtualMachineTO.class);
 
         when(vmTO.getNics()).thenReturn(null);
-        when(updatedTO.getNics()).thenReturn(new NicTO[]{mock(NicTO.class)});
+        when(updatedTO.getNics()).thenReturn(new NicTO[]{nicTOMock});
 
         virtualMachineManagerImpl.updateExternalVmNicsFromPrepareAnswer(vmTO, updatedTO);
 
@@ -1683,7 +1686,7 @@
         VirtualMachineProfile vmProfile = mock(VirtualMachineProfile.class);
         VirtualMachineTemplate template = mock(VirtualMachineTemplate.class);
         when(vmProfile.getTemplate()).thenReturn(template);
-        NicTO[] nics = new NicTO[]{mock(NicTO.class)};
+        NicTO[] nics = new NicTO[]{nicTOMock};
         VirtualMachineTO vmTO = mock(VirtualMachineTO.class);
         when(vmTO.getNics()).thenReturn(nics);
         doReturn(vmTO).when(virtualMachineManagerImpl).toVmTO(vmProfile);
