--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/vrouter/VRouterApiConnectorImplTest.java
@@ -55,6 +55,10 @@
         entityUtilsMocked = Mockito.mockStatic(EntityUtils.class);
     }
 
+    private Gateway createGatewayMock() {
+        return mock(Gateway.class);
+    }
+
     @After
     public void tearDown() throws Exception {
         httpClientsMocked.close();
@@ -177,7 +181,7 @@
     @Test
     public void addGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
+        Gateway gateway2 = createGatewayMock();
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
@@ -193,7 +197,7 @@
     @Test
     public void addGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
+        Gateway gateway2 = createGatewayMock();
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
@@ -205,7 +209,7 @@
     @Test
     public void deleteGatewayTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
+        Gateway gateway2 = createGatewayMock();
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
         CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
         HttpEntity httpEntity = mock(HttpEntity.class);
@@ -221,7 +225,7 @@
     @Test
     public void deleteGatewayWithExceptionTest() throws Exception {
         Gateway gateway1 = mock(Gateway.class);
-        Gateway gateway2 = mock(Gateway.class);
+        Gateway gateway2 = createGatewayMock();
         CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
 
         httpClientsMocked.when(HttpClients::createDefault).thenReturn(httpClient);
