--- a/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionServiceTest.java
+++ b/plugins/integrations/kubernetes-service/src/test/java/com/cloud/kubernetes/version/KubernetesVersionServiceTest.java
@@ -106,6 +106,8 @@
     @Mock
     private Account accountMock;
 
+    private ListKubernetesSupportedVersionsCmd listKubernetesSupportedVersionsCmd;
+
     AutoCloseable closeable;
 
 
@@ -118,6 +120,7 @@
     @Before
     public void setUp() throws Exception {
         closeable = MockitoAnnotations.openMocks(this);
+        listKubernetesSupportedVersionsCmd = Mockito.mock(ListKubernetesSupportedVersionsCmd.class);
 
         overrideDefaultConfigValue(KubernetesClusterService.KubernetesServiceEnabled, "_defaultValue", "true");
 
@@ -160,7 +163,6 @@
             callContextMockedStatic.when(CallContext::current).thenReturn(callContextMock);
             final SearchCriteria<KubernetesSupportedVersionVO> versionSearchCriteria = Mockito.mock(SearchCriteria.class);
             when(callContextMock.getCallingAccount()).thenReturn(accountMock);
-            ListKubernetesSupportedVersionsCmd cmd = Mockito.mock(ListKubernetesSupportedVersionsCmd.class);
             List<KubernetesSupportedVersionVO> versionVOs = new ArrayList<>();
             KubernetesSupportedVersionVO versionVO = Mockito.mock(KubernetesSupportedVersionVO.class);
             when(versionVO.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
@@ -169,7 +171,7 @@
             when(kubernetesSupportedVersionDao.searchAndCount(Mockito.any(), Mockito.any(Filter.class)))
                     .thenReturn(new Pair<>(versionVOs, versionVOs.size()));
             ListResponse<KubernetesSupportedVersionResponse> versionsResponse =
-                    kubernetesVersionService.listKubernetesSupportedVersions(cmd);
+                    kubernetesVersionService.listKubernetesSupportedVersions(listKubernetesSupportedVersionsCmd);
             Assert.assertEquals(versionVOs.size(), versionsResponse.getCount().intValue());
             Assert.assertTrue(CollectionUtils.isNotEmpty(versionsResponse.getResponses()));
             Assert.assertEquals(versionVOs.size(), versionsResponse.getResponses().size());
@@ -181,7 +183,6 @@
         CallContext callContextMock = Mockito.mock(CallContext.class);
         try (MockedStatic<CallContext> callContextMockedStatic = Mockito.mockStatic(CallContext.class)) {
             callContextMockedStatic.when(CallContext::current).thenReturn(callContextMock);
-            ListKubernetesSupportedVersionsCmd cmd = Mockito.mock(ListKubernetesSupportedVersionsCmd.class);
             List<KubernetesSupportedVersionVO> versionVOs = new ArrayList<>();
             KubernetesSupportedVersionVO versionVO = Mockito.mock(KubernetesSupportedVersionVO.class);
             when(versionVO.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
@@ -191,7 +192,7 @@
             when(kubernetesSupportedVersionDao.searchAndCount(Mockito.any(), Mockito.any(Filter.class)))
                     .thenReturn(new Pair<>(versionVOs, versionVOs.size()));
             when(accountManager.isRootAdmin(anyLong())).thenReturn(true);
-            ListResponse<KubernetesSupportedVersionResponse> response = kubernetesVersionService.listKubernetesSupportedVersions(cmd);
+            ListResponse<KubernetesSupportedVersionResponse> response = kubernetesVersionService.listKubernetesSupportedVersions(listKubernetesSupportedVersionsCmd);
             assertNotNull(response.getResponses().get(0).getIsoUrl());
         }
     }
@@ -201,7 +202,6 @@
         CallContext callContextMock = Mockito.mock(CallContext.class);
         try (MockedStatic<CallContext> callContextMockedStatic = Mockito.mockStatic(CallContext.class)) {
             callContextMockedStatic.when(CallContext::current).thenReturn(callContextMock);
-            ListKubernetesSupportedVersionsCmd cmd = Mockito.mock(ListKubernetesSupportedVersionsCmd.class);
             List<KubernetesSupportedVersionVO> versionVOs = new ArrayList<>();
             KubernetesSupportedVersionVO versionVO = Mockito.mock(KubernetesSupportedVersionVO.class);
             when(versionVO.getSemanticVersion()).thenReturn(KubernetesVersionService.MIN_KUBERNETES_VERSION);
@@ -212,7 +212,7 @@
                     .thenReturn(new Pair<>(versionVOs, versionVOs.size()));
             when(accountManager.isRootAdmin(anyLong())).thenReturn(false);
             when(accountMock.getId()).thenReturn(2L);
-            ListResponse<KubernetesSupportedVersionResponse> response = kubernetesVersionService.listKubernetesSupportedVersions(cmd);
+            ListResponse<KubernetesSupportedVersionResponse> response = kubernetesVersionService.listKubernetesSupportedVersions(listKubernetesSupportedVersionsCmd);
             assertNull(response.getResponses().get(0).getIsoUrl());
         }
     }
