--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -152,6 +152,9 @@
     private SnapshotDataStoreVO snapshotDataStoreVoMock;
 
     @Mock
+    private SnapshotDataStoreVO snapshotDataStoreVOMock;
+
+    @Mock
     private VMSnapshotDao vmSnapshotDaoMock;
 
     @Mock
@@ -2216,12 +2219,11 @@
     @Test
     public void gatherSnapshotReferencesOfChildrenSnapshotTestSingleChildWithSingleSnapshotReference() {
         VMSnapshotVO childSnapshot = Mockito.mock(VMSnapshotVO.class);
-        SnapshotDataStoreVO snapshotDataStoreVO = Mockito.mock(SnapshotDataStoreVO.class);
 
         doReturn(100L).when(vmSnapshotVoMock).getId();
         doReturn(List.of(childSnapshot)).when(vmSnapshotDaoMock).listByParent(100L);
         doReturn(200L).when(childSnapshot).getId();
-        doReturn(List.of(snapshotDataStoreVO)).when(vmSnapshotHelperMock).getVolumeSnapshotsAssociatedWithKvmDiskOnlyVmSnapshot(200L);
+        doReturn(List.of(snapshotDataStoreVOMock)).when(vmSnapshotHelperMock).getVolumeSnapshotsAssociatedWithKvmDiskOnlyVmSnapshot(200L);
         doNothing().when(kbossBackupProviderSpy).mapVolumesToSnapshotReferences(anyList(), anyList(), anyMap());
 
         List<VolumeObjectTO> volumeObjectTOs = List.of(volumeObjectToMock);
@@ -2239,14 +2241,13 @@
     public void gatherSnapshotReferencesOfChildrenSnapshotTestMultipleChildrenAggregatesSnapshotReferences() {
         VMSnapshotVO childSnapshot1 = Mockito.mock(VMSnapshotVO.class);
         VMSnapshotVO childSnapshot2 = Mockito.mock(VMSnapshotVO.class);
-        SnapshotDataStoreVO snapshotDataStoreVO1 = Mockito.mock(SnapshotDataStoreVO.class);
         SnapshotDataStoreVO snapshotDataStoreVO2 = Mockito.mock(SnapshotDataStoreVO.class);
 
         doReturn(100L).when(vmSnapshotVoMock).getId();
         doReturn(List.of(childSnapshot1, childSnapshot2)).when(vmSnapshotDaoMock).listByParent(100L);
         doReturn(201L).when(childSnapshot1).getId();
         doReturn(202L).when(childSnapshot2).getId();
-        doReturn(List.of(snapshotDataStoreVO1)).when(vmSnapshotHelperMock)
+        doReturn(List.of(snapshotDataStoreVOMock)).when(vmSnapshotHelperMock)
                 .getVolumeSnapshotsAssociatedWithKvmDiskOnlyVmSnapshot(201L);
         doReturn(List.of(snapshotDataStoreVO2)).when(vmSnapshotHelperMock)
                 .getVolumeSnapshotsAssociatedWithKvmDiskOnlyVmSnapshot(202L);
