--- a/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
+++ b/server/src/test/java/com/cloud/network/router/CommandSetupHelperTest.java
@@ -102,6 +102,7 @@
     ASNumberDao asNumberDao;
     @Mock
     BgpPeerDetailsDao bgpPeerDetailsDao;
+    private BgpPeerVO bgpPeer2Mock;
 
     @Before
     public void setUp() {
@@ -116,6 +117,7 @@
         ReflectionTestUtils.setField(commandSetupHelper, "_vpcDao", vpcDao);
         ReflectionTestUtils.setField(commandSetupHelper, "_routerControlHelper", routerControlHelper);
         ReflectionTestUtils.setField(commandSetupHelper, "_dcDao", dcDao);
+        bgpPeer2Mock = Mockito.mock(BgpPeerVO.class);
     }
 
     @Test
@@ -211,8 +213,7 @@
     @Test
     public void testCreateBgpPeersCommandsForNetwork() {
         BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
-        BgpPeerVO bgpPeer2 = Mockito.mock(BgpPeerVO.class);
-        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2);
+        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2Mock);
         Commands cmds = new Commands(Command.OnError.Stop);
         VirtualRouter router = Mockito.mock(VirtualRouter.class);
         NetworkVO network = Mockito.mock(NetworkVO.class);
@@ -239,8 +240,7 @@
     @Test
     public void testCreateBgpPeersCommandsForVpc() {
         BgpPeerVO bgpPeer1 = Mockito.mock(BgpPeerVO.class);
-        BgpPeerVO bgpPeer2 = Mockito.mock(BgpPeerVO.class);
-        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2);
+        List<BgpPeerVO> bgpPeers = Arrays.asList(bgpPeer1, bgpPeer2Mock);
         Commands cmds = new Commands(Command.OnError.Stop);
         VirtualRouter router = Mockito.mock(VirtualRouter.class);
         NetworkVO network = Mockito.mock(NetworkVO.class);
