--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/XenServerGuruTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/XenServerGuruTest.java
@@ -79,8 +79,7 @@
     public void beforeTest() {
         Mockito.when(sourceDataMock.getHypervisorType()).thenReturn(HypervisorType.XenServer);
 
-        Mockito.when(copyCommandMock.getSrcTO()).thenReturn(sourceDataMock);
-        Mockito.when(copyCommandMock.getDestTO()).thenReturn(destinationDataMock);
+        copyCommandMock = createMockCopyCommand(sourceDataMock, destinationDataMock);
 
         Mockito.when(changedHost.getId()).thenReturn(changedHostId);
         Mockito.lenient().when(defaultHost.getId()).thenReturn(defaultHostId);
@@ -100,6 +99,13 @@
     private void assertPairOfHostToExecuteCommandIsTheDefaultHostId(Pair<Boolean, Long> pairHostToExecuteCommand) {
         Assert.assertFalse(pairHostToExecuteCommand.first());
         Assert.assertEquals(defaultHostId, pairHostToExecuteCommand.second());
+    }
+
+    private static CopyCommand createMockCopyCommand(DataTO sourceData, DataTO destinationData) {
+        CopyCommand copyCommand = Mockito.mock(CopyCommand.class);
+        Mockito.when(copyCommand.getSrcTO()).thenReturn(sourceData);
+        Mockito.when(copyCommand.getDestTO()).thenReturn(destinationData);
+        return copyCommand;
     }
 
     @Test
