--- a/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
+++ b/server/src/test/java/com/cloud/vm/UserVmManagerImplTest.java
@@ -2853,14 +2853,12 @@
 
     @Test
     public void implementNetworkTestImplementedNetworkIsNullReturnCurrentNewNetwork() throws ResourceUnavailableException, InsufficientCapacityException {
-        CallContext callContextMock = Mockito.mock(CallContext.class);
+        CallContext callContextMock = createMockCallContext();
         NetworkVO currentNetwork = Mockito.mock(NetworkVO.class);
 
         try (MockedStatic<CallContext> ignored = mockStatic(CallContext.class)) {
             Mockito.when(CallContext.current()).thenReturn(callContextMock);
 
-            Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
-
             Mockito.doReturn(callerUser).when(userDao).findById(Mockito.anyLong());
             Mockito.doReturn(null).when(_networkMgr).implementNetwork(Mockito.anyLong(), Mockito.any(), Mockito.any());
 
@@ -2872,13 +2870,11 @@
 
     @Test
     public void implementNetworkTestImplementedNetworkFirstIsNullReturnCurrentNewNetwork() throws ResourceUnavailableException, InsufficientCapacityException {
-        CallContext callContextMock = Mockito.mock(CallContext.class);
+        CallContext callContextMock = createMockCallContext();
         NetworkVO currentNetwork = Mockito.mock(NetworkVO.class);
 
         try (MockedStatic<CallContext> ignored = mockStatic(CallContext.class)) {
             Mockito.when(CallContext.current()).thenReturn(callContextMock);
-
-            Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
 
             Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
 
@@ -2894,13 +2890,11 @@
 
     @Test
     public void implementNetworkTestImplementedNetworkSecondIsNullReturnCurrentNewNetwork() throws ResourceUnavailableException, InsufficientCapacityException {
-        CallContext callContextMock = Mockito.mock(CallContext.class);
+        CallContext callContextMock = createMockCallContext();
         NetworkVO currentNetwork = Mockito.mock(NetworkVO.class);
 
         try (MockedStatic<CallContext> ignored = mockStatic(CallContext.class)) {
             Mockito.when(CallContext.current()).thenReturn(callContextMock);
-
-            Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
 
             Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
 
@@ -2917,13 +2911,11 @@
 
     @Test
     public void implementNetworkTestImplementedNetworkSecondIsNotNullReturnImplementedNetworkSecond() throws ResourceUnavailableException, InsufficientCapacityException {
-        CallContext callContextMock = Mockito.mock(CallContext.class);
+        CallContext callContextMock = createMockCallContext();
         NetworkVO currentNetwork = Mockito.mock(NetworkVO.class);
 
         try (MockedStatic<CallContext> ignored = mockStatic(CallContext.class)) {
             Mockito.when(CallContext.current()).thenReturn(callContextMock);
-
-            Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
 
             Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
 
@@ -2942,12 +2934,10 @@
     public void implementNetworkTestImplementedNetworkCatchException() throws ResourceUnavailableException, InsufficientCapacityException {
         String expectedMessage = String.format("Failed to implement network [%s] elements and resources as a part of network provision.", networkMock);
 
-        CallContext callContextMock = Mockito.mock(CallContext.class);
+        CallContext callContextMock = createMockCallContext();
 
         try (MockedStatic<CallContext> ignored = mockStatic(CallContext.class)) {
             Mockito.when(CallContext.current()).thenReturn(callContextMock);
-
-            Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
 
             Pair<? extends NetworkGuru, ? extends Network> implementedNetwork = Mockito.mock(Pair.class);
 
@@ -4047,6 +4037,12 @@
         leaseDetails.put(VmDetailConstants.INSTANCE_LEASE_EXPIRY_DATE, getLeaseExpiryDate(leaseDuration));
         leaseDetails.put(VmDetailConstants.INSTANCE_LEASE_EXECUTION, leaseExecution);
         return leaseDetails;
+    }
+
+    private static CallContext createMockCallContext() {
+        CallContext callContextMock = Mockito.mock(CallContext.class);
+        Mockito.doReturn(1L).when(callContextMock).getCallingUserId();
+        return callContextMock;
     }
 
     private static DeployVMCmd buildDeployVMCmd(Long volumeId, Long snapshotId) {
