--- a/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
+++ b/plugins/event-bus/inmemory/src/test/java/org/apache/cloudstack/mom/inmemory/InMemoryEventBusTest.java
@@ -51,7 +51,7 @@
     @Test
     public void testSubscribe() throws Exception {
         EventTopic topic = mock(EventTopic.class);
-        EventSubscriber subscriber = mock(EventSubscriber.class);
+        EventSubscriber subscriber = createSubscriber();
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
@@ -68,7 +68,7 @@
 
     @Test(expected = EventBusException.class)
     public void testSubscribeFailTopic() throws Exception {
-        EventSubscriber subscriber = mock(EventSubscriber.class);
+        EventSubscriber subscriber = createSubscriber();
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
@@ -87,7 +87,7 @@
     @Test
     public void testUnsubscribe() throws Exception {
         EventTopic topic = mock(EventTopic.class);
-        EventSubscriber subscriber = mock(EventSubscriber.class);
+        EventSubscriber subscriber = createSubscriber();
 
         InMemoryEventBus bus = new InMemoryEventBus();
 
@@ -119,7 +119,7 @@
 
     @Test(expected = EventBusException.class)
     public void testUnsubscribeFailWrongUUID() throws Exception {
-        EventSubscriber subscriber = mock(EventSubscriber.class);
+        EventSubscriber subscriber = createSubscriber();
         InMemoryEventBus bus = new InMemoryEventBus();
         UUID uuid = UUID.randomUUID();
 
@@ -129,7 +129,7 @@
     @Test
     public void testPublish() throws Exception {
         EventTopic topic = mock(EventTopic.class);
-        EventSubscriber subscriber = mock(EventSubscriber.class);
+        EventSubscriber subscriber = createSubscriber();
         Event event = mock(Event.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
@@ -151,7 +151,7 @@
 
     @Test
     public void testPublishEmpty() throws Exception {
-        EventSubscriber subscriber = mock(EventSubscriber.class);
+        EventSubscriber subscriber = createSubscriber();
         Event event = mock(Event.class);
 
         InMemoryEventBus bus = new InMemoryEventBus();
@@ -159,4 +159,8 @@
 
         verify(subscriber, times(0)).onEvent(event);
     }
+
+    private EventSubscriber createSubscriber() {
+        return mock(EventSubscriber.class);
+    }
 }
