--- a/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
@@ -258,6 +258,13 @@
         f.set(configKey, o);
     }
 
+    private static BgpPeerNetworkMapVO createMockBgpPeerNetworkMap2() {
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
+        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(12L);
+        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        return bgpPeerNetworkMap2;
+    }
+
     @Test
     public void testCreateDataCenterIpv4GuestSubnet() {
         CreateIpv4SubnetForZoneCmd cmd = new CreateIpv4SubnetForZoneCmd();
@@ -991,12 +998,10 @@
         when(network.getId()).thenReturn(networkId);
         when(networkDao.findById(networkId)).thenReturn(network);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap2();
         when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
         when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap2();
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
@@ -1024,12 +1029,10 @@
 
         when(network.getId()).thenReturn(networkId);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap2();
         when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
         when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap2();
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
@@ -1139,9 +1142,7 @@
         BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
         when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap2();
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
@@ -1169,12 +1170,10 @@
 
         when(vpc.getId()).thenReturn(vpcId);
 
-        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = Mockito.mock(BgpPeerNetworkMapVO.class);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap1 = createMockBgpPeerNetworkMap2();
         when(bgpPeerNetworkMap1.getBgpPeerId()).thenReturn(bgpPeerId1);
         when(bgpPeerNetworkMap1.getState()).thenReturn(BgpPeer.State.Active);
-        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = Mockito.mock(BgpPeerNetworkMapVO.class);
-        when(bgpPeerNetworkMap2.getBgpPeerId()).thenReturn(bgpPeerId2);
-        when(bgpPeerNetworkMap2.getState()).thenReturn(BgpPeer.State.Revoke);
+        BgpPeerNetworkMapVO bgpPeerNetworkMap2 = createMockBgpPeerNetworkMap2();
         BgpPeerNetworkMapVO bgpPeerNetworkMap3 = Mockito.mock(BgpPeerNetworkMapVO.class);
         when(bgpPeerNetworkMap3.getState()).thenReturn(BgpPeer.State.Add);
 
