--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -129,11 +129,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockStoragePoolDetail();
         String srcPoolSystemId = "610204d03e3ad60f";
         when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
 
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockStoragePoolDetail();
         String destPoolSystemId = "610204d03e3ad60f";
         when(destPoolSystemIdDetail.getValue()).thenReturn(destPoolSystemId);
 
@@ -152,11 +152,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockStoragePoolDetail();
         String srcPoolSystemId = "610204d03e3ad60f";
         when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
 
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockStoragePoolDetail();
         String destPoolSystemId = "7332760565f6340f";
         when(destPoolSystemIdDetail.getValue()).thenReturn(destPoolSystemId);
 
@@ -175,11 +175,11 @@
         when(srcStore.getId()).thenReturn(1L);
         when(destStore.getId()).thenReturn(2L);
 
-        StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcPoolSystemIdDetail = createMockStoragePoolDetail();
         String srcPoolSystemId = "610204d03e3ad60f";
         when(srcPoolSystemIdDetail.getValue()).thenReturn(srcPoolSystemId);
 
-        StoragePoolDetailVO destPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO destPoolSystemIdDetail = createMockStoragePoolDetail();
         when(destPoolSystemIdDetail.getValue()).thenReturn(null);
 
         when(storagePoolDetailsDao.findDetail(1L,ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcPoolSystemIdDetail);
@@ -297,7 +297,7 @@
         when(volumeDetailsDao.findDetail(1L,  DiskTO.SCSI_NAA_DEVICE_ID)).thenReturn(null);
         when(volumeService.getChapInfo(srcData, srcStore)).thenReturn(null);
 
-        StoragePoolDetailVO srcStoragePoolDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcStoragePoolDetail = createMockStoragePoolDetail();
         when(srcStoragePoolDetail.getValue()).thenReturn("610204d03e3ad60f");
         when(storagePoolDetailsDao.findDetail(1L, ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcStoragePoolDetail);
     }
@@ -316,7 +316,7 @@
         when(volumeDetailsDao.findDetail(2L,  DiskTO.SCSI_NAA_DEVICE_ID)).thenReturn(null);
         when(volumeService.getChapInfo(srcData, srcStore)).thenReturn(null);
 
-        StoragePoolDetailVO srcStoragePoolDetail = Mockito.mock(StoragePoolDetailVO.class);
+        StoragePoolDetailVO srcStoragePoolDetail = createMockStoragePoolDetail();
         when(srcStoragePoolDetail.getValue()).thenReturn("7332760565f6340f");
         when(storagePoolDetailsDao.findDetail(2L, ScaleIOGatewayClient.STORAGE_POOL_SYSTEM_ID)).thenReturn(srcStoragePoolDetail);
     }
@@ -597,4 +597,8 @@
                         16L * (1024 * 1024 * 1024),
                         true));
     }
+
+    private static StoragePoolDetailVO createMockStoragePoolDetail() {
+        return Mockito.mock(StoragePoolDetailVO.class);
+    }
 }
