--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -97,6 +97,7 @@
     HostDao hostDao;
     @Mock
     ConfigurationDao configDao;
+    private DataStore destStoreMock;
 
     static MockedStatic<RemoteHostEndPoint> remoteHostEndPointMock;
 
@@ -115,6 +116,7 @@
     @Before
     public void initMocks() {
         closeable = MockitoAnnotations.openMocks(this);
+        destStoreMock = Mockito.mock(DataStore.class);
     }
 
     @After
@@ -194,13 +196,12 @@
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
 
         when(srcData.getDataStore()).thenReturn(srcStore);
-        when(destData.getDataStore()).thenReturn(destStore);
+        when(destData.getDataStore()).thenReturn(destStoreMock);
 
         fillSrcVolumeDetails(srcData, srcStore);
-        fillDestVolumeDetails(destData, destStore);
+        fillDestVolumeDetails(destData, destStoreMock);
 
         VolumeObjectTO destVolTO = Mockito.mock(VolumeObjectTO.class);
         when(destData.getTO()).thenReturn(destVolTO);
@@ -245,13 +246,12 @@
         VolumeInfo destData = Mockito.mock(VolumeInfo.class);
 
         DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
 
         when(srcData.getDataStore()).thenReturn(srcStore);
-        when(destData.getDataStore()).thenReturn(destStore);
+        when(destData.getDataStore()).thenReturn(destStoreMock);
 
         fillSrcVolumeDetails(srcData, srcStore);
-        fillDestVolumeDetails(destData, destStore);
+        fillDestVolumeDetails(destData, destStoreMock);
 
         VolumeObjectTO destVolTO = Mockito.mock(VolumeObjectTO.class);
         when(destData.getTO()).thenReturn(destVolTO);
