--- a/server/src/test/java/org/apache/cloudstack/annotation/AnnotationManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/annotation/AnnotationManagerImplTest.java
@@ -152,8 +152,7 @@
         Mockito.when(volume.getId()).thenReturn(ENTITY_ID);
         Mockito.when(volumeDao.findByUuid(uuid)).thenReturn(volume);
         Mockito.doNothing().when(accountService).checkAccess(user, volume);
-        Role role = Mockito.mock(Role.class);
-        Mockito.when(role.getRoleType()).thenReturn(RoleType.User);
+        Role role = createMockRole();
         Mockito.when(roleService.findRole(Mockito.anyLong())).thenReturn(role);
         AnnotationVO annotationVO = Mockito.mock(AnnotationVO.class);
         AnnotationService.EntityType type = AnnotationService.EntityType.VOLUME;
@@ -172,8 +171,7 @@
         String annotationUuid = UUID.randomUUID().toString();
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         Mockito.when(vm.getId()).thenReturn(ENTITY_ID);
-        Role role = Mockito.mock(Role.class);
-        Mockito.when(role.getRoleType()).thenReturn(RoleType.User);
+        Role role = createMockRole();
         Mockito.when(roleService.findRole(Mockito.anyLong())).thenReturn(role);
         AnnotationVO annotationVO = Mockito.mock(AnnotationVO.class);
         AnnotationService.EntityType type = AnnotationService.EntityType.VM;
@@ -189,4 +187,10 @@
         Assert.assertEquals(ENTITY_ID, CallContext.current().getEventResourceId());
         Assert.assertEquals(ApiCommandResourceType.VirtualMachine, CallContext.current().getEventResourceType());
     }
+
+    private static Role createMockRole() {
+        Role role = Mockito.mock(Role.class);
+        Mockito.when(role.getRoleType()).thenReturn(RoleType.User);
+        return role;
+    }
 }
