--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -208,6 +208,8 @@
     @Mock
     private BackupOfferingVO backupOfferingVOMock;
 
+    private BackupOfferingVO backupOfferingEntityMock;
+
     @Mock
     private AsyncJobVO asyncJobVOMock;
 
@@ -293,6 +295,7 @@
         gson = new Gson();
 
         closeable = MockitoAnnotations.openMocks(this);
+        backupOfferingEntityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.findById(null)).thenReturn(null);
         when(backupOfferingDao.findById(123l)).thenReturn(null);
 
@@ -1598,8 +1601,7 @@
         when(vmInstanceDao.findByIdIncludingRemoved(vmId)).thenReturn(vm);
         when(backupDao.listByVmIdAndOffering(zoneId, vmId, backupOfferingId)).thenReturn(new ArrayList<>());
 
-        BackupOfferingVO offering = mock(BackupOfferingVO.class);
-        when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(offering);
+        when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(backupOfferingEntityMock);
 
         when(backupProvider.deleteBackup(backup, false)).thenReturn(true);
 
@@ -2291,8 +2293,7 @@
     @Test
     public void getBackupOfferingDomainsTestReturnsDomains() {
         Long offeringId = 1L;
-        BackupOfferingVO offering = Mockito.mock(BackupOfferingVO.class);
-        when(backupOfferingDao.findById(offeringId)).thenReturn(offering);
+        when(backupOfferingDao.findById(offeringId)).thenReturn(backupOfferingEntityMock);
         when(backupOfferingDetailsDao.findDomainIds(offeringId)).thenReturn(List.of(10L, 20L));
 
         List<Long> result = backupManager.getBackupOfferingDomains(offeringId);
@@ -2330,9 +2331,8 @@
         when(domainHelper.filterChildSubDomains(List.of(domainId))).thenReturn(new ArrayList<>(List.of(domainId)));
         when(backupOfferingDetailsDao.findDomainIds(id)).thenReturn(new ArrayList<>());
 
-        BackupOfferingVO offering = Mockito.mock(BackupOfferingVO.class);
         BackupOfferingVO offeringUpdate = Mockito.mock(BackupOfferingVO.class);
-        when(backupOfferingDao.findById(id)).thenReturn(offering);
+        when(backupOfferingDao.findById(id)).thenReturn(backupOfferingEntityMock);
         when(backupOfferingDao.createForUpdate(id)).thenReturn(offeringUpdate);
         when(backupOfferingDao.update(id, offeringUpdate)).thenReturn(true);
 
@@ -2359,9 +2359,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(backupOfferingEntityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2403,9 +2402,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(backupOfferingEntityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2448,9 +2446,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(backupOfferingEntityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2494,9 +2491,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(backupOfferingEntityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2538,9 +2534,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(backupOfferingEntityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
@@ -2587,9 +2582,8 @@
 
         SearchBuilder<BackupOfferingVO> sb = Mockito.mock(SearchBuilder.class);
         SearchCriteria<BackupOfferingVO> sc = Mockito.mock(SearchCriteria.class);
-        BackupOfferingVO entityMock = Mockito.mock(BackupOfferingVO.class);
         when(backupOfferingDao.createSearchBuilder()).thenReturn(sb);
-        when(sb.entity()).thenReturn(entityMock);
+        when(sb.entity()).thenReturn(backupOfferingEntityMock);
         when(sb.and(Mockito.anyString(), Mockito.any(), Mockito.any(SearchCriteria.Op.class))).thenReturn(sb);
         when(sb.create()).thenReturn(sc);
         when(backupOfferingDao.searchAndCount(Mockito.any(), Mockito.any()))
