--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -1549,6 +1549,13 @@
         verify(extensionCustomActionDao).remove(actionId);
     }
 
+    private static Extension createMockExtension(Extension.State state) {
+        Extension extension = mock(Extension.class);
+        when(extension.getType()).thenReturn(Extension.Type.Orchestrator);
+        when(extension.getState()).thenReturn(state);
+        return extension;
+    }
+
     private void mockCallerRole(RoleType roleType) {
         CallContext callContextMock = mock(CallContext.class);
         when(CallContext.current()).thenReturn(callContextMock);
@@ -2420,9 +2427,7 @@
 
     @Test
     public void getInstanceConsole_whenValid() {
-        Extension extension = mock(Extension.class);
-        when(extension.getType()).thenReturn(Extension.Type.Orchestrator);
-        when(extension.getState()).thenReturn(Extension.State.Enabled);
+        Extension extension = createMockExtension(Extension.State.Enabled);
         when(extensionsManager.getExtensionForCluster(anyLong())).thenReturn(extension);
         VirtualMachine vm = mock(VirtualMachine.class);
         Host host = mock(Host.class);
@@ -2460,9 +2465,7 @@
 
     @Test
     public void getInstanceConsole_whenNullExtensionNotEnabled() {
-        Extension extension = mock(Extension.class);
-        when(extension.getType()).thenReturn(Extension.Type.Orchestrator);
-        when(extension.getState()).thenReturn(Extension.State.Disabled);
+        Extension extension = createMockExtension(Extension.State.Disabled);
         when(extensionsManager.getExtensionForCluster(anyLong())).thenReturn(extension);
         VirtualMachine vm = mock(VirtualMachine.class);
         Host host = mock(Host.class);
@@ -2474,9 +2477,7 @@
 
     @Test
     public void getInstanceConsole_whenAgentManagerFails() {
-        Extension extension = mock(Extension.class);
-        when(extension.getType()).thenReturn(Extension.Type.Orchestrator);
-        when(extension.getState()).thenReturn(Extension.State.Enabled);
+        Extension extension = createMockExtension(Extension.State.Enabled);
         when(extensionsManager.getExtensionForCluster(anyLong())).thenReturn(extension);
         VirtualMachine vm = mock(VirtualMachine.class);
         Host host = mock(Host.class);
