--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -712,8 +712,8 @@
         when(accountVOMock.getDomainId()).thenReturn(domainId);
 
         BackupScheduleVO schedule = mock(BackupScheduleVO.class);
+        configureMockBackupSchedule(schedule, 2);
         when(backupScheduleDao.findById(scheduleId)).thenReturn(schedule);
-        when(schedule.getMaxBackups()).thenReturn(2);
 
         VolumeVO volume = mock(VolumeVO.class);
         when(volumeDao.findByInstance(vmId)).thenReturn(List.of(volume));
@@ -1855,7 +1855,7 @@
     @Test
     public void deleteOldestBackupFromScheduleIfRequiredTestSkipDeletionWhenRetentionIsEqualToZero() throws ResourceAllocationException {
         Mockito.when(backupScheduleDao.findById(1L)).thenReturn(backupScheduleVOMock);
-        Mockito.when(backupScheduleVOMock.getMaxBackups()).thenReturn(0);
+        configureMockBackupSchedule(backupScheduleVOMock, 0);
         backupManager.deleteOldestBackupFromScheduleIfRequired(1L, 1L);
         Mockito.verify(backupManager, Mockito.never()).deleteExcessBackups(Mockito.anyList(), Mockito.anyInt(), Mockito.anyLong());
     }
@@ -1863,8 +1863,8 @@
     @Test
     public void deleteOldestBackupFromScheduleIfRequiredTestSkipDeletionWhenAmountOfBackupsToBeDeletedIsLessThanOne() throws ResourceAllocationException {
         List<BackupVO> backups = List.of(Mockito.mock(BackupVO.class), Mockito.mock(BackupVO.class));
+        configureMockBackupSchedule(backupScheduleVOMock, 2);
         Mockito.when(backupScheduleDao.findById(1L)).thenReturn(backupScheduleVOMock);
-        Mockito.when(backupScheduleVOMock.getMaxBackups()).thenReturn(2);
         Mockito.when(backupDao.listBySchedule(1L)).thenReturn(backups);
         backupManager.deleteOldestBackupFromScheduleIfRequired(1L, 1L);
         Mockito.verify(backupManager, Mockito.never()).deleteExcessBackups(Mockito.anyList(), Mockito.anyInt(), Mockito.anyLong());
@@ -1874,7 +1874,7 @@
     public void deleteOldestBackupFromScheduleIfRequiredTestDeleteBackupsWhenRequired() throws ResourceAllocationException {
         List<BackupVO> backups = List.of(Mockito.mock(BackupVO.class), Mockito.mock(BackupVO.class));
         Mockito.when(backupScheduleDao.findById(1L)).thenReturn(backupScheduleVOMock);
-        Mockito.when(backupScheduleVOMock.getMaxBackups()).thenReturn(1);
+        configureMockBackupSchedule(backupScheduleVOMock, 1);
         Mockito.when(backupDao.listBySchedule(1L)).thenReturn(backups);
         Mockito.doNothing().when(backupManager).deleteExcessBackups(Mockito.anyList(), Mockito.anyInt(), Mockito.anyLong());
         backupManager.deleteOldestBackupFromScheduleIfRequired(1L, 1L);
@@ -2623,6 +2623,10 @@
         when(offering.getId()).thenReturn(id);
         when(offering.getName()).thenReturn(name);
         return offering;
+    }
+
+    private void configureMockBackupSchedule(BackupScheduleVO backupSchedule, int maxBackups) {
+        when(backupSchedule.getMaxBackups()).thenReturn(maxBackups);
     }
 
     @Test
