--- a/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/network/RoutedIpv4ManagerImplTest.java
@@ -258,6 +258,12 @@
         f.set(configKey, o);
     }
 
+    private static DataCenterIpv4GuestSubnetVO createMockDataCenterIpv4GuestSubnet() {
+        DataCenterIpv4GuestSubnetVO existingSubnet = Mockito.mock(DataCenterIpv4GuestSubnetVO.class);
+        when(existingSubnet.getSubnet()).thenReturn("172.16.1.0/24");
+        return existingSubnet;
+    }
+
     @Test
     public void testCreateDataCenterIpv4GuestSubnet() {
         CreateIpv4SubnetForZoneCmd cmd = new CreateIpv4SubnetForZoneCmd();
@@ -308,7 +314,7 @@
 
     @Test
     public void testCheckConflicts1() {
-        DataCenterIpv4GuestSubnetVO existingSubnet = Mockito.mock(DataCenterIpv4GuestSubnetVO.class);
+        DataCenterIpv4GuestSubnetVO existingSubnet = createMockDataCenterIpv4GuestSubnet();
         when(existingSubnet.getSubnet()).thenReturn(subnet);
         List<DataCenterIpv4GuestSubnetVO> existingSubnets = Arrays.asList(existingSubnet);
         routedIpv4Manager.checkConflicts(existingSubnets, newSubnet, null);
@@ -316,7 +322,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testCheckConflicts2() {
-        DataCenterIpv4GuestSubnetVO existingSubnet = Mockito.mock(DataCenterIpv4GuestSubnetVO.class);
+        DataCenterIpv4GuestSubnetVO existingSubnet = createMockDataCenterIpv4GuestSubnet();
         when(existingSubnet.getSubnet()).thenReturn(subnet);
         List<DataCenterIpv4GuestSubnetVO> existingSubnets = Arrays.asList(existingSubnet);
         routedIpv4Manager.checkConflicts(existingSubnets, subnet, null);
@@ -470,6 +476,7 @@
         when(vpc.getZoneId()).thenReturn(zoneId);
 
         when(ipv4GuestSubnetNetworkMap.getParentId()).thenReturn(zoneSubnetId);
+        subnetVO = createMockDataCenterIpv4GuestSubnet();
         when(dataCenterIpv4GuestSubnetDao.findById(zoneSubnetId)).thenReturn(subnetVO);
         when(subnetVO.getSubnet()).thenReturn(subnet);
         when(subnetVO.getUuid()).thenReturn(zoneSubnetUuid);
