--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -959,10 +959,9 @@
 
     @Test
     public void testDeleteGuestOsCategory_Successful() {
-        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
-        Mockito.when(deleteCmd.getId()).thenReturn(id);
+        DeleteGuestOsCategoryCmd deleteCmd = createMockDeleteGuestOsCategoryCmd();
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList());
         Mockito.when(guestOSCategoryDao.remove(id)).thenReturn(true);
@@ -975,8 +974,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteGuestOsCategory_ThrowsExceptionWhenCategoryNotFound() {
-        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
         long id = 1L;
+        DeleteGuestOsCategoryCmd deleteCmd = createMockDeleteGuestOsCategoryCmd();
         Mockito.when(deleteCmd.getId()).thenReturn(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(null);
         spy.deleteGuestOsCategory(deleteCmd);
@@ -984,13 +983,19 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteGuestOsCategory_ThrowsExceptionWhenGuestOsExists() {
-        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         long id = 1L;
+        DeleteGuestOsCategoryCmd deleteCmd = createMockDeleteGuestOsCategoryCmd();
         Mockito.when(deleteCmd.getId()).thenReturn(id);
         Mockito.when(guestOSCategoryDao.findById(id)).thenReturn(guestOSCategory);
         Mockito.when(guestOSDao.listIdsByCategoryId(id)).thenReturn(Arrays.asList(1L));
         spy.deleteGuestOsCategory(deleteCmd);
+    }
+
+    private static DeleteGuestOsCategoryCmd createMockDeleteGuestOsCategoryCmd() {
+        DeleteGuestOsCategoryCmd deleteCmd = Mockito.mock(DeleteGuestOsCategoryCmd.class);
+        Mockito.when(deleteCmd.getId()).thenReturn(1L);
+        return deleteCmd;
     }
 
     private void mockGuestOsJoin() {
