--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -993,6 +993,19 @@
         spy.deleteGuestOsCategory(deleteCmd);
     }
 
+    private static ListGuestOsCategoriesCmd createMockListGuestOsCategoriesCmd(String name, Boolean featured, Boolean isIso) {
+        ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
+        Mockito.when(listCmd.getArch()).thenReturn(CPU.CPUArch.getDefault());
+        Mockito.when(listCmd.getId()).thenReturn(1L);
+        Mockito.when(listCmd.getKeyword()).thenReturn("Linux");
+        Mockito.when(listCmd.getName()).thenReturn(name);
+        Mockito.when(listCmd.getZoneId()).thenReturn(1L);
+        Mockito.when(listCmd.isFeatured()).thenReturn(featured);
+        Mockito.when(listCmd.isIso()).thenReturn(isIso);
+        Mockito.when(listCmd.isVnf()).thenReturn(false);
+        return listCmd;
+    }
+
     private void mockGuestOsJoin() {
         GuestOSVO vo = mock(GuestOSVO.class);
         SearchBuilder<GuestOSVO> sb = mock(SearchBuilder.class);
@@ -1002,7 +1015,6 @@
 
     @Test
     public void testListGuestOSCategoriesByCriteria_Success() {
-        ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Filter filter = Mockito.mock(Filter.class);
         Long id = 1L;
@@ -1013,14 +1025,7 @@
         CPU.CPUArch arch = CPU.CPUArch.getDefault();
         Boolean isIso = true;
         Boolean isVnf = false;
-        Mockito.when(listCmd.getId()).thenReturn(id);
-        Mockito.when(listCmd.getName()).thenReturn(name);
-        Mockito.when(listCmd.getKeyword()).thenReturn(keyword);
-        Mockito.when(listCmd.isFeatured()).thenReturn(featured);
-        Mockito.when(listCmd.getZoneId()).thenReturn(zoneId);
-        Mockito.when(listCmd.getArch()).thenReturn(arch);
-        Mockito.when(listCmd.isIso()).thenReturn(isIso);
-        Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
+        ListGuestOsCategoriesCmd listCmd = createMockListGuestOsCategoriesCmd(name, featured, isIso);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
@@ -1041,7 +1046,6 @@
 
     @Test
     public void testListGuestOSCategoriesByCriteria_NoResults() {
-        ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         String name = "CentOS";
@@ -1051,14 +1055,7 @@
         CPU.CPUArch arch = CPU.CPUArch.getDefault();
         Boolean isIso = false;
         Boolean isVnf = false;
-        Mockito.when(listCmd.getId()).thenReturn(id);
-        Mockito.when(listCmd.getName()).thenReturn(name);
-        Mockito.when(listCmd.getKeyword()).thenReturn(keyword);
-        Mockito.when(listCmd.isFeatured()).thenReturn(featured);
-        Mockito.when(listCmd.getZoneId()).thenReturn(zoneId);
-        Mockito.when(listCmd.getArch()).thenReturn(arch);
-        Mockito.when(listCmd.isIso()).thenReturn(isIso);
-        Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
+        ListGuestOsCategoriesCmd listCmd = createMockListGuestOsCategoriesCmd(name, featured, isIso);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
@@ -1079,7 +1076,6 @@
 
     @Test
     public void testListGuestOSCategoriesByCriteria_NoGuestOsIdsFound() {
-        ListGuestOsCategoriesCmd listCmd = Mockito.mock(ListGuestOsCategoriesCmd.class);
         GuestOSCategoryVO guestOSCategory = Mockito.mock(GuestOSCategoryVO.class);
         Long id = 1L;
         String name = "Ubuntu";
@@ -1089,14 +1085,7 @@
         CPU.CPUArch arch = CPU.CPUArch.getDefault();
         Boolean isIso = true;
         Boolean isVnf = false;
-        Mockito.when(listCmd.getId()).thenReturn(id);
-        Mockito.when(listCmd.getName()).thenReturn(name);
-        Mockito.when(listCmd.getKeyword()).thenReturn(keyword);
-        Mockito.when(listCmd.isFeatured()).thenReturn(featured);
-        Mockito.when(listCmd.getZoneId()).thenReturn(zoneId);
-        Mockito.when(listCmd.getArch()).thenReturn(arch);
-        Mockito.when(listCmd.isIso()).thenReturn(isIso);
-        Mockito.when(listCmd.isVnf()).thenReturn(isVnf);
+        ListGuestOsCategoriesCmd listCmd = createMockListGuestOsCategoriesCmd(name, featured, isIso);
         SearchBuilder<GuestOSCategoryVO> searchBuilder = Mockito.mock(SearchBuilder.class);
         Mockito.when(searchBuilder.entity()).thenReturn(guestOSCategory);
         SearchCriteria<GuestOSCategoryVO> searchCriteria = Mockito.mock(SearchCriteria.class);
