--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenVRouterApiTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenVRouterApiTest.java
@@ -38,11 +38,13 @@
 
 @RunWith(MockitoJUnitRunner.class)
 public class TungstenVRouterApiTest {
+    private Port portMock;
     MockedStatic<VRouterApiConnectorFactory> vRouterApiConnectorFactoryMocked;
 
     @Before
     public void setup() {
         vRouterApiConnectorFactoryMocked = Mockito.mockStatic(VRouterApiConnectorFactory.class);
+        portMock = mock(Port.class);
     }
 
     @After
@@ -52,24 +54,22 @@
 
     @Test
     public void addTungstenVrouterPortSuccessTest() throws IOException {
-        Port port = mock(Port.class);
         VRouterApiConnector vRouterApiConnector = mock(VRouterApiConnector.class);
 
         when(VRouterApiConnectorFactory.getInstance(anyString(), anyString())).thenReturn(vRouterApiConnector);
         when(vRouterApiConnector.addPort(any(Port.class))).thenReturn(true);
 
-        assertTrue(TungstenVRouterApi.addTungstenVrouterPort("192.168.100.100", "9091", port));
+        assertTrue(TungstenVRouterApi.addTungstenVrouterPort("192.168.100.100", "9091", portMock));
     }
 
     @Test
     public void addTungstenVrouterPortFailTest() throws IOException {
-        Port port = mock(Port.class);
         VRouterApiConnector vRouterApiConnector = mock(VRouterApiConnector.class);
 
         when(VRouterApiConnectorFactory.getInstance(anyString(), anyString())).thenReturn(vRouterApiConnector);
         when(vRouterApiConnector.addPort(any(Port.class))).thenThrow(IOException.class);
 
-        assertFalse(TungstenVRouterApi.addTungstenVrouterPort("192.168.100.100", "9091", port));
+        assertFalse(TungstenVRouterApi.addTungstenVrouterPort("192.168.100.100", "9091", portMock));
     }
 
 
