--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -1353,8 +1353,7 @@
         when(vm.getHostId()).thenReturn(hostId);
 
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
-        BackupProvider backupProvider = mock(BackupProvider.class);
-        when(backupProvider.supportsInstanceFromBackup()).thenReturn(true);
+        BackupProvider backupProvider = createMockBackupProvider(vm, backup, true);
 
         overrideBackupFrameworkConfigValue();
 
@@ -1376,7 +1375,6 @@
         when(rootVolume.getPoolId()).thenReturn(poolId);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
-        when(backupProvider.restoreBackupToVM(vm, backup, null, null, false)).thenReturn(new Pair<>(true, null));
 
         try (MockedStatic<ActionEventUtils> utils = Mockito.mockStatic(ActionEventUtils.class)) {
             boolean result = backupManager.restoreBackupToVM(backupId, vmId, false);
@@ -1409,8 +1407,7 @@
         when(vm.getHostId()).thenReturn(hostId);
 
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
-        BackupProvider backupProvider = mock(BackupProvider.class);
-        when(backupProvider.supportsInstanceFromBackup()).thenReturn(true);
+        BackupProvider backupProvider = createMockBackupProvider(vm, backup, false);
 
         overrideBackupFrameworkConfigValue();
 
@@ -1432,7 +1429,6 @@
         when(rootVolume.getPoolId()).thenReturn(poolId);
         when(volumeDao.findIncludingRemovedByInstanceAndType(vmId, Volume.Type.ROOT)).thenReturn(List.of(rootVolume));
         when(primaryDataStoreDao.findById(poolId)).thenReturn(pool);
-        when(backupProvider.restoreBackupToVM(vm, backup, null, null, false)).thenReturn(new Pair<>(false, null));
 
         try (MockedStatic<ActionEventUtils> utils = Mockito.mockStatic(ActionEventUtils.class)) {
             CloudRuntimeException exception = Assert.assertThrows(CloudRuntimeException.class,
@@ -2616,6 +2612,13 @@
             assertTrue(result.first().stream().anyMatch(o -> o.getName().equals("Global Offering")));
             assertTrue(result.first().stream().anyMatch(o -> o.getName().equals("User Domain Offering")));
         }
+    }
+
+    private static BackupProvider createMockBackupProvider(VMInstanceVO vm, BackupVO backup, boolean restoreResult) {
+        BackupProvider backupProvider = mock(BackupProvider.class);
+        when(backupProvider.supportsInstanceFromBackup()).thenReturn(true);
+        when(backupProvider.restoreBackupToVM(vm, backup, null, null, false)).thenReturn(new Pair<>(restoreResult, null));
+        return backupProvider;
     }
 
     private BackupOfferingVO createMockOffering(Long id, String name) {
