--- a/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
+++ b/server/src/test/java/com/cloud/storage/snapshot/SnapshotManagerImplTest.java
@@ -567,11 +567,16 @@
         Mockito.verify(snapshotPolicyDao).remove(policyId);
     }
 
+    private static DeleteSnapshotPoliciesCmd createMockDeleteSnapshotPoliciesCmd() {
+        DeleteSnapshotPoliciesCmd cmd = Mockito.mock(DeleteSnapshotPoliciesCmd.class);
+        Mockito.when(cmd.getIds()).thenReturn(null);
+        return cmd;
+    }
+
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteSnapshotPoliciesNoPolicyId() {
-        DeleteSnapshotPoliciesCmd cmd = Mockito.mock(DeleteSnapshotPoliciesCmd.class);
+        DeleteSnapshotPoliciesCmd cmd = createMockDeleteSnapshotPoliciesCmd();
         Mockito.when(cmd.getId()).thenReturn(null);
-        Mockito.when(cmd.getIds()).thenReturn(null);
 
         snapshotManager.deleteSnapshotPolicies(cmd);
     }
@@ -610,9 +615,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testDeleteSnapshotPoliciesManualPolicyId() {
-        DeleteSnapshotPoliciesCmd cmd = Mockito.mock(DeleteSnapshotPoliciesCmd.class);
+        DeleteSnapshotPoliciesCmd cmd = createMockDeleteSnapshotPoliciesCmd();
         Mockito.when(cmd.getId()).thenReturn(Snapshot.MANUAL_POLICY_ID);
-        Mockito.when(cmd.getIds()).thenReturn(null);
 
         snapshotManager.deleteSnapshotPolicies(cmd);
     }
