--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -2520,10 +2520,8 @@
     public void processRemoveBackupFailuresTestFailureOnCurrentBackupNotForcedSetsError() {
         doReturn(backupId).when(internalBackupJoinVoMock).getId();
 
-        BackupDeleteAnswer failedCurrentBackupAnswer = Mockito.mock(BackupDeleteAnswer.class);
-        doReturn(false).when(failedCurrentBackupAnswer).getResult();
+        BackupDeleteAnswer failedCurrentBackupAnswer = createMockBackupDeleteAnswer();
         doReturn(backupId).when(failedCurrentBackupAnswer).getBackupId();
-        doReturn("delete failed").when(failedCurrentBackupAnswer).getDetails();
 
         doReturn(backupId).when(backupVoMock).getId();
         doReturn(backupVoMock).when(backupDaoMock).findByIdIncludingRemoved(backupId);
@@ -2541,10 +2539,8 @@
 
     @Test
     public void processRemoveBackupFailuresTestFailureOnCurrentBackupForcedSetBackupAsExpunged() {
-        BackupDeleteAnswer failedCurrentBackupAnswer = Mockito.mock(BackupDeleteAnswer.class);
-        doReturn(false).when(failedCurrentBackupAnswer).getResult();
+        BackupDeleteAnswer failedCurrentBackupAnswer = createMockBackupDeleteAnswer();
         doReturn(backupId).when(failedCurrentBackupAnswer).getBackupId();
-        doReturn("delete failed").when(failedCurrentBackupAnswer).getDetails();
         doReturn(backupVoMock).when(backupDaoMock).findByIdIncludingRemoved(backupId);
 
         List<Long> removedBackupIds = new ArrayList<>(List.of(backupId, 200L));
@@ -2562,10 +2558,8 @@
     public void processRemoveBackupFailuresTestFailureOnOtherBackupMarksItExpunged() {
         doReturn(backupId).when(internalBackupJoinVoMock).getId();
 
-        BackupDeleteAnswer failedOtherBackupAnswer = Mockito.mock(BackupDeleteAnswer.class);
-        doReturn(false).when(failedOtherBackupAnswer).getResult();
+        BackupDeleteAnswer failedOtherBackupAnswer = createMockBackupDeleteAnswer();
         doReturn(200L).when(failedOtherBackupAnswer).getBackupId();
-        doReturn("delete failed").when(failedOtherBackupAnswer).getDetails();
 
         BackupVO failedBackup = Mockito.mock(BackupVO.class);
         doReturn(failedBackup).when(backupDaoMock).findByIdIncludingRemoved(200L);
@@ -3113,4 +3107,11 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+    private static BackupDeleteAnswer createMockBackupDeleteAnswer() {
+        BackupDeleteAnswer backupDeleteAnswer = Mockito.mock(BackupDeleteAnswer.class);
+        doReturn(false).when(backupDeleteAnswer).getResult();
+        doReturn("delete failed").when(backupDeleteAnswer).getDetails();
+        return backupDeleteAnswer;
+    }
+
 }
