--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -643,12 +643,8 @@
             when(CallContext.current()).thenReturn(callContextMock);
             when(callContextMock.getCallingAccount()).thenReturn(account);
 
-            ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
+            ListUserDataCmd cmd = createMockListUserDataCmd();
             when(cmd.getName()).thenReturn("testSearchUserdataName");
-            when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
@@ -676,12 +672,8 @@
             when(CallContext.current()).thenReturn(callContextMock);
             when(callContextMock.getCallingAccount()).thenReturn(account);
 
-            ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
-            when(cmd.getAccountName()).thenReturn("testAccountName");
-            when(cmd.getDomainId()).thenReturn(1L);
-            when(cmd.getProjectId()).thenReturn(2L);
+            ListUserDataCmd cmd = createMockListUserDataCmd();
             when(cmd.getKeyword()).thenReturn("testSearchUserdataKeyword");
-            when(cmd.isRecursive()).thenReturn(false);
             UserDataVO userData = Mockito.mock(UserDataVO.class);
 
             SearchBuilder<UserDataVO> sb = Mockito.mock(SearchBuilder.class);
@@ -696,7 +688,7 @@
             Pair<List<UserDataVO>, Integer> result = new Pair(userDataList, 1);
             when(userDataDao.searchAndCount(nullable(SearchCriteria.class), nullable(Filter.class))).thenReturn(result);
 
-            Pair<List<? extends UserData>, Integer> userdataResultList = spy.listUserDatas(cmd, false);
+            Pair<List<? extends UserData>, Integer> userdataResultList = spy.listUserData(cmd, false);
 
             Assert.assertEquals(userdataResultList.first().get(0), userDataList.get(0));
         }
@@ -2466,6 +2458,15 @@
         return diskOffering;
     }
 
+    private static ListUserDataCmd createMockListUserDataCmd() {
+        ListUserDataCmd cmd = Mockito.mock(ListUserDataCmd.class);
+        when(cmd.getAccountName()).thenReturn("testAccountName");
+        when(cmd.getDomainId()).thenReturn(1L);
+        when(cmd.getProjectId()).thenReturn(2L);
+        when(cmd.isRecursive()).thenReturn(false);
+        return cmd;
+    }
+
     @Test
     public void createDeploymentPlanForMigrationListingTestAllocatesInAnyClusterWhenStorageMigrationIsSupported() {
         VMInstanceVO vm = mockRunningVM(1L, HypervisorType.KVM);
