--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -2629,9 +2629,7 @@
         doReturn(false).when(answer).isBootValidated();
         doReturn("false").when(answer).getScriptResult();
 
-        ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
-        doReturn(true).when(cmd).isWaitForBoot();
-        doReturn(true).when(cmd).isExecuteScript();
+        ValidateKbossVmCommand cmd = createMockValidateKbossVmCommand();
 
         boolean result = kbossBackupProviderSpy.processValidationAnswer(answer, backupVoMock, userVmVOMock, hostVOMock, cmd);
 
@@ -2650,9 +2648,7 @@
         doReturn("false").when(answer).getScriptResult();
         doReturn(null).when(answer).getScreenshotPath();
 
-        ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
-        doReturn(true).when(cmd).isWaitForBoot();
-        doReturn(true).when(cmd).isExecuteScript();
+        ValidateKbossVmCommand cmd = createMockValidateKbossVmCommand();
         doReturn(true).when(cmd).isTakeScreenshot();
 
         boolean result = kbossBackupProviderSpy.processValidationAnswer(answer, backupVoMock, userVmVOMock, hostVOMock, cmd);
@@ -2671,9 +2667,7 @@
         doReturn(null).when(answer).getScriptResult();
         doReturn("snap-path").when(answer).getScreenshotPath();
 
-        ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
-        doReturn(true).when(cmd).isWaitForBoot();
-        doReturn(true).when(cmd).isExecuteScript();
+        ValidateKbossVmCommand cmd = createMockValidateKbossVmCommand();
         doReturn(true).when(cmd).isTakeScreenshot();
 
         boolean result = kbossBackupProviderSpy.processValidationAnswer(answer, backupVoMock, userVmVOMock, hostVOMock, cmd);
@@ -3113,4 +3107,11 @@
         verify(kbossBackupProviderSpy).releaseBackup(backupId);
     }
 
+    private static ValidateKbossVmCommand createMockValidateKbossVmCommand() {
+        ValidateKbossVmCommand cmd = Mockito.mock(ValidateKbossVmCommand.class);
+        doReturn(true).when(cmd).isExecuteScript();
+        doReturn(true).when(cmd).isWaitForBoot();
+        return cmd;
+    }
+
 }
