--- a/server/src/test/java/org/apache/cloudstack/storage/object/BucketApiServiceImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/storage/object/BucketApiServiceImplTest.java
@@ -140,11 +140,9 @@
         Long objectStoreId = 3L;
         int quota = 1;
 
-        CreateBucketCmd cmd = Mockito.mock(CreateBucketCmd.class);
+        CreateBucketCmd cmd = createMockCreateBucketCmd(quota);
         Mockito.when(cmd.getBucketName()).thenReturn(bucketName);
         Mockito.when(cmd.getEntityOwnerId()).thenReturn(ACCOUNT_ID);
-        Mockito.when(cmd.getObjectStoragePoolId()).thenReturn(poolId);
-        Mockito.when(cmd.getQuota()).thenReturn(quota);
 
         ObjectStoreVO objectStoreVO = Mockito.mock(ObjectStoreVO.class);
         Mockito.when(objectStoreVO.getId()).thenReturn(objectStoreId);
@@ -259,4 +257,11 @@
                 .decrementResourceCount(ACCOUNT_ID, Resource.ResourceType.object_storage,
                         (bucketQuota - cmdQuota) * Resource.ResourceType.bytesToGiB);
     }
+
+    private static CreateBucketCmd createMockCreateBucketCmd(int quota) {
+        CreateBucketCmd cmd = Mockito.mock(CreateBucketCmd.class);
+        Mockito.when(cmd.getObjectStoragePoolId()).thenReturn(2L);
+        Mockito.when(cmd.getQuota()).thenReturn(quota);
+        return cmd;
+    }
 }
