--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -438,8 +438,7 @@
         Mockito.when(vm.getState()).thenReturn(vmState);
         Pair<String, VirtualMachine.State> vmNameAndState = new Pair<>(vmName, vmState);
 
-        Backup.VolumeInfo volumeInfo = mock(Backup.VolumeInfo.class);
-        when(volumeInfo.getUuid()).thenReturn(volumeUuid);
+        Backup.VolumeInfo volumeInfo = createMockVolumeInfo();
 
         doReturn(new Pair<Boolean, String>(Boolean.TRUE, "Success"))
             .when(backupProvider).restoreBackedUpVolume(any(Backup.class), any(Backup.VolumeInfo.class),
@@ -466,8 +465,7 @@
         Mockito.when(vm.getState()).thenReturn(vmState);
         Pair<String, VirtualMachine.State> vmNameAndState = new Pair<>(vmName, vmState);
 
-        Backup.VolumeInfo volumeInfo = mock(Backup.VolumeInfo.class);
-        when(volumeInfo.getUuid()).thenReturn(volumeUuid);
+        Backup.VolumeInfo volumeInfo = createMockVolumeInfo();
 
         doReturn(new Pair<Boolean, String>(Boolean.TRUE, "Success2"))
             .when(backupProvider).restoreBackedUpVolume(any(Backup.class), any(Backup.VolumeInfo.class),
@@ -494,8 +492,7 @@
         Mockito.when(vm.getState()).thenReturn(vmState);
         Pair<String, VirtualMachine.State> vmNameAndState = new Pair<>(vmName, vmState);
 
-        Backup.VolumeInfo volumeInfo = mock(Backup.VolumeInfo.class);
-        when(volumeInfo.getUuid()).thenReturn(volumeUuid);
+        Backup.VolumeInfo volumeInfo = createMockVolumeInfo();
 
         doReturn(new Pair<Boolean, String>(Boolean.TRUE, "Success3"))
             .when(backupProvider).restoreBackedUpVolume(any(Backup.class), any(Backup.VolumeInfo.class),
@@ -522,8 +519,7 @@
         Mockito.when(vm.getState()).thenReturn(vmState);
         Pair<String, VirtualMachine.State> vmNameAndState = new Pair<>(vmName, vmState);
 
-        Backup.VolumeInfo volumeInfo = mock(Backup.VolumeInfo.class);
-        when(volumeInfo.getUuid()).thenReturn(volumeUuid);
+        Backup.VolumeInfo volumeInfo = createMockVolumeInfo();
 
         doReturn(new Pair<Boolean, String>(Boolean.TRUE, "Success4"))
             .when(backupProvider).restoreBackedUpVolume(any(Backup.class), any(Backup.VolumeInfo.class),
@@ -2616,6 +2612,12 @@
             assertTrue(result.first().stream().anyMatch(o -> o.getName().equals("Global Offering")));
             assertTrue(result.first().stream().anyMatch(o -> o.getName().equals("User Domain Offering")));
         }
+    }
+
+    private static Backup.VolumeInfo createMockVolumeInfo() {
+        Backup.VolumeInfo volumeInfo = mock(Backup.VolumeInfo.class);
+        when(volumeInfo.getUuid()).thenReturn("5f4ed903-ac23-4f8a-b595-69c73c40593f");
+        return volumeInfo;
     }
 
     private BackupOfferingVO createMockOffering(Long id, String name) {
