--- a/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
+++ b/engine/orchestration/src/test/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestratorTest.java
@@ -383,8 +383,7 @@
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
 
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(pool.getScope()).thenReturn(ScopeType.CLUSTER); // Shared scope
+        StoragePoolVO pool = createMockStoragePoolVO(ScopeType.CLUSTER); // Shared scope
         Mockito.when(storagePoolDao.findById(MOCK_POOL_ID)).thenReturn(pool);
 
         assertTrue(volumeOrchestrator.volumeOnSharedStoragePool(volume));
@@ -395,8 +394,7 @@
         VolumeVO volume = Mockito.mock(VolumeVO.class);
         Mockito.when(volume.getPoolId()).thenReturn(MOCK_POOL_ID);
 
-        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
-        Mockito.when(pool.getScope()).thenReturn(ScopeType.HOST); // Local scope
+        StoragePoolVO pool = createMockStoragePoolVO(ScopeType.HOST);
         Mockito.when(storagePoolDao.findById(MOCK_POOL_ID)).thenReturn(pool);
 
         Assert.assertFalse(volumeOrchestrator.volumeOnSharedStoragePool(volume));
@@ -922,6 +920,12 @@
         } catch (InvocationTargetException e) {
             throw e.getCause();
         }
+    }
+
+    private static StoragePoolVO createMockStoragePoolVO(ScopeType scope) {
+        StoragePoolVO pool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(pool.getScope()).thenReturn(scope);
+        return pool;
     }
 
     private void setField(Object target, String fieldName, Object value) throws Exception {
