--- a/plugins/storage/object/cloudian/src/test/java/org/apache/cloudstack/storage/datastore/driver/CloudianHyperStoreObjectStoreDriverImplTest.java
+++ b/plugins/storage/object/cloudian/src/test/java/org/apache/cloudstack/storage/datastore/driver/CloudianHyperStoreObjectStoreDriverImplTest.java
@@ -293,10 +293,8 @@
         d2bucketList.add(bu3);
         d2bucketList.add(bu4);
         d2bucketList.add(bu5);
-        CloudianUserBucketUsage d1U1Usage = mock(CloudianUserBucketUsage.class);
-        when(d1U1Usage.getBuckets()).thenReturn(d1bucketList);
-        CloudianUserBucketUsage d2U1Usage = mock(CloudianUserBucketUsage.class);
-        when(d2U1Usage.getBuckets()).thenReturn(d2bucketList);
+        CloudianUserBucketUsage d1U1Usage = createMockCloudianUserBucketUsage(d1bucketList);
+        CloudianUserBucketUsage d2U1Usage = createMockCloudianUserBucketUsage(d2bucketList);
         List<CloudianUserBucketUsage> d1Usage = new ArrayList<CloudianUserBucketUsage>();
         d1Usage.add(d1U1Usage);
         List<CloudianUserBucketUsage> d2Usage = new ArrayList<CloudianUserBucketUsage>();
@@ -683,4 +681,10 @@
         CloudRuntimeException thrown = assertThrows(CloudRuntimeException.class, () -> cloudianHyperStoreObjectStoreDriverImpl.getS3ClientByBucketAndStore(bucket, TEST_STORE_ID));
         assertTrue(thrown.getMessage().contains("not found"));
     }
+
+    private static CloudianUserBucketUsage createMockCloudianUserBucketUsage(List<CloudianBucketUsage> bucketUsages) {
+        CloudianUserBucketUsage usage = mock(CloudianUserBucketUsage.class);
+        when(usage.getBuckets()).thenReturn(bucketUsages);
+        return usage;
+    }
 }
