--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/snapshot/DefaultSnapshotStrategyTest.java
@@ -81,6 +81,8 @@
     @Mock
     private SnapshotDataStoreVO snapshotDataStoreVOMock;
 
+    private SnapshotDataStoreVO snapshotStore3Mock;
+
     @Mock
     SnapshotZoneDao snapshotZoneDaoMock;
 
@@ -97,6 +99,7 @@
 
     @Before
     public void setup() {
+        snapshotStore3Mock = Mockito.mock(SnapshotDataStoreVO.class);
         mockSnapshotInfos.add(snapshotInfo1Mock);
         mockSnapshotInfos.add(snapshotInfo2Mock);
     }
@@ -468,10 +471,9 @@
         Mockito.when(snapshotStore2.getDataStoreId()).thenReturn(202L);
         Mockito.when(snapshotStore2.getRole()).thenReturn(DataStoreRole.Image);
 
-        SnapshotDataStoreVO snapshotStore3 = Mockito.mock(SnapshotDataStoreVO.class);
 
         Mockito.when(snapshotDataStoreDao.listReadyBySnapshot(1L, DataStoreRole.Image))
-            .thenReturn(List.of(snapshotStore1, snapshotStore2, snapshotStore3));
+            .thenReturn(List.of(snapshotStore1, snapshotStore2, snapshotStore3Mock));
 
         Mockito.when(dataStoreManager.getStoreZoneId(201L, DataStoreRole.Image)).thenReturn(111L);
         Mockito.when(dataStoreManager.getStoreZoneId(202L, DataStoreRole.Image)).thenReturn(100L);
@@ -546,10 +548,9 @@
         Mockito.when(snapshotStore1.getDataStoreId()).thenReturn(201L);
         Mockito.when(snapshotStore1.getRole()).thenReturn(DataStoreRole.Image);
 
-        SnapshotDataStoreVO snapshotStore2 = Mockito.mock(SnapshotDataStoreVO.class);
 
         Mockito.when(snapshotDataStoreDao.listReadyBySnapshot(1L, DataStoreRole.Image))
-            .thenReturn(List.of(snapshotStore1, snapshotStore2));
+            .thenReturn(List.of(snapshotStore1, snapshotStore3Mock));
 
         Mockito.when(dataStoreManager.getStoreZoneId(201L, DataStoreRole.Image)).thenReturn(100L);
 
