--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -1165,12 +1165,11 @@
         AccountManagerImpl.enableUserTwoFactorAuthentication = enableUserTwoFactorAuthentication;
         Mockito.when(enableUserTwoFactorAuthentication.valueIn(1L)).thenReturn(true);
 
-        UserTwoFactorAuthenticator totpProvider = Mockito.mock(UserTwoFactorAuthenticator.class);
+        UserTwoFactorAuthenticator totpProvider = createMockUserTwoFactorAuthenticator(userAccount, "EUJEAEDVOURFZTE6OGWVTJZMI54QGMIL");
         Map<String, UserTwoFactorAuthenticator> userTwoFactorAuthenticationProvidersMap = Mockito.mock(HashMap.class);
         Mockito.when(userTwoFactorAuthenticationProvidersMap.containsKey("totp")).thenReturn( true);
         Mockito.when(userTwoFactorAuthenticationProvidersMap.get("totp")).thenReturn(totpProvider);
         AccountManagerImpl.userTwoFactorAuthenticationProvidersMap = userTwoFactorAuthenticationProvidersMap;
-        Mockito.when(totpProvider.setup2FAKey(userAccount)).thenReturn("EUJEAEDVOURFZTE6OGWVTJZMI54QGMIL");
         Mockito.when(userDaoMock.createForUpdate()).thenReturn(userVoMock);
         Mockito.when(userDaoMock.update(userId, userVoMock)).thenReturn(true);
 
@@ -1194,12 +1193,11 @@
         AccountManagerImpl.enableUserTwoFactorAuthentication = enableUserTwoFactorAuthentication;
         Mockito.when(enableUserTwoFactorAuthentication.valueIn(1L)).thenReturn(true);
 
-        UserTwoFactorAuthenticator totpProvider = Mockito.mock(UserTwoFactorAuthenticator.class);
+        UserTwoFactorAuthenticator totpProvider = createMockUserTwoFactorAuthenticator(userAccount, "EUJEAEDVOURFZTE6OGWVTJZMI54QGMIL");
         Map<String, UserTwoFactorAuthenticator> userTwoFactorAuthenticationProvidersMap = Mockito.mock(HashMap.class);
         Mockito.when(userTwoFactorAuthenticationProvidersMap.containsKey("totp")).thenReturn( true);
         Mockito.when(userTwoFactorAuthenticationProvidersMap.get("totp")).thenReturn(totpProvider);
         AccountManagerImpl.userTwoFactorAuthenticationProvidersMap = userTwoFactorAuthenticationProvidersMap;
-        Mockito.when(totpProvider.setup2FAKey(userAccount)).thenReturn("EUJEAEDVOURFZTE6OGWVTJZMI54QGMIL");
         Mockito.when(userDaoMock.createForUpdate()).thenReturn(userVoMock);
         Mockito.when(userDaoMock.update(userId, userVoMock)).thenReturn(true);
 
@@ -1327,11 +1325,10 @@
         Mockito.when(enableUserTwoFactorAuthenticationMock.valueIn(1L)).thenReturn(true);
         Mockito.when(cmd.getEnable()).thenReturn(true);
 
-        UserTwoFactorAuthenticator staticpinProvider = Mockito.mock(UserTwoFactorAuthenticator.class);
+        UserTwoFactorAuthenticator staticpinProvider = createMockUserTwoFactorAuthenticator(userAccountVO, "345543");
         Map<String, UserTwoFactorAuthenticator> userTwoFactorAuthenticationProvidersMap = Mockito.mock(HashMap.class);
         Mockito.when(userTwoFactorAuthenticationProvidersMap.containsKey("staticpin")).thenReturn( true);
         Mockito.when(userTwoFactorAuthenticationProvidersMap.get("staticpin")).thenReturn(staticpinProvider);
-        Mockito.when(staticpinProvider.setup2FAKey(userAccountVO)).thenReturn("345543");
         Mockito.when(userDaoMock.createForUpdate()).thenReturn(userVoMock);
         AccountManagerImpl.userTwoFactorAuthenticationProvidersMap = userTwoFactorAuthenticationProvidersMap;
 
@@ -2289,4 +2286,10 @@
 
         accountManagerImpl.checkRoleEscalation(caller, requested);
     }
+
+    private static UserTwoFactorAuthenticator createMockUserTwoFactorAuthenticator(UserAccountVO userAccount, String setup2FAKeyReturn) {
+        UserTwoFactorAuthenticator provider = Mockito.mock(UserTwoFactorAuthenticator.class);
+        Mockito.when(provider.setup2FAKey(userAccount)).thenReturn(setup2FAKeyReturn);
+        return provider;
+    }
 }
