--- a/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/KvmNonManagedStorageSystemDataMotionTest.java
+++ b/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/KvmNonManagedStorageSystemDataMotionTest.java
@@ -231,11 +231,16 @@
     }
 
     private String prepareTestGetTemplateUuid() {
-        TemplateInfo templateImage = Mockito.mock(TemplateInfo.class);
         String expectedTemplateUuid = "template uuid";
-        Mockito.when(templateImage.getUuid()).thenReturn(expectedTemplateUuid);
+        TemplateInfo templateImage = createMockTemplateInfo(expectedTemplateUuid);
         Mockito.doReturn(templateImage).when(templateDataFactory).getTemplate(0l, DataStoreRole.Image);
         return expectedTemplateUuid;
+    }
+
+    private static TemplateInfo createMockTemplateInfo(String uuid) {
+        TemplateInfo templateInfo = Mockito.mock(TemplateInfo.class);
+        Mockito.when(templateInfo.getUuid()).thenReturn(uuid);
+        return templateInfo;
     }
 
     @Test
@@ -370,9 +375,8 @@
         DataStore sourceTemplateDataStore = Mockito.mock(DataStore.class);
         Mockito.lenient().when(sourceTemplateDataStore.getName()).thenReturn("sourceTemplateName");
 
-        TemplateInfo sourceTemplateInfo = Mockito.mock(TemplateInfo.class);
+        TemplateInfo sourceTemplateInfo = createMockTemplateInfo("uuid");
         Mockito.when(sourceTemplateInfo.getInstallPath()).thenReturn("installPath");
-        Mockito.when(sourceTemplateInfo.getUuid()).thenReturn("uuid");
         Mockito.when(sourceTemplateInfo.getId()).thenReturn(0l);
         Mockito.when(sourceTemplateInfo.getUrl()).thenReturn("url");
         Mockito.when(sourceTemplateInfo.getDisplayText()).thenReturn("display text");
