--- a/plugins/storage/volume/linstor/src/test/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImplTest.java
+++ b/plugins/storage/volume/linstor/src/test/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImplTest.java
@@ -111,9 +111,8 @@
 
     @Test
     public void testCanCopySnapshotToVolumeOnSamePrimary() {
-        DataStore primaryStore = mock(DataStore.class);
+        DataStore primaryStore = MockDataStore.createMockDataStore();
         when(primaryStore.getRole()).thenReturn(DataStoreRole.Primary);
-        when(primaryStore.getId()).thenReturn(1L);
 
         SnapshotInfo snapshot = mock(SnapshotInfo.class);
         when(snapshot.getType()).thenReturn(DataObjectType.SNAPSHOT);
@@ -133,9 +132,8 @@
 
     @Test
     public void testCanCopySnapshotToVolumeRejectsNonLinstor() {
-        DataStore primaryStore = mock(DataStore.class);
+        DataStore primaryStore = MockDataStore.createMockDataStore();
         when(primaryStore.getRole()).thenReturn(DataStoreRole.Primary);
-        when(primaryStore.getId()).thenReturn(1L);
 
         SnapshotInfo snapshot = mock(SnapshotInfo.class);
         when(snapshot.getType()).thenReturn(DataObjectType.SNAPSHOT);
@@ -155,11 +153,10 @@
 
     @Test
     public void testCanCopySnapshotToVolumeRejectsCrossPrimary() {
-        DataStore srcStore = mock(DataStore.class);
+        DataStore srcStore = MockDataStore.createMockDataStore();
         when(srcStore.getRole()).thenReturn(DataStoreRole.Primary);
-        when(srcStore.getId()).thenReturn(1L);
 
-        DataStore destStore = mock(DataStore.class);
+        DataStore destStore = MockDataStore.createMockDataStore();
         when(destStore.getRole()).thenReturn(DataStoreRole.Primary);
         when(destStore.getId()).thenReturn(2L);
 
--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/ScaleIOPrimaryDataStoreDriverTest.java
@@ -124,9 +124,8 @@
 
     @Test
     public void testSameScaleIOStorageInstance() {
-        DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
-        when(srcStore.getId()).thenReturn(1L);
+        DataStore srcStore = MockDataStore.createMockDataStore();
+        DataStore destStore = MockDataStore.createMockDataStore();
         when(destStore.getId()).thenReturn(2L);
 
         StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
@@ -147,9 +146,8 @@
 
     @Test
     public void testDifferentScaleIOStorageInstance() {
-        DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
-        when(srcStore.getId()).thenReturn(1L);
+        DataStore srcStore = MockDataStore.createMockDataStore();
+        DataStore destStore = MockDataStore.createMockDataStore();
         when(destStore.getId()).thenReturn(2L);
 
         StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
@@ -170,9 +168,8 @@
 
     @Test (expected = CloudRuntimeException.class)
     public void testCheckVolumeOnDifferentScaleIOStorageInstanceSystemIdShouldNotBeNull() {
-        DataStore srcStore = Mockito.mock(DataStore.class);
-        DataStore destStore = Mockito.mock(DataStore.class);
-        when(srcStore.getId()).thenReturn(1L);
+        DataStore srcStore = MockDataStore.createMockDataStore();
+        DataStore destStore = MockDataStore.createMockDataStore();
         when(destStore.getId()).thenReturn(2L);
 
         StoragePoolDetailVO srcPoolSystemIdDetail = Mockito.mock(StoragePoolDetailVO.class);
--- a/plugins/storage/volume/linstor/src/test/java/org/apache/cloudstack/storage/datastore/driver/MockDataStore.java
+++ b/plugins/storage/volume/linstor/src/test/java/org/apache/cloudstack/storage/datastore/driver/MockDataStore.java
@@ -0,0 +1,14 @@
+package org.apache.cloudstack.storage.datastore.driver;
+
+import org.apache.cloudstack.engine.subsystem.api.storage.DataStore;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockDataStore {
+    public static DataStore createMockDataStore() {
+        DataStore srcStore = Mockito.mock(DataStore.class);
+        when(srcStore.getId()).thenReturn(1L);
+        return srcStore;
+    }
+}
--- a/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/MockDataStore.java
+++ b/plugins/storage/volume/scaleio/src/test/java/org/apache/cloudstack/storage/datastore/driver/MockDataStore.java
@@ -0,0 +1,14 @@
+package org.apache.cloudstack.storage.datastore.driver;
+
+import org.apache.cloudstack.engine.subsystem.api.storage.DataStore;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.when;
+
+public class MockDataStore {
+    public static DataStore createMockDataStore() {
+        DataStore srcStore = Mockito.mock(DataStore.class);
+        when(srcStore.getId()).thenReturn(1L);
+        return srcStore;
+    }
+}
