--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -175,6 +175,8 @@
     @Mock
     private InternalBackupDataStoreVO internalBackupDataStoreVoMock;
 
+    private InternalBackupDataStoreVO parentDeltaMock;
+
     @Mock
     private InternalBackupStoragePoolDao internalBackupStoragePoolDaoMock;
 
@@ -303,6 +305,7 @@
 
     @Before
     public void setup() {
+        parentDeltaMock = mock(InternalBackupDataStoreVO.class);
         doReturn(vmId).when(virtualMachineMock).getId();
         doReturn(vmId).when(backupVoMock).getVmId();
         doReturn(vmId).when(internalBackupJoinVoMock).getVmId();
@@ -1239,8 +1242,7 @@
         InternalBackupJoinVO parentVo = Mockito.mock(InternalBackupJoinVO.class);
         doReturn(parentVo).when(internalBackupJoinDaoMock).findById(parentId);
         doReturn(List.of(internalBackupDataStoreVoMock)).when(internalBackupDataStoreDaoMock).listByBackupId(backupId);
-        InternalBackupDataStoreVO parentDelta = mock(InternalBackupDataStoreVO.class);
-        doReturn(List.of(parentDelta)).when(internalBackupDataStoreDaoMock).listByBackupId(parentId);
+        doReturn(List.of(parentDeltaMock)).when(internalBackupDataStoreDaoMock).listByBackupId(parentId);
         doReturn(dataStoreMock).when(dataStoreManagerMock).getDataStore(anyLong(), any());
         doReturn(hostVOMock).when(hostDaoMock).findById(anyLong());
         doReturn(backupOfferingMock).when(backupOfferingDaoMock).findByIdIncludingRemoved(anyLong());
@@ -1266,8 +1268,7 @@
         InternalBackupJoinVO parentVo = Mockito.mock(InternalBackupJoinVO.class);
         doReturn(parentVo).when(internalBackupJoinDaoMock).findById(parentId);
         doReturn(List.of(internalBackupDataStoreVoMock)).when(internalBackupDataStoreDaoMock).listByBackupId(backupId);
-        InternalBackupDataStoreVO parentDelta = mock(InternalBackupDataStoreVO.class);
-        doReturn(List.of(parentDelta)).when(internalBackupDataStoreDaoMock).listByBackupId(parentId);
+        doReturn(List.of(parentDeltaMock)).when(internalBackupDataStoreDaoMock).listByBackupId(parentId);
         doReturn(dataStoreMock).when(dataStoreManagerMock).getDataStore(anyLong(), any());
         doReturn(hostVOMock).when(hostDaoMock).findById(anyLong());
         doReturn(backupOfferingMock).when(backupOfferingDaoMock).findByIdIncludingRemoved(anyLong());
