--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricApplicationPolicySetCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricApplicationPolicySetCmdTest.java
@@ -41,6 +41,9 @@
     @Mock
     TungstenService tungstenService;
 
+    @Mock
+    private BaseResponse baseResponseMock;
+
     ListTungstenFabricApplictionPolicySetCmd listTungstenFabricApplictionPolicySetCmd;
 
     AutoCloseable closeable;
@@ -64,8 +67,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricApplictionPolicySetCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         Mockito.when(tungstenService.listTungstenApplicationPolicySet(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricApplictionPolicySetCmd.execute();
@@ -76,8 +78,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
