--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -259,6 +259,8 @@
     @Mock
     private UserDataVO userDataMock;
 
+    private StoragePoolVO mockStoragePool;
+
     public class CustomThreadPoolExecutor extends ThreadPoolExecutor {
         AtomicInteger ai = new AtomicInteger(0);
         public CustomThreadPoolExecutor(int corePoolSize, int maximumPoolSize, long keepAliveTime, TimeUnit unit,
@@ -286,6 +288,7 @@
 
     @Before
     public void setUp() {
+        mockStoragePool = mock(StoragePoolVO.class);
         AccountVO account = new AccountVO("admin", 1L, "networkDomain", Account.Type.NORMAL, "uuid");
         UserVO user = new UserVO(1, "testuser", "password", "firstname", "lastName", "email", "timezone", UUID.randomUUID().toString(), User.Source.UNKNOWN);
         CallContext.register(user, account);
@@ -354,7 +357,6 @@
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         when(mockTemplate.getId()).thenReturn(202l);
 
-        StoragePoolVO mockPool = mock(StoragePoolVO.class);
 
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
@@ -376,7 +378,6 @@
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         when(mockTemplate.getId()).thenReturn(202l);
 
-        StoragePoolVO mockPool = mock(StoragePoolVO.class);
 
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
@@ -395,7 +396,6 @@
         VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
         when(mockTemplate.getId()).thenReturn(202l);
 
-        StoragePoolVO mockPool = mock(StoragePoolVO.class);
 
         PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
         when(mockPrimaryDataStore.getId()).thenReturn(2l);
