--- a/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
+++ b/engine/storage/snapshot/src/test/java/org/apache/cloudstack/storage/vmsnapshot/VMSnapshotStrategyKVMTest.java
@@ -146,11 +146,13 @@
     StorageVMSnapshotStrategy vmStrategy;
     @Inject
     VMSnapshotDetailsDao vmSnapshotDetailsDao;
+    private SnapshotInfo snapshotInfoMock;
 
     @Override
     @Before
     public void setUp() throws Exception {
         ComponentContext.initComponentsLifeCycle();
+        snapshotInfoMock = Mockito.mock(SnapshotInfo.class);
     }
 
     @Test
@@ -158,7 +160,6 @@
         VMSnapshotVO vmSnapshot = Mockito.mock(VMSnapshotVO.class);
         List<SnapshotInfo> snapshotsForRollback = new ArrayList<>();
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
-        SnapshotInfo snapshotInfo = Mockito.mock(SnapshotInfo.class);
         SnapshotStrategy strategy = Mockito.mock(SnapshotStrategy.class);
         DataStore dataStore = Mockito.mock(DataStore.class);
         String volUuid = UUID.randomUUID().toString();
@@ -171,12 +172,12 @@
         when(vol.getUuid()).thenReturn(volUuid);
         when(_snapshotDao.persist(any())).thenReturn(snapshot);
         when(vol.getDataStore()).thenReturn(dataStore);
-        when(_snapshotDataFactory.getSnapshot(snapshot.getId(), vol.getDataStore())).thenReturn(snapshotInfo);
-        when(_storageStrategyFactory.getSnapshotStrategy(snapshotInfo, SnapshotOperation.TAKE)).thenReturn(strategy);
+        when(_snapshotDataFactory.getSnapshot(snapshot.getId(), vol.getDataStore())).thenReturn(snapshotInfoMock);
+        when(_storageStrategyFactory.getSnapshotStrategy(snapshotInfoMock, SnapshotOperation.TAKE)).thenReturn(strategy);
 
         SnapshotInfo info = null;
 
-        when(strategy.takeSnapshot(any())).thenReturn(snapshotInfo);
+        when(strategy.takeSnapshot(any())).thenReturn(snapshotInfoMock);
         VMSnapshotDetailsVO vmDetails = new VMSnapshotDetailsVO(vmSnapshot.getId(), volUuid, String.valueOf(snapshot.getId()), false);
         when(vmSnapshotDetailsDao.persist(any())).thenReturn(vmDetails);
 
@@ -247,7 +248,6 @@
         VolumeInfo vol = Mockito.mock(VolumeInfo.class);
         SnapshotVO snapshotVO = Mockito.mock(SnapshotVO.class);
         Snapshot snap = Mockito.mock(Snapshot.class);
-        SnapshotInfo info = Mockito.mock(SnapshotInfo.class);
         SnapshotStrategy strategy = Mockito.mock(SnapshotStrategy.class);
         String volUuid = UUID.randomUUID().toString();
         String vmUuid = UUID.randomUUID().toString();
@@ -255,8 +255,8 @@
         when(vol.getUuid()).thenReturn(volUuid);
         when(vmSnapshot.getUuid()).thenReturn(vmUuid);
         when(snapshotVO.getId()).thenReturn(1L);
-        when( _snapshotDataFactory.getSnapshot(snapshotVO.getId(), vol.getDataStore())).thenReturn(info);
-        when(_storageStrategyFactory.getSnapshotStrategy(info, SnapshotOperation.DELETE)).thenReturn(strategy);
+        when( _snapshotDataFactory.getSnapshot(snapshotVO.getId(), vol.getDataStore())).thenReturn(snapshotInfoMock);
+        when(_storageStrategyFactory.getSnapshotStrategy(snapshotInfoMock, SnapshotOperation.DELETE)).thenReturn(strategy);
         testFindSnapshotByName(name);
         vmStrategy.deleteDiskSnapshot(vmSnapshot);
     }
