--- a/server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java
@@ -54,8 +54,7 @@
     private RolePermissionsDao rolePermissionsDaoMock;
     @Mock
     private RolePermission rolePermission1Mock;
-    @Mock
-    private RolePermission rolePermission2Mock;
+    private RolePermission rolePermission2Mock = createMockRolePermission2();
     @Mock
     private RolePermission rolePermission2MockWithDeniedPermission;
     @Mock
@@ -88,10 +87,8 @@
         Mockito.doReturn(callerAccountMock).when(roleManagerImpl).getCurrentAccount();
         Mockito.doReturn(callerAccountRoleMock.getId()).when(callerAccountMock).getRoleId();
 
+        rolePermission1Mock = createMockRolePermission2();
         Mockito.when(rolePermission1Mock.getRule()).thenReturn(new Rule("api1"));
-        Mockito.when(rolePermission2Mock.getRule()).thenReturn(new Rule("api2"));
-        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission1Mock).getPermission();
-        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission2Mock).getPermission();
         Mockito.doReturn(RolePermissionEntity.Permission.DENY).when(rolePermission2MockWithDeniedPermission).getPermission();
 
         Mockito.doReturn(1L).when(lessPermissionsRoleMock).getId();
@@ -379,4 +376,11 @@
         Mockito.verify(roleDaoMock).listAll();
         Mockito.verify(roleManagerImpl).removeRolesIfNeeded(roles);
     }
+
+    private static RolePermission createMockRolePermission2() {
+        RolePermission rolePermission2Mock = Mockito.mock(RolePermission.class);
+        Mockito.when(rolePermission2Mock.getRule()).thenReturn(new Rule("api2"));
+        Mockito.doReturn(RolePermissionEntity.Permission.ALLOW).when(rolePermission2Mock).getPermission();
+        return rolePermission2Mock;
+    }
 }
