--- a/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
+++ b/server/src/test/java/com/cloud/server/ManagementServerImplTest.java
@@ -304,6 +304,12 @@
         closeable.close();
     }
 
+    private static UpdateGuestOsCategoryCmd createMockUpdateGuestOsCategory() {
+        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
+        Mockito.when(updateCmd.getId()).thenReturn(1L);
+        return updateCmd;
+    }
+
     private void overrideDefaultConfigValue(final ConfigKey configKey, final String name, final Object o) throws IllegalAccessException, NoSuchFieldException {
         Field f = ConfigKey.class.getDeclaredField(name);
         f.setAccessible(true);
@@ -890,12 +896,12 @@
 
     @Test
     public void testUpdateGuestOsCategory() {
-        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
         GuestOSCategoryVO guestOSCategory = new GuestOSCategoryVO("Old name", false);
         long id = 1L;
         String name = "Updated Name";
         Boolean featured = true;
         Integer sortKey = 10;
+        UpdateGuestOsCategoryCmd updateCmd = createMockUpdateGuestOsCategory();
         Mockito.when(updateCmd.getId()).thenReturn(id);
         Mockito.when(updateCmd.getName()).thenReturn(name);
         Mockito.when(updateCmd.isFeatured()).thenReturn(featured);
@@ -912,9 +918,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testUpdateGuestOsCategory_ThrowsExceptionWhenCategoryNotFound() {
-        UpdateGuestOsCategoryCmd updateCmd = Mockito.mock(UpdateGuestOsCategoryCmd.class);
         long id = 1L;
-        when(updateCmd.getId()).thenReturn(id);
+        UpdateGuestOsCategoryCmd updateCmd = createMockUpdateGuestOsCategory();
         when(guestOSCategoryDao.findById(id)).thenReturn(null);
         spy.updateGuestOsCategory(updateCmd);
     }
