--- a/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
+++ b/server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java
@@ -705,6 +705,12 @@
         Assert.assertEquals(uuid, rule.getUuid());
     }
 
+    private static CreateIpv6FirewallRuleCmd createMockCreateIpv6FirewallRuleCmd() {
+        CreateIpv6FirewallRuleCmd cmd = Mockito.mock(CreateIpv6FirewallRuleCmd.class);
+        Mockito.when(cmd.getSourcePortEnd()).thenReturn(800000);
+        return cmd;
+    }
+
     private void registerCallContext() {
         account = new AccountVO("testaccount", 1L, "networkdomain", Account.Type.NORMAL, "uuid");
         account.setId(ACCOUNT_ID);
@@ -752,8 +758,7 @@
     @Test(expected = InvalidParameterValueException.class)
     public void testEndPortCidrCreateIpv6FirewallRule() {
         registerCallContext();
-        CreateIpv6FirewallRuleCmd cmd = Mockito.mock(CreateIpv6FirewallRuleCmd.class);
-        Mockito.when(cmd.getSourcePortEnd()).thenReturn(800000);
+        CreateIpv6FirewallRuleCmd cmd = createMockCreateIpv6FirewallRuleCmd();
         try {
             ipv6Service.createIpv6FirewallRule(cmd);
         } catch (NetworkRuleConflictException e) {
@@ -764,7 +769,7 @@
     @Test(expected = InvalidParameterValueException.class)
     public void testPortRangeCidrCreateIpv6FirewallRule() {
         registerCallContext();
-        CreateIpv6FirewallRuleCmd cmd = Mockito.mock(CreateIpv6FirewallRuleCmd.class);
+        CreateIpv6FirewallRuleCmd cmd = createMockCreateIpv6FirewallRuleCmd();
         Mockito.when(cmd.getSourcePortStart()).thenReturn(900);
         Mockito.when(cmd.getSourcePortEnd()).thenReturn(800);
         try {
