--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNetworkCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNetworkCmdTest.java
@@ -41,6 +41,8 @@
     @Mock
     TungstenService tungstenService;
 
+    private BaseResponse baseResponseMock;
+
     ListTungstenFabricNetworkCmd listTungstenFabricNetworkCmd;
 
     AutoCloseable closeable;
@@ -48,6 +50,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        baseResponseMock = Mockito.mock(BaseResponse.class);
         listTungstenFabricNetworkCmd = new ListTungstenFabricNetworkCmd();
         listTungstenFabricNetworkCmd.tungstenService = tungstenService;
         ReflectionTestUtils.setField(listTungstenFabricNetworkCmd, "networkUuid", "test");
@@ -65,8 +68,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricNetworkCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
 
         Mockito.when(tungstenService.listTungstenNetwork(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString(), ArgumentMatchers.anyBoolean())).thenReturn(baseResponseList);
@@ -79,8 +81,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
