--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -573,11 +573,9 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testLinkUserDataToTemplateByProvidingBothISOAndTemplateId() {
-        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
+        LinkUserDataToTemplateCmd cmd = createMockLinkUserDataToTemplateCmd();
         when(cmd.getTemplateId()).thenReturn(1L);
         when(cmd.getIsoId()).thenReturn(1L);
-        when(cmd.getUserdataId()).thenReturn(2L);
-        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
 
@@ -586,11 +584,9 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testLinkUserDataToTemplateByNotProvidingBothISOAndTemplateId() {
-        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
+        LinkUserDataToTemplateCmd cmd = createMockLinkUserDataToTemplateCmd();
         when(cmd.getTemplateId()).thenReturn(null);
         when(cmd.getIsoId()).thenReturn(null);
-        when(cmd.getUserdataId()).thenReturn(2L);
-        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
 
@@ -674,6 +670,13 @@
         when(adapter.getName()).thenReturn(TemplateAdapter.TemplateAdapterType.Hypervisor.getName());
         templateManager.setTemplateAdapters(new ArrayList<>(List.of(adapter)));
         return adapter;
+    }
+
+    private static LinkUserDataToTemplateCmd createMockLinkUserDataToTemplateCmd() {
+        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
+        when(cmd.getUserdataId()).thenReturn(2L);
+        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
+        return cmd;
     }
 
     @Test
