--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -208,6 +208,8 @@
     @Mock
     private VolumeObjectTO volumeObjectToMock;
 
+    private VolumeObjectTO vol2Mock;
+
     @Mock
     private VirtualMachineManager virtualMachineManagerMock;
 
@@ -303,6 +305,7 @@
 
     @Before
     public void setup() {
+        vol2Mock = Mockito.mock(VolumeObjectTO.class);
         doReturn(vmId).when(virtualMachineMock).getId();
         doReturn(vmId).when(backupVoMock).getVmId();
         doReturn(vmId).when(internalBackupJoinVoMock).getVmId();
@@ -781,8 +784,7 @@
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
         VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
-        VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(vol1, vol2Mock)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
@@ -806,8 +808,7 @@
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
         VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
-        VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(vol1, vol2Mock)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
@@ -839,8 +840,7 @@
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
         VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
-        VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(vol1, vol2Mock)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
