--- a/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
+++ b/server/src/test/java/com/cloud/configuration/ConfigurationManagerTest.java
@@ -1284,7 +1284,7 @@
     @Test(expected = CloudRuntimeException.class)
     public void testUsedDeleteDataCenterGuestIpv6Prefix() {
         final Long prefixId = 1L;
-        DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
+        DeleteGuestNetworkIpv6PrefixCmd cmd = createMockDeleteGuestNetworkIpv6PrefixCmd();
         Mockito.when(cmd.getId()).thenReturn(prefixId);
         DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
         Mockito.when(prefixVO.getId()).thenReturn(prefixId);
@@ -1297,7 +1297,7 @@
     @Test
     public void testDeleteDataCenterGuestIpv6Prefix() {
         final Long prefixId = 1L;
-        DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
+        DeleteGuestNetworkIpv6PrefixCmd cmd = createMockDeleteGuestNetworkIpv6PrefixCmd();
         Mockito.when(cmd.getId()).thenReturn(prefixId);
         DataCenterGuestIpv6PrefixVO prefixVO = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
         Mockito.when(prefixVO.getId()).thenReturn(prefixId);
@@ -1508,4 +1508,10 @@
                 "Validation failed: Command 'DhcpEntryCommand600' does not contain '='.",
                 result.second());
     }
+
+    private static DeleteGuestNetworkIpv6PrefixCmd createMockDeleteGuestNetworkIpv6PrefixCmd() {
+        DeleteGuestNetworkIpv6PrefixCmd cmd = Mockito.mock(DeleteGuestNetworkIpv6PrefixCmd.class);
+        Mockito.when(cmd.getId()).thenReturn(1L);
+        return cmd;
+    }
 }
