--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -292,6 +292,8 @@
     @Mock
     private DeltaMergeTreeTO deltaMergeTreeToMock;
 
+    private CallContext callContextMockShared;
+
     @Spy
     @InjectMocks
     private KbossBackupProvider kbossBackupProviderSpy;
@@ -303,6 +305,7 @@
 
     @Before
     public void setup() {
+        callContextMockShared = Mockito.mock(CallContext.class);
         doReturn(vmId).when(virtualMachineMock).getId();
         doReturn(vmId).when(backupVoMock).getVmId();
         doReturn(vmId).when(internalBackupJoinVoMock).getVmId();
@@ -1817,8 +1820,7 @@
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
-            callContextMocked.when(CallContext::current).thenReturn(callContextMock);
+            callContextMocked.when(CallContext::current).thenReturn(callContextMockShared);
 
             kbossBackupProviderSpy.cleanupValidation(false, userVmVOMock, backupVoMock, List.of(dataVolume));
 
@@ -1841,8 +1843,7 @@
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
-            callContextMocked.when(CallContext::current).thenReturn(callContextMock);
+            callContextMocked.when(CallContext::current).thenReturn(callContextMockShared);
 
             kbossBackupProviderSpy.cleanupValidation(true, userVmVOMock, backupVoMock, List.of(dataVolume));
 
@@ -1863,8 +1864,7 @@
         doNothing().when(kbossBackupProviderSpy).sendCleanupFailedEmail(any(), any());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
-            callContextMocked.when(CallContext::current).thenReturn(callContextMock);
+            callContextMocked.when(CallContext::current).thenReturn(callContextMockShared);
 
             kbossBackupProviderSpy.cleanupValidation(true, userVmVOMock, backupVoMock, List.of(dataVolume));
 
@@ -1891,8 +1891,7 @@
         doReturn(volumeVoMock).when(volumeApiServiceMock).destroyVolume(Mockito.eq(52L), any(), Mockito.eq(true), Mockito.eq(true), Mockito.isNull());
 
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
-            CallContext callContextMock = Mockito.mock(CallContext.class);
-            callContextMocked.when(CallContext::current).thenReturn(callContextMock);
+            callContextMocked.when(CallContext::current).thenReturn(callContextMockShared);
 
             kbossBackupProviderSpy.cleanupValidation(true, userVmVOMock, backupVoMock, List.of(rootVolume, dataVolume));
 
