--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookApiServiceImplTest.java
@@ -139,6 +139,13 @@
         return webhookJoinVO;
     }
 
+    private static WebhookVO createMockWebhookVO() {
+        WebhookVO webhookVO = Mockito.mock(WebhookVO.class);
+        Mockito.when(webhookVO.getUuid()).thenReturn("webhookUuid");
+        Mockito.when(webhookVO.getName()).thenReturn("webhookName");
+        return webhookVO;
+    }
+
     private void validateWebhookResponseWithWebhookJoinVO(WebhookResponse response, WebhookJoinVO webhookJoinVO) {
         Assert.assertEquals(webhookJoinVO.getName(), ReflectionTestUtils.getField(response, "name"));
         Assert.assertEquals(webhookJoinVO.getDescription(), ReflectionTestUtils.getField(response, "description"));
@@ -697,9 +704,7 @@
         Mockito.when(webhookFilter.getValue()).thenReturn("value");
         Mockito.when(webhookFilter.getCreated()).thenReturn(new Date(1000));
 
-        WebhookVO webhookVO = Mockito.mock(WebhookVO.class);
-        Mockito.when(webhookVO.getUuid()).thenReturn("webhookUuid");
-        Mockito.when(webhookVO.getName()).thenReturn("webhookName");
+        WebhookVO webhookVO = createMockWebhookVO();
 
         WebhookFilterResponse response = webhookApiServiceImpl.createWebhookFilterResponse(webhookFilter, webhookVO);
 
@@ -725,9 +730,7 @@
         Mockito.when(webhookFilter.getValue()).thenReturn("value");
         Mockito.when(webhookFilter.getCreated()).thenReturn(new Date(1000));
 
-        WebhookVO webhookVO = Mockito.mock(WebhookVO.class);
-        Mockito.when(webhookVO.getUuid()).thenReturn("webhookUuid");
-        Mockito.when(webhookVO.getName()).thenReturn("webhookName");
+        WebhookVO webhookVO = createMockWebhookVO();
         Mockito.when(webhookDao.findById(1L)).thenReturn(webhookVO);
 
         WebhookFilterResponse response = webhookApiServiceImpl.createWebhookFilterResponse(webhookFilter, null);
