--- a/server/src/test/java/org/apache/cloudstack/schedule/ResourceScheduleManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/schedule/ResourceScheduleManagerImplTest.java
@@ -88,11 +88,14 @@
     @Mock
     EntityManager entityManager;
 
+    private ResourceScheduleVO schedule2Mock;
+
     private AutoCloseable closeable;
 
     @Before
     public void setUp() {
         closeable = MockitoAnnotations.openMocks(this);
+        schedule2Mock = Mockito.mock(ResourceScheduleVO.class);
         Account callingAccount = Mockito.mock(Account.class);
         User callingUser = Mockito.mock(User.class);
         CallContext.register(callingUser, callingAccount);
@@ -173,10 +176,9 @@
     public void listSchedule() {
         VirtualMachine vm = Mockito.mock(VirtualMachine.class);
         ResourceScheduleVO schedule1 = Mockito.mock(ResourceScheduleVO.class);
-        ResourceScheduleVO schedule2 = Mockito.mock(ResourceScheduleVO.class);
         List<ResourceScheduleVO> scheduleList = new ArrayList<>();
         scheduleList.add(schedule1);
-        scheduleList.add(schedule2);
+        scheduleList.add(schedule2Mock);
 
         Mockito.when(vm.getUuid()).thenReturn(UUID.randomUUID().toString());
         Mockito.when(vm.getId()).thenReturn(1L);
@@ -351,10 +353,9 @@
                 Mockito.any(), Mockito.anyLong(), Mockito.anyList())).thenReturn(1L);
 
         ResourceScheduleVO schedule1 = Mockito.mock(ResourceScheduleVO.class);
-        ResourceScheduleVO schedule2 = Mockito.mock(ResourceScheduleVO.class);
         List<ResourceScheduleVO> scheduleList = new ArrayList<>();
         scheduleList.add(schedule1);
-        scheduleList.add(schedule2);
+        scheduleList.add(schedule2Mock);
 
         Mockito.when(resourceScheduleDao.searchAndCount(
                 Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(),
