--- a/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
+++ b/plugins/database/quota/src/test/java/org/apache/cloudstack/api/response/QuotaResponseBuilderImplTest.java
@@ -225,6 +225,12 @@
         return tariffVO;
     }
 
+    private static QuotaCreditsCmd createMockQuotaCreditsCmd(double value) {
+        QuotaCreditsCmd cmd = Mockito.mock(QuotaCreditsCmd.class);
+        Mockito.doReturn(value).when(cmd).getValue();
+        return cmd;
+    }
+
     @Test
     public void testQuotaResponse() {
         QuotaTariffVO tariffVO = makeTariffTestData();
@@ -1170,8 +1176,7 @@
 
     @Test
     public void addQuotaCreditsTestValidParameters() {
-        QuotaCreditsCmd cmd = Mockito.mock(QuotaCreditsCmd.class);
-        Mockito.doReturn(10D).when(cmd).getValue();
+        QuotaCreditsCmd cmd = createMockQuotaCreditsCmd(10D);
         Mockito.doReturn(BigDecimal.TEN).when(quotaCreditsVoMock).getCredit();
         Mockito.doReturn(accountMock).when(accountDaoMock).findById(Mockito.anyLong());
         Mockito.doReturn(null).when(quotaBalanceDaoMock).findLaterBalanceEntry(Mockito.anyLong(), Mockito.anyLong(),
@@ -1187,7 +1192,7 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void addQuotaCreditsTestThrowsExceptionWhenValueIsNull() {
-        QuotaCreditsCmd cmd = Mockito.mock(QuotaCreditsCmd.class);
+        QuotaCreditsCmd cmd = createMockQuotaCreditsCmd(100.0);
         Mockito.doReturn(null).when(cmd).getValue();
 
         quotaResponseBuilderSpy.addQuotaCredits(cmd);
