--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/KbossBackupProviderTest.java
@@ -208,6 +208,8 @@
     @Mock
     private VolumeObjectTO volumeObjectToMock;
 
+    private VolumeObjectTO vol1Mock;
+
     @Mock
     private VirtualMachineManager virtualMachineManagerMock;
 
@@ -303,6 +305,7 @@
 
     @Before
     public void setup() {
+        vol1Mock = Mockito.mock(VolumeObjectTO.class);
         doReturn(vmId).when(virtualMachineMock).getId();
         doReturn(vmId).when(backupVoMock).getVmId();
         doReturn(vmId).when(internalBackupJoinVoMock).getVmId();
@@ -780,9 +783,8 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(vol1Mock, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
@@ -805,9 +807,8 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(vol1Mock, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
@@ -838,9 +839,8 @@
         Mockito.when(hostVOMock.getResourceState()).thenReturn(ResourceState.Enabled);
         doNothing().when(kbossBackupProviderSpy).validateVmState(any(), any());
 
-        VolumeObjectTO vol1 = Mockito.mock(VolumeObjectTO.class);
         VolumeObjectTO vol2 = Mockito.mock(VolumeObjectTO.class);
-        doReturn(List.of(vol1, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
+        doReturn(List.of(vol1Mock, vol2)).when(vmSnapshotHelperMock).getVolumeTOList(Mockito.anyLong());
 
         doNothing().when(kbossBackupProviderSpy).validateStorages(any(), any());
         doReturn(internalBackupJoinVoMock).when(internalBackupJoinDaoMock).findById(any());
