--- a/server/src/test/java/org/apache/cloudstack/annotation/AnnotationManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/annotation/AnnotationManagerImplTest.java
@@ -155,9 +155,8 @@
         Role role = Mockito.mock(Role.class);
         Mockito.when(role.getRoleType()).thenReturn(RoleType.User);
         Mockito.when(roleService.findRole(Mockito.anyLong())).thenReturn(role);
-        AnnotationVO annotationVO = Mockito.mock(AnnotationVO.class);
         AnnotationService.EntityType type = AnnotationService.EntityType.VOLUME;
-        Mockito.when(annotationVO.getEntityType()).thenReturn(type);
+        AnnotationVO annotationVO = createMockAnnotationVO(type);
         Mockito.when(annotationDao.persist(Mockito.any())).thenReturn(annotationVO);
         Mockito.when(entityManager.findByUuid(Volume.class, uuid)).thenReturn(volume);
         annotationManager.addAnnotation("Some text", type, uuid, false);
@@ -175,8 +174,8 @@
         Role role = Mockito.mock(Role.class);
         Mockito.when(role.getRoleType()).thenReturn(RoleType.User);
         Mockito.when(roleService.findRole(Mockito.anyLong())).thenReturn(role);
-        AnnotationVO annotationVO = Mockito.mock(AnnotationVO.class);
         AnnotationService.EntityType type = AnnotationService.EntityType.VM;
+        AnnotationVO annotationVO = createMockAnnotationVO(type);
         Mockito.when(annotationVO.getUuid()).thenReturn(annotationUuid);
         Mockito.when(annotationVO.getUserUuid()).thenReturn(user.getUuid());
         Mockito.when(annotationVO.getEntityType()).thenReturn(type);
@@ -189,4 +188,10 @@
         Assert.assertEquals(ENTITY_ID, CallContext.current().getEventResourceId());
         Assert.assertEquals(ApiCommandResourceType.VirtualMachine, CallContext.current().getEventResourceType());
     }
+
+    private static AnnotationVO createMockAnnotationVO(AnnotationService.EntityType entityType) {
+        AnnotationVO annotationVO = Mockito.mock(AnnotationVO.class);
+        Mockito.when(annotationVO.getEntityType()).thenReturn(entityType);
+        return annotationVO;
+    }
 }
