--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirTakeKbossBackupCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirTakeKbossBackupCommandWrapperTest.java
@@ -369,9 +369,15 @@
     }
 
     private void setupKbossTos() {
+        volumeObjectToMock1 = createMockVolumeObjectTo(volUuid1);
         doReturn(volumeObjectToMock1).when(kbossTO1).getVolumeObjectTO();
-        doReturn(volUuid1).when(volumeObjectToMock1).getUuid();
+        volumeObjectToMock2 = createMockVolumeObjectTo(volUuid2);
         doReturn(volumeObjectToMock2).when(kbossTO2).getVolumeObjectTO();
-        doReturn(volUuid2).when(volumeObjectToMock2).getUuid();
+    }
+
+    private static VolumeObjectTO createMockVolumeObjectTo(String uuid) {
+        VolumeObjectTO volumeObjectToMock = Mockito.mock(VolumeObjectTO.class);
+        doReturn(uuid).when(volumeObjectToMock).getUuid();
+        return volumeObjectToMock;
     }
 }
