--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNicCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricNicCmdTest.java
@@ -45,11 +45,14 @@
     @InjectMocks
     ListTungstenFabricNicCmd listTungstenFabricNicCmd;
 
+    private BaseResponse baseResponseMock;
+
     AutoCloseable closeable;
 
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        baseResponseMock = Mockito.mock(BaseResponse.class);
         listTungstenFabricNicCmd = new ListTungstenFabricNicCmd();
         listTungstenFabricNicCmd.tungstenService = tungstenService;
         ReflectionTestUtils.setField(listTungstenFabricNicCmd, "nicUuid", "test");
@@ -66,8 +69,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricNicCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         Mockito.when(tungstenService.listTungstenNic(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricNicCmd.execute();
@@ -78,8 +80,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
