--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -541,7 +541,7 @@
 
     @Test
     public void tryRestoreVMTestRestoreSucceeded() throws NoTransitionException {
-        BackupOffering offering = mock(BackupOffering.class);
+        BackupOffering offering = createMockBackupOffering();
         VolumeVO volumeVO = mock(VolumeVO.class);
         VMInstanceVO vm = mock(VMInstanceVO.class);
         BackupVO backup = mock(BackupVO.class);
@@ -569,7 +569,7 @@
 
     @Test
     public void tryRestoreVMTestRestoreFails() throws NoTransitionException {
-        BackupOffering offering = mock(BackupOffering.class);
+        BackupOffering offering = createMockBackupOffering();
         VolumeVO volumeVO = mock(VolumeVO.class);
         VMInstanceVO vm = mock(VMInstanceVO.class);
         BackupVO backup = mock(BackupVO.class);
@@ -2874,4 +2874,8 @@
         verify(backupOfferingDao).persist(any());
         verify(backupOfferingDetailsDao, never()).saveDetails(any());
     }
+
+    private static BackupOffering createMockBackupOffering() {
+        return mock(BackupOffering.class);
+    }
 }
