--- a/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategyTest.java
+++ b/engine/storage/datamotion/src/test/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategyTest.java
@@ -107,6 +107,12 @@
         return clvmPoolManager;
     }
 
+    private static DataStore createMockDataStore() {
+        DataStore srcDataStore = Mockito.mock(DataStore.class);
+        Mockito.doReturn(new HostScope(1L, 1L, 1L)).when(srcDataStore).getScope();
+        return srcDataStore;
+    }
+
     @Test
     public void testAddFullCloneFlagOnVMwareDest(){
         strategy.addFullCloneAndDiskprovisiongStrictnessFlagOnVMwareDest(dataTO);
@@ -250,8 +256,7 @@
     public void testCanBypassSecondaryStorageWithLocalAndClusterWideNFSPoolsInSameCluster() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new HostScope(1L, 1L, 1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.Filesystem).when(srcVolumeInfo).getStoragePoolType();
 
@@ -276,8 +281,7 @@
     public void testCanBypassSecondaryStorageWithLocalAndZoneWideNFSPoolsInSameZone() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         VolumeObject srcVolumeInfo = Mockito.spy(new VolumeObject());
         Mockito.doReturn(HypervisorType.KVM).when(srcVolumeInfo).getHypervisorType();
-        DataStore srcDataStore = Mockito.mock(DataStore.class);
-        Mockito.doReturn(new HostScope(1L, 1L, 1L)).when(srcDataStore).getScope();
+        DataStore srcDataStore = createMockDataStore();
         Mockito.doReturn(srcDataStore).when(srcVolumeInfo).getDataStore();
         Mockito.doReturn(Storage.StoragePoolType.Filesystem).when(srcVolumeInfo).getStoragePoolType();
 
