--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -221,6 +221,7 @@
     private DiskOfferingDao _diskOfferingDao;
     @Mock
     private DataStoreManager dataStoreMgr;
+    private DataStore dataStoreMock;
     @Mock
     private SnapshotHelper snapshotHelper;
     @Mock
@@ -300,6 +301,7 @@
 
     @Before
     public void setup() throws InterruptedException, ExecutionException {
+        dataStoreMock = Mockito.mock(DataStore.class);
         Mockito.lenient().doReturn(volumeMockId).when(volumeDataStoreVoMock).getVolumeId();
         Mockito.doReturn(volumeMockId).when(volumeVoMock).getId();
         Mockito.lenient().doReturn(accountMockId).when(accountMock).getId();
@@ -1162,9 +1164,8 @@
         VolumeInfo volumeInfoMock1 = Mockito.mock(VolumeInfo.class);
         VolumeInfo volumeInfoMock2 = Mockito.mock(VolumeInfo.class);
 
-        DataStore dataStoreMock1 = Mockito.mock(DataStore.class);
         DataStore dataStoreMock2 = Mockito.mock(DataStore.class);
-        Mockito.doReturn(dataStoreMock1).when(volumeInfoMock1).getDataStore();
+        Mockito.doReturn(dataStoreMock).when(volumeInfoMock1).getDataStore();
         Mockito.doReturn(dataStoreMock2).when(volumeInfoMock2).getDataStore();
 
         volumeInfos.add(volumeInfoMock1);
@@ -1986,8 +1987,7 @@
 
         VolumeInfo volInfo = Mockito.mock(VolumeInfo.class);
         when(volumeDataFactoryMock.getVolume(volumeMockId)).thenReturn(volInfo);
-        DataStore dataStore = Mockito.mock(DataStore.class);
-        when((volInfo.getDataStore())).thenReturn(dataStore);
+        when((volInfo.getDataStore())).thenReturn(dataStoreMock);
 
         when(volumeVoMock.getPoolId()).thenReturn(poolId);
         StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
