--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricServiceGroupCmdTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/api/command/ListTungstenFabricServiceGroupCmdTest.java
@@ -43,11 +43,14 @@
 
     ListTungstenFabricServiceGroupCmd listTungstenFabricServiceGroupCmd;
 
+    private BaseResponse baseResponseMock;
+
     AutoCloseable closeable;
 
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        baseResponseMock = Mockito.mock(BaseResponse.class);
         listTungstenFabricServiceGroupCmd = new ListTungstenFabricServiceGroupCmd();
         listTungstenFabricServiceGroupCmd.tungstenService = tungstenService;
         ReflectionTestUtils.setField(listTungstenFabricServiceGroupCmd, "serviceGroupUuid", "test");
@@ -64,8 +67,7 @@
     @Test
     public void executeTest() throws Exception {
         ReflectionTestUtils.setField(listTungstenFabricServiceGroupCmd, "zoneId", 1L);
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         Mockito.when(tungstenService.listTungstenServiceGroup(ArgumentMatchers.anyLong(),
                 ArgumentMatchers.anyString())).thenReturn(baseResponseList);
         listTungstenFabricServiceGroupCmd.execute();
@@ -77,8 +79,7 @@
 
     @Test
     public void executeAllZoneTest() throws Exception {
-        BaseResponse baseResponse = Mockito.mock(BaseResponse.class);
-        List<BaseResponse> baseResponseList = Arrays.asList(baseResponse);
+        List<BaseResponse> baseResponseList = Arrays.asList(baseResponseMock);
         TungstenProviderVO tungstenProviderVO = Mockito.mock(TungstenProviderVO.class);
         List<TungstenProviderVO> tungstenProviderVOList = Arrays.asList(tungstenProviderVO);
         Mockito.when(tungstenService.getTungstenProviders()).thenReturn(tungstenProviderVOList);
