--- a/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
+++ b/server/src/test/java/com/cloud/network/element/VpcVirtualRouterElementTest.java
@@ -74,6 +74,8 @@
     @Mock
     NetworkTopologyContext networkTopologyContext;
 
+    private final BasicNetworkTopology sharedBasicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
+
     @InjectMocks
     VpcVirtualNetworkApplianceManagerImpl _vpcRouterMgr;
 
@@ -86,10 +88,9 @@
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
         final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
-        final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
         networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
-        networkTopologyContext.setBasicNetworkTopology(basicNetworkTopology);
+        networkTopologyContext.setBasicNetworkTopology(sharedBasicNetworkTopology);
         networkTopologyContext.init();
 
         final Vpc vpc = Mockito.mock(Vpc.class);
@@ -148,10 +149,9 @@
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
         final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
-        final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
         networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
-        networkTopologyContext.setBasicNetworkTopology(basicNetworkTopology);
+        networkTopologyContext.setBasicNetworkTopology(sharedBasicNetworkTopology);
         networkTopologyContext.init();
 
         final RemoteAccessVpn remoteAccessVpn = Mockito.mock(RemoteAccessVpn.class);
@@ -174,10 +174,9 @@
         vpcVirtualRouterElement._vpcRouterMgr = _vpcRouterMgr;
 
         final AdvancedNetworkTopology advancedNetworkTopology = Mockito.mock(AdvancedNetworkTopology.class);
-        final BasicNetworkTopology basicNetworkTopology = Mockito.mock(BasicNetworkTopology.class);
 
         networkTopologyContext.setAdvancedNetworkTopology(advancedNetworkTopology);
-        networkTopologyContext.setBasicNetworkTopology(basicNetworkTopology);
+        networkTopologyContext.setBasicNetworkTopology(sharedBasicNetworkTopology);
         networkTopologyContext.init();
 
         final RemoteAccessVpn remoteAccessVpn = Mockito.mock(RemoteAccessVpn.class);
