--- a/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
+++ b/server/src/test/java/com/cloud/user/AccountManagerImplTest.java
@@ -315,10 +315,9 @@
     public void deleteUserTestIfUserIdIsEqualToCallerIdShouldThrowException() {
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
             DeleteUserCmd cmd = Mockito.mock(DeleteUserCmd.class);
-            CallContext callContextMock = Mockito.mock(CallContext.class);
+            CallContext callContextMock = createMockCallContext(userVoMock);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
 
-            Mockito.doReturn(userVoMock).when(callContextMock).getCallingUser();
             Mockito.doReturn(1L).when(cmd).getId();
             Mockito.doReturn(userVoMock).when(accountManagerImpl).getValidUserVO(Mockito.anyLong());
             Mockito.doReturn(accountVoMock).when(_accountDao).findById(Mockito.anyLong());
@@ -333,10 +332,8 @@
     public void deleteUserTestIfUserIdIsNotEqualToCallerIdShouldNotThrowException() {
         try (MockedStatic<CallContext> callContextMocked = Mockito.mockStatic(CallContext.class)) {
             DeleteUserCmd cmd = Mockito.mock(DeleteUserCmd.class);
-            CallContext callContextMock = Mockito.mock(CallContext.class);
+            CallContext callContextMock = createMockCallContext(userVoMock);
             callContextMocked.when(CallContext::current).thenReturn(callContextMock);
-
-            Mockito.doReturn(userVoMock).when(callContextMock).getCallingUser();
             Mockito.doReturn(1L).when(cmd).getId();
             Mockito.doReturn(userVoMock).when(accountManagerImpl).getValidUserVO(Mockito.anyLong());
             Mockito.doReturn(accountVoMock).when(_accountDao).findById(Mockito.anyLong());
@@ -498,6 +495,12 @@
         Mockito.when(userVoMock.getState()).thenReturn(State.ENABLED);
         Mockito.when(accountMock.getState()).thenReturn(State.ENABLED);
         updateUserPwdChange();
+    }
+
+    private static CallContext createMockCallContext(UserVO callingUser) {
+        CallContext callContextMock = Mockito.mock(CallContext.class);
+        Mockito.doReturn(callingUser).when(callContextMock).getCallingUser();
+        return callContextMock;
     }
 
     private void updateUserPwdChange() {
