--- a/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
+++ b/server/src/test/java/com/cloud/storage/VolumeApiServiceImplTest.java
@@ -179,6 +179,7 @@
     private ImageStoreVO imageStoreVO;
     @Mock
     private StoragePoolVO storagePoolVO;
+    private StoragePoolVO volumePoolMock;
     @Mock
     private VMSnapshotDao _vmSnapshotDao;
     @Mock
@@ -300,6 +301,7 @@
 
     @Before
     public void setup() throws InterruptedException, ExecutionException {
+        volumePoolMock = Mockito.mock(StoragePoolVO.class);
         Mockito.lenient().doReturn(volumeMockId).when(volumeDataStoreVoMock).getVolumeId();
         Mockito.doReturn(volumeMockId).when(volumeVoMock).getId();
         Mockito.lenient().doReturn(accountMockId).when(accountMock).getId();
@@ -1990,8 +1992,7 @@
         when((volInfo.getDataStore())).thenReturn(dataStore);
 
         when(volumeVoMock.getPoolId()).thenReturn(poolId);
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
-        when(primaryDataStoreDaoMock.findById(poolId)).thenReturn(storagePool);
+        when(primaryDataStoreDaoMock.findById(poolId)).thenReturn(volumePoolMock);
 
         Mockito.lenient().doReturn(asyncCallFutureVolumeapiResultMock).when(volumeServiceMock).resize(any(VolumeInfo.class));
         Mockito.doReturn(Mockito.mock(VolumeApiResult.class)).when(asyncCallFutureVolumeapiResultMock).get();
@@ -2024,8 +2025,7 @@
         testResizeVolumeSetup();
 
         when(storageMgr.storagePoolHasEnoughSpaceForResize(any(), nullable(Long.class), nullable(Long.class))).thenReturn(false).thenReturn(true);
-        StoragePoolVO suitableStoragePool = Mockito.mock(StoragePoolVO.class);
-        Pair<List<? extends StoragePool>, List<? extends StoragePool>> poolsPair = new Pair<>(Arrays.asList(suitableStoragePool), Arrays.asList(suitableStoragePool));
+        Pair<List<? extends StoragePool>, List<? extends StoragePool>> poolsPair = new Pair<>(Arrays.asList(volumePoolMock), Arrays.asList(volumePoolMock));
         when(managementService.listStoragePoolsForSystemMigrationOfVolume(anyLong(), anyLong(), anyLong(), anyLong(), anyLong(), anyBoolean(), anyBoolean())).thenReturn(poolsPair);
         doReturn(volumeInfoMock).when(volumeApiServiceImpl).migrateVolume(any());
         when(volumeInfoMock.getId()).thenReturn(volumeMockId);
@@ -2532,7 +2532,6 @@
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
         Long volumePoolId = 100L;
@@ -2542,7 +2541,7 @@
         Mockito.when(volumeInfo.getPoolId()).thenReturn(volumePoolId);
         Mockito.when(vmExistingVolume.getPoolId()).thenReturn(vmPoolId);
         Mockito.when(vm.getHostId()).thenReturn(vmHostId);
-        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePool);
+        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePoolMock);
         Mockito.when(primaryDataStoreDaoMock.findById(vmPoolId)).thenReturn(vmPool);
 
         Mockito.when(vmPool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
@@ -2567,7 +2566,6 @@
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
         Long volumePoolId = 100L;
@@ -2577,7 +2575,7 @@
         Mockito.when(volumeInfo.getPoolId()).thenReturn(volumePoolId);
         Mockito.when(vmExistingVolume.getPoolId()).thenReturn(vmPoolId);
         Mockito.when(vm.getHostId()).thenReturn(vmHostId);
-        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePool);
+        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePoolMock);
         Mockito.when(primaryDataStoreDaoMock.findById(vmPoolId)).thenReturn(vmPool);
 
         Mockito.when(vmPool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
@@ -2633,7 +2631,6 @@
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
         Long volumePoolId = 100L;
@@ -2643,7 +2640,7 @@
         Mockito.when(volumeInfo.getPoolId()).thenReturn(volumePoolId);
         Mockito.when(vmExistingVolume.getPoolId()).thenReturn(vmPoolId);
         Mockito.when(vm.getHostId()).thenReturn(vmHostId);
-        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePool);
+        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePoolMock);
         Mockito.when(primaryDataStoreDaoMock.findById(vmPoolId)).thenReturn(vmPool);
 
         Mockito.when(vmPool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
@@ -2673,7 +2670,6 @@
         VolumeInfo volumeInfo = Mockito.mock(VolumeInfo.class);
         VolumeVO vmExistingVolume = Mockito.mock(VolumeVO.class);
         UserVmVO vm = Mockito.mock(UserVmVO.class);
-        StoragePoolVO volumePool = Mockito.mock(StoragePoolVO.class);
         StoragePoolVO vmPool = Mockito.mock(StoragePoolVO.class);
 
         Long volumePoolId = 100L;
@@ -2684,7 +2680,7 @@
         Mockito.when(vmExistingVolume.getPoolId()).thenReturn(vmPoolId);
         Mockito.when(vm.getHostId()).thenReturn(null); // VM is stopped
         Mockito.when(vm.getLastHostId()).thenReturn(lastHostId);
-        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePool);
+        Mockito.when(primaryDataStoreDaoMock.findById(volumePoolId)).thenReturn(volumePoolMock);
         Mockito.when(primaryDataStoreDaoMock.findById(vmPoolId)).thenReturn(vmPool);
 
         Mockito.when(vmPool.getPoolType()).thenReturn(Storage.StoragePoolType.CLVM);
