--- a/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
+++ b/server/src/test/java/com/cloud/template/TemplateManagerImplTest.java
@@ -555,11 +555,8 @@
 
     @Test
     public void testLinkUserDataToTemplate() {
-        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
-        when(cmd.getTemplateId()).thenReturn(1L);
-        when(cmd.getIsoId()).thenReturn(null);
+        LinkUserDataToTemplateCmd cmd = createMockLinkUserDataToTemplateCmd(1L, null);
         when(cmd.getUserdataId()).thenReturn(2L);
-        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
         when(vmTemplateDao.findById(anyLong())).thenReturn(template);
@@ -599,11 +596,8 @@
 
     @Test(expected = InvalidParameterValueException.class)
     public void testLinkUserDataToTemplateWhenNoTemplate() {
-        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
-        when(cmd.getTemplateId()).thenReturn(1L);
-        when(cmd.getIsoId()).thenReturn(null);
+        LinkUserDataToTemplateCmd cmd = createMockLinkUserDataToTemplateCmd(1L, null);
         when(cmd.getUserdataId()).thenReturn(2L);
-        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
         when(vmTemplateDao.findById(anyLong())).thenReturn(null);
 
@@ -612,11 +606,8 @@
 
     @Test
     public void testUnLinkUserDataToTemplate() {
-        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
-        when(cmd.getTemplateId()).thenReturn(1L);
-        when(cmd.getIsoId()).thenReturn(null);
+        LinkUserDataToTemplateCmd cmd = createMockLinkUserDataToTemplateCmd(1L, null);
         when(cmd.getUserdataId()).thenReturn(null);
-        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
 
         VMTemplateVO template = Mockito.mock(VMTemplateVO.class);
         when(template.getId()).thenReturn(1L);
@@ -667,6 +658,14 @@
 
         templateManager.getImageStore(null, 1L, volumeVO);
         Mockito.verify(dataStoreManager, Mockito.times(0)).getImageStoreWithFreeCapacity(Mockito.anyLong());
+    }
+
+    private static LinkUserDataToTemplateCmd createMockLinkUserDataToTemplateCmd(Long templateId, Long isoId) {
+        LinkUserDataToTemplateCmd cmd = Mockito.mock(LinkUserDataToTemplateCmd.class);
+        when(cmd.getTemplateId()).thenReturn(templateId);
+        when(cmd.getIsoId()).thenReturn(isoId);
+        when(cmd.getUserdataPolicy()).thenReturn(UserData.UserDataOverridePolicy.ALLOWOVERRIDE);
+        return cmd;
     }
 
     private HypervisorTemplateAdapter mockHypervisorTemplateAdapter() {
