--- a/server/src/test/java/org/apache/cloudstack/schedule/vm/VMScheduleWorkerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/schedule/vm/VMScheduleWorkerTest.java
@@ -118,10 +118,10 @@
     }
 
     private void executeProcessJobWithVMStateAndActionNonSkipped(VirtualMachine.State state, VMScheduleAction action) {
-        ResourceScheduledJobVO job = Mockito.mock(ResourceScheduledJobVO.class);
-        UserVm vm = Mockito.mock(UserVm.class);
-
-        Mockito.when(job.getResourceId()).thenReturn(1L);
+        ResourceScheduledJobVO job = createMockResourceScheduledJob();
+        UserVm vm = Mockito.mock(UserVm.class);
+
+
         Mockito.when(job.getActionName()).thenReturn(action.name());
         Mockito.when(vm.getState()).thenReturn(state);
         Mockito.when(userVmManager.getUserVm(1L)).thenReturn(vm);
@@ -134,13 +134,20 @@
         Assert.assertEquals(1L, (long) jobId);
     }
 
+    private static ResourceScheduledJobVO createMockResourceScheduledJob() {
+        ResourceScheduledJobVO job = createMockResourceScheduledJob();
+
+
+        return job;
+    }
+
     @Test
     public void testProcessJobInvalidAction() {
-        ResourceScheduledJobVO job = Mockito.mock(ResourceScheduledJobVO.class);
-        UserVm vm = Mockito.mock(UserVm.class);
-
-        Mockito.when(job.getResourceId()).thenReturn(1L);
-        Mockito.when(job.getActionName()).thenReturn(VMScheduleAction.START.name());
+        ResourceScheduledJobVO job = createMockResourceScheduledJob();
+        UserVm vm = Mockito.mock(UserVm.class);
+
+
+
         Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Running);
         Mockito.when(userVmManager.getUserVm(1L)).thenReturn(vm);
 
@@ -151,11 +158,11 @@
 
     @Test
     public void testProcessJobVMInInvalidState() {
-        ResourceScheduledJobVO job = Mockito.mock(ResourceScheduledJobVO.class);
-        UserVm vm = Mockito.mock(UserVm.class);
-
-        Mockito.when(job.getResourceId()).thenReturn(1L);
-        Mockito.when(job.getActionName()).thenReturn(VMScheduleAction.START.name());
+        ResourceScheduledJobVO job = createMockResourceScheduledJob();
+        UserVm vm = Mockito.mock(UserVm.class);
+
+
+
         Mockito.when(vm.getState()).thenReturn(VirtualMachine.State.Unknown);
         Mockito.when(userVmManager.getUserVm(1L)).thenReturn(vm);
 
