--- a/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookServiceImplTest.java
+++ b/plugins/event-bus/webhook/src/test/java/org/apache/cloudstack/mom/webhook/WebhookServiceImplTest.java
@@ -492,8 +492,7 @@
         Mockito.when(result.isSuccess()).thenReturn(true);
         Mockito.when(result.getResult()).thenReturn("result");
 
-        AsyncCallbackDispatcher<WebhookServiceImpl, WebhookDeliveryThread.WebhookDeliveryResult> callback = Mockito.mock(AsyncCallbackDispatcher.class);
-        Mockito.when(callback.getResult()).thenReturn(result);
+        AsyncCallbackDispatcher<WebhookServiceImpl, WebhookDeliveryThread.WebhookDeliveryResult> callback = createMockAsyncCallback(result);
 
         webhookServiceImpl.deliveryCompleteCallback(callback, context);
 
@@ -507,8 +506,7 @@
         CompletableFuture<WebhookDeliveryThread.WebhookDeliveryResult> future = Mockito.mock(CompletableFuture.class);
 
         Mockito.when(context.getFuture()).thenReturn(future);
-        AsyncCallbackDispatcher<WebhookServiceImpl, WebhookDeliveryThread.WebhookDeliveryResult> callback = Mockito.mock(AsyncCallbackDispatcher.class);
-        Mockito.when(callback.getResult()).thenReturn(result);
+        AsyncCallbackDispatcher<WebhookServiceImpl, WebhookDeliveryThread.WebhookDeliveryResult> callback = createMockAsyncCallback(result);
 
         webhookServiceImpl.manualDeliveryCompleteCallback(callback, context);
 
@@ -670,4 +668,10 @@
 
         Mockito.verify(cache, Mockito.times(1)).invalidate(123L);
     }
+
+    private static AsyncCallbackDispatcher<WebhookServiceImpl, WebhookDeliveryThread.WebhookDeliveryResult> createMockAsyncCallback(WebhookDeliveryThread.WebhookDeliveryResult result) {
+        AsyncCallbackDispatcher<WebhookServiceImpl, WebhookDeliveryThread.WebhookDeliveryResult> callback = Mockito.mock(AsyncCallbackDispatcher.class);
+        Mockito.when(callback.getResult()).thenReturn(result);
+        return callback;
+    }
 }
