--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -2018,8 +2018,7 @@
         Long backupOfferingId = 2L;
         String providerName = "testbackupprovider";
 
-        BackupVO backup = mock(BackupVO.class);
-        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
+        BackupVO backup = createMockBackup(backupOfferingId);
 
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
@@ -2046,8 +2045,7 @@
         Long backupOfferingId = 2L;
         String providerName = "testbackupprovider";
 
-        BackupVO backup = mock(BackupVO.class);
-        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
+        BackupVO backup = createMockBackup(backupOfferingId);
 
         BackupOfferingVO offering = mock(BackupOfferingVO.class);
         when(offering.getProvider()).thenReturn(providerName);
@@ -2073,8 +2071,7 @@
         Long backupId = 1L;
         Long backupOfferingId = 2L;
 
-        BackupVO backup = mock(BackupVO.class);
-        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
+        BackupVO backup = createMockBackup(backupOfferingId);
 
         when(backupDao.findById(backupId)).thenReturn(backup);
         when(backupOfferingDao.findByIdIncludingRemoved(backupOfferingId)).thenReturn(null);
@@ -2623,6 +2620,12 @@
         when(offering.getId()).thenReturn(id);
         when(offering.getName()).thenReturn(name);
         return offering;
+    }
+
+    private BackupVO createMockBackup(Long backupOfferingId) {
+        BackupVO backup = Mockito.mock(BackupVO.class);
+        when(backup.getBackupOfferingId()).thenReturn(backupOfferingId);
+        return backup;
     }
 
     @Test
