--- a/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/MockContrailManager.java
+++ b/plugins/network-elements/juniper-contrail/src/test/java/org/apache/cloudstack/network/contrail/model/MockContrailManager.java
@@ -0,0 +1,18 @@
+package org.apache.cloudstack.network.contrail.model;
+
+import com.cloud.network.dao.NetworkVO;
+import org.apache.cloudstack.network.contrail.management.ContrailManagerImpl;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static org.mockito.Mockito.when;
+
+public class MockContrailManager {
+    public static ContrailManagerImpl createMockContrailManager(NetworkVO network) throws IOException {
+        ContrailManagerImpl contrailMgr = Mockito.mock(ContrailManagerImpl.class);
+        when(contrailMgr.getCanonicalName(network)).thenReturn("testnetwork");
+        when(contrailMgr.getProjectId(network.getDomainId(), network.getAccountId())).thenReturn("testProjectId");
+        return contrailMgr;
+    }
+}
