--- a/server/src/test/java/org/apache/cloudstack/schedule/ResourceScheduleManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/schedule/ResourceScheduleManagerImplTest.java
@@ -318,8 +318,8 @@
 
     @Test
     public void removeSchedulesForResource() {
-        ResourceScheduleVO schedule1 = Mockito.mock(ResourceScheduleVO.class);
-        ResourceScheduleVO schedule2 = Mockito.mock(ResourceScheduleVO.class);
+        ResourceScheduleVO schedule1 = createMockResourceScheduleVO(1L);
+        ResourceScheduleVO schedule2 = createMockResourceScheduleVO(2L);
         List<ResourceScheduleVO> scheduleList = new ArrayList<>();
         scheduleList.add(schedule1);
         scheduleList.add(schedule2);
@@ -328,8 +328,6 @@
         Mockito.when(resourceScheduleDao.getSearchCriteriaForResource(
                 ApiCommandResourceType.VirtualMachine, 1L)).thenReturn(sc);
         Mockito.when(resourceScheduleDao.search(sc, null)).thenReturn(scheduleList);
-        Mockito.when(schedule1.getId()).thenReturn(1L);
-        Mockito.when(schedule2.getId()).thenReturn(2L);
         Mockito.when(resourceScheduleDao.removeAllSchedulesForResource(Mockito.any(), Mockito.anyLong())).thenReturn(2L);
 
         resourceScheduleManager.removeSchedulesForResource(ApiCommandResourceType.VirtualMachine, 1L);
@@ -424,4 +422,10 @@
         Assert.assertEquals(1, configKeys.length);
         Assert.assertEquals(BaseScheduleWorker.ScheduledJobExpireInterval.key(), configKeys[0].key());
     }
+
+    private static ResourceScheduleVO createMockResourceScheduleVO(long scheduleId) {
+        ResourceScheduleVO schedule = Mockito.mock(ResourceScheduleVO.class);
+        Mockito.when(schedule.getId()).thenReturn(scheduleId);
+        return schedule;
+    }
 }
