--- a/api/src/test/java/org/apache/cloudstack/api/command/user/vm/ListVMsCmdTest.java
+++ b/api/src/test/java/org/apache/cloudstack/api/command/user/vm/ListVMsCmdTest.java
@@ -72,11 +72,9 @@
     @Test
     public void testUpdateVMResponse_withMixedIcons() {
         String vm1Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm1 = mock(UserVmResponse.class);
-        when(vm1.getId()).thenReturn(vm1Uuid);
+        UserVmResponse vm1 = createMockUserVmResponse(vm1Uuid);
         String vm2Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm2 = mock(UserVmResponse.class);
-        when(vm2.getId()).thenReturn(vm2Uuid);
+        UserVmResponse vm2 = createMockUserVmResponse(vm2Uuid);
         List<UserVmResponse> responses = Arrays.asList(vm1, vm2);
         ResourceIcon icon1 = mock(ResourceIcon.class);
         ResourceIcon icon2 = mock(ResourceIcon.class);
@@ -105,14 +103,12 @@
     public void testGetResourceIconsForUsingTemplateIso_withValidData() {
         String vm1Uuid = UUID.randomUUID().toString();
         String template1Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm1 = mock(UserVmResponse.class);
-        when(vm1.getId()).thenReturn(vm1Uuid);
+        UserVmResponse vm1 = createMockUserVmResponse(vm1Uuid);
         when(vm1.getTemplateId()).thenReturn(template1Uuid);
         when(vm1.getIsoId()).thenReturn(null);
         String vm2Uuid = UUID.randomUUID().toString();
         String iso2Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm2 = mock(UserVmResponse.class);
-        when(vm2.getId()).thenReturn(vm2Uuid);
+        UserVmResponse vm2 = createMockUserVmResponse(vm2Uuid);
         when(vm2.getTemplateId()).thenReturn(null);
         when(vm2.getIsoId()).thenReturn(iso2Uuid);
         List<UserVmResponse> responses = Arrays.asList(vm1, vm2);
@@ -137,8 +133,7 @@
     public void testGetResourceIconsForUsingTemplateIso_withMissingIcons() {
         String vm1Uuid = UUID.randomUUID().toString();
         String template1Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm1 = mock(UserVmResponse.class);
-        when(vm1.getId()).thenReturn(vm1Uuid);
+        UserVmResponse vm1 = createMockUserVmResponse(vm1Uuid);
         when(vm1.getTemplateId()).thenReturn(template1Uuid);
         when(vm1.getIsoId()).thenReturn(null);
         List<UserVmResponse> responses = List.of(vm1);
@@ -157,14 +152,12 @@
     public void testGetResourceIconsUsingOsCategory_withValidData() {
         String vm1Uuid = UUID.randomUUID().toString();
         String os1Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm1 = mock(UserVmResponse.class);
+        UserVmResponse vm1 = createMockUserVmResponse(vm1Uuid);
         when(vm1.getGuestOsId()).thenReturn(os1Uuid);
-        when(vm1.getId()).thenReturn(vm1Uuid);
         String vm2Uuid = UUID.randomUUID().toString();
         String os2Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm2 = mock(UserVmResponse.class);
+        UserVmResponse vm2 = createMockUserVmResponse(vm2Uuid);
         when(vm2.getGuestOsId()).thenReturn(os2Uuid);
-        when(vm2.getId()).thenReturn(vm2Uuid);
         List<UserVmResponse> responses = Arrays.asList(vm1, vm2);
         GuestOS guestOS1 = mock(GuestOS.class);
         when(guestOS1.getUuid()).thenReturn(os1Uuid);
@@ -191,9 +184,8 @@
     public void testGetResourceIconsUsingOsCategory_missingGuestOS() {
         String vm1Uuid = UUID.randomUUID().toString();
         String os1Uuid = UUID.randomUUID().toString();
-        UserVmResponse vm1 = mock(UserVmResponse.class);
+        UserVmResponse vm1 = createMockUserVmResponse(os1Uuid);
         when(vm1.getGuestOsId()).thenReturn(vm1Uuid);
-        when(vm1.getId()).thenReturn(os1Uuid);
         List<UserVmResponse> responses = Collections.singletonList(vm1);
         when(_entityMgr.listByUuids(eq(GuestOS.class), anySet()))
                 .thenReturn(Collections.emptyList());
@@ -203,11 +195,10 @@
 
     @Test
     public void testGetResourceIconsUsingOsCategory_missingIcon() {
-        UserVmResponse vm1 = mock(UserVmResponse.class);
         String vmUuid = UUID.randomUUID().toString();
+        UserVmResponse vm1 = createMockUserVmResponse(vmUuid);
         String osUuid = UUID.randomUUID().toString();
         when(vm1.getGuestOsId()).thenReturn(osUuid);
-        when(vm1.getId()).thenReturn(vmUuid);
         List<UserVmResponse> responses = Collections.singletonList(vm1);
         GuestOS guestOS1 = mock(GuestOS.class);
         when(guestOS1.getCategoryId()).thenReturn(10L);
@@ -220,4 +211,10 @@
         assertTrue(result.containsKey(vmUuid));
         assertNull(result.get(vmUuid));
     }
+
+    private static UserVmResponse createMockUserVmResponse(String id) {
+        UserVmResponse vm = mock(UserVmResponse.class);
+        when(vm.getId()).thenReturn(id);
+        return vm;
+    }
 }
