--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -1779,9 +1779,8 @@
         when(actionVO.getAllowedRoleTypes()).thenReturn(
                 RoleType.toCombinedMask(List.of(RoleType.Admin, RoleType.DomainAdmin, RoleType.User)));
 
-        ExtensionVO extensionVO = mock(ExtensionVO.class);
+        ExtensionVO extensionVO = createMockExtensionVO();
         when(extensionDao.findById(anyLong())).thenReturn(extensionVO);
-        when(extensionVO.getState()).thenReturn(Extension.State.Enabled);
 
         RunCustomActionAnswer answer = mock(RunCustomActionAnswer.class);
         when(answer.getResult()).thenReturn(true);
@@ -1858,8 +1857,7 @@
         when(actionVO.isEnabled()).thenReturn(true);
         when(actionVO.getResourceType()).thenReturn(null);
         when(actionVO.getExtensionId()).thenReturn(1L);
-        ExtensionVO extensionVO = mock(ExtensionVO.class);
-        when(extensionVO.getState()).thenReturn(Extension.State.Enabled);
+        ExtensionVO extensionVO = createMockExtensionVO();
         when(extensionDao.findById(1L)).thenReturn(extensionVO);
 
 
@@ -1881,9 +1879,8 @@
         when(actionVO.isEnabled()).thenReturn(true);
         when(actionVO.getResourceType()).thenReturn(ExtensionCustomAction.ResourceType.VirtualMachine);
 
-        ExtensionVO extensionVO = mock(ExtensionVO.class);
+        ExtensionVO extensionVO = createMockExtensionVO();
         when(extensionDao.findById(anyLong())).thenReturn(extensionVO);
-        when(extensionVO.getState()).thenReturn(Extension.State.Enabled);
 
         VirtualMachine vm = mock(VirtualMachine.class);
         when(vm.getHypervisorType()).thenReturn(Hypervisor.HypervisorType.External);
@@ -1916,9 +1913,8 @@
         when(actionVO.getResourceType()).thenReturn(ExtensionCustomAction.ResourceType.VirtualMachine);
         when(actionVO.getAllowedRoleTypes()).thenReturn(RoleType.toCombinedMask(List.of(RoleType.Admin, RoleType.DomainAdmin, RoleType.User)));
 
-        ExtensionVO extensionVO = mock(ExtensionVO.class);
+        ExtensionVO extensionVO = createMockExtensionVO();
         when(extensionDao.findById(anyLong())).thenReturn(extensionVO);
-        when(extensionVO.getState()).thenReturn(Extension.State.Enabled);
 
         VirtualMachine vm = mock(VirtualMachine.class);
         when(entityManager.findByUuid(eq(VirtualMachine.class), anyString())).thenReturn(vm);
@@ -3062,4 +3058,10 @@
         assertTrue(resp.getResult().get(ApiConstants.DETAILS).contains("Action failed"));
     }
 
+    private static ExtensionVO createMockExtensionVO() {
+        ExtensionVO extensionVO = mock(ExtensionVO.class);
+        when(extensionVO.getState()).thenReturn(Extension.State.Enabled);
+        return extensionVO;
+    }
+
 }
