--- a/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/MockInternalBackupStoragePoolDao.java
+++ b/plugins/backup/kboss/src/test/java/org/apache/cloudstack/backup/MockInternalBackupStoragePoolDao.java
@@ -0,0 +1,19 @@
+package org.apache.cloudstack.backup;
+
+import org.apache.cloudstack.backup.dao.InternalBackupStoragePoolDao;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.doReturn;
+
+public final class MockInternalBackupStoragePoolDao {
+    private MockInternalBackupStoragePoolDao() {
+    }
+
+    public static InternalBackupStoragePoolDao createMockInternalBackupStoragePoolDao(
+            InternalBackupStoragePoolVO storagePoolVo, long volumeId) {
+        InternalBackupStoragePoolDao internalBackupStoragePoolDaoMock =
+                Mockito.mock(InternalBackupStoragePoolDao.class);
+        doReturn(storagePoolVo).when(internalBackupStoragePoolDaoMock).findOneByVolumeId(volumeId);
+        return internalBackupStoragePoolDaoMock;
+    }
+}
