--- a/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
+++ b/server/src/test/java/com/cloud/resource/ResourceManagerImplTest.java
@@ -182,6 +182,14 @@
     private MockedConstruction<GetVncPortCommand> getVncPortCommandMockedConstruction;
     private AutoCloseable closeable;
 
+    private static StoragePoolVO createMockStoragePool(ScopeType scope) {
+        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        Mockito.when(storagePool.getId()).thenReturn(1L);
+        Mockito.when(storagePool.getScope()).thenReturn(scope);
+        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
+        return storagePool;
+    }
+
     private void overrideDefaultConfigValue(final ConfigKey configKey, final String name, final Object o) throws IllegalAccessException, NoSuchFieldException {
         Field f = ConfigKey.class.getDeclaredField(name);
         f.setAccessible(true);
@@ -940,12 +948,10 @@
         Long poolId = 1L;
         List<String> storageAccessGroups = Arrays.asList("sag1", "sag2");
 
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePool = createMockStoragePool(ScopeType.CLUSTER);
         Mockito.when(storagePool.getId()).thenReturn(poolId);
-        Mockito.when(storagePool.getScope()).thenReturn(ScopeType.CLUSTER);
         Mockito.when(storagePool.getClusterId()).thenReturn(1L);
         Mockito.when(storagePool.getPodId()).thenReturn(1L);
-        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
 
         Mockito.when(storagePoolDao.findById(poolId)).thenReturn(storagePool);
 
@@ -983,10 +989,8 @@
         Long poolId = 1L;
         List<String> storageAccessGroups = new ArrayList<>();
 
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePool = createMockStoragePool(ScopeType.ZONE);
         Mockito.when(storagePool.getId()).thenReturn(poolId);
-        Mockito.when(storagePool.getScope()).thenReturn(ScopeType.ZONE);
-        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
 
         Mockito.when(storagePoolDao.findById(poolId)).thenReturn(storagePool);
 
@@ -1020,12 +1024,10 @@
         Long poolId = 1L;
         List<String> storageAccessGroups = Arrays.asList("sag1", "sag2");
 
-        StoragePoolVO storagePool = Mockito.mock(StoragePoolVO.class);
+        StoragePoolVO storagePool = createMockStoragePool(ScopeType.CLUSTER);
         Mockito.when(storagePool.getId()).thenReturn(poolId);
-        Mockito.when(storagePool.getScope()).thenReturn(ScopeType.CLUSTER);
         Mockito.when(storagePool.getClusterId()).thenReturn(1L);
         Mockito.when(storagePool.getPodId()).thenReturn(1L);
-        Mockito.when(storagePool.getDataCenterId()).thenReturn(1L);
 
         Mockito.when(storagePoolDao.findById(poolId)).thenReturn(storagePool);
 
