--- a/server/src/test/java/org/apache/cloudstack/user/UserPasswordResetManagerImplTest.java
+++ b/server/src/test/java/org/apache/cloudstack/user/UserPasswordResetManagerImplTest.java
@@ -89,7 +89,7 @@
 
         Mockito.doNothing().when(passwordReset).resetPassword(userAccount, "new_password");
 
-        UserDetailVO resetTokenDetail = Mockito.mock(UserDetailVO.class);
+        UserDetailVO resetTokenDetail = createMockUserDetailVO();
         UserDetailVO resetTokenExpiryDate = Mockito.mock(UserDetailVO.class);
         Mockito.when(userDetailsDao.findDetail(1L, PasswordResetToken)).thenReturn(resetTokenDetail);
         Mockito.when(userDetailsDao.findDetail(1L, PasswordResetTokenExpiryDate)).thenReturn(resetTokenExpiryDate);
@@ -174,4 +174,8 @@
         Mockito.verify(userDetailsDao).removeDetail(userId, UserDetailVO.PasswordChangeRequired);
         Mockito.verify(userDao).persist(userVO);
     }
+
+    private static UserDetailVO createMockUserDetailVO() {
+        return Mockito.mock(UserDetailVO.class);
+    }
 }
