--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -341,8 +341,7 @@
 
     @Test
     public void getActionMessageReturnsCustomSuccessMessage() {
-        ExtensionCustomAction action = mock(ExtensionCustomAction.class);
-        when(action.getName()).thenReturn("actionName");
+        ExtensionCustomAction action = createMockExtensionCustomAction();
         Extension extension = mock(Extension.class);
         when(extension.getName()).thenReturn("extension");
         when(action.getSuccessMessage()).thenReturn("Custom success message");
@@ -363,8 +362,7 @@
 
     @Test
     public void getActionMessageReturnsCustomFailureMessage() {
-        ExtensionCustomAction action = mock(ExtensionCustomAction.class);
-        when(action.getName()).thenReturn("actionName");
+        ExtensionCustomAction action = createMockExtensionCustomAction();
         Extension extension = mock(Extension.class);
         when(extension.getName()).thenReturn("extension");
         when(action.getErrorMessage()).thenReturn("Custom failure message");
@@ -1549,6 +1547,12 @@
         verify(extensionCustomActionDao).remove(actionId);
     }
 
+    private static ExtensionCustomAction createMockExtensionCustomAction() {
+        ExtensionCustomAction action = mock(ExtensionCustomAction.class);
+        when(action.getName()).thenReturn("actionName");
+        return action;
+    }
+
     private void mockCallerRole(RoleType roleType) {
         CallContext callContextMock = mock(CallContext.class);
         when(CallContext.current()).thenReturn(callContextMock);
