--- a/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedNASStrategyTest.java
+++ b/plugins/storage/volume/ontap/src/test/java/org/apache/cloudstack/storage/service/UnifiedNASStrategyTest.java
@@ -127,7 +127,11 @@
         strategy = new TestableUnifiedNASStrategy(ontapStorage, nasFeignClient, volumeFeignClient, jobFeignClient, aggregateFeignClient, svmFeignClient, networkFeignClient, sanFeignClient);
         injectField("volumeDao", volumeDao);
         injectField("epSelector", epSelector);
-        injectField("storagePoolDetailsDao", storagePoolDetailsDao);
+        try {
+            injectField("storagePoolDetailsDao", storagePoolDetailsDao);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
     }
 
     private void injectField(String fieldName, Object mockedField) throws Exception {
