--- a/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
+++ b/server/src/test/java/com/cloud/api/query/QueryManagerImplTest.java
@@ -782,6 +782,12 @@
         return sc;
     }
 
+    private static Extension createMockExtension() {
+        Extension extension = mock(Extension.class);
+        when(extension.getUuid()).thenReturn("b");
+        return extension;
+    }
+
     @Test
     public void testSearchForAccounts() {
         ListAccountsCmd cmd = mock(ListAccountsCmd.class);
@@ -841,10 +847,9 @@
         when(host2.getClusterInternalId()).thenReturn(2L);
         when(host2.getHypervisor()).thenReturn(Hypervisor.HypervisorType.External.name());
 
-        Extension ext1 = mock(Extension.class);
+        Extension ext1 = createMockExtension();
         when(ext1.getUuid()).thenReturn("a");
-        Extension ext2 = mock(Extension.class);
-        when(ext2.getUuid()).thenReturn("b");
+        Extension ext2 = createMockExtension();
 
         when(extensionHelper.getExtensionForCluster(1L)).thenReturn(ext1);
         when(extensionHelper.getExtensionForCluster(2L)).thenReturn(ext2);
