--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateVolumeCommandWrapperTest.java
@@ -30,6 +30,7 @@
 import org.apache.cloudstack.storage.to.PrimaryDataStoreTO;
 import org.apache.cloudstack.storage.to.VolumeObjectTO;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.libvirt.Connect;
@@ -67,6 +68,13 @@
 
     @Mock
     LibvirtUtilitiesHelper libvirtUtilitiesHelper;
+
+    private PrimaryDataStoreTO primaryDataStore;
+
+    @Before
+    public void setUp() {
+        primaryDataStore = Mockito.mock(PrimaryDataStoreTO.class);
+    }
 
     private String domxml = "<domain type='kvm' id='1'>\n" +
             "  <name>i-2-27-VM</name>\n" +
@@ -287,8 +295,7 @@
 
         KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         Mockito.doReturn(storagePoolMgr).when(libvirtComputingResource).getStoragePoolMgr();
-        PrimaryDataStoreTO spool = Mockito.mock(PrimaryDataStoreTO.class);
-        Mockito.doReturn(spool).when(destVolumeObjectTO).getDataStore();
+        Mockito.doReturn(primaryDataStore).when(destVolumeObjectTO).getDataStore();
         KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
         Mockito.doReturn(pool).when(storagePoolMgr).getStoragePool(Mockito.any(), Mockito.any());
         Mockito.doReturn(true).when(pool).connectPhysicalDisk(Mockito.any(), Mockito.any());
@@ -334,8 +341,7 @@
 
         KVMStoragePoolManager storagePoolMgr = Mockito.mock(KVMStoragePoolManager.class);
         Mockito.doReturn(storagePoolMgr).when(libvirtComputingResource).getStoragePoolMgr();
-        PrimaryDataStoreTO spool = Mockito.mock(PrimaryDataStoreTO.class);
-        Mockito.doReturn(spool).when(destVolumeObjectTO).getDataStore();
+        Mockito.doReturn(primaryDataStore).when(destVolumeObjectTO).getDataStore();
         KVMStoragePool pool = Mockito.mock(KVMStoragePool.class);
         Mockito.doReturn(pool).when(storagePoolMgr).getStoragePool(Mockito.any(), Mockito.any());
         Mockito.doReturn(true).when(pool).connectPhysicalDisk(Mockito.any(), Mockito.any());
