--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAffinityProcessorTest.java
@@ -79,7 +79,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -88,7 +87,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        _affinityGroupVMMapDao = MockAffinityGroupVMMapDao.createMockAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -109,7 +108,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         plan.adjustHostPriority(host2Id, DeploymentPlan.HostPriorityAdjustment.DEFAULT);
@@ -120,7 +118,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id, vm3Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        _affinityGroupVMMapDao = MockAffinityGroupVMMapDao.createMockAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -146,7 +144,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -155,7 +152,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        _affinityGroupVMMapDao = MockAffinityGroupVMMapDao.createMockAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         when(vm2.getHostId()).thenReturn(null);
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/NonStrictHostAntiAffinityProcessorTest.java
@@ -78,7 +78,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -87,7 +86,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        _affinityGroupVMMapDao = MockAffinityGroupVMMapDao.createMockAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -108,7 +107,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         plan.adjustHostPriority(host2Id, DeploymentPlan.HostPriorityAdjustment.DEFAULT);
@@ -119,7 +117,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id, vm3Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        _affinityGroupVMMapDao = MockAffinityGroupVMMapDao.createMockAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = new VMInstanceVO();
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         vm2.setHostId(host2Id);
@@ -145,7 +143,6 @@
 
         List<AffinityGroupVMMapVO> vmGroupMappings = new ArrayList<>();
         vmGroupMappings.add(new AffinityGroupVMMapVO(affinityGroupId, vmId));
-        when(_affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
 
         DataCenterDeployment plan = new DataCenterDeployment(zoneId);
         ExcludeList avoid = new ExcludeList();
@@ -154,7 +151,7 @@
         when(affinityGroupDao.findById(affinityGroupId)).thenReturn(affinityGroupVO);
         when(affinityGroupVO.getId()).thenReturn(affinityGroupId);
         List<Long> groupVMIds = new ArrayList<>(Arrays.asList(vmId, vm2Id));
-        when(_affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        _affinityGroupVMMapDao = MockAffinityGroupVMMapDao.createMockAffinityGroupVMMapDao(vmId, vmGroupMappings, affinityGroupId, groupVMIds);
         VMInstanceVO vm2 = Mockito.mock(VMInstanceVO.class);
         when(vmInstanceDao.findById(vm2Id)).thenReturn(vm2);
         when(vm2.getHostId()).thenReturn(null);
--- a/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/MockAffinityGroupVMMapDao.java
+++ b/plugins/affinity-group-processors/non-strict-host-affinity/src/test/java/org/apache/cloudstack/affinity/MockAffinityGroupVMMapDao.java
@@ -0,0 +1,20 @@
+package org.apache.cloudstack.affinity;
+
+import java.util.List;
+
+import org.apache.cloudstack.affinity.dao.AffinityGroupVMMapDao;
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.nullable;
+import static org.mockito.Mockito.when;
+
+public class MockAffinityGroupVMMapDao {
+    public static AffinityGroupVMMapDao createMockAffinityGroupVMMapDao(long vmId,
+            List<AffinityGroupVMMapVO> vmGroupMappings, long affinityGroupId, List<Long> groupVMIds) {
+        AffinityGroupVMMapDao affinityGroupVMMapDao = Mockito.mock(AffinityGroupVMMapDao.class);
+        when(affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
+        when(affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        return affinityGroupVMMapDao;
+    }
+}
--- a/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/MockAffinityGroupVMMapDao.java
+++ b/plugins/affinity-group-processors/non-strict-host-anti-affinity/src/test/java/org/apache/cloudstack/affinity/MockAffinityGroupVMMapDao.java
@@ -0,0 +1,20 @@
+package org.apache.cloudstack.affinity;
+
+import java.util.List;
+
+import org.apache.cloudstack.affinity.dao.AffinityGroupVMMapDao;
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.nullable;
+import static org.mockito.Mockito.when;
+
+public class MockAffinityGroupVMMapDao {
+    public static AffinityGroupVMMapDao createMockAffinityGroupVMMapDao(long vmId,
+            List<AffinityGroupVMMapVO> vmGroupMappings, long affinityGroupId, List<Long> groupVMIds) {
+        AffinityGroupVMMapDao affinityGroupVMMapDao = Mockito.mock(AffinityGroupVMMapDao.class);
+        when(affinityGroupVMMapDao.findByVmIdType(eq(vmId), nullable(String.class))).thenReturn(vmGroupMappings);
+        when(affinityGroupVMMapDao.listVmIdsByAffinityGroup(affinityGroupId)).thenReturn(groupVMIds);
+        return affinityGroupVMMapDao;
+    }
+}
