--- a/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
+++ b/server/src/test/java/org/apache/cloudstack/backup/BackupManagerTest.java
@@ -2625,6 +2625,13 @@
         return offering;
     }
 
+    private static CreateBackupOfferingCmd createMockCreateBackupOfferingCmd() {
+        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
+        doReturn(1L).when(cmd).getZoneId();
+        doReturn("name").when(cmd).getName();
+        return cmd;
+    }
+
     @Test
     public void testCloneBackupOfferingUsesProvidedDomainIds() {
         Long sourceOfferingId = 1L;
@@ -2738,9 +2745,7 @@
 
         doNothing().when(backupManager).validateBackupForZone(zoneId);
         doReturn(backupOfferingVOMock).when(backupOfferingDao).findByName(offeringName, zoneId);
-        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
-        doReturn(zoneId).when(cmd).getZoneId();
-        doReturn(offeringName).when(cmd).getName();
+        CreateBackupOfferingCmd cmd = createMockCreateBackupOfferingCmd();
 
         backupManager.createBackupOffering(cmd);
 
@@ -2755,9 +2760,7 @@
 
         doNothing().when(backupManager).validateBackupForZone(zoneId);
         doReturn(null).when(backupOfferingDao).findByName(offeringName, zoneId);
-        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
-        doReturn(zoneId).when(cmd).getZoneId();
-        doReturn(offeringName).when(cmd).getName();
+        CreateBackupOfferingCmd cmd = createMockCreateBackupOfferingCmd();
         doReturn(List.of(domainId)).when(cmd).getDomainIds();
         doReturn(null).when(domainDao).findById(domainId);
 
@@ -2773,9 +2776,7 @@
 
         doNothing().when(backupManager).validateBackupForZone(zoneId);
         doReturn(null).when(backupOfferingDao).findByName(offeringName, zoneId);
-        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
-        doReturn(zoneId).when(cmd).getZoneId();
-        doReturn(offeringName).when(cmd).getName();
+        CreateBackupOfferingCmd cmd = createMockCreateBackupOfferingCmd();
         doReturn(backupProvider).when(backupManager).getBackupProvider(zoneId);
         doReturn("dummy").when(backupProvider).getName();
 
@@ -2791,9 +2792,7 @@
 
         doNothing().when(backupManager).validateBackupForZone(zoneId);
         doReturn(null).when(backupOfferingDao).findByName(offeringName, zoneId);
-        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
-        doReturn(zoneId).when(cmd).getZoneId();
-        doReturn(offeringName).when(cmd).getName();
+        CreateBackupOfferingCmd cmd = createMockCreateBackupOfferingCmd();
         doReturn(backupProvider).when(backupManager).getBackupProvider(zoneId);
         doReturn("kboss").when(backupProvider).getName();
         doReturn(false).when(backupProvider).isValidProviderOffering(zoneId, null);
@@ -2811,7 +2810,7 @@
 
         doNothing().when(backupManager).validateBackupForZone(zoneId);
         doReturn(null).when(backupOfferingDao).findByName(offeringName, zoneId);
-        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
+        CreateBackupOfferingCmd cmd = createMockCreateBackupOfferingCmd();
         doReturn(zoneId).when(cmd).getZoneId();
         doReturn(List.of(domainId)).when(cmd).getDomainIds();
         doReturn(domainVOMock).when(domainDao).findById(domainId);
@@ -2853,7 +2852,7 @@
 
         doNothing().when(backupManager).validateBackupForZone(zoneId);
         doReturn(null).when(backupOfferingDao).findByName(offeringName, zoneId);
-        CreateBackupOfferingCmd cmd = Mockito.mock(CreateBackupOfferingCmd.class);
+        CreateBackupOfferingCmd cmd = createMockCreateBackupOfferingCmd();
         doReturn(zoneId).when(cmd).getZoneId();
         doReturn(offeringName).when(cmd).getName();
         doReturn(backupProvider).when(backupManager).getBackupProvider(zoneId);
