--- a/plugins/backup/nas/src/test/java/org/apache/cloudstack/backup/NASBackupProviderTest.java
+++ b/plugins/backup/nas/src/test/java/org/apache/cloudstack/backup/NASBackupProviderTest.java
@@ -470,8 +470,8 @@
         Mockito.when(smpVol.getPoolId()).thenReturn(3L);
         Mockito.when(volumeDao.findByInstance(vmId)).thenReturn(List.of(nfsVol, hostVol, smpVol));
 
-        StoragePoolVO nfs = mock(StoragePoolVO.class);
-        Mockito.when(nfs.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
+        StoragePoolVO nfs = createMockStoragePool();
+
         StoragePoolVO host = mock(StoragePoolVO.class);
         Mockito.when(host.getScope()).thenReturn(ScopeType.HOST);
         StoragePoolVO smp = mock(StoragePoolVO.class);
@@ -499,8 +499,8 @@
         Mockito.when(rbdVol.getPoolId()).thenReturn(9L);
         Mockito.when(volumeDao.findByInstance(vmId)).thenReturn(List.of(nfsVol, rbdVol));
 
-        StoragePoolVO nfs = mock(StoragePoolVO.class);
-        Mockito.when(nfs.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
+        StoragePoolVO nfs = createMockStoragePool();
+
         StoragePoolVO rbd = mock(StoragePoolVO.class);
         Mockito.when(rbd.getPoolType()).thenReturn(Storage.StoragePoolType.RBD);
         Mockito.when(storagePoolDao.findById(1L)).thenReturn(nfs);
@@ -1034,4 +1034,10 @@
         Mockito.verify(backupDao, Mockito.never()).remove(51L);
         Mockito.verify(backupDao).remove(50L);
     }
+
+    private static StoragePoolVO createMockStoragePool() {
+        StoragePoolVO storagePool = mock(StoragePoolVO.class);
+        Mockito.when(storagePool.getPoolType()).thenReturn(Storage.StoragePoolType.NetworkFilesystem);
+        return storagePool;
+    }
 }
