--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -892,7 +892,7 @@
         TungstenAnswer addTungstenPolicyRuleAnswer = MockTungstenAnswerFactory.get(true);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
-        PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
+        PolicyRuleType policyRuleType = createMockPolicyRuleType();
         ActionListType actionListType = mock(ActionListType.class);
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
@@ -999,7 +999,7 @@
         TungstenAnswer listTungstenPolicyRuleAnswer = MockTungstenAnswerFactory.get(true);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
-        PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
+        PolicyRuleType policyRuleType = createMockPolicyRuleType();
         ActionListType actionListType = mock(ActionListType.class);
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
@@ -1028,7 +1028,7 @@
         TungstenAnswer listTungstenPolicyRuleAnswer = MockTungstenAnswerFactory.get(true);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         PolicyEntriesType policyEntriesType = mock(PolicyEntriesType.class);
-        PolicyRuleType policyRuleType = mock(PolicyRuleType.class);
+        PolicyRuleType policyRuleType = createMockPolicyRuleType();
         ActionListType actionListType = mock(ActionListType.class);
         AddressType addressType = mock(AddressType.class);
         SubnetType subnetType = mock(SubnetType.class);
@@ -1668,4 +1668,8 @@
 
         assertTrue(tungstenService.deleteLogicalRouter(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
+
+    private static PolicyRuleType createMockPolicyRuleType() {
+        return mock(PolicyRuleType.class);
+    }
 }
