--- a/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
+++ b/framework/extensions/src/test/java/org/apache/cloudstack/framework/extensions/manager/ExtensionsManagerImplTest.java
@@ -214,6 +214,8 @@
     @Mock
     private NetworkModel networkModel;
 
+    private PrepareExtensionPathCommand prepareExtensionPathCommand;
+
     @Mock
     private PhysicalNetworkServiceProviderDao physicalNetworkServiceProviderDao;
 
@@ -226,6 +228,7 @@
     @Before
     public void setUp() {
         MockitoAnnotations.openMocks(this);
+        prepareExtensionPathCommand = mock(PrepareExtensionPathCommand.class);
     }
 
     @Test
@@ -254,7 +257,7 @@
     @Test
     public void getResultFromAnswersStringReturnsSuccess() {
         Extension ext = mock(Extension.class);
-        Answer[] answers = new Answer[]{new Answer(mock(PrepareExtensionPathCommand.class), true, "ok")};
+        Answer[] answers = new Answer[]{new Answer(prepareExtensionPathCommand, true, "ok")};
         String json = GsonHelper.getGson().toJson(answers);
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Pair<Boolean, String> result = extensionsManager.getResultFromAnswersString(json, ext, msHost, "op");
@@ -265,7 +268,7 @@
     @Test
     public void getResultFromAnswersStringReturnsFailure() {
         Extension ext = mock(Extension.class);
-        Answer[] answers = new Answer[]{new Answer(mock(PrepareExtensionPathCommand.class), false, "fail")};
+        Answer[] answers = new Answer[]{new Answer(prepareExtensionPathCommand, false, "fail")};
         String json = GsonHelper.getGson().toJson(answers);
         ManagementServerHostVO msHost = mock(ManagementServerHostVO.class);
         Pair<Boolean, String> result = extensionsManager.getResultFromAnswersString(json, ext, msHost, "op");
