--- a/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
+++ b/plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessorTest.java
@@ -112,6 +112,7 @@
     @Mock
     Logger loggerMock;
 
+    private Rados radosMockShared;
 
     private static final String directDownloadTemporaryPath = "/var/lib/libvirt/images/dd";
     private static final long templateSize = 80000L;
@@ -126,6 +127,7 @@
     @Before
     public void setUp() throws ConfigurationException {
         closeable = MockitoAnnotations.openMocks(this);
+        radosMockShared = Mockito.mock(Rados.class);
         storageProcessor = new KVMStorageProcessor(storagePoolManager, resource);
     }
 
@@ -540,10 +542,9 @@
      */
     @Test
     public void takeRbdVolumeSnapshotOfStoppedVmTestCreatesSnapshotExactlyOnce() throws Exception {
-        Rados radosMock = Mockito.mock(Rados.class);
         IoCTX ioCtxMock = Mockito.mock(IoCTX.class);
         RbdImage rbdImageMock = Mockito.mock(RbdImage.class);
-        KVMPhysicalDisk diskMock = prepareRbdSnapshotMocks(radosMock, ioCtxMock);
+        KVMPhysicalDisk diskMock = prepareRbdSnapshotMocks(radosMockShared, ioCtxMock);
 
         try (MockedConstruction<Rbd> rbd = Mockito.mockConstruction(Rbd.class, ((mock, context) ->
                 Mockito.doReturn(rbdImageMock).when(mock).open(RBD_IMAGE_NAME)))) {
@@ -553,7 +554,7 @@
             Assert.assertEquals(Long.valueOf(SNAPSHOT_SIZE), result);
             Mockito.verify(rbdImageMock, Mockito.times(1)).snapCreate(SNAPSHOT_NAME);
             Mockito.verify(rbd.constructed().get(0)).close(rbdImageMock);
-            Mockito.verify(radosMock).ioCtxDestroy(ioCtxMock);
+            Mockito.verify(radosMockShared).ioCtxDestroy(ioCtxMock);
         }
     }
 
@@ -563,10 +564,9 @@
      */
     @Test
     public void takeRbdVolumeSnapshotOfStoppedVmTestReleasesHandlesWhenSnapshotFails() throws Exception {
-        Rados radosMock = Mockito.mock(Rados.class);
         IoCTX ioCtxMock = Mockito.mock(IoCTX.class);
         RbdImage rbdImageMock = Mockito.mock(RbdImage.class);
-        KVMPhysicalDisk diskMock = prepareRbdSnapshotMocks(radosMock, ioCtxMock);
+        KVMPhysicalDisk diskMock = prepareRbdSnapshotMocks(radosMockShared, ioCtxMock);
         Mockito.doThrow(new RbdException("Failed to create snapshot")).when(rbdImageMock).snapCreate(SNAPSHOT_NAME);
 
         try (MockedConstruction<Rbd> rbd = Mockito.mockConstruction(Rbd.class, ((mock, context) ->
@@ -576,7 +576,7 @@
 
             Assert.assertNull(result);
             Mockito.verify(rbd.constructed().get(0)).close(rbdImageMock);
-            Mockito.verify(radosMock).ioCtxDestroy(ioCtxMock);
+            Mockito.verify(radosMockShared).ioCtxDestroy(ioCtxMock);
         }
     }
 
