--- a/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
+++ b/plugins/hypervisors/vmware/src/test/java/com/cloud/hypervisor/guru/VMwareGuruTest.java
@@ -98,6 +98,8 @@
     @InjectMocks
     private VMwareGuru vMwareGuru = new VMwareGuru();
 
+    private ServiceContent serviceContentMock;
+
     @Mock
     PrimaryDataStoreDao _storagePoolDao;
 
@@ -136,6 +138,7 @@
     @Before
     public void testSetUp() throws Exception {
         closeable = MockitoAnnotations.openMocks(this);
+        serviceContentMock = Mockito.mock(ServiceContent.class);
     }
 
     @After
@@ -230,18 +233,17 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
-        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
-
-        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
-            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
+
+        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
+            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -267,7 +269,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -285,7 +286,7 @@
 
         try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -311,7 +312,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -332,7 +332,7 @@
 
         try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -358,7 +358,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -384,7 +383,7 @@
 
         try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -412,7 +411,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -444,7 +442,7 @@
 
         try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -474,18 +472,17 @@
 
         DataStoreTO dataStore = Mockito.mock(DataStoreTO.class);
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
-        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
-
-        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
-            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
+
+        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
+            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -511,7 +508,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -526,7 +522,7 @@
 
         try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -554,18 +550,17 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
-        VimPortType vimPort = Mockito.mock(VimPortType.class);
-        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
-        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
-        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
-        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
-        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
-        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
-
-        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
-            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+        VimPortType vimPort = Mockito.mock(VimPortType.class);
+        VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
+        VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
+        Mockito.doReturn(vimClient).when(vmwareContext).getVimClient();
+        Mockito.doReturn(mor).when(vmwareContext).getRootFolder();
+        Mockito.doReturn(mor).when(vimClient).getDecendentMoRef(any(ManagedObjectReference.class), anyString(), anyString());
+        DatacenterMO dataCenterMO = spy(new DatacenterMO(vmwareContext, datacenterName));
+
+        try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
+            Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
@@ -592,7 +587,6 @@
         params.put(VmDetailConstants.VMWARE_VCENTER_PASSWORD, "password");
 
         ManagedObjectReference mor = Mockito.mock(ManagedObjectReference.class);
-        ServiceContent serviceContent = Mockito.mock(ServiceContent.class);
         VimPortType vimPort = Mockito.mock(VimPortType.class);
         VmwareClient vimClient = spy(new VmwareClient(vCenterHost));
         VmwareContext vmwareContext = spy(new VmwareContext(vimClient, vCenterHost));
@@ -605,7 +599,7 @@
 
         try (MockedConstruction<VmwareClient> ignored1 = Mockito.mockConstruction(VmwareClient.class, withSettings().spiedInstance(vimClient), (mockVmwareClient, contextVmwareClient) -> {
             Mockito.doReturn(vimPort).when(mockVmwareClient).getService();
-            Mockito.doReturn(serviceContent).when(mockVmwareClient).getServiceContent();
+            Mockito.doReturn(serviceContentMock).when(mockVmwareClient).getServiceContent();
             Mockito.doNothing().when(mockVmwareClient).connect(anyString(), anyString(), anyString());
             Mockito.doReturn(mor).when(mockVmwareClient).getRootFolder();
         }); MockedConstruction<VmwareContext> ignored2 = Mockito.mockConstruction(VmwareContext.class, withSettings().spiedInstance(vmwareContext), (mockVmwareContext, contextVmwareContext) -> {
