--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -274,6 +274,8 @@
     @Mock
     IpAddressManager ipAddressManager;
 
+    private Tag tagMock;
+
     TungstenServiceImpl tungstenService;
 
     AutoCloseable closeable;
@@ -281,6 +283,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        tagMock = mock(Tag.class);
         tungstenService = new TungstenServiceImpl();
         tungstenService.projectDao = projectDao;
         tungstenService.tungstenProviderDao = tungstenProviderDao;
@@ -1073,7 +1076,6 @@
     public void createTungstenTagTest() {
         TungstenAnswer createTungstenTagAnswer = MockTungstenAnswerFactory.get(true);
         TungstenTag tungstenTag = mock(TungstenTag.class);
-        Tag tag = mock(Tag.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
@@ -1083,7 +1085,7 @@
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(CreateTungstenTagCommand.class), anyLong())).thenReturn(createTungstenTagAnswer);
         when(createTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
-        when(tungstenTag.getTag()).thenReturn(tag);
+        when(tungstenTag.getTag()).thenReturn(tagMock);
         doReturn(List.of(virtualNetwork)).when(tungstenTag).getVirtualNetworkList();
         doReturn(List.of(virtualMachine)).when(tungstenTag).getVirtualMachineList();
         doReturn(List.of(virtualMachineInterface)).when(tungstenTag).getVirtualMachineInterfaceList();
@@ -1109,7 +1111,6 @@
     public void listTungstenTagsTest() {
         TungstenAnswer listTungstenTagAnswer = MockTungstenAnswerFactory.get(true);
         TungstenTag tungstenTag = mock(TungstenTag.class);
-        Tag tag = mock(Tag.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
@@ -1120,7 +1121,7 @@
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenTagCommand.class), anyLong())).thenReturn(listTungstenTagAnswer);
         when(listTungstenTagAnswer.getTungstenModelList()).thenReturn(List.of(tungstenTag));
-        when(tungstenTag.getTag()).thenReturn(tag);
+        when(tungstenTag.getTag()).thenReturn(tagMock);
         doReturn(List.of(virtualNetwork)).when(tungstenTag).getVirtualNetworkList();
         doReturn(List.of(virtualMachine)).when(tungstenTag).getVirtualMachineList();
         doReturn(List.of(virtualMachineInterface)).when(tungstenTag).getVirtualMachineInterfaceList();
@@ -1183,7 +1184,6 @@
     public void applyTungstenTagTest() {
         TungstenAnswer applyTungstenTagAnswer = MockTungstenAnswerFactory.get(true);
         TungstenTag tungstenTag = mock(TungstenTag.class);
-        Tag tag = mock(Tag.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
@@ -1193,7 +1193,7 @@
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ApplyTungstenTagCommand.class), anyLong())).thenReturn(applyTungstenTagAnswer);
         when(applyTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
-        when(tungstenTag.getTag()).thenReturn(tag);
+        when(tungstenTag.getTag()).thenReturn(tagMock);
         when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
         when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
         when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
@@ -1225,7 +1225,6 @@
     public void removeTungstenTagTest() {
         TungstenAnswer removeTungstenTagAnswer = MockTungstenAnswerFactory.get(true);
         TungstenTag tungstenTag = mock(TungstenTag.class);
-        Tag tag = mock(Tag.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
@@ -1235,7 +1234,7 @@
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(RemoveTungstenTagCommand.class), anyLong())).thenReturn(removeTungstenTagAnswer);
         when(removeTungstenTagAnswer.getTungstenModel()).thenReturn(tungstenTag);
-        when(tungstenTag.getTag()).thenReturn(tag);
+        when(tungstenTag.getTag()).thenReturn(tagMock);
         when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
         when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
         when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
