--- a/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
+++ b/plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/XenServer610WrapperTest.java
@@ -70,6 +70,8 @@
     @Mock
     protected XenServer610Resource xenServer610Resource;
 
+    private final VDI vdi1Mock = Mockito.mock(VDI.class);
+
     @Test
     public void testCheckNetworkCommandFailure() {
         final XenServer610Resource xenServer610Resource = new XenServer610Resource();
@@ -265,7 +267,6 @@
         final SR sr1 = Mockito.mock(SR.class);
         final SR sr2 = Mockito.mock(SR.class);
 
-        final VDI vdi1 = Mockito.mock(VDI.class);
         final VDI vdi2 = Mockito.mock(VDI.class);
 
         final NicTO nic1 = Mockito.mock(NicTO.class);
@@ -298,7 +299,7 @@
         when(nic1.getMac()).thenReturn(mac);
         when(nic2.getMac()).thenReturn(mac);
 
-        when(xenServer610Resource.getVDIbyUuid(conn, volume1.getPath())).thenReturn(vdi1);
+        when(xenServer610Resource.getVDIbyUuid(conn, volume1.getPath())).thenReturn(vdi1Mock);
         when(xenServer610Resource.getVDIbyUuid(conn, volume2.getPath())).thenReturn(vdi2);
 
         try {
@@ -372,7 +373,6 @@
         final SR sr1 = Mockito.mock(SR.class);
         final SR sr2 = Mockito.mock(SR.class);
 
-        final VDI vdi1 = Mockito.mock(VDI.class);
         final VDI vdi2 = Mockito.mock(VDI.class);
 
         final NicTO nic1 = Mockito.mock(NicTO.class);
@@ -396,7 +396,7 @@
         when(xenServer610Resource.getConnection()).thenReturn(conn);
         when(vmSpec.getName()).thenReturn(vmName);
 
-        when(xenServer610Resource.getVDIbyUuid(conn, volume1.getPath())).thenReturn(vdi1);
+        when(xenServer610Resource.getVDIbyUuid(conn, volume1.getPath())).thenReturn(vdi1Mock);
         when(xenServer610Resource.getVDIbyUuid(conn, volume2.getPath())).thenReturn(vdi2);
 
         final Answer answer = wrapper.execute(migrateStorageCommand, xenServer610Resource);
@@ -437,7 +437,6 @@
     public void testXenServer610MigrateVolumeCommandWrapper() {
 
         final Connection conn = Mockito.mock(Connection.class);
-        final VDI srcVolume = Mockito.mock(VDI.class);
 
         final long volumeId = 1l;
         final String volumePath = "206b21a7-c6ec-40e2-b5e2-f861b9612f04";
@@ -453,7 +452,7 @@
         assertNotNull(wrapper);
 
         when(xenServer610Resource.getConnection()).thenReturn(conn);
-        when(xenServer610Resource.getVDIbyUuid(conn, volumePath)).thenReturn(srcVolume);
+        when(xenServer610Resource.getVDIbyUuid(conn, volumePath)).thenReturn(vdi1Mock);
 
 
         when(xenServer610Resource.getMigrateWait()).thenReturn(120);
