--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -183,6 +183,8 @@
     @Mock
     TungstenApi tungstenApi;
 
+    private VirtualMachine virtualMachine2Mock;
+
     TungstenResource tungstenResource;
 
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
@@ -192,6 +194,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        virtualMachine2Mock = mock(VirtualMachine.class);
         tungstenResource = new TungstenResource();
         tungstenResource.tungstenApi = tungstenApi;
         ReflectionTestUtils.setField(tungstenResource, "vrouterPort", "9091");
@@ -989,8 +992,7 @@
     public void executeRequestListTungstenVmCommandTest() {
         TungstenCommand command = new ListTungstenVmCommand("projectFqn", "ca86c658-1096-11ec-82a8-0242ac130003");
         VirtualMachine virtualMachine1 = mock(VirtualMachine.class);
-        VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
-        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
+        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2Mock);
         when(tungstenApi.listTungstenVm(anyString(), anyString())).thenAnswer(virtualMachinesAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1015,8 +1017,7 @@
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         Answer<List<ApiObjectBase>> virtualNetworksAnswer = setupApiObjectBaseListAnswer(virtualNetwork1, virtualNetwork2);
         VirtualMachine virtualMachine1 = mock(VirtualMachine.class);
-        VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
-        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
+        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2Mock);
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
         VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
         Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
@@ -1110,8 +1111,7 @@
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         Answer<List<ApiObjectBase>> virtualNetworksAnswer = setupApiObjectBaseListAnswer(virtualNetwork1, virtualNetwork2);
         VirtualMachine virtualMachine1 = mock(VirtualMachine.class);
-        VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
-        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
+        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2Mock);
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
         VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
         Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
@@ -1161,8 +1161,7 @@
         VirtualNetwork virtualNetwork2 = mock(VirtualNetwork.class);
         Answer<List<ApiObjectBase>> virtualNetworksAnswer = setupApiObjectBaseListAnswer(virtualNetwork1, virtualNetwork2);
         VirtualMachine virtualMachine1 = mock(VirtualMachine.class);
-        VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
-        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
+        Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2Mock);
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
         VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
         Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
