--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/resource/TungstenResourceTest.java
@@ -185,6 +185,8 @@
 
     TungstenResource tungstenResource;
 
+    private VirtualMachineInterface virtualMachineInterface2Mock;
+
     MockedStatic<TungstenVRouterApi> tungstenVRouterApiMocked;
 
     AutoCloseable closeable;
@@ -192,6 +194,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        virtualMachineInterface2Mock = mock(VirtualMachineInterface.class);
         tungstenResource = new TungstenResource();
         tungstenResource.tungstenApi = tungstenApi;
         ReflectionTestUtils.setField(tungstenResource, "vrouterPort", "9091");
@@ -1000,8 +1003,7 @@
     public void executeRequestListTungstenNicCommandTest() {
         TungstenCommand command = new ListTungstenNicCommand("projectFqn", "5b3ffba2-1096-11ec-82a8-0242ac130003");
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
-        VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
-        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
+        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2Mock);
         when(tungstenApi.listTungstenNic(anyString(), anyString())).thenAnswer(virtualMachineInterfacesAnswer);
         TungstenAnswer answer = (TungstenAnswer) tungstenResource.executeRequest(command);
         assertTrue(answer.getResult());
@@ -1018,8 +1020,7 @@
         VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
         Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
-        VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
-        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
+        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2Mock);
         NetworkPolicy networkPolicy1 = mock(NetworkPolicy.class);
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
@@ -1113,8 +1114,7 @@
         VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
         Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
-        VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
-        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
+        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2Mock);
         NetworkPolicy networkPolicy1 = mock(NetworkPolicy.class);
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
@@ -1164,8 +1164,7 @@
         VirtualMachine virtualMachine2 = mock(VirtualMachine.class);
         Answer<List<ApiObjectBase>> virtualMachinesAnswer = setupApiObjectBaseListAnswer(virtualMachine1, virtualMachine2);
         VirtualMachineInterface virtualMachineInterface1 = mock(VirtualMachineInterface.class);
-        VirtualMachineInterface virtualMachineInterface2 = mock(VirtualMachineInterface.class);
-        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2);
+        Answer<List<ApiObjectBase>> virtualMachineInterfacesAnswer = setupApiObjectBaseListAnswer(virtualMachineInterface1, virtualMachineInterface2Mock);
         NetworkPolicy networkPolicy1 = mock(NetworkPolicy.class);
         NetworkPolicy networkPolicy2 = mock(NetworkPolicy.class);
         Answer<List<ApiObjectBase>> networkPoliciesAnswer = setupApiObjectBaseListAnswer(networkPolicy1, networkPolicy2);
