--- a/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
+++ b/plugins/network-elements/tungsten/src/test/java/org/apache/cloudstack/network/tungsten/service/TungstenServiceImplTest.java
@@ -274,6 +274,8 @@
     @Mock
     IpAddressManager ipAddressManager;
 
+    private VirtualMachineInterface virtualMachineInterfaceMock;
+
     TungstenServiceImpl tungstenService;
 
     AutoCloseable closeable;
@@ -281,6 +283,7 @@
     @Before
     public void setup() {
         closeable = MockitoAnnotations.openMocks(this);
+        virtualMachineInterfaceMock = mock(VirtualMachineInterface.class);
         tungstenService = new TungstenServiceImpl();
         tungstenService.projectDao = projectDao;
         tungstenService.tungstenProviderDao = tungstenProviderDao;
@@ -962,12 +965,11 @@
     @Test
     public void listTungstenNicTest() {
         TungstenAnswer listTungstenNicAnswer = MockTungstenAnswerFactory.get(true);
-        VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
         when(tungstenFabricUtils.sendTungstenCommand(any(ListTungstenNicCommand.class), anyLong())).thenReturn(listTungstenNicAnswer);
-        doReturn(List.of(virtualMachineInterface)).when(listTungstenNicAnswer).getApiObjectBaseList();
+        doReturn(List.of(virtualMachineInterfaceMock)).when(listTungstenNicAnswer).getApiObjectBaseList();
 
         assertNotNull(tungstenService.listTungstenNic(1L, "948f421c-edde-4518-a391-09299cc25dc2"));
     }
@@ -1075,7 +1077,6 @@
         TungstenTag tungstenTag = mock(TungstenTag.class);
         Tag tag = mock(Tag.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
@@ -1086,7 +1087,7 @@
         when(tungstenTag.getTag()).thenReturn(tag);
         doReturn(List.of(virtualNetwork)).when(tungstenTag).getVirtualNetworkList();
         doReturn(List.of(virtualMachine)).when(tungstenTag).getVirtualMachineList();
-        doReturn(List.of(virtualMachineInterface)).when(tungstenTag).getVirtualMachineInterfaceList();
+        doReturn(List.of(virtualMachineInterfaceMock)).when(tungstenTag).getVirtualMachineInterfaceList();
         doReturn(List.of(networkPolicy)).when(tungstenTag).getNetworkPolicyList();
 
         assertNotNull(tungstenService.createTungstenTag(1L, "testTag", "testTagType"));
@@ -1111,7 +1112,6 @@
         TungstenTag tungstenTag = mock(TungstenTag.class);
         Tag tag = mock(Tag.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
-        VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         ApplicationPolicySet applicationPolicySet = mock(ApplicationPolicySet.class);
@@ -1123,7 +1123,7 @@
         when(tungstenTag.getTag()).thenReturn(tag);
         doReturn(List.of(virtualNetwork)).when(tungstenTag).getVirtualNetworkList();
         doReturn(List.of(virtualMachine)).when(tungstenTag).getVirtualMachineList();
-        doReturn(List.of(virtualMachineInterface)).when(tungstenTag).getVirtualMachineInterfaceList();
+        doReturn(List.of(virtualMachineInterfaceMock)).when(tungstenTag).getVirtualMachineInterfaceList();
         doReturn(List.of(networkPolicy)).when(tungstenTag).getNetworkPolicyList();
 
         assertNotNull(tungstenService.listTungstenTags(1L, "948f421c-edde-4518-a391-09299cc25dc2"
@@ -1187,7 +1187,6 @@
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
-        VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
@@ -1197,7 +1196,7 @@
         when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
         when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
         when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
-        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterface));
+        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterfaceMock));
 
         assertNotNull(tungstenService.applyTungstenTag(1L, List.of("948f421c-edde-4518-a391-09299cc25dc2"), List.of("8b4637b6-5629-46de-8fb2-d0b0502bfa85")
                 , List.of("8d097a79-a38d-4db4-8a41-16f15d9c5afa"), "a329662e-1805-4a89-9b05-2b818ea35978", "d5e3f5c5-97ed-41b6-9b6f-7f696b9eddeb"
@@ -1229,7 +1228,6 @@
         NetworkPolicy networkPolicy = mock(NetworkPolicy.class);
         VirtualNetwork virtualNetwork = mock(VirtualNetwork.class);
         VirtualMachine virtualMachine = mock(VirtualMachine.class);
-        VirtualMachineInterface virtualMachineInterface = mock(VirtualMachineInterface.class);
         DataCenterVO dataCenterVO = mock(DataCenterVO.class);
 
         when(dataCenterDao.findById(anyLong())).thenReturn(dataCenterVO);
@@ -1239,7 +1237,7 @@
         when(tungstenTag.getNetworkPolicyList()).thenReturn(List.of(networkPolicy));
         when(tungstenTag.getVirtualNetworkList()).thenReturn(List.of(virtualNetwork));
         when(tungstenTag.getVirtualMachineList()).thenReturn(List.of(virtualMachine));
-        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterface));
+        when(tungstenTag.getVirtualMachineInterfaceList()).thenReturn(List.of(virtualMachineInterfaceMock));
 
         assertNotNull(tungstenService.removeTungstenTag(1L, List.of("948f421c-edde-4518-a391-09299cc25dc2"),
                 List.of("8b4637b6-5629-46de-8fb2-d0b0502bfa85"),
