--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/HelpTelnetHandlerTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/HelpTelnetHandlerTest.java
@@ -28,8 +28,7 @@
 class HelpTelnetHandlerTest {
     @Test
     void test() {
-        Channel channel = Mockito.mock(Channel.class);
-        Mockito.when(channel.getUrl()).thenReturn(URL.valueOf("dubbo://127.0.0.1:12345"));
+        Channel channel = MockChannel.createMockChannel(URL.valueOf("dubbo://127.0.0.1:12345"));
 
         HelpTelnetHandler helpTelnetHandler = new HelpTelnetHandler(FrameworkModel.defaultModel());
         // default output
--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/StatusTelnetHandlerTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/StatusTelnetHandlerTest.java
@@ -27,8 +27,7 @@
 class StatusTelnetHandlerTest {
     @Test
     void test() {
-        Channel channel = Mockito.mock(Channel.class);
-        Mockito.when(channel.getUrl()).thenReturn(URL.valueOf("dubbo://127.0.0.1:12345"));
+        Channel channel = MockChannel.createMockChannel(URL.valueOf("dubbo://127.0.0.1:12345"));
 
         StatusTelnetHandler statusTelnetHandler = new StatusTelnetHandler();
         Assertions.assertNotNull(statusTelnetHandler.telnet(channel, ""));
--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/TelnetHandlerAdapterTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/TelnetHandlerAdapterTest.java
@@ -34,11 +34,10 @@
     @Test
     void testTelnet() throws RemotingException {
 
-        Channel channel = Mockito.mock(Channel.class);
         Map<String, String> param = new HashMap<>();
         param.put("telnet", "status");
         URL url = new URL("p1", "127.0.0.1", 12345, "path1", param);
-        Mockito.when(channel.getUrl()).thenReturn(url);
+        Channel channel = MockChannel.createMockChannel(url);
         TelnetHandlerAdapter telnetHandlerAdapter = new TelnetHandlerAdapter(FrameworkModel.defaultModel());
 
         String message = "--no-prompt status ";
--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/MockChannel.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/telnet/support/MockChannel.java
@@ -0,0 +1,15 @@
+package org.apache.dubbo.remoting.telnet.support;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.remoting.Channel;
+
+import org.mockito.Mockito;
+
+public class MockChannel {
+
+    public static Channel createMockChannel(URL url) {
+        Channel channel = Mockito.mock(Channel.class);
+        Mockito.when(channel.getUrl()).thenReturn(url);
+        return channel;
+    }
+}
