--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvokerTest.java
@@ -516,26 +516,32 @@
         given(firstInvoker.getInterface()).willReturn(MenuService.class);
         given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
 
+        secondInvoker = createMockSecondInvoker(url, invocation);
+
+        given(directory.list(invocation)).willReturn(new ArrayList() {
+
+            {
+                add(firstInvoker);
+                add(secondInvoker);
+            }
+        });
+        given(directory.getUrl()).willReturn(url);
+        given(directory.getConsumerUrl()).willReturn(url);
+        given(directory.getConsumerUrl()).willReturn(url);
+        given(directory.getInterface()).willReturn(MenuService.class);
+
+        mergeableClusterInvoker = new MergeableClusterInvoker<MenuService>(directory);
+        mergeableClusterInvoker.destroy();
+
+        assertFalse(firstInvoker.isAvailable());
+        assertFalse(secondInvoker.isAvailable());
+    }
+
+    private static Invoker createMockSecondInvoker(URL url, Invocation invocation) {
+        Invoker secondInvoker = mock(Invoker.class);
         given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
         given(secondInvoker.getInterface()).willReturn(MenuService.class);
         given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
-
-        given(directory.list(invocation)).willReturn(new ArrayList() {
-
-            {
-                add(firstInvoker);
-                add(secondInvoker);
-            }
-        });
-        given(directory.getUrl()).willReturn(url);
-        given(directory.getConsumerUrl()).willReturn(url);
-        given(directory.getConsumerUrl()).willReturn(url);
-        given(directory.getInterface()).willReturn(MenuService.class);
-
-        mergeableClusterInvoker = new MergeableClusterInvoker<MenuService>(directory);
-        mergeableClusterInvoker.destroy();
-
-        assertFalse(firstInvoker.isAvailable());
-        assertFalse(secondInvoker.isAvailable());
+        return secondInvoker;
     }
 }
