--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/loadbalance/AbstractLoadBalanceTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/loadbalance/AbstractLoadBalanceTest.java
@@ -49,15 +49,13 @@
         RpcInvocation invocation = new RpcInvocation();
         invocation.setMethodName("say");
 
-        Invoker invoker1 = mock(Invoker.class, Mockito.withSettings().stubOnly());
         URL url1 = new ServiceConfigURL("", "", 0, "DemoService", new HashMap<>());
         url1 = url1.addParameter(TIMESTAMP_KEY, System.currentTimeMillis() - Integer.MAX_VALUE - 1);
-        given(invoker1.getUrl()).willReturn(url1);
+        Invoker invoker1 = createMockInvoker(url1);
 
-        Invoker invoker2 = mock(Invoker.class, Mockito.withSettings().stubOnly());
         URL url2 = new ServiceConfigURL("", "", 0, "DemoService", new HashMap<>());
         url2 = url2.addParameter(TIMESTAMP_KEY, System.currentTimeMillis() - 10 * 60 * 1000L - 1);
-        given(invoker2.getUrl()).willReturn(url2);
+        Invoker invoker2 = createMockInvoker(url2);
 
         Assertions.assertEquals(balance.getWeight(invoker1, invocation), balance.getWeight(invoker2, invocation));
     }
@@ -67,9 +65,8 @@
         RpcInvocation invocation = new RpcInvocation();
         invocation.setMethodName("say");
 
-        Invoker invoker1 = mock(Invoker.class, Mockito.withSettings().stubOnly());
         URL url1 = new ServiceConfigURL("", "", 0, "DemoService", new HashMap<>());
-        given(invoker1.getUrl()).willReturn(url1);
+        Invoker invoker1 = createMockInvoker(url1);
 
         ClusterInvoker invoker2 =
                 mock(ClusterInvoker.class, Mockito.withSettings().stubOnly());
@@ -84,4 +81,10 @@
         Assertions.assertEquals(100, balance.getWeight(invoker1, invocation));
         Assertions.assertEquals(30, balance.getWeight(invoker2, invocation));
     }
+
+    private static Invoker createMockInvoker(URL urlReturn) {
+        Invoker invoker1 = mock(Invoker.class, Mockito.withSettings().stubOnly());
+        given(invoker1.getUrl()).willReturn(urlReturn);
+        return invoker1;
+    }
 }
