--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeServerTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeServerTest.java
@@ -48,7 +48,7 @@
 
         // test getChannels() and getExchangeChannels()
         Channel channel1 = Mockito.mock(Channel.class);
-        Channel channel2 = Mockito.mock(Channel.class);
+        Channel channel2 = createMockChannel();
         Channel exchangeChannel1 = new HeaderExchangeChannel(channel1);
         Channel exchangeChannel2 = new HeaderExchangeChannel(channel2);
         Mockito.when(channel1.getAttribute(HeaderExchangeChannel.class.getName() + ".CHANNEL"))
@@ -89,4 +89,8 @@
         Assertions.assertThrows(RemotingException.class, () -> headerExchangeServer.send("test"));
         Assertions.assertThrows(RemotingException.class, () -> headerExchangeServer.send("test", true));
     }
+
+    private static Channel createMockChannel() {
+        return Mockito.mock(Channel.class);
+    }
 }
