--- a/spring-integration-core/src/test/java/org/springframework/integration/config/GlobalChannelInterceptorProcessorTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/config/GlobalChannelInterceptorProcessorTests.java
@@ -44,10 +44,13 @@
 
 	private ListableBeanFactory beanFactory;
 
+	private ChannelInterceptor sharedChannelInterceptor;
+
 	@BeforeEach
 	public void setup() {
 		this.globalChannelInterceptorProcessor = new GlobalChannelInterceptorProcessor();
 		this.beanFactory = mock(ListableBeanFactory.class);
+		this.sharedChannelInterceptor = mock();
 		this.globalChannelInterceptorProcessor.setBeanFactory(this.beanFactory);
 	}
 
@@ -66,9 +69,8 @@
 	public void testProcessorWithInterceptorDefaultPattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
-				new GlobalChannelInterceptorWrapper(channelInterceptor);
+				new GlobalChannelInterceptorWrapper(sharedChannelInterceptor);
 
 		InterceptableChannel channel = mock();
 
@@ -82,16 +84,15 @@
 		this.globalChannelInterceptorProcessor.afterSingletonsInstantiated();
 
 		verify(channel)
-				.addInterceptor(channelInterceptor);
+				.addInterceptor(sharedChannelInterceptor);
 	}
 
 	@Test
 	public void testProcessorWithInterceptorMatchingPattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
-				new GlobalChannelInterceptorWrapper(channelInterceptor);
+				new GlobalChannelInterceptorWrapper(sharedChannelInterceptor);
 
 		InterceptableChannel channel = mock();
 
@@ -105,16 +106,15 @@
 		this.globalChannelInterceptorProcessor.afterSingletonsInstantiated();
 
 		verify(channel)
-				.addInterceptor(channelInterceptor);
+				.addInterceptor(sharedChannelInterceptor);
 	}
 
 	@Test
 	public void testProcessorWithInterceptorNotMatchingPattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
-				new GlobalChannelInterceptorWrapper(channelInterceptor);
+				new GlobalChannelInterceptorWrapper(sharedChannelInterceptor);
 
 		InterceptableChannel channel = mock();
 
@@ -129,16 +129,15 @@
 		this.globalChannelInterceptorProcessor.afterSingletonsInstantiated();
 
 		verify(channel, never())
-				.addInterceptor(channelInterceptor);
+				.addInterceptor(sharedChannelInterceptor);
 	}
 
 	@Test
 	public void testProcessorWithInterceptorMatchingNegativePattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
-				new GlobalChannelInterceptorWrapper(channelInterceptor);
+				new GlobalChannelInterceptorWrapper(sharedChannelInterceptor);
 
 		InterceptableChannel channel = mock();
 
@@ -152,7 +151,7 @@
 		this.globalChannelInterceptorProcessor.afterSingletonsInstantiated();
 
 		verify(channel, never())
-				.addInterceptor(channelInterceptor);
+				.addInterceptor(sharedChannelInterceptor);
 	}
 
 }
