--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -19,6 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.expression.Expression;
@@ -55,6 +56,13 @@
  */
 public class WsDslTests {
 
+	private ClientInterceptor clientInterceptor;
+
+	@BeforeEach
+	void setUp() {
+		clientInterceptor = mock();
+	}
+
 	@Test
 	void marshallingInbound() {
 		Marshaller marshaller = mock();
@@ -93,7 +101,6 @@
 		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
-		ClientInterceptor interceptor = mock();
 		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
@@ -108,7 +115,7 @@
 						.faultMessageResolver(faultMessageResolver)
 						.headerMapper(headerMapper)
 						.ignoreEmptyResponses(true)
-						.interceptors(interceptor)
+						.interceptors(clientInterceptor)
 						.messageSenders(messageSender)
 						.requestCallback(requestCallback)
 						.uriVariableExpressions(uriVariableExpressions)
@@ -122,7 +129,7 @@
 				.isSameAs(faultMessageResolver);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(headerMapper);
 		assertThat(TestUtils.<ClientInterceptor[]>getPropertyValue(gateway, "webServiceTemplate.interceptors")[0])
-				.isSameAs(interceptor);
+				.isSameAs(clientInterceptor);
 		assertThat(TestUtils.<WebServiceMessageSender[]>getPropertyValue(gateway, "webServiceTemplate.messageSenders")[0])
 				.isSameAs(messageSender);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "requestCallback")).isSameAs(requestCallback);
@@ -136,7 +143,6 @@
 		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
-		ClientInterceptor interceptor = mock();
 		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
@@ -152,7 +158,7 @@
 						.faultMessageResolver(faultMessageResolver)
 						.headerMapper(headerMapper)
 						.ignoreEmptyResponses(true)
-						.interceptors(interceptor)
+						.interceptors(clientInterceptor)
 						.messageSenders(messageSender)
 						.requestCallback(requestCallback)
 						.uriVariableExpressions(uriVariableExpressions)
@@ -164,7 +170,7 @@
 				.isSameAs(faultMessageResolver);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(headerMapper);
 		assertThat(TestUtils.<ClientInterceptor[]>getPropertyValue(gateway, "webServiceTemplate.interceptors")[0])
-				.isSameAs(interceptor);
+				.isSameAs(clientInterceptor);
 		assertThat(TestUtils.<WebServiceMessageSender[]>getPropertyValue(gateway, "webServiceTemplate.messageSenders")[0])
 				.isSameAs(messageSender);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "requestCallback")).isSameAs(requestCallback);
