--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
@@ -95,8 +95,7 @@
 
 	@Test
 	public void testInt2809JavaTypePropertiesToAmqp() throws Exception {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -140,8 +139,7 @@
 
 	@Test
 	public void testInt2809JavaTypePropertiesFromAmqp() throws Exception {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -178,8 +176,7 @@
 
 	@Test
 	public void testMessageConverterJsonHeadersHavePrecedenceOverMessageHeaders() throws Exception {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -246,8 +243,7 @@
 
 	@Test
 	public void testAdapterConversionError() throws Exception {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -297,8 +293,7 @@
 
 	@Test
 	public void testGatewayConversionError() throws Exception {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -639,8 +634,7 @@
 
 	@Test
 	public void testAdapterConversionErrorConsumerBatchExtract() {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -697,8 +691,7 @@
 
 	@Test
 	public void testAdapterConversionErrorConsumerBatch() {
-		Connection connection = mock();
-		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection();
 		ConnectionFactory connectionFactory = mock();
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -851,6 +844,12 @@
 		assertThat(recoveredMessages.get()).isSameAs(messages);
 	}
 
+	private static Connection createMockConnection() {
+		Connection connection = mock();
+		willReturn(mock(Channel.class)).given(connection).createChannel(anyBoolean());
+		return connection;
+	}
+
 	public record Foo(String bar) {
 
 	}
