--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/inbound/MessageSourceTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/inbound/MessageSourceTests.java
@@ -369,19 +369,8 @@
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	void testAckOutOfOrder() {
-		Consumer consumer = mock(Consumer.class);
+		Consumer consumer = createMockConsumer();
 		TopicPartition topicPartition = new TopicPartition("foo", 0);
-		willAnswer(i -> {
-			((ConsumerRebalanceListener) i.getArgument(1))
-					.onPartitionsAssigned(Collections.singletonList(topicPartition));
-			return null;
-		}).given(consumer).subscribe(anyCollection(), any(ConsumerRebalanceListener.class));
-		AtomicReference<Set<TopicPartition>> paused = new AtomicReference<>(new HashSet<>());
-		willAnswer(i -> {
-			paused.set(new HashSet<>(i.getArgument(0)));
-			return null;
-		}).given(consumer).pause(anyCollection());
-		willAnswer(i -> paused.get()).given(consumer).paused();
 		Map<TopicPartition, List<ConsumerRecord>> records1 = new LinkedHashMap<>();
 		records1.put(topicPartition, Collections.singletonList(
 				new ConsumerRecord("foo", 0, 0L, 0L, TimestampType.NO_TIMESTAMP_TYPE, 0, 0, null, "foo",
@@ -465,19 +454,8 @@
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	void testNack() {
-		Consumer consumer = mock(Consumer.class);
+		Consumer consumer = createMockConsumer();
 		TopicPartition topicPartition = new TopicPartition("foo", 0);
-		willAnswer(i -> {
-			((ConsumerRebalanceListener) i.getArgument(1))
-					.onPartitionsAssigned(Collections.singletonList(topicPartition));
-			return null;
-		}).given(consumer).subscribe(anyCollection(), any(ConsumerRebalanceListener.class));
-		AtomicReference<Set<TopicPartition>> paused = new AtomicReference<>(new HashSet<>());
-		willAnswer(i -> {
-			paused.set(new HashSet<>(i.getArgument(0)));
-			return null;
-		}).given(consumer).pause(anyCollection());
-		willAnswer(i -> paused.get()).given(consumer).paused();
 		Map<TopicPartition, List<ConsumerRecord>> records1 = new LinkedHashMap<>();
 		records1.put(topicPartition, Collections.singletonList(
 				new ConsumerRecord("foo", 0, 0L, 0L, TimestampType.NO_TIMESTAMP_TYPE, 0, 0, null, "foo",
@@ -536,19 +514,8 @@
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	void testNackWithLaterInflight() {
-		Consumer consumer = mock(Consumer.class);
+		Consumer consumer = createMockConsumer();
 		TopicPartition topicPartition = new TopicPartition("foo", 0);
-		willAnswer(i -> {
-			((ConsumerRebalanceListener) i.getArgument(1))
-					.onPartitionsAssigned(Collections.singletonList(topicPartition));
-			return null;
-		}).given(consumer).subscribe(anyCollection(), any(ConsumerRebalanceListener.class));
-		AtomicReference<Set<TopicPartition>> paused = new AtomicReference<>(new HashSet<>());
-		willAnswer(i -> {
-			paused.set(new HashSet<>(i.getArgument(0)));
-			return null;
-		}).given(consumer).pause(anyCollection());
-		willAnswer(i -> paused.get()).given(consumer).paused();
 		Map<TopicPartition, List<ConsumerRecord>> records1 = new LinkedHashMap<>();
 		records1.put(topicPartition, Collections.singletonList(
 				new ConsumerRecord("foo", 0, 0L, 0L, TimestampType.NO_TIMESTAMP_TYPE, 0, 0, null, "foo",
@@ -950,4 +917,22 @@
 		inOrder.verify(consumer).close(any(CloseOptions.class));
 	}
 
+	@SuppressWarnings({"rawtypes", "unchecked"})
+	private static Consumer createMockConsumer() {
+		Consumer consumer = mock(Consumer.class);
+		TopicPartition topicPartition = new TopicPartition("foo", 0);
+		AtomicReference<Set<TopicPartition>> paused = new AtomicReference<>(new HashSet<>());
+		willAnswer(i -> paused.get()).given(consumer).paused();
+		willAnswer(i -> {
+			((ConsumerRebalanceListener) i.getArgument(1))
+					.onPartitionsAssigned(Collections.singletonList(topicPartition));
+			return null;
+		}).given(consumer).subscribe(anyCollection(), any(ConsumerRebalanceListener.class));
+		willAnswer(i -> {
+			paused.set(new HashSet<>(i.getArgument(0)));
+			return null;
+		}).given(consumer).pause(anyCollection());
+		return consumer;
+	}
+
 }
