--- a/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
+++ b/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
@@ -675,12 +675,10 @@
 		for (int i = 0; i < 100; i++) {
 			final ImapMailReceiver receiver = new ImapMailReceiver("imap://foo");
 			Store store = mock(Store.class);
-			Folder folder = mock(Folder.class);
+			Folder folder = createMockFolder();
 			given(folder.exists()).willReturn(true);
-			given(folder.isOpen()).willReturn(true);
 			given(folder.search(Mockito.any())).willReturn(new Message[] {});
 			given(store.getFolder(Mockito.any(URLName.class))).willReturn(folder);
-			given(folder.getPermanentFlags()).willReturn(new Flags(Flags.Flag.USER));
 
 			DirectFieldAccessor df = new DirectFieldAccessor(receiver);
 			df.setPropertyValue("store", store);
@@ -754,14 +752,12 @@
 
 	private Folder testAttachmentsGuts(final ImapMailReceiver receiver) throws MessagingException, IOException {
 		Store store = mock(Store.class);
-		Folder folder = mock(Folder.class);
+		Folder folder = createMockFolder();
 		given(folder.exists()).willReturn(true);
-		given(folder.isOpen()).willReturn(true);
 
 		Message message = GreenMailUtil.newMimeMessage(new ClassPathResource("test.mail").getInputStream());
 		given(folder.search(Mockito.any())).willReturn(new Message[] {message});
 		given(store.getFolder(Mockito.any(URLName.class))).willReturn(folder);
-		given(folder.getPermanentFlags()).willReturn(new Flags(Flags.Flag.USER));
 		DirectFieldAccessor df = new DirectFieldAccessor(receiver);
 		df.setPropertyValue("store", store);
 		receiver.setBeanFactory(TEST_INTEGRATION_CONTEXT);
@@ -877,6 +873,13 @@
 		// msg2 is marked with the user and seen flags
 		verify(msg1, times(2)).setFlags(Mockito.any(), Mockito.anyBoolean());
 		verify(receiver, times(0)).deleteMessages(Mockito.any());
+	}
+
+	private static Folder createMockFolder() {
+		Folder folder = mock(Folder.class);
+		given(folder.getPermanentFlags()).willReturn(new Flags(Flags.Flag.USER));
+		given(folder.isOpen()).willReturn(true);
+		return folder;
 	}
 
 	private void testIdleWithServerGuts(ImapMailReceiver receiver, boolean mapped) throws Exception {
