--- a/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
+++ b/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
@@ -541,10 +541,8 @@
 
 		ImapMailReceiver receiver = new ImapMailReceiver("imap:foo");
 
-		final IMAPFolder folder = mock(IMAPFolder.class);
+		final IMAPFolder folder = createMockImapFolder();
 		given(folder.getPermanentFlags()).willReturn(new Flags(Flags.Flag.USER));
-		given(folder.isOpen()).willReturn(false).willReturn(true);
-		given(folder.exists()).willReturn(true);
 
 		Field storeField = AbstractMailReceiver.class.getDeclaredField("store");
 		storeField.setAccessible(true);
@@ -607,10 +605,8 @@
 
 		ImapMailReceiver receiver = new ImapMailReceiver("imap:foo");
 		receiver.setCancelIdleInterval(10);
-		IMAPFolder folder = mock(IMAPFolder.class);
+		IMAPFolder folder = createMockImapFolder();
 		given(folder.getPermanentFlags()).willReturn(new Flags(Flags.Flag.RECENT));
-		given(folder.isOpen()).willReturn(false).willReturn(true);
-		given(folder.exists()).willReturn(true);
 
 		Field storeField = AbstractMailReceiver.class.getDeclaredField("store");
 		storeField.setAccessible(true);
@@ -752,6 +748,13 @@
 		assertThat(((Multipart) content).getBodyPart(1).getContent().toString().trim()).isEqualTo("foo");
 	}
 
+	private static IMAPFolder createMockImapFolder() {
+		IMAPFolder folder = mock(IMAPFolder.class);
+		given(folder.isOpen()).willReturn(false).willReturn(true);
+		given(folder.exists()).willReturn(true);
+		return folder;
+	}
+
 	private Folder testAttachmentsGuts(final ImapMailReceiver receiver) throws MessagingException, IOException {
 		Store store = mock(Store.class);
 		Folder folder = mock(Folder.class);
@@ -822,10 +825,8 @@
 		ImapMailReceiver receiver = spy(new ImapMailReceiver("imap:foo"));
 		receiver.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		receiver.afterPropertiesSet();
-		IMAPFolder folder = mock(IMAPFolder.class);
+		IMAPFolder folder = createMockImapFolder();
 		given(folder.getPermanentFlags()).willReturn(new Flags(Flags.Flag.USER));
-		given(folder.isOpen()).willReturn(false).willReturn(true);
-		given(folder.exists()).willReturn(true);
 		given(folder.hasNewMessages()).willReturn(true);
 		Field storeField = AbstractMailReceiver.class.getDeclaredField("store");
 		storeField.setAccessible(true);
