--- a/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
+++ b/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
@@ -280,7 +280,7 @@
 		willReturn(client).given(factory).getAsyncClientInstance(anyString(), anyString());
 
 		final AtomicBoolean connectCalled = new AtomicBoolean();
-		IMqttToken token = mock(IMqttToken.class);
+		IMqttToken token = createMockMqttToken(new int[] {2});
 		willAnswer(invocation -> {
 			MqttConnectOptions options = invocation.getArgument(0);
 			assertThat(options.getConnectionTimeout()).isEqualTo(23);
@@ -296,7 +296,6 @@
 			return token;
 		}).given(client).connect(any(MqttConnectOptions.class));
 		given(client.subscribe(any(String[].class), any(int[].class), any())).willReturn(token);
-		given(token.getGrantedQos()).willReturn(new int[] {2});
 
 		final AtomicReference<MqttCallbackExtended> callback = new AtomicReference<>();
 		willAnswer(invocation -> {
@@ -440,8 +439,7 @@
 		given(client.isConnected()).willReturn(true);
 		willReturn(alwaysComplete).given(client).connect(any(MqttConnectOptions.class));
 
-		IMqttToken token = mock(IMqttToken.class);
-		given(token.getGrantedQos()).willReturn(new int[] {0x80});
+		IMqttToken token = createMockMqttToken(new int[] {0x80});
 		willReturn(token).given(client).subscribe(any(String[].class), any(int[].class), any());
 
 		MqttPahoMessageDrivenChannelAdapter adapter = new MqttPahoMessageDrivenChannelAdapter("foo", "bar", factory,
@@ -487,8 +485,7 @@
 		given(client.isConnected()).willReturn(true);
 		willReturn(alwaysComplete).given(client).connect(any(MqttConnectOptions.class));
 
-		IMqttToken token = mock(IMqttToken.class);
-		given(token.getGrantedQos()).willReturn(new int[] {2, 0});
+		IMqttToken token = createMockMqttToken(new int[] {2, 0});
 		willReturn(token).given(client).subscribe(any(String[].class), any(int[].class), any());
 
 		MqttPahoMessageDrivenChannelAdapter adapter =
@@ -534,6 +531,12 @@
 				.withMessage("The topic to subscribe cannot be empty string");
 	}
 
+	private static IMqttToken createMockMqttToken(int[] grantedQos) {
+		IMqttToken token = mock(IMqttToken.class);
+		given(token.getGrantedQos()).willReturn(grantedQos);
+		return token;
+	}
+
 	private MqttPahoMessageDrivenChannelAdapter buildAdapterIn(final IMqttAsyncClient client, Boolean cleanSession)
 			throws MqttException {
 
