--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
@@ -74,15 +74,14 @@
 	@Test
 	public void testNetClient() throws Exception {
 		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
-		SocketFactory factory = Mockito.mock(SocketFactory.class);
+		Socket socket = mock(Socket.class);
+		SocketFactory factory = createMockSocketFactory(socket);
 		when(factorySupport.getSocketFactory()).thenReturn(factory);
-		Socket socket = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
 		when(socket.getInputStream()).thenReturn(is);
 		InetAddress inetAddress = InetAddress.getLocalHost();
 		when(socket.getInetAddress()).thenReturn(inetAddress);
-		when(factory.createSocket()).thenReturn(socket);
 		TcpSocketSupport socketSupport = Mockito.mock(TcpSocketSupport.class);
 
 		TcpNetClientConnectionFactory connectionFactory = new TcpNetClientConnectionFactory("x", 0);
@@ -101,15 +100,14 @@
 	@Test
 	public void testNetClientSocketTimeout() throws Exception {
 		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
-		SocketFactory factory = Mockito.mock(SocketFactory.class);
+		Socket socket = mock(Socket.class);
+		SocketFactory factory = createMockSocketFactory(socket);
 		when(factorySupport.getSocketFactory()).thenReturn(factory);
-		Socket socket = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
 		when(socket.getInputStream()).thenReturn(is);
 		InetAddress inetAddress = InetAddress.getLocalHost();
 		when(socket.getInetAddress()).thenReturn(inetAddress);
-		when(factory.createSocket()).thenReturn(socket);
 		doThrow(new SocketTimeoutException()).when(socket).connect(any(), eq(1000));
 		TcpSocketSupport socketSupport = Mockito.mock(TcpSocketSupport.class);
 
@@ -720,4 +718,10 @@
 
 	}
 
+	private static SocketFactory createMockSocketFactory(Socket socket) {
+		SocketFactory factory = Mockito.mock(SocketFactory.class);
+		when(factory.createSocket()).thenReturn(socket);
+		return factory;
+	}
+
 }
