--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
@@ -76,12 +76,10 @@
 		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
 		SocketFactory factory = Mockito.mock(SocketFactory.class);
 		when(factorySupport.getSocketFactory()).thenReturn(factory);
-		Socket socket = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
-		when(socket.getInputStream()).thenReturn(is);
 		InetAddress inetAddress = InetAddress.getLocalHost();
-		when(socket.getInetAddress()).thenReturn(inetAddress);
+		Socket socket = createMockSocket(is, inetAddress);
 		when(factory.createSocket()).thenReturn(socket);
 		TcpSocketSupport socketSupport = Mockito.mock(TcpSocketSupport.class);
 
@@ -103,12 +101,10 @@
 		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
 		SocketFactory factory = Mockito.mock(SocketFactory.class);
 		when(factorySupport.getSocketFactory()).thenReturn(factory);
-		Socket socket = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
-		when(socket.getInputStream()).thenReturn(is);
 		InetAddress inetAddress = InetAddress.getLocalHost();
-		when(socket.getInetAddress()).thenReturn(inetAddress);
+		Socket socket = createMockSocket(is, inetAddress);
 		when(factory.createSocket()).thenReturn(socket);
 		doThrow(new SocketTimeoutException()).when(socket).connect(any(), eq(1000));
 		TcpSocketSupport socketSupport = Mockito.mock(TcpSocketSupport.class);
@@ -133,15 +129,11 @@
 		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
 		ServerSocketFactory factory = mock(ServerSocketFactory.class);
 		when(factorySupport.getServerSocketFactory()).thenReturn(factory);
-		Socket socket = mock(Socket.class);
-		Socket socket1 = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
-		when(socket.getInputStream()).thenReturn(is);
-		when(socket1.getInputStream()).thenReturn(is);
 		InetAddress inetAddress = InetAddress.getLocalHost();
-		when(socket.getInetAddress()).thenReturn(inetAddress);
-		when(socket1.getInetAddress()).thenReturn(inetAddress);
+		Socket socket1 = createMockSocket(is, inetAddress);
+		Socket socket = createMockSocket(is, inetAddress);
 		ServerSocket serverSocket = mock(ServerSocket.class);
 		AtomicBoolean closed = new AtomicBoolean();
 		doAnswer(invoc -> {
@@ -688,6 +680,13 @@
 		server.stop();
 	}
 
+	private static Socket createMockSocket(InputStream inputStream, InetAddress inetAddress) {
+		Socket socket = mock(Socket.class);
+		when(socket.getInputStream()).thenReturn(inputStream);
+		when(socket.getInetAddress()).thenReturn(inetAddress);
+		return socket;
+	}
+
 	private static class Replier implements TcpSender {
 
 		private TcpConnection connection;
