--- a/spring-integration-core/src/test/java/org/springframework/integration/channel/P2pChannelTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/channel/P2pChannelTests.java
@@ -92,13 +92,18 @@
 		verify(logger, times(4)).info(Mockito.anyString());
 	}
 
+	private static LogAccessor createMockLogger() {
+		final LogAccessor logger = mock(LogAccessor.class);
+		when(logger.isInfoEnabled()).thenReturn(true);
+		return logger;
+	}
+
 	@Test
 	public void testExecutorChannelLoggingWithMoreThenOneSubscriber() {
 		final ExecutorChannel channel = new ExecutorChannel(mock(Executor.class));
 		channel.setBeanName("executorChannel");
 
-		final LogAccessor logger = mock(LogAccessor.class);
-		when(logger.isInfoEnabled()).thenReturn(true);
+		final LogAccessor logger = createMockLogger();
 		ReflectionUtils.doWithFields(AbstractMessageChannel.class, field -> {
 			if ("logger".equals(field.getName())) {
 				field.setAccessible(true);
@@ -115,8 +120,7 @@
 		final PublishSubscribeChannel channel = new PublishSubscribeChannel();
 		channel.setBeanName("pubSubChannel");
 
-		final LogAccessor logger = mock(LogAccessor.class);
-		when(logger.isInfoEnabled()).thenReturn(true);
+		final LogAccessor logger = createMockLogger();
 		ReflectionUtils.doWithFields(AbstractMessageChannel.class, field -> {
 			if ("logger".equals(field.getName())) {
 				field.setAccessible(true);
