--- a/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
+++ b/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
@@ -277,10 +277,9 @@
 		new DirectFieldAccessor(receiver).setPropertyValue("logger", logger);
 		when(logger.isDebugEnabled()).thenReturn(true);
 
-		Message msg1 = mock(MimeMessage.class);
+		Message msg1 = createMockMessageWithSubject();
 		Message msg2 = mock(MimeMessage.class);
 		given(msg1.isExpunged()).willReturn(true);
-		given(msg1.getSubject()).willReturn("msg1");
 		given(msg2.getSubject()).willReturn("msg2");
 		Expression selectorExpression = new SpelExpressionParser().parseExpression("false");
 		receiver.setSelectorExpression(selectorExpression);
@@ -750,6 +749,12 @@
 		assertThat(content).isInstanceOf(Multipart.class);
 		assertThat(((Multipart) content).getBodyPart(0).getContent().toString().trim()).isEqualTo("bar");
 		assertThat(((Multipart) content).getBodyPart(1).getContent().toString().trim()).isEqualTo("foo");
+	}
+
+	private static Message createMockMessageWithSubject() {
+		Message message = mock(MimeMessage.class);
+		given(message.getSubject()).willReturn("msg1");
+		return message;
 	}
 
 	private Folder testAttachmentsGuts(final ImapMailReceiver receiver) throws MessagingException, IOException {
