--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandlerTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandlerTests.java
@@ -21,6 +21,7 @@
 import java.util.Set;
 import java.util.concurrent.atomic.AtomicReference;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -57,6 +58,13 @@
  */
 public class FileTransferringMessageHandlerTests implements TestApplicationContextAware {
 
+	private Session sessionMock;
+
+	@BeforeEach
+	void setUp() {
+		sessionMock = mock(Session.class);
+	}
+
 	@SuppressWarnings("unchecked")
 	@Test
 	public <F> void testRemoteDirWithEmptyString() throws Exception {
@@ -129,8 +137,7 @@
 	@Test
 	public <F> void testEmptyTemporaryFileSuffixCannotBeNull() throws Exception {
 		SessionFactory<F> sf = mock(SessionFactory.class);
-		Session<F> session = mock(Session.class);
-		when(sf.getSession()).thenReturn(session);
+		when(sf.getSession()).thenReturn(sessionMock);
 		FileTransferringMessageHandler<F> handler = new FileTransferringMessageHandler<F>(sf);
 		handler.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		handler.setRemoteDirectoryExpressionString("headers['path']");
@@ -142,9 +149,8 @@
 	@Test
 	public <F> void testUseTemporaryFileNameFalse() throws Exception {
 		SessionFactory<F> sf = mock(SessionFactory.class);
-		Session<F> session = mock(Session.class);
-
-		when(sf.getSession()).thenReturn(session);
+
+		when(sf.getSession()).thenReturn(sessionMock);
 
 		ExpressionParser parser = new SpelExpressionParser();
 		FileTransferringMessageHandler<F> handler = new FileTransferringMessageHandler<F>(sf);
@@ -154,8 +160,8 @@
 		handler.afterPropertiesSet();
 		Message<?> message = MessageBuilder.withPayload("hello").setHeader("path", null).build();
 		handler.handleMessage(message);
-		verify(session, times(1)).write(Mockito.any(InputStream.class), Mockito.anyString());
-		verify(session, times(0)).rename(Mockito.anyString(), Mockito.anyString());
+		verify(sessionMock, times(1)).write(Mockito.any(InputStream.class), Mockito.anyString());
+		verify(sessionMock, times(0)).rename(Mockito.anyString(), Mockito.anyString());
 	}
 
 	@SuppressWarnings("unchecked")
