--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
@@ -32,6 +32,7 @@
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicReference;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 import org.mockito.ArgumentCaptor;
@@ -82,20 +83,25 @@
 
 	private final String tmpDir = System.getProperty("java.io.tmpdir");
 
+	private SessionFactory sessionFactoryMock;
+
+	@BeforeEach
+	public void setUp() {
+		this.sessionFactoryMock = mock(SessionFactory.class);
+	}
+
 	@TempDir
 	public static File tempFolder;
 
 	@Test
 	public void testBad() {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		assertThatIllegalArgumentException()
-				.isThrownBy(() -> new TestRemoteFileOutboundGateway(sessionFactory, "bad", "payload"));
+				.isThrownBy(() -> new TestRemoteFileOutboundGateway(sessionFactoryMock, "bad", "payload"));
 	}
 
 	@Test
 	public void testBadFilterGet() {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
-		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "get", "payload");
+		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactoryMock, "get", "payload");
 		gw.setFilter(new TestPatternFilter(""));
 
 		assertThatIllegalArgumentException()
@@ -105,8 +111,7 @@
 
 	@Test
 	public void testBadFilterRm() {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
-		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "rm", "payload");
+		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactoryMock, "rm", "payload");
 		gw.setFilter(new TestPatternFilter(""));
 
 		assertThatIllegalArgumentException()
