--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
@@ -708,6 +708,14 @@
 		reader.close();
 	}
 
+	private static Session<TestLsEntry> createMockSession() throws IOException {
+		Session<TestLsEntry> session = mock(Session.class);
+		willReturn(Boolean.TRUE)
+				.given(session)
+				.exists(anyString());
+		return session;
+	}
+
 	@Test
 	public void testGetTempFileDelete() {
 		@SuppressWarnings("unchecked")
@@ -865,11 +873,7 @@
 	public void testPutExists() throws Exception {
 		@SuppressWarnings("unchecked")
 		SessionFactory<TestLsEntry> sessionFactory = mock(SessionFactory.class);
-		@SuppressWarnings("unchecked")
-		Session<TestLsEntry> session = mock(Session.class);
-		willReturn(Boolean.TRUE)
-				.given(session)
-				.exists(anyString());
+		Session<TestLsEntry> session = createMockSession();
 		RemoteFileTemplate<TestLsEntry> template = new RemoteFileTemplate<>(sessionFactory);
 		template.setRemoteDirectoryExpression(new LiteralExpression("foo/"));
 		template.setBeanFactory(TEST_INTEGRATION_CONTEXT);
@@ -926,10 +930,7 @@
 	@SuppressWarnings("unchecked")
 	public void testPutExistsExpression() throws Exception {
 		SessionFactory<TestLsEntry> sessionFactory = mock(SessionFactory.class);
-		Session<TestLsEntry> session = mock(Session.class);
-		willReturn(Boolean.TRUE)
-				.given(session)
-				.exists(anyString());
+		Session<TestLsEntry> session = createMockSession();
 		RemoteFileTemplate<TestLsEntry> template = new RemoteFileTemplate<>(sessionFactory);
 		template.setRemoteDirectoryExpression(new LiteralExpression("foo/"));
 		template.setBeanFactory(TEST_INTEGRATION_CONTEXT);
