--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/CachingClientConnectionFactoryTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/CachingClientConnectionFactoryTests.java
@@ -133,8 +133,7 @@
 
 	@Test
 	public void testReuse() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1");
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2");
 		when(factory.getConnection()).thenReturn(mockConn1).thenReturn(mockConn2);
@@ -163,8 +162,7 @@
 
 	@Test
 	public void testReuseNoLimit() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1");
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2");
 		when(factory.getConnection()).thenReturn(mockConn1).thenReturn(mockConn2);
@@ -183,8 +181,7 @@
 
 	@Test
 	public void testReuseClosed() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1");
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2");
 		doAnswer(invocation -> null).when(mockConn1).close();
@@ -212,8 +209,7 @@
 
 	@Test
 	public void testLimit() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1");
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2");
 		when(factory.getConnection()).thenReturn(mockConn1).thenReturn(mockConn2);
@@ -233,8 +229,7 @@
 
 	@Test
 	public void testStop() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1");
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2");
 		int i = 3;
@@ -271,8 +266,7 @@
 
 	@Test
 	public void testEnlargePool() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1");
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2");
 		TcpConnectionSupport mockConn3 = makeMockConnection("conn3");
@@ -300,8 +294,7 @@
 
 	@Test
 	public void testReducePool() throws Exception {
-		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
-		when(factory.isRunning()).thenReturn(true);
+		AbstractClientConnectionFactory factory = createMockClientConnectionFactory();
 		TcpConnectionSupport mockConn1 = makeMockConnection("conn1", true);
 		TcpConnectionSupport mockConn2 = makeMockConnection("conn2", true);
 		TcpConnectionSupport mockConn3 = makeMockConnection("conn3", true);
@@ -353,6 +346,12 @@
 
 		CachingClientConnectionFactory cccf = createCCCFWith2Connections(conn1, conn2);
 		doTestCloseOnSendError(conn1, conn2, cccf);
+	}
+
+	private static AbstractClientConnectionFactory createMockClientConnectionFactory() {
+		AbstractClientConnectionFactory factory = mock(AbstractClientConnectionFactory.class);
+		when(factory.isRunning()).thenReturn(true);
+		return factory;
 	}
 
 	private void doTestCloseOnSendError(TcpConnection conn1, TcpConnection conn2,
