--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
@@ -116,11 +116,10 @@
 
 	@Test
 	public void testLs() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/x/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -323,14 +322,19 @@
 		return files;
 	}
 
+private static SessionFactory createMockSessionFactory(Session session) {
+	SessionFactory sessionFactory = mock(SessionFactory.class);
+	when(sessionFactory.getSession()).thenReturn(session);
+	return sessionFactory;
+}
+
 	@Test
 	public void testLs_f() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-f");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/x/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -366,12 +370,11 @@
 
 	@Test
 	public void testLs_f_R() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-f -R");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] level1 = level1List();
 		TestLsEntry[] level2 = level2List();
 		TestLsEntry[] level3 = level3List();
@@ -392,12 +395,11 @@
 
 	@Test
 	public void testLs_f_R_dirs() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-f -R -dirs");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] level1 = level1List();
 		TestLsEntry[] level2 = level2List();
 		TestLsEntry[] level3 = level3List();
@@ -420,11 +422,10 @@
 
 	@Test
 	public void testLs_None() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = new TestLsEntry[0];
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -436,12 +437,11 @@
 
 	@Test
 	public void testLs_1() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -455,12 +455,11 @@
 
 	@Test
 	public void testLs_1_f() throws Exception { //no sort
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -f");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -473,12 +472,11 @@
 
 	@Test
 	public void testLs_1_dirs() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -dirs");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -492,12 +490,11 @@
 
 	@Test
 	public void testLs_1_dirs_links() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -dirs -links");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -512,12 +509,11 @@
 
 	@Test
 	public void testLs_1_a_f_dirs_links() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -a -f -dirs -links");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -534,13 +530,12 @@
 
 	@Test
 	public void testLs_1_a_f_dirs_links_filtered() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -a -f -dirs -links");
 		gw.setFilter(new TestPatternFilter("*4"));
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -700,6 +695,12 @@
 		assertContents("testfile", outFile);
 
 		outFile.delete();
+	}
+
+	private static SessionFactory createMockSessionFactory(Session session) {
+		SessionFactory sessionFactory = mock(SessionFactory.class);
+		when(sessionFactory.getSession()).thenReturn(session);
+		return sessionFactory;
 	}
 
 	private void assertContents(String expected, File outFile) throws Exception {
@@ -811,11 +812,10 @@
 
 	@Test
 	public void testRm() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "rm", "payload");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		when(session.remove("testremote/x/f1")).thenReturn(Boolean.TRUE);
 		@SuppressWarnings("unchecked")
 		MessageBuilder<Boolean> out = (MessageBuilder<Boolean>) gw
