--- a/spring-integration-core/src/test/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiatorTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiatorTests.java
@@ -165,9 +165,8 @@
 		AtomicBoolean firstLocked = new AtomicBoolean(true);
 
 		// set up first registry instance - this one will be able to obtain lock initially
-		LockRegistry firstRegistry = mock(LockRegistry.class);
 		Lock firstLock = mock(Lock.class);
-		given(firstRegistry.obtain(anyString())).willReturn(firstLock);
+		LockRegistry firstRegistry = createMockLockRegistry(firstLock);
 		given(firstLock.tryLock(anyLong(), any(TimeUnit.class))).willAnswer(i -> firstLocked.get());
 
 		// set up first initiator instance using first LockRegistry
@@ -182,9 +181,8 @@
 		first.setLeaderEventPublisher(new CountingPublisher(firstGranted, firstRevoked, firstAquireLockFailed));
 
 		// set up second registry instance - this one will NOT be able to obtain lock initially
-		LockRegistry secondRegistry = mock(LockRegistry.class);
 		Lock secondLock = mock(Lock.class);
-		given(secondRegistry.obtain(anyString())).willReturn(secondLock);
+		LockRegistry secondRegistry = createMockLockRegistry(secondLock);
 		given(secondLock.tryLock(anyLong(), any(TimeUnit.class))).willAnswer(i -> !firstLocked.get());
 
 		// set up second initiator instance using second LockRegistry
@@ -225,8 +223,6 @@
 	public void testGracefulLeaderSelectorExit() throws Exception {
 		AtomicReference<Throwable> throwableAtomicReference = new AtomicReference<>();
 
-		LockRegistry registry = mock(LockRegistry.class);
-
 		Lock lock = spy(new ReentrantLock());
 
 		willAnswer(invocation -> {
@@ -241,8 +237,7 @@
 				.given(lock)
 				.unlock();
 
-		given(registry.obtain(anyString()))
-				.willReturn(lock);
+		LockRegistry registry = createMockLockRegistry(lock);
 
 		LockRegistryLeaderInitiator another = new LockRegistryLeaderInitiator(registry);
 
@@ -294,6 +289,12 @@
 		another.stop();
 	}
 
+	private static LockRegistry createMockLockRegistry(Lock lock) {
+		LockRegistry registry = mock(LockRegistry.class);
+		given(registry.obtain(anyString())).willReturn(lock);
+		return registry;
+	}
+
 	private static class CountingPublisher implements LeaderEventPublisher {
 
 		private final CountDownLatch granted;
