--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
@@ -360,8 +360,7 @@
 		RuntimeException rte = new RuntimeException("test.immediate");
 		cf.completeExceptionally(rte);
 		given(producer.send(any(), any())).willReturn(cf);
-		ProducerFactory pf = mock();
-		given(pf.createProducer()).willReturn(producer);
+		ProducerFactory pf = createMockProducerFactory(producer);
 		KafkaTemplate template = new KafkaTemplate(pf);
 		template.setDefaultTopic("foo");
 		KafkaProducerMessageHandler handler = new KafkaProducerMessageHandler<>(template);
@@ -746,9 +745,8 @@
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	void testFlush() {
-		ProducerFactory pf = mock();
 		Producer producer = mock();
-		given(pf.createProducer()).willReturn(producer);
+		ProducerFactory pf = createMockProducerFactory(producer);
 		willReturn(mock(Future.class)).given(producer).send(any(ProducerRecord.class), any(Callback.class));
 		KafkaTemplate template = new KafkaTemplate(pf);
 		KafkaProducerMessageHandler handler = new KafkaProducerMessageHandler(template);
@@ -769,9 +767,8 @@
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	void testNoFlush() {
-		ProducerFactory pf = mock();
 		Producer producer = mock();
-		given(pf.createProducer()).willReturn(producer);
+		ProducerFactory pf = createMockProducerFactory(producer);
 		willReturn(mock(Future.class)).given(producer).send(any(ProducerRecord.class), any(Callback.class));
 		KafkaTemplate template = new KafkaTemplate(pf);
 		KafkaProducerMessageHandler handler = new KafkaProducerMessageHandler(template);
@@ -789,9 +786,8 @@
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	void conversion() {
-		ProducerFactory pf = mock();
 		Producer producer = mock();
-		given(pf.createProducer()).willReturn(producer);
+		ProducerFactory pf = createMockProducerFactory(producer);
 		willReturn(mock(Future.class)).given(producer).send(any(ProducerRecord.class), any(Callback.class));
 		KafkaTemplate template = new KafkaTemplate(pf);
 		RecordMessageConverter converter = mock(RecordMessageConverter.class);
@@ -818,6 +814,12 @@
 		handler.stop();
 	}
 
+	private static ProducerFactory createMockProducerFactory(Producer producer) {
+		ProducerFactory pf = mock();
+		given(pf.createProducer()).willReturn(producer);
+		return pf;
+	}
+
 	@SuppressWarnings("serial")
 	static class SomeOtherTransactionManager extends AbstractPlatformTransactionManager {
 
