--- a/spring-integration-ftp/src/test/java/org/springframework/integration/ftp/session/SessionFactoryTests.java
+++ b/spring-integration-ftp/src/test/java/org/springframework/integration/ftp/session/SessionFactoryTests.java
@@ -165,9 +165,8 @@
 	@Test
 	@SuppressWarnings("unchecked")
 	void testSameSessionFromThePool() {
-		SessionFactory<FTPFile> sessionFactory = Mockito.mock(SessionFactory.class);
 		Session<FTPFile> session = Mockito.mock(Session.class);
-		Mockito.when(sessionFactory.getSession()).thenReturn(session);
+		SessionFactory<FTPFile> sessionFactory = createMockSessionFactory(session);
 
 		CachingSessionFactory<FTPFile> cachingFactory = new CachingSessionFactory<>(sessionFactory, 2);
 
@@ -183,9 +182,8 @@
 	@Test
 	@SuppressWarnings("unchecked")
 	void testSessionWaitExpire() {
-		SessionFactory<FTPFile> sessionFactory = Mockito.mock(SessionFactory.class);
 		Session<FTPFile> session = Mockito.mock(Session.class);
-		Mockito.when(sessionFactory.getSession()).thenReturn(session);
+		SessionFactory<FTPFile> sessionFactory = createMockSessionFactory(session);
 
 		CachingSessionFactory<FTPFile> cachingFactory = new CachingSessionFactory<>(sessionFactory, 2);
 
@@ -228,4 +226,11 @@
 		assertThat(failures.get()).isEqualTo(0);
 	}
 
+	@SuppressWarnings("unchecked")
+	private static SessionFactory<FTPFile> createMockSessionFactory(Session<FTPFile> session) {
+		SessionFactory<FTPFile> sessionFactory = Mockito.mock(SessionFactory.class);
+		Mockito.when(sessionFactory.getSession()).thenReturn(session);
+		return sessionFactory;
+	}
+
 }
