--- a/spring-integration-core/src/test/java/org/springframework/integration/support/SmartLifecycleRoleControllerTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/support/SmartLifecycleRoleControllerTests.java
@@ -40,10 +40,9 @@
 
 	@Test
 	public void testOrder() {
-		SmartLifecycle lc1 = mock(SmartLifecycle.class);
+		SmartLifecycle lc1 = createMockSmartLifecycle();
 		when(lc1.getPhase()).thenReturn(2);
-		SmartLifecycle lc2 = mock(SmartLifecycle.class);
-		when(lc2.getPhase()).thenReturn(1);
+		SmartLifecycle lc2 = createMockSmartLifecycle();
 		MultiValueMap<String, SmartLifecycle> map = new LinkedMultiValueMap<>();
 		map.add("foo", lc1);
 		map.add("foo", lc2);
@@ -72,6 +71,12 @@
 					.contains("Cannot add the Lifecycle 'bar' to the role 'foo' " +
 							"because a Lifecycle with the name 'bar' is already present.");
 		}
+	}
+
+	private static SmartLifecycle createMockSmartLifecycle() {
+		SmartLifecycle lifecycle = mock(SmartLifecycle.class);
+		when(lifecycle.getPhase()).thenReturn(1);
+		return lifecycle;
 	}
 
 	private static class PseudoSmartLifecycle implements SmartLifecycle, NamedComponent {
