--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeServerTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeServerTest.java
@@ -89,4 +89,11 @@
         Assertions.assertThrows(RemotingException.class, () -> headerExchangeServer.send("test"));
         Assertions.assertThrows(RemotingException.class, () -> headerExchangeServer.send("test", true));
     }
+
+    private static Channel createMockChannel(Channel exchangeChannel) {
+        Channel channel = Mockito.mock(Channel.class);
+        Mockito.when(channel.getAttribute(HeaderExchangeChannel.class.getName() + ".CHANNEL"))
+                .thenReturn(exchangeChannel);
+        return channel;
+    }
 }
