--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
@@ -96,9 +96,7 @@
         applicationConfig.setName("demo-consumer");
         ApplicationModel.defaultModel().getApplicationConfigManager().setApplication(applicationConfig);
 
-        URL consumerURL = Mockito.mock(URL.class);
-        Mockito.when(consumerURL.getServiceKey()).thenReturn("Test");
-        Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
+        URL consumerURL = createMockConsumerUrl();
 
         System.setProperty("dubbo.application.migration.delay", "1");
         MigrationRuleHandler<?> handler =
@@ -142,9 +140,7 @@
         applicationConfig.setName("demo-consumer");
         ApplicationModel.defaultModel().getApplicationConfigManager().setApplication(applicationConfig);
 
-        URL consumerURL = Mockito.mock(URL.class);
-        Mockito.when(consumerURL.getServiceKey()).thenReturn("Test");
-        Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
+        URL consumerURL = createMockConsumerUrl();
 
         System.setProperty("dubbo.application.migration.delay", "1000");
         MigrationRuleHandler<?> handler =
@@ -183,11 +179,9 @@
         applicationConfig.setName("demo-consumer");
         ApplicationModel.defaultModel().getApplicationConfigManager().setApplication(applicationConfig);
 
-        URL consumerURL = Mockito.mock(URL.class);
-        Mockito.when(consumerURL.getServiceKey()).thenReturn("Test");
-        Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
-
-        URL consumerURL2 = Mockito.mock(URL.class);
+        URL consumerURL = createMockConsumerUrl();
+
+        URL consumerURL2 = createMockConsumerUrl();
         Mockito.when(consumerURL2.getServiceKey()).thenReturn("Test2");
         Mockito.when(consumerURL2.getParameter("timestamp")).thenReturn("2");
 
@@ -243,4 +237,11 @@
         Assertions.assertEquals(
                 "APPLICATION_FIRST", captor_event.getValue().getStep().toString());
     }
+
+    private static URL createMockConsumerUrl() {
+        URL consumerURL = Mockito.mock(URL.class);
+        Mockito.when(consumerURL.getServiceKey()).thenReturn("Test");
+        Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
+        return consumerURL;
+    }
 }
