--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
@@ -84,10 +84,9 @@
     @Test
     void testGetLogger() {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            Logger failLogger = mock(Logger.class);
+            Logger failLogger = MockLoggerWithErrorFailure.createMockLoggerWithErrorFailure();
             FailsafeErrorTypeAwareLogger failsafeLogger = new FailsafeErrorTypeAwareLogger(failLogger);
 
-            doThrow(new RuntimeException()).when(failLogger).error(anyString());
             failsafeLogger.getLogger().error("should get error");
         });
     }
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
@@ -103,10 +103,9 @@
     @Test
     void testGetLogger() {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            Logger failLogger = mock(Logger.class);
+            Logger failLogger = MockLoggerWithErrorFailure.createMockLoggerWithErrorFailure();
             FailsafeLogger failsafeLogger = new FailsafeLogger(failLogger);
 
-            doThrow(new RuntimeException()).when(failLogger).error(anyString());
             failsafeLogger.getLogger().error("should get error");
         });
     }
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/MockLoggerWithErrorFailure.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/MockLoggerWithErrorFailure.java
@@ -0,0 +1,16 @@
+package org.apache.dubbo.common.logger.support;
+
+import org.apache.dubbo.common.logger.Logger;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+public class MockLoggerWithErrorFailure {
+
+    public static Logger createMockLoggerWithErrorFailure() {
+        Logger failLogger = mock(Logger.class);
+        doThrow(new RuntimeException()).when(failLogger).error(anyString());
+        return failLogger;
+    }
+}
