--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvokerTest.java
@@ -195,10 +195,7 @@
         given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
         given(firstInvoker.isAvailable()).willReturn(true);
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
-        given(secondInvoker.isAvailable()).willReturn(true);
+        secondInvoker = createMockSecondInvoker(url, invocation);
 
         given(directory.list(invocation)).willReturn(new ArrayList() {
 
@@ -307,10 +304,7 @@
         given(firstInvoker.invoke(invocation))
                 .willThrow(new RpcException(RpcException.NO_INVOKER_AVAILABLE_AFTER_FILTER));
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
-        given(secondInvoker.isAvailable()).willReturn(true);
+        secondInvoker = createMockSecondInvoker(url, invocation);
         given(secondInvoker.invoke(invocation))
                 .willThrow(new RpcException(RpcException.NO_INVOKER_AVAILABLE_AFTER_FILTER));
 
@@ -363,10 +357,7 @@
         given(firstInvoker.isAvailable()).willReturn(true);
         given(firstInvoker.invoke(invocation)).willThrow(new RpcException(RpcException.NETWORK_EXCEPTION));
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
-        given(secondInvoker.isAvailable()).willReturn(true);
+        secondInvoker = createMockSecondInvoker(url, invocation);
         given(secondInvoker.invoke(invocation)).willThrow(new RpcException(RpcException.NETWORK_EXCEPTION));
 
         given(directory.list(invocation)).willReturn(new ArrayList() {
@@ -410,10 +401,7 @@
         given(firstInvoker.invoke(invocation)).willReturn(new AppResponse());
         given(firstInvoker.isAvailable()).willReturn(true);
 
-        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
-        given(secondInvoker.getInterface()).willReturn(MenuService.class);
-        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
-        given(secondInvoker.isAvailable()).willReturn(true);
+        secondInvoker = createMockSecondInvoker(url, invocation);
 
         given(directory.list(invocation)).willReturn(new ArrayList() {
 
@@ -538,4 +526,13 @@
         assertFalse(firstInvoker.isAvailable());
         assertFalse(secondInvoker.isAvailable());
     }
+
+    private static Invoker createMockSecondInvoker(URL url, Invocation invocation) {
+        Invoker secondInvoker = mock(Invoker.class);
+        given(secondInvoker.getUrl()).willReturn(url.addParameter(GROUP_KEY, "second"));
+        given(secondInvoker.getInterface()).willReturn(MenuService.class);
+        given(secondInvoker.invoke(invocation)).willReturn(new AppResponse());
+        given(secondInvoker.isAvailable()).willReturn(true);
+        return secondInvoker;
+    }
 }
