--- a/extensions-core/azure-extensions/src/test/java/org/apache/druid/storage/azure/AzureStorageTest.java
+++ b/extensions-core/azure-extensions/src/test/java/org/apache/druid/storage/azure/AzureStorageTest.java
@@ -230,21 +230,18 @@
   public void testBatchDeleteFiles_emptyResponse() throws BlobStorageException
   {
     String containerUrl = "https://storageaccount.blob.core.windows.net/container";
-    BlobBatchClient blobBatchClient = Mockito.mock(BlobBatchClient.class);
 
     SettableSupplier<PagedResponse<BlobItem>> supplier = new SettableSupplier<>();
     supplier.set(new TestPagedResponse<>(ImmutableList.of()));
     PagedIterable<BlobItem> pagedIterable = new PagedIterable<>(supplier);
 
     ArgumentCaptor<List<String>> captor = ArgumentCaptor.forClass(List.class);
+    final BlobBatchClient blobBatchClient = createMockBlobBatchClient(pagedIterable, captor);
 
     Mockito.doReturn(containerUrl).when(blobContainerClient).getBlobContainerUrl();
     Mockito.doReturn(blobContainerClient).when(blobServiceClient).getBlobContainerClient(CONTAINER);
     Mockito.doReturn(blobServiceClient).when(azureClientFactory).getBlobServiceClient(null, STORAGE_ACCOUNT);
     Mockito.doReturn(blobBatchClient).when(azureClientFactory).getBlobBatchClient(blobContainerClient);
-    Mockito.doReturn(pagedIterable).when(blobBatchClient).deleteBlobs(
-        captor.capture(), ArgumentMatchers.eq(DeleteSnapshotsOptionType.INCLUDE)
-    );
 
     boolean deleteSuccessful = azureStorage.batchDeleteFiles(CONTAINER, ImmutableList.of(BLOB_NAME), null);
     assertEquals(captor.getValue().get(0), containerUrl + "/" + BLOB_NAME);
@@ -279,21 +276,18 @@
   public void testBatchDeleteFiles_emptyResponse_multipleResponses() throws BlobStorageException
   {
     String containerUrl = "https://storageaccount.blob.core.windows.net/container";
-    BlobBatchClient blobBatchClient = Mockito.mock(BlobBatchClient.class);
 
     SettableSupplier<PagedResponse<BlobItem>> supplier = new SettableSupplier<>();
     supplier.set(new TestPagedResponse<>(ImmutableList.of()));
     PagedIterable<BlobItem> pagedIterable = new PagedIterable<>(supplier);
 
     ArgumentCaptor<List<String>> captor = ArgumentCaptor.forClass(List.class);
+    final BlobBatchClient blobBatchClient = createMockBlobBatchClient(pagedIterable, captor);
 
     Mockito.doReturn(containerUrl).when(blobContainerClient).getBlobContainerUrl();
     Mockito.doReturn(blobContainerClient).when(blobServiceClient).getBlobContainerClient(CONTAINER);
     Mockito.doReturn(blobServiceClient).when(azureClientFactory).getBlobServiceClient(null, STORAGE_ACCOUNT);
     Mockito.doReturn(blobBatchClient).when(azureClientFactory).getBlobBatchClient(blobContainerClient);
-    Mockito.doReturn(pagedIterable).when(blobBatchClient).deleteBlobs(
-        captor.capture(), ArgumentMatchers.eq(DeleteSnapshotsOptionType.INCLUDE)
-    );
 
 
     List<String> blobNameList = new ArrayList<>();
@@ -328,5 +322,17 @@
     Mockito.verify(blobClient).upload(captor.capture(), captor2.capture(), overrideArgument.capture());
     assertTrue(overrideArgument.getValue());
   }
+
+  private static BlobBatchClient createMockBlobBatchClient(
+      PagedIterable<BlobItem> pagedIterable,
+      ArgumentCaptor<List<String>> captor
+  )
+  {
+    final BlobBatchClient blobBatchClient = Mockito.mock(BlobBatchClient.class);
+    Mockito.doReturn(pagedIterable).when(blobBatchClient).deleteBlobs(
+        captor.capture(), ArgumentMatchers.eq(DeleteSnapshotsOptionType.INCLUDE)
+    );
+    return blobBatchClient;
+  }
 }
 
