--- a/sql/src/test/java/org/apache/druid/sql/avatica/ErrorHandlerTest.java
+++ b/sql/src/test/java/org/apache/druid/sql/avatica/ErrorHandlerTest.java
@@ -34,12 +34,7 @@
   @Test
   public void testErrorHandlerSanitizesErrorAsExpected()
   {
-    ServerConfig serverConfig = Mockito.mock(ServerConfig.class);
-    AllowedRegexErrorResponseTransformStrategy emptyAllowedRegexErrorResponseTransformStrategy = new AllowedRegexErrorResponseTransformStrategy(
-        ImmutableList.of());
-
-    Mockito.when(serverConfig.getErrorResponseTransformStrategy())
-           .thenReturn(emptyAllowedRegexErrorResponseTransformStrategy);
+    final ServerConfig serverConfig = createMockServerConfigWithEmptyAllowedRegexErrorResponseTransformStrategy();
     ErrorHandler errorHandler = new ErrorHandler(serverConfig);
     QueryException input = new QueryException("error", "error message", "error class", "host");
 
@@ -61,12 +56,7 @@
   @Test
   public void testErrorHandlerHasAffectingErrorResponseTransformStrategyReturnsTrueWhenNotUsingNoErrorResponseTransformStrategy()
   {
-    ServerConfig serverConfig = Mockito.mock(ServerConfig.class);
-    AllowedRegexErrorResponseTransformStrategy emptyAllowedRegexErrorResponseTransformStrategy = new AllowedRegexErrorResponseTransformStrategy(
-        ImmutableList.of());
-
-    Mockito.when(serverConfig.getErrorResponseTransformStrategy())
-           .thenReturn(emptyAllowedRegexErrorResponseTransformStrategy);
+    final ServerConfig serverConfig = createMockServerConfigWithEmptyAllowedRegexErrorResponseTransformStrategy();
     ErrorHandler errorHandler = new ErrorHandler(serverConfig);
     Assert.assertTrue(errorHandler.hasAffectingErrorResponseTransformStrategy());
   }
@@ -82,16 +72,19 @@
   @Test
   public void testErrorHandlerHandlesNonSanitizableExceptionCorrectly()
   {
-    ServerConfig serverConfig = Mockito.mock(ServerConfig.class);
-    AllowedRegexErrorResponseTransformStrategy emptyAllowedRegexErrorResponseTransformStrategy = new AllowedRegexErrorResponseTransformStrategy(
-        ImmutableList.of());
-
-    Mockito.when(serverConfig.getErrorResponseTransformStrategy())
-           .thenReturn(emptyAllowedRegexErrorResponseTransformStrategy);
+    final ServerConfig serverConfig = createMockServerConfigWithEmptyAllowedRegexErrorResponseTransformStrategy();
     ErrorHandler errorHandler = new ErrorHandler(serverConfig);
 
     Exception input = new Exception("message");
     RuntimeException output = errorHandler.sanitize(input);
     Assert.assertEquals(null, output.getMessage());
   }
+
+  private static ServerConfig createMockServerConfigWithEmptyAllowedRegexErrorResponseTransformStrategy()
+  {
+    final ServerConfig serverConfig = Mockito.mock(ServerConfig.class);
+    Mockito.when(serverConfig.getErrorResponseTransformStrategy())
+           .thenReturn(new AllowedRegexErrorResponseTransformStrategy(ImmutableList.of()));
+    return serverConfig;
+  }
 }
