--- a/indexing-service/src/test/java/org/apache/druid/indexing/overlord/duty/OverlordDutyExecutorTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/overlord/duty/OverlordDutyExecutorTest.java
@@ -35,13 +35,8 @@
   @Test
   public void testStartAndStop()
   {
-    OverlordDuty testDuty1 = Mockito.mock(OverlordDuty.class);
-    Mockito.when(testDuty1.isEnabled()).thenReturn(true);
-    Mockito.when(testDuty1.getSchedule()).thenReturn(new DutySchedule(1, 0));
-
-    OverlordDuty testDuty2 = Mockito.mock(OverlordDuty.class);
-    Mockito.when(testDuty2.isEnabled()).thenReturn(true);
-    Mockito.when(testDuty2.getSchedule()).thenReturn(new DutySchedule(1, 0));
+    final OverlordDuty testDuty1 = createMockOverlordDuty();
+    final OverlordDuty testDuty2 = createMockOverlordDuty();
 
     ScheduledExecutorFactory executorFactory = Mockito.mock(ScheduledExecutorFactory.class);
     ScheduledExecutorService executorService = Mockito.mock(ScheduledExecutorService.class);
@@ -74,6 +69,14 @@
     Mockito.verify(executorService, Mockito.times(1)).shutdownNow();
   }
 
+  private static OverlordDuty createMockOverlordDuty()
+  {
+    final OverlordDuty testDuty = Mockito.mock(OverlordDuty.class);
+    Mockito.when(testDuty.isEnabled()).thenReturn(true);
+    Mockito.when(testDuty.getSchedule()).thenReturn(new DutySchedule(1, 0));
+    return testDuty;
+  }
+
   @Test
   public void testStartWithNoEnabledDuty()
   {
