--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -32,12 +32,13 @@
 public class RunWorkOrderTest
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
+  private final WorkOrder sharedWorkOrder = Mockito.mock(WorkOrder.class);
 
   @Test
   public void test_stop()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
+    final WorkOrder workOrder = sharedWorkOrder;
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -73,7 +74,7 @@
   public void test_stop_error()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
+    final WorkOrder workOrder = sharedWorkOrder;
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -114,7 +115,7 @@
   public void test_stop_errorDuringExecCancel()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
+    final WorkOrder workOrder = sharedWorkOrder;
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -152,7 +153,7 @@
   public void test_stop_errorDuringFrameContextClose()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
+    final WorkOrder workOrder = sharedWorkOrder;
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
