--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateStatusActionTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateStatusActionTest.java
@@ -26,6 +26,7 @@
 import org.apache.druid.indexing.common.task.Task;
 import org.apache.druid.indexing.overlord.TaskRunner;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 import static org.mockito.ArgumentMatchers.any;
@@ -41,16 +42,23 @@
 
   private static final String ID = "id";
 
+  private TaskRunner taskRunner;
+
+  @Before
+  public void setUp()
+  {
+    taskRunner = mock(TaskRunner.class);
+  }
+
   @Test
   public void testActionCallsTaskRunner()
   {
     UpdateStatusAction action = new UpdateStatusAction("successful");
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
-    when(toolbox.getTaskRunner()).thenReturn(Optional.of(runner));
+    when(toolbox.getTaskRunner()).thenReturn(Optional.of(taskRunner));
     action.perform(task, toolbox);
-    verify(runner, times(1)).updateStatus(eq(task), eq(TaskStatus.success(task.getId())));
+    verify(taskRunner, times(1)).updateStatus(eq(task), eq(TaskStatus.success(task.getId())));
   }
 
   @Test
@@ -59,10 +67,9 @@
     UpdateStatusAction action = new UpdateStatusAction("failure");
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
-    when(toolbox.getTaskRunner()).thenReturn(Optional.of(runner));
+    when(toolbox.getTaskRunner()).thenReturn(Optional.of(taskRunner));
     action.perform(task, toolbox);
-    verify(runner, times(1)).updateStatus(eq(task), eq(TaskStatus.failure(task.getId(), "Error with task")));
+    verify(taskRunner, times(1)).updateStatus(eq(task), eq(TaskStatus.failure(task.getId(), "Error with task")));
   }
 
   @Test
@@ -70,12 +77,11 @@
   {
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
-    when(toolbox.getTaskRunner()).thenReturn(Optional.of(runner));
+    when(toolbox.getTaskRunner()).thenReturn(Optional.of(taskRunner));
     TaskStatus taskStatus = TaskStatus.failure(task.getId(), "custom error message");
     UpdateStatusAction action = new UpdateStatusAction("failure", taskStatus);
     action.perform(task, toolbox);
-    verify(runner, times(1)).updateStatus(eq(task), eq(taskStatus));
+    verify(taskRunner, times(1)).updateStatus(eq(task), eq(taskStatus));
   }
 
   @Test
@@ -84,10 +90,9 @@
     UpdateStatusAction action = new UpdateStatusAction("", TaskStatus.success(ID));
     Task task = NoopTask.create();
     TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
-    TaskRunner runner = mock(TaskRunner.class);
     when(toolbox.getTaskRunner()).thenReturn(Optional.absent());
     action.perform(task, toolbox);
-    verify(runner, never()).updateStatus(any(), any());
+    verify(taskRunner, never()).updateStatus(any(), any());
   }
 
   @Test
