--- a/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
@@ -153,8 +153,7 @@
 
     Monitor monitor = Mockito.mock(Monitor.class);
 
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
 
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
@@ -217,8 +216,7 @@
 
     Monitor monitor = Mockito.mock(Monitor.class);
 
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
 
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
@@ -248,8 +246,7 @@
     Mockito.when(monitor.monitor(ArgumentMatchers.any(ServiceEmitter.class)))
            .thenThrow(new RuntimeException("Test throwing exception while monitoring"));
 
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
 
     CountDownLatch latch = new CountDownLatch(1);
     AtomicBoolean monitorResultHolder = new AtomicBoolean(false);
@@ -310,8 +307,7 @@
   {
     ExecutorService executor = Mockito.mock(ExecutorService.class);
     Monitor monitor = Mockito.mock(Monitor.class);
-    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
-    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
+    DruidMonitorSchedulerConfig config = createMockDruidMonitorSchedulerConfig();
 
     CountDownLatch latch = new CountDownLatch(1);
     Mockito.doAnswer(new Answer<Future<?>>()
@@ -393,6 +389,13 @@
 
     return future;
   }
+
+  private static DruidMonitorSchedulerConfig createMockDruidMonitorSchedulerConfig()
+  {
+    DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
+    Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
+    return config;
+  }
   
   
   private static class NoopMonitor implements Monitor
