--- a/sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java
+++ b/sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java
@@ -2721,8 +2721,7 @@
   public void testPresenceOfOverIsInvalid()
   {
     final RexBuilder rexBuilder = new RexBuilder(DruidTypeSystem.TYPE_FACTORY);
-    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
-    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+    final PlannerContext plannerContext = createMockPlannerContext();
 
     RexNode rexNode = rexBuilder.makeOver(
         testHelper.createSqlType(SqlTypeName.BIGINT),
@@ -2751,8 +2750,7 @@
   public void testCalciteLiteralToDruidLiteral()
   {
     final RexBuilder rexBuilder = new RexBuilder(DruidTypeSystem.TYPE_FACTORY);
-    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
-    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+    final PlannerContext plannerContext = createMockPlannerContext();
 
     assertDruidLiteral(
         new DruidLiteral(ExpressionType.STRING, null),
@@ -2915,4 +2913,11 @@
         StringUtils.format("%s: %s", actual.type(), actual.value())
     );
   }
+
+  private static PlannerContext createMockPlannerContext()
+  {
+    final PlannerContext plannerContext = Mockito.mock(PlannerContext.class);
+    Mockito.when(plannerContext.getTimeZone()).thenReturn(DateTimeZone.UTC);
+    return plannerContext;
+  }
 }
