--- a/extensions-contrib/cloudfiles-extensions/src/test/java/org/apache/druid/storage/cloudfiles/CloudFilesByteSourceTest.java
+++ b/extensions-contrib/cloudfiles-extensions/src/test/java/org/apache/druid/storage/cloudfiles/CloudFilesByteSourceTest.java
@@ -39,12 +39,11 @@
     final String path = "path";
 
     CloudFilesObjectApiProxy objectApi = mock(CloudFilesObjectApiProxy.class);
-    CloudFilesObject cloudFilesObject = mock(CloudFilesObject.class);
     Payload payload = mock(Payload.class);
+    CloudFilesObject cloudFilesObject = createMockCloudFilesObject(payload);
     InputStream stream = mock(InputStream.class);
 
     when(objectApi.get(path, 0)).thenReturn(cloudFilesObject);
-    when(cloudFilesObject.getPayload()).thenReturn(payload);
     when(payload.openStream()).thenReturn(stream);
     payload.close();
 
@@ -63,12 +62,11 @@
     final String path = "path";
 
     CloudFilesObjectApiProxy objectApi = mock(CloudFilesObjectApiProxy.class);
-    CloudFilesObject cloudFilesObject = mock(CloudFilesObject.class);
     Payload payload = mock(Payload.class);
+    CloudFilesObject cloudFilesObject = createMockCloudFilesObject(payload);
     InputStream stream = mock(InputStream.class);
 
     when(objectApi.get(path, 0)).thenReturn(cloudFilesObject);
-    when(cloudFilesObject.getPayload()).thenReturn(payload);
     when(payload.openStream()).thenThrow(new IOException())
                               .thenReturn(stream);
     payload.close();
@@ -88,4 +86,11 @@
     verify(cloudFilesObject).getPayload();
     verify(payload, times(2)).openStream();
   }
+
+  private static CloudFilesObject createMockCloudFilesObject(final Payload payload)
+  {
+    final CloudFilesObject cloudFilesObject = mock(CloudFilesObject.class);
+    when(cloudFilesObject.getPayload()).thenReturn(payload);
+    return cloudFilesObject;
+  }
 }
