--- a/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunnerTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunnerTest.java
@@ -122,18 +122,18 @@
                   .build();
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
-    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
+    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = createMockSeekableStreamIndexTaskIOConfig(
+        sequenceNumbers,
+        endSequenceNumbers
+    );
 
     DateTime now = DateTimes.nowUtc();
 
     Mockito.when(ioConfig.getRefreshRejectionPeriodsInMinutes()).thenReturn(120L);
     Mockito.when(ioConfig.getMaximumMessageTime()).thenReturn(DateTimes.nowUtc().plusHours(2));
     Mockito.when(ioConfig.getMinimumMessageTime()).thenReturn(DateTimes.nowUtc().minusHours(2));
-    Mockito.when(ioConfig.getInputFormat()).thenReturn(new JsonInputFormat(null, null, null, null, null));
-    Mockito.when(ioConfig.getStartSequenceNumbers()).thenReturn(sequenceNumbers);
-    Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
     Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
@@ -177,9 +177,12 @@
                   .build();
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
-    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
+    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = createMockSeekableStreamIndexTaskIOConfig(
+        sequenceNumbers,
+        endSequenceNumbers
+    );
 
     DateTime now = DateTimes.nowUtc();
 
@@ -187,9 +190,6 @@
     // min max time not populated.
     Mockito.when(ioConfig.getMaximumMessageTime()).thenReturn(null);
     Mockito.when(ioConfig.getMinimumMessageTime()).thenReturn(null);
-    Mockito.when(ioConfig.getInputFormat()).thenReturn(new JsonInputFormat(null, null, null, null, null));
-    Mockito.when(ioConfig.getStartSequenceNumbers()).thenReturn(sequenceNumbers);
-    Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
     Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
@@ -232,16 +232,16 @@
                   .build();
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
-    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
+    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = createMockSeekableStreamIndexTaskIOConfig(
+        sequenceNumbers,
+        endSequenceNumbers
+    );
 
     Mockito.when(ioConfig.getRefreshRejectionPeriodsInMinutes()).thenReturn(null);
     Mockito.when(ioConfig.getMaximumMessageTime()).thenReturn(null);
     Mockito.when(ioConfig.getMinimumMessageTime()).thenReturn(null);
-    Mockito.when(ioConfig.getInputFormat()).thenReturn(new JsonInputFormat(null, null, null, null, null));
-    Mockito.when(ioConfig.getStartSequenceNumbers()).thenReturn(sequenceNumbers);
-    Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
     Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
@@ -278,14 +278,14 @@
                   .build();
 
     SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
-    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
+    SeekableStreamIndexTaskIOConfig<String, String> ioConfig = createMockSeekableStreamIndexTaskIOConfig(
+        sequenceNumbers,
+        endSequenceNumbers
+    );
 
     Mockito.when(ioConfig.getRefreshRejectionPeriodsInMinutes()).thenReturn(null);
-    Mockito.when(ioConfig.getInputFormat()).thenReturn(new JsonInputFormat(null, null, null, null, null));
-    Mockito.when(ioConfig.getStartSequenceNumbers()).thenReturn(sequenceNumbers);
-    Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
 
     Mockito.when(endSequenceNumbers.getPartitionSequenceNumberMap()).thenReturn(ImmutableMap.of());
     Mockito.when(sequenceNumbers.getStream()).thenReturn("test");
@@ -335,6 +335,18 @@
     runner.run(createTaskToolbox());
     emitter.verifyValue("ingest/segments/count", 10);
     emitter.verifyValue("ingest/rows/published", 10_000L);
+  }
+
+  private static SeekableStreamIndexTaskIOConfig<String, String> createMockSeekableStreamIndexTaskIOConfig(
+      final SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers,
+      final SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers
+  )
+  {
+    final SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
+    Mockito.when(ioConfig.getInputFormat()).thenReturn(new JsonInputFormat(null, null, null, null, null));
+    Mockito.when(ioConfig.getStartSequenceNumbers()).thenReturn(sequenceNumbers);
+    Mockito.when(ioConfig.getEndSequenceNumbers()).thenReturn(endSequenceNumbers);
+    return ioConfig;
   }
 
   private TaskToolbox createTaskToolbox()
