--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
@@ -52,8 +52,7 @@
   @Test
   public void testArchiveSegmentUnknowType()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+    final DataSegment segment = MockDataSegmentForUnknownType.createMockDataSegmentForUnknownType();
 
     final Injector injector = createInjector(null);
     final OmniDataSegmentArchiver segmentArchiver = injector.getInstance(OmniDataSegmentArchiver.class);
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentKillerTest.java
@@ -63,8 +63,7 @@
   @Test
   public void testKillSegmentUnknowType()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+    final DataSegment segment = MockDataSegmentForUnknownType.createMockDataSegmentForUnknownType();
 
     final Injector injector = createInjector(null);
     final OmniDataSegmentKiller segmentKiller = injector.getInstance(OmniDataSegmentKiller.class);
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
@@ -54,8 +54,7 @@
   @Test
   public void testMoveSegmentUnknownType()
   {
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+    final DataSegment segment = MockDataSegmentForUnknownType.createMockDataSegmentForUnknownType();
 
     final Injector injector = createInjector(null);
     final OmniDataSegmentMover segmentMover = injector.getInstance(OmniDataSegmentMover.class);
--- a/server/src/test/java/org/apache/druid/segment/loading/MockDataSegmentForUnknownType.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/MockDataSegmentForUnknownType.java
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.loading;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.timeline.DataSegment;
+import org.mockito.Mockito;
+
+public class MockDataSegmentForUnknownType
+{
+  public static DataSegment createMockDataSegmentForUnknownType()
+  {
+    final DataSegment segment = Mockito.mock(DataSegment.class);
+    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "unknown-type"));
+    return segment;
+  }
+}
