--- a/multi-stage-query/src/test/java/org/apache/druid/msq/dart/controller/http/DartSqlResourceTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/dart/controller/http/DartSqlResourceTest.java
@@ -328,8 +328,7 @@
   @Test
   public void test_getRunningQueries_selfOnly_superUser()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(CalciteTests.TEST_SUPERUSER_NAME));
+    httpServletRequest = createMockHttpServletRequest(CalciteTests.TEST_SUPERUSER_NAME);
 
     final ControllerHolder holder = setUpMockRunningQuery(REGULAR_USER_NAME);
 
@@ -348,8 +347,7 @@
   @Test
   public void test_getRunningQueries_selfOnly_regularUser()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
 
     final ControllerHolder holder = setUpMockRunningQuery(REGULAR_USER_NAME);
     final ControllerHolder holder2 = setUpMockRunningQuery(DIFFERENT_REGULAR_USER_NAME);
@@ -372,8 +370,7 @@
   @Test
   public void test_getRunningQueries_global_superUser()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(CalciteTests.TEST_SUPERUSER_NAME));
+    httpServletRequest = createMockHttpServletRequest(CalciteTests.TEST_SUPERUSER_NAME);
 
     // REGULAR_USER_NAME runs a query locally.
     final ControllerHolder localHolder = setUpMockRunningQuery(REGULAR_USER_NAME);
@@ -414,8 +411,7 @@
   @Test
   public void test_getRunningQueries_global_remoteError_superUser()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(CalciteTests.TEST_SUPERUSER_NAME));
+    httpServletRequest = createMockHttpServletRequest(CalciteTests.TEST_SUPERUSER_NAME);
 
     // REGULAR_USER_NAME runs a query locally.
     final ControllerHolder localHolder = setUpMockRunningQuery(REGULAR_USER_NAME);
@@ -441,8 +437,7 @@
   @Test
   public void test_getRunningQueries_global_regularUser()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
 
     // REGULAR_USER_NAME runs a query locally.
     final ControllerHolder localHolder = setUpMockRunningQuery(REGULAR_USER_NAME);
@@ -479,8 +474,7 @@
   @Test
   public void test_getRunningQueries_global_differentRegularUser()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(DIFFERENT_REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(DIFFERENT_REGULAR_USER_NAME);
 
     // REGULAR_USER_NAME runs a query locally.
     final ControllerHolder holder = setUpMockRunningQuery(REGULAR_USER_NAME);
@@ -516,8 +510,7 @@
     final MockHttpServletResponse asyncResponse = new MockHttpServletResponse();
     asyncContext.response = asyncResponse;
 
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
@@ -543,8 +536,7 @@
     final MockHttpServletResponse asyncResponse = new MockHttpServletResponse();
     asyncContext.response = asyncResponse;
 
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
@@ -571,8 +563,7 @@
     final MockHttpServletResponse asyncResponse = new MockHttpServletResponse();
     asyncContext.response = asyncResponse;
 
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
@@ -606,8 +597,7 @@
     final MockHttpServletResponse asyncResponse = new MockHttpServletResponse();
     asyncContext.response = asyncResponse;
 
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
@@ -647,8 +637,7 @@
     final MockHttpServletResponse asyncResponse = new MockHttpServletResponse();
     asyncContext.response = asyncResponse;
 
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
@@ -692,8 +681,7 @@
     final MockHttpServletResponse asyncResponse = new MockHttpServletResponse();
     asyncContext.response = asyncResponse;
 
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
@@ -749,14 +737,12 @@
     asyncContext.response = asyncResponse;
 
     // POST SQL query request.
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
     Mockito.when(httpServletRequest.startAsync())
            .thenReturn(asyncContext);
 
     // Cancellation request.
-    Mockito.when(httpServletRequest2.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    httpServletRequest2 = createMockHttpServletRequest(REGULAR_USER_NAME);
 
     // Block up the controllerExecutor so the controller runs long enough to cancel it.
     final Future<?> sleepFuture = controllerThreadPool.getRunExecutorService().submit(() -> {
@@ -850,8 +836,7 @@
   @Test
   public void test_cancelQuery_regularUser_unknownQuery()
   {
-    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
-           .thenReturn(makeAuthenticationResult(REGULAR_USER_NAME));
+    final HttpServletRequest httpServletRequest = createMockHttpServletRequest(REGULAR_USER_NAME);
 
     final Response cancellationResponse = sqlResource.cancelQuery("nonexistent", httpServletRequest);
     Assertions.assertEquals(Response.Status.NOT_FOUND.getStatusCode(), cancellationResponse.getStatus());
@@ -892,4 +877,16 @@
   {
     return new AuthenticationResult(identity, null, AUTHENTICATOR_NAME, Collections.emptyMap());
   }
+
+  /**
+   * Create a mock {@link HttpServletRequest} whose {@link AuthConfig#DRUID_AUTHENTICATION_RESULT} attribute is
+   * the authentication result for the given identity.
+   */
+  private static HttpServletRequest createMockHttpServletRequest(final String identity)
+  {
+    final HttpServletRequest httpServletRequest = Mockito.mock(HttpServletRequest.class);
+    Mockito.when(httpServletRequest.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
+           .thenReturn(makeAuthenticationResult(identity));
+    return httpServletRequest;
+  }
 }
