--- a/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
+++ b/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
@@ -33,6 +33,7 @@
 import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.eclipse.jetty.util.thread.QueuedThreadPool;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
@@ -41,6 +42,14 @@
 
 public class JettyServerModuleTest
 {
+  private JettyServerInitializer jettyServerInitializer;
+
+  @Before
+  public void setUp()
+  {
+    jettyServerInitializer = Mockito.mock(JettyServerInitializer.class);
+  }
+
   @Test
   public void testJettyServerModule()
   {
@@ -101,8 +110,7 @@
     Injector injector = Mockito.mock(Injector.class);
     TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
-    JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
-    Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
+    Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(jettyServerInitializer);
 
     Server server = JettyServerModule.makeAndInitializeServer(
         injector,
@@ -143,8 +151,7 @@
     Injector injector = Mockito.mock(Injector.class);
     TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
-    JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
-    Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
+    Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(jettyServerInitializer);
 
     // Create a custom SslContextFactory via binding
     SslContextFactory.Server customSslContextFactory = Mockito.mock(SslContextFactory.Server.class);
@@ -207,8 +214,7 @@
     Injector injector = Mockito.mock(Injector.class);
     TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
-    JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
-    Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
+    Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(jettyServerInitializer);
 
     // Create a custom SslContextFactory via binding
     SslContextFactory.Server customSslContextFactory = Mockito.mock(SslContextFactory.Server.class);
