--- a/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
+++ b/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
@@ -151,11 +151,9 @@
 
     @SuppressWarnings("unchecked")
     Binding<SslContextFactory.Server> sslContextFactoryBinding = Mockito.mock(Binding.class);
-    @SuppressWarnings("unchecked")
-    Provider<SslContextFactory.Server> provider = Mockito.mock(Provider.class);
+    Provider<SslContextFactory.Server> provider = createMockProvider(customSslContextFactory);
 
     Mockito.when(sslContextFactoryBinding.getProvider()).thenReturn(provider);
-    Mockito.when(provider.get()).thenReturn(customSslContextFactory);
 
     Server server = JettyServerModule.makeAndInitializeServer(
         injector,
@@ -215,11 +213,9 @@
 
     @SuppressWarnings("unchecked")
     Binding<SslContextFactory.Server> sslContextFactoryBinding = Mockito.mock(Binding.class);
-    @SuppressWarnings("unchecked")
-    Provider<SslContextFactory.Server> provider = Mockito.mock(Provider.class);
+    Provider<SslContextFactory.Server> provider = createMockProvider(customSslContextFactory);
 
     Mockito.when(sslContextFactoryBinding.getProvider()).thenReturn(provider);
-    Mockito.when(provider.get()).thenReturn(customSslContextFactory);
 
     Server server = JettyServerModule.makeAndInitializeServer(
         injector,
@@ -244,4 +240,12 @@
     Mockito.verify(customSslContextFactory, Mockito.never()).setExcludeCipherSuites("TLS_RSA_WITH_NULL_SHA256");
     Mockito.verify(customSslContextFactory, Mockito.never()).setExcludeProtocols("TLSv1", "TLSv1.1");
   }
+
+  private static Provider<SslContextFactory.Server> createMockProvider(SslContextFactory.Server customSslContextFactory)
+  {
+    @SuppressWarnings("unchecked")
+    Provider<SslContextFactory.Server> provider = Mockito.mock(Provider.class);
+    Mockito.when(provider.get()).thenReturn(customSslContextFactory);
+    return provider;
+  }
 }
