--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/ForeignHostPermitHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/ForeignHostPermitHandlerTest.java
@@ -44,13 +44,12 @@
     void shouldShowIpNotPermittedMsg_GivenAcceptForeignIpFalseAndEmptyWhiteList() throws Exception {
         ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         Channel channel = mock(Channel.class);
-        when(context.channel()).thenReturn(channel);
         InetAddress addr = mock(InetAddress.class);
         when(addr.isLoopbackAddress()).thenReturn(false);
         InetSocketAddress address = new InetSocketAddress(addr, 12345);
         when(channel.remoteAddress()).thenReturn(address);
         ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(ByteBuf.class))).thenReturn(future);
+        mockContext(context, channel, future);
         ForeignHostPermitHandler handler = new ForeignHostPermitHandler(QosConfiguration.builder()
                 .acceptForeignIp(false)
                 .acceptForeignIpWhitelist(StringUtils.EMPTY_STRING)
@@ -69,14 +68,13 @@
     void shouldShowIpNotPermittedMsg_GivenAcceptForeignIpFalseAndNotMatchWhiteList() throws Exception {
         ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         Channel channel = mock(Channel.class);
-        when(context.channel()).thenReturn(channel);
         InetAddress addr = mock(InetAddress.class);
         when(addr.isLoopbackAddress()).thenReturn(false);
         when(addr.getHostAddress()).thenReturn("179.23.44.1");
         InetSocketAddress address = new InetSocketAddress(addr, 12345);
         when(channel.remoteAddress()).thenReturn(address);
         ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(ByteBuf.class))).thenReturn(future);
+        mockContext(context, channel, future);
         ForeignHostPermitHandler handler = new ForeignHostPermitHandler(QosConfiguration.builder()
                 .acceptForeignIp(false)
                 .acceptForeignIpWhitelist("175.23.44.1 ,  192.168.1.192/26")
@@ -135,9 +133,8 @@
             throws Exception {
         ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         Channel channel = mock(Channel.class);
-        when(context.channel()).thenReturn(channel);
         ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(ByteBuf.class))).thenReturn(future);
+        mockContext(context, channel, future);
         ForeignHostPermitHandler handler = new ForeignHostPermitHandler(QosConfiguration.builder()
                 .acceptForeignIp(false)
                 .acceptForeignIpWhitelist("175.23.44.1 ,  192.168.1.192/26")
@@ -147,4 +144,9 @@
         handler.handlerAdded(context);
         verify(future, never()).addListener(ChannelFutureListener.CLOSE);
     }
+
+    private void mockContext(ChannelHandlerContext context, Channel channel, ChannelFuture future) {
+        when(context.channel()).thenReturn(channel);
+        when(context.writeAndFlush(any(ByteBuf.class))).thenReturn(future);
+    }
 }
