--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/HttpProcessHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/HttpProcessHandlerTest.java
@@ -40,9 +40,8 @@
 class HttpProcessHandlerTest {
     @Test
     void test1() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        ChannelHandlerContext context = mockContext(future);
         HttpRequest message = Mockito.mock(HttpRequest.class);
         when(message.uri()).thenReturn("test");
         HttpProcessHandler handler = new HttpProcessHandler(
@@ -57,9 +56,8 @@
 
     @Test
     void test2() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        ChannelHandlerContext context = mockContext(future);
         HttpRequest message = Mockito.mock(HttpRequest.class);
         when(message.uri()).thenReturn("localhost:80/greeting");
         when(message.method()).thenReturn(HttpMethod.GET);
@@ -78,9 +76,8 @@
 
     @Test
     void test3() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        ChannelHandlerContext context = mockContext(future);
         HttpRequest message = Mockito.mock(HttpRequest.class);
         when(message.uri()).thenReturn("localhost:80/test");
         when(message.method()).thenReturn(HttpMethod.GET);
@@ -96,4 +93,10 @@
         FullHttpResponse response = captor.getValue();
         assertThat(response.status().code(), equalTo(404));
     }
+
+    private ChannelHandlerContext mockContext(ChannelFuture future) {
+        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        return context;
+    }
 }
