--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationInvokerTest.java
@@ -75,11 +75,11 @@
         Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
 
         List<Invoker<?>> invokers = new LinkedList<>();
-        invokers.add(Mockito.mock(Invoker.class));
-        invokers.add(Mockito.mock(Invoker.class));
+        invokers.add(createInvokerMock());
+        invokers.add(createInvokerMock());
         List<Invoker<?>> serviceDiscoveryInvokers = new LinkedList<>();
-        serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
-        serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
+        serviceDiscoveryInvokers.add(createInvokerMock());
+        serviceDiscoveryInvokers.add(createInvokerMock());
         Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
         Mockito.when(serviceDiscoveryDirectory.getAllInvokers()).thenReturn(serviceDiscoveryInvokers);
 
@@ -212,7 +212,7 @@
         }
         Mockito.verify(invoker, Mockito.times(27)).invoke(null);
 
-        serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
+        serviceDiscoveryInvokers.add(createInvokerMock());
         Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
         argument.getAllValues().get(argument.getAllValues().size() - 1).onChange();
 
@@ -251,11 +251,11 @@
         Mockito.when(serviceDiscoveryInvoker.hasProxyInvokers()).thenReturn(true);
 
         List<Invoker<?>> invokers = new LinkedList<>();
-        invokers.add(Mockito.mock(Invoker.class));
-        invokers.add(Mockito.mock(Invoker.class));
+        invokers.add(createInvokerMock());
+        invokers.add(createInvokerMock());
         List<Invoker<?>> serviceDiscoveryInvokers = new LinkedList<>();
-        serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
-        serviceDiscoveryInvokers.add(Mockito.mock(Invoker.class));
+        serviceDiscoveryInvokers.add(createInvokerMock());
+        serviceDiscoveryInvokers.add(createInvokerMock());
         Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
         Mockito.when(serviceDiscoveryDirectory.getAllInvokers()).thenReturn(serviceDiscoveryInvokers);
 
@@ -302,4 +302,8 @@
         // 独立线程invoker状态切换
 
     }
+
+    private Invoker<?> createInvokerMock() {
+        return Mockito.mock(Invoker.class);
+    }
 }
