--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapperTest.java
@@ -38,7 +38,7 @@
     @Test
     void testSubscribe() throws NacosException {
         NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
-        NamingService namingService = Mockito.mock(NamingService.class);
+        NamingService namingService = mockNamingService();
         Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
@@ -63,8 +63,8 @@
     @Test
     void testSubscribeMultiManager() throws NacosException {
         NacosConnectionManager connectionManager = Mockito.mock(NacosConnectionManager.class);
-        NamingService namingService1 = Mockito.mock(NamingService.class);
-        NamingService namingService2 = Mockito.mock(NamingService.class);
+        NamingService namingService1 = mockNamingService();
+        NamingService namingService2 = mockNamingService();
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper = new NacosNamingServiceWrapper(connectionManager, 0, 0);
 
@@ -94,7 +94,7 @@
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -147,7 +147,7 @@
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 try {
                     Mockito.doThrow(new NacosException())
                             .when(namingService)
@@ -221,7 +221,7 @@
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -391,7 +391,7 @@
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = mockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -559,7 +559,7 @@
 
         CountDownLatch startLatch = new CountDownLatch(1);
         CountDownLatch stopLatch = new CountDownLatch(1);
-        NamingService namingService = Mockito.mock(NamingService.class);
+        NamingService namingService = mockNamingService();
         Mockito.when(connectionManager.getNamingService()).thenReturn(namingService);
 
         NacosNamingServiceWrapper nacosNamingServiceWrapper =
@@ -748,4 +748,8 @@
             Assertions.fail(e);
         }
     }
+
+    private NamingService mockNamingService() {
+        return Mockito.mock(NamingService.class);
+    }
 }
