--- a/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosConnectionsManagerTest.java
+++ b/dubbo-registry/dubbo-registry-nacos/src/test/java/org/apache/dubbo/registry/nacos/NacosConnectionsManagerTest.java
@@ -41,7 +41,7 @@
 public class NacosConnectionsManagerTest {
     @Test
     public void testGet() {
-        NamingService namingService = Mockito.mock(NamingService.class);
+        NamingService namingService = createMockNamingService();
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(namingService);
         Assertions.assertEquals(namingService, nacosConnectionManager.getNamingService());
         Assertions.assertEquals(namingService, nacosConnectionManager.getNamingService());
@@ -54,7 +54,7 @@
         NacosConnectionManager nacosConnectionManager = new NacosConnectionManager(URL.valueOf(""), false, 0, 0) {
             @Override
             protected NamingService createNamingService() {
-                NamingService namingService = Mockito.mock(NamingService.class);
+                NamingService namingService = createMockNamingService();
                 namingServiceList.add(namingService);
                 return namingService;
             }
@@ -200,4 +200,8 @@
             }
         }
     }
+
+    private NamingService createMockNamingService() {
+        return Mockito.mock(NamingService.class);
+    }
 }
