--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
@@ -31,7 +31,7 @@
 
     @Test
     void addDescriptor() {
-        ServiceDescriptor descriptor = Mockito.mock(ServiceDescriptor.class);
+        ServiceDescriptor descriptor = mockServiceDescriptor();
         StubSuppliers.addDescriptor(serviceName, descriptor);
         Assertions.assertEquals(descriptor, StubSuppliers.getServiceDescriptor(serviceName));
     }
@@ -39,7 +39,7 @@
     @Test
     void addSupplier() {
         Invoker<?> invoker = Mockito.mock(Invoker.class);
-        ServiceDescriptor descriptor = Mockito.mock(ServiceDescriptor.class);
+        ServiceDescriptor descriptor = mockServiceDescriptor();
         StubSuppliers.addSupplier(serviceName, i -> invoker);
         Assertions.assertEquals(invoker, StubSuppliers.createStub(serviceName, invoker));
     }
@@ -64,4 +64,8 @@
             // pass
         }
     }
+
+    private ServiceDescriptor mockServiceDescriptor() {
+        return Mockito.mock(ServiceDescriptor.class);
+    }
 }
