--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java
@@ -99,7 +99,7 @@
             ClusterInvoker serviceDiscoveryInvoker = Mockito.mock(ClusterInvoker.class);
             Mockito.when(migrationInvoker.getServiceDiscoveryInvoker()).thenReturn(serviceDiscoveryInvoker);
             Directory<?> sdDirectory = Mockito.mock(Directory.class);
-            Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(sdDirectory);
+            mockDirectory(serviceDiscoveryInvoker, sdDirectory);
             List sdInvokers = Mockito.mock(List.class);
             Mockito.when(sdDirectory.getAllInvokers()).thenReturn(sdInvokers);
             Mockito.when(sdInvokers.size()).thenReturn(5);
@@ -114,7 +114,7 @@
             ClusterInvoker serviceDiscoveryInvoker = Mockito.mock(ClusterInvoker.class);
             Mockito.when(migrationInvoker.getServiceDiscoveryInvoker()).thenReturn(serviceDiscoveryInvoker);
             Directory<?> sdDirectory = Mockito.mock(Directory.class);
-            Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(sdDirectory);
+            mockDirectory(serviceDiscoveryInvoker, sdDirectory);
             List sdInvokers = Mockito.mock(List.class);
             Mockito.when(sdDirectory.getAllInvokers()).thenReturn(sdInvokers);
             Mockito.when(sdInvokers.size()).thenReturn(0);
@@ -122,7 +122,7 @@
             ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
             Mockito.when(migrationInvoker.getInvoker()).thenReturn(invoker);
             Directory<?> directory = Mockito.mock(Directory.class);
-            Mockito.when(invoker.getDirectory()).thenReturn(directory);
+            mockDirectory(invoker, directory);
             List invokers = Mockito.mock(List.class);
             Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
             Mockito.when(invokers.size()).thenReturn(10);
@@ -137,7 +137,7 @@
             ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
             Mockito.when(migrationInvoker.getInvoker()).thenReturn(invoker);
             Directory<?> directory = Mockito.mock(Directory.class);
-            Mockito.when(invoker.getDirectory()).thenReturn(directory);
+            mockDirectory(invoker, directory);
             List invokers = Mockito.mock(List.class);
             Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
             Mockito.when(invokers.size()).thenReturn(10);
@@ -150,4 +150,8 @@
                 "zookeeper-A(5)/zookeeper-AF(I-10,A-0)/nacos-I(10)",
                 ServiceCheckUtils.getConsumerAddressNum(consumerModel));
     }
+
+    private void mockDirectory(ClusterInvoker invoker, Directory<?> directory) {
+        Mockito.when(invoker.getDirectory()).thenReturn(directory);
+    }
 }
