--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/migration/MigrationRuleListenerTest.java
@@ -101,8 +101,7 @@
         Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
 
         System.setProperty("dubbo.application.migration.delay", "1");
-        MigrationRuleHandler<?> handler =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
+        MigrationRuleHandler<?> handler = createMigrationRuleHandler();
 
         CountDownLatch countDownLatch = new CountDownLatch(1);
         MigrationRuleListener migrationRuleListener =
@@ -147,8 +146,7 @@
         Mockito.when(consumerURL.getParameter("timestamp")).thenReturn("1");
 
         System.setProperty("dubbo.application.migration.delay", "1000");
-        MigrationRuleHandler<?> handler =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
+        MigrationRuleHandler<?> handler = createMigrationRuleHandler();
 
         MigrationRuleListener migrationRuleListener =
                 new MigrationRuleListener(ApplicationModel.defaultModel().getDefaultModule());
@@ -192,10 +190,8 @@
         Mockito.when(consumerURL2.getParameter("timestamp")).thenReturn("2");
 
         System.setProperty("dubbo.application.migration.delay", "10");
-        MigrationRuleHandler<?> handler =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
-        MigrationRuleHandler<?> handler2 =
-                Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
+        MigrationRuleHandler<?> handler = createMigrationRuleHandler();
+        MigrationRuleHandler<?> handler2 = createMigrationRuleHandler();
 
         // Both local rule and remote rule are here
         // Local rule with one delayed task started to apply
@@ -243,4 +239,8 @@
         Assertions.assertEquals(
                 "APPLICATION_FIRST", captor_event.getValue().getStep().toString());
     }
+
+    private MigrationRuleHandler<?> createMigrationRuleHandler() {
+        return Mockito.mock(MigrationRuleHandler.class, Mockito.withSettings().verboseLogging());
+    }
 }
