--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -71,6 +71,10 @@
         ApplicationModel.defaultModel().destroy();
     }
 
+    private MigrationRuleListener mockMigrationRuleListener() {
+        return mock(MigrationRuleListener.class);
+    }
+
     /**
      * verify the generated consumer url information
      */
@@ -105,7 +109,7 @@
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
+        MigrationRuleListener migrationRuleListener = mockMigrationRuleListener();
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
         registryProtocolListeners.add(migrationRuleListener);
 
@@ -173,7 +177,7 @@
         RegistryProtocol registryProtocol = new RegistryProtocol();
         Registry registry = mock(Registry.class);
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
+        MigrationRuleListener migrationRuleListener = mockMigrationRuleListener();
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
         registryProtocolListeners.add(migrationRuleListener);
 
@@ -240,7 +244,7 @@
         RegistryFactory registryFactory = mock(RegistryFactory.class);
         Registry registry = mock(Registry.class);
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
+        MigrationRuleListener migrationRuleListener = mockMigrationRuleListener();
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
         registryProtocolListeners.add(migrationRuleListener);
 
@@ -307,7 +311,7 @@
         RegistryFactory registryFactory = mock(RegistryFactory.class);
         Registry registry = mock(Registry.class);
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
+        MigrationRuleListener migrationRuleListener = mockMigrationRuleListener();
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
         registryProtocolListeners.add(migrationRuleListener);
 
@@ -385,7 +389,7 @@
                 parameters,
                 consumerAttribute);
         url = url.putAttribute(CONSUMER_URL_KEY, consumerUrl);
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
+        MigrationRuleListener migrationRuleListener = mockMigrationRuleListener();
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
         registryProtocolListeners.add(migrationRuleListener);
 
