--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ProviderAuthFilterTest.java
@@ -44,7 +44,7 @@
     void testAuthDisabled() {
         URL url = mock(URL.class);
         Invoker invoker = mock(Invoker.class);
-        Invocation invocation = mock(RpcInvocation.class);
+        Invocation invocation = mockInvocation();
         when(invoker.getUrl()).thenReturn(url);
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         providerAuthFilter.invoke(invoker, invocation);
@@ -60,7 +60,7 @@
                 .addParameter(Constants.AUTHENTICATOR_KEY, "accesskey")
                 .addParameter(Constants.AUTH_KEY, true);
         Invoker invoker = mock(Invoker.class);
-        Invocation invocation = mock(RpcInvocation.class);
+        Invocation invocation = mockInvocation();
         when(invoker.getUrl()).thenReturn(url);
         ProviderAuthFilter providerAuthFilter = new ProviderAuthFilter(FrameworkModel.defaultModel());
         providerAuthFilter.invoke(invoker, invocation);
@@ -196,4 +196,8 @@
         Result result = providerAuthFilter.invoke(invoker, invocation);
         assertNull(result);
     }
+
+    private Invocation mockInvocation() {
+        return mock(RpcInvocation.class);
+    }
 }
