--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mesh/route/MeshRuleManagerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mesh/route/MeshRuleManagerTest.java
@@ -224,8 +224,8 @@
             }
         };
 
-        when(meshEnvListener1.isEnable()).thenReturn(false);
-        when(meshEnvListener2.isEnable()).thenReturn(true);
+        stubEnablement(meshEnvListener1, false);
+        stubEnablement(meshEnvListener2, true);
 
         meshRuleManager.register("dubbo-demo", meshRuleListener1);
 
@@ -270,4 +270,8 @@
         verify(ruleRepository, times(1)).removeListener("dubbo-demo.MESHAPPRULE", "dubbo", meshAppRuleListener);
         verify(meshEnvListener2, times(1)).onUnSubscribe("dubbo-demo");
     }
+
+    private void stubEnablement(MeshEnvListener meshEnvListener, boolean enabled) {
+        when(meshEnvListener.isEnable()).thenReturn(enabled);
+    }
 }
