--- a/dubbo-plugin/dubbo-mutiny/src/test/java/org/apache/dubbo/mutiny/MutinyServerCallsTest.java
+++ b/dubbo-plugin/dubbo-mutiny/src/test/java/org/apache/dubbo/mutiny/MutinyServerCallsTest.java
@@ -45,7 +45,7 @@
 
     @Test
     void testOneToOne_success() {
-        StreamObserver<String> responseObserver = mock(StreamObserver.class);
+        StreamObserver<String> responseObserver = mockResponseObserver();
 
         Function<Uni<String>, Uni<String>> func = reqUni -> reqUni.onItem().transform(i -> i + "-resp");
 
@@ -59,7 +59,7 @@
 
     @Test
     void testOneToOne_exception() {
-        StreamObserver<String> responseObserver = mock(StreamObserver.class);
+        StreamObserver<String> responseObserver = mockResponseObserver();
 
         // mock func error
         Function<Uni<String>, Uni<String>> func = reqUni -> {
@@ -175,4 +175,8 @@
 
         verify(responseObserver, times(1)).onError(any());
     }
+
+    private StreamObserver<String> mockResponseObserver() {
+        return mock(StreamObserver.class);
+    }
 }
