--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java
@@ -48,7 +48,7 @@
     void test() {
         Codec2 codec2 = Mockito.mock(Codec2.class);
         URL url = Mockito.mock(URL.class);
-        ChannelHandler handler = Mockito.mock(ChannelHandler.class);
+        ChannelHandler handler = createChannelHandler();
         NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, handler);
         io.netty.channel.ChannelHandler decoder = nettyCodecAdapter.getDecoder();
         io.netty.channel.ChannelHandler encoder = nettyCodecAdapter.getEncoder();
@@ -64,7 +64,7 @@
         URL url = Mockito.mock(URL.class);
         doReturn("default").when(url).getParameter(eq(Constants.CODEC_KEY));
 
-        ChannelHandler handler = Mockito.mock(ChannelHandler.class);
+        ChannelHandler handler = createChannelHandler();
         NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, handler);
         io.netty.channel.ChannelHandler decoder = nettyCodecAdapter.getDecoder();
         EmbeddedChannel embeddedChannel = new EmbeddedChannel();
@@ -81,4 +81,8 @@
 
         Assertions.assertEquals(0, input.refCnt());
     }
+
+    private ChannelHandler createChannelHandler() {
+        return Mockito.mock(ChannelHandler.class);
+    }
 }
