--- a/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/ExchangersTest.java
+++ b/dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/exchange/ExchangersTest.java
@@ -31,8 +31,8 @@
     @Test
     void testBind() throws RemotingException {
         String url = "dubbo://127.0.0.1:12345?exchanger=mockExchanger";
-        Exchangers.bind(url, Mockito.mock(Replier.class));
-        Exchangers.bind(url, new ChannelHandlerAdapter(), Mockito.mock(Replier.class));
+        Exchangers.bind(url, createReplier());
+        Exchangers.bind(url, new ChannelHandlerAdapter(), createReplier());
         Exchangers.bind(url, new ExchangeHandlerDispatcher());
 
         Assertions.assertThrows(
@@ -44,9 +44,9 @@
     void testConnect() throws RemotingException {
         String url = "dubbo://127.0.0.1:12345?exchanger=mockExchanger";
         Exchangers.connect(url);
-        Exchangers.connect(url, Mockito.mock(Replier.class));
-        Exchangers.connect(URL.valueOf(url), Mockito.mock(Replier.class));
-        Exchangers.connect(url, new ChannelHandlerAdapter(), Mockito.mock(Replier.class));
+        Exchangers.connect(url, createReplier());
+        Exchangers.connect(URL.valueOf(url), createReplier());
+        Exchangers.connect(url, new ChannelHandlerAdapter(), createReplier());
         Exchangers.connect(url, new ExchangeHandlerDispatcher());
 
         Assertions.assertThrows(
@@ -54,4 +54,8 @@
         Assertions.assertThrows(
                 RuntimeException.class, () -> Exchangers.connect(url, (ExchangeHandlerDispatcher) null));
     }
+
+    private Replier<?> createReplier() {
+        return Mockito.mock(Replier.class);
+    }
 }
