--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeErrorTypeAwareLoggerTest.java
@@ -38,23 +38,11 @@
 class FailsafeErrorTypeAwareLoggerTest {
     @Test
     void testFailsafeErrorTypeAwareForLoggingMethod() {
-        Logger failLogger = mock(Logger.class);
+        Logger failLogger = MockFailLogger.createMockFailLogger();
         FailsafeErrorTypeAwareLogger failsafeLogger = new FailsafeErrorTypeAwareLogger(failLogger);
-
-        doThrow(new RuntimeException()).when(failLogger).error(anyString());
-        doThrow(new RuntimeException()).when(failLogger).warn(anyString());
-        doThrow(new RuntimeException()).when(failLogger).info(anyString());
-        doThrow(new RuntimeException()).when(failLogger).debug(anyString());
-        doThrow(new RuntimeException()).when(failLogger).trace(anyString());
 
         failsafeLogger.error(REGISTRY_ADDRESS_INVALID, "Registry center", "May be it's offline.", "error");
         failsafeLogger.warn(REGISTRY_ADDRESS_INVALID, "Registry center", "May be it's offline.", "warn");
-
-        doThrow(new RuntimeException()).when(failLogger).error(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).warn(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).info(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).debug(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).trace(any(Throwable.class));
 
         failsafeLogger.error(
                 REGISTRY_ADDRESS_INVALID, "Registry center", "May be it's offline.", "error", new Exception("error"));
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/FailsafeLoggerTest.java
@@ -31,26 +31,14 @@
 class FailsafeLoggerTest {
     @Test
     void testFailSafeForLoggingMethod() {
-        Logger failLogger = mock(Logger.class);
+        Logger failLogger = MockFailLogger.createMockFailLogger();
         FailsafeLogger failsafeLogger = new FailsafeLogger(failLogger);
-
-        doThrow(new RuntimeException()).when(failLogger).error(anyString());
-        doThrow(new RuntimeException()).when(failLogger).warn(anyString());
-        doThrow(new RuntimeException()).when(failLogger).info(anyString());
-        doThrow(new RuntimeException()).when(failLogger).debug(anyString());
-        doThrow(new RuntimeException()).when(failLogger).trace(anyString());
 
         failsafeLogger.error("error");
         failsafeLogger.warn("warn");
         failsafeLogger.info("info");
         failsafeLogger.debug("debug");
         failsafeLogger.trace("info");
-
-        doThrow(new RuntimeException()).when(failLogger).error(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).warn(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).info(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).debug(any(Throwable.class));
-        doThrow(new RuntimeException()).when(failLogger).trace(any(Throwable.class));
 
         failsafeLogger.error(new Exception("error"));
         failsafeLogger.warn(new Exception("warn"));
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/MockFailLogger.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/logger/support/MockFailLogger.java
@@ -0,0 +1,25 @@
+package org.apache.dubbo.common.logger.support;
+
+import org.apache.dubbo.common.logger.Logger;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+public class MockFailLogger {
+    public static Logger createMockFailLogger() {
+        Logger failLogger = mock(Logger.class);
+        doThrow(new RuntimeException()).when(failLogger).error(anyString());
+        doThrow(new RuntimeException()).when(failLogger).warn(anyString());
+        doThrow(new RuntimeException()).when(failLogger).info(anyString());
+        doThrow(new RuntimeException()).when(failLogger).debug(anyString());
+        doThrow(new RuntimeException()).when(failLogger).trace(anyString());
+        doThrow(new RuntimeException()).when(failLogger).error(any(Throwable.class));
+        doThrow(new RuntimeException()).when(failLogger).warn(any(Throwable.class));
+        doThrow(new RuntimeException()).when(failLogger).info(any(Throwable.class));
+        doThrow(new RuntimeException()).when(failLogger).debug(any(Throwable.class));
+        doThrow(new RuntimeException()).when(failLogger).trace(any(Throwable.class));
+        return failLogger;
+    }
+}
