--- a/spring-integration-syslog/src/test/java/org/springframework/integration/syslog/inbound/SyslogReceivingChannelAdapterTests.java
+++ b/spring-integration-syslog/src/test/java/org/springframework/integration/syslog/inbound/SyslogReceivingChannelAdapterTests.java
@@ -105,12 +105,8 @@
 		factory.setPort(0);
 		PollableChannel outputChannel = new QueueChannel();
 		factory.setOutputChannel(outputChannel);
-		ApplicationEventPublisher publisher = mock(ApplicationEventPublisher.class);
 		final CountDownLatch latch = new CountDownLatch(2);
-		doAnswer(invocation -> {
-			latch.countDown();
-			return null;
-		}).when(publisher).publishEvent(any(ApplicationEvent.class));
+		ApplicationEventPublisher publisher = createMockApplicationEventPublisher(latch);
 		factory.setApplicationEventPublisher(publisher);
 		factory.setBeanFactory(getBeanFactory());
 		factory.afterPropertiesSet();
@@ -184,12 +180,8 @@
 		factory.setBeanFactory(getBeanFactory());
 		PollableChannel outputChannel = new QueueChannel();
 		factory.setOutputChannel(outputChannel);
-		ApplicationEventPublisher publisher = mock(ApplicationEventPublisher.class);
 		final CountDownLatch latch = new CountDownLatch(2);
-		doAnswer(invocation -> {
-			latch.countDown();
-			return null;
-		}).when(publisher).publishEvent(any(ApplicationEvent.class));
+		ApplicationEventPublisher publisher = createMockApplicationEventPublisher(latch);
 		factory.setBeanFactory(getBeanFactory());
 		factory.setApplicationEventPublisher(publisher);
 		AbstractServerConnectionFactory connectionFactory = new TcpNioServerConnectionFactory(0);
@@ -275,4 +267,13 @@
 		return beanFactory;
 	}
 
+
+    private static ApplicationEventPublisher createMockApplicationEventPublisher(CountDownLatch latch) {
+        ApplicationEventPublisher publisher = mock(ApplicationEventPublisher.class);
+        doAnswer(invocation -> {
+            latch.countDown();
+            return null;
+        }).when(publisher).publishEvent(any(ApplicationEvent.class));
+        return publisher;
+    }
 }
