--- a/spring-integration-core/src/test/java/org/springframework/integration/config/GlobalChannelInterceptorProcessorTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/config/GlobalChannelInterceptorProcessorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.config;
 
+import org.mockito.Mock;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
@@ -39,6 +40,8 @@
  * @since 5.0
  */
 public class GlobalChannelInterceptorProcessorTests {
+	@Mock
+	private ChannelInterceptor channelInterceptor;
 
 	private GlobalChannelInterceptorProcessor globalChannelInterceptorProcessor;
 
@@ -66,7 +69,6 @@
 	public void testProcessorWithInterceptorDefaultPattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
 				new GlobalChannelInterceptorWrapper(channelInterceptor);
 
@@ -89,7 +91,6 @@
 	public void testProcessorWithInterceptorMatchingPattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
 				new GlobalChannelInterceptorWrapper(channelInterceptor);
 
@@ -112,7 +113,6 @@
 	public void testProcessorWithInterceptorNotMatchingPattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
 				new GlobalChannelInterceptorWrapper(channelInterceptor);
 
@@ -136,7 +136,6 @@
 	public void testProcessorWithInterceptorMatchingNegativePattern() {
 		Map<String, GlobalChannelInterceptorWrapper> interceptors = new HashMap<>();
 		Map<String, InterceptableChannel> channels = new HashMap<>();
-		ChannelInterceptor channelInterceptor = mock();
 		GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper =
 				new GlobalChannelInterceptorWrapper(channelInterceptor);
 
