--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/channel/DispatcherHasNoSubscribersTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/channel/DispatcherHasNoSubscribersTests.java
@@ -53,8 +53,7 @@
 		when(declareOk.getQueue()).thenReturn("noSubscribersChannel");
 		when(channel.queueDeclare(anyString(), anyBoolean(), anyBoolean(), anyBoolean(), isNull()))
 				.thenReturn(declareOk);
-		Connection connection = mock(Connection.class);
-		doAnswer(invocation -> channel).when(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection(channel);
 		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -77,8 +76,7 @@
 	@Test
 	public void testPubSub() {
 		final Channel channel = mock();
-		Connection connection = mock();
-		doAnswer(invocation -> channel).when(connection).createChannel(anyBoolean());
+		Connection connection = createMockConnection(channel);
 		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
@@ -96,4 +94,10 @@
 				.withMessageContaining("Dispatcher has no subscribers for amqp-channel 'noSubscribersChannel'.");
 	}
 
+
+	private static Connection createMockConnection(Channel channel) {
+		Connection connection = mock(Connection.class);
+		doAnswer(invocation -> channel).when(connection).createChannel(anyBoolean());
+		return connection;
+	}
 }
