--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
@@ -529,8 +529,7 @@
 		}).given(mockConsumer).poll(any(Duration.class));
 		ConsumerGroupMetadata meta = mock();
 		given(mockConsumer.groupMetadata()).willReturn(meta);
-		ConsumerFactory cf = mock();
-		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
+		ConsumerFactory cf = createMockConsumerFactory(mockConsumer);
 		Producer producer = mock(Producer.class);
 		given(producer.send(any(), any())).willReturn(mock(Future.class));
 		final CountDownLatch closeLatch = new CountDownLatch(2);
@@ -666,8 +665,7 @@
 		}).given(mockConsumer).poll(any(Duration.class));
 		ConsumerGroupMetadata meta = mock();
 		given(mockConsumer.groupMetadata()).willReturn(meta);
-		ConsumerFactory cf = mock();
-		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
+		ConsumerFactory cf = createMockConsumerFactory(mockConsumer);
 		Producer producer = mock(Producer.class);
 		given(producer.send(any(), any())).willReturn(mock(Future.class));
 		final CountDownLatch closeLatch = new CountDownLatch(2);
@@ -840,4 +838,10 @@
 
 	}
 
+
+	private static ConsumerFactory createMockConsumerFactory(Consumer mockConsumer) {
+		ConsumerFactory cf = mock();
+		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
+		return cf;
+	}
 }
