--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.kafka.outbound;
 
+import org.mockito.Mock;
 import java.time.Duration;
 import java.util.Collection;
 import java.util.Collections;
@@ -123,6 +124,8 @@
  * @since 5.4
  */
 class KafkaProducerMessageHandlerTests implements TestApplicationContextAware {
+	@Mock
+	private ConsumerGroupMetadata meta;
 
 	private static final String topic1 = "testTopic1out";
 
@@ -527,7 +530,6 @@
 				return null;
 			}
 		}).given(mockConsumer).poll(any(Duration.class));
-		ConsumerGroupMetadata meta = mock();
 		given(mockConsumer.groupMetadata()).willReturn(meta);
 		ConsumerFactory cf = mock();
 		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
@@ -664,7 +666,6 @@
 				return null;
 			}
 		}).given(mockConsumer).poll(any(Duration.class));
-		ConsumerGroupMetadata meta = mock();
 		given(mockConsumer.groupMetadata()).willReturn(meta);
 		ConsumerFactory cf = mock();
 		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
