--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/inbound/MessageSourceTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/inbound/MessageSourceTests.java
@@ -120,10 +120,7 @@
 			return null;
 		}).given(consumer).subscribe(anyCollection(), any(ConsumerRebalanceListener.class));
 
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		ConsumerProperties consumerProperties = new ConsumerProperties("foo");
 		AtomicBoolean partitionsAssignedCalled = new AtomicBoolean();
 		AtomicReference<Consumer> partitionsAssignedConsumer = new AtomicReference<>();
@@ -172,10 +169,7 @@
 			return null;
 		}).given(consumer).subscribe(anyCollection(), any(ConsumerRebalanceListener.class));
 
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		ConsumerProperties consumerProperties = new ConsumerProperties("foo");
 		AtomicBoolean partitionsAssignedCalled = new AtomicBoolean();
 		AtomicBoolean partitionsRevokedCalled = new AtomicBoolean();
@@ -275,10 +269,7 @@
 		ConsumerRecords cr4 = new ConsumerRecords(records4, Map.of());
 		ConsumerRecords cr5 = new ConsumerRecords(Collections.emptyMap(), Map.of());
 		given(consumer.poll(any(Duration.class))).willReturn(cr1, cr2, cr3, cr4, cr5);
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		ConsumerProperties consumerProperties = new ConsumerProperties("foo");
 		AtomicInteger callbackCount = new AtomicInteger();
 		OffsetCommitCallback commitCallback = (offsets, ex) -> {
@@ -414,10 +405,7 @@
 		ConsumerRecords cr6 = new ConsumerRecords(records6, Map.of());
 		ConsumerRecords cr7 = new ConsumerRecords(Collections.emptyMap(), Map.of());
 		given(consumer.poll(any(Duration.class))).willReturn(cr1, cr2, cr3, cr4, cr5, cr6, cr7);
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		KafkaMessageSource source = new KafkaMessageSource(consumerFactory, new ConsumerProperties("foo"));
 
 		Message<?> received1 = source.receive();
@@ -490,10 +478,7 @@
 		ConsumerRecords cr2 = new ConsumerRecords(records2, Map.of());
 		ConsumerRecords cr3 = new ConsumerRecords(Collections.emptyMap(), Map.of());
 		given(consumer.poll(any(Duration.class))).willReturn(cr1, cr1, cr2, cr2, cr3);
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		ConsumerProperties consumerProperties = new ConsumerProperties("foo");
 		consumerProperties.setSyncCommitTimeout(Duration.ofSeconds(30));
 		KafkaMessageSource source = new KafkaMessageSource(consumerFactory, consumerProperties);
@@ -561,10 +546,7 @@
 		ConsumerRecords cr2 = new ConsumerRecords(records2, Map.of());
 		ConsumerRecords cr3 = new ConsumerRecords(Collections.emptyMap(), Map.of());
 		given(consumer.poll(any(Duration.class))).willReturn(cr1, cr2, cr1, cr2, cr3);
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		KafkaMessageSource source = new KafkaMessageSource(consumerFactory, new ConsumerProperties("foo"));
 
 		Message<?> received1 = source.receive();
@@ -666,10 +648,7 @@
 						new RecordHeaders(), Optional.empty())));
 		ConsumerRecords cr2 = new ConsumerRecords(records2, Map.of());
 		given(consumer.poll(Duration.of(5000, ChronoUnit.MILLIS))).willReturn(cr2, ConsumerRecords.EMPTY);
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		KafkaMessageSource source = new KafkaMessageSource(consumerFactory, new ConsumerProperties("foo"));
 		source.setRawMessageHeader(true);
 
@@ -861,11 +840,8 @@
 		consumer.seek(negativeRelativeToCurrent, 3);
 		consumer.seek(positiveRelativeToCurrent, 1);
 
-		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
-		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)).given(consumerFactory)
-				.getConfigurationProperties();
-		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any())).willReturn(consumer);
-
+
+		ConsumerFactory consumerFactory = createConsumerFactory(consumer);
 		TopicPartitionOffset beginningTpo = new TopicPartitionOffset(beginning, null,
 				TopicPartitionOffset.SeekPosition.BEGINNING);
 		TopicPartitionOffset endTpo = new TopicPartitionOffset(end, null, TopicPartitionOffset.SeekPosition.END);
@@ -950,4 +926,14 @@
 		inOrder.verify(consumer).close(any(CloseOptions.class));
 	}
 
+
+	private static ConsumerFactory createConsumerFactory(Consumer consumer) {
+		ConsumerFactory consumerFactory = mock(ConsumerFactory.class);
+		willReturn(Collections.singletonMap(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1))
+				.given(consumerFactory)
+				.getConfigurationProperties();
+		given(consumerFactory.createConsumer(isNull(), anyString(), isNull(), any()))
+				.willReturn(consumer);
+		return consumerFactory;
+	}
 }
