--- a/spring-integration-jpa/src/test/java/org/springframework/integration/jpa/core/JpaExecutorTests.java
+++ b/spring-integration-jpa/src/test/java/org/springframework/integration/jpa/core/JpaExecutorTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.jpa.core;
 
+import org.mockito.Mock;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
@@ -55,6 +56,8 @@
 @SpringJUnitConfig
 @DirtiesContext
 public class JpaExecutorTests implements TestApplicationContextAware {
+	@Mock
+	private EntityManager entityManager;
 
 	@Autowired
 	protected EntityManager entityManager;
@@ -68,7 +71,7 @@
 	 */
 	@Test
 	public void testExecutePollWithNoEntityClassSpecified() {
-		JpaExecutor jpaExecutor = new JpaExecutor(mock(EntityManager.class));
+		JpaExecutor jpaExecutor = new JpaExecutor(entityManager);
 		jpaExecutor.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		jpaExecutor.afterPropertiesSet();
 
@@ -88,7 +91,7 @@
 
 	@Test
 	public void testSetMultipleQueryTypes() {
-		final JpaExecutor executor = new JpaExecutor(mock(EntityManager.class));
+		final JpaExecutor executor = new JpaExecutor(entityManager);
 		executor.setJpaQuery("select s from Student s");
 		assertThat(TestUtils.<String>getPropertyValue(executor, "jpaQuery")).isNotNull();
 
@@ -104,7 +107,7 @@
 
 		assertThat(TestUtils.<Object>getPropertyValue(executor, "nativeQuery")).isNull();
 
-		final JpaExecutor executor2 = new JpaExecutor(mock(EntityManager.class));
+		final JpaExecutor executor2 = new JpaExecutor(entityManager);
 		executor2.setNamedQuery("NamedQuery");
 		assertThat(TestUtils.<String>getPropertyValue(executor2, "namedQuery")).isNotNull();
 
@@ -243,7 +246,7 @@
 
 	@Test
 	public void withNullMaxResultsExpression() {
-		final JpaExecutor jpaExecutor = new JpaExecutor(mock(EntityManager.class));
+		final JpaExecutor jpaExecutor = new JpaExecutor(entityManager);
 		assertThatIllegalArgumentException()
 				.isThrownBy(() -> jpaExecutor.setMaxResultsExpression(null))
 				.withMessage("maxResultsExpression cannot be null");
