--- a/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
+++ b/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.mqtt;
 
+import org.mockito.Mock;
 import java.lang.reflect.Method;
 import java.util.Properties;
 import java.util.concurrent.BlockingQueue;
@@ -103,6 +104,8 @@
  *
  */
 public class MqttAdapterTests implements TestApplicationContextAware {
+	@Mock
+	private final IMqttAsyncClient client;
 
 	private final IMqttToken alwaysComplete;
 
@@ -375,7 +378,6 @@
 
 	@Test
 	public void testStopActionDefault() throws Exception {
-		final IMqttAsyncClient client = mock(IMqttAsyncClient.class);
 		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapterIn(client, null);
 
 		adapter.start();
@@ -386,7 +388,6 @@
 
 	@Test
 	public void testStopActionDefaultNotClean() throws Exception {
-		final IMqttAsyncClient client = mock(IMqttAsyncClient.class);
 		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapterIn(client, false);
 
 		adapter.start();
