--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpMessageMapperTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpMessageMapperTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ip.tcp.connection;
 
+import org.mockito.Mock;
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.net.InetAddress;
@@ -60,6 +61,8 @@
  *
  */
 public class TcpMessageMapperTests {
+	@Mock
+	private InetAddress local;
 
 	private static final String TEST_PAYLOAD = "abcdefghijkl";
 
@@ -75,7 +78,6 @@
 		TcpMessageMapper mapper = new TcpMessageMapper();
 		mapper.setBeanFactory(mock());
 		TcpConnection connection = creatMockTcpConcnection(TEST_PAYLOAD.getBytes(), "MyHost", "1.1.1.1", 1234);
-		InetAddress local = mock(InetAddress.class);
 		Socket socket = creatMockSocket(local);
 		SocketInfo info = new SocketInfo(socket);
 		when(connection.getSocketInfo()).thenReturn(info);
@@ -94,7 +96,6 @@
 		mapper.setAddContentTypeHeader(true);
 		mapper.setBeanFactory(mock());
 		TcpConnection connection = creatMockTcpConcnection(TEST_PAYLOAD.getBytes(), "MyHost", "1.1.1.1", 1234);
-		InetAddress local = mock(InetAddress.class);
 		Socket socket = creatMockSocket(local);
 		SocketInfo info = new SocketInfo(socket);
 		when(connection.getSocketInfo()).thenReturn(info);
@@ -117,7 +118,6 @@
 		mapper.setContentType("application/octet-stream;charset=ISO-8859-1");
 		mapper.setBeanFactory(mock());
 		TcpConnection connection = creatMockTcpConcnection(TEST_PAYLOAD.getBytes(), "MyHost", "1.1.1.1", 1234);
-		InetAddress local = mock(InetAddress.class);
 		Socket socket = creatMockSocket(local);
 		SocketInfo info = new SocketInfo(socket);
 		when(connection.getSocketInfo()).thenReturn(info);
