--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpNioConnectionTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpNioConnectionTests.java
@@ -246,18 +246,15 @@
 		TcpNioClientConnectionFactory factory = new TcpNioClientConnectionFactory("localhost", 0);
 		factory.setApplicationEventPublisher(nullPublisher);
 		Map<SocketChannel, TcpNioConnection> connections = new HashMap<>();
-		SocketChannel chan1 = mock();
-		SocketChannel chan2 = mock();
-		SocketChannel chan3 = mock();
+		SocketChannel chan1 = createMockSocketChannel(true);
+		SocketChannel chan2 = createMockSocketChannel(true);
 		TcpNioConnection conn1 = mock();
 		TcpNioConnection conn2 = mock();
 		TcpNioConnection conn3 = mock();
 		connections.put(chan1, conn1);
 		connections.put(chan2, conn2);
 		connections.put(chan3, conn3);
-		willReturn(true).given(chan1).isOpen();
-		willReturn(true).given(chan2).isOpen();
-		willReturn(true).given(chan3).isOpen();
+		SocketChannel chan3 = createMockSocketChannel(true);
 		Selector selector = mock();
 		HashSet<SelectionKey> keys = new HashSet<>();
 		when(selector.selectedKeys()).thenReturn(keys);
@@ -867,8 +864,7 @@
 				new ThreadPoolExecutor(1, 2, 0, TimeUnit.MILLISECONDS, new SynchronousQueue<>());
 		factory.setTaskExecutor(threadPoolExecutor);
 		factory.start();
-		SocketChannel chan1 = mock();
-		willReturn(true).given(chan1).isOpen();
+		SocketChannel chan1 = createMockSocketChannel(true);
 		when(chan1.socket()).thenReturn(mock());
 		TcpNioConnection conn1 = new TcpNioConnection(chan1, false, false, null, null);
 		conn1.setTaskExecutor(threadPoolExecutor);
@@ -950,4 +946,10 @@
 		}
 	}
 
+
+	private static SocketChannel createMockSocketChannel(boolean isOpen) {
+		SocketChannel channel = mock();
+		willReturn(isOpen).given(channel).isOpen();
+		return channel;
+	}
 }
