--- a/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
+++ b/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
@@ -277,11 +277,9 @@
 		new DirectFieldAccessor(receiver).setPropertyValue("logger", logger);
 		when(logger.isDebugEnabled()).thenReturn(true);
 
-		Message msg1 = mock(MimeMessage.class);
-		Message msg2 = mock(MimeMessage.class);
+		Message msg2 = createMockMessage("msg2");
+		Message msg1 = createMockMessage("msg1");
 		given(msg1.isExpunged()).willReturn(true);
-		given(msg1.getSubject()).willReturn("msg1");
-		given(msg2.getSubject()).willReturn("msg2");
 		Expression selectorExpression = new SpelExpressionParser().parseExpression("false");
 		receiver.setSelectorExpression(selectorExpression);
 		receiveAndMarkAsReadDontDeleteGuts(receiver, msg1, msg2);
@@ -1043,4 +1041,10 @@
 
 	}
 
+
+	private static Message createMockMessage(String subject) throws MessagingException {
+		Message message = mock(Message.class);
+		when(message.getSubject()).thenReturn(subject);
+		return message;
+	}
 }
