--- a/spring-integration-mail/src/test/java/org/springframework/integration/mail/config/MailToStringTransformerParserTests.java
+++ b/spring-integration-mail/src/test/java/org/springframework/integration/mail/config/MailToStringTransformerParserTests.java
@@ -45,8 +45,7 @@
 			MessageChannel input = new BeanFactoryChannelResolver(context).resolveDestination("input");
 			PollableChannel output =
 					(PollableChannel) new BeanFactoryChannelResolver(context).resolveDestination("output");
-			MimeMessage mimeMessage = Mockito.mock(MimeMessage.class);
-			Mockito.when(mimeMessage.getContent()).thenReturn("hello");
+			MimeMessage mimeMessage = createMockMimeMessage("hello");
 			input.send(new GenericMessage<jakarta.mail.Message>(mimeMessage));
 			Message<?> result = output.receive(10_000);
 			assertThat(result).isNotNull();
@@ -63,8 +62,7 @@
 			MessageChannel input = new BeanFactoryChannelResolver(context).resolveDestination("input");
 			PollableChannel output =
 					(PollableChannel) new BeanFactoryChannelResolver(context).resolveDestination("output");
-			MimeMessage mimeMessage = Mockito.mock(MimeMessage.class);
-			Mockito.when(mimeMessage.getContent()).thenReturn("foo");
+			MimeMessage mimeMessage = createMockMimeMessage("foo");
 			input.send(new GenericMessage<jakarta.mail.Message>(mimeMessage));
 			Message<?> result = output.receive(0);
 			assertThat(result).isNotNull();
@@ -82,4 +80,10 @@
 				.withMessageContaining("input-channel");
 	}
 
+
+	private static MimeMessage createMockMimeMessage(String content) {
+		MimeMessage mimeMessage = Mockito.mock(MimeMessage.class);
+		Mockito.when(mimeMessage.getContent()).thenReturn(content);
+		return mimeMessage;
+	}
 }
