--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
@@ -866,10 +866,7 @@
 		@SuppressWarnings("unchecked")
 		SessionFactory<TestLsEntry> sessionFactory = mock(SessionFactory.class);
 		@SuppressWarnings("unchecked")
-		Session<TestLsEntry> session = mock(Session.class);
-		willReturn(Boolean.TRUE)
-				.given(session)
-				.exists(anyString());
+		Session<TestLsEntry> session = createMockSession();
 		RemoteFileTemplate<TestLsEntry> template = new RemoteFileTemplate<>(sessionFactory);
 		template.setRemoteDirectoryExpression(new LiteralExpression("foo/"));
 		template.setBeanFactory(TEST_INTEGRATION_CONTEXT);
@@ -926,10 +923,7 @@
 	@SuppressWarnings("unchecked")
 	public void testPutExistsExpression() throws Exception {
 		SessionFactory<TestLsEntry> sessionFactory = mock(SessionFactory.class);
-		Session<TestLsEntry> session = mock(Session.class);
-		willReturn(Boolean.TRUE)
-				.given(session)
-				.exists(anyString());
+		Session<TestLsEntry> session = createMockSession();
 		RemoteFileTemplate<TestLsEntry> template = new RemoteFileTemplate<>(sessionFactory);
 		template.setRemoteDirectoryExpression(new LiteralExpression("foo/"));
 		template.setBeanFactory(TEST_INTEGRATION_CONTEXT);
@@ -1311,4 +1305,11 @@
 
 	}
 
+
+	@SuppressWarnings("unchecked")
+	private static Session<TestLsEntry> createMockSession() {
+		Session<TestLsEntry> session = mock(Session.class);
+		willReturn(Boolean.TRUE).given(session).exists(anyString());
+		return session;
+	}
 }
