--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpConnectionEventListenerTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/TcpConnectionEventListenerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ip.tcp.connection;
 
+import org.mockito.Mock;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.context.event.ApplicationEventMulticaster;
@@ -36,6 +37,8 @@
  *
  */
 public class TcpConnectionEventListenerTests implements TestApplicationContextAware {
+	@Mock
+	private TcpConnectionSupport connection;
 
 	@Test
 	public void testNoFilter() {
@@ -48,7 +51,6 @@
 		eventProducer.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		eventProducer.afterPropertiesSet();
 		eventProducer.start();
-		TcpConnectionSupport connection = mock(TcpConnectionSupport.class);
 
 		assertThat(eventProducer.supportsEventType(ResolvableType.forClass(TcpConnectionOpenEvent.class))).isTrue();
 		TcpConnectionEvent event1 = new TcpConnectionOpenEvent(connection, "foo");
@@ -86,7 +88,6 @@
 		eventProducer.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		eventProducer.afterPropertiesSet();
 		eventProducer.start();
-		TcpConnectionSupport connection = mock(TcpConnectionSupport.class);
 
 		assertThat(eventProducer.supportsEventType(ResolvableType.forClass(TcpConnectionOpenEvent.class))).isFalse();
 
