--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFactoryTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFactoryTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ip.dsl;
 
+import org.mockito.Mock;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
@@ -54,6 +55,8 @@
  *
  */
 public class ConnectionFactoryTests implements TestApplicationContextAware {
+	@Mock
+	private TcpSocketSupport sockSupp;
 
 	@Test
 	public void test() throws Exception {
@@ -104,7 +107,6 @@
 
 	@Test
 	void netCustomServer() {
-		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
 		TcpNetConnectionSupport conSupp = mock(TcpNetConnectionSupport.class);
 		TcpSocketFactorySupport factSupp = mock(TcpSocketFactorySupport.class);
 		TcpNetServerConnectionFactory server = Tcp.netServer(0)
@@ -119,7 +121,6 @@
 
 	@Test
 	void nioCustomServer() {
-		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
 		TcpNioConnectionSupport conSupp = mock(TcpNioConnectionSupport.class);
 		TcpNioServerConnectionFactory server = Tcp.nioServer(0)
 				.socketSupport(sockSupp)
@@ -133,7 +134,6 @@
 
 	@Test
 	void netCustomClient() {
-		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
 		TcpNetConnectionSupport conSupp = mock(TcpNetConnectionSupport.class);
 		TcpSocketFactorySupport factSupp = mock(TcpSocketFactorySupport.class);
 		TcpNetClientConnectionFactory client = Tcp.netClient("localhost", 0)
@@ -148,7 +148,6 @@
 
 	@Test
 	void nioCustomClient() {
-		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
 		TcpNioConnectionSupport conSupp = mock(TcpNioConnectionSupport.class);
 		TcpNioClientConnectionFactory client = Tcp.nioClient("localhost", 0)
 				.socketSupport(sockSupp)
