--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFactoryTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFactoryTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ip.dsl;
 
+import org.mockito.Mock;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
@@ -54,6 +55,8 @@
  *
  */
 public class ConnectionFactoryTests implements TestApplicationContextAware {
+	@Mock
+	private TcpNioConnectionSupport conSupp;
 
 	@Test
 	public void test() throws Exception {
@@ -120,7 +123,6 @@
 	@Test
 	void nioCustomServer() {
 		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
-		TcpNioConnectionSupport conSupp = mock(TcpNioConnectionSupport.class);
 		TcpNioServerConnectionFactory server = Tcp.nioServer(0)
 				.socketSupport(sockSupp)
 				.directBuffers(true)
@@ -149,7 +151,6 @@
 	@Test
 	void nioCustomClient() {
 		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
-		TcpNioConnectionSupport conSupp = mock(TcpNioConnectionSupport.class);
 		TcpNioClientConnectionFactory client = Tcp.nioClient("localhost", 0)
 				.socketSupport(sockSupp)
 				.directBuffers(true)
