--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ip.tcp.connection;
 
+import org.mockito.Mock;
 import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.net.InetAddress;
@@ -70,6 +71,8 @@
  * @since 2.2
  */
 public class SocketSupportTests implements TestApplicationContextAware {
+	@Mock
+	private TcpSocketSupport socketSupport;
 
 	@Test
 	public void testNetClient() throws Exception {
@@ -83,7 +86,6 @@
 		InetAddress inetAddress = InetAddress.getLocalHost();
 		when(socket.getInetAddress()).thenReturn(inetAddress);
 		when(factory.createSocket()).thenReturn(socket);
-		TcpSocketSupport socketSupport = Mockito.mock(TcpSocketSupport.class);
 
 		TcpNetClientConnectionFactory connectionFactory = new TcpNetClientConnectionFactory("x", 0);
 		connectionFactory.setTcpSocketFactorySupport(factorySupport);
@@ -111,7 +113,6 @@
 		when(socket.getInetAddress()).thenReturn(inetAddress);
 		when(factory.createSocket()).thenReturn(socket);
 		doThrow(new SocketTimeoutException()).when(socket).connect(any(), eq(1000));
-		TcpSocketSupport socketSupport = Mockito.mock(TcpSocketSupport.class);
 
 		TcpNetClientConnectionFactory connectionFactory = new TcpNetClientConnectionFactory("x", 0);
 		connectionFactory.setConnectTimeout(1);
@@ -161,7 +162,6 @@
 			Thread.sleep(50);
 			return socket1;
 		});
-		TcpSocketSupport socketSupport = mock(TcpSocketSupport.class);
 
 		TcpNetServerConnectionFactory connectionFactory = new TcpNetServerConnectionFactory(0);
 		connectionFactory.setTcpSocketFactorySupport(factorySupport);
