--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandlerTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandlerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.file.remote.handler;
 
+import org.mockito.Mock;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Set;
@@ -56,6 +57,8 @@
  * @author Glenn Renfro
  */
 public class FileTransferringMessageHandlerTests implements TestApplicationContextAware {
+	@Mock
+	private Session<F> session;
 
 	@SuppressWarnings("unchecked")
 	@Test
@@ -129,7 +132,6 @@
 	@Test
 	public <F> void testEmptyTemporaryFileSuffixCannotBeNull() throws Exception {
 		SessionFactory<F> sf = mock(SessionFactory.class);
-		Session<F> session = mock(Session.class);
 		when(sf.getSession()).thenReturn(session);
 		FileTransferringMessageHandler<F> handler = new FileTransferringMessageHandler<F>(sf);
 		handler.setBeanFactory(TEST_INTEGRATION_CONTEXT);
@@ -142,7 +144,6 @@
 	@Test
 	public <F> void testUseTemporaryFileNameFalse() throws Exception {
 		SessionFactory<F> sf = mock(SessionFactory.class);
-		Session<F> session = mock(Session.class);
 
 		when(sf.getSession()).thenReturn(session);
 
