--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java
@@ -73,9 +73,8 @@
 
 	@Test
 	public void testNetClient() throws Exception {
-		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
 		SocketFactory factory = Mockito.mock(SocketFactory.class);
-		when(factorySupport.getSocketFactory()).thenReturn(factory);
+		TcpSocketFactorySupport factorySupport = createTcpSocketFactorySupport(factory);
 		Socket socket = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
@@ -100,9 +99,8 @@
 
 	@Test
 	public void testNetClientSocketTimeout() throws Exception {
-		TcpSocketFactorySupport factorySupport = mock(TcpSocketFactorySupport.class);
 		SocketFactory factory = Mockito.mock(SocketFactory.class);
-		when(factorySupport.getSocketFactory()).thenReturn(factory);
+		TcpSocketFactorySupport factorySupport = createTcpSocketFactorySupport(factory);
 		Socket socket = mock(Socket.class);
 		InputStream is = mock(InputStream.class);
 		when(is.read()).thenReturn(-1);
@@ -720,4 +718,10 @@
 
 	}
 
+
+    private static TcpSocketFactorySupport createTcpSocketFactorySupport(SocketFactory factory) {
+        TcpSocketFactorySupport factorySupport = Mockito.mock(TcpSocketFactorySupport.class);
+        when(factorySupport.getSocketFactory()).thenReturn(factory);
+        return factorySupport;
+    }
 }
