--- a/spring-integration-ftp/src/test/java/org/springframework/integration/ftp/session/SessionFactoryTests.java
+++ b/spring-integration-ftp/src/test/java/org/springframework/integration/ftp/session/SessionFactoryTests.java
@@ -144,10 +144,8 @@
 	@SuppressWarnings("unchecked")
 	void testStaleConnection() {
 		SessionFactory<FTPFile> sessionFactory = Mockito.mock(SessionFactory.class);
-		Session<FTPFile> sessionA = Mockito.mock(Session.class);
-		Session<FTPFile> sessionB = Mockito.mock(Session.class);
-		Mockito.when(sessionA.isOpen()).thenReturn(true);
-		Mockito.when(sessionB.isOpen()).thenReturn(false);
+		Session<FTPFile> sessionB = createMockSession(false);
+		Session<FTPFile> sessionA = createMockSession(true);
 
 		Mockito.when(sessionFactory.getSession()).thenReturn(sessionA);
 		Mockito.when(sessionFactory.getSession()).thenReturn(sessionB);
@@ -228,4 +226,11 @@
 		assertThat(failures.get()).isEqualTo(0);
 	}
 
+
+	@SuppressWarnings("unchecked")
+	private static Session<FTPFile> createMockSession(boolean isOpen) {
+		Session<FTPFile> session = Mockito.mock(Session.class);
+		Mockito.when(session.isOpen()).thenReturn(isOpen);
+		return session;
+	}
 }
