--- a/spring-integration-core/src/test/java/org/springframework/integration/support/SmartLifecycleRoleControllerTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/support/SmartLifecycleRoleControllerTests.java
@@ -40,10 +40,8 @@
 
 	@Test
 	public void testOrder() {
-		SmartLifecycle lc1 = mock(SmartLifecycle.class);
-		when(lc1.getPhase()).thenReturn(2);
-		SmartLifecycle lc2 = mock(SmartLifecycle.class);
-		when(lc2.getPhase()).thenReturn(1);
+		SmartLifecycle lc1 = createMockSmartLifecycle(2);
+		SmartLifecycle lc2 = createMockSmartLifecycle(1);
 		MultiValueMap<String, SmartLifecycle> map = new LinkedMultiValueMap<>();
 		map.add("foo", lc1);
 		map.add("foo", lc2);
@@ -120,4 +118,10 @@
 
 	}
 
+
+	private static SmartLifecycle createMockSmartLifecycle(int phase) {
+		SmartLifecycle lifecycle = mock(SmartLifecycle.class);
+		when(lifecycle.getPhase()).thenReturn(phase);
+		return lifecycle;
+	}
 }
