--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/DefaultSoapHeaderMapperTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/DefaultSoapHeaderMapperTests.java
@@ -151,18 +151,16 @@
 	@Test
 	public void testExtractStandardHeadersEmptySoapAction() {
 		DefaultSoapHeaderMapper mapper = new DefaultSoapHeaderMapper();
-		SoapMessage soapMessage = mock(SoapMessage.class);
-		when(soapMessage.getSoapAction()).thenReturn("");
-
+
+		SoapMessage soapMessage = createMockSoapMessage("");
 		assertThat(mapper.extractStandardHeaders(soapMessage).isEmpty()).isTrue();
 	}
 
 	@Test
 	public void testExtractStandardHeadersNonEmptySoapAction() {
 		DefaultSoapHeaderMapper mapper = new DefaultSoapHeaderMapper();
-		SoapMessage soapMessage = mock(SoapMessage.class);
-		when(soapMessage.getSoapAction()).thenReturn("foo");
-
+
+		SoapMessage soapMessage = createMockSoapMessage("foo");
 		Map<String, Object> standardHeaders = mapper.toHeadersFromRequest(soapMessage);
 
 		assertThat(standardHeaders.size()).isEqualTo(1);
@@ -245,4 +243,10 @@
 		assertThat(soapMessage.getSoapAction()).isEqualTo(testSoapAction);
 	}
 
+
+    private static SoapMessage createMockSoapMessage(String soapAction) {
+        SoapMessage soapMessage = mock(SoapMessage.class);
+        when(soapMessage.getSoapAction()).thenReturn(soapAction);
+        return soapMessage;
+    }
 }
