--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ws.dsl;
 
+import org.mockito.Mock;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -54,11 +55,12 @@
  *
  */
 public class WsDslTests {
+	@Mock
+	private Unmarshaller unmarshaller;
 
 	@Test
 	void marshallingInbound() {
 		Marshaller marshaller = mock();
-		Unmarshaller unmarshaller = mock();
 		MarshallingWebServiceInboundGateway gateway = Ws.marshallingInboundGateway(marshaller)
 				.unmarshaller(unmarshaller)
 				.getObject();
@@ -89,7 +91,6 @@
 	void marshallingOutbound() {
 		DestinationProvider destinationProvider = mock();
 		Marshaller marshaller = mock();
-		Unmarshaller unmarshaller = mock();
 		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
