--- a/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
+++ b/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
@@ -200,11 +200,9 @@
 	@Test
 	void testClientManagerIsNotConnectedAndClosedInHandler() throws Exception {
 		// given
-		var clientManager = mock(Mqttv3ClientManager.class);
-		when(clientManager.getConnectionInfo()).thenReturn(new MqttConnectOptions());
 		var client = mock(MqttAsyncClient.class);
-		given(clientManager.getClient()).willReturn(client);
-
+
+		var clientManager = createMqttv3ClientManager(client);
 		var deliveryToken = mock(MqttDeliveryToken.class);
 		given(client.publish(anyString(), any(), any(), any())).willReturn(deliveryToken);
 
@@ -229,11 +227,9 @@
 	@Test
 	void testClientManagerIsNotConnectedAndClosedInAdapter() throws Exception {
 		// given
-		var clientManager = mock(Mqttv3ClientManager.class);
-		when(clientManager.getConnectionInfo()).thenReturn(new MqttConnectOptions());
 		var client = mock(MqttAsyncClient.class);
-		given(clientManager.getClient()).willReturn(client);
-
+
+		var clientManager = createMqttv3ClientManager(client);
 		var subscribeToken = mock(MqttToken.class);
 		given(subscribeToken.getGrantedQos()).willReturn(new int[] {2});
 		given(client.subscribe(any(String[].class), any(int[].class), any()))
@@ -638,4 +634,11 @@
 
 	}
 
+
+	private static Mqttv3ClientManager createMqttv3ClientManager(MqttAsyncClient client) {
+		Mqttv3ClientManager clientManager = mock(Mqttv3ClientManager.class);
+		when(clientManager.getClient()).thenReturn(client);
+		when(clientManager.getConnectionInfo()).thenReturn(new MqttConnectOptions());
+		return clientManager;
+	}
 }
