--- a/spring-integration-websocket/src/test/java/org/springframework/integration/websocket/support/SubProtocolHandlerRegistryTests.java
+++ b/spring-integration-websocket/src/test/java/org/springframework/integration/websocket/support/SubProtocolHandlerRegistryTests.java
@@ -47,8 +47,7 @@
 				new SubProtocolHandlerRegistry(
 						Collections.singletonList(new StompSubProtocolHandler()),
 						defaultProtocolHandler);
-		WebSocketSession session = mock(WebSocketSession.class);
-		when(session.getAcceptedProtocol()).thenReturn("v10.stomp", (String) null);
+		WebSocketSession session = createMockWebSocketSession("v10.stomp");
 		SubProtocolHandler protocolHandler = subProtocolHandlerRegistry.findProtocolHandler(session);
 		assertThat(protocolHandler).isNotNull();
 		assertThat(protocolHandler).isInstanceOf(StompSubProtocolHandler.class);
@@ -66,8 +65,7 @@
 		when(testProtocolHandler.getSupportedProtocols()).thenReturn(Collections.singletonList("foo"));
 		SubProtocolHandlerRegistry subProtocolHandlerRegistry =
 				new SubProtocolHandlerRegistry(Collections.singletonList(testProtocolHandler));
-		WebSocketSession session = mock(WebSocketSession.class);
-		when(session.getAcceptedProtocol()).thenReturn("foo", (String) null);
+		WebSocketSession session = createMockWebSocketSession("foo");
 		SubProtocolHandler protocolHandler = subProtocolHandlerRegistry.findProtocolHandler(session);
 		assertThat(protocolHandler).isNotNull();
 		assertThat(testProtocolHandler).isSameAs(protocolHandler);
@@ -118,4 +116,10 @@
 		assertThat(sessionId).isNull();
 	}
 
+
+	private static WebSocketSession createMockWebSocketSession(String acceptedProtocol) {
+		WebSocketSession session = mock(WebSocketSession.class);
+		when(session.getAcceptedProtocol()).thenReturn(acceptedProtocol, (String) null);
+		return session;
+	}
 }
