--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.ws.dsl;
 
+import org.mockito.Mock;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -54,6 +55,8 @@
  *
  */
 public class WsDslTests {
+	@Mock
+	private WebServiceMessageSender messageSender;
 
 	@Test
 	void marshallingInbound() {
@@ -94,7 +97,6 @@
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
-		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
@@ -137,7 +139,6 @@
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
-		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
