--- a/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/outbound/PresenceSendingMessageHandlerTests.java
+++ b/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/outbound/PresenceSendingMessageHandlerTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.xmpp.outbound;
 
+import org.mockito.Mock;
 import org.jivesoftware.smack.XMPPConnection;
 import org.jivesoftware.smack.packet.StanzaBuilder;
 import org.junit.jupiter.api.Test;
@@ -40,10 +41,12 @@
  * @author Glenn Renfro
  */
 public class PresenceSendingMessageHandlerTests {
+	@Mock
+	private XMPPConnection xmppConnection;
 
 	@Test
 	public void testPresencePayload() {
-		PresenceSendingMessageHandler handler = new PresenceSendingMessageHandler(mock(XMPPConnection.class));
+		PresenceSendingMessageHandler handler = new PresenceSendingMessageHandler(xmppConnection);
 		handler.setBeanFactory(mock(BeanFactory.class));
 		handler.afterPropertiesSet();
 		handler.handleMessage(new GenericMessage<>(StanzaBuilder.buildPresence().build()));
@@ -51,7 +54,7 @@
 
 	@Test
 	public void testWrongPayload() {
-		PresenceSendingMessageHandler handler = new PresenceSendingMessageHandler(mock(XMPPConnection.class));
+		PresenceSendingMessageHandler handler = new PresenceSendingMessageHandler(xmppConnection);
 		handler.setBeanFactory(mock(BeanFactory.class));
 		handler.afterPropertiesSet();
 		assertThatExceptionOfType(MessageHandlingException.class)
