--- a/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/PresenceListeningEndpointTests.java
+++ b/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/PresenceListeningEndpointTests.java
@@ -16,6 +16,7 @@
 
 package org.springframework.integration.xmpp.inbound;
 
+import org.mockito.Mock;
 import java.util.Set;
 
 import org.jivesoftware.smack.XMPPConnection;
@@ -49,10 +50,11 @@
  * @author Glenn Renfro
  */
 public class PresenceListeningEndpointTests {
+	@Mock
+	private XMPPConnection connection;
 
 	@Test
 	public void testEndpointLifecycle() {
-		XMPPConnection connection = mock(XMPPConnection.class);
 		Roster roster = Roster.getInstanceFor(connection);
 
 		Set<RosterListener> rosterSet = TestUtils.getPropertyValue(roster, "rosterListeners");
@@ -70,14 +72,13 @@
 
 	@Test
 	public void testNonInitializedFailure() {
-		PresenceListeningEndpoint rosterEndpoint = new PresenceListeningEndpoint(mock(XMPPConnection.class));
+		PresenceListeningEndpoint rosterEndpoint = new PresenceListeningEndpoint(connection);
 		assertThatIllegalArgumentException()
 				.isThrownBy(rosterEndpoint::start);
 	}
 
 	@Test
 	public void testRosterPresenceChangeEvent() {
-		XMPPConnection connection = mock(XMPPConnection.class);
 		PresenceListeningEndpoint rosterEndpoint = new PresenceListeningEndpoint(connection);
 		QueueChannel channel = new QueueChannel();
 		rosterEndpoint.setOutputChannel(channel);
@@ -117,7 +118,6 @@
 	@Test
 	public void testWithErrorChannel() {
 		DefaultListableBeanFactory bf = new DefaultListableBeanFactory();
-		XMPPConnection connection = mock(XMPPConnection.class);
 		bf.registerSingleton(XmppContextUtils.XMPP_CONNECTION_BEAN_NAME, connection);
 
 		PresenceListeningEndpoint endpoint = new PresenceListeningEndpoint();
