--- a/processing/src/test/java/org/apache/druid/java/util/metrics/OshiSysMonitorTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/OshiSysMonitorTest.java
@@ -262,22 +262,8 @@
   public void testDiskStats()
   {
     StubServiceEmitter emitter = new StubServiceEmitter("dev/monitor-test", "localhost:0000");
-    HWDiskStore disk1 = Mockito.mock(HWDiskStore.class);
-    HWDiskStore disk2 = Mockito.mock(HWDiskStore.class);
-    Mockito.when(disk1.getReadBytes()).thenReturn(300L);
-    Mockito.when(disk1.getReads()).thenReturn(200L);
-    Mockito.when(disk1.getWriteBytes()).thenReturn(400L);
-    Mockito.when(disk1.getWrites()).thenReturn(500L);
-    Mockito.when(disk1.getCurrentQueueLength()).thenReturn(100L);
-    Mockito.when(disk1.getTransferTime()).thenReturn(150L);
-    Mockito.when(disk1.getName()).thenReturn("disk1");
-    Mockito.when(disk2.getReadBytes()).thenReturn(2000L);
-    Mockito.when(disk2.getReads()).thenReturn(3000L);
-    Mockito.when(disk2.getWriteBytes()).thenReturn(1000L);
-    Mockito.when(disk2.getWrites()).thenReturn(4000L);
-    Mockito.when(disk2.getCurrentQueueLength()).thenReturn(750L);
-    Mockito.when(disk2.getTransferTime()).thenReturn(800L);
-    Mockito.when(disk2.getName()).thenReturn("disk2");
+    HWDiskStore disk1 = createMockHWDiskStore(300L, 200L, 400L, 500L, 100L, 150L, "disk1");
+    HWDiskStore disk2 = createMockHWDiskStore(2000L, 3000L, 1000L, 4000L, 750L, 800L, "disk2");
     List<HWDiskStore> hwDiskStores = ImmutableList.of(disk1, disk2);
     Mockito.when(hal.getDiskStores()).thenReturn(hwDiskStores);
 
@@ -649,4 +635,25 @@
   {
     return new OshiSysMonitor(new OshiSysMonitorConfig(categories), si);
   }
+
+  private static HWDiskStore createMockHWDiskStore(
+      long readBytes,
+      long reads,
+      long writeBytes,
+      long writes,
+      long currentQueueLength,
+      long transferTime,
+      String name
+  )
+  {
+    HWDiskStore diskStore = Mockito.mock(HWDiskStore.class);
+    Mockito.when(diskStore.getReadBytes()).thenReturn(readBytes);
+    Mockito.when(diskStore.getReads()).thenReturn(reads);
+    Mockito.when(diskStore.getWriteBytes()).thenReturn(writeBytes);
+    Mockito.when(diskStore.getWrites()).thenReturn(writes);
+    Mockito.when(diskStore.getCurrentQueueLength()).thenReturn(currentQueueLength);
+    Mockito.when(diskStore.getTransferTime()).thenReturn(transferTime);
+    Mockito.when(diskStore.getName()).thenReturn(name);
+    return diskStore;
+  }
 }
