--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -40,12 +40,10 @@
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
-    final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
+    final FrameContext frameContext = createMockFrameContext(storageParameters);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
-
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
@@ -76,12 +74,10 @@
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
-    final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
+    final FrameContext frameContext = createMockFrameContext(storageParameters);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
-
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
@@ -117,13 +113,12 @@
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
-    final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    final FrameContext frameContext = createMockFrameContext(storageParameters);
     Mockito.doThrow(exception).when(exec).cancel(CANCELLATION_ID);
 
     final RunWorkOrder runWorkOrder =
@@ -155,13 +150,12 @@
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
-    final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
-    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    final FrameContext frameContext = createMockFrameContext(storageParameters);
     Mockito.doThrow(exception).when(frameContext).close();
 
     final RunWorkOrder runWorkOrder =
@@ -185,4 +179,11 @@
     Mockito.verify(frameContext).close();
     Mockito.verify(listener).onFailure(ArgumentMatchers.eq(exception));
   }
+
+  private static FrameContext createMockFrameContext(final WorkerStorageParameters storageParameters)
+  {
+    final FrameContext frameContext = Mockito.mock(FrameContext.class);
+    Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
+    return frameContext;
+  }
 }
