--- a/server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java
+++ b/server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java
@@ -196,6 +196,7 @@
   private final List<DataSegment> allSegments = new ArrayList<>();
   private DataSourcesSnapshot dataSources;
   private CompactionStatusTracker statusTracker;
+  private final OverlordClient mockOverlordClient = Mockito.mock(OverlordClient.class);
   private final Map<String, List<DataSegment>> datasourceToSegments = new HashMap<>();
 
   public CompactSegmentsTest(
@@ -769,9 +770,8 @@
   @Test
   public void testCompactWithoutGranularitySpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -800,9 +800,8 @@
   @Test
   public void testCompactWithNotNullIOConfig()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -824,9 +823,8 @@
   @Test
   public void testCompactWithNullIOConfig()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -851,9 +849,8 @@
   @Test
   public void testCompactWithGranularitySpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -888,9 +885,8 @@
   @Test
   public void testCompactWithDimensionSpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -919,9 +915,8 @@
   @Test
   public void testCompactWithoutDimensionSpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -942,9 +937,8 @@
   @Test
   public void testCompactWithProjections()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     final List<AggregateProjectionSpec> projections = List.of(
@@ -988,7 +982,6 @@
   public void testCompactWithCatalogProjections()
   {
     final String dataSource = DATA_SOURCE_PREFIX + 0;
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
     final ObjectMapper mapper = new DefaultObjectMapper((DefaultObjectMapper) JSON_MAPPER);
     final MapMetadataCatalog metadataCatalog = new MapMetadataCatalog(mapper);
 
@@ -998,7 +991,7 @@
             .addValue(ExprMacroTable.class, TestExprMacroTable.INSTANCE)
     );
 
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
     final AggregateProjectionSpec projectionSpec =
         AggregateProjectionSpec.builder(dataSource + "_projection")
                                .virtualColumns(
@@ -1019,7 +1012,7 @@
                     )
                     .buildSpec()
     );
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     compactionConfigs.add(
         new CatalogDataSourceCompactionConfig(
@@ -1043,9 +1036,8 @@
   @Test
   public void testCompactWithRollupInGranularitySpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -1283,9 +1275,8 @@
   @Test
   public void testCompactWithTransformSpec()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -1313,9 +1304,8 @@
   @Test
   public void testCompactWithoutCustomSpecs()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -1338,9 +1328,8 @@
   public void testCompactWithMetricsSpec()
   {
     AggregatorFactory[] aggregatorFactories = new AggregatorFactory[] {new CountAggregatorFactory("cnt")};
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -1394,9 +1383,8 @@
     );
     dataSources = DataSourcesSnapshot.fromUsedSegments(segments);
 
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     compactionConfigs.add(
         InlineSchemaDataSourceCompactionConfig.builder()
@@ -1454,9 +1442,8 @@
     );
     dataSources = DataSourcesSnapshot.fromUsedSegments(segments);
 
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     compactionConfigs.add(
         InlineSchemaDataSourceCompactionConfig.builder()
@@ -1491,9 +1478,8 @@
   @Test
   public void testCompactWithMetricsSpecShouldSetPreserveExistingMetricsTrue()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
@@ -1518,9 +1504,8 @@
   @Test
   public void testCompactWithoutMetricsSpecShouldSetPreserveExistingMetricsFalse()
   {
-    final OverlordClient mockClient = Mockito.mock(OverlordClient.class);
-    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockClient);
-    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockClient);
+    final ArgumentCaptor<Object> payloadCaptor = setUpMockClient(mockOverlordClient);
+    final CompactSegments compactSegments = new CompactSegments(statusTracker, mockOverlordClient);
     final List<DataSourceCompactionConfig> compactionConfigs = new ArrayList<>();
     final String dataSource = DATA_SOURCE_PREFIX + 0;
     compactionConfigs.add(
