--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/task/AbstractTaskTest.java
@@ -55,6 +55,7 @@
 public class AbstractTaskTest
 {
   private ObjectMapper objectMapper;
+  private TaskLogPusher taskLogPusher;
 
   @Rule
   public TemporaryFolder temporaryFolder = new TemporaryFolder();
@@ -63,6 +64,7 @@
   public void setup()
   {
     objectMapper = new TestUtils().getTestObjectMapper();
+    taskLogPusher = mock(TaskLogPusher.class);
   }
 
   @Test
@@ -78,8 +80,7 @@
     DruidNode node = new DruidNode("foo", "foo", false, 1, 2, true, true);
     when(toolbox.getTaskExecutorNode()).thenReturn(node);
 
-    TaskLogPusher pusher = mock(TaskLogPusher.class);
-    when(toolbox.getTaskLogPusher()).thenReturn(pusher);
+    when(toolbox.getTaskLogPusher()).thenReturn(taskLogPusher);
 
     TaskConfig config = mock(TaskConfig.class);
     when(config.isEncapsulatedTask()).thenReturn(true);
@@ -114,8 +115,8 @@
 
     // call it 3 times, once to update location in setup, then one for status and location in cleanup
     Mockito.verify(taskActionClient, times(3)).submit(any());
-    verify(pusher, times(1)).pushTaskReports(eq("myID"), any());
-    verify(pusher, times(1)).pushTaskStatus(eq("myID"), any());
+    verify(taskLogPusher, times(1)).pushTaskReports(eq("myID"), any());
+    verify(taskLogPusher, times(1)).pushTaskStatus(eq("myID"), any());
   }
 
   @Test
@@ -127,8 +128,7 @@
     DruidNode node = new DruidNode("foo", "foo", false, 1, 2, true, true);
     when(toolbox.getTaskExecutorNode()).thenReturn(node);
 
-    TaskLogPusher pusher = mock(TaskLogPusher.class);
-    when(toolbox.getTaskLogPusher()).thenReturn(pusher);
+    when(toolbox.getTaskLogPusher()).thenReturn(taskLogPusher);
 
     TaskConfig config = mock(TaskConfig.class);
     when(config.isEncapsulatedTask()).thenReturn(false);
@@ -160,7 +160,7 @@
 
     // encapsulated task is set to false, should never get called
     Mockito.verify(taskActionClient, never()).submit(any());
-    verify(pusher, never()).pushTaskReports(eq("myID"), any());
+    verify(taskLogPusher, never()).pushTaskReports(eq("myID"), any());
   }
 
   @Test
@@ -172,8 +172,7 @@
     DruidNode node = new DruidNode("foo", "foo", false, 1, 2, true, true);
     when(toolbox.getTaskExecutorNode()).thenReturn(node);
 
-    TaskLogPusher pusher = mock(TaskLogPusher.class);
-    when(toolbox.getTaskLogPusher()).thenReturn(pusher);
+    when(toolbox.getTaskLogPusher()).thenReturn(taskLogPusher);
 
     TaskConfig config = mock(TaskConfig.class);
     when(config.isEncapsulatedTask()).thenReturn(true);
@@ -211,8 +210,7 @@
     DruidNode node = new DruidNode("foo", "foo", false, 1, 2, true, true);
     when(toolbox.getTaskExecutorNode()).thenReturn(node);
 
-    TaskLogPusher pusher = mock(TaskLogPusher.class);
-    when(toolbox.getTaskLogPusher()).thenReturn(pusher);
+    when(toolbox.getTaskLogPusher()).thenReturn(taskLogPusher);
 
     TaskConfig config = mock(TaskConfig.class);
     when(config.isEncapsulatedTask()).thenReturn(true);
@@ -248,8 +246,7 @@
     DruidNode node = new DruidNode("foo", "foo", false, 1, 2, true, true);
     when(toolbox.getTaskExecutorNode()).thenReturn(node);
 
-    TaskLogPusher pusher = mock(TaskLogPusher.class);
-    when(toolbox.getTaskLogPusher()).thenReturn(pusher);
+    when(toolbox.getTaskLogPusher()).thenReturn(taskLogPusher);
 
     TaskConfig config = mock(TaskConfig.class);
     when(config.isEncapsulatedTask()).thenReturn(true);
