--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -33,13 +33,14 @@
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
 
+  private final WorkerContext mockWorkerContext = Mockito.mock(WorkerContext.class);
+
   @Test
   public void test_stop()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -54,7 +55,7 @@
             counters,
             exec,
             CANCELLATION_ID,
-            workerContext,
+            mockWorkerContext,
             frameContext,
             listener
         );
@@ -75,7 +76,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -90,7 +90,7 @@
             counters,
             exec,
             CANCELLATION_ID,
-            workerContext,
+            mockWorkerContext,
             frameContext,
             listener
         );
@@ -116,7 +116,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -133,7 +132,7 @@
             counters,
             exec,
             CANCELLATION_ID,
-            workerContext,
+            mockWorkerContext,
             frameContext,
             listener
         );
@@ -154,7 +153,6 @@
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
-    final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
     final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
@@ -171,7 +169,7 @@
             counters,
             exec,
             CANCELLATION_ID,
-            workerContext,
+            mockWorkerContext,
             frameContext,
             listener
         );
