--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -33,6 +33,8 @@
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
 
+  private final InputChannelFactory sharedInputChannelFactory = Mockito.mock(InputChannelFactory.class);
+
   @Test
   public void test_stop()
   {
@@ -42,7 +44,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
@@ -50,7 +51,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            sharedInputChannelFactory,
             counters,
             exec,
             CANCELLATION_ID,
@@ -78,7 +79,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
@@ -86,7 +86,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            sharedInputChannelFactory,
             counters,
             exec,
             CANCELLATION_ID,
@@ -119,7 +119,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
@@ -129,7 +128,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            sharedInputChannelFactory,
             counters,
             exec,
             CANCELLATION_ID,
@@ -157,7 +156,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
@@ -167,7 +165,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            sharedInputChannelFactory,
             counters,
             exec,
             CANCELLATION_ID,
