--- a/server/src/test/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCacheTest.java
+++ b/server/src/test/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCacheTest.java
@@ -1899,10 +1899,9 @@
         segmentsMetadataManagerConfigSupplier
     );
 
-    SegmentReplicaCount zeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
-    SegmentReplicaCount nonZeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
+    SegmentReplicaCount zeroSegmentReplicaCount = createMockNonZeroSegmentReplicaCount();
     Mockito.when(zeroSegmentReplicaCount.required()).thenReturn(0);
-    Mockito.when(nonZeroSegmentReplicaCount.required()).thenReturn(1);
+    SegmentReplicaCount nonZeroSegmentReplicaCount = createMockNonZeroSegmentReplicaCount();
     SegmentReplicationStatus segmentReplicationStatus = Mockito.mock(SegmentReplicationStatus.class);
     Mockito.when(segmentReplicationStatus.getReplicaCountsInCluster(ArgumentMatchers.eq(coldSegment.getId())))
            .thenReturn(zeroSegmentReplicaCount);
@@ -2118,10 +2117,9 @@
         segmentsMetadataManagerConfigSupplier
     );
 
-    SegmentReplicaCount zeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
-    SegmentReplicaCount nonZeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
+    SegmentReplicaCount zeroSegmentReplicaCount = createMockNonZeroSegmentReplicaCount();
+    SegmentReplicaCount nonZeroSegmentReplicaCount = createMockNonZeroSegmentReplicaCount();
     Mockito.when(zeroSegmentReplicaCount.required()).thenReturn(0);
-    Mockito.when(nonZeroSegmentReplicaCount.required()).thenReturn(1);
     SegmentReplicationStatus segmentReplicationStatus = Mockito.mock(SegmentReplicationStatus.class);
     Mockito.when(segmentReplicationStatus.getReplicaCountsInCluster(ArgumentMatchers.eq(coldSegmentAlpha.getId())))
            .thenReturn(zeroSegmentReplicaCount);
@@ -2455,4 +2453,11 @@
     Assert.assertEquals("m1", columnNames.get(2));
     Assert.assertEquals(ColumnType.LONG, fooRowSignature.getColumnType(columnNames.get(2)).get());
   }
+
+  private static SegmentReplicaCount createMockNonZeroSegmentReplicaCount()
+  {
+    SegmentReplicaCount nonZeroSegmentReplicaCount = Mockito.mock(SegmentReplicaCount.class);
+    Mockito.when(nonZeroSegmentReplicaCount.required()).thenReturn(1);
+    return nonZeroSegmentReplicaCount;
+  }
 }
