--- a/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
+++ b/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
@@ -486,9 +486,7 @@
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
     final Request proxyRequestMock = Mockito.mock(Request.class);
-    final Response responseMock = Mockito.mock(Response.class);
-    Mockito.when(responseMock.getStatus()).thenReturn(504); // Gateway Timeout
-    Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
+    final Response responseMock = createMockResponse(504);
     Mockito.when(responseMock.getRequest()).thenReturn(proxyRequestMock);
 
     final Result result = new Result(proxyRequestMock, responseMock)
@@ -550,9 +548,7 @@
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
     final Request proxyRequestMock = Mockito.mock(Request.class);
-    final Response responseMock = Mockito.mock(Response.class);
-    Mockito.when(responseMock.getStatus()).thenReturn(200); // Status OK
-    Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
+    final Response responseMock = createMockResponse(200); // Status OK
     Mockito.when(responseMock.getRequest()).thenReturn(proxyRequestMock);
 
     // Result where connection is closed prematurely
@@ -620,9 +616,7 @@
     Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
-    final Response responseMock = Mockito.mock(Response.class);
-    Mockito.when(responseMock.getStatus()).thenReturn(0); // Test unassigned http status code case from server
-    Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
+    final Response responseMock = createMockResponse(0); // Test unassigned http status code case from server
 
     final StubServiceEmitter stubServiceEmitter = StubServiceEmitter.createStarted();
     final AsyncQueryForwardingServlet servlet = new AsyncQueryForwardingServlet(
@@ -721,6 +715,14 @@
                       )
                   );
     verifyServletCallsForQuery(query, false, false, hostFinder, new Properties(), false, jsonMapper);
+  }
+
+  private static Response createMockResponse(int status)
+  {
+    final Response responseMock = Mockito.mock(Response.class);
+    Mockito.when(responseMock.getStatus()).thenReturn(status);
+    Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
+    return responseMock;
   }
 
   private void verifyServletCallsForQuery(
