--- a/services/src/test/java/org/apache/druid/cli/DumpSegmentTest.java
+++ b/services/src/test/java/org/apache/druid/cli/DumpSegmentTest.java
@@ -125,7 +125,6 @@
   @Test
   public void testDumpRows() throws Exception
   {
-    Injector injector = Mockito.mock(Injector.class);
     ObjectMapper mapper = TestHelper.makeJsonMapper();
     mapper.registerModules(BuiltInTypesModule.getJacksonModulesList());
     mapper.setInjectableValues(
@@ -134,8 +133,7 @@
             .addValue(ObjectMapper.class.getName(), mapper)
             .addValue(DefaultColumnFormatConfig.class, new DefaultColumnFormatConfig(null, null, null, null))
     );
-    Mockito.when(injector.getInstance(Key.get(ObjectMapper.class, Json.class))).thenReturn(mapper);
-    Mockito.when(injector.getInstance(DefaultColumnFormatConfig.class)).thenReturn(new DefaultColumnFormatConfig(null, null, null, null));
+    Injector injector = createMockInjector(mapper);
 
     List<Segment> segments = createSegments(tempFolder, closer);
     QueryableIndex queryableIndex = segments.get(0).as(QueryableIndex.class);
@@ -163,9 +161,9 @@
   @Test
   public void testDumpBitmap() throws IOException
   {
-    Injector injector = Mockito.mock(Injector.class);
     QueryableIndex queryableIndex = Mockito.mock(QueryableIndex.class);
     ObjectMapper mapper = new DefaultObjectMapper();
+    Injector injector = createMockInjector(mapper);
     BitmapFactory bitmapFactory = new RoaringBitmapFactory();
     BaseColumnHolder xHolder = Mockito.mock(BaseColumnHolder.class);
     BaseColumnHolder yHolder = Mockito.mock(BaseColumnHolder.class);
@@ -173,7 +171,6 @@
     DictionaryEncodedStringValueIndex valueIndex = Mockito.mock(DictionaryEncodedStringValueIndex.class);
 
     ImmutableBitmap bitmap = bitmapFactory.complement(bitmapFactory.makeEmptyImmutableBitmap(), 10);
-    Mockito.when(injector.getInstance(Key.get(ObjectMapper.class, Json.class))).thenReturn(mapper);
 
     Mockito.when(queryableIndex.getBitmapFactoryForDimensions()).thenReturn(bitmapFactory);
 
@@ -199,7 +196,6 @@
   @Test
   public void testDumpNestedColumn() throws Exception
   {
-    Injector injector = Mockito.mock(Injector.class);
     ObjectMapper mapper = TestHelper.makeJsonMapper();
     mapper.registerModules(BuiltInTypesModule.getJacksonModulesList());
     mapper.setInjectableValues(
@@ -208,8 +204,7 @@
             .addValue(ObjectMapper.class.getName(), mapper)
             .addValue(DefaultColumnFormatConfig.class, new DefaultColumnFormatConfig(null, null, null, null))
     );
-    Mockito.when(injector.getInstance(Key.get(ObjectMapper.class, Json.class))).thenReturn(mapper);
-    Mockito.when(injector.getInstance(DefaultColumnFormatConfig.class)).thenReturn(new DefaultColumnFormatConfig(null, null, null, null));
+    Injector injector = createMockInjector(mapper);
 
     List<Segment> segments = createSegments(tempFolder, closer);
     QueryableIndex queryableIndex = segments.get(0).as(QueryableIndex.class);
@@ -232,7 +227,6 @@
   @Test
   public void testDumpNestedColumnPath() throws Exception
   {
-    Injector injector = Mockito.mock(Injector.class);
     ObjectMapper mapper = TestHelper.makeJsonMapper();
     mapper.registerModules(BuiltInTypesModule.getJacksonModulesList());
     mapper.setInjectableValues(
@@ -241,8 +235,7 @@
             .addValue(ObjectMapper.class.getName(), mapper)
             .addValue(DefaultColumnFormatConfig.class, new DefaultColumnFormatConfig(null, null, null, null))
     );
-    Mockito.when(injector.getInstance(Key.get(ObjectMapper.class, Json.class))).thenReturn(mapper);
-    Mockito.when(injector.getInstance(DefaultColumnFormatConfig.class)).thenReturn(new DefaultColumnFormatConfig(null, null, null, null));
+    Injector injector = createMockInjector(mapper);
 
     List<Segment> segments = createSegments(tempFolder, closer);
     QueryableIndex queryableIndex = segments.get(0).as(QueryableIndex.class);
@@ -281,7 +274,6 @@
   @Test
   public void testDumpV10Metadata() throws IOException
   {
-    Injector injector = Mockito.mock(Injector.class);
     ObjectMapper mapper = TestHelper.makeJsonMapper();
     mapper.registerModules(BuiltInTypesModule.getJacksonModulesList());
     mapper.setInjectableValues(
@@ -290,8 +282,7 @@
             .addValue(ObjectMapper.class.getName(), mapper)
             .addValue(DefaultColumnFormatConfig.class, new DefaultColumnFormatConfig(null, null, null, null))
     );
-    Mockito.when(injector.getInstance(Key.get(ObjectMapper.class, Json.class))).thenReturn(mapper);
-    Mockito.when(injector.getInstance(DefaultColumnFormatConfig.class)).thenReturn(new DefaultColumnFormatConfig(null, null, null, null));
+    Injector injector = createMockInjector(mapper);
 
     File f = buildV10Segment();
 
@@ -309,6 +300,12 @@
     Assert.assertEquals(12, dumped.getFiles().size());
   }
 
+
+  private static Injector createMockInjector(ObjectMapper mapper)
+  {
+    Injector injector = createMockInjector(mapper);
+    return injector;
+  }
 
   private File buildV10Segment() throws IOException
   {
