--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -32,13 +32,13 @@
 public class RunWorkOrderTest
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
+  private final CounterTracker counterTracker = Mockito.mock(CounterTracker.class);
 
   @Test
   public void test_stop()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -51,7 +51,7 @@
         new RunWorkOrder(
             workOrder,
             inputChannelFactory,
-            counters,
+            counterTracker,
             exec,
             CANCELLATION_ID,
             workerContext,
@@ -74,7 +74,6 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -87,7 +86,7 @@
         new RunWorkOrder(
             workOrder,
             inputChannelFactory,
-            counters,
+            counterTracker,
             exec,
             CANCELLATION_ID,
             workerContext,
@@ -115,7 +114,6 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -130,7 +128,7 @@
         new RunWorkOrder(
             workOrder,
             inputChannelFactory,
-            counters,
+            counterTracker,
             exec,
             CANCELLATION_ID,
             workerContext,
@@ -153,7 +151,6 @@
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
     final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
-    final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
@@ -168,7 +165,7 @@
         new RunWorkOrder(
             workOrder,
             inputChannelFactory,
-            counters,
+            counterTracker,
             exec,
             CANCELLATION_ID,
             workerContext,
