--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentArchiverTest.java
@@ -40,8 +40,7 @@
   public void testArchiveSegmentWithType() throws SegmentLoadingException
   {
     final DataSegmentArchiver archiver = Mockito.mock(DataSegmentArchiver.class);
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "sane"));
+    final DataSegment segment = createMockDataSegment();
 
     final Injector injector = createInjector(archiver);
     final DataSegmentArchiver segmentArchiver = injector.getInstance(OmniDataSegmentArchiver.class);
--- a/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/OmniDataSegmentMoverTest.java
@@ -42,8 +42,7 @@
   public void testMoveSegmentWithType() throws SegmentLoadingException
   {
     final DataSegmentMover mover = Mockito.mock(DataSegmentMover.class);
-    final DataSegment segment = Mockito.mock(DataSegment.class);
-    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "sane"));
+    final DataSegment segment = createMockDataSegment();
 
     final Injector injector = createInjector(mover);
     final OmniDataSegmentMover segmentMover = injector.getInstance(OmniDataSegmentMover.class);
--- a/server/src/test/java/org/apache/druid/segment/loading/MockDataSegment.java
+++ b/server/src/test/java/org/apache/druid/segment/loading/MockDataSegment.java
@@ -0,0 +1,16 @@
+package org.apache.druid.segment.loading;
+
+import org.apache.druid.timeline.DataSegment;
+import org.mockito.Mockito;
+
+import com.google.common.collect.ImmutableMap;
+
+public class MockDataSegment
+{
+  public static DataSegment createMockDataSegment()
+  {
+    final DataSegment segment = Mockito.mock(DataSegment.class);
+    Mockito.when(segment.getLoadSpec()).thenReturn(ImmutableMap.of("type", "sane"));
+    return segment;
+  }
+}
