--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -32,6 +32,7 @@
 public class RunWorkOrderTest
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
+  private final InputChannelFactory inputChannelFactoryMock = Mockito.mock(InputChannelFactory.class);
 
   @Test
   public void test_stop()
@@ -42,7 +43,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
@@ -50,7 +50,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            inputChannelFactoryMock,
             counters,
             exec,
             CANCELLATION_ID,
@@ -78,7 +78,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     Mockito.when(frameContext.storageParameters()).thenReturn(storageParameters);
@@ -86,7 +85,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            inputChannelFactoryMock,
             counters,
             exec,
             CANCELLATION_ID,
@@ -119,7 +118,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
@@ -129,7 +127,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            inputChannelFactoryMock,
             counters,
             exec,
             CANCELLATION_ID,
@@ -157,7 +155,6 @@
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
     final WorkerStorageParameters storageParameters = Mockito.mock(WorkerStorageParameters.class);
-    final InputChannelFactory inputChannelFactory = Mockito.mock(InputChannelFactory.class);
     final RunWorkOrderListener listener = Mockito.mock(RunWorkOrderListener.class);
 
     final ISE exception = new ISE("oops");
@@ -167,7 +164,7 @@
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
             workOrder,
-            inputChannelFactory,
+            inputChannelFactoryMock,
             counters,
             exec,
             CANCELLATION_ID,
