--- a/processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/HashVectorGrouperTest.java
+++ b/processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/HashVectorGrouperTest.java
@@ -97,8 +97,7 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
 
     int startingNumBuckets = 4;
     int maxBuckets = 16;
@@ -138,8 +137,7 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
 
     int startingNumBuckets = 4;
     int maxBuckets = 32;
@@ -185,8 +183,7 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
 
     int startingNumBuckets = 4;
     int maxBuckets = 64;
@@ -239,8 +236,7 @@
     final int aggSize = 8;
     final WritableMemory keySpace = WritableMemory.allocate(keySize * maxVectorSize);
 
-    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
-    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
+    final AggregatorAdapters aggregatorAdapters = createMockAggregatorAdapters(aggSize);
 
     int startingNumBuckets = 4;
     int maxBuckets = 128;
@@ -290,6 +286,13 @@
     result = grouper.aggregateVector(keySpace, 0, maxVectorSize);
     Assert.assertTrue(result.isOk());
     Assert.assertEquals(0, grouper.getTableStart());
+  }
+
+  private static AggregatorAdapters createMockAggregatorAdapters(int aggSize)
+  {
+    final AggregatorAdapters aggregatorAdapters = Mockito.mock(AggregatorAdapters.class);
+    Mockito.when(aggregatorAdapters.spaceNeeded()).thenReturn(aggSize);
+    return aggregatorAdapters;
   }
 
   private void fillKeyspace(WritableMemory keySpace, int maxVectorSize, int distinctKeys)
