--- a/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
+++ b/multi-stage-query/src/test/java/org/apache/druid/msq/exec/RunWorkOrderTest.java
@@ -32,12 +32,12 @@
 public class RunWorkOrderTest
 {
   private static final String CANCELLATION_ID = "my-cancellation-id";
+  private final WorkOrder workOrderMock = Mockito.mock(WorkOrder.class);
 
   @Test
   public void test_stop()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -49,7 +49,7 @@
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
-            workOrder,
+            workOrderMock,
             inputChannelFactory,
             counters,
             exec,
@@ -73,7 +73,6 @@
   public void test_stop_error()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -85,7 +84,7 @@
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
-            workOrder,
+            workOrderMock,
             inputChannelFactory,
             counters,
             exec,
@@ -114,7 +113,6 @@
   public void test_stop_errorDuringExecCancel()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -128,7 +126,7 @@
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
-            workOrder,
+            workOrderMock,
             inputChannelFactory,
             counters,
             exec,
@@ -152,7 +150,6 @@
   public void test_stop_errorDuringFrameContextClose()
   {
     final FrameProcessorExecutor exec = Mockito.mock(FrameProcessorExecutor.class);
-    final WorkOrder workOrder = Mockito.mock(WorkOrder.class);
     final CounterTracker counters = Mockito.mock(CounterTracker.class);
     final WorkerContext workerContext = Mockito.mock(WorkerContext.class);
     final FrameContext frameContext = Mockito.mock(FrameContext.class);
@@ -166,7 +163,7 @@
 
     final RunWorkOrder runWorkOrder =
         new RunWorkOrder(
-            workOrder,
+            workOrderMock,
             inputChannelFactory,
             counters,
             exec,
