--- a/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
+++ b/server/src/test/java/org/apache/druid/server/initialization/jetty/JettyServerModuleTest.java
@@ -33,6 +33,7 @@
 import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.eclipse.jetty.util.thread.QueuedThreadPool;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
@@ -41,6 +42,14 @@
 
 public class JettyServerModuleTest
 {
+  private TLSCertificateChecker tlsCertificateChecker;
+
+  @Before
+  public void setUp()
+  {
+    tlsCertificateChecker = Mockito.mock(TLSCertificateChecker.class);
+  }
+
   @Test
   public void testJettyServerModule()
   {
@@ -99,7 +108,6 @@
     ServerConfig serverConfig = new ServerConfig();
     Lifecycle lifecycle = new Lifecycle();
     Injector injector = Mockito.mock(Injector.class);
-    TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
     JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
@@ -111,7 +119,7 @@
         serverConfig,
         tlsServerConfig,
         null,
-        certificateChecker
+        tlsCertificateChecker
     );
 
     Assert.assertNotNull(server);
@@ -141,7 +149,6 @@
     ServerConfig serverConfig = new ServerConfig();
     Lifecycle lifecycle = new Lifecycle();
     Injector injector = Mockito.mock(Injector.class);
-    TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
     JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
@@ -164,7 +171,7 @@
         serverConfig,
         tlsServerConfig,
         sslContextFactoryBinding,
-        certificateChecker
+        tlsCertificateChecker
     );
 
     Assert.assertNotNull(server);
@@ -205,7 +212,6 @@
     ServerConfig serverConfig = new ServerConfig();
     Lifecycle lifecycle = new Lifecycle();
     Injector injector = Mockito.mock(Injector.class);
-    TLSCertificateChecker certificateChecker = Mockito.mock(TLSCertificateChecker.class);
 
     JettyServerInitializer initializer = Mockito.mock(JettyServerInitializer.class);
     Mockito.when(injector.getInstance(JettyServerInitializer.class)).thenReturn(initializer);
@@ -228,7 +234,7 @@
         serverConfig,
         tlsServerConfig,
         sslContextFactoryBinding,
-        certificateChecker
+        tlsCertificateChecker
     );
 
     Assert.assertNotNull(server);
