--- a/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunnerTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunnerTest.java
@@ -94,6 +94,9 @@
   @Mock
   private SeekableStreamIndexTask task;
 
+  @Mock
+  private SeekableStreamIndexTaskTuningConfig taskTuningConfig;
+
   private StubServiceEmitter emitter;
 
   @Before
@@ -121,7 +124,6 @@
                   )
                   .build();
 
-    SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
@@ -140,7 +142,7 @@
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
-    Mockito.when(task.getTuningConfig()).thenReturn(tuningConfig);
+    Mockito.when(task.getTuningConfig()).thenReturn(taskTuningConfig);
 
     TestSeekableStreamIndexTaskRunner runner = new TestSeekableStreamIndexTaskRunner(
         task,
@@ -176,7 +178,6 @@
                   )
                   .build();
 
-    SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
@@ -196,7 +197,7 @@
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
-    Mockito.when(task.getTuningConfig()).thenReturn(tuningConfig);
+    Mockito.when(task.getTuningConfig()).thenReturn(taskTuningConfig);
     TestSeekableStreamIndexTaskRunner runner = new TestSeekableStreamIndexTaskRunner(
         task,
         LockGranularity.TIME_CHUNK
@@ -231,7 +232,6 @@
                   )
                   .build();
 
-    SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
@@ -248,7 +248,7 @@
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
-    Mockito.when(task.getTuningConfig()).thenReturn(tuningConfig);
+    Mockito.when(task.getTuningConfig()).thenReturn(taskTuningConfig);
 
     TestSeekableStreamIndexTaskRunner runner = new TestSeekableStreamIndexTaskRunner(
         task,
@@ -277,7 +277,6 @@
                   )
                   .build();
 
-    SeekableStreamIndexTaskTuningConfig tuningConfig = Mockito.mock(SeekableStreamIndexTaskTuningConfig.class);
     SeekableStreamIndexTaskIOConfig<String, String> ioConfig = Mockito.mock(SeekableStreamIndexTaskIOConfig.class);
     SeekableStreamStartSequenceNumbers<String, String> sequenceNumbers = Mockito.mock(SeekableStreamStartSequenceNumbers.class);
     SeekableStreamEndSequenceNumbers<String, String> endSequenceNumbers = Mockito.mock(SeekableStreamEndSequenceNumbers.class);
@@ -292,7 +291,7 @@
 
     Mockito.when(task.getDataSchema()).thenReturn(schema);
     Mockito.when(task.getIOConfig()).thenReturn(ioConfig);
-    Mockito.when(task.getTuningConfig()).thenReturn(tuningConfig);
+    Mockito.when(task.getTuningConfig()).thenReturn(taskTuningConfig);
     Mockito.when(task.getId()).thenReturn("task1");
     Mockito.when(task.getSupervisorId()).thenReturn("supervisorId");
     TestSeekableStreamIndexTaskRunner runner = new TestSeekableStreamIndexTaskRunner(
