--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateLocationActionTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateLocationActionTest.java
@@ -66,9 +66,8 @@
     TaskLocation myLocation = TaskLocation.create(hostName.getHostAddress(), 1, 2);
     UpdateLocationAction action = new UpdateLocationAction(myLocation);
     Task task = NoopTask.create();
-    TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
+    TaskActionToolbox toolbox = createMockTaskActionToolbox();
     TaskRunner runner = mock(TaskRunner.class);
-    when(toolbox.getTaskRunner()).thenReturn(Optional.absent());
     action.perform(task, toolbox);
     verify(runner, never()).updateStatus(any(), any());
   }
--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateStatusActionTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/UpdateStatusActionTest.java
@@ -83,9 +83,8 @@
   {
     UpdateStatusAction action = new UpdateStatusAction("", TaskStatus.success(ID));
     Task task = NoopTask.create();
-    TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
+    TaskActionToolbox toolbox = createMockTaskActionToolbox();
     TaskRunner runner = mock(TaskRunner.class);
-    when(toolbox.getTaskRunner()).thenReturn(Optional.absent());
     action.perform(task, toolbox);
     verify(runner, never()).updateStatus(any(), any());
   }
--- a/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/MockTaskActionToolbox.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/common/actions/MockTaskActionToolbox.java
@@ -0,0 +1,16 @@
+package org.apache.druid.indexing.common.actions;
+
+import com.google.common.base.Optional;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class MockTaskActionToolbox
+{
+  public static TaskActionToolbox createMockTaskActionToolbox()
+  {
+    TaskActionToolbox toolbox = mock(TaskActionToolbox.class);
+    when(toolbox.getTaskRunner()).thenReturn(Optional.absent());
+    return toolbox;
+  }
+}
