--- a/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
+++ b/services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java
@@ -129,6 +129,7 @@
 {
   private static int port1;
   private static int port2;
+  private final Request proxyRequestMockField = Mockito.mock(Request.class);
 
   @Override
   @Before
@@ -485,13 +486,12 @@
     Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
-    final Request proxyRequestMock = Mockito.mock(Request.class);
     final Response responseMock = Mockito.mock(Response.class);
     Mockito.when(responseMock.getStatus()).thenReturn(504); // Gateway Timeout
     Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
-    Mockito.when(responseMock.getRequest()).thenReturn(proxyRequestMock);
-
-    final Result result = new Result(proxyRequestMock, responseMock)
+    Mockito.when(responseMock.getRequest()).thenReturn(proxyRequestMockField);
+
+    final Result result = new Result(proxyRequestMockField, responseMock)
     {
       @Override
       public Throwable getFailure()
@@ -549,14 +549,13 @@
     Mockito.when(requestMock.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT))
            .thenReturn(new AuthenticationResult("testUser", "basic", "basic", null));
 
-    final Request proxyRequestMock = Mockito.mock(Request.class);
     final Response responseMock = Mockito.mock(Response.class);
     Mockito.when(responseMock.getStatus()).thenReturn(200); // Status OK
     Mockito.when(responseMock.getHeaders()).thenReturn(HttpFields.build());
-    Mockito.when(responseMock.getRequest()).thenReturn(proxyRequestMock);
+    Mockito.when(responseMock.getRequest()).thenReturn(proxyRequestMockField);
 
     // Result where connection is closed prematurely
-    final Result result = new Result(proxyRequestMock, responseMock)
+    final Result result = new Result(proxyRequestMockField, responseMock)
     {
       @Override
       public boolean isSucceeded()
