--- a/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
+++ b/processing/src/test/java/org/apache/druid/java/util/metrics/ClockDriftSafeMonitorSchedulerTest.java
@@ -58,12 +58,14 @@
   private ExecutorService cronTaskRunner;
   @Mock
   private CronScheduler cronScheduler;
+  private ServiceEmitter serviceEmitter;
   private AutoCloseable mocks;
 
   @Before
   public void setUp()
   {
     cronTaskRunner = Execs.singleThreaded("monitor-scheduler-test");
+    serviceEmitter = Mockito.mock(ServiceEmitter.class);
     mocks = MockitoAnnotations.openMocks(this);
   }
 
@@ -94,7 +96,7 @@
 
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         Mockito.mock(DruidMonitorSchedulerConfig.class),
-        Mockito.mock(ServiceEmitter.class),
+        serviceEmitter,
         ImmutableList.of(monitor1, monitor2),
         CronScheduler.newBuilder(Duration.ofSeconds(1L)).setThreadName("monitor-scheduler-test").build(),
         executor
@@ -156,9 +158,10 @@
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
 
+    ServiceEmitter serviceEmitter = Mockito.mock(ServiceEmitter.class);
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
-        Mockito.mock(ServiceEmitter.class),
+        serviceEmitter,
         ImmutableList.of(monitor),
         cronScheduler,
         executor
@@ -220,9 +223,10 @@
     DruidMonitorSchedulerConfig config = Mockito.mock(DruidMonitorSchedulerConfig.class);
     Mockito.when(config.getEmissionDuration()).thenReturn(new org.joda.time.Duration(1000L));
 
+    ServiceEmitter serviceEmitter = Mockito.mock(ServiceEmitter.class);
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
-        Mockito.mock(ServiceEmitter.class),
+        serviceEmitter,
         ImmutableList.of(monitor),
         cronScheduler,
         executor
@@ -287,7 +291,7 @@
     
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
-        Mockito.mock(ServiceEmitter.class),
+        serviceEmitter,
         ImmutableList.of(monitor),
         cronScheduler,
         executor
@@ -338,7 +342,7 @@
     
     final MonitorScheduler scheduler = new ClockDriftSafeMonitorScheduler(
         config,
-        Mockito.mock(ServiceEmitter.class),
+        serviceEmitter,
         ImmutableList.of(monitor),
         cronScheduler,
         executor
