--- a/components/abstractions/src/test/java/com/microsoft/kiota/serialization/DeserializationHelpersTest.java
+++ b/components/abstractions/src/test/java/com/microsoft/kiota/serialization/DeserializationHelpersTest.java
@@ -33,6 +33,18 @@
 
     private static ParsableFactory<TestEntity> nullFactory() {
         return nullValue();
+    }
+
+    private static ParseNode createMockParseNode() {
+        final ParseNode mockParseNode = mock(ParseNode.class);
+        when(mockParseNode.getObjectValue(any()))
+                .thenReturn(
+                        new TestEntity() {
+                            {
+                                setId("123");
+                            }
+                        });
+        return mockParseNode;
     }
 
     @Test
@@ -88,14 +100,7 @@
     @Test
     void deserializesObjectWithoutReflection() throws IOException {
         final String strValue = "{'id':'123'}";
-        final ParseNode mockParseNode = mock(ParseNode.class);
-        when(mockParseNode.getObjectValue(any()))
-                .thenReturn(
-                        new TestEntity() {
-                            {
-                                setId("123");
-                            }
-                        });
+        final ParseNode mockParseNode = createMockParseNode();
         final ParseNodeFactory mockParseNodeFactory = mock(ParseNodeFactory.class);
         when(mockParseNodeFactory.getParseNode(any(), any())).thenReturn(mockParseNode);
         ParseNodeFactoryRegistry.defaultInstance.contentTypeAssociatedFactories.put(
@@ -110,14 +115,7 @@
     @Test
     void deserializesObjectWithReflection() throws IOException {
         final String strValue = "{'id':'123'}";
-        final ParseNode mockParseNode = mock(ParseNode.class);
-        when(mockParseNode.getObjectValue(any()))
-                .thenReturn(
-                        new TestEntity() {
-                            {
-                                setId("123");
-                            }
-                        });
+        final ParseNode mockParseNode = createMockParseNode();
         final ParseNodeFactory mockParseNodeFactory = mock(ParseNodeFactory.class);
         when(mockParseNodeFactory.getParseNode(any(), any())).thenReturn(mockParseNode);
         ParseNodeFactoryRegistry.defaultInstance.contentTypeAssociatedFactories.put(
