--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/HttpProcessHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/HttpProcessHandlerTest.java
@@ -26,6 +26,7 @@
 import io.netty.handler.codec.http.FullHttpResponse;
 import io.netty.handler.codec.http.HttpMethod;
 import io.netty.handler.codec.http.HttpRequest;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
@@ -38,17 +39,23 @@
 import static org.mockito.Mockito.when;
 
 class HttpProcessHandlerTest {
+    private ChannelFuture channelFuture;
+
+    @BeforeEach
+    void setUp() {
+        channelFuture = mock(ChannelFuture.class);
+    }
+
     @Test
     void test1() throws Exception {
         ChannelHandlerContext context = mock(ChannelHandlerContext.class);
-        ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(channelFuture);
         HttpRequest message = Mockito.mock(HttpRequest.class);
         when(message.uri()).thenReturn("test");
         HttpProcessHandler handler = new HttpProcessHandler(
                 FrameworkModel.defaultModel(), QosConfiguration.builder().build());
         handler.channelRead0(context, message);
-        verify(future).addListener(ChannelFutureListener.CLOSE);
+        verify(channelFuture).addListener(ChannelFutureListener.CLOSE);
         ArgumentCaptor<FullHttpResponse> captor = ArgumentCaptor.forClass(FullHttpResponse.class);
         verify(context).writeAndFlush(captor.capture());
         FullHttpResponse response = captor.getValue();
@@ -58,8 +65,7 @@
     @Test
     void test2() throws Exception {
         ChannelHandlerContext context = mock(ChannelHandlerContext.class);
-        ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(channelFuture);
         HttpRequest message = Mockito.mock(HttpRequest.class);
         when(message.uri()).thenReturn("localhost:80/greeting");
         when(message.method()).thenReturn(HttpMethod.GET);
@@ -69,7 +75,7 @@
                         .anonymousAccessPermissionLevel(PermissionLevel.NONE.name())
                         .build());
         handler.channelRead0(context, message);
-        verify(future).addListener(ChannelFutureListener.CLOSE);
+        verify(channelFuture).addListener(ChannelFutureListener.CLOSE);
         ArgumentCaptor<FullHttpResponse> captor = ArgumentCaptor.forClass(FullHttpResponse.class);
         verify(context).writeAndFlush(captor.capture());
         FullHttpResponse response = captor.getValue();
@@ -79,8 +85,7 @@
     @Test
     void test3() throws Exception {
         ChannelHandlerContext context = mock(ChannelHandlerContext.class);
-        ChannelFuture future = mock(ChannelFuture.class);
-        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(future);
+        when(context.writeAndFlush(any(FullHttpResponse.class))).thenReturn(channelFuture);
         HttpRequest message = Mockito.mock(HttpRequest.class);
         when(message.uri()).thenReturn("localhost:80/test");
         when(message.method()).thenReturn(HttpMethod.GET);
@@ -90,7 +95,7 @@
                         .anonymousAccessPermissionLevel(PermissionLevel.NONE.name())
                         .build());
         handler.channelRead0(context, message);
-        verify(future).addListener(ChannelFutureListener.CLOSE);
+        verify(channelFuture).addListener(ChannelFutureListener.CLOSE);
         ArgumentCaptor<FullHttpResponse> captor = ArgumentCaptor.forClass(FullHttpResponse.class);
         verify(context).writeAndFlush(captor.capture());
         FullHttpResponse response = captor.getValue();
