--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/ForeignHostPermitHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/ForeignHostPermitHandlerTest.java
@@ -94,9 +94,8 @@
 
     @Test
     void shouldNotShowIpNotPermittedMsg_GivenAcceptForeignIpFalseAndMatchWhiteList() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         Channel channel = mock(Channel.class);
-        when(context.channel()).thenReturn(channel);
+        ChannelHandlerContext context = createMockChannelHandlerContext(channel);
         InetAddress addr = mock(InetAddress.class);
         when(addr.isLoopbackAddress()).thenReturn(false);
         when(addr.getHostAddress()).thenReturn("175.23.44.1");
@@ -113,9 +112,8 @@
 
     @Test
     void shouldNotShowIpNotPermittedMsg_GivenAcceptForeignIpFalseAndMatchWhiteListRange() throws Exception {
-        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
         Channel channel = mock(Channel.class);
-        when(context.channel()).thenReturn(channel);
+        ChannelHandlerContext context = createMockChannelHandlerContext(channel);
         InetAddress addr = mock(InetAddress.class);
         when(addr.isLoopbackAddress()).thenReturn(false);
         when(addr.getHostAddress()).thenReturn("192.168.1.199");
@@ -147,4 +145,9 @@
         handler.handlerAdded(context);
         verify(future, never()).addListener(ChannelFutureListener.CLOSE);
     }
+    private static ChannelHandlerContext createMockChannelHandlerContext(Channel channel) {
+        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        when(context.channel()).thenReturn(channel);
+        return context;
+    }
 }
