--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/server/handler/QosProcessHandlerTest.java
@@ -41,9 +41,8 @@
     @Test
     void testDecodeHttp() throws Exception {
         ByteBuf buf = Unpooled.wrappedBuffer(new byte[] {'G'});
-        ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
         ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
-        Mockito.when(context.pipeline()).thenReturn(pipeline);
+        ChannelHandlerContext context = createMockChannelHandlerContext(pipeline);
         QosProcessHandler handler = new QosProcessHandler(
                 FrameworkModel.defaultModel(),
                 QosConfiguration.builder()
@@ -61,9 +60,8 @@
     @Test
     void testDecodeTelnet() throws Exception {
         ByteBuf buf = Unpooled.wrappedBuffer(new byte[] {'A'});
-        ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
         ChannelPipeline pipeline = Mockito.mock(ChannelPipeline.class);
-        Mockito.when(context.pipeline()).thenReturn(pipeline);
+        ChannelHandlerContext context = createMockChannelHandlerContext(pipeline);
         QosProcessHandler handler = new QosProcessHandler(
                 FrameworkModel.defaultModel(),
                 QosConfiguration.builder()
@@ -78,4 +76,9 @@
         verify(pipeline).addLast(any(TelnetProcessHandler.class));
         verify(pipeline).remove(handler);
     }
+    private static ChannelHandlerContext createMockChannelHandlerContext(ChannelPipeline pipeline) {
+        ChannelHandlerContext context = Mockito.mock(ChannelHandlerContext.class);
+        Mockito.when(context.pipeline()).thenReturn(pipeline);
+        return context;
+    }
 }
