--- a/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java
+++ b/dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java
@@ -100,9 +100,8 @@
             Mockito.when(migrationInvoker.getServiceDiscoveryInvoker()).thenReturn(serviceDiscoveryInvoker);
             Directory<?> sdDirectory = Mockito.mock(Directory.class);
             Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(sdDirectory);
-            List sdInvokers = Mockito.mock(List.class);
+            List sdInvokers = createMockSdInvokers(5);
             Mockito.when(sdDirectory.getAllInvokers()).thenReturn(sdInvokers);
-            Mockito.when(sdInvokers.size()).thenReturn(5);
             invokerMap.put(registry, migrationInvoker);
         }
 
@@ -115,17 +114,15 @@
             Mockito.when(migrationInvoker.getServiceDiscoveryInvoker()).thenReturn(serviceDiscoveryInvoker);
             Directory<?> sdDirectory = Mockito.mock(Directory.class);
             Mockito.when(serviceDiscoveryInvoker.getDirectory()).thenReturn(sdDirectory);
-            List sdInvokers = Mockito.mock(List.class);
+            List sdInvokers = createMockSdInvokers(0);
             Mockito.when(sdDirectory.getAllInvokers()).thenReturn(sdInvokers);
-            Mockito.when(sdInvokers.size()).thenReturn(0);
 
             ClusterInvoker invoker = Mockito.mock(ClusterInvoker.class);
             Mockito.when(migrationInvoker.getInvoker()).thenReturn(invoker);
             Directory<?> directory = Mockito.mock(Directory.class);
             Mockito.when(invoker.getDirectory()).thenReturn(directory);
-            List invokers = Mockito.mock(List.class);
+            List invokers = createMockSdInvokers(10);
             Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
-            Mockito.when(invokers.size()).thenReturn(10);
             invokerMap.put(registry, migrationInvoker);
         }
 
@@ -138,9 +135,8 @@
             Mockito.when(migrationInvoker.getInvoker()).thenReturn(invoker);
             Directory<?> directory = Mockito.mock(Directory.class);
             Mockito.when(invoker.getDirectory()).thenReturn(directory);
-            List invokers = Mockito.mock(List.class);
+            List invokers = createMockSdInvokers(10);
             Mockito.when(directory.getAllInvokers()).thenReturn(invokers);
-            Mockito.when(invokers.size()).thenReturn(10);
             invokerMap.put(registry, migrationInvoker);
         }
 
@@ -150,4 +146,10 @@
                 "zookeeper-A(5)/zookeeper-AF(I-10,A-0)/nacos-I(10)",
                 ServiceCheckUtils.getConsumerAddressNum(consumerModel));
     }
+
+    private static List createMockSdInvokers(int size) {
+        List sdInvokers = Mockito.mock(List.class);
+        Mockito.when(sdInvokers.size()).thenReturn(size);
+        return sdInvokers;
+    }
 }
