--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/FutureFilterTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/FutureFilterTest.java
@@ -52,10 +52,7 @@
 
     @Test
     void testSyncCallback() {
-        @SuppressWarnings("unchecked")
-        Invoker<DemoService> invoker = mock(Invoker.class);
-        given(invoker.isAvailable()).willReturn(true);
-        given(invoker.getInterface()).willReturn(DemoService.class);
+        Invoker<DemoService> invoker = createMockInvoker();
         AppResponse result = new AppResponse();
         result.setValue("High");
         given(invoker.invoke(invocation)).willReturn(result);
@@ -69,10 +66,7 @@
     @Test
     void testSyncCallbackHasException() throws RpcException, Throwable {
         Assertions.assertThrows(RuntimeException.class, () -> {
-            @SuppressWarnings("unchecked")
-            Invoker<DemoService> invoker = mock(Invoker.class);
-            given(invoker.isAvailable()).willReturn(true);
-            given(invoker.getInterface()).willReturn(DemoService.class);
+            Invoker<DemoService> invoker = createMockInvoker();
             AppResponse result = new AppResponse();
             result.setException(new RuntimeException());
             given(invoker.invoke(invocation)).willReturn(result);
@@ -82,4 +76,12 @@
             eventFilter.invoke(invoker, invocation).recreate();
         });
     }
+
+    @SuppressWarnings("unchecked")
+    private static Invoker<DemoService> createMockInvoker() {
+        Invoker<DemoService> invoker = mock(Invoker.class);
+        given(invoker.isAvailable()).willReturn(true);
+        given(invoker.getInterface()).willReturn(DemoService.class);
+        return invoker;
+    }
 }
