--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelectorTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/router/mock/MockInvokersSelectorTest.java
@@ -38,11 +38,9 @@
         MockInvokersSelector selector = new MockInvokersSelector(URL.valueOf(""));
 
         // Data preparation
-        Invoker<DemoService> invoker1 = Mockito.mock(Invoker.class);
-        Invoker<DemoService> invoker2 = Mockito.mock(Invoker.class);
-        Invoker<DemoService> invoker3 = Mockito.mock(Invoker.class);
-        Mockito.when(invoker1.getUrl()).thenReturn(URL.valueOf("mock://127.0.0.1/test"));
-        Mockito.when(invoker2.getUrl()).thenReturn(URL.valueOf("mock://127.0.0.1/test"));
+        Invoker<DemoService> invoker1 = createMockInvoker();
+        Invoker<DemoService> invoker2 = createMockInvoker();
+        Invoker<DemoService> invoker3 = createMockInvoker();
         Mockito.when(invoker3.getUrl()).thenReturn(URL.valueOf("dubbo://127.0.0.1/test"));
         BitList<Invoker<DemoService>> providers = new BitList<>(Arrays.asList(invoker1, invoker2, invoker3));
 
@@ -68,5 +66,11 @@
         Assertions.assertTrue(invokers.contains(invoker2));
     }
 
+    private static Invoker<DemoService> createMockInvoker() {
+        Invoker<DemoService> invoker = Mockito.mock(Invoker.class);
+        Mockito.when(invoker.getUrl()).thenReturn(URL.valueOf("mock://127.0.0.1/test"));
+        return invoker;
+    }
+
     class DemoService {}
 }
