--- a/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailbackClusterInvokerTest.java
+++ b/dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/support/FailbackClusterInvokerTest.java
@@ -71,11 +71,7 @@
     public void setUp() throws Exception {
         RpcContext.removeServiceContext();
 
-        dic = mock(Directory.class);
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
-        given(dic.list(invocation)).willReturn(invokers);
-        given(dic.getInterface()).willReturn(FailbackClusterInvokerTest.class);
+        dic = createMockDirectory(url, invocation, invokers);
 
         invocation.setMethodName("method1");
 
@@ -100,6 +96,12 @@
         given(invoker.invoke(invocation)).willReturn(result);
         given(invoker.getUrl()).willReturn(url);
         given(invoker.getInterface()).willReturn(FailbackClusterInvokerTest.class);
+    }
+
+    private static Directory<FailbackClusterInvokerTest> createMockDirectory(
+            URL url, RpcInvocation invocation, List<Invoker<FailbackClusterInvokerTest>> invokers) {
+        Directory<FailbackClusterInvokerTest> dic = createMockDirectory(url, invocation, invokers);
+        return dic;
     }
 
     @Test
@@ -121,11 +123,7 @@
     @Test
     void testInvokeWithIllegalFailbacktasksParam() {
         URL url = URL.valueOf("test://test:11/test?retries=2&failbacktasks=-1");
-        Directory<FailbackClusterInvokerTest> dic = mock(Directory.class);
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
-        given(dic.getInterface()).willReturn(FailbackClusterInvokerTest.class);
-        given(dic.list(invocation)).willReturn(invokers);
+        Directory<FailbackClusterInvokerTest> dic = createMockDirectory(url, invocation, invokers);
         given(invoker.getUrl()).willReturn(url);
 
         FailbackClusterInvoker<FailbackClusterInvokerTest> invoker = new FailbackClusterInvoker<>(dic);
@@ -158,12 +156,8 @@
     @Test
     @Order(3)
     public void testNoInvoke() {
-        dic = mock(Directory.class);
-
-        given(dic.getUrl()).willReturn(url);
-        given(dic.getConsumerUrl()).willReturn(url);
+        dic = createMockDirectory(url, invocation, invokers);
         given(dic.list(invocation)).willReturn(null);
-        given(dic.getInterface()).willReturn(FailbackClusterInvokerTest.class);
 
         invocation.setMethodName("method1");
         invokers.add(invoker);
@@ -221,6 +215,7 @@
             throws InterruptedException, NoSuchFieldException, IllegalAccessException {
         int retries = 0;
         resetInvokerToException();
+        dic = createMockDirectory(url, invocation, invokers);
         given(dic.getConsumerUrl()).willReturn(url.addParameter(RETRIES_KEY, retries));
 
         FailbackClusterInvoker<FailbackClusterInvokerTest> invoker = new FailbackClusterInvoker<>(dic);
@@ -246,6 +241,7 @@
             throws InterruptedException, NoSuchFieldException, IllegalAccessException {
         int retries = 2;
         resetInvokerToException();
+        dic = createMockDirectory(url, invocation, invokers);
         given(dic.getConsumerUrl()).willReturn(url.addParameter(RETRIES_KEY, retries));
 
         FailbackClusterInvoker<FailbackClusterInvokerTest> invoker = new FailbackClusterInvoker<>(dic);
@@ -294,6 +290,7 @@
     public void testInvokeRetryTimesWithIllegalValue()
             throws InterruptedException, NoSuchFieldException, IllegalAccessException {
         resetInvokerToException();
+        dic = createMockDirectory(url, invocation, invokers);
         given(dic.getConsumerUrl()).willReturn(url.addParameter(RETRIES_KEY, -100));
 
         FailbackClusterInvoker<FailbackClusterInvokerTest> invoker = new FailbackClusterInvoker<>(dic);
