--- a/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubSuppliersTest.java
@@ -20,6 +20,7 @@
 import org.apache.dubbo.rpc.model.ServiceDescriptor;
 
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -28,18 +29,22 @@
 class StubSuppliersTest {
 
     private final String serviceName = "service";
+    private ServiceDescriptor descriptorMock;
+
+    @BeforeEach
+    void setUp() {
+        descriptorMock = Mockito.mock(ServiceDescriptor.class);
+    }
 
     @Test
     void addDescriptor() {
-        ServiceDescriptor descriptor = Mockito.mock(ServiceDescriptor.class);
-        StubSuppliers.addDescriptor(serviceName, descriptor);
-        Assertions.assertEquals(descriptor, StubSuppliers.getServiceDescriptor(serviceName));
+        StubSuppliers.addDescriptor(serviceName, descriptorMock);
+        Assertions.assertEquals(descriptorMock, StubSuppliers.getServiceDescriptor(serviceName));
     }
 
     @Test
     void addSupplier() {
         Invoker<?> invoker = Mockito.mock(Invoker.class);
-        ServiceDescriptor descriptor = Mockito.mock(ServiceDescriptor.class);
         StubSuppliers.addSupplier(serviceName, i -> invoker);
         Assertions.assertEquals(invoker, StubSuppliers.createStub(serviceName, invoker));
     }
