--- a/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
+++ b/dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/integration/RegistryProtocolTest.java
@@ -41,6 +41,7 @@
 import java.util.Map;
 
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
@@ -65,6 +66,13 @@
 
 class RegistryProtocolTest {
 
+    private MigrationRuleListener migrationRuleListenerMock;
+
+    @BeforeEach
+    void setUp() {
+        migrationRuleListenerMock = mock(MigrationRuleListener.class);
+    }
+
     @AfterEach
     public void tearDown() throws IOException {
         Mockito.framework().clearInlineMocks();
@@ -105,9 +113,8 @@
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
-        registryProtocolListeners.add(migrationRuleListener);
+        registryProtocolListeners.add(migrationRuleListenerMock);
 
         ModuleModel moduleModel = Mockito.spy(ApplicationModel.defaultModel().getDefaultModule());
         moduleModel
@@ -173,9 +180,8 @@
         RegistryProtocol registryProtocol = new RegistryProtocol();
         Registry registry = mock(Registry.class);
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
-        registryProtocolListeners.add(migrationRuleListener);
+        registryProtocolListeners.add(migrationRuleListenerMock);
 
         ModuleModel moduleModel = Mockito.spy(ApplicationModel.defaultModel().getDefaultModule());
         moduleModel
@@ -240,9 +246,8 @@
         RegistryFactory registryFactory = mock(RegistryFactory.class);
         Registry registry = mock(Registry.class);
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
-        registryProtocolListeners.add(migrationRuleListener);
+        registryProtocolListeners.add(migrationRuleListenerMock);
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
         ModuleModel moduleModel = Mockito.spy(ApplicationModel.defaultModel().getDefaultModule());
@@ -307,9 +312,8 @@
         RegistryFactory registryFactory = mock(RegistryFactory.class);
         Registry registry = mock(Registry.class);
 
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
-        registryProtocolListeners.add(migrationRuleListener);
+        registryProtocolListeners.add(migrationRuleListenerMock);
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
         ModuleModel moduleModel = Mockito.spy(ApplicationModel.defaultModel().getDefaultModule());
@@ -385,9 +389,8 @@
                 parameters,
                 consumerAttribute);
         url = url.putAttribute(CONSUMER_URL_KEY, consumerUrl);
-        MigrationRuleListener migrationRuleListener = mock(MigrationRuleListener.class);
         List<RegistryProtocolListener> registryProtocolListeners = new ArrayList<>();
-        registryProtocolListeners.add(migrationRuleListener);
+        registryProtocolListeners.add(migrationRuleListenerMock);
 
         RegistryProtocol registryProtocol = new RegistryProtocol();
         ModuleModel moduleModel = Mockito.spy(ApplicationModel.defaultModel().getDefaultModule());
@@ -403,7 +406,7 @@
         url = url.setScopeModel(moduleModel);
 
         registryProtocol.interceptInvoker(clusterInvoker, url, consumerUrl);
-        verify(migrationRuleListener, times(1)).onRefer(registryProtocol, clusterInvoker, consumerUrl, url);
+        verify(migrationRuleListenerMock, times(1)).onRefer(registryProtocol, clusterInvoker, consumerUrl, url);
     }
 
     /**
