--- a/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
+++ b/dubbo-plugin/dubbo-auth/src/test/java/org/apache/dubbo/auth/filter/ConsumerSignFilterTest.java
@@ -23,6 +23,7 @@
 import org.apache.dubbo.rpc.Invoker;
 import org.apache.dubbo.rpc.model.FrameworkModel;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import static org.mockito.ArgumentMatchers.anyString;
@@ -34,6 +35,13 @@
 import static org.mockito.Mockito.when;
 
 class ConsumerSignFilterTest {
+
+    private Invocation invocationMock;
+
+    @BeforeEach
+    void setUp() {
+        invocationMock = mock(Invocation.class);
+    }
 
     @Test
     void testAuthDisabled() {
