--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/channel/DispatcherHasNoSubscribersTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/channel/DispatcherHasNoSubscribersTests.java
@@ -55,8 +55,7 @@
 				.thenReturn(declareOk);
 		Connection connection = mock(Connection.class);
 		doAnswer(invocation -> channel).when(connection).createChannel(anyBoolean());
-		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
-		when(connectionFactory.createConnection()).thenReturn(connection);
+		ConnectionFactory connectionFactory = createMockConnectionFactory(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
 		container.setConnectionFactory(connectionFactory);
 		AmqpTemplate amqpTemplate = mock(AmqpTemplate.class);
@@ -79,8 +78,7 @@
 		final Channel channel = mock();
 		Connection connection = mock();
 		doAnswer(invocation -> channel).when(connection).createChannel(anyBoolean());
-		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
-		when(connectionFactory.createConnection()).thenReturn(connection);
+		ConnectionFactory connectionFactory = createMockConnectionFactory(connection);
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
 		container.setConnectionFactory(connectionFactory);
 		AmqpTemplate amqpTemplate = mock(AmqpTemplate.class);
@@ -96,4 +94,10 @@
 				.withMessageContaining("Dispatcher has no subscribers for amqp-channel 'noSubscribersChannel'.");
 	}
 
+	private static ConnectionFactory createMockConnectionFactory(Connection connection) {
+		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
+		when(connectionFactory.createConnection()).thenReturn(connection);
+		return connectionFactory;
+	}
+
 }
