--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/AmqpMessageSourceTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/AmqpMessageSourceTests.java
@@ -69,8 +69,7 @@
 		Connection connection = mock(Connection.class);
 		willReturn(true).given(connection).isOpen();
 		willReturn(channel).given(connection).createChannel();
-		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
-		willReturn(connection).given(connectionFactory).newConnection((ExecutorService) isNull(), anyString());
+		ConnectionFactory connectionFactory = createMockConnectionFactory(connection);
 
 		CachingConnectionFactory ccf = new CachingConnectionFactory(connectionFactory);
 		AmqpMessageSource source = new AmqpMessageSource(ccf, "foo");
@@ -120,8 +119,7 @@
 		Connection connection = mock(Connection.class);
 		willReturn(true).given(connection).isOpen();
 		willReturn(channel).given(connection).createChannel();
-		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
-		willReturn(connection).given(connectionFactory).newConnection((ExecutorService) isNull(), anyString());
+		ConnectionFactory connectionFactory = createMockConnectionFactory(connection);
 
 		CachingConnectionFactory ccf = new CachingConnectionFactory(connectionFactory);
 		AmqpMessageSource source = new AmqpMessageSource(ccf, "foo");
@@ -160,8 +158,7 @@
 		Connection connection = mock(Connection.class);
 		willReturn(true).given(connection).isOpen();
 		willReturn(channel).given(connection).createChannel();
-		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
-		willReturn(connection).given(connectionFactory).newConnection((ExecutorService) isNull(), anyString());
+		ConnectionFactory connectionFactory = createMockConnectionFactory(connection);
 
 		CachingConnectionFactory ccf = new CachingConnectionFactory(connectionFactory);
 		AmqpMessageSource source = new AmqpMessageSource(ccf, "foo");
@@ -170,4 +167,10 @@
 		assertThat(((List<String>) received.getPayload())).contains("test1", "test2");
 	}
 
+	private static ConnectionFactory createMockConnectionFactory(Connection connection) throws Exception {
+		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
+		willReturn(connection).given(connectionFactory).newConnection((ExecutorService) isNull(), anyString());
+		return connectionFactory;
+	}
+
 }
