--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
@@ -26,6 +26,7 @@
 
 import com.rabbitmq.client.Channel;
 import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -93,6 +94,13 @@
  */
 public class InboundEndpointTests implements TestApplicationContextAware {
 
+	private ConnectionFactory mockConnectionFactory;
+
+	@BeforeEach
+	void setUp() {
+		this.mockConnectionFactory = mock();
+	}
+
 	@Test
 	public void testInt2809JavaTypePropertiesToAmqp() throws Exception {
 		Connection connection = mock();
@@ -352,8 +360,7 @@
 
 	@Test
 	public void testRetryWithinOnMessageAdapter() throws Exception {
-		ConnectionFactory connectionFactory = mock();
-		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(mockConnectionFactory);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		adapter.setOutputChannel(new DirectChannel());
 		adapter.setRetryTemplate(
@@ -386,8 +393,7 @@
 
 	@Test
 	public void testRetryWithMessageRecovererOnMessageAdapter() throws Exception {
-		ConnectionFactory connectionFactory = mock();
-		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(mockConnectionFactory);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		adapter.setOutputChannel(new DirectChannel());
 		adapter.setRetryTemplate(
@@ -426,8 +432,7 @@
 
 	@Test
 	public void testRetryWithinOnMessageGateway() throws Exception {
-		ConnectionFactory connectionFactory = mock();
-		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(mockConnectionFactory);
 		AmqpInboundGateway adapter = new AmqpInboundGateway(container);
 		adapter.setRequestChannel(new DirectChannel());
 		adapter.setRetryTemplate(
@@ -465,8 +470,7 @@
 
 	@Test
 	public void testRetryWithMessageRecovererOnMessageGateway() throws Exception {
-		ConnectionFactory connectionFactory = mock();
-		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		AbstractMessageListenerContainer container = new SimpleMessageListenerContainer(mockConnectionFactory);
 		AmqpInboundGateway adapter = new AmqpInboundGateway(container);
 		adapter.setRequestChannel(new DirectChannel());
 		adapter.setRetryTemplate(
@@ -754,8 +758,7 @@
 
 	@Test
 	public void testRetryWithinOnMessageAdapterConsumerBatch() {
-		ConnectionFactory connectionFactory = mock();
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mockConnectionFactory);
 		container.setConsumerBatchEnabled(true);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		adapter.setOutputChannel(new DirectChannel());
@@ -806,8 +809,7 @@
 
 	@Test
 	public void testRetryWithMessageRecovererOnMessageAdapterConsumerBatch() throws InterruptedException {
-		ConnectionFactory connectionFactory = mock();
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mockConnectionFactory);
 		container.setConsumerBatchEnabled(true);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		adapter.setOutputChannel(new DirectChannel());
