--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
@@ -45,6 +45,7 @@
 import org.jspecify.annotations.Nullable;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.InOrder;
@@ -140,6 +141,8 @@
 
 	private static Consumer<Integer, String> consumer;
 
+	private ConsumerGroupMetadata consumerGroupMetadata;
+
 	@BeforeAll
 	static void setup() {
 		embeddedKafka = new EmbeddedKafkaKraftBroker(1, 2, topic1, topic2, topic3, topic4, topic5, topic6);
@@ -149,6 +152,11 @@
 		ConsumerFactory<Integer, String> cf = new DefaultKafkaConsumerFactory<>(consumerProps);
 		consumer = cf.createConsumer();
 		embeddedKafka.consumeFromAllEmbeddedTopics(consumer);
+	}
+
+	@BeforeEach
+	void setupMock() {
+		this.consumerGroupMetadata = mock();
 	}
 
 	@AfterAll
@@ -527,8 +535,7 @@
 				return null;
 			}
 		}).given(mockConsumer).poll(any(Duration.class));
-		ConsumerGroupMetadata meta = mock();
-		given(mockConsumer.groupMetadata()).willReturn(meta);
+		given(mockConsumer.groupMetadata()).willReturn(consumerGroupMetadata);
 		ConsumerFactory cf = mock();
 		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
 		Producer producer = mock(Producer.class);
@@ -564,7 +571,7 @@
 		InOrder inOrder = inOrder(producer);
 		inOrder.verify(producer).beginTransaction();
 		inOrder.verify(producer).sendOffsetsToTransaction(Collections.singletonMap(topicPartition,
-				new OffsetAndMetadata(0)), meta);
+				new OffsetAndMetadata(0)), consumerGroupMetadata);
 		inOrder.verify(producer).commitTransaction();
 		inOrder.verify(producer).close(any());
 		inOrder.verify(producer).beginTransaction();
@@ -572,7 +579,7 @@
 		inOrder.verify(producer).send(captor.capture(), any(Callback.class));
 		assertThat(captor.getValue()).isEqualTo(new ProducerRecord("topic", null, "bar", "value"));
 		inOrder.verify(producer).sendOffsetsToTransaction(Collections.singletonMap(topicPartition,
-				new OffsetAndMetadata(1)), meta);
+				new OffsetAndMetadata(1)), consumerGroupMetadata);
 		inOrder.verify(producer).commitTransaction();
 		inOrder.verify(producer).close(any());
 		container.stop();
@@ -664,8 +671,7 @@
 				return null;
 			}
 		}).given(mockConsumer).poll(any(Duration.class));
-		ConsumerGroupMetadata meta = mock();
-		given(mockConsumer.groupMetadata()).willReturn(meta);
+		given(mockConsumer.groupMetadata()).willReturn(consumerGroupMetadata);
 		ConsumerFactory cf = mock();
 		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
 		Producer producer = mock(Producer.class);
@@ -711,7 +717,7 @@
 		InOrder inOrder = inOrder(producer);
 		inOrder.verify(producer).beginTransaction();
 		inOrder.verify(producer).sendOffsetsToTransaction(Collections.singletonMap(topicPartition,
-				new OffsetAndMetadata(0)), meta);
+				new OffsetAndMetadata(0)), consumerGroupMetadata);
 		inOrder.verify(producer).commitTransaction();
 		inOrder.verify(producer).close(any());
 		inOrder.verify(producer).beginTransaction();
@@ -719,7 +725,7 @@
 		inOrder.verify(producer).send(captor.capture(), any(Callback.class));
 		assertThat(captor.getValue()).isEqualTo(new ProducerRecord("topic", null, "bar", "value"));
 		inOrder.verify(producer).sendOffsetsToTransaction(Collections.singletonMap(topicPartition,
-				new OffsetAndMetadata(1)), meta);
+				new OffsetAndMetadata(1)), consumerGroupMetadata);
 		inOrder.verify(producer).commitTransaction();
 		inOrder.verify(producer).close(any());
 		container.stop();
