--- a/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
+++ b/spring-integration-kafka/src/test/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandlerTests.java
@@ -481,6 +481,30 @@
 		KafkaTestUtils.getSingleRecord(consumer, topic6);
 
 		producerFactory.destroy();
+	}
+
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	private static Consumer createMockConsumer(TopicPartition topicPartition, ConsumerGroupMetadata meta) {
+		Consumer mockConsumer = mock();
+		willAnswer(i -> {
+			((ConsumerRebalanceListener) i.getArgument(1))
+					.onPartitionsAssigned(Collections.singletonList(topicPartition));
+			return null;
+		}).given(mockConsumer).subscribe(any(Collection.class), any(ConsumerRebalanceListener.class));
+		ConsumerRecords records = new ConsumerRecords(Collections.singletonMap(topicPartition,
+				Collections.singletonList(new ConsumerRecord<>("foo", 0, 0, "key", "value"))), Map.of());
+		final AtomicBoolean done = new AtomicBoolean();
+		willAnswer(i -> {
+			if (done.compareAndSet(false, true)) {
+				return records;
+			}
+			else {
+				Thread.sleep(500);
+				return null;
+			}
+		}).given(mockConsumer).poll(any(Duration.class));
+		given(mockConsumer.groupMetadata()).willReturn(meta);
+		return mockConsumer;
 	}
 
 	@SuppressWarnings({"rawtypes", "unchecked"})
@@ -645,27 +669,12 @@
 	@SuppressWarnings({"unchecked", "rawtypes"})
 	@Test
 	void testConsumeAndProduceTransactionTxIdOverride() throws Exception {
-		Consumer mockConsumer = mock();
 		final TopicPartition topicPartition = new TopicPartition("foo", 0);
-		willAnswer(i -> {
-			((ConsumerRebalanceListener) i.getArgument(1))
-					.onPartitionsAssigned(Collections.singletonList(topicPartition));
-			return null;
-		}).given(mockConsumer).subscribe(any(Collection.class), any(ConsumerRebalanceListener.class));
 		ConsumerRecords records = new ConsumerRecords(Collections.singletonMap(topicPartition,
 				Collections.singletonList(new ConsumerRecord<>("foo", 0, 0, "key", "value"))), Map.of());
 		final AtomicBoolean done = new AtomicBoolean();
-		willAnswer(i -> {
-			if (done.compareAndSet(false, true)) {
-				return records;
-			}
-			else {
-				Thread.sleep(500);
-				return null;
-			}
-		}).given(mockConsumer).poll(any(Duration.class));
 		ConsumerGroupMetadata meta = mock();
-		given(mockConsumer.groupMetadata()).willReturn(meta);
+		Consumer mockConsumer = createMockConsumer(topicPartition, meta);
 		ConsumerFactory cf = mock();
 		willReturn(mockConsumer).given(cf).createConsumer("group", "", null, KafkaTestUtils.defaultPropertyOverrides());
 		Producer producer = mock(Producer.class);
