--- a/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
+++ b/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java
@@ -106,6 +106,8 @@
 
 	private final IMqttToken alwaysComplete;
 
+	private final IMqttAsyncClient mqttClient = mock(IMqttAsyncClient.class);
+
 	{
 		ProxyFactoryBean pfb = new ProxyFactoryBean();
 		pfb.addAdvice((MethodInterceptor) invocation -> null);
@@ -375,24 +377,22 @@
 
 	@Test
 	public void testStopActionDefault() throws Exception {
-		final IMqttAsyncClient client = mock(IMqttAsyncClient.class);
-		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapterIn(client, null);
+		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapterIn(mqttClient, null);
 
 		adapter.start();
 		adapter.connectComplete(false, null);
 		adapter.stop();
-		verifyUnsubscribe(client);
+		verifyUnsubscribe(mqttClient);
 	}
 
 	@Test
 	public void testStopActionDefaultNotClean() throws Exception {
-		final IMqttAsyncClient client = mock(IMqttAsyncClient.class);
-		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapterIn(client, false);
+		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapterIn(mqttClient, false);
 
 		adapter.start();
 		adapter.connectComplete(false, null);
 		adapter.stop();
-		verifyNotUnsubscribe(client);
+		verifyNotUnsubscribe(mqttClient);
 	}
 
 	@SuppressWarnings("unchecked")
