--- a/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/Mqttv5HandlerTests.java
+++ b/spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/Mqttv5HandlerTests.java
@@ -53,10 +53,7 @@
 
 	@Test
 	void publishPojoWithDefaultConverter() throws Exception {
-		IMqttAsyncClient mockClient = mock();
-		IMqttToken token = mock();
-		given(mockClient.isConnected()).willReturn(true);
-		given(mockClient.publish(anyString(), any(MqttMessage.class), any(), any())).willReturn(token);
+		IMqttAsyncClient mockClient = createMockClient();
 
 		Mqttv5PahoMessageHandler handler = createHandler(mockClient, new ConfigurableCompositeMessageConverter());
 
@@ -75,10 +72,7 @@
 
 	@Test
 	void publishPojoWithCustomConverter() throws Exception {
-		IMqttAsyncClient mockClient = mock();
-		IMqttToken token = mock();
-		given(mockClient.isConnected()).willReturn(true);
-		given(mockClient.publish(anyString(), any(MqttMessage.class), any(), any())).willReturn(token);
+		IMqttAsyncClient mockClient = createMockClient();
 
 		MessageConverter converter = mock(MessageConverter.class);
 		given(converter.toMessage(any(), any(MessageHeaders.class)))
@@ -92,6 +86,14 @@
 		ArgumentCaptor<MqttMessage> captor = ArgumentCaptor.forClass(MqttMessage.class);
 		verify(mockClient).publish(anyString(), captor.capture(), any(), any());
 		assertThat(captor.getValue().getPayload()).isEqualTo("test".getBytes());
+	}
+
+	private static IMqttAsyncClient createMockClient() throws Exception {
+		IMqttAsyncClient mockClient = mock();
+		IMqttToken token = mock();
+		given(mockClient.isConnected()).willReturn(true);
+		given(mockClient.publish(anyString(), any(MqttMessage.class), any(), any())).willReturn(token);
+		return mockClient;
 	}
 
 	private static Mqttv5PahoMessageHandler createHandler(IMqttAsyncClient mockClient, MessageConverter converter) {
