--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/RemoteFileStreamingMessageSourceTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/RemoteFileStreamingMessageSourceTests.java
@@ -23,6 +23,7 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -54,6 +55,13 @@
  */
 public class RemoteFileStreamingMessageSourceTests implements TestApplicationContextAware {
 
+	private Comparator<String> comparatorMock;
+
+	@BeforeEach
+	void setup() {
+		this.comparatorMock = mock();
+	}
+
 	@Test
 	public void fetchFilesFromRemoteAfterClearingFetchedCache() throws IOException {
 		RemoteFileTemplate<String> remoteFileTemplate = mock();
@@ -62,9 +70,8 @@
 		when(session.readRaw(anyString())).thenReturn(mock());
 		when(remoteFileTemplate.getSession()).thenReturn(session);
 
-		Comparator<String> comparator = mock();
 		TestRemoteFileStreamingMessageSource testRemoteFileStreamingMessageSource =
-				new TestRemoteFileStreamingMessageSource(remoteFileTemplate, comparator);
+				new TestRemoteFileStreamingMessageSource(remoteFileTemplate, comparatorMock);
 
 		testRemoteFileStreamingMessageSource.setRemoteDirectory("remoteDirectory");
 		testRemoteFileStreamingMessageSource.setBeanFactory(TEST_INTEGRATION_CONTEXT);
@@ -95,9 +102,8 @@
 		when(fileListFilter.accept("file1")).thenReturn(false);
 		when(fileListFilter.accept("file2")).thenReturn(false);
 
-		Comparator<String> comparator = mock();
 		TestRemoteFileStreamingMessageSource testRemoteFileStreamingMessageSource =
-				new TestRemoteFileStreamingMessageSource(remoteFileTemplate, comparator);
+				new TestRemoteFileStreamingMessageSource(remoteFileTemplate, comparatorMock);
 
 		testRemoteFileStreamingMessageSource.setFilter(fileListFilter);
 		testRemoteFileStreamingMessageSource.setRemoteDirectory("remoteDirectory");
