--- a/spring-integration-core/src/test/java/org/springframework/integration/gateway/GatewayInterfaceTests.java
+++ b/spring-integration-core/src/test/java/org/springframework/integration/gateway/GatewayInterfaceTests.java
@@ -31,6 +31,7 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.jspecify.annotations.Nullable;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
@@ -169,6 +170,13 @@
 	@Autowired(required = false)
 	private NotActivatedByProfileGateway notActivatedByProfileGateway;
 
+	private MessageHandler messageHandlerMock;
+
+	@BeforeEach
+	void setUp() {
+		this.messageHandlerMock = mock(MessageHandler.class);
+	}
+
 	@Test
 	public void testWithServiceSuperclassAnnotatedMethod() throws Exception {
 		ConfigurableApplicationContext ac =
@@ -223,11 +231,10 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelBar", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Bar bar = ac.getBean(Bar.class);
 		bar.bar("hello");
-		verify(handler, times(1)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(1)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -282,11 +289,10 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelFoo", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Foo foo = ac.getBean(Foo.class);
 		foo.foo("hello");
-		verify(handler, times(1)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(1)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -295,11 +301,10 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelBaz", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Foo foo = ac.getBean(Foo.class);
 		foo.baz("hello");
-		verify(handler, times(1)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(1)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -308,11 +313,10 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelBaz", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Bar bar = ac.getBean(Bar.class);
 		assertThat(ac.getBean(Bar.class).hashCode()).isEqualTo(bar.hashCode());
-		verify(handler, times(0)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(0)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -321,11 +325,10 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelBaz", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Bar bar = ac.getBean(Bar.class);
 		bar.toString();
-		verify(handler, times(0)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(0)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -334,8 +337,7 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelBaz", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Bar bar = ac.getBean(Bar.class);
 		assertThat(ac.getBean(Bar.class)).isSameAs(bar);
 		GatewayProxyFactoryBean<Bar> fb = new GatewayProxyFactoryBean<>(Bar.class);
@@ -349,7 +351,7 @@
 		fb.setBeanFactory(bf);
 		fb.afterPropertiesSet();
 		assertThat(fb.getObject()).isNotSameAs(bar);
-		verify(handler, times(0)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(0)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -358,11 +360,10 @@
 		ConfigurableApplicationContext ac =
 				new ClassPathXmlApplicationContext("GatewayInterfaceTests-context.xml", getClass());
 		DirectChannel channel = ac.getBean("requestChannelBaz", DirectChannel.class);
-		MessageHandler handler = mock(MessageHandler.class);
-		channel.subscribe(handler);
+		channel.subscribe(messageHandlerMock);
 		Bar bar = ac.getBean(Bar.class);
 		bar.getClass();
-		verify(handler, times(0)).handleMessage(Mockito.any(Message.class));
+		verify(messageHandlerMock, times(0)).handleMessage(Mockito.any(Message.class));
 		ac.close();
 	}
 
@@ -503,20 +504,19 @@
 	@Test
 	@SuppressWarnings("unchecked")
 	public void testIgnoredHeader() {
-		MessageHandler messageHandler = mock(MessageHandler.class);
-
-		((SubscribableChannel) this.errorChannel).subscribe(messageHandler);
+
+		((SubscribableChannel) this.errorChannel).subscribe(messageHandlerMock);
 		this.ignoredHeaderGateway.service("foo", "theHeaderValue");
 
 		ArgumentCaptor<Message<?>> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
 
-		verify(messageHandler).handleMessage(messageArgumentCaptor.capture());
+		verify(messageHandlerMock).handleMessage(messageArgumentCaptor.capture());
 
 		Message<?> message = messageArgumentCaptor.getValue();
 
 		assertThat(message.getHeaders().containsKey(IGNORE_HEADER)).isFalse();
 
-		((SubscribableChannel) this.errorChannel).unsubscribe(messageHandler);
+		((SubscribableChannel) this.errorChannel).unsubscribe(messageHandlerMock);
 	}
 
 	@Test
@@ -546,15 +546,14 @@
 		PrimaryGateway primaryGateway = this.beanFactory.getBean(PrimaryGateway.class);
 		assertThat(AopUtils.isAopProxy(primaryGateway)).isTrue();
 
-		MessageHandler messageHandler = mock(MessageHandler.class);
-
-		((SubscribableChannel) this.errorChannel).subscribe(messageHandler);
+
+		((SubscribableChannel) this.errorChannel).subscribe(messageHandlerMock);
 
 		primaryGateway.testGateway("test data");
 
 		ArgumentCaptor<Message<?>> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
 
-		verify(messageHandler).handleMessage(messageArgumentCaptor.capture());
+		verify(messageHandlerMock).handleMessage(messageArgumentCaptor.capture());
 
 		Message<?> receive = messageArgumentCaptor.getValue();
 
@@ -565,7 +564,7 @@
 		PrimaryGateway notPrimaryGateway = this.beanFactory.getBean("notPrimaryGatewayInstance", PrimaryGateway.class);
 		assertThat(AopUtils.isAopProxy(notPrimaryGateway)).isFalse();
 
-		((SubscribableChannel) this.errorChannel).unsubscribe(messageHandler);
+		((SubscribableChannel) this.errorChannel).unsubscribe(messageHandlerMock);
 	}
 
 	@Autowired
