--- a/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
+++ b/spring-integration-mail/src/test/java/org/springframework/integration/mail/inbound/ImapMailReceiverTests.java
@@ -278,10 +278,9 @@
 		when(logger.isDebugEnabled()).thenReturn(true);
 
 		Message msg1 = mock(MimeMessage.class);
-		Message msg2 = mock(MimeMessage.class);
+		Message msg2 = createMockMsg2();
 		given(msg1.isExpunged()).willReturn(true);
 		given(msg1.getSubject()).willReturn("msg1");
-		given(msg2.getSubject()).willReturn("msg2");
 		Expression selectorExpression = new SpelExpressionParser().parseExpression("false");
 		receiver.setSelectorExpression(selectorExpression);
 		receiveAndMarkAsReadDontDeleteGuts(receiver, msg1, msg2);
@@ -990,6 +989,12 @@
 		return receiver;
 	}
 
+	private static Message createMockMsg2() throws Exception {
+		Message msg2 = mock(MimeMessage.class);
+		given(msg2.getSubject()).willReturn("msg2");
+		return msg2;
+	}
+
 	private static class ImapSearchLoggingHandler extends Handler {
 
 		private final List<String> searches = new ArrayList<>();
