--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
@@ -26,6 +26,7 @@
 
 import com.rabbitmq.client.Channel;
 import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
@@ -93,6 +94,13 @@
  */
 public class InboundEndpointTests implements TestApplicationContextAware {
 
+	private ConnectionFactory connectionFactoryMock;
+
+	@BeforeEach
+	void setUp() {
+		connectionFactoryMock = mock();
+	}
+
 	@Test
 	public void testInt2809JavaTypePropertiesToAmqp() throws Exception {
 		Connection connection = mock();
@@ -507,7 +515,7 @@
 	@SuppressWarnings({"unchecked"})
 	@Test
 	public void testBatchAdapter() throws Exception {
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock());
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactoryMock);
 		container.setDeBatchingEnabled(false);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		QueueChannel out = new QueueChannel();
@@ -532,7 +540,7 @@
 	@SuppressWarnings({"unchecked"})
 	@Test
 	public void testBatchGateway() throws Exception {
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock());
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactoryMock);
 		container.setDeBatchingEnabled(false);
 		AmqpInboundGateway gateway = new AmqpInboundGateway(container);
 		QueueChannel out = new QueueChannel();
@@ -561,7 +569,7 @@
 	@SuppressWarnings({"unchecked"})
 	@Test
 	public void testConsumerBatchExtract() {
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock());
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactoryMock);
 		container.setConsumerBatchEnabled(true);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		QueueChannel out = new QueueChannel();
@@ -587,7 +595,7 @@
 	@SuppressWarnings({"unchecked"})
 	@Test
 	public void testConsumerBatch() {
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock());
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactoryMock);
 		container.setConsumerBatchEnabled(true);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		QueueChannel out = new QueueChannel();
@@ -610,7 +618,7 @@
 
 	@Test
 	public void testConsumerBatchAndWrongMessageRecoverer() {
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(mock());
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactoryMock);
 		container.setConsumerBatchEnabled(true);
 		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter((MessageListenerContainer) container);
 		adapter.setBeanFactory(TEST_INTEGRATION_CONTEXT);
