--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -19,6 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.expression.Expression;
@@ -55,6 +56,13 @@
  */
 public class WsDslTests {
 
+	private SoapHeaderMapper soapHeaderMapper;
+
+	@BeforeEach
+	void setup() {
+		this.soapHeaderMapper = mock();
+	}
+
 	@Test
 	void marshallingInbound() {
 		Marshaller marshaller = mock();
@@ -92,7 +100,6 @@
 		Unmarshaller unmarshaller = mock();
 		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
-		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
 		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
@@ -106,7 +113,7 @@
 						.unmarshaller(unmarshaller)
 						.messageFactory(messageFactory)
 						.faultMessageResolver(faultMessageResolver)
-						.headerMapper(headerMapper)
+						.headerMapper(soapHeaderMapper)
 						.ignoreEmptyResponses(true)
 						.interceptors(interceptor)
 						.messageSenders(messageSender)
@@ -120,7 +127,7 @@
 				.isSameAs(messageFactory);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "webServiceTemplate.faultMessageResolver"))
 				.isSameAs(faultMessageResolver);
-		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(headerMapper);
+		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(soapHeaderMapper);
 		assertThat(TestUtils.<ClientInterceptor[]>getPropertyValue(gateway, "webServiceTemplate.interceptors")[0])
 				.isSameAs(interceptor);
 		assertThat(TestUtils.<WebServiceMessageSender[]>getPropertyValue(gateway, "webServiceTemplate.messageSenders")[0])
@@ -135,7 +142,6 @@
 		DestinationProvider destinationProvider = mock();
 		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
-		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
 		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
@@ -150,7 +156,7 @@
 						.messageFactory(messageFactory)
 						.encodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY)
 						.faultMessageResolver(faultMessageResolver)
-						.headerMapper(headerMapper)
+						.headerMapper(soapHeaderMapper)
 						.ignoreEmptyResponses(true)
 						.interceptors(interceptor)
 						.messageSenders(messageSender)
@@ -162,7 +168,7 @@
 				.isSameAs(messageFactory);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "webServiceTemplate.faultMessageResolver"))
 				.isSameAs(faultMessageResolver);
-		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(headerMapper);
+		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(soapHeaderMapper);
 		assertThat(TestUtils.<ClientInterceptor[]>getPropertyValue(gateway, "webServiceTemplate.interceptors")[0])
 				.isSameAs(interceptor);
 		assertThat(TestUtils.<WebServiceMessageSender[]>getPropertyValue(gateway, "webServiceTemplate.messageSenders")[0])
@@ -175,7 +181,6 @@
 
 	@Test
 	void marshallingOutboundTemplate() {
-		SoapHeaderMapper headerMapper = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
@@ -186,13 +191,13 @@
 		MarshallingWebServiceOutboundGateway gateway =
 				Ws.marshallingOutboundGateway(template)
 						.uri(uri)
-						.headerMapper(headerMapper)
+						.headerMapper(soapHeaderMapper)
 						.ignoreEmptyResponses(true)
 						.requestCallback(requestCallback)
 						.uriVariableExpressions(uriVariableExpressions)
 						.getObject();
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "uri")).isSameAs(uri);
-		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(headerMapper);
+		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(soapHeaderMapper);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "requestCallback")).isSameAs(requestCallback);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "uriVariableExpressions"))
 				.isEqualTo(uriVariableExpressions);
@@ -203,7 +208,6 @@
 
 	@Test
 	void simpleOutboundTemplate() {
-		SoapHeaderMapper headerMapper = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
@@ -215,13 +219,13 @@
 				Ws.simpleOutboundGateway(template)
 						.sourceExtractor(sourceExtractor)
 						.encodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY)
-						.headerMapper(headerMapper)
+						.headerMapper(soapHeaderMapper)
 						.ignoreEmptyResponses(true)
 						.requestCallback(requestCallback)
 						.uriVariableExpressions(uriVariableExpressions)
 						.extractPayload(false)
 						.getObject();
-		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(headerMapper);
+		assertThat(TestUtils.<Object>getPropertyValue(gateway, "headerMapper")).isSameAs(soapHeaderMapper);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "requestCallback")).isSameAs(requestCallback);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "uriVariableExpressions"))
 				.isEqualTo(uriVariableExpressions);
