--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/outbound/TcpOutboundGatewayTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/outbound/TcpOutboundGatewayTests.java
@@ -500,9 +500,8 @@
 		assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
 
 		// Failover
-		AbstractClientConnectionFactory factory1 = mock();
 		TcpConnectionSupport mockConn1 = makeMockConnection();
-		when(factory1.getConnection()).thenReturn(mockConn1);
+		AbstractClientConnectionFactory factory1 = createMockClientConnectionFactory(mockConn1);
 		doThrow(new UncheckedIOException(new IOException("fail")))
 				.when(mockConn1).send(any(Message.class));
 
@@ -585,9 +584,8 @@
 		assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
 
 		// Cache
-		AbstractClientConnectionFactory factory1 = mock();
 		TcpConnectionSupport mockConn1 = makeMockConnection();
-		when(factory1.getConnection()).thenReturn(mockConn1);
+		AbstractClientConnectionFactory factory1 = createMockClientConnectionFactory(mockConn1);
 		when(factory1.isSingleUse()).thenReturn(true);
 		doThrow(new UncheckedIOException(new IOException("fail")))
 				.when(mockConn1).send(any(Message.class));
@@ -638,6 +636,14 @@
 		TcpConnectionSupport connection = mock();
 		when(connection.isOpen()).thenReturn(true);
 		return connection;
+	}
+
+	private static AbstractClientConnectionFactory createMockClientConnectionFactory(
+			TcpConnectionSupport connection) throws InterruptedException {
+
+		AbstractClientConnectionFactory factory = mock();
+		when(factory.getConnection()).thenReturn(connection);
+		return factory;
 	}
 
 	@Test
