--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/gateway/RemoteFileOutboundGatewayTests.java
@@ -116,11 +116,10 @@
 
 	@Test
 	public void testLs() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/x/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -325,12 +324,11 @@
 
 	@Test
 	public void testLs_f() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-f");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/x/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -366,12 +364,11 @@
 
 	@Test
 	public void testLs_f_R() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-f -R");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] level1 = level1List();
 		TestLsEntry[] level2 = level2List();
 		TestLsEntry[] level3 = level3List();
@@ -392,12 +389,11 @@
 
 	@Test
 	public void testLs_f_R_dirs() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-f -R -dirs");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] level1 = level1List();
 		TestLsEntry[] level2 = level2List();
 		TestLsEntry[] level3 = level3List();
@@ -420,11 +416,10 @@
 
 	@Test
 	public void testLs_None() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = new TestLsEntry[0];
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -436,12 +431,11 @@
 
 	@Test
 	public void testLs_1() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -455,12 +449,11 @@
 
 	@Test
 	public void testLs_1_f() throws Exception { //no sort
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -f");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -473,12 +466,11 @@
 
 	@Test
 	public void testLs_1_dirs() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -dirs");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -492,12 +484,11 @@
 
 	@Test
 	public void testLs_1_dirs_links() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -dirs -links");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -512,12 +503,11 @@
 
 	@Test
 	public void testLs_1_a_f_dirs_links() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "ls", "payload");
 		gw.setOptions("-1 -a -f -dirs -links");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		TestLsEntry[] files = fileList();
 		when(session.list("testremote/")).thenReturn(files);
 		@SuppressWarnings("unchecked")
@@ -552,12 +542,7 @@
 
 	@Test
 	public void testGet() throws IOException {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
-		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "get", "payload");
-		gw.setLocalDirectory(new File(this.tmpDir));
-		gw.afterPropertiesSet();
-		new File(this.tmpDir + "/f1").delete();
-		when(sessionFactory.getSession()).thenReturn(new TestSession() {
+		Session session = new TestSession() {
 
 			@Override
 			public TestLsEntry[] list(String path) {
@@ -572,7 +557,12 @@
 				outputStream.write("testfile".getBytes());
 			}
 
-		});
+		};
+		SessionFactory sessionFactory = createMockSessionFactory(session);
+		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "get", "payload");
+		gw.setLocalDirectory(new File(this.tmpDir));
+		gw.afterPropertiesSet();
+		new File(this.tmpDir + "/f1").delete();
 		@SuppressWarnings("unchecked")
 		MessageBuilder<File> out = (MessageBuilder<File>) gw.handleRequestMessage(new GenericMessage<>("f1"));
 		File outFile = new File(this.tmpDir + "/f1").getCanonicalFile();
@@ -706,6 +696,12 @@
 		BufferedReader reader = new BufferedReader(new FileReader(outFile));
 		assertThat(reader.readLine()).isEqualTo(expected);
 		reader.close();
+	}
+
+	private static SessionFactory createMockSessionFactory(Session session) {
+		SessionFactory sessionFactory = mock(SessionFactory.class);
+		when(sessionFactory.getSession()).thenReturn(session);
+		return sessionFactory;
 	}
 
 	@Test
@@ -811,11 +807,10 @@
 
 	@Test
 	public void testRm() throws Exception {
-		SessionFactory sessionFactory = mock(SessionFactory.class);
 		Session session = mock(Session.class);
+		SessionFactory sessionFactory = createMockSessionFactory(session);
 		TestRemoteFileOutboundGateway gw = new TestRemoteFileOutboundGateway(sessionFactory, "rm", "payload");
 		gw.afterPropertiesSet();
-		when(sessionFactory.getSession()).thenReturn(session);
 		when(session.remove("testremote/x/f1")).thenReturn(Boolean.TRUE);
 		@SuppressWarnings("unchecked")
 		MessageBuilder<Boolean> out = (MessageBuilder<Boolean>) gw
