--- a/spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFactoryTests.java
+++ b/spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/ConnectionFactoryTests.java
@@ -20,6 +20,7 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.context.ApplicationEventPublisher;
@@ -54,6 +55,13 @@
  *
  */
 public class ConnectionFactoryTests implements TestApplicationContextAware {
+
+	private TcpNetConnectionSupport netConnectionSupport;
+
+	@BeforeEach
+	void setUp() {
+		this.netConnectionSupport = mock(TcpNetConnectionSupport.class);
+	}
 
 	@Test
 	public void test() throws Exception {
@@ -105,15 +113,14 @@
 	@Test
 	void netCustomServer() {
 		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
-		TcpNetConnectionSupport conSupp = mock(TcpNetConnectionSupport.class);
 		TcpSocketFactorySupport factSupp = mock(TcpSocketFactorySupport.class);
 		TcpNetServerConnectionFactory server = Tcp.netServer(0)
 				.socketSupport(sockSupp)
-				.connectionSupport(conSupp)
+				.connectionSupport(netConnectionSupport)
 				.socketFactorySupport(factSupp)
 				.getObject();
 		assertThat(TestUtils.<Object>getPropertyValue(server, "tcpSocketSupport")).isSameAs(sockSupp);
-		assertThat(TestUtils.<Object>getPropertyValue(server, "tcpNetConnectionSupport")).isSameAs(conSupp);
+		assertThat(TestUtils.<Object>getPropertyValue(server, "tcpNetConnectionSupport")).isSameAs(netConnectionSupport);
 		assertThat(TestUtils.<Object>getPropertyValue(server, "tcpSocketFactorySupport")).isSameAs(factSupp);
 	}
 
@@ -134,15 +141,14 @@
 	@Test
 	void netCustomClient() {
 		TcpSocketSupport sockSupp = mock(TcpSocketSupport.class);
-		TcpNetConnectionSupport conSupp = mock(TcpNetConnectionSupport.class);
 		TcpSocketFactorySupport factSupp = mock(TcpSocketFactorySupport.class);
 		TcpNetClientConnectionFactory client = Tcp.netClient("localhost", 0)
 				.socketSupport(sockSupp)
-				.connectionSupport(conSupp)
+				.connectionSupport(netConnectionSupport)
 				.socketFactorySupport(factSupp)
 				.getObject();
 		assertThat(TestUtils.<Object>getPropertyValue(client, "tcpSocketSupport")).isSameAs(sockSupp);
-		assertThat(TestUtils.<Object>getPropertyValue(client, "tcpNetConnectionSupport")).isSameAs(conSupp);
+		assertThat(TestUtils.<Object>getPropertyValue(client, "tcpNetConnectionSupport")).isSameAs(netConnectionSupport);
 		assertThat(TestUtils.<Object>getPropertyValue(client, "tcpSocketFactorySupport")).isSameAs(factSupp);
 	}
 
