--- a/spring-integration-file/src/test/java/org/springframework/integration/file/remote/RemoteFileStreamingMessageSourceTests.java
+++ b/spring-integration-file/src/test/java/org/springframework/integration/file/remote/RemoteFileStreamingMessageSourceTests.java
@@ -56,11 +56,9 @@
 
 	@Test
 	public void fetchFilesFromRemoteAfterClearingFetchedCache() throws IOException {
-		RemoteFileTemplate<String> remoteFileTemplate = mock();
-		when(remoteFileTemplate.list("remoteDirectory")).thenReturn(new String[] {"file1", "file2"});
 		Session<String> session = mock();
 		when(session.readRaw(anyString())).thenReturn(mock());
-		when(remoteFileTemplate.getSession()).thenReturn(session);
+		RemoteFileTemplate<String> remoteFileTemplate = createMockRemoteFileTemplate(session);
 
 		Comparator<String> comparator = mock();
 		TestRemoteFileStreamingMessageSource testRemoteFileStreamingMessageSource =
@@ -84,11 +82,9 @@
 
 	@Test
 	public void filterOutFilesNotAcceptedByFilter() throws IOException {
-		RemoteFileTemplate<String> remoteFileTemplate = mock();
-		when(remoteFileTemplate.list("remoteDirectory")).thenReturn(new String[] {"file1", "file2"});
 		Session<String> session = mock();
 		when(session.readRaw(anyString())).thenReturn(mock());
-		when(remoteFileTemplate.getSession()).thenReturn(session);
+		RemoteFileTemplate<String> remoteFileTemplate = createMockRemoteFileTemplate(session);
 
 		FileListFilter<String> fileListFilter = mock();
 		when(fileListFilter.supportsSingleFileFiltering()).thenReturn(true);
@@ -145,6 +141,13 @@
 		return beanFactory;
 	}
 
+	private static RemoteFileTemplate<String> createMockRemoteFileTemplate(Session<String> session) {
+		RemoteFileTemplate<String> remoteFileTemplate = mock();
+		when(remoteFileTemplate.list("remoteDirectory")).thenReturn(new String[] {"file1", "file2"});
+		when(remoteFileTemplate.getSession()).thenReturn(session);
+		return remoteFileTemplate;
+	}
+
 	static class TestRemoteFileStreamingMessageSource extends AbstractRemoteFileStreamingMessageSource<String> {
 
 		TestRemoteFileStreamingMessageSource(RemoteFileTemplate<String> template, Comparator<String> comparator) {
