--- a/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
+++ b/spring-integration-amqp/src/test/java/org/springframework/integration/amqp/inbound/InboundEndpointTests.java
@@ -267,10 +267,9 @@
 		});
 		adapter.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		adapter.afterPropertiesSet();
-		org.springframework.amqp.core.Message message = mock();
 		MessageProperties props = new MessageProperties();
 		props.setDeliveryTag(42L);
-		given(message.getMessageProperties()).willReturn(props);
+		org.springframework.amqp.core.Message message = createMockAmqpMessage(props);
 		((ChannelAwareMessageListener) container.getMessageListener())
 				.onMessage(message, null);
 		assertThat(outputChannel.receive(0)).isNull();
@@ -324,10 +323,9 @@
 		});
 		adapter.setBeanFactory(TEST_INTEGRATION_CONTEXT);
 		adapter.afterPropertiesSet();
-		org.springframework.amqp.core.Message message = mock();
 		MessageProperties props = new MessageProperties();
 		props.setDeliveryTag(42L);
-		given(message.getMessageProperties()).willReturn(props);
+		org.springframework.amqp.core.Message message = createMockAmqpMessage(props);
 		((ChannelAwareMessageListener) container.getMessageListener())
 				.onMessage(message, null);
 		assertThat(outputChannel.receive(0)).isNull();
@@ -851,6 +849,12 @@
 		assertThat(recoveredMessages.get()).isSameAs(messages);
 	}
 
+	private static org.springframework.amqp.core.Message createMockAmqpMessage(MessageProperties props) {
+		org.springframework.amqp.core.Message message = mock();
+		given(message.getMessageProperties()).willReturn(props);
+		return message;
+	}
+
 	public record Foo(String bar) {
 
 	}
