--- a/spring-integration-ftp/src/test/java/org/springframework/integration/ftp/session/SessionFactoryTests.java
+++ b/spring-integration-ftp/src/test/java/org/springframework/integration/ftp/session/SessionFactoryTests.java
@@ -26,6 +26,7 @@
 
 import org.apache.commons.net.ftp.FTPClient;
 import org.apache.commons.net.ftp.FTPFile;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
@@ -55,6 +56,14 @@
  *
  */
 class SessionFactoryTests {
+
+	private Session<FTPFile> mockedSession;
+
+	@BeforeEach
+	@SuppressWarnings("unchecked")
+	void setUp() {
+		this.mockedSession = Mockito.mock(Session.class);
+	}
 
 	@Test
 	void testFtpClientInteraction() throws Exception {
@@ -166,8 +175,7 @@
 	@SuppressWarnings("unchecked")
 	void testSameSessionFromThePool() {
 		SessionFactory<FTPFile> sessionFactory = Mockito.mock(SessionFactory.class);
-		Session<FTPFile> session = Mockito.mock(Session.class);
-		Mockito.when(sessionFactory.getSession()).thenReturn(session);
+		Mockito.when(sessionFactory.getSession()).thenReturn(mockedSession);
 
 		CachingSessionFactory<FTPFile> cachingFactory = new CachingSessionFactory<>(sessionFactory, 2);
 
@@ -184,8 +192,7 @@
 	@SuppressWarnings("unchecked")
 	void testSessionWaitExpire() {
 		SessionFactory<FTPFile> sessionFactory = Mockito.mock(SessionFactory.class);
-		Session<FTPFile> session = Mockito.mock(Session.class);
-		Mockito.when(sessionFactory.getSession()).thenReturn(session);
+		Mockito.when(sessionFactory.getSession()).thenReturn(mockedSession);
 
 		CachingSessionFactory<FTPFile> cachingFactory = new CachingSessionFactory<>(sessionFactory, 2);
 
