--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -19,6 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.expression.Expression;
@@ -55,6 +56,13 @@
  */
 public class WsDslTests {
 
+	private SourceExtractor<?> sourceExtractorMock;
+
+	@BeforeEach
+	void setUp() {
+		sourceExtractorMock = mock();
+	}
+
 	@Test
 	void marshallingInbound() {
 		Marshaller marshaller = mock();
@@ -142,11 +150,10 @@
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
 		uriVariableExpressions.put("foo", new LiteralExpression("bar"));
-		SourceExtractor<?> sourceExtractor = mock();
 		SimpleWebServiceOutboundGateway gateway =
 				Ws.simpleOutboundGateway()
 						.destinationProvider(destinationProvider)
-						.sourceExtractor(sourceExtractor)
+						.sourceExtractor(sourceExtractorMock)
 						.messageFactory(messageFactory)
 						.encodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY)
 						.faultMessageResolver(faultMessageResolver)
@@ -208,12 +215,11 @@
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
 		uriVariableExpressions.put("testVariable", new LiteralExpression("testValue"));
-		SourceExtractor<?> sourceExtractor = mock();
 		WebServiceTemplate template = new WebServiceTemplate();
 		template.setDefaultUri("testUri");
 		SimpleWebServiceOutboundGateway gateway =
 				Ws.simpleOutboundGateway(template)
-						.sourceExtractor(sourceExtractor)
+						.sourceExtractor(sourceExtractorMock)
 						.encodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY)
 						.headerMapper(headerMapper)
 						.ignoreEmptyResponses(true)
