--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -19,6 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.expression.Expression;
@@ -55,15 +56,21 @@
  */
 public class WsDslTests {
 
+	private Unmarshaller unmarshallerMock;
+
+	@BeforeEach
+	void setup() {
+		this.unmarshallerMock = mock();
+	}
+
 	@Test
 	void marshallingInbound() {
 		Marshaller marshaller = mock();
-		Unmarshaller unmarshaller = mock();
 		MarshallingWebServiceInboundGateway gateway = Ws.marshallingInboundGateway(marshaller)
-				.unmarshaller(unmarshaller)
+				.unmarshaller(unmarshallerMock)
 				.getObject();
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "marshaller")).isSameAs(marshaller);
-		assertThat(TestUtils.<Object>getPropertyValue(gateway, "unmarshaller")).isSameAs(unmarshaller);
+		assertThat(TestUtils.<Object>getPropertyValue(gateway, "unmarshaller")).isSameAs(unmarshallerMock);
 
 		marshaller = mock(Both.class);
 		gateway = Ws.marshallingInboundGateway(marshaller).getObject();
@@ -89,7 +96,6 @@
 	void marshallingOutbound() {
 		DestinationProvider destinationProvider = mock();
 		Marshaller marshaller = mock();
-		Unmarshaller unmarshaller = mock();
 		WebServiceMessageFactory messageFactory = mock();
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
@@ -103,7 +109,7 @@
 				Ws.marshallingOutboundGateway()
 						.destinationProvider(destinationProvider)
 						.marshaller(marshaller)
-						.unmarshaller(unmarshaller)
+						.unmarshaller(unmarshallerMock)
 						.messageFactory(messageFactory)
 						.faultMessageResolver(faultMessageResolver)
 						.headerMapper(headerMapper)
@@ -115,7 +121,7 @@
 						.getObject();
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "webServiceTemplate.marshaller")).isSameAs(marshaller);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "webServiceTemplate.unmarshaller"))
-				.isSameAs(unmarshaller);
+				.isSameAs(unmarshallerMock);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "webServiceTemplate.messageFactory"))
 				.isSameAs(messageFactory);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "webServiceTemplate.faultMessageResolver"))
