--- a/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
+++ b/spring-integration-ws/src/test/java/org/springframework/integration/ws/dsl/WsDslTests.java
@@ -19,6 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.expression.Expression;
@@ -55,6 +56,13 @@
  */
 public class WsDslTests {
 
+	private WebServiceMessageSender webServiceMessageSender;
+
+	@BeforeEach
+	void setup() {
+		webServiceMessageSender = mock();
+	}
+
 	@Test
 	void marshallingInbound() {
 		Marshaller marshaller = mock();
@@ -94,7 +102,6 @@
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
-		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
@@ -109,7 +116,7 @@
 						.headerMapper(headerMapper)
 						.ignoreEmptyResponses(true)
 						.interceptors(interceptor)
-						.messageSenders(messageSender)
+						.messageSenders(webServiceMessageSender)
 						.requestCallback(requestCallback)
 						.uriVariableExpressions(uriVariableExpressions)
 						.getObject();
@@ -124,7 +131,7 @@
 		assertThat(TestUtils.<ClientInterceptor[]>getPropertyValue(gateway, "webServiceTemplate.interceptors")[0])
 				.isSameAs(interceptor);
 		assertThat(TestUtils.<WebServiceMessageSender[]>getPropertyValue(gateway, "webServiceTemplate.messageSenders")[0])
-				.isSameAs(messageSender);
+				.isSameAs(webServiceMessageSender);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "requestCallback")).isSameAs(requestCallback);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "uriVariableExpressions"))
 				.isEqualTo(uriVariableExpressions);
@@ -137,7 +144,6 @@
 		FaultMessageResolver faultMessageResolver = mock();
 		SoapHeaderMapper headerMapper = mock();
 		ClientInterceptor interceptor = mock();
-		WebServiceMessageSender messageSender = mock();
 		WebServiceMessageCallback requestCallback = msg -> {
 		};
 		Map<String, Expression> uriVariableExpressions = new HashMap<>();
@@ -153,7 +159,7 @@
 						.headerMapper(headerMapper)
 						.ignoreEmptyResponses(true)
 						.interceptors(interceptor)
-						.messageSenders(messageSender)
+						.messageSenders(webServiceMessageSender)
 						.requestCallback(requestCallback)
 						.uriVariableExpressions(uriVariableExpressions)
 						.extractPayload(false)
@@ -166,7 +172,7 @@
 		assertThat(TestUtils.<ClientInterceptor[]>getPropertyValue(gateway, "webServiceTemplate.interceptors")[0])
 				.isSameAs(interceptor);
 		assertThat(TestUtils.<WebServiceMessageSender[]>getPropertyValue(gateway, "webServiceTemplate.messageSenders")[0])
-				.isSameAs(messageSender);
+				.isSameAs(webServiceMessageSender);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "requestCallback")).isSameAs(requestCallback);
 		assertThat(TestUtils.<Object>getPropertyValue(gateway, "uriVariableExpressions"))
 				.isEqualTo(uriVariableExpressions);
