--- a/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/outbound/PresenceSendingMessageHandlerTests.java
+++ b/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/outbound/PresenceSendingMessageHandlerTests.java
@@ -18,6 +18,7 @@
 
 import org.jivesoftware.smack.XMPPConnection;
 import org.jivesoftware.smack.packet.StanzaBuilder;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import org.springframework.beans.factory.BeanFactory;
@@ -40,6 +41,13 @@
  * @author Glenn Renfro
  */
 public class PresenceSendingMessageHandlerTests {
+
+	private XMPPConnection xmppConnection;
+
+	@BeforeEach
+	void setUp() {
+		xmppConnection = mock(XMPPConnection.class);
+	}
 
 	@Test
 	public void testPresencePayload() {
