--- a/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/outbound/ChatMessageSendingMessageHandlerTests.java
+++ b/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/outbound/ChatMessageSendingMessageHandlerTests.java
@@ -20,6 +20,7 @@
 import org.jivesoftware.smack.packet.StanzaBuilder;
 import org.jivesoftware.smackx.gcm.packet.GcmPacketExtension;
 import org.jivesoftware.smackx.gcm.provider.GcmExtensionProvider;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
@@ -53,10 +54,16 @@
  */
 public class ChatMessageSendingMessageHandlerTests {
 
+	private XMPPConnection xmppConnection;
+
+	@BeforeEach
+	public void setup() {
+		this.xmppConnection = mock(XMPPConnection.class);
+	}
+
 	@Test
 	public void testSendMessages() throws Exception {
-		XMPPConnection connection = mock(XMPPConnection.class);
-		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(connection);
+		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(xmppConnection);
 		handler.setBeanFactory(mock(BeanFactory.class));
 		handler.afterPropertiesSet();
 		Message<?> message = MessageBuilder.withPayload("Test Message").
@@ -65,7 +72,7 @@
 		// first Message new
 		handler.handleMessage(message);
 
-		verify(connection, times(1))
+		verify(xmppConnection, times(1))
 				.sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message smackMessage) -> {
 					boolean bodyMatches = smackMessage.getBody().equals("Test Message");
 					boolean toMatches = smackMessage.getTo().equals("kermit@frog.com");
@@ -78,11 +85,11 @@
 				setHeader(XmppHeaders.THREAD, "123").
 				build();
 
-		reset(connection);
+		reset(xmppConnection);
 		handler.handleMessage(message);
 
 		// in threaded conversation we need to look for existing chat
-		verify(connection, times(1))
+		verify(xmppConnection, times(1))
 				.sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message smackMessage) -> {
 					boolean bodyMatches = smackMessage.getBody().equals("Hello Kitty");
 					boolean toMatches = smackMessage.getTo().equals("kermit@frog.com");
@@ -90,14 +97,14 @@
 					return bodyMatches & toMatches & threadIdMatches;
 				}));
 
-		reset(connection);
+		reset(xmppConnection);
 		final String json = "{\"foo\": \"bar\"}";
 		message = MessageBuilder.withPayload(new GcmPacketExtension(json))
 				.setHeader(XmppHeaders.TO, "kermit@frog.com")
 				.build();
 		handler.handleMessage(message);
 
-		verify(connection, times(1))
+		verify(xmppConnection, times(1))
 				.sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message smackMessage) -> {
 					boolean bodyMatches = smackMessage.getBody() == null;
 					boolean toMatches = smackMessage.getTo().equals("kermit@frog.com");
@@ -109,8 +116,7 @@
 
 	@Test
 	public void validateMessagePostAsSmackMessage() throws Exception {
-		XMPPConnection connection = mock(XMPPConnection.class);
-		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(connection);
+		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(xmppConnection);
 		handler.setBeanFactory(mock(BeanFactory.class));
 		handler.afterPropertiesSet();
 
@@ -123,8 +129,8 @@
 		// first Message new
 		handler.handleMessage(message);
 
-		verify(connection).isConnected();
-		verify(connection).sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message m) -> {
+		verify(xmppConnection).isConnected();
+		verify(xmppConnection).sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message m) -> {
 			boolean bodyMatches = "Test Message".equals(m.getBody());
 			boolean toMatches = m.getTo().toString().equals("kermit@frog.com");
 			return bodyMatches && toMatches;
@@ -139,12 +145,12 @@
 				.build();
 		message = MessageBuilder.withPayload(smackMessage).build();
 
-		reset(connection);
+		reset(xmppConnection);
 		handler.handleMessage(message);
 
 		// in threaded conversation we need to look for existing chat
-		verify(connection).isConnected();
-		verify(connection).sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message m) -> {
+		verify(xmppConnection).isConnected();
+		verify(xmppConnection).sendStanza(Mockito.argThat((org.jivesoftware.smack.packet.Message m) -> {
 			boolean bodyMatches = "Hello Kitty".equals(m.getBody());
 			boolean toMatches = "kermit@frog.com".equals(m.getTo().toString());
 			boolean threadMatches = "123".equals(m.getThread());
@@ -154,8 +160,7 @@
 
 	@Test
 	public void testExtensionProvider() throws Exception {
-		XMPPConnection connection = mock(XMPPConnection.class);
-		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(connection);
+		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(xmppConnection);
 		GcmExtensionProvider extensionElementProvider = spy(new GcmExtensionProvider());
 		handler.setExtensionProvider(extensionElementProvider);
 		handler.setBeanFactory(mock(BeanFactory.class));
@@ -171,7 +176,7 @@
 		ArgumentCaptor<org.jivesoftware.smack.packet.Message> argumentCaptor =
 				ArgumentCaptor.forClass(org.jivesoftware.smack.packet.Message.class);
 
-		verify(connection).sendStanza(argumentCaptor.capture());
+		verify(xmppConnection).sendStanza(argumentCaptor.capture());
 
 		org.jivesoftware.smack.packet.Message smackMessage = argumentCaptor.getValue();
 
@@ -186,14 +191,14 @@
 
 	@Test
 	public void validateFailureNoChatToUser() {
-		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(mock(XMPPConnection.class));
+		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(xmppConnection);
 		assertThatExceptionOfType(MessageHandlingException.class)
 				.isThrownBy(() -> handler.handleMessage(new GenericMessage<>("hello")));
 	}
 
 	@Test
 	public void validateMessageWithUnsupportedPayload() {
-		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(mock(XMPPConnection.class));
+		ChatMessageSendingMessageHandler handler = new ChatMessageSendingMessageHandler(xmppConnection);
 		assertThatExceptionOfType(MessageHandlingException.class)
 				.isThrownBy(() -> handler.handleMessage(new GenericMessage<>(123)));
 	}
