--- a/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpointTests.java
+++ b/spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpointTests.java
@@ -34,6 +34,7 @@
 import org.jivesoftware.smack.xml.XmlPullParser;
 import org.jivesoftware.smack.xml.XmlPullParserException;
 import org.jivesoftware.smackx.gcm.packet.GcmPacketExtension;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.jxmpp.jid.impl.JidCreate;
 import org.jxmpp.stringprep.XmppStringprepException;
@@ -72,6 +73,13 @@
  */
 public class ChatMessageListeningEndpointTests implements TestApplicationContextAware {
 
+	private XMPPConnection xmppConnection;
+
+	@BeforeEach
+	public void setup() {
+		this.xmppConnection = mock(XMPPConnection.class);
+	}
+
 	/*
 	 * Should add/remove StanzaListener when endpoint started/stopped
 	 */
@@ -105,7 +113,7 @@
 
 	@Test
 	public void testNonInitializationFailure() {
-		ChatMessageListeningEndpoint endpoint = new ChatMessageListeningEndpoint(mock(XMPPConnection.class));
+		ChatMessageListeningEndpoint endpoint = new ChatMessageListeningEndpoint(this.xmppConnection);
 		assertThatIllegalArgumentException()
 				.isThrownBy(endpoint::start);
 	}
@@ -129,8 +137,7 @@
 
 	@Test
 	public void testWithErrorChannel() throws Exception {
-		XMPPConnection connection = mock(XMPPConnection.class);
-		TEST_INTEGRATION_CONTEXT.registerBean(XmppContextUtils.XMPP_CONNECTION_BEAN_NAME, connection);
+		TEST_INTEGRATION_CONTEXT.registerBean(XmppContextUtils.XMPP_CONNECTION_BEAN_NAME, xmppConnection);
 		ChatMessageListeningEndpoint endpoint = new ChatMessageListeningEndpoint();
 
 		DirectChannel outChannel = new DirectChannel();
